System.register(["./vendor/svelte-CiOtOeYy.js","./vendor/carbon-components-svelte-B9ucfJDX.js","./vendor/carbon-icons-svelte-DVphRbRG.js"],(function(e){"use strict";var t,n,a,s,r,i,l,o,c,p,_,d,m,u,g,f,b,$,h,v,x,y,C,S,M,w,L,R,U,N,T,P,A,E,I,B,V,z,k,q,G,D,O,F,H,j,K,W,Y,X,J,Q,Z,ee,te,ne,ae,se,re,ie,le,oe,ce,pe,_e,de,me,ue,ge,fe,be,$e,he,ve,xe,ye,Ce,Se,Me,we,Le,Re,Ue,Ne,Te,Pe,Ae,Ee,Ie,Be;return{setters:[function(e){t=e.S,n=e.i,a=e.s,s=e.a1,r=e.l,i=e.O,l=e.a,o=e.b,c=e.g,p=e.c,_=e.t,d=e.d,m=e.a5,u=e.B,g=e.e,f=e.G,b=e.H,$=e.I,h=e.y,v=e.K,x=e.T,y=e.U,C=e.L,S=e.D,M=e.M,w=e.z,L=e.Q,R=e.k,U=e.u,N=e.n,T=e.o,P=e.f,A=e.A,E=e.h,I=e.j,B=e.aa,V=e.x,z=e.W,k=e.ab,q=e.m,G=e.p,D=e.a0,O=e.q,F=e.R,H=e.$,j=e.v,K=e.Y,W=e.Z,Y=e.F,X=e.ac,J=e.w,Q=e.J,Z=e.r},function(e){ee=e.S,te=e.a,ne=e.b,ae=e.H,se=e.c,re=e.T,ie=e.d,le=e.e,oe=e.f,ce=e.g,pe=e.h,_e=e.M,de=e.I,me=e.i,ue=e.j,ge=e.k,fe=e.C,be=e.l,$e=e.m,he=e.G,ve=e.R,xe=e.n,ye=e.B,Ce=e.A,Se=e.o,Me=e.D,we=e.P,Le=e.p,Re=e.q,Ue=e.r},function(e){Ne=e.C,Te=e.a,Pe=e.O,Ae=e.b,Ee=e.c,Ie=e.D,Be=e.d}],execute:function(){function Ve(e,t,n){const a=e.slice();return a[4]=t[n],a}function ze(e,t,n){const a=e.slice();return a[7]=t[n],a}function ke(e){let t,n;return t=new ee({props:{href:`${e[1](e[4].page)}#${e[4].slug}`,class:e[2](e[4].page),title:e[4].text,text:e[4].text,expanded:!0,$$slots:{default:[De]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.href=`${e[1](e[4].page)}#${e[4].slug}`),1&n&&(a.class=e[2](e[4].page)),1&n&&(a.title=e[4].text),1&n&&(a.text=e[4].text),1025&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function qe(e){let t,n;return t=new te({props:{href:`${e[1](e[4].page)}#${e[4].slug}`,class:e[2](e[4].page),title:e[4].text,text:e[4].text}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.href=`${e[1](e[4].page)}#${e[4].slug}`),1&n&&(a.class=e[2](e[4].page)),1&n&&(a.title=e[4].text),1&n&&(a.text=e[4].text),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Ge(e){let t,n;return t=new ne({props:{href:`${e[1](e[7].page)}#${e[7].slug}`,class:e[2](e[7].page),title:e[7].text,text:e[7].text}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.href=`${e[1](e[7].page)}#${e[7].slug}`),1&n&&(a.class=e[2](e[7].page)),1&n&&(a.title=e[7].text),1&n&&(a.text=e[7].text),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function De(e){let t,n,a=s(e[4].children),r=[];for(let t=0;t<a.length;t+=1)r[t]=Ge(ze(e,a,t));const i=e=>_(r[e],1,1,(()=>{r[e]=null}));return{c(){for(let e=0;e<r.length;e+=1)r[e].c();t=h()},m(e,a){for(let t=0;t<r.length;t+=1)r[t]&&r[t].m(e,a);l(e,t,a),n=!0},p(e,n){if(7&n){let l;for(a=s(e[4].children),l=0;l<a.length;l+=1){const s=ze(e,a,l);r[l]?(r[l].p(s,n),o(r[l],1)):(r[l]=Ge(s),r[l].c(),o(r[l],1),r[l].m(t.parentNode,t))}for(c(),l=a.length;l<r.length;l+=1)i(l);p()}},i(e){if(!n){for(let e=0;e<a.length;e+=1)o(r[e]);n=!0}},o(e){r=r.filter(Boolean);for(let e=0;e<r.length;e+=1)_(r[e]);n=!1},d(e){e&&d(t),m(r,e)}}}function Oe(e){let t,n,a,s;const r=[qe,ke],i=[];function m(e,t){return 0==e[4].children.length?0:1}return t=m(e),n=i[t]=r[t](e),{c(){n.c(),a=g()},m(e,n){i[t].m(e,n),l(e,a,n),s=!0},p(e,s){let l=t;t=m(e),t===l?i[t].p(e,s):(c(),_(i[l],1,1,(()=>{i[l]=null})),p(),n=i[t],n?n.p(e,s):(n=i[t]=r[t](e),n.c()),o(n,1),n.m(a.parentNode,a))},i(e){s||(o(n),s=!0)},o(e){_(n),s=!1},d(e){e&&d(a),i[t].d(e)}}}function Fe(e){let t,n,a=s(e[0]),u=[];for(let t=0;t<a.length;t+=1)u[t]=Oe(Ve(e,a,t));const g=e=>_(u[e],1,1,(()=>{u[e]=null}));return{c(){t=r("ul");for(let e=0;e<u.length;e+=1)u[e].c();i(t,"class","bx--side-nav__items pipen-report-sidenav svelte-12gpq3")},m(e,a){l(e,t,a);for(let e=0;e<u.length;e+=1)u[e]&&u[e].m(t,null);n=!0},p(e,[n]){if(7&n){let r;for(a=s(e[0]),r=0;r<a.length;r+=1){const s=Ve(e,a,r);u[r]?(u[r].p(s,n),o(u[r],1)):(u[r]=Oe(s),u[r].c(),o(u[r],1),u[r].m(t,null))}for(c(),r=a.length;r<u.length;r+=1)g(r);p()}},i(e){if(!n){for(let e=0;e<a.length;e+=1)o(u[e]);n=!0}},o(e){u=u.filter(Boolean);for(let e=0;e<u.length;e+=1)_(u[e]);n=!1},d(e){e&&d(t),m(u,e)}}}function He(e,t,n){let{toc:a}=t,{page:s}=t;return u((()=>{let e,t=new URL(window.location.href);e=t.hash.length>0?document.querySelector(`[href="${t.hash}"]`):document.querySelector(".toc-current-page"),e.scrollIntoView({behavior:"smooth"})})),e.$$set=e=>{"toc"in e&&n(0,a=e.toc),"page"in e&&n(3,s=e.page)},[a,e=>{if(s===e)return"";let t=new URL(window.location.href);return t.hash="",t.searchParams.set("page",e),t.toString()},e=>e!=s?"toc-noncurrent-page":"toc-current-page",s]}class je extends t{constructor(e){super(),n(this,e,He,Fe,a,{toc:0,page:3})}}function Ke(e,t,n){const a=e.slice();return a[7]=t[n],a}function We(e,t,n){const a=e.slice();return a[10]=t[n],a}function Ye(e){let t,n;return t=new le({props:{isOpen:ut,transition:e[5],$$slots:{default:[lt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8197&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Xe(e){let t;return{c(){t=C("Go to home:")},m(e,n){l(e,t,n)},d(e){e&&d(t)}}}function Je(e){let t;return{c(){t=C(e[0])},m(e,n){l(e,t,n)},p(e,n){1&n&&M(t,e[0])},d(e){e&&d(t)}}}function Qe(e){let t;return{c(){t=C("Switch to process:")},m(e,n){l(e,t,n)},d(e){e&&d(t)}}}function Ze(e){let t,n;return t=new pe({props:{title:e[7].name,href:`?proc=${e[7].name}`,$$slots:{default:[tt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};4&n&&(a.title=e[7].name),4&n&&(a.href=`?proc=${e[7].name}`),8196&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function et(e){let t,n,a,r;t=new pe({props:{class:"procgroup",$$slots:{default:[nt]},$$scope:{ctx:e}}});let i=s(e[7].procs),u=[];for(let t=0;t<i.length;t+=1)u[t]=st(We(e,i,t));const v=e=>_(u[e],1,1,(()=>{u[e]=null}));return{c(){f(t.$$.fragment),n=h();for(let e=0;e<u.length;e+=1)u[e].c();a=g()},m(e,s){b(t,e,s),l(e,n,s);for(let t=0;t<u.length;t+=1)u[t]&&u[t].m(e,s);l(e,a,s),r=!0},p(e,n){const r={};if(8196&n&&(r.$$scope={dirty:n,ctx:e}),t.$set(r),4&n){let t;for(i=s(e[7].procs),t=0;t<i.length;t+=1){const s=We(e,i,t);u[t]?(u[t].p(s,n),o(u[t],1)):(u[t]=st(s),u[t].c(),o(u[t],1),u[t].m(a.parentNode,a))}for(c(),t=i.length;t<u.length;t+=1)v(t);p()}},i(e){if(!r){o(t.$$.fragment,e);for(let e=0;e<i.length;e+=1)o(u[e]);r=!0}},o(e){_(t.$$.fragment,e),u=u.filter(Boolean);for(let e=0;e<u.length;e+=1)_(u[e]);r=!1},d(e){e&&(d(n),d(a)),$(t,e),m(u,e)}}}function tt(e){let t,n,a=e[7].name+"";return{c(){t=C(a),n=h()},m(e,a){l(e,t,a),l(e,n,a)},p(e,n){4&n&&a!==(a=e[7].name+"")&&M(t,a)},d(e){e&&(d(t),d(n))}}}function nt(e){let t,n=e[7].name+"";return{c(){t=C(n)},m(e,n){l(e,t,n)},p(e,a){4&a&&n!==(n=e[7].name+"")&&M(t,n)},d(e){e&&d(t)}}}function at(e){let t,n,a,s=e[10].name+"";return{c(){t=C("- "),n=C(s),a=h()},m(e,s){l(e,t,s),l(e,n,s),l(e,a,s)},p(e,t){4&t&&s!==(s=e[10].name+"")&&M(n,s)},d(e){e&&(d(t),d(n),d(a))}}}function st(e){let t,n;return t=new pe({props:{title:e[10].name,class:"procgroup-proc",href:`?proc=${e[10].name}`,$$slots:{default:[at]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};4&n&&(a.title=e[10].name),4&n&&(a.href=`?proc=${e[10].name}`),8196&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function rt(e){let t,n,a,s;const r=[et,Ze],i=[];function m(e,t){return e[7].procs?0:1}return t=m(e),n=i[t]=r[t](e),{c(){n.c(),a=g()},m(e,n){i[t].m(e,n),l(e,a,n),s=!0},p(e,s){let l=t;t=m(e),t===l?i[t].p(e,s):(c(),_(i[l],1,1,(()=>{i[l]=null})),p(),n=i[t],n?n.p(e,s):(n=i[t]=r[t](e),n.c()),o(n,1),n.m(a.parentNode,a))},i(e){s||(o(n),s=!0)},o(e){_(n),s=!1},d(e){e&&d(a),i[t].d(e)}}}function it(e){let t,n,a,r,i,u,v,x;t=new ce({props:{$$slots:{default:[Xe]},$$scope:{ctx:e}}}),a=new pe({props:{href:"index.html",$$slots:{default:[Je]},$$scope:{ctx:e}}}),i=new ce({props:{$$slots:{default:[Qe]},$$scope:{ctx:e}}});let y=s(e[2]),C=[];for(let t=0;t<y.length;t+=1)C[t]=rt(Ke(e,y,t));const S=e=>_(C[e],1,1,(()=>{C[e]=null}));return{c(){f(t.$$.fragment),n=h(),f(a.$$.fragment),r=h(),f(i.$$.fragment),u=h();for(let e=0;e<C.length;e+=1)C[e].c();v=g()},m(e,s){b(t,e,s),l(e,n,s),b(a,e,s),l(e,r,s),b(i,e,s),l(e,u,s);for(let t=0;t<C.length;t+=1)C[t]&&C[t].m(e,s);l(e,v,s),x=!0},p(e,n){const r={};8192&n&&(r.$$scope={dirty:n,ctx:e}),t.$set(r);const l={};8193&n&&(l.$$scope={dirty:n,ctx:e}),a.$set(l);const _={};if(8192&n&&(_.$$scope={dirty:n,ctx:e}),i.$set(_),4&n){let t;for(y=s(e[2]),t=0;t<y.length;t+=1){const a=Ke(e,y,t);C[t]?(C[t].p(a,n),o(C[t],1)):(C[t]=rt(a),C[t].c(),o(C[t],1),C[t].m(v.parentNode,v))}for(c(),t=y.length;t<C.length;t+=1)S(t);p()}},i(e){if(!x){o(t.$$.fragment,e),o(a.$$.fragment,e),o(i.$$.fragment,e);for(let e=0;e<y.length;e+=1)o(C[e]);x=!0}},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),_(i.$$.fragment,e),C=C.filter(Boolean);for(let e=0;e<C.length;e+=1)_(C[e]);x=!1},d(e){e&&(d(n),d(r),d(u),d(v)),$(t,e),$(a,e),$(i,e),m(C,e)}}}function lt(e){let t,n;return t=new oe({props:{$$slots:{default:[it]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8197&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function ot(e){let t,n,a,s;t=new ie({props:{render:"select",theme:"g10",persist:!0,persistKey:"__pipen-report-theme",select:{themes:["g10","white","g80","g90","g100"],labelText:"Theme",inline:!0}}});let r=e[2].length>0&&Ye(e);return{c(){f(t.$$.fragment),n=h(),r&&r.c(),a=g()},m(e,i){b(t,e,i),l(e,n,i),r&&r.m(e,i),l(e,a,i),s=!0},p(e,t){e[2].length>0?r?(r.p(e,t),4&t&&o(r,1)):(r=Ye(e),r.c(),o(r,1),r.m(a.parentNode,a)):r&&(c(),_(r,1,1,(()=>{r=null})),p())},i(e){s||(o(t.$$.fragment,e),o(r),s=!0)},o(e){_(t.$$.fragment,e),_(r),s=!1},d(e){e&&(d(n),d(a)),$(t,e),r&&r.d(e)}}}function ct(e){let t,n;return t=new se({props:{class:"header-util",$$slots:{default:[ot]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8197&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function pt(e){let t;return{c(){t=C(e[4])},m(e,n){l(e,t,n)},p(e,n){16&n&&M(t,e[4])},d(e){e&&d(t)}}}function _t(e){let t,n,a,s,i=null!==e[4]&&pt(e);return{c(){t=r("h1"),n=C(e[3]),a=h(),i&&i.c(),s=g()},m(e,r){l(e,t,r),S(t,n),l(e,a,r),i&&i.m(e,r),l(e,s,r)},p(e,t){8&t&&M(n,e[3]),null!==e[4]?i?i.p(e,t):(i=pt(e),i.c(),i.m(s.parentNode,s)):i&&(i.d(1),i=null)},d(e){e&&(d(t),d(a),d(s)),i&&i.d(e)}}}function dt(e){let t,n;return t=new re({props:{class:"header-tile",slot:"platform",$$slots:{default:[_t]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8216&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function mt(e){let t,n,a;function s(t){e[6](t)}let r={$$slots:{platform:[dt],default:[ct]},$$scope:{ctx:e}};return void 0!==e[1]&&(r.isSideNavOpen=e[1]),t=new ae({props:r}),v.push((()=>x(t,"isSideNavOpen",s))),{c(){f(t.$$.fragment)},m(e,n){b(t,e,n),a=!0},p(e,[a]){const s={};8221&a&&(s.$$scope={dirty:a,ctx:e}),!n&&2&a&&(n=!0,s.isSideNavOpen=e[1],y((()=>n=!1))),t.$set(s)},i(e){a||(o(t.$$.fragment,e),a=!0)},o(e){_(t.$$.fragment,e),a=!1},d(e){$(t,e)}}}let ut=!1;function gt(e,t,n){let{entries:a}=t,{logo:s}=t,{logotext:r=null}=t,{pipeline_name:i}=t,{isSideNavOpen:l}=t;return i=i||s,e.$$set=e=>{"entries"in e&&n(2,a=e.entries),"logo"in e&&n(3,s=e.logo),"logotext"in e&&n(4,r=e.logotext),"pipeline_name"in e&&n(0,i=e.pipeline_name),"isSideNavOpen"in e&&n(1,l=e.isSideNavOpen)},[i,l,a,s,r,{duration:200},function(e){l=e,n(1,l)}]}class ft extends t{constructor(e){super(),n(this,e,gt,mt,a,{entries:2,logo:3,logotext:4,pipeline_name:0,isSideNavOpen:1})}}function bt(e){let t,n,a;function s(t){e[3](t)}let r={modalHeading:"Running Information",passiveModal:!0,hasScrollingContent:!0,class:"runinfo",$$slots:{default:[Ct]},$$scope:{ctx:e}};return void 0!==e[2]&&(r.open=e[2]),t=new _e({props:r}),v.push((()=>x(t,"open",s))),{c(){f(t.$$.fragment)},m(e,n){b(t,e,n),a=!0},p(e,a){const s={};34&a&&(s.$$scope={dirty:a,ctx:e}),!n&&4&a&&(n=!0,s.open=e[2],y((()=>n=!1))),t.$set(s)},i(e){a||(o(t.$$.fragment,e),a=!0)},o(e){_(t.$$.fragment,e),a=!1},d(e){$(t,e)}}}function $t(e){let t,n,a,s,r,i;return t=new ue({props:{label:"Session Information"}}),a=new ue({props:{label:"Time Spent"}}),r=new ue({props:{label:"Device Information"}}),{c(){f(t.$$.fragment),n=h(),f(a.$$.fragment),s=h(),f(r.$$.fragment)},m(e,o){b(t,e,o),l(e,n,o),b(a,e,o),l(e,s,o),b(r,e,o),i=!0},p:L,i(e){i||(o(t.$$.fragment,e),o(a.$$.fragment,e),o(r.$$.fragment,e),i=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),_(r.$$.fragment,e),i=!1},d(e){e&&(d(n),d(s)),$(t,e),$(a,e),$(r,e)}}}function ht(e){let t,n;return t=new fe({props:{type:"multi",code:e[1].session}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};2&n&&(a.code=e[1].session),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function vt(e){let t,n;return t=new fe({props:{type:"multi",code:e[1].time}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};2&n&&(a.code=e[1].time),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function xt(e){let t,n;return t=new fe({props:{type:"multi",code:e[1].device}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};2&n&&(a.code=e[1].device),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function yt(e){let t,n,a,s,r,i;return t=new ge({props:{$$slots:{default:[ht]},$$scope:{ctx:e}}}),a=new ge({props:{$$slots:{default:[vt]},$$scope:{ctx:e}}}),r=new ge({props:{$$slots:{default:[xt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=h(),f(a.$$.fragment),s=h(),f(r.$$.fragment)},m(e,o){b(t,e,o),l(e,n,o),b(a,e,o),l(e,s,o),b(r,e,o),i=!0},p(e,n){const s={};34&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const i={};34&n&&(i.$$scope={dirty:n,ctx:e}),a.$set(i);const l={};34&n&&(l.$$scope={dirty:n,ctx:e}),r.$set(l)},i(e){i||(o(t.$$.fragment,e),o(a.$$.fragment,e),o(r.$$.fragment,e),i=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),_(r.$$.fragment,e),i=!1},d(e){e&&(d(n),d(s)),$(t,e),$(a,e),$(r,e)}}}function Ct(e){let t,n,a,s;return t=new de({props:{kind:"warning",lowContrast:!0,hideCloseButton:!0,title:"Showing the first job only. Check the workdir for information of other jobs if any."}}),a=new me({props:{$$slots:{content:[yt],default:[$t]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=h(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,t){const n={};34&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function St(e){let t,n,a;return{c(){t=r("a"),t.textContent="> Running Information",i(t,"href","javascript:void(0);")},m(s,r){l(s,t,r),n||(a=w(t,"click",e[4]),n=!0)},p:L,d(e){e&&d(t),n=!1,a()}}}function Mt(e){let t,n,a,s,m,u,g,f,b=e[1]&&bt(e),$=e[1]&&St(e);return{c(){b&&b.c(),t=h(),n=r("div"),a=r("div"),$&&$.c(),s=h(),m=r("div"),u=C("Powered by "),g=r("span"),i(a,"class","runinfo"),i(g,"class","pipen-report-footer-versions"),i(m,"class","versions"),i(n,"class","pipen-report-footer bx--grid")},m(r,i){b&&b.m(r,i),l(r,t,i),l(r,n,i),S(n,a),$&&$.m(a,null),S(n,s),S(n,m),S(m,u),S(m,g),g.innerHTML=e[0],f=!0},p(e,[n]){e[1]?b?(b.p(e,n),2&n&&o(b,1)):(b=bt(e),b.c(),o(b,1),b.m(t.parentNode,t)):b&&(c(),_(b,1,1,(()=>{b=null})),p()),e[1]?$?$.p(e,n):($=St(e),$.c(),$.m(a,null)):$&&($.d(1),$=null),(!f||1&n)&&(g.innerHTML=e[0])},i(e){f||(o(b),f=!0)},o(e){_(b),f=!1},d(e){e&&(d(t),d(n)),b&&b.d(e),$&&$.d()}}}function wt(e,t,n){let{versions:a}=t,{runinfo:s}=t,r=!1;return e.$$set=e=>{"versions"in e&&n(0,a=e.versions),"runinfo"in e&&n(1,s=e.runinfo)},[a,s,r,function(e){r=e,n(2,r)},()=>{n(2,r=!0)}]}class Lt extends t{constructor(e){super(),n(this,e,wt,Mt,a,{versions:0,runinfo:1})}}function Rt(e){let t,n,a;function s(t){e[12](t)}let r={$$slots:{default:[Ut]},$$scope:{ctx:e}};return void 0!==e[0]&&(r.isOpen=e[0]),t=new $e({props:r}),v.push((()=>x(t,"isOpen",s))),{c(){f(t.$$.fragment)},m(e,n){b(t,e,n),a=!0},p(e,a){const s={};8576&a&&(s.$$scope={dirty:a,ctx:e}),!n&&1&a&&(n=!0,s.isOpen=e[0],y((()=>n=!1))),t.$set(s)},i(e){a||(o(t.$$.fragment,e),a=!0)},o(e){_(t.$$.fragment,e),a=!1},d(e){$(t,e)}}}function Ut(e){let t,n;return t=new je({props:{toc:e[7],page:e[8]}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};128&n&&(a.toc=e[7]),256&n&&(a.page=e[8]),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Nt(e){let t;const n=e[10].default,a=R(n,e,e[13],null);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||8192&s)&&U(a,n,e,e[13],t?T(n,e[13],s,null):N(e[13]),null)},i(e){t||(o(a,e),t=!0)},o(e){_(a,e),t=!1},d(e){a&&a.d(e)}}}function Tt(e){let t,n;return t=new xe({props:{$$slots:{default:[Nt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8192&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Pt(e){let t,n;return t=new ve({props:{$$slots:{default:[Tt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8192&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function At(e){let t,n,a,s;return t=new he({props:{$$slots:{default:[Pt]},$$scope:{ctx:e}}}),a=new Lt({props:{versions:e[3],runinfo:e[9]}}),{c(){f(t.$$.fragment),n=h(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const s={};8192&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};8&n&&(r.versions=e[3]),512&n&&(r.runinfo=e[9]),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function Et(e){let t,n,a,s,r,i;function m(t){e[11](t)}let u={logo:e[1],logotext:e[2],entries:e[4],pipeline_name:e[5]};void 0!==e[0]&&(u.isSideNavOpen=e[0]),t=new ft({props:u}),v.push((()=>x(t,"isSideNavOpen",m)));let g=e[6]&&e[7].length>0&&Rt(e);return r=new be({props:{$$slots:{default:[At]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),a=h(),g&&g.c(),s=h(),f(r.$$.fragment)},m(e,n){b(t,e,n),l(e,a,n),g&&g.m(e,n),l(e,s,n),b(r,e,n),i=!0},p(e,[a]){const i={};2&a&&(i.logo=e[1]),4&a&&(i.logotext=e[2]),16&a&&(i.entries=e[4]),32&a&&(i.pipeline_name=e[5]),!n&&1&a&&(n=!0,i.isSideNavOpen=e[0],y((()=>n=!1))),t.$set(i),e[6]&&e[7].length>0?g?(g.p(e,a),192&a&&o(g,1)):(g=Rt(e),g.c(),o(g,1),g.m(s.parentNode,s)):g&&(c(),_(g,1,1,(()=>{g=null})),p());const l={};8712&a&&(l.$$scope={dirty:a,ctx:e}),r.$set(l)},i(e){i||(o(t.$$.fragment,e),o(g),o(r.$$.fragment,e),i=!0)},o(e){_(t.$$.fragment,e),_(g),_(r.$$.fragment,e),i=!1},d(e){e&&(d(a),d(s)),$(t,e),g&&g.d(e),$(r,e)}}}function It(e,t,n){let a,{$$slots:s={},$$scope:r}=t,{logo:i}=t,{logotext:l=null}=t,{versions:o}=t,{entries:c}=t,{pipeline_name:p}=t,{report_toc:_}=t,{toc:d}=t,{page:m}=t,{isSideNavOpen:g}=t,{runinfo:f}=t;return u((()=>{const e=new URL(window.location.href);let t;t=e.hash.length>0?document.querySelector(`li:has(>[href="${e.hash}"])`):document.querySelector("li:has(>.toc-current-page)"),t&&(t.classList.add("toc-active"),t.scrollIntoView());const n=document.querySelectorAll(".pipen-report-toc-anchor");a=new IntersectionObserver((e=>{e.forEach((e=>{const t=e.target.getAttribute("id"),n=document.querySelector(`li:has(>[href="#${t}"])`);n&&(e.isIntersecting?n.classList.add("toc-active"):n.classList.remove("toc-active"))}))}),{root:null,rootMargin:"0px",threshold:.5}),n.forEach((e=>{a.observe(e)}))})),e.$$set=e=>{"logo"in e&&n(1,i=e.logo),"logotext"in e&&n(2,l=e.logotext),"versions"in e&&n(3,o=e.versions),"entries"in e&&n(4,c=e.entries),"pipeline_name"in e&&n(5,p=e.pipeline_name),"report_toc"in e&&n(6,_=e.report_toc),"toc"in e&&n(7,d=e.toc),"page"in e&&n(8,m=e.page),"isSideNavOpen"in e&&n(0,g=e.isSideNavOpen),"runinfo"in e&&n(9,f=e.runinfo),"$$scope"in e&&n(13,r=e.$$scope)},[g,i,l,o,c,p,_,d,m,f,s,function(e){g=e,n(0,g)},function(e){g=e,n(0,g)},r]}class Bt extends t{constructor(e){super(),n(this,e,It,Et,a,{logo:1,logotext:2,versions:3,entries:4,pipeline_name:5,report_toc:6,toc:7,page:8,isSideNavOpen:0,runinfo:9})}}function Vt(e){let t;return{c(){t=C(e[2])},m(e,n){l(e,t,n)},p:L,d(e){e&&d(t)}}}function zt(e){let t,n,a,s;return n=new ye({props:{icon:e[1],kind:"tertiary",href:e[3](),$$slots:{default:[Vt]},$$scope:{ctx:e}}}),{c(){t=r("div"),f(n.$$.fragment),i(t,"class",a=`pagenav ${e[0]}`)},m(e,a){l(e,t,a),b(n,t,null),s=!0},p(e,[r]){const l={};16&r&&(l.$$scope={dirty:r,ctx:e}),n.$set(l),(!s||1&r&&a!==(a=`pagenav ${e[0]}`))&&i(t,"class",a)},i(e){s||(o(n.$$.fragment,e),s=!0)},o(e){_(n.$$.fragment,e),s=!1},d(e){e&&d(t),$(n)}}}function kt(e,t,n){let{dir:a}=t,s="up"===a?Ne:Te,r="up"===a?"Go to previous page":"Go to next page";return e.$$set=e=>{"dir"in e&&n(0,a=e.dir)},[a,s,r,()=>{let e=new URL(window.location.href);e.hash="";const t=parseInt(e.searchParams.get("page")||0),n="up"===a?t-1:t+1;return 0===n?e.searchParams.delete("page"):e.searchParams.set("page",n),e.toString()}]}class qt extends t{constructor(e){super(),n(this,e,kt,zt,a,{dir:0})}}const Gt=e=>({}),Dt=e=>({}),Ot=e=>({}),Ft=e=>({}),Ht=e=>({}),jt=e=>({}),Kt=e=>({}),Wt=e=>({});function Yt(e){let t,n;return t=new Ce({props:{ratio:e[5],$$slots:{default:[en]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};32&n&&(a.ratio=e[5]),65631&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Xt(e){let t,n,a,s,r=e[2]&&tn(e),i=e[0]&&nn(e),m=e[1]&&an(e);return{c(){r&&r.c(),t=h(),i&&i.c(),n=h(),m&&m.c(),a=g()},m(e,o){r&&r.m(e,o),l(e,t,o),i&&i.m(e,o),l(e,n,o),m&&m.m(e,o),l(e,a,o),s=!0},p(e,s){e[2]?r?(r.p(e,s),4&s&&o(r,1)):(r=tn(e),r.c(),o(r,1),r.m(t.parentNode,t)):r&&(c(),_(r,1,1,(()=>{r=null})),p()),e[0]?i?(i.p(e,s),1&s&&o(i,1)):(i=nn(e),i.c(),o(i,1),i.m(n.parentNode,n)):i&&(c(),_(i,1,1,(()=>{i=null})),p()),e[1]?m?(m.p(e,s),2&s&&o(m,1)):(m=an(e),m.c(),o(m,1),m.m(a.parentNode,a)):m&&(c(),_(m,1,1,(()=>{m=null})),p())},i(e){s||(o(r),o(i),o(m),s=!0)},o(e){_(r),_(i),_(m),s=!1},d(e){e&&(d(t),d(n),d(a)),r&&r.d(e),i&&i.d(e),m&&m.d(e)}}}function Jt(e){let t;const n=e[12].loading,a=R(n,e,e[16],Ft);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||65536&s)&&U(a,n,e,e[16],t?T(n,e[16],s,Ot):N(e[16]),Ft)},i(e){t||(o(a,e),t=!0)},o(e){_(a,e),t=!1},d(e){a&&a.d(e)}}}function Qt(e){let t,n,a,s,i,o,c=[e[9],{src:n=e[3]},{alt:e[4]},{title:"Click to zoom in ..."}],p={};for(let e=0;e<c.length;e+=1)p=E(p,c[e]);return{c(){t=r("img"),q(t,p)},m(n,a){l(n,t,a),s=!0,i||(o=w(t,"click",e[8]),i=!0)},p(a,r){q(t,p=G(c,[(e=a)[9],(!s||8&r&&!B(t.src,n=e[3]))&&{src:n},(!s||16&r)&&{alt:e[4]},{title:"Click to zoom in ..."}]))},i(n){s||(n&&V((()=>{s&&(a||(a=z(t,k,{duration:e[6]?ln:0},!0)),a.run(1))})),s=!0)},o(n){n&&(a||(a=z(t,k,{duration:e[6]?ln:0},!1)),a.run(0)),s=!1},d(e){e&&d(t),e&&a&&a.end(),i=!1,o()}}}function Zt(e){let t;const n=e[12].error,a=R(n,e,e[16],Dt);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||65536&s)&&U(a,n,e,e[16],t?T(n,e[16],s,Gt):N(e[16]),Dt)},i(e){t||(o(a,e),t=!0)},o(e){_(a,e),t=!1},d(e){a&&a.d(e)}}}function en(e){let t,n,a,s,r=e[2]&&Jt(e),i=e[0]&&Qt(e),m=e[1]&&Zt(e);return{c(){r&&r.c(),t=h(),i&&i.c(),n=h(),m&&m.c(),a=g()},m(e,o){r&&r.m(e,o),l(e,t,o),i&&i.m(e,o),l(e,n,o),m&&m.m(e,o),l(e,a,o),s=!0},p(e,s){e[2]?r?(r.p(e,s),4&s&&o(r,1)):(r=Jt(e),r.c(),o(r,1),r.m(t.parentNode,t)):r&&(c(),_(r,1,1,(()=>{r=null})),p()),e[0]?i?(i.p(e,s),1&s&&o(i,1)):(i=Qt(e),i.c(),o(i,1),i.m(n.parentNode,n)):i&&(c(),_(i,1,1,(()=>{i=null})),p()),e[1]?m?(m.p(e,s),2&s&&o(m,1)):(m=Zt(e),m.c(),o(m,1),m.m(a.parentNode,a)):m&&(c(),_(m,1,1,(()=>{m=null})),p())},i(e){s||(o(r),o(i),o(m),s=!0)},o(e){_(r),_(i),_(m),s=!1},d(e){e&&(d(t),d(n),d(a)),r&&r.d(e),i&&i.d(e),m&&m.d(e)}}}function tn(e){let t;const n=e[12].loading,a=R(n,e,e[16],Wt);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||65536&s)&&U(a,n,e,e[16],t?T(n,e[16],s,Kt):N(e[16]),Wt)},i(e){t||(o(a,e),t=!0)},o(e){_(a,e),t=!1},d(e){a&&a.d(e)}}}function nn(e){let t,n,a,s,i,o,c=[e[9],{src:n=e[3]},{alt:e[4]},{title:"Click to zoom in ..."}],p={};for(let e=0;e<c.length;e+=1)p=E(p,c[e]);return{c(){t=r("img"),q(t,p)},m(n,a){l(n,t,a),s=!0,i||(o=w(t,"click",e[8]),i=!0)},p(a,r){q(t,p=G(c,[(e=a)[9],(!s||8&r&&!B(t.src,n=e[3]))&&{src:n},(!s||16&r)&&{alt:e[4]},{title:"Click to zoom in ..."}]))},i(n){s||(n&&V((()=>{s&&(a||(a=z(t,k,{duration:e[6]?ln:0},!0)),a.run(1))})),s=!0)},o(n){n&&(a||(a=z(t,k,{duration:e[6]?ln:0},!1)),a.run(0)),s=!1},d(e){e&&d(t),e&&a&&a.end(),i=!1,o()}}}function an(e){let t;const n=e[12].error,a=R(n,e,e[16],jt);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||65536&s)&&U(a,n,e,e[16],t?T(n,e[16],s,Ht):N(e[16]),jt)},i(e){t||(o(a,e),t=!0)},o(e){_(a,e),t=!1},d(e){a&&a.d(e)}}}function sn(e){let t,n,a,s;return{c(){t=r("img"),i(t,"class","pipen-report-image-lg"),B(t.src,n=e[3])||i(t,"src",n),i(t,"alt",e[4])},m(e,n){l(e,t,n),s=!0},p(a,r){e=a,(!s||8&r&&!B(t.src,n=e[3]))&&i(t,"src",n),(!s||16&r)&&i(t,"alt",e[4])},i(n){s||(n&&V((()=>{s&&(a||(a=z(t,k,{duration:e[6]?ln:0},!0)),a.run(1))})),s=!0)},o(n){n&&(a||(a=z(t,k,{duration:e[6]?ln:0},!1)),a.run(0)),s=!1},d(e){e&&d(t),e&&a&&a.end()}}}function rn(e){let t,n,a,s,r,i;const m=[Xt,Yt],u=[];function g(e,t){return void 0===e[5]?0:1}function C(t){e[13](t)}t=g(e),n=u[t]=m[t](e);let S={modalHeading:e[4],primaryButtonText:"Open in new window ...",secondaryButtonText:"Close",$$slots:{default:[sn]},$$scope:{ctx:e}};return void 0!==e[7]&&(S.open=e[7]),s=new _e({props:S}),v.push((()=>x(s,"open",C))),s.$on("click:button--secondary",e[14]),s.$on("click:button--primary",e[15]),{c(){n.c(),a=h(),f(s.$$.fragment)},m(e,n){u[t].m(e,n),l(e,a,n),b(s,e,n),i=!0},p(e,[i]){let l=t;t=g(e),t===l?u[t].p(e,i):(c(),_(u[l],1,1,(()=>{u[l]=null})),p(),n=u[t],n?n.p(e,i):(n=u[t]=m[t](e),n.c()),o(n,1),n.m(a.parentNode,a));const d={};16&i&&(d.modalHeading=e[4]),65624&i&&(d.$$scope={dirty:i,ctx:e}),!r&&128&i&&(r=!0,d.open=e[7],y((()=>r=!1))),s.$set(d)},i(e){i||(o(n),o(s.$$.fragment,e),i=!0)},o(e){_(n),_(s.$$.fragment,e),i=!1},d(e){e&&d(a),u[t].d(e),$(s,e)}}}const ln=110;function on(e,t,n){const a=["src","alt","ratio","loading","loaded","error","fadeIn","border","loadImage"];let s=P(t,a),{$$slots:r={},$$scope:i}=t,{src:l=""}=t,{alt:o=""}=t,{ratio:c}=t,{loading:p=!1}=t,{loaded:_=!1}=t,{error:d=!1}=t,{fadeIn:m=!1}=t,{border:g=!0}=t,f=!1;const b=e=>{null!=h&&(h=null),n(0,_=!1),n(1,d=!1),h=new Image,h.src=e||l,h.onload=()=>n(0,_=!0),h.onerror=()=>n(1,d=!0)},$=A();let h=null;u((()=>()=>h=null));const v={...s,class:[s.class,"pipen-report-image",g?"":"pipen-report-image-noborder"].filter(Boolean).join(" ")};return e.$$set=e=>{t=E(E({},t),I(e)),n(19,s=P(t,a)),"src"in e&&n(3,l=e.src),"alt"in e&&n(4,o=e.alt),"ratio"in e&&n(5,c=e.ratio),"loading"in e&&n(2,p=e.loading),"loaded"in e&&n(0,_=e.loaded),"error"in e&&n(1,d=e.error),"fadeIn"in e&&n(6,m=e.fadeIn),"border"in e&&n(10,g=e.border),"$$scope"in e&&n(16,i=e.$$scope)},e.$$.update=()=>{3&e.$$.dirty&&n(2,p=!_&&!d),8&e.$$.dirty&&l&&"undefined"!=typeof window&&b(),1&e.$$.dirty&&_&&$("load"),2&e.$$.dirty&&d&&$("error")},[_,d,p,l,o,c,m,f,()=>{n(7,f=!0)},v,g,b,r,function(e){f=e,n(7,f)},()=>{n(7,f=!1)},()=>{n(7,f=!1),window.open(l)},i]}class cn extends t{constructor(e){super(),n(this,e,on,rn,a,{src:3,alt:4,ratio:5,loading:2,loaded:0,error:1,fadeIn:6,border:10,loadImage:11})}get loadImage(){return this.$$.ctx[11]}}function pn(e){let t,n;const a=[{src:e[1]},e[4]];let s={$$slots:{error:[mn],loading:[dn]},$$scope:{ctx:e}};for(let e=0;e<a.length;e+=1)s=E(s,a[e]);return t=new cn({props:s}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const s=18&n?G(a,[2&n&&{src:e[1]},16&n&&D(e[4])]):{};256&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function _n(e){let t,n,a;const s=[{src:e[1]},e[4]];let i={$$slots:{error:[gn],loading:[un]},$$scope:{ctx:e}};for(let e=0;e<s.length;e+=1)i=E(i,s[e]);n=new cn({props:i});let c=[e[0]],p={};for(let e=0;e<c.length;e+=1)p=E(p,c[e]);return{c(){t=r("div"),f(n.$$.fragment),q(t,p)},m(e,s){l(e,t,s),b(n,t,null),a=!0},p(e,a){const r=18&a?G(s,[2&a&&{src:e[1]},16&a&&D(e[4])]):{};256&a&&(r.$$scope={dirty:a,ctx:e}),n.$set(r),q(t,p=G(c,[1&a&&e[0]]))},i(e){a||(o(n.$$.fragment,e),a=!0)},o(e){_(n.$$.fragment,e),a=!1},d(e){e&&d(t),$(n)}}}function dn(e){let t,n,a=[e[2]],s={};for(let e=0;e<a.length;e+=1)s=E(s,a[e]);return{c(){t=r("div"),n=C("Image loading ..."),q(t,s)},m(e,a){l(e,t,a),S(t,n)},p:L,d(e){e&&d(t)}}}function mn(e){let t,n,a=[e[3]],s={};for(let e=0;e<a.length;e+=1)s=E(s,a[e]);return{c(){t=r("div"),n=C("Image loading error!"),q(t,s)},m(e,a){l(e,t,a),S(t,n)},p:L,d(e){e&&d(t)}}}function un(e){let t,n,a=[e[2]],s={};for(let e=0;e<a.length;e+=1)s=E(s,a[e]);return{c(){t=r("div"),n=C("Image loading ..."),q(t,s)},m(e,a){l(e,t,a),S(t,n)},p:L,d(e){e&&d(t)}}}function gn(e){let t,n,a=[e[3]],s={};for(let e=0;e<a.length;e+=1)s=E(s,a[e]);return{c(){t=r("div"),n=C("Image loading error!"),q(t,s)},m(e,a){l(e,t,a),S(t,n)},p:L,d(e){e&&d(t)}}}function fn(e){let t,n,a,s;const r=[_n,pn],i=[];function m(e,t){return e[0]?0:1}return t=m(e),n=i[t]=r[t](e),{c(){n.c(),a=g()},m(e,n){i[t].m(e,n),l(e,a,n),s=!0},p(e,[s]){let l=t;t=m(e),t===l?i[t].p(e,s):(c(),_(i[l],1,1,(()=>{i[l]=null})),p(),n=i[t],n?n.p(e,s):(n=i[t]=r[t](e),n.c()),o(n,1),n.m(a.parentNode,a))},i(e){s||(o(n),s=!0)},o(e){_(n),s=!1},d(e){e&&d(a),i[t].d(e)}}}function bn(e,t,n){const a=["frameProps","width","height","src"];let s=P(t,a),{frameProps:r={class:"pipen-report-image-frame"}}=t,{width:i=300}=t,{height:l=200}=t,{src:o}=t,c=`width: ${i}px; height: ${l}px; aspect-ratio: ${i} / ${l}`,p={...s,class:[s.class,"pipen-report-image-loading"].filter(Boolean).join(" "),style:s.style?`${s.style}; ${c}`:c},_={...p,class:[p.class,"error"].join(" ")};return e.$$set=e=>{t=E(E({},t),I(e)),n(4,s=P(t,a)),"frameProps"in e&&n(0,r=e.frameProps),"width"in e&&n(5,i=e.width),"height"in e&&n(6,l=e.height),"src"in e&&n(1,o=e.src)},[r,o,p,_,s,i,l]}let $n=class extends t{constructor(e){super(),n(this,e,bn,fn,a,{frameProps:0,width:5,height:6,src:1})}};const hn=e=>({}),vn=e=>({});function xn(e){let t,n;const a=e[25].default,s=R(a,e,e[24],null);return{c(){t=r("ul"),s&&s.c(),i(t,"role","menu"),i(t,"tabindex","-1"),i(t,"aria-label",e[14]),i(t,"data-floating-menu-direction",e[5]),i(t,"class",e[9]),O(t,"bx--overflow-menu-options",!0),O(t,"bx--overflow-menu--flip",e[7]),O(t,"bx--overflow-menu-options--open",e[0]),O(t,"bx--overflow-menu-options--light",e[6]),O(t,"bx--overflow-menu-options--sm","sm"===e[4]),O(t,"bx--overflow-menu-options--xl","xl"===e[4]),O(t,"bx--breadcrumb-menu-options",!!e[15])},m(a,r){l(a,t,r),s&&s.m(t,null),e[32](t),n=!0},p(e,r){s&&s.p&&(!n||16777216&r[0])&&U(s,a,e,e[24],n?T(a,e[24],r,null):N(e[24]),null),(!n||16384&r[0])&&i(t,"aria-label",e[14]),(!n||32&r[0])&&i(t,"data-floating-menu-direction",e[5]),(!n||512&r[0])&&i(t,"class",e[9]),(!n||512&r[0])&&O(t,"bx--overflow-menu-options",!0),(!n||640&r[0])&&O(t,"bx--overflow-menu--flip",e[7]),(!n||513&r[0])&&O(t,"bx--overflow-menu-options--open",e[0]),(!n||576&r[0])&&O(t,"bx--overflow-menu-options--light",e[6]),(!n||528&r[0])&&O(t,"bx--overflow-menu-options--sm","sm"===e[4]),(!n||528&r[0])&&O(t,"bx--overflow-menu-options--xl","xl"===e[4]),(!n||33280&r[0])&&O(t,"bx--breadcrumb-menu-options",!!e[15])},i(e){n||(o(s,e),n=!0)},o(e){_(s,e),n=!1},d(n){n&&d(t),s&&s.d(n),e[32](null)}}}function yn(e){let t,n,a,s,i,m,u,v;const x=e[25].menu,y=R(x,e,e[24],vn),C=y||function(e){let t,n,a;var s=e[1];function r(e,t){return{props:{"aria-label":e[11],title:e[11],class:"bx--overflow-menu__icon "+e[10]}}}return s&&(t=Y(s,r(e))),{c(){t&&f(t.$$.fragment),n=g()},m(e,s){t&&b(t,e,s),l(e,n,s),a=!0},p(e,a){if(2&a[0]&&s!==(s=e[1])){if(t){c();const e=t;_(e.$$.fragment,1,0,(()=>{$(e,1)})),p()}s?(t=Y(s,r(e)),f(t.$$.fragment),o(t.$$.fragment,1),b(t,n.parentNode,n)):t=null}else if(s){const n={};2048&a[0]&&(n["aria-label"]=e[11]),2048&a[0]&&(n.title=e[11]),1024&a[0]&&(n.class="bx--overflow-menu__icon "+e[10]),t.$set(n)}},i(e){a||(t&&o(t.$$.fragment,e),a=!0)},o(e){t&&_(t.$$.fragment,e),a=!1},d(e){e&&d(n),t&&$(t,e)}}}(e);let M=e[0]&&xn(e),L=[{type:"button"},{"aria-haspopup":""},{"aria-expanded":e[0]},{"aria-label":e[14]},{id:e[12]},e[20]],P={};for(let e=0;e<L.length;e+=1)P=E(P,L[e]);return{c(){t=new X(!1),n=g(),a=h(),s=r("button"),C&&C.c(),i=h(),M&&M.c(),t.a=n,q(s,P),O(s,"bx--overflow-menu",!0),O(s,"bx--overflow-menu--open",e[0]),O(s,"bx--overflow-menu--light",e[6]),O(s,"bx--overflow-menu--sm","sm"===e[4]),O(s,"bx--overflow-menu--xl","xl"===e[4])},m(r,o){t.m(e[13],document.head),S(document.head,n),l(r,a,o),l(r,s,o),C&&C.m(s,null),S(s,i),M&&M.m(s,null),s.autofocus&&s.focus(),e[33](s),m=!0,u||(v=[w(window,"click",e[31]),w(s,"click",e[26]),w(s,"click",e[34]),w(s,"mouseover",e[27]),w(s,"mouseenter",e[28]),w(s,"mouseleave",e[29]),w(s,"keydown",e[30]),w(s,"keydown",e[35]),w(s,"focusout",e[36])],u=!0)},p(e,n){(!m||8192&n[0])&&t.p(e[13]),y?y.p&&(!m||16777216&n[0])&&U(y,x,e,e[24],m?T(x,e[24],n,hn):N(e[24]),vn):C&&C.p&&(!m||3074&n[0])&&C.p(e,m?n:[-1,-1]),e[0]?M?(M.p(e,n),1&n[0]&&o(M,1)):(M=xn(e),M.c(),o(M,1),M.m(s,null)):M&&(c(),_(M,1,1,(()=>{M=null})),p()),q(s,P=G(L,[{type:"button"},{"aria-haspopup":""},(!m||1&n[0])&&{"aria-expanded":e[0]},(!m||16384&n[0])&&{"aria-label":e[14]},(!m||4096&n[0])&&{id:e[12]},1048576&n[0]&&e[20]])),O(s,"bx--overflow-menu",!0),O(s,"bx--overflow-menu--open",e[0]),O(s,"bx--overflow-menu--light",e[6]),O(s,"bx--overflow-menu--sm","sm"===e[4]),O(s,"bx--overflow-menu--xl","xl"===e[4])},i(e){m||(o(C,e),o(M),m=!0)},o(e){_(C,e),_(M),m=!1},d(r){r&&(t.d(),d(a),d(s)),d(n),C&&C.d(r),M&&M.d(),e[33](null),u=!1,F(v)}}}function Cn(e,t,n){let a,s;const r=["size","direction","open","light","flipped","persistentClickItems","menuOptionsClass","icon","iconClass","iconDescription","id","buttonRef","menuRef"];let i,l,o,c=P(t,r),{$$slots:p={},$$scope:_}=t,{size:d}=t,{direction:m="bottom"}=t,{open:u=!1}=t,{light:g=!1}=t,{flipped:f=!1}=t,{persistentClickItems:b=!1}=t,{menuOptionsClass:$}=t,{icon:h=Ae}=t,{iconClass:x}=t,{iconDescription:y="Open and close list of options"}=t,{id:C="ccs-"+Math.random().toString(36)}=t,{buttonRef:S=null}=t,{menuRef:M=null}=t;const w=H("BreadcrumbItem"),L=A(),R=J([]);j(e,R,(e=>n(23,l=e)));const U=J(void 0);j(e,U,(e=>n(38,o=e)));const N=J(void 0),T=J(-1);let B;j(e,T,(e=>n(22,i=e)));let V=!0;K("OverflowMenu",{focusedId:N,add:({id:e,text:t,primaryFocus:n,disabled:a})=>{R.update((s=>(n&&T.set(s.length),[...s,{id:e,text:t,primaryFocus:n,disabled:a,index:s.length}])))},update:e=>{U.set(e)},change:e=>{let t=i+e;t<0?t=l.length-1:t>=l.length&&(t=0);let n=l[t].disabled;for(;n;)t+=e,t<0?t=l.length-1:t>=l.length&&(t=0),n=l[t].disabled;T.set(t)}}),W((()=>{if(o){const{index:e,text:t}=l.filter((e=>e.id===o))[0];L("close",{index:e,text:t}),n(0,u=!1)}if(u){const{width:e,height:t}=S.getBoundingClientRect();n(21,B=e),!V&&i<0&&M.focus(),f&&(n(3,M.style.left="auto",M),n(3,M.style.right=0,M)),"top"===m?(n(3,M.style.top="auto",M),n(3,M.style.bottom=t+"px",M)):"bottom"===m&&n(3,M.style.top=t+"px",M),w&&(n(3,M.style.top=t+10+"px",M),n(3,M.style.left="-11px",M))}u||(R.set([]),U.set(void 0),T.set(0)),V=!1}));return e.$$set=e=>{n(40,t=E(E({},t),I(e))),n(20,c=P(t,r)),"size"in e&&n(4,d=e.size),"direction"in e&&n(5,m=e.direction),"open"in e&&n(0,u=e.open),"light"in e&&n(6,g=e.light),"flipped"in e&&n(7,f=e.flipped),"persistentClickItems"in e&&n(8,b=e.persistentClickItems),"menuOptionsClass"in e&&n(9,$=e.menuOptionsClass),"icon"in e&&n(1,h=e.icon),"iconClass"in e&&n(10,x=e.iconClass),"iconDescription"in e&&n(11,y=e.iconDescription),"id"in e&&n(12,C=e.id),"buttonRef"in e&&n(2,S=e.buttonRef),"menuRef"in e&&n(3,M=e.menuRef),"$$scope"in e&&n(24,_=e.$$scope)},e.$$.update=()=>{n(14,a=t["aria-label"]||"menu"),12582912&e.$$.dirty[0]&&l[i]&&N.set(l[i].id),2101248&e.$$.dirty[0]&&n(13,s=`<style>\n    #${C} .bx--overflow-menu-options.bx--overflow-menu-options:after {\n      width: ${B?B+"px":"2rem"};\n    }\n  </style>`)},w&&n(1,h=Pe),t=I(t),[u,h,S,M,d,m,g,f,b,$,x,y,C,s,a,w,L,R,U,T,c,B,i,l,_,p,function(t){Q.call(this,e,t)},function(t){Q.call(this,e,t)},function(t){Q.call(this,e,t)},function(t){Q.call(this,e,t)},function(t){Q.call(this,e,t)},({target:e})=>{S&&S.contains(e)||M&&!M.contains(e)&&n(0,u=!1)},function(e){v[e?"unshift":"push"]((()=>{M=e,n(3,M)}))},function(e){v[e?"unshift":"push"]((()=>{S=e,n(2,S)}))},e=>{b&&e.stopPropagation(),M&&M.contains(e.target)||(n(0,u=!u),u||L("close"))},e=>{u&&(["ArrowDown","ArrowLeft","ArrowRight","ArrowUp"].includes(e.key)?e.preventDefault():"Escape"===e.key&&(e.stopPropagation(),L("close"),n(0,u=!1),S.focus()))},e=>{u&&(S.contains(e.relatedTarget)||(L("close"),n(0,u=!1)))}]}class Sn extends t{constructor(e){super(),n(this,e,Cn,yn,a,{size:4,direction:5,open:0,light:6,flipped:7,persistentClickItems:8,menuOptionsClass:9,icon:1,iconClass:10,iconDescription:11,id:12,buttonRef:2,menuRef:3},null,[-1,-1])}}function Mn(e,t,n){const a=e.slice();return a[3]=t[n],a[4]=t,a[5]=n,a}function wn(e){let t,n,a,s,c;function p(t){e[1](t,e[3])}let m={labelText:e[3].value};return void 0!==e[3].selected&&(m.checked=e[3].selected),n=new Se({props:m}),v.push((()=>x(n,"checked",p))),n.$on("change",e[2]),{c(){t=r("li"),f(n.$$.fragment),s=h(),i(t,"class","pipen-report-column-selector-item svelte-147jw2r")},m(e,a){l(e,t,a),b(n,t,null),S(t,s),c=!0},p(t,s){e=t;const r={};1&s&&(r.labelText=e[3].value),!a&&1&s&&(a=!0,r.checked=e[3].selected,y((()=>a=!1))),n.$set(r)},i(e){c||(o(n.$$.fragment,e),c=!0)},o(e){_(n.$$.fragment,e),c=!1},d(e){e&&d(t),$(n)}}}function Ln(e){let t,n,a=s(e[0]),r=[];for(let t=0;t<a.length;t+=1)r[t]=wn(Mn(e,a,t));const i=e=>_(r[e],1,1,(()=>{r[e]=null}));return{c(){for(let e=0;e<r.length;e+=1)r[e].c();t=g()},m(e,a){for(let t=0;t<r.length;t+=1)r[t]&&r[t].m(e,a);l(e,t,a),n=!0},p(e,n){if(1&n){let l;for(a=s(e[0]),l=0;l<a.length;l+=1){const s=Mn(e,a,l);r[l]?(r[l].p(s,n),o(r[l],1)):(r[l]=wn(s),r[l].c(),o(r[l],1),r[l].m(t.parentNode,t))}for(c(),l=a.length;l<r.length;l+=1)i(l);p()}},i(e){if(!n){for(let e=0;e<a.length;e+=1)o(r[e]);n=!0}},o(e){r=r.filter(Boolean);for(let e=0;e<r.length;e+=1)_(r[e]);n=!1},d(e){e&&d(t),m(r,e)}}}function Rn(e){let t,n;return t=new Sn({props:{class:"pipen-report-overflowmenu",persistentClickItems:!0,size:"sm",icon:Ee,iconDescription:"Select visible columns ...",$$slots:{default:[Ln]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,[n]){const a={};65&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Un(e,t,n){let{allColumns:a}=t;return e.$$set=e=>{"allColumns"in e&&n(0,a=e.allColumns)},[a,function(t,s){e.$$.not_equal(s.selected,t)&&(s.selected=t,n(0,a))},()=>{n(0,a)}]}class Nn extends t{constructor(e){super(),n(this,e,Un,Rn,a,{allColumns:0})}}function Tn(e){let t,n;return t=new ye({props:{iconDescription:"Right click and save as to download the entire data",tooltipPosition:"top",icon:Be,href:e[2]}}),t.$on("click",e[13]),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};4&n&&(a.href=e[2]),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Pn(e){let t,n,a,s,r,i,m,u,C,S;function M(t){e[17](t)}let w={persistent:!0};function L(t){e[18](t)}void 0!==e[7]&&(w.value=e[7]),t=new Ue({props:w}),v.push((()=>x(t,"value",M)));let R={};void 0!==e[6]&&(R.allColumns=e[6]),s=new Nn({props:R}),v.push((()=>x(s,"allColumns",L))),m=new ye({props:{iconDescription:`Download current ${e[11].length} records (may be partial of the entire data)`,tooltipPosition:"top",icon:Ie}}),m.$on("click",e[14]);let U=!!e[2]&&Tn(e);return{c(){f(t.$$.fragment),a=h(),f(s.$$.fragment),i=h(),f(m.$$.fragment),u=h(),U&&U.c(),C=g()},m(e,n){b(t,e,n),l(e,a,n),b(s,e,n),l(e,i,n),b(m,e,n),l(e,u,n),U&&U.m(e,n),l(e,C,n),S=!0},p(e,a){const i={};!n&&128&a&&(n=!0,i.value=e[7],y((()=>n=!1))),t.$set(i);const l={};!r&&64&a&&(r=!0,l.allColumns=e[6],y((()=>r=!1))),s.$set(l);const d={};2048&a&&(d.iconDescription=`Download current ${e[11].length} records (may be partial of the entire data)`),m.$set(d),e[2]?U?(U.p(e,a),4&a&&o(U,1)):(U=Tn(e),U.c(),o(U,1),U.m(C.parentNode,C)):U&&(c(),_(U,1,1,(()=>{U=null})),p())},i(e){S||(o(t.$$.fragment,e),o(s.$$.fragment,e),o(m.$$.fragment,e),o(U),S=!0)},o(e){_(t.$$.fragment,e),_(s.$$.fragment,e),_(m.$$.fragment,e),_(U),S=!1},d(e){e&&(d(a),d(i),d(u),d(C)),$(t,e),$(s,e),$(m,e),U&&U.d(e)}}}function An(e){let t,n;return t=new Re({props:{$$slots:{default:[Pn]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};134219972&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function En(e){let t,n;return t=new Le({props:{size:e[12],$$slots:{default:[An]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};134219972&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function In(e){let t;return{c(){t=r("p"),t.textContent='You must use "Save as" or "Save link as" from the context menu (by right-clicking the button) to download the entire data.'},m(e,n){l(e,t,n)},p:L,d(e){e&&d(t)}}}function Bn(e){let t,n,a,s,r,i,c,p,m;const u=[{size:e[0]},{zebra:e[3]},{sortable:e[4]},{page:e[8]},{pageSize:e[9]},{rows:e[11]},{headers:e[1]},e[15]];let g={$$slots:{default:[En]},$$scope:{ctx:e}};for(let e=0;e<u.length;e+=1)g=E(g,u[e]);function C(t){e[19](t)}function S(t){e[20](t)}t=new Me({props:g});let M={pageSizes:e[5],totalItems:e[11].length};function w(t){e[21](t)}void 0!==e[9]&&(M.pageSize=e[9]),void 0!==e[8]&&(M.page=e[8]),a=new we({props:M}),v.push((()=>x(a,"pageSize",C))),v.push((()=>x(a,"page",S)));let L={passiveModal:!0,modalHeading:"Downloading entire data",$$slots:{default:[In]},$$scope:{ctx:e}};return void 0!==e[10]&&(L.open=e[10]),c=new _e({props:L}),v.push((()=>x(c,"open",w))),c.$on("open",e[22]),c.$on("close",e[23]),{c(){f(t.$$.fragment),n=h(),f(a.$$.fragment),i=h(),f(c.$$.fragment)},m(e,s){b(t,e,s),l(e,n,s),b(a,e,s),l(e,i,s),b(c,e,s),m=!0},p(e,[n]){const i=35611&n?G(u,[1&n&&{size:e[0]},8&n&&{zebra:e[3]},16&n&&{sortable:e[4]},256&n&&{page:e[8]},512&n&&{pageSize:e[9]},2048&n&&{rows:e[11]},2&n&&{headers:e[1]},32768&n&&D(e[15])]):{};134219972&n&&(i.$$scope={dirty:n,ctx:e}),t.$set(i);const l={};32&n&&(l.pageSizes=e[5]),2048&n&&(l.totalItems=e[11].length),!s&&512&n&&(s=!0,l.pageSize=e[9],y((()=>s=!1))),!r&&256&n&&(r=!0,l.page=e[8],y((()=>r=!1))),a.$set(l);const o={};134217728&n&&(o.$$scope={dirty:n,ctx:e}),!p&&1024&n&&(p=!0,o.open=e[10],y((()=>p=!1))),c.$set(o)},i(e){m||(o(t.$$.fragment,e),o(a.$$.fragment,e),o(c.$$.fragment,e),m=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),_(c.$$.fragment,e),m=!1},d(e){e&&(d(n),d(i)),$(t,e),$(a,e),$(c,e)}}}function Vn(e,t,n){const a=["size","src","zebra","sortable","rows","headers","pageSizes"];let s=P(t,a),{size:r="sm"}=t,{src:i}=t;r={sm:"short",small:"short",short:"short",md:"medium",medium:"medium",lg:"tall",large:"tall",tall:"tall",compact:"compact"}[r];let l={short:"sm",compact:"sm",medium:"default",tall:"default"}[r],{zebra:o=!0}=t,{sortable:c=!0}=t,{rows:p=[]}=t,{headers:_=[]}=t,{pageSizes:d=[20,50,75,100]}=t,m=_.map((e=>({...e,selected:!0}))),u=1,g=d[0],f="",b=!1,$=p;return e.$$set=e=>{t=E(E({},t),I(e)),n(15,s=P(t,a)),"size"in e&&n(0,r=e.size),"src"in e&&n(2,i=e.src),"zebra"in e&&n(3,o=e.zebra),"sortable"in e&&n(4,c=e.sortable),"rows"in e&&n(16,p=e.rows),"headers"in e&&n(1,_=e.headers),"pageSizes"in e&&n(5,d=e.pageSizes)},e.$$.update=()=>{64&e.$$.dirty&&n(1,_=m.filter((e=>e.selected))),65664&e.$$.dirty&&(f.length>0?n(11,$=p.filter((e=>Object.values(e).filter((e=>!!e&&e.toString().includes(f))).length>0))):n(11,$=p))},[r,_,i,o,c,d,m,f,u,g,b,$,l,e=>{e.preventDefault(),n(10,b=!0)},()=>{((e,t)=>{const n=document.createElement("a");n.href=e,n.download=t,document.body.appendChild(n),n.click(),document.body.removeChild(n)})("data:application/vnd.ms-excel;charset=utf-8,\ufeff"+encodeURIComponent(function(){const e=_.map((e=>e.value)),t=[];return $.forEach((e=>{const n=_.map((t=>`"${e[t.key]}"`));t.push(n.join(","))})),e.join(",")+"\r\n"+t.join("\r\n")}()),i&&i.length>0?i.split("/").pop().split(".")[0]+".xls":"data-subset.xls")},s,p,function(e){f=e,n(7,f)},function(e){m=e,n(6,m)},function(e){g=e,n(9,g)},function(e){u=e,n(8,u)},function(e){b=e,n(10,b)},function(t){Q.call(this,e,t)},function(t){Q.call(this,e,t)}]}class zn extends t{constructor(e){super(),n(this,e,Vn,Bn,a,{size:0,src:2,zebra:3,sortable:4,rows:16,headers:1,pageSizes:5})}}function kn(e){let t,n,a;const s=[{headers:e[0]},{rows:e[1]},e[3]];let i={};for(let e=0;e<s.length;e+=1)i=E(i,s[e]);n=new zn({props:i});let c=[e[2]],p={};for(let e=0;e<c.length;e+=1)p=E(p,c[e]);return{c(){t=r("div"),f(n.$$.fragment),q(t,p)},m(e,s){l(e,t,s),b(n,t,null),a=!0},p(e,a){const r=11&a?G(s,[1&a&&{headers:e[0]},2&a&&{rows:e[1]},8&a&&D(e[3])]):{};n.$set(r),q(t,p=G(c,[4&a&&e[2]]))},i(e){a||(o(n.$$.fragment,e),a=!0)},o(e){_(n.$$.fragment,e),a=!1},d(e){e&&d(t),$(n)}}}function qn(e){let t,n;const a=[{headers:e[0]},{rows:e[1]},e[3]];let s={};for(let e=0;e<a.length;e+=1)s=E(s,a[e]);return t=new zn({props:s}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const s=11&n?G(a,[1&n&&{headers:e[0]},2&n&&{rows:e[1]},8&n&&D(e[3])]):{};t.$set(s)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Gn(e){let t,n,a,s;const r=[qn,kn],i=[];function m(e,t){return e[2]?1:0}return t=m(e),n=i[t]=r[t](e),{c(){n.c(),a=g()},m(e,n){i[t].m(e,n),l(e,a,n),s=!0},p(e,[s]){let l=t;t=m(e),t===l?i[t].p(e,s):(c(),_(i[l],1,1,(()=>{i[l]=null})),p(),n=i[t],n?n.p(e,s):(n=i[t]=r[t](e),n.c()),o(n,1),n.m(a.parentNode,a))},i(e){s||(o(n),s=!0)},o(e){_(n),s=!1},d(e){e&&d(a),i[t].d(e)}}}function Dn(e,t,n){const a=["frameProps","data","headers","rows"];let s=P(t,a),{frameProps:r={class:"pipen-report-datatable-frame"}}=t,{data:i={}}=t,{headers:l=[]}=t,{rows:o=[]}=t;if(l.length+o.length>0&&i.length>0)throw"Can only specify `data` or `headers`/`rows` for `DataTable`!";return i.length>0&&(o=i,l=Object.keys(o[0]).filter((e=>"id"!==e)).map((e=>({key:e,value:e})))),e.$$set=e=>{t=E(E({},t),I(e)),n(3,s=P(t,a)),"frameProps"in e&&n(2,r=e.frameProps),"data"in e&&n(4,i=e.data),"headers"in e&&n(0,l=e.headers),"rows"in e&&n(1,o=e.rows)},[l,o,r,s,i]}class On extends t{constructor(e){super(),n(this,e,Dn,Gn,a,{frameProps:2,data:4,headers:0,rows:1})}}function Fn(e){let t,n;return{c(){t=r("div"),n=C(e[0]),i(t,"class","pipen-report-descr-title svelte-1xzxuva")},m(e,a){l(e,t,a),S(t,n)},p(e,t){1&t&&M(n,e[0])},d(e){e&&d(t)}}}function Hn(e){let t,n,a,s=e[0]&&Fn(e);const i=e[3].default,c=R(i,e,e[2],null);let p=[e[1]],m={};for(let e=0;e<p.length;e+=1)m=E(m,p[e]);return{c(){t=r("div"),s&&s.c(),n=h(),c&&c.c(),q(t,m),O(t,"svelte-1xzxuva",!0)},m(e,r){l(e,t,r),s&&s.m(t,null),S(t,n),c&&c.m(t,null),a=!0},p(e,[r]){e[0]?s?s.p(e,r):(s=Fn(e),s.c(),s.m(t,n)):s&&(s.d(1),s=null),c&&c.p&&(!a||4&r)&&U(c,i,e,e[2],a?T(i,e[2],r,null):N(e[2]),null),q(t,m=G(p,[2&r&&e[1]])),O(t,"svelte-1xzxuva",!0)},i(e){a||(o(c,e),a=!0)},o(e){_(c,e),a=!1},d(e){e&&d(t),s&&s.d(),c&&c.d(e)}}}function jn(e,t,n){const a=["title"];let s=P(t,a),{$$slots:r={},$$scope:i}=t,{title:l}=t,{klass:o,...c}=s;return o=o?`pipen-report-descr ${o}`:"pipen-report-descr",c||(c={}),c.class=o,e.$$set=e=>{t=E(E({},t),I(e)),n(5,s=P(t,a)),"title"in e&&n(0,l=e.title),"$$scope"in e&&n(2,i=e.$$scope)},[l,c,i,r]}class Kn extends t{constructor(e){super(),n(this,e,jn,Hn,a,{title:0})}}function Wn(e){let t,n,a;return{c(){t=C("Distribution of known gene segments following the "),n=r("a"),n.textContent="IMGT",a=C(" nomenclature."),i(n,"href","http://www.imgt.org/IMGTrepertoire/LocusGenes/"),i(n,"target","_blank")},m(e,s){l(e,t,s),l(e,n,s),l(e,a,s)},p:L,d(e){e&&(d(t),d(n),d(a))}}}function Yn(e){let t;return{c(){t=C("Spectratype is a useful way to represent distributions of genes per sequence length.")},m(e,n){l(e,t,n)},d(e){e&&d(t)}}}function Xn(e){let t,n,a;return{c(){t=C("Diversity estimation using "),n=r("code"),n.textContent="gini",a=C(": the Gini coefficient measures the inequality among values of a frequency distribution (for example levels of income). A Gini coefficient of zero expresses perfect equality, where all values are the same (for example, where everyone has the same income). A Gini coefficient of one (or 100 percents ) expresses maximal inequality among values (for example where only one person has all the income).")},m(e,s){l(e,t,s),l(e,n,s),l(e,a,s)},p:L,d(e){e&&(d(t),d(n),d(a))}}}function Jn(e){let t,n,a,s;return t=new ue({props:{label:"Diversity Plot"}}),a=new ue({props:{label:"Diversity Table"}}),{c(){f(t.$$.fragment),n=h(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p:L,i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function Qn(e){let t,n;return t=new $n({props:{src:"../Immunarch/samples.immunarch.immunarch/diversity/Gini_Coef/diversity.png",width:800,height:800}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p:L,i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Zn(e){let t,n;return t=new On({props:{src:"../Immunarch/samples.immunarch.immunarch/diversity/Gini_Coef/diversity.txt",data:[{Sample:"C1",V1:.2426,id:0},{Sample:"C3",V1:.3241,id:1},{Sample:"CT1",V1:.2044,id:2},{Sample:"CT2",V1:.1856,id:3},{Sample:"NC1",V1:.0653,id:4},{Sample:"NC3",V1:.307,id:5}]}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p:L,i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function ea(e){let t,n,a,s;return t=new ge({props:{$$slots:{default:[Qn]},$$scope:{ctx:e}}}),a=new ge({props:{$$slots:{default:[Zn]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=h(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function ta(e){let t,n,a;return{c(){t=C("Diversity estimation using "),n=r("code"),n.textContent="gini",a=C(": the Gini coefficient measures the inequality among values of a frequency distribution (for example levels of income). A Gini coefficient of zero expresses perfect equality, where all values are the same (for example, where everyone has the same income). A Gini coefficient of one (or 100 percents ) expresses maximal inequality among values (for example where only one person has all the income).")},m(e,s){l(e,t,s),l(e,n,s),l(e,a,s)},p:L,d(e){e&&(d(t),d(n),d(a))}}}function na(e){let t,n,a,s;return t=new ue({props:{label:"Diversity Plot"}}),a=new ue({props:{label:"Diversity Table"}}),{c(){f(t.$$.fragment),n=h(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p:L,i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function aa(e){let t,n;return t=new $n({props:{src:"../Immunarch/samples.immunarch.immunarch/diversity/Gini_Coef_in_Treg/diversity.png",width:800,height:800}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p:L,i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function sa(e){let t,n;return t=new On({props:{src:"../Immunarch/samples.immunarch.immunarch/diversity/Gini_Coef_in_Treg/diversity.txt",data:[{Sample:"C1",V1:.0705,id:0},{Sample:"C3",V1:.0326,id:1},{Sample:"CT1",V1:.0221,id:2},{Sample:"CT2",V1:.0227,id:3},{Sample:"NC1",V1:.0177,id:4},{Sample:"NC3",V1:.0086,id:5}]}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p:L,i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function ra(e){let t,n,a,s;return t=new ge({props:{$$slots:{default:[aa]},$$scope:{ctx:e}}}),a=new ge({props:{$$slots:{default:[sa]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=h(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function ia(e){let t,n,a,s,c,p,m,u,g,v,x,y,C,M,w,L,R,U,N,T,P,A,E,I,B,V,z,k,q,G,D,O,F,H,j,K,W,Y,X,J,Q,ee,te,ne,ae,se,re,ie,le,oe,ce,pe,_e,de,ue,ge,fe,be,$e,he,ve,xe,ye,Ce,Se,Me,we,Le,Re,Ue,Ne,Te,Pe,Ae,Ee,Ie,Be,Ve,ze,ke,qe,Ge,De,Oe,Fe,He,je,Ke,We,Ye,Xe,Je,Qe,Ze,et,tt,nt,at;return c=new Kn({props:{class:"pipen-report-table-image-descr",$$slots:{default:[Wn]},$$scope:{ctx:e}}}),m=new $n({props:{src:"../Immunarch/samples.immunarch.immunarch/gene_usage/DEFAULT/DEFAULT.png",class:"pipen-report-table-image",width:1e3,height:1e3}}),y=new Kn({props:{$$slots:{default:[Yn]},$$scope:{ctx:e}}}),N=new Kn({props:{title:"C1",class:"pipen-report-table-image-descr"}}),P=new $n({props:{src:"../Immunarch/samples.immunarch.immunarch/spectratyping/By_Clonotype/C1.spect.png",class:"pipen-report-table-image",width:1e3,height:1e3}}),I=new Kn({props:{title:"C3",class:"pipen-report-table-image-descr"}}),V=new $n({props:{src:"../Immunarch/samples.immunarch.immunarch/spectratyping/By_Clonotype/C3.spect.png",class:"pipen-report-table-image",width:1e3,height:1e3}}),q=new Kn({props:{title:"CT1",class:"pipen-report-table-image-descr"}}),D=new $n({props:{src:"../Immunarch/samples.immunarch.immunarch/spectratyping/By_Clonotype/CT1.spect.png",class:"pipen-report-table-image",width:1e3,height:1e3}}),H=new Kn({props:{title:"CT2",class:"pipen-report-table-image-descr"}}),K=new $n({props:{src:"../Immunarch/samples.immunarch.immunarch/spectratyping/By_Clonotype/CT2.spect.png",class:"pipen-report-table-image",width:1e3,height:1e3}}),X=new Kn({props:{title:"NC1",class:"pipen-report-table-image-descr"}}),Q=new $n({props:{src:"../Immunarch/samples.immunarch.immunarch/spectratyping/By_Clonotype/NC1.spect.png",class:"pipen-report-table-image",width:1e3,height:1e3}}),ne=new Kn({props:{title:"NC3",class:"pipen-report-table-image-descr"}}),se=new $n({props:{src:"../Immunarch/samples.immunarch.immunarch/spectratyping/By_Clonotype/NC3.spect.png",class:"pipen-report-table-image",width:1e3,height:1e3}}),_e=new Kn({props:{title:"C1",class:"pipen-report-table-image-descr"}}),ue=new $n({props:{src:"../Immunarch/samples.immunarch.immunarch/spectratyping/By_Num_Clones/C1.spect.png",class:"pipen-report-table-image",width:1e3,height:1e3}}),be=new Kn({props:{title:"C3",class:"pipen-report-table-image-descr"}}),he=new $n({props:{src:"../Immunarch/samples.immunarch.immunarch/spectratyping/By_Num_Clones/C3.spect.png",class:"pipen-report-table-image",width:1e3,height:1e3}}),ye=new Kn({props:{title:"CT1",class:"pipen-report-table-image-descr"}}),Se=new $n({props:{src:"../Immunarch/samples.immunarch.immunarch/spectratyping/By_Num_Clones/CT1.spect.png",class:"pipen-report-table-image",width:1e3,height:1e3}}),Le=new Kn({props:{title:"CT2",class:"pipen-report-table-image-descr"}}),Ue=new $n({props:{src:"../Immunarch/samples.immunarch.immunarch/spectratyping/By_Num_Clones/CT2.spect.png",class:"pipen-report-table-image",width:1e3,height:1e3}}),Pe=new Kn({props:{title:"NC1",class:"pipen-report-table-image-descr"}}),Ee=new $n({props:{src:"../Immunarch/samples.immunarch.immunarch/spectratyping/By_Num_Clones/NC1.spect.png",class:"pipen-report-table-image",width:1e3,height:1e3}}),Ve=new Kn({props:{title:"NC3",class:"pipen-report-table-image-descr"}}),ke=new $n({props:{src:"../Immunarch/samples.immunarch.immunarch/spectratyping/By_Num_Clones/NC3.spect.png",class:"pipen-report-table-image",width:1e3,height:1e3}}),Ke=new Kn({props:{$$slots:{default:[Xn]},$$scope:{ctx:e}}}),Ye=new me({props:{$$slots:{content:[ea],default:[Jn]},$$scope:{ctx:e}}}),et=new Kn({props:{$$slots:{default:[ta]},$$scope:{ctx:e}}}),nt=new me({props:{$$slots:{content:[ra],default:[na]},$$scope:{ctx:e}}}),{c(){t=r("h1"),t.textContent="Gene Usage",n=r("a"),n.innerHTML="",a=h(),s=r("div"),f(c.$$.fragment),p=h(),f(m.$$.fragment),u=h(),g=r("h1"),g.textContent="Spectratyping",v=r("a"),v.innerHTML="",x=h(),f(y.$$.fragment),C=h(),M=r("h2"),M.textContent="By_Clonotype",w=r("a"),w.innerHTML="",L=h(),R=r("div"),U=r("div"),f(N.$$.fragment),T=h(),f(P.$$.fragment),A=h(),E=r("div"),f(I.$$.fragment),B=h(),f(V.$$.fragment),z=h(),k=r("div"),f(q.$$.fragment),G=h(),f(D.$$.fragment),O=h(),F=r("div"),f(H.$$.fragment),j=h(),f(K.$$.fragment),W=h(),Y=r("div"),f(X.$$.fragment),J=h(),f(Q.$$.fragment),ee=h(),te=r("div"),f(ne.$$.fragment),ae=h(),f(se.$$.fragment),re=h(),ie=r("h2"),ie.textContent="By_Num_Clones",le=r("a"),le.innerHTML="",oe=h(),ce=r("div"),pe=r("div"),f(_e.$$.fragment),de=h(),f(ue.$$.fragment),ge=h(),fe=r("div"),f(be.$$.fragment),$e=h(),f(he.$$.fragment),ve=h(),xe=r("div"),f(ye.$$.fragment),Ce=h(),f(Se.$$.fragment),Me=h(),we=r("div"),f(Le.$$.fragment),Re=h(),f(Ue.$$.fragment),Ne=h(),Te=r("div"),f(Pe.$$.fragment),Ae=h(),f(Ee.$$.fragment),Ie=h(),Be=r("div"),f(Ve.$$.fragment),ze=h(),f(ke.$$.fragment),qe=h(),Ge=r("h1"),Ge.textContent="Diversity Estimation",De=r("a"),De.innerHTML="",Oe=h(),Fe=r("h2"),Fe.textContent="Gini_Coef",He=r("a"),He.innerHTML="",je=h(),f(Ke.$$.fragment),We=h(),f(Ye.$$.fragment),Xe=h(),Je=r("h2"),Je.textContent="Gini_Coef_in_Treg",Qe=r("a"),Qe.innerHTML="",Ze=h(),f(et.$$.fragment),tt=h(),f(nt.$$.fragment),i(n,"id","prt-h1-3-gene-usage"),i(n,"class","pipen-report-toc-anchor"),i(v,"id","prt-h1-4-spectratyping"),i(v,"class","pipen-report-toc-anchor"),i(w,"id","prt-h2-7-by-clonotype"),i(w,"class","pipen-report-toc-anchor"),i(R,"class","pipen-report-table-of-images"),Z(R,"grid-template-columns","repeat(2, auto)"),i(le,"id","prt-h2-8-by-num-clones"),i(le,"class","pipen-report-toc-anchor"),i(ce,"class","pipen-report-table-of-images"),Z(ce,"grid-template-columns","repeat(2, auto)"),i(De,"id","prt-h1-5-diversity-estimation"),i(De,"class","pipen-report-toc-anchor"),i(He,"id","prt-h2-9-gini-coef"),i(He,"class","pipen-report-toc-anchor"),i(Qe,"id","prt-h2-10-gini-coef-in-treg"),i(Qe,"class","pipen-report-toc-anchor")},m(e,r){l(e,t,r),l(e,n,r),l(e,a,r),l(e,s,r),b(c,s,null),S(s,p),b(m,s,null),l(e,u,r),l(e,g,r),l(e,v,r),l(e,x,r),b(y,e,r),l(e,C,r),l(e,M,r),l(e,w,r),l(e,L,r),l(e,R,r),S(R,U),b(N,U,null),S(U,T),b(P,U,null),S(R,A),S(R,E),b(I,E,null),S(E,B),b(V,E,null),S(R,z),S(R,k),b(q,k,null),S(k,G),b(D,k,null),S(R,O),S(R,F),b(H,F,null),S(F,j),b(K,F,null),S(R,W),S(R,Y),b(X,Y,null),S(Y,J),b(Q,Y,null),S(R,ee),S(R,te),b(ne,te,null),S(te,ae),b(se,te,null),l(e,re,r),l(e,ie,r),l(e,le,r),l(e,oe,r),l(e,ce,r),S(ce,pe),b(_e,pe,null),S(pe,de),b(ue,pe,null),S(ce,ge),S(ce,fe),b(be,fe,null),S(fe,$e),b(he,fe,null),S(ce,ve),S(ce,xe),b(ye,xe,null),S(xe,Ce),b(Se,xe,null),S(ce,Me),S(ce,we),b(Le,we,null),S(we,Re),b(Ue,we,null),S(ce,Ne),S(ce,Te),b(Pe,Te,null),S(Te,Ae),b(Ee,Te,null),S(ce,Ie),S(ce,Be),b(Ve,Be,null),S(Be,ze),b(ke,Be,null),l(e,qe,r),l(e,Ge,r),l(e,De,r),l(e,Oe,r),l(e,Fe,r),l(e,He,r),l(e,je,r),b(Ke,e,r),l(e,We,r),b(Ye,e,r),l(e,Xe,r),l(e,Je,r),l(e,Qe,r),l(e,Ze,r),b(et,e,r),l(e,tt,r),b(nt,e,r),at=!0},p(e,[t]){const n={};1&t&&(n.$$scope={dirty:t,ctx:e}),c.$set(n);const a={};1&t&&(a.$$scope={dirty:t,ctx:e}),y.$set(a);const s={};1&t&&(s.$$scope={dirty:t,ctx:e}),Ke.$set(s);const r={};1&t&&(r.$$scope={dirty:t,ctx:e}),Ye.$set(r);const i={};1&t&&(i.$$scope={dirty:t,ctx:e}),et.$set(i);const l={};1&t&&(l.$$scope={dirty:t,ctx:e}),nt.$set(l)},i(e){at||(o(c.$$.fragment,e),o(m.$$.fragment,e),o(y.$$.fragment,e),o(N.$$.fragment,e),o(P.$$.fragment,e),o(I.$$.fragment,e),o(V.$$.fragment,e),o(q.$$.fragment,e),o(D.$$.fragment,e),o(H.$$.fragment,e),o(K.$$.fragment,e),o(X.$$.fragment,e),o(Q.$$.fragment,e),o(ne.$$.fragment,e),o(se.$$.fragment,e),o(_e.$$.fragment,e),o(ue.$$.fragment,e),o(be.$$.fragment,e),o(he.$$.fragment,e),o(ye.$$.fragment,e),o(Se.$$.fragment,e),o(Le.$$.fragment,e),o(Ue.$$.fragment,e),o(Pe.$$.fragment,e),o(Ee.$$.fragment,e),o(Ve.$$.fragment,e),o(ke.$$.fragment,e),o(Ke.$$.fragment,e),o(Ye.$$.fragment,e),o(et.$$.fragment,e),o(nt.$$.fragment,e),at=!0)},o(e){_(c.$$.fragment,e),_(m.$$.fragment,e),_(y.$$.fragment,e),_(N.$$.fragment,e),_(P.$$.fragment,e),_(I.$$.fragment,e),_(V.$$.fragment,e),_(q.$$.fragment,e),_(D.$$.fragment,e),_(H.$$.fragment,e),_(K.$$.fragment,e),_(X.$$.fragment,e),_(Q.$$.fragment,e),_(ne.$$.fragment,e),_(se.$$.fragment,e),_(_e.$$.fragment,e),_(ue.$$.fragment,e),_(be.$$.fragment,e),_(he.$$.fragment,e),_(ye.$$.fragment,e),_(Se.$$.fragment,e),_(Le.$$.fragment,e),_(Ue.$$.fragment,e),_(Pe.$$.fragment,e),_(Ee.$$.fragment,e),_(Ve.$$.fragment,e),_(ke.$$.fragment,e),_(Ke.$$.fragment,e),_(Ye.$$.fragment,e),_(et.$$.fragment,e),_(nt.$$.fragment,e),at=!1},d(e){e&&(d(t),d(n),d(a),d(s),d(u),d(g),d(v),d(x),d(C),d(M),d(w),d(L),d(R),d(re),d(ie),d(le),d(oe),d(ce),d(qe),d(Ge),d(De),d(Oe),d(Fe),d(He),d(je),d(We),d(Xe),d(Je),d(Qe),d(Ze),d(tt)),$(c),$(m),$(y,e),$(N),$(P),$(I),$(V),$(q),$(D),$(H),$(K),$(X),$(Q),$(ne),$(se),$(_e),$(ue),$(be),$(he),$(ye),$(Se),$(Le),$(Ue),$(Pe),$(Ee),$(Ve),$(ke),$(Ke,e),$(Ye,e),$(et,e),$(nt,e)}}}class la extends t{constructor(e){super(),n(this,e,null,ia,a,{})}}var oa=[{slug:"prt-h1-0-exploratory-analysis",text:"Exploratory Analysis",children:[{slug:"prt-h2-0-cdr3-length-distribution",text:"CDR3 length distribution",children:[],page:0},{slug:"prt-h2-1-clonotype-volume-number-of-clonotypes",text:"Clonotype volume (Number of clonotypes)",children:[],page:0},{slug:"prt-h2-2-clonotype-abundances",text:"Clonotype abundances",children:[],page:0}],page:0},{slug:"prt-h1-1-clonality-analysis",text:"Clonality Analysis",children:[{slug:"prt-h2-3-top-clones",text:"Top Clones",children:[],page:0},{slug:"prt-h2-4-rare-clones",text:"Rare Clones",children:[],page:0},{slug:"prt-h2-5-clonal-space-homeostasis",text:"Clonal Space Homeostasis",children:[],page:0}],page:0},{slug:"prt-h1-2-repertoire-overlaps",text:"Repertoire Overlaps",children:[{slug:"prt-h2-6-public",text:"public",children:[],page:0}],page:0},{slug:"prt-h1-3-gene-usage",text:"Gene Usage",children:[],page:1},{slug:"prt-h1-4-spectratyping",text:"Spectratyping",children:[{slug:"prt-h2-7-by-clonotype",text:"By_Clonotype",children:[],page:1},{slug:"prt-h2-8-by-num-clones",text:"By_Num_Clones",children:[],page:1}],page:1},{slug:"prt-h1-5-diversity-estimation",text:"Diversity Estimation",children:[{slug:"prt-h2-9-gini-coef",text:"Gini_Coef",children:[],page:1},{slug:"prt-h2-10-gini-coef-in-treg",text:"Gini_Coef_in_Treg",children:[],page:1}],page:1},{slug:"prt-h1-6-kmer-and-sequence-motif-analysis",text:"Kmer and sequence motif analysis",children:[],page:2},{slug:"prt-h1-7-v-j-junction-circos-plots",text:"V-J Junction Circos Plots",children:[],page:2}],ca={pipeline:{name:"Immunopipe",desc:"Immunopipe (v1.4.2): Integrative analysis for scRNA-seq and scTCR-seq data"},versions:'\n    <a href="https://github.com/pwwang/pipen" target="_blank">\n        pipen v0.15.2\n    </a> and\n    <a href="https://github.com/pwwang/pipen-report" target="_blank">\n        pipen-report v0.20.1\n    </a>\n',entries:[{name:"SampleInfo",desc:"List sample information and perform statistics",npages:1,report_toc:!0,order:0,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n[1] ggrepel_0.9.5    ggprism_1.0.5    ggplot2_3.5.1    tidyr_1.3.1     \n[5] dplyr_1.1.4      tidyselect_1.2.1 jsonlite_1.8.8   rlang_1.1.4     \n[9] logger_0.3.0    \n\nloaded via a namespace (and not attached):\n [1] vctrs_0.6.5      cli_3.6.3        purrr_1.0.2      generics_0.1.3  \n [5] labeling_0.4.3   glue_1.7.0       colorspace_2.1-1 scales_1.3.0    \n [9] fansi_1.0.6      grid_4.3.3       munsell_0.5.1    tibble_3.2.1    \n[13] lifecycle_1.0.4  compiler_4.3.3   Rcpp_1.0.13      pkgconfig_2.0.3 \n[17] farver_2.1.2     digest_0.6.36    R6_2.5.1         utf8_1.2.4      \n[21] pillar_1.9.0     magrittr_2.0.3   withr_3.0.1      gtable_0.3.5    \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 1.73\nuser 10.62\nsys 0.10\npercent 617%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"SeuratPreparing",desc:"Load, prepare and apply QC to data, using `Seurat`",npages:1,report_toc:!0,order:2,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats4    stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] scDblFinder_1.16.0          SingleCellExperiment_1.24.0\n [3] SummarizedExperiment_1.32.0 Biobase_2.62.0             \n [5] GenomicRanges_1.54.1        GenomeInfoDb_1.38.1        \n [7] IRanges_2.36.0              S4Vectors_0.40.2           \n [9] BiocGenerics_0.48.1         MatrixGenerics_1.14.0      \n[11] matrixStats_1.1.0           dplyr_1.1.4                \n[13] ggplot2_3.5.1               bracer_1.2.1               \n[15] future_1.34.0               Seurat_5.1.0               \n[17] SeuratObject_5.0.2          sp_2.1-4                   \n[19] digest_0.6.36               jsonlite_1.8.8             \n[21] rlang_1.1.4                 logger_0.3.0               \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22          splines_4.3.3            \n  [3] later_1.3.2               BiocIO_1.12.0            \n  [5] bitops_1.0-8              tibble_3.2.1             \n  [7] R.oo_1.26.0               polyclip_1.10-7          \n  [9] XML_3.99-0.17             fastDummies_1.7.3        \n [11] lifecycle_1.0.4           edgeR_4.0.16             \n [13] globals_0.16.3            lattice_0.22-6           \n [15] MASS_7.3-60.0.1           magrittr_2.0.3           \n [17] limma_3.58.1              plotly_4.10.4            \n [19] yaml_2.3.10               metapod_1.10.0           \n [21] httpuv_1.6.15             sctransform_0.4.1        \n [23] spam_2.10-0               spatstat.sparse_3.1-0    \n [25] reticulate_1.38.0         cowplot_1.1.3            \n [27] pbapply_1.7-2             RColorBrewer_1.1-3       \n [29] abind_1.4-5               zlibbioc_1.48.0          \n [31] Rtsne_0.17                purrr_1.0.2              \n [33] R.utils_2.12.3            RCurl_1.98-1.16          \n [35] GenomeInfoDbData_1.2.11   ggrepel_0.9.5            \n [37] irlba_2.3.5.1             listenv_0.9.1            \n [39] spatstat.utils_3.0-5      goftest_1.2-3            \n [41] RSpectra_0.16-1           dqrng_0.3.2              \n [43] spatstat.random_3.2-3     fitdistrplus_1.1-11      \n [45] parallelly_1.38.0         DelayedMatrixStats_1.24.0\n [47] leiden_0.4.3.1            codetools_0.2-20         \n [49] DelayedArray_0.28.0       scuttle_1.12.0           \n [51] tidyselect_1.2.1          farver_2.1.2             \n [53] viridis_0.6.5             ScaledMatrix_1.10.0      \n [55] spatstat.explore_3.2-6    GenomicAlignments_1.38.0 \n [57] BiocNeighbors_1.20.0      progressr_0.14.0         \n [59] scater_1.30.1             ggridges_0.5.6           \n [61] survival_3.7-0            systemfonts_1.1.0        \n [63] tools_4.3.3               ragg_1.3.2               \n [65] ica_1.0-3                 Rcpp_1.0.13              \n [67] glue_1.7.0                gridExtra_2.3            \n [69] SparseArray_1.2.2         withr_3.0.1              \n [71] fastmap_1.2.0             bluster_1.12.0           \n [73] fansi_1.0.6               rsvd_1.0.5               \n [75] R6_2.5.1                  mime_0.12                \n [77] textshaping_0.4.0         colorspace_2.1-1         \n [79] scattermore_1.2           tensor_1.5               \n [81] spatstat.data_3.1-2       R.methodsS3_1.8.2        \n [83] RhpcBLASctl_0.23-42       utf8_1.2.4               \n [85] tidyr_1.3.1               generics_0.1.3           \n [87] data.table_1.15.4         rtracklayer_1.62.0       \n [89] httr_1.4.7                htmlwidgets_1.6.4        \n [91] S4Arrays_1.2.0            uwot_0.1.16              \n [93] pkgconfig_2.0.3           gtable_0.3.5             \n [95] lmtest_0.9-40             XVector_0.42.0           \n [97] htmltools_0.5.8.1         dotCall64_1.1-1          \n [99] scales_1.3.0              png_0.1-8                \n[101] harmony_1.2.0             scran_1.30.0             \n[103] reshape2_1.4.4            rjson_0.2.21             \n[105] nlme_3.1-165              zoo_1.8-12               \n[107] stringr_1.5.1             KernSmooth_2.23-24       \n[109] vipor_0.4.7               parallel_4.3.3           \n[111] miniUI_0.1.1.1            restfulr_0.0.15          \n[113] pillar_1.9.0              grid_4.3.3               \n[115] vctrs_0.6.5               RANN_2.6.1               \n[117] promises_1.3.0            BiocSingular_1.18.0      \n[119] beachmat_2.18.0           xtable_1.8-4             \n[121] cluster_2.1.6             beeswarm_0.4.0           \n[123] locfit_1.5-9.9            cli_3.6.3                \n[125] compiler_4.3.3            Rsamtools_2.18.0         \n[127] crayon_1.5.3              future.apply_1.11.2      \n[129] labeling_0.4.3            ggbeeswarm_0.7.2         \n[131] plyr_1.8.9                stringi_1.8.4            \n[133] viridisLite_0.4.2         deldir_2.0-4             \n[135] BiocParallel_1.36.0       munsell_0.5.1            \n[137] Biostrings_2.70.1         lazyeval_0.2.2           \n[139] spatstat.geom_3.2-9       Matrix_1.6-5             \n[141] RcppHNSW_0.6.0            patchwork_1.2.0          \n[143] sparseMatrixStats_1.14.0  statmod_1.5.0            \n[145] shiny_1.9.1               ROCR_1.0-11              \n[147] igraph_2.0.3              xgboost_2.1.1.1          \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 202.98\nuser 1582.98\nsys 7.06\npercent 783%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ClusterMarkersOfAllCells",desc:"Markers for clusters of all cells.",npages:3,report_toc:!0,order:4,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] UpSetR_1.4.0        ggVennDiagram_1.2.3 tidyseurat_0.8.0   \n [4] ttservice_0.4.1     future_1.34.0       ggrepel_0.9.5      \n [7] ggprism_1.0.5       ggplot2_3.5.1       enrichR_3.2        \n[10] Seurat_5.1.0        SeuratObject_5.0.2  sp_2.1-4           \n[13] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[16] tidyselect_1.2.1    digest_0.6.36       jsonlite_1.8.8     \n[19] rlang_1.1.4         logger_0.3.0       \n\nloaded via a namespace (and not attached):\n  [1] RColorBrewer_1.1-3     magrittr_2.0.3         spatstat.utils_3.0-5  \n  [4] farver_2.1.2           vctrs_0.6.5            ROCR_1.0-11           \n  [7] spatstat.explore_3.2-6 htmltools_0.5.8.1      curl_5.2.1            \n [10] sctransform_0.4.1      parallelly_1.38.0      KernSmooth_2.23-24    \n [13] htmlwidgets_1.6.4      ica_1.0-3              plyr_1.8.9            \n [16] plotly_4.10.4          zoo_1.8-12             igraph_2.0.3          \n [19] mime_0.12              lifecycle_1.0.4        pkgconfig_2.0.3       \n [22] Matrix_1.6-5           R6_2.5.1               fastmap_1.2.0         \n [25] fitdistrplus_1.1-11    shiny_1.9.1            colorspace_2.1-1      \n [28] patchwork_1.2.0        tensor_1.5             RSpectra_0.16-1       \n [31] irlba_2.3.5.1          WriteXLS_6.7.0         labeling_0.4.3        \n [34] progressr_0.14.0       fansi_1.0.6            spatstat.sparse_3.1-0 \n [37] httr_1.4.7             polyclip_1.10-7        abind_1.4-5           \n [40] compiler_4.3.3         withr_3.0.1            fastDummies_1.7.3     \n [43] MASS_7.3-60.0.1        rjson_0.2.21           tools_4.3.3           \n [46] lmtest_0.9-40          httpuv_1.6.15          future.apply_1.11.2   \n [49] goftest_1.2-3          glue_1.7.0             nlme_3.1-165          \n [52] promises_1.3.0         grid_4.3.3             Rtsne_0.17            \n [55] cluster_2.1.6          reshape2_1.4.4         generics_0.1.3        \n [58] gtable_0.3.5           spatstat.data_3.1-2    data.table_1.15.4     \n [61] utf8_1.2.4             spatstat.geom_3.2-9    RcppAnnoy_0.0.22      \n [64] RANN_2.6.1             pillar_1.9.0           stringr_1.5.1         \n [67] spam_2.10-0            RcppHNSW_0.6.0         limma_3.58.1          \n [70] later_1.3.2            splines_4.3.3          lattice_0.22-6        \n [73] survival_3.7-0         deldir_2.0-4           miniUI_0.1.1.1        \n [76] pbapply_1.7-2          gridExtra_2.3          scattermore_1.2       \n [79] statmod_1.5.0          matrixStats_1.1.0      stringi_1.8.4         \n [82] lazyeval_0.2.2         codetools_0.2-20       RVenn_1.1.0           \n [85] cli_3.6.3              uwot_0.1.16            xtable_1.8-4          \n [88] reticulate_1.38.0      munsell_0.5.1          Rcpp_1.0.13           \n [91] globals_0.16.3         spatstat.random_3.2-3  png_0.1-8             \n [94] parallel_4.3.3         ellipsis_0.3.2         presto_1.0.0          \n [97] dotCall64_1.1-1        listenv_0.9.1          viridisLite_0.4.2     \n[100] scales_1.3.0           ggridges_0.5.6         leiden_0.4.3.1        \n[103] purrr_1.0.2            cowplot_1.1.3         \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 216.89\nuser 139.48\nsys 7.23\npercent 67%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TopExpressingGenesOfAllCells",desc:"Top expressing genes for clusters of all cells.",npages:3,report_toc:!0,order:5,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ggprism_1.0.5      dplyr_1.1.4        enrichR_3.2        tibble_3.2.1      \n [5] Seurat_5.1.0       SeuratObject_5.0.2 sp_2.1-4           jsonlite_1.8.8    \n [9] rlang_1.1.4        logger_0.3.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.2-9   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] labeling_0.4.3         utf8_1.2.4             promises_1.3.0        \n [19] purrr_1.0.2            WriteXLS_6.7.0         goftest_1.2-3         \n [22] later_1.3.2            spatstat.utils_3.0-5   irlba_2.3.5.1         \n [25] parallel_4.3.3         cluster_2.1.6          R6_2.5.1              \n [28] ica_1.0-3              stringi_1.8.4          RColorBrewer_1.1-3    \n [31] spatstat.data_3.1-2    reticulate_1.38.0      parallelly_1.38.0     \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.13           \n [37] tensor_1.5             future.apply_1.11.2    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [43] splines_4.3.3          igraph_2.0.3           tidyselect_1.2.1      \n [46] abind_1.4-5            spatstat.random_3.2-3  codetools_0.2-20      \n [49] miniUI_0.1.1.1         spatstat.explore_3.2-6 curl_5.2.1            \n [52] listenv_0.9.1          lattice_0.22-6         plyr_1.8.9            \n [55] withr_3.0.1            shiny_1.9.1            ROCR_1.0-11           \n [58] Rtsne_0.17             future_1.34.0          fastDummies_1.7.3     \n [61] survival_3.7-0         polyclip_1.10-7        fitdistrplus_1.1-11   \n [64] pillar_1.9.0           KernSmooth_2.23-24     plotly_4.10.4         \n [67] generics_0.1.3         RcppHNSW_0.6.0         ggplot2_3.5.1         \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.7.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.15.4      RSpectra_0.16-1       \n [79] RANN_2.6.1             leiden_0.4.3.1         dotCall64_1.1-1       \n [82] cowplot_1.1.3          grid_4.3.3             tidyr_1.3.1           \n [85] colorspace_2.1-1       nlme_3.1-165           patchwork_1.2.0       \n [88] cli_3.6.3              spatstat.sparse_3.1-0  spam_2.10-0           \n [91] fansi_1.0.6            viridisLite_0.4.2      uwot_0.1.16           \n [94] gtable_0.3.5           digest_0.6.36          progressr_0.14.0      \n [97] ggrepel_0.9.5          farver_2.1.2           rjson_0.2.21          \n[100] htmlwidgets_1.6.4      htmltools_0.5.8.1      lifecycle_1.0.4       \n[103] httr_1.4.7             mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 52.61\nuser 38.55\nsys 2.30\npercent 77%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TCellSelection",desc:"Separate T and non-T cells and select T cells.",npages:1,report_toc:!0,order:6,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] glue_1.7.0         factoextra_1.0.7   patchwork_1.2.0    ggpubr_0.6.0      \n [5] ggrepel_0.9.5      ggprism_1.0.5      ggplot2_3.5.1      tibble_3.2.1      \n [9] tidyr_1.3.1        dplyr_1.1.4        Seurat_5.1.0       SeuratObject_5.0.2\n[13] sp_2.1-4           jsonlite_1.8.8     rlang_1.1.4        logger_0.3.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.2-9   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] backports_1.5.0        labeling_0.4.3         utf8_1.2.4            \n [19] promises_1.3.0         purrr_1.0.2            goftest_1.2-3         \n [22] later_1.3.2            spatstat.utils_3.0-5   broom_1.0.6           \n [25] irlba_2.3.5.1          parallel_4.3.3         cluster_2.1.6         \n [28] R6_2.5.1               ica_1.0-3              stringi_1.8.4         \n [31] RColorBrewer_1.1-3     spatstat.data_3.1-2    reticulate_1.38.0     \n [34] car_3.1-2              parallelly_1.38.0      lmtest_0.9-40         \n [37] scattermore_1.2        Rcpp_1.0.13            tensor_1.5            \n [40] future.apply_1.11.2    zoo_1.8-12             sctransform_0.4.1     \n [43] httpuv_1.6.15          Matrix_1.6-5           splines_4.3.3         \n [46] igraph_2.0.3           tidyselect_1.2.1       abind_1.4-5           \n [49] spatstat.random_3.2-3  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.2-6 listenv_0.9.1          lattice_0.22-6        \n [55] plyr_1.8.9             withr_3.0.1            shiny_1.9.1           \n [58] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n [61] fastDummies_1.7.3      survival_3.7-0         polyclip_1.10-7       \n [64] fitdistrplus_1.1-11    pillar_1.9.0           carData_3.0-5         \n [67] KernSmooth_2.23-24     plotly_4.10.4          generics_0.1.3        \n [70] RcppHNSW_0.6.0         munsell_0.5.1          scales_1.3.0          \n [73] globals_0.16.3         xtable_1.8-4           lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.15.4      RSpectra_0.16-1       \n [79] ggsignif_0.6.4         RANN_2.6.1             leiden_0.4.3.1        \n [82] dotCall64_1.1-1        cowplot_1.1.3          grid_4.3.3            \n [85] colorspace_2.1-1       nlme_3.1-165           cli_3.6.3             \n [88] spatstat.sparse_3.1-0  spam_2.10-0            fansi_1.0.6           \n [91] viridisLite_0.4.2      uwot_0.1.16            gtable_0.3.5          \n [94] rstatix_0.7.2          digest_0.6.36          progressr_0.14.0      \n [97] farver_2.1.2           htmlwidgets_1.6.4      htmltools_0.5.8.1     \n[100] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[103] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 38.03\nuser 45.75\nsys 0.83\npercent 122%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3435.798\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ClusterMarkers",desc:"Markers for clusters of all or selected T cells.",npages:2,report_toc:!0,order:10,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] UpSetR_1.4.0        ggVennDiagram_1.2.3 tidyseurat_0.8.0   \n [4] ttservice_0.4.1     future_1.34.0       ggrepel_0.9.5      \n [7] ggprism_1.0.5       ggplot2_3.5.1       enrichR_3.2        \n[10] Seurat_5.1.0        SeuratObject_5.0.2  sp_2.1-4           \n[13] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[16] tidyselect_1.2.1    digest_0.6.36       jsonlite_1.8.8     \n[19] rlang_1.1.4         logger_0.3.0       \n\nloaded via a namespace (and not attached):\n  [1] RColorBrewer_1.1-3     magrittr_2.0.3         spatstat.utils_3.0-5  \n  [4] farver_2.1.2           vctrs_0.6.5            ROCR_1.0-11           \n  [7] spatstat.explore_3.2-6 htmltools_0.5.8.1      curl_5.2.1            \n [10] sctransform_0.4.1      parallelly_1.38.0      KernSmooth_2.23-24    \n [13] htmlwidgets_1.6.4      ica_1.0-3              plyr_1.8.9            \n [16] plotly_4.10.4          zoo_1.8-12             igraph_2.0.3          \n [19] mime_0.12              lifecycle_1.0.4        pkgconfig_2.0.3       \n [22] Matrix_1.6-5           R6_2.5.1               fastmap_1.2.0         \n [25] fitdistrplus_1.1-11    shiny_1.9.1            colorspace_2.1-1      \n [28] patchwork_1.2.0        tensor_1.5             RSpectra_0.16-1       \n [31] irlba_2.3.5.1          WriteXLS_6.7.0         labeling_0.4.3        \n [34] progressr_0.14.0       fansi_1.0.6            spatstat.sparse_3.1-0 \n [37] httr_1.4.7             polyclip_1.10-7        abind_1.4-5           \n [40] compiler_4.3.3         withr_3.0.1            fastDummies_1.7.3     \n [43] MASS_7.3-60.0.1        rjson_0.2.21           tools_4.3.3           \n [46] lmtest_0.9-40          httpuv_1.6.15          future.apply_1.11.2   \n [49] goftest_1.2-3          glue_1.7.0             nlme_3.1-165          \n [52] promises_1.3.0         grid_4.3.3             Rtsne_0.17            \n [55] cluster_2.1.6          reshape2_1.4.4         generics_0.1.3        \n [58] gtable_0.3.5           spatstat.data_3.1-2    data.table_1.15.4     \n [61] utf8_1.2.4             spatstat.geom_3.2-9    RcppAnnoy_0.0.22      \n [64] RANN_2.6.1             pillar_1.9.0           stringr_1.5.1         \n [67] spam_2.10-0            RcppHNSW_0.6.0         limma_3.58.1          \n [70] later_1.3.2            splines_4.3.3          lattice_0.22-6        \n [73] survival_3.7-0         deldir_2.0-4           miniUI_0.1.1.1        \n [76] pbapply_1.7-2          gridExtra_2.3          scattermore_1.2       \n [79] statmod_1.5.0          matrixStats_1.1.0      stringi_1.8.4         \n [82] lazyeval_0.2.2         codetools_0.2-20       RVenn_1.1.0           \n [85] cli_3.6.3              uwot_0.1.16            xtable_1.8-4          \n [88] reticulate_1.38.0      munsell_0.5.1          Rcpp_1.0.13           \n [91] globals_0.16.3         spatstat.random_3.2-3  png_0.1-8             \n [94] parallel_4.3.3         ellipsis_0.3.2         presto_1.0.0          \n [97] dotCall64_1.1-1        listenv_0.9.1          viridisLite_0.4.2     \n[100] scales_1.3.0           ggridges_0.5.6         leiden_0.4.3.1        \n[103] purrr_1.0.2            cowplot_1.1.3         \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 136.36\nuser 90.28\nsys 4.48\npercent 69%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TopExpressingGenes",desc:"Top expressing genes for clusters of all or selected T cells.",npages:2,report_toc:!0,order:11,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ggprism_1.0.5      dplyr_1.1.4        enrichR_3.2        tibble_3.2.1      \n [5] Seurat_5.1.0       SeuratObject_5.0.2 sp_2.1-4           jsonlite_1.8.8    \n [9] rlang_1.1.4        logger_0.3.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.2-9   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] labeling_0.4.3         utf8_1.2.4             promises_1.3.0        \n [19] purrr_1.0.2            WriteXLS_6.7.0         goftest_1.2-3         \n [22] later_1.3.2            spatstat.utils_3.0-5   irlba_2.3.5.1         \n [25] parallel_4.3.3         cluster_2.1.6          R6_2.5.1              \n [28] ica_1.0-3              stringi_1.8.4          RColorBrewer_1.1-3    \n [31] spatstat.data_3.1-2    reticulate_1.38.0      parallelly_1.38.0     \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.13           \n [37] tensor_1.5             future.apply_1.11.2    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [43] splines_4.3.3          igraph_2.0.3           tidyselect_1.2.1      \n [46] abind_1.4-5            spatstat.random_3.2-3  codetools_0.2-20      \n [49] miniUI_0.1.1.1         spatstat.explore_3.2-6 curl_5.2.1            \n [52] listenv_0.9.1          lattice_0.22-6         plyr_1.8.9            \n [55] withr_3.0.1            shiny_1.9.1            ROCR_1.0-11           \n [58] Rtsne_0.17             future_1.34.0          fastDummies_1.7.3     \n [61] survival_3.7-0         polyclip_1.10-7        fitdistrplus_1.1-11   \n [64] pillar_1.9.0           KernSmooth_2.23-24     plotly_4.10.4         \n [67] generics_0.1.3         RcppHNSW_0.6.0         ggplot2_3.5.1         \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.7.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.15.4      RSpectra_0.16-1       \n [79] RANN_2.6.1             leiden_0.4.3.1         dotCall64_1.1-1       \n [82] cowplot_1.1.3          grid_4.3.3             tidyr_1.3.1           \n [85] colorspace_2.1-1       nlme_3.1-165           patchwork_1.2.0       \n [88] cli_3.6.3              spatstat.sparse_3.1-0  spam_2.10-0           \n [91] fansi_1.0.6            viridisLite_0.4.2      uwot_0.1.16           \n [94] gtable_0.3.5           digest_0.6.36          progressr_0.14.0      \n [97] ggrepel_0.9.5          farver_2.1.2           rjson_0.2.21          \n[100] htmlwidgets_1.6.4      htmltools_0.5.8.1      lifecycle_1.0.4       \n[103] httr_1.4.7             mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 31.30\nuser 27.76\nsys 1.57\npercent 93%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TCRClusterStats",desc:"Statistics of TCR clusters, generated by `TCRClustering`.",npages:1,report_toc:!0,order:13,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] grid      stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] ComplexHeatmap_2.18.0 ggprism_1.0.5         immunarch_0.9.1      \n [4] patchwork_1.2.0       data.table_1.15.4     dtplyr_1.3.1         \n [7] dplyr_1.1.4           tidyr_1.3.1           tibble_3.2.1         \n[10] ggplot2_3.5.1         jsonlite_1.8.8        rlang_1.1.4          \n[13] logger_0.3.0         \n\nloaded via a namespace (and not attached):\n  [1] gridExtra_2.3       phangorn_2.11.1     readxl_1.4.3       \n  [4] magrittr_2.0.3      clue_0.3-65         GetoptLong_1.0.5   \n  [7] matrixStats_1.1.0   compiler_4.3.3      flexmix_2.3-19     \n [10] png_0.1-8           vctrs_0.6.5         reshape2_1.4.4     \n [13] ggalluvial_0.12.5   quadprog_1.5-8      stringr_1.5.1      \n [16] crayon_1.5.3        pkgconfig_2.0.3     shape_1.4.6.1      \n [19] fastmap_1.2.0       backports_1.5.0     labeling_0.4.3     \n [22] ggraph_2.1.0        utf8_1.2.4          promises_1.3.0     \n [25] tzdb_0.4.0          UpSetR_1.4.0        purrr_1.0.2        \n [28] modeltools_0.2-23   ggseqlogo_0.2       later_1.3.2        \n [31] uuid_1.2-1          fpc_2.2-12          tweenr_2.0.3       \n [34] broom_1.0.6         parallel_4.3.3      prabclus_2.3-3     \n [37] cluster_2.1.6       R6_2.5.1            stringi_1.8.4      \n [40] RColorBrewer_1.1-3  rlist_0.4.6.2       car_3.1-2          \n [43] diptest_0.77-1      cellranger_1.1.0    Rcpp_1.0.13        \n [46] iterators_1.0.14    IRanges_2.36.0      readr_2.1.5        \n [49] httpuv_1.6.15       Matrix_1.6-5        nnet_7.3-19        \n [52] igraph_2.0.3        tidyselect_1.2.1    stringdist_0.9.12  \n [55] abind_1.4-5         viridis_0.6.5       doParallel_1.0.17  \n [58] codetools_0.2-20    lattice_0.22-6      plyr_1.8.9         \n [61] shiny_1.9.1         withr_3.0.1         polyclip_1.10-7    \n [64] circlize_0.4.16     mclust_6.1.1        kernlab_0.9-32     \n [67] pillar_1.9.0        ggpubr_0.6.0        carData_3.0-5      \n [70] foreach_1.5.2       stats4_4.3.3        generics_0.1.3     \n [73] S4Vectors_0.40.2    hms_1.1.3           munsell_0.5.1      \n [76] scales_1.3.0        xtable_1.8-4        class_7.3-22       \n [79] glue_1.7.0          pheatmap_1.0.12     tools_4.3.3        \n [82] robustbase_0.99-3   ggsignif_0.6.4      graphlayouts_1.1.0 \n [85] Cairo_1.6-2         fastmatch_1.1-4     tidygraph_1.3.0    \n [88] ape_5.8             colorspace_2.1-1    nlme_3.1-165       \n [91] ggforce_0.4.2       cli_3.6.3           fansi_1.0.6        \n [94] viridisLite_0.4.2   gtable_0.3.5        DEoptimR_1.1-3     \n [97] rstatix_0.7.2       digest_0.6.36       BiocGenerics_0.48.1\n[100] ggrepel_0.9.5       rjson_0.2.21        farver_2.1.2       \n[103] htmltools_0.5.8.1   lifecycle_1.0.4     factoextra_1.0.7   \n[106] GlobalOptions_0.1.2 mime_0.12           shinythemes_1.2.0  \n[109] MASS_7.3-60.0.1    \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 5.36\nuser 14.04\nsys 0.30\npercent 267%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TESSA",desc:"Tessa is a Bayesian model to integrate T cell receptor (TCR) sequence profiling with transcriptomes of T cells.",npages:1,report_toc:!0,order:14,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] igraph_2.0.3         Rtsne_0.17           LaplacesDemon_16.1.6\n [4] MASS_7.3-60.0.1      ggprism_1.0.5        Seurat_5.1.0        \n [7] SeuratObject_5.0.2   sp_2.1-4             tibble_3.2.1        \n[10] glue_1.7.0           immunarch_0.9.1      patchwork_1.2.0     \n[13] data.table_1.15.4    dtplyr_1.3.1         ggplot2_3.5.1       \n[16] tidyr_1.3.1          dplyr_1.1.4          jsonlite_1.8.8      \n[19] rlang_1.1.4          logger_0.3.0        \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22       shinythemes_1.2.0      splines_4.3.3         \n  [4] later_1.3.2            cellranger_1.1.0       polyclip_1.10-7       \n  [7] fastDummies_1.7.3      factoextra_1.0.7       lifecycle_1.0.4       \n [10] rstatix_0.7.2          doParallel_1.0.17      globals_0.16.3        \n [13] lattice_0.22-6         prabclus_2.3-3         backports_1.5.0       \n [16] magrittr_2.0.3         plotly_4.10.4          rlist_0.4.6.2         \n [19] httpuv_1.6.15          sctransform_0.4.1      spam_2.10-0           \n [22] flexmix_2.3-19         spatstat.sparse_3.1-0  reticulate_1.38.0     \n [25] cowplot_1.1.3          pbapply_1.7-2          RColorBrewer_1.1-3    \n [28] abind_1.4-5            quadprog_1.5-8         purrr_1.0.2           \n [31] ggraph_2.1.0           nnet_7.3-19            tweenr_2.0.3          \n [34] circlize_0.4.16        ggrepel_0.9.5          irlba_2.3.5.1         \n [37] listenv_0.9.1          spatstat.utils_3.0-5   pheatmap_1.0.12       \n [40] goftest_1.2-3          RSpectra_0.16-1        spatstat.random_3.2-3 \n [43] fitdistrplus_1.1-11    parallelly_1.38.0      leiden_0.4.3.1        \n [46] codetools_0.2-20       ggforce_0.4.2          tidyselect_1.2.1      \n [49] shape_1.4.6.1          ggseqlogo_0.2          farver_2.1.2          \n [52] viridis_0.6.5          matrixStats_1.1.0      stats4_4.3.3          \n [55] spatstat.explore_3.2-6 tidygraph_1.3.0        progressr_0.14.0      \n [58] ggridges_0.5.6         ggalluvial_0.12.5      survival_3.7-0        \n [61] iterators_1.0.14       foreach_1.5.2          tools_4.3.3           \n [64] stringdist_0.9.12      ica_1.0-3              Rcpp_1.0.13           \n [67] gridExtra_2.3          withr_3.0.1            fastmap_1.2.0         \n [70] fansi_1.0.6            digest_0.6.36          R6_2.5.1              \n [73] mime_0.12              colorspace_2.1-1       scattermore_1.2       \n [76] tensor_1.5             spatstat.data_3.1-2    diptest_0.77-1        \n [79] UpSetR_1.4.0           utf8_1.2.4             generics_0.1.3        \n [82] robustbase_0.99-3      class_7.3-22           graphlayouts_1.1.0    \n [85] httr_1.4.7             htmlwidgets_1.6.4      uwot_0.1.16           \n [88] pkgconfig_2.0.3        gtable_0.3.5           modeltools_0.2-23     \n [91] lmtest_0.9-40          htmltools_0.5.8.1      carData_3.0-5         \n [94] dotCall64_1.1-1        scales_1.3.0           png_0.1-8             \n [97] tzdb_0.4.0             reshape2_1.4.4         uuid_1.2-1            \n[100] nlme_3.1-165           zoo_1.8-12             GlobalOptions_0.1.2   \n[103] stringr_1.5.1          KernSmooth_2.23-24     parallel_4.3.3        \n[106] miniUI_0.1.1.1         pillar_1.9.0           grid_4.3.3            \n[109] vctrs_0.6.5            RANN_2.6.1             promises_1.3.0        \n[112] ggpubr_0.6.0           car_3.1-2              xtable_1.8-4          \n[115] cluster_2.1.6          readr_2.1.5            cli_3.6.3             \n[118] compiler_4.3.3         future.apply_1.11.2    ggsignif_0.6.4        \n[121] labeling_0.4.3         mclust_6.1.1           plyr_1.8.9            \n[124] stringi_1.8.4          deldir_2.0-4           viridisLite_0.4.2     \n[127] munsell_0.5.1          lazyeval_0.2.2         spatstat.geom_3.2-9   \n[130] Matrix_1.6-5           RcppHNSW_0.6.0         hms_1.1.3             \n[133] future_1.34.0          fpc_2.2-12             shiny_1.9.1           \n[136] kernlab_0.9-32         ROCR_1.0-11            broom_1.0.6           \n[139] phangorn_2.11.1        fastmatch_1.1-4        DEoptimR_1.1-3        \n[142] readxl_1.4.3           ape_5.8               \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 1774.00\nuser 8784.42\nsys 20.96\npercent 496%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"SeuratClusterStats",desc:"Statistics of the clustering.",npages:1,report_toc:!0,order:17,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] grid      stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] clustree_0.5.1        ggraph_2.1.0          ComplexHeatmap_2.18.0\n [4] circlize_0.4.16       tidyseurat_0.8.0      ttservice_0.4.1      \n [7] ggrepel_0.9.5         ggprism_1.0.5         tibble_3.2.1         \n[10] Seurat_5.1.0          SeuratObject_5.0.2    sp_2.1-4             \n[13] ggplot2_3.5.1         tidyr_1.3.1           dplyr_1.1.4          \n[16] tidyselect_1.2.1      jsonlite_1.8.8        rlang_1.1.4          \n[19] logger_0.3.0         \n\nloaded via a namespace (and not attached):\n  [1] RColorBrewer_1.1-3     shape_1.4.6.1          magrittr_2.0.3        \n  [4] ggbeeswarm_0.7.2       spatstat.utils_3.0-5   farver_2.1.2          \n  [7] GlobalOptions_0.1.2    vctrs_0.6.5            ROCR_1.0-11           \n [10] spatstat.explore_3.2-6 htmltools_0.5.8.1      sctransform_0.4.1     \n [13] parallelly_1.38.0      KernSmooth_2.23-24     htmlwidgets_1.6.4     \n [16] ica_1.0-3              plyr_1.8.9             plotly_4.10.4         \n [19] zoo_1.8-12             igraph_2.0.3           mime_0.12             \n [22] lifecycle_1.0.4        iterators_1.0.14       pkgconfig_2.0.3       \n [25] Matrix_1.6-5           R6_2.5.1               fastmap_1.2.0         \n [28] clue_0.3-65            fitdistrplus_1.1-11    future_1.34.0         \n [31] shiny_1.9.1            digest_0.6.36          colorspace_2.1-1      \n [34] patchwork_1.2.0        S4Vectors_0.40.2       tensor_1.5            \n [37] RSpectra_0.16-1        irlba_2.3.5.1          labeling_0.4.3        \n [40] progressr_0.14.0       fansi_1.0.6            spatstat.sparse_3.1-0 \n [43] httr_1.4.7             polyclip_1.10-7        abind_1.4-5           \n [46] compiler_4.3.3         withr_3.0.1            doParallel_1.0.17     \n [49] viridis_0.6.5          fastDummies_1.7.3      ggforce_0.4.2         \n [52] MASS_7.3-60.0.1        rjson_0.2.21           tools_4.3.3           \n [55] vipor_0.4.7            lmtest_0.9-40          beeswarm_0.4.0        \n [58] httpuv_1.6.15          future.apply_1.11.2    goftest_1.2-3         \n [61] glue_1.7.0             nlme_3.1-165           promises_1.3.0        \n [64] Rtsne_0.17             cluster_2.1.6          reshape2_1.4.4        \n [67] generics_0.1.3         gtable_0.3.5           spatstat.data_3.1-2   \n [70] data.table_1.15.4      tidygraph_1.3.0        utf8_1.2.4            \n [73] BiocGenerics_0.48.1    spatstat.geom_3.2-9    RcppAnnoy_0.0.22      \n [76] RANN_2.6.1             foreach_1.5.2          pillar_1.9.0          \n [79] stringr_1.5.1          spam_2.10-0            RcppHNSW_0.6.0        \n [82] later_1.3.2            splines_4.3.3          tweenr_2.0.3          \n [85] lattice_0.22-6         survival_3.7-0         deldir_2.0-4          \n [88] miniUI_0.1.1.1         pbapply_1.7-2          gridExtra_2.3         \n [91] IRanges_2.36.0         scattermore_1.2        stats4_4.3.3          \n [94] graphlayouts_1.1.0     matrixStats_1.1.0      stringi_1.8.4         \n [97] lazyeval_0.2.2         codetools_0.2-20       cli_3.6.3             \n[100] uwot_0.1.16            xtable_1.8-4           reticulate_1.38.0     \n[103] munsell_0.5.1          Rcpp_1.0.13            globals_0.16.3        \n[106] spatstat.random_3.2-3  png_0.1-8              ggrastr_1.0.2         \n[109] parallel_4.3.3         ellipsis_0.3.2         dotCall64_1.1-1       \n[112] listenv_0.9.1          viridisLite_0.4.2      scales_1.3.0          \n[115] ggridges_0.5.6         leiden_0.4.3.1         purrr_1.0.2           \n[118] crayon_1.5.3           GetoptLong_1.0.5       cowplot_1.1.3         \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 15.10\nuser 23.39\nsys 0.59\npercent 158%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"Immunarch",desc:"Exploration of Single-cell and Bulk T-cell/Antibody Immune Repertoires",npages:3,report_toc:!0,order:18,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] circlize_0.4.16   tibble_3.2.1      glue_1.7.0        ggprism_1.0.5    \n [5] immunarch_0.9.1   patchwork_1.2.0   data.table_1.15.4 dtplyr_1.3.1     \n [9] ggplot2_3.5.1     tidyr_1.3.1       dplyr_1.1.4       jsonlite_1.8.8   \n[13] rlang_1.1.4       logger_0.3.0     \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22       shinythemes_1.2.0      splines_4.3.3         \n  [4] later_1.3.2            cellranger_1.1.0       polyclip_1.10-7       \n  [7] fastDummies_1.7.3      factoextra_1.0.7       lifecycle_1.0.4       \n [10] rstatix_0.7.2          doParallel_1.0.17      globals_0.16.3        \n [13] lattice_0.22-6         prabclus_2.3-3         MASS_7.3-60.0.1       \n [16] backports_1.5.0        magrittr_2.0.3         plotly_4.10.4         \n [19] rlist_0.4.6.2          httpuv_1.6.15          Seurat_5.1.0          \n [22] sctransform_0.4.1      spam_2.10-0            flexmix_2.3-19        \n [25] spatstat.sparse_3.1-0  sp_2.1-4               reticulate_1.38.0     \n [28] cowplot_1.1.3          pbapply_1.7-2          RColorBrewer_1.1-3    \n [31] abind_1.4-5            Rtsne_0.17             quadprog_1.5-8        \n [34] purrr_1.0.2            ggraph_2.1.0           nnet_7.3-19           \n [37] tweenr_2.0.3           ggrepel_0.9.5          irlba_2.3.5.1         \n [40] listenv_0.9.1          spatstat.utils_3.0-5   pheatmap_1.0.12       \n [43] goftest_1.2-3          RSpectra_0.16-1        spatstat.random_3.2-3 \n [46] fitdistrplus_1.1-11    parallelly_1.38.0      leiden_0.4.3.1        \n [49] codetools_0.2-20       ggforce_0.4.2          tidyselect_1.2.1      \n [52] shape_1.4.6.1          ggseqlogo_0.2          farver_2.1.2          \n [55] viridis_0.6.5          matrixStats_1.1.0      stats4_4.3.3          \n [58] spatstat.explore_3.2-6 tidygraph_1.3.0        progressr_0.14.0      \n [61] ggridges_0.5.6         ggalluvial_0.12.5      survival_3.7-0        \n [64] iterators_1.0.14       foreach_1.5.2          tools_4.3.3           \n [67] stringdist_0.9.12      ica_1.0-3              Rcpp_1.0.13           \n [70] gridExtra_2.3          withr_3.0.1            fastmap_1.2.0         \n [73] fansi_1.0.6            digest_0.6.36          R6_2.5.1              \n [76] mime_0.12              colorspace_2.1-1       scattermore_1.2       \n [79] tensor_1.5             spatstat.data_3.1-2    diptest_0.77-1        \n [82] UpSetR_1.4.0           utf8_1.2.4             generics_0.1.3        \n [85] robustbase_0.99-3      class_7.3-22           graphlayouts_1.1.0    \n [88] httr_1.4.7             htmlwidgets_1.6.4      uwot_0.1.16           \n [91] pkgconfig_2.0.3        gtable_0.3.5           modeltools_0.2-23     \n [94] lmtest_0.9-40          htmltools_0.5.8.1      carData_3.0-5         \n [97] dotCall64_1.1-1        SeuratObject_5.0.2     scales_1.3.0          \n[100] png_0.1-8              tzdb_0.4.0             reshape2_1.4.4        \n[103] uuid_1.2-1             nlme_3.1-165           zoo_1.8-12            \n[106] GlobalOptions_0.1.2    stringr_1.5.1          KernSmooth_2.23-24    \n[109] parallel_4.3.3         miniUI_0.1.1.1         pillar_1.9.0          \n[112] grid_4.3.3             vctrs_0.6.5            RANN_2.6.1            \n[115] promises_1.3.0         ggpubr_0.6.0           car_3.1-2             \n[118] xtable_1.8-4           cluster_2.1.6          readr_2.1.5           \n[121] cli_3.6.3              compiler_4.3.3         future.apply_1.11.2   \n[124] ggsignif_0.6.4         labeling_0.4.3         mclust_6.1.1          \n[127] plyr_1.8.9             stringi_1.8.4          deldir_2.0-4          \n[130] viridisLite_0.4.2      assertthat_0.2.1       munsell_0.5.1         \n[133] lazyeval_0.2.2         spatstat.geom_3.2-9    Matrix_1.6-5          \n[136] RcppHNSW_0.6.0         hms_1.1.3              future_1.34.0         \n[139] fpc_2.2-12             shiny_1.9.1            kernlab_0.9-32        \n[142] ROCR_1.0-11            igraph_2.0.3           broom_1.0.6           \n[145] phangorn_2.11.1        fastmatch_1.1-4        DEoptimR_1.1-3        \n[148] readxl_1.4.3           ape_5.8               \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 91.14\nuser 113.47\nsys 1.29\npercent 125%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"RadarPlots",desc:"Radar plots for cell proportion in different clusters.",npages:1,report_toc:!0,order:19},{name:"CloneResidency",desc:"Identification of clone residency",npages:1,report_toc:!0,order:20},{name:"ScFGSEA",desc:"Gene set enrichment analysis for cells in different groups using `fgsea`",npages:1,report_toc:!0,order:21},{name:"MarkersFinder",desc:"Find markers between different groups of cells",npages:1,report_toc:!0,order:22},{name:"MetaMarkers",desc:"Find markers between three or more groups of cells, using one-way ANOVA or Kruskal-Wallis test.",npages:1,report_toc:!0,order:23},{name:"CellsDistribution",desc:"Distribution of cells (i.e. in a TCR clone) from different groups for each cluster",npages:1,report_toc:!0,order:24},{name:"CDR3AAPhyschem",desc:"CDR3 AA physicochemical feature analysis",npages:1,report_toc:!0,order:25},{name:"ScrnaMetabolicLandscape",desc:"Metabolic landscape analysis for scRNA-seq data",order:28,procs:[{name:"MetabolicPathwayActivity",desc:"This process calculates the pathway activities in different groups and subsets.",npages:1,report_toc:!0,order:28},{name:"MetabolicFeatures",desc:"This process performs enrichment analysis for the metabolic pathways for each group in each subset.",npages:1,report_toc:!0,order:29},{name:"MetabolicFeaturesIntraSubset",desc:"Intra-subset metabolic features - Enrichment analysis in details",npages:1,report_toc:!0,order:30},{name:"MetabolicPathwayHeterogeneity",desc:"Calculate Metabolic Pathway heterogeneity.",npages:1,report_toc:!0,order:31}]}]};function pa(e){let t,n;return t=new qt({props:{dir:"up"}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function _a(e){let t,n;return t=new qt({props:{dir:"down"}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function da(e){let t,n,a,s,r,i=e[1]>0&&pa();n=new la({});let m=e[1]<e[3].npages-1&&_a();return{c(){i&&i.c(),t=h(),f(n.$$.fragment),a=h(),m&&m.c(),s=g()},m(e,o){i&&i.m(e,o),l(e,t,o),b(n,e,o),l(e,a,o),m&&m.m(e,o),l(e,s,o),r=!0},p(e,n){e[1]>0?i?2&n&&o(i,1):(i=pa(),i.c(),o(i,1),i.m(t.parentNode,t)):i&&(c(),_(i,1,1,(()=>{i=null})),p()),e[1]<e[3].npages-1?m?10&n&&o(m,1):(m=_a(),m.c(),o(m,1),m.m(s.parentNode,s)):m&&(c(),_(m,1,1,(()=>{m=null})),p())},i(e){r||(o(i),o(n.$$.fragment,e),o(m),r=!0)},o(e){_(i),_(n.$$.fragment,e),_(m),r=!1},d(e){e&&(d(t),d(a),d(s)),i&&i.d(e),$(n,e),m&&m.d(e)}}}function ma(e){let t,n,a;function s(t){e[4](t)}let r={logo:e[0],logotext:e[3].desc,versions:ca.versions,entries:ca.entries,pipeline_name:ca.pipeline.name,report_toc:e[3].report_toc,runinfo:e[3].runinfo,toc:oa,page:e[1],$$slots:{default:[da]},$$scope:{ctx:e}};return void 0!==e[2]&&(r.isSideNavOpen=e[2]),t=new Bt({props:r}),v.push((()=>x(t,"isSideNavOpen",s))),{c(){f(t.$$.fragment)},m(e,n){b(t,e,n),a=!0},p(e,[a]){const s={};1&a&&(s.logo=e[0]),8&a&&(s.logotext=e[3].desc),8&a&&(s.report_toc=e[3].report_toc),8&a&&(s.runinfo=e[3].runinfo),2&a&&(s.page=e[1]),42&a&&(s.$$scope={dirty:a,ctx:e}),!n&&4&a&&(n=!0,s.isSideNavOpen=e[2],y((()=>n=!1))),t.$set(s)},i(e){a||(o(t.$$.fragment,e),a=!0)},o(e){_(t.$$.fragment,e),a=!1},d(e){$(t,e)}}}function ua(e,t,n){let a,{name:s}=t,{page:r=0}=t,i=!1;for(const e of ca.entries)if(e.procs){for(const t of e.procs)if(t.name==s){a=t;break}}else if(e.name==s){a=e;break}return u((()=>{const e=new URL(window.location.href);let t;e.hash.length>1&&(t=document.getElementById(`${e.hash.slice(1)}`)),t&&t.scrollIntoView({block:"start",behavior:"smooth"})})),e.$$set=e=>{"name"in e&&n(0,s=e.name),"page"in e&&n(1,r=e.page)},[s,r,i,a,function(e){i=e,n(2,i)}]}e("default",new class extends t{constructor(e){super(),n(this,e,ua,ma,a,{name:0,page:1})}}({target:document.body,props:{name:"Immunarch",page:1}}))}}}));
