System.register(["./vendor/svelte-CiOtOeYy.js","./vendor/carbon-components-svelte-BhQO6q4b.js","./vendor/carbon-icons-svelte-DVphRbRG.js"],(function(e){"use strict";var a,t,n,_,l,d,s,i,r,o,p,c,u,v,m,g,P,A,f,b,C,O,S,R,h,$,T,y,L,j,x,M,B,G,F,N,U,w,D,E,H,I,k,V,z,q,Y,K,X,Q,W,Z,J,ee,ae,te,ne,_e,le,de,se,ie,re,oe,pe,ce,ue,ve,me,ge,Pe,Ae,fe,be,Ce,Oe,Se,Re,he,$e,Te,ye,Le,je,xe,Me,Be,Ge,Fe,Ne,Ue;return{setters:[function(e){a=e.S,t=e.i,n=e.s,_=e.a1,l=e.l,d=e.O,s=e.a,i=e.b,r=e.g,o=e.c,p=e.t,c=e.d,u=e.a5,v=e.B,m=e.e,g=e.G,P=e.H,A=e.I,f=e.y,b=e.K,C=e.T,O=e.U,S=e.L,R=e.D,h=e.M,$=e.z,T=e.Q,y=e.k,L=e.u,j=e.n,x=e.o,M=e.f,B=e.A,G=e.h,F=e.j,N=e.aa,U=e.x,w=e.W,D=e.ab,E=e.m,H=e.p,I=e.a0,k=e.q,V=e.R,z=e.$,q=e.v,Y=e.Y,K=e.Z,X=e.F,Q=e.ac,W=e.w,Z=e.J},function(e){J=e.S,ee=e.a,ae=e.b,te=e.H,ne=e.c,_e=e.T,le=e.d,de=e.e,se=e.f,ie=e.g,re=e.h,oe=e.M,pe=e.I,ce=e.i,ue=e.j,ve=e.k,me=e.C,ge=e.l,Pe=e.m,Ae=e.G,fe=e.R,be=e.n,Ce=e.B,Oe=e.A,Se=e.o,Re=e.D,he=e.P,$e=e.p,Te=e.q,ye=e.r,Le=e.s,je=e.t},function(e){xe=e.C,Me=e.a,Be=e.O,Ge=e.b,Fe=e.c,Ne=e.D,Ue=e.d}],execute:function(){function we(e,a,t){const n=e.slice();return n[4]=a[t],n}function De(e,a,t){const n=e.slice();return n[7]=a[t],n}function Ee(e){let a,t;return a=new J({props:{href:`${e[1](e[4].page)}#${e[4].slug}`,class:e[2](e[4].page),title:e[4].text,text:e[4].text,expanded:!0,$$slots:{default:[ke]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};1&t&&(n.href=`${e[1](e[4].page)}#${e[4].slug}`),1&t&&(n.class=e[2](e[4].page)),1&t&&(n.title=e[4].text),1&t&&(n.text=e[4].text),1025&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function He(e){let a,t;return a=new ee({props:{href:`${e[1](e[4].page)}#${e[4].slug}`,class:e[2](e[4].page),title:e[4].text,text:e[4].text}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};1&t&&(n.href=`${e[1](e[4].page)}#${e[4].slug}`),1&t&&(n.class=e[2](e[4].page)),1&t&&(n.title=e[4].text),1&t&&(n.text=e[4].text),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function Ie(e){let a,t;return a=new ae({props:{href:`${e[1](e[7].page)}#${e[7].slug}`,class:e[2](e[7].page),title:e[7].text,text:e[7].text}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};1&t&&(n.href=`${e[1](e[7].page)}#${e[7].slug}`),1&t&&(n.class=e[2](e[7].page)),1&t&&(n.title=e[7].text),1&t&&(n.text=e[7].text),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function ke(e){let a,t,n=_(e[4].children),l=[];for(let a=0;a<n.length;a+=1)l[a]=Ie(De(e,n,a));const d=e=>p(l[e],1,1,(()=>{l[e]=null}));return{c(){for(let e=0;e<l.length;e+=1)l[e].c();a=f()},m(e,n){for(let a=0;a<l.length;a+=1)l[a]&&l[a].m(e,n);s(e,a,n),t=!0},p(e,t){if(7&t){let s;for(n=_(e[4].children),s=0;s<n.length;s+=1){const _=De(e,n,s);l[s]?(l[s].p(_,t),i(l[s],1)):(l[s]=Ie(_),l[s].c(),i(l[s],1),l[s].m(a.parentNode,a))}for(r(),s=n.length;s<l.length;s+=1)d(s);o()}},i(e){if(!t){for(let e=0;e<n.length;e+=1)i(l[e]);t=!0}},o(e){l=l.filter(Boolean);for(let e=0;e<l.length;e+=1)p(l[e]);t=!1},d(e){e&&c(a),u(l,e)}}}function Ve(e){let a,t,n,_;const l=[He,Ee],d=[];function u(e,a){return 0==e[4].children.length?0:1}return a=u(e),t=d[a]=l[a](e),{c(){t.c(),n=m()},m(e,t){d[a].m(e,t),s(e,n,t),_=!0},p(e,_){let s=a;a=u(e),a===s?d[a].p(e,_):(r(),p(d[s],1,1,(()=>{d[s]=null})),o(),t=d[a],t?t.p(e,_):(t=d[a]=l[a](e),t.c()),i(t,1),t.m(n.parentNode,n))},i(e){_||(i(t),_=!0)},o(e){p(t),_=!1},d(e){e&&c(n),d[a].d(e)}}}function ze(e){let a,t,n=_(e[0]),v=[];for(let a=0;a<n.length;a+=1)v[a]=Ve(we(e,n,a));const m=e=>p(v[e],1,1,(()=>{v[e]=null}));return{c(){a=l("ul");for(let e=0;e<v.length;e+=1)v[e].c();d(a,"class","bx--side-nav__items pipen-report-sidenav svelte-12gpq3")},m(e,n){s(e,a,n);for(let e=0;e<v.length;e+=1)v[e]&&v[e].m(a,null);t=!0},p(e,[t]){if(7&t){let l;for(n=_(e[0]),l=0;l<n.length;l+=1){const _=we(e,n,l);v[l]?(v[l].p(_,t),i(v[l],1)):(v[l]=Ve(_),v[l].c(),i(v[l],1),v[l].m(a,null))}for(r(),l=n.length;l<v.length;l+=1)m(l);o()}},i(e){if(!t){for(let e=0;e<n.length;e+=1)i(v[e]);t=!0}},o(e){v=v.filter(Boolean);for(let e=0;e<v.length;e+=1)p(v[e]);t=!1},d(e){e&&c(a),u(v,e)}}}function qe(e,a,t){let{toc:n}=a,{page:_}=a;return v((()=>{let e,a=new URL(window.location.href);e=a.hash.length>0?document.querySelector(`[href="${a.hash}"]`):document.querySelector(".toc-current-page"),e.scrollIntoView({behavior:"smooth"})})),e.$$set=e=>{"toc"in e&&t(0,n=e.toc),"page"in e&&t(3,_=e.page)},[n,e=>{if(_===e)return"";let a=new URL(window.location.href);return a.hash="",a.searchParams.set("page",e),a.toString()},e=>e!=_?"toc-noncurrent-page":"toc-current-page",_]}class Ye extends a{constructor(e){super(),t(this,e,qe,ze,n,{toc:0,page:3})}}function Ke(e,a,t){const n=e.slice();return n[7]=a[t],n}function Xe(e,a,t){const n=e.slice();return n[10]=a[t],n}function Qe(e){let a,t;return a=new de({props:{isOpen:ma,transition:e[5],$$slots:{default:[ia]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};8197&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function We(e){let a;return{c(){a=S("Go to home:")},m(e,t){s(e,a,t)},d(e){e&&c(a)}}}function Ze(e){let a;return{c(){a=S(e[0])},m(e,t){s(e,a,t)},p(e,t){1&t&&h(a,e[0])},d(e){e&&c(a)}}}function Je(e){let a;return{c(){a=S("Switch to process:")},m(e,t){s(e,a,t)},d(e){e&&c(a)}}}function ea(e){let a,t;return a=new re({props:{title:e[7].name,href:`?proc=${e[7].name}`,$$slots:{default:[ta]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};4&t&&(n.title=e[7].name),4&t&&(n.href=`?proc=${e[7].name}`),8196&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function aa(e){let a,t,n,l;a=new re({props:{class:"procgroup",$$slots:{default:[na]},$$scope:{ctx:e}}});let d=_(e[7].procs),v=[];for(let a=0;a<d.length;a+=1)v[a]=la(Xe(e,d,a));const b=e=>p(v[e],1,1,(()=>{v[e]=null}));return{c(){g(a.$$.fragment),t=f();for(let e=0;e<v.length;e+=1)v[e].c();n=m()},m(e,_){P(a,e,_),s(e,t,_);for(let a=0;a<v.length;a+=1)v[a]&&v[a].m(e,_);s(e,n,_),l=!0},p(e,t){const l={};if(8196&t&&(l.$$scope={dirty:t,ctx:e}),a.$set(l),4&t){let a;for(d=_(e[7].procs),a=0;a<d.length;a+=1){const _=Xe(e,d,a);v[a]?(v[a].p(_,t),i(v[a],1)):(v[a]=la(_),v[a].c(),i(v[a],1),v[a].m(n.parentNode,n))}for(r(),a=d.length;a<v.length;a+=1)b(a);o()}},i(e){if(!l){i(a.$$.fragment,e);for(let e=0;e<d.length;e+=1)i(v[e]);l=!0}},o(e){p(a.$$.fragment,e),v=v.filter(Boolean);for(let e=0;e<v.length;e+=1)p(v[e]);l=!1},d(e){e&&(c(t),c(n)),A(a,e),u(v,e)}}}function ta(e){let a,t,n=e[7].name+"";return{c(){a=S(n),t=f()},m(e,n){s(e,a,n),s(e,t,n)},p(e,t){4&t&&n!==(n=e[7].name+"")&&h(a,n)},d(e){e&&(c(a),c(t))}}}function na(e){let a,t=e[7].name+"";return{c(){a=S(t)},m(e,t){s(e,a,t)},p(e,n){4&n&&t!==(t=e[7].name+"")&&h(a,t)},d(e){e&&c(a)}}}function _a(e){let a,t,n,_=e[10].name+"";return{c(){a=S("- "),t=S(_),n=f()},m(e,_){s(e,a,_),s(e,t,_),s(e,n,_)},p(e,a){4&a&&_!==(_=e[10].name+"")&&h(t,_)},d(e){e&&(c(a),c(t),c(n))}}}function la(e){let a,t;return a=new re({props:{title:e[10].name,class:"procgroup-proc",href:`?proc=${e[10].name}`,$$slots:{default:[_a]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};4&t&&(n.title=e[10].name),4&t&&(n.href=`?proc=${e[10].name}`),8196&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function da(e){let a,t,n,_;const l=[aa,ea],d=[];function u(e,a){return e[7].procs?0:1}return a=u(e),t=d[a]=l[a](e),{c(){t.c(),n=m()},m(e,t){d[a].m(e,t),s(e,n,t),_=!0},p(e,_){let s=a;a=u(e),a===s?d[a].p(e,_):(r(),p(d[s],1,1,(()=>{d[s]=null})),o(),t=d[a],t?t.p(e,_):(t=d[a]=l[a](e),t.c()),i(t,1),t.m(n.parentNode,n))},i(e){_||(i(t),_=!0)},o(e){p(t),_=!1},d(e){e&&c(n),d[a].d(e)}}}function sa(e){let a,t,n,l,d,v,b,C;a=new ie({props:{$$slots:{default:[We]},$$scope:{ctx:e}}}),n=new re({props:{href:"index.html",$$slots:{default:[Ze]},$$scope:{ctx:e}}}),d=new ie({props:{$$slots:{default:[Je]},$$scope:{ctx:e}}});let O=_(e[2]),S=[];for(let a=0;a<O.length;a+=1)S[a]=da(Ke(e,O,a));const R=e=>p(S[e],1,1,(()=>{S[e]=null}));return{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment),l=f(),g(d.$$.fragment),v=f();for(let e=0;e<S.length;e+=1)S[e].c();b=m()},m(e,_){P(a,e,_),s(e,t,_),P(n,e,_),s(e,l,_),P(d,e,_),s(e,v,_);for(let a=0;a<S.length;a+=1)S[a]&&S[a].m(e,_);s(e,b,_),C=!0},p(e,t){const l={};8192&t&&(l.$$scope={dirty:t,ctx:e}),a.$set(l);const s={};8193&t&&(s.$$scope={dirty:t,ctx:e}),n.$set(s);const p={};if(8192&t&&(p.$$scope={dirty:t,ctx:e}),d.$set(p),4&t){let a;for(O=_(e[2]),a=0;a<O.length;a+=1){const n=Ke(e,O,a);S[a]?(S[a].p(n,t),i(S[a],1)):(S[a]=da(n),S[a].c(),i(S[a],1),S[a].m(b.parentNode,b))}for(r(),a=O.length;a<S.length;a+=1)R(a);o()}},i(e){if(!C){i(a.$$.fragment,e),i(n.$$.fragment,e),i(d.$$.fragment,e);for(let e=0;e<O.length;e+=1)i(S[e]);C=!0}},o(e){p(a.$$.fragment,e),p(n.$$.fragment,e),p(d.$$.fragment,e),S=S.filter(Boolean);for(let e=0;e<S.length;e+=1)p(S[e]);C=!1},d(e){e&&(c(t),c(l),c(v),c(b)),A(a,e),A(n,e),A(d,e),u(S,e)}}}function ia(e){let a,t;return a=new se({props:{$$slots:{default:[sa]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};8197&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function ra(e){let a,t,n,_;a=new le({props:{render:"select",theme:"g10",persist:!0,persistKey:"__pipen-report-theme",select:{themes:["g10","white","g80","g90","g100"],labelText:"Theme",inline:!0}}});let l=e[2].length>0&&Qe(e);return{c(){g(a.$$.fragment),t=f(),l&&l.c(),n=m()},m(e,d){P(a,e,d),s(e,t,d),l&&l.m(e,d),s(e,n,d),_=!0},p(e,a){e[2].length>0?l?(l.p(e,a),4&a&&i(l,1)):(l=Qe(e),l.c(),i(l,1),l.m(n.parentNode,n)):l&&(r(),p(l,1,1,(()=>{l=null})),o())},i(e){_||(i(a.$$.fragment,e),i(l),_=!0)},o(e){p(a.$$.fragment,e),p(l),_=!1},d(e){e&&(c(t),c(n)),A(a,e),l&&l.d(e)}}}function oa(e){let a,t;return a=new ne({props:{class:"header-util",$$slots:{default:[ra]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};8197&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function pa(e){let a;return{c(){a=S(e[4])},m(e,t){s(e,a,t)},p(e,t){16&t&&h(a,e[4])},d(e){e&&c(a)}}}function ca(e){let a,t,n,_,d=null!==e[4]&&pa(e);return{c(){a=l("h1"),t=S(e[3]),n=f(),d&&d.c(),_=m()},m(e,l){s(e,a,l),R(a,t),s(e,n,l),d&&d.m(e,l),s(e,_,l)},p(e,a){8&a&&h(t,e[3]),null!==e[4]?d?d.p(e,a):(d=pa(e),d.c(),d.m(_.parentNode,_)):d&&(d.d(1),d=null)},d(e){e&&(c(a),c(n),c(_)),d&&d.d(e)}}}function ua(e){let a,t;return a=new _e({props:{class:"header-tile",slot:"platform",$$slots:{default:[ca]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};8216&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function va(e){let a,t,n;function _(a){e[6](a)}let l={$$slots:{platform:[ua],default:[oa]},$$scope:{ctx:e}};return void 0!==e[1]&&(l.isSideNavOpen=e[1]),a=new te({props:l}),b.push((()=>C(a,"isSideNavOpen",_))),{c(){g(a.$$.fragment)},m(e,t){P(a,e,t),n=!0},p(e,[n]){const _={};8221&n&&(_.$$scope={dirty:n,ctx:e}),!t&&2&n&&(t=!0,_.isSideNavOpen=e[1],O((()=>t=!1))),a.$set(_)},i(e){n||(i(a.$$.fragment,e),n=!0)},o(e){p(a.$$.fragment,e),n=!1},d(e){A(a,e)}}}let ma=!1;function ga(e,a,t){let{entries:n}=a,{logo:_}=a,{logotext:l=null}=a,{pipeline_name:d}=a,{isSideNavOpen:s}=a;return d=d||_,e.$$set=e=>{"entries"in e&&t(2,n=e.entries),"logo"in e&&t(3,_=e.logo),"logotext"in e&&t(4,l=e.logotext),"pipeline_name"in e&&t(0,d=e.pipeline_name),"isSideNavOpen"in e&&t(1,s=e.isSideNavOpen)},[d,s,n,_,l,{duration:200},function(e){s=e,t(1,s)}]}class Pa extends a{constructor(e){super(),t(this,e,ga,va,n,{entries:2,logo:3,logotext:4,pipeline_name:0,isSideNavOpen:1})}}function Aa(e){let a,t,n;function _(a){e[3](a)}let l={modalHeading:"Running Information",passiveModal:!0,hasScrollingContent:!0,class:"runinfo",$$slots:{default:[Ra]},$$scope:{ctx:e}};return void 0!==e[2]&&(l.open=e[2]),a=new oe({props:l}),b.push((()=>C(a,"open",_))),{c(){g(a.$$.fragment)},m(e,t){P(a,e,t),n=!0},p(e,n){const _={};34&n&&(_.$$scope={dirty:n,ctx:e}),!t&&4&n&&(t=!0,_.open=e[2],O((()=>t=!1))),a.$set(_)},i(e){n||(i(a.$$.fragment,e),n=!0)},o(e){p(a.$$.fragment,e),n=!1},d(e){A(a,e)}}}function fa(e){let a,t,n,_,l,d;return a=new ue({props:{label:"Session Information"}}),n=new ue({props:{label:"Time Spent"}}),l=new ue({props:{label:"Device Information"}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment),_=f(),g(l.$$.fragment)},m(e,i){P(a,e,i),s(e,t,i),P(n,e,i),s(e,_,i),P(l,e,i),d=!0},p:T,i(e){d||(i(a.$$.fragment,e),i(n.$$.fragment,e),i(l.$$.fragment,e),d=!0)},o(e){p(a.$$.fragment,e),p(n.$$.fragment,e),p(l.$$.fragment,e),d=!1},d(e){e&&(c(t),c(_)),A(a,e),A(n,e),A(l,e)}}}function ba(e){let a,t;return a=new me({props:{type:"multi",code:e[1].session}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};2&t&&(n.code=e[1].session),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function Ca(e){let a,t;return a=new me({props:{type:"multi",code:e[1].time}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};2&t&&(n.code=e[1].time),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function Oa(e){let a,t;return a=new me({props:{type:"multi",code:e[1].device}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};2&t&&(n.code=e[1].device),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function Sa(e){let a,t,n,_,l,d;return a=new ve({props:{$$slots:{default:[ba]},$$scope:{ctx:e}}}),n=new ve({props:{$$slots:{default:[Ca]},$$scope:{ctx:e}}}),l=new ve({props:{$$slots:{default:[Oa]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment),_=f(),g(l.$$.fragment)},m(e,i){P(a,e,i),s(e,t,i),P(n,e,i),s(e,_,i),P(l,e,i),d=!0},p(e,t){const _={};34&t&&(_.$$scope={dirty:t,ctx:e}),a.$set(_);const d={};34&t&&(d.$$scope={dirty:t,ctx:e}),n.$set(d);const s={};34&t&&(s.$$scope={dirty:t,ctx:e}),l.$set(s)},i(e){d||(i(a.$$.fragment,e),i(n.$$.fragment,e),i(l.$$.fragment,e),d=!0)},o(e){p(a.$$.fragment,e),p(n.$$.fragment,e),p(l.$$.fragment,e),d=!1},d(e){e&&(c(t),c(_)),A(a,e),A(n,e),A(l,e)}}}function Ra(e){let a,t,n,_;return a=new pe({props:{kind:"warning",lowContrast:!0,hideCloseButton:!0,title:"Showing the first job only. Check the workdir for information of other jobs if any."}}),n=new ce({props:{$$slots:{content:[Sa],default:[fa]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment)},m(e,l){P(a,e,l),s(e,t,l),P(n,e,l),_=!0},p(e,a){const t={};34&a&&(t.$$scope={dirty:a,ctx:e}),n.$set(t)},i(e){_||(i(a.$$.fragment,e),i(n.$$.fragment,e),_=!0)},o(e){p(a.$$.fragment,e),p(n.$$.fragment,e),_=!1},d(e){e&&c(t),A(a,e),A(n,e)}}}function ha(e){let a,t,n;return{c(){a=l("a"),a.textContent="> Running Information",d(a,"href","javascript:void(0);")},m(_,l){s(_,a,l),t||(n=$(a,"click",e[4]),t=!0)},p:T,d(e){e&&c(a),t=!1,n()}}}function $a(e){let a,t,n,_,u,v,m,g,P=e[1]&&Aa(e),A=e[1]&&ha(e);return{c(){P&&P.c(),a=f(),t=l("div"),n=l("div"),A&&A.c(),_=f(),u=l("div"),v=S("Powered by "),m=l("span"),d(n,"class","runinfo"),d(m,"class","pipen-report-footer-versions"),d(u,"class","versions"),d(t,"class","pipen-report-footer bx--grid")},m(l,d){P&&P.m(l,d),s(l,a,d),s(l,t,d),R(t,n),A&&A.m(n,null),R(t,_),R(t,u),R(u,v),R(u,m),m.innerHTML=e[0],g=!0},p(e,[t]){e[1]?P?(P.p(e,t),2&t&&i(P,1)):(P=Aa(e),P.c(),i(P,1),P.m(a.parentNode,a)):P&&(r(),p(P,1,1,(()=>{P=null})),o()),e[1]?A?A.p(e,t):(A=ha(e),A.c(),A.m(n,null)):A&&(A.d(1),A=null),(!g||1&t)&&(m.innerHTML=e[0])},i(e){g||(i(P),g=!0)},o(e){p(P),g=!1},d(e){e&&(c(a),c(t)),P&&P.d(e),A&&A.d()}}}function Ta(e,a,t){let{versions:n}=a,{runinfo:_}=a,l=!1;return e.$$set=e=>{"versions"in e&&t(0,n=e.versions),"runinfo"in e&&t(1,_=e.runinfo)},[n,_,l,function(e){l=e,t(2,l)},()=>{t(2,l=!0)}]}class ya extends a{constructor(e){super(),t(this,e,Ta,$a,n,{versions:0,runinfo:1})}}function La(e){let a,t,n;function _(a){e[12](a)}let l={$$slots:{default:[ja]},$$scope:{ctx:e}};return void 0!==e[0]&&(l.isOpen=e[0]),a=new Pe({props:l}),b.push((()=>C(a,"isOpen",_))),{c(){g(a.$$.fragment)},m(e,t){P(a,e,t),n=!0},p(e,n){const _={};8576&n&&(_.$$scope={dirty:n,ctx:e}),!t&&1&n&&(t=!0,_.isOpen=e[0],O((()=>t=!1))),a.$set(_)},i(e){n||(i(a.$$.fragment,e),n=!0)},o(e){p(a.$$.fragment,e),n=!1},d(e){A(a,e)}}}function ja(e){let a,t;return a=new Ye({props:{toc:e[7],page:e[8]}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};128&t&&(n.toc=e[7]),256&t&&(n.page=e[8]),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function xa(e){let a;const t=e[10].default,n=y(t,e,e[13],null);return{c(){n&&n.c()},m(e,t){n&&n.m(e,t),a=!0},p(e,_){n&&n.p&&(!a||8192&_)&&L(n,t,e,e[13],a?x(t,e[13],_,null):j(e[13]),null)},i(e){a||(i(n,e),a=!0)},o(e){p(n,e),a=!1},d(e){n&&n.d(e)}}}function Ma(e){let a,t;return a=new be({props:{$$slots:{default:[xa]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};8192&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function Ba(e){let a,t;return a=new fe({props:{$$slots:{default:[Ma]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};8192&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function Ga(e){let a,t,n,_;return a=new Ae({props:{$$slots:{default:[Ba]},$$scope:{ctx:e}}}),n=new ya({props:{versions:e[3],runinfo:e[9]}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment)},m(e,l){P(a,e,l),s(e,t,l),P(n,e,l),_=!0},p(e,t){const _={};8192&t&&(_.$$scope={dirty:t,ctx:e}),a.$set(_);const l={};8&t&&(l.versions=e[3]),512&t&&(l.runinfo=e[9]),n.$set(l)},i(e){_||(i(a.$$.fragment,e),i(n.$$.fragment,e),_=!0)},o(e){p(a.$$.fragment,e),p(n.$$.fragment,e),_=!1},d(e){e&&c(t),A(a,e),A(n,e)}}}function Fa(e){let a,t,n,_,l,d;function u(a){e[11](a)}let v={logo:e[1],logotext:e[2],entries:e[4],pipeline_name:e[5]};void 0!==e[0]&&(v.isSideNavOpen=e[0]),a=new Pa({props:v}),b.push((()=>C(a,"isSideNavOpen",u)));let m=e[6]&&e[7].length>0&&La(e);return l=new ge({props:{$$slots:{default:[Ga]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment),n=f(),m&&m.c(),_=f(),g(l.$$.fragment)},m(e,t){P(a,e,t),s(e,n,t),m&&m.m(e,t),s(e,_,t),P(l,e,t),d=!0},p(e,[n]){const d={};2&n&&(d.logo=e[1]),4&n&&(d.logotext=e[2]),16&n&&(d.entries=e[4]),32&n&&(d.pipeline_name=e[5]),!t&&1&n&&(t=!0,d.isSideNavOpen=e[0],O((()=>t=!1))),a.$set(d),e[6]&&e[7].length>0?m?(m.p(e,n),192&n&&i(m,1)):(m=La(e),m.c(),i(m,1),m.m(_.parentNode,_)):m&&(r(),p(m,1,1,(()=>{m=null})),o());const s={};8712&n&&(s.$$scope={dirty:n,ctx:e}),l.$set(s)},i(e){d||(i(a.$$.fragment,e),i(m),i(l.$$.fragment,e),d=!0)},o(e){p(a.$$.fragment,e),p(m),p(l.$$.fragment,e),d=!1},d(e){e&&(c(n),c(_)),A(a,e),m&&m.d(e),A(l,e)}}}function Na(e,a,t){let n,{$$slots:_={},$$scope:l}=a,{logo:d}=a,{logotext:s=null}=a,{versions:i}=a,{entries:r}=a,{pipeline_name:o}=a,{report_toc:p}=a,{toc:c}=a,{page:u}=a,{isSideNavOpen:m}=a,{runinfo:g}=a;return v((()=>{const e=new URL(window.location.href);let a;a=e.hash.length>0?document.querySelector(`li:has(>[href="${e.hash}"])`):document.querySelector("li:has(>.toc-current-page)"),a&&(a.classList.add("toc-active"),a.scrollIntoView());const t=document.querySelectorAll(".pipen-report-toc-anchor");n=new IntersectionObserver((e=>{e.forEach((e=>{const a=e.target.getAttribute("id"),t=document.querySelector(`li:has(>[href="#${a}"])`);t&&(e.isIntersecting?t.classList.add("toc-active"):t.classList.remove("toc-active"))}))}),{root:null,rootMargin:"0px",threshold:.5}),t.forEach((e=>{n.observe(e)}))})),e.$$set=e=>{"logo"in e&&t(1,d=e.logo),"logotext"in e&&t(2,s=e.logotext),"versions"in e&&t(3,i=e.versions),"entries"in e&&t(4,r=e.entries),"pipeline_name"in e&&t(5,o=e.pipeline_name),"report_toc"in e&&t(6,p=e.report_toc),"toc"in e&&t(7,c=e.toc),"page"in e&&t(8,u=e.page),"isSideNavOpen"in e&&t(0,m=e.isSideNavOpen),"runinfo"in e&&t(9,g=e.runinfo),"$$scope"in e&&t(13,l=e.$$scope)},[m,d,s,i,r,o,p,c,u,g,_,function(e){m=e,t(0,m)},function(e){m=e,t(0,m)},l]}class Ua extends a{constructor(e){super(),t(this,e,Na,Fa,n,{logo:1,logotext:2,versions:3,entries:4,pipeline_name:5,report_toc:6,toc:7,page:8,isSideNavOpen:0,runinfo:9})}}function wa(e){let a;return{c(){a=S(e[2])},m(e,t){s(e,a,t)},p:T,d(e){e&&c(a)}}}function Da(e){let a,t,n,_;return t=new Ce({props:{icon:e[1],kind:"tertiary",href:e[3](),$$slots:{default:[wa]},$$scope:{ctx:e}}}),{c(){a=l("div"),g(t.$$.fragment),d(a,"class",n=`pagenav ${e[0]}`)},m(e,n){s(e,a,n),P(t,a,null),_=!0},p(e,[l]){const s={};16&l&&(s.$$scope={dirty:l,ctx:e}),t.$set(s),(!_||1&l&&n!==(n=`pagenav ${e[0]}`))&&d(a,"class",n)},i(e){_||(i(t.$$.fragment,e),_=!0)},o(e){p(t.$$.fragment,e),_=!1},d(e){e&&c(a),A(t)}}}function Ea(e,a,t){let{dir:n}=a,_="up"===n?xe:Me,l="up"===n?"Go to previous page":"Go to next page";return e.$$set=e=>{"dir"in e&&t(0,n=e.dir)},[n,_,l,()=>{let e=new URL(window.location.href);e.hash="";const a=parseInt(e.searchParams.get("page")||0),t="up"===n?a-1:a+1;return 0===t?e.searchParams.delete("page"):e.searchParams.set("page",t),e.toString()}]}class Ha extends a{constructor(e){super(),t(this,e,Ea,Da,n,{dir:0})}}const Ia=e=>({}),ka=e=>({}),Va=e=>({}),za=e=>({}),qa=e=>({}),Ya=e=>({}),Ka=e=>({}),Xa=e=>({});function Qa(e){let a,t;return a=new Oe({props:{ratio:e[5],$$slots:{default:[at]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};32&t&&(n.ratio=e[5]),65631&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function Wa(e){let a,t,n,_,l=e[2]&&tt(e),d=e[0]&&nt(e),u=e[1]&&_t(e);return{c(){l&&l.c(),a=f(),d&&d.c(),t=f(),u&&u.c(),n=m()},m(e,i){l&&l.m(e,i),s(e,a,i),d&&d.m(e,i),s(e,t,i),u&&u.m(e,i),s(e,n,i),_=!0},p(e,_){e[2]?l?(l.p(e,_),4&_&&i(l,1)):(l=tt(e),l.c(),i(l,1),l.m(a.parentNode,a)):l&&(r(),p(l,1,1,(()=>{l=null})),o()),e[0]?d?(d.p(e,_),1&_&&i(d,1)):(d=nt(e),d.c(),i(d,1),d.m(t.parentNode,t)):d&&(r(),p(d,1,1,(()=>{d=null})),o()),e[1]?u?(u.p(e,_),2&_&&i(u,1)):(u=_t(e),u.c(),i(u,1),u.m(n.parentNode,n)):u&&(r(),p(u,1,1,(()=>{u=null})),o())},i(e){_||(i(l),i(d),i(u),_=!0)},o(e){p(l),p(d),p(u),_=!1},d(e){e&&(c(a),c(t),c(n)),l&&l.d(e),d&&d.d(e),u&&u.d(e)}}}function Za(e){let a;const t=e[12].loading,n=y(t,e,e[16],za);return{c(){n&&n.c()},m(e,t){n&&n.m(e,t),a=!0},p(e,_){n&&n.p&&(!a||65536&_)&&L(n,t,e,e[16],a?x(t,e[16],_,Va):j(e[16]),za)},i(e){a||(i(n,e),a=!0)},o(e){p(n,e),a=!1},d(e){n&&n.d(e)}}}function Ja(e){let a,t,n,_,d,i,r=[e[9],{src:t=e[3]},{alt:e[4]},{title:"Click to zoom in ..."}],o={};for(let e=0;e<r.length;e+=1)o=G(o,r[e]);return{c(){a=l("img"),E(a,o)},m(t,n){s(t,a,n),_=!0,d||(i=$(a,"click",e[8]),d=!0)},p(n,l){E(a,o=H(r,[(e=n)[9],(!_||8&l&&!N(a.src,t=e[3]))&&{src:t},(!_||16&l)&&{alt:e[4]},{title:"Click to zoom in ..."}]))},i(t){_||(t&&U((()=>{_&&(n||(n=w(a,D,{duration:e[6]?st:0},!0)),n.run(1))})),_=!0)},o(t){t&&(n||(n=w(a,D,{duration:e[6]?st:0},!1)),n.run(0)),_=!1},d(e){e&&c(a),e&&n&&n.end(),d=!1,i()}}}function et(e){let a;const t=e[12].error,n=y(t,e,e[16],ka);return{c(){n&&n.c()},m(e,t){n&&n.m(e,t),a=!0},p(e,_){n&&n.p&&(!a||65536&_)&&L(n,t,e,e[16],a?x(t,e[16],_,Ia):j(e[16]),ka)},i(e){a||(i(n,e),a=!0)},o(e){p(n,e),a=!1},d(e){n&&n.d(e)}}}function at(e){let a,t,n,_,l=e[2]&&Za(e),d=e[0]&&Ja(e),u=e[1]&&et(e);return{c(){l&&l.c(),a=f(),d&&d.c(),t=f(),u&&u.c(),n=m()},m(e,i){l&&l.m(e,i),s(e,a,i),d&&d.m(e,i),s(e,t,i),u&&u.m(e,i),s(e,n,i),_=!0},p(e,_){e[2]?l?(l.p(e,_),4&_&&i(l,1)):(l=Za(e),l.c(),i(l,1),l.m(a.parentNode,a)):l&&(r(),p(l,1,1,(()=>{l=null})),o()),e[0]?d?(d.p(e,_),1&_&&i(d,1)):(d=Ja(e),d.c(),i(d,1),d.m(t.parentNode,t)):d&&(r(),p(d,1,1,(()=>{d=null})),o()),e[1]?u?(u.p(e,_),2&_&&i(u,1)):(u=et(e),u.c(),i(u,1),u.m(n.parentNode,n)):u&&(r(),p(u,1,1,(()=>{u=null})),o())},i(e){_||(i(l),i(d),i(u),_=!0)},o(e){p(l),p(d),p(u),_=!1},d(e){e&&(c(a),c(t),c(n)),l&&l.d(e),d&&d.d(e),u&&u.d(e)}}}function tt(e){let a;const t=e[12].loading,n=y(t,e,e[16],Xa);return{c(){n&&n.c()},m(e,t){n&&n.m(e,t),a=!0},p(e,_){n&&n.p&&(!a||65536&_)&&L(n,t,e,e[16],a?x(t,e[16],_,Ka):j(e[16]),Xa)},i(e){a||(i(n,e),a=!0)},o(e){p(n,e),a=!1},d(e){n&&n.d(e)}}}function nt(e){let a,t,n,_,d,i,r=[e[9],{src:t=e[3]},{alt:e[4]},{title:"Click to zoom in ..."}],o={};for(let e=0;e<r.length;e+=1)o=G(o,r[e]);return{c(){a=l("img"),E(a,o)},m(t,n){s(t,a,n),_=!0,d||(i=$(a,"click",e[8]),d=!0)},p(n,l){E(a,o=H(r,[(e=n)[9],(!_||8&l&&!N(a.src,t=e[3]))&&{src:t},(!_||16&l)&&{alt:e[4]},{title:"Click to zoom in ..."}]))},i(t){_||(t&&U((()=>{_&&(n||(n=w(a,D,{duration:e[6]?st:0},!0)),n.run(1))})),_=!0)},o(t){t&&(n||(n=w(a,D,{duration:e[6]?st:0},!1)),n.run(0)),_=!1},d(e){e&&c(a),e&&n&&n.end(),d=!1,i()}}}function _t(e){let a;const t=e[12].error,n=y(t,e,e[16],Ya);return{c(){n&&n.c()},m(e,t){n&&n.m(e,t),a=!0},p(e,_){n&&n.p&&(!a||65536&_)&&L(n,t,e,e[16],a?x(t,e[16],_,qa):j(e[16]),Ya)},i(e){a||(i(n,e),a=!0)},o(e){p(n,e),a=!1},d(e){n&&n.d(e)}}}function lt(e){let a,t,n,_;return{c(){a=l("img"),d(a,"class","pipen-report-image-lg"),N(a.src,t=e[3])||d(a,"src",t),d(a,"alt",e[4])},m(e,t){s(e,a,t),_=!0},p(n,l){e=n,(!_||8&l&&!N(a.src,t=e[3]))&&d(a,"src",t),(!_||16&l)&&d(a,"alt",e[4])},i(t){_||(t&&U((()=>{_&&(n||(n=w(a,D,{duration:e[6]?st:0},!0)),n.run(1))})),_=!0)},o(t){t&&(n||(n=w(a,D,{duration:e[6]?st:0},!1)),n.run(0)),_=!1},d(e){e&&c(a),e&&n&&n.end()}}}function dt(e){let a,t,n,_,l,d;const u=[Wa,Qa],v=[];function m(e,a){return void 0===e[5]?0:1}function S(a){e[13](a)}a=m(e),t=v[a]=u[a](e);let R={modalHeading:e[4],primaryButtonText:"Open in new window ...",secondaryButtonText:"Close",$$slots:{default:[lt]},$$scope:{ctx:e}};return void 0!==e[7]&&(R.open=e[7]),_=new oe({props:R}),b.push((()=>C(_,"open",S))),_.$on("click:button--secondary",e[14]),_.$on("click:button--primary",e[15]),{c(){t.c(),n=f(),g(_.$$.fragment)},m(e,t){v[a].m(e,t),s(e,n,t),P(_,e,t),d=!0},p(e,[d]){let s=a;a=m(e),a===s?v[a].p(e,d):(r(),p(v[s],1,1,(()=>{v[s]=null})),o(),t=v[a],t?t.p(e,d):(t=v[a]=u[a](e),t.c()),i(t,1),t.m(n.parentNode,n));const c={};16&d&&(c.modalHeading=e[4]),65624&d&&(c.$$scope={dirty:d,ctx:e}),!l&&128&d&&(l=!0,c.open=e[7],O((()=>l=!1))),_.$set(c)},i(e){d||(i(t),i(_.$$.fragment,e),d=!0)},o(e){p(t),p(_.$$.fragment,e),d=!1},d(e){e&&c(n),v[a].d(e),A(_,e)}}}const st=110;function it(e,a,t){const n=["src","alt","ratio","loading","loaded","error","fadeIn","border","loadImage"];let _=M(a,n),{$$slots:l={},$$scope:d}=a,{src:s=""}=a,{alt:i=""}=a,{ratio:r}=a,{loading:o=!1}=a,{loaded:p=!1}=a,{error:c=!1}=a,{fadeIn:u=!1}=a,{border:m=!0}=a,g=!1;const P=e=>{null!=f&&(f=null),t(0,p=!1),t(1,c=!1),f=new Image,f.src=e||s,f.onload=()=>t(0,p=!0),f.onerror=()=>t(1,c=!0)},A=B();let f=null;v((()=>()=>f=null));const b={..._,class:[_.class,"pipen-report-image",m?"":"pipen-report-image-noborder"].filter(Boolean).join(" ")};return e.$$set=e=>{a=G(G({},a),F(e)),t(19,_=M(a,n)),"src"in e&&t(3,s=e.src),"alt"in e&&t(4,i=e.alt),"ratio"in e&&t(5,r=e.ratio),"loading"in e&&t(2,o=e.loading),"loaded"in e&&t(0,p=e.loaded),"error"in e&&t(1,c=e.error),"fadeIn"in e&&t(6,u=e.fadeIn),"border"in e&&t(10,m=e.border),"$$scope"in e&&t(16,d=e.$$scope)},e.$$.update=()=>{3&e.$$.dirty&&t(2,o=!p&&!c),8&e.$$.dirty&&s&&"undefined"!=typeof window&&P(),1&e.$$.dirty&&p&&A("load"),2&e.$$.dirty&&c&&A("error")},[p,c,o,s,i,r,u,g,()=>{t(7,g=!0)},b,m,P,l,function(e){g=e,t(7,g)},()=>{t(7,g=!1)},()=>{t(7,g=!1),window.open(s)},d]}class rt extends a{constructor(e){super(),t(this,e,it,dt,n,{src:3,alt:4,ratio:5,loading:2,loaded:0,error:1,fadeIn:6,border:10,loadImage:11})}get loadImage(){return this.$$.ctx[11]}}function ot(e){let a,t;const n=[{src:e[1]},e[4]];let _={$$slots:{error:[ut],loading:[ct]},$$scope:{ctx:e}};for(let e=0;e<n.length;e+=1)_=G(_,n[e]);return a=new rt({props:_}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const _=18&t?H(n,[2&t&&{src:e[1]},16&t&&I(e[4])]):{};256&t&&(_.$$scope={dirty:t,ctx:e}),a.$set(_)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function pt(e){let a,t,n;const _=[{src:e[1]},e[4]];let d={$$slots:{error:[mt],loading:[vt]},$$scope:{ctx:e}};for(let e=0;e<_.length;e+=1)d=G(d,_[e]);t=new rt({props:d});let r=[e[0]],o={};for(let e=0;e<r.length;e+=1)o=G(o,r[e]);return{c(){a=l("div"),g(t.$$.fragment),E(a,o)},m(e,_){s(e,a,_),P(t,a,null),n=!0},p(e,n){const l=18&n?H(_,[2&n&&{src:e[1]},16&n&&I(e[4])]):{};256&n&&(l.$$scope={dirty:n,ctx:e}),t.$set(l),E(a,o=H(r,[1&n&&e[0]]))},i(e){n||(i(t.$$.fragment,e),n=!0)},o(e){p(t.$$.fragment,e),n=!1},d(e){e&&c(a),A(t)}}}function ct(e){let a,t,n=[e[2]],_={};for(let e=0;e<n.length;e+=1)_=G(_,n[e]);return{c(){a=l("div"),t=S("Image loading ..."),E(a,_)},m(e,n){s(e,a,n),R(a,t)},p:T,d(e){e&&c(a)}}}function ut(e){let a,t,n=[e[3]],_={};for(let e=0;e<n.length;e+=1)_=G(_,n[e]);return{c(){a=l("div"),t=S("Image loading error!"),E(a,_)},m(e,n){s(e,a,n),R(a,t)},p:T,d(e){e&&c(a)}}}function vt(e){let a,t,n=[e[2]],_={};for(let e=0;e<n.length;e+=1)_=G(_,n[e]);return{c(){a=l("div"),t=S("Image loading ..."),E(a,_)},m(e,n){s(e,a,n),R(a,t)},p:T,d(e){e&&c(a)}}}function mt(e){let a,t,n=[e[3]],_={};for(let e=0;e<n.length;e+=1)_=G(_,n[e]);return{c(){a=l("div"),t=S("Image loading error!"),E(a,_)},m(e,n){s(e,a,n),R(a,t)},p:T,d(e){e&&c(a)}}}function gt(e){let a,t,n,_;const l=[pt,ot],d=[];function u(e,a){return e[0]?0:1}return a=u(e),t=d[a]=l[a](e),{c(){t.c(),n=m()},m(e,t){d[a].m(e,t),s(e,n,t),_=!0},p(e,[_]){let s=a;a=u(e),a===s?d[a].p(e,_):(r(),p(d[s],1,1,(()=>{d[s]=null})),o(),t=d[a],t?t.p(e,_):(t=d[a]=l[a](e),t.c()),i(t,1),t.m(n.parentNode,n))},i(e){_||(i(t),_=!0)},o(e){p(t),_=!1},d(e){e&&c(n),d[a].d(e)}}}function Pt(e,a,t){const n=["frameProps","width","height","src"];let _=M(a,n),{frameProps:l={class:"pipen-report-image-frame"}}=a,{width:d=300}=a,{height:s=200}=a,{src:i}=a,r=`width: ${d}px; height: ${s}px; aspect-ratio: ${d} / ${s}`,o={..._,class:[_.class,"pipen-report-image-loading"].filter(Boolean).join(" "),style:_.style?`${_.style}; ${r}`:r},p={...o,class:[o.class,"error"].join(" ")};return e.$$set=e=>{a=G(G({},a),F(e)),t(4,_=M(a,n)),"frameProps"in e&&t(0,l=e.frameProps),"width"in e&&t(5,d=e.width),"height"in e&&t(6,s=e.height),"src"in e&&t(1,i=e.src)},[l,i,o,p,_,d,s]}let At=class extends a{constructor(e){super(),t(this,e,Pt,gt,n,{frameProps:0,width:5,height:6,src:1})}};const ft=e=>({}),bt=e=>({});function Ct(e){let a,t;const n=e[25].default,_=y(n,e,e[24],null);return{c(){a=l("ul"),_&&_.c(),d(a,"role","menu"),d(a,"tabindex","-1"),d(a,"aria-label",e[14]),d(a,"data-floating-menu-direction",e[5]),d(a,"class",e[9]),k(a,"bx--overflow-menu-options",!0),k(a,"bx--overflow-menu--flip",e[7]),k(a,"bx--overflow-menu-options--open",e[0]),k(a,"bx--overflow-menu-options--light",e[6]),k(a,"bx--overflow-menu-options--sm","sm"===e[4]),k(a,"bx--overflow-menu-options--xl","xl"===e[4]),k(a,"bx--breadcrumb-menu-options",!!e[15])},m(n,l){s(n,a,l),_&&_.m(a,null),e[32](a),t=!0},p(e,l){_&&_.p&&(!t||16777216&l[0])&&L(_,n,e,e[24],t?x(n,e[24],l,null):j(e[24]),null),(!t||16384&l[0])&&d(a,"aria-label",e[14]),(!t||32&l[0])&&d(a,"data-floating-menu-direction",e[5]),(!t||512&l[0])&&d(a,"class",e[9]),(!t||512&l[0])&&k(a,"bx--overflow-menu-options",!0),(!t||640&l[0])&&k(a,"bx--overflow-menu--flip",e[7]),(!t||513&l[0])&&k(a,"bx--overflow-menu-options--open",e[0]),(!t||576&l[0])&&k(a,"bx--overflow-menu-options--light",e[6]),(!t||528&l[0])&&k(a,"bx--overflow-menu-options--sm","sm"===e[4]),(!t||528&l[0])&&k(a,"bx--overflow-menu-options--xl","xl"===e[4]),(!t||33280&l[0])&&k(a,"bx--breadcrumb-menu-options",!!e[15])},i(e){t||(i(_,e),t=!0)},o(e){p(_,e),t=!1},d(t){t&&c(a),_&&_.d(t),e[32](null)}}}function Ot(e){let a,t,n,_,d,u,v,b;const C=e[25].menu,O=y(C,e,e[24],bt),S=O||function(e){let a,t,n;var _=e[1];function l(e,a){return{props:{"aria-label":e[11],title:e[11],class:"bx--overflow-menu__icon "+e[10]}}}return _&&(a=X(_,l(e))),{c(){a&&g(a.$$.fragment),t=m()},m(e,_){a&&P(a,e,_),s(e,t,_),n=!0},p(e,n){if(2&n[0]&&_!==(_=e[1])){if(a){r();const e=a;p(e.$$.fragment,1,0,(()=>{A(e,1)})),o()}_?(a=X(_,l(e)),g(a.$$.fragment),i(a.$$.fragment,1),P(a,t.parentNode,t)):a=null}else if(_){const t={};2048&n[0]&&(t["aria-label"]=e[11]),2048&n[0]&&(t.title=e[11]),1024&n[0]&&(t.class="bx--overflow-menu__icon "+e[10]),a.$set(t)}},i(e){n||(a&&i(a.$$.fragment,e),n=!0)},o(e){a&&p(a.$$.fragment,e),n=!1},d(e){e&&c(t),a&&A(a,e)}}}(e);let h=e[0]&&Ct(e),T=[{type:"button"},{"aria-haspopup":""},{"aria-expanded":e[0]},{"aria-label":e[14]},{id:e[12]},e[20]],M={};for(let e=0;e<T.length;e+=1)M=G(M,T[e]);return{c(){a=new Q(!1),t=m(),n=f(),_=l("button"),S&&S.c(),d=f(),h&&h.c(),a.a=t,E(_,M),k(_,"bx--overflow-menu",!0),k(_,"bx--overflow-menu--open",e[0]),k(_,"bx--overflow-menu--light",e[6]),k(_,"bx--overflow-menu--sm","sm"===e[4]),k(_,"bx--overflow-menu--xl","xl"===e[4])},m(l,i){a.m(e[13],document.head),R(document.head,t),s(l,n,i),s(l,_,i),S&&S.m(_,null),R(_,d),h&&h.m(_,null),_.autofocus&&_.focus(),e[33](_),u=!0,v||(b=[$(window,"click",e[31]),$(_,"click",e[26]),$(_,"click",e[34]),$(_,"mouseover",e[27]),$(_,"mouseenter",e[28]),$(_,"mouseleave",e[29]),$(_,"keydown",e[30]),$(_,"keydown",e[35]),$(_,"focusout",e[36])],v=!0)},p(e,t){(!u||8192&t[0])&&a.p(e[13]),O?O.p&&(!u||16777216&t[0])&&L(O,C,e,e[24],u?x(C,e[24],t,ft):j(e[24]),bt):S&&S.p&&(!u||3074&t[0])&&S.p(e,u?t:[-1,-1]),e[0]?h?(h.p(e,t),1&t[0]&&i(h,1)):(h=Ct(e),h.c(),i(h,1),h.m(_,null)):h&&(r(),p(h,1,1,(()=>{h=null})),o()),E(_,M=H(T,[{type:"button"},{"aria-haspopup":""},(!u||1&t[0])&&{"aria-expanded":e[0]},(!u||16384&t[0])&&{"aria-label":e[14]},(!u||4096&t[0])&&{id:e[12]},1048576&t[0]&&e[20]])),k(_,"bx--overflow-menu",!0),k(_,"bx--overflow-menu--open",e[0]),k(_,"bx--overflow-menu--light",e[6]),k(_,"bx--overflow-menu--sm","sm"===e[4]),k(_,"bx--overflow-menu--xl","xl"===e[4])},i(e){u||(i(S,e),i(h),u=!0)},o(e){p(S,e),p(h),u=!1},d(l){l&&(a.d(),c(n),c(_)),c(t),S&&S.d(l),h&&h.d(),e[33](null),v=!1,V(b)}}}function St(e,a,t){let n,_;const l=["size","direction","open","light","flipped","persistentClickItems","menuOptionsClass","icon","iconClass","iconDescription","id","buttonRef","menuRef"];let d,s,i,r=M(a,l),{$$slots:o={},$$scope:p}=a,{size:c}=a,{direction:u="bottom"}=a,{open:v=!1}=a,{light:m=!1}=a,{flipped:g=!1}=a,{persistentClickItems:P=!1}=a,{menuOptionsClass:A}=a,{icon:f=Ge}=a,{iconClass:C}=a,{iconDescription:O="Open and close list of options"}=a,{id:S="ccs-"+Math.random().toString(36)}=a,{buttonRef:R=null}=a,{menuRef:h=null}=a;const $=z("BreadcrumbItem"),T=B(),y=W([]);q(e,y,(e=>t(23,s=e)));const L=W(void 0);q(e,L,(e=>t(38,i=e)));const j=W(void 0),x=W(-1);let N;q(e,x,(e=>t(22,d=e)));let U=!0;Y("OverflowMenu",{focusedId:j,add:({id:e,text:a,primaryFocus:t,disabled:n})=>{y.update((_=>(t&&x.set(_.length),[..._,{id:e,text:a,primaryFocus:t,disabled:n,index:_.length}])))},update:e=>{L.set(e)},change:e=>{let a=d+e;a<0?a=s.length-1:a>=s.length&&(a=0);let t=s[a].disabled;for(;t;)a+=e,a<0?a=s.length-1:a>=s.length&&(a=0),t=s[a].disabled;x.set(a)}}),K((()=>{if(i){const{index:e,text:a}=s.filter((e=>e.id===i))[0];T("close",{index:e,text:a}),t(0,v=!1)}if(v){const{width:e,height:a}=R.getBoundingClientRect();t(21,N=e),!U&&d<0&&h.focus(),g&&(t(3,h.style.left="auto",h),t(3,h.style.right=0,h)),"top"===u?(t(3,h.style.top="auto",h),t(3,h.style.bottom=a+"px",h)):"bottom"===u&&t(3,h.style.top=a+"px",h),$&&(t(3,h.style.top=a+10+"px",h),t(3,h.style.left="-11px",h))}v||(y.set([]),L.set(void 0),x.set(0)),U=!1}));return e.$$set=e=>{t(40,a=G(G({},a),F(e))),t(20,r=M(a,l)),"size"in e&&t(4,c=e.size),"direction"in e&&t(5,u=e.direction),"open"in e&&t(0,v=e.open),"light"in e&&t(6,m=e.light),"flipped"in e&&t(7,g=e.flipped),"persistentClickItems"in e&&t(8,P=e.persistentClickItems),"menuOptionsClass"in e&&t(9,A=e.menuOptionsClass),"icon"in e&&t(1,f=e.icon),"iconClass"in e&&t(10,C=e.iconClass),"iconDescription"in e&&t(11,O=e.iconDescription),"id"in e&&t(12,S=e.id),"buttonRef"in e&&t(2,R=e.buttonRef),"menuRef"in e&&t(3,h=e.menuRef),"$$scope"in e&&t(24,p=e.$$scope)},e.$$.update=()=>{t(14,n=a["aria-label"]||"menu"),12582912&e.$$.dirty[0]&&s[d]&&j.set(s[d].id),2101248&e.$$.dirty[0]&&t(13,_=`<style>\n    #${S} .bx--overflow-menu-options.bx--overflow-menu-options:after {\n      width: ${N?N+"px":"2rem"};\n    }\n  </style>`)},$&&t(1,f=Be),a=F(a),[v,f,R,h,c,u,m,g,P,A,C,O,S,_,n,$,T,y,L,x,r,N,d,s,p,o,function(a){Z.call(this,e,a)},function(a){Z.call(this,e,a)},function(a){Z.call(this,e,a)},function(a){Z.call(this,e,a)},function(a){Z.call(this,e,a)},({target:e})=>{R&&R.contains(e)||h&&!h.contains(e)&&t(0,v=!1)},function(e){b[e?"unshift":"push"]((()=>{h=e,t(3,h)}))},function(e){b[e?"unshift":"push"]((()=>{R=e,t(2,R)}))},e=>{P&&e.stopPropagation(),h&&h.contains(e.target)||(t(0,v=!v),v||T("close"))},e=>{v&&(["ArrowDown","ArrowLeft","ArrowRight","ArrowUp"].includes(e.key)?e.preventDefault():"Escape"===e.key&&(e.stopPropagation(),T("close"),t(0,v=!1),R.focus()))},e=>{v&&(R.contains(e.relatedTarget)||(T("close"),t(0,v=!1)))}]}class Rt extends a{constructor(e){super(),t(this,e,St,Ot,n,{size:4,direction:5,open:0,light:6,flipped:7,persistentClickItems:8,menuOptionsClass:9,icon:1,iconClass:10,iconDescription:11,id:12,buttonRef:2,menuRef:3},null,[-1,-1])}}function ht(e,a,t){const n=e.slice();return n[3]=a[t],n[4]=a,n[5]=t,n}function $t(e){let a,t,n,_,r;function o(a){e[1](a,e[3])}let u={labelText:e[3].value};return void 0!==e[3].selected&&(u.checked=e[3].selected),t=new Se({props:u}),b.push((()=>C(t,"checked",o))),t.$on("change",e[2]),{c(){a=l("li"),g(t.$$.fragment),_=f(),d(a,"class","pipen-report-column-selector-item svelte-147jw2r")},m(e,n){s(e,a,n),P(t,a,null),R(a,_),r=!0},p(a,_){e=a;const l={};1&_&&(l.labelText=e[3].value),!n&&1&_&&(n=!0,l.checked=e[3].selected,O((()=>n=!1))),t.$set(l)},i(e){r||(i(t.$$.fragment,e),r=!0)},o(e){p(t.$$.fragment,e),r=!1},d(e){e&&c(a),A(t)}}}function Tt(e){let a,t,n=_(e[0]),l=[];for(let a=0;a<n.length;a+=1)l[a]=$t(ht(e,n,a));const d=e=>p(l[e],1,1,(()=>{l[e]=null}));return{c(){for(let e=0;e<l.length;e+=1)l[e].c();a=m()},m(e,n){for(let a=0;a<l.length;a+=1)l[a]&&l[a].m(e,n);s(e,a,n),t=!0},p(e,t){if(1&t){let s;for(n=_(e[0]),s=0;s<n.length;s+=1){const _=ht(e,n,s);l[s]?(l[s].p(_,t),i(l[s],1)):(l[s]=$t(_),l[s].c(),i(l[s],1),l[s].m(a.parentNode,a))}for(r(),s=n.length;s<l.length;s+=1)d(s);o()}},i(e){if(!t){for(let e=0;e<n.length;e+=1)i(l[e]);t=!0}},o(e){l=l.filter(Boolean);for(let e=0;e<l.length;e+=1)p(l[e]);t=!1},d(e){e&&c(a),u(l,e)}}}function yt(e){let a,t;return a=new Rt({props:{class:"pipen-report-overflowmenu",persistentClickItems:!0,size:"sm",icon:Fe,iconDescription:"Select visible columns ...",$$slots:{default:[Tt]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,[t]){const n={};65&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function Lt(e,a,t){let{allColumns:n}=a;return e.$$set=e=>{"allColumns"in e&&t(0,n=e.allColumns)},[n,function(a,_){e.$$.not_equal(_.selected,a)&&(_.selected=a,t(0,n))},()=>{t(0,n)}]}class jt extends a{constructor(e){super(),t(this,e,Lt,yt,n,{allColumns:0})}}function xt(e){let a,t;return a=new Ce({props:{iconDescription:"Right click and save as to download the entire data",tooltipPosition:"top",icon:Ue,href:e[2]}}),a.$on("click",e[13]),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};4&t&&(n.href=e[2]),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function Mt(e){let a,t,n,_,l,d,u,v,S,R;function h(a){e[17](a)}let $={persistent:!0};function T(a){e[18](a)}void 0!==e[7]&&($.value=e[7]),a=new ye({props:$}),b.push((()=>C(a,"value",h)));let y={};void 0!==e[6]&&(y.allColumns=e[6]),_=new jt({props:y}),b.push((()=>C(_,"allColumns",T))),u=new Ce({props:{iconDescription:`Download current ${e[11].length} records (may be partial of the entire data)`,tooltipPosition:"top",icon:Ne}}),u.$on("click",e[14]);let L=!!e[2]&&xt(e);return{c(){g(a.$$.fragment),n=f(),g(_.$$.fragment),d=f(),g(u.$$.fragment),v=f(),L&&L.c(),S=m()},m(e,t){P(a,e,t),s(e,n,t),P(_,e,t),s(e,d,t),P(u,e,t),s(e,v,t),L&&L.m(e,t),s(e,S,t),R=!0},p(e,n){const d={};!t&&128&n&&(t=!0,d.value=e[7],O((()=>t=!1))),a.$set(d);const s={};!l&&64&n&&(l=!0,s.allColumns=e[6],O((()=>l=!1))),_.$set(s);const c={};2048&n&&(c.iconDescription=`Download current ${e[11].length} records (may be partial of the entire data)`),u.$set(c),e[2]?L?(L.p(e,n),4&n&&i(L,1)):(L=xt(e),L.c(),i(L,1),L.m(S.parentNode,S)):L&&(r(),p(L,1,1,(()=>{L=null})),o())},i(e){R||(i(a.$$.fragment,e),i(_.$$.fragment,e),i(u.$$.fragment,e),i(L),R=!0)},o(e){p(a.$$.fragment,e),p(_.$$.fragment,e),p(u.$$.fragment,e),p(L),R=!1},d(e){e&&(c(n),c(d),c(v),c(S)),A(a,e),A(_,e),A(u,e),L&&L.d(e)}}}function Bt(e){let a,t;return a=new Te({props:{$$slots:{default:[Mt]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};134219972&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function Gt(e){let a,t;return a=new $e({props:{size:e[12],$$slots:{default:[Bt]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};134219972&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function Ft(e){let a;return{c(){a=l("p"),a.textContent='You must use "Save as" or "Save link as" from the context menu (by right-clicking the button) to download the entire data.'},m(e,t){s(e,a,t)},p:T,d(e){e&&c(a)}}}function Nt(e){let a,t,n,_,l,d,r,o,u;const v=[{size:e[0]},{zebra:e[3]},{sortable:e[4]},{page:e[8]},{pageSize:e[9]},{rows:e[11]},{headers:e[1]},e[15]];let m={$$slots:{default:[Gt]},$$scope:{ctx:e}};for(let e=0;e<v.length;e+=1)m=G(m,v[e]);function S(a){e[19](a)}function R(a){e[20](a)}a=new Re({props:m});let h={pageSizes:e[5],totalItems:e[11].length};function $(a){e[21](a)}void 0!==e[9]&&(h.pageSize=e[9]),void 0!==e[8]&&(h.page=e[8]),n=new he({props:h}),b.push((()=>C(n,"pageSize",S))),b.push((()=>C(n,"page",R)));let T={passiveModal:!0,modalHeading:"Downloading entire data",$$slots:{default:[Ft]},$$scope:{ctx:e}};return void 0!==e[10]&&(T.open=e[10]),r=new oe({props:T}),b.push((()=>C(r,"open",$))),r.$on("open",e[22]),r.$on("close",e[23]),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment),d=f(),g(r.$$.fragment)},m(e,_){P(a,e,_),s(e,t,_),P(n,e,_),s(e,d,_),P(r,e,_),u=!0},p(e,[t]){const d=35611&t?H(v,[1&t&&{size:e[0]},8&t&&{zebra:e[3]},16&t&&{sortable:e[4]},256&t&&{page:e[8]},512&t&&{pageSize:e[9]},2048&t&&{rows:e[11]},2&t&&{headers:e[1]},32768&t&&I(e[15])]):{};134219972&t&&(d.$$scope={dirty:t,ctx:e}),a.$set(d);const s={};32&t&&(s.pageSizes=e[5]),2048&t&&(s.totalItems=e[11].length),!_&&512&t&&(_=!0,s.pageSize=e[9],O((()=>_=!1))),!l&&256&t&&(l=!0,s.page=e[8],O((()=>l=!1))),n.$set(s);const i={};134217728&t&&(i.$$scope={dirty:t,ctx:e}),!o&&1024&t&&(o=!0,i.open=e[10],O((()=>o=!1))),r.$set(i)},i(e){u||(i(a.$$.fragment,e),i(n.$$.fragment,e),i(r.$$.fragment,e),u=!0)},o(e){p(a.$$.fragment,e),p(n.$$.fragment,e),p(r.$$.fragment,e),u=!1},d(e){e&&(c(t),c(d)),A(a,e),A(n,e),A(r,e)}}}function Ut(e,a,t){const n=["size","src","zebra","sortable","rows","headers","pageSizes"];let _=M(a,n),{size:l="sm"}=a,{src:d}=a;l={sm:"short",small:"short",short:"short",md:"medium",medium:"medium",lg:"tall",large:"tall",tall:"tall",compact:"compact"}[l];let s={short:"sm",compact:"sm",medium:"default",tall:"default"}[l],{zebra:i=!0}=a,{sortable:r=!0}=a,{rows:o=[]}=a,{headers:p=[]}=a,{pageSizes:c=[20,50,75,100]}=a,u=p.map((e=>({...e,selected:!0}))),v=1,m=c[0],g="",P=!1,A=o;return e.$$set=e=>{a=G(G({},a),F(e)),t(15,_=M(a,n)),"size"in e&&t(0,l=e.size),"src"in e&&t(2,d=e.src),"zebra"in e&&t(3,i=e.zebra),"sortable"in e&&t(4,r=e.sortable),"rows"in e&&t(16,o=e.rows),"headers"in e&&t(1,p=e.headers),"pageSizes"in e&&t(5,c=e.pageSizes)},e.$$.update=()=>{64&e.$$.dirty&&t(1,p=u.filter((e=>e.selected))),65664&e.$$.dirty&&(g.length>0?t(11,A=o.filter((e=>Object.values(e).filter((e=>!!e&&e.toString().includes(g))).length>0))):t(11,A=o))},[l,p,d,i,r,c,u,g,v,m,P,A,s,e=>{e.preventDefault(),t(10,P=!0)},()=>{((e,a)=>{const t=document.createElement("a");t.href=e,t.download=a,document.body.appendChild(t),t.click(),document.body.removeChild(t)})("data:application/vnd.ms-excel;charset=utf-8,\ufeff"+encodeURIComponent(function(){const e=p.map((e=>e.value)),a=[];return A.forEach((e=>{const t=p.map((a=>`"${e[a.key]}"`));a.push(t.join(","))})),e.join(",")+"\r\n"+a.join("\r\n")}()),d&&d.length>0?d.split("/").pop().split(".")[0]+".xls":"data-subset.xls")},_,o,function(e){g=e,t(7,g)},function(e){u=e,t(6,u)},function(e){m=e,t(9,m)},function(e){v=e,t(8,v)},function(e){P=e,t(10,P)},function(a){Z.call(this,e,a)},function(a){Z.call(this,e,a)}]}class wt extends a{constructor(e){super(),t(this,e,Ut,Nt,n,{size:0,src:2,zebra:3,sortable:4,rows:16,headers:1,pageSizes:5})}}function Dt(e){let a,t,n;const _=[{headers:e[0]},{rows:e[1]},e[3]];let d={};for(let e=0;e<_.length;e+=1)d=G(d,_[e]);t=new wt({props:d});let r=[e[2]],o={};for(let e=0;e<r.length;e+=1)o=G(o,r[e]);return{c(){a=l("div"),g(t.$$.fragment),E(a,o)},m(e,_){s(e,a,_),P(t,a,null),n=!0},p(e,n){const l=11&n?H(_,[1&n&&{headers:e[0]},2&n&&{rows:e[1]},8&n&&I(e[3])]):{};t.$set(l),E(a,o=H(r,[4&n&&e[2]]))},i(e){n||(i(t.$$.fragment,e),n=!0)},o(e){p(t.$$.fragment,e),n=!1},d(e){e&&c(a),A(t)}}}function Et(e){let a,t;const n=[{headers:e[0]},{rows:e[1]},e[3]];let _={};for(let e=0;e<n.length;e+=1)_=G(_,n[e]);return a=new wt({props:_}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const _=11&t?H(n,[1&t&&{headers:e[0]},2&t&&{rows:e[1]},8&t&&I(e[3])]):{};a.$set(_)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function Ht(e){let a,t,n,_;const l=[Et,Dt],d=[];function u(e,a){return e[2]?1:0}return a=u(e),t=d[a]=l[a](e),{c(){t.c(),n=m()},m(e,t){d[a].m(e,t),s(e,n,t),_=!0},p(e,[_]){let s=a;a=u(e),a===s?d[a].p(e,_):(r(),p(d[s],1,1,(()=>{d[s]=null})),o(),t=d[a],t?t.p(e,_):(t=d[a]=l[a](e),t.c()),i(t,1),t.m(n.parentNode,n))},i(e){_||(i(t),_=!0)},o(e){p(t),_=!1},d(e){e&&c(n),d[a].d(e)}}}function It(e,a,t){const n=["frameProps","data","headers","rows"];let _=M(a,n),{frameProps:l={class:"pipen-report-datatable-frame"}}=a,{data:d={}}=a,{headers:s=[]}=a,{rows:i=[]}=a;if(s.length+i.length>0&&d.length>0)throw"Can only specify `data` or `headers`/`rows` for `DataTable`!";return d.length>0&&(i=d,s=Object.keys(i[0]).filter((e=>"id"!==e)).map((e=>({key:e,value:e})))),e.$$set=e=>{a=G(G({},a),F(e)),t(3,_=M(a,n)),"frameProps"in e&&t(2,l=e.frameProps),"data"in e&&t(4,d=e.data),"headers"in e&&t(0,s=e.headers),"rows"in e&&t(1,i=e.rows)},[s,i,l,_,d]}class kt extends a{constructor(e){super(),t(this,e,It,Ht,n,{frameProps:2,data:4,headers:0,rows:1})}}function Vt(e){let a,t;return{c(){a=l("div"),t=S(e[0]),d(a,"class","pipen-report-descr-title svelte-1xzxuva")},m(e,n){s(e,a,n),R(a,t)},p(e,a){1&a&&h(t,e[0])},d(e){e&&c(a)}}}function zt(e){let a,t,n,_=e[0]&&Vt(e);const d=e[3].default,r=y(d,e,e[2],null);let o=[e[1]],u={};for(let e=0;e<o.length;e+=1)u=G(u,o[e]);return{c(){a=l("div"),_&&_.c(),t=f(),r&&r.c(),E(a,u),k(a,"svelte-1xzxuva",!0)},m(e,l){s(e,a,l),_&&_.m(a,null),R(a,t),r&&r.m(a,null),n=!0},p(e,[l]){e[0]?_?_.p(e,l):(_=Vt(e),_.c(),_.m(a,t)):_&&(_.d(1),_=null),r&&r.p&&(!n||4&l)&&L(r,d,e,e[2],n?x(d,e[2],l,null):j(e[2]),null),E(a,u=H(o,[2&l&&e[1]])),k(a,"svelte-1xzxuva",!0)},i(e){n||(i(r,e),n=!0)},o(e){p(r,e),n=!1},d(e){e&&c(a),_&&_.d(),r&&r.d(e)}}}function qt(e,a,t){const n=["title"];let _=M(a,n),{$$slots:l={},$$scope:d}=a,{title:s}=a,{klass:i,...r}=_;return i=i?`pipen-report-descr ${i}`:"pipen-report-descr",r||(r={}),r.class=i,e.$$set=e=>{a=G(G({},a),F(e)),t(5,_=M(a,n)),"title"in e&&t(0,s=e.title),"$$scope"in e&&t(2,d=e.$$scope)},[s,r,d,l]}class Yt extends a{constructor(e){super(),t(this,e,qt,zt,n,{title:0})}}function Kt(e){let a,t,n,_,l,d;return a=new ue({props:{label:"Significant Markers"}}),n=new ue({props:{label:"Volcano Plot"}}),l=new ue({props:{label:"Dot Plot"}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment),_=f(),g(l.$$.fragment)},m(e,i){P(a,e,i),s(e,t,i),P(n,e,i),s(e,_,i),P(l,e,i),d=!0},p:T,i(e){d||(i(a.$$.fragment,e),i(n.$$.fragment,e),i(l.$$.fragment,e),d=!0)},o(e){p(a.$$.fragment,e),p(n.$$.fragment,e),p(l.$$.fragment,e),d=!1},d(e){e&&(c(t),c(_)),A(a,e),A(n,e),A(l,e)}}}function Xt(e){let a;return{c(){a=S("The markers are found using Seurat's FindMarkers function, and filtered by: p_val_adj < 0.05")},m(e,t){s(e,a,t)},d(e){e&&c(a)}}}function Qt(e){let a,t,n,_;return a=new Yt({props:{$$slots:{default:[Xt]},$$scope:{ctx:e}}}),n=new kt({props:{src:"../MarkersFinder/samples.markers/c1/markers.txt",data:[{gene:"HLA-B",p_val:3104e-90,avg_log2FC:.9855,pct_1:1,pct_2:1,p_val_adj:1041e-85,id:0},{gene:"RPS14",p_val:2289e-51,avg_log2FC:-.5403,pct_1:1,pct_2:1,p_val_adj:7676e-47,id:1},{gene:"RPL32",p_val:1537e-44,avg_log2FC:-.5014,pct_1:.997,pct_2:1,p_val_adj:5155e-40,id:2},{gene:"PSMB9",p_val:1822e-40,avg_log2FC:1.1103,pct_1:.889,pct_2:.704,p_val_adj:6109e-36,id:3},{gene:"RPL30",p_val:4346e-39,avg_log2FC:-.4639,pct_1:1,pct_2:1,p_val_adj:1458e-34,id:4},{gene:"DUSP4",p_val:4766e-39,avg_log2FC:1.9484,pct_1:.649,pct_2:.293,p_val_adj:1598e-34,id:5},{gene:"CXCR3",p_val:7118e-36,avg_log2FC:2.2574,pct_1:.465,pct_2:.135,p_val_adj:2387e-31,id:6},{gene:"ISG20",p_val:1374e-35,avg_log2FC:1.0657,pct_1:.933,pct_2:.751,p_val_adj:4606e-31,id:7},{gene:"RPS3A",p_val:4295e-34,avg_log2FC:-.4227,pct_1:.997,pct_2:1,p_val_adj:144e-28,id:8},{gene:"CST7",p_val:6.836e-30,avg_log2FC:1.7166,pct_1:.738,pct_2:.474,p_val_adj:2293e-28,id:9},{gene:"RPL19",p_val:1577e-32,avg_log2FC:-.3967,pct_1:1,pct_2:1,p_val_adj:529e-27,id:10},{gene:"RPS12",p_val:1772e-32,avg_log2FC:-.4429,pct_1:.997,pct_2:1,p_val_adj:5942e-28,id:11},{gene:"DUSP1",p_val:1932e-32,avg_log2FC:-.9563,pct_1:.933,pct_2:.976,p_val_adj:6478e-28,id:12},{gene:"RPL21",p_val:2215e-32,avg_log2FC:-.5329,pct_1:.992,pct_2:.994,p_val_adj:7428e-28,id:13},{gene:"RPL13",p_val:3286e-31,avg_log2FC:-.3868,pct_1:.997,pct_2:1,p_val_adj:1102e-26,id:14},{gene:"RPL10A",p_val:5132e-31,avg_log2FC:-.5052,pct_1:.992,pct_2:.996,p_val_adj:1721e-26,id:15},{gene:"GPR171",p_val:9615e-30,avg_log2FC:3.2452,pct_1:.259,pct_2:.038,p_val_adj:3225e-25,id:16},{gene:"RPL34",p_val:1519e-29,avg_log2FC:-.4338,pct_1:.997,pct_2:1,p_val_adj:5096e-25,id:17},{gene:"FOS",p_val:2417e-29,avg_log2FC:-1.473,pct_1:.646,pct_2:.847,p_val_adj:8105e-25,id:18},{gene:"PDE3B",p_val:1534e-28,avg_log2FC:1.8866,pct_1:.351,pct_2:.088,p_val_adj:5144e-24,id:19},{gene:"RNF213",p_val:8151e-28,avg_log2FC:1.0996,pct_1:.822,pct_2:.581,p_val_adj:2.734e-20,id:20},{gene:"CCL5",p_val:8168e-28,avg_log2FC:4.1162,pct_1:.217,pct_2:.025,p_val_adj:2.739e-20,id:21},{gene:"FKBP5",p_val:1267e-27,avg_log2FC:1.8825,pct_1:.552,pct_2:.262,p_val_adj:4.251e-20,id:22},{gene:"RPS18",p_val:2537e-27,avg_log2FC:-.4335,pct_1:.997,pct_2:1,p_val_adj:8.508e-20,id:23},{gene:"RPL10",p_val:8472e-27,avg_log2FC:-.3391,pct_1:1,pct_2:1,p_val_adj:2841e-22,id:24},{gene:"RPS23",p_val:3243e-26,avg_log2FC:-.3647,pct_1:1,pct_2:1,p_val_adj:1088e-21,id:25},{gene:"RPS3",p_val:4351e-26,avg_log2FC:-.3425,pct_1:.997,pct_2:1,p_val_adj:1459e-21,id:26},{gene:"RPL3",p_val:3462e-25,avg_log2FC:-.4044,pct_1:.994,pct_2:1,p_val_adj:1161e-20,id:27},{gene:"FOSB",p_val:3693e-25,avg_log2FC:-1.8982,pct_1:.337,pct_2:.597,p_val_adj:1238e-20,id:28},{gene:"RPL22",p_val:6976e-25,avg_log2FC:-.4233,pct_1:.986,pct_2:.994,p_val_adj:234e-19,id:29},{gene:"LMO4",p_val:1855e-24,avg_log2FC:4.2601,pct_1:.178,pct_2:.018,p_val_adj:622e-19,id:30},{gene:"RPL17",p_val:2735e-24,avg_log2FC:-.367,pct_1:.992,pct_2:.999,p_val_adj:9173e-20,id:31},{gene:"TAP1",p_val:2862e-24,avg_log2FC:1.0883,pct_1:.749,pct_2:.535,p_val_adj:9599e-20,id:32},{gene:"EEF1A1",p_val:3561e-24,avg_log2FC:-.2877,pct_1:1,pct_2:1,p_val_adj:1194e-19,id:33},{gene:"RPS6",p_val:3595e-24,avg_log2FC:-.3766,pct_1:.997,pct_2:1,p_val_adj:1206e-19,id:34},{gene:"RPS27A",p_val:3628e-24,avg_log2FC:-.3312,pct_1:1,pct_2:1,p_val_adj:1217e-19,id:35},{gene:"PSME2",p_val:1.601e-20,avg_log2FC:.8151,pct_1:.872,pct_2:.706,p_val_adj:537e-18,id:36},{gene:"RPLP2",p_val:4.32e-20,avg_log2FC:-.3636,pct_1:.997,pct_2:.999,p_val_adj:0,id:37},{gene:"RPL26",p_val:5.442e-20,avg_log2FC:-.3125,pct_1:.994,pct_2:1,p_val_adj:0,id:38},{gene:"RPS5",p_val:6.595e-20,avg_log2FC:-.4127,pct_1:.997,pct_2:1,p_val_adj:0,id:39},{gene:"IFITM2",p_val:6.971e-20,avg_log2FC:.9357,pct_1:.83,pct_2:.65,p_val_adj:0,id:40},{gene:"UBE2L6",p_val:1396e-22,avg_log2FC:1.0317,pct_1:.696,pct_2:.457,p_val_adj:0,id:41},{gene:"SAT1",p_val:1503e-22,avg_log2FC:1.1055,pct_1:.894,pct_2:.781,p_val_adj:0,id:42},{gene:"FKBP11",p_val:1649e-22,avg_log2FC:2.4745,pct_1:.253,pct_2:.06,p_val_adj:0,id:43},{gene:"IFNG",p_val:2272e-22,avg_log2FC:4.3227,pct_1:.139,pct_2:.007,p_val_adj:0,id:44},{gene:"RPL12",p_val:2356e-22,avg_log2FC:-.3396,pct_1:.997,pct_2:1,p_val_adj:0,id:45},{gene:"CD74",p_val:4395e-22,avg_log2FC:.8801,pct_1:.972,pct_2:.941,p_val_adj:0,id:46},{gene:"B2M",p_val:4574e-22,avg_log2FC:.3083,pct_1:1,pct_2:1,p_val_adj:0,id:47},{gene:"WARS",p_val:5206e-22,avg_log2FC:1.9718,pct_1:.323,pct_2:.103,p_val_adj:0,id:48},{gene:"SLC27A2",p_val:1664e-21,avg_log2FC:3.5301,pct_1:.153,pct_2:.015,p_val_adj:0,id:49},{gene:"SRGN",p_val:3619e-21,avg_log2FC:.5455,pct_1:.994,pct_2:.996,p_val_adj:0,id:50},{gene:"LY6E",p_val:468e-20,avg_log2FC:.8654,pct_1:.808,pct_2:.616,p_val_adj:0,id:51},{gene:"RPL9",p_val:5026e-21,avg_log2FC:-.3628,pct_1:.994,pct_2:.997,p_val_adj:0,id:52},{gene:"RPS8",p_val:5898e-21,avg_log2FC:-.3367,pct_1:1,pct_2:1,p_val_adj:0,id:53},{gene:"RPS13",p_val:6743e-21,avg_log2FC:-.3418,pct_1:.997,pct_2:1,p_val_adj:0,id:54},{gene:"STAT1",p_val:6966e-21,avg_log2FC:1.1925,pct_1:.677,pct_2:.435,p_val_adj:0,id:55},{gene:"RPL11",p_val:1061e-20,avg_log2FC:-.2963,pct_1:1,pct_2:1,p_val_adj:0,id:56},{gene:"ETV7",p_val:2328e-20,avg_log2FC:1.6641,pct_1:.312,pct_2:.101,p_val_adj:0,id:57},{gene:"RPS15",p_val:2523e-20,avg_log2FC:-.2924,pct_1:.997,pct_2:1,p_val_adj:0,id:58},{gene:"AL157402.2",p_val:2945e-20,avg_log2FC:2.8371,pct_1:.22,pct_2:.051,p_val_adj:0,id:59},{gene:"RPS21",p_val:4547e-20,avg_log2FC:-.4003,pct_1:.992,pct_2:.996,p_val_adj:0,id:60},{gene:"RARRES3",p_val:5167e-20,avg_log2FC:1.0253,pct_1:.816,pct_2:.649,p_val_adj:0,id:61},{gene:"TXNIP",p_val:7543e-20,avg_log2FC:-.9098,pct_1:.529,pct_2:.747,p_val_adj:0,id:62},{gene:"DUSP2",p_val:1672e-19,avg_log2FC:.809,pct_1:.861,pct_2:.706,p_val_adj:0,id:63},{gene:"GNA15",p_val:1783e-19,avg_log2FC:2.1936,pct_1:.212,pct_2:.047,p_val_adj:0,id:64},{gene:"MT-ND5",p_val:3551e-19,avg_log2FC:-.4642,pct_1:.975,pct_2:.988,p_val_adj:0,id:65},{gene:"BHLHE40",p_val:4775e-19,avg_log2FC:1.9665,pct_1:.281,pct_2:.09,p_val_adj:0,id:66},{gene:"GAPDH",p_val:4896e-19,avg_log2FC:.4954,pct_1:1,pct_2:.997,p_val_adj:0,id:67},{gene:"RPS28",p_val:5021e-19,avg_log2FC:-.3324,pct_1:.997,pct_2:1,p_val_adj:0,id:68},{gene:"RPS25",p_val:6316e-19,avg_log2FC:-.3085,pct_1:1,pct_2:1,p_val_adj:0,id:69},{gene:"RPL35A",p_val:6522e-19,avg_log2FC:-.315,pct_1:.997,pct_2:1,p_val_adj:0,id:70},{gene:"SPINT2",p_val:6699e-19,avg_log2FC:-1.4845,pct_1:.153,pct_2:.403,p_val_adj:0,id:71},{gene:"IL12RB2",p_val:0,avg_log2FC:2.0678,pct_1:.22,pct_2:.054,p_val_adj:0,id:72},{gene:"EEF1B2",p_val:0,avg_log2FC:-.4358,pct_1:.967,pct_2:.984,p_val_adj:0,id:73},{gene:"RPL7",p_val:0,avg_log2FC:-.3847,pct_1:.967,pct_2:.981,p_val_adj:0,id:74},{gene:"RPL5",p_val:0,avg_log2FC:-.3126,pct_1:.994,pct_2:1,p_val_adj:0,id:75},{gene:"CREM",p_val:0,avg_log2FC:.8055,pct_1:.733,pct_2:.515,p_val_adj:0,id:76},{gene:"NPDC1",p_val:0,avg_log2FC:1.6766,pct_1:.276,pct_2:.091,p_val_adj:0,id:77},{gene:"ADAM19",p_val:0,avg_log2FC:2.0483,pct_1:.242,pct_2:.072,p_val_adj:0,id:78},{gene:"HLA-A",p_val:0,avg_log2FC:.4523,pct_1:1,pct_2:.999,p_val_adj:0,id:79},{gene:"RPL29",p_val:0,avg_log2FC:-.285,pct_1:.997,pct_2:.999,p_val_adj:0,id:80},{gene:"LGALS3BP",p_val:0,avg_log2FC:2.6144,pct_1:.175,pct_2:.035,p_val_adj:0,id:81},{gene:"RGS2",p_val:0,avg_log2FC:1.2165,pct_1:.454,pct_2:.219,p_val_adj:0,id:82},{gene:"MAP3K8",p_val:0,avg_log2FC:1.6692,pct_1:.331,pct_2:.135,p_val_adj:0,id:83},{gene:"SATB1",p_val:0,avg_log2FC:1.455,pct_1:.351,pct_2:.146,p_val_adj:0,id:84},{gene:"RPL18A",p_val:0,avg_log2FC:-.2682,pct_1:1,pct_2:1,p_val_adj:0,id:85},{gene:"HLA-DRB1",p_val:0,avg_log2FC:.9837,pct_1:.409,pct_2:.179,p_val_adj:0,id:86},{gene:"DYNLT1",p_val:0,avg_log2FC:.9522,pct_1:.602,pct_2:.385,p_val_adj:0,id:87},{gene:"RPL39",p_val:0,avg_log2FC:-.2898,pct_1:1,pct_2:.999,p_val_adj:0,id:88},{gene:"SH2D2A",p_val:0,avg_log2FC:.8556,pct_1:.721,pct_2:.559,p_val_adj:0,id:89},{gene:"MCOLN2",p_val:0,avg_log2FC:2.6274,pct_1:.153,pct_2:.029,p_val_adj:0,id:90},{gene:"CYTIP",p_val:0,avg_log2FC:.5718,pct_1:.928,pct_2:.841,p_val_adj:0,id:91},{gene:"RPL8",p_val:0,avg_log2FC:-.2589,pct_1:1,pct_2:1,p_val_adj:0,id:92},{gene:"EIF4B",p_val:0,avg_log2FC:-.584,pct_1:.752,pct_2:.857,p_val_adj:0,id:93},{gene:"RPS7",p_val:0,avg_log2FC:-.2424,pct_1:1,pct_2:1,p_val_adj:0,id:94},{gene:"CXCL13",p_val:0,avg_log2FC:7.4242,pct_1:.103,pct_2:.009,p_val_adj:0,id:95},{gene:"RPS16",p_val:0,avg_log2FC:-.292,pct_1:.997,pct_2:.999,p_val_adj:0,id:96},{gene:"FAU",p_val:0,avg_log2FC:-.2621,pct_1:.997,pct_2:1,p_val_adj:0,id:97},{gene:"OAZ1",p_val:0,avg_log2FC:.392,pct_1:.986,pct_2:.972,p_val_adj:0,id:98},{gene:"MT-ATP6",p_val:0,avg_log2FC:-.3113,pct_1:.994,pct_2:.994,p_val_adj:0,id:99}]}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment)},m(e,l){P(a,e,l),s(e,t,l),P(n,e,l),_=!0},p(e,t){const n={};1&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){_||(i(a.$$.fragment,e),i(n.$$.fragment,e),_=!0)},o(e){p(a.$$.fragment,e),p(n.$$.fragment,e),_=!1},d(e){e&&c(t),A(a,e),A(n,e)}}}function Wt(e){let a,t;return a=new At({props:{src:"../MarkersFinder/samples.markers/c1/volcano.png",width:900,height:1200}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p:T,i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function Zt(e){let a,t;return a=new At({props:{src:"../MarkersFinder/samples.markers/c1/dotplot.png",width:600,height:1200}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p:T,i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function Jt(e){let a,t,n,_,l,d;return a=new ve({props:{$$slots:{default:[Qt]},$$scope:{ctx:e}}}),n=new ve({props:{$$slots:{default:[Wt]},$$scope:{ctx:e}}}),l=new ve({props:{$$slots:{default:[Zt]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment),_=f(),g(l.$$.fragment)},m(e,i){P(a,e,i),s(e,t,i),P(n,e,i),s(e,_,i),P(l,e,i),d=!0},p(e,t){const _={};1&t&&(_.$$scope={dirty:t,ctx:e}),a.$set(_);const d={};1&t&&(d.$$scope={dirty:t,ctx:e}),n.$set(d);const s={};1&t&&(s.$$scope={dirty:t,ctx:e}),l.$set(s)},i(e){d||(i(a.$$.fragment,e),i(n.$$.fragment,e),i(l.$$.fragment,e),d=!0)},o(e){p(a.$$.fragment,e),p(n.$$.fragment,e),p(l.$$.fragment,e),d=!1},d(e){e&&(c(t),c(_)),A(a,e),A(n,e),A(l,e)}}}function en(e){let a;return{c(){a=S("The enrichment analysis is done using Enrichr. The significant markers are used as input.")},m(e,t){s(e,a,t)},d(e){e&&c(a)}}}function an(e){let a,t,n,_;return a=new ue({props:{label:"Plot"}}),n=new ue({props:{label:"Table"}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment)},m(e,l){P(a,e,l),s(e,t,l),P(n,e,l),_=!0},p:T,i(e){_||(i(a.$$.fragment,e),i(n.$$.fragment,e),_=!0)},o(e){p(a.$$.fragment,e),p(n.$$.fragment,e),_=!1},d(e){e&&c(t),A(a,e),A(n,e)}}}function tn(e){let a,t;return a=new At({props:{src:"../MarkersFinder/samples.markers/c1/Enrichr-KEGG_2021_Human.png",width:1e3,height:1e3}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p:T,i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function nn(e){let a,t;return a=new kt({props:{src:"../MarkersFinder/samples.markers/c1/Enrichr-KEGG_2021_Human.txt",data:[{Term:"Ribosome",Overlap:"61/158",P_value:5.843e-70,Adjusted_P_value:1134e-70,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:49.8892,Combined_Score:7953.1203,Genes:"RPL5;RPL30;RPL3;RPL32;RPL34;RPL10A;RPL8;RPL9;RPL6;RPL7;RPS15;RPS4X;RPS14;RPL7A;RPS16;RPL18A;RPS18;RPL36;RPL35;RPLP2;RPL38;RPL37;RPS11;RPL39;RPS13;RPS12;RPL21;RPS7;RPS8;RPS5;RPL22;RPS6;RPS3A;RPL24;RPL26;RPL29;RPL28;UBA52;RPL10;RPL12;RPL11;RPL36A;MRPL10;RPS3;RPL13;RPL15;RPL18;RPS27A;RPL17;RPL19;RPL41;RPL35A;RPL23A;RPS25;RPS28;RPS27;RPL27A;RPS20;FAU;RPS21;RPS23",id:0},{Term:"Coronavirus disease",Overlap:"64/232",P_value:1603e-65,Adjusted_P_value:1.555e-60,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:30.4824,Combined_Score:4337.3015,Genes:"RPL5;RPL30;RPL3;CSF2;RPL32;RPL34;RPL8;RPL10A;RPL9;RPL6;RPL7;RPS15;RPS4X;RPS14;RPL7A;RPS16;RPL18A;RPS18;RPL36;RPL35;RPLP2;RPL38;RPL37;RPS11;RPL39;RPS13;RPS12;RPL21;RPS7;RPS8;RPS5;RPL22;RPS6;RPS3A;FOS;RPL24;RPL26;RPL29;RPL28;UBA52;RPL10;RPL12;RPL11;RPL36A;RPS3;RPL13;RPL15;RPL18;RPS27A;RPL17;RPL19;RPL41;STAT1;RPL35A;ISG15;RPL23A;RPS25;RPS28;RPS27;RPL27A;RPS20;FAU;RPS21;RPS23",id:1},{Term:"Antigen processing and presentation",Overlap:"13/78",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:13.3064,Combined_Score:297.7508,Genes:"CD74;HLA-B;TAP1;HLA-A;CD8B;IFNG;CD8A;PSME1;PSME2;HLA-DRA;B2M;HLA-DRB1;HLA-DPA1",id:2},{Term:"Proteasome",Overlap:"9/46",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:15.9898,Combined_Score:277.5386,Genes:"PSMA5;PSMB2;IFNG;PSMA1;PSME1;PSME2;PSMB10;PSMB8;PSMB9",id:3},{Term:"Type I diabetes mellitus",Overlap:"8/43",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:14.9768,Combined_Score:226.9598,Genes:"IFNG;ICA1;HLA-B;GZMB;HLA-DRA;HLA-A;HLA-DRB1;HLA-DPA1",id:4},{Term:"Epstein-Barr virus infection",Overlap:"15/202",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:5.3398,Combined_Score:76.5454,Genes:"STAT1;HLA-B;TAP1;TNFAIP3;ISG15;HLA-A;RUNX3;RELB;BCL2L11;IRF7;HLA-DRA;VIM;B2M;HLA-DRB1;HLA-DPA1",id:5},{Term:"Inflammatory bowel disease",Overlap:"9/65",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:10.5545,Combined_Score:150.3555,Genes:"IL22;IL18RAP;IFNG;STAT1;TBX21;HLA-DRA;HLA-DRB1;HLA-DPA1;IL12RB2",id:6},{Term:"Th1 and Th2 cell differentiation",Overlap:"10/92",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:8.025,Combined_Score:107.4409,Genes:"IFNG;STAT1;TBX21;NFATC2;HLA-DRA;FOS;RUNX3;HLA-DRB1;HLA-DPA1;IL12RB2",id:7},{Term:"Allograft rejection",Overlap:"7/38",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:14.7494,Combined_Score:196.7667,Genes:"IFNG;HLA-B;GZMB;HLA-DRA;HLA-A;HLA-DRB1;HLA-DPA1",id:8},{Term:"Graft-versus-host disease",Overlap:"7/42",P_value:0,Adjusted_P_value:1e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:13.0611,Combined_Score:165.0093,Genes:"IFNG;HLA-B;GZMB;HLA-DRA;HLA-A;HLA-DRB1;HLA-DPA1",id:9},{Term:"Osteoclast differentiation",Overlap:"11/127",P_value:0,Adjusted_P_value:1e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:6.2503,Combined_Score:77.0815,Genes:"SOCS1;IFNG;STAT1;NFATC2;ACP5;FOSB;CYBA;FYN;FOS;JUNB;RELB",id:10},{Term:"T cell receptor signaling pathway",Overlap:"10/104",P_value:0,Adjusted_P_value:1e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:6.9963,Combined_Score:85.805,Genes:"CSF2;CD8B;IFNG;CD8A;NFATC2;CBLB;MAP3K8;FYN;FOS;ICOS",id:11},{Term:"Hematopoietic cell lineage",Overlap:"9/99",P_value:0,Adjusted_P_value:3e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:6.5559,Combined_Score:70.2206,Genes:"CD2;CSF2;CD8B;CD8A;CD7;HLA-DRA;CD38;HLA-DRB1;HLA-DPA1",id:12},{Term:"Leishmaniasis",Overlap:"8/77",P_value:0,Adjusted_P_value:3e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:7.5838,Combined_Score:80.6151,Genes:"EEF1A1;IFNG;STAT1;HLA-DRA;CYBA;FOS;HLA-DRB1;HLA-DPA1",id:13},{Term:"Viral myocarditis",Overlap:"7/60",P_value:0,Adjusted_P_value:5e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:8.6174,Combined_Score:87.9754,Genes:"RAC2;HLA-B;HLA-DRA;FYN;HLA-A;HLA-DRB1;HLA-DPA1",id:14},{Term:"Th17 cell differentiation",Overlap:"9/107",P_value:0,Adjusted_P_value:5e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:6.0182,Combined_Score:60.7256,Genes:"IL22;IFNG;STAT1;TBX21;NFATC2;HLA-DRA;FOS;HLA-DRB1;HLA-DPA1",id:15},{Term:"Rheumatoid arthritis",Overlap:"8/93",P_value:1e-4,Adjusted_P_value:.0011,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:6.1512,Combined_Score:56.9946,Genes:"CSF2;IFNG;CCL5;ACP5;HLA-DRA;FOS;HLA-DRB1;HLA-DPA1",id:16},{Term:"Autoimmune thyroid disease",Overlap:"6/53",P_value:2e-4,Adjusted_P_value:.0017,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:8.3042,Combined_Score:72.6495,Genes:"HLA-B;GZMB;HLA-DRA;HLA-A;HLA-DRB1;HLA-DPA1",id:17},{Term:"Fluid shear stress and atherosclerosis",Overlap:"9/139",P_value:3e-4,Adjusted_P_value:.0032,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.5294,Combined_Score:36.6081,Genes:"IFNG;DUSP1;MGST3;SUMO2;RAC2;PLAT;CYBA;TXN;FOS",id:18},{Term:"TNF signaling pathway",Overlap:"8/112",P_value:3e-4,Adjusted_P_value:.0033,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:5.0226,Combined_Score:40.0615,Genes:"CSF2;IRF1;CCL5;TNFAIP3;MAP3K8;FOS;CFLAR;JUNB",id:19},{Term:"Cell adhesion molecules",Overlap:"9/148",P_value:5e-4,Adjusted_P_value:.0044,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.2342,Combined_Score:32.2702,Genes:"CD2;CD8B;CD8A;HLA-B;HLA-DRA;HLA-A;ICOS;HLA-DRB1;HLA-DPA1",id:20},{Term:"Parkinson disease",Overlap:"12/249",P_value:5e-4,Adjusted_P_value:.0044,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.3279,Combined_Score:25.2389,Genes:"PSMA5;PSMA1;PSMB2;DUSP1;TUBB;NDUFS5;UBE2L6;TXN;RPS27A;SLC25A5;UBA52;COX5A",id:21},{Term:"NOD-like receptor signaling pathway",Overlap:"10/181",P_value:5e-4,Adjusted_P_value:.0044,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.8308,Combined_Score:28.9928,Genes:"PYCARD;STAT1;CCL5;IRF7;TXNIP;TNFAIP3;CYBA;TXN;GBP1;GBP4",id:22},{Term:"Human T-cell leukemia virus 1 infection",Overlap:"11/219",P_value:6e-4,Adjusted_P_value:.005,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.4694,Combined_Score:25.6591,Genes:"CSF2;HLA-B;NFATC2;HLA-DRA;HLA-A;FOS;SLC25A5;B2M;HLA-DRB1;HLA-DPA1;RELB",id:23},{Term:"Cytokine-cytokine receptor interaction",Overlap:"13/295",P_value:7e-4,Adjusted_P_value:.0054,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.0332,Combined_Score:22.0418,Genes:"IL32;IL22;CSF2;TNFRSF18;IL16;CXCL13;IL18RAP;IFNG;CXCR3;CCL5;TNFRSF25;TNFRSF4;IL12RB2",id:24},{Term:"Natural killer cell mediated cytotoxicity",Overlap:"8/131",P_value:.001,Adjusted_P_value:.0073,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.2426,Combined_Score:29.4254,Genes:"CSF2;IFNG;HLA-B;RAC2;NFATC2;GZMB;FYN;HLA-A",id:25},{Term:"Prion disease",Overlap:"12/273",P_value:.0011,Adjusted_P_value:.0082,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.0182,Combined_Score:20.4666,Genes:"PSMA5;PSMA1;PSMB2;TUBB;NDUFS5;CCL5;RAC2;CYBA;FYN;SLC25A5;COX5A;SOD1",id:26},{Term:"Influenza A",Overlap:"9/172",P_value:.0014,Adjusted_P_value:.0099,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.6063,Combined_Score:23.6322,Genes:"PYCARD;IFNG;STAT1;CCL5;IRF7;HLA-DRA;SLC25A5;HLA-DRB1;HLA-DPA1",id:27},{Term:"Phagosome",Overlap:"8/152",P_value:.0025,Adjusted_P_value:.0167,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.62,Combined_Score:21.6903,Genes:"TUBB;HLA-B;TAP1;HLA-DRA;CYBA;HLA-A;HLA-DRB1;HLA-DPA1",id:28},{Term:"Primary immunodeficiency",Overlap:"4/38",P_value:.0027,Adjusted_P_value:.0174,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:7.6076,Combined_Score:45.008,Genes:"CD8B;CD8A;TAP1;ICOS",id:29},{Term:"Kaposi sarcoma-associated herpesvirus infection",Overlap:"9/193",P_value:.0031,Adjusted_P_value:.0195,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.1913,Combined_Score:18.4176,Genes:"CSF2;STAT1;HLA-B;IRF7;NFATC2;HLA-A;FOS;RPS27A;UBA52",id:30},{Term:"JAK-STAT signaling pathway",Overlap:"8/162",P_value:.0037,Adjusted_P_value:.0224,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.3832,Combined_Score:18.9508,Genes:"IL22;CSF2;SOCS1;IFNG;STAT1;PIM1;MCL1;IL12RB2",id:31},{Term:"C-type lectin receptor signaling pathway",Overlap:"6/104",P_value:.0055,Adjusted_P_value:.0321,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.9723,Combined_Score:20.6939,Genes:"PYCARD;STAT1;IRF1;NFATC2;CBLB;RELB",id:32},{Term:"Intestinal immune network for IgA production",Overlap:"4/48",P_value:.0063,Adjusted_P_value:.036,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:5.8756,Combined_Score:29.7712,Genes:"HLA-DRA;ICOS;HLA-DRB1;HLA-DPA1",id:33},{Term:"Apoptosis",Overlap:"7/142",P_value:.0065,Adjusted_P_value:.0361,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.369,Combined_Score:16.9555,Genes:"BCL2L11;PMAIP1;GZMB;FOS;CFLAR;MCL1;LMNB1",id:34},{Term:"Toxoplasmosis",Overlap:"6/112",P_value:.0078,Adjusted_P_value:.0419,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.671,Combined_Score:17.825,Genes:"SOCS1;IFNG;STAT1;HLA-DRA;HLA-DRB1;HLA-DPA1",id:35},{Term:"Necroptosis",Overlap:"7/159",P_value:.0118,Adjusted_P_value:.0603,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.9896,Combined_Score:13.2846,Genes:"PYCARD;IFNG;STAT1;TNFAIP3;HMGB1;CFLAR;SLC25A5",id:36},{Term:"Asthma",Overlap:"3/31",P_value:.0118,Adjusted_P_value:.0603,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:6.9077,Combined_Score:30.656,Genes:"HLA-DRA;HLA-DRB1;HLA-DPA1",id:37},{Term:"IL-17 signaling pathway",Overlap:"5/94",P_value:.0151,Adjusted_P_value:.0752,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.6346,Combined_Score:15.2348,Genes:"CSF2;IFNG;TNFAIP3;FOSB;FOS",id:38},{Term:"Human immunodeficiency virus 1 infection",Overlap:"8/212",P_value:.0173,Adjusted_P_value:.0837,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.5475,Combined_Score:10.3401,Genes:"APOBEC3G;HLA-B;RAC2;NFATC2;TAP1;HLA-A;FOS;B2M",id:39},{Term:"Yersinia infection",Overlap:"6/137",P_value:.0195,Adjusted_P_value:.0923,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.9666,Combined_Score:11.6786,Genes:"PYCARD;CD8B;CD8A;RAC2;NFATC2;FOS",id:40},{Term:"Chagas disease",Overlap:"5/102",P_value:.0208,Adjusted_P_value:.0961,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.3335,Combined_Score:12.9098,Genes:"GNA15;IFNG;CCL5;FOS;CFLAR",id:41},{Term:"Toll-like receptor signaling pathway",Overlap:"5/104",P_value:.0224,Adjusted_P_value:.1008,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.2658,Combined_Score:12.4042,Genes:"STAT1;CCL5;IRF7;MAP3K8;FOS",id:42},{Term:"Prolactin signaling pathway",Overlap:"4/70",P_value:.0229,Adjusted_P_value:.1008,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.9127,Combined_Score:14.7837,Genes:"SOCS1;STAT1;IRF1;FOS",id:43},{Term:"Parathyroid hormone synthesis, secretion and action",Overlap:"5/106",P_value:.0241,Adjusted_P_value:.1039,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.2008,Combined_Score:11.9248,Genes:"NR4A2;NACA;PDE4B;FOS;RUNX2",id:44},{Term:"Transcriptional misregulation in cancer",Overlap:"7/192",P_value:.0296,Adjusted_P_value:.1246,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.4522,Combined_Score:8.6355,Genes:"ETV7;CSF2;NFKBIZ;PRCC;GZMB;PLAT;RUNX2",id:45},{Term:"Pathogenic Escherichia coli infection",Overlap:"7/197",P_value:.0333,Adjusted_P_value:.1374,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.387,Combined_Score:8.1213,Genes:"PYCARD;TUBB;F2R;RPS3;FYN;FOS;GAPDH",id:46},{Term:"B cell receptor signaling pathway",Overlap:"4/81",P_value:.0364,Adjusted_P_value:.1473,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.3518,Combined_Score:11.1018,Genes:"IFITM1;RAC2;NFATC2;FOS",id:47},{Term:"Diabetic cardiomyopathy",Overlap:"7/203",P_value:.0382,Adjusted_P_value:.1512,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.3132,Combined_Score:7.553,Genes:"TBC1D4;NDUFS5;RAC2;CYBA;SLC25A5;COX5A;GAPDH",id:48},{Term:"Salmonella infection",Overlap:"8/249",P_value:.0396,Adjusted_P_value:.1536,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.1523,Combined_Score:6.9499,Genes:"PYCARD;AHNAK;DYNLT1;TUBB;RPS3;TXN;FOS;GAPDH",id:49},{Term:"Colorectal cancer",Overlap:"4/86",P_value:.0439,Adjusted_P_value:.1668,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.1467,Combined_Score:9.8388,Genes:"BCL2L11;RAC2;PMAIP1;FOS",id:50},{Term:"Herpes simplex virus 1 infection",Overlap:"13/498",P_value:.0457,Adjusted_P_value:.1703,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.7452,Combined_Score:5.3866,Genes:"CD74;STAT1;HLA-B;TAP1;HLA-A;IFNG;CCL5;IRF7;HLA-DRA;B2M;SRSF7;HLA-DRB1;HLA-DPA1",id:51},{Term:"PD-L1 expression and PD-1 checkpoint pathway in cancer",Overlap:"4/89",P_value:.0487,Adjusted_P_value:.1782,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.0351,Combined_Score:9.1729,Genes:"IFNG;STAT1;NFATC2;FOS",id:52},{Term:"Legionellosis",Overlap:"3/57",P_value:.0576,Adjusted_P_value:.2069,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.577,Combined_Score:10.2102,Genes:"PYCARD;EEF1A1;EEF1G",id:53},{Term:"Human cytomegalovirus infection",Overlap:"7/225",P_value:.0601,Adjusted_P_value:.2076,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.0774,Combined_Score:5.8421,Genes:"CCL5;HLA-B;RAC2;NFATC2;TAP1;HLA-A;B2M",id:54},{Term:"Tuberculosis",Overlap:"6/180",P_value:.0604,Adjusted_P_value:.2076,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.2286,Combined_Score:6.2564,Genes:"CD74;IFNG;STAT1;HLA-DRA;HLA-DRB1;HLA-DPA1",id:55},{Term:"Insulin signaling pathway",Overlap:"5/137",P_value:.061,Adjusted_P_value:.2076,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.4452,Combined_Score:6.8397,Genes:"SOCS1;PRKAR1B;RPS6;PDE3B;CBLB",id:56},{Term:"VEGF signaling pathway",Overlap:"3/59",P_value:.0626,Adjusted_P_value:.2094,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.4489,Combined_Score:9.5573,Genes:"SH2D2A;RAC2;NFATC2",id:57},{Term:"Measles",Overlap:"5/139",P_value:.0641,Adjusted_P_value:.2107,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.4085,Combined_Score:6.6178,Genes:"STAT1;IRF7;TNFAIP3;CBLB;FOS",id:58},{Term:"Ubiquitin mediated proteolysis",Overlap:"5/140",P_value:.0657,Adjusted_P_value:.2123,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.3905,Combined_Score:6.5102,Genes:"SOCS1;CBLB;UBE2L6;RPS27A;UBA52",id:59},{Term:"Viral protein interaction with cytokine and cytokine receptor",Overlap:"4/100",P_value:.0688,Adjusted_P_value:.2188,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.6859,Combined_Score:7.1888,Genes:"IL18RAP;CCL5;CXCR3;CXCL13",id:60},{Term:"Cytosolic DNA-sensing pathway",Overlap:"3/63",P_value:.0732,Adjusted_P_value:.229,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.2184,Combined_Score:8.4156,Genes:"PYCARD;CCL5;IRF7",id:61},{Term:"Glycolysis / Gluconeogenesis",Overlap:"3/67",P_value:.0845,Adjusted_P_value:.2586,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.0166,Combined_Score:7.4542,Genes:"LDHA;GALM;GAPDH",id:62},{Term:"MAPK signaling pathway",Overlap:"8/294",P_value:.0857,Adjusted_P_value:.2586,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.8094,Combined_Score:4.4447,Genes:"DUSP4;DUSP5;DUSP2;DUSP1;RAC2;MAP3K8;FOS;RELB",id:63},{Term:"Fc epsilon RI signaling pathway",Overlap:"3/68",P_value:.0874,Adjusted_P_value:.2586,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.97,Combined_Score:7.2375,Genes:"CSF2;RAC2;FYN",id:64},{Term:"HIF-1 signaling pathway",Overlap:"4/109",P_value:.088,Adjusted_P_value:.2586,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.4545,Combined_Score:5.9665,Genes:"LDHA;IFNG;RPS6;GAPDH",id:65},{Term:"Oxytocin signaling pathway",Overlap:"5/154",P_value:.09,Adjusted_P_value:.2606,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.1644,Combined_Score:5.2116,Genes:"RGS2;MYL6;NFATC2;CD38;FOS",id:66},{Term:"Hepatitis C",Overlap:"5/157",P_value:.0958,Adjusted_P_value:.2732,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.1213,Combined_Score:4.9766,Genes:"IFNG;YWHAB;STAT1;IRF7;CFLAR",id:67},{Term:"Huntington disease",Overlap:"8/306",P_value:.1018,Adjusted_P_value:.2863,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.7355,Combined_Score:3.9644,Genes:"PSMA5;PSMA1;PSMB2;TUBB;NDUFS5;SLC25A5;COX5A;SOD1",id:68},{Term:"Hepatitis B",Overlap:"5/162",P_value:.1057,Adjusted_P_value:.2931,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.0532,Combined_Score:4.6131,Genes:"YWHAB;STAT1;IRF7;NFATC2;FOS",id:69},{Term:"MicroRNAs in cancer",Overlap:"8/310",P_value:.1076,Adjusted_P_value:.2939,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.7121,Combined_Score:3.8176,Genes:"ZEB2;BCL2L11;SOCS1;DDIT4;PIM1;VIM;FOXP1;MCL1",id:70},{Term:"Growth hormone synthesis, secretion and action",Overlap:"4/119",P_value:.1119,Adjusted_P_value:.2988,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.2399,Combined_Score:4.9052,Genes:"SOCS1;STAT1;FOS;JUNB",id:71},{Term:"Pertussis",Overlap:"3/76",P_value:.1124,Adjusted_P_value:.2988,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.6435,Combined_Score:5.7769,Genes:"PYCARD;IRF1;FOS",id:72},{Term:"Lipid and atherosclerosis",Overlap:"6/215",P_value:.1157,Adjusted_P_value:.3014,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.8521,Combined_Score:3.9937,Genes:"PYCARD;CCL5;IRF7;NFATC2;CYBA;FOS",id:73},{Term:"Riboflavin metabolism",Overlap:"1/8",P_value:.1168,Adjusted_P_value:.3014,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:9.1601,Combined_Score:19.671,Genes:"ACP5",id:74},{Term:"Pathways of neurodegeneration",Overlap:"11/475",P_value:.1181,Adjusted_P_value:.3014,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.5348,Combined_Score:3.2788,Genes:"PSMA5;PSMA1;PSMB2;TUBB;NDUFS5;UBE2L6;RPS27A;SLC25A5;UBA52;COX5A;SOD1",id:75},{Term:"Systemic lupus erythematosus",Overlap:"4/135",P_value:.1555,Adjusted_P_value:.3917,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.9647,Combined_Score:3.6571,Genes:"IFNG;HLA-DRA;HLA-DRB1;HLA-DPA1",id:76},{Term:"Chemokine signaling pathway",Overlap:"5/192",P_value:.1753,Adjusted_P_value:.4361,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.7212,Combined_Score:2.9967,Genes:"STAT1;CCL5;CXCR3;RAC2;CXCL13",id:77},{Term:"Spinocerebellar ataxia",Overlap:"4/143",P_value:.1793,Adjusted_P_value:.4378,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.8509,Combined_Score:3.1814,Genes:"PSMA5;PSMA1;PSMB2;SLC25A5",id:78},{Term:"Staphylococcus aureus infection",Overlap:"3/95",P_value:.1806,Adjusted_P_value:.4378,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.0955,Combined_Score:3.587,Genes:"HLA-DRA;HLA-DRB1;HLA-DPA1",id:79},{Term:"Endocytosis",Overlap:"6/252",P_value:.1941,Adjusted_P_value:.4649,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.5705,Combined_Score:2.5746,Genes:"CAPZB;HLA-B;ARAP2;CBLB;HLA-A;SMAP2",id:80},{Term:"Amyotrophic lateral sclerosis",Overlap:"8/364",P_value:.2007,Adjusted_P_value:.4748,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.4484,Combined_Score:2.3261,Genes:"PSMA5;PSMA1;PSMB2;TUBB;NDUFS5;COX5A;SRSF7;SOD1",id:81},{Term:"Viral carcinogenesis",Overlap:"5/203",P_value:.2044,Adjusted_P_value:.4777,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.6247,Combined_Score:2.5796,Genes:"YWHAB;HLA-B;IRF7;PMAIP1;HLA-A",id:82},{Term:"Amoebiasis",Overlap:"3/102",P_value:.208,Adjusted_P_value:.4785,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.9466,Combined_Score:3.057,Genes:"GNA15;CSF2;IFNG",id:83},{Term:"Alzheimer disease",Overlap:"8/369",P_value:.2106,Adjusted_P_value:.4785,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.428,Combined_Score:2.2241,Genes:"PSMA5;PSMA1;PSMB2;TUBB;NDUFS5;SLC25A5;COX5A;GAPDH",id:84},{Term:"mTOR signaling pathway",Overlap:"4/154",P_value:.2138,Adjusted_P_value:.4785,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.7142,Combined_Score:2.6444,Genes:"RPS6;DDIT4;EIF4B;LAMTOR5",id:85},{Term:"NF-kappa B signaling pathway",Overlap:"3/104",P_value:.216,Adjusted_P_value:.4785,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.9079,Combined_Score:2.9241,Genes:"TNFAIP3;CFLAR;RELB",id:86},{Term:"Non-alcoholic fatty liver disease",Overlap:"4/155",P_value:.2171,Adjusted_P_value:.4785,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.7028,Combined_Score:2.6012,Genes:"BCL2L11;NDUFS5;FOS;COX5A",id:87},{Term:"Cellular senescence",Overlap:"4/156",P_value:.2203,Adjusted_P_value:.4802,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.6915,Combined_Score:2.5589,Genes:"HLA-B;NFATC2;HLA-A;SLC25A5",id:88},{Term:"cAMP signaling pathway",Overlap:"5/216",P_value:.2406,Adjusted_P_value:.5186,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.5235,Combined_Score:2.1706,Genes:"F2R;PDE3B;RAC2;PDE4B;FOS",id:89},{Term:"cGMP-PKG signaling pathway",Overlap:"4/167",P_value:.2567,Adjusted_P_value:.5473,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.5764,Combined_Score:2.1435,Genes:"RGS2;PDE3B;NFATC2;SLC25A5",id:90},{Term:"One carbon pool by folate",Overlap:"1/20",P_value:.267,Adjusted_P_value:.5629,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.3727,Combined_Score:4.4543,Genes:"MTHFD2",id:91},{Term:"Acute myeloid leukemia",Overlap:"2/67",P_value:.276,Adjusted_P_value:.5758,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.9736,Combined_Score:2.5404,Genes:"CSF2;PIM1",id:92},{Term:"Mitophagy",Overlap:"2/68",P_value:.2817,Adjusted_P_value:.5815,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.9436,Combined_Score:2.462,Genes:"RPS27A;UBA52",id:93},{Term:"Amphetamine addiction",Overlap:"2/69",P_value:.2875,Adjusted_P_value:.587,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.9144,Combined_Score:2.3867,Genes:"FOSB;FOS",id:94},{Term:"RIG-I-like receptor signaling pathway",Overlap:"2/70",P_value:.2932,Adjusted_P_value:.5924,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.8862,Combined_Score:2.3144,Genes:"IRF7;ISG15",id:95},{Term:"Adherens junction",Overlap:"2/71",P_value:.2989,Adjusted_P_value:.5975,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.8588,Combined_Score:2.245,Genes:"RAC2;FYN",id:96},{Term:"Pathways in cancer",Overlap:"10/531",P_value:.3033,Adjusted_P_value:.5975,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.2348,Combined_Score:1.4729,Genes:"BCL2L11;IFNG;STAT1;MGST3;F2R;RAC2;PIM1;PMAIP1;FOS;IL12RB2",id:97},{Term:"PI3K-Akt signaling pathway",Overlap:"7/354",P_value:.3049,Adjusted_P_value:.5975,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.2965,Combined_Score:1.5399,Genes:"BCL2L11;YWHAB;DDIT4;RPS6;F2R;EIF4B;MCL1",id:98},{Term:"Calcium signaling pathway",Overlap:"5/240",P_value:.3112,Adjusted_P_value:.6037,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.3663,Combined_Score:1.595,Genes:"GNA15;F2R;CD38;SLC25A5;MCOLN2",id:99},{Term:"PPAR signaling pathway",Overlap:"2/74",P_value:.3159,Adjusted_P_value:.6068,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.781,Combined_Score:2.0523,Genes:"GK;SLC27A2",id:100},{Term:"Pancreatic cancer",Overlap:"2/76",P_value:.3272,Adjusted_P_value:.6202,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.7327,Combined_Score:1.9356,Genes:"STAT1;RAC2",id:101},{Term:"Shigellosis",Overlap:"5/246",P_value:.3293,Adjusted_P_value:.6202,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.3318,Combined_Score:1.4794,Genes:"PYCARD;CSF2;CCL5;RPS27A;UBA52",id:102},{Term:"Neutrophil extracellular trap formation",Overlap:"4/189",P_value:.3325,Adjusted_P_value:.6202,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.3874,Combined_Score:1.5278,Genes:"RAC2;CYBA;HMGB1;SLC25A5",id:103},{Term:"Oxidative phosphorylation",Overlap:"3/133",P_value:.3365,Adjusted_P_value:.6217,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.4801,Combined_Score:1.6121,Genes:"PPA1;NDUFS5;COX5A",id:104},{Term:"RNA degradation",Overlap:"2/79",P_value:.3441,Adjusted_P_value:.6298,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.665,Combined_Score:1.7762,Genes:"BTG3;BTG2",id:105},{Term:"Apelin signaling pathway",Overlap:"3/137",P_value:.3533,Adjusted_P_value:.6347,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.4356,Combined_Score:1.4936,Genes:"RPS6;PDE3B;PLAT",id:106},{Term:"Autophagy",Overlap:"3/137",P_value:.3533,Adjusted_P_value:.6347,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.4356,Combined_Score:1.4936,Genes:"DDIT4;HMGB1;CFLAR",id:107},{Term:"Peroxisome",Overlap:"2/82",P_value:.3608,Adjusted_P_value:.6422,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.6023,Combined_Score:1.6332,Genes:"SLC27A2;SOD1",id:108},{Term:"Complement and coagulation cascades",Overlap:"2/85",P_value:.3774,Adjusted_P_value:.6619,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.5441,Combined_Score:1.5046,Genes:"F2R;PLAT",id:109},{Term:"Circadian rhythm",Overlap:"1/31",P_value:.3821,Adjusted_P_value:.6619,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.1349,Combined_Score:2.0537,Genes:"BHLHE40",id:110},{Term:"Galactose metabolism",Overlap:"1/31",P_value:.3821,Adjusted_P_value:.6619,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.1349,Combined_Score:2.0537,Genes:"GALM",id:111},{Term:"Base excision repair",Overlap:"1/33",P_value:.401,Adjusted_P_value:.683,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.0012,Combined_Score:1.8285,Genes:"HMGB1",id:112},{Term:"Human papillomavirus infection",Overlap:"6/331",P_value:.4014,Adjusted_P_value:.683,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.1839,Combined_Score:1.0808,Genes:"STAT1;IRF1;HLA-B;ISG15;HLA-A;OASL",id:113},{Term:"Morphine addiction",Overlap:"2/91",P_value:.41,Adjusted_P_value:.6862,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.4396,Combined_Score:1.2834,Genes:"PDE3B;PDE4B",id:114},{Term:"Propanoate metabolism",Overlap:"1/34",P_value:.4103,Adjusted_P_value:.6862,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.9405,Combined_Score:1.7288,Genes:"LDHA",id:115},{Term:"Nicotinate and nicotinamide metabolism",Overlap:"1/35",P_value:.4194,Adjusted_P_value:.6954,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.8833,Combined_Score:1.6366,Genes:"CD38",id:116},{Term:"Starch and sucrose metabolism",Overlap:"1/36",P_value:.4283,Adjusted_P_value:.7042,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.8294,Combined_Score:1.5511,Genes:"GYG1",id:117},{Term:"African trypanosomiasis",Overlap:"1/37",P_value:.4372,Adjusted_P_value:.7127,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.7785,Combined_Score:1.4716,Genes:"IFNG",id:118},{Term:"Choline metabolism in cancer",Overlap:"2/98",P_value:.447,Adjusted_P_value:.7226,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.3342,Combined_Score:1.0743,Genes:"RAC2;FOS",id:119},{Term:"Various types of N-glycan biosynthesis",Overlap:"1/39",P_value:.4544,Adjusted_P_value:.7272,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.6847,Combined_Score:1.3289,Genes:"MGAT4A",id:120},{Term:"AGE-RAGE signaling pathway in diabetic complications",Overlap:"2/100",P_value:.4573,Adjusted_P_value:.7272,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.3068,Combined_Score:1.0225,Genes:"STAT1;PIM1",id:121},{Term:"Ferroptosis",Overlap:"1/41",P_value:.4711,Adjusted_P_value:.743,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.6003,Combined_Score:1.2045,Genes:"SAT1",id:122},{Term:"Protein processing in endoplasmic reticulum",Overlap:"3/171",P_value:.4918,Adjusted_P_value:.7594,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.1431,Combined_Score:.8112,Genes:"DNAJC1;SELENOS;HERPUD1",id:123},{Term:"Glucagon signaling pathway",Overlap:"2/107",P_value:.4925,Adjusted_P_value:.7594,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.2192,Combined_Score:.8635,Genes:"LDHA;PDE3B",id:124},{Term:"Insulin resistance",Overlap:"2/108",P_value:.4974,Adjusted_P_value:.7594,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.2077,Combined_Score:.8433,Genes:"TBC1D4;SLC27A2",id:125},{Term:"ABC transporters",Overlap:"1/45",P_value:.503,Adjusted_P_value:.7594,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.4545,Combined_Score:.9995,Genes:"TAP1",id:126},{Term:"Type II diabetes mellitus",Overlap:"1/46",P_value:.5107,Adjusted_P_value:.7594,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.4221,Combined_Score:.9557,Genes:"SOCS1",id:127},{Term:"Pyruvate metabolism",Overlap:"1/47",P_value:.5182,Adjusted_P_value:.7594,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.3912,Combined_Score:.9145,Genes:"LDHA",id:128},{Term:"Cholinergic synapse",Overlap:"2/113",P_value:.5215,Adjusted_P_value:.7594,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.153,Combined_Score:.7506,Genes:"FYN;FOS",id:129},{Term:"Leukocyte transendothelial migration",Overlap:"2/114",P_value:.5262,Adjusted_P_value:.7594,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.1426,Combined_Score:.7335,Genes:"RAC2;CYBA",id:130},{Term:"Cocaine addiction",Overlap:"1/49",P_value:.533,Adjusted_P_value:.7594,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.3331,Combined_Score:.8388,Genes:"FOSB",id:131},{Term:"Sphingolipid metabolism",Overlap:"1/49",P_value:.533,Adjusted_P_value:.7594,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.3331,Combined_Score:.8388,Genes:"PSAP",id:132},{Term:"Axon guidance",Overlap:"3/182",P_value:.5336,Adjusted_P_value:.7594,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.0722,Combined_Score:.6736,Genes:"RAC2;NFATC2;FYN",id:133},{Term:"Arginine and proline metabolism",Overlap:"1/50",P_value:.5402,Adjusted_P_value:.7594,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.3058,Combined_Score:.8041,Genes:"SAT1",id:134},{Term:"Cysteine and methionine metabolism",Overlap:"1/50",P_value:.5402,Adjusted_P_value:.7594,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.3058,Combined_Score:.8041,Genes:"LDHA",id:135},{Term:"Malaria",Overlap:"1/50",P_value:.5402,Adjusted_P_value:.7594,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.3058,Combined_Score:.8041,Genes:"IFNG",id:136},{Term:"N-Glycan biosynthesis",Overlap:"1/50",P_value:.5402,Adjusted_P_value:.7594,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.3058,Combined_Score:.8041,Genes:"MGAT4A",id:137},{Term:"RNA transport",Overlap:"3/186",P_value:.5483,Adjusted_P_value:.7613,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.0486,Combined_Score:.6302,Genes:"EEF1A1;SUMO2;EIF4B",id:138},{Term:"Sphingolipid signaling pathway",Overlap:"2/119",P_value:.5494,Adjusted_P_value:.7613,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.0935,Combined_Score:.6549,Genes:"RAC2;FYN",id:139},{Term:"Thyroid hormone signaling pathway",Overlap:"2/121",P_value:.5584,Adjusted_P_value:.7684,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.075,Combined_Score:.6263,Genes:"TBC1D4;STAT1",id:140},{Term:"Platelet activation",Overlap:"2/124",P_value:.5718,Adjusted_P_value:.7795,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.0484,Combined_Score:.5861,Genes:"F2R;FYN",id:141},{Term:"Regulation of lipolysis in adipocytes",Overlap:"1/55",P_value:.5746,Adjusted_P_value:.7795,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.1846,Combined_Score:.6563,Genes:"PDE3B",id:142},{Term:"Glutathione metabolism",Overlap:"1/57",P_value:.5876,Adjusted_P_value:.783,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.1422,Combined_Score:.6072,Genes:"MGST3",id:143},{Term:"Lysosome",Overlap:"2/128",P_value:.5891,Adjusted_P_value:.783,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.0149,Combined_Score:.5371,Genes:"PSAP;ACP5",id:144},{Term:"Purine metabolism",Overlap:"2/129",P_value:.5933,Adjusted_P_value:.783,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.0069,Combined_Score:.5257,Genes:"PDE3B;PDE4B",id:145},{Term:"Relaxin signaling pathway",Overlap:"2/129",P_value:.5933,Adjusted_P_value:.783,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.0069,Combined_Score:.5257,Genes:"GNA15;FOS",id:146},{Term:"Mineral absorption",Overlap:"1/60",P_value:.6065,Adjusted_P_value:.795,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.0839,Combined_Score:.5421,Genes:"MT2A",id:147},{Term:"Glycerolipid metabolism",Overlap:"1/61",P_value:.6125,Adjusted_P_value:.7975,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.0658,Combined_Score:.5224,Genes:"GK",id:148},{Term:"Estrogen signaling pathway",Overlap:"2/137",P_value:.6261,Adjusted_P_value:.8097,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.9468,Combined_Score:.4434,Genes:"FOS;FKBP5",id:149},{Term:"Renal cell carcinoma",Overlap:"1/69",P_value:.6579,Adjusted_P_value:.8355,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.94,Combined_Score:.3936,Genes:"PRCC",id:150},{Term:"Renin secretion",Overlap:"1/69",P_value:.6579,Adjusted_P_value:.8355,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.94,Combined_Score:.3936,Genes:"PDE3B",id:151},{Term:"Central carbon metabolism in cancer",Overlap:"1/70",P_value:.6632,Adjusted_P_value:.8355,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.9264,Combined_Score:.3804,Genes:"LDHA",id:152},{Term:"Epithelial cell signaling in Helicobacter pylori infection",Overlap:"1/70",P_value:.6632,Adjusted_P_value:.8355,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.9264,Combined_Score:.3804,Genes:"CCL5",id:153},{Term:"Phospholipase D signaling pathway",Overlap:"2/148",P_value:.6677,Adjusted_P_value:.8357,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.875,Combined_Score:.3534,Genes:"F2R;FYN",id:154},{Term:"p53 signaling pathway",Overlap:"1/73",P_value:.6786,Adjusted_P_value:.8439,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.8876,Combined_Score:.3442,Genes:"PMAIP1",id:155},{Term:"Metabolism of xenobiotics by cytochrome P450",Overlap:"1/76",P_value:.6933,Adjusted_P_value:.8495,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.852,Combined_Score:.3121,Genes:"MGST3",id:156},{Term:"Ras signaling pathway",Overlap:"3/232",P_value:.6962,Adjusted_P_value:.8495,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.836,Combined_Score:.3027,Genes:"RASA3;RAC2;RASAL3",id:157},{Term:"Thermogenesis",Overlap:"3/232",P_value:.6962,Adjusted_P_value:.8495,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.836,Combined_Score:.3027,Genes:"NDUFS5;RPS6;COX5A",id:158},{Term:"Wnt signaling pathway",Overlap:"2/166",P_value:.7278,Adjusted_P_value:.8824,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.7783,Combined_Score:.2473,Genes:"RAC2;NFATC2",id:159},{Term:"ErbB signaling pathway",Overlap:"1/85",P_value:.7334,Adjusted_P_value:.8837,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.7604,Combined_Score:.2358,Genes:"CBLB",id:160},{Term:"Cardiac muscle contraction",Overlap:"1/87",P_value:.7416,Adjusted_P_value:.8874,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.7426,Combined_Score:.222,Genes:"COX5A",id:161},{Term:"Gap junction",Overlap:"1/88",P_value:.7456,Adjusted_P_value:.8874,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.734,Combined_Score:.2155,Genes:"TUBB",id:162},{Term:"Salivary secretion",Overlap:"1/93",P_value:.7647,Adjusted_P_value:.8972,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.694,Combined_Score:.1862,Genes:"CD38",id:163},{Term:"TGF-beta signaling pathway",Overlap:"1/94",P_value:.7683,Adjusted_P_value:.8972,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.6865,Combined_Score:.1809,Genes:"IFNG",id:164},{Term:"Circadian entrainment",Overlap:"1/97",P_value:.7789,Adjusted_P_value:.8972,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.6649,Combined_Score:.1662,Genes:"FOS",id:165},{Term:"Fc gamma R-mediated phagocytosis",Overlap:"1/97",P_value:.7789,Adjusted_P_value:.8972,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.6649,Combined_Score:.1662,Genes:"RAC2",id:166},{Term:"Prostate cancer",Overlap:"1/97",P_value:.7789,Adjusted_P_value:.8972,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.6649,Combined_Score:.1662,Genes:"PLAT",id:167},{Term:"Aldosterone synthesis and secretion",Overlap:"1/98",P_value:.7823,Adjusted_P_value:.8972,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.658,Combined_Score:.1615,Genes:"NR4A2",id:168},{Term:"Progesterone-mediated oocyte maturation",Overlap:"1/100",P_value:.789,Adjusted_P_value:.8972,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.6447,Combined_Score:.1528,Genes:"PDE3B",id:169},{Term:"Longevity regulating pathway",Overlap:"1/102",P_value:.7955,Adjusted_P_value:.8972,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.6318,Combined_Score:.1446,Genes:"SOD1",id:170},{Term:"Pancreatic secretion",Overlap:"1/102",P_value:.7955,Adjusted_P_value:.8972,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.6318,Combined_Score:.1446,Genes:"CD38",id:171},{Term:"Drug metabolism",Overlap:"1/108",P_value:.8138,Adjusted_P_value:.9124,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.5962,Combined_Score:.1229,Genes:"MGST3",id:172},{Term:"Focal adhesion",Overlap:"2/201",P_value:.8184,Adjusted_P_value:.9124,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.6402,Combined_Score:.1283,Genes:"RAC2;FYN",id:173},{Term:"Proteoglycans in cancer",Overlap:"2/205",P_value:.8268,Adjusted_P_value:.9124,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.6275,Combined_Score:.1193,Genes:"RPS6;EIF4B",id:174},{Term:"Serotonergic synapse",Overlap:"1/113",P_value:.8277,Adjusted_P_value:.9124,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.5695,Combined_Score:.1077,Genes:"DUSP1",id:175},{Term:"Rap1 signaling pathway",Overlap:"2/210",P_value:.8369,Adjusted_P_value:.9172,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.6122,Combined_Score:.109,Genes:"F2R;RAC2",id:176},{Term:"Regulation of actin cytoskeleton",Overlap:"2/218",P_value:.8519,Adjusted_P_value:.9265,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.5893,Combined_Score:.0945,Genes:"F2R;RAC2",id:177},{Term:"Cell cycle",Overlap:"1/124",P_value:.8549,Adjusted_P_value:.9265,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.5182,Combined_Score:.0812,Genes:"YWHAB",id:178},{Term:"Oocyte meiosis",Overlap:"1/129",P_value:.8658,Adjusted_P_value:.9265,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.4979,Combined_Score:.0717,Genes:"YWHAB",id:179},{Term:"FoxO signaling pathway",Overlap:"1/131",P_value:.8699,Adjusted_P_value:.9265,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.4902,Combined_Score:.0683,Genes:"BCL2L11",id:180},{Term:"Dopaminergic synapse",Overlap:"1/132",P_value:.872,Adjusted_P_value:.9265,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.4864,Combined_Score:.0666,Genes:"FOS",id:181},{Term:"Vascular smooth muscle contraction",Overlap:"1/133",P_value:.8739,Adjusted_P_value:.9265,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.4827,Combined_Score:.065,Genes:"MYL6",id:182},{Term:"Chemical carcinogenesis",Overlap:"2/239",P_value:.8854,Adjusted_P_value:.9322,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.5365,Combined_Score:.0653,Genes:"MGST3;FOS",id:183},{Term:"Breast cancer",Overlap:"1/147",P_value:.8987,Adjusted_P_value:.9322,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.4361,Combined_Score:.0466,Genes:"FOS",id:184},{Term:"Retrograde endocannabinoid signaling",Overlap:"1/148",P_value:.9003,Adjusted_P_value:.9322,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.4331,Combined_Score:.0455,Genes:"NDUFS5",id:185},{Term:"Adrenergic signaling in cardiomyocytes",Overlap:"1/150",P_value:.9034,Adjusted_P_value:.9322,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.4272,Combined_Score:.0434,Genes:"CREM",id:186},{Term:"Spliceosome",Overlap:"1/150",P_value:.9034,Adjusted_P_value:.9322,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.4272,Combined_Score:.0434,Genes:"SRSF7",id:187},{Term:"Hippo signaling pathway",Overlap:"1/163",P_value:.9211,Adjusted_P_value:.9428,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.3927,Combined_Score:.0323,Genes:"YWHAB",id:188},{Term:"Hepatocellular carcinoma",Overlap:"1/168",P_value:.9271,Adjusted_P_value:.9428,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.3808,Combined_Score:.0288,Genes:"MGST3",id:189},{Term:"Tight junction",Overlap:"1/169",P_value:.9282,Adjusted_P_value:.9428,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.3785,Combined_Score:.0282,Genes:"MYL6",id:190},{Term:"Alcoholism",Overlap:"1/186",P_value:.945,Adjusted_P_value:.9548,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.3435,Combined_Score:.0194,Genes:"FOSB",id:191},{Term:"Neuroactive ligand-receptor interaction",Overlap:"2/341",P_value:.9691,Adjusted_P_value:.9741,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.3731,Combined_Score:.0117,Genes:"GZMA;F2R",id:192},{Term:"Olfactory transduction",Overlap:"1/440",P_value:.999,Adjusted_P_value:.999,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.1429,Combined_Score:1e-4,Genes:"RGS2",id:193}]}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p:T,i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function _n(e){let a,t,n,_;return a=new ve({props:{$$slots:{default:[tn]},$$scope:{ctx:e}}}),n=new ve({props:{$$slots:{default:[nn]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment)},m(e,l){P(a,e,l),s(e,t,l),P(n,e,l),_=!0},p(e,t){const _={};1&t&&(_.$$scope={dirty:t,ctx:e}),a.$set(_);const l={};1&t&&(l.$$scope={dirty:t,ctx:e}),n.$set(l)},i(e){_||(i(a.$$.fragment,e),i(n.$$.fragment,e),_=!0)},o(e){p(a.$$.fragment,e),p(n.$$.fragment,e),_=!1},d(e){e&&c(t),A(a,e),A(n,e)}}}function ln(e){let a,t;return a=new ce({props:{$$slots:{content:[_n],default:[an]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};1&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function dn(e){let a,t;return a=new je({props:{title:"KEGG_2021_Human",open:!0,$$slots:{default:[ln]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};1&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function sn(e){let a,t,n,_,l,d;return a=new ue({props:{label:"Significant Markers"}}),n=new ue({props:{label:"Volcano Plot"}}),l=new ue({props:{label:"Dot Plot"}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment),_=f(),g(l.$$.fragment)},m(e,i){P(a,e,i),s(e,t,i),P(n,e,i),s(e,_,i),P(l,e,i),d=!0},p:T,i(e){d||(i(a.$$.fragment,e),i(n.$$.fragment,e),i(l.$$.fragment,e),d=!0)},o(e){p(a.$$.fragment,e),p(n.$$.fragment,e),p(l.$$.fragment,e),d=!1},d(e){e&&(c(t),c(_)),A(a,e),A(n,e),A(l,e)}}}function rn(e){let a;return{c(){a=S("The markers are found using Seurat's FindMarkers function, and filtered by: p_val_adj < 0.05")},m(e,t){s(e,a,t)},d(e){e&&c(a)}}}function on(e){let a,t,n,_;return a=new Yt({props:{$$slots:{default:[rn]},$$scope:{ctx:e}}}),n=new kt({props:{src:"../MarkersFinder/samples.markers/c2/markers.txt",data:[{gene:"HLA-B",p_val:8455e-34,avg_log2FC:.7613,pct_1:1,pct_2:1,p_val_adj:2836e-29,id:0},{gene:"CTSW",p_val:1539e-25,avg_log2FC:-1.7545,pct_1:.456,pct_2:.889,p_val_adj:516e-20,id:1},{gene:"SRGN",p_val:3.571e-20,avg_log2FC:.8186,pct_1:1,pct_2:.981,p_val_adj:0,id:2},{gene:"FOS",p_val:1026e-22,avg_log2FC:-1.4244,pct_1:.796,pct_2:.974,p_val_adj:0,id:3},{gene:"FKBP5",p_val:525e-20,avg_log2FC:2.5237,pct_1:.485,pct_2:.151,p_val_adj:0,id:4},{gene:"S100A4",p_val:1016e-20,avg_log2FC:-.996,pct_1:.913,pct_2:.986,p_val_adj:0,id:5},{gene:"VIM",p_val:5323e-20,avg_log2FC:-.8107,pct_1:.932,pct_2:.998,p_val_adj:0,id:6},{gene:"HLA-DRB1",p_val:2976e-19,avg_log2FC:1.9955,pct_1:.612,pct_2:.257,p_val_adj:0,id:7},{gene:"CTLA4",p_val:9031e-19,avg_log2FC:3.9532,pct_1:.165,pct_2:.01,p_val_adj:0,id:8},{gene:"HLA-DRA",p_val:9589e-19,avg_log2FC:3.275,pct_1:.272,pct_2:.045,p_val_adj:0,id:9},{gene:"ISG20",p_val:0,avg_log2FC:1.2375,pct_1:.864,pct_2:.589,p_val_adj:0,id:10},{gene:"MT-ND5",p_val:0,avg_log2FC:-.57,pct_1:1,pct_2:.995,p_val_adj:0,id:11},{gene:"GZMB",p_val:0,avg_log2FC:2.5014,pct_1:.621,pct_2:.308,p_val_adj:0,id:12},{gene:"CAPG",p_val:0,avg_log2FC:-1.5487,pct_1:.34,pct_2:.728,p_val_adj:0,id:13},{gene:"TXNIP",p_val:0,avg_log2FC:-1.0542,pct_1:.689,pct_2:.892,p_val_adj:0,id:14},{gene:"CD7",p_val:0,avg_log2FC:.5956,pct_1:1,pct_2:.988,p_val_adj:0,id:15},{gene:"HAPLN3",p_val:0,avg_log2FC:3.1094,pct_1:.175,pct_2:.021,p_val_adj:0,id:16},{gene:"CCL4L2",p_val:0,avg_log2FC:-2.1244,pct_1:.243,pct_2:.656,p_val_adj:0,id:17},{gene:"HLA-A",p_val:0,avg_log2FC:.5357,pct_1:1,pct_2:.997,p_val_adj:0,id:18},{gene:"RPS14",p_val:0,avg_log2FC:-.3977,pct_1:1,pct_2:1,p_val_adj:0,id:19},{gene:"LGALS1",p_val:0,avg_log2FC:-1.9842,pct_1:.291,pct_2:.632,p_val_adj:0,id:20},{gene:"AL157402.2",p_val:0,avg_log2FC:2.7847,pct_1:.262,pct_2:.06,p_val_adj:0,id:21},{gene:"IGKV3D-20",p_val:0,avg_log2FC:7.471,pct_1:.078,pct_2:0,p_val_adj:0,id:22},{gene:"EEF1A1",p_val:0,avg_log2FC:-.2836,pct_1:1,pct_2:1,p_val_adj:0,id:23},{gene:"TOX2",p_val:0,avg_log2FC:3.3578,pct_1:.117,pct_2:.009,p_val_adj:0,id:24},{gene:"MIR181A1HG",p_val:0,avg_log2FC:2.9498,pct_1:.136,pct_2:.014,p_val_adj:0,id:25},{gene:"SUN2",p_val:0,avg_log2FC:-.9405,pct_1:.573,pct_2:.849,p_val_adj:0,id:26},{gene:"B2M",p_val:0,avg_log2FC:.2534,pct_1:1,pct_2:1,p_val_adj:0,id:27},{gene:"STAT1",p_val:0,avg_log2FC:1.4734,pct_1:.553,pct_2:.296,p_val_adj:0,id:28},{gene:"GNLY",p_val:0,avg_log2FC:2.1482,pct_1:.252,pct_2:.06,p_val_adj:0,id:29},{gene:"CD74",p_val:0,avg_log2FC:1.2909,pct_1:.942,pct_2:.865,p_val_adj:0,id:30},{gene:"DUSP2",p_val:0,avg_log2FC:.9801,pct_1:.903,pct_2:.661,p_val_adj:0,id:31},{gene:"SPINK2",p_val:0,avg_log2FC:-2.373,pct_1:.068,pct_2:.396,p_val_adj:0,id:32},{gene:"AC020916.1",p_val:0,avg_log2FC:-1.2786,pct_1:.34,pct_2:.663,p_val_adj:0,id:33},{gene:"DUSP1",p_val:0,avg_log2FC:-.7356,pct_1:.874,pct_2:.961,p_val_adj:0,id:34},{gene:"BTG1",p_val:0,avg_log2FC:.8834,pct_1:1,pct_2:.988,p_val_adj:0,id:35},{gene:"CREM",p_val:0,avg_log2FC:1.4153,pct_1:.699,pct_2:.498,p_val_adj:0,id:36},{gene:"RGS1",p_val:0,avg_log2FC:1.1722,pct_1:.854,pct_2:.661,p_val_adj:1e-4,id:37},{gene:"TUBB",p_val:0,avg_log2FC:-1.0049,pct_1:.602,pct_2:.791,p_val_adj:1e-4,id:38},{gene:"PSMB9",p_val:0,avg_log2FC:.6847,pct_1:.922,pct_2:.846,p_val_adj:1e-4,id:39},{gene:"MT-CO3",p_val:0,avg_log2FC:.2861,pct_1:1,pct_2:.998,p_val_adj:1e-4,id:40},{gene:"RPS26",p_val:0,avg_log2FC:.5713,pct_1:1,pct_2:.95,p_val_adj:1e-4,id:41},{gene:"ITM2C",p_val:0,avg_log2FC:-.982,pct_1:.515,pct_2:.781,p_val_adj:1e-4,id:42},{gene:"MYL12A",p_val:0,avg_log2FC:-.4721,pct_1:.971,pct_2:.997,p_val_adj:1e-4,id:43},{gene:"RARRES3",p_val:0,avg_log2FC:.9709,pct_1:.913,pct_2:.921,p_val_adj:1e-4,id:44},{gene:"LRRN3",p_val:0,avg_log2FC:-2.4191,pct_1:.068,pct_2:.368,p_val_adj:1e-4,id:45},{gene:"TRAV19",p_val:0,avg_log2FC:1.6228,pct_1:.311,pct_2:.099,p_val_adj:1e-4,id:46},{gene:"P2RX5",p_val:0,avg_log2FC:3.0211,pct_1:.126,pct_2:.015,p_val_adj:1e-4,id:47},{gene:"FOSB",p_val:0,avg_log2FC:-1.0327,pct_1:.621,pct_2:.851,p_val_adj:2e-4,id:48},{gene:"CERS6",p_val:0,avg_log2FC:5.6885,pct_1:.058,pct_2:0,p_val_adj:2e-4,id:49},{gene:"IGHV3-21",p_val:0,avg_log2FC:6.5077,pct_1:.058,pct_2:0,p_val_adj:2e-4,id:50},{gene:"HLA-DQA1",p_val:0,avg_log2FC:3.864,pct_1:.107,pct_2:.01,p_val_adj:2e-4,id:51},{gene:"GEM",p_val:0,avg_log2FC:2.192,pct_1:.107,pct_2:.01,p_val_adj:2e-4,id:52},{gene:"EMP3",p_val:0,avg_log2FC:-.9439,pct_1:.515,pct_2:.798,p_val_adj:3e-4,id:53},{gene:"UBE2L6",p_val:0,avg_log2FC:.9367,pct_1:.767,pct_2:.502,p_val_adj:4e-4,id:54},{gene:"RPS12",p_val:0,avg_log2FC:-.2948,pct_1:1,pct_2:1,p_val_adj:4e-4,id:55},{gene:"MT-CO2",p_val:0,avg_log2FC:.2536,pct_1:1,pct_2:1,p_val_adj:8e-4,id:56},{gene:"RPS27",p_val:0,avg_log2FC:.3352,pct_1:1,pct_2:1,p_val_adj:8e-4,id:57},{gene:"ATP1A1",p_val:0,avg_log2FC:-1.1524,pct_1:.408,pct_2:.673,p_val_adj:9e-4,id:58},{gene:"ACTB",p_val:0,avg_log2FC:-.4295,pct_1:1,pct_2:1,p_val_adj:.0011,id:59},{gene:"GBP2",p_val:0,avg_log2FC:1.3417,pct_1:.544,pct_2:.318,p_val_adj:.0014,id:60},{gene:"LAYN",p_val:0,avg_log2FC:2.7129,pct_1:.146,pct_2:.027,p_val_adj:.0026,id:61},{gene:"LY6E",p_val:0,avg_log2FC:.7291,pct_1:.893,pct_2:.781,p_val_adj:.0026,id:62},{gene:"ATF3",p_val:0,avg_log2FC:1.5955,pct_1:.388,pct_2:.166,p_val_adj:.0027,id:63},{gene:"NFKBIZ",p_val:0,avg_log2FC:1.6469,pct_1:.592,pct_2:.404,p_val_adj:.0027,id:64},{gene:"CLIP3",p_val:0,avg_log2FC:5.4301,pct_1:.049,pct_2:0,p_val_adj:.0031,id:65},{gene:"AREG",p_val:0,avg_log2FC:2.1049,pct_1:.408,pct_2:.19,p_val_adj:.0037,id:66},{gene:"NOP53",p_val:0,avg_log2FC:.5708,pct_1:.913,pct_2:.86,p_val_adj:.0039,id:67},{gene:"CISH",p_val:0,avg_log2FC:-1.712,pct_1:.194,pct_2:.459,p_val_adj:.0044,id:68},{gene:"HLA-DQB1",p_val:0,avg_log2FC:1.7561,pct_1:.223,pct_2:.065,p_val_adj:.0054,id:69},{gene:"CST7",p_val:0,avg_log2FC:.7462,pct_1:.903,pct_2:.868,p_val_adj:.0057,id:70},{gene:"SOCS3",p_val:0,avg_log2FC:1.5698,pct_1:.388,pct_2:.176,p_val_adj:.0059,id:71},{gene:"GBP5",p_val:0,avg_log2FC:1.5848,pct_1:.515,pct_2:.33,p_val_adj:.0068,id:72},{gene:"PKM",p_val:0,avg_log2FC:-.6538,pct_1:.553,pct_2:.795,p_val_adj:.0081,id:73},{gene:"FAM129A",p_val:0,avg_log2FC:1.2721,pct_1:.456,pct_2:.245,p_val_adj:.0086,id:74},{gene:"LTB",p_val:0,avg_log2FC:-.8873,pct_1:.447,pct_2:.711,p_val_adj:.0089,id:75},{gene:"TUBA1B",p_val:0,avg_log2FC:-.9338,pct_1:.456,pct_2:.711,p_val_adj:.0092,id:76},{gene:"CHI3L2",p_val:0,avg_log2FC:1.8326,pct_1:.223,pct_2:.068,p_val_adj:.0096,id:77},{gene:"LCP1",p_val:0,avg_log2FC:-.6093,pct_1:.816,pct_2:.935,p_val_adj:.0147,id:78},{gene:"RPS27A",p_val:0,avg_log2FC:-.2098,pct_1:1,pct_2:1,p_val_adj:.0147,id:79},{gene:"AC020571.1",p_val:0,avg_log2FC:-1.9908,pct_1:.184,pct_2:.44,p_val_adj:.0161,id:80},{gene:"RBL2",p_val:0,avg_log2FC:-.9146,pct_1:.544,pct_2:.735,p_val_adj:.0179,id:81},{gene:"RPL5",p_val:0,avg_log2FC:-.329,pct_1:1,pct_2:1,p_val_adj:.0185,id:82},{gene:"CD160",p_val:0,avg_log2FC:-1.0194,pct_1:.35,pct_2:.634,p_val_adj:.0209,id:83},{gene:"CBLB",p_val:0,avg_log2FC:.9969,pct_1:.738,pct_2:.563,p_val_adj:.0217,id:84},{gene:"ZC3H12D",p_val:0,avg_log2FC:1.8953,pct_1:.184,pct_2:.05,p_val_adj:.0219,id:85},{gene:"KLRC1",p_val:0,avg_log2FC:1.6493,pct_1:.262,pct_2:.092,p_val_adj:.0241,id:86},{gene:"SARAF",p_val:0,avg_log2FC:.326,pct_1:1,pct_2:.993,p_val_adj:.028,id:87},{gene:"RPS3A",p_val:0,avg_log2FC:-.2475,pct_1:1,pct_2:1,p_val_adj:.0317,id:88},{gene:"PDE3B",p_val:0,avg_log2FC:1.4413,pct_1:.427,pct_2:.238,p_val_adj:.0319,id:89},{gene:"PSMB8",p_val:0,avg_log2FC:.6523,pct_1:.816,pct_2:.707,p_val_adj:.0324,id:90},{gene:"FIBP",p_val:0,avg_log2FC:1.1163,pct_1:.476,pct_2:.265,p_val_adj:.0425,id:91},{gene:"HIST1H2AC",p_val:0,avg_log2FC:1.7952,pct_1:.204,pct_2:.062,p_val_adj:.0438,id:92},{gene:"RPS5",p_val:0,avg_log2FC:-.2673,pct_1:1,pct_2:1,p_val_adj:.0462,id:93}]}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment)},m(e,l){P(a,e,l),s(e,t,l),P(n,e,l),_=!0},p(e,t){const n={};1&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){_||(i(a.$$.fragment,e),i(n.$$.fragment,e),_=!0)},o(e){p(a.$$.fragment,e),p(n.$$.fragment,e),_=!1},d(e){e&&c(t),A(a,e),A(n,e)}}}function pn(e){let a,t;return a=new At({props:{src:"../MarkersFinder/samples.markers/c2/volcano.png",width:900,height:1200}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p:T,i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function cn(e){let a,t;return a=new At({props:{src:"../MarkersFinder/samples.markers/c2/dotplot.png",width:600,height:1200}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p:T,i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function un(e){let a,t,n,_,l,d;return a=new ve({props:{$$slots:{default:[on]},$$scope:{ctx:e}}}),n=new ve({props:{$$slots:{default:[pn]},$$scope:{ctx:e}}}),l=new ve({props:{$$slots:{default:[cn]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment),_=f(),g(l.$$.fragment)},m(e,i){P(a,e,i),s(e,t,i),P(n,e,i),s(e,_,i),P(l,e,i),d=!0},p(e,t){const _={};1&t&&(_.$$scope={dirty:t,ctx:e}),a.$set(_);const d={};1&t&&(d.$$scope={dirty:t,ctx:e}),n.$set(d);const s={};1&t&&(s.$$scope={dirty:t,ctx:e}),l.$set(s)},i(e){d||(i(a.$$.fragment,e),i(n.$$.fragment,e),i(l.$$.fragment,e),d=!0)},o(e){p(a.$$.fragment,e),p(n.$$.fragment,e),p(l.$$.fragment,e),d=!1},d(e){e&&(c(t),c(_)),A(a,e),A(n,e),A(l,e)}}}function vn(e){let a;return{c(){a=S("The enrichment analysis is done using Enrichr. The significant markers are used as input.")},m(e,t){s(e,a,t)},d(e){e&&c(a)}}}function mn(e){let a,t,n,_;return a=new ue({props:{label:"Plot"}}),n=new ue({props:{label:"Table"}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment)},m(e,l){P(a,e,l),s(e,t,l),P(n,e,l),_=!0},p:T,i(e){_||(i(a.$$.fragment,e),i(n.$$.fragment,e),_=!0)},o(e){p(a.$$.fragment,e),p(n.$$.fragment,e),_=!1},d(e){e&&c(t),A(a,e),A(n,e)}}}function gn(e){let a,t;return a=new At({props:{src:"../MarkersFinder/samples.markers/c2/Enrichr-KEGG_2021_Human.png",width:1e3,height:1e3}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p:T,i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function Pn(e){let a,t;return a=new kt({props:{src:"../MarkersFinder/samples.markers/c2/Enrichr-KEGG_2021_Human.txt",data:[{Term:"Graft-versus-host disease",Overlap:"8/42",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:54.3694,Combined_Score:1344.5737,Genes:"HLA-B;GZMB;HLA-DRA;KLRC1;HLA-A;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:0},{Term:"Antigen processing and presentation",Overlap:"9/78",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:30.4404,Combined_Score:699.2342,Genes:"CD74;HLA-B;HLA-DRA;HLA-A;KLRC1;B2M;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:1},{Term:"Autoimmune thyroid disease",Overlap:"8/53",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:41.0563,Combined_Score:934.2906,Genes:"HLA-B;CTLA4;GZMB;HLA-DRA;HLA-A;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:2},{Term:"Allograft rejection",Overlap:"7/38",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:51.5851,Combined_Score:1109.9398,Genes:"HLA-B;GZMB;HLA-DRA;HLA-A;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:3},{Term:"Type I diabetes mellitus",Overlap:"7/43",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:44.4093,Combined_Score:914.757,Genes:"HLA-B;GZMB;HLA-DRA;HLA-A;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:4},{Term:"Viral myocarditis",Overlap:"7/60",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:30.139,Combined_Score:547.9095,Genes:"HLA-B;HLA-DRA;HLA-A;ACTB;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:5},{Term:"Phagosome",Overlap:"9/152",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:14.6332,Combined_Score:248.9394,Genes:"TUBA1B;TUBB;HLA-B;HLA-DRA;HLA-A;ACTB;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:6},{Term:"Leishmaniasis",Overlap:"7/77",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:22.8,Combined_Score:374.2691,Genes:"EEF1A1;STAT1;HLA-DRA;FOS;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:7},{Term:"Coronavirus disease",Overlap:"10/232",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:10.5556,Combined_Score:166.5673,Genes:"RPS26;RPL5;RPS14;RPS27;STAT1;RPS5;RPS3A;FOS;RPS27A;RPS12",id:8},{Term:"Rheumatoid arthritis",Overlap:"7/93",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:18.5432,Combined_Score:280.0983,Genes:"CTLA4;HLA-DRA;FOS;LTB;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:9},{Term:"Epstein-Barr virus infection",Overlap:"9/202",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:10.8148,Combined_Score:157.7191,Genes:"STAT1;HLA-B;HLA-DRA;HLA-A;VIM;B2M;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:10},{Term:"Ribosome",Overlap:"8/158",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:12.2518,Combined_Score:172.0732,Genes:"RPS26;RPL5;RPS14;RPS27;RPS5;RPS3A;RPS27A;RPS12",id:11},{Term:"Th1 and Th2 cell differentiation",Overlap:"6/92",P_value:0,Adjusted_P_value:1e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:15.7135,Combined_Score:192.6905,Genes:"STAT1;HLA-DRA;FOS;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:12},{Term:"Cell adhesion molecules",Overlap:"7/148",P_value:0,Adjusted_P_value:1e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:11.2786,Combined_Score:135.0615,Genes:"HLA-B;CTLA4;HLA-DRA;HLA-A;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:13},{Term:"Human T-cell leukemia virus 1 infection",Overlap:"8/219",P_value:0,Adjusted_P_value:1e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:8.6829,Combined_Score:100.857,Genes:"HLA-B;HLA-DRA;FOS;HLA-A;B2M;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:14},{Term:"Th17 cell differentiation",Overlap:"6/107",P_value:0,Adjusted_P_value:1e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:13.3697,Combined_Score:152.2705,Genes:"STAT1;HLA-DRA;FOS;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:15},{Term:"Asthma",Overlap:"4/31",P_value:0,Adjusted_P_value:1e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:32.7226,Combined_Score:367.992,Genes:"HLA-DRA;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:16},{Term:"Inflammatory bowel disease",Overlap:"5/65",P_value:0,Adjusted_P_value:1e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:18.5824,Combined_Score:208.2021,Genes:"STAT1;HLA-DRA;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:17},{Term:"Influenza A",Overlap:"7/172",P_value:0,Adjusted_P_value:1e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:9.6264,Combined_Score:105.8319,Genes:"SOCS3;STAT1;HLA-DRA;ACTB;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:18},{Term:"Intestinal immune network for IgA production",Overlap:"4/48",P_value:1e-4,Adjusted_P_value:6e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:20.0626,Combined_Score:190.2918,Genes:"HLA-DRA;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:19},{Term:"Hematopoietic cell lineage",Overlap:"5/99",P_value:1e-4,Adjusted_P_value:8e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:11.8408,Combined_Score:108.5958,Genes:"CD7;HLA-DRA;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:20},{Term:"Herpes simplex virus 1 infection",Overlap:"10/498",P_value:1e-4,Adjusted_P_value:8e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.737,Combined_Score:42.9039,Genes:"SOCS3;CD74;STAT1;HLA-B;HLA-DRA;HLA-A;B2M;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:21},{Term:"Toxoplasmosis",Overlap:"5/112",P_value:2e-4,Adjusted_P_value:.0013,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:10.3954,Combined_Score:89.2997,Genes:"STAT1;HLA-DRA;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:22},{Term:"Tuberculosis",Overlap:"6/180",P_value:2e-4,Adjusted_P_value:.0014,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:7.732,Combined_Score:65.6008,Genes:"CD74;STAT1;HLA-DRA;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:23},{Term:"Prolactin signaling pathway",Overlap:"4/70",P_value:3e-4,Adjusted_P_value:.0021,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:13.3603,Combined_Score:107.0639,Genes:"SOCS3;CISH;STAT1;FOS",id:24},{Term:"Apoptosis",Overlap:"5/142",P_value:6e-4,Adjusted_P_value:.0034,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:8.1067,Combined_Score:60.7589,Genes:"TUBA1B;GZMB;CTSW;FOS;ACTB",id:25},{Term:"Staphylococcus aureus infection",Overlap:"4/95",P_value:.0011,Adjusted_P_value:.0061,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:9.6777,Combined_Score:66.3714,Genes:"HLA-DRA;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:26},{Term:"Kaposi sarcoma-associated herpesvirus infection",Overlap:"5/193",P_value:.0022,Adjusted_P_value:.0123,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:5.8923,Combined_Score:36.1135,Genes:"STAT1;HLA-B;FOS;HLA-A;RPS27A",id:27},{Term:"Osteoclast differentiation",Overlap:"4/127",P_value:.003,Adjusted_P_value:.0166,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:7.1483,Combined_Score:41.4223,Genes:"SOCS3;STAT1;FOSB;FOS",id:28},{Term:"Natural killer cell mediated cytotoxicity",Overlap:"4/131",P_value:.0034,Adjusted_P_value:.0179,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:6.9218,Combined_Score:39.3392,Genes:"HLA-B;GZMB;HLA-A;KLRC1",id:29},{Term:"Systemic lupus erythematosus",Overlap:"4/135",P_value:.0038,Adjusted_P_value:.0193,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:6.7091,Combined_Score:37.4092,Genes:"HLA-DRA;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:30},{Term:"Ubiquitin mediated proteolysis",Overlap:"4/140",P_value:.0043,Adjusted_P_value:.0213,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:6.4608,Combined_Score:35.1891,Genes:"SOCS3;UBE2L6;CBLB;RPS27A",id:31},{Term:"Salmonella infection",Overlap:"5/249",P_value:.0064,Adjusted_P_value:.0298,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.5271,Combined_Score:22.8528,Genes:"TUBA1B;TUBB;FOS;ACTB;MYL12A",id:32},{Term:"Parkinson disease",Overlap:"5/249",P_value:.0064,Adjusted_P_value:.0298,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.5271,Combined_Score:22.8528,Genes:"TUBA1B;DUSP1;TUBB;UBE2L6;RPS27A",id:33},{Term:"NOD-like receptor signaling pathway",Overlap:"4/181",P_value:.0105,Adjusted_P_value:.0475,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.9539,Combined_Score:22.5622,Genes:"GBP5;STAT1;TXNIP;GBP2",id:34},{Term:"T cell receptor signaling pathway",Overlap:"3/104",P_value:.013,Adjusted_P_value:.0555,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:6.4645,Combined_Score:28.0702,Genes:"CTLA4;CBLB;FOS",id:35},{Term:"Toll-like receptor signaling pathway",Overlap:"3/104",P_value:.013,Adjusted_P_value:.0555,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:6.4645,Combined_Score:28.0702,Genes:"STAT1;CCL4L2;FOS",id:36},{Term:"Pathogenic Escherichia coli infection",Overlap:"4/197",P_value:.014,Adjusted_P_value:.0582,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.5396,Combined_Score:19.3805,Genes:"TUBA1B;TUBB;FOS;ACTB",id:37},{Term:"Viral carcinogenesis",Overlap:"4/203",P_value:.0155,Adjusted_P_value:.0626,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.4013,Combined_Score:18.3512,Genes:"RBL2;PKM;HLA-B;HLA-A",id:38},{Term:"Human immunodeficiency virus 1 infection",Overlap:"4/212",P_value:.0178,Adjusted_P_value:.0702,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.209,Combined_Score:16.9466,Genes:"HLA-B;FOS;HLA-A;B2M",id:39},{Term:"Growth hormone synthesis, secretion and action",Overlap:"3/119",P_value:.0186,Adjusted_P_value:.0702,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:5.6243,Combined_Score:22.41,Genes:"SOCS3;STAT1;FOS",id:40},{Term:"Thyroid hormone signaling pathway",Overlap:"3/121",P_value:.0194,Adjusted_P_value:.0702,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:5.5284,Combined_Score:21.7861,Genes:"STAT1;ATP1A1;ACTB",id:41},{Term:"Type II diabetes mellitus",Overlap:"2/46",P_value:.0198,Adjusted_P_value:.0702,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:9.8132,Combined_Score:38.4975,Genes:"SOCS3;PKM",id:42},{Term:"Proteasome",Overlap:"2/46",P_value:.0198,Adjusted_P_value:.0702,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:9.8132,Combined_Score:38.4975,Genes:"PSMB8;PSMB9",id:43},{Term:"Human papillomavirus infection",Overlap:"5/331",P_value:.02,Adjusted_P_value:.0702,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.3742,Combined_Score:13.2017,Genes:"RBL2;PKM;STAT1;HLA-B;HLA-A",id:44},{Term:"Human cytomegalovirus infection",Overlap:"4/225",P_value:.0217,Adjusted_P_value:.0744,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.9588,Combined_Score:15.171,Genes:"CCL4L2;HLA-B;HLA-A;B2M",id:45},{Term:"Insulin signaling pathway",Overlap:"3/137",P_value:.0268,Adjusted_P_value:.0898,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.8644,Combined_Score:17.602,Genes:"SOCS3;PDE3B;CBLB",id:46},{Term:"Fluid shear stress and atherosclerosis",Overlap:"3/139",P_value:.0278,Adjusted_P_value:.0898,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.7923,Combined_Score:17.1634,Genes:"DUSP1;FOS;ACTB",id:47},{Term:"Measles",Overlap:"3/139",P_value:.0278,Adjusted_P_value:.0898,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.7923,Combined_Score:17.1634,Genes:"STAT1;CBLB;FOS",id:48},{Term:"Cellular senescence",Overlap:"3/156",P_value:.0373,Adjusted_P_value:.1178,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.2562,Combined_Score:13.9999,Genes:"RBL2;HLA-B;HLA-A",id:49},{Term:"JAK-STAT signaling pathway",Overlap:"3/162",P_value:.041,Adjusted_P_value:.1269,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.0943,Combined_Score:13.0823,Genes:"SOCS3;CISH;STAT1",id:50},{Term:"Amphetamine addiction",Overlap:"2/69",P_value:.0418,Adjusted_P_value:.1271,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:6.4371,Combined_Score:20.431,Genes:"FOSB;FOS",id:51},{Term:"Tight junction",Overlap:"3/169",P_value:.0455,Adjusted_P_value:.1356,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.9203,Combined_Score:12.1164,Genes:"TUBA1B;ACTB;MYL12A",id:52},{Term:"Gastric acid secretion",Overlap:"2/76",P_value:.0498,Adjusted_P_value:.1435,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:5.8261,Combined_Score:17.4801,Genes:"ATP1A1;ACTB",id:53},{Term:"MAPK signaling pathway",Overlap:"4/294",P_value:.05,Adjusted_P_value:.1435,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.0063,Combined_Score:9.009,Genes:"DUSP2;DUSP1;FOS;AREG",id:54},{Term:"ErbB signaling pathway",Overlap:"2/85",P_value:.0607,Adjusted_P_value:.1712,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:5.192,Combined_Score:14.5477,Genes:"CBLB;AREG",id:55},{Term:"Colorectal cancer",Overlap:"2/86",P_value:.062,Adjusted_P_value:.1717,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:5.1299,Combined_Score:14.2683,Genes:"FOS;AREG",id:56},{Term:"Gap junction",Overlap:"2/88",P_value:.0645,Adjusted_P_value:.1757,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:5.0101,Combined_Score:13.7333,Genes:"TUBA1B;TUBB",id:57},{Term:"PD-L1 expression and PD-1 checkpoint pathway in cancer",Overlap:"2/89",P_value:.0658,Adjusted_P_value:.1762,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.9523,Combined_Score:13.4769,Genes:"STAT1;FOS",id:58},{Term:"IL-17 signaling pathway",Overlap:"2/94",P_value:.0723,Adjusted_P_value:.1905,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.6819,Combined_Score:12.2962,Genes:"FOSB;FOS",id:59},{Term:"cAMP signaling pathway",Overlap:"3/216",P_value:.0817,Adjusted_P_value:.2116,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.048,Combined_Score:7.6351,Genes:"PDE3B;FOS;ATP1A1",id:60},{Term:"C-type lectin receptor signaling pathway",Overlap:"2/104",P_value:.0861,Adjusted_P_value:.2158,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.2208,Combined_Score:10.3524,Genes:"STAT1;CBLB",id:61},{Term:"NF-kappa B signaling pathway",Overlap:"2/104",P_value:.0861,Adjusted_P_value:.2158,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.2208,Combined_Score:10.3524,Genes:"CCL4L2;LTB",id:62},{Term:"Glucagon signaling pathway",Overlap:"2/107",P_value:.0903,Adjusted_P_value:.223,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.0996,Combined_Score:9.8572,Genes:"PKM;PDE3B",id:63},{Term:"TNF signaling pathway",Overlap:"2/112",P_value:.0975,Adjusted_P_value:.2371,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.9123,Combined_Score:9.1055,Genes:"SOCS3;FOS",id:64},{Term:"Leukocyte transendothelial migration",Overlap:"2/114",P_value:.1005,Adjusted_P_value:.2405,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.842,Combined_Score:8.8281,Genes:"ACTB;MYL12A",id:65},{Term:"Proximal tubule bicarbonate reclamation",Overlap:"1/23",P_value:.1027,Adjusted_P_value:.2423,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:9.7185,Combined_Score:22.1145,Genes:"ATP1A1",id:66},{Term:"Shigellosis",Overlap:"3/246",P_value:.1096,Adjusted_P_value:.2547,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.6676,Combined_Score:5.8972,Genes:"RPS27A;ACTB;MYL12A",id:67},{Term:"Platelet activation",Overlap:"2/124",P_value:.1155,Adjusted_P_value:.261,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.5253,Combined_Score:7.6087,Genes:"ACTB;MYL12A",id:68},{Term:"Endocytosis",Overlap:"3/252",P_value:.1156,Adjusted_P_value:.261,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.6025,Combined_Score:5.615,Genes:"HLA-B;CBLB;HLA-A",id:69},{Term:"Purine metabolism",Overlap:"2/129",P_value:.1232,Adjusted_P_value:.2743,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.3857,Combined_Score:7.0882,Genes:"PKM;PDE3B",id:70},{Term:"Yersinia infection",Overlap:"2/137",P_value:.1359,Adjusted_P_value:.294,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.1837,Combined_Score:6.3554,Genes:"FOS;ACTB",id:71},{Term:"Estrogen signaling pathway",Overlap:"2/137",P_value:.1359,Adjusted_P_value:.294,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.1837,Combined_Score:6.3554,Genes:"FOS;FKBP5",id:72},{Term:"Adrenergic signaling in cardiomyocytes",Overlap:"2/150",P_value:.1569,Adjusted_P_value:.335,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.9022,Combined_Score:5.3752,Genes:"CREM;ATP1A1",id:73},{Term:"Aldosterone-regulated sodium reabsorption",Overlap:"1/37",P_value:.1601,Adjusted_P_value:.3373,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:5.9349,Combined_Score:10.8726,Genes:"ATP1A1",id:74},{Term:"Oxytocin signaling pathway",Overlap:"2/154",P_value:.1635,Adjusted_P_value:.3389,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.8252,Combined_Score:5.1163,Genes:"FOS;ACTB",id:75},{Term:"Non-alcoholic fatty liver disease",Overlap:"2/155",P_value:.1652,Adjusted_P_value:.3389,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.8066,Combined_Score:5.0542,Genes:"SOCS3;FOS",id:76},{Term:"Hepatitis C",Overlap:"2/157",P_value:.1685,Adjusted_P_value:.3413,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.7701,Combined_Score:4.9333,Genes:"SOCS3;STAT1",id:77},{Term:"Hepatitis B",Overlap:"2/162",P_value:.1768,Adjusted_P_value:.3526,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.6829,Combined_Score:4.648,Genes:"STAT1;FOS",id:78},{Term:"Hippo signaling pathway",Overlap:"2/163",P_value:.1785,Adjusted_P_value:.3526,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.6661,Combined_Score:4.5937,Genes:"AREG;ACTB",id:79},{Term:"Pathways of neurodegeneration",Overlap:"4/475",P_value:.1848,Adjusted_P_value:.357,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.8339,Combined_Score:3.0961,Genes:"TUBA1B;TUBB;UBE2L6;RPS27A",id:80},{Term:"cGMP-PKG signaling pathway",Overlap:"2/167",P_value:.1853,Adjusted_P_value:.357,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.6009,Combined_Score:4.385,Genes:"PDE3B;ATP1A1",id:81},{Term:"Carbohydrate digestion and absorption",Overlap:"1/47",P_value:.1988,Adjusted_P_value:.374,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.6424,Combined_Score:7.4989,Genes:"ATP1A1",id:82},{Term:"Pyruvate metabolism",Overlap:"1/47",P_value:.1988,Adjusted_P_value:.374,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.6424,Combined_Score:7.4989,Genes:"PKM",id:83},{Term:"Sphingolipid metabolism",Overlap:"1/49",P_value:.2064,Adjusted_P_value:.3791,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.4485,Combined_Score:7.0204,Genes:"CERS6",id:84},{Term:"Cocaine addiction",Overlap:"1/49",P_value:.2064,Adjusted_P_value:.3791,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.4485,Combined_Score:7.0204,Genes:"FOSB",id:85},{Term:"Vibrio cholerae infection",Overlap:"1/50",P_value:.2101,Adjusted_P_value:.3816,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.3575,Combined_Score:6.7985,Genes:"ACTB",id:86},{Term:"Endocrine and other factor-regulated calcium reabsorption",Overlap:"1/53",P_value:.2212,Adjusted_P_value:.3968,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.1055,Combined_Score:6.1937,Genes:"ATP1A1",id:87},{Term:"Transcriptional misregulation in cancer",Overlap:"2/192",P_value:.2281,Adjusted_P_value:.3968,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.2558,Combined_Score:3.3338,Genes:"NFKBIZ;GZMB",id:88},{Term:"Chemokine signaling pathway",Overlap:"2/192",P_value:.2281,Adjusted_P_value:.3968,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.2558,Combined_Score:3.3338,Genes:"STAT1;CCL4L2",id:89},{Term:"Regulation of lipolysis in adipocytes",Overlap:"1/55",P_value:.2285,Adjusted_P_value:.3968,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.953,Combined_Score:5.835,Genes:"PDE3B",id:90},{Term:"Legionellosis",Overlap:"1/57",P_value:.2358,Adjusted_P_value:.4049,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.8114,Combined_Score:5.5069,Genes:"EEF1A1",id:91},{Term:"Focal adhesion",Overlap:"2/201",P_value:.2437,Adjusted_P_value:.41,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.1528,Combined_Score:3.039,Genes:"ACTB;MYL12A",id:92},{Term:"Amyotrophic lateral sclerosis",Overlap:"3/364",P_value:.2447,Adjusted_P_value:.41,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.7849,Combined_Score:2.5125,Genes:"TUBA1B;TUBB;ACTB",id:93},{Term:"Mineral absorption",Overlap:"1/60",P_value:.2465,Adjusted_P_value:.41,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.6171,Combined_Score:5.0647,Genes:"ATP1A1",id:94},{Term:"Cytosolic DNA-sensing pathway",Overlap:"1/63",P_value:.2571,Adjusted_P_value:.4232,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.4416,Combined_Score:4.674,Genes:"CCL4L2",id:95},{Term:"Glycolysis / Gluconeogenesis",Overlap:"1/67",P_value:.2711,Adjusted_P_value:.4348,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.2323,Combined_Score:4.2195,Genes:"PKM",id:96},{Term:"Regulation of actin cytoskeleton",Overlap:"2/218",P_value:.2733,Adjusted_P_value:.4348,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.9817,Combined_Score:2.5703,Genes:"ACTB;MYL12A",id:97},{Term:"Mitophagy",Overlap:"1/68",P_value:.2745,Adjusted_P_value:.4348,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.1839,Combined_Score:4.1162,Genes:"RPS27A",id:98},{Term:"Adipocytokine signaling pathway",Overlap:"1/69",P_value:.2779,Adjusted_P_value:.4348,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.1369,Combined_Score:4.0166,Genes:"SOCS3",id:99},{Term:"Renin secretion",Overlap:"1/69",P_value:.2779,Adjusted_P_value:.4348,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.1369,Combined_Score:4.0166,Genes:"PDE3B",id:100},{Term:"Central carbon metabolism in cancer",Overlap:"1/70",P_value:.2813,Adjusted_P_value:.4358,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.0913,Combined_Score:3.9205,Genes:"PKM",id:101},{Term:"Adherens junction",Overlap:"1/71",P_value:.2847,Adjusted_P_value:.4367,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.047,Combined_Score:3.8279,Genes:"ACTB",id:102},{Term:"Thyroid hormone synthesis",Overlap:"1/75",P_value:.2981,Adjusted_P_value:.4458,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.8817,Combined_Score:3.4877,Genes:"ATP1A1",id:103},{Term:"Pancreatic cancer",Overlap:"1/76",P_value:.3014,Adjusted_P_value:.4458,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.8432,Combined_Score:3.4096,Genes:"STAT1",id:104},{Term:"Pertussis",Overlap:"1/76",P_value:.3014,Adjusted_P_value:.4458,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.8432,Combined_Score:3.4096,Genes:"FOS",id:105},{Term:"Arrhythmogenic right ventricular cardiomyopathy",Overlap:"1/77",P_value:.3047,Adjusted_P_value:.4458,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.8056,Combined_Score:3.334,Genes:"ACTB",id:106},{Term:"Bacterial invasion of epithelial cells",Overlap:"1/77",P_value:.3047,Adjusted_P_value:.4458,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.8056,Combined_Score:3.334,Genes:"ACTB",id:107},{Term:"RNA degradation",Overlap:"1/79",P_value:.3113,Adjusted_P_value:.4512,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.7334,Combined_Score:3.1901,Genes:"BTG1",id:108},{Term:"B cell receptor signaling pathway",Overlap:"1/81",P_value:.3178,Adjusted_P_value:.4564,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.6648,Combined_Score:3.0551,Genes:"FOS",id:109},{Term:"Insulin secretion",Overlap:"1/86",P_value:.3337,Adjusted_P_value:.475,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.5074,Combined_Score:2.7519,Genes:"ATP1A1",id:110},{Term:"Cardiac muscle contraction",Overlap:"1/87",P_value:.3368,Adjusted_P_value:.4752,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.4781,Combined_Score:2.6965,Genes:"ATP1A1",id:111},{Term:"Bile secretion",Overlap:"1/90",P_value:.3462,Adjusted_P_value:.4792,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.3942,Combined_Score:2.5397,Genes:"ATP1A1",id:112},{Term:"Hypertrophic cardiomyopathy",Overlap:"1/90",P_value:.3462,Adjusted_P_value:.4792,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.3942,Combined_Score:2.5397,Genes:"ACTB",id:113},{Term:"Morphine addiction",Overlap:"1/91",P_value:.3493,Adjusted_P_value:.4792,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.3675,Combined_Score:2.4903,Genes:"PDE3B",id:114},{Term:"Salivary secretion",Overlap:"1/93",P_value:.3554,Adjusted_P_value:.4792,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.3158,Combined_Score:2.3956,Genes:"ATP1A1",id:115},{Term:"Dilated cardiomyopathy",Overlap:"1/96",P_value:.3645,Adjusted_P_value:.4792,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.2423,Combined_Score:2.263,Genes:"ACTB",id:116},{Term:"Circadian entrainment",Overlap:"1/97",P_value:.3675,Adjusted_P_value:.4792,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.2189,Combined_Score:2.2211,Genes:"FOS",id:117},{Term:"Prion disease",Overlap:"2/273",P_value:.3681,Adjusted_P_value:.4792,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.5751,Combined_Score:1.5742,Genes:"TUBA1B;TUBB",id:118},{Term:"Aldosterone synthesis and secretion",Overlap:"1/98",P_value:.3705,Adjusted_P_value:.4792,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.1959,Combined_Score:2.1804,Genes:"ATP1A1",id:119},{Term:"Choline metabolism in cancer",Overlap:"1/98",P_value:.3705,Adjusted_P_value:.4792,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.1959,Combined_Score:2.1804,Genes:"FOS",id:120},{Term:"AGE-RAGE signaling pathway in diabetic complications",Overlap:"1/100",P_value:.3764,Adjusted_P_value:.4792,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.1513,Combined_Score:2.1019,Genes:"STAT1",id:121},{Term:"Viral protein interaction with cytokine and cytokine receptor",Overlap:"1/100",P_value:.3764,Adjusted_P_value:.4792,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.1513,Combined_Score:2.1019,Genes:"CCL4L2",id:122},{Term:"Progesterone-mediated oocyte maturation",Overlap:"1/100",P_value:.3764,Adjusted_P_value:.4792,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.1513,Combined_Score:2.1019,Genes:"PDE3B",id:123},{Term:"Chagas disease",Overlap:"1/102",P_value:.3823,Adjusted_P_value:.4792,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.1085,Combined_Score:2.0274,Genes:"FOS",id:124},{Term:"Pancreatic secretion",Overlap:"1/102",P_value:.3823,Adjusted_P_value:.4792,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.1085,Combined_Score:2.0274,Genes:"ATP1A1",id:125},{Term:"Protein digestion and absorption",Overlap:"1/103",P_value:.3852,Adjusted_P_value:.4792,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.0877,Combined_Score:1.9916,Genes:"ATP1A1",id:126},{Term:"Parathyroid hormone synthesis, secretion and action",Overlap:"1/106",P_value:.3939,Adjusted_P_value:.4862,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.0278,Combined_Score:1.8892,Genes:"FOS",id:127},{Term:"Insulin resistance",Overlap:"1/108",P_value:.3996,Adjusted_P_value:.4894,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.9896,Combined_Score:1.8251,Genes:"SOCS3",id:128},{Term:"Cytokine-cytokine receptor interaction",Overlap:"2/295",P_value:.4048,Adjusted_P_value:.492,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.4552,Combined_Score:1.316,Genes:"CCL4L2;LTB",id:129},{Term:"Serotonergic synapse",Overlap:"1/113",P_value:.4137,Adjusted_P_value:.4951,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.9003,Combined_Score:1.6775,Genes:"DUSP1",id:130},{Term:"Cholinergic synapse",Overlap:"1/113",P_value:.4137,Adjusted_P_value:.4951,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.9003,Combined_Score:1.6775,Genes:"FOS",id:131},{Term:"Huntington disease",Overlap:"2/306",P_value:.4228,Adjusted_P_value:.5022,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.4017,Combined_Score:1.2068,Genes:"TUBA1B;TUBB",id:132},{Term:"Sphingolipid signaling pathway",Overlap:"1/119",P_value:.4301,Adjusted_P_value:.5071,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.8032,Combined_Score:1.5214,Genes:"CERS6",id:133},{Term:"Cell cycle",Overlap:"1/124",P_value:.4434,Adjusted_P_value:.519,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.7294,Combined_Score:1.4064,Genes:"RBL2",id:134},{Term:"Lysosome",Overlap:"1/128",P_value:.4539,Adjusted_P_value:.5264,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.6746,Combined_Score:1.3228,Genes:"CTSW",id:135},{Term:"Relaxin signaling pathway",Overlap:"1/129",P_value:.4565,Adjusted_P_value:.5264,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.6615,Combined_Score:1.3029,Genes:"FOS",id:136},{Term:"FoxO signaling pathway",Overlap:"1/131",P_value:.4616,Adjusted_P_value:.5276,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.6357,Combined_Score:1.2645,Genes:"RBL2",id:137},{Term:"Dopaminergic synapse",Overlap:"1/132",P_value:.4642,Adjusted_P_value:.5276,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.6232,Combined_Score:1.2458,Genes:"FOS",id:138},{Term:"Apelin signaling pathway",Overlap:"1/137",P_value:.4767,Adjusted_P_value:.538,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.5631,Combined_Score:1.158,Genes:"PDE3B",id:139},{Term:"PI3K-Akt signaling pathway",Overlap:"2/354",P_value:.4978,Adjusted_P_value:.5574,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.2076,Combined_Score:.8424,Genes:"RBL2;AREG",id:140},{Term:"Breast cancer",Overlap:"1/147",P_value:.501,Adjusted_P_value:.5574,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.4553,Combined_Score:1.0059,Genes:"FOS",id:141},{Term:"Alzheimer disease",Overlap:"2/369",P_value:.5199,Adjusted_P_value:.5745,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.1574,Combined_Score:.757,Genes:"TUBA1B;TUBB",id:142},{Term:"Necroptosis",Overlap:"1/159",P_value:.5286,Adjusted_P_value:.58,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.3439,Combined_Score:.8568,Genes:"STAT1",id:143},{Term:"Hepatocellular carcinoma",Overlap:"1/168",P_value:.5483,Adjusted_P_value:.5975,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.2709,Combined_Score:.7637,Genes:"ACTB",id:144},{Term:"Axon guidance",Overlap:"1/182",P_value:.5774,Adjusted_P_value:.6249,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.1718,Combined_Score:.6436,Genes:"MYL12A",id:145},{Term:"Alcoholism",Overlap:"1/186",P_value:.5854,Adjusted_P_value:.6249,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.1462,Combined_Score:.6138,Genes:"FOSB",id:146},{Term:"RNA transport",Overlap:"1/186",P_value:.5854,Adjusted_P_value:.6249,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.1462,Combined_Score:.6138,Genes:"EEF1A1",id:147},{Term:"Neutrophil extracellular trap formation",Overlap:"1/189",P_value:.5912,Adjusted_P_value:.6269,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.1278,Combined_Score:.5927,Genes:"ACTB",id:148},{Term:"Proteoglycans in cancer",Overlap:"1/205",P_value:.6212,Adjusted_P_value:.6543,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.0385,Combined_Score:.4944,Genes:"ACTB",id:149},{Term:"Rap1 signaling pathway",Overlap:"1/210",P_value:.6301,Adjusted_P_value:.6593,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.0134,Combined_Score:.468,Genes:"ACTB",id:150},{Term:"Lipid and atherosclerosis",Overlap:"1/215",P_value:.6388,Adjusted_P_value:.664,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.9894,Combined_Score:.4434,Genes:"FOS",id:151},{Term:"Thermogenesis",Overlap:"1/232",P_value:.6669,Adjusted_P_value:.6887,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.9158,Combined_Score:.371,Genes:"ACTB",id:152},{Term:"Chemical carcinogenesis",Overlap:"1/239",P_value:.6778,Adjusted_P_value:.6925,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.8886,Combined_Score:.3455,Genes:"FOS",id:153},{Term:"Calcium signaling pathway",Overlap:"1/240",P_value:.6794,Adjusted_P_value:.6925,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.8848,Combined_Score:.3421,Genes:"P2RX5",id:154},{Term:"Pathways in cancer",Overlap:"2/531",P_value:.7167,Adjusted_P_value:.7259,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.7963,Combined_Score:.2653,Genes:"STAT1;FOS",id:155},{Term:"MicroRNAs in cancer",Overlap:"1/310",P_value:.7705,Adjusted_P_value:.7754,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.6819,Combined_Score:.1778,Genes:"VIM",id:156},{Term:"Neuroactive ligand-receptor interaction",Overlap:"1/341",P_value:.8022,Adjusted_P_value:.8022,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.6188,Combined_Score:.1364,Genes:"P2RX5",id:157}]}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p:T,i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function An(e){let a,t,n,_;return a=new ve({props:{$$slots:{default:[gn]},$$scope:{ctx:e}}}),n=new ve({props:{$$slots:{default:[Pn]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment)},m(e,l){P(a,e,l),s(e,t,l),P(n,e,l),_=!0},p(e,t){const _={};1&t&&(_.$$scope={dirty:t,ctx:e}),a.$set(_);const l={};1&t&&(l.$$scope={dirty:t,ctx:e}),n.$set(l)},i(e){_||(i(a.$$.fragment,e),i(n.$$.fragment,e),_=!0)},o(e){p(a.$$.fragment,e),p(n.$$.fragment,e),_=!1},d(e){e&&c(t),A(a,e),A(n,e)}}}function fn(e){let a,t;return a=new ce({props:{$$slots:{content:[An],default:[mn]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};1&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function bn(e){let a,t;return a=new je({props:{title:"KEGG_2021_Human",open:!0,$$slots:{default:[fn]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};1&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function Cn(e){let a,t,n,_,r,o,u,v,m,b,C,O,S,R,h,$,T,y,L,j,x,M,B,G,F,N,U,w,D,E;return u=new ce({props:{$$slots:{content:[Jt],default:[Kt]},$$scope:{ctx:e}}}),O=new Yt({props:{$$slots:{default:[en]},$$scope:{ctx:e}}}),R=new Le({props:{align:"start",$$slots:{default:[dn]},$$scope:{ctx:e}}}),M=new ce({props:{$$slots:{content:[un],default:[sn]},$$scope:{ctx:e}}}),U=new Yt({props:{$$slots:{default:[vn]},$$scope:{ctx:e}}}),D=new Le({props:{align:"start",$$slots:{default:[bn]},$$scope:{ctx:e}}}),{c(){a=l("h1"),a.textContent="c1",t=l("a"),t.innerHTML="",n=f(),_=l("h2"),_.textContent="Markers",r=l("a"),r.innerHTML="",o=f(),g(u.$$.fragment),v=f(),m=l("h2"),m.textContent="Enrichment Analysis",b=l("a"),b.innerHTML="",C=f(),g(O.$$.fragment),S=f(),g(R.$$.fragment),h=f(),$=l("h1"),$.textContent="c2",T=l("a"),T.innerHTML="",y=f(),L=l("h2"),L.textContent="Markers",j=l("a"),j.innerHTML="",x=f(),g(M.$$.fragment),B=f(),G=l("h2"),G.textContent="Enrichment Analysis",F=l("a"),F.innerHTML="",N=f(),g(U.$$.fragment),w=f(),g(D.$$.fragment),d(t,"id","prt-h1-0-c1"),d(t,"class","pipen-report-toc-anchor"),d(r,"id","prt-h2-0-markers"),d(r,"class","pipen-report-toc-anchor"),d(b,"id","prt-h2-1-enrichment-analysis"),d(b,"class","pipen-report-toc-anchor"),d(T,"id","prt-h1-1-c2"),d(T,"class","pipen-report-toc-anchor"),d(j,"id","prt-h2-2-markers"),d(j,"class","pipen-report-toc-anchor"),d(F,"id","prt-h2-3-enrichment-analysis"),d(F,"class","pipen-report-toc-anchor")},m(e,l){s(e,a,l),s(e,t,l),s(e,n,l),s(e,_,l),s(e,r,l),s(e,o,l),P(u,e,l),s(e,v,l),s(e,m,l),s(e,b,l),s(e,C,l),P(O,e,l),s(e,S,l),P(R,e,l),s(e,h,l),s(e,$,l),s(e,T,l),s(e,y,l),s(e,L,l),s(e,j,l),s(e,x,l),P(M,e,l),s(e,B,l),s(e,G,l),s(e,F,l),s(e,N,l),P(U,e,l),s(e,w,l),P(D,e,l),E=!0},p(e,[a]){const t={};1&a&&(t.$$scope={dirty:a,ctx:e}),u.$set(t);const n={};1&a&&(n.$$scope={dirty:a,ctx:e}),O.$set(n);const _={};1&a&&(_.$$scope={dirty:a,ctx:e}),R.$set(_);const l={};1&a&&(l.$$scope={dirty:a,ctx:e}),M.$set(l);const d={};1&a&&(d.$$scope={dirty:a,ctx:e}),U.$set(d);const s={};1&a&&(s.$$scope={dirty:a,ctx:e}),D.$set(s)},i(e){E||(i(u.$$.fragment,e),i(O.$$.fragment,e),i(R.$$.fragment,e),i(M.$$.fragment,e),i(U.$$.fragment,e),i(D.$$.fragment,e),E=!0)},o(e){p(u.$$.fragment,e),p(O.$$.fragment,e),p(R.$$.fragment,e),p(M.$$.fragment,e),p(U.$$.fragment,e),p(D.$$.fragment,e),E=!1},d(e){e&&(c(a),c(t),c(n),c(_),c(r),c(o),c(v),c(m),c(b),c(C),c(S),c(h),c($),c(T),c(y),c(L),c(j),c(x),c(B),c(G),c(F),c(N),c(w)),A(u,e),A(O,e),A(R,e),A(M,e),A(U,e),A(D,e)}}}class On extends a{constructor(e){super(),t(this,e,null,Cn,n,{})}}var Sn=[{slug:"prt-h1-0-c1",text:"c1",children:[{slug:"prt-h2-0-markers",text:"Markers",children:[],page:0},{slug:"prt-h2-1-enrichment-analysis",text:"Enrichment Analysis",children:[],page:0}],page:0},{slug:"prt-h1-1-c2",text:"c2",children:[{slug:"prt-h2-2-markers",text:"Markers",children:[],page:0},{slug:"prt-h2-3-enrichment-analysis",text:"Enrichment Analysis",children:[],page:0}],page:0}],Rn={pipeline:{name:"Immunopipe",desc:"Immunopipe (v1.4.2): Integrative analysis for scRNA-seq and scTCR-seq data"},versions:'\n    <a href="https://github.com/pwwang/pipen" target="_blank">\n        pipen v0.15.2\n    </a> and\n    <a href="https://github.com/pwwang/pipen-report" target="_blank">\n        pipen-report v0.20.1\n    </a>\n',entries:[{name:"SampleInfo",desc:"List sample information and perform statistics",npages:1,report_toc:!0,order:0,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n[1] ggrepel_0.9.5    ggprism_1.0.5    ggplot2_3.5.1    tidyr_1.3.1     \n[5] dplyr_1.1.4      tidyselect_1.2.1 jsonlite_1.8.8   rlang_1.1.4     \n[9] logger_0.3.0    \n\nloaded via a namespace (and not attached):\n [1] vctrs_0.6.5      cli_3.6.3        purrr_1.0.2      generics_0.1.3  \n [5] labeling_0.4.3   glue_1.7.0       colorspace_2.1-1 scales_1.3.0    \n [9] fansi_1.0.6      grid_4.3.3       munsell_0.5.1    tibble_3.2.1    \n[13] lifecycle_1.0.4  compiler_4.3.3   Rcpp_1.0.13      pkgconfig_2.0.3 \n[17] farver_2.1.2     digest_0.6.36    R6_2.5.1         utf8_1.2.4      \n[21] pillar_1.9.0     magrittr_2.0.3   withr_3.0.1      gtable_0.3.5    \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 1.73\nuser 10.62\nsys 0.10\npercent 617%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"SeuratPreparing",desc:"Load, prepare and apply QC to data, using `Seurat`",npages:1,report_toc:!0,order:2,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats4    stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] scDblFinder_1.16.0          SingleCellExperiment_1.24.0\n [3] SummarizedExperiment_1.32.0 Biobase_2.62.0             \n [5] GenomicRanges_1.54.1        GenomeInfoDb_1.38.1        \n [7] IRanges_2.36.0              S4Vectors_0.40.2           \n [9] BiocGenerics_0.48.1         MatrixGenerics_1.14.0      \n[11] matrixStats_1.1.0           dplyr_1.1.4                \n[13] ggplot2_3.5.1               bracer_1.2.1               \n[15] future_1.34.0               Seurat_5.1.0               \n[17] SeuratObject_5.0.2          sp_2.1-4                   \n[19] digest_0.6.36               jsonlite_1.8.8             \n[21] rlang_1.1.4                 logger_0.3.0               \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22          splines_4.3.3            \n  [3] later_1.3.2               BiocIO_1.12.0            \n  [5] bitops_1.0-8              tibble_3.2.1             \n  [7] R.oo_1.26.0               polyclip_1.10-7          \n  [9] XML_3.99-0.17             fastDummies_1.7.3        \n [11] lifecycle_1.0.4           edgeR_4.0.16             \n [13] globals_0.16.3            lattice_0.22-6           \n [15] MASS_7.3-60.0.1           magrittr_2.0.3           \n [17] limma_3.58.1              plotly_4.10.4            \n [19] yaml_2.3.10               metapod_1.10.0           \n [21] httpuv_1.6.15             sctransform_0.4.1        \n [23] spam_2.10-0               spatstat.sparse_3.1-0    \n [25] reticulate_1.38.0         cowplot_1.1.3            \n [27] pbapply_1.7-2             RColorBrewer_1.1-3       \n [29] abind_1.4-5               zlibbioc_1.48.0          \n [31] Rtsne_0.17                purrr_1.0.2              \n [33] R.utils_2.12.3            RCurl_1.98-1.16          \n [35] GenomeInfoDbData_1.2.11   ggrepel_0.9.5            \n [37] irlba_2.3.5.1             listenv_0.9.1            \n [39] spatstat.utils_3.0-5      goftest_1.2-3            \n [41] RSpectra_0.16-1           dqrng_0.3.2              \n [43] spatstat.random_3.2-3     fitdistrplus_1.1-11      \n [45] parallelly_1.38.0         DelayedMatrixStats_1.24.0\n [47] leiden_0.4.3.1            codetools_0.2-20         \n [49] DelayedArray_0.28.0       scuttle_1.12.0           \n [51] tidyselect_1.2.1          farver_2.1.2             \n [53] viridis_0.6.5             ScaledMatrix_1.10.0      \n [55] spatstat.explore_3.2-6    GenomicAlignments_1.38.0 \n [57] BiocNeighbors_1.20.0      progressr_0.14.0         \n [59] scater_1.30.1             ggridges_0.5.6           \n [61] survival_3.7-0            systemfonts_1.1.0        \n [63] tools_4.3.3               ragg_1.3.2               \n [65] ica_1.0-3                 Rcpp_1.0.13              \n [67] glue_1.7.0                gridExtra_2.3            \n [69] SparseArray_1.2.2         withr_3.0.1              \n [71] fastmap_1.2.0             bluster_1.12.0           \n [73] fansi_1.0.6               rsvd_1.0.5               \n [75] R6_2.5.1                  mime_0.12                \n [77] textshaping_0.4.0         colorspace_2.1-1         \n [79] scattermore_1.2           tensor_1.5               \n [81] spatstat.data_3.1-2       R.methodsS3_1.8.2        \n [83] RhpcBLASctl_0.23-42       utf8_1.2.4               \n [85] tidyr_1.3.1               generics_0.1.3           \n [87] data.table_1.15.4         rtracklayer_1.62.0       \n [89] httr_1.4.7                htmlwidgets_1.6.4        \n [91] S4Arrays_1.2.0            uwot_0.1.16              \n [93] pkgconfig_2.0.3           gtable_0.3.5             \n [95] lmtest_0.9-40             XVector_0.42.0           \n [97] htmltools_0.5.8.1         dotCall64_1.1-1          \n [99] scales_1.3.0              png_0.1-8                \n[101] harmony_1.2.0             scran_1.30.0             \n[103] reshape2_1.4.4            rjson_0.2.21             \n[105] nlme_3.1-165              zoo_1.8-12               \n[107] stringr_1.5.1             KernSmooth_2.23-24       \n[109] vipor_0.4.7               parallel_4.3.3           \n[111] miniUI_0.1.1.1            restfulr_0.0.15          \n[113] pillar_1.9.0              grid_4.3.3               \n[115] vctrs_0.6.5               RANN_2.6.1               \n[117] promises_1.3.0            BiocSingular_1.18.0      \n[119] beachmat_2.18.0           xtable_1.8-4             \n[121] cluster_2.1.6             beeswarm_0.4.0           \n[123] locfit_1.5-9.9            cli_3.6.3                \n[125] compiler_4.3.3            Rsamtools_2.18.0         \n[127] crayon_1.5.3              future.apply_1.11.2      \n[129] labeling_0.4.3            ggbeeswarm_0.7.2         \n[131] plyr_1.8.9                stringi_1.8.4            \n[133] viridisLite_0.4.2         deldir_2.0-4             \n[135] BiocParallel_1.36.0       munsell_0.5.1            \n[137] Biostrings_2.70.1         lazyeval_0.2.2           \n[139] spatstat.geom_3.2-9       Matrix_1.6-5             \n[141] RcppHNSW_0.6.0            patchwork_1.2.0          \n[143] sparseMatrixStats_1.14.0  statmod_1.5.0            \n[145] shiny_1.9.1               ROCR_1.0-11              \n[147] igraph_2.0.3              xgboost_2.1.1.1          \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 202.98\nuser 1582.98\nsys 7.06\npercent 783%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ClusterMarkersOfAllCells",desc:"Markers for clusters of all cells.",npages:3,report_toc:!0,order:4,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] UpSetR_1.4.0        ggVennDiagram_1.2.3 tidyseurat_0.8.0   \n [4] ttservice_0.4.1     future_1.34.0       ggrepel_0.9.5      \n [7] ggprism_1.0.5       ggplot2_3.5.1       enrichR_3.2        \n[10] Seurat_5.1.0        SeuratObject_5.0.2  sp_2.1-4           \n[13] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[16] tidyselect_1.2.1    digest_0.6.36       jsonlite_1.8.8     \n[19] rlang_1.1.4         logger_0.3.0       \n\nloaded via a namespace (and not attached):\n  [1] RColorBrewer_1.1-3     magrittr_2.0.3         spatstat.utils_3.0-5  \n  [4] farver_2.1.2           vctrs_0.6.5            ROCR_1.0-11           \n  [7] spatstat.explore_3.2-6 htmltools_0.5.8.1      curl_5.2.1            \n [10] sctransform_0.4.1      parallelly_1.38.0      KernSmooth_2.23-24    \n [13] htmlwidgets_1.6.4      ica_1.0-3              plyr_1.8.9            \n [16] plotly_4.10.4          zoo_1.8-12             igraph_2.0.3          \n [19] mime_0.12              lifecycle_1.0.4        pkgconfig_2.0.3       \n [22] Matrix_1.6-5           R6_2.5.1               fastmap_1.2.0         \n [25] fitdistrplus_1.1-11    shiny_1.9.1            colorspace_2.1-1      \n [28] patchwork_1.2.0        tensor_1.5             RSpectra_0.16-1       \n [31] irlba_2.3.5.1          WriteXLS_6.7.0         labeling_0.4.3        \n [34] progressr_0.14.0       fansi_1.0.6            spatstat.sparse_3.1-0 \n [37] httr_1.4.7             polyclip_1.10-7        abind_1.4-5           \n [40] compiler_4.3.3         withr_3.0.1            fastDummies_1.7.3     \n [43] MASS_7.3-60.0.1        rjson_0.2.21           tools_4.3.3           \n [46] lmtest_0.9-40          httpuv_1.6.15          future.apply_1.11.2   \n [49] goftest_1.2-3          glue_1.7.0             nlme_3.1-165          \n [52] promises_1.3.0         grid_4.3.3             Rtsne_0.17            \n [55] cluster_2.1.6          reshape2_1.4.4         generics_0.1.3        \n [58] gtable_0.3.5           spatstat.data_3.1-2    data.table_1.15.4     \n [61] utf8_1.2.4             spatstat.geom_3.2-9    RcppAnnoy_0.0.22      \n [64] RANN_2.6.1             pillar_1.9.0           stringr_1.5.1         \n [67] spam_2.10-0            RcppHNSW_0.6.0         limma_3.58.1          \n [70] later_1.3.2            splines_4.3.3          lattice_0.22-6        \n [73] survival_3.7-0         deldir_2.0-4           miniUI_0.1.1.1        \n [76] pbapply_1.7-2          gridExtra_2.3          scattermore_1.2       \n [79] statmod_1.5.0          matrixStats_1.1.0      stringi_1.8.4         \n [82] lazyeval_0.2.2         codetools_0.2-20       RVenn_1.1.0           \n [85] cli_3.6.3              uwot_0.1.16            xtable_1.8-4          \n [88] reticulate_1.38.0      munsell_0.5.1          Rcpp_1.0.13           \n [91] globals_0.16.3         spatstat.random_3.2-3  png_0.1-8             \n [94] parallel_4.3.3         ellipsis_0.3.2         presto_1.0.0          \n [97] dotCall64_1.1-1        listenv_0.9.1          viridisLite_0.4.2     \n[100] scales_1.3.0           ggridges_0.5.6         leiden_0.4.3.1        \n[103] purrr_1.0.2            cowplot_1.1.3         \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 216.89\nuser 139.48\nsys 7.23\npercent 67%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TopExpressingGenesOfAllCells",desc:"Top expressing genes for clusters of all cells.",npages:3,report_toc:!0,order:5,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ggprism_1.0.5      dplyr_1.1.4        enrichR_3.2        tibble_3.2.1      \n [5] Seurat_5.1.0       SeuratObject_5.0.2 sp_2.1-4           jsonlite_1.8.8    \n [9] rlang_1.1.4        logger_0.3.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.2-9   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] labeling_0.4.3         utf8_1.2.4             promises_1.3.0        \n [19] purrr_1.0.2            WriteXLS_6.7.0         goftest_1.2-3         \n [22] later_1.3.2            spatstat.utils_3.0-5   irlba_2.3.5.1         \n [25] parallel_4.3.3         cluster_2.1.6          R6_2.5.1              \n [28] ica_1.0-3              stringi_1.8.4          RColorBrewer_1.1-3    \n [31] spatstat.data_3.1-2    reticulate_1.38.0      parallelly_1.38.0     \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.13           \n [37] tensor_1.5             future.apply_1.11.2    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [43] splines_4.3.3          igraph_2.0.3           tidyselect_1.2.1      \n [46] abind_1.4-5            spatstat.random_3.2-3  codetools_0.2-20      \n [49] miniUI_0.1.1.1         spatstat.explore_3.2-6 curl_5.2.1            \n [52] listenv_0.9.1          lattice_0.22-6         plyr_1.8.9            \n [55] withr_3.0.1            shiny_1.9.1            ROCR_1.0-11           \n [58] Rtsne_0.17             future_1.34.0          fastDummies_1.7.3     \n [61] survival_3.7-0         polyclip_1.10-7        fitdistrplus_1.1-11   \n [64] pillar_1.9.0           KernSmooth_2.23-24     plotly_4.10.4         \n [67] generics_0.1.3         RcppHNSW_0.6.0         ggplot2_3.5.1         \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.7.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.15.4      RSpectra_0.16-1       \n [79] RANN_2.6.1             leiden_0.4.3.1         dotCall64_1.1-1       \n [82] cowplot_1.1.3          grid_4.3.3             tidyr_1.3.1           \n [85] colorspace_2.1-1       nlme_3.1-165           patchwork_1.2.0       \n [88] cli_3.6.3              spatstat.sparse_3.1-0  spam_2.10-0           \n [91] fansi_1.0.6            viridisLite_0.4.2      uwot_0.1.16           \n [94] gtable_0.3.5           digest_0.6.36          progressr_0.14.0      \n [97] ggrepel_0.9.5          farver_2.1.2           rjson_0.2.21          \n[100] htmlwidgets_1.6.4      htmltools_0.5.8.1      lifecycle_1.0.4       \n[103] httr_1.4.7             mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 52.61\nuser 38.55\nsys 2.30\npercent 77%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TCellSelection",desc:"Separate T and non-T cells and select T cells.",npages:1,report_toc:!0,order:6,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] glue_1.7.0         factoextra_1.0.7   patchwork_1.2.0    ggpubr_0.6.0      \n [5] ggrepel_0.9.5      ggprism_1.0.5      ggplot2_3.5.1      tibble_3.2.1      \n [9] tidyr_1.3.1        dplyr_1.1.4        Seurat_5.1.0       SeuratObject_5.0.2\n[13] sp_2.1-4           jsonlite_1.8.8     rlang_1.1.4        logger_0.3.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.2-9   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] backports_1.5.0        labeling_0.4.3         utf8_1.2.4            \n [19] promises_1.3.0         purrr_1.0.2            goftest_1.2-3         \n [22] later_1.3.2            spatstat.utils_3.0-5   broom_1.0.6           \n [25] irlba_2.3.5.1          parallel_4.3.3         cluster_2.1.6         \n [28] R6_2.5.1               ica_1.0-3              stringi_1.8.4         \n [31] RColorBrewer_1.1-3     spatstat.data_3.1-2    reticulate_1.38.0     \n [34] car_3.1-2              parallelly_1.38.0      lmtest_0.9-40         \n [37] scattermore_1.2        Rcpp_1.0.13            tensor_1.5            \n [40] future.apply_1.11.2    zoo_1.8-12             sctransform_0.4.1     \n [43] httpuv_1.6.15          Matrix_1.6-5           splines_4.3.3         \n [46] igraph_2.0.3           tidyselect_1.2.1       abind_1.4-5           \n [49] spatstat.random_3.2-3  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.2-6 listenv_0.9.1          lattice_0.22-6        \n [55] plyr_1.8.9             withr_3.0.1            shiny_1.9.1           \n [58] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n [61] fastDummies_1.7.3      survival_3.7-0         polyclip_1.10-7       \n [64] fitdistrplus_1.1-11    pillar_1.9.0           carData_3.0-5         \n [67] KernSmooth_2.23-24     plotly_4.10.4          generics_0.1.3        \n [70] RcppHNSW_0.6.0         munsell_0.5.1          scales_1.3.0          \n [73] globals_0.16.3         xtable_1.8-4           lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.15.4      RSpectra_0.16-1       \n [79] ggsignif_0.6.4         RANN_2.6.1             leiden_0.4.3.1        \n [82] dotCall64_1.1-1        cowplot_1.1.3          grid_4.3.3            \n [85] colorspace_2.1-1       nlme_3.1-165           cli_3.6.3             \n [88] spatstat.sparse_3.1-0  spam_2.10-0            fansi_1.0.6           \n [91] viridisLite_0.4.2      uwot_0.1.16            gtable_0.3.5          \n [94] rstatix_0.7.2          digest_0.6.36          progressr_0.14.0      \n [97] farver_2.1.2           htmlwidgets_1.6.4      htmltools_0.5.8.1     \n[100] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[103] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 38.03\nuser 45.75\nsys 0.83\npercent 122%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3435.798\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ClusterMarkers",desc:"Markers for clusters of all or selected T cells.",npages:2,report_toc:!0,order:10,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] UpSetR_1.4.0        ggVennDiagram_1.2.3 tidyseurat_0.8.0   \n [4] ttservice_0.4.1     future_1.34.0       ggrepel_0.9.5      \n [7] ggprism_1.0.5       ggplot2_3.5.1       enrichR_3.2        \n[10] Seurat_5.1.0        SeuratObject_5.0.2  sp_2.1-4           \n[13] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[16] tidyselect_1.2.1    digest_0.6.36       jsonlite_1.8.8     \n[19] rlang_1.1.4         logger_0.3.0       \n\nloaded via a namespace (and not attached):\n  [1] RColorBrewer_1.1-3     magrittr_2.0.3         spatstat.utils_3.0-5  \n  [4] farver_2.1.2           vctrs_0.6.5            ROCR_1.0-11           \n  [7] spatstat.explore_3.2-6 htmltools_0.5.8.1      curl_5.2.1            \n [10] sctransform_0.4.1      parallelly_1.38.0      KernSmooth_2.23-24    \n [13] htmlwidgets_1.6.4      ica_1.0-3              plyr_1.8.9            \n [16] plotly_4.10.4          zoo_1.8-12             igraph_2.0.3          \n [19] mime_0.12              lifecycle_1.0.4        pkgconfig_2.0.3       \n [22] Matrix_1.6-5           R6_2.5.1               fastmap_1.2.0         \n [25] fitdistrplus_1.1-11    shiny_1.9.1            colorspace_2.1-1      \n [28] patchwork_1.2.0        tensor_1.5             RSpectra_0.16-1       \n [31] irlba_2.3.5.1          WriteXLS_6.7.0         labeling_0.4.3        \n [34] progressr_0.14.0       fansi_1.0.6            spatstat.sparse_3.1-0 \n [37] httr_1.4.7             polyclip_1.10-7        abind_1.4-5           \n [40] compiler_4.3.3         withr_3.0.1            fastDummies_1.7.3     \n [43] MASS_7.3-60.0.1        rjson_0.2.21           tools_4.3.3           \n [46] lmtest_0.9-40          httpuv_1.6.15          future.apply_1.11.2   \n [49] goftest_1.2-3          glue_1.7.0             nlme_3.1-165          \n [52] promises_1.3.0         grid_4.3.3             Rtsne_0.17            \n [55] cluster_2.1.6          reshape2_1.4.4         generics_0.1.3        \n [58] gtable_0.3.5           spatstat.data_3.1-2    data.table_1.15.4     \n [61] utf8_1.2.4             spatstat.geom_3.2-9    RcppAnnoy_0.0.22      \n [64] RANN_2.6.1             pillar_1.9.0           stringr_1.5.1         \n [67] spam_2.10-0            RcppHNSW_0.6.0         limma_3.58.1          \n [70] later_1.3.2            splines_4.3.3          lattice_0.22-6        \n [73] survival_3.7-0         deldir_2.0-4           miniUI_0.1.1.1        \n [76] pbapply_1.7-2          gridExtra_2.3          scattermore_1.2       \n [79] statmod_1.5.0          matrixStats_1.1.0      stringi_1.8.4         \n [82] lazyeval_0.2.2         codetools_0.2-20       RVenn_1.1.0           \n [85] cli_3.6.3              uwot_0.1.16            xtable_1.8-4          \n [88] reticulate_1.38.0      munsell_0.5.1          Rcpp_1.0.13           \n [91] globals_0.16.3         spatstat.random_3.2-3  png_0.1-8             \n [94] parallel_4.3.3         ellipsis_0.3.2         presto_1.0.0          \n [97] dotCall64_1.1-1        listenv_0.9.1          viridisLite_0.4.2     \n[100] scales_1.3.0           ggridges_0.5.6         leiden_0.4.3.1        \n[103] purrr_1.0.2            cowplot_1.1.3         \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 136.36\nuser 90.28\nsys 4.48\npercent 69%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TopExpressingGenes",desc:"Top expressing genes for clusters of all or selected T cells.",npages:2,report_toc:!0,order:11,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ggprism_1.0.5      dplyr_1.1.4        enrichR_3.2        tibble_3.2.1      \n [5] Seurat_5.1.0       SeuratObject_5.0.2 sp_2.1-4           jsonlite_1.8.8    \n [9] rlang_1.1.4        logger_0.3.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.2-9   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] labeling_0.4.3         utf8_1.2.4             promises_1.3.0        \n [19] purrr_1.0.2            WriteXLS_6.7.0         goftest_1.2-3         \n [22] later_1.3.2            spatstat.utils_3.0-5   irlba_2.3.5.1         \n [25] parallel_4.3.3         cluster_2.1.6          R6_2.5.1              \n [28] ica_1.0-3              stringi_1.8.4          RColorBrewer_1.1-3    \n [31] spatstat.data_3.1-2    reticulate_1.38.0      parallelly_1.38.0     \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.13           \n [37] tensor_1.5             future.apply_1.11.2    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [43] splines_4.3.3          igraph_2.0.3           tidyselect_1.2.1      \n [46] abind_1.4-5            spatstat.random_3.2-3  codetools_0.2-20      \n [49] miniUI_0.1.1.1         spatstat.explore_3.2-6 curl_5.2.1            \n [52] listenv_0.9.1          lattice_0.22-6         plyr_1.8.9            \n [55] withr_3.0.1            shiny_1.9.1            ROCR_1.0-11           \n [58] Rtsne_0.17             future_1.34.0          fastDummies_1.7.3     \n [61] survival_3.7-0         polyclip_1.10-7        fitdistrplus_1.1-11   \n [64] pillar_1.9.0           KernSmooth_2.23-24     plotly_4.10.4         \n [67] generics_0.1.3         RcppHNSW_0.6.0         ggplot2_3.5.1         \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.7.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.15.4      RSpectra_0.16-1       \n [79] RANN_2.6.1             leiden_0.4.3.1         dotCall64_1.1-1       \n [82] cowplot_1.1.3          grid_4.3.3             tidyr_1.3.1           \n [85] colorspace_2.1-1       nlme_3.1-165           patchwork_1.2.0       \n [88] cli_3.6.3              spatstat.sparse_3.1-0  spam_2.10-0           \n [91] fansi_1.0.6            viridisLite_0.4.2      uwot_0.1.16           \n [94] gtable_0.3.5           digest_0.6.36          progressr_0.14.0      \n [97] ggrepel_0.9.5          farver_2.1.2           rjson_0.2.21          \n[100] htmlwidgets_1.6.4      htmltools_0.5.8.1      lifecycle_1.0.4       \n[103] httr_1.4.7             mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 31.30\nuser 27.76\nsys 1.57\npercent 93%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TCRClusterStats",desc:"Statistics of TCR clusters, generated by `TCRClustering`.",npages:1,report_toc:!0,order:13,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] grid      stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] ComplexHeatmap_2.18.0 ggprism_1.0.5         immunarch_0.9.1      \n [4] patchwork_1.2.0       data.table_1.15.4     dtplyr_1.3.1         \n [7] dplyr_1.1.4           tidyr_1.3.1           tibble_3.2.1         \n[10] ggplot2_3.5.1         jsonlite_1.8.8        rlang_1.1.4          \n[13] logger_0.3.0         \n\nloaded via a namespace (and not attached):\n  [1] gridExtra_2.3       phangorn_2.11.1     readxl_1.4.3       \n  [4] magrittr_2.0.3      clue_0.3-65         GetoptLong_1.0.5   \n  [7] matrixStats_1.1.0   compiler_4.3.3      flexmix_2.3-19     \n [10] png_0.1-8           vctrs_0.6.5         reshape2_1.4.4     \n [13] ggalluvial_0.12.5   quadprog_1.5-8      stringr_1.5.1      \n [16] crayon_1.5.3        pkgconfig_2.0.3     shape_1.4.6.1      \n [19] fastmap_1.2.0       backports_1.5.0     labeling_0.4.3     \n [22] ggraph_2.1.0        utf8_1.2.4          promises_1.3.0     \n [25] tzdb_0.4.0          UpSetR_1.4.0        purrr_1.0.2        \n [28] modeltools_0.2-23   ggseqlogo_0.2       later_1.3.2        \n [31] uuid_1.2-1          fpc_2.2-12          tweenr_2.0.3       \n [34] broom_1.0.6         parallel_4.3.3      prabclus_2.3-3     \n [37] cluster_2.1.6       R6_2.5.1            stringi_1.8.4      \n [40] RColorBrewer_1.1-3  rlist_0.4.6.2       car_3.1-2          \n [43] diptest_0.77-1      cellranger_1.1.0    Rcpp_1.0.13        \n [46] iterators_1.0.14    IRanges_2.36.0      readr_2.1.5        \n [49] httpuv_1.6.15       Matrix_1.6-5        nnet_7.3-19        \n [52] igraph_2.0.3        tidyselect_1.2.1    stringdist_0.9.12  \n [55] abind_1.4-5         viridis_0.6.5       doParallel_1.0.17  \n [58] codetools_0.2-20    lattice_0.22-6      plyr_1.8.9         \n [61] shiny_1.9.1         withr_3.0.1         polyclip_1.10-7    \n [64] circlize_0.4.16     mclust_6.1.1        kernlab_0.9-32     \n [67] pillar_1.9.0        ggpubr_0.6.0        carData_3.0-5      \n [70] foreach_1.5.2       stats4_4.3.3        generics_0.1.3     \n [73] S4Vectors_0.40.2    hms_1.1.3           munsell_0.5.1      \n [76] scales_1.3.0        xtable_1.8-4        class_7.3-22       \n [79] glue_1.7.0          pheatmap_1.0.12     tools_4.3.3        \n [82] robustbase_0.99-3   ggsignif_0.6.4      graphlayouts_1.1.0 \n [85] Cairo_1.6-2         fastmatch_1.1-4     tidygraph_1.3.0    \n [88] ape_5.8             colorspace_2.1-1    nlme_3.1-165       \n [91] ggforce_0.4.2       cli_3.6.3           fansi_1.0.6        \n [94] viridisLite_0.4.2   gtable_0.3.5        DEoptimR_1.1-3     \n [97] rstatix_0.7.2       digest_0.6.36       BiocGenerics_0.48.1\n[100] ggrepel_0.9.5       rjson_0.2.21        farver_2.1.2       \n[103] htmltools_0.5.8.1   lifecycle_1.0.4     factoextra_1.0.7   \n[106] GlobalOptions_0.1.2 mime_0.12           shinythemes_1.2.0  \n[109] MASS_7.3-60.0.1    \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 5.36\nuser 14.04\nsys 0.30\npercent 267%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TESSA",desc:"Tessa is a Bayesian model to integrate T cell receptor (TCR) sequence profiling with transcriptomes of T cells.",npages:1,report_toc:!0,order:14,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] igraph_2.0.3         Rtsne_0.17           LaplacesDemon_16.1.6\n [4] MASS_7.3-60.0.1      ggprism_1.0.5        Seurat_5.1.0        \n [7] SeuratObject_5.0.2   sp_2.1-4             tibble_3.2.1        \n[10] glue_1.7.0           immunarch_0.9.1      patchwork_1.2.0     \n[13] data.table_1.15.4    dtplyr_1.3.1         ggplot2_3.5.1       \n[16] tidyr_1.3.1          dplyr_1.1.4          jsonlite_1.8.8      \n[19] rlang_1.1.4          logger_0.3.0        \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22       shinythemes_1.2.0      splines_4.3.3         \n  [4] later_1.3.2            cellranger_1.1.0       polyclip_1.10-7       \n  [7] fastDummies_1.7.3      factoextra_1.0.7       lifecycle_1.0.4       \n [10] rstatix_0.7.2          doParallel_1.0.17      globals_0.16.3        \n [13] lattice_0.22-6         prabclus_2.3-3         backports_1.5.0       \n [16] magrittr_2.0.3         plotly_4.10.4          rlist_0.4.6.2         \n [19] httpuv_1.6.15          sctransform_0.4.1      spam_2.10-0           \n [22] flexmix_2.3-19         spatstat.sparse_3.1-0  reticulate_1.38.0     \n [25] cowplot_1.1.3          pbapply_1.7-2          RColorBrewer_1.1-3    \n [28] abind_1.4-5            quadprog_1.5-8         purrr_1.0.2           \n [31] ggraph_2.1.0           nnet_7.3-19            tweenr_2.0.3          \n [34] circlize_0.4.16        ggrepel_0.9.5          irlba_2.3.5.1         \n [37] listenv_0.9.1          spatstat.utils_3.0-5   pheatmap_1.0.12       \n [40] goftest_1.2-3          RSpectra_0.16-1        spatstat.random_3.2-3 \n [43] fitdistrplus_1.1-11    parallelly_1.38.0      leiden_0.4.3.1        \n [46] codetools_0.2-20       ggforce_0.4.2          tidyselect_1.2.1      \n [49] shape_1.4.6.1          ggseqlogo_0.2          farver_2.1.2          \n [52] viridis_0.6.5          matrixStats_1.1.0      stats4_4.3.3          \n [55] spatstat.explore_3.2-6 tidygraph_1.3.0        progressr_0.14.0      \n [58] ggridges_0.5.6         ggalluvial_0.12.5      survival_3.7-0        \n [61] iterators_1.0.14       foreach_1.5.2          tools_4.3.3           \n [64] stringdist_0.9.12      ica_1.0-3              Rcpp_1.0.13           \n [67] gridExtra_2.3          withr_3.0.1            fastmap_1.2.0         \n [70] fansi_1.0.6            digest_0.6.36          R6_2.5.1              \n [73] mime_0.12              colorspace_2.1-1       scattermore_1.2       \n [76] tensor_1.5             spatstat.data_3.1-2    diptest_0.77-1        \n [79] UpSetR_1.4.0           utf8_1.2.4             generics_0.1.3        \n [82] robustbase_0.99-3      class_7.3-22           graphlayouts_1.1.0    \n [85] httr_1.4.7             htmlwidgets_1.6.4      uwot_0.1.16           \n [88] pkgconfig_2.0.3        gtable_0.3.5           modeltools_0.2-23     \n [91] lmtest_0.9-40          htmltools_0.5.8.1      carData_3.0-5         \n [94] dotCall64_1.1-1        scales_1.3.0           png_0.1-8             \n [97] tzdb_0.4.0             reshape2_1.4.4         uuid_1.2-1            \n[100] nlme_3.1-165           zoo_1.8-12             GlobalOptions_0.1.2   \n[103] stringr_1.5.1          KernSmooth_2.23-24     parallel_4.3.3        \n[106] miniUI_0.1.1.1         pillar_1.9.0           grid_4.3.3            \n[109] vctrs_0.6.5            RANN_2.6.1             promises_1.3.0        \n[112] ggpubr_0.6.0           car_3.1-2              xtable_1.8-4          \n[115] cluster_2.1.6          readr_2.1.5            cli_3.6.3             \n[118] compiler_4.3.3         future.apply_1.11.2    ggsignif_0.6.4        \n[121] labeling_0.4.3         mclust_6.1.1           plyr_1.8.9            \n[124] stringi_1.8.4          deldir_2.0-4           viridisLite_0.4.2     \n[127] munsell_0.5.1          lazyeval_0.2.2         spatstat.geom_3.2-9   \n[130] Matrix_1.6-5           RcppHNSW_0.6.0         hms_1.1.3             \n[133] future_1.34.0          fpc_2.2-12             shiny_1.9.1           \n[136] kernlab_0.9-32         ROCR_1.0-11            broom_1.0.6           \n[139] phangorn_2.11.1        fastmatch_1.1-4        DEoptimR_1.1-3        \n[142] readxl_1.4.3           ape_5.8               \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 1774.00\nuser 8784.42\nsys 20.96\npercent 496%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"SeuratClusterStats",desc:"Statistics of the clustering.",npages:1,report_toc:!0,order:17,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] grid      stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] clustree_0.5.1        ggraph_2.1.0          ComplexHeatmap_2.18.0\n [4] circlize_0.4.16       tidyseurat_0.8.0      ttservice_0.4.1      \n [7] ggrepel_0.9.5         ggprism_1.0.5         tibble_3.2.1         \n[10] Seurat_5.1.0          SeuratObject_5.0.2    sp_2.1-4             \n[13] ggplot2_3.5.1         tidyr_1.3.1           dplyr_1.1.4          \n[16] tidyselect_1.2.1      jsonlite_1.8.8        rlang_1.1.4          \n[19] logger_0.3.0         \n\nloaded via a namespace (and not attached):\n  [1] RColorBrewer_1.1-3     shape_1.4.6.1          magrittr_2.0.3        \n  [4] ggbeeswarm_0.7.2       spatstat.utils_3.0-5   farver_2.1.2          \n  [7] GlobalOptions_0.1.2    vctrs_0.6.5            ROCR_1.0-11           \n [10] spatstat.explore_3.2-6 htmltools_0.5.8.1      sctransform_0.4.1     \n [13] parallelly_1.38.0      KernSmooth_2.23-24     htmlwidgets_1.6.4     \n [16] ica_1.0-3              plyr_1.8.9             plotly_4.10.4         \n [19] zoo_1.8-12             igraph_2.0.3           mime_0.12             \n [22] lifecycle_1.0.4        iterators_1.0.14       pkgconfig_2.0.3       \n [25] Matrix_1.6-5           R6_2.5.1               fastmap_1.2.0         \n [28] clue_0.3-65            fitdistrplus_1.1-11    future_1.34.0         \n [31] shiny_1.9.1            digest_0.6.36          colorspace_2.1-1      \n [34] patchwork_1.2.0        S4Vectors_0.40.2       tensor_1.5            \n [37] RSpectra_0.16-1        irlba_2.3.5.1          labeling_0.4.3        \n [40] progressr_0.14.0       fansi_1.0.6            spatstat.sparse_3.1-0 \n [43] httr_1.4.7             polyclip_1.10-7        abind_1.4-5           \n [46] compiler_4.3.3         withr_3.0.1            doParallel_1.0.17     \n [49] viridis_0.6.5          fastDummies_1.7.3      ggforce_0.4.2         \n [52] MASS_7.3-60.0.1        rjson_0.2.21           tools_4.3.3           \n [55] vipor_0.4.7            lmtest_0.9-40          beeswarm_0.4.0        \n [58] httpuv_1.6.15          future.apply_1.11.2    goftest_1.2-3         \n [61] glue_1.7.0             nlme_3.1-165           promises_1.3.0        \n [64] Rtsne_0.17             cluster_2.1.6          reshape2_1.4.4        \n [67] generics_0.1.3         gtable_0.3.5           spatstat.data_3.1-2   \n [70] data.table_1.15.4      tidygraph_1.3.0        utf8_1.2.4            \n [73] BiocGenerics_0.48.1    spatstat.geom_3.2-9    RcppAnnoy_0.0.22      \n [76] RANN_2.6.1             foreach_1.5.2          pillar_1.9.0          \n [79] stringr_1.5.1          spam_2.10-0            RcppHNSW_0.6.0        \n [82] later_1.3.2            splines_4.3.3          tweenr_2.0.3          \n [85] lattice_0.22-6         survival_3.7-0         deldir_2.0-4          \n [88] miniUI_0.1.1.1         pbapply_1.7-2          gridExtra_2.3         \n [91] IRanges_2.36.0         scattermore_1.2        stats4_4.3.3          \n [94] graphlayouts_1.1.0     matrixStats_1.1.0      stringi_1.8.4         \n [97] lazyeval_0.2.2         codetools_0.2-20       cli_3.6.3             \n[100] uwot_0.1.16            xtable_1.8-4           reticulate_1.38.0     \n[103] munsell_0.5.1          Rcpp_1.0.13            globals_0.16.3        \n[106] spatstat.random_3.2-3  png_0.1-8              ggrastr_1.0.2         \n[109] parallel_4.3.3         ellipsis_0.3.2         dotCall64_1.1-1       \n[112] listenv_0.9.1          viridisLite_0.4.2      scales_1.3.0          \n[115] ggridges_0.5.6         leiden_0.4.3.1         purrr_1.0.2           \n[118] crayon_1.5.3           GetoptLong_1.0.5       cowplot_1.1.3         \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 15.10\nuser 23.39\nsys 0.59\npercent 158%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"Immunarch",desc:"Exploration of Single-cell and Bulk T-cell/Antibody Immune Repertoires",npages:3,report_toc:!0,order:18,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] circlize_0.4.16   tibble_3.2.1      glue_1.7.0        ggprism_1.0.5    \n [5] immunarch_0.9.1   patchwork_1.2.0   data.table_1.15.4 dtplyr_1.3.1     \n [9] ggplot2_3.5.1     tidyr_1.3.1       dplyr_1.1.4       jsonlite_1.8.8   \n[13] rlang_1.1.4       logger_0.3.0     \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22       shinythemes_1.2.0      splines_4.3.3         \n  [4] later_1.3.2            cellranger_1.1.0       polyclip_1.10-7       \n  [7] fastDummies_1.7.3      factoextra_1.0.7       lifecycle_1.0.4       \n [10] rstatix_0.7.2          doParallel_1.0.17      globals_0.16.3        \n [13] lattice_0.22-6         prabclus_2.3-3         MASS_7.3-60.0.1       \n [16] backports_1.5.0        magrittr_2.0.3         plotly_4.10.4         \n [19] rlist_0.4.6.2          httpuv_1.6.15          Seurat_5.1.0          \n [22] sctransform_0.4.1      spam_2.10-0            flexmix_2.3-19        \n [25] spatstat.sparse_3.1-0  sp_2.1-4               reticulate_1.38.0     \n [28] cowplot_1.1.3          pbapply_1.7-2          RColorBrewer_1.1-3    \n [31] abind_1.4-5            Rtsne_0.17             quadprog_1.5-8        \n [34] purrr_1.0.2            ggraph_2.1.0           nnet_7.3-19           \n [37] tweenr_2.0.3           ggrepel_0.9.5          irlba_2.3.5.1         \n [40] listenv_0.9.1          spatstat.utils_3.0-5   pheatmap_1.0.12       \n [43] goftest_1.2-3          RSpectra_0.16-1        spatstat.random_3.2-3 \n [46] fitdistrplus_1.1-11    parallelly_1.38.0      leiden_0.4.3.1        \n [49] codetools_0.2-20       ggforce_0.4.2          tidyselect_1.2.1      \n [52] shape_1.4.6.1          ggseqlogo_0.2          farver_2.1.2          \n [55] viridis_0.6.5          matrixStats_1.1.0      stats4_4.3.3          \n [58] spatstat.explore_3.2-6 tidygraph_1.3.0        progressr_0.14.0      \n [61] ggridges_0.5.6         ggalluvial_0.12.5      survival_3.7-0        \n [64] iterators_1.0.14       foreach_1.5.2          tools_4.3.3           \n [67] stringdist_0.9.12      ica_1.0-3              Rcpp_1.0.13           \n [70] gridExtra_2.3          withr_3.0.1            fastmap_1.2.0         \n [73] fansi_1.0.6            digest_0.6.36          R6_2.5.1              \n [76] mime_0.12              colorspace_2.1-1       scattermore_1.2       \n [79] tensor_1.5             spatstat.data_3.1-2    diptest_0.77-1        \n [82] UpSetR_1.4.0           utf8_1.2.4             generics_0.1.3        \n [85] robustbase_0.99-3      class_7.3-22           graphlayouts_1.1.0    \n [88] httr_1.4.7             htmlwidgets_1.6.4      uwot_0.1.16           \n [91] pkgconfig_2.0.3        gtable_0.3.5           modeltools_0.2-23     \n [94] lmtest_0.9-40          htmltools_0.5.8.1      carData_3.0-5         \n [97] dotCall64_1.1-1        SeuratObject_5.0.2     scales_1.3.0          \n[100] png_0.1-8              tzdb_0.4.0             reshape2_1.4.4        \n[103] uuid_1.2-1             nlme_3.1-165           zoo_1.8-12            \n[106] GlobalOptions_0.1.2    stringr_1.5.1          KernSmooth_2.23-24    \n[109] parallel_4.3.3         miniUI_0.1.1.1         pillar_1.9.0          \n[112] grid_4.3.3             vctrs_0.6.5            RANN_2.6.1            \n[115] promises_1.3.0         ggpubr_0.6.0           car_3.1-2             \n[118] xtable_1.8-4           cluster_2.1.6          readr_2.1.5           \n[121] cli_3.6.3              compiler_4.3.3         future.apply_1.11.2   \n[124] ggsignif_0.6.4         labeling_0.4.3         mclust_6.1.1          \n[127] plyr_1.8.9             stringi_1.8.4          deldir_2.0-4          \n[130] viridisLite_0.4.2      assertthat_0.2.1       munsell_0.5.1         \n[133] lazyeval_0.2.2         spatstat.geom_3.2-9    Matrix_1.6-5          \n[136] RcppHNSW_0.6.0         hms_1.1.3              future_1.34.0         \n[139] fpc_2.2-12             shiny_1.9.1            kernlab_0.9-32        \n[142] ROCR_1.0-11            igraph_2.0.3           broom_1.0.6           \n[145] phangorn_2.11.1        fastmatch_1.1-4        DEoptimR_1.1-3        \n[148] readxl_1.4.3           ape_5.8               \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 91.14\nuser 113.47\nsys 1.29\npercent 125%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"RadarPlots",desc:"Radar plots for cell proportion in different clusters.",npages:1,report_toc:!0,order:19,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ggprism_1.0.5      ggradar_0.2        ggplot2_3.5.1      tibble_3.2.1      \n [5] tidyr_1.3.1        dplyr_1.1.4        Seurat_5.1.0       SeuratObject_5.0.2\n [9] sp_2.1-4           jsonlite_1.8.8     rlang_1.1.4        logger_0.3.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.2-9   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] labeling_0.4.3         utf8_1.2.4             promises_1.3.0        \n [19] purrr_1.0.2            goftest_1.2-3          later_1.3.2           \n [22] spatstat.utils_3.0-5   irlba_2.3.5.1          parallel_4.3.3        \n [25] cluster_2.1.6          R6_2.5.1               ica_1.0-3             \n [28] stringi_1.8.4          RColorBrewer_1.1-3     spatstat.data_3.1-2   \n [31] reticulate_1.38.0      parallelly_1.38.0      lmtest_0.9-40         \n [34] scattermore_1.2        Rcpp_1.0.13            tensor_1.5            \n [37] future.apply_1.11.2    zoo_1.8-12             sctransform_0.4.1     \n [40] httpuv_1.6.15          Matrix_1.6-5           splines_4.3.3         \n [43] igraph_2.0.3           tidyselect_1.2.1       abind_1.4-5           \n [46] spatstat.random_3.2-3  codetools_0.2-20       miniUI_0.1.1.1        \n [49] spatstat.explore_3.2-6 listenv_0.9.1          lattice_0.22-6        \n [52] plyr_1.8.9             withr_3.0.1            shiny_1.9.1           \n [55] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n [58] fastDummies_1.7.3      survival_3.7-0         polyclip_1.10-7       \n [61] fitdistrplus_1.1-11    pillar_1.9.0           KernSmooth_2.23-24    \n [64] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n [67] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [70] xtable_1.8-4           glue_1.7.0             lazyeval_0.2.2        \n [73] tools_4.3.3            data.table_1.15.4      RSpectra_0.16-1       \n [76] RANN_2.6.1             leiden_0.4.3.1         dotCall64_1.1-1       \n [79] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n [82] nlme_3.1-165           patchwork_1.2.0        cli_3.6.3             \n [85] spatstat.sparse_3.1-0  spam_2.10-0            fansi_1.0.6           \n [88] viridisLite_0.4.2      uwot_0.1.16            gtable_0.3.5          \n [91] digest_0.6.36          progressr_0.14.0       ggrepel_0.9.5         \n [94] farver_2.1.2           htmlwidgets_1.6.4      htmltools_0.5.8.1     \n [97] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[100] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 7.79\nuser 16.36\nsys 0.40\npercent 215%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"CloneResidency",desc:"Identification of clone residency",npages:1,report_toc:!0,order:20,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ComplexUpset_1.3.3  ggVennDiagram_1.2.3 ggprism_1.0.5      \n [4] ggplot2_3.5.1       ggnewscale_0.5.0    glue_1.7.0         \n [7] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[10] jsonlite_1.8.8      rlang_1.1.4         logger_0.3.0       \n\nloaded via a namespace (and not attached):\n  [1] DBI_1.2.3              deldir_2.0-4           pbapply_1.7-2         \n  [4] gridExtra_2.3          magrittr_2.0.3         RcppAnnoy_0.0.22      \n  [7] e1071_1.7-14           spatstat.geom_3.2-9    matrixStats_1.1.0     \n [10] ggridges_0.5.6         compiler_4.3.3         png_0.1-8             \n [13] vctrs_0.6.5            reshape2_1.4.4         stringr_1.5.1         \n [16] pkgconfig_2.0.3        fastmap_1.2.0          labeling_0.4.3        \n [19] utf8_1.2.4             promises_1.3.0         purrr_1.0.2           \n [22] RVenn_1.1.0            goftest_1.2-3          later_1.3.2           \n [25] spatstat.utils_3.0-5   irlba_2.3.5.1          parallel_4.3.3        \n [28] cluster_2.1.6          R6_2.5.1               ica_1.0-3             \n [31] stringi_1.8.4          RColorBrewer_1.1-3     spatstat.data_3.1-2   \n [34] reticulate_1.38.0      parallelly_1.38.0      lmtest_0.9-40         \n [37] scattermore_1.2        Rcpp_1.0.13            tensor_1.5            \n [40] future.apply_1.11.2    zoo_1.8-12             sctransform_0.4.1     \n [43] httpuv_1.6.15          Matrix_1.6-5           splines_4.3.3         \n [46] igraph_2.0.3           tidyselect_1.2.1       abind_1.4-5           \n [49] spatstat.random_3.2-3  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.2-6 listenv_0.9.1          lattice_0.22-6        \n [55] plyr_1.8.9             shiny_1.9.1            withr_3.0.1           \n [58] ROCR_1.0-11            Rtsne_0.17             sf_1.0-16             \n [61] future_1.34.0          fastDummies_1.7.3      survival_3.7-0        \n [64] units_0.8-5            proxy_0.4-27           polyclip_1.10-7       \n [67] fitdistrplus_1.1-11    pillar_1.9.0           Seurat_5.1.0          \n [70] KernSmooth_2.23-24     plotly_4.10.4          generics_0.1.3        \n [73] RcppHNSW_0.6.0         sp_2.1-4               munsell_0.5.1         \n [76] scales_1.3.0           globals_0.16.3         xtable_1.8-4          \n [79] class_7.3-22           lazyeval_0.2.2         tools_4.3.3           \n [82] data.table_1.15.4      RSpectra_0.16-1        RANN_2.6.1            \n [85] leiden_0.4.3.1         dotCall64_1.1-1        cowplot_1.1.3         \n [88] grid_4.3.3             colorspace_2.1-1       nlme_3.1-165          \n [91] patchwork_1.2.0        cli_3.6.3              spatstat.sparse_3.1-0 \n [94] spam_2.10-0            fansi_1.0.6            viridisLite_0.4.2     \n [97] uwot_0.1.16            gtable_0.3.5           digest_0.6.36         \n[100] classInt_0.4-10        progressr_0.14.0       ggrepel_0.9.5         \n[103] farver_2.1.2           htmlwidgets_1.6.4      SeuratObject_5.0.2    \n[106] htmltools_0.5.8.1      lifecycle_1.0.4        httr_1.4.7            \n[109] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 16.51\nuser 24.80\nsys 0.51\npercent 153%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ScFGSEA",desc:"Gene set enrichment analysis for cells in different groups using `fgsea`",npages:2,report_toc:!0,order:21,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] fgsea_1.28.0       data.table_1.15.4  matrixStats_1.1.0  tidyseurat_0.8.0  \n [5] ttservice_0.4.1    Seurat_5.1.0       SeuratObject_5.0.2 sp_2.1-4          \n [9] tidyr_1.3.1        tidyselect_1.2.1   tibble_3.2.1       dplyr_1.1.4       \n[13] ggplot2_3.5.1      jsonlite_1.8.8     rlang_1.1.4        logger_0.3.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       ggridges_0.5.6        \n  [7] compiler_4.3.3         spatstat.geom_3.2-9    png_0.1-8             \n [10] vctrs_0.6.5            reshape2_1.4.4         stringr_1.5.1         \n [13] pkgconfig_2.0.3        fastmap_1.2.0          ellipsis_0.3.2        \n [16] labeling_0.4.3         utf8_1.2.4             promises_1.3.0        \n [19] purrr_1.0.2            goftest_1.2-3          later_1.3.2           \n [22] BiocParallel_1.36.0    spatstat.utils_3.0-5   irlba_2.3.5.1         \n [25] parallel_4.3.3         cluster_2.1.6          R6_2.5.1              \n [28] ica_1.0-3              stringi_1.8.4          RColorBrewer_1.1-3    \n [31] spatstat.data_3.1-2    reticulate_1.38.0      parallelly_1.38.0     \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.13           \n [37] tensor_1.5             future.apply_1.11.2    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [43] splines_4.3.3          igraph_2.0.3           abind_1.4-5           \n [46] spatstat.random_3.2-3  codetools_0.2-20       miniUI_0.1.1.1        \n [49] spatstat.explore_3.2-6 listenv_0.9.1          lattice_0.22-6        \n [52] plyr_1.8.9             shiny_1.9.1            withr_3.0.1           \n [55] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n [58] fastDummies_1.7.3      survival_3.7-0         polyclip_1.10-7       \n [61] fitdistrplus_1.1-11    pillar_1.9.0           KernSmooth_2.23-24    \n [64] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n [67] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [70] xtable_1.8-4           glue_1.7.0             lazyeval_0.2.2        \n [73] tools_4.3.3            RSpectra_0.16-1        RANN_2.6.1            \n [76] leiden_0.4.3.1         dotCall64_1.1-1        fastmatch_1.1-4       \n [79] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n [82] nlme_3.1-165           patchwork_1.2.0        cli_3.6.3             \n [85] spatstat.sparse_3.1-0  spam_2.10-0            fansi_1.0.6           \n [88] viridisLite_0.4.2      uwot_0.1.16            gtable_0.3.5          \n [91] digest_0.6.36          progressr_0.14.0       ggrepel_0.9.5         \n [94] farver_2.1.2           htmlwidgets_1.6.4      htmltools_0.5.8.1     \n [97] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[100] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 96.79\nuser 103.64\nsys 1.43\npercent 108%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"MarkersFinder",desc:"Find markers between different groups of cells",npages:1,report_toc:!0,order:22,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] UpSetR_1.4.0        ggVennDiagram_1.2.3 tidyseurat_0.8.0   \n [4] ttservice_0.4.1     future_1.34.0       ggrepel_0.9.5      \n [7] ggprism_1.0.5       ggplot2_3.5.1       enrichR_3.2        \n[10] Seurat_5.1.0        SeuratObject_5.0.2  sp_2.1-4           \n[13] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[16] tidyselect_1.2.1    digest_0.6.36       jsonlite_1.8.8     \n[19] rlang_1.1.4         logger_0.3.0       \n\nloaded via a namespace (and not attached):\n  [1] RColorBrewer_1.1-3     magrittr_2.0.3         spatstat.utils_3.0-5  \n  [4] farver_2.1.2           vctrs_0.6.5            ROCR_1.0-11           \n  [7] spatstat.explore_3.2-6 htmltools_0.5.8.1      curl_5.2.1            \n [10] sctransform_0.4.1      parallelly_1.38.0      KernSmooth_2.23-24    \n [13] htmlwidgets_1.6.4      ica_1.0-3              plyr_1.8.9            \n [16] plotly_4.10.4          zoo_1.8-12             igraph_2.0.3          \n [19] mime_0.12              lifecycle_1.0.4        pkgconfig_2.0.3       \n [22] Matrix_1.6-5           R6_2.5.1               fastmap_1.2.0         \n [25] fitdistrplus_1.1-11    shiny_1.9.1            colorspace_2.1-1      \n [28] patchwork_1.2.0        tensor_1.5             RSpectra_0.16-1       \n [31] irlba_2.3.5.1          WriteXLS_6.7.0         labeling_0.4.3        \n [34] progressr_0.14.0       fansi_1.0.6            spatstat.sparse_3.1-0 \n [37] httr_1.4.7             polyclip_1.10-7        abind_1.4-5           \n [40] compiler_4.3.3         withr_3.0.1            fastDummies_1.7.3     \n [43] MASS_7.3-60.0.1        rjson_0.2.21           tools_4.3.3           \n [46] lmtest_0.9-40          httpuv_1.6.15          future.apply_1.11.2   \n [49] goftest_1.2-3          glue_1.7.0             nlme_3.1-165          \n [52] promises_1.3.0         grid_4.3.3             Rtsne_0.17            \n [55] cluster_2.1.6          reshape2_1.4.4         generics_0.1.3        \n [58] gtable_0.3.5           spatstat.data_3.1-2    data.table_1.15.4     \n [61] utf8_1.2.4             spatstat.geom_3.2-9    RcppAnnoy_0.0.22      \n [64] RANN_2.6.1             pillar_1.9.0           stringr_1.5.1         \n [67] spam_2.10-0            RcppHNSW_0.6.0         limma_3.58.1          \n [70] later_1.3.2            splines_4.3.3          lattice_0.22-6        \n [73] survival_3.7-0         deldir_2.0-4           miniUI_0.1.1.1        \n [76] pbapply_1.7-2          gridExtra_2.3          scattermore_1.2       \n [79] statmod_1.5.0          matrixStats_1.1.0      stringi_1.8.4         \n [82] lazyeval_0.2.2         codetools_0.2-20       RVenn_1.1.0           \n [85] cli_3.6.3              uwot_0.1.16            xtable_1.8-4          \n [88] reticulate_1.38.0      munsell_0.5.1          Rcpp_1.0.13           \n [91] globals_0.16.3         spatstat.random_3.2-3  png_0.1-8             \n [94] parallel_4.3.3         ellipsis_0.3.2         presto_1.0.0          \n [97] dotCall64_1.1-1        listenv_0.9.1          viridisLite_0.4.2     \n[100] scales_1.3.0           ggridges_0.5.6         leiden_0.4.3.1        \n[103] purrr_1.0.2            cowplot_1.1.3         \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 15.26\nuser 20.73\nsys 0.60\npercent 139%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"MetaMarkers",desc:"Find markers between three or more groups of cells, using one-way ANOVA or Kruskal-Wallis test.",npages:1,report_toc:!0,order:23},{name:"CellsDistribution",desc:"Distribution of cells (i.e. in a TCR clone) from different groups for each cluster",npages:1,report_toc:!0,order:24},{name:"CDR3AAPhyschem",desc:"CDR3 AA physicochemical feature analysis",npages:1,report_toc:!0,order:25},{name:"ScrnaMetabolicLandscape",desc:"Metabolic landscape analysis for scRNA-seq data",order:28,procs:[{name:"MetabolicPathwayActivity",desc:"This process calculates the pathway activities in different groups and subsets.",npages:1,report_toc:!0,order:28},{name:"MetabolicFeatures",desc:"This process performs enrichment analysis for the metabolic pathways for each group in each subset.",npages:1,report_toc:!0,order:29},{name:"MetabolicFeaturesIntraSubset",desc:"Intra-subset metabolic features - Enrichment analysis in details",npages:1,report_toc:!0,order:30},{name:"MetabolicPathwayHeterogeneity",desc:"Calculate Metabolic Pathway heterogeneity.",npages:1,report_toc:!0,order:31}]}]};function hn(e){let a,t;return a=new Ha({props:{dir:"up"}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function $n(e){let a,t;return a=new Ha({props:{dir:"down"}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){p(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function Tn(e){let a,t,n,_,l,d=e[1]>0&&hn();t=new On({});let u=e[1]<e[3].npages-1&&$n();return{c(){d&&d.c(),a=f(),g(t.$$.fragment),n=f(),u&&u.c(),_=m()},m(e,i){d&&d.m(e,i),s(e,a,i),P(t,e,i),s(e,n,i),u&&u.m(e,i),s(e,_,i),l=!0},p(e,t){e[1]>0?d?2&t&&i(d,1):(d=hn(),d.c(),i(d,1),d.m(a.parentNode,a)):d&&(r(),p(d,1,1,(()=>{d=null})),o()),e[1]<e[3].npages-1?u?10&t&&i(u,1):(u=$n(),u.c(),i(u,1),u.m(_.parentNode,_)):u&&(r(),p(u,1,1,(()=>{u=null})),o())},i(e){l||(i(d),i(t.$$.fragment,e),i(u),l=!0)},o(e){p(d),p(t.$$.fragment,e),p(u),l=!1},d(e){e&&(c(a),c(n),c(_)),d&&d.d(e),A(t,e),u&&u.d(e)}}}function yn(e){let a,t,n;function _(a){e[4](a)}let l={logo:e[0],logotext:e[3].desc,versions:Rn.versions,entries:Rn.entries,pipeline_name:Rn.pipeline.name,report_toc:e[3].report_toc,runinfo:e[3].runinfo,toc:Sn,page:e[1],$$slots:{default:[Tn]},$$scope:{ctx:e}};return void 0!==e[2]&&(l.isSideNavOpen=e[2]),a=new Ua({props:l}),b.push((()=>C(a,"isSideNavOpen",_))),{c(){g(a.$$.fragment)},m(e,t){P(a,e,t),n=!0},p(e,[n]){const _={};1&n&&(_.logo=e[0]),8&n&&(_.logotext=e[3].desc),8&n&&(_.report_toc=e[3].report_toc),8&n&&(_.runinfo=e[3].runinfo),2&n&&(_.page=e[1]),42&n&&(_.$$scope={dirty:n,ctx:e}),!t&&4&n&&(t=!0,_.isSideNavOpen=e[2],O((()=>t=!1))),a.$set(_)},i(e){n||(i(a.$$.fragment,e),n=!0)},o(e){p(a.$$.fragment,e),n=!1},d(e){A(a,e)}}}function Ln(e,a,t){let n,{name:_}=a,{page:l=0}=a,d=!1;for(const e of Rn.entries)if(e.procs){for(const a of e.procs)if(a.name==_){n=a;break}}else if(e.name==_){n=e;break}return v((()=>{const e=new URL(window.location.href);let a;e.hash.length>1&&(a=document.getElementById(`${e.hash.slice(1)}`)),a&&a.scrollIntoView({block:"start",behavior:"smooth"})})),e.$$set=e=>{"name"in e&&t(0,_=e.name),"page"in e&&t(1,l=e.page)},[_,l,d,n,function(e){d=e,t(2,d)}]}e("default",new class extends a{constructor(e){super(),t(this,e,Ln,yn,n,{name:0,page:1})}}({target:document.body,props:{name:"MarkersFinder",page:0}}))}}}));
