System.register(["./vendor/svelte-C_apNiMW.js","./vendor/carbon-components-svelte-CB8VNYQX.js","./vendor/carbon-icons-svelte-DkM4cpjm.js"],(function(e){"use strict";var a,t,n,l,_,s,d,i,r,o,c,p,u,v,m,g,P,A,f,b,C,O,S,h,R,$,T,y,L,j,x,M,B,G,F,w,N,U,D,E,H,I,k,V,z,q,K,Y,W,X,Q,Z,J,ee,ae,te,ne,le,_e,se,de,ie,re,oe,ce,pe,ue,ve,me,ge,Pe,Ae,fe,be,Ce,Oe,Se,he,Re,$e,Te,ye,Le,je,xe,Me,Be,Ge,Fe,we,Ne,Ue,De,Ee,He,Ie,ke,Ve,ze,qe,Ke,Ye,We,Xe,Qe,Ze;return{setters:[function(e){a=e.S,t=e.i,n=e.s,l=e.a1,_=e.l,s=e.O,d=e.a,i=e.b,r=e.g,o=e.c,c=e.t,p=e.d,u=e.a5,v=e.B,m=e.e,g=e.G,P=e.H,A=e.I,f=e.y,b=e.K,C=e.T,O=e.U,S=e.L,h=e.D,R=e.M,$=e.z,T=e.Q,y=e.k,L=e.u,j=e.n,x=e.o,M=e.h,B=e.p,G=e.a0,F=e.f,w=e.j,N=e.A,U=e.m,D=e.q,E=e.aa,H=e.ab,I=e.x,k=e.W,V=e.ac,z=e.R,q=e.$,K=e.v,Y=e.Y,W=e.Z,X=e.F,Q=e.ad,Z=e.w,J=e.J},function(e){ee=e.S,ae=e.a,te=e.b,ne=e.H,le=e.c,_e=e.T,se=e.d,de=e.e,ie=e.f,re=e.g,oe=e.h,ce=e.M,pe=e.I,ue=e.i,ve=e.j,me=e.k,ge=e.C,Pe=e.l,Ae=e.m,fe=e.G,be=e.R,Ce=e.n,Oe=e.B,Se=e.o,he=e.A,Re=e.p,$e=e.D,Te=e.P,ye=e.q,Le=e.r,je=e.s,xe=e.t,Me=e.u},function(e){Be=e.C,Ge=e.a,Fe=e.D,we=e.A,Ne=e.S,Ue=e.I,De=e.b,Ee=e.c,He=e.V,Ie=e.d,ke=e.T,Ve=e.G,ze=e.J,qe=e.P,Ke=e.e,Ye=e.f,We=e.O,Xe=e.g,Qe=e.h,Ze=e.i}],execute:function(){function Je(e,a,t){const n=e.slice();return n[4]=a[t],n}function ea(e,a,t){const n=e.slice();return n[7]=a[t],n}function aa(e){let a,t;return a=new ee({props:{href:`${e[1](e[4].page)}#${e[4].slug}`,class:e[2](e[4].page),title:e[4].text,text:e[4].text,expanded:!0,$$slots:{default:[la]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};1&t&&(n.href=`${e[1](e[4].page)}#${e[4].slug}`),1&t&&(n.class=e[2](e[4].page)),1&t&&(n.title=e[4].text),1&t&&(n.text=e[4].text),1025&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function ta(e){let a,t;return a=new ae({props:{href:`${e[1](e[4].page)}#${e[4].slug}`,class:e[2](e[4].page),title:e[4].text,text:e[4].text}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};1&t&&(n.href=`${e[1](e[4].page)}#${e[4].slug}`),1&t&&(n.class=e[2](e[4].page)),1&t&&(n.title=e[4].text),1&t&&(n.text=e[4].text),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function na(e){let a,t;return a=new te({props:{href:`${e[1](e[7].page)}#${e[7].slug}`,class:e[2](e[7].page),title:e[7].text,text:e[7].text}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};1&t&&(n.href=`${e[1](e[7].page)}#${e[7].slug}`),1&t&&(n.class=e[2](e[7].page)),1&t&&(n.title=e[7].text),1&t&&(n.text=e[7].text),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function la(e){let a,t,n=l(e[4].children),_=[];for(let a=0;a<n.length;a+=1)_[a]=na(ea(e,n,a));const s=e=>c(_[e],1,1,(()=>{_[e]=null}));return{c(){for(let e=0;e<_.length;e+=1)_[e].c();a=f()},m(e,n){for(let a=0;a<_.length;a+=1)_[a]&&_[a].m(e,n);d(e,a,n),t=!0},p(e,t){if(7&t){let d;for(n=l(e[4].children),d=0;d<n.length;d+=1){const l=ea(e,n,d);_[d]?(_[d].p(l,t),i(_[d],1)):(_[d]=na(l),_[d].c(),i(_[d],1),_[d].m(a.parentNode,a))}for(r(),d=n.length;d<_.length;d+=1)s(d);o()}},i(e){if(!t){for(let e=0;e<n.length;e+=1)i(_[e]);t=!0}},o(e){_=_.filter(Boolean);for(let e=0;e<_.length;e+=1)c(_[e]);t=!1},d(e){e&&p(a),u(_,e)}}}function _a(e){let a,t,n,l;const _=[ta,aa],s=[];function u(e,a){return 0==e[4].children.length?0:1}return a=u(e),t=s[a]=_[a](e),{c(){t.c(),n=m()},m(e,t){s[a].m(e,t),d(e,n,t),l=!0},p(e,l){let d=a;a=u(e),a===d?s[a].p(e,l):(r(),c(s[d],1,1,(()=>{s[d]=null})),o(),t=s[a],t?t.p(e,l):(t=s[a]=_[a](e),t.c()),i(t,1),t.m(n.parentNode,n))},i(e){l||(i(t),l=!0)},o(e){c(t),l=!1},d(e){e&&p(n),s[a].d(e)}}}function sa(e){let a,t,n=l(e[0]),v=[];for(let a=0;a<n.length;a+=1)v[a]=_a(Je(e,n,a));const m=e=>c(v[e],1,1,(()=>{v[e]=null}));return{c(){a=_("ul");for(let e=0;e<v.length;e+=1)v[e].c();s(a,"class","bx--side-nav__items pipen-report-sidenav svelte-12gpq3")},m(e,n){d(e,a,n);for(let e=0;e<v.length;e+=1)v[e]&&v[e].m(a,null);t=!0},p(e,[t]){if(7&t){let _;for(n=l(e[0]),_=0;_<n.length;_+=1){const l=Je(e,n,_);v[_]?(v[_].p(l,t),i(v[_],1)):(v[_]=_a(l),v[_].c(),i(v[_],1),v[_].m(a,null))}for(r(),_=n.length;_<v.length;_+=1)m(_);o()}},i(e){if(!t){for(let e=0;e<n.length;e+=1)i(v[e]);t=!0}},o(e){v=v.filter(Boolean);for(let e=0;e<v.length;e+=1)c(v[e]);t=!1},d(e){e&&p(a),u(v,e)}}}function da(e,a,t){let{toc:n}=a,{page:l}=a;return v((()=>{let e,a=new URL(window.location.href);e=a.hash.length>0?document.querySelector(`[href="${a.hash}"]`):document.querySelector(".toc-current-page"),e.scrollIntoView({behavior:"smooth"})})),e.$$set=e=>{"toc"in e&&t(0,n=e.toc),"page"in e&&t(3,l=e.page)},[n,e=>{if(l===e)return"";let a=new URL(window.location.href);return a.hash="",a.searchParams.set("page",e),a.toString()},e=>e!=l?"toc-noncurrent-page":"toc-current-page",l]}class ia extends a{constructor(e){super(),t(this,e,da,sa,n,{toc:0,page:3})}}function ra(e,a,t){const n=e.slice();return n[7]=a[t],n}function oa(e,a,t){const n=e.slice();return n[10]=a[t],n}function ca(e){let a,t;return a=new de({props:{isOpen:ja,transition:e[5],$$slots:{default:[Sa]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};8197&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function pa(e){let a;return{c(){a=S("Go to home:")},m(e,t){d(e,a,t)},d(e){e&&p(a)}}}function ua(e){let a;return{c(){a=S(e[0])},m(e,t){d(e,a,t)},p(e,t){1&t&&R(a,e[0])},d(e){e&&p(a)}}}function va(e){let a;return{c(){a=S("Switch to process:")},m(e,t){d(e,a,t)},d(e){e&&p(a)}}}function ma(e){let a,t;return a=new oe({props:{title:e[7].name,href:`?proc=${e[7].name}`,$$slots:{default:[Pa]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};4&t&&(n.title=e[7].name),4&t&&(n.href=`?proc=${e[7].name}`),8196&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function ga(e){let a,t,n,_;a=new oe({props:{class:"procgroup",$$slots:{default:[Aa]},$$scope:{ctx:e}}});let s=l(e[7].procs),v=[];for(let a=0;a<s.length;a+=1)v[a]=ba(oa(e,s,a));const b=e=>c(v[e],1,1,(()=>{v[e]=null}));return{c(){g(a.$$.fragment),t=f();for(let e=0;e<v.length;e+=1)v[e].c();n=m()},m(e,l){P(a,e,l),d(e,t,l);for(let a=0;a<v.length;a+=1)v[a]&&v[a].m(e,l);d(e,n,l),_=!0},p(e,t){const _={};if(8196&t&&(_.$$scope={dirty:t,ctx:e}),a.$set(_),4&t){let a;for(s=l(e[7].procs),a=0;a<s.length;a+=1){const l=oa(e,s,a);v[a]?(v[a].p(l,t),i(v[a],1)):(v[a]=ba(l),v[a].c(),i(v[a],1),v[a].m(n.parentNode,n))}for(r(),a=s.length;a<v.length;a+=1)b(a);o()}},i(e){if(!_){i(a.$$.fragment,e);for(let e=0;e<s.length;e+=1)i(v[e]);_=!0}},o(e){c(a.$$.fragment,e),v=v.filter(Boolean);for(let e=0;e<v.length;e+=1)c(v[e]);_=!1},d(e){e&&(p(t),p(n)),A(a,e),u(v,e)}}}function Pa(e){let a,t,n=e[7].name+"";return{c(){a=S(n),t=f()},m(e,n){d(e,a,n),d(e,t,n)},p(e,t){4&t&&n!==(n=e[7].name+"")&&R(a,n)},d(e){e&&(p(a),p(t))}}}function Aa(e){let a,t=e[7].name+"";return{c(){a=S(t)},m(e,t){d(e,a,t)},p(e,n){4&n&&t!==(t=e[7].name+"")&&R(a,t)},d(e){e&&p(a)}}}function fa(e){let a,t,n,l=e[10].name+"";return{c(){a=S("- "),t=S(l),n=f()},m(e,l){d(e,a,l),d(e,t,l),d(e,n,l)},p(e,a){4&a&&l!==(l=e[10].name+"")&&R(t,l)},d(e){e&&(p(a),p(t),p(n))}}}function ba(e){let a,t;return a=new oe({props:{title:e[10].name,class:"procgroup-proc",href:`?proc=${e[10].name}`,$$slots:{default:[fa]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};4&t&&(n.title=e[10].name),4&t&&(n.href=`?proc=${e[10].name}`),8196&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function Ca(e){let a,t,n,l;const _=[ga,ma],s=[];function u(e,a){return e[7].procs?0:1}return a=u(e),t=s[a]=_[a](e),{c(){t.c(),n=m()},m(e,t){s[a].m(e,t),d(e,n,t),l=!0},p(e,l){let d=a;a=u(e),a===d?s[a].p(e,l):(r(),c(s[d],1,1,(()=>{s[d]=null})),o(),t=s[a],t?t.p(e,l):(t=s[a]=_[a](e),t.c()),i(t,1),t.m(n.parentNode,n))},i(e){l||(i(t),l=!0)},o(e){c(t),l=!1},d(e){e&&p(n),s[a].d(e)}}}function Oa(e){let a,t,n,_,s,v,b,C;a=new re({props:{$$slots:{default:[pa]},$$scope:{ctx:e}}}),n=new oe({props:{href:"index.html",$$slots:{default:[ua]},$$scope:{ctx:e}}}),s=new re({props:{$$slots:{default:[va]},$$scope:{ctx:e}}});let O=l(e[2]),S=[];for(let a=0;a<O.length;a+=1)S[a]=Ca(ra(e,O,a));const h=e=>c(S[e],1,1,(()=>{S[e]=null}));return{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment),_=f(),g(s.$$.fragment),v=f();for(let e=0;e<S.length;e+=1)S[e].c();b=m()},m(e,l){P(a,e,l),d(e,t,l),P(n,e,l),d(e,_,l),P(s,e,l),d(e,v,l);for(let a=0;a<S.length;a+=1)S[a]&&S[a].m(e,l);d(e,b,l),C=!0},p(e,t){const _={};8192&t&&(_.$$scope={dirty:t,ctx:e}),a.$set(_);const d={};8193&t&&(d.$$scope={dirty:t,ctx:e}),n.$set(d);const c={};if(8192&t&&(c.$$scope={dirty:t,ctx:e}),s.$set(c),4&t){let a;for(O=l(e[2]),a=0;a<O.length;a+=1){const n=ra(e,O,a);S[a]?(S[a].p(n,t),i(S[a],1)):(S[a]=Ca(n),S[a].c(),i(S[a],1),S[a].m(b.parentNode,b))}for(r(),a=O.length;a<S.length;a+=1)h(a);o()}},i(e){if(!C){i(a.$$.fragment,e),i(n.$$.fragment,e),i(s.$$.fragment,e);for(let e=0;e<O.length;e+=1)i(S[e]);C=!0}},o(e){c(a.$$.fragment,e),c(n.$$.fragment,e),c(s.$$.fragment,e),S=S.filter(Boolean);for(let e=0;e<S.length;e+=1)c(S[e]);C=!1},d(e){e&&(p(t),p(_),p(v),p(b)),A(a,e),A(n,e),A(s,e),u(S,e)}}}function Sa(e){let a,t;return a=new ie({props:{$$slots:{default:[Oa]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};8197&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function ha(e){let a,t,n,l;a=new se({props:{render:"select",theme:"g10",persist:!0,persistKey:"__pipen-report-theme",select:{themes:["g10","white","g80","g90","g100"],labelText:"Theme",inline:!0}}});let _=e[2].length>0&&ca(e);return{c(){g(a.$$.fragment),t=f(),_&&_.c(),n=m()},m(e,s){P(a,e,s),d(e,t,s),_&&_.m(e,s),d(e,n,s),l=!0},p(e,a){e[2].length>0?_?(_.p(e,a),4&a&&i(_,1)):(_=ca(e),_.c(),i(_,1),_.m(n.parentNode,n)):_&&(r(),c(_,1,1,(()=>{_=null})),o())},i(e){l||(i(a.$$.fragment,e),i(_),l=!0)},o(e){c(a.$$.fragment,e),c(_),l=!1},d(e){e&&(p(t),p(n)),A(a,e),_&&_.d(e)}}}function Ra(e){let a,t;return a=new le({props:{class:"header-util",$$slots:{default:[ha]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};8197&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function $a(e){let a;return{c(){a=S(e[4])},m(e,t){d(e,a,t)},p(e,t){16&t&&R(a,e[4])},d(e){e&&p(a)}}}function Ta(e){let a,t,n,l,s=null!==e[4]&&$a(e);return{c(){a=_("h1"),t=S(e[3]),n=f(),s&&s.c(),l=m()},m(e,_){d(e,a,_),h(a,t),d(e,n,_),s&&s.m(e,_),d(e,l,_)},p(e,a){8&a&&R(t,e[3]),null!==e[4]?s?s.p(e,a):(s=$a(e),s.c(),s.m(l.parentNode,l)):s&&(s.d(1),s=null)},d(e){e&&(p(a),p(n),p(l)),s&&s.d(e)}}}function ya(e){let a,t;return a=new _e({props:{class:"header-tile",slot:"platform",$$slots:{default:[Ta]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};8216&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function La(e){let a,t,n;function l(a){e[6](a)}let _={$$slots:{platform:[ya],default:[Ra]},$$scope:{ctx:e}};return void 0!==e[1]&&(_.isSideNavOpen=e[1]),a=new ne({props:_}),b.push((()=>C(a,"isSideNavOpen",l))),{c(){g(a.$$.fragment)},m(e,t){P(a,e,t),n=!0},p(e,[n]){const l={};8221&n&&(l.$$scope={dirty:n,ctx:e}),!t&&2&n&&(t=!0,l.isSideNavOpen=e[1],O((()=>t=!1))),a.$set(l)},i(e){n||(i(a.$$.fragment,e),n=!0)},o(e){c(a.$$.fragment,e),n=!1},d(e){A(a,e)}}}let ja=!1;function xa(e,a,t){let{entries:n}=a,{logo:l}=a,{logotext:_=null}=a,{pipeline_name:s}=a,{isSideNavOpen:d}=a;return s=s||l,e.$$set=e=>{"entries"in e&&t(2,n=e.entries),"logo"in e&&t(3,l=e.logo),"logotext"in e&&t(4,_=e.logotext),"pipeline_name"in e&&t(0,s=e.pipeline_name),"isSideNavOpen"in e&&t(1,d=e.isSideNavOpen)},[s,d,n,l,_,{duration:200},function(e){d=e,t(1,d)}]}class Ma extends a{constructor(e){super(),t(this,e,xa,La,n,{entries:2,logo:3,logotext:4,pipeline_name:0,isSideNavOpen:1})}}function Ba(e){let a,t,n;function l(a){e[3](a)}let _={modalHeading:"Running Information",passiveModal:!0,hasScrollingContent:!0,class:"runinfo",$$slots:{default:[Da]},$$scope:{ctx:e}};return void 0!==e[2]&&(_.open=e[2]),a=new ce({props:_}),b.push((()=>C(a,"open",l))),{c(){g(a.$$.fragment)},m(e,t){P(a,e,t),n=!0},p(e,n){const l={};34&n&&(l.$$scope={dirty:n,ctx:e}),!t&&4&n&&(t=!0,l.open=e[2],O((()=>t=!1))),a.$set(l)},i(e){n||(i(a.$$.fragment,e),n=!0)},o(e){c(a.$$.fragment,e),n=!1},d(e){A(a,e)}}}function Ga(e){let a,t,n,l,_,s;return a=new ve({props:{label:"Session Information"}}),n=new ve({props:{label:"Time Spent"}}),_=new ve({props:{label:"Device Information"}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment),l=f(),g(_.$$.fragment)},m(e,i){P(a,e,i),d(e,t,i),P(n,e,i),d(e,l,i),P(_,e,i),s=!0},p:T,i(e){s||(i(a.$$.fragment,e),i(n.$$.fragment,e),i(_.$$.fragment,e),s=!0)},o(e){c(a.$$.fragment,e),c(n.$$.fragment,e),c(_.$$.fragment,e),s=!1},d(e){e&&(p(t),p(l)),A(a,e),A(n,e),A(_,e)}}}function Fa(e){let a,t;return a=new ge({props:{type:"multi",code:e[1].session}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};2&t&&(n.code=e[1].session),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function wa(e){let a,t;return a=new ge({props:{type:"multi",code:e[1].time}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};2&t&&(n.code=e[1].time),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function Na(e){let a,t;return a=new ge({props:{type:"multi",code:e[1].device}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};2&t&&(n.code=e[1].device),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function Ua(e){let a,t,n,l,_,s;return a=new me({props:{$$slots:{default:[Fa]},$$scope:{ctx:e}}}),n=new me({props:{$$slots:{default:[wa]},$$scope:{ctx:e}}}),_=new me({props:{$$slots:{default:[Na]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment),l=f(),g(_.$$.fragment)},m(e,i){P(a,e,i),d(e,t,i),P(n,e,i),d(e,l,i),P(_,e,i),s=!0},p(e,t){const l={};34&t&&(l.$$scope={dirty:t,ctx:e}),a.$set(l);const s={};34&t&&(s.$$scope={dirty:t,ctx:e}),n.$set(s);const d={};34&t&&(d.$$scope={dirty:t,ctx:e}),_.$set(d)},i(e){s||(i(a.$$.fragment,e),i(n.$$.fragment,e),i(_.$$.fragment,e),s=!0)},o(e){c(a.$$.fragment,e),c(n.$$.fragment,e),c(_.$$.fragment,e),s=!1},d(e){e&&(p(t),p(l)),A(a,e),A(n,e),A(_,e)}}}function Da(e){let a,t,n,l;return a=new pe({props:{kind:"warning",lowContrast:!0,hideCloseButton:!0,title:"Showing the first job only. Check the workdir for information of other jobs if any."}}),n=new ue({props:{$$slots:{content:[Ua],default:[Ga]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment)},m(e,_){P(a,e,_),d(e,t,_),P(n,e,_),l=!0},p(e,a){const t={};34&a&&(t.$$scope={dirty:a,ctx:e}),n.$set(t)},i(e){l||(i(a.$$.fragment,e),i(n.$$.fragment,e),l=!0)},o(e){c(a.$$.fragment,e),c(n.$$.fragment,e),l=!1},d(e){e&&p(t),A(a,e),A(n,e)}}}function Ea(e){let a,t,n;return{c(){a=_("a"),a.textContent="> Running Information",s(a,"href","javascript:void(0);")},m(l,_){d(l,a,_),t||(n=$(a,"click",e[4]),t=!0)},p:T,d(e){e&&p(a),t=!1,n()}}}function Ha(e){let a,t,n,l,u,v,m,g,P=e[1]&&Ba(e),A=e[1]&&Ea(e);return{c(){P&&P.c(),a=f(),t=_("div"),n=_("div"),A&&A.c(),l=f(),u=_("div"),v=S("Powered by "),m=_("span"),s(n,"class","runinfo"),s(m,"class","pipen-report-footer-versions"),s(u,"class","versions"),s(t,"class","pipen-report-footer bx--grid")},m(_,s){P&&P.m(_,s),d(_,a,s),d(_,t,s),h(t,n),A&&A.m(n,null),h(t,l),h(t,u),h(u,v),h(u,m),m.innerHTML=e[0],g=!0},p(e,[t]){e[1]?P?(P.p(e,t),2&t&&i(P,1)):(P=Ba(e),P.c(),i(P,1),P.m(a.parentNode,a)):P&&(r(),c(P,1,1,(()=>{P=null})),o()),e[1]?A?A.p(e,t):(A=Ea(e),A.c(),A.m(n,null)):A&&(A.d(1),A=null),(!g||1&t)&&(m.innerHTML=e[0])},i(e){g||(i(P),g=!0)},o(e){c(P),g=!1},d(e){e&&(p(a),p(t)),P&&P.d(e),A&&A.d()}}}function Ia(e,a,t){let{versions:n}=a,{runinfo:l}=a,_=!1;return e.$$set=e=>{"versions"in e&&t(0,n=e.versions),"runinfo"in e&&t(1,l=e.runinfo)},[n,l,_,function(e){_=e,t(2,_)},()=>{t(2,_=!0)}]}class ka extends a{constructor(e){super(),t(this,e,Ia,Ha,n,{versions:0,runinfo:1})}}function Va(e){let a,t,n;function l(a){e[12](a)}let _={$$slots:{default:[za]},$$scope:{ctx:e}};return void 0!==e[0]&&(_.isOpen=e[0]),a=new Ae({props:_}),b.push((()=>C(a,"isOpen",l))),{c(){g(a.$$.fragment)},m(e,t){P(a,e,t),n=!0},p(e,n){const l={};8576&n&&(l.$$scope={dirty:n,ctx:e}),!t&&1&n&&(t=!0,l.isOpen=e[0],O((()=>t=!1))),a.$set(l)},i(e){n||(i(a.$$.fragment,e),n=!0)},o(e){c(a.$$.fragment,e),n=!1},d(e){A(a,e)}}}function za(e){let a,t;return a=new ia({props:{toc:e[7],page:e[8]}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};128&t&&(n.toc=e[7]),256&t&&(n.page=e[8]),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function qa(e){let a;const t=e[10].default,n=y(t,e,e[13],null);return{c(){n&&n.c()},m(e,t){n&&n.m(e,t),a=!0},p(e,l){n&&n.p&&(!a||8192&l)&&L(n,t,e,e[13],a?x(t,e[13],l,null):j(e[13]),null)},i(e){a||(i(n,e),a=!0)},o(e){c(n,e),a=!1},d(e){n&&n.d(e)}}}function Ka(e){let a,t;return a=new Ce({props:{$$slots:{default:[qa]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};8192&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function Ya(e){let a,t;return a=new be({props:{$$slots:{default:[Ka]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};8192&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function Wa(e){let a,t,n,l;return a=new fe({props:{$$slots:{default:[Ya]},$$scope:{ctx:e}}}),n=new ka({props:{versions:e[3],runinfo:e[9]}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment)},m(e,_){P(a,e,_),d(e,t,_),P(n,e,_),l=!0},p(e,t){const l={};8192&t&&(l.$$scope={dirty:t,ctx:e}),a.$set(l);const _={};8&t&&(_.versions=e[3]),512&t&&(_.runinfo=e[9]),n.$set(_)},i(e){l||(i(a.$$.fragment,e),i(n.$$.fragment,e),l=!0)},o(e){c(a.$$.fragment,e),c(n.$$.fragment,e),l=!1},d(e){e&&p(t),A(a,e),A(n,e)}}}function Xa(e){let a,t,n,l,_,s;function u(a){e[11](a)}let v={logo:e[1],logotext:e[2],entries:e[4],pipeline_name:e[5]};void 0!==e[0]&&(v.isSideNavOpen=e[0]),a=new Ma({props:v}),b.push((()=>C(a,"isSideNavOpen",u)));let m=e[6]&&e[7].length>0&&Va(e);return _=new Pe({props:{$$slots:{default:[Wa]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment),n=f(),m&&m.c(),l=f(),g(_.$$.fragment)},m(e,t){P(a,e,t),d(e,n,t),m&&m.m(e,t),d(e,l,t),P(_,e,t),s=!0},p(e,[n]){const s={};2&n&&(s.logo=e[1]),4&n&&(s.logotext=e[2]),16&n&&(s.entries=e[4]),32&n&&(s.pipeline_name=e[5]),!t&&1&n&&(t=!0,s.isSideNavOpen=e[0],O((()=>t=!1))),a.$set(s),e[6]&&e[7].length>0?m?(m.p(e,n),192&n&&i(m,1)):(m=Va(e),m.c(),i(m,1),m.m(l.parentNode,l)):m&&(r(),c(m,1,1,(()=>{m=null})),o());const d={};8712&n&&(d.$$scope={dirty:n,ctx:e}),_.$set(d)},i(e){s||(i(a.$$.fragment,e),i(m),i(_.$$.fragment,e),s=!0)},o(e){c(a.$$.fragment,e),c(m),c(_.$$.fragment,e),s=!1},d(e){e&&(p(n),p(l)),A(a,e),m&&m.d(e),A(_,e)}}}function Qa(e,a,t){let n,{$$slots:l={},$$scope:_}=a,{logo:s}=a,{logotext:d=null}=a,{versions:i}=a,{entries:r}=a,{pipeline_name:o}=a,{report_toc:c}=a,{toc:p}=a,{page:u}=a,{isSideNavOpen:m}=a,{runinfo:g}=a;return v((()=>{const e=new URL(window.location.href);let a;a=e.hash.length>0?document.querySelector(`li:has(>[href="${e.hash}"])`):document.querySelector("li:has(>.toc-current-page)"),a&&(a.classList.add("toc-active"),a.scrollIntoView());const t=document.querySelectorAll(".pipen-report-toc-anchor");n=new IntersectionObserver((e=>{e.forEach((e=>{const a=e.target.getAttribute("id"),t=document.querySelector(`li:has(>[href="#${a}"])`);t&&(e.isIntersecting?t.classList.add("toc-active"):t.classList.remove("toc-active"))}))}),{root:null,rootMargin:"0px",threshold:.5}),t.forEach((e=>{n.observe(e)}))})),e.$$set=e=>{"logo"in e&&t(1,s=e.logo),"logotext"in e&&t(2,d=e.logotext),"versions"in e&&t(3,i=e.versions),"entries"in e&&t(4,r=e.entries),"pipeline_name"in e&&t(5,o=e.pipeline_name),"report_toc"in e&&t(6,c=e.report_toc),"toc"in e&&t(7,p=e.toc),"page"in e&&t(8,u=e.page),"isSideNavOpen"in e&&t(0,m=e.isSideNavOpen),"runinfo"in e&&t(9,g=e.runinfo),"$$scope"in e&&t(13,_=e.$$scope)},[m,s,d,i,r,o,c,p,u,g,l,function(e){m=e,t(0,m)},function(e){m=e,t(0,m)},_]}class Za extends a{constructor(e){super(),t(this,e,Qa,Xa,n,{logo:1,logotext:2,versions:3,entries:4,pipeline_name:5,report_toc:6,toc:7,page:8,isSideNavOpen:0,runinfo:9})}}function Ja(e){let a;return{c(){a=S(e[2])},m(e,t){d(e,a,t)},p:T,d(e){e&&p(a)}}}function et(e){let a,t,n,l;return t=new Oe({props:{icon:e[1],kind:"tertiary",href:e[3](),$$slots:{default:[Ja]},$$scope:{ctx:e}}}),{c(){a=_("div"),g(t.$$.fragment),s(a,"class",n=`pagenav ${e[0]}`)},m(e,n){d(e,a,n),P(t,a,null),l=!0},p(e,[_]){const d={};16&_&&(d.$$scope={dirty:_,ctx:e}),t.$set(d),(!l||1&_&&n!==(n=`pagenav ${e[0]}`))&&s(a,"class",n)},i(e){l||(i(t.$$.fragment,e),l=!0)},o(e){c(t.$$.fragment,e),l=!1},d(e){e&&p(a),A(t)}}}function at(e,a,t){let{dir:n}=a,l="up"===n?Be:Ge,_="up"===n?"Go to previous page":"Go to next page";return e.$$set=e=>{"dir"in e&&t(0,n=e.dir)},[n,l,_,()=>{let e=new URL(window.location.href);e.hash="";const a=parseInt(e.searchParams.get("page")||0),t="up"===n?a-1:a+1;return 0===t?e.searchParams.delete("page"):e.searchParams.set("page",t),e.toString()}]}class tt extends a{constructor(e){super(),t(this,e,at,et,n,{dir:0})}}function nt(e){let a,t;const n=[{tooltipText:e[0]},{icon:e[1]},{direction:"bottom"},{align:"start"},e[3]];let l={};for(let e=0;e<n.length;e+=1)l=M(l,n[e]);return a=new Se({props:l}),a.$on("click",e[2]),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,[t]){const l=11&t?B(n,[1&t&&{tooltipText:e[0]},2&t&&{icon:e[1]},n[2],n[3],8&t&&G(e[3])]):{};a.$set(l)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function lt(e,a,t){const n=["src","tip","icon"];let l,_=F(a,n),{src:s=""}=a,{tip:d=""}=a,{icon:i="auto"}=a,r=s.split(".").at(-1).toLocaleLowerCase();if("string"==typeof i)switch("auto"===i&&(i=r),i){case"pdf":case"DocumentPdf":l=Ye;break;case"html":case"htm":case"ChartLineData":l=Ke;break;case"png":case"Png":l=qe;break;case"jpg":case"jpeg":case"Jpg":l=ze;break;case"gif":case"Gif":l=Ve;break;case"tif":case"tiff":case"Tif":l=ke;break;case"eps":case"DatabaseElastic":l=Ie;break;case"mp4":case"avi":case"mov":case"mkv":case"webm":case"VideoPlayer":l=He;break;case"Image":l=Ee;break;case"Data":case"DataVolume":l=De;break;case"Code":case"IbmWatsonxCodeAssistantForZRefactor":l=Ue;break;case"svg":case"Svg":l=Ne;break;case"zip":case"tar":case"gz":case"bz2":case"7z":case"rar":case"xz":case"Archive":l=we;break;default:l=Fe}const o={..._,class:[_.class,"pipen-report-image-download"].filter(Boolean).join(" ")};return e.$$set=e=>{a=M(M({},a),w(e)),t(7,_=F(a,n)),"src"in e&&t(5,s=e.src),"tip"in e&&t(0,d=e.tip),"icon"in e&&t(4,i=e.icon)},[d,l,()=>{const e=document.createElement("a");e.href=s,e.download=s.split("/").pop(),e.click()},o,i,s]}class _t extends a{constructor(e){super(),t(this,e,lt,nt,n,{src:5,tip:0,icon:4})}}const{Boolean:st}=E;function dt(e,a,t){const n=e.slice();return n[5]=a[t].src,n[6]=a[t].tip,n[7]=a[t].icon,n[9]=t,n}function it(e){let a,t,n,s,v=l(e[0]),m=[];for(let a=0;a<v.length;a+=1)m[a]=rt(dt(e,v,a));const g=e=>c(m[e],1,1,(()=>{m[e]=null}));let P=[e[1],{role:"button"},{tabindex:"0"}],A={};for(let e=0;e<P.length;e+=1)A=M(A,P[e]);return{c(){a=_("div");for(let e=0;e<m.length;e+=1)m[e].c();U(a,A),D(a,"svelte-14lcnln",!0)},m(l,_){d(l,a,_);for(let e=0;e<m.length;e+=1)m[e]&&m[e].m(a,null);t=!0,n||(s=$(a,"mouseenter",e[3]),n=!0)},p(e,t){if(1&t){let n;for(v=l(e[0]),n=0;n<v.length;n+=1){const l=dt(e,v,n);m[n]?(m[n].p(l,t),i(m[n],1)):(m[n]=rt(l),m[n].c(),i(m[n],1),m[n].m(a,null))}for(r(),n=v.length;n<m.length;n+=1)g(n);o()}D(a,"svelte-14lcnln",!0)},i(e){if(!t){for(let e=0;e<v.length;e+=1)i(m[e]);t=!0}},o(e){m=m.filter(st);for(let e=0;e<m.length;e+=1)c(m[e]);t=!1},d(e){e&&p(a),u(m,e),n=!1,s()}}}function rt(e){let a,t;return a=new _t({props:{src:e[5],tip:e[6],icon:e[7]}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};1&t&&(n.src=e[5]),1&t&&(n.tip=e[6]),1&t&&(n.icon=e[7]),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function ot(e){let a,t,n=e[0].length>0&&it(e);return{c(){n&&n.c(),a=m()},m(e,l){n&&n.m(e,l),d(e,a,l),t=!0},p(e,[t]){e[0].length>0?n?(n.p(e,t),1&t&&i(n,1)):(n=it(e),n.c(),i(n,1),n.m(a.parentNode,a)):n&&(r(),c(n,1,1,(()=>{n=null})),o())},i(e){t||(i(n),t=!0)},o(e){c(n),t=!1},d(e){e&&p(a),n&&n.d(e)}}}function ct(e,a,t){const n=["data"];let l=F(a,n),{data:_=[]}=a;Array.isArray(_)||(_=[_]),_=_.map((e=>"string"==typeof e?{src:e,tip:`Download the ${e.split(".").at(-1)} format.`,icon:"auto"}:(e.tip=e.tip||`Download the ${e.src.split(".").at(-1)} format.`,e.icon=e.icon||"auto",e)));const s={...l,class:[l.class,"pipen-report-image-download-list"].filter(Boolean).join(" ")},d=N();return e.$$set=e=>{a=M(M({},a),w(e)),t(4,l=F(a,n)),"data"in e&&t(0,_=e.data)},[_,s,d,()=>d("mouseenter")]}class pt extends a{constructor(e){super(),t(this,e,ct,ot,n,{data:0})}}const ut=e=>({}),vt=e=>({}),mt=e=>({}),gt=e=>({}),Pt=e=>({}),At=e=>({}),ft=e=>({}),bt=e=>({});function Ct(e){let a,t;return a=new he({props:{ratio:e[6],$$slots:{default:[Tt]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};64&t&&(n.ratio=e[6]),33556159&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function Ot(e){let a,t,n,l,_=e[2]&&yt(e),s=e[0]&&Lt(e),u=e[1]&&xt(e);return{c(){_&&_.c(),a=f(),s&&s.c(),t=f(),u&&u.c(),n=m()},m(e,i){_&&_.m(e,i),d(e,a,i),s&&s.m(e,i),d(e,t,i),u&&u.m(e,i),d(e,n,i),l=!0},p(e,l){e[2]?_?(_.p(e,l),4&l&&i(_,1)):(_=yt(e),_.c(),i(_,1),_.m(a.parentNode,a)):_&&(r(),c(_,1,1,(()=>{_=null})),o()),e[0]?s?(s.p(e,l),1&l&&i(s,1)):(s=Lt(e),s.c(),i(s,1),s.m(t.parentNode,t)):s&&(r(),c(s,1,1,(()=>{s=null})),o()),e[1]?u?(u.p(e,l),2&l&&i(u,1)):(u=xt(e),u.c(),i(u,1),u.m(n.parentNode,n)):u&&(r(),c(u,1,1,(()=>{u=null})),o())},i(e){l||(i(_),i(s),i(u),l=!0)},o(e){c(_),c(s),c(u),l=!1},d(e){e&&(p(a),p(t),p(n)),_&&_.d(e),s&&s.d(e),u&&u.d(e)}}}function St(e){let a;const t=e[15].loading,n=y(t,e,e[25],gt);return{c(){n&&n.c()},m(e,t){n&&n.m(e,t),a=!0},p(e,l){n&&n.p&&(!a||33554432&l)&&L(n,t,e,e[25],a?x(t,e[25],l,mt):j(e[25]),gt)},i(e){a||(i(n,e),a=!0)},o(e){c(n,e),a=!1},d(e){n&&n.d(e)}}}function ht(e){let a,t,n,l,s,u,v,g,P=[e[12],{src:t=e[3]},{alt:e[4]},{title:"Click to zoom in ..."}],A={};for(let e=0;e<P.length;e+=1)A=M(A,P[e]);let b=e[9]&&Rt(e);return{c(){a=_("img"),l=f(),b&&b.c(),s=m(),U(a,A)},m(t,n){d(t,a,n),d(t,l,n),b&&b.m(t,n),d(t,s,n),u=!0,v||(g=[$(a,"click",e[11]),$(a,"mouseenter",e[19]),$(a,"mouseleave",e[20])],v=!0)},p(n,l){U(a,A=B(P,[(e=n)[12],(!u||8&l&&!H(a.src,t=e[3]))&&{src:t},(!u||16&l)&&{alt:e[4]},{title:"Click to zoom in ..."}])),e[9]?b?(b.p(e,l),512&l&&i(b,1)):(b=Rt(e),b.c(),i(b,1),b.m(s.parentNode,s)):b&&(r(),c(b,1,1,(()=>{b=null})),o())},i(t){u||(t&&I((()=>{u&&(n||(n=k(a,V,{duration:e[7]?Gt:0},!0)),n.run(1))})),i(b),u=!0)},o(t){t&&(n||(n=k(a,V,{duration:e[7]?Gt:0},!1)),n.run(0)),c(b),u=!1},d(e){e&&(p(a),p(l),p(s)),e&&n&&n.end(),b&&b.d(e),v=!1,z(g)}}}function Rt(e){let a,t;return a=new pt({props:{data:e[5]}}),a.$on("mouseenter",e[21]),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};32&t&&(n.data=e[5]),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function $t(e){let a;const t=e[15].error,n=y(t,e,e[25],vt);return{c(){n&&n.c()},m(e,t){n&&n.m(e,t),a=!0},p(e,l){n&&n.p&&(!a||33554432&l)&&L(n,t,e,e[25],a?x(t,e[25],l,ut):j(e[25]),vt)},i(e){a||(i(n,e),a=!0)},o(e){c(n,e),a=!1},d(e){n&&n.d(e)}}}function Tt(e){let a,t,n,l,_=e[2]&&St(e),s=e[0]&&ht(e),u=e[1]&&$t(e);return{c(){_&&_.c(),a=f(),s&&s.c(),t=f(),u&&u.c(),n=m()},m(e,i){_&&_.m(e,i),d(e,a,i),s&&s.m(e,i),d(e,t,i),u&&u.m(e,i),d(e,n,i),l=!0},p(e,l){e[2]?_?(_.p(e,l),4&l&&i(_,1)):(_=St(e),_.c(),i(_,1),_.m(a.parentNode,a)):_&&(r(),c(_,1,1,(()=>{_=null})),o()),e[0]?s?(s.p(e,l),1&l&&i(s,1)):(s=ht(e),s.c(),i(s,1),s.m(t.parentNode,t)):s&&(r(),c(s,1,1,(()=>{s=null})),o()),e[1]?u?(u.p(e,l),2&l&&i(u,1)):(u=$t(e),u.c(),i(u,1),u.m(n.parentNode,n)):u&&(r(),c(u,1,1,(()=>{u=null})),o())},i(e){l||(i(_),i(s),i(u),l=!0)},o(e){c(_),c(s),c(u),l=!1},d(e){e&&(p(a),p(t),p(n)),_&&_.d(e),s&&s.d(e),u&&u.d(e)}}}function yt(e){let a;const t=e[15].loading,n=y(t,e,e[25],bt);return{c(){n&&n.c()},m(e,t){n&&n.m(e,t),a=!0},p(e,l){n&&n.p&&(!a||33554432&l)&&L(n,t,e,e[25],a?x(t,e[25],l,ft):j(e[25]),bt)},i(e){a||(i(n,e),a=!0)},o(e){c(n,e),a=!1},d(e){n&&n.d(e)}}}function Lt(e){let a,t,n,l,s,u,v,g,P=[e[12],{src:t=e[3]},{alt:e[4]},{title:"Click to zoom in ..."}],A={};for(let e=0;e<P.length;e+=1)A=M(A,P[e]);let b=e[9]&&jt(e);return{c(){a=_("img"),l=f(),b&&b.c(),s=m(),U(a,A)},m(t,n){d(t,a,n),d(t,l,n),b&&b.m(t,n),d(t,s,n),u=!0,v||(g=[$(a,"click",e[11]),$(a,"mouseenter",e[16]),$(a,"mouseleave",e[17])],v=!0)},p(n,l){U(a,A=B(P,[(e=n)[12],(!u||8&l&&!H(a.src,t=e[3]))&&{src:t},(!u||16&l)&&{alt:e[4]},{title:"Click to zoom in ..."}])),e[9]?b?(b.p(e,l),512&l&&i(b,1)):(b=jt(e),b.c(),i(b,1),b.m(s.parentNode,s)):b&&(r(),c(b,1,1,(()=>{b=null})),o())},i(t){u||(t&&I((()=>{u&&(n||(n=k(a,V,{duration:e[7]?Gt:0},!0)),n.run(1))})),i(b),u=!0)},o(t){t&&(n||(n=k(a,V,{duration:e[7]?Gt:0},!1)),n.run(0)),c(b),u=!1},d(e){e&&(p(a),p(l),p(s)),e&&n&&n.end(),b&&b.d(e),v=!1,z(g)}}}function jt(e){let a,t;return a=new pt({props:{data:e[5]}}),a.$on("mouseenter",e[18]),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};32&t&&(n.data=e[5]),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function xt(e){let a;const t=e[15].error,n=y(t,e,e[25],At);return{c(){n&&n.c()},m(e,t){n&&n.m(e,t),a=!0},p(e,l){n&&n.p&&(!a||33554432&l)&&L(n,t,e,e[25],a?x(t,e[25],l,Pt):j(e[25]),At)},i(e){a||(i(n,e),a=!0)},o(e){c(n,e),a=!1},d(e){n&&n.d(e)}}}function Mt(e){let a,t,n,l;return{c(){a=_("img"),s(a,"class","pipen-report-image-lg"),H(a.src,t=e[3])||s(a,"src",t),s(a,"alt",e[4])},m(e,t){d(e,a,t),l=!0},p(n,_){e=n,(!l||8&_&&!H(a.src,t=e[3]))&&s(a,"src",t),(!l||16&_)&&s(a,"alt",e[4])},i(t){l||(t&&I((()=>{l&&(n||(n=k(a,V,{duration:e[7]?Gt:0},!0)),n.run(1))})),l=!0)},o(t){t&&(n||(n=k(a,V,{duration:e[7]?Gt:0},!1)),n.run(0)),l=!1},d(e){e&&p(a),e&&n&&n.end()}}}function Bt(e){let a,t,n,l,_,s;const u=[Ot,Ct],v=[];function m(e,a){return void 0===e[6]?0:1}function S(a){e[22](a)}a=m(e),t=v[a]=u[a](e);let h={modalHeading:e[4],primaryButtonText:"Open in new window ...",secondaryButtonText:"Close",$$slots:{default:[Mt]},$$scope:{ctx:e}};return void 0!==e[8]&&(h.open=e[8]),l=new ce({props:h}),b.push((()=>C(l,"open",S))),l.$on("click:button--secondary",e[23]),l.$on("click:button--primary",e[24]),{c(){t.c(),n=f(),g(l.$$.fragment)},m(e,t){v[a].m(e,t),d(e,n,t),P(l,e,t),s=!0},p(e,[s]){let d=a;a=m(e),a===d?v[a].p(e,s):(r(),c(v[d],1,1,(()=>{v[d]=null})),o(),t=v[a],t?t.p(e,s):(t=v[a]=u[a](e),t.c()),i(t,1),t.m(n.parentNode,n));const p={};16&s&&(p.modalHeading=e[4]),33554584&s&&(p.$$scope={dirty:s,ctx:e}),!_&&256&s&&(_=!0,p.open=e[8],O((()=>_=!1))),l.$set(p)},i(e){s||(i(t),i(l.$$.fragment,e),s=!0)},o(e){c(t),c(l.$$.fragment,e),s=!1},d(e){e&&p(n),v[a].d(e),A(l,e)}}}const Gt=110;function Ft(e,a,t){const n=["src","alt","download","ratio","loading","loaded","error","fadeIn","border","loadImage"];let l=F(a,n),{$$slots:_={},$$scope:s}=a,{src:d=""}=a,{alt:i=""}=a,{download:r=[]}=a,{ratio:o}=a,{loading:c=!1}=a,{loaded:p=!1}=a,{error:u=!1}=a,{fadeIn:m=!1}=a,{border:g=!0}=a,P=!1,A=!1;const f=e=>{null!=O&&(O=null),t(0,p=!1),t(1,u=!1),O=new Image,O.src=e||d,O.onload=()=>t(0,p=!0),O.onerror=()=>t(1,u=!0)},b=N();let C,O=null;v((()=>()=>O=null));const S={...l,class:[l.class,"pipen-report-image",g?"":"pipen-report-image-noborder"].filter(Boolean).join(" ")};return e.$$set=e=>{a=M(M({},a),w(e)),t(28,l=F(a,n)),"src"in e&&t(3,d=e.src),"alt"in e&&t(4,i=e.alt),"download"in e&&t(5,r=e.download),"ratio"in e&&t(6,o=e.ratio),"loading"in e&&t(2,c=e.loading),"loaded"in e&&t(0,p=e.loaded),"error"in e&&t(1,u=e.error),"fadeIn"in e&&t(7,m=e.fadeIn),"border"in e&&t(13,g=e.border),"$$scope"in e&&t(25,s=e.$$scope)},e.$$.update=()=>{3&e.$$.dirty&&t(2,c=!p&&!u),8&e.$$.dirty&&d&&"undefined"!=typeof window&&f(),1&e.$$.dirty&&p&&b("load"),2&e.$$.dirty&&u&&b("error")},[p,u,c,d,i,r,o,m,P,A,C,()=>{t(8,P=!0)},S,g,f,_,()=>{t(9,A=!0)},()=>{t(10,C=setTimeout((()=>{t(9,A=!1)}),100))},()=>{clearTimeout(C),t(9,A=!0)},()=>{t(9,A=!0)},()=>{t(10,C=setTimeout((()=>{t(9,A=!1)}),100))},()=>{clearTimeout(C),t(9,A=!0)},function(e){P=e,t(8,P)},()=>{t(8,P=!1)},()=>{t(8,P=!1),window.open(d)},s]}class wt extends a{constructor(e){super(),t(this,e,Ft,Bt,n,{src:3,alt:4,download:5,ratio:6,loading:2,loaded:0,error:1,fadeIn:7,border:13,loadImage:14})}get loadImage(){return this.$$.ctx[14]}}function Nt(e){let a,t,n=[e[2]],l={};for(let e=0;e<n.length;e+=1)l=M(l,n[e]);return{c(){a=_("div"),t=S("Image loading ..."),U(a,l)},m(e,n){d(e,a,n),h(a,t)},p:T,d(e){e&&p(a)}}}function Ut(e){let a,t,n=[e[3]],l={};for(let e=0;e<n.length;e+=1)l=M(l,n[e]);return{c(){a=_("div"),t=S("Image loading error!"),U(a,l)},m(e,n){d(e,a,n),h(a,t)},p:T,d(e){e&&p(a)}}}function Dt(e){let a,t,n;const l=[{src:e[1]},e[4]];let s={$$slots:{error:[Ut],loading:[Nt]},$$scope:{ctx:e}};for(let e=0;e<l.length;e+=1)s=M(s,l[e]);t=new wt({props:s});let r=[e[0]],o={};for(let e=0;e<r.length;e+=1)o=M(o,r[e]);return{c(){a=_("div"),g(t.$$.fragment),U(a,o)},m(e,l){d(e,a,l),P(t,a,null),n=!0},p(e,[n]){const _=18&n?B(l,[2&n&&{src:e[1]},16&n&&G(e[4])]):{};256&n&&(_.$$scope={dirty:n,ctx:e}),t.$set(_),U(a,o=B(r,[1&n&&e[0]]))},i(e){n||(i(t.$$.fragment,e),n=!0)},o(e){c(t.$$.fragment,e),n=!1},d(e){e&&p(a),A(t)}}}function Et(e,a,t){const n=["frameProps","width","height","src"];let l=F(a,n),{frameProps:_={class:"pipen-report-image-frame"}}=a,{width:s=300}=a,{height:d=200}=a,{src:i}=a;_||(_={"style:position":"relative"});let r=`width: ${s}px; height: ${d}px; aspect-ratio: ${s} / ${d}`,o={...l,class:[l.class,"pipen-report-image-loading"].filter(Boolean).join(" "),style:l.style?`${l.style}; ${r}`:r},c={...o,class:[o.class,"error"].join(" ")};return e.$$set=e=>{a=M(M({},a),w(e)),t(4,l=F(a,n)),"frameProps"in e&&t(0,_=e.frameProps),"width"in e&&t(5,s=e.width),"height"in e&&t(6,d=e.height),"src"in e&&t(1,i=e.src)},[_,i,o,c,l,s,d]}let Ht=class extends a{constructor(e){super(),t(this,e,Et,Dt,n,{frameProps:0,width:5,height:6,src:1})}};const It=e=>({}),kt=e=>({});function Vt(e){let a,t;const n=e[25].default,l=y(n,e,e[24],null);return{c(){a=_("ul"),l&&l.c(),s(a,"role","menu"),s(a,"tabindex","-1"),s(a,"aria-label",e[14]),s(a,"data-floating-menu-direction",e[5]),s(a,"class",e[9]),D(a,"bx--overflow-menu-options",!0),D(a,"bx--overflow-menu--flip",e[7]),D(a,"bx--overflow-menu-options--open",e[0]),D(a,"bx--overflow-menu-options--light",e[6]),D(a,"bx--overflow-menu-options--sm","sm"===e[4]),D(a,"bx--overflow-menu-options--xl","xl"===e[4]),D(a,"bx--breadcrumb-menu-options",!!e[15])},m(n,_){d(n,a,_),l&&l.m(a,null),e[32](a),t=!0},p(e,_){l&&l.p&&(!t||16777216&_[0])&&L(l,n,e,e[24],t?x(n,e[24],_,null):j(e[24]),null),(!t||16384&_[0])&&s(a,"aria-label",e[14]),(!t||32&_[0])&&s(a,"data-floating-menu-direction",e[5]),(!t||512&_[0])&&s(a,"class",e[9]),(!t||512&_[0])&&D(a,"bx--overflow-menu-options",!0),(!t||640&_[0])&&D(a,"bx--overflow-menu--flip",e[7]),(!t||513&_[0])&&D(a,"bx--overflow-menu-options--open",e[0]),(!t||576&_[0])&&D(a,"bx--overflow-menu-options--light",e[6]),(!t||528&_[0])&&D(a,"bx--overflow-menu-options--sm","sm"===e[4]),(!t||528&_[0])&&D(a,"bx--overflow-menu-options--xl","xl"===e[4]),(!t||33280&_[0])&&D(a,"bx--breadcrumb-menu-options",!!e[15])},i(e){t||(i(l,e),t=!0)},o(e){c(l,e),t=!1},d(t){t&&p(a),l&&l.d(t),e[32](null)}}}function zt(e){let a,t,n,l,s,u,v,b;const C=e[25].menu,O=y(C,e,e[24],kt),S=O||function(e){let a,t,n;var l=e[1];function _(e,a){return{props:{"aria-label":e[11],title:e[11],class:"bx--overflow-menu__icon "+e[10]}}}return l&&(a=X(l,_(e))),{c(){a&&g(a.$$.fragment),t=m()},m(e,l){a&&P(a,e,l),d(e,t,l),n=!0},p(e,n){if(2&n[0]&&l!==(l=e[1])){if(a){r();const e=a;c(e.$$.fragment,1,0,(()=>{A(e,1)})),o()}l?(a=X(l,_(e)),g(a.$$.fragment),i(a.$$.fragment,1),P(a,t.parentNode,t)):a=null}else if(l){const t={};2048&n[0]&&(t["aria-label"]=e[11]),2048&n[0]&&(t.title=e[11]),1024&n[0]&&(t.class="bx--overflow-menu__icon "+e[10]),a.$set(t)}},i(e){n||(a&&i(a.$$.fragment,e),n=!0)},o(e){a&&c(a.$$.fragment,e),n=!1},d(e){e&&p(t),a&&A(a,e)}}}(e);let R=e[0]&&Vt(e),T=[{type:"button"},{"aria-haspopup":""},{"aria-expanded":e[0]},{"aria-label":e[14]},{id:e[12]},e[20]],G={};for(let e=0;e<T.length;e+=1)G=M(G,T[e]);return{c(){a=new Q(!1),t=m(),n=f(),l=_("button"),S.c(),s=f(),R&&R.c(),a.a=t,U(l,G),D(l,"bx--overflow-menu",!0),D(l,"bx--overflow-menu--open",e[0]),D(l,"bx--overflow-menu--light",e[6]),D(l,"bx--overflow-menu--sm","sm"===e[4]),D(l,"bx--overflow-menu--xl","xl"===e[4])},m(_,i){a.m(e[13],document.head),h(document.head,t),d(_,n,i),d(_,l,i),S.m(l,null),h(l,s),R&&R.m(l,null),l.autofocus&&l.focus(),e[33](l),u=!0,v||(b=[$(window,"click",e[31]),$(l,"click",e[26]),$(l,"click",e[34]),$(l,"mouseover",e[27]),$(l,"mouseenter",e[28]),$(l,"mouseleave",e[29]),$(l,"keydown",e[30]),$(l,"keydown",e[35]),$(l,"focusout",e[36])],v=!0)},p(e,t){(!u||8192&t[0])&&a.p(e[13]),O?O.p&&(!u||16777216&t[0])&&L(O,C,e,e[24],u?x(C,e[24],t,It):j(e[24]),kt):S.p&&(!u||3074&t[0])&&S.p(e,u?t:[-1,-1]),e[0]?R?(R.p(e,t),1&t[0]&&i(R,1)):(R=Vt(e),R.c(),i(R,1),R.m(l,null)):R&&(r(),c(R,1,1,(()=>{R=null})),o()),U(l,G=B(T,[{type:"button"},{"aria-haspopup":""},(!u||1&t[0])&&{"aria-expanded":e[0]},(!u||16384&t[0])&&{"aria-label":e[14]},(!u||4096&t[0])&&{id:e[12]},1048576&t[0]&&e[20]])),D(l,"bx--overflow-menu",!0),D(l,"bx--overflow-menu--open",e[0]),D(l,"bx--overflow-menu--light",e[6]),D(l,"bx--overflow-menu--sm","sm"===e[4]),D(l,"bx--overflow-menu--xl","xl"===e[4])},i(e){u||(i(S,e),i(R),u=!0)},o(e){c(S,e),c(R),u=!1},d(_){_&&(a.d(),p(n),p(l)),p(t),S.d(_),R&&R.d(),e[33](null),v=!1,z(b)}}}function qt(e,a,t){let n,l;const _=["size","direction","open","light","flipped","persistentClickItems","menuOptionsClass","icon","iconClass","iconDescription","id","buttonRef","menuRef"];let s,d,i,r=F(a,_),{$$slots:o={},$$scope:c}=a,{size:p}=a,{direction:u="bottom"}=a,{open:v=!1}=a,{light:m=!1}=a,{flipped:g=!1}=a,{persistentClickItems:P=!1}=a,{menuOptionsClass:A}=a,{icon:f=Xe}=a,{iconClass:C}=a,{iconDescription:O="Open and close list of options"}=a,{id:S="ccs-"+Math.random().toString(36)}=a,{buttonRef:h=null}=a,{menuRef:R=null}=a;const $=q("BreadcrumbItem"),T=N(),y=Z([]);K(e,y,(e=>t(23,d=e)));const L=Z(void 0);K(e,L,(e=>t(38,i=e)));const j=Z(void 0),x=Z(-1);let B;K(e,x,(e=>t(22,s=e)));let G=!0;Y("OverflowMenu",{focusedId:j,add:({id:e,text:a,primaryFocus:t,disabled:n})=>{y.update((l=>(t&&x.set(l.length),[...l,{id:e,text:a,primaryFocus:t,disabled:n,index:l.length}])))},update:e=>{L.set(e)},change:e=>{let a=s+e;a<0?a=d.length-1:a>=d.length&&(a=0);let t=d[a].disabled;for(;t;)a+=e,a<0?a=d.length-1:a>=d.length&&(a=0),t=d[a].disabled;x.set(a)}}),W((()=>{if(i){const{index:e,text:a}=d.filter((e=>e.id===i))[0];T("close",{index:e,text:a}),t(0,v=!1)}if(v){const{width:e,height:a}=h.getBoundingClientRect();t(21,B=e),!G&&s<0&&R.focus(),g&&(t(3,R.style.left="auto",R),t(3,R.style.right=0,R)),"top"===u?(t(3,R.style.top="auto",R),t(3,R.style.bottom=a+"px",R)):"bottom"===u&&t(3,R.style.top=a+"px",R),$&&(t(3,R.style.top=a+10+"px",R),t(3,R.style.left="-11px",R))}v||(y.set([]),L.set(void 0),x.set(0)),G=!1}));return e.$$set=e=>{t(40,a=M(M({},a),w(e))),t(20,r=F(a,_)),"size"in e&&t(4,p=e.size),"direction"in e&&t(5,u=e.direction),"open"in e&&t(0,v=e.open),"light"in e&&t(6,m=e.light),"flipped"in e&&t(7,g=e.flipped),"persistentClickItems"in e&&t(8,P=e.persistentClickItems),"menuOptionsClass"in e&&t(9,A=e.menuOptionsClass),"icon"in e&&t(1,f=e.icon),"iconClass"in e&&t(10,C=e.iconClass),"iconDescription"in e&&t(11,O=e.iconDescription),"id"in e&&t(12,S=e.id),"buttonRef"in e&&t(2,h=e.buttonRef),"menuRef"in e&&t(3,R=e.menuRef),"$$scope"in e&&t(24,c=e.$$scope)},e.$$.update=()=>{t(14,n=a["aria-label"]||"menu"),12582912&e.$$.dirty[0]&&d[s]&&j.set(d[s].id),2101248&e.$$.dirty[0]&&t(13,l=`<style>\n    #${S} .bx--overflow-menu-options.bx--overflow-menu-options:after {\n      width: ${B?B+"px":"2rem"};\n    }\n  </style>`)},$&&t(1,f=We),a=w(a),[v,f,h,R,p,u,m,g,P,A,C,O,S,l,n,$,T,y,L,x,r,B,s,d,c,o,function(a){J.call(this,e,a)},function(a){J.call(this,e,a)},function(a){J.call(this,e,a)},function(a){J.call(this,e,a)},function(a){J.call(this,e,a)},({target:e})=>{h&&h.contains(e)||R&&!R.contains(e)&&t(0,v=!1)},function(e){b[e?"unshift":"push"]((()=>{R=e,t(3,R)}))},function(e){b[e?"unshift":"push"]((()=>{h=e,t(2,h)}))},e=>{P&&e.stopPropagation(),R&&R.contains(e.target)||(t(0,v=!v),v||T("close"))},e=>{v&&(["ArrowDown","ArrowLeft","ArrowRight","ArrowUp"].includes(e.key)?e.preventDefault():"Escape"===e.key&&(e.stopPropagation(),T("close"),t(0,v=!1),h.focus()))},e=>{v&&(h.contains(e.relatedTarget)||(T("close"),t(0,v=!1)))}]}class Kt extends a{constructor(e){super(),t(this,e,qt,zt,n,{size:4,direction:5,open:0,light:6,flipped:7,persistentClickItems:8,menuOptionsClass:9,icon:1,iconClass:10,iconDescription:11,id:12,buttonRef:2,menuRef:3},null,[-1,-1])}}function Yt(e,a,t){const n=e.slice();return n[3]=a[t],n[4]=a,n[5]=t,n}function Wt(e){let a,t,n,l,r;function o(a){e[1](a,e[3])}let u={labelText:e[3].value};return void 0!==e[3].selected&&(u.checked=e[3].selected),t=new Re({props:u}),b.push((()=>C(t,"checked",o))),t.$on("change",e[2]),{c(){a=_("li"),g(t.$$.fragment),l=f(),s(a,"class","pipen-report-column-selector-item svelte-147jw2r")},m(e,n){d(e,a,n),P(t,a,null),h(a,l),r=!0},p(a,l){e=a;const _={};1&l&&(_.labelText=e[3].value),!n&&1&l&&(n=!0,_.checked=e[3].selected,O((()=>n=!1))),t.$set(_)},i(e){r||(i(t.$$.fragment,e),r=!0)},o(e){c(t.$$.fragment,e),r=!1},d(e){e&&p(a),A(t)}}}function Xt(e){let a,t,n=l(e[0]),_=[];for(let a=0;a<n.length;a+=1)_[a]=Wt(Yt(e,n,a));const s=e=>c(_[e],1,1,(()=>{_[e]=null}));return{c(){for(let e=0;e<_.length;e+=1)_[e].c();a=m()},m(e,n){for(let a=0;a<_.length;a+=1)_[a]&&_[a].m(e,n);d(e,a,n),t=!0},p(e,t){if(1&t){let d;for(n=l(e[0]),d=0;d<n.length;d+=1){const l=Yt(e,n,d);_[d]?(_[d].p(l,t),i(_[d],1)):(_[d]=Wt(l),_[d].c(),i(_[d],1),_[d].m(a.parentNode,a))}for(r(),d=n.length;d<_.length;d+=1)s(d);o()}},i(e){if(!t){for(let e=0;e<n.length;e+=1)i(_[e]);t=!0}},o(e){_=_.filter(Boolean);for(let e=0;e<_.length;e+=1)c(_[e]);t=!1},d(e){e&&p(a),u(_,e)}}}function Qt(e){let a,t;return a=new Kt({props:{class:"pipen-report-overflowmenu",persistentClickItems:!0,size:"sm",icon:Qe,iconDescription:"Select visible columns ...",$$slots:{default:[Xt]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,[t]){const n={};65&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function Zt(e,a,t){let{allColumns:n}=a;return e.$$set=e=>{"allColumns"in e&&t(0,n=e.allColumns)},[n,function(a,l){e.$$.not_equal(l.selected,a)&&(l.selected=a,t(0,n))},()=>{t(0,n)}]}class Jt extends a{constructor(e){super(),t(this,e,Zt,Qt,n,{allColumns:0})}}function en(e){let a,t;return a=new Oe({props:{iconDescription:"Right click and save as to download the entire data",tooltipPosition:"top",icon:Ze,href:e[2]}}),a.$on("click",e[13]),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};4&t&&(n.href=e[2]),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function an(e){let a,t,n,l,_,s,u,v,S,h;function R(a){e[17](a)}let $={persistent:!0};function T(a){e[18](a)}void 0!==e[7]&&($.value=e[7]),a=new je({props:$}),b.push((()=>C(a,"value",R)));let y={};void 0!==e[6]&&(y.allColumns=e[6]),l=new Jt({props:y}),b.push((()=>C(l,"allColumns",T))),u=new Oe({props:{iconDescription:`Download current ${e[11].length} records (may be partial of the entire data)`,tooltipPosition:"top",icon:Fe}}),u.$on("click",e[14]);let L=!!e[2]&&en(e);return{c(){g(a.$$.fragment),n=f(),g(l.$$.fragment),s=f(),g(u.$$.fragment),v=f(),L&&L.c(),S=m()},m(e,t){P(a,e,t),d(e,n,t),P(l,e,t),d(e,s,t),P(u,e,t),d(e,v,t),L&&L.m(e,t),d(e,S,t),h=!0},p(e,n){const s={};!t&&128&n&&(t=!0,s.value=e[7],O((()=>t=!1))),a.$set(s);const d={};!_&&64&n&&(_=!0,d.allColumns=e[6],O((()=>_=!1))),l.$set(d);const p={};2048&n&&(p.iconDescription=`Download current ${e[11].length} records (may be partial of the entire data)`),u.$set(p),e[2]?L?(L.p(e,n),4&n&&i(L,1)):(L=en(e),L.c(),i(L,1),L.m(S.parentNode,S)):L&&(r(),c(L,1,1,(()=>{L=null})),o())},i(e){h||(i(a.$$.fragment,e),i(l.$$.fragment,e),i(u.$$.fragment,e),i(L),h=!0)},o(e){c(a.$$.fragment,e),c(l.$$.fragment,e),c(u.$$.fragment,e),c(L),h=!1},d(e){e&&(p(n),p(s),p(v),p(S)),A(a,e),A(l,e),A(u,e),L&&L.d(e)}}}function tn(e){let a,t;return a=new Le({props:{$$slots:{default:[an]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};134219972&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function nn(e){let a,t;return a=new ye({props:{size:e[12],$$slots:{default:[tn]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};134219972&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function ln(e){let a;return{c(){a=_("p"),a.textContent='You must use "Save as" or "Save link as" from the context menu (by right-clicking the button) to download the entire data.'},m(e,t){d(e,a,t)},p:T,d(e){e&&p(a)}}}function _n(e){let a,t,n,l,_,s,r,o,u;const v=[{size:e[0]},{zebra:e[3]},{sortable:e[4]},{page:e[8]},{pageSize:e[9]},{rows:e[11]},{headers:e[1]},e[15]];let m={$$slots:{default:[nn]},$$scope:{ctx:e}};for(let e=0;e<v.length;e+=1)m=M(m,v[e]);function S(a){e[19](a)}function h(a){e[20](a)}a=new $e({props:m});let R={pageSizes:e[5],totalItems:e[11].length};function $(a){e[21](a)}void 0!==e[9]&&(R.pageSize=e[9]),void 0!==e[8]&&(R.page=e[8]),n=new Te({props:R}),b.push((()=>C(n,"pageSize",S))),b.push((()=>C(n,"page",h)));let T={passiveModal:!0,modalHeading:"Downloading entire data",$$slots:{default:[ln]},$$scope:{ctx:e}};return void 0!==e[10]&&(T.open=e[10]),r=new ce({props:T}),b.push((()=>C(r,"open",$))),r.$on("open",e[22]),r.$on("close",e[23]),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment),s=f(),g(r.$$.fragment)},m(e,l){P(a,e,l),d(e,t,l),P(n,e,l),d(e,s,l),P(r,e,l),u=!0},p(e,[t]){const s=35611&t?B(v,[1&t&&{size:e[0]},8&t&&{zebra:e[3]},16&t&&{sortable:e[4]},256&t&&{page:e[8]},512&t&&{pageSize:e[9]},2048&t&&{rows:e[11]},2&t&&{headers:e[1]},32768&t&&G(e[15])]):{};134219972&t&&(s.$$scope={dirty:t,ctx:e}),a.$set(s);const d={};32&t&&(d.pageSizes=e[5]),2048&t&&(d.totalItems=e[11].length),!l&&512&t&&(l=!0,d.pageSize=e[9],O((()=>l=!1))),!_&&256&t&&(_=!0,d.page=e[8],O((()=>_=!1))),n.$set(d);const i={};134217728&t&&(i.$$scope={dirty:t,ctx:e}),!o&&1024&t&&(o=!0,i.open=e[10],O((()=>o=!1))),r.$set(i)},i(e){u||(i(a.$$.fragment,e),i(n.$$.fragment,e),i(r.$$.fragment,e),u=!0)},o(e){c(a.$$.fragment,e),c(n.$$.fragment,e),c(r.$$.fragment,e),u=!1},d(e){e&&(p(t),p(s)),A(a,e),A(n,e),A(r,e)}}}function sn(e,a,t){const n=["size","src","zebra","sortable","rows","headers","pageSizes"];let l=F(a,n),{size:_="sm"}=a,{src:s}=a;_={sm:"short",small:"short",short:"short",md:"medium",medium:"medium",lg:"tall",large:"tall",tall:"tall",compact:"compact"}[_];let d={short:"sm",compact:"sm",medium:"default",tall:"default"}[_],{zebra:i=!0}=a,{sortable:r=!0}=a,{rows:o=[]}=a,{headers:c=[]}=a,{pageSizes:p=[20,50,75,100]}=a,u=c.map((e=>({...e,selected:!0}))),v=1,m=p[0],g="",P=!1,A=o;return e.$$set=e=>{a=M(M({},a),w(e)),t(15,l=F(a,n)),"size"in e&&t(0,_=e.size),"src"in e&&t(2,s=e.src),"zebra"in e&&t(3,i=e.zebra),"sortable"in e&&t(4,r=e.sortable),"rows"in e&&t(16,o=e.rows),"headers"in e&&t(1,c=e.headers),"pageSizes"in e&&t(5,p=e.pageSizes)},e.$$.update=()=>{64&e.$$.dirty&&t(1,c=u.filter((e=>e.selected))),65664&e.$$.dirty&&(g.length>0?t(11,A=o.filter((e=>Object.values(e).filter((e=>!!e&&e.toString().includes(g))).length>0))):t(11,A=o))},[_,c,s,i,r,p,u,g,v,m,P,A,d,e=>{e.preventDefault(),t(10,P=!0)},()=>{((e,a)=>{const t=document.createElement("a");t.href=e,t.download=a,document.body.appendChild(t),t.click(),document.body.removeChild(t)})("data:application/vnd.ms-excel;charset=utf-8,\ufeff"+encodeURIComponent(function(){const e=c.map((e=>e.value)),a=[];return A.forEach((e=>{const t=c.map((a=>`"${e[a.key]}"`));a.push(t.join(","))})),e.join(",")+"\r\n"+a.join("\r\n")}()),s&&s.length>0?s.split("/").pop().split(".")[0]+".xls":"data-subset.xls")},l,o,function(e){g=e,t(7,g)},function(e){u=e,t(6,u)},function(e){m=e,t(9,m)},function(e){v=e,t(8,v)},function(e){P=e,t(10,P)},function(a){J.call(this,e,a)},function(a){J.call(this,e,a)}]}class dn extends a{constructor(e){super(),t(this,e,sn,_n,n,{size:0,src:2,zebra:3,sortable:4,rows:16,headers:1,pageSizes:5})}}function rn(e){let a,t,n;const l=[{headers:e[0]},{rows:e[1]},e[3]];let s={};for(let e=0;e<l.length;e+=1)s=M(s,l[e]);t=new dn({props:s});let r=[e[2]],o={};for(let e=0;e<r.length;e+=1)o=M(o,r[e]);return{c(){a=_("div"),g(t.$$.fragment),U(a,o)},m(e,l){d(e,a,l),P(t,a,null),n=!0},p(e,n){const _=11&n?B(l,[1&n&&{headers:e[0]},2&n&&{rows:e[1]},8&n&&G(e[3])]):{};t.$set(_),U(a,o=B(r,[4&n&&e[2]]))},i(e){n||(i(t.$$.fragment,e),n=!0)},o(e){c(t.$$.fragment,e),n=!1},d(e){e&&p(a),A(t)}}}function on(e){let a,t;const n=[{headers:e[0]},{rows:e[1]},e[3]];let l={};for(let e=0;e<n.length;e+=1)l=M(l,n[e]);return a=new dn({props:l}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const l=11&t?B(n,[1&t&&{headers:e[0]},2&t&&{rows:e[1]},8&t&&G(e[3])]):{};a.$set(l)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function cn(e){let a,t,n,l;const _=[on,rn],s=[];function u(e,a){return e[2]?1:0}return a=u(e),t=s[a]=_[a](e),{c(){t.c(),n=m()},m(e,t){s[a].m(e,t),d(e,n,t),l=!0},p(e,[l]){let d=a;a=u(e),a===d?s[a].p(e,l):(r(),c(s[d],1,1,(()=>{s[d]=null})),o(),t=s[a],t?t.p(e,l):(t=s[a]=_[a](e),t.c()),i(t,1),t.m(n.parentNode,n))},i(e){l||(i(t),l=!0)},o(e){c(t),l=!1},d(e){e&&p(n),s[a].d(e)}}}function pn(e,a,t){const n=["frameProps","data","headers","rows"];let l=F(a,n),{frameProps:_={class:"pipen-report-datatable-frame"}}=a,{data:s={}}=a,{headers:d=[]}=a,{rows:i=[]}=a;if(d.length+i.length>0&&s.length>0)throw"Can only specify `data` or `headers`/`rows` for `DataTable`!";return s.length>0&&(i=s,d=Object.keys(i[0]).filter((e=>"id"!==e)).map((e=>({key:e,value:e})))),e.$$set=e=>{a=M(M({},a),w(e)),t(3,l=F(a,n)),"frameProps"in e&&t(2,_=e.frameProps),"data"in e&&t(4,s=e.data),"headers"in e&&t(0,d=e.headers),"rows"in e&&t(1,i=e.rows)},[d,i,_,l,s]}class un extends a{constructor(e){super(),t(this,e,pn,cn,n,{frameProps:2,data:4,headers:0,rows:1})}}function vn(e){let a,t;return{c(){a=_("div"),t=S(e[0]),s(a,"class","pipen-report-descr-title svelte-1xzxuva")},m(e,n){d(e,a,n),h(a,t)},p(e,a){1&a&&R(t,e[0])},d(e){e&&p(a)}}}function mn(e){let a,t,n,l=e[0]&&vn(e);const s=e[3].default,r=y(s,e,e[2],null);let o=[e[1]],u={};for(let e=0;e<o.length;e+=1)u=M(u,o[e]);return{c(){a=_("div"),l&&l.c(),t=f(),r&&r.c(),U(a,u),D(a,"svelte-1xzxuva",!0)},m(e,_){d(e,a,_),l&&l.m(a,null),h(a,t),r&&r.m(a,null),n=!0},p(e,[_]){e[0]?l?l.p(e,_):(l=vn(e),l.c(),l.m(a,t)):l&&(l.d(1),l=null),r&&r.p&&(!n||4&_)&&L(r,s,e,e[2],n?x(s,e[2],_,null):j(e[2]),null),U(a,u=B(o,[2&_&&e[1]])),D(a,"svelte-1xzxuva",!0)},i(e){n||(i(r,e),n=!0)},o(e){c(r,e),n=!1},d(e){e&&p(a),l&&l.d(),r&&r.d(e)}}}function gn(e,a,t){const n=["title"];let l=F(a,n),{$$slots:_={},$$scope:s}=a,{title:d}=a,{klass:i,...r}=l;return i=i?`pipen-report-descr ${i}`:"pipen-report-descr",r||(r={}),r.class=i,e.$$set=e=>{a=M(M({},a),w(e)),t(5,l=F(a,n)),"title"in e&&t(0,d=e.title),"$$scope"in e&&t(2,s=e.$$scope)},[d,r,s,_]}class Pn extends a{constructor(e){super(),t(this,e,gn,mn,n,{title:0})}}function An(e){let a,t,n,l,_,s;return a=new ve({props:{label:"Significant Markers"}}),n=new ve({props:{label:"Volcano Plot"}}),_=new ve({props:{label:"Dot Plot"}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment),l=f(),g(_.$$.fragment)},m(e,i){P(a,e,i),d(e,t,i),P(n,e,i),d(e,l,i),P(_,e,i),s=!0},p:T,i(e){s||(i(a.$$.fragment,e),i(n.$$.fragment,e),i(_.$$.fragment,e),s=!0)},o(e){c(a.$$.fragment,e),c(n.$$.fragment,e),c(_.$$.fragment,e),s=!1},d(e){e&&(p(t),p(l)),A(a,e),A(n,e),A(_,e)}}}function fn(e){let a;return{c(){a=S("The markers are found using Seurat's FindMarkers function, and filtered by: p_val_adj < 0.05")},m(e,t){d(e,a,t)},d(e){e&&p(a)}}}function bn(e){let a,t,n,l;return a=new Pn({props:{$$slots:{default:[fn]},$$scope:{ctx:e}}}),n=new un({props:{src:"../MarkersFinder/samples.markers/c1/markers.txt",data:[{gene:"HLA-B",p_val:3104e-90,avg_log2FC:.9855,pct_1:1,pct_2:1,p_val_adj:1041e-85,id:0},{gene:"RPS14",p_val:2289e-51,avg_log2FC:-.5403,pct_1:1,pct_2:1,p_val_adj:7676e-47,id:1},{gene:"RPL32",p_val:1537e-44,avg_log2FC:-.5014,pct_1:.997,pct_2:1,p_val_adj:5155e-40,id:2},{gene:"PSMB9",p_val:1822e-40,avg_log2FC:1.1103,pct_1:.889,pct_2:.704,p_val_adj:6109e-36,id:3},{gene:"RPL30",p_val:4346e-39,avg_log2FC:-.4639,pct_1:1,pct_2:1,p_val_adj:1458e-34,id:4},{gene:"DUSP4",p_val:4766e-39,avg_log2FC:1.9484,pct_1:.649,pct_2:.293,p_val_adj:1598e-34,id:5},{gene:"CXCR3",p_val:7118e-36,avg_log2FC:2.2574,pct_1:.465,pct_2:.135,p_val_adj:2387e-31,id:6},{gene:"ISG20",p_val:1374e-35,avg_log2FC:1.0657,pct_1:.933,pct_2:.751,p_val_adj:4606e-31,id:7},{gene:"RPS3A",p_val:4295e-34,avg_log2FC:-.4227,pct_1:.997,pct_2:1,p_val_adj:144e-28,id:8},{gene:"CST7",p_val:6.836e-30,avg_log2FC:1.7166,pct_1:.738,pct_2:.474,p_val_adj:2293e-28,id:9},{gene:"RPL19",p_val:1577e-32,avg_log2FC:-.3967,pct_1:1,pct_2:1,p_val_adj:529e-27,id:10},{gene:"RPS12",p_val:1772e-32,avg_log2FC:-.4429,pct_1:.997,pct_2:1,p_val_adj:5942e-28,id:11},{gene:"DUSP1",p_val:1932e-32,avg_log2FC:-.9563,pct_1:.933,pct_2:.976,p_val_adj:6478e-28,id:12},{gene:"RPL21",p_val:2215e-32,avg_log2FC:-.5329,pct_1:.992,pct_2:.994,p_val_adj:7428e-28,id:13},{gene:"RPL13",p_val:3286e-31,avg_log2FC:-.3868,pct_1:.997,pct_2:1,p_val_adj:1102e-26,id:14},{gene:"RPL10A",p_val:5132e-31,avg_log2FC:-.5052,pct_1:.992,pct_2:.996,p_val_adj:1721e-26,id:15},{gene:"GPR171",p_val:9615e-30,avg_log2FC:3.2452,pct_1:.259,pct_2:.038,p_val_adj:3225e-25,id:16},{gene:"RPL34",p_val:1519e-29,avg_log2FC:-.4338,pct_1:.997,pct_2:1,p_val_adj:5096e-25,id:17},{gene:"FOS",p_val:2417e-29,avg_log2FC:-1.473,pct_1:.646,pct_2:.847,p_val_adj:8105e-25,id:18},{gene:"PDE3B",p_val:1534e-28,avg_log2FC:1.8866,pct_1:.351,pct_2:.088,p_val_adj:5144e-24,id:19},{gene:"RNF213",p_val:8151e-28,avg_log2FC:1.0996,pct_1:.822,pct_2:.581,p_val_adj:2.734e-20,id:20},{gene:"CCL5",p_val:8168e-28,avg_log2FC:4.1162,pct_1:.217,pct_2:.025,p_val_adj:2.739e-20,id:21},{gene:"FKBP5",p_val:1267e-27,avg_log2FC:1.8825,pct_1:.552,pct_2:.262,p_val_adj:4.251e-20,id:22},{gene:"RPS18",p_val:2537e-27,avg_log2FC:-.4335,pct_1:.997,pct_2:1,p_val_adj:8.508e-20,id:23},{gene:"RPL10",p_val:8472e-27,avg_log2FC:-.3391,pct_1:1,pct_2:1,p_val_adj:2841e-22,id:24},{gene:"RPS23",p_val:3243e-26,avg_log2FC:-.3647,pct_1:1,pct_2:1,p_val_adj:1088e-21,id:25},{gene:"RPS3",p_val:4351e-26,avg_log2FC:-.3425,pct_1:.997,pct_2:1,p_val_adj:1459e-21,id:26},{gene:"RPL3",p_val:3462e-25,avg_log2FC:-.4044,pct_1:.994,pct_2:1,p_val_adj:1161e-20,id:27},{gene:"FOSB",p_val:3693e-25,avg_log2FC:-1.8982,pct_1:.337,pct_2:.597,p_val_adj:1238e-20,id:28},{gene:"RPL22",p_val:6976e-25,avg_log2FC:-.4233,pct_1:.986,pct_2:.994,p_val_adj:234e-19,id:29},{gene:"LMO4",p_val:1855e-24,avg_log2FC:4.2601,pct_1:.178,pct_2:.018,p_val_adj:622e-19,id:30},{gene:"RPL17",p_val:2735e-24,avg_log2FC:-.367,pct_1:.992,pct_2:.999,p_val_adj:9173e-20,id:31},{gene:"TAP1",p_val:2862e-24,avg_log2FC:1.0883,pct_1:.749,pct_2:.535,p_val_adj:9599e-20,id:32},{gene:"EEF1A1",p_val:3561e-24,avg_log2FC:-.2877,pct_1:1,pct_2:1,p_val_adj:1194e-19,id:33},{gene:"RPS6",p_val:3595e-24,avg_log2FC:-.3766,pct_1:.997,pct_2:1,p_val_adj:1206e-19,id:34},{gene:"RPS27A",p_val:3628e-24,avg_log2FC:-.3312,pct_1:1,pct_2:1,p_val_adj:1217e-19,id:35},{gene:"PSME2",p_val:1.601e-20,avg_log2FC:.8151,pct_1:.872,pct_2:.706,p_val_adj:537e-18,id:36},{gene:"RPLP2",p_val:4.32e-20,avg_log2FC:-.3636,pct_1:.997,pct_2:.999,p_val_adj:0,id:37},{gene:"RPL26",p_val:5.442e-20,avg_log2FC:-.3125,pct_1:.994,pct_2:1,p_val_adj:0,id:38},{gene:"RPS5",p_val:6.595e-20,avg_log2FC:-.4127,pct_1:.997,pct_2:1,p_val_adj:0,id:39},{gene:"IFITM2",p_val:6.971e-20,avg_log2FC:.9357,pct_1:.83,pct_2:.65,p_val_adj:0,id:40},{gene:"UBE2L6",p_val:1396e-22,avg_log2FC:1.0317,pct_1:.696,pct_2:.457,p_val_adj:0,id:41},{gene:"SAT1",p_val:1503e-22,avg_log2FC:1.1055,pct_1:.894,pct_2:.781,p_val_adj:0,id:42},{gene:"FKBP11",p_val:1649e-22,avg_log2FC:2.4745,pct_1:.253,pct_2:.06,p_val_adj:0,id:43},{gene:"IFNG",p_val:2272e-22,avg_log2FC:4.3227,pct_1:.139,pct_2:.007,p_val_adj:0,id:44},{gene:"RPL12",p_val:2356e-22,avg_log2FC:-.3396,pct_1:.997,pct_2:1,p_val_adj:0,id:45},{gene:"CD74",p_val:4395e-22,avg_log2FC:.8801,pct_1:.972,pct_2:.941,p_val_adj:0,id:46},{gene:"B2M",p_val:4574e-22,avg_log2FC:.3083,pct_1:1,pct_2:1,p_val_adj:0,id:47},{gene:"WARS",p_val:5206e-22,avg_log2FC:1.9718,pct_1:.323,pct_2:.103,p_val_adj:0,id:48},{gene:"SLC27A2",p_val:1664e-21,avg_log2FC:3.5301,pct_1:.153,pct_2:.015,p_val_adj:0,id:49},{gene:"SRGN",p_val:3619e-21,avg_log2FC:.5455,pct_1:.994,pct_2:.996,p_val_adj:0,id:50},{gene:"LY6E",p_val:468e-20,avg_log2FC:.8654,pct_1:.808,pct_2:.616,p_val_adj:0,id:51},{gene:"RPL9",p_val:5026e-21,avg_log2FC:-.3628,pct_1:.994,pct_2:.997,p_val_adj:0,id:52},{gene:"RPS8",p_val:5898e-21,avg_log2FC:-.3367,pct_1:1,pct_2:1,p_val_adj:0,id:53},{gene:"RPS13",p_val:6743e-21,avg_log2FC:-.3418,pct_1:.997,pct_2:1,p_val_adj:0,id:54},{gene:"STAT1",p_val:6966e-21,avg_log2FC:1.1925,pct_1:.677,pct_2:.435,p_val_adj:0,id:55},{gene:"RPL11",p_val:1061e-20,avg_log2FC:-.2963,pct_1:1,pct_2:1,p_val_adj:0,id:56},{gene:"ETV7",p_val:2328e-20,avg_log2FC:1.6641,pct_1:.312,pct_2:.101,p_val_adj:0,id:57},{gene:"RPS15",p_val:2523e-20,avg_log2FC:-.2924,pct_1:.997,pct_2:1,p_val_adj:0,id:58},{gene:"AL157402.2",p_val:2945e-20,avg_log2FC:2.8371,pct_1:.22,pct_2:.051,p_val_adj:0,id:59},{gene:"RPS21",p_val:4547e-20,avg_log2FC:-.4003,pct_1:.992,pct_2:.996,p_val_adj:0,id:60},{gene:"RARRES3",p_val:5167e-20,avg_log2FC:1.0253,pct_1:.816,pct_2:.649,p_val_adj:0,id:61},{gene:"TXNIP",p_val:7543e-20,avg_log2FC:-.9098,pct_1:.529,pct_2:.747,p_val_adj:0,id:62},{gene:"DUSP2",p_val:1672e-19,avg_log2FC:.809,pct_1:.861,pct_2:.706,p_val_adj:0,id:63},{gene:"GNA15",p_val:1783e-19,avg_log2FC:2.1936,pct_1:.212,pct_2:.047,p_val_adj:0,id:64},{gene:"MT-ND5",p_val:3551e-19,avg_log2FC:-.4642,pct_1:.975,pct_2:.988,p_val_adj:0,id:65},{gene:"BHLHE40",p_val:4775e-19,avg_log2FC:1.9665,pct_1:.281,pct_2:.09,p_val_adj:0,id:66},{gene:"GAPDH",p_val:4896e-19,avg_log2FC:.4954,pct_1:1,pct_2:.997,p_val_adj:0,id:67},{gene:"RPS28",p_val:5021e-19,avg_log2FC:-.3324,pct_1:.997,pct_2:1,p_val_adj:0,id:68},{gene:"RPS25",p_val:6316e-19,avg_log2FC:-.3085,pct_1:1,pct_2:1,p_val_adj:0,id:69},{gene:"RPL35A",p_val:6522e-19,avg_log2FC:-.315,pct_1:.997,pct_2:1,p_val_adj:0,id:70},{gene:"SPINT2",p_val:6699e-19,avg_log2FC:-1.4845,pct_1:.153,pct_2:.403,p_val_adj:0,id:71},{gene:"IL12RB2",p_val:0,avg_log2FC:2.0678,pct_1:.22,pct_2:.054,p_val_adj:0,id:72},{gene:"EEF1B2",p_val:0,avg_log2FC:-.4358,pct_1:.967,pct_2:.984,p_val_adj:0,id:73},{gene:"RPL7",p_val:0,avg_log2FC:-.3847,pct_1:.967,pct_2:.981,p_val_adj:0,id:74},{gene:"RPL5",p_val:0,avg_log2FC:-.3126,pct_1:.994,pct_2:1,p_val_adj:0,id:75},{gene:"CREM",p_val:0,avg_log2FC:.8055,pct_1:.733,pct_2:.515,p_val_adj:0,id:76},{gene:"NPDC1",p_val:0,avg_log2FC:1.6766,pct_1:.276,pct_2:.091,p_val_adj:0,id:77},{gene:"ADAM19",p_val:0,avg_log2FC:2.0483,pct_1:.242,pct_2:.072,p_val_adj:0,id:78},{gene:"HLA-A",p_val:0,avg_log2FC:.4523,pct_1:1,pct_2:.999,p_val_adj:0,id:79},{gene:"RPL29",p_val:0,avg_log2FC:-.285,pct_1:.997,pct_2:.999,p_val_adj:0,id:80},{gene:"LGALS3BP",p_val:0,avg_log2FC:2.6144,pct_1:.175,pct_2:.035,p_val_adj:0,id:81},{gene:"RGS2",p_val:0,avg_log2FC:1.2165,pct_1:.454,pct_2:.219,p_val_adj:0,id:82},{gene:"MAP3K8",p_val:0,avg_log2FC:1.6692,pct_1:.331,pct_2:.135,p_val_adj:0,id:83},{gene:"SATB1",p_val:0,avg_log2FC:1.455,pct_1:.351,pct_2:.146,p_val_adj:0,id:84},{gene:"RPL18A",p_val:0,avg_log2FC:-.2682,pct_1:1,pct_2:1,p_val_adj:0,id:85},{gene:"HLA-DRB1",p_val:0,avg_log2FC:.9837,pct_1:.409,pct_2:.179,p_val_adj:0,id:86},{gene:"DYNLT1",p_val:0,avg_log2FC:.9522,pct_1:.602,pct_2:.385,p_val_adj:0,id:87},{gene:"RPL39",p_val:0,avg_log2FC:-.2898,pct_1:1,pct_2:.999,p_val_adj:0,id:88},{gene:"SH2D2A",p_val:0,avg_log2FC:.8556,pct_1:.721,pct_2:.559,p_val_adj:0,id:89},{gene:"MCOLN2",p_val:0,avg_log2FC:2.6274,pct_1:.153,pct_2:.029,p_val_adj:0,id:90},{gene:"CYTIP",p_val:0,avg_log2FC:.5718,pct_1:.928,pct_2:.841,p_val_adj:0,id:91},{gene:"RPL8",p_val:0,avg_log2FC:-.2589,pct_1:1,pct_2:1,p_val_adj:0,id:92},{gene:"EIF4B",p_val:0,avg_log2FC:-.584,pct_1:.752,pct_2:.857,p_val_adj:0,id:93},{gene:"RPS7",p_val:0,avg_log2FC:-.2424,pct_1:1,pct_2:1,p_val_adj:0,id:94},{gene:"CXCL13",p_val:0,avg_log2FC:7.4242,pct_1:.103,pct_2:.009,p_val_adj:0,id:95},{gene:"RPS16",p_val:0,avg_log2FC:-.292,pct_1:.997,pct_2:.999,p_val_adj:0,id:96},{gene:"FAU",p_val:0,avg_log2FC:-.2621,pct_1:.997,pct_2:1,p_val_adj:0,id:97},{gene:"OAZ1",p_val:0,avg_log2FC:.392,pct_1:.986,pct_2:.972,p_val_adj:0,id:98},{gene:"MT-ATP6",p_val:0,avg_log2FC:-.3113,pct_1:.994,pct_2:.994,p_val_adj:0,id:99}]}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment)},m(e,_){P(a,e,_),d(e,t,_),P(n,e,_),l=!0},p(e,t){const n={};1&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){l||(i(a.$$.fragment,e),i(n.$$.fragment,e),l=!0)},o(e){c(a.$$.fragment,e),c(n.$$.fragment,e),l=!1},d(e){e&&p(t),A(a,e),A(n,e)}}}function Cn(e){let a;return{c(){a=S("The volcano plot is generated using ggplot2. The significant markers are highlighted in blue and red, indicating up-regulated and down-regulated genes, respectively. The non-significant genes are shown in grey.")},m(e,t){d(e,a,t)},d(e){e&&p(a)}}}function On(e){let a,t,n,l;return a=new Pn({props:{$$slots:{default:[Cn]},$$scope:{ctx:e}}}),n=new Ht({props:{src:"../MarkersFinder/samples.markers/c1/volcano.png",download:"../MarkersFinder/samples.markers/c1/volcano.pdf",width:900,height:1200}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment)},m(e,_){P(a,e,_),d(e,t,_),P(n,e,_),l=!0},p(e,t){const n={};1&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){l||(i(a.$$.fragment,e),i(n.$$.fragment,e),l=!0)},o(e){c(a.$$.fragment,e),c(n.$$.fragment,e),l=!1},d(e){e&&p(t),A(a,e),A(n,e)}}}function Sn(e){let a;return{c(){a=S("The dot plot is generated using Seurat's DotPlot function. The top significant markers are used as input. The sizes of the dots are proportional to the percentage of cells expressing the gene in each group. The colors of the dots are proportional to the average expression of the gene in each group.")},m(e,t){d(e,a,t)},d(e){e&&p(a)}}}function hn(e){let a,t,n,l;return a=new Pn({props:{$$slots:{default:[Sn]},$$scope:{ctx:e}}}),n=new Ht({props:{src:"../MarkersFinder/samples.markers/c1/dotplot.png",download:"../MarkersFinder/samples.markers/c1/dotplot.pdf",width:600,height:1200}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment)},m(e,_){P(a,e,_),d(e,t,_),P(n,e,_),l=!0},p(e,t){const n={};1&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){l||(i(a.$$.fragment,e),i(n.$$.fragment,e),l=!0)},o(e){c(a.$$.fragment,e),c(n.$$.fragment,e),l=!1},d(e){e&&p(t),A(a,e),A(n,e)}}}function Rn(e){let a,t,n,l,_,s;return a=new me({props:{$$slots:{default:[bn]},$$scope:{ctx:e}}}),n=new me({props:{$$slots:{default:[On]},$$scope:{ctx:e}}}),_=new me({props:{$$slots:{default:[hn]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment),l=f(),g(_.$$.fragment)},m(e,i){P(a,e,i),d(e,t,i),P(n,e,i),d(e,l,i),P(_,e,i),s=!0},p(e,t){const l={};1&t&&(l.$$scope={dirty:t,ctx:e}),a.$set(l);const s={};1&t&&(s.$$scope={dirty:t,ctx:e}),n.$set(s);const d={};1&t&&(d.$$scope={dirty:t,ctx:e}),_.$set(d)},i(e){s||(i(a.$$.fragment,e),i(n.$$.fragment,e),i(_.$$.fragment,e),s=!0)},o(e){c(a.$$.fragment,e),c(n.$$.fragment,e),c(_.$$.fragment,e),s=!1},d(e){e&&(p(t),p(l)),A(a,e),A(n,e),A(_,e)}}}function $n(e){let a;return{c(){a=S("The enrichment analysis is done using Enrichr. The significant markers are used as input.")},m(e,t){d(e,a,t)},d(e){e&&p(a)}}}function Tn(e){let a,t,n,l;return a=new ve({props:{label:"Plot"}}),n=new ve({props:{label:"Table"}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment)},m(e,_){P(a,e,_),d(e,t,_),P(n,e,_),l=!0},p:T,i(e){l||(i(a.$$.fragment,e),i(n.$$.fragment,e),l=!0)},o(e){c(a.$$.fragment,e),c(n.$$.fragment,e),l=!1},d(e){e&&p(t),A(a,e),A(n,e)}}}function yn(e){let a;return{c(){a=S("The enrichment analysis results of the top biological pathways associated with the input gene set. Each bar represents a pathway, with the length of the bar indicating the number of input genes overlapping with genes in that pathway. The color intensity of the bars reflects the statistical significance of the enrichment (p-value).")},m(e,t){d(e,a,t)},d(e){e&&p(a)}}}function Ln(e){let a,t,n,l;return a=new Pn({props:{$$slots:{default:[yn]},$$scope:{ctx:e}}}),n=new Ht({props:{src:"../MarkersFinder/samples.markers/c1/Enrichr-KEGG_2021_Human.png",download:"../MarkersFinder/samples.markers/c1/Enrichr-KEGG_2021_Human.pdf",width:1e3,height:1e3}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment)},m(e,_){P(a,e,_),d(e,t,_),P(n,e,_),l=!0},p(e,t){const n={};1&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){l||(i(a.$$.fragment,e),i(n.$$.fragment,e),l=!0)},o(e){c(a.$$.fragment,e),c(n.$$.fragment,e),l=!1},d(e){e&&p(t),A(a,e),A(n,e)}}}function jn(e){let a,t;return a=new un({props:{src:"../MarkersFinder/samples.markers/c1/Enrichr-KEGG_2021_Human.txt",data:[{Term:"Ribosome",Overlap:"61/158",P_value:5.843e-70,Adjusted_P_value:1134e-70,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:49.8892,Combined_Score:7953.1203,Genes:"RPL5;RPL30;RPL3;RPL32;RPL34;RPL10A;RPL8;RPL9;RPL6;RPL7;RPS15;RPS4X;RPS14;RPL7A;RPS16;RPL18A;RPS18;RPL36;RPL35;RPLP2;RPL38;RPL37;RPS11;RPL39;RPS13;RPS12;RPL21;RPS7;RPS8;RPS5;RPL22;RPS6;RPS3A;RPL24;RPL26;RPL29;RPL28;UBA52;RPL10;RPL12;RPL11;RPL36A;MRPL10;RPS3;RPL13;RPL15;RPL18;RPS27A;RPL17;RPL19;RPL41;RPL35A;RPL23A;RPS25;RPS28;RPS27;RPL27A;RPS20;FAU;RPS21;RPS23",id:0},{Term:"Coronavirus disease",Overlap:"64/232",P_value:1603e-65,Adjusted_P_value:1.555e-60,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:30.4824,Combined_Score:4337.3015,Genes:"RPL5;RPL30;RPL3;CSF2;RPL32;RPL34;RPL8;RPL10A;RPL9;RPL6;RPL7;RPS15;RPS4X;RPS14;RPL7A;RPS16;RPL18A;RPS18;RPL36;RPL35;RPLP2;RPL38;RPL37;RPS11;RPL39;RPS13;RPS12;RPL21;RPS7;RPS8;RPS5;RPL22;RPS6;RPS3A;FOS;RPL24;RPL26;RPL29;RPL28;UBA52;RPL10;RPL12;RPL11;RPL36A;RPS3;RPL13;RPL15;RPL18;RPS27A;RPL17;RPL19;RPL41;STAT1;RPL35A;ISG15;RPL23A;RPS25;RPS28;RPS27;RPL27A;RPS20;FAU;RPS21;RPS23",id:1},{Term:"Antigen processing and presentation",Overlap:"13/78",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:13.3064,Combined_Score:297.7508,Genes:"CD74;HLA-B;TAP1;HLA-A;CD8B;IFNG;CD8A;PSME1;PSME2;HLA-DRA;B2M;HLA-DRB1;HLA-DPA1",id:2},{Term:"Proteasome",Overlap:"9/46",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:15.9898,Combined_Score:277.5386,Genes:"PSMA5;PSMB2;IFNG;PSMA1;PSME1;PSME2;PSMB10;PSMB8;PSMB9",id:3},{Term:"Type I diabetes mellitus",Overlap:"8/43",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:14.9768,Combined_Score:226.9598,Genes:"IFNG;ICA1;HLA-B;GZMB;HLA-DRA;HLA-A;HLA-DRB1;HLA-DPA1",id:4},{Term:"Epstein-Barr virus infection",Overlap:"15/202",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:5.3398,Combined_Score:76.5454,Genes:"STAT1;HLA-B;TAP1;TNFAIP3;ISG15;HLA-A;RUNX3;RELB;BCL2L11;IRF7;HLA-DRA;VIM;B2M;HLA-DRB1;HLA-DPA1",id:5},{Term:"Inflammatory bowel disease",Overlap:"9/65",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:10.5545,Combined_Score:150.3555,Genes:"IL22;IL18RAP;IFNG;STAT1;TBX21;HLA-DRA;HLA-DRB1;HLA-DPA1;IL12RB2",id:6},{Term:"Th1 and Th2 cell differentiation",Overlap:"10/92",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:8.025,Combined_Score:107.4409,Genes:"IFNG;STAT1;TBX21;NFATC2;HLA-DRA;FOS;RUNX3;HLA-DRB1;HLA-DPA1;IL12RB2",id:7},{Term:"Allograft rejection",Overlap:"7/38",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:14.7494,Combined_Score:196.7667,Genes:"IFNG;HLA-B;GZMB;HLA-DRA;HLA-A;HLA-DRB1;HLA-DPA1",id:8},{Term:"Graft-versus-host disease",Overlap:"7/42",P_value:0,Adjusted_P_value:1e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:13.0611,Combined_Score:165.0093,Genes:"IFNG;HLA-B;GZMB;HLA-DRA;HLA-A;HLA-DRB1;HLA-DPA1",id:9},{Term:"Osteoclast differentiation",Overlap:"11/127",P_value:0,Adjusted_P_value:1e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:6.2503,Combined_Score:77.0815,Genes:"SOCS1;IFNG;STAT1;NFATC2;ACP5;FOSB;CYBA;FYN;FOS;JUNB;RELB",id:10},{Term:"T cell receptor signaling pathway",Overlap:"10/104",P_value:0,Adjusted_P_value:1e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:6.9963,Combined_Score:85.805,Genes:"CSF2;CD8B;IFNG;CD8A;NFATC2;CBLB;MAP3K8;FYN;FOS;ICOS",id:11},{Term:"Hematopoietic cell lineage",Overlap:"9/99",P_value:0,Adjusted_P_value:3e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:6.5559,Combined_Score:70.2206,Genes:"CD2;CSF2;CD8B;CD8A;CD7;HLA-DRA;CD38;HLA-DRB1;HLA-DPA1",id:12},{Term:"Leishmaniasis",Overlap:"8/77",P_value:0,Adjusted_P_value:3e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:7.5838,Combined_Score:80.6151,Genes:"EEF1A1;IFNG;STAT1;HLA-DRA;CYBA;FOS;HLA-DRB1;HLA-DPA1",id:13},{Term:"Viral myocarditis",Overlap:"7/60",P_value:0,Adjusted_P_value:5e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:8.6174,Combined_Score:87.9754,Genes:"RAC2;HLA-B;HLA-DRA;FYN;HLA-A;HLA-DRB1;HLA-DPA1",id:14},{Term:"Th17 cell differentiation",Overlap:"9/107",P_value:0,Adjusted_P_value:5e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:6.0182,Combined_Score:60.7256,Genes:"IL22;IFNG;STAT1;TBX21;NFATC2;HLA-DRA;FOS;HLA-DRB1;HLA-DPA1",id:15},{Term:"Rheumatoid arthritis",Overlap:"8/93",P_value:1e-4,Adjusted_P_value:.0011,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:6.1512,Combined_Score:56.9946,Genes:"CSF2;IFNG;CCL5;ACP5;HLA-DRA;FOS;HLA-DRB1;HLA-DPA1",id:16},{Term:"Autoimmune thyroid disease",Overlap:"6/53",P_value:2e-4,Adjusted_P_value:.0017,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:8.3042,Combined_Score:72.6495,Genes:"HLA-B;GZMB;HLA-DRA;HLA-A;HLA-DRB1;HLA-DPA1",id:17},{Term:"Fluid shear stress and atherosclerosis",Overlap:"9/139",P_value:3e-4,Adjusted_P_value:.0032,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.5294,Combined_Score:36.6081,Genes:"IFNG;DUSP1;MGST3;SUMO2;RAC2;PLAT;CYBA;TXN;FOS",id:18},{Term:"TNF signaling pathway",Overlap:"8/112",P_value:3e-4,Adjusted_P_value:.0033,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:5.0226,Combined_Score:40.0615,Genes:"CSF2;IRF1;CCL5;TNFAIP3;MAP3K8;FOS;CFLAR;JUNB",id:19},{Term:"Cell adhesion molecules",Overlap:"9/148",P_value:5e-4,Adjusted_P_value:.0044,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.2342,Combined_Score:32.2702,Genes:"CD2;CD8B;CD8A;HLA-B;HLA-DRA;HLA-A;ICOS;HLA-DRB1;HLA-DPA1",id:20},{Term:"Parkinson disease",Overlap:"12/249",P_value:5e-4,Adjusted_P_value:.0044,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.3279,Combined_Score:25.2389,Genes:"PSMA5;PSMA1;PSMB2;DUSP1;TUBB;NDUFS5;UBE2L6;TXN;RPS27A;SLC25A5;UBA52;COX5A",id:21},{Term:"NOD-like receptor signaling pathway",Overlap:"10/181",P_value:5e-4,Adjusted_P_value:.0044,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.8308,Combined_Score:28.9928,Genes:"PYCARD;STAT1;CCL5;IRF7;TXNIP;TNFAIP3;CYBA;TXN;GBP1;GBP4",id:22},{Term:"Human T-cell leukemia virus 1 infection",Overlap:"11/219",P_value:6e-4,Adjusted_P_value:.005,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.4694,Combined_Score:25.6591,Genes:"CSF2;HLA-B;NFATC2;HLA-DRA;HLA-A;FOS;SLC25A5;B2M;HLA-DRB1;HLA-DPA1;RELB",id:23},{Term:"Cytokine-cytokine receptor interaction",Overlap:"13/295",P_value:7e-4,Adjusted_P_value:.0054,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.0332,Combined_Score:22.0418,Genes:"IL32;IL22;CSF2;TNFRSF18;IL16;CXCL13;IL18RAP;IFNG;CXCR3;CCL5;TNFRSF25;TNFRSF4;IL12RB2",id:24},{Term:"Natural killer cell mediated cytotoxicity",Overlap:"8/131",P_value:.001,Adjusted_P_value:.0073,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.2426,Combined_Score:29.4254,Genes:"CSF2;IFNG;HLA-B;RAC2;NFATC2;GZMB;FYN;HLA-A",id:25},{Term:"Prion disease",Overlap:"12/273",P_value:.0011,Adjusted_P_value:.0082,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.0182,Combined_Score:20.4666,Genes:"PSMA5;PSMA1;PSMB2;TUBB;NDUFS5;CCL5;RAC2;CYBA;FYN;SLC25A5;COX5A;SOD1",id:26},{Term:"Influenza A",Overlap:"9/172",P_value:.0014,Adjusted_P_value:.0099,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.6063,Combined_Score:23.6322,Genes:"PYCARD;IFNG;STAT1;CCL5;IRF7;HLA-DRA;SLC25A5;HLA-DRB1;HLA-DPA1",id:27},{Term:"Phagosome",Overlap:"8/152",P_value:.0025,Adjusted_P_value:.0167,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.62,Combined_Score:21.6903,Genes:"TUBB;HLA-B;TAP1;HLA-DRA;CYBA;HLA-A;HLA-DRB1;HLA-DPA1",id:28},{Term:"Primary immunodeficiency",Overlap:"4/38",P_value:.0027,Adjusted_P_value:.0174,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:7.6076,Combined_Score:45.008,Genes:"CD8B;CD8A;TAP1;ICOS",id:29},{Term:"Kaposi sarcoma-associated herpesvirus infection",Overlap:"9/193",P_value:.0031,Adjusted_P_value:.0195,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.1913,Combined_Score:18.4176,Genes:"CSF2;STAT1;HLA-B;IRF7;NFATC2;HLA-A;FOS;RPS27A;UBA52",id:30},{Term:"JAK-STAT signaling pathway",Overlap:"8/162",P_value:.0037,Adjusted_P_value:.0224,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.3832,Combined_Score:18.9508,Genes:"IL22;CSF2;SOCS1;IFNG;STAT1;PIM1;MCL1;IL12RB2",id:31},{Term:"C-type lectin receptor signaling pathway",Overlap:"6/104",P_value:.0055,Adjusted_P_value:.0321,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.9723,Combined_Score:20.6939,Genes:"PYCARD;STAT1;IRF1;NFATC2;CBLB;RELB",id:32},{Term:"Intestinal immune network for IgA production",Overlap:"4/48",P_value:.0063,Adjusted_P_value:.036,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:5.8756,Combined_Score:29.7712,Genes:"HLA-DRA;ICOS;HLA-DRB1;HLA-DPA1",id:33},{Term:"Apoptosis",Overlap:"7/142",P_value:.0065,Adjusted_P_value:.0361,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.369,Combined_Score:16.9555,Genes:"BCL2L11;PMAIP1;GZMB;FOS;CFLAR;MCL1;LMNB1",id:34},{Term:"Toxoplasmosis",Overlap:"6/112",P_value:.0078,Adjusted_P_value:.0419,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.671,Combined_Score:17.825,Genes:"SOCS1;IFNG;STAT1;HLA-DRA;HLA-DRB1;HLA-DPA1",id:35},{Term:"Necroptosis",Overlap:"7/159",P_value:.0118,Adjusted_P_value:.0603,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.9896,Combined_Score:13.2846,Genes:"PYCARD;IFNG;STAT1;TNFAIP3;HMGB1;CFLAR;SLC25A5",id:36},{Term:"Asthma",Overlap:"3/31",P_value:.0118,Adjusted_P_value:.0603,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:6.9077,Combined_Score:30.656,Genes:"HLA-DRA;HLA-DRB1;HLA-DPA1",id:37},{Term:"IL-17 signaling pathway",Overlap:"5/94",P_value:.0151,Adjusted_P_value:.0752,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.6346,Combined_Score:15.2348,Genes:"CSF2;IFNG;TNFAIP3;FOSB;FOS",id:38},{Term:"Human immunodeficiency virus 1 infection",Overlap:"8/212",P_value:.0173,Adjusted_P_value:.0837,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.5475,Combined_Score:10.3401,Genes:"APOBEC3G;HLA-B;RAC2;NFATC2;TAP1;HLA-A;FOS;B2M",id:39},{Term:"Yersinia infection",Overlap:"6/137",P_value:.0195,Adjusted_P_value:.0923,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.9666,Combined_Score:11.6786,Genes:"PYCARD;CD8B;CD8A;RAC2;NFATC2;FOS",id:40},{Term:"Chagas disease",Overlap:"5/102",P_value:.0208,Adjusted_P_value:.0961,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.3335,Combined_Score:12.9098,Genes:"GNA15;IFNG;CCL5;FOS;CFLAR",id:41},{Term:"Toll-like receptor signaling pathway",Overlap:"5/104",P_value:.0224,Adjusted_P_value:.1008,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.2658,Combined_Score:12.4042,Genes:"STAT1;CCL5;IRF7;MAP3K8;FOS",id:42},{Term:"Prolactin signaling pathway",Overlap:"4/70",P_value:.0229,Adjusted_P_value:.1008,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.9127,Combined_Score:14.7837,Genes:"SOCS1;STAT1;IRF1;FOS",id:43},{Term:"Parathyroid hormone synthesis, secretion and action",Overlap:"5/106",P_value:.0241,Adjusted_P_value:.1039,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.2008,Combined_Score:11.9248,Genes:"NR4A2;NACA;PDE4B;FOS;RUNX2",id:44},{Term:"Transcriptional misregulation in cancer",Overlap:"7/192",P_value:.0296,Adjusted_P_value:.1246,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.4522,Combined_Score:8.6355,Genes:"ETV7;CSF2;NFKBIZ;PRCC;GZMB;PLAT;RUNX2",id:45},{Term:"Pathogenic Escherichia coli infection",Overlap:"7/197",P_value:.0333,Adjusted_P_value:.1374,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.387,Combined_Score:8.1213,Genes:"PYCARD;TUBB;F2R;RPS3;FYN;FOS;GAPDH",id:46},{Term:"B cell receptor signaling pathway",Overlap:"4/81",P_value:.0364,Adjusted_P_value:.1473,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.3518,Combined_Score:11.1018,Genes:"IFITM1;RAC2;NFATC2;FOS",id:47},{Term:"Diabetic cardiomyopathy",Overlap:"7/203",P_value:.0382,Adjusted_P_value:.1512,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.3132,Combined_Score:7.553,Genes:"TBC1D4;NDUFS5;RAC2;CYBA;SLC25A5;COX5A;GAPDH",id:48},{Term:"Salmonella infection",Overlap:"8/249",P_value:.0396,Adjusted_P_value:.1536,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.1523,Combined_Score:6.9499,Genes:"PYCARD;AHNAK;DYNLT1;TUBB;RPS3;TXN;FOS;GAPDH",id:49},{Term:"Colorectal cancer",Overlap:"4/86",P_value:.0439,Adjusted_P_value:.1668,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.1467,Combined_Score:9.8388,Genes:"BCL2L11;RAC2;PMAIP1;FOS",id:50},{Term:"Herpes simplex virus 1 infection",Overlap:"13/498",P_value:.0457,Adjusted_P_value:.1703,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.7452,Combined_Score:5.3866,Genes:"CD74;STAT1;HLA-B;TAP1;HLA-A;IFNG;CCL5;IRF7;HLA-DRA;B2M;SRSF7;HLA-DRB1;HLA-DPA1",id:51},{Term:"PD-L1 expression and PD-1 checkpoint pathway in cancer",Overlap:"4/89",P_value:.0487,Adjusted_P_value:.1782,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.0351,Combined_Score:9.1729,Genes:"IFNG;STAT1;NFATC2;FOS",id:52},{Term:"Legionellosis",Overlap:"3/57",P_value:.0576,Adjusted_P_value:.2069,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.577,Combined_Score:10.2102,Genes:"PYCARD;EEF1A1;EEF1G",id:53},{Term:"Human cytomegalovirus infection",Overlap:"7/225",P_value:.0601,Adjusted_P_value:.2076,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.0774,Combined_Score:5.8421,Genes:"CCL5;HLA-B;RAC2;NFATC2;TAP1;HLA-A;B2M",id:54},{Term:"Tuberculosis",Overlap:"6/180",P_value:.0604,Adjusted_P_value:.2076,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.2286,Combined_Score:6.2564,Genes:"CD74;IFNG;STAT1;HLA-DRA;HLA-DRB1;HLA-DPA1",id:55},{Term:"Insulin signaling pathway",Overlap:"5/137",P_value:.061,Adjusted_P_value:.2076,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.4452,Combined_Score:6.8397,Genes:"SOCS1;PRKAR1B;RPS6;PDE3B;CBLB",id:56},{Term:"VEGF signaling pathway",Overlap:"3/59",P_value:.0626,Adjusted_P_value:.2094,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.4489,Combined_Score:9.5573,Genes:"SH2D2A;RAC2;NFATC2",id:57},{Term:"Measles",Overlap:"5/139",P_value:.0641,Adjusted_P_value:.2107,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.4085,Combined_Score:6.6178,Genes:"STAT1;IRF7;TNFAIP3;CBLB;FOS",id:58},{Term:"Ubiquitin mediated proteolysis",Overlap:"5/140",P_value:.0657,Adjusted_P_value:.2123,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.3905,Combined_Score:6.5102,Genes:"SOCS1;CBLB;UBE2L6;RPS27A;UBA52",id:59},{Term:"Viral protein interaction with cytokine and cytokine receptor",Overlap:"4/100",P_value:.0688,Adjusted_P_value:.2188,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.6859,Combined_Score:7.1888,Genes:"IL18RAP;CCL5;CXCR3;CXCL13",id:60},{Term:"Cytosolic DNA-sensing pathway",Overlap:"3/63",P_value:.0732,Adjusted_P_value:.229,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.2184,Combined_Score:8.4156,Genes:"PYCARD;CCL5;IRF7",id:61},{Term:"Glycolysis / Gluconeogenesis",Overlap:"3/67",P_value:.0845,Adjusted_P_value:.2586,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.0166,Combined_Score:7.4542,Genes:"LDHA;GALM;GAPDH",id:62},{Term:"MAPK signaling pathway",Overlap:"8/294",P_value:.0857,Adjusted_P_value:.2586,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.8094,Combined_Score:4.4447,Genes:"DUSP4;DUSP5;DUSP2;DUSP1;RAC2;MAP3K8;FOS;RELB",id:63},{Term:"Fc epsilon RI signaling pathway",Overlap:"3/68",P_value:.0874,Adjusted_P_value:.2586,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.97,Combined_Score:7.2375,Genes:"CSF2;RAC2;FYN",id:64},{Term:"HIF-1 signaling pathway",Overlap:"4/109",P_value:.088,Adjusted_P_value:.2586,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.4545,Combined_Score:5.9665,Genes:"LDHA;IFNG;RPS6;GAPDH",id:65},{Term:"Oxytocin signaling pathway",Overlap:"5/154",P_value:.09,Adjusted_P_value:.2606,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.1644,Combined_Score:5.2116,Genes:"RGS2;MYL6;NFATC2;CD38;FOS",id:66},{Term:"Hepatitis C",Overlap:"5/157",P_value:.0958,Adjusted_P_value:.2732,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.1213,Combined_Score:4.9766,Genes:"IFNG;YWHAB;STAT1;IRF7;CFLAR",id:67},{Term:"Huntington disease",Overlap:"8/306",P_value:.1018,Adjusted_P_value:.2863,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.7355,Combined_Score:3.9644,Genes:"PSMA5;PSMA1;PSMB2;TUBB;NDUFS5;SLC25A5;COX5A;SOD1",id:68},{Term:"Hepatitis B",Overlap:"5/162",P_value:.1057,Adjusted_P_value:.2931,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.0532,Combined_Score:4.6131,Genes:"YWHAB;STAT1;IRF7;NFATC2;FOS",id:69},{Term:"MicroRNAs in cancer",Overlap:"8/310",P_value:.1076,Adjusted_P_value:.2939,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.7121,Combined_Score:3.8176,Genes:"ZEB2;BCL2L11;SOCS1;DDIT4;PIM1;VIM;FOXP1;MCL1",id:70},{Term:"Growth hormone synthesis, secretion and action",Overlap:"4/119",P_value:.1119,Adjusted_P_value:.2988,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.2399,Combined_Score:4.9052,Genes:"SOCS1;STAT1;FOS;JUNB",id:71},{Term:"Pertussis",Overlap:"3/76",P_value:.1124,Adjusted_P_value:.2988,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.6435,Combined_Score:5.7769,Genes:"PYCARD;IRF1;FOS",id:72},{Term:"Lipid and atherosclerosis",Overlap:"6/215",P_value:.1157,Adjusted_P_value:.3014,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.8521,Combined_Score:3.9937,Genes:"PYCARD;CCL5;IRF7;NFATC2;CYBA;FOS",id:73},{Term:"Riboflavin metabolism",Overlap:"1/8",P_value:.1168,Adjusted_P_value:.3014,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:9.1601,Combined_Score:19.671,Genes:"ACP5",id:74},{Term:"Pathways of neurodegeneration",Overlap:"11/475",P_value:.1181,Adjusted_P_value:.3014,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.5348,Combined_Score:3.2788,Genes:"PSMA5;PSMA1;PSMB2;TUBB;NDUFS5;UBE2L6;RPS27A;SLC25A5;UBA52;COX5A;SOD1",id:75},{Term:"Systemic lupus erythematosus",Overlap:"4/135",P_value:.1555,Adjusted_P_value:.3917,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.9647,Combined_Score:3.6571,Genes:"IFNG;HLA-DRA;HLA-DRB1;HLA-DPA1",id:76},{Term:"Chemokine signaling pathway",Overlap:"5/192",P_value:.1753,Adjusted_P_value:.4361,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.7212,Combined_Score:2.9967,Genes:"STAT1;CCL5;CXCR3;RAC2;CXCL13",id:77},{Term:"Spinocerebellar ataxia",Overlap:"4/143",P_value:.1793,Adjusted_P_value:.4378,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.8509,Combined_Score:3.1814,Genes:"PSMA5;PSMA1;PSMB2;SLC25A5",id:78},{Term:"Staphylococcus aureus infection",Overlap:"3/95",P_value:.1806,Adjusted_P_value:.4378,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.0955,Combined_Score:3.587,Genes:"HLA-DRA;HLA-DRB1;HLA-DPA1",id:79},{Term:"Endocytosis",Overlap:"6/252",P_value:.1941,Adjusted_P_value:.4649,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.5705,Combined_Score:2.5746,Genes:"CAPZB;HLA-B;ARAP2;CBLB;HLA-A;SMAP2",id:80},{Term:"Amyotrophic lateral sclerosis",Overlap:"8/364",P_value:.2007,Adjusted_P_value:.4748,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.4484,Combined_Score:2.3261,Genes:"PSMA5;PSMA1;PSMB2;TUBB;NDUFS5;COX5A;SRSF7;SOD1",id:81},{Term:"Viral carcinogenesis",Overlap:"5/203",P_value:.2044,Adjusted_P_value:.4777,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.6247,Combined_Score:2.5796,Genes:"YWHAB;HLA-B;IRF7;PMAIP1;HLA-A",id:82},{Term:"Amoebiasis",Overlap:"3/102",P_value:.208,Adjusted_P_value:.4785,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.9466,Combined_Score:3.057,Genes:"GNA15;CSF2;IFNG",id:83},{Term:"Alzheimer disease",Overlap:"8/369",P_value:.2106,Adjusted_P_value:.4785,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.428,Combined_Score:2.2241,Genes:"PSMA5;PSMA1;PSMB2;TUBB;NDUFS5;SLC25A5;COX5A;GAPDH",id:84},{Term:"mTOR signaling pathway",Overlap:"4/154",P_value:.2138,Adjusted_P_value:.4785,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.7142,Combined_Score:2.6444,Genes:"RPS6;DDIT4;EIF4B;LAMTOR5",id:85},{Term:"NF-kappa B signaling pathway",Overlap:"3/104",P_value:.216,Adjusted_P_value:.4785,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.9079,Combined_Score:2.9241,Genes:"TNFAIP3;CFLAR;RELB",id:86},{Term:"Non-alcoholic fatty liver disease",Overlap:"4/155",P_value:.2171,Adjusted_P_value:.4785,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.7028,Combined_Score:2.6012,Genes:"BCL2L11;NDUFS5;FOS;COX5A",id:87},{Term:"Cellular senescence",Overlap:"4/156",P_value:.2203,Adjusted_P_value:.4802,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.6915,Combined_Score:2.5589,Genes:"HLA-B;NFATC2;HLA-A;SLC25A5",id:88},{Term:"cAMP signaling pathway",Overlap:"5/216",P_value:.2406,Adjusted_P_value:.5186,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.5235,Combined_Score:2.1706,Genes:"F2R;PDE3B;RAC2;PDE4B;FOS",id:89},{Term:"cGMP-PKG signaling pathway",Overlap:"4/167",P_value:.2567,Adjusted_P_value:.5473,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.5764,Combined_Score:2.1435,Genes:"RGS2;PDE3B;NFATC2;SLC25A5",id:90},{Term:"One carbon pool by folate",Overlap:"1/20",P_value:.267,Adjusted_P_value:.5629,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.3727,Combined_Score:4.4543,Genes:"MTHFD2",id:91},{Term:"Acute myeloid leukemia",Overlap:"2/67",P_value:.276,Adjusted_P_value:.5758,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.9736,Combined_Score:2.5404,Genes:"CSF2;PIM1",id:92},{Term:"Mitophagy",Overlap:"2/68",P_value:.2817,Adjusted_P_value:.5815,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.9436,Combined_Score:2.462,Genes:"RPS27A;UBA52",id:93},{Term:"Amphetamine addiction",Overlap:"2/69",P_value:.2875,Adjusted_P_value:.587,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.9144,Combined_Score:2.3867,Genes:"FOSB;FOS",id:94},{Term:"RIG-I-like receptor signaling pathway",Overlap:"2/70",P_value:.2932,Adjusted_P_value:.5924,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.8862,Combined_Score:2.3144,Genes:"IRF7;ISG15",id:95},{Term:"Adherens junction",Overlap:"2/71",P_value:.2989,Adjusted_P_value:.5975,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.8588,Combined_Score:2.245,Genes:"RAC2;FYN",id:96},{Term:"Pathways in cancer",Overlap:"10/531",P_value:.3033,Adjusted_P_value:.5975,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.2348,Combined_Score:1.4729,Genes:"BCL2L11;IFNG;STAT1;MGST3;F2R;RAC2;PIM1;PMAIP1;FOS;IL12RB2",id:97},{Term:"PI3K-Akt signaling pathway",Overlap:"7/354",P_value:.3049,Adjusted_P_value:.5975,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.2965,Combined_Score:1.5399,Genes:"BCL2L11;YWHAB;DDIT4;RPS6;F2R;EIF4B;MCL1",id:98},{Term:"Calcium signaling pathway",Overlap:"5/240",P_value:.3112,Adjusted_P_value:.6037,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.3663,Combined_Score:1.595,Genes:"GNA15;F2R;CD38;SLC25A5;MCOLN2",id:99},{Term:"PPAR signaling pathway",Overlap:"2/74",P_value:.3159,Adjusted_P_value:.6068,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.781,Combined_Score:2.0523,Genes:"GK;SLC27A2",id:100},{Term:"Pancreatic cancer",Overlap:"2/76",P_value:.3272,Adjusted_P_value:.6202,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.7327,Combined_Score:1.9356,Genes:"STAT1;RAC2",id:101},{Term:"Shigellosis",Overlap:"5/246",P_value:.3293,Adjusted_P_value:.6202,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.3318,Combined_Score:1.4794,Genes:"PYCARD;CSF2;CCL5;RPS27A;UBA52",id:102},{Term:"Neutrophil extracellular trap formation",Overlap:"4/189",P_value:.3325,Adjusted_P_value:.6202,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.3874,Combined_Score:1.5278,Genes:"RAC2;CYBA;HMGB1;SLC25A5",id:103},{Term:"Oxidative phosphorylation",Overlap:"3/133",P_value:.3365,Adjusted_P_value:.6217,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.4801,Combined_Score:1.6121,Genes:"PPA1;NDUFS5;COX5A",id:104},{Term:"RNA degradation",Overlap:"2/79",P_value:.3441,Adjusted_P_value:.6298,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.665,Combined_Score:1.7762,Genes:"BTG3;BTG2",id:105},{Term:"Apelin signaling pathway",Overlap:"3/137",P_value:.3533,Adjusted_P_value:.6347,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.4356,Combined_Score:1.4936,Genes:"RPS6;PDE3B;PLAT",id:106},{Term:"Autophagy",Overlap:"3/137",P_value:.3533,Adjusted_P_value:.6347,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.4356,Combined_Score:1.4936,Genes:"DDIT4;HMGB1;CFLAR",id:107},{Term:"Peroxisome",Overlap:"2/82",P_value:.3608,Adjusted_P_value:.6422,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.6023,Combined_Score:1.6332,Genes:"SLC27A2;SOD1",id:108},{Term:"Complement and coagulation cascades",Overlap:"2/85",P_value:.3774,Adjusted_P_value:.6619,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.5441,Combined_Score:1.5046,Genes:"F2R;PLAT",id:109},{Term:"Circadian rhythm",Overlap:"1/31",P_value:.3821,Adjusted_P_value:.6619,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.1349,Combined_Score:2.0537,Genes:"BHLHE40",id:110},{Term:"Galactose metabolism",Overlap:"1/31",P_value:.3821,Adjusted_P_value:.6619,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.1349,Combined_Score:2.0537,Genes:"GALM",id:111},{Term:"Base excision repair",Overlap:"1/33",P_value:.401,Adjusted_P_value:.683,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.0012,Combined_Score:1.8285,Genes:"HMGB1",id:112},{Term:"Human papillomavirus infection",Overlap:"6/331",P_value:.4014,Adjusted_P_value:.683,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.1839,Combined_Score:1.0808,Genes:"STAT1;IRF1;HLA-B;ISG15;HLA-A;OASL",id:113},{Term:"Morphine addiction",Overlap:"2/91",P_value:.41,Adjusted_P_value:.6862,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.4396,Combined_Score:1.2834,Genes:"PDE3B;PDE4B",id:114},{Term:"Propanoate metabolism",Overlap:"1/34",P_value:.4103,Adjusted_P_value:.6862,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.9405,Combined_Score:1.7288,Genes:"LDHA",id:115},{Term:"Nicotinate and nicotinamide metabolism",Overlap:"1/35",P_value:.4194,Adjusted_P_value:.6954,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.8833,Combined_Score:1.6366,Genes:"CD38",id:116},{Term:"Starch and sucrose metabolism",Overlap:"1/36",P_value:.4283,Adjusted_P_value:.7042,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.8294,Combined_Score:1.5511,Genes:"GYG1",id:117},{Term:"African trypanosomiasis",Overlap:"1/37",P_value:.4372,Adjusted_P_value:.7127,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.7785,Combined_Score:1.4716,Genes:"IFNG",id:118},{Term:"Choline metabolism in cancer",Overlap:"2/98",P_value:.447,Adjusted_P_value:.7226,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.3342,Combined_Score:1.0743,Genes:"RAC2;FOS",id:119},{Term:"Various types of N-glycan biosynthesis",Overlap:"1/39",P_value:.4544,Adjusted_P_value:.7272,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.6847,Combined_Score:1.3289,Genes:"MGAT4A",id:120},{Term:"AGE-RAGE signaling pathway in diabetic complications",Overlap:"2/100",P_value:.4573,Adjusted_P_value:.7272,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.3068,Combined_Score:1.0225,Genes:"STAT1;PIM1",id:121},{Term:"Ferroptosis",Overlap:"1/41",P_value:.4711,Adjusted_P_value:.743,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.6003,Combined_Score:1.2045,Genes:"SAT1",id:122},{Term:"Protein processing in endoplasmic reticulum",Overlap:"3/171",P_value:.4918,Adjusted_P_value:.7594,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.1431,Combined_Score:.8112,Genes:"DNAJC1;SELENOS;HERPUD1",id:123},{Term:"Glucagon signaling pathway",Overlap:"2/107",P_value:.4925,Adjusted_P_value:.7594,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.2192,Combined_Score:.8635,Genes:"LDHA;PDE3B",id:124},{Term:"Insulin resistance",Overlap:"2/108",P_value:.4974,Adjusted_P_value:.7594,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.2077,Combined_Score:.8433,Genes:"TBC1D4;SLC27A2",id:125},{Term:"ABC transporters",Overlap:"1/45",P_value:.503,Adjusted_P_value:.7594,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.4545,Combined_Score:.9995,Genes:"TAP1",id:126},{Term:"Type II diabetes mellitus",Overlap:"1/46",P_value:.5107,Adjusted_P_value:.7594,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.4221,Combined_Score:.9557,Genes:"SOCS1",id:127},{Term:"Pyruvate metabolism",Overlap:"1/47",P_value:.5182,Adjusted_P_value:.7594,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.3912,Combined_Score:.9145,Genes:"LDHA",id:128},{Term:"Cholinergic synapse",Overlap:"2/113",P_value:.5215,Adjusted_P_value:.7594,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.153,Combined_Score:.7506,Genes:"FYN;FOS",id:129},{Term:"Leukocyte transendothelial migration",Overlap:"2/114",P_value:.5262,Adjusted_P_value:.7594,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.1426,Combined_Score:.7335,Genes:"RAC2;CYBA",id:130},{Term:"Cocaine addiction",Overlap:"1/49",P_value:.533,Adjusted_P_value:.7594,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.3331,Combined_Score:.8388,Genes:"FOSB",id:131},{Term:"Sphingolipid metabolism",Overlap:"1/49",P_value:.533,Adjusted_P_value:.7594,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.3331,Combined_Score:.8388,Genes:"PSAP",id:132},{Term:"Axon guidance",Overlap:"3/182",P_value:.5336,Adjusted_P_value:.7594,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.0722,Combined_Score:.6736,Genes:"RAC2;NFATC2;FYN",id:133},{Term:"Arginine and proline metabolism",Overlap:"1/50",P_value:.5402,Adjusted_P_value:.7594,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.3058,Combined_Score:.8041,Genes:"SAT1",id:134},{Term:"Cysteine and methionine metabolism",Overlap:"1/50",P_value:.5402,Adjusted_P_value:.7594,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.3058,Combined_Score:.8041,Genes:"LDHA",id:135},{Term:"Malaria",Overlap:"1/50",P_value:.5402,Adjusted_P_value:.7594,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.3058,Combined_Score:.8041,Genes:"IFNG",id:136},{Term:"N-Glycan biosynthesis",Overlap:"1/50",P_value:.5402,Adjusted_P_value:.7594,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.3058,Combined_Score:.8041,Genes:"MGAT4A",id:137},{Term:"RNA transport",Overlap:"3/186",P_value:.5483,Adjusted_P_value:.7613,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.0486,Combined_Score:.6302,Genes:"EEF1A1;SUMO2;EIF4B",id:138},{Term:"Sphingolipid signaling pathway",Overlap:"2/119",P_value:.5494,Adjusted_P_value:.7613,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.0935,Combined_Score:.6549,Genes:"RAC2;FYN",id:139},{Term:"Thyroid hormone signaling pathway",Overlap:"2/121",P_value:.5584,Adjusted_P_value:.7684,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.075,Combined_Score:.6263,Genes:"TBC1D4;STAT1",id:140},{Term:"Platelet activation",Overlap:"2/124",P_value:.5718,Adjusted_P_value:.7795,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.0484,Combined_Score:.5861,Genes:"F2R;FYN",id:141},{Term:"Regulation of lipolysis in adipocytes",Overlap:"1/55",P_value:.5746,Adjusted_P_value:.7795,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.1846,Combined_Score:.6563,Genes:"PDE3B",id:142},{Term:"Glutathione metabolism",Overlap:"1/57",P_value:.5876,Adjusted_P_value:.783,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.1422,Combined_Score:.6072,Genes:"MGST3",id:143},{Term:"Lysosome",Overlap:"2/128",P_value:.5891,Adjusted_P_value:.783,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.0149,Combined_Score:.5371,Genes:"PSAP;ACP5",id:144},{Term:"Purine metabolism",Overlap:"2/129",P_value:.5933,Adjusted_P_value:.783,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.0069,Combined_Score:.5257,Genes:"PDE3B;PDE4B",id:145},{Term:"Relaxin signaling pathway",Overlap:"2/129",P_value:.5933,Adjusted_P_value:.783,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.0069,Combined_Score:.5257,Genes:"GNA15;FOS",id:146},{Term:"Mineral absorption",Overlap:"1/60",P_value:.6065,Adjusted_P_value:.795,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.0839,Combined_Score:.5421,Genes:"MT2A",id:147},{Term:"Glycerolipid metabolism",Overlap:"1/61",P_value:.6125,Adjusted_P_value:.7975,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.0658,Combined_Score:.5224,Genes:"GK",id:148},{Term:"Estrogen signaling pathway",Overlap:"2/137",P_value:.6261,Adjusted_P_value:.8097,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.9468,Combined_Score:.4434,Genes:"FOS;FKBP5",id:149},{Term:"Renal cell carcinoma",Overlap:"1/69",P_value:.6579,Adjusted_P_value:.8355,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.94,Combined_Score:.3936,Genes:"PRCC",id:150},{Term:"Renin secretion",Overlap:"1/69",P_value:.6579,Adjusted_P_value:.8355,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.94,Combined_Score:.3936,Genes:"PDE3B",id:151},{Term:"Central carbon metabolism in cancer",Overlap:"1/70",P_value:.6632,Adjusted_P_value:.8355,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.9264,Combined_Score:.3804,Genes:"LDHA",id:152},{Term:"Epithelial cell signaling in Helicobacter pylori infection",Overlap:"1/70",P_value:.6632,Adjusted_P_value:.8355,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.9264,Combined_Score:.3804,Genes:"CCL5",id:153},{Term:"Phospholipase D signaling pathway",Overlap:"2/148",P_value:.6677,Adjusted_P_value:.8357,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.875,Combined_Score:.3534,Genes:"F2R;FYN",id:154},{Term:"p53 signaling pathway",Overlap:"1/73",P_value:.6786,Adjusted_P_value:.8439,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.8876,Combined_Score:.3442,Genes:"PMAIP1",id:155},{Term:"Metabolism of xenobiotics by cytochrome P450",Overlap:"1/76",P_value:.6933,Adjusted_P_value:.8495,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.852,Combined_Score:.3121,Genes:"MGST3",id:156},{Term:"Ras signaling pathway",Overlap:"3/232",P_value:.6962,Adjusted_P_value:.8495,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.836,Combined_Score:.3027,Genes:"RASA3;RAC2;RASAL3",id:157},{Term:"Thermogenesis",Overlap:"3/232",P_value:.6962,Adjusted_P_value:.8495,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.836,Combined_Score:.3027,Genes:"NDUFS5;RPS6;COX5A",id:158},{Term:"Wnt signaling pathway",Overlap:"2/166",P_value:.7278,Adjusted_P_value:.8824,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.7783,Combined_Score:.2473,Genes:"RAC2;NFATC2",id:159},{Term:"ErbB signaling pathway",Overlap:"1/85",P_value:.7334,Adjusted_P_value:.8837,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.7604,Combined_Score:.2358,Genes:"CBLB",id:160},{Term:"Cardiac muscle contraction",Overlap:"1/87",P_value:.7416,Adjusted_P_value:.8874,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.7426,Combined_Score:.222,Genes:"COX5A",id:161},{Term:"Gap junction",Overlap:"1/88",P_value:.7456,Adjusted_P_value:.8874,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.734,Combined_Score:.2155,Genes:"TUBB",id:162},{Term:"Salivary secretion",Overlap:"1/93",P_value:.7647,Adjusted_P_value:.8972,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.694,Combined_Score:.1862,Genes:"CD38",id:163},{Term:"TGF-beta signaling pathway",Overlap:"1/94",P_value:.7683,Adjusted_P_value:.8972,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.6865,Combined_Score:.1809,Genes:"IFNG",id:164},{Term:"Circadian entrainment",Overlap:"1/97",P_value:.7789,Adjusted_P_value:.8972,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.6649,Combined_Score:.1662,Genes:"FOS",id:165},{Term:"Fc gamma R-mediated phagocytosis",Overlap:"1/97",P_value:.7789,Adjusted_P_value:.8972,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.6649,Combined_Score:.1662,Genes:"RAC2",id:166},{Term:"Prostate cancer",Overlap:"1/97",P_value:.7789,Adjusted_P_value:.8972,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.6649,Combined_Score:.1662,Genes:"PLAT",id:167},{Term:"Aldosterone synthesis and secretion",Overlap:"1/98",P_value:.7823,Adjusted_P_value:.8972,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.658,Combined_Score:.1615,Genes:"NR4A2",id:168},{Term:"Progesterone-mediated oocyte maturation",Overlap:"1/100",P_value:.789,Adjusted_P_value:.8972,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.6447,Combined_Score:.1528,Genes:"PDE3B",id:169},{Term:"Longevity regulating pathway",Overlap:"1/102",P_value:.7955,Adjusted_P_value:.8972,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.6318,Combined_Score:.1446,Genes:"SOD1",id:170},{Term:"Pancreatic secretion",Overlap:"1/102",P_value:.7955,Adjusted_P_value:.8972,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.6318,Combined_Score:.1446,Genes:"CD38",id:171},{Term:"Drug metabolism",Overlap:"1/108",P_value:.8138,Adjusted_P_value:.9124,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.5962,Combined_Score:.1229,Genes:"MGST3",id:172},{Term:"Focal adhesion",Overlap:"2/201",P_value:.8184,Adjusted_P_value:.9124,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.6402,Combined_Score:.1283,Genes:"RAC2;FYN",id:173},{Term:"Proteoglycans in cancer",Overlap:"2/205",P_value:.8268,Adjusted_P_value:.9124,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.6275,Combined_Score:.1193,Genes:"RPS6;EIF4B",id:174},{Term:"Serotonergic synapse",Overlap:"1/113",P_value:.8277,Adjusted_P_value:.9124,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.5695,Combined_Score:.1077,Genes:"DUSP1",id:175},{Term:"Rap1 signaling pathway",Overlap:"2/210",P_value:.8369,Adjusted_P_value:.9172,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.6122,Combined_Score:.109,Genes:"F2R;RAC2",id:176},{Term:"Regulation of actin cytoskeleton",Overlap:"2/218",P_value:.8519,Adjusted_P_value:.9265,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.5893,Combined_Score:.0945,Genes:"F2R;RAC2",id:177},{Term:"Cell cycle",Overlap:"1/124",P_value:.8549,Adjusted_P_value:.9265,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.5182,Combined_Score:.0812,Genes:"YWHAB",id:178},{Term:"Oocyte meiosis",Overlap:"1/129",P_value:.8658,Adjusted_P_value:.9265,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.4979,Combined_Score:.0717,Genes:"YWHAB",id:179},{Term:"FoxO signaling pathway",Overlap:"1/131",P_value:.8699,Adjusted_P_value:.9265,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.4902,Combined_Score:.0683,Genes:"BCL2L11",id:180},{Term:"Dopaminergic synapse",Overlap:"1/132",P_value:.872,Adjusted_P_value:.9265,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.4864,Combined_Score:.0666,Genes:"FOS",id:181},{Term:"Vascular smooth muscle contraction",Overlap:"1/133",P_value:.8739,Adjusted_P_value:.9265,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.4827,Combined_Score:.065,Genes:"MYL6",id:182},{Term:"Chemical carcinogenesis",Overlap:"2/239",P_value:.8854,Adjusted_P_value:.9322,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.5365,Combined_Score:.0653,Genes:"MGST3;FOS",id:183},{Term:"Breast cancer",Overlap:"1/147",P_value:.8987,Adjusted_P_value:.9322,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.4361,Combined_Score:.0466,Genes:"FOS",id:184},{Term:"Retrograde endocannabinoid signaling",Overlap:"1/148",P_value:.9003,Adjusted_P_value:.9322,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.4331,Combined_Score:.0455,Genes:"NDUFS5",id:185},{Term:"Adrenergic signaling in cardiomyocytes",Overlap:"1/150",P_value:.9034,Adjusted_P_value:.9322,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.4272,Combined_Score:.0434,Genes:"CREM",id:186},{Term:"Spliceosome",Overlap:"1/150",P_value:.9034,Adjusted_P_value:.9322,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.4272,Combined_Score:.0434,Genes:"SRSF7",id:187},{Term:"Hippo signaling pathway",Overlap:"1/163",P_value:.9211,Adjusted_P_value:.9428,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.3927,Combined_Score:.0323,Genes:"YWHAB",id:188},{Term:"Hepatocellular carcinoma",Overlap:"1/168",P_value:.9271,Adjusted_P_value:.9428,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.3808,Combined_Score:.0288,Genes:"MGST3",id:189},{Term:"Tight junction",Overlap:"1/169",P_value:.9282,Adjusted_P_value:.9428,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.3785,Combined_Score:.0282,Genes:"MYL6",id:190},{Term:"Alcoholism",Overlap:"1/186",P_value:.945,Adjusted_P_value:.9548,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.3435,Combined_Score:.0194,Genes:"FOSB",id:191},{Term:"Neuroactive ligand-receptor interaction",Overlap:"2/341",P_value:.9691,Adjusted_P_value:.9741,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.3731,Combined_Score:.0117,Genes:"GZMA;F2R",id:192},{Term:"Olfactory transduction",Overlap:"1/440",P_value:.999,Adjusted_P_value:.999,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.1429,Combined_Score:1e-4,Genes:"RGS2",id:193}]}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p:T,i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function xn(e){let a,t,n,l;return a=new me({props:{$$slots:{default:[Ln]},$$scope:{ctx:e}}}),n=new me({props:{$$slots:{default:[jn]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment)},m(e,_){P(a,e,_),d(e,t,_),P(n,e,_),l=!0},p(e,t){const l={};1&t&&(l.$$scope={dirty:t,ctx:e}),a.$set(l);const _={};1&t&&(_.$$scope={dirty:t,ctx:e}),n.$set(_)},i(e){l||(i(a.$$.fragment,e),i(n.$$.fragment,e),l=!0)},o(e){c(a.$$.fragment,e),c(n.$$.fragment,e),l=!1},d(e){e&&p(t),A(a,e),A(n,e)}}}function Mn(e){let a,t;return a=new ue({props:{$$slots:{content:[xn],default:[Tn]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};1&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function Bn(e){let a,t;return a=new Me({props:{title:"KEGG_2021_Human",open:!0,$$slots:{default:[Mn]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};1&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function Gn(e){let a,t,n,l,_,s;return a=new ve({props:{label:"Significant Markers"}}),n=new ve({props:{label:"Volcano Plot"}}),_=new ve({props:{label:"Dot Plot"}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment),l=f(),g(_.$$.fragment)},m(e,i){P(a,e,i),d(e,t,i),P(n,e,i),d(e,l,i),P(_,e,i),s=!0},p:T,i(e){s||(i(a.$$.fragment,e),i(n.$$.fragment,e),i(_.$$.fragment,e),s=!0)},o(e){c(a.$$.fragment,e),c(n.$$.fragment,e),c(_.$$.fragment,e),s=!1},d(e){e&&(p(t),p(l)),A(a,e),A(n,e),A(_,e)}}}function Fn(e){let a;return{c(){a=S("The markers are found using Seurat's FindMarkers function, and filtered by: p_val_adj < 0.05")},m(e,t){d(e,a,t)},d(e){e&&p(a)}}}function wn(e){let a,t,n,l;return a=new Pn({props:{$$slots:{default:[Fn]},$$scope:{ctx:e}}}),n=new un({props:{src:"../MarkersFinder/samples.markers/c2/markers.txt",data:[{gene:"HLA-B",p_val:8455e-34,avg_log2FC:.7613,pct_1:1,pct_2:1,p_val_adj:2836e-29,id:0},{gene:"CTSW",p_val:1539e-25,avg_log2FC:-1.7545,pct_1:.456,pct_2:.889,p_val_adj:516e-20,id:1},{gene:"SRGN",p_val:3.571e-20,avg_log2FC:.8186,pct_1:1,pct_2:.981,p_val_adj:0,id:2},{gene:"FOS",p_val:1026e-22,avg_log2FC:-1.4244,pct_1:.796,pct_2:.974,p_val_adj:0,id:3},{gene:"FKBP5",p_val:525e-20,avg_log2FC:2.5237,pct_1:.485,pct_2:.151,p_val_adj:0,id:4},{gene:"S100A4",p_val:1016e-20,avg_log2FC:-.996,pct_1:.913,pct_2:.986,p_val_adj:0,id:5},{gene:"VIM",p_val:5323e-20,avg_log2FC:-.8107,pct_1:.932,pct_2:.998,p_val_adj:0,id:6},{gene:"HLA-DRB1",p_val:2976e-19,avg_log2FC:1.9955,pct_1:.612,pct_2:.257,p_val_adj:0,id:7},{gene:"CTLA4",p_val:9031e-19,avg_log2FC:3.9532,pct_1:.165,pct_2:.01,p_val_adj:0,id:8},{gene:"HLA-DRA",p_val:9589e-19,avg_log2FC:3.275,pct_1:.272,pct_2:.045,p_val_adj:0,id:9},{gene:"ISG20",p_val:0,avg_log2FC:1.2375,pct_1:.864,pct_2:.589,p_val_adj:0,id:10},{gene:"MT-ND5",p_val:0,avg_log2FC:-.57,pct_1:1,pct_2:.995,p_val_adj:0,id:11},{gene:"GZMB",p_val:0,avg_log2FC:2.5014,pct_1:.621,pct_2:.308,p_val_adj:0,id:12},{gene:"CAPG",p_val:0,avg_log2FC:-1.5487,pct_1:.34,pct_2:.728,p_val_adj:0,id:13},{gene:"TXNIP",p_val:0,avg_log2FC:-1.0542,pct_1:.689,pct_2:.892,p_val_adj:0,id:14},{gene:"CD7",p_val:0,avg_log2FC:.5956,pct_1:1,pct_2:.988,p_val_adj:0,id:15},{gene:"HAPLN3",p_val:0,avg_log2FC:3.1094,pct_1:.175,pct_2:.021,p_val_adj:0,id:16},{gene:"CCL4L2",p_val:0,avg_log2FC:-2.1244,pct_1:.243,pct_2:.656,p_val_adj:0,id:17},{gene:"HLA-A",p_val:0,avg_log2FC:.5357,pct_1:1,pct_2:.997,p_val_adj:0,id:18},{gene:"RPS14",p_val:0,avg_log2FC:-.3977,pct_1:1,pct_2:1,p_val_adj:0,id:19},{gene:"LGALS1",p_val:0,avg_log2FC:-1.9842,pct_1:.291,pct_2:.632,p_val_adj:0,id:20},{gene:"AL157402.2",p_val:0,avg_log2FC:2.7847,pct_1:.262,pct_2:.06,p_val_adj:0,id:21},{gene:"IGKV3D-20",p_val:0,avg_log2FC:7.471,pct_1:.078,pct_2:0,p_val_adj:0,id:22},{gene:"EEF1A1",p_val:0,avg_log2FC:-.2836,pct_1:1,pct_2:1,p_val_adj:0,id:23},{gene:"TOX2",p_val:0,avg_log2FC:3.3578,pct_1:.117,pct_2:.009,p_val_adj:0,id:24},{gene:"MIR181A1HG",p_val:0,avg_log2FC:2.9498,pct_1:.136,pct_2:.014,p_val_adj:0,id:25},{gene:"SUN2",p_val:0,avg_log2FC:-.9405,pct_1:.573,pct_2:.849,p_val_adj:0,id:26},{gene:"B2M",p_val:0,avg_log2FC:.2534,pct_1:1,pct_2:1,p_val_adj:0,id:27},{gene:"STAT1",p_val:0,avg_log2FC:1.4734,pct_1:.553,pct_2:.296,p_val_adj:0,id:28},{gene:"GNLY",p_val:0,avg_log2FC:2.1482,pct_1:.252,pct_2:.06,p_val_adj:0,id:29},{gene:"CD74",p_val:0,avg_log2FC:1.2909,pct_1:.942,pct_2:.865,p_val_adj:0,id:30},{gene:"DUSP2",p_val:0,avg_log2FC:.9801,pct_1:.903,pct_2:.661,p_val_adj:0,id:31},{gene:"SPINK2",p_val:0,avg_log2FC:-2.373,pct_1:.068,pct_2:.396,p_val_adj:0,id:32},{gene:"AC020916.1",p_val:0,avg_log2FC:-1.2786,pct_1:.34,pct_2:.663,p_val_adj:0,id:33},{gene:"DUSP1",p_val:0,avg_log2FC:-.7356,pct_1:.874,pct_2:.961,p_val_adj:0,id:34},{gene:"BTG1",p_val:0,avg_log2FC:.8834,pct_1:1,pct_2:.988,p_val_adj:0,id:35},{gene:"CREM",p_val:0,avg_log2FC:1.4153,pct_1:.699,pct_2:.498,p_val_adj:0,id:36},{gene:"RGS1",p_val:0,avg_log2FC:1.1722,pct_1:.854,pct_2:.661,p_val_adj:1e-4,id:37},{gene:"TUBB",p_val:0,avg_log2FC:-1.0049,pct_1:.602,pct_2:.791,p_val_adj:1e-4,id:38},{gene:"PSMB9",p_val:0,avg_log2FC:.6847,pct_1:.922,pct_2:.846,p_val_adj:1e-4,id:39},{gene:"MT-CO3",p_val:0,avg_log2FC:.2861,pct_1:1,pct_2:.998,p_val_adj:1e-4,id:40},{gene:"RPS26",p_val:0,avg_log2FC:.5713,pct_1:1,pct_2:.95,p_val_adj:1e-4,id:41},{gene:"ITM2C",p_val:0,avg_log2FC:-.982,pct_1:.515,pct_2:.781,p_val_adj:1e-4,id:42},{gene:"MYL12A",p_val:0,avg_log2FC:-.4721,pct_1:.971,pct_2:.997,p_val_adj:1e-4,id:43},{gene:"RARRES3",p_val:0,avg_log2FC:.9709,pct_1:.913,pct_2:.921,p_val_adj:1e-4,id:44},{gene:"LRRN3",p_val:0,avg_log2FC:-2.4191,pct_1:.068,pct_2:.368,p_val_adj:1e-4,id:45},{gene:"TRAV19",p_val:0,avg_log2FC:1.6228,pct_1:.311,pct_2:.099,p_val_adj:1e-4,id:46},{gene:"P2RX5",p_val:0,avg_log2FC:3.0211,pct_1:.126,pct_2:.015,p_val_adj:1e-4,id:47},{gene:"FOSB",p_val:0,avg_log2FC:-1.0327,pct_1:.621,pct_2:.851,p_val_adj:2e-4,id:48},{gene:"CERS6",p_val:0,avg_log2FC:5.6885,pct_1:.058,pct_2:0,p_val_adj:2e-4,id:49},{gene:"IGHV3-21",p_val:0,avg_log2FC:6.5077,pct_1:.058,pct_2:0,p_val_adj:2e-4,id:50},{gene:"HLA-DQA1",p_val:0,avg_log2FC:3.864,pct_1:.107,pct_2:.01,p_val_adj:2e-4,id:51},{gene:"GEM",p_val:0,avg_log2FC:2.192,pct_1:.107,pct_2:.01,p_val_adj:2e-4,id:52},{gene:"EMP3",p_val:0,avg_log2FC:-.9439,pct_1:.515,pct_2:.798,p_val_adj:3e-4,id:53},{gene:"UBE2L6",p_val:0,avg_log2FC:.9367,pct_1:.767,pct_2:.502,p_val_adj:4e-4,id:54},{gene:"RPS12",p_val:0,avg_log2FC:-.2948,pct_1:1,pct_2:1,p_val_adj:4e-4,id:55},{gene:"MT-CO2",p_val:0,avg_log2FC:.2536,pct_1:1,pct_2:1,p_val_adj:8e-4,id:56},{gene:"RPS27",p_val:0,avg_log2FC:.3352,pct_1:1,pct_2:1,p_val_adj:8e-4,id:57},{gene:"ATP1A1",p_val:0,avg_log2FC:-1.1524,pct_1:.408,pct_2:.673,p_val_adj:9e-4,id:58},{gene:"ACTB",p_val:0,avg_log2FC:-.4295,pct_1:1,pct_2:1,p_val_adj:.0011,id:59},{gene:"GBP2",p_val:0,avg_log2FC:1.3417,pct_1:.544,pct_2:.318,p_val_adj:.0014,id:60},{gene:"LAYN",p_val:0,avg_log2FC:2.7129,pct_1:.146,pct_2:.027,p_val_adj:.0026,id:61},{gene:"LY6E",p_val:0,avg_log2FC:.7291,pct_1:.893,pct_2:.781,p_val_adj:.0026,id:62},{gene:"ATF3",p_val:0,avg_log2FC:1.5955,pct_1:.388,pct_2:.166,p_val_adj:.0027,id:63},{gene:"NFKBIZ",p_val:0,avg_log2FC:1.6469,pct_1:.592,pct_2:.404,p_val_adj:.0027,id:64},{gene:"CLIP3",p_val:0,avg_log2FC:5.4301,pct_1:.049,pct_2:0,p_val_adj:.0031,id:65},{gene:"AREG",p_val:0,avg_log2FC:2.1049,pct_1:.408,pct_2:.19,p_val_adj:.0037,id:66},{gene:"NOP53",p_val:0,avg_log2FC:.5708,pct_1:.913,pct_2:.86,p_val_adj:.0039,id:67},{gene:"CISH",p_val:0,avg_log2FC:-1.712,pct_1:.194,pct_2:.459,p_val_adj:.0044,id:68},{gene:"HLA-DQB1",p_val:0,avg_log2FC:1.7561,pct_1:.223,pct_2:.065,p_val_adj:.0054,id:69},{gene:"CST7",p_val:0,avg_log2FC:.7462,pct_1:.903,pct_2:.868,p_val_adj:.0057,id:70},{gene:"SOCS3",p_val:0,avg_log2FC:1.5698,pct_1:.388,pct_2:.176,p_val_adj:.0059,id:71},{gene:"GBP5",p_val:0,avg_log2FC:1.5848,pct_1:.515,pct_2:.33,p_val_adj:.0068,id:72},{gene:"PKM",p_val:0,avg_log2FC:-.6538,pct_1:.553,pct_2:.795,p_val_adj:.0081,id:73},{gene:"FAM129A",p_val:0,avg_log2FC:1.2721,pct_1:.456,pct_2:.245,p_val_adj:.0086,id:74},{gene:"LTB",p_val:0,avg_log2FC:-.8873,pct_1:.447,pct_2:.711,p_val_adj:.0089,id:75},{gene:"TUBA1B",p_val:0,avg_log2FC:-.9338,pct_1:.456,pct_2:.711,p_val_adj:.0092,id:76},{gene:"CHI3L2",p_val:0,avg_log2FC:1.8326,pct_1:.223,pct_2:.068,p_val_adj:.0096,id:77},{gene:"LCP1",p_val:0,avg_log2FC:-.6093,pct_1:.816,pct_2:.935,p_val_adj:.0147,id:78},{gene:"RPS27A",p_val:0,avg_log2FC:-.2098,pct_1:1,pct_2:1,p_val_adj:.0147,id:79},{gene:"AC020571.1",p_val:0,avg_log2FC:-1.9908,pct_1:.184,pct_2:.44,p_val_adj:.0161,id:80},{gene:"RBL2",p_val:0,avg_log2FC:-.9146,pct_1:.544,pct_2:.735,p_val_adj:.0179,id:81},{gene:"RPL5",p_val:0,avg_log2FC:-.329,pct_1:1,pct_2:1,p_val_adj:.0185,id:82},{gene:"CD160",p_val:0,avg_log2FC:-1.0194,pct_1:.35,pct_2:.634,p_val_adj:.0209,id:83},{gene:"CBLB",p_val:0,avg_log2FC:.9969,pct_1:.738,pct_2:.563,p_val_adj:.0217,id:84},{gene:"ZC3H12D",p_val:0,avg_log2FC:1.8953,pct_1:.184,pct_2:.05,p_val_adj:.0219,id:85},{gene:"KLRC1",p_val:0,avg_log2FC:1.6493,pct_1:.262,pct_2:.092,p_val_adj:.0241,id:86},{gene:"SARAF",p_val:0,avg_log2FC:.326,pct_1:1,pct_2:.993,p_val_adj:.028,id:87},{gene:"RPS3A",p_val:0,avg_log2FC:-.2475,pct_1:1,pct_2:1,p_val_adj:.0317,id:88},{gene:"PDE3B",p_val:0,avg_log2FC:1.4413,pct_1:.427,pct_2:.238,p_val_adj:.0319,id:89},{gene:"PSMB8",p_val:0,avg_log2FC:.6523,pct_1:.816,pct_2:.707,p_val_adj:.0324,id:90},{gene:"FIBP",p_val:0,avg_log2FC:1.1163,pct_1:.476,pct_2:.265,p_val_adj:.0425,id:91},{gene:"HIST1H2AC",p_val:0,avg_log2FC:1.7952,pct_1:.204,pct_2:.062,p_val_adj:.0438,id:92},{gene:"RPS5",p_val:0,avg_log2FC:-.2673,pct_1:1,pct_2:1,p_val_adj:.0462,id:93}]}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment)},m(e,_){P(a,e,_),d(e,t,_),P(n,e,_),l=!0},p(e,t){const n={};1&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){l||(i(a.$$.fragment,e),i(n.$$.fragment,e),l=!0)},o(e){c(a.$$.fragment,e),c(n.$$.fragment,e),l=!1},d(e){e&&p(t),A(a,e),A(n,e)}}}function Nn(e){let a;return{c(){a=S("The volcano plot is generated using ggplot2. The significant markers are highlighted in blue and red, indicating up-regulated and down-regulated genes, respectively. The non-significant genes are shown in grey.")},m(e,t){d(e,a,t)},d(e){e&&p(a)}}}function Un(e){let a,t,n,l;return a=new Pn({props:{$$slots:{default:[Nn]},$$scope:{ctx:e}}}),n=new Ht({props:{src:"../MarkersFinder/samples.markers/c2/volcano.png",download:"../MarkersFinder/samples.markers/c2/volcano.pdf",width:900,height:1200}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment)},m(e,_){P(a,e,_),d(e,t,_),P(n,e,_),l=!0},p(e,t){const n={};1&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){l||(i(a.$$.fragment,e),i(n.$$.fragment,e),l=!0)},o(e){c(a.$$.fragment,e),c(n.$$.fragment,e),l=!1},d(e){e&&p(t),A(a,e),A(n,e)}}}function Dn(e){let a;return{c(){a=S("The dot plot is generated using Seurat's DotPlot function. The top significant markers are used as input. The sizes of the dots are proportional to the percentage of cells expressing the gene in each group. The colors of the dots are proportional to the average expression of the gene in each group.")},m(e,t){d(e,a,t)},d(e){e&&p(a)}}}function En(e){let a,t,n,l;return a=new Pn({props:{$$slots:{default:[Dn]},$$scope:{ctx:e}}}),n=new Ht({props:{src:"../MarkersFinder/samples.markers/c2/dotplot.png",download:"../MarkersFinder/samples.markers/c2/dotplot.pdf",width:600,height:1200}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment)},m(e,_){P(a,e,_),d(e,t,_),P(n,e,_),l=!0},p(e,t){const n={};1&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){l||(i(a.$$.fragment,e),i(n.$$.fragment,e),l=!0)},o(e){c(a.$$.fragment,e),c(n.$$.fragment,e),l=!1},d(e){e&&p(t),A(a,e),A(n,e)}}}function Hn(e){let a,t,n,l,_,s;return a=new me({props:{$$slots:{default:[wn]},$$scope:{ctx:e}}}),n=new me({props:{$$slots:{default:[Un]},$$scope:{ctx:e}}}),_=new me({props:{$$slots:{default:[En]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment),l=f(),g(_.$$.fragment)},m(e,i){P(a,e,i),d(e,t,i),P(n,e,i),d(e,l,i),P(_,e,i),s=!0},p(e,t){const l={};1&t&&(l.$$scope={dirty:t,ctx:e}),a.$set(l);const s={};1&t&&(s.$$scope={dirty:t,ctx:e}),n.$set(s);const d={};1&t&&(d.$$scope={dirty:t,ctx:e}),_.$set(d)},i(e){s||(i(a.$$.fragment,e),i(n.$$.fragment,e),i(_.$$.fragment,e),s=!0)},o(e){c(a.$$.fragment,e),c(n.$$.fragment,e),c(_.$$.fragment,e),s=!1},d(e){e&&(p(t),p(l)),A(a,e),A(n,e),A(_,e)}}}function In(e){let a;return{c(){a=S("The enrichment analysis is done using Enrichr. The significant markers are used as input.")},m(e,t){d(e,a,t)},d(e){e&&p(a)}}}function kn(e){let a,t,n,l;return a=new ve({props:{label:"Plot"}}),n=new ve({props:{label:"Table"}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment)},m(e,_){P(a,e,_),d(e,t,_),P(n,e,_),l=!0},p:T,i(e){l||(i(a.$$.fragment,e),i(n.$$.fragment,e),l=!0)},o(e){c(a.$$.fragment,e),c(n.$$.fragment,e),l=!1},d(e){e&&p(t),A(a,e),A(n,e)}}}function Vn(e){let a;return{c(){a=S("The enrichment analysis results of the top biological pathways associated with the input gene set. Each bar represents a pathway, with the length of the bar indicating the number of input genes overlapping with genes in that pathway. The color intensity of the bars reflects the statistical significance of the enrichment (p-value).")},m(e,t){d(e,a,t)},d(e){e&&p(a)}}}function zn(e){let a,t,n,l;return a=new Pn({props:{$$slots:{default:[Vn]},$$scope:{ctx:e}}}),n=new Ht({props:{src:"../MarkersFinder/samples.markers/c2/Enrichr-KEGG_2021_Human.png",download:"../MarkersFinder/samples.markers/c2/Enrichr-KEGG_2021_Human.pdf",width:1e3,height:1e3}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment)},m(e,_){P(a,e,_),d(e,t,_),P(n,e,_),l=!0},p(e,t){const n={};1&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){l||(i(a.$$.fragment,e),i(n.$$.fragment,e),l=!0)},o(e){c(a.$$.fragment,e),c(n.$$.fragment,e),l=!1},d(e){e&&p(t),A(a,e),A(n,e)}}}function qn(e){let a,t;return a=new un({props:{src:"../MarkersFinder/samples.markers/c2/Enrichr-KEGG_2021_Human.txt",data:[{Term:"Graft-versus-host disease",Overlap:"8/42",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:54.3694,Combined_Score:1344.5737,Genes:"HLA-B;GZMB;HLA-DRA;KLRC1;HLA-A;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:0},{Term:"Antigen processing and presentation",Overlap:"9/78",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:30.4404,Combined_Score:699.2342,Genes:"CD74;HLA-B;HLA-DRA;HLA-A;KLRC1;B2M;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:1},{Term:"Autoimmune thyroid disease",Overlap:"8/53",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:41.0563,Combined_Score:934.2906,Genes:"HLA-B;CTLA4;GZMB;HLA-DRA;HLA-A;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:2},{Term:"Allograft rejection",Overlap:"7/38",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:51.5851,Combined_Score:1109.9398,Genes:"HLA-B;GZMB;HLA-DRA;HLA-A;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:3},{Term:"Type I diabetes mellitus",Overlap:"7/43",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:44.4093,Combined_Score:914.757,Genes:"HLA-B;GZMB;HLA-DRA;HLA-A;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:4},{Term:"Viral myocarditis",Overlap:"7/60",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:30.139,Combined_Score:547.9095,Genes:"HLA-B;HLA-DRA;HLA-A;ACTB;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:5},{Term:"Phagosome",Overlap:"9/152",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:14.6332,Combined_Score:248.9394,Genes:"TUBA1B;TUBB;HLA-B;HLA-DRA;HLA-A;ACTB;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:6},{Term:"Leishmaniasis",Overlap:"7/77",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:22.8,Combined_Score:374.2691,Genes:"EEF1A1;STAT1;HLA-DRA;FOS;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:7},{Term:"Coronavirus disease",Overlap:"10/232",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:10.5556,Combined_Score:166.5673,Genes:"RPS26;RPL5;RPS14;RPS27;STAT1;RPS5;RPS3A;FOS;RPS27A;RPS12",id:8},{Term:"Rheumatoid arthritis",Overlap:"7/93",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:18.5432,Combined_Score:280.0983,Genes:"CTLA4;HLA-DRA;FOS;LTB;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:9},{Term:"Epstein-Barr virus infection",Overlap:"9/202",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:10.8148,Combined_Score:157.7191,Genes:"STAT1;HLA-B;HLA-DRA;HLA-A;VIM;B2M;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:10},{Term:"Ribosome",Overlap:"8/158",P_value:0,Adjusted_P_value:0,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:12.2518,Combined_Score:172.0732,Genes:"RPS26;RPL5;RPS14;RPS27;RPS5;RPS3A;RPS27A;RPS12",id:11},{Term:"Th1 and Th2 cell differentiation",Overlap:"6/92",P_value:0,Adjusted_P_value:1e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:15.7135,Combined_Score:192.6905,Genes:"STAT1;HLA-DRA;FOS;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:12},{Term:"Cell adhesion molecules",Overlap:"7/148",P_value:0,Adjusted_P_value:1e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:11.2786,Combined_Score:135.0615,Genes:"HLA-B;CTLA4;HLA-DRA;HLA-A;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:13},{Term:"Human T-cell leukemia virus 1 infection",Overlap:"8/219",P_value:0,Adjusted_P_value:1e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:8.6829,Combined_Score:100.857,Genes:"HLA-B;HLA-DRA;FOS;HLA-A;B2M;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:14},{Term:"Th17 cell differentiation",Overlap:"6/107",P_value:0,Adjusted_P_value:1e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:13.3697,Combined_Score:152.2705,Genes:"STAT1;HLA-DRA;FOS;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:15},{Term:"Asthma",Overlap:"4/31",P_value:0,Adjusted_P_value:1e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:32.7226,Combined_Score:367.992,Genes:"HLA-DRA;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:16},{Term:"Inflammatory bowel disease",Overlap:"5/65",P_value:0,Adjusted_P_value:1e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:18.5824,Combined_Score:208.2021,Genes:"STAT1;HLA-DRA;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:17},{Term:"Influenza A",Overlap:"7/172",P_value:0,Adjusted_P_value:1e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:9.6264,Combined_Score:105.8319,Genes:"SOCS3;STAT1;HLA-DRA;ACTB;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:18},{Term:"Intestinal immune network for IgA production",Overlap:"4/48",P_value:1e-4,Adjusted_P_value:6e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:20.0626,Combined_Score:190.2918,Genes:"HLA-DRA;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:19},{Term:"Hematopoietic cell lineage",Overlap:"5/99",P_value:1e-4,Adjusted_P_value:8e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:11.8408,Combined_Score:108.5958,Genes:"CD7;HLA-DRA;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:20},{Term:"Herpes simplex virus 1 infection",Overlap:"10/498",P_value:1e-4,Adjusted_P_value:8e-4,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.737,Combined_Score:42.9039,Genes:"SOCS3;CD74;STAT1;HLA-B;HLA-DRA;HLA-A;B2M;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:21},{Term:"Toxoplasmosis",Overlap:"5/112",P_value:2e-4,Adjusted_P_value:.0013,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:10.3954,Combined_Score:89.2997,Genes:"STAT1;HLA-DRA;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:22},{Term:"Tuberculosis",Overlap:"6/180",P_value:2e-4,Adjusted_P_value:.0014,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:7.732,Combined_Score:65.6008,Genes:"CD74;STAT1;HLA-DRA;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:23},{Term:"Prolactin signaling pathway",Overlap:"4/70",P_value:3e-4,Adjusted_P_value:.0021,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:13.3603,Combined_Score:107.0639,Genes:"SOCS3;CISH;STAT1;FOS",id:24},{Term:"Apoptosis",Overlap:"5/142",P_value:6e-4,Adjusted_P_value:.0034,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:8.1067,Combined_Score:60.7589,Genes:"TUBA1B;GZMB;CTSW;FOS;ACTB",id:25},{Term:"Staphylococcus aureus infection",Overlap:"4/95",P_value:.0011,Adjusted_P_value:.0061,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:9.6777,Combined_Score:66.3714,Genes:"HLA-DRA;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:26},{Term:"Kaposi sarcoma-associated herpesvirus infection",Overlap:"5/193",P_value:.0022,Adjusted_P_value:.0123,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:5.8923,Combined_Score:36.1135,Genes:"STAT1;HLA-B;FOS;HLA-A;RPS27A",id:27},{Term:"Osteoclast differentiation",Overlap:"4/127",P_value:.003,Adjusted_P_value:.0166,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:7.1483,Combined_Score:41.4223,Genes:"SOCS3;STAT1;FOSB;FOS",id:28},{Term:"Natural killer cell mediated cytotoxicity",Overlap:"4/131",P_value:.0034,Adjusted_P_value:.0179,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:6.9218,Combined_Score:39.3392,Genes:"HLA-B;GZMB;HLA-A;KLRC1",id:29},{Term:"Systemic lupus erythematosus",Overlap:"4/135",P_value:.0038,Adjusted_P_value:.0193,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:6.7091,Combined_Score:37.4092,Genes:"HLA-DRA;HLA-DQA1;HLA-DRB1;HLA-DQB1",id:30},{Term:"Ubiquitin mediated proteolysis",Overlap:"4/140",P_value:.0043,Adjusted_P_value:.0213,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:6.4608,Combined_Score:35.1891,Genes:"SOCS3;UBE2L6;CBLB;RPS27A",id:31},{Term:"Salmonella infection",Overlap:"5/249",P_value:.0064,Adjusted_P_value:.0298,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.5271,Combined_Score:22.8528,Genes:"TUBA1B;TUBB;FOS;ACTB;MYL12A",id:32},{Term:"Parkinson disease",Overlap:"5/249",P_value:.0064,Adjusted_P_value:.0298,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.5271,Combined_Score:22.8528,Genes:"TUBA1B;DUSP1;TUBB;UBE2L6;RPS27A",id:33},{Term:"NOD-like receptor signaling pathway",Overlap:"4/181",P_value:.0105,Adjusted_P_value:.0475,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.9539,Combined_Score:22.5622,Genes:"GBP5;STAT1;TXNIP;GBP2",id:34},{Term:"T cell receptor signaling pathway",Overlap:"3/104",P_value:.013,Adjusted_P_value:.0555,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:6.4645,Combined_Score:28.0702,Genes:"CTLA4;CBLB;FOS",id:35},{Term:"Toll-like receptor signaling pathway",Overlap:"3/104",P_value:.013,Adjusted_P_value:.0555,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:6.4645,Combined_Score:28.0702,Genes:"STAT1;CCL4L2;FOS",id:36},{Term:"Pathogenic Escherichia coli infection",Overlap:"4/197",P_value:.014,Adjusted_P_value:.0582,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.5396,Combined_Score:19.3805,Genes:"TUBA1B;TUBB;FOS;ACTB",id:37},{Term:"Viral carcinogenesis",Overlap:"4/203",P_value:.0155,Adjusted_P_value:.0626,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.4013,Combined_Score:18.3512,Genes:"RBL2;PKM;HLA-B;HLA-A",id:38},{Term:"Human immunodeficiency virus 1 infection",Overlap:"4/212",P_value:.0178,Adjusted_P_value:.0702,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.209,Combined_Score:16.9466,Genes:"HLA-B;FOS;HLA-A;B2M",id:39},{Term:"Growth hormone synthesis, secretion and action",Overlap:"3/119",P_value:.0186,Adjusted_P_value:.0702,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:5.6243,Combined_Score:22.41,Genes:"SOCS3;STAT1;FOS",id:40},{Term:"Thyroid hormone signaling pathway",Overlap:"3/121",P_value:.0194,Adjusted_P_value:.0702,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:5.5284,Combined_Score:21.7861,Genes:"STAT1;ATP1A1;ACTB",id:41},{Term:"Type II diabetes mellitus",Overlap:"2/46",P_value:.0198,Adjusted_P_value:.0702,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:9.8132,Combined_Score:38.4975,Genes:"SOCS3;PKM",id:42},{Term:"Proteasome",Overlap:"2/46",P_value:.0198,Adjusted_P_value:.0702,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:9.8132,Combined_Score:38.4975,Genes:"PSMB8;PSMB9",id:43},{Term:"Human papillomavirus infection",Overlap:"5/331",P_value:.02,Adjusted_P_value:.0702,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.3742,Combined_Score:13.2017,Genes:"RBL2;PKM;STAT1;HLA-B;HLA-A",id:44},{Term:"Human cytomegalovirus infection",Overlap:"4/225",P_value:.0217,Adjusted_P_value:.0744,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.9588,Combined_Score:15.171,Genes:"CCL4L2;HLA-B;HLA-A;B2M",id:45},{Term:"Insulin signaling pathway",Overlap:"3/137",P_value:.0268,Adjusted_P_value:.0898,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.8644,Combined_Score:17.602,Genes:"SOCS3;PDE3B;CBLB",id:46},{Term:"Fluid shear stress and atherosclerosis",Overlap:"3/139",P_value:.0278,Adjusted_P_value:.0898,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.7923,Combined_Score:17.1634,Genes:"DUSP1;FOS;ACTB",id:47},{Term:"Measles",Overlap:"3/139",P_value:.0278,Adjusted_P_value:.0898,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.7923,Combined_Score:17.1634,Genes:"STAT1;CBLB;FOS",id:48},{Term:"Cellular senescence",Overlap:"3/156",P_value:.0373,Adjusted_P_value:.1178,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.2562,Combined_Score:13.9999,Genes:"RBL2;HLA-B;HLA-A",id:49},{Term:"JAK-STAT signaling pathway",Overlap:"3/162",P_value:.041,Adjusted_P_value:.1269,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.0943,Combined_Score:13.0823,Genes:"SOCS3;CISH;STAT1",id:50},{Term:"Amphetamine addiction",Overlap:"2/69",P_value:.0418,Adjusted_P_value:.1271,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:6.4371,Combined_Score:20.431,Genes:"FOSB;FOS",id:51},{Term:"Tight junction",Overlap:"3/169",P_value:.0455,Adjusted_P_value:.1356,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.9203,Combined_Score:12.1164,Genes:"TUBA1B;ACTB;MYL12A",id:52},{Term:"Gastric acid secretion",Overlap:"2/76",P_value:.0498,Adjusted_P_value:.1435,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:5.8261,Combined_Score:17.4801,Genes:"ATP1A1;ACTB",id:53},{Term:"MAPK signaling pathway",Overlap:"4/294",P_value:.05,Adjusted_P_value:.1435,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.0063,Combined_Score:9.009,Genes:"DUSP2;DUSP1;FOS;AREG",id:54},{Term:"ErbB signaling pathway",Overlap:"2/85",P_value:.0607,Adjusted_P_value:.1712,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:5.192,Combined_Score:14.5477,Genes:"CBLB;AREG",id:55},{Term:"Colorectal cancer",Overlap:"2/86",P_value:.062,Adjusted_P_value:.1717,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:5.1299,Combined_Score:14.2683,Genes:"FOS;AREG",id:56},{Term:"Gap junction",Overlap:"2/88",P_value:.0645,Adjusted_P_value:.1757,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:5.0101,Combined_Score:13.7333,Genes:"TUBA1B;TUBB",id:57},{Term:"PD-L1 expression and PD-1 checkpoint pathway in cancer",Overlap:"2/89",P_value:.0658,Adjusted_P_value:.1762,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.9523,Combined_Score:13.4769,Genes:"STAT1;FOS",id:58},{Term:"IL-17 signaling pathway",Overlap:"2/94",P_value:.0723,Adjusted_P_value:.1905,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.6819,Combined_Score:12.2962,Genes:"FOSB;FOS",id:59},{Term:"cAMP signaling pathway",Overlap:"3/216",P_value:.0817,Adjusted_P_value:.2116,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.048,Combined_Score:7.6351,Genes:"PDE3B;FOS;ATP1A1",id:60},{Term:"C-type lectin receptor signaling pathway",Overlap:"2/104",P_value:.0861,Adjusted_P_value:.2158,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.2208,Combined_Score:10.3524,Genes:"STAT1;CBLB",id:61},{Term:"NF-kappa B signaling pathway",Overlap:"2/104",P_value:.0861,Adjusted_P_value:.2158,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.2208,Combined_Score:10.3524,Genes:"CCL4L2;LTB",id:62},{Term:"Glucagon signaling pathway",Overlap:"2/107",P_value:.0903,Adjusted_P_value:.223,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.0996,Combined_Score:9.8572,Genes:"PKM;PDE3B",id:63},{Term:"TNF signaling pathway",Overlap:"2/112",P_value:.0975,Adjusted_P_value:.2371,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.9123,Combined_Score:9.1055,Genes:"SOCS3;FOS",id:64},{Term:"Leukocyte transendothelial migration",Overlap:"2/114",P_value:.1005,Adjusted_P_value:.2405,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.842,Combined_Score:8.8281,Genes:"ACTB;MYL12A",id:65},{Term:"Proximal tubule bicarbonate reclamation",Overlap:"1/23",P_value:.1027,Adjusted_P_value:.2423,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:9.7185,Combined_Score:22.1145,Genes:"ATP1A1",id:66},{Term:"Shigellosis",Overlap:"3/246",P_value:.1096,Adjusted_P_value:.2547,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.6676,Combined_Score:5.8972,Genes:"RPS27A;ACTB;MYL12A",id:67},{Term:"Platelet activation",Overlap:"2/124",P_value:.1155,Adjusted_P_value:.261,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.5253,Combined_Score:7.6087,Genes:"ACTB;MYL12A",id:68},{Term:"Endocytosis",Overlap:"3/252",P_value:.1156,Adjusted_P_value:.261,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.6025,Combined_Score:5.615,Genes:"HLA-B;CBLB;HLA-A",id:69},{Term:"Purine metabolism",Overlap:"2/129",P_value:.1232,Adjusted_P_value:.2743,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.3857,Combined_Score:7.0882,Genes:"PKM;PDE3B",id:70},{Term:"Yersinia infection",Overlap:"2/137",P_value:.1359,Adjusted_P_value:.294,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.1837,Combined_Score:6.3554,Genes:"FOS;ACTB",id:71},{Term:"Estrogen signaling pathway",Overlap:"2/137",P_value:.1359,Adjusted_P_value:.294,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.1837,Combined_Score:6.3554,Genes:"FOS;FKBP5",id:72},{Term:"Adrenergic signaling in cardiomyocytes",Overlap:"2/150",P_value:.1569,Adjusted_P_value:.335,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.9022,Combined_Score:5.3752,Genes:"CREM;ATP1A1",id:73},{Term:"Aldosterone-regulated sodium reabsorption",Overlap:"1/37",P_value:.1601,Adjusted_P_value:.3373,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:5.9349,Combined_Score:10.8726,Genes:"ATP1A1",id:74},{Term:"Oxytocin signaling pathway",Overlap:"2/154",P_value:.1635,Adjusted_P_value:.3389,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.8252,Combined_Score:5.1163,Genes:"FOS;ACTB",id:75},{Term:"Non-alcoholic fatty liver disease",Overlap:"2/155",P_value:.1652,Adjusted_P_value:.3389,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.8066,Combined_Score:5.0542,Genes:"SOCS3;FOS",id:76},{Term:"Hepatitis C",Overlap:"2/157",P_value:.1685,Adjusted_P_value:.3413,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.7701,Combined_Score:4.9333,Genes:"SOCS3;STAT1",id:77},{Term:"Hepatitis B",Overlap:"2/162",P_value:.1768,Adjusted_P_value:.3526,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.6829,Combined_Score:4.648,Genes:"STAT1;FOS",id:78},{Term:"Hippo signaling pathway",Overlap:"2/163",P_value:.1785,Adjusted_P_value:.3526,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.6661,Combined_Score:4.5937,Genes:"AREG;ACTB",id:79},{Term:"Pathways of neurodegeneration",Overlap:"4/475",P_value:.1848,Adjusted_P_value:.357,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.8339,Combined_Score:3.0961,Genes:"TUBA1B;TUBB;UBE2L6;RPS27A",id:80},{Term:"cGMP-PKG signaling pathway",Overlap:"2/167",P_value:.1853,Adjusted_P_value:.357,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.6009,Combined_Score:4.385,Genes:"PDE3B;ATP1A1",id:81},{Term:"Carbohydrate digestion and absorption",Overlap:"1/47",P_value:.1988,Adjusted_P_value:.374,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.6424,Combined_Score:7.4989,Genes:"ATP1A1",id:82},{Term:"Pyruvate metabolism",Overlap:"1/47",P_value:.1988,Adjusted_P_value:.374,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.6424,Combined_Score:7.4989,Genes:"PKM",id:83},{Term:"Sphingolipid metabolism",Overlap:"1/49",P_value:.2064,Adjusted_P_value:.3791,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.4485,Combined_Score:7.0204,Genes:"CERS6",id:84},{Term:"Cocaine addiction",Overlap:"1/49",P_value:.2064,Adjusted_P_value:.3791,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.4485,Combined_Score:7.0204,Genes:"FOSB",id:85},{Term:"Vibrio cholerae infection",Overlap:"1/50",P_value:.2101,Adjusted_P_value:.3816,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.3575,Combined_Score:6.7985,Genes:"ACTB",id:86},{Term:"Endocrine and other factor-regulated calcium reabsorption",Overlap:"1/53",P_value:.2212,Adjusted_P_value:.3968,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:4.1055,Combined_Score:6.1937,Genes:"ATP1A1",id:87},{Term:"Transcriptional misregulation in cancer",Overlap:"2/192",P_value:.2281,Adjusted_P_value:.3968,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.2558,Combined_Score:3.3338,Genes:"NFKBIZ;GZMB",id:88},{Term:"Chemokine signaling pathway",Overlap:"2/192",P_value:.2281,Adjusted_P_value:.3968,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.2558,Combined_Score:3.3338,Genes:"STAT1;CCL4L2",id:89},{Term:"Regulation of lipolysis in adipocytes",Overlap:"1/55",P_value:.2285,Adjusted_P_value:.3968,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.953,Combined_Score:5.835,Genes:"PDE3B",id:90},{Term:"Legionellosis",Overlap:"1/57",P_value:.2358,Adjusted_P_value:.4049,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.8114,Combined_Score:5.5069,Genes:"EEF1A1",id:91},{Term:"Focal adhesion",Overlap:"2/201",P_value:.2437,Adjusted_P_value:.41,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.1528,Combined_Score:3.039,Genes:"ACTB;MYL12A",id:92},{Term:"Amyotrophic lateral sclerosis",Overlap:"3/364",P_value:.2447,Adjusted_P_value:.41,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.7849,Combined_Score:2.5125,Genes:"TUBA1B;TUBB;ACTB",id:93},{Term:"Mineral absorption",Overlap:"1/60",P_value:.2465,Adjusted_P_value:.41,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.6171,Combined_Score:5.0647,Genes:"ATP1A1",id:94},{Term:"Cytosolic DNA-sensing pathway",Overlap:"1/63",P_value:.2571,Adjusted_P_value:.4232,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.4416,Combined_Score:4.674,Genes:"CCL4L2",id:95},{Term:"Glycolysis / Gluconeogenesis",Overlap:"1/67",P_value:.2711,Adjusted_P_value:.4348,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.2323,Combined_Score:4.2195,Genes:"PKM",id:96},{Term:"Regulation of actin cytoskeleton",Overlap:"2/218",P_value:.2733,Adjusted_P_value:.4348,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.9817,Combined_Score:2.5703,Genes:"ACTB;MYL12A",id:97},{Term:"Mitophagy",Overlap:"1/68",P_value:.2745,Adjusted_P_value:.4348,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.1839,Combined_Score:4.1162,Genes:"RPS27A",id:98},{Term:"Adipocytokine signaling pathway",Overlap:"1/69",P_value:.2779,Adjusted_P_value:.4348,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.1369,Combined_Score:4.0166,Genes:"SOCS3",id:99},{Term:"Renin secretion",Overlap:"1/69",P_value:.2779,Adjusted_P_value:.4348,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.1369,Combined_Score:4.0166,Genes:"PDE3B",id:100},{Term:"Central carbon metabolism in cancer",Overlap:"1/70",P_value:.2813,Adjusted_P_value:.4358,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.0913,Combined_Score:3.9205,Genes:"PKM",id:101},{Term:"Adherens junction",Overlap:"1/71",P_value:.2847,Adjusted_P_value:.4367,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:3.047,Combined_Score:3.8279,Genes:"ACTB",id:102},{Term:"Thyroid hormone synthesis",Overlap:"1/75",P_value:.2981,Adjusted_P_value:.4458,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.8817,Combined_Score:3.4877,Genes:"ATP1A1",id:103},{Term:"Pancreatic cancer",Overlap:"1/76",P_value:.3014,Adjusted_P_value:.4458,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.8432,Combined_Score:3.4096,Genes:"STAT1",id:104},{Term:"Pertussis",Overlap:"1/76",P_value:.3014,Adjusted_P_value:.4458,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.8432,Combined_Score:3.4096,Genes:"FOS",id:105},{Term:"Arrhythmogenic right ventricular cardiomyopathy",Overlap:"1/77",P_value:.3047,Adjusted_P_value:.4458,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.8056,Combined_Score:3.334,Genes:"ACTB",id:106},{Term:"Bacterial invasion of epithelial cells",Overlap:"1/77",P_value:.3047,Adjusted_P_value:.4458,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.8056,Combined_Score:3.334,Genes:"ACTB",id:107},{Term:"RNA degradation",Overlap:"1/79",P_value:.3113,Adjusted_P_value:.4512,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.7334,Combined_Score:3.1901,Genes:"BTG1",id:108},{Term:"B cell receptor signaling pathway",Overlap:"1/81",P_value:.3178,Adjusted_P_value:.4564,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.6648,Combined_Score:3.0551,Genes:"FOS",id:109},{Term:"Insulin secretion",Overlap:"1/86",P_value:.3337,Adjusted_P_value:.475,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.5074,Combined_Score:2.7519,Genes:"ATP1A1",id:110},{Term:"Cardiac muscle contraction",Overlap:"1/87",P_value:.3368,Adjusted_P_value:.4752,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.4781,Combined_Score:2.6965,Genes:"ATP1A1",id:111},{Term:"Bile secretion",Overlap:"1/90",P_value:.3462,Adjusted_P_value:.4792,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.3942,Combined_Score:2.5397,Genes:"ATP1A1",id:112},{Term:"Hypertrophic cardiomyopathy",Overlap:"1/90",P_value:.3462,Adjusted_P_value:.4792,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.3942,Combined_Score:2.5397,Genes:"ACTB",id:113},{Term:"Morphine addiction",Overlap:"1/91",P_value:.3493,Adjusted_P_value:.4792,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.3675,Combined_Score:2.4903,Genes:"PDE3B",id:114},{Term:"Salivary secretion",Overlap:"1/93",P_value:.3554,Adjusted_P_value:.4792,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.3158,Combined_Score:2.3956,Genes:"ATP1A1",id:115},{Term:"Dilated cardiomyopathy",Overlap:"1/96",P_value:.3645,Adjusted_P_value:.4792,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.2423,Combined_Score:2.263,Genes:"ACTB",id:116},{Term:"Circadian entrainment",Overlap:"1/97",P_value:.3675,Adjusted_P_value:.4792,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.2189,Combined_Score:2.2211,Genes:"FOS",id:117},{Term:"Prion disease",Overlap:"2/273",P_value:.3681,Adjusted_P_value:.4792,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.5751,Combined_Score:1.5742,Genes:"TUBA1B;TUBB",id:118},{Term:"Aldosterone synthesis and secretion",Overlap:"1/98",P_value:.3705,Adjusted_P_value:.4792,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.1959,Combined_Score:2.1804,Genes:"ATP1A1",id:119},{Term:"Choline metabolism in cancer",Overlap:"1/98",P_value:.3705,Adjusted_P_value:.4792,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.1959,Combined_Score:2.1804,Genes:"FOS",id:120},{Term:"AGE-RAGE signaling pathway in diabetic complications",Overlap:"1/100",P_value:.3764,Adjusted_P_value:.4792,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.1513,Combined_Score:2.1019,Genes:"STAT1",id:121},{Term:"Viral protein interaction with cytokine and cytokine receptor",Overlap:"1/100",P_value:.3764,Adjusted_P_value:.4792,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.1513,Combined_Score:2.1019,Genes:"CCL4L2",id:122},{Term:"Progesterone-mediated oocyte maturation",Overlap:"1/100",P_value:.3764,Adjusted_P_value:.4792,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.1513,Combined_Score:2.1019,Genes:"PDE3B",id:123},{Term:"Chagas disease",Overlap:"1/102",P_value:.3823,Adjusted_P_value:.4792,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.1085,Combined_Score:2.0274,Genes:"FOS",id:124},{Term:"Pancreatic secretion",Overlap:"1/102",P_value:.3823,Adjusted_P_value:.4792,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.1085,Combined_Score:2.0274,Genes:"ATP1A1",id:125},{Term:"Protein digestion and absorption",Overlap:"1/103",P_value:.3852,Adjusted_P_value:.4792,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.0877,Combined_Score:1.9916,Genes:"ATP1A1",id:126},{Term:"Parathyroid hormone synthesis, secretion and action",Overlap:"1/106",P_value:.3939,Adjusted_P_value:.4862,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:2.0278,Combined_Score:1.8892,Genes:"FOS",id:127},{Term:"Insulin resistance",Overlap:"1/108",P_value:.3996,Adjusted_P_value:.4894,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.9896,Combined_Score:1.8251,Genes:"SOCS3",id:128},{Term:"Cytokine-cytokine receptor interaction",Overlap:"2/295",P_value:.4048,Adjusted_P_value:.492,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.4552,Combined_Score:1.316,Genes:"CCL4L2;LTB",id:129},{Term:"Serotonergic synapse",Overlap:"1/113",P_value:.4137,Adjusted_P_value:.4951,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.9003,Combined_Score:1.6775,Genes:"DUSP1",id:130},{Term:"Cholinergic synapse",Overlap:"1/113",P_value:.4137,Adjusted_P_value:.4951,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.9003,Combined_Score:1.6775,Genes:"FOS",id:131},{Term:"Huntington disease",Overlap:"2/306",P_value:.4228,Adjusted_P_value:.5022,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.4017,Combined_Score:1.2068,Genes:"TUBA1B;TUBB",id:132},{Term:"Sphingolipid signaling pathway",Overlap:"1/119",P_value:.4301,Adjusted_P_value:.5071,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.8032,Combined_Score:1.5214,Genes:"CERS6",id:133},{Term:"Cell cycle",Overlap:"1/124",P_value:.4434,Adjusted_P_value:.519,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.7294,Combined_Score:1.4064,Genes:"RBL2",id:134},{Term:"Lysosome",Overlap:"1/128",P_value:.4539,Adjusted_P_value:.5264,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.6746,Combined_Score:1.3228,Genes:"CTSW",id:135},{Term:"Relaxin signaling pathway",Overlap:"1/129",P_value:.4565,Adjusted_P_value:.5264,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.6615,Combined_Score:1.3029,Genes:"FOS",id:136},{Term:"FoxO signaling pathway",Overlap:"1/131",P_value:.4616,Adjusted_P_value:.5276,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.6357,Combined_Score:1.2645,Genes:"RBL2",id:137},{Term:"Dopaminergic synapse",Overlap:"1/132",P_value:.4642,Adjusted_P_value:.5276,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.6232,Combined_Score:1.2458,Genes:"FOS",id:138},{Term:"Apelin signaling pathway",Overlap:"1/137",P_value:.4767,Adjusted_P_value:.538,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.5631,Combined_Score:1.158,Genes:"PDE3B",id:139},{Term:"PI3K-Akt signaling pathway",Overlap:"2/354",P_value:.4978,Adjusted_P_value:.5574,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.2076,Combined_Score:.8424,Genes:"RBL2;AREG",id:140},{Term:"Breast cancer",Overlap:"1/147",P_value:.501,Adjusted_P_value:.5574,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.4553,Combined_Score:1.0059,Genes:"FOS",id:141},{Term:"Alzheimer disease",Overlap:"2/369",P_value:.5199,Adjusted_P_value:.5745,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.1574,Combined_Score:.757,Genes:"TUBA1B;TUBB",id:142},{Term:"Necroptosis",Overlap:"1/159",P_value:.5286,Adjusted_P_value:.58,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.3439,Combined_Score:.8568,Genes:"STAT1",id:143},{Term:"Hepatocellular carcinoma",Overlap:"1/168",P_value:.5483,Adjusted_P_value:.5975,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.2709,Combined_Score:.7637,Genes:"ACTB",id:144},{Term:"Axon guidance",Overlap:"1/182",P_value:.5774,Adjusted_P_value:.6249,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.1718,Combined_Score:.6436,Genes:"MYL12A",id:145},{Term:"Alcoholism",Overlap:"1/186",P_value:.5854,Adjusted_P_value:.6249,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.1462,Combined_Score:.6138,Genes:"FOSB",id:146},{Term:"RNA transport",Overlap:"1/186",P_value:.5854,Adjusted_P_value:.6249,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.1462,Combined_Score:.6138,Genes:"EEF1A1",id:147},{Term:"Neutrophil extracellular trap formation",Overlap:"1/189",P_value:.5912,Adjusted_P_value:.6269,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.1278,Combined_Score:.5927,Genes:"ACTB",id:148},{Term:"Proteoglycans in cancer",Overlap:"1/205",P_value:.6212,Adjusted_P_value:.6543,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.0385,Combined_Score:.4944,Genes:"ACTB",id:149},{Term:"Rap1 signaling pathway",Overlap:"1/210",P_value:.6301,Adjusted_P_value:.6593,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:1.0134,Combined_Score:.468,Genes:"ACTB",id:150},{Term:"Lipid and atherosclerosis",Overlap:"1/215",P_value:.6388,Adjusted_P_value:.664,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.9894,Combined_Score:.4434,Genes:"FOS",id:151},{Term:"Thermogenesis",Overlap:"1/232",P_value:.6669,Adjusted_P_value:.6887,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.9158,Combined_Score:.371,Genes:"ACTB",id:152},{Term:"Chemical carcinogenesis",Overlap:"1/239",P_value:.6778,Adjusted_P_value:.6925,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.8886,Combined_Score:.3455,Genes:"FOS",id:153},{Term:"Calcium signaling pathway",Overlap:"1/240",P_value:.6794,Adjusted_P_value:.6925,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.8848,Combined_Score:.3421,Genes:"P2RX5",id:154},{Term:"Pathways in cancer",Overlap:"2/531",P_value:.7167,Adjusted_P_value:.7259,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.7963,Combined_Score:.2653,Genes:"STAT1;FOS",id:155},{Term:"MicroRNAs in cancer",Overlap:"1/310",P_value:.7705,Adjusted_P_value:.7754,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.6819,Combined_Score:.1778,Genes:"VIM",id:156},{Term:"Neuroactive ligand-receptor interaction",Overlap:"1/341",P_value:.8022,Adjusted_P_value:.8022,Old_P_value:0,Old_Adjusted_P_value:0,Odds_Ratio:.6188,Combined_Score:.1364,Genes:"P2RX5",id:157}]}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p:T,i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function Kn(e){let a,t,n,l;return a=new me({props:{$$slots:{default:[zn]},$$scope:{ctx:e}}}),n=new me({props:{$$slots:{default:[qn]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment),t=f(),g(n.$$.fragment)},m(e,_){P(a,e,_),d(e,t,_),P(n,e,_),l=!0},p(e,t){const l={};1&t&&(l.$$scope={dirty:t,ctx:e}),a.$set(l);const _={};1&t&&(_.$$scope={dirty:t,ctx:e}),n.$set(_)},i(e){l||(i(a.$$.fragment,e),i(n.$$.fragment,e),l=!0)},o(e){c(a.$$.fragment,e),c(n.$$.fragment,e),l=!1},d(e){e&&p(t),A(a,e),A(n,e)}}}function Yn(e){let a,t;return a=new ue({props:{$$slots:{content:[Kn],default:[kn]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};1&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function Wn(e){let a,t;return a=new Me({props:{title:"KEGG_2021_Human",open:!0,$$slots:{default:[Yn]},$$scope:{ctx:e}}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},p(e,t){const n={};1&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function Xn(e){let a,t,n,l,r,o,u,v,m,b,C,O,S,h,R,$,T,y,L,j,x,M,B,G,F,w,N,U,D,E;return u=new ue({props:{$$slots:{content:[Rn],default:[An]},$$scope:{ctx:e}}}),O=new Pn({props:{$$slots:{default:[$n]},$$scope:{ctx:e}}}),h=new xe({props:{align:"start",$$slots:{default:[Bn]},$$scope:{ctx:e}}}),M=new ue({props:{$$slots:{content:[Hn],default:[Gn]},$$scope:{ctx:e}}}),N=new Pn({props:{$$slots:{default:[In]},$$scope:{ctx:e}}}),D=new xe({props:{align:"start",$$slots:{default:[Wn]},$$scope:{ctx:e}}}),{c(){a=_("h1"),a.textContent="c1",t=_("a"),t.innerHTML="",n=f(),l=_("h2"),l.textContent="Markers",r=_("a"),r.innerHTML="",o=f(),g(u.$$.fragment),v=f(),m=_("h2"),m.textContent="Enrichment Analysis",b=_("a"),b.innerHTML="",C=f(),g(O.$$.fragment),S=f(),g(h.$$.fragment),R=f(),$=_("h1"),$.textContent="c2",T=_("a"),T.innerHTML="",y=f(),L=_("h2"),L.textContent="Markers",j=_("a"),j.innerHTML="",x=f(),g(M.$$.fragment),B=f(),G=_("h2"),G.textContent="Enrichment Analysis",F=_("a"),F.innerHTML="",w=f(),g(N.$$.fragment),U=f(),g(D.$$.fragment),s(t,"id","prt-h1-0-c1"),s(t,"class","pipen-report-toc-anchor"),s(r,"id","prt-h2-0-markers"),s(r,"class","pipen-report-toc-anchor"),s(b,"id","prt-h2-1-enrichment-analysis"),s(b,"class","pipen-report-toc-anchor"),s(T,"id","prt-h1-1-c2"),s(T,"class","pipen-report-toc-anchor"),s(j,"id","prt-h2-2-markers"),s(j,"class","pipen-report-toc-anchor"),s(F,"id","prt-h2-3-enrichment-analysis"),s(F,"class","pipen-report-toc-anchor")},m(e,_){d(e,a,_),d(e,t,_),d(e,n,_),d(e,l,_),d(e,r,_),d(e,o,_),P(u,e,_),d(e,v,_),d(e,m,_),d(e,b,_),d(e,C,_),P(O,e,_),d(e,S,_),P(h,e,_),d(e,R,_),d(e,$,_),d(e,T,_),d(e,y,_),d(e,L,_),d(e,j,_),d(e,x,_),P(M,e,_),d(e,B,_),d(e,G,_),d(e,F,_),d(e,w,_),P(N,e,_),d(e,U,_),P(D,e,_),E=!0},p(e,[a]){const t={};1&a&&(t.$$scope={dirty:a,ctx:e}),u.$set(t);const n={};1&a&&(n.$$scope={dirty:a,ctx:e}),O.$set(n);const l={};1&a&&(l.$$scope={dirty:a,ctx:e}),h.$set(l);const _={};1&a&&(_.$$scope={dirty:a,ctx:e}),M.$set(_);const s={};1&a&&(s.$$scope={dirty:a,ctx:e}),N.$set(s);const d={};1&a&&(d.$$scope={dirty:a,ctx:e}),D.$set(d)},i(e){E||(i(u.$$.fragment,e),i(O.$$.fragment,e),i(h.$$.fragment,e),i(M.$$.fragment,e),i(N.$$.fragment,e),i(D.$$.fragment,e),E=!0)},o(e){c(u.$$.fragment,e),c(O.$$.fragment,e),c(h.$$.fragment,e),c(M.$$.fragment,e),c(N.$$.fragment,e),c(D.$$.fragment,e),E=!1},d(e){e&&(p(a),p(t),p(n),p(l),p(r),p(o),p(v),p(m),p(b),p(C),p(S),p(R),p($),p(T),p(y),p(L),p(j),p(x),p(B),p(G),p(F),p(w),p(U)),A(u,e),A(O,e),A(h,e),A(M,e),A(N,e),A(D,e)}}}class Qn extends a{constructor(e){super(),t(this,e,null,Xn,n,{})}}var Zn=[{slug:"prt-h1-0-c1",text:"c1",children:[{slug:"prt-h2-0-markers",text:"Markers",children:[],page:0},{slug:"prt-h2-1-enrichment-analysis",text:"Enrichment Analysis",children:[],page:0}],page:0},{slug:"prt-h1-1-c2",text:"c2",children:[{slug:"prt-h2-2-markers",text:"Markers",children:[],page:0},{slug:"prt-h2-3-enrichment-analysis",text:"Enrichment Analysis",children:[],page:0}],page:0}],Jn={pipeline:{name:"Immunopipe",desc:"Immunopipe (v1.4.3): Integrative analysis for scRNA-seq and scTCR-seq data"},versions:'\n    <a href="https://github.com/pwwang/pipen" target="_blank">\n        pipen v0.15.7\n    </a> and\n    <a href="https://github.com/pwwang/pipen-report" target="_blank">\n        pipen-report v0.21.4\n    </a>\n',entries:[{name:"SampleInfo",desc:"List sample information and perform statistics",npages:1,report_toc:!0,order:0,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n[1] ggrepel_0.9.6    ggprism_1.0.5    ggplot2_3.5.1    tidyr_1.3.1     \n[5] dplyr_1.1.4      tidyselect_1.2.1 jsonlite_1.8.9   rlang_1.1.5     \n[9] logger_0.4.0    \n\nloaded via a namespace (and not attached):\n [1] vctrs_0.6.5      cli_3.6.4        purrr_1.0.2      generics_0.1.3  \n [5] glue_1.8.0       labeling_0.4.3   colorspace_2.1-1 scales_1.3.0    \n [9] grid_4.3.3       munsell_0.5.1    tibble_3.2.1     lifecycle_1.0.4 \n[13] compiler_4.3.3   Rcpp_1.0.14      pkgconfig_2.0.3  farver_2.1.2    \n[17] digest_0.6.37    R6_2.6.0         pillar_1.10.1    magrittr_2.0.3  \n[21] withr_3.0.2      gtable_0.3.6    \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 2.19\nuser 11.15\nsys 0.11\npercent 513%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3421.613\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"SeuratPreparing",desc:"Load, prepare and apply QC to data, using `Seurat`",npages:1,report_toc:!0,order:2,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats4    stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] scDblFinder_1.16.0          SingleCellExperiment_1.24.0\n [3] SummarizedExperiment_1.32.0 Biobase_2.62.0             \n [5] GenomicRanges_1.54.1        GenomeInfoDb_1.38.1        \n [7] IRanges_2.36.0              S4Vectors_0.40.2           \n [9] BiocGenerics_0.48.1         MatrixGenerics_1.14.0      \n[11] matrixStats_1.1.0           dplyr_1.1.4                \n[13] ggplot2_3.5.1               bracer_1.2.1               \n[15] future_1.34.0               Seurat_5.1.0               \n[17] SeuratObject_5.0.2          sp_2.2-0                   \n[19] digest_0.6.37               jsonlite_1.8.9             \n[21] rlang_1.1.5                 logger_0.4.0               \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22          splines_4.3.3            \n  [3] later_1.4.1               BiocIO_1.12.0            \n  [5] bitops_1.0-9              tibble_3.2.1             \n  [7] R.oo_1.27.0               polyclip_1.10-7          \n  [9] XML_3.99-0.17             fastDummies_1.7.5        \n [11] lifecycle_1.0.4           edgeR_4.0.16             \n [13] globals_0.16.3            lattice_0.22-6           \n [15] MASS_7.3-60.0.1           magrittr_2.0.3           \n [17] limma_3.58.1              plotly_4.10.4            \n [19] yaml_2.3.10               metapod_1.10.0           \n [21] httpuv_1.6.15             sctransform_0.4.1        \n [23] spam_2.11-1               spatstat.sparse_3.1-0    \n [25] reticulate_1.40.0         cowplot_1.1.3            \n [27] pbapply_1.7-2             RColorBrewer_1.1-3       \n [29] abind_1.4-5               zlibbioc_1.48.0          \n [31] Rtsne_0.17                purrr_1.0.2              \n [33] R.utils_2.12.3            RCurl_1.98-1.16          \n [35] GenomeInfoDbData_1.2.11   ggrepel_0.9.6            \n [37] irlba_2.3.5.1             listenv_0.9.1            \n [39] spatstat.utils_3.1-2      goftest_1.2-3            \n [41] RSpectra_0.16-2           dqrng_0.3.2              \n [43] spatstat.random_3.3-2     fitdistrplus_1.2-2       \n [45] parallelly_1.42.0         DelayedMatrixStats_1.24.0\n [47] leiden_0.4.3.1            codetools_0.2-20         \n [49] DelayedArray_0.28.0       scuttle_1.12.0           \n [51] tidyselect_1.2.1          farver_2.1.2             \n [53] viridis_0.6.5             ScaledMatrix_1.10.0      \n [55] spatstat.explore_3.3-4    GenomicAlignments_1.38.0 \n [57] BiocNeighbors_1.20.0      progressr_0.15.1         \n [59] scater_1.30.1             ggridges_0.5.6           \n [61] survival_3.8-3            systemfonts_1.2.1        \n [63] tools_4.3.3               ragg_1.3.3               \n [65] ica_1.0-3                 Rcpp_1.0.14              \n [67] glue_1.8.0                gridExtra_2.3            \n [69] SparseArray_1.2.2         withr_3.0.2              \n [71] fastmap_1.2.0             bluster_1.12.0           \n [73] rsvd_1.0.5                R6_2.6.0                 \n [75] mime_0.12                 textshaping_0.4.0        \n [77] colorspace_2.1-1          scattermore_1.2          \n [79] tensor_1.5                spatstat.data_3.1-4      \n [81] R.methodsS3_1.8.2         RhpcBLASctl_0.23-42      \n [83] tidyr_1.3.1               generics_0.1.3           \n [85] data.table_1.16.4         rtracklayer_1.62.0       \n [87] httr_1.4.7                htmlwidgets_1.6.4        \n [89] S4Arrays_1.2.0            uwot_0.2.2               \n [91] pkgconfig_2.0.3           gtable_0.3.6             \n [93] lmtest_0.9-40             XVector_0.42.0           \n [95] htmltools_0.5.8.1         dotCall64_1.2            \n [97] scales_1.3.0              png_0.1-8                \n [99] harmony_1.2.0             spatstat.univar_3.1-1    \n[101] scran_1.30.0              reshape2_1.4.4           \n[103] rjson_0.2.23              nlme_3.1-167             \n[105] zoo_1.8-12                stringr_1.5.1            \n[107] KernSmooth_2.23-26        vipor_0.4.7              \n[109] parallel_4.3.3            miniUI_0.1.1.1           \n[111] restfulr_0.0.15           pillar_1.10.1            \n[113] grid_4.3.3                vctrs_0.6.5              \n[115] RANN_2.6.2                promises_1.3.2           \n[117] BiocSingular_1.18.0       beachmat_2.18.0          \n[119] xtable_1.8-4              cluster_2.1.8            \n[121] beeswarm_0.4.0            locfit_1.5-9.11          \n[123] cli_3.6.4                 compiler_4.3.3           \n[125] Rsamtools_2.18.0          crayon_1.5.3             \n[127] future.apply_1.11.3       labeling_0.4.3           \n[129] ggbeeswarm_0.7.2          plyr_1.8.9               \n[131] stringi_1.8.4             viridisLite_0.4.2        \n[133] deldir_2.0-4              BiocParallel_1.36.0      \n[135] munsell_0.5.1             Biostrings_2.70.1        \n[137] lazyeval_0.2.2            spatstat.geom_3.3-5      \n[139] Matrix_1.6-5              RcppHNSW_0.6.0           \n[141] patchwork_1.3.0           sparseMatrixStats_1.14.0 \n[143] statmod_1.5.0             shiny_1.10.0             \n[145] ROCR_1.0-11               igraph_2.0.3             \n[147] xgboost_2.1.4.1          \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 209.65\nuser 1454.55\nsys 7.57\npercent 697%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ClusterMarkersOfAllCells",desc:"Markers for clusters of all cells.",npages:3,report_toc:!0,order:4,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] UpSetR_1.4.0        ggVennDiagram_1.2.3 tidyseurat_0.8.0   \n [4] ttservice_0.4.1     future_1.34.0       ggrepel_0.9.6      \n [7] ggprism_1.0.5       ggplot2_3.5.1       enrichR_3.4        \n[10] Seurat_5.1.0        SeuratObject_5.0.2  sp_2.2-0           \n[13] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[16] tidyselect_1.2.1    digest_0.6.37       jsonlite_1.8.9     \n[19] rlang_1.1.5         logger_0.4.0       \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] ellipsis_0.3.2         labeling_0.4.3         promises_1.3.2        \n [19] purrr_1.0.2            WriteXLS_6.7.0         RVenn_1.1.0           \n [22] goftest_1.2-3          later_1.4.1            spatstat.utils_3.1-2  \n [25] irlba_2.3.5.1          parallel_4.3.3         cluster_2.1.8         \n [28] R6_2.6.0               ica_1.0-3              stringi_1.8.4         \n [31] RColorBrewer_1.1-3     spatstat.data_3.1-4    limma_3.58.1          \n [34] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [37] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [40] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [43] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [46] splines_4.3.3          igraph_2.0.3           abind_1.4-5           \n [49] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.3-4 curl_6.0.1             listenv_0.9.1         \n [55] lattice_0.22-6         plyr_1.8.9             withr_3.0.2           \n [58] shiny_1.10.0           ROCR_1.0-11            Rtsne_0.17            \n [61] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n [64] fitdistrplus_1.2-2     pillar_1.10.1          KernSmooth_2.23-26    \n [67] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n [82] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n [85] nlme_3.1-167           patchwork_1.3.0        presto_1.0.0          \n [88] cli_3.6.4              spatstat.sparse_3.1-0  fansi_1.0.6           \n [91] spam_2.11-1            viridisLite_0.4.2      uwot_0.2.2            \n [94] gtable_0.3.6           progressr_0.15.1       rjson_0.2.23          \n [97] htmlwidgets_1.6.4      farver_2.1.2           htmltools_0.5.8.1     \n[100] lifecycle_1.0.4        httr_1.4.7             statmod_1.5.0         \n[103] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 259.76\nuser 186.00\nsys 8.62\npercent 74%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TopExpressingGenesOfAllCells",desc:"Top expressing genes for clusters of all cells.",npages:3,report_toc:!0,order:5,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ggprism_1.0.5      dplyr_1.1.4        enrichR_3.4        tibble_3.2.1      \n [5] Seurat_5.1.0       SeuratObject_5.0.2 sp_2.2-0           jsonlite_1.8.9    \n [9] rlang_1.1.5        logger_0.4.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] labeling_0.4.3         promises_1.3.2         purrr_1.0.2           \n [19] WriteXLS_6.7.0         goftest_1.2-3          later_1.4.1           \n [22] spatstat.utils_3.1-2   irlba_2.3.5.1          parallel_4.3.3        \n [25] cluster_2.1.8          R6_2.6.0               ica_1.0-3             \n [28] stringi_1.8.4          RColorBrewer_1.1-3     spatstat.data_3.1-4   \n [31] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [37] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [43] splines_4.3.3          igraph_2.0.3           tidyselect_1.2.1      \n [46] abind_1.4-5            spatstat.random_3.3-2  codetools_0.2-20      \n [49] miniUI_0.1.1.1         spatstat.explore_3.3-4 curl_6.0.1            \n [52] listenv_0.9.1          lattice_0.22-6         plyr_1.8.9            \n [55] withr_3.0.2            shiny_1.10.0           ROCR_1.0-11           \n [58] Rtsne_0.17             future_1.34.0          fastDummies_1.7.5     \n [61] survival_3.8-3         polyclip_1.10-7        fitdistrplus_1.2-2    \n [64] pillar_1.10.1          KernSmooth_2.23-26     plotly_4.10.4         \n [67] generics_0.1.3         RcppHNSW_0.6.0         ggplot2_3.5.1         \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n [82] cowplot_1.1.3          grid_4.3.3             tidyr_1.3.1           \n [85] colorspace_2.1-1       nlme_3.1-167           patchwork_1.3.0       \n [88] cli_3.6.4              spatstat.sparse_3.1-0  spam_2.11-1           \n [91] viridisLite_0.4.2      uwot_0.2.2             gtable_0.3.6          \n [94] digest_0.6.37          progressr_0.15.1       ggrepel_0.9.6         \n [97] rjson_0.2.23           htmlwidgets_1.6.4      farver_2.1.2          \n[100] htmltools_0.5.8.1      lifecycle_1.0.4        httr_1.4.7            \n[103] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 73.70\nuser 41.45\nsys 2.53\npercent 59%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TCellSelection",desc:"Separate T and non-T cells and select T cells.",npages:1,report_toc:!0,order:6,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] glue_1.8.0         factoextra_1.0.7   patchwork_1.3.0    ggpubr_0.6.0      \n [5] ggrepel_0.9.6      ggprism_1.0.5      ggplot2_3.5.1      tibble_3.2.1      \n [9] tidyr_1.3.1        dplyr_1.1.4        Seurat_5.1.0       SeuratObject_5.0.2\n[13] sp_2.2-0           jsonlite_1.8.9     rlang_1.1.5        logger_0.4.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] backports_1.5.0        labeling_0.4.3         promises_1.3.2        \n [19] purrr_1.0.2            goftest_1.2-3          later_1.4.1           \n [22] spatstat.utils_3.1-2   broom_1.0.7            irlba_2.3.5.1         \n [25] parallel_4.3.3         cluster_2.1.8          R6_2.6.0              \n [28] ica_1.0-3              stringi_1.8.4          RColorBrewer_1.1-3    \n [31] spatstat.data_3.1-4    reticulate_1.40.0      car_3.1-3             \n [34] parallelly_1.42.0      spatstat.univar_3.1-1  lmtest_0.9-40         \n [37] scattermore_1.2        Rcpp_1.0.14            tensor_1.5            \n [40] future.apply_1.11.3    zoo_1.8-12             sctransform_0.4.1     \n [43] httpuv_1.6.15          Matrix_1.6-5           splines_4.3.3         \n [46] igraph_2.0.3           tidyselect_1.2.1       abind_1.4-5           \n [49] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.3-4 listenv_0.9.1          lattice_0.22-6        \n [55] plyr_1.8.9             withr_3.0.2            shiny_1.10.0          \n [58] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n [61] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n [64] fitdistrplus_1.2-2     pillar_1.10.1          carData_3.0-5         \n [67] KernSmooth_2.23-26     plotly_4.10.4          generics_0.1.3        \n [70] RcppHNSW_0.6.0         munsell_0.5.1          scales_1.3.0          \n [73] globals_0.16.3         xtable_1.8-4           lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] ggsignif_0.6.4         RANN_2.6.2             leiden_0.4.3.1        \n [82] dotCall64_1.2          cowplot_1.1.3          grid_4.3.3            \n [85] colorspace_2.1-1       nlme_3.1-167           Formula_1.2-5         \n [88] cli_3.6.4              spatstat.sparse_3.1-0  spam_2.11-1           \n [91] viridisLite_0.4.2      uwot_0.2.2             gtable_0.3.6          \n [94] rstatix_0.7.2          digest_0.6.37          progressr_0.15.1      \n [97] htmlwidgets_1.6.4      farver_2.1.2           htmltools_0.5.8.1     \n[100] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[103] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 41.63\nuser 49.85\nsys 0.77\npercent 121%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ClusterMarkers",desc:"Markers for clusters of all or selected T cells.",npages:2,report_toc:!0,order:10,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] UpSetR_1.4.0        ggVennDiagram_1.2.3 tidyseurat_0.8.0   \n [4] ttservice_0.4.1     future_1.34.0       ggrepel_0.9.6      \n [7] ggprism_1.0.5       ggplot2_3.5.1       enrichR_3.4        \n[10] Seurat_5.1.0        SeuratObject_5.0.2  sp_2.2-0           \n[13] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[16] tidyselect_1.2.1    digest_0.6.37       jsonlite_1.8.9     \n[19] rlang_1.1.5         logger_0.4.0       \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] ellipsis_0.3.2         labeling_0.4.3         promises_1.3.2        \n [19] purrr_1.0.2            WriteXLS_6.7.0         RVenn_1.1.0           \n [22] goftest_1.2-3          later_1.4.1            spatstat.utils_3.1-2  \n [25] irlba_2.3.5.1          parallel_4.3.3         cluster_2.1.8         \n [28] R6_2.6.0               ica_1.0-3              stringi_1.8.4         \n [31] RColorBrewer_1.1-3     spatstat.data_3.1-4    limma_3.58.1          \n [34] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [37] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [40] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [43] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [46] splines_4.3.3          igraph_2.0.3           abind_1.4-5           \n [49] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.3-4 curl_6.0.1             listenv_0.9.1         \n [55] lattice_0.22-6         plyr_1.8.9             withr_3.0.2           \n [58] shiny_1.10.0           ROCR_1.0-11            Rtsne_0.17            \n [61] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n [64] fitdistrplus_1.2-2     pillar_1.10.1          KernSmooth_2.23-26    \n [67] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n [82] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n [85] nlme_3.1-167           patchwork_1.3.0        presto_1.0.0          \n [88] cli_3.6.4              spatstat.sparse_3.1-0  fansi_1.0.6           \n [91] spam_2.11-1            viridisLite_0.4.2      uwot_0.2.2            \n [94] gtable_0.3.6           progressr_0.15.1       rjson_0.2.23          \n [97] htmlwidgets_1.6.4      farver_2.1.2           htmltools_0.5.8.1     \n[100] lifecycle_1.0.4        httr_1.4.7             statmod_1.5.0         \n[103] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 162.57\nuser 119.15\nsys 5.40\npercent 76%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TopExpressingGenes",desc:"Top expressing genes for clusters of all or selected T cells.",npages:2,report_toc:!0,order:11,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ggprism_1.0.5      dplyr_1.1.4        enrichR_3.4        tibble_3.2.1      \n [5] Seurat_5.1.0       SeuratObject_5.0.2 sp_2.2-0           jsonlite_1.8.9    \n [9] rlang_1.1.5        logger_0.4.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] labeling_0.4.3         promises_1.3.2         purrr_1.0.2           \n [19] WriteXLS_6.7.0         goftest_1.2-3          later_1.4.1           \n [22] spatstat.utils_3.1-2   irlba_2.3.5.1          parallel_4.3.3        \n [25] cluster_2.1.8          R6_2.6.0               ica_1.0-3             \n [28] stringi_1.8.4          RColorBrewer_1.1-3     spatstat.data_3.1-4   \n [31] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [37] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [43] splines_4.3.3          igraph_2.0.3           tidyselect_1.2.1      \n [46] abind_1.4-5            spatstat.random_3.3-2  codetools_0.2-20      \n [49] miniUI_0.1.1.1         spatstat.explore_3.3-4 curl_6.0.1            \n [52] listenv_0.9.1          lattice_0.22-6         plyr_1.8.9            \n [55] withr_3.0.2            shiny_1.10.0           ROCR_1.0-11           \n [58] Rtsne_0.17             future_1.34.0          fastDummies_1.7.5     \n [61] survival_3.8-3         polyclip_1.10-7        fitdistrplus_1.2-2    \n [64] pillar_1.10.1          KernSmooth_2.23-26     plotly_4.10.4         \n [67] generics_0.1.3         RcppHNSW_0.6.0         ggplot2_3.5.1         \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n [82] cowplot_1.1.3          grid_4.3.3             tidyr_1.3.1           \n [85] colorspace_2.1-1       nlme_3.1-167           patchwork_1.3.0       \n [88] cli_3.6.4              spatstat.sparse_3.1-0  spam_2.11-1           \n [91] viridisLite_0.4.2      uwot_0.2.2             gtable_0.3.6          \n [94] digest_0.6.37          progressr_0.15.1       ggrepel_0.9.6         \n [97] rjson_0.2.23           htmlwidgets_1.6.4      farver_2.1.2          \n[100] htmltools_0.5.8.1      lifecycle_1.0.4        httr_1.4.7            \n[103] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 45.21\nuser 28.74\nsys 1.92\npercent 67%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TCRClusterStats",desc:"Statistics of TCR clusters, generated by `TCRClustering`.",npages:1,report_toc:!0,order:13,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] grid      stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] ComplexHeatmap_2.18.0 ggprism_1.0.5         immunarch_0.9.1      \n [4] patchwork_1.3.0       data.table_1.16.4     dtplyr_1.3.1         \n [7] dplyr_1.1.4           tidyr_1.3.1           tibble_3.2.1         \n[10] ggplot2_3.5.1         jsonlite_1.8.9        rlang_1.1.5          \n[13] logger_0.4.0         \n\nloaded via a namespace (and not attached):\n  [1] gridExtra_2.3       phangorn_2.12.1     readxl_1.4.3       \n  [4] magrittr_2.0.3      clue_0.3-66         GetoptLong_1.0.5   \n  [7] matrixStats_1.1.0   compiler_4.3.3      flexmix_2.3-19     \n [10] png_0.1-8           vctrs_0.6.5         reshape2_1.4.4     \n [13] ggalluvial_0.12.5   quadprog_1.5-8      stringr_1.5.1      \n [16] crayon_1.5.3        pkgconfig_2.0.3     shape_1.4.6.1      \n [19] fastmap_1.2.0       backports_1.5.0     labeling_0.4.3     \n [22] ggraph_2.2.1        promises_1.3.2      tzdb_0.4.0         \n [25] UpSetR_1.4.0        purrr_1.0.2         modeltools_0.2-23  \n [28] ggseqlogo_0.2       cachem_1.1.0        later_1.4.1        \n [31] uuid_1.2-1          fpc_2.2-13          tweenr_2.0.3       \n [34] broom_1.0.7         parallel_4.3.3      prabclus_2.3-4     \n [37] cluster_2.1.8       R6_2.6.0            stringi_1.8.4      \n [40] RColorBrewer_1.1-3  rlist_0.4.6.2       car_3.1-3          \n [43] diptest_0.77-1      cellranger_1.1.0    Rcpp_1.0.14        \n [46] iterators_1.0.14    IRanges_2.36.0      readr_2.1.5        \n [49] httpuv_1.6.15       Matrix_1.6-5        nnet_7.3-20        \n [52] igraph_2.0.3        tidyselect_1.2.1    stringdist_0.9.15  \n [55] abind_1.4-5         viridis_0.6.5       doParallel_1.0.17  \n [58] codetools_0.2-20    lattice_0.22-6      plyr_1.8.9         \n [61] shiny_1.10.0        withr_3.0.2         polyclip_1.10-7    \n [64] circlize_0.4.16     mclust_6.1.1        kernlab_0.9-33     \n [67] pillar_1.10.1       ggpubr_0.6.0        carData_3.0-5      \n [70] foreach_1.5.2       stats4_4.3.3        generics_0.1.3     \n [73] S4Vectors_0.40.2    hms_1.1.3           munsell_0.5.1      \n [76] scales_1.3.0        xtable_1.8-4        class_7.3-23       \n [79] glue_1.8.0          pheatmap_1.0.12     tools_4.3.3        \n [82] robustbase_0.99-4-1 ggsignif_0.6.4      graphlayouts_1.2.2 \n [85] Cairo_1.6-2         fastmatch_1.1-6     tidygraph_1.3.0    \n [88] ape_5.8-1           colorspace_2.1-1    nlme_3.1-167       \n [91] ggforce_0.4.2       Formula_1.2-5       cli_3.6.4          \n [94] viridisLite_0.4.2   gtable_0.3.6        DEoptimR_1.1-3-1   \n [97] rstatix_0.7.2       digest_0.6.37       BiocGenerics_0.48.1\n[100] ggrepel_0.9.6       rjson_0.2.23        farver_2.1.2       \n[103] memoise_2.0.1       htmltools_0.5.8.1   lifecycle_1.0.4    \n[106] factoextra_1.0.7    mime_0.12           GlobalOptions_0.1.2\n[109] shinythemes_1.2.0   MASS_7.3-60.0.1    \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 7.17\nuser 15.93\nsys 0.32\npercent 226%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3287.293\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TESSA",desc:"Tessa is a Bayesian model to integrate T cell receptor (TCR) sequence profiling with transcriptomes of T cells.",npages:1,report_toc:!0,order:14,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] igraph_2.0.3         Rtsne_0.17           LaplacesDemon_16.1.6\n [4] MASS_7.3-60.0.1      ggprism_1.0.5        Seurat_5.1.0        \n [7] SeuratObject_5.0.2   sp_2.2-0             tibble_3.2.1        \n[10] glue_1.8.0           immunarch_0.9.1      patchwork_1.3.0     \n[13] data.table_1.16.4    dtplyr_1.3.1         ggplot2_3.5.1       \n[16] tidyr_1.3.1          dplyr_1.1.4          jsonlite_1.8.9      \n[19] rlang_1.1.5          logger_0.4.0        \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22       shinythemes_1.2.0      splines_4.3.3         \n  [4] later_1.4.1            cellranger_1.1.0       polyclip_1.10-7       \n  [7] fastDummies_1.7.5      factoextra_1.0.7       lifecycle_1.0.4       \n [10] rstatix_0.7.2          doParallel_1.0.17      globals_0.16.3        \n [13] lattice_0.22-6         prabclus_2.3-4         backports_1.5.0       \n [16] magrittr_2.0.3         plotly_4.10.4          rlist_0.4.6.2         \n [19] httpuv_1.6.15          sctransform_0.4.1      spam_2.11-1           \n [22] flexmix_2.3-19         spatstat.sparse_3.1-0  reticulate_1.40.0     \n [25] cowplot_1.1.3          pbapply_1.7-2          RColorBrewer_1.1-3    \n [28] abind_1.4-5            quadprog_1.5-8         purrr_1.0.2           \n [31] ggraph_2.2.1           nnet_7.3-20            tweenr_2.0.3          \n [34] circlize_0.4.16        ggrepel_0.9.6          irlba_2.3.5.1         \n [37] listenv_0.9.1          spatstat.utils_3.1-2   pheatmap_1.0.12       \n [40] goftest_1.2-3          RSpectra_0.16-2        spatstat.random_3.3-2 \n [43] fitdistrplus_1.2-2     parallelly_1.42.0      leiden_0.4.3.1        \n [46] codetools_0.2-20       ggforce_0.4.2          tidyselect_1.2.1      \n [49] shape_1.4.6.1          ggseqlogo_0.2          farver_2.1.2          \n [52] viridis_0.6.5          matrixStats_1.1.0      stats4_4.3.3          \n [55] spatstat.explore_3.3-4 tidygraph_1.3.0        progressr_0.15.1      \n [58] Formula_1.2-5          ggridges_0.5.6         ggalluvial_0.12.5     \n [61] survival_3.8-3         iterators_1.0.14       foreach_1.5.2         \n [64] tools_4.3.3            stringdist_0.9.15      ica_1.0-3             \n [67] Rcpp_1.0.14            gridExtra_2.3          withr_3.0.2           \n [70] fastmap_1.2.0          digest_0.6.37          R6_2.6.0              \n [73] mime_0.12              colorspace_2.1-1       scattermore_1.2       \n [76] tensor_1.5             spatstat.data_3.1-4    diptest_0.77-1        \n [79] UpSetR_1.4.0           generics_0.1.3         robustbase_0.99-4-1   \n [82] class_7.3-23           graphlayouts_1.2.2     httr_1.4.7            \n [85] htmlwidgets_1.6.4      uwot_0.2.2             pkgconfig_2.0.3       \n [88] gtable_0.3.6           modeltools_0.2-23      lmtest_0.9-40         \n [91] htmltools_0.5.8.1      carData_3.0-5          dotCall64_1.2         \n [94] scales_1.3.0           png_0.1-8              spatstat.univar_3.1-1 \n [97] tzdb_0.4.0             reshape2_1.4.4         uuid_1.2-1            \n[100] nlme_3.1-167           cachem_1.1.0           zoo_1.8-12            \n[103] GlobalOptions_0.1.2    stringr_1.5.1          KernSmooth_2.23-26    \n[106] parallel_4.3.3         miniUI_0.1.1.1         pillar_1.10.1         \n[109] grid_4.3.3             vctrs_0.6.5            RANN_2.6.2            \n[112] promises_1.3.2         ggpubr_0.6.0           car_3.1-3             \n[115] xtable_1.8-4           cluster_2.1.8          readr_2.1.5           \n[118] cli_3.6.4              compiler_4.3.3         future.apply_1.11.3   \n[121] ggsignif_0.6.4         labeling_0.4.3         mclust_6.1.1          \n[124] plyr_1.8.9             stringi_1.8.4          deldir_2.0-4          \n[127] viridisLite_0.4.2      munsell_0.5.1          lazyeval_0.2.2        \n[130] spatstat.geom_3.3-5    Matrix_1.6-5           RcppHNSW_0.6.0        \n[133] hms_1.1.3              future_1.34.0          fpc_2.2-13            \n[136] shiny_1.10.0           kernlab_0.9-33         ROCR_1.0-11           \n[139] broom_1.0.7            memoise_2.0.1          phangorn_2.12.1       \n[142] fastmatch_1.1-6        DEoptimR_1.1-3-1       readxl_1.4.3          \n[145] ape_5.8-1             \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 1679.73\nuser 2269.05\nsys 21.01\npercent 136%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"SeuratClusterStats",desc:"Statistics of the clustering.",npages:1,report_toc:!0,order:17,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] grid      stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] clustree_0.5.1        ggraph_2.2.1          ComplexHeatmap_2.18.0\n [4] circlize_0.4.16       tidyseurat_0.8.0      ttservice_0.4.1      \n [7] ggrepel_0.9.6         ggprism_1.0.5         tibble_3.2.1         \n[10] Seurat_5.1.0          SeuratObject_5.0.2    sp_2.2-0             \n[13] ggplot2_3.5.1         tidyr_1.3.1           dplyr_1.1.4          \n[16] tidyselect_1.2.1      jsonlite_1.8.9        rlang_1.1.5          \n[19] logger_0.4.0         \n\nloaded via a namespace (and not attached):\n  [1] RColorBrewer_1.1-3     shape_1.4.6.1          magrittr_2.0.3        \n  [4] ggbeeswarm_0.7.2       spatstat.utils_3.1-2   farver_2.1.2          \n  [7] GlobalOptions_0.1.2    vctrs_0.6.5            ROCR_1.0-11           \n [10] memoise_2.0.1          spatstat.explore_3.3-4 htmltools_0.5.8.1     \n [13] sctransform_0.4.1      parallelly_1.42.0      KernSmooth_2.23-26    \n [16] htmlwidgets_1.6.4      ica_1.0-3              plyr_1.8.9            \n [19] cachem_1.1.0           plotly_4.10.4          zoo_1.8-12            \n [22] igraph_2.0.3           mime_0.12              lifecycle_1.0.4       \n [25] iterators_1.0.14       pkgconfig_2.0.3        Matrix_1.6-5          \n [28] R6_2.6.0               fastmap_1.2.0          clue_0.3-66           \n [31] fitdistrplus_1.2-2     future_1.34.0          shiny_1.10.0          \n [34] digest_0.6.37          colorspace_2.1-1       patchwork_1.3.0       \n [37] S4Vectors_0.40.2       tensor_1.5             RSpectra_0.16-2       \n [40] irlba_2.3.5.1          labeling_0.4.3         progressr_0.15.1      \n [43] fansi_1.0.6            spatstat.sparse_3.1-0  httr_1.4.7            \n [46] polyclip_1.10-7        abind_1.4-5            compiler_4.3.3        \n [49] withr_3.0.2            doParallel_1.0.17      viridis_0.6.5         \n [52] fastDummies_1.7.5      ggforce_0.4.2          MASS_7.3-60.0.1       \n [55] rjson_0.2.23           tools_4.3.3            vipor_0.4.7           \n [58] lmtest_0.9-40          beeswarm_0.4.0         httpuv_1.6.15         \n [61] future.apply_1.11.3    goftest_1.2-3          glue_1.8.0            \n [64] nlme_3.1-167           promises_1.3.2         Rtsne_0.17            \n [67] cluster_2.1.8          reshape2_1.4.4         generics_0.1.3        \n [70] gtable_0.3.6           spatstat.data_3.1-4    data.table_1.16.4     \n [73] tidygraph_1.3.0        BiocGenerics_0.48.1    spatstat.geom_3.3-5   \n [76] RcppAnnoy_0.0.22       RANN_2.6.2             foreach_1.5.2         \n [79] pillar_1.10.1          stringr_1.5.1          spam_2.11-1           \n [82] RcppHNSW_0.6.0         later_1.4.1            splines_4.3.3         \n [85] tweenr_2.0.3           lattice_0.22-6         survival_3.8-3        \n [88] deldir_2.0-4           miniUI_0.1.1.1         pbapply_1.7-2         \n [91] gridExtra_2.3          IRanges_2.36.0         scattermore_1.2       \n [94] stats4_4.3.3           graphlayouts_1.2.2     matrixStats_1.1.0     \n [97] stringi_1.8.4          lazyeval_0.2.2         codetools_0.2-20      \n[100] cli_3.6.4              uwot_0.2.2             xtable_1.8-4          \n[103] reticulate_1.40.0      munsell_0.5.1          Rcpp_1.0.14           \n[106] globals_0.16.3         spatstat.random_3.3-2  png_0.1-8             \n[109] ggrastr_1.0.2          spatstat.univar_3.1-1  parallel_4.3.3        \n[112] ellipsis_0.3.2         dotCall64_1.2          listenv_0.9.1         \n[115] viridisLite_0.4.2      scales_1.3.0           ggridges_0.5.6        \n[118] leiden_0.4.3.1         purrr_1.0.2            crayon_1.5.3          \n[121] GetoptLong_1.0.5       cowplot_1.1.3         \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 18.76\nuser 27.07\nsys 0.71\npercent 148%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"Immunarch",desc:"Exploration of Single-cell and Bulk T-cell/Antibody Immune Repertoires",npages:3,report_toc:!0,order:18,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] circlize_0.4.16   tibble_3.2.1      glue_1.8.0        ggprism_1.0.5    \n [5] immunarch_0.9.1   patchwork_1.3.0   data.table_1.16.4 dtplyr_1.3.1     \n [9] ggplot2_3.5.1     tidyr_1.3.1       dplyr_1.1.4       jsonlite_1.8.9   \n[13] rlang_1.1.5       logger_0.4.0     \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22       shinythemes_1.2.0      splines_4.3.3         \n  [4] later_1.4.1            cellranger_1.1.0       polyclip_1.10-7       \n  [7] fastDummies_1.7.5      factoextra_1.0.7       lifecycle_1.0.4       \n [10] rstatix_0.7.2          doParallel_1.0.17      globals_0.16.3        \n [13] lattice_0.22-6         prabclus_2.3-4         MASS_7.3-60.0.1       \n [16] backports_1.5.0        magrittr_2.0.3         plotly_4.10.4         \n [19] rlist_0.4.6.2          httpuv_1.6.15          Seurat_5.1.0          \n [22] sctransform_0.4.1      spam_2.11-1            flexmix_2.3-19        \n [25] spatstat.sparse_3.1-0  sp_2.2-0               reticulate_1.40.0     \n [28] cowplot_1.1.3          pbapply_1.7-2          RColorBrewer_1.1-3    \n [31] abind_1.4-5            Rtsne_0.17             quadprog_1.5-8        \n [34] purrr_1.0.2            ggraph_2.2.1           nnet_7.3-20           \n [37] tweenr_2.0.3           ggrepel_0.9.6          irlba_2.3.5.1         \n [40] spatstat.utils_3.1-2   listenv_0.9.1          pheatmap_1.0.12       \n [43] goftest_1.2-3          RSpectra_0.16-2        spatstat.random_3.3-2 \n [46] fitdistrplus_1.2-2     parallelly_1.42.0      leiden_0.4.3.1        \n [49] codetools_0.2-20       ggforce_0.4.2          tidyselect_1.2.1      \n [52] shape_1.4.6.1          ggseqlogo_0.2          farver_2.1.2          \n [55] viridis_0.6.5          matrixStats_1.1.0      stats4_4.3.3          \n [58] spatstat.explore_3.3-4 tidygraph_1.3.0        progressr_0.15.1      \n [61] Formula_1.2-5          ggridges_0.5.6         ggalluvial_0.12.5     \n [64] survival_3.8-3         iterators_1.0.14       foreach_1.5.2         \n [67] tools_4.3.3            stringdist_0.9.15      ica_1.0-3             \n [70] Rcpp_1.0.14            gridExtra_2.3          withr_3.0.2           \n [73] fastmap_1.2.0          digest_0.6.37          R6_2.6.0              \n [76] mime_0.12              colorspace_2.1-1       scattermore_1.2       \n [79] tensor_1.5             spatstat.data_3.1-4    diptest_0.77-1        \n [82] UpSetR_1.4.0           generics_0.1.3         robustbase_0.99-4-1   \n [85] class_7.3-23           graphlayouts_1.2.2     httr_1.4.7            \n [88] htmlwidgets_1.6.4      uwot_0.2.2             pkgconfig_2.0.3       \n [91] gtable_0.3.6           modeltools_0.2-23      lmtest_0.9-40         \n [94] htmltools_0.5.8.1      carData_3.0-5          dotCall64_1.2         \n [97] SeuratObject_5.0.2     scales_1.3.0           png_0.1-8             \n[100] spatstat.univar_3.1-1  tzdb_0.4.0             reshape2_1.4.4        \n[103] uuid_1.2-1             nlme_3.1-167           cachem_1.1.0          \n[106] zoo_1.8-12             GlobalOptions_0.1.2    stringr_1.5.1         \n[109] KernSmooth_2.23-26     parallel_4.3.3         miniUI_0.1.1.1        \n[112] pillar_1.10.1          grid_4.3.3             vctrs_0.6.5           \n[115] RANN_2.6.2             promises_1.3.2         ggpubr_0.6.0          \n[118] car_3.1-3              xtable_1.8-4           cluster_2.1.8         \n[121] readr_2.1.5            cli_3.6.4              compiler_4.3.3        \n[124] future.apply_1.11.3    ggsignif_0.6.4         labeling_0.4.3        \n[127] mclust_6.1.1           plyr_1.8.9             stringi_1.8.4         \n[130] deldir_2.0-4           viridisLite_0.4.2      assertthat_0.2.1      \n[133] munsell_0.5.1          lazyeval_0.2.2         spatstat.geom_3.3-5   \n[136] Matrix_1.6-5           RcppHNSW_0.6.0         hms_1.1.3             \n[139] future_1.34.0          fpc_2.2-13             shiny_1.10.0          \n[142] kernlab_0.9-33         ROCR_1.0-11            igraph_2.0.3          \n[145] broom_1.0.7            memoise_2.0.1          phangorn_2.12.1       \n[148] fastmatch_1.1-6        DEoptimR_1.1-3-1       readxl_1.4.3          \n[151] ape_5.8-1             \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 102.43\nuser 256.71\nsys 5.18\npercent 255%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"RadarPlots",desc:"Radar plots for cell proportion in different clusters.",npages:1,report_toc:!0,order:19,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ggprism_1.0.5      ggradar_0.2        ggplot2_3.5.1      tibble_3.2.1      \n [5] tidyr_1.3.1        dplyr_1.1.4        Seurat_5.1.0       SeuratObject_5.0.2\n [9] sp_2.2-0           jsonlite_1.8.9     rlang_1.1.5        logger_0.4.0      \n\nloaded via a namespace (and not attached):\n [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n[10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n[13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n[16] labeling_0.4.3         promises_1.3.2         purrr_1.0.2           \n[19] goftest_1.2-3          later_1.4.1            spatstat.utils_3.1-2  \n[22] irlba_2.3.5.1          parallel_4.3.3         cluster_2.1.8         \n[25] R6_2.6.0               ica_1.0-3              stringi_1.8.4         \n[28] RColorBrewer_1.1-3     spatstat.data_3.1-4    reticulate_1.40.0     \n[31] parallelly_1.42.0      spatstat.univar_3.1-1  lmtest_0.9-40         \n[34] scattermore_1.2        Rcpp_1.0.14            tensor_1.5            \n[37] future.apply_1.11.3    zoo_1.8-12             sctransform_0.4.1     \n[40] httpuv_1.6.15          Matrix_1.6-5           splines_4.3.3         \n[43] igraph_2.0.3           tidyselect_1.2.1       abind_1.4-5           \n[46] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n[49] spatstat.explore_3.3-4 listenv_0.9.1          lattice_0.22-6        \n[52] plyr_1.8.9             withr_3.0.2            shiny_1.10.0          \n[55] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n[58] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n[61] fitdistrplus_1.2-2     pillar_1.10.1          KernSmooth_2.23-26    \n[64] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n[67] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n[70] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n[73] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n[76] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n[79] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n[82] nlme_3.1-167           patchwork_1.3.0        cli_3.6.4             \n[85] spatstat.sparse_3.1-0  spam_2.11-1            viridisLite_0.4.2     \n[88] uwot_0.2.2             gtable_0.3.6           digest_0.6.37         \n[91] progressr_0.15.1       ggrepel_0.9.6          htmlwidgets_1.6.4     \n[94] farver_2.1.2           htmltools_0.5.8.1      lifecycle_1.0.4       \n[97] httr_1.4.7             mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 8.07\nuser 16.74\nsys 0.38\npercent 212%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"CloneResidency",desc:"Identification of clone residency",npages:1,report_toc:!0,order:20,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ComplexUpset_1.3.3  ggVennDiagram_1.2.3 ggprism_1.0.5      \n [4] ggplot2_3.5.1       ggnewscale_0.5.0    glue_1.8.0         \n [7] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[10] jsonlite_1.8.9      rlang_1.1.5         logger_0.4.0       \n\nloaded via a namespace (and not attached):\n  [1] DBI_1.2.3              deldir_2.0-4           pbapply_1.7-2         \n  [4] gridExtra_2.3          magrittr_2.0.3         RcppAnnoy_0.0.22      \n  [7] e1071_1.7-16           spatstat.geom_3.3-5    matrixStats_1.1.0     \n [10] ggridges_0.5.6         compiler_4.3.3         png_0.1-8             \n [13] vctrs_0.6.5            reshape2_1.4.4         stringr_1.5.1         \n [16] pkgconfig_2.0.3        fastmap_1.2.0          labeling_0.4.3        \n [19] promises_1.3.2         purrr_1.0.2            RVenn_1.1.0           \n [22] goftest_1.2-3          later_1.4.1            spatstat.utils_3.1-2  \n [25] irlba_2.3.5.1          parallel_4.3.3         cluster_2.1.8         \n [28] R6_2.6.0               ica_1.0-3              stringi_1.8.4         \n [31] RColorBrewer_1.1-3     spatstat.data_3.1-4    reticulate_1.40.0     \n [34] parallelly_1.42.0      spatstat.univar_3.1-1  lmtest_0.9-40         \n [37] scattermore_1.2        Rcpp_1.0.14            tensor_1.5            \n [40] future.apply_1.11.3    zoo_1.8-12             sctransform_0.4.1     \n [43] httpuv_1.6.15          Matrix_1.6-5           splines_4.3.3         \n [46] igraph_2.0.3           tidyselect_1.2.1       abind_1.4-5           \n [49] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.3-4 listenv_0.9.1          lattice_0.22-6        \n [55] plyr_1.8.9             shiny_1.10.0           withr_3.0.2           \n [58] ROCR_1.0-11            Rtsne_0.17             sf_1.0-19             \n [61] future_1.34.0          fastDummies_1.7.5      survival_3.8-3        \n [64] units_0.8-5            proxy_0.4-27           polyclip_1.10-7       \n [67] fitdistrplus_1.2-2     pillar_1.10.1          Seurat_5.1.0          \n [70] KernSmooth_2.23-26     plotly_4.10.4          generics_0.1.3        \n [73] RcppHNSW_0.6.0         sp_2.2-0               munsell_0.5.1         \n [76] scales_1.3.0           globals_0.16.3         xtable_1.8-4          \n [79] class_7.3-23           lazyeval_0.2.2         tools_4.3.3           \n [82] data.table_1.16.4      RSpectra_0.16-2        RANN_2.6.2            \n [85] leiden_0.4.3.1         dotCall64_1.2          cowplot_1.1.3         \n [88] grid_4.3.3             colorspace_2.1-1       nlme_3.1-167          \n [91] patchwork_1.3.0        cli_3.6.4              spatstat.sparse_3.1-0 \n [94] spam_2.11-1            viridisLite_0.4.2      uwot_0.2.2            \n [97] gtable_0.3.6           digest_0.6.37          classInt_0.4-11       \n[100] progressr_0.15.1       ggrepel_0.9.6          htmlwidgets_1.6.4     \n[103] SeuratObject_5.0.2     farver_2.1.2           htmltools_0.5.8.1     \n[106] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[109] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 19.89\nuser 28.24\nsys 0.66\npercent 145%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ScFGSEA",desc:"Gene set enrichment analysis for cells in different groups using `fgsea`",npages:2,report_toc:!0,order:21,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] fgsea_1.28.0       data.table_1.16.4  matrixStats_1.1.0  tidyseurat_0.8.0  \n [5] ttservice_0.4.1    Seurat_5.1.0       SeuratObject_5.0.2 sp_2.2-0          \n [9] tidyr_1.3.1        tidyselect_1.2.1   tibble_3.2.1       dplyr_1.1.4       \n[13] ggplot2_3.5.1      jsonlite_1.8.9     rlang_1.1.5        logger_0.4.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       ggridges_0.5.6        \n  [7] compiler_4.3.3         spatstat.geom_3.3-5    png_0.1-8             \n [10] vctrs_0.6.5            reshape2_1.4.4         stringr_1.5.1         \n [13] pkgconfig_2.0.3        fastmap_1.2.0          ellipsis_0.3.2        \n [16] labeling_0.4.3         promises_1.3.2         purrr_1.0.2           \n [19] goftest_1.2-3          later_1.4.1            BiocParallel_1.36.0   \n [22] spatstat.utils_3.1-2   irlba_2.3.5.1          parallel_4.3.3        \n [25] cluster_2.1.8          R6_2.6.0               ica_1.0-3             \n [28] stringi_1.8.4          RColorBrewer_1.1-3     spatstat.data_3.1-4   \n [31] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [37] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [43] splines_4.3.3          igraph_2.0.3           abind_1.4-5           \n [46] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n [49] spatstat.explore_3.3-4 listenv_0.9.1          lattice_0.22-6        \n [52] plyr_1.8.9             shiny_1.10.0           withr_3.0.2           \n [55] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n [58] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n [61] fitdistrplus_1.2-2     pillar_1.10.1          KernSmooth_2.23-26    \n [64] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n [67] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [70] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [73] tools_4.3.3            RSpectra_0.16-2        RANN_2.6.2            \n [76] leiden_0.4.3.1         dotCall64_1.2          fastmatch_1.1-6       \n [79] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n [82] nlme_3.1-167           patchwork_1.3.0        cli_3.6.4             \n [85] spatstat.sparse_3.1-0  fansi_1.0.6            spam_2.11-1           \n [88] viridisLite_0.4.2      uwot_0.2.2             gtable_0.3.6          \n [91] digest_0.6.37          progressr_0.15.1       ggrepel_0.9.6         \n [94] htmlwidgets_1.6.4      farver_2.1.2           htmltools_0.5.8.1     \n [97] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[100] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 128.45\nuser 135.35\nsys 1.52\npercent 106%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"MarkersFinder",desc:"Find markers between different groups of cells",npages:1,report_toc:!0,order:22,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] UpSetR_1.4.0        ggVennDiagram_1.2.3 tidyseurat_0.8.0   \n [4] ttservice_0.4.1     future_1.34.0       ggrepel_0.9.6      \n [7] ggprism_1.0.5       ggplot2_3.5.1       enrichR_3.4        \n[10] Seurat_5.1.0        SeuratObject_5.0.2  sp_2.2-0           \n[13] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[16] tidyselect_1.2.1    digest_0.6.37       jsonlite_1.8.9     \n[19] rlang_1.1.5         logger_0.4.0       \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] ellipsis_0.3.2         labeling_0.4.3         promises_1.3.2        \n [19] purrr_1.0.2            WriteXLS_6.7.0         RVenn_1.1.0           \n [22] goftest_1.2-3          later_1.4.1            spatstat.utils_3.1-2  \n [25] irlba_2.3.5.1          parallel_4.3.3         cluster_2.1.8         \n [28] R6_2.6.0               ica_1.0-3              stringi_1.8.4         \n [31] RColorBrewer_1.1-3     spatstat.data_3.1-4    limma_3.58.1          \n [34] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [37] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [40] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [43] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [46] splines_4.3.3          igraph_2.0.3           abind_1.4-5           \n [49] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.3-4 curl_6.0.1             listenv_0.9.1         \n [55] lattice_0.22-6         plyr_1.8.9             withr_3.0.2           \n [58] shiny_1.10.0           ROCR_1.0-11            Rtsne_0.17            \n [61] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n [64] fitdistrplus_1.2-2     pillar_1.10.1          KernSmooth_2.23-26    \n [67] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n [82] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n [85] nlme_3.1-167           patchwork_1.3.0        presto_1.0.0          \n [88] cli_3.6.4              spatstat.sparse_3.1-0  fansi_1.0.6           \n [91] spam_2.11-1            viridisLite_0.4.2      uwot_0.2.2            \n [94] gtable_0.3.6           progressr_0.15.1       rjson_0.2.23          \n [97] htmlwidgets_1.6.4      farver_2.1.2           htmltools_0.5.8.1     \n[100] lifecycle_1.0.4        httr_1.4.7             statmod_1.5.0         \n[103] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 19.16\nuser 23.15\nsys 0.56\npercent 123%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"MetaMarkers",desc:"Find markers between three or more groups of cells, using one-way ANOVA or Kruskal-Wallis test.",npages:1,report_toc:!0,order:23},{name:"CellsDistribution",desc:"Distribution of cells (i.e. in a TCR clone) from different groups for each cluster",npages:1,report_toc:!0,order:24},{name:"CDR3AAPhyschem",desc:"CDR3 AA physicochemical feature analysis",npages:1,report_toc:!0,order:25},{name:"ScrnaMetabolicLandscape",desc:"Metabolic landscape analysis for scRNA-seq data",order:28,procs:[{name:"MetabolicPathwayActivity",desc:"This process calculates the pathway activities in different groups and subsets.",npages:1,report_toc:!0,order:28},{name:"MetabolicFeatures",desc:"This process performs enrichment analysis for the metabolic pathways for each group in each subset.",npages:1,report_toc:!0,order:29},{name:"MetabolicFeaturesIntraSubset",desc:"Intra-subset metabolic features - Enrichment analysis in details",npages:1,report_toc:!0,order:30},{name:"MetabolicPathwayHeterogeneity",desc:"Calculate Metabolic Pathway heterogeneity.",npages:1,report_toc:!0,order:31}]}]};function el(e){let a,t;return a=new tt({props:{dir:"up"}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function al(e){let a,t;return a=new tt({props:{dir:"down"}}),{c(){g(a.$$.fragment)},m(e,n){P(a,e,n),t=!0},i(e){t||(i(a.$$.fragment,e),t=!0)},o(e){c(a.$$.fragment,e),t=!1},d(e){A(a,e)}}}function tl(e){let a,t,n,l,_,s=e[1]>0&&el();t=new Qn({});let u=e[1]<e[3].npages-1&&al();return{c(){s&&s.c(),a=f(),g(t.$$.fragment),n=f(),u&&u.c(),l=m()},m(e,i){s&&s.m(e,i),d(e,a,i),P(t,e,i),d(e,n,i),u&&u.m(e,i),d(e,l,i),_=!0},p(e,t){e[1]>0?s?2&t&&i(s,1):(s=el(),s.c(),i(s,1),s.m(a.parentNode,a)):s&&(r(),c(s,1,1,(()=>{s=null})),o()),e[1]<e[3].npages-1?u?10&t&&i(u,1):(u=al(),u.c(),i(u,1),u.m(l.parentNode,l)):u&&(r(),c(u,1,1,(()=>{u=null})),o())},i(e){_||(i(s),i(t.$$.fragment,e),i(u),_=!0)},o(e){c(s),c(t.$$.fragment,e),c(u),_=!1},d(e){e&&(p(a),p(n),p(l)),s&&s.d(e),A(t,e),u&&u.d(e)}}}function nl(e){let a,t,n;function l(a){e[4](a)}let _={logo:e[0],logotext:e[3].desc,versions:Jn.versions,entries:Jn.entries,pipeline_name:Jn.pipeline.name,report_toc:e[3].report_toc,runinfo:e[3].runinfo,toc:Zn,page:e[1],$$slots:{default:[tl]},$$scope:{ctx:e}};return void 0!==e[2]&&(_.isSideNavOpen=e[2]),a=new Za({props:_}),b.push((()=>C(a,"isSideNavOpen",l))),{c(){g(a.$$.fragment)},m(e,t){P(a,e,t),n=!0},p(e,[n]){const l={};1&n&&(l.logo=e[0]),8&n&&(l.logotext=e[3].desc),8&n&&(l.report_toc=e[3].report_toc),8&n&&(l.runinfo=e[3].runinfo),2&n&&(l.page=e[1]),42&n&&(l.$$scope={dirty:n,ctx:e}),!t&&4&n&&(t=!0,l.isSideNavOpen=e[2],O((()=>t=!1))),a.$set(l)},i(e){n||(i(a.$$.fragment,e),n=!0)},o(e){c(a.$$.fragment,e),n=!1},d(e){A(a,e)}}}function ll(e,a,t){let n,{name:l}=a,{page:_=0}=a,s=!1;for(const e of Jn.entries)if(e.procs){for(const a of e.procs)if(a.name==l){n=a;break}}else if(e.name==l){n=e;break}return v((()=>{const e=new URL(window.location.href);let a;e.hash.length>1&&(a=document.getElementById(`${e.hash.slice(1)}`)),a&&a.scrollIntoView({block:"start",behavior:"smooth"})})),e.$$set=e=>{"name"in e&&t(0,l=e.name),"page"in e&&t(1,_=e.page)},[l,_,s,n,function(e){s=e,t(2,s)}]}e("default",new class extends a{constructor(e){super(),t(this,e,ll,nl,n,{name:0,page:1})}}({target:document.body,props:{name:"MarkersFinder",page:0}}))}}}));
