System.register(["./vendor/svelte-C_apNiMW.js","./vendor/carbon-components-svelte-CB8VNYQX.js","./vendor/carbon-icons-svelte-DkM4cpjm.js"],(function(e){"use strict";var t,n,a,s,r,i,l,o,p,c,d,_,m,u,g,f,b,C,$,h,v,y,S,x,A,M,P,w,T,L,E,N,U,R,D,I,B,F,G,H,V,z,O,k,q,K,j,X,Y,W,Q,J,Z,ee,te,ne,ae,se,re,ie,le,oe,pe,ce,de,_e,me,ue,ge,fe,be,Ce,$e,he,ve,ye,Se,xe,Ae,Me,Pe,we,Te,Le,Ee,Ne,Ue,Re,De,Ie,Be,Fe,Ge,He,Ve,ze,Oe,ke,qe,Ke,je,Xe,Ye,We,Qe,Je,Ze;return{setters:[function(e){t=e.S,n=e.i,a=e.s,s=e.a1,r=e.l,i=e.O,l=e.a,o=e.b,p=e.g,c=e.c,d=e.t,_=e.d,m=e.a5,u=e.B,g=e.e,f=e.G,b=e.H,C=e.I,$=e.y,h=e.K,v=e.T,y=e.U,S=e.L,x=e.D,A=e.M,M=e.z,P=e.Q,w=e.k,T=e.u,L=e.n,E=e.o,N=e.h,U=e.p,R=e.a0,D=e.f,I=e.j,B=e.A,F=e.m,G=e.q,H=e.aa,V=e.ab,z=e.x,O=e.W,k=e.ac,q=e.R,K=e.$,j=e.v,X=e.Y,Y=e.Z,W=e.F,Q=e.ad,J=e.w,Z=e.J,ee=e.r},function(e){te=e.S,ne=e.a,ae=e.b,se=e.H,re=e.c,ie=e.T,le=e.d,oe=e.e,pe=e.f,ce=e.g,de=e.h,_e=e.M,me=e.I,ue=e.i,ge=e.j,fe=e.k,be=e.C,Ce=e.l,$e=e.m,he=e.G,ve=e.R,ye=e.n,Se=e.B,xe=e.o,Ae=e.A,Me=e.p,Pe=e.D,we=e.P,Te=e.q,Le=e.r,Ee=e.s,Ne=e.t,Ue=e.u},function(e){Re=e.C,De=e.a,Ie=e.D,Be=e.A,Fe=e.S,Ge=e.I,He=e.b,Ve=e.c,ze=e.V,Oe=e.d,ke=e.T,qe=e.G,Ke=e.J,je=e.P,Xe=e.e,Ye=e.f,We=e.O,Qe=e.g,Je=e.h,Ze=e.i}],execute:function(){function et(e,t,n){const a=e.slice();return a[4]=t[n],a}function tt(e,t,n){const a=e.slice();return a[7]=t[n],a}function nt(e){let t,n;return t=new te({props:{href:`${e[1](e[4].page)}#${e[4].slug}`,class:e[2](e[4].page),title:e[4].text,text:e[4].text,expanded:!0,$$slots:{default:[rt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.href=`${e[1](e[4].page)}#${e[4].slug}`),1&n&&(a.class=e[2](e[4].page)),1&n&&(a.title=e[4].text),1&n&&(a.text=e[4].text),1025&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function at(e){let t,n;return t=new ne({props:{href:`${e[1](e[4].page)}#${e[4].slug}`,class:e[2](e[4].page),title:e[4].text,text:e[4].text}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.href=`${e[1](e[4].page)}#${e[4].slug}`),1&n&&(a.class=e[2](e[4].page)),1&n&&(a.title=e[4].text),1&n&&(a.text=e[4].text),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function st(e){let t,n;return t=new ae({props:{href:`${e[1](e[7].page)}#${e[7].slug}`,class:e[2](e[7].page),title:e[7].text,text:e[7].text}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.href=`${e[1](e[7].page)}#${e[7].slug}`),1&n&&(a.class=e[2](e[7].page)),1&n&&(a.title=e[7].text),1&n&&(a.text=e[7].text),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function rt(e){let t,n,a=s(e[4].children),r=[];for(let t=0;t<a.length;t+=1)r[t]=st(tt(e,a,t));const i=e=>d(r[e],1,1,(()=>{r[e]=null}));return{c(){for(let e=0;e<r.length;e+=1)r[e].c();t=$()},m(e,a){for(let t=0;t<r.length;t+=1)r[t]&&r[t].m(e,a);l(e,t,a),n=!0},p(e,n){if(7&n){let l;for(a=s(e[4].children),l=0;l<a.length;l+=1){const s=tt(e,a,l);r[l]?(r[l].p(s,n),o(r[l],1)):(r[l]=st(s),r[l].c(),o(r[l],1),r[l].m(t.parentNode,t))}for(p(),l=a.length;l<r.length;l+=1)i(l);c()}},i(e){if(!n){for(let e=0;e<a.length;e+=1)o(r[e]);n=!0}},o(e){r=r.filter(Boolean);for(let e=0;e<r.length;e+=1)d(r[e]);n=!1},d(e){e&&_(t),m(r,e)}}}function it(e){let t,n,a,s;const r=[at,nt],i=[];function m(e,t){return 0==e[4].children.length?0:1}return t=m(e),n=i[t]=r[t](e),{c(){n.c(),a=g()},m(e,n){i[t].m(e,n),l(e,a,n),s=!0},p(e,s){let l=t;t=m(e),t===l?i[t].p(e,s):(p(),d(i[l],1,1,(()=>{i[l]=null})),c(),n=i[t],n?n.p(e,s):(n=i[t]=r[t](e),n.c()),o(n,1),n.m(a.parentNode,a))},i(e){s||(o(n),s=!0)},o(e){d(n),s=!1},d(e){e&&_(a),i[t].d(e)}}}function lt(e){let t,n,a=s(e[0]),u=[];for(let t=0;t<a.length;t+=1)u[t]=it(et(e,a,t));const g=e=>d(u[e],1,1,(()=>{u[e]=null}));return{c(){t=r("ul");for(let e=0;e<u.length;e+=1)u[e].c();i(t,"class","bx--side-nav__items pipen-report-sidenav svelte-12gpq3")},m(e,a){l(e,t,a);for(let e=0;e<u.length;e+=1)u[e]&&u[e].m(t,null);n=!0},p(e,[n]){if(7&n){let r;for(a=s(e[0]),r=0;r<a.length;r+=1){const s=et(e,a,r);u[r]?(u[r].p(s,n),o(u[r],1)):(u[r]=it(s),u[r].c(),o(u[r],1),u[r].m(t,null))}for(p(),r=a.length;r<u.length;r+=1)g(r);c()}},i(e){if(!n){for(let e=0;e<a.length;e+=1)o(u[e]);n=!0}},o(e){u=u.filter(Boolean);for(let e=0;e<u.length;e+=1)d(u[e]);n=!1},d(e){e&&_(t),m(u,e)}}}function ot(e,t,n){let{toc:a}=t,{page:s}=t;return u((()=>{let e,t=new URL(window.location.href);e=t.hash.length>0?document.querySelector(`[href="${t.hash}"]`):document.querySelector(".toc-current-page"),e.scrollIntoView({behavior:"smooth"})})),e.$$set=e=>{"toc"in e&&n(0,a=e.toc),"page"in e&&n(3,s=e.page)},[a,e=>{if(s===e)return"";let t=new URL(window.location.href);return t.hash="",t.searchParams.set("page",e),t.toString()},e=>e!=s?"toc-noncurrent-page":"toc-current-page",s]}class pt extends t{constructor(e){super(),n(this,e,ot,lt,a,{toc:0,page:3})}}function ct(e,t,n){const a=e.slice();return a[7]=t[n],a}function dt(e,t,n){const a=e.slice();return a[10]=t[n],a}function _t(e){let t,n;return t=new oe({props:{isOpen:Et,transition:e[5],$$slots:{default:[xt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8197&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function mt(e){let t;return{c(){t=S("Go to home:")},m(e,n){l(e,t,n)},d(e){e&&_(t)}}}function ut(e){let t;return{c(){t=S(e[0])},m(e,n){l(e,t,n)},p(e,n){1&n&&A(t,e[0])},d(e){e&&_(t)}}}function gt(e){let t;return{c(){t=S("Switch to process:")},m(e,n){l(e,t,n)},d(e){e&&_(t)}}}function ft(e){let t,n;return t=new de({props:{title:e[7].name,href:`?proc=${e[7].name}`,$$slots:{default:[Ct]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};4&n&&(a.title=e[7].name),4&n&&(a.href=`?proc=${e[7].name}`),8196&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function bt(e){let t,n,a,r;t=new de({props:{class:"procgroup",$$slots:{default:[$t]},$$scope:{ctx:e}}});let i=s(e[7].procs),u=[];for(let t=0;t<i.length;t+=1)u[t]=vt(dt(e,i,t));const h=e=>d(u[e],1,1,(()=>{u[e]=null}));return{c(){f(t.$$.fragment),n=$();for(let e=0;e<u.length;e+=1)u[e].c();a=g()},m(e,s){b(t,e,s),l(e,n,s);for(let t=0;t<u.length;t+=1)u[t]&&u[t].m(e,s);l(e,a,s),r=!0},p(e,n){const r={};if(8196&n&&(r.$$scope={dirty:n,ctx:e}),t.$set(r),4&n){let t;for(i=s(e[7].procs),t=0;t<i.length;t+=1){const s=dt(e,i,t);u[t]?(u[t].p(s,n),o(u[t],1)):(u[t]=vt(s),u[t].c(),o(u[t],1),u[t].m(a.parentNode,a))}for(p(),t=i.length;t<u.length;t+=1)h(t);c()}},i(e){if(!r){o(t.$$.fragment,e);for(let e=0;e<i.length;e+=1)o(u[e]);r=!0}},o(e){d(t.$$.fragment,e),u=u.filter(Boolean);for(let e=0;e<u.length;e+=1)d(u[e]);r=!1},d(e){e&&(_(n),_(a)),C(t,e),m(u,e)}}}function Ct(e){let t,n,a=e[7].name+"";return{c(){t=S(a),n=$()},m(e,a){l(e,t,a),l(e,n,a)},p(e,n){4&n&&a!==(a=e[7].name+"")&&A(t,a)},d(e){e&&(_(t),_(n))}}}function $t(e){let t,n=e[7].name+"";return{c(){t=S(n)},m(e,n){l(e,t,n)},p(e,a){4&a&&n!==(n=e[7].name+"")&&A(t,n)},d(e){e&&_(t)}}}function ht(e){let t,n,a,s=e[10].name+"";return{c(){t=S("- "),n=S(s),a=$()},m(e,s){l(e,t,s),l(e,n,s),l(e,a,s)},p(e,t){4&t&&s!==(s=e[10].name+"")&&A(n,s)},d(e){e&&(_(t),_(n),_(a))}}}function vt(e){let t,n;return t=new de({props:{title:e[10].name,class:"procgroup-proc",href:`?proc=${e[10].name}`,$$slots:{default:[ht]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};4&n&&(a.title=e[10].name),4&n&&(a.href=`?proc=${e[10].name}`),8196&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function yt(e){let t,n,a,s;const r=[bt,ft],i=[];function m(e,t){return e[7].procs?0:1}return t=m(e),n=i[t]=r[t](e),{c(){n.c(),a=g()},m(e,n){i[t].m(e,n),l(e,a,n),s=!0},p(e,s){let l=t;t=m(e),t===l?i[t].p(e,s):(p(),d(i[l],1,1,(()=>{i[l]=null})),c(),n=i[t],n?n.p(e,s):(n=i[t]=r[t](e),n.c()),o(n,1),n.m(a.parentNode,a))},i(e){s||(o(n),s=!0)},o(e){d(n),s=!1},d(e){e&&_(a),i[t].d(e)}}}function St(e){let t,n,a,r,i,u,h,v;t=new ce({props:{$$slots:{default:[mt]},$$scope:{ctx:e}}}),a=new de({props:{href:"index.html",$$slots:{default:[ut]},$$scope:{ctx:e}}}),i=new ce({props:{$$slots:{default:[gt]},$$scope:{ctx:e}}});let y=s(e[2]),S=[];for(let t=0;t<y.length;t+=1)S[t]=yt(ct(e,y,t));const x=e=>d(S[e],1,1,(()=>{S[e]=null}));return{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment),r=$(),f(i.$$.fragment),u=$();for(let e=0;e<S.length;e+=1)S[e].c();h=g()},m(e,s){b(t,e,s),l(e,n,s),b(a,e,s),l(e,r,s),b(i,e,s),l(e,u,s);for(let t=0;t<S.length;t+=1)S[t]&&S[t].m(e,s);l(e,h,s),v=!0},p(e,n){const r={};8192&n&&(r.$$scope={dirty:n,ctx:e}),t.$set(r);const l={};8193&n&&(l.$$scope={dirty:n,ctx:e}),a.$set(l);const d={};if(8192&n&&(d.$$scope={dirty:n,ctx:e}),i.$set(d),4&n){let t;for(y=s(e[2]),t=0;t<y.length;t+=1){const a=ct(e,y,t);S[t]?(S[t].p(a,n),o(S[t],1)):(S[t]=yt(a),S[t].c(),o(S[t],1),S[t].m(h.parentNode,h))}for(p(),t=y.length;t<S.length;t+=1)x(t);c()}},i(e){if(!v){o(t.$$.fragment,e),o(a.$$.fragment,e),o(i.$$.fragment,e);for(let e=0;e<y.length;e+=1)o(S[e]);v=!0}},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),d(i.$$.fragment,e),S=S.filter(Boolean);for(let e=0;e<S.length;e+=1)d(S[e]);v=!1},d(e){e&&(_(n),_(r),_(u),_(h)),C(t,e),C(a,e),C(i,e),m(S,e)}}}function xt(e){let t,n;return t=new pe({props:{$$slots:{default:[St]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8197&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function At(e){let t,n,a,s;t=new le({props:{render:"select",theme:"g10",persist:!0,persistKey:"__pipen-report-theme",select:{themes:["g10","white","g80","g90","g100"],labelText:"Theme",inline:!0}}});let r=e[2].length>0&&_t(e);return{c(){f(t.$$.fragment),n=$(),r&&r.c(),a=g()},m(e,i){b(t,e,i),l(e,n,i),r&&r.m(e,i),l(e,a,i),s=!0},p(e,t){e[2].length>0?r?(r.p(e,t),4&t&&o(r,1)):(r=_t(e),r.c(),o(r,1),r.m(a.parentNode,a)):r&&(p(),d(r,1,1,(()=>{r=null})),c())},i(e){s||(o(t.$$.fragment,e),o(r),s=!0)},o(e){d(t.$$.fragment,e),d(r),s=!1},d(e){e&&(_(n),_(a)),C(t,e),r&&r.d(e)}}}function Mt(e){let t,n;return t=new re({props:{class:"header-util",$$slots:{default:[At]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8197&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function Pt(e){let t;return{c(){t=S(e[4])},m(e,n){l(e,t,n)},p(e,n){16&n&&A(t,e[4])},d(e){e&&_(t)}}}function wt(e){let t,n,a,s,i=null!==e[4]&&Pt(e);return{c(){t=r("h1"),n=S(e[3]),a=$(),i&&i.c(),s=g()},m(e,r){l(e,t,r),x(t,n),l(e,a,r),i&&i.m(e,r),l(e,s,r)},p(e,t){8&t&&A(n,e[3]),null!==e[4]?i?i.p(e,t):(i=Pt(e),i.c(),i.m(s.parentNode,s)):i&&(i.d(1),i=null)},d(e){e&&(_(t),_(a),_(s)),i&&i.d(e)}}}function Tt(e){let t,n;return t=new ie({props:{class:"header-tile",slot:"platform",$$slots:{default:[wt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8216&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function Lt(e){let t,n,a;function s(t){e[6](t)}let r={$$slots:{platform:[Tt],default:[Mt]},$$scope:{ctx:e}};return void 0!==e[1]&&(r.isSideNavOpen=e[1]),t=new se({props:r}),h.push((()=>v(t,"isSideNavOpen",s))),{c(){f(t.$$.fragment)},m(e,n){b(t,e,n),a=!0},p(e,[a]){const s={};8221&a&&(s.$$scope={dirty:a,ctx:e}),!n&&2&a&&(n=!0,s.isSideNavOpen=e[1],y((()=>n=!1))),t.$set(s)},i(e){a||(o(t.$$.fragment,e),a=!0)},o(e){d(t.$$.fragment,e),a=!1},d(e){C(t,e)}}}let Et=!1;function Nt(e,t,n){let{entries:a}=t,{logo:s}=t,{logotext:r=null}=t,{pipeline_name:i}=t,{isSideNavOpen:l}=t;return i=i||s,e.$$set=e=>{"entries"in e&&n(2,a=e.entries),"logo"in e&&n(3,s=e.logo),"logotext"in e&&n(4,r=e.logotext),"pipeline_name"in e&&n(0,i=e.pipeline_name),"isSideNavOpen"in e&&n(1,l=e.isSideNavOpen)},[i,l,a,s,r,{duration:200},function(e){l=e,n(1,l)}]}class Ut extends t{constructor(e){super(),n(this,e,Nt,Lt,a,{entries:2,logo:3,logotext:4,pipeline_name:0,isSideNavOpen:1})}}function Rt(e){let t,n,a;function s(t){e[3](t)}let r={modalHeading:"Running Information",passiveModal:!0,hasScrollingContent:!0,class:"runinfo",$$slots:{default:[Ht]},$$scope:{ctx:e}};return void 0!==e[2]&&(r.open=e[2]),t=new _e({props:r}),h.push((()=>v(t,"open",s))),{c(){f(t.$$.fragment)},m(e,n){b(t,e,n),a=!0},p(e,a){const s={};34&a&&(s.$$scope={dirty:a,ctx:e}),!n&&4&a&&(n=!0,s.open=e[2],y((()=>n=!1))),t.$set(s)},i(e){a||(o(t.$$.fragment,e),a=!0)},o(e){d(t.$$.fragment,e),a=!1},d(e){C(t,e)}}}function Dt(e){let t,n,a,s,r,i;return t=new ge({props:{label:"Session Information"}}),a=new ge({props:{label:"Time Spent"}}),r=new ge({props:{label:"Device Information"}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment),s=$(),f(r.$$.fragment)},m(e,o){b(t,e,o),l(e,n,o),b(a,e,o),l(e,s,o),b(r,e,o),i=!0},p:P,i(e){i||(o(t.$$.fragment,e),o(a.$$.fragment,e),o(r.$$.fragment,e),i=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),d(r.$$.fragment,e),i=!1},d(e){e&&(_(n),_(s)),C(t,e),C(a,e),C(r,e)}}}function It(e){let t,n;return t=new be({props:{type:"multi",code:e[1].session}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};2&n&&(a.code=e[1].session),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function Bt(e){let t,n;return t=new be({props:{type:"multi",code:e[1].time}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};2&n&&(a.code=e[1].time),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function Ft(e){let t,n;return t=new be({props:{type:"multi",code:e[1].device}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};2&n&&(a.code=e[1].device),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function Gt(e){let t,n,a,s,r,i;return t=new fe({props:{$$slots:{default:[It]},$$scope:{ctx:e}}}),a=new fe({props:{$$slots:{default:[Bt]},$$scope:{ctx:e}}}),r=new fe({props:{$$slots:{default:[Ft]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment),s=$(),f(r.$$.fragment)},m(e,o){b(t,e,o),l(e,n,o),b(a,e,o),l(e,s,o),b(r,e,o),i=!0},p(e,n){const s={};34&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const i={};34&n&&(i.$$scope={dirty:n,ctx:e}),a.$set(i);const l={};34&n&&(l.$$scope={dirty:n,ctx:e}),r.$set(l)},i(e){i||(o(t.$$.fragment,e),o(a.$$.fragment,e),o(r.$$.fragment,e),i=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),d(r.$$.fragment,e),i=!1},d(e){e&&(_(n),_(s)),C(t,e),C(a,e),C(r,e)}}}function Ht(e){let t,n,a,s;return t=new me({props:{kind:"warning",lowContrast:!0,hideCloseButton:!0,title:"Showing the first job only. Check the workdir for information of other jobs if any."}}),a=new ue({props:{$$slots:{content:[Gt],default:[Dt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,t){const n={};34&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function Vt(e){let t,n,a;return{c(){t=r("a"),t.textContent="> Running Information",i(t,"href","javascript:void(0);")},m(s,r){l(s,t,r),n||(a=M(t,"click",e[4]),n=!0)},p:P,d(e){e&&_(t),n=!1,a()}}}function zt(e){let t,n,a,s,m,u,g,f,b=e[1]&&Rt(e),C=e[1]&&Vt(e);return{c(){b&&b.c(),t=$(),n=r("div"),a=r("div"),C&&C.c(),s=$(),m=r("div"),u=S("Powered by "),g=r("span"),i(a,"class","runinfo"),i(g,"class","pipen-report-footer-versions"),i(m,"class","versions"),i(n,"class","pipen-report-footer bx--grid")},m(r,i){b&&b.m(r,i),l(r,t,i),l(r,n,i),x(n,a),C&&C.m(a,null),x(n,s),x(n,m),x(m,u),x(m,g),g.innerHTML=e[0],f=!0},p(e,[n]){e[1]?b?(b.p(e,n),2&n&&o(b,1)):(b=Rt(e),b.c(),o(b,1),b.m(t.parentNode,t)):b&&(p(),d(b,1,1,(()=>{b=null})),c()),e[1]?C?C.p(e,n):(C=Vt(e),C.c(),C.m(a,null)):C&&(C.d(1),C=null),(!f||1&n)&&(g.innerHTML=e[0])},i(e){f||(o(b),f=!0)},o(e){d(b),f=!1},d(e){e&&(_(t),_(n)),b&&b.d(e),C&&C.d()}}}function Ot(e,t,n){let{versions:a}=t,{runinfo:s}=t,r=!1;return e.$$set=e=>{"versions"in e&&n(0,a=e.versions),"runinfo"in e&&n(1,s=e.runinfo)},[a,s,r,function(e){r=e,n(2,r)},()=>{n(2,r=!0)}]}class kt extends t{constructor(e){super(),n(this,e,Ot,zt,a,{versions:0,runinfo:1})}}function qt(e){let t,n,a;function s(t){e[12](t)}let r={$$slots:{default:[Kt]},$$scope:{ctx:e}};return void 0!==e[0]&&(r.isOpen=e[0]),t=new $e({props:r}),h.push((()=>v(t,"isOpen",s))),{c(){f(t.$$.fragment)},m(e,n){b(t,e,n),a=!0},p(e,a){const s={};8576&a&&(s.$$scope={dirty:a,ctx:e}),!n&&1&a&&(n=!0,s.isOpen=e[0],y((()=>n=!1))),t.$set(s)},i(e){a||(o(t.$$.fragment,e),a=!0)},o(e){d(t.$$.fragment,e),a=!1},d(e){C(t,e)}}}function Kt(e){let t,n;return t=new pt({props:{toc:e[7],page:e[8]}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};128&n&&(a.toc=e[7]),256&n&&(a.page=e[8]),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function jt(e){let t;const n=e[10].default,a=w(n,e,e[13],null);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||8192&s)&&T(a,n,e,e[13],t?E(n,e[13],s,null):L(e[13]),null)},i(e){t||(o(a,e),t=!0)},o(e){d(a,e),t=!1},d(e){a&&a.d(e)}}}function Xt(e){let t,n;return t=new ye({props:{$$slots:{default:[jt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8192&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function Yt(e){let t,n;return t=new ve({props:{$$slots:{default:[Xt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8192&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function Wt(e){let t,n,a,s;return t=new he({props:{$$slots:{default:[Yt]},$$scope:{ctx:e}}}),a=new kt({props:{versions:e[3],runinfo:e[9]}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const s={};8192&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};8&n&&(r.versions=e[3]),512&n&&(r.runinfo=e[9]),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function Qt(e){let t,n,a,s,r,i;function m(t){e[11](t)}let u={logo:e[1],logotext:e[2],entries:e[4],pipeline_name:e[5]};void 0!==e[0]&&(u.isSideNavOpen=e[0]),t=new Ut({props:u}),h.push((()=>v(t,"isSideNavOpen",m)));let g=e[6]&&e[7].length>0&&qt(e);return r=new Ce({props:{$$slots:{default:[Wt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),a=$(),g&&g.c(),s=$(),f(r.$$.fragment)},m(e,n){b(t,e,n),l(e,a,n),g&&g.m(e,n),l(e,s,n),b(r,e,n),i=!0},p(e,[a]){const i={};2&a&&(i.logo=e[1]),4&a&&(i.logotext=e[2]),16&a&&(i.entries=e[4]),32&a&&(i.pipeline_name=e[5]),!n&&1&a&&(n=!0,i.isSideNavOpen=e[0],y((()=>n=!1))),t.$set(i),e[6]&&e[7].length>0?g?(g.p(e,a),192&a&&o(g,1)):(g=qt(e),g.c(),o(g,1),g.m(s.parentNode,s)):g&&(p(),d(g,1,1,(()=>{g=null})),c());const l={};8712&a&&(l.$$scope={dirty:a,ctx:e}),r.$set(l)},i(e){i||(o(t.$$.fragment,e),o(g),o(r.$$.fragment,e),i=!0)},o(e){d(t.$$.fragment,e),d(g),d(r.$$.fragment,e),i=!1},d(e){e&&(_(a),_(s)),C(t,e),g&&g.d(e),C(r,e)}}}function Jt(e,t,n){let a,{$$slots:s={},$$scope:r}=t,{logo:i}=t,{logotext:l=null}=t,{versions:o}=t,{entries:p}=t,{pipeline_name:c}=t,{report_toc:d}=t,{toc:_}=t,{page:m}=t,{isSideNavOpen:g}=t,{runinfo:f}=t;return u((()=>{const e=new URL(window.location.href);let t;t=e.hash.length>0?document.querySelector(`li:has(>[href="${e.hash}"])`):document.querySelector("li:has(>.toc-current-page)"),t&&(t.classList.add("toc-active"),t.scrollIntoView());const n=document.querySelectorAll(".pipen-report-toc-anchor");a=new IntersectionObserver((e=>{e.forEach((e=>{const t=e.target.getAttribute("id"),n=document.querySelector(`li:has(>[href="#${t}"])`);n&&(e.isIntersecting?n.classList.add("toc-active"):n.classList.remove("toc-active"))}))}),{root:null,rootMargin:"0px",threshold:.5}),n.forEach((e=>{a.observe(e)}))})),e.$$set=e=>{"logo"in e&&n(1,i=e.logo),"logotext"in e&&n(2,l=e.logotext),"versions"in e&&n(3,o=e.versions),"entries"in e&&n(4,p=e.entries),"pipeline_name"in e&&n(5,c=e.pipeline_name),"report_toc"in e&&n(6,d=e.report_toc),"toc"in e&&n(7,_=e.toc),"page"in e&&n(8,m=e.page),"isSideNavOpen"in e&&n(0,g=e.isSideNavOpen),"runinfo"in e&&n(9,f=e.runinfo),"$$scope"in e&&n(13,r=e.$$scope)},[g,i,l,o,p,c,d,_,m,f,s,function(e){g=e,n(0,g)},function(e){g=e,n(0,g)},r]}class Zt extends t{constructor(e){super(),n(this,e,Jt,Qt,a,{logo:1,logotext:2,versions:3,entries:4,pipeline_name:5,report_toc:6,toc:7,page:8,isSideNavOpen:0,runinfo:9})}}function en(e){let t;return{c(){t=S(e[2])},m(e,n){l(e,t,n)},p:P,d(e){e&&_(t)}}}function tn(e){let t,n,a,s;return n=new Se({props:{icon:e[1],kind:"tertiary",href:e[3](),$$slots:{default:[en]},$$scope:{ctx:e}}}),{c(){t=r("div"),f(n.$$.fragment),i(t,"class",a=`pagenav ${e[0]}`)},m(e,a){l(e,t,a),b(n,t,null),s=!0},p(e,[r]){const l={};16&r&&(l.$$scope={dirty:r,ctx:e}),n.$set(l),(!s||1&r&&a!==(a=`pagenav ${e[0]}`))&&i(t,"class",a)},i(e){s||(o(n.$$.fragment,e),s=!0)},o(e){d(n.$$.fragment,e),s=!1},d(e){e&&_(t),C(n)}}}function nn(e,t,n){let{dir:a}=t,s="up"===a?Re:De,r="up"===a?"Go to previous page":"Go to next page";return e.$$set=e=>{"dir"in e&&n(0,a=e.dir)},[a,s,r,()=>{let e=new URL(window.location.href);e.hash="";const t=parseInt(e.searchParams.get("page")||0),n="up"===a?t-1:t+1;return 0===n?e.searchParams.delete("page"):e.searchParams.set("page",n),e.toString()}]}class an extends t{constructor(e){super(),n(this,e,nn,tn,a,{dir:0})}}function sn(e){let t,n;const a=[{tooltipText:e[0]},{icon:e[1]},{direction:"bottom"},{align:"start"},e[3]];let s={};for(let e=0;e<a.length;e+=1)s=N(s,a[e]);return t=new xe({props:s}),t.$on("click",e[2]),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,[n]){const s=11&n?U(a,[1&n&&{tooltipText:e[0]},2&n&&{icon:e[1]},a[2],a[3],8&n&&R(e[3])]):{};t.$set(s)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function rn(e,t,n){const a=["src","tip","icon"];let s,r=D(t,a),{src:i=""}=t,{tip:l=""}=t,{icon:o="auto"}=t,p=i.split(".").at(-1).toLocaleLowerCase();if("string"==typeof o)switch("auto"===o&&(o=p),o){case"pdf":case"DocumentPdf":s=Ye;break;case"html":case"htm":case"ChartLineData":s=Xe;break;case"png":case"Png":s=je;break;case"jpg":case"jpeg":case"Jpg":s=Ke;break;case"gif":case"Gif":s=qe;break;case"tif":case"tiff":case"Tif":s=ke;break;case"eps":case"DatabaseElastic":s=Oe;break;case"mp4":case"avi":case"mov":case"mkv":case"webm":case"VideoPlayer":s=ze;break;case"Image":s=Ve;break;case"Data":case"DataVolume":s=He;break;case"Code":case"IbmWatsonxCodeAssistantForZRefactor":s=Ge;break;case"svg":case"Svg":s=Fe;break;case"zip":case"tar":case"gz":case"bz2":case"7z":case"rar":case"xz":case"Archive":s=Be;break;default:s=Ie}const c={...r,class:[r.class,"pipen-report-image-download"].filter(Boolean).join(" ")};return e.$$set=e=>{t=N(N({},t),I(e)),n(7,r=D(t,a)),"src"in e&&n(5,i=e.src),"tip"in e&&n(0,l=e.tip),"icon"in e&&n(4,o=e.icon)},[l,s,()=>{const e=document.createElement("a");e.href=i,e.download=i.split("/").pop(),e.click()},c,o,i]}class ln extends t{constructor(e){super(),n(this,e,rn,sn,a,{src:5,tip:0,icon:4})}}const{Boolean:on}=H;function pn(e,t,n){const a=e.slice();return a[5]=t[n].src,a[6]=t[n].tip,a[7]=t[n].icon,a[9]=n,a}function cn(e){let t,n,a,i,u=s(e[0]),g=[];for(let t=0;t<u.length;t+=1)g[t]=dn(pn(e,u,t));const f=e=>d(g[e],1,1,(()=>{g[e]=null}));let b=[e[1],{role:"button"},{tabindex:"0"}],C={};for(let e=0;e<b.length;e+=1)C=N(C,b[e]);return{c(){t=r("div");for(let e=0;e<g.length;e+=1)g[e].c();F(t,C),G(t,"svelte-14lcnln",!0)},m(s,r){l(s,t,r);for(let e=0;e<g.length;e+=1)g[e]&&g[e].m(t,null);n=!0,a||(i=M(t,"mouseenter",e[3]),a=!0)},p(e,n){if(1&n){let a;for(u=s(e[0]),a=0;a<u.length;a+=1){const s=pn(e,u,a);g[a]?(g[a].p(s,n),o(g[a],1)):(g[a]=dn(s),g[a].c(),o(g[a],1),g[a].m(t,null))}for(p(),a=u.length;a<g.length;a+=1)f(a);c()}G(t,"svelte-14lcnln",!0)},i(e){if(!n){for(let e=0;e<u.length;e+=1)o(g[e]);n=!0}},o(e){g=g.filter(on);for(let e=0;e<g.length;e+=1)d(g[e]);n=!1},d(e){e&&_(t),m(g,e),a=!1,i()}}}function dn(e){let t,n;return t=new ln({props:{src:e[5],tip:e[6],icon:e[7]}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.src=e[5]),1&n&&(a.tip=e[6]),1&n&&(a.icon=e[7]),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function _n(e){let t,n,a=e[0].length>0&&cn(e);return{c(){a&&a.c(),t=g()},m(e,s){a&&a.m(e,s),l(e,t,s),n=!0},p(e,[n]){e[0].length>0?a?(a.p(e,n),1&n&&o(a,1)):(a=cn(e),a.c(),o(a,1),a.m(t.parentNode,t)):a&&(p(),d(a,1,1,(()=>{a=null})),c())},i(e){n||(o(a),n=!0)},o(e){d(a),n=!1},d(e){e&&_(t),a&&a.d(e)}}}function mn(e,t,n){const a=["data"];let s=D(t,a),{data:r=[]}=t;Array.isArray(r)||(r=[r]),r=r.map((e=>"string"==typeof e?{src:e,tip:`Download the ${e.split(".").at(-1)} format.`,icon:"auto"}:(e.tip=e.tip||`Download the ${e.src.split(".").at(-1)} format.`,e.icon=e.icon||"auto",e)));const i={...s,class:[s.class,"pipen-report-image-download-list"].filter(Boolean).join(" ")},l=B();return e.$$set=e=>{t=N(N({},t),I(e)),n(4,s=D(t,a)),"data"in e&&n(0,r=e.data)},[r,i,l,()=>l("mouseenter")]}class un extends t{constructor(e){super(),n(this,e,mn,_n,a,{data:0})}}const gn=e=>({}),fn=e=>({}),bn=e=>({}),Cn=e=>({}),$n=e=>({}),hn=e=>({}),vn=e=>({}),yn=e=>({});function Sn(e){let t,n;return t=new Ae({props:{ratio:e[6],$$slots:{default:[Tn]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};64&n&&(a.ratio=e[6]),33556159&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function xn(e){let t,n,a,s,r=e[2]&&Ln(e),i=e[0]&&En(e),m=e[1]&&Un(e);return{c(){r&&r.c(),t=$(),i&&i.c(),n=$(),m&&m.c(),a=g()},m(e,o){r&&r.m(e,o),l(e,t,o),i&&i.m(e,o),l(e,n,o),m&&m.m(e,o),l(e,a,o),s=!0},p(e,s){e[2]?r?(r.p(e,s),4&s&&o(r,1)):(r=Ln(e),r.c(),o(r,1),r.m(t.parentNode,t)):r&&(p(),d(r,1,1,(()=>{r=null})),c()),e[0]?i?(i.p(e,s),1&s&&o(i,1)):(i=En(e),i.c(),o(i,1),i.m(n.parentNode,n)):i&&(p(),d(i,1,1,(()=>{i=null})),c()),e[1]?m?(m.p(e,s),2&s&&o(m,1)):(m=Un(e),m.c(),o(m,1),m.m(a.parentNode,a)):m&&(p(),d(m,1,1,(()=>{m=null})),c())},i(e){s||(o(r),o(i),o(m),s=!0)},o(e){d(r),d(i),d(m),s=!1},d(e){e&&(_(t),_(n),_(a)),r&&r.d(e),i&&i.d(e),m&&m.d(e)}}}function An(e){let t;const n=e[15].loading,a=w(n,e,e[25],Cn);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||33554432&s)&&T(a,n,e,e[25],t?E(n,e[25],s,bn):L(e[25]),Cn)},i(e){t||(o(a,e),t=!0)},o(e){d(a,e),t=!1},d(e){a&&a.d(e)}}}function Mn(e){let t,n,a,s,i,m,u,f,b=[e[12],{src:n=e[3]},{alt:e[4]},{title:"Click to zoom in ..."}],C={};for(let e=0;e<b.length;e+=1)C=N(C,b[e]);let h=e[9]&&Pn(e);return{c(){t=r("img"),s=$(),h&&h.c(),i=g(),F(t,C)},m(n,a){l(n,t,a),l(n,s,a),h&&h.m(n,a),l(n,i,a),m=!0,u||(f=[M(t,"click",e[11]),M(t,"mouseenter",e[19]),M(t,"mouseleave",e[20])],u=!0)},p(a,s){F(t,C=U(b,[(e=a)[12],(!m||8&s&&!V(t.src,n=e[3]))&&{src:n},(!m||16&s)&&{alt:e[4]},{title:"Click to zoom in ..."}])),e[9]?h?(h.p(e,s),512&s&&o(h,1)):(h=Pn(e),h.c(),o(h,1),h.m(i.parentNode,i)):h&&(p(),d(h,1,1,(()=>{h=null})),c())},i(n){m||(n&&z((()=>{m&&(a||(a=O(t,k,{duration:e[7]?In:0},!0)),a.run(1))})),o(h),m=!0)},o(n){n&&(a||(a=O(t,k,{duration:e[7]?In:0},!1)),a.run(0)),d(h),m=!1},d(e){e&&(_(t),_(s),_(i)),e&&a&&a.end(),h&&h.d(e),u=!1,q(f)}}}function Pn(e){let t,n;return t=new un({props:{data:e[5]}}),t.$on("mouseenter",e[21]),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};32&n&&(a.data=e[5]),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function wn(e){let t;const n=e[15].error,a=w(n,e,e[25],fn);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||33554432&s)&&T(a,n,e,e[25],t?E(n,e[25],s,gn):L(e[25]),fn)},i(e){t||(o(a,e),t=!0)},o(e){d(a,e),t=!1},d(e){a&&a.d(e)}}}function Tn(e){let t,n,a,s,r=e[2]&&An(e),i=e[0]&&Mn(e),m=e[1]&&wn(e);return{c(){r&&r.c(),t=$(),i&&i.c(),n=$(),m&&m.c(),a=g()},m(e,o){r&&r.m(e,o),l(e,t,o),i&&i.m(e,o),l(e,n,o),m&&m.m(e,o),l(e,a,o),s=!0},p(e,s){e[2]?r?(r.p(e,s),4&s&&o(r,1)):(r=An(e),r.c(),o(r,1),r.m(t.parentNode,t)):r&&(p(),d(r,1,1,(()=>{r=null})),c()),e[0]?i?(i.p(e,s),1&s&&o(i,1)):(i=Mn(e),i.c(),o(i,1),i.m(n.parentNode,n)):i&&(p(),d(i,1,1,(()=>{i=null})),c()),e[1]?m?(m.p(e,s),2&s&&o(m,1)):(m=wn(e),m.c(),o(m,1),m.m(a.parentNode,a)):m&&(p(),d(m,1,1,(()=>{m=null})),c())},i(e){s||(o(r),o(i),o(m),s=!0)},o(e){d(r),d(i),d(m),s=!1},d(e){e&&(_(t),_(n),_(a)),r&&r.d(e),i&&i.d(e),m&&m.d(e)}}}function Ln(e){let t;const n=e[15].loading,a=w(n,e,e[25],yn);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||33554432&s)&&T(a,n,e,e[25],t?E(n,e[25],s,vn):L(e[25]),yn)},i(e){t||(o(a,e),t=!0)},o(e){d(a,e),t=!1},d(e){a&&a.d(e)}}}function En(e){let t,n,a,s,i,m,u,f,b=[e[12],{src:n=e[3]},{alt:e[4]},{title:"Click to zoom in ..."}],C={};for(let e=0;e<b.length;e+=1)C=N(C,b[e]);let h=e[9]&&Nn(e);return{c(){t=r("img"),s=$(),h&&h.c(),i=g(),F(t,C)},m(n,a){l(n,t,a),l(n,s,a),h&&h.m(n,a),l(n,i,a),m=!0,u||(f=[M(t,"click",e[11]),M(t,"mouseenter",e[16]),M(t,"mouseleave",e[17])],u=!0)},p(a,s){F(t,C=U(b,[(e=a)[12],(!m||8&s&&!V(t.src,n=e[3]))&&{src:n},(!m||16&s)&&{alt:e[4]},{title:"Click to zoom in ..."}])),e[9]?h?(h.p(e,s),512&s&&o(h,1)):(h=Nn(e),h.c(),o(h,1),h.m(i.parentNode,i)):h&&(p(),d(h,1,1,(()=>{h=null})),c())},i(n){m||(n&&z((()=>{m&&(a||(a=O(t,k,{duration:e[7]?In:0},!0)),a.run(1))})),o(h),m=!0)},o(n){n&&(a||(a=O(t,k,{duration:e[7]?In:0},!1)),a.run(0)),d(h),m=!1},d(e){e&&(_(t),_(s),_(i)),e&&a&&a.end(),h&&h.d(e),u=!1,q(f)}}}function Nn(e){let t,n;return t=new un({props:{data:e[5]}}),t.$on("mouseenter",e[18]),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};32&n&&(a.data=e[5]),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function Un(e){let t;const n=e[15].error,a=w(n,e,e[25],hn);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||33554432&s)&&T(a,n,e,e[25],t?E(n,e[25],s,$n):L(e[25]),hn)},i(e){t||(o(a,e),t=!0)},o(e){d(a,e),t=!1},d(e){a&&a.d(e)}}}function Rn(e){let t,n,a,s;return{c(){t=r("img"),i(t,"class","pipen-report-image-lg"),V(t.src,n=e[3])||i(t,"src",n),i(t,"alt",e[4])},m(e,n){l(e,t,n),s=!0},p(a,r){e=a,(!s||8&r&&!V(t.src,n=e[3]))&&i(t,"src",n),(!s||16&r)&&i(t,"alt",e[4])},i(n){s||(n&&z((()=>{s&&(a||(a=O(t,k,{duration:e[7]?In:0},!0)),a.run(1))})),s=!0)},o(n){n&&(a||(a=O(t,k,{duration:e[7]?In:0},!1)),a.run(0)),s=!1},d(e){e&&_(t),e&&a&&a.end()}}}function Dn(e){let t,n,a,s,r,i;const m=[xn,Sn],u=[];function g(e,t){return void 0===e[6]?0:1}function S(t){e[22](t)}t=g(e),n=u[t]=m[t](e);let x={modalHeading:e[4],primaryButtonText:"Open in new window ...",secondaryButtonText:"Close",$$slots:{default:[Rn]},$$scope:{ctx:e}};return void 0!==e[8]&&(x.open=e[8]),s=new _e({props:x}),h.push((()=>v(s,"open",S))),s.$on("click:button--secondary",e[23]),s.$on("click:button--primary",e[24]),{c(){n.c(),a=$(),f(s.$$.fragment)},m(e,n){u[t].m(e,n),l(e,a,n),b(s,e,n),i=!0},p(e,[i]){let l=t;t=g(e),t===l?u[t].p(e,i):(p(),d(u[l],1,1,(()=>{u[l]=null})),c(),n=u[t],n?n.p(e,i):(n=u[t]=m[t](e),n.c()),o(n,1),n.m(a.parentNode,a));const _={};16&i&&(_.modalHeading=e[4]),33554584&i&&(_.$$scope={dirty:i,ctx:e}),!r&&256&i&&(r=!0,_.open=e[8],y((()=>r=!1))),s.$set(_)},i(e){i||(o(n),o(s.$$.fragment,e),i=!0)},o(e){d(n),d(s.$$.fragment,e),i=!1},d(e){e&&_(a),u[t].d(e),C(s,e)}}}const In=110;function Bn(e,t,n){const a=["src","alt","download","ratio","loading","loaded","error","fadeIn","border","loadImage"];let s=D(t,a),{$$slots:r={},$$scope:i}=t,{src:l=""}=t,{alt:o=""}=t,{download:p=[]}=t,{ratio:c}=t,{loading:d=!1}=t,{loaded:_=!1}=t,{error:m=!1}=t,{fadeIn:g=!1}=t,{border:f=!0}=t,b=!1,C=!1;const $=e=>{null!=y&&(y=null),n(0,_=!1),n(1,m=!1),y=new Image,y.src=e||l,y.onload=()=>n(0,_=!0),y.onerror=()=>n(1,m=!0)},h=B();let v,y=null;u((()=>()=>y=null));const S={...s,class:[s.class,"pipen-report-image",f?"":"pipen-report-image-noborder"].filter(Boolean).join(" ")};return e.$$set=e=>{t=N(N({},t),I(e)),n(28,s=D(t,a)),"src"in e&&n(3,l=e.src),"alt"in e&&n(4,o=e.alt),"download"in e&&n(5,p=e.download),"ratio"in e&&n(6,c=e.ratio),"loading"in e&&n(2,d=e.loading),"loaded"in e&&n(0,_=e.loaded),"error"in e&&n(1,m=e.error),"fadeIn"in e&&n(7,g=e.fadeIn),"border"in e&&n(13,f=e.border),"$$scope"in e&&n(25,i=e.$$scope)},e.$$.update=()=>{3&e.$$.dirty&&n(2,d=!_&&!m),8&e.$$.dirty&&l&&"undefined"!=typeof window&&$(),1&e.$$.dirty&&_&&h("load"),2&e.$$.dirty&&m&&h("error")},[_,m,d,l,o,p,c,g,b,C,v,()=>{n(8,b=!0)},S,f,$,r,()=>{n(9,C=!0)},()=>{n(10,v=setTimeout((()=>{n(9,C=!1)}),100))},()=>{clearTimeout(v),n(9,C=!0)},()=>{n(9,C=!0)},()=>{n(10,v=setTimeout((()=>{n(9,C=!1)}),100))},()=>{clearTimeout(v),n(9,C=!0)},function(e){b=e,n(8,b)},()=>{n(8,b=!1)},()=>{n(8,b=!1),window.open(l)},i]}class Fn extends t{constructor(e){super(),n(this,e,Bn,Dn,a,{src:3,alt:4,download:5,ratio:6,loading:2,loaded:0,error:1,fadeIn:7,border:13,loadImage:14})}get loadImage(){return this.$$.ctx[14]}}function Gn(e){let t,n,a=[e[2]],s={};for(let e=0;e<a.length;e+=1)s=N(s,a[e]);return{c(){t=r("div"),n=S("Image loading ..."),F(t,s)},m(e,a){l(e,t,a),x(t,n)},p:P,d(e){e&&_(t)}}}function Hn(e){let t,n,a=[e[3]],s={};for(let e=0;e<a.length;e+=1)s=N(s,a[e]);return{c(){t=r("div"),n=S("Image loading error!"),F(t,s)},m(e,a){l(e,t,a),x(t,n)},p:P,d(e){e&&_(t)}}}function Vn(e){let t,n,a;const s=[{src:e[1]},e[4]];let i={$$slots:{error:[Hn],loading:[Gn]},$$scope:{ctx:e}};for(let e=0;e<s.length;e+=1)i=N(i,s[e]);n=new Fn({props:i});let p=[e[0]],c={};for(let e=0;e<p.length;e+=1)c=N(c,p[e]);return{c(){t=r("div"),f(n.$$.fragment),F(t,c)},m(e,s){l(e,t,s),b(n,t,null),a=!0},p(e,[a]){const r=18&a?U(s,[2&a&&{src:e[1]},16&a&&R(e[4])]):{};256&a&&(r.$$scope={dirty:a,ctx:e}),n.$set(r),F(t,c=U(p,[1&a&&e[0]]))},i(e){a||(o(n.$$.fragment,e),a=!0)},o(e){d(n.$$.fragment,e),a=!1},d(e){e&&_(t),C(n)}}}function zn(e,t,n){const a=["frameProps","width","height","src"];let s=D(t,a),{frameProps:r={class:"pipen-report-image-frame"}}=t,{width:i=300}=t,{height:l=200}=t,{src:o}=t;r||(r={"style:position":"relative"});let p=`width: ${i}px; height: ${l}px; aspect-ratio: ${i} / ${l}`,c={...s,class:[s.class,"pipen-report-image-loading"].filter(Boolean).join(" "),style:s.style?`${s.style}; ${p}`:p},d={...c,class:[c.class,"error"].join(" ")};return e.$$set=e=>{t=N(N({},t),I(e)),n(4,s=D(t,a)),"frameProps"in e&&n(0,r=e.frameProps),"width"in e&&n(5,i=e.width),"height"in e&&n(6,l=e.height),"src"in e&&n(1,o=e.src)},[r,o,c,d,s,i,l]}let On=class extends t{constructor(e){super(),n(this,e,zn,Vn,a,{frameProps:0,width:5,height:6,src:1})}};const kn=e=>({}),qn=e=>({});function Kn(e){let t,n;const a=e[25].default,s=w(a,e,e[24],null);return{c(){t=r("ul"),s&&s.c(),i(t,"role","menu"),i(t,"tabindex","-1"),i(t,"aria-label",e[14]),i(t,"data-floating-menu-direction",e[5]),i(t,"class",e[9]),G(t,"bx--overflow-menu-options",!0),G(t,"bx--overflow-menu--flip",e[7]),G(t,"bx--overflow-menu-options--open",e[0]),G(t,"bx--overflow-menu-options--light",e[6]),G(t,"bx--overflow-menu-options--sm","sm"===e[4]),G(t,"bx--overflow-menu-options--xl","xl"===e[4]),G(t,"bx--breadcrumb-menu-options",!!e[15])},m(a,r){l(a,t,r),s&&s.m(t,null),e[32](t),n=!0},p(e,r){s&&s.p&&(!n||16777216&r[0])&&T(s,a,e,e[24],n?E(a,e[24],r,null):L(e[24]),null),(!n||16384&r[0])&&i(t,"aria-label",e[14]),(!n||32&r[0])&&i(t,"data-floating-menu-direction",e[5]),(!n||512&r[0])&&i(t,"class",e[9]),(!n||512&r[0])&&G(t,"bx--overflow-menu-options",!0),(!n||640&r[0])&&G(t,"bx--overflow-menu--flip",e[7]),(!n||513&r[0])&&G(t,"bx--overflow-menu-options--open",e[0]),(!n||576&r[0])&&G(t,"bx--overflow-menu-options--light",e[6]),(!n||528&r[0])&&G(t,"bx--overflow-menu-options--sm","sm"===e[4]),(!n||528&r[0])&&G(t,"bx--overflow-menu-options--xl","xl"===e[4]),(!n||33280&r[0])&&G(t,"bx--breadcrumb-menu-options",!!e[15])},i(e){n||(o(s,e),n=!0)},o(e){d(s,e),n=!1},d(n){n&&_(t),s&&s.d(n),e[32](null)}}}function jn(e){let t,n,a,s,i,m,u,h;const v=e[25].menu,y=w(v,e,e[24],qn),S=y||function(e){let t,n,a;var s=e[1];function r(e,t){return{props:{"aria-label":e[11],title:e[11],class:"bx--overflow-menu__icon "+e[10]}}}return s&&(t=W(s,r(e))),{c(){t&&f(t.$$.fragment),n=g()},m(e,s){t&&b(t,e,s),l(e,n,s),a=!0},p(e,a){if(2&a[0]&&s!==(s=e[1])){if(t){p();const e=t;d(e.$$.fragment,1,0,(()=>{C(e,1)})),c()}s?(t=W(s,r(e)),f(t.$$.fragment),o(t.$$.fragment,1),b(t,n.parentNode,n)):t=null}else if(s){const n={};2048&a[0]&&(n["aria-label"]=e[11]),2048&a[0]&&(n.title=e[11]),1024&a[0]&&(n.class="bx--overflow-menu__icon "+e[10]),t.$set(n)}},i(e){a||(t&&o(t.$$.fragment,e),a=!0)},o(e){t&&d(t.$$.fragment,e),a=!1},d(e){e&&_(n),t&&C(t,e)}}}(e);let A=e[0]&&Kn(e),P=[{type:"button"},{"aria-haspopup":""},{"aria-expanded":e[0]},{"aria-label":e[14]},{id:e[12]},e[20]],R={};for(let e=0;e<P.length;e+=1)R=N(R,P[e]);return{c(){t=new Q(!1),n=g(),a=$(),s=r("button"),S.c(),i=$(),A&&A.c(),t.a=n,F(s,R),G(s,"bx--overflow-menu",!0),G(s,"bx--overflow-menu--open",e[0]),G(s,"bx--overflow-menu--light",e[6]),G(s,"bx--overflow-menu--sm","sm"===e[4]),G(s,"bx--overflow-menu--xl","xl"===e[4])},m(r,o){t.m(e[13],document.head),x(document.head,n),l(r,a,o),l(r,s,o),S.m(s,null),x(s,i),A&&A.m(s,null),s.autofocus&&s.focus(),e[33](s),m=!0,u||(h=[M(window,"click",e[31]),M(s,"click",e[26]),M(s,"click",e[34]),M(s,"mouseover",e[27]),M(s,"mouseenter",e[28]),M(s,"mouseleave",e[29]),M(s,"keydown",e[30]),M(s,"keydown",e[35]),M(s,"focusout",e[36])],u=!0)},p(e,n){(!m||8192&n[0])&&t.p(e[13]),y?y.p&&(!m||16777216&n[0])&&T(y,v,e,e[24],m?E(v,e[24],n,kn):L(e[24]),qn):S.p&&(!m||3074&n[0])&&S.p(e,m?n:[-1,-1]),e[0]?A?(A.p(e,n),1&n[0]&&o(A,1)):(A=Kn(e),A.c(),o(A,1),A.m(s,null)):A&&(p(),d(A,1,1,(()=>{A=null})),c()),F(s,R=U(P,[{type:"button"},{"aria-haspopup":""},(!m||1&n[0])&&{"aria-expanded":e[0]},(!m||16384&n[0])&&{"aria-label":e[14]},(!m||4096&n[0])&&{id:e[12]},1048576&n[0]&&e[20]])),G(s,"bx--overflow-menu",!0),G(s,"bx--overflow-menu--open",e[0]),G(s,"bx--overflow-menu--light",e[6]),G(s,"bx--overflow-menu--sm","sm"===e[4]),G(s,"bx--overflow-menu--xl","xl"===e[4])},i(e){m||(o(S,e),o(A),m=!0)},o(e){d(S,e),d(A),m=!1},d(r){r&&(t.d(),_(a),_(s)),_(n),S.d(r),A&&A.d(),e[33](null),u=!1,q(h)}}}function Xn(e,t,n){let a,s;const r=["size","direction","open","light","flipped","persistentClickItems","menuOptionsClass","icon","iconClass","iconDescription","id","buttonRef","menuRef"];let i,l,o,p=D(t,r),{$$slots:c={},$$scope:d}=t,{size:_}=t,{direction:m="bottom"}=t,{open:u=!1}=t,{light:g=!1}=t,{flipped:f=!1}=t,{persistentClickItems:b=!1}=t,{menuOptionsClass:C}=t,{icon:$=Qe}=t,{iconClass:v}=t,{iconDescription:y="Open and close list of options"}=t,{id:S="ccs-"+Math.random().toString(36)}=t,{buttonRef:x=null}=t,{menuRef:A=null}=t;const M=K("BreadcrumbItem"),P=B(),w=J([]);j(e,w,(e=>n(23,l=e)));const T=J(void 0);j(e,T,(e=>n(38,o=e)));const L=J(void 0),E=J(-1);let U;j(e,E,(e=>n(22,i=e)));let R=!0;X("OverflowMenu",{focusedId:L,add:({id:e,text:t,primaryFocus:n,disabled:a})=>{w.update((s=>(n&&E.set(s.length),[...s,{id:e,text:t,primaryFocus:n,disabled:a,index:s.length}])))},update:e=>{T.set(e)},change:e=>{let t=i+e;t<0?t=l.length-1:t>=l.length&&(t=0);let n=l[t].disabled;for(;n;)t+=e,t<0?t=l.length-1:t>=l.length&&(t=0),n=l[t].disabled;E.set(t)}}),Y((()=>{if(o){const{index:e,text:t}=l.filter((e=>e.id===o))[0];P("close",{index:e,text:t}),n(0,u=!1)}if(u){const{width:e,height:t}=x.getBoundingClientRect();n(21,U=e),!R&&i<0&&A.focus(),f&&(n(3,A.style.left="auto",A),n(3,A.style.right=0,A)),"top"===m?(n(3,A.style.top="auto",A),n(3,A.style.bottom=t+"px",A)):"bottom"===m&&n(3,A.style.top=t+"px",A),M&&(n(3,A.style.top=t+10+"px",A),n(3,A.style.left="-11px",A))}u||(w.set([]),T.set(void 0),E.set(0)),R=!1}));return e.$$set=e=>{n(40,t=N(N({},t),I(e))),n(20,p=D(t,r)),"size"in e&&n(4,_=e.size),"direction"in e&&n(5,m=e.direction),"open"in e&&n(0,u=e.open),"light"in e&&n(6,g=e.light),"flipped"in e&&n(7,f=e.flipped),"persistentClickItems"in e&&n(8,b=e.persistentClickItems),"menuOptionsClass"in e&&n(9,C=e.menuOptionsClass),"icon"in e&&n(1,$=e.icon),"iconClass"in e&&n(10,v=e.iconClass),"iconDescription"in e&&n(11,y=e.iconDescription),"id"in e&&n(12,S=e.id),"buttonRef"in e&&n(2,x=e.buttonRef),"menuRef"in e&&n(3,A=e.menuRef),"$$scope"in e&&n(24,d=e.$$scope)},e.$$.update=()=>{n(14,a=t["aria-label"]||"menu"),12582912&e.$$.dirty[0]&&l[i]&&L.set(l[i].id),2101248&e.$$.dirty[0]&&n(13,s=`<style>\n    #${S} .bx--overflow-menu-options.bx--overflow-menu-options:after {\n      width: ${U?U+"px":"2rem"};\n    }\n  </style>`)},M&&n(1,$=We),t=I(t),[u,$,x,A,_,m,g,f,b,C,v,y,S,s,a,M,P,w,T,E,p,U,i,l,d,c,function(t){Z.call(this,e,t)},function(t){Z.call(this,e,t)},function(t){Z.call(this,e,t)},function(t){Z.call(this,e,t)},function(t){Z.call(this,e,t)},({target:e})=>{x&&x.contains(e)||A&&!A.contains(e)&&n(0,u=!1)},function(e){h[e?"unshift":"push"]((()=>{A=e,n(3,A)}))},function(e){h[e?"unshift":"push"]((()=>{x=e,n(2,x)}))},e=>{b&&e.stopPropagation(),A&&A.contains(e.target)||(n(0,u=!u),u||P("close"))},e=>{u&&(["ArrowDown","ArrowLeft","ArrowRight","ArrowUp"].includes(e.key)?e.preventDefault():"Escape"===e.key&&(e.stopPropagation(),P("close"),n(0,u=!1),x.focus()))},e=>{u&&(x.contains(e.relatedTarget)||(P("close"),n(0,u=!1)))}]}class Yn extends t{constructor(e){super(),n(this,e,Xn,jn,a,{size:4,direction:5,open:0,light:6,flipped:7,persistentClickItems:8,menuOptionsClass:9,icon:1,iconClass:10,iconDescription:11,id:12,buttonRef:2,menuRef:3},null,[-1,-1])}}function Wn(e,t,n){const a=e.slice();return a[3]=t[n],a[4]=t,a[5]=n,a}function Qn(e){let t,n,a,s,p;function c(t){e[1](t,e[3])}let m={labelText:e[3].value};return void 0!==e[3].selected&&(m.checked=e[3].selected),n=new Me({props:m}),h.push((()=>v(n,"checked",c))),n.$on("change",e[2]),{c(){t=r("li"),f(n.$$.fragment),s=$(),i(t,"class","pipen-report-column-selector-item svelte-147jw2r")},m(e,a){l(e,t,a),b(n,t,null),x(t,s),p=!0},p(t,s){e=t;const r={};1&s&&(r.labelText=e[3].value),!a&&1&s&&(a=!0,r.checked=e[3].selected,y((()=>a=!1))),n.$set(r)},i(e){p||(o(n.$$.fragment,e),p=!0)},o(e){d(n.$$.fragment,e),p=!1},d(e){e&&_(t),C(n)}}}function Jn(e){let t,n,a=s(e[0]),r=[];for(let t=0;t<a.length;t+=1)r[t]=Qn(Wn(e,a,t));const i=e=>d(r[e],1,1,(()=>{r[e]=null}));return{c(){for(let e=0;e<r.length;e+=1)r[e].c();t=g()},m(e,a){for(let t=0;t<r.length;t+=1)r[t]&&r[t].m(e,a);l(e,t,a),n=!0},p(e,n){if(1&n){let l;for(a=s(e[0]),l=0;l<a.length;l+=1){const s=Wn(e,a,l);r[l]?(r[l].p(s,n),o(r[l],1)):(r[l]=Qn(s),r[l].c(),o(r[l],1),r[l].m(t.parentNode,t))}for(p(),l=a.length;l<r.length;l+=1)i(l);c()}},i(e){if(!n){for(let e=0;e<a.length;e+=1)o(r[e]);n=!0}},o(e){r=r.filter(Boolean);for(let e=0;e<r.length;e+=1)d(r[e]);n=!1},d(e){e&&_(t),m(r,e)}}}function Zn(e){let t,n;return t=new Yn({props:{class:"pipen-report-overflowmenu",persistentClickItems:!0,size:"sm",icon:Je,iconDescription:"Select visible columns ...",$$slots:{default:[Jn]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,[n]){const a={};65&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function ea(e,t,n){let{allColumns:a}=t;return e.$$set=e=>{"allColumns"in e&&n(0,a=e.allColumns)},[a,function(t,s){e.$$.not_equal(s.selected,t)&&(s.selected=t,n(0,a))},()=>{n(0,a)}]}class ta extends t{constructor(e){super(),n(this,e,ea,Zn,a,{allColumns:0})}}function na(e){let t,n;return t=new Se({props:{iconDescription:"Right click and save as to download the entire data",tooltipPosition:"top",icon:Ze,href:e[2]}}),t.$on("click",e[13]),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};4&n&&(a.href=e[2]),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function aa(e){let t,n,a,s,r,i,m,u,S,x;function A(t){e[17](t)}let M={persistent:!0};function P(t){e[18](t)}void 0!==e[7]&&(M.value=e[7]),t=new Ee({props:M}),h.push((()=>v(t,"value",A)));let w={};void 0!==e[6]&&(w.allColumns=e[6]),s=new ta({props:w}),h.push((()=>v(s,"allColumns",P))),m=new Se({props:{iconDescription:`Download current ${e[11].length} records (may be partial of the entire data)`,tooltipPosition:"top",icon:Ie}}),m.$on("click",e[14]);let T=!!e[2]&&na(e);return{c(){f(t.$$.fragment),a=$(),f(s.$$.fragment),i=$(),f(m.$$.fragment),u=$(),T&&T.c(),S=g()},m(e,n){b(t,e,n),l(e,a,n),b(s,e,n),l(e,i,n),b(m,e,n),l(e,u,n),T&&T.m(e,n),l(e,S,n),x=!0},p(e,a){const i={};!n&&128&a&&(n=!0,i.value=e[7],y((()=>n=!1))),t.$set(i);const l={};!r&&64&a&&(r=!0,l.allColumns=e[6],y((()=>r=!1))),s.$set(l);const _={};2048&a&&(_.iconDescription=`Download current ${e[11].length} records (may be partial of the entire data)`),m.$set(_),e[2]?T?(T.p(e,a),4&a&&o(T,1)):(T=na(e),T.c(),o(T,1),T.m(S.parentNode,S)):T&&(p(),d(T,1,1,(()=>{T=null})),c())},i(e){x||(o(t.$$.fragment,e),o(s.$$.fragment,e),o(m.$$.fragment,e),o(T),x=!0)},o(e){d(t.$$.fragment,e),d(s.$$.fragment,e),d(m.$$.fragment,e),d(T),x=!1},d(e){e&&(_(a),_(i),_(u),_(S)),C(t,e),C(s,e),C(m,e),T&&T.d(e)}}}function sa(e){let t,n;return t=new Le({props:{$$slots:{default:[aa]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};134219972&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function ra(e){let t,n;return t=new Te({props:{size:e[12],$$slots:{default:[sa]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};134219972&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function ia(e){let t;return{c(){t=r("p"),t.textContent='You must use "Save as" or "Save link as" from the context menu (by right-clicking the button) to download the entire data.'},m(e,n){l(e,t,n)},p:P,d(e){e&&_(t)}}}function la(e){let t,n,a,s,r,i,p,c,m;const u=[{size:e[0]},{zebra:e[3]},{sortable:e[4]},{page:e[8]},{pageSize:e[9]},{rows:e[11]},{headers:e[1]},e[15]];let g={$$slots:{default:[ra]},$$scope:{ctx:e}};for(let e=0;e<u.length;e+=1)g=N(g,u[e]);function S(t){e[19](t)}function x(t){e[20](t)}t=new Pe({props:g});let A={pageSizes:e[5],totalItems:e[11].length};function M(t){e[21](t)}void 0!==e[9]&&(A.pageSize=e[9]),void 0!==e[8]&&(A.page=e[8]),a=new we({props:A}),h.push((()=>v(a,"pageSize",S))),h.push((()=>v(a,"page",x)));let P={passiveModal:!0,modalHeading:"Downloading entire data",$$slots:{default:[ia]},$$scope:{ctx:e}};return void 0!==e[10]&&(P.open=e[10]),p=new _e({props:P}),h.push((()=>v(p,"open",M))),p.$on("open",e[22]),p.$on("close",e[23]),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment),i=$(),f(p.$$.fragment)},m(e,s){b(t,e,s),l(e,n,s),b(a,e,s),l(e,i,s),b(p,e,s),m=!0},p(e,[n]){const i=35611&n?U(u,[1&n&&{size:e[0]},8&n&&{zebra:e[3]},16&n&&{sortable:e[4]},256&n&&{page:e[8]},512&n&&{pageSize:e[9]},2048&n&&{rows:e[11]},2&n&&{headers:e[1]},32768&n&&R(e[15])]):{};134219972&n&&(i.$$scope={dirty:n,ctx:e}),t.$set(i);const l={};32&n&&(l.pageSizes=e[5]),2048&n&&(l.totalItems=e[11].length),!s&&512&n&&(s=!0,l.pageSize=e[9],y((()=>s=!1))),!r&&256&n&&(r=!0,l.page=e[8],y((()=>r=!1))),a.$set(l);const o={};134217728&n&&(o.$$scope={dirty:n,ctx:e}),!c&&1024&n&&(c=!0,o.open=e[10],y((()=>c=!1))),p.$set(o)},i(e){m||(o(t.$$.fragment,e),o(a.$$.fragment,e),o(p.$$.fragment,e),m=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),d(p.$$.fragment,e),m=!1},d(e){e&&(_(n),_(i)),C(t,e),C(a,e),C(p,e)}}}function oa(e,t,n){const a=["size","src","zebra","sortable","rows","headers","pageSizes"];let s=D(t,a),{size:r="sm"}=t,{src:i}=t;r={sm:"short",small:"short",short:"short",md:"medium",medium:"medium",lg:"tall",large:"tall",tall:"tall",compact:"compact"}[r];let l={short:"sm",compact:"sm",medium:"default",tall:"default"}[r],{zebra:o=!0}=t,{sortable:p=!0}=t,{rows:c=[]}=t,{headers:d=[]}=t,{pageSizes:_=[20,50,75,100]}=t,m=d.map((e=>({...e,selected:!0}))),u=1,g=_[0],f="",b=!1,C=c;return e.$$set=e=>{t=N(N({},t),I(e)),n(15,s=D(t,a)),"size"in e&&n(0,r=e.size),"src"in e&&n(2,i=e.src),"zebra"in e&&n(3,o=e.zebra),"sortable"in e&&n(4,p=e.sortable),"rows"in e&&n(16,c=e.rows),"headers"in e&&n(1,d=e.headers),"pageSizes"in e&&n(5,_=e.pageSizes)},e.$$.update=()=>{64&e.$$.dirty&&n(1,d=m.filter((e=>e.selected))),65664&e.$$.dirty&&(f.length>0?n(11,C=c.filter((e=>Object.values(e).filter((e=>!!e&&e.toString().includes(f))).length>0))):n(11,C=c))},[r,d,i,o,p,_,m,f,u,g,b,C,l,e=>{e.preventDefault(),n(10,b=!0)},()=>{((e,t)=>{const n=document.createElement("a");n.href=e,n.download=t,document.body.appendChild(n),n.click(),document.body.removeChild(n)})("data:application/vnd.ms-excel;charset=utf-8,\ufeff"+encodeURIComponent(function(){const e=d.map((e=>e.value)),t=[];return C.forEach((e=>{const n=d.map((t=>`"${e[t.key]}"`));t.push(n.join(","))})),e.join(",")+"\r\n"+t.join("\r\n")}()),i&&i.length>0?i.split("/").pop().split(".")[0]+".xls":"data-subset.xls")},s,c,function(e){f=e,n(7,f)},function(e){m=e,n(6,m)},function(e){g=e,n(9,g)},function(e){u=e,n(8,u)},function(e){b=e,n(10,b)},function(t){Z.call(this,e,t)},function(t){Z.call(this,e,t)}]}class pa extends t{constructor(e){super(),n(this,e,oa,la,a,{size:0,src:2,zebra:3,sortable:4,rows:16,headers:1,pageSizes:5})}}function ca(e){let t,n,a;const s=[{headers:e[0]},{rows:e[1]},e[3]];let i={};for(let e=0;e<s.length;e+=1)i=N(i,s[e]);n=new pa({props:i});let p=[e[2]],c={};for(let e=0;e<p.length;e+=1)c=N(c,p[e]);return{c(){t=r("div"),f(n.$$.fragment),F(t,c)},m(e,s){l(e,t,s),b(n,t,null),a=!0},p(e,a){const r=11&a?U(s,[1&a&&{headers:e[0]},2&a&&{rows:e[1]},8&a&&R(e[3])]):{};n.$set(r),F(t,c=U(p,[4&a&&e[2]]))},i(e){a||(o(n.$$.fragment,e),a=!0)},o(e){d(n.$$.fragment,e),a=!1},d(e){e&&_(t),C(n)}}}function da(e){let t,n;const a=[{headers:e[0]},{rows:e[1]},e[3]];let s={};for(let e=0;e<a.length;e+=1)s=N(s,a[e]);return t=new pa({props:s}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const s=11&n?U(a,[1&n&&{headers:e[0]},2&n&&{rows:e[1]},8&n&&R(e[3])]):{};t.$set(s)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function _a(e){let t,n,a,s;const r=[da,ca],i=[];function m(e,t){return e[2]?1:0}return t=m(e),n=i[t]=r[t](e),{c(){n.c(),a=g()},m(e,n){i[t].m(e,n),l(e,a,n),s=!0},p(e,[s]){let l=t;t=m(e),t===l?i[t].p(e,s):(p(),d(i[l],1,1,(()=>{i[l]=null})),c(),n=i[t],n?n.p(e,s):(n=i[t]=r[t](e),n.c()),o(n,1),n.m(a.parentNode,a))},i(e){s||(o(n),s=!0)},o(e){d(n),s=!1},d(e){e&&_(a),i[t].d(e)}}}function ma(e,t,n){const a=["frameProps","data","headers","rows"];let s=D(t,a),{frameProps:r={class:"pipen-report-datatable-frame"}}=t,{data:i={}}=t,{headers:l=[]}=t,{rows:o=[]}=t;if(l.length+o.length>0&&i.length>0)throw"Can only specify `data` or `headers`/`rows` for `DataTable`!";return i.length>0&&(o=i,l=Object.keys(o[0]).filter((e=>"id"!==e)).map((e=>({key:e,value:e})))),e.$$set=e=>{t=N(N({},t),I(e)),n(3,s=D(t,a)),"frameProps"in e&&n(2,r=e.frameProps),"data"in e&&n(4,i=e.data),"headers"in e&&n(0,l=e.headers),"rows"in e&&n(1,o=e.rows)},[l,o,r,s,i]}class ua extends t{constructor(e){super(),n(this,e,ma,_a,a,{frameProps:2,data:4,headers:0,rows:1})}}function ga(e){let t,n;return{c(){t=r("div"),n=S(e[0]),i(t,"class","pipen-report-descr-title svelte-1xzxuva")},m(e,a){l(e,t,a),x(t,n)},p(e,t){1&t&&A(n,e[0])},d(e){e&&_(t)}}}function fa(e){let t,n,a,s=e[0]&&ga(e);const i=e[3].default,p=w(i,e,e[2],null);let c=[e[1]],m={};for(let e=0;e<c.length;e+=1)m=N(m,c[e]);return{c(){t=r("div"),s&&s.c(),n=$(),p&&p.c(),F(t,m),G(t,"svelte-1xzxuva",!0)},m(e,r){l(e,t,r),s&&s.m(t,null),x(t,n),p&&p.m(t,null),a=!0},p(e,[r]){e[0]?s?s.p(e,r):(s=ga(e),s.c(),s.m(t,n)):s&&(s.d(1),s=null),p&&p.p&&(!a||4&r)&&T(p,i,e,e[2],a?E(i,e[2],r,null):L(e[2]),null),F(t,m=U(c,[2&r&&e[1]])),G(t,"svelte-1xzxuva",!0)},i(e){a||(o(p,e),a=!0)},o(e){d(p,e),a=!1},d(e){e&&_(t),s&&s.d(),p&&p.d(e)}}}function ba(e,t,n){const a=["title"];let s=D(t,a),{$$slots:r={},$$scope:i}=t,{title:l}=t,{klass:o,...p}=s;return o=o?`pipen-report-descr ${o}`:"pipen-report-descr",p||(p={}),p.class=o,e.$$set=e=>{t=N(N({},t),I(e)),n(5,s=D(t,a)),"title"in e&&n(0,l=e.title),"$$scope"in e&&n(2,i=e.$$scope)},[l,p,i,r]}class Ca extends t{constructor(e){super(),n(this,e,ba,fa,a,{title:0})}}function $a(e){let t,n,a,s;return t=new ge({props:{label:"Plot"}}),a=new ge({props:{label:"Table"}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p:P,i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function ha(e){let t;return{c(){t=S("This table presents a comprehensive summary of the top enriched pathways derived from the fgsea. Each row corresponds to a pathway, and the gene ranks are shown based on the ranking metric used in the analysis. The enrichment score, p-value, and adjusted p-value are also provided to assess the significance of the enrichment.")},m(e,n){l(e,t,n)},d(e){e&&_(t)}}}function va(e){let t,n,a,s;return t=new Ca({props:{$$slots:{default:[ha]},$$scope:{ctx:e}}}),a=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells-1/Colitis_vs_Control/gsea_table.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells-1/Colitis_vs_Control/gsea_table.pdf",width:1e3,height:600}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function ya(e){let t;return{c(){t=S("This plot represents the GSEA results for a specified gene set, illustrating the distribution and impact of the gene set along the ranked list of genes. The running enrichment score curve shows the cumulative enrichment score as genes from the input list are encountered. Positive peaks on the curve indicate regions where members of the gene set are predominantly found.")},m(e,n){l(e,t,n)},d(e){e&&_(t)}}}function Sa(e){let t,n,a,s;return t=new Ca({props:{$$slots:{default:[ya]},$$scope:{ctx:e}}}),a=new ua({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells-1/Colitis_vs_Control/fgsea.txt",data:[{pathway:"Oxidative phosphorylation",pval:0,padj:0,log2err:.7749,ES:.5865,NES:2.1348,size:120,leadingEdge:"NDUFS5,NDUFA6,ATP6V0B,UQCRFS1,ATP5ME,UQCR10,ATP5MG,COX5A,NDUFB11,NDUFB4,NDUFB5,UQCRQ,COX7B,COX10,ATP5F1D,ATP5PD,NDUFA12,NDUFA2,ATP5MF,ATP5MC3,COX7A2,ATP6V1F,NDUFA1,COX7A2L,ATP6V0D1,NDUFB8,NDUFA7,NDUFB1,COX7C,UQCRH,NDUFS2,COX8A,PPA1,ATP5F1C,UQCR11,ATP5F1E,COX5B,COX6C,NDUFB10,LHPP,NDUFB3,NDUFB9,TCIRG1,ATP5MC2,NDUFS4,UQCRHL,ATP6V1A,SDHC,NDUFA10,UQCRC2,SDHA,ATP5F1A,COX6A1,COX4I1,NDUFA8,CYC1,NDUFC1,COX6B1,NDUFS3,NDUFV2,SDHD,ATP6V1C2,NDUFS8,ATP6V0C,NDUFA9,NDUFA3,COX11,NDUFAB1,ATP6V1D,NDUFC2,SDHB,ATP12A,NDUFB2,ATP5PB,NDUFS7,NDUFB6,NDUFA13",id:0},{pathway:"Ascorbate and aldarate metabolism",pval:0,padj:0,log2err:.675,ES:-.7893,NES:-2.3461,size:27,leadingEdge:"UGDH,ALDH9A1,ALDH1B1,ALDH3A2",id:1},{pathway:"Starch and sucrose metabolism",pval:3e-4,padj:.0022,log2err:.4985,ES:-.598,NES:-1.8959,size:35,leadingEdge:"GAA,AMY2B,GBE1,GPI,AGL,PYGB,UGP2",id:2},{pathway:"Pentose and glucuronate interconversions",pval:.001,padj:.005,log2err:.4551,ES:-.5919,NES:-1.8594,size:34,leadingEdge:"UGDH,UGP2,DCXR,SORD,CRYL1,AKR1B1,XYLB,GUSB",id:3},{pathway:"Amino sugar and nucleotide sugar metabolism",pval:.0023,padj:.0092,log2err:.4317,ES:-.5057,NES:-1.7041,size:48,leadingEdge:"FPGT,CYB5R1,GNPNAT1,UGDH,GNPDA2,NPL,AMDHD2,GPI,CYB5R3,HEXB,FUK,NAGK,UGP2,PGM3,GFPT1,PMM2,CHIT1,GALE,MPI,GNE,PMM1",id:4},{pathway:"Sulfur metabolism",pval:.0054,padj:.0181,log2err:.407,ES:.7874,NES:1.6843,size:9,leadingEdge:"SQOR,ETHE1,MPST,TST,BPNT1,IMPAD1",id:5},{pathway:"Fatty acid degradation",pval:.0227,padj:.0648,log2err:.3525,ES:-.4309,NES:-1.4278,size:44,leadingEdge:"ACADSB,ACSL3,ALDH9A1,ACADVL,ECI2,CPT1A,ALDH1B1,ALDH3A2,ACAT2,CPT1C,ECI1,ACAA2,HADH,ECHS1,CPT2",id:6},{pathway:"Fructose and mannose metabolism",pval:.0611,padj:.1528,log2err:.3218,ES:-.4431,NES:-1.3868,size:33,leadingEdge:"ENOSF1,FPGT,PFKL,KHK,FUK,ALDOC,ALDOA,PFKFB4,SORD,PMM2,AKR1B1,MPI,FBP1,PMM1",id:7},{pathway:"Glycolysis / Gluconeogenesis",pval:.1197,padj:.2404,log2err:.2114,ES:-.3337,NES:-1.1998,size:68,leadingEdge:"ENO2,ALDH9A1,PKM,PFKL,GPI,ALDOC,ALDOA,DLAT,ENO1,ALDH1B1,ALDH3A2",id:8},{pathway:"Butanoate metabolism",pval:.1202,padj:.2404,log2err:.2066,ES:-.4203,NES:-1.2528,size:28,leadingEdge:"ACSM3,AACS,ACAT2,BDH2,L2HGDH,HADH,ECHS1,BDH1",id:9},{pathway:"Pyruvate metabolism",pval:.2487,padj:.4522,log2err:.1388,ES:-.349,NES:-1.1319,size:39,leadingEdge:"ALDH9A1,PKM,GLO1,FH,ACACA,DLAT,ALDH1B1,ME3,ALDH3A2,ACAT2,ACYP1,ACSS2",id:10},{pathway:"Fatty acid elongation",pval:.4709,padj:.7848,log2err:.0956,ES:-.3272,NES:-.9889,size:30,leadingEdge:"HACD3,HSD17B12,HACD2,TECR,PPT1,ACOT4,MECR,HACD1,ACAA2,HADH,ECHS1",id:11},{pathway:"Pentose phosphate pathway",pval:.5519,padj:.8491,log2err:.0865,ES:-.3108,NES:-.9393,size:30,leadingEdge:"PRPS1,PFKL,GPI,ALDOC,ALDOA,RPIA",id:12},{pathway:"Inositol phosphate metabolism",pval:.7709,padj:.9639,log2err:.0468,ES:.248,NES:.845,size:73,leadingEdge:"MTMR3,MTMR14,PI4KA,INPPL1,INPP5K,SYNJ2,PTEN,PIK3CA,CDIPT,INPP5D,PIK3C2B,PIK3C3,PLCD1,PIP5K1A,ITPKB,PIP5K1C,IMPAD1",id:13},{pathway:"Galactose metabolism",pval:.7924,padj:.9639,log2err:.0677,ES:-.2728,NES:-.8288,size:31,leadingEdge:"GAA,PFKL,GLA,UGP2,AKR1B1,GANC,GALE",id:14},{pathway:"Citrate cycle (TCA cycle)",pval:.8122,padj:.9639,log2err:.0463,ES:.2707,NES:.7818,size:30,leadingEdge:"IDH2,IDH3G,SUCLG1,IDH1,PDHB,SDHC,MDH2,SDHA,SDHD,SDHB,DLD,OGDH",id:15},{pathway:"Glyoxylate and dicarboxylate metabolism",pval:.9156,padj:.9639,log2err:.0612,ES:-.2457,NES:-.7325,size:28,leadingEdge:"ACO1,HYI,GCSH,CAT,ACAT2,MUT",id:16},{pathway:"Fatty acid biosynthesis",pval:.9315,padj:.9639,log2err:.0582,ES:-.2746,NES:-.6798,size:13,leadingEdge:"ACSL3,ACACA",id:17},{pathway:"Propanoate metabolism",pval:.9425,padj:.9639,log2err:.0587,ES:-.2312,NES:-.7126,size:32,leadingEdge:"MLYCD,BCKDHB,ACACA,SUCLA2,ACAT2,HIBCH,ACSS2,MUT,ECHS1",id:18},{pathway:"Nitrogen metabolism",pval:.9639,padj:.9639,log2err:.0558,ES:-.2382,NES:-.6425,size:17,leadingEdge:"CA13,CA9,CA2,CA1,CA4,CA8,CA5A,CA3,CA12,CA7,CPS1,CA6,CA14,GLUD2,GLUL,GLUD1,CA5B",id:19}]}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function xa(e){let t,n,a,s;return t=new fe({props:{$$slots:{default:[va]},$$scope:{ctx:e}}}),a=new fe({props:{$$slots:{default:[Sa]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function Aa(e){let t,n;return t=new ue({props:{$$slots:{content:[xa],default:[$a]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function Ma(e){let t,n,a,s,p,c,m,u,g,h,v,y,S,A,M,w,T,L,E,N,U,R,D,I,B,F,G,H,V,z,O,k,q,K,j,X,Y,W,Q,J,Z,te,ne,ae,se,re,ie,le,oe,pe,ce;return a=new Ca({props:{title:"Oxidative phosphorylation",class:"pipen-report-table-image-descr"}}),p=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells-1/Colitis_vs_Control/fgsea_Oxidative-phosphorylation.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells-1/Colitis_vs_Control/fgsea_Oxidative-phosphorylation.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),u=new Ca({props:{title:"Ascorbate and aldarate metabolism",class:"pipen-report-table-image-descr"}}),h=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells-1/Colitis_vs_Control/fgsea_Ascorbate-and-aldarate-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells-1/Colitis_vs_Control/fgsea_Ascorbate-and-aldarate-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),S=new Ca({props:{title:"Starch and sucrose metabolism",class:"pipen-report-table-image-descr"}}),M=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells-1/Colitis_vs_Control/fgsea_Starch-and-sucrose-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells-1/Colitis_vs_Control/fgsea_Starch-and-sucrose-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),L=new Ca({props:{title:"Pentose and glucuronate interconversions",class:"pipen-report-table-image-descr"}}),N=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells-1/Colitis_vs_Control/fgsea_Pentose-and-glucuronate-interconversions.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells-1/Colitis_vs_Control/fgsea_Pentose-and-glucuronate-interconversions.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),D=new Ca({props:{title:"Amino sugar and nucleotide sugar metabolism",class:"pipen-report-table-image-descr"}}),B=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells-1/Colitis_vs_Control/fgsea_Amino-sugar-and-nucleotide-sugar-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells-1/Colitis_vs_Control/fgsea_Amino-sugar-and-nucleotide-sugar-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),H=new Ca({props:{title:"Sulfur metabolism",class:"pipen-report-table-image-descr"}}),z=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells-1/Colitis_vs_Control/fgsea_Sulfur-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells-1/Colitis_vs_Control/fgsea_Sulfur-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),q=new Ca({props:{title:"Fatty acid degradation",class:"pipen-report-table-image-descr"}}),j=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells-1/Colitis_vs_Control/fgsea_Fatty-acid-degradation.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells-1/Colitis_vs_Control/fgsea_Fatty-acid-degradation.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),W=new Ca({props:{title:"Fructose and mannose metabolism",class:"pipen-report-table-image-descr"}}),J=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells-1/Colitis_vs_Control/fgsea_Fructose-and-mannose-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells-1/Colitis_vs_Control/fgsea_Fructose-and-mannose-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new Ca({props:{title:"Glycolysis / Gluconeogenesis",class:"pipen-report-table-image-descr"}}),se=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells-1/Colitis_vs_Control/fgsea_Glycolysis-Gluconeogenesis.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells-1/Colitis_vs_Control/fgsea_Glycolysis-Gluconeogenesis.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),le=new Ca({props:{title:"Butanoate metabolism",class:"pipen-report-table-image-descr"}}),pe=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells-1/Colitis_vs_Control/fgsea_Butanoate-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells-1/Colitis_vs_Control/fgsea_Butanoate-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=r("div"),n=r("div"),f(a.$$.fragment),s=$(),f(p.$$.fragment),c=$(),m=r("div"),f(u.$$.fragment),g=$(),f(h.$$.fragment),v=$(),y=r("div"),f(S.$$.fragment),A=$(),f(M.$$.fragment),w=$(),T=r("div"),f(L.$$.fragment),E=$(),f(N.$$.fragment),U=$(),R=r("div"),f(D.$$.fragment),I=$(),f(B.$$.fragment),F=$(),G=r("div"),f(H.$$.fragment),V=$(),f(z.$$.fragment),O=$(),k=r("div"),f(q.$$.fragment),K=$(),f(j.$$.fragment),X=$(),Y=r("div"),f(W.$$.fragment),Q=$(),f(J.$$.fragment),Z=$(),te=r("div"),f(ne.$$.fragment),ae=$(),f(se.$$.fragment),re=$(),ie=r("div"),f(le.$$.fragment),oe=$(),f(pe.$$.fragment),i(t,"class","pipen-report-table-of-images"),ee(t,"grid-template-columns","repeat(2, auto)")},m(e,r){l(e,t,r),x(t,n),b(a,n,null),x(n,s),b(p,n,null),x(t,c),x(t,m),b(u,m,null),x(m,g),b(h,m,null),x(t,v),x(t,y),b(S,y,null),x(y,A),b(M,y,null),x(t,w),x(t,T),b(L,T,null),x(T,E),b(N,T,null),x(t,U),x(t,R),b(D,R,null),x(R,I),b(B,R,null),x(t,F),x(t,G),b(H,G,null),x(G,V),b(z,G,null),x(t,O),x(t,k),b(q,k,null),x(k,K),b(j,k,null),x(t,X),x(t,Y),b(W,Y,null),x(Y,Q),b(J,Y,null),x(t,Z),x(t,te),b(ne,te,null),x(te,ae),b(se,te,null),x(t,re),x(t,ie),b(le,ie,null),x(ie,oe),b(pe,ie,null),ce=!0},p:P,i(e){ce||(o(a.$$.fragment,e),o(p.$$.fragment,e),o(u.$$.fragment,e),o(h.$$.fragment,e),o(S.$$.fragment,e),o(M.$$.fragment,e),o(L.$$.fragment,e),o(N.$$.fragment,e),o(D.$$.fragment,e),o(B.$$.fragment,e),o(H.$$.fragment,e),o(z.$$.fragment,e),o(q.$$.fragment,e),o(j.$$.fragment,e),o(W.$$.fragment,e),o(J.$$.fragment,e),o(ne.$$.fragment,e),o(se.$$.fragment,e),o(le.$$.fragment,e),o(pe.$$.fragment,e),ce=!0)},o(e){d(a.$$.fragment,e),d(p.$$.fragment,e),d(u.$$.fragment,e),d(h.$$.fragment,e),d(S.$$.fragment,e),d(M.$$.fragment,e),d(L.$$.fragment,e),d(N.$$.fragment,e),d(D.$$.fragment,e),d(B.$$.fragment,e),d(H.$$.fragment,e),d(z.$$.fragment,e),d(q.$$.fragment,e),d(j.$$.fragment,e),d(W.$$.fragment,e),d(J.$$.fragment,e),d(ne.$$.fragment,e),d(se.$$.fragment,e),d(le.$$.fragment,e),d(pe.$$.fragment,e),ce=!1},d(e){e&&_(t),C(a),C(p),C(u),C(h),C(S),C(M),C(L),C(N),C(D),C(B),C(H),C(z),C(q),C(j),C(W),C(J),C(ne),C(se),C(le),C(pe)}}}function Pa(e){let t,n,a,s;return t=new Ue({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[Aa]},$$scope:{ctx:e}}}),a=new Ue({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[Ma]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function wa(e){let t,n,a,s;return t=new ge({props:{label:"Plot"}}),a=new ge({props:{label:"Table"}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p:P,i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function Ta(e){let t;return{c(){t=S("This table presents a comprehensive summary of the top enriched pathways derived from the fgsea. Each row corresponds to a pathway, and the gene ranks are shown based on the ranking metric used in the analysis. The enrichment score, p-value, and adjusted p-value are also provided to assess the significance of the enrichment.")},m(e,n){l(e,t,n)},d(e){e&&_(t)}}}function La(e){let t,n,a,s;return t=new Ca({props:{$$slots:{default:[Ta]},$$scope:{ctx:e}}}),a=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tcm-Naive-helper-T-cells/Colitis_vs_Control/gsea_table.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tcm-Naive-helper-T-cells/Colitis_vs_Control/gsea_table.pdf",width:1e3,height:600}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function Ea(e){let t;return{c(){t=S("This plot represents the GSEA results for a specified gene set, illustrating the distribution and impact of the gene set along the ranked list of genes. The running enrichment score curve shows the cumulative enrichment score as genes from the input list are encountered. Positive peaks on the curve indicate regions where members of the gene set are predominantly found.")},m(e,n){l(e,t,n)},d(e){e&&_(t)}}}function Na(e){let t,n,a,s;return t=new Ca({props:{$$slots:{default:[Ea]},$$scope:{ctx:e}}}),a=new ua({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tcm-Naive-helper-T-cells/Colitis_vs_Control/fgsea.txt",data:[{pathway:"Nitrogen metabolism",pval:7e-4,padj:.0127,log2err:.4773,ES:.7027,NES:2.2075,size:17,leadingEdge:"GLUL,CPS1,CA5B,GLUD2,CA13",id:0},{pathway:"Fatty acid degradation",pval:.0147,padj:.1097,log2err:.3807,ES:-.575,NES:-1.3259,size:44,leadingEdge:"ALDH9A1,ACADVL,ACOX1,ADH5,ACSL3,HADHA,CPT1A,ACADSB,ACAA2,ALDH3A2,ACSL1,GCDH,CPT1C,HADHB,ACADS,ACSL5,ECHS1",id:1},{pathway:"Citrate cycle (TCA cycle)",pval:.0173,padj:.1097,log2err:.3525,ES:-.6147,NES:-1.3822,size:30,leadingEdge:"DLST,ACO1,SUCLA2,CS,FH,MDH1,ACLY,ACO2,SUCLG2,DLAT,PCK2,IDH3B,SDHC,PDHB,DLD,IDH3G,OGDH",id:2},{pathway:"Inositol phosphate metabolism",pval:.05,padj:.2045,log2err:.1999,ES:-.5203,NES:-1.2411,size:73,leadingEdge:"PIK3CG,PLCB2,INPP5B,IMPA1,MTMR2,PIP4K2C,INPP4A,MINPP1,MTMR4,IPPK,PLCG1,MTMR6,SACM1L,PI4K2A,PIP5K1A,MTMR1,MTM1,MTMR8,SYNJ1,ISYNA1,PIK3CD,PIK3C2A,PI4KA,ALDH6A1,PIK3CB,PIK3C2B,ITPK1,INPP4B,INPP5D,INPP5F,PTEN,PLCH2,PLCB4,PIK3CA,INPP5E,PI4K2B,PIK3C3",id:3},{pathway:"Glycolysis / Gluconeogenesis",pval:.07,padj:.2045,log2err:.1669,ES:-.5115,NES:-1.2161,size:68,leadingEdge:"ALDH9A1,PKM,ADH5,PGM2,MINPP1,GPI,PFKL,ENO2,ALDH3A2,ALDOA,ENO1,DLAT,GCK,PCK2,PGM1,ADPGK,PGAM1,PDHB,ALDOC,DLD,ACSS1,GAPDH,ALDH1B1,HK1",id:4},{pathway:"Ascorbate and aldarate metabolism",pval:.0779,padj:.2045,log2err:.1596,ES:-.5761,NES:-1.2811,size:27,leadingEdge:"ALDH9A1,ALDH3A2",id:5},{pathway:"Propanoate metabolism",pval:.0842,padj:.2045,log2err:.1521,ES:-.5589,NES:-1.2662,size:32,leadingEdge:"MLYCD,MUT,HADHA,SUCLA2,PCCB,HIBCH,PCCA,DBT,SUCLG2,ECHS1,BCKDHB,ALDH6A1,DLD,ACSS1,BCKDHA,LDHB,SUCLG1",id:6},{pathway:"Starch and sucrose metabolism",pval:.0861,padj:.2045,log2err:.1502,ES:-.5468,NES:-1.2474,size:35,leadingEdge:"GAA,PGM2,PYGB,GPI,AGL,GYS1,AMY2B,GCK,GANC,PGM1",id:7},{pathway:"Butanoate metabolism",pval:.1073,padj:.2115,log2err:.1336,ES:-.5611,NES:-1.2556,size:28,leadingEdge:"ALDH5A1,HADHA,OXCT1,AACS,BDH2,ACADS,HMGCL,ECHS1",id:8},{pathway:"Glyoxylate and dicarboxylate metabolism",pval:.1113,padj:.2115,log2err:.1308,ES:-.5596,NES:-1.2522,size:28,leadingEdge:"CAT,MUT,ACO1,PCCB,CS,PCCA,MDH1,GCSH,ACO2,HYI",id:9},{pathway:"Fatty acid elongation",pval:.261,padj:.4507,log2err:.0779,ES:-.5069,NES:-1.1396,size:30,leadingEdge:"HACD2,HADHA,HACD3,ACAA2,HSD17B12,ABHD17B,PPT1,ACOT2,HADHB,HACD4,ECHS1",id:10},{pathway:"Amino sugar and nucleotide sugar metabolism",pval:.3048,padj:.4826,log2err:.0693,ES:-.4706,NES:-1.0934,size:48,leadingEdge:"FPGT,GNPNAT1,GNPDA2,PGM2,HEXA,GPI,RENBP,HEXB,GNE,GCK,PGM1,CYB5R4,GMPPA,AMDHD2,CHIT1,GALT,GNPDA1,HK1,MPI,PMM2,GALE,NPL,CMAS,CYB5RL,CYB5R3",id:11},{pathway:"Pentose phosphate pathway",pval:.4444,padj:.6496,log2err:.052,ES:-.4612,NES:-1.0371,size:30,leadingEdge:"PGM2,GPI,PFKL,PRPS1,IDNK,G6PD,ALDOA,RPIA,PRPS2,PGM1,PGD,ALDOC",id:12},{pathway:"Pyruvate metabolism",pval:.5302,padj:.7196,log2err:.0434,ES:-.4332,NES:-.9964,size:39,leadingEdge:"ALDH9A1,PKM,ME2,FH,ALDH3A2,MDH1,DLAT,PCK2,PDHB,DLD,ACSS1,ALDH1B1,LDHB,ME3",id:13},{pathway:"Sulfur metabolism",pval:.6198,padj:.7497,log2err:.0417,ES:-.4901,NES:-.9263,size:9,leadingEdge:"BPNT1,PAPSS1",id:14},{pathway:"Galactose metabolism",pval:.6314,padj:.7497,log2err:.0358,ES:-.4156,NES:-.9355,size:31,leadingEdge:"GAA,PGM2,PFKL,GLB1,GCK,GANC,AKR1B1,PGM1,GALT,HK1,GLA,GALE",id:15},{pathway:"Fructose and mannose metabolism",pval:.7771,padj:.8246,log2err:.0252,ES:-.3765,NES:-.8549,size:33,leadingEdge:"FPGT,ENOSF1,PFKL,ALDOA,TIGAR,AKR1B1,GMPPA,ALDOC,SORD,HK1,MPI,PMM2",id:16},{pathway:"Pentose and glucuronate interconversions",pval:.7812,padj:.8246,log2err:.0249,ES:-.3744,NES:-.8511,size:34,leadingEdge:"GUSB,AKR1B1,CRYL1,SORD,DHDH,ISPD,XYLB,KL,RPE,DCXR,UGDH",id:17},{pathway:"Fatty acid biosynthesis",pval:.9342,padj:.9342,log2err:.018,ES:-.3187,NES:-.6484,size:13,leadingEdge:"ACSL3,ACSL1,ACSL5,ACSL4",id:18},{pathway:"Oxidative phosphorylation",pval:null,padj:null,log2err:null,ES:.2274,NES:null,size:120,leadingEdge:"NDUFS5,ATP5PD,ATP6V0C,UQCRH,COX7C,NDUFB5,ATP5ME,NDUFA2,COX17,NDUFB4,UQCRQ,UQCRB,ATP5F1E,NDUFA3,UQCR10,NDUFA6,COX7B,NDUFA1,NDUFB8,NDUFB1,COX7A2,NDUFA7,NDUFA12,PPA1,NDUFC1,COX5B,ATP6V0B,UQCRHL,ATP5MG,UQCR11,UQCRFS1,NDUFB6",id:19}]}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function Ua(e){let t,n,a,s;return t=new fe({props:{$$slots:{default:[La]},$$scope:{ctx:e}}}),a=new fe({props:{$$slots:{default:[Na]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function Ra(e){let t,n;return t=new ue({props:{$$slots:{content:[Ua],default:[wa]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function Da(e){let t,n,a,s,p,c,m,u,g,h,v,y,S,A,M,w,T,L,E,N,U,R,D,I,B,F,G,H,V,z,O,k,q,K,j,X,Y,W,Q,J,Z,te,ne,ae,se,re,ie,le,oe,pe,ce;return a=new Ca({props:{title:"Nitrogen metabolism",class:"pipen-report-table-image-descr"}}),p=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tcm-Naive-helper-T-cells/Colitis_vs_Control/fgsea_Nitrogen-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tcm-Naive-helper-T-cells/Colitis_vs_Control/fgsea_Nitrogen-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),u=new Ca({props:{title:"Fatty acid degradation",class:"pipen-report-table-image-descr"}}),h=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tcm-Naive-helper-T-cells/Colitis_vs_Control/fgsea_Fatty-acid-degradation.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tcm-Naive-helper-T-cells/Colitis_vs_Control/fgsea_Fatty-acid-degradation.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),S=new Ca({props:{title:"Citrate cycle (TCA cycle)",class:"pipen-report-table-image-descr"}}),M=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tcm-Naive-helper-T-cells/Colitis_vs_Control/fgsea_Citrate-cycle-TCA-cycle-.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tcm-Naive-helper-T-cells/Colitis_vs_Control/fgsea_Citrate-cycle-TCA-cycle-.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),L=new Ca({props:{title:"Inositol phosphate metabolism",class:"pipen-report-table-image-descr"}}),N=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tcm-Naive-helper-T-cells/Colitis_vs_Control/fgsea_Inositol-phosphate-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tcm-Naive-helper-T-cells/Colitis_vs_Control/fgsea_Inositol-phosphate-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),D=new Ca({props:{title:"Glycolysis / Gluconeogenesis",class:"pipen-report-table-image-descr"}}),B=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tcm-Naive-helper-T-cells/Colitis_vs_Control/fgsea_Glycolysis-Gluconeogenesis.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tcm-Naive-helper-T-cells/Colitis_vs_Control/fgsea_Glycolysis-Gluconeogenesis.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),H=new Ca({props:{title:"Ascorbate and aldarate metabolism",class:"pipen-report-table-image-descr"}}),z=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tcm-Naive-helper-T-cells/Colitis_vs_Control/fgsea_Ascorbate-and-aldarate-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tcm-Naive-helper-T-cells/Colitis_vs_Control/fgsea_Ascorbate-and-aldarate-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),q=new Ca({props:{title:"Propanoate metabolism",class:"pipen-report-table-image-descr"}}),j=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tcm-Naive-helper-T-cells/Colitis_vs_Control/fgsea_Propanoate-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tcm-Naive-helper-T-cells/Colitis_vs_Control/fgsea_Propanoate-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),W=new Ca({props:{title:"Starch and sucrose metabolism",class:"pipen-report-table-image-descr"}}),J=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tcm-Naive-helper-T-cells/Colitis_vs_Control/fgsea_Starch-and-sucrose-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tcm-Naive-helper-T-cells/Colitis_vs_Control/fgsea_Starch-and-sucrose-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new Ca({props:{title:"Butanoate metabolism",class:"pipen-report-table-image-descr"}}),se=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tcm-Naive-helper-T-cells/Colitis_vs_Control/fgsea_Butanoate-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tcm-Naive-helper-T-cells/Colitis_vs_Control/fgsea_Butanoate-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),le=new Ca({props:{title:"Glyoxylate and dicarboxylate metabolism",class:"pipen-report-table-image-descr"}}),pe=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tcm-Naive-helper-T-cells/Colitis_vs_Control/fgsea_Glyoxylate-and-dicarboxylate-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tcm-Naive-helper-T-cells/Colitis_vs_Control/fgsea_Glyoxylate-and-dicarboxylate-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=r("div"),n=r("div"),f(a.$$.fragment),s=$(),f(p.$$.fragment),c=$(),m=r("div"),f(u.$$.fragment),g=$(),f(h.$$.fragment),v=$(),y=r("div"),f(S.$$.fragment),A=$(),f(M.$$.fragment),w=$(),T=r("div"),f(L.$$.fragment),E=$(),f(N.$$.fragment),U=$(),R=r("div"),f(D.$$.fragment),I=$(),f(B.$$.fragment),F=$(),G=r("div"),f(H.$$.fragment),V=$(),f(z.$$.fragment),O=$(),k=r("div"),f(q.$$.fragment),K=$(),f(j.$$.fragment),X=$(),Y=r("div"),f(W.$$.fragment),Q=$(),f(J.$$.fragment),Z=$(),te=r("div"),f(ne.$$.fragment),ae=$(),f(se.$$.fragment),re=$(),ie=r("div"),f(le.$$.fragment),oe=$(),f(pe.$$.fragment),i(t,"class","pipen-report-table-of-images"),ee(t,"grid-template-columns","repeat(2, auto)")},m(e,r){l(e,t,r),x(t,n),b(a,n,null),x(n,s),b(p,n,null),x(t,c),x(t,m),b(u,m,null),x(m,g),b(h,m,null),x(t,v),x(t,y),b(S,y,null),x(y,A),b(M,y,null),x(t,w),x(t,T),b(L,T,null),x(T,E),b(N,T,null),x(t,U),x(t,R),b(D,R,null),x(R,I),b(B,R,null),x(t,F),x(t,G),b(H,G,null),x(G,V),b(z,G,null),x(t,O),x(t,k),b(q,k,null),x(k,K),b(j,k,null),x(t,X),x(t,Y),b(W,Y,null),x(Y,Q),b(J,Y,null),x(t,Z),x(t,te),b(ne,te,null),x(te,ae),b(se,te,null),x(t,re),x(t,ie),b(le,ie,null),x(ie,oe),b(pe,ie,null),ce=!0},p:P,i(e){ce||(o(a.$$.fragment,e),o(p.$$.fragment,e),o(u.$$.fragment,e),o(h.$$.fragment,e),o(S.$$.fragment,e),o(M.$$.fragment,e),o(L.$$.fragment,e),o(N.$$.fragment,e),o(D.$$.fragment,e),o(B.$$.fragment,e),o(H.$$.fragment,e),o(z.$$.fragment,e),o(q.$$.fragment,e),o(j.$$.fragment,e),o(W.$$.fragment,e),o(J.$$.fragment,e),o(ne.$$.fragment,e),o(se.$$.fragment,e),o(le.$$.fragment,e),o(pe.$$.fragment,e),ce=!0)},o(e){d(a.$$.fragment,e),d(p.$$.fragment,e),d(u.$$.fragment,e),d(h.$$.fragment,e),d(S.$$.fragment,e),d(M.$$.fragment,e),d(L.$$.fragment,e),d(N.$$.fragment,e),d(D.$$.fragment,e),d(B.$$.fragment,e),d(H.$$.fragment,e),d(z.$$.fragment,e),d(q.$$.fragment,e),d(j.$$.fragment,e),d(W.$$.fragment,e),d(J.$$.fragment,e),d(ne.$$.fragment,e),d(se.$$.fragment,e),d(le.$$.fragment,e),d(pe.$$.fragment,e),ce=!1},d(e){e&&_(t),C(a),C(p),C(u),C(h),C(S),C(M),C(L),C(N),C(D),C(B),C(H),C(z),C(q),C(j),C(W),C(J),C(ne),C(se),C(le),C(pe)}}}function Ia(e){let t,n,a,s;return t=new Ue({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[Ra]},$$scope:{ctx:e}}}),a=new Ue({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[Da]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function Ba(e){let t,n,a,s;return t=new ge({props:{label:"Plot"}}),a=new ge({props:{label:"Table"}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p:P,i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function Fa(e){let t;return{c(){t=S("This table presents a comprehensive summary of the top enriched pathways derived from the fgsea. Each row corresponds to a pathway, and the gene ranks are shown based on the ranking metric used in the analysis. The enrichment score, p-value, and adjusted p-value are also provided to assess the significance of the enrichment.")},m(e,n){l(e,t,n)},d(e){e&&_(t)}}}function Ga(e){let t,n,a,s;return t=new Ca({props:{$$slots:{default:[Fa]},$$scope:{ctx:e}}}),a=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells/Colitis_vs_Control/gsea_table.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells/Colitis_vs_Control/gsea_table.pdf",width:1e3,height:600}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function Ha(e){let t;return{c(){t=S("This plot represents the GSEA results for a specified gene set, illustrating the distribution and impact of the gene set along the ranked list of genes. The running enrichment score curve shows the cumulative enrichment score as genes from the input list are encountered. Positive peaks on the curve indicate regions where members of the gene set are predominantly found.")},m(e,n){l(e,t,n)},d(e){e&&_(t)}}}function Va(e){let t,n,a,s;return t=new Ca({props:{$$slots:{default:[Ha]},$$scope:{ctx:e}}}),a=new ua({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea.txt",data:[{pathway:"Oxidative phosphorylation",pval:0,padj:0,log2err:.9436,ES:.6902,NES:2.0622,size:120,leadingEdge:"ATP5MG,ATP5F1D,UQCRFS1,COX7A2L,NDUFB4,NDUFB11,NDUFB5,NDUFA6,ATP6V0D1,ATP5MC2,COX5A,UQCRH,COX7C,ATP5ME,COX4I1,NDUFA7,ATP6V0B,NDUFS5,PPA1,ATP5PD,ATP6V1A,ATP5MC3,NDUFS8,COX7B,NDUFB1,ATP5MF,COX6B1,ATP5F1C,UQCRQ,COX15,NDUFA1,NDUFS2,UQCRB,NDUFA10,NDUFA12,UQCR10,COX7A2,COX10,NDUFB3,ATP5F1A,NDUFA2,SDHC,UQCR11,COX6C,UQCRC2,ATP6V1D,ATP6V1G1,COX8A,COX5B,NDUFS7,ATP6V1F,UQCRHL,NDUFB10,NDUFA8,LHPP,NDUFV2,ATP5PF,NDUFA11,ATP6V0C,NDUFAB1,NDUFC1,COX6A1,ATP5F1E,NDUFS4,NDUFS3,NDUFB9,CYC1,NDUFA13,ATP6V1C1,ATP5PO,NDUFB6,ATP6V0A2,ATP5F1B,ATP5PB,NDUFS6,COX11,NDUFV1",id:0},{pathway:"Ascorbate and aldarate metabolism",pval:1e-4,padj:8e-4,log2err:.5384,ES:-.6588,NES:-2.1478,size:27,leadingEdge:"UGDH,ALDH9A1,ALDH1B1",id:1},{pathway:"Pentose and glucuronate interconversions",pval:2e-4,padj:.0015,log2err:.5188,ES:-.5902,NES:-2.0375,size:34,leadingEdge:"UGDH,CRYL1,SORD,AKR1B1,UGP2",id:2},{pathway:"Amino sugar and nucleotide sugar metabolism",pval:5e-4,padj:.0024,log2err:.4985,ES:-.4946,NES:-1.7559,size:48,leadingEdge:"UGDH,FPGT,NAGK,CYB5R1,CYB5R3,NPL,GNPDA2,FUK,PGM3,PGM1,HEXB,UAP1,GPI,GALE,PMM2,GNPNAT1,UGP2",id:3},{pathway:"Starch and sucrose metabolism",pval:.0229,padj:.0914,log2err:.3525,ES:-.4279,NES:-1.4718,size:35,leadingEdge:"GBE1,AGL,GANC,PGM1,GPI,UGP2,GAA",id:4},{pathway:"Fructose and mannose metabolism",pval:.1331,padj:.4436,log2err:.2492,ES:-.3672,NES:-1.2511,size:33,leadingEdge:"KHK,FPGT,PFKL,FUK,SORD,PMM2,AKR1B1,FBP1,ENOSF1,ALDOA",id:5},{pathway:"Sulfur metabolism",pval:.1992,padj:.5691,log2err:.111,ES:.6162,NES:1.2385,size:9,leadingEdge:"MPST,TST,ETHE1,SQOR,IMPAD1,PAPSS2,BPNT1",id:6},{pathway:"Propanoate metabolism",pval:.2896,padj:.724,log2err:.0802,ES:.4383,NES:1.1363,size:32,leadingEdge:"LDHB,SUCLG1,ABAT,ACADM,BCKDHA,HADHA,DLD,ACSS1,EHHADH,HIBCH,LDHA,PCCA",id:7},{pathway:"Galactose metabolism",pval:.4684,padj:.8988,log2err:.162,ES:-.2984,NES:-.9883,size:31,leadingEdge:"PFKL,GANC,PGM1,GALE,AKR1B1,UGP2,GAA",id:8},{pathway:"Fatty acid elongation",pval:.5422,padj:.8988,log2err:.1455,ES:-.2817,NES:-.9331,size:30,leadingEdge:"HACD3,HSD17B12,HACD2,PPT1,MECR,TECR",id:9},{pathway:"Citrate cycle (TCA cycle)",pval:.5586,padj:.8988,log2err:.0488,ES:.3718,NES:.9499,size:30,leadingEdge:"IDH2,SUCLG1,SDHC,MDH2,IDH1,IDH3G,PCK2,DLD,CS,PDHB,OGDHL,DLST,SDHB,OGDH,SDHD",id:10},{pathway:"Fatty acid degradation",pval:.5641,padj:.8988,log2err:.1723,ES:-.2743,NES:-.9625,size:44,leadingEdge:"ACSL3,ACADSB,ALDH9A1,CPT1A,CPT1C,ACADVL,ECI1,ACAT2,ALDH1B1",id:11},{pathway:"Glyoxylate and dicarboxylate metabolism",pval:.5842,padj:.8988,log2err:.0474,ES:.3701,NES:.9355,size:28,leadingEdge:"GLDC,GLYCTK,SHMT2,GRHPR,AFMID,MDH2,DLD,PCCA,CS,MCEE",id:12},{pathway:"Inositol phosphate metabolism",pval:.6562,padj:.9375,log2err:.2193,ES:-.2502,NES:-.9491,size:73,leadingEdge:"INPP5E,PIP4K2C,ISYNA1,IMPA1,PIP4K2B,PLCB3,PIK3CB,MTMR4,MTMR8,PIK3CG,PI4K2B,PLCH2,INPP4A,INPP5B,INPP4B",id:13},{pathway:"Glycolysis / Gluconeogenesis",pval:.8286,padj:.9958,log2err:.1847,ES:-.226,NES:-.8669,size:68,leadingEdge:"ENO2,PFKL,ALDH9A1,BPGM,PGM1,GPI,FBP1,PKM,ALDH1B1,ALDOA,DLAT,ENO1",id:14},{pathway:"Butanoate metabolism",pval:.8776,padj:.9958,log2err:.0282,ES:.2907,NES:.7348,size:28,leadingEdge:"ABAT,HMGCS1,HADHA,EHHADH,OXCT1,ACADS,HMGCL,OXCT2,HADH,ACSM1",id:15},{pathway:"Nitrogen metabolism",pval:.8966,padj:.9958,log2err:.0969,ES:-.2533,NES:-.7262,size:17,leadingEdge:"CA5B,CA9,GLUL,CA8,GLUD2,CA4,CA5A,CA12,CA7,CPS1,CA3,CA2,CA14,CA13,CA6,GLUD1,CA1",id:16},{pathway:"Pentose phosphate pathway",pval:.9337,padj:.9958,log2err:.1067,ES:-.223,NES:-.7389,size:30,leadingEdge:"PFKL,PRPS1,PGM1,GPI,FBP1,ALDOA",id:17},{pathway:"Pyruvate metabolism",pval:.9793,padj:.9958,log2err:.019,ES:.2314,NES:.621,size:39,leadingEdge:"LDHB,ACYP2,GRHPR,MDH2,PCK2,DLD,HAGH,ACSS1,LDHD,LDHA",id:18},{pathway:"Fatty acid biosynthesis",pval:.9958,padj:.9958,log2err:.0815,ES:-.2108,NES:-.5674,size:13,leadingEdge:"ACSL3",id:19}]}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function za(e){let t,n,a,s;return t=new fe({props:{$$slots:{default:[Ga]},$$scope:{ctx:e}}}),a=new fe({props:{$$slots:{default:[Va]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function Oa(e){let t,n;return t=new ue({props:{$$slots:{content:[za],default:[Ba]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function ka(e){let t,n,a,s,p,c,m,u,g,h,v,y,S,A,M,w,T,L,E,N,U,R,D,I,B,F,G,H,V,z,O,k,q,K,j,X,Y,W,Q,J,Z,te,ne,ae,se,re,ie,le,oe,pe,ce;return a=new Ca({props:{title:"Oxidative phosphorylation",class:"pipen-report-table-image-descr"}}),p=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Oxidative-phosphorylation.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Oxidative-phosphorylation.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),u=new Ca({props:{title:"Ascorbate and aldarate metabolism",class:"pipen-report-table-image-descr"}}),h=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Ascorbate-and-aldarate-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Ascorbate-and-aldarate-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),S=new Ca({props:{title:"Pentose and glucuronate interconversions",class:"pipen-report-table-image-descr"}}),M=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Pentose-and-glucuronate-interconversions.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Pentose-and-glucuronate-interconversions.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),L=new Ca({props:{title:"Amino sugar and nucleotide sugar metabolism",class:"pipen-report-table-image-descr"}}),N=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Amino-sugar-and-nucleotide-sugar-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Amino-sugar-and-nucleotide-sugar-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),D=new Ca({props:{title:"Starch and sucrose metabolism",class:"pipen-report-table-image-descr"}}),B=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Starch-and-sucrose-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Starch-and-sucrose-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),H=new Ca({props:{title:"Fructose and mannose metabolism",class:"pipen-report-table-image-descr"}}),z=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Fructose-and-mannose-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Fructose-and-mannose-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),q=new Ca({props:{title:"Sulfur metabolism",class:"pipen-report-table-image-descr"}}),j=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Sulfur-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Sulfur-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),W=new Ca({props:{title:"Propanoate metabolism",class:"pipen-report-table-image-descr"}}),J=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Propanoate-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Propanoate-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new Ca({props:{title:"Galactose metabolism",class:"pipen-report-table-image-descr"}}),se=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Galactose-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Galactose-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),le=new Ca({props:{title:"Fatty acid elongation",class:"pipen-report-table-image-descr"}}),pe=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Fatty-acid-elongation.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Fatty-acid-elongation.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=r("div"),n=r("div"),f(a.$$.fragment),s=$(),f(p.$$.fragment),c=$(),m=r("div"),f(u.$$.fragment),g=$(),f(h.$$.fragment),v=$(),y=r("div"),f(S.$$.fragment),A=$(),f(M.$$.fragment),w=$(),T=r("div"),f(L.$$.fragment),E=$(),f(N.$$.fragment),U=$(),R=r("div"),f(D.$$.fragment),I=$(),f(B.$$.fragment),F=$(),G=r("div"),f(H.$$.fragment),V=$(),f(z.$$.fragment),O=$(),k=r("div"),f(q.$$.fragment),K=$(),f(j.$$.fragment),X=$(),Y=r("div"),f(W.$$.fragment),Q=$(),f(J.$$.fragment),Z=$(),te=r("div"),f(ne.$$.fragment),ae=$(),f(se.$$.fragment),re=$(),ie=r("div"),f(le.$$.fragment),oe=$(),f(pe.$$.fragment),i(t,"class","pipen-report-table-of-images"),ee(t,"grid-template-columns","repeat(2, auto)")},m(e,r){l(e,t,r),x(t,n),b(a,n,null),x(n,s),b(p,n,null),x(t,c),x(t,m),b(u,m,null),x(m,g),b(h,m,null),x(t,v),x(t,y),b(S,y,null),x(y,A),b(M,y,null),x(t,w),x(t,T),b(L,T,null),x(T,E),b(N,T,null),x(t,U),x(t,R),b(D,R,null),x(R,I),b(B,R,null),x(t,F),x(t,G),b(H,G,null),x(G,V),b(z,G,null),x(t,O),x(t,k),b(q,k,null),x(k,K),b(j,k,null),x(t,X),x(t,Y),b(W,Y,null),x(Y,Q),b(J,Y,null),x(t,Z),x(t,te),b(ne,te,null),x(te,ae),b(se,te,null),x(t,re),x(t,ie),b(le,ie,null),x(ie,oe),b(pe,ie,null),ce=!0},p:P,i(e){ce||(o(a.$$.fragment,e),o(p.$$.fragment,e),o(u.$$.fragment,e),o(h.$$.fragment,e),o(S.$$.fragment,e),o(M.$$.fragment,e),o(L.$$.fragment,e),o(N.$$.fragment,e),o(D.$$.fragment,e),o(B.$$.fragment,e),o(H.$$.fragment,e),o(z.$$.fragment,e),o(q.$$.fragment,e),o(j.$$.fragment,e),o(W.$$.fragment,e),o(J.$$.fragment,e),o(ne.$$.fragment,e),o(se.$$.fragment,e),o(le.$$.fragment,e),o(pe.$$.fragment,e),ce=!0)},o(e){d(a.$$.fragment,e),d(p.$$.fragment,e),d(u.$$.fragment,e),d(h.$$.fragment,e),d(S.$$.fragment,e),d(M.$$.fragment,e),d(L.$$.fragment,e),d(N.$$.fragment,e),d(D.$$.fragment,e),d(B.$$.fragment,e),d(H.$$.fragment,e),d(z.$$.fragment,e),d(q.$$.fragment,e),d(j.$$.fragment,e),d(W.$$.fragment,e),d(J.$$.fragment,e),d(ne.$$.fragment,e),d(se.$$.fragment,e),d(le.$$.fragment,e),d(pe.$$.fragment,e),ce=!1},d(e){e&&_(t),C(a),C(p),C(u),C(h),C(S),C(M),C(L),C(N),C(D),C(B),C(H),C(z),C(q),C(j),C(W),C(J),C(ne),C(se),C(le),C(pe)}}}function qa(e){let t,n,a,s;return t=new Ue({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[Oa]},$$scope:{ctx:e}}}),a=new Ue({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[ka]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function Ka(e){let t,n,a,s;return t=new ge({props:{label:"Plot"}}),a=new ge({props:{label:"Table"}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p:P,i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function ja(e){let t;return{c(){t=S("This table presents a comprehensive summary of the top enriched pathways derived from the fgsea. Each row corresponds to a pathway, and the gene ranks are shown based on the ranking metric used in the analysis. The enrichment score, p-value, and adjusted p-value are also provided to assess the significance of the enrichment.")},m(e,n){l(e,t,n)},d(e){e&&_(t)}}}function Xa(e){let t,n,a,s;return t=new Ca({props:{$$slots:{default:[ja]},$$scope:{ctx:e}}}),a=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells-1/Colitis_vs_Control/gsea_table.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells-1/Colitis_vs_Control/gsea_table.pdf",width:1e3,height:600}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function Ya(e){let t;return{c(){t=S("This plot represents the GSEA results for a specified gene set, illustrating the distribution and impact of the gene set along the ranked list of genes. The running enrichment score curve shows the cumulative enrichment score as genes from the input list are encountered. Positive peaks on the curve indicate regions where members of the gene set are predominantly found.")},m(e,n){l(e,t,n)},d(e){e&&_(t)}}}function Wa(e){let t,n,a,s;return t=new Ca({props:{$$slots:{default:[Ya]},$$scope:{ctx:e}}}),a=new ua({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells-1/Colitis_vs_Control/fgsea.txt",data:[{pathway:"Oxidative phosphorylation",pval:0,padj:0,log2err:.8987,ES:.6577,NES:2.2249,size:120,leadingEdge:"ATP5MG,ATP5F1D,NDUFB5,NDUFB4,UQCRFS1,ATP6V0B,COX5A,ATP5PD,NDUFS5,COX7A2L,ATP5MC2,NDUFA12,NDUFB11,NDUFS8,COX7B,NDUFA6,UQCRH,ATP6V0D1,ATP5MF,ATP6V1F,ATP5MC3,NDUFB10,ATP5F1C,NDUFA7,UQCR10,UQCRQ,LHPP,COX7C,COX10,COX7A2,COX4I1,NDUFS2,NDUFA2,NDUFS3,ATP6V1D,UQCRHL,ATP5F1A,NDUFA10,NDUFA1,NDUFB3,ATP5ME,SDHC,NDUFB1,COX5B,COX8A,NDUFS7,COX6C,PPA1,UQCRB,NDUFB9,ATP6V1A,UQCRC2,COX6B1,UQCR11,ATP6V1C1,ATP6V0C,NDUFV2,ATP6V1G1,COX15,COX6A1,NDUFB8,CYC1,NDUFS4,NDUFB6",id:0},{pathway:"Ascorbate and aldarate metabolism",pval:2e-4,padj:.0025,log2err:.4985,ES:-.681,NES:-2.0731,size:27,leadingEdge:"UGDH,ALDH3A2,ALDH9A1,ALDH1B1",id:1},{pathway:"Amino sugar and nucleotide sugar metabolism",pval:.0042,padj:.0281,log2err:.407,ES:-.4868,NES:-1.677,size:48,leadingEdge:"NPL,CYB5R1,UGDH,PMM2,CYB5R3,GNPNAT1,NAGK,FPGT,FUK,PGM1,GNPDA2,PGM3,GALE,UAP1,GPI",id:2},{pathway:"Sulfur metabolism",pval:.018,padj:.0902,log2err:.3525,ES:.7558,NES:1.561,size:9,leadingEdge:"MPST,TST,SQOR,ETHE1,IMPAD1",id:3},{pathway:"Pentose and glucuronate interconversions",pval:.0227,padj:.0906,log2err:.3525,ES:-.4674,NES:-1.511,size:34,leadingEdge:"UGDH,CRYL1,SORD",id:4},{pathway:"Glycolysis / Gluconeogenesis",pval:.1356,padj:.4125,log2err:.253,ES:-.3189,NES:-1.1694,size:68,leadingEdge:"ENO2,PKM,DLAT,PGM1,PFKL,ALDH3A2,ALDOA,ALDH9A1,GPI,PDHA1,ALDOC,FBP1,BPGM,ALDH1B1",id:5},{pathway:"Starch and sucrose metabolism",pval:.1509,padj:.4125,log2err:.2165,ES:-.3785,NES:-1.2324,size:35,leadingEdge:"AGL,PGM1,GANC,GPI,GBE1,AMY2B,PYGM",id:6},{pathway:"Fructose and mannose metabolism",pval:.165,padj:.4125,log2err:.2021,ES:-.3738,NES:-1.207,size:33,leadingEdge:"PMM2,ENOSF1,FPGT,FUK,PFKL,SORD,ALDOA,KHK,ALDOC,FBP1",id:7},{pathway:"Fatty acid degradation",pval:.214,padj:.4756,log2err:.1847,ES:-.3447,NES:-1.1582,size:44,leadingEdge:"ACSL3,CPT1A,ACADVL,CPT2,ACADSB,CPT1C,ALDH3A2,ALDH9A1",id:8},{pathway:"Fatty acid biosynthesis",pval:.2857,padj:.5714,log2err:.1388,ES:-.4401,NES:-1.1139,size:13,leadingEdge:"ACSL3,ACACA,ACACB,ACSL1,OXSM,ACSBG2,ACSBG1,ACSL5",id:9},{pathway:"Nitrogen metabolism",pval:.3724,padj:.6771,log2err:.0789,ES:.4473,NES:1.0892,size:17,leadingEdge:"GLUD1,GLUL,GLUD2,CA5B,CA1,CA6",id:10},{pathway:"Inositol phosphate metabolism",pval:.4286,padj:.7143,log2err:.1429,ES:-.2719,NES:-1.0053,size:73,leadingEdge:"ISYNA1,INPP5E,PIP4K2B,PIP4K2C,MTMR8,MTMR4,PIK3CG,IMPA1,MTM1,INPP5B,PIK3CB,IPPK,PLCG2,PIKFYVE,PLCB3,PLCH2",id:11},{pathway:"Pentose phosphate pathway",pval:.6242,padj:.931,log2err:.0941,ES:-.2926,NES:-.9142,size:30,leadingEdge:"PRPS1,PGM1,PFKL,RPIA,ALDOA,GPI,ALDOC,FBP1,PRPS2,H6PD,PFKP",id:12},{pathway:"Pyruvate metabolism",pval:.6643,padj:.931,log2err:.0962,ES:-.273,NES:-.901,size:39,leadingEdge:"GLO1,PKM,DLAT,ACACA,ALDH3A2,ALDH9A1,PDHA1,ME3,ACACB,ALDH1B1,FH,ME2,ACYP1,MDH1",id:13},{pathway:"Propanoate metabolism",pval:.6983,padj:.931,log2err:.047,ES:.3114,NES:.8611,size:32,leadingEdge:"LDHB,ACSS1,ABAT,BCKDHA,ACADM,SUCLG1,ACAT1,LDHA,HADHA,DBT,DLD,EHHADH,ALDH6A1,LDHAL6A,PCCB,SUCLG2,MCEE",id:14},{pathway:"Galactose metabolism",pval:.8241,padj:.99,log2err:.0785,ES:-.256,NES:-.8056,size:31,leadingEdge:"PGM1,PFKL,GALE,GANC,GLA",id:15},{pathway:"Citrate cycle (TCA cycle)",pval:.908,padj:.99,log2err:.0349,ES:.2667,NES:.7283,size:30,leadingEdge:"IDH2,SUCLG1,SDHC,IDH3G,MDH2,OGDH,IDH3B,DLST,IDH1,SDHB,DLD,PDHB,SDHA,SUCLG2,SDHD,PCK2",id:16},{pathway:"Fatty acid elongation",pval:.915,padj:.99,log2err:.0733,ES:-.2394,NES:-.748,size:30,leadingEdge:"HSD17B12,HACD3,MECR,HACD2,TECR,ACOT4,PPT1,ECHS1,PPT2,ELOVL4,THEM5",id:17},{pathway:"Glyoxylate and dicarboxylate metabolism",pval:.9486,padj:.99,log2err:.0324,ES:.2411,NES:.655,size:28,leadingEdge:"GLDC,GLYCTK,SHMT2,MDH2,GRHPR,ACAT1,GLUL",id:18},{pathway:"Butanoate metabolism",pval:.99,padj:.99,log2err:.0701,ES:-.204,NES:-.6234,size:28,leadingEdge:"ALDH5A1,AACS,L2HGDH,ACSM3,BDH2,ECHS1",id:19}]}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function Qa(e){let t,n,a,s;return t=new fe({props:{$$slots:{default:[Xa]},$$scope:{ctx:e}}}),a=new fe({props:{$$slots:{default:[Wa]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function Ja(e){let t,n;return t=new ue({props:{$$slots:{content:[Qa],default:[Ka]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function Za(e){let t,n,a,s,p,c,m,u,g,h,v,y,S,A,M,w,T,L,E,N,U,R,D,I,B,F,G,H,V,z,O,k,q,K,j,X,Y,W,Q,J,Z,te,ne,ae,se,re,ie,le,oe,pe,ce;return a=new Ca({props:{title:"Oxidative phosphorylation",class:"pipen-report-table-image-descr"}}),p=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells-1/Colitis_vs_Control/fgsea_Oxidative-phosphorylation.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells-1/Colitis_vs_Control/fgsea_Oxidative-phosphorylation.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),u=new Ca({props:{title:"Ascorbate and aldarate metabolism",class:"pipen-report-table-image-descr"}}),h=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells-1/Colitis_vs_Control/fgsea_Ascorbate-and-aldarate-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells-1/Colitis_vs_Control/fgsea_Ascorbate-and-aldarate-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),S=new Ca({props:{title:"Amino sugar and nucleotide sugar metabolism",class:"pipen-report-table-image-descr"}}),M=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells-1/Colitis_vs_Control/fgsea_Amino-sugar-and-nucleotide-sugar-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells-1/Colitis_vs_Control/fgsea_Amino-sugar-and-nucleotide-sugar-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),L=new Ca({props:{title:"Sulfur metabolism",class:"pipen-report-table-image-descr"}}),N=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells-1/Colitis_vs_Control/fgsea_Sulfur-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells-1/Colitis_vs_Control/fgsea_Sulfur-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),D=new Ca({props:{title:"Pentose and glucuronate interconversions",class:"pipen-report-table-image-descr"}}),B=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells-1/Colitis_vs_Control/fgsea_Pentose-and-glucuronate-interconversions.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells-1/Colitis_vs_Control/fgsea_Pentose-and-glucuronate-interconversions.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),H=new Ca({props:{title:"Glycolysis / Gluconeogenesis",class:"pipen-report-table-image-descr"}}),z=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells-1/Colitis_vs_Control/fgsea_Glycolysis-Gluconeogenesis.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells-1/Colitis_vs_Control/fgsea_Glycolysis-Gluconeogenesis.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),q=new Ca({props:{title:"Starch and sucrose metabolism",class:"pipen-report-table-image-descr"}}),j=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells-1/Colitis_vs_Control/fgsea_Starch-and-sucrose-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells-1/Colitis_vs_Control/fgsea_Starch-and-sucrose-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),W=new Ca({props:{title:"Fructose and mannose metabolism",class:"pipen-report-table-image-descr"}}),J=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells-1/Colitis_vs_Control/fgsea_Fructose-and-mannose-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells-1/Colitis_vs_Control/fgsea_Fructose-and-mannose-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new Ca({props:{title:"Fatty acid degradation",class:"pipen-report-table-image-descr"}}),se=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells-1/Colitis_vs_Control/fgsea_Fatty-acid-degradation.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells-1/Colitis_vs_Control/fgsea_Fatty-acid-degradation.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),le=new Ca({props:{title:"Fatty acid biosynthesis",class:"pipen-report-table-image-descr"}}),pe=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells-1/Colitis_vs_Control/fgsea_Fatty-acid-biosynthesis.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Tem-Trm-cytotoxic-T-cells-1/Colitis_vs_Control/fgsea_Fatty-acid-biosynthesis.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=r("div"),n=r("div"),f(a.$$.fragment),s=$(),f(p.$$.fragment),c=$(),m=r("div"),f(u.$$.fragment),g=$(),f(h.$$.fragment),v=$(),y=r("div"),f(S.$$.fragment),A=$(),f(M.$$.fragment),w=$(),T=r("div"),f(L.$$.fragment),E=$(),f(N.$$.fragment),U=$(),R=r("div"),f(D.$$.fragment),I=$(),f(B.$$.fragment),F=$(),G=r("div"),f(H.$$.fragment),V=$(),f(z.$$.fragment),O=$(),k=r("div"),f(q.$$.fragment),K=$(),f(j.$$.fragment),X=$(),Y=r("div"),f(W.$$.fragment),Q=$(),f(J.$$.fragment),Z=$(),te=r("div"),f(ne.$$.fragment),ae=$(),f(se.$$.fragment),re=$(),ie=r("div"),f(le.$$.fragment),oe=$(),f(pe.$$.fragment),i(t,"class","pipen-report-table-of-images"),ee(t,"grid-template-columns","repeat(2, auto)")},m(e,r){l(e,t,r),x(t,n),b(a,n,null),x(n,s),b(p,n,null),x(t,c),x(t,m),b(u,m,null),x(m,g),b(h,m,null),x(t,v),x(t,y),b(S,y,null),x(y,A),b(M,y,null),x(t,w),x(t,T),b(L,T,null),x(T,E),b(N,T,null),x(t,U),x(t,R),b(D,R,null),x(R,I),b(B,R,null),x(t,F),x(t,G),b(H,G,null),x(G,V),b(z,G,null),x(t,O),x(t,k),b(q,k,null),x(k,K),b(j,k,null),x(t,X),x(t,Y),b(W,Y,null),x(Y,Q),b(J,Y,null),x(t,Z),x(t,te),b(ne,te,null),x(te,ae),b(se,te,null),x(t,re),x(t,ie),b(le,ie,null),x(ie,oe),b(pe,ie,null),ce=!0},p:P,i(e){ce||(o(a.$$.fragment,e),o(p.$$.fragment,e),o(u.$$.fragment,e),o(h.$$.fragment,e),o(S.$$.fragment,e),o(M.$$.fragment,e),o(L.$$.fragment,e),o(N.$$.fragment,e),o(D.$$.fragment,e),o(B.$$.fragment,e),o(H.$$.fragment,e),o(z.$$.fragment,e),o(q.$$.fragment,e),o(j.$$.fragment,e),o(W.$$.fragment,e),o(J.$$.fragment,e),o(ne.$$.fragment,e),o(se.$$.fragment,e),o(le.$$.fragment,e),o(pe.$$.fragment,e),ce=!0)},o(e){d(a.$$.fragment,e),d(p.$$.fragment,e),d(u.$$.fragment,e),d(h.$$.fragment,e),d(S.$$.fragment,e),d(M.$$.fragment,e),d(L.$$.fragment,e),d(N.$$.fragment,e),d(D.$$.fragment,e),d(B.$$.fragment,e),d(H.$$.fragment,e),d(z.$$.fragment,e),d(q.$$.fragment,e),d(j.$$.fragment,e),d(W.$$.fragment,e),d(J.$$.fragment,e),d(ne.$$.fragment,e),d(se.$$.fragment,e),d(le.$$.fragment,e),d(pe.$$.fragment,e),ce=!1},d(e){e&&_(t),C(a),C(p),C(u),C(h),C(S),C(M),C(L),C(N),C(D),C(B),C(H),C(z),C(q),C(j),C(W),C(J),C(ne),C(se),C(le),C(pe)}}}function es(e){let t,n,a,s;return t=new Ue({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[Ja]},$$scope:{ctx:e}}}),a=new Ue({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[Za]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function ts(e){let t,n,a,s;return t=new ge({props:{label:"Plot"}}),a=new ge({props:{label:"Table"}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p:P,i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function ns(e){let t;return{c(){t=S("This table presents a comprehensive summary of the top enriched pathways derived from the fgsea. Each row corresponds to a pathway, and the gene ranks are shown based on the ranking metric used in the analysis. The enrichment score, p-value, and adjusted p-value are also provided to assess the significance of the enrichment.")},m(e,n){l(e,t,n)},d(e){e&&_(t)}}}function as(e){let t,n,a,s;return t=new Ca({props:{$$slots:{default:[ns]},$$scope:{ctx:e}}}),a=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Trm-cytotoxic-T-cells/Colitis_vs_Control/gsea_table.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Trm-cytotoxic-T-cells/Colitis_vs_Control/gsea_table.pdf",width:1e3,height:600}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function ss(e){let t;return{c(){t=S("This plot represents the GSEA results for a specified gene set, illustrating the distribution and impact of the gene set along the ranked list of genes. The running enrichment score curve shows the cumulative enrichment score as genes from the input list are encountered. Positive peaks on the curve indicate regions where members of the gene set are predominantly found.")},m(e,n){l(e,t,n)},d(e){e&&_(t)}}}function rs(e){let t,n,a,s;return t=new Ca({props:{$$slots:{default:[ss]},$$scope:{ctx:e}}}),a=new ua({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea.txt",data:[{pathway:"Oxidative phosphorylation",pval:0,padj:0,log2err:.7749,ES:.5008,NES:1.9836,size:120,leadingEdge:"ATP6V0C,NDUFA6,ATP6V0B,NDUFS5,NDUFB4,COX7C,NDUFS7,ATP5ME,ATP5MG,UQCR10,NDUFA3,NDUFB1,COX7B,NDUFA2,UQCRH,ATP5F1E,UQCRFS1,NDUFC1,COX5B,UQCR11,UQCRB,UQCRQ,NDUFA1,NDUFB11,ATP5MF,ATP6V0A2,COX7A2,NDUFA12,COX6C,ATP6V1F,COX5A,NDUFS8,ATP5PD,NDUFA13,PPA1,ATP5MC3,NDUFA4L2,NDUFS2,NDUFAB1,NDUFA7,ATP6V0D1,ATP5F1D,COX6A1,ATP5F1C,ATP6V0A4,COX6B1,SDHB,ATP6V1G1,COX10,COX6B2,ATP5MC2,CYC1,NDUFV1,UQCRHL",id:0},{pathway:"Pentose and glucuronate interconversions",pval:1e-4,padj:.0013,log2err:.5188,ES:-.7098,NES:-1.7782,size:34,leadingEdge:"UGDH,CRYL1,AKR1B1,SORD",id:1},{pathway:"Amino sugar and nucleotide sugar metabolism",pval:3e-4,padj:.0018,log2err:.4985,ES:-.645,NES:-1.6854,size:48,leadingEdge:"FPGT,GNPDA2,PGM2,GNPNAT1,UGDH,HEXB,GNPDA1,GPI,GALT,CYB5R3,CYB5R1,PGM1,CMAS,UAP1,HK1,GALE,PGM3,NPL,MPI,AMDHD2,GNE,FUK,PMM2,NAGK,HEXA",id:2},{pathway:"Ascorbate and aldarate metabolism",pval:9e-4,padj:.0044,log2err:.4773,ES:-.7012,NES:-1.6979,size:27,leadingEdge:"UGDH,ALDH9A1,ALDH1B1",id:3},{pathway:"Starch and sucrose metabolism",pval:.0157,padj:.0627,log2err:.3525,ES:-.5948,NES:-1.4962,size:35,leadingEdge:"AGL,GAA,PGM2,GPI,PGM1,GYG1,HK1,GBE1,GYS1,GANC",id:4},{pathway:"Glycolysis / Gluconeogenesis",pval:.0322,padj:.095,log2err:.2617,ES:-.5071,NES:-1.3629,size:68,leadingEdge:"ENO2,PGM2,ADH5,PKM,GPI,ALDH9A1,PGM1,PFKL,DLAT,ALDOA,FBP1,HK1,PDHA1,ALDH1B1,ENO1,PGAM1",id:5},{pathway:"Pentose phosphate pathway",pval:.0333,padj:.095,log2err:.2713,ES:-.5815,NES:-1.4232,size:30,leadingEdge:"PGM2,GPI,PRPS2,PGM1,PFKL,ALDOA,FBP1,PRPS1,G6PD,H6PD,TALDO1,IDNK,ALDOC,PGD",id:6},{pathway:"Fatty acid degradation",pval:.0407,padj:.1016,log2err:.2344,ES:-.5397,NES:-1.3888,size:44,leadingEdge:"ACSL3,ACADVL,ADH5,ACAA2,CPT1A,HADHA,ALDH9A1,ACADSB,ACSL6,ALDH1B1,CPT1C,ECHS1,ACOX1,ACSL5",id:7},{pathway:"Pyruvate metabolism",pval:.0471,padj:.1048,log2err:.2193,ES:-.5407,NES:-1.3786,size:39,leadingEdge:"GLO1,FH,PKM,ME2,ALDH9A1,DLAT,PDHA1,ALDH1B1,ME1,ACYP1",id:8},{pathway:"Galactose metabolism",pval:.0539,padj:.1079,log2err:.209,ES:-.5633,NES:-1.3861,size:31,leadingEdge:"GAA,PGM2,GLB1,GALT,AKR1B1,PGM1,PFKL,HK1,GLA,GALE,GANC",id:9},{pathway:"Citrate cycle (TCA cycle)",pval:.0879,padj:.1475,log2err:.162,ES:-.5457,NES:-1.3355,size:30,leadingEdge:"ACO1,FH,DLST,SUCLA2,DLAT,ACLY,PDHA1,CS,OGDH,SUCLG2,ACO2,SDHD",id:10},{pathway:"Propanoate metabolism",pval:.0885,padj:.1475,log2err:.1596,ES:-.5347,NES:-1.3223,size:32,leadingEdge:"MLYCD,SUCLA2,HADHA,BCKDHB,MUT,ECHDC1,DBT,ECHS1,SUCLG2",id:11},{pathway:"Butanoate metabolism",pval:.2237,padj:.3442,log2err:.0959,ES:-.4964,NES:-1.1999,size:28,leadingEdge:"HADHA,ACSM3,AACS,BDH2,OXCT1,HMGCL,ECHS1,ALDH5A1",id:12},{pathway:"Nitrogen metabolism",pval:.2582,padj:.3689,log2err:.19,ES:.4144,NES:1.1379,size:17,leadingEdge:"GLUL,CA1,CA14,GLUD1,CA13",id:13},{pathway:"Fructose and mannose metabolism",pval:.3016,padj:.3919,log2err:.0765,ES:-.4516,NES:-1.1279,size:33,leadingEdge:"FPGT,KHK,AKR1B1,PFKL,ALDOA,SORD,FBP1,HK1,MPI,FUK,PMM2,ENOSF1,GMPPA,PFKFB4,ALDOC",id:14},{pathway:"Fatty acid elongation",pval:.3135,padj:.3919,log2err:.0763,ES:-.4557,NES:-1.1154,size:30,leadingEdge:"HSD17B12,PPT1,ACAA2,HADHA,HACD2,HACD4,ACOT4,HACD3,ECHS1,ELOVL1",id:15},{pathway:"Glyoxylate and dicarboxylate metabolism",pval:.3543,padj:.4168,log2err:.071,ES:-.4513,NES:-1.0909,size:28,leadingEdge:"ACO1,CAT,GCSH,HYI,CS,MUT,ACO2,MDH1,MDH2,PCCA,SHMT1,ACAT2,GLYCTK,PCCB,AMT",id:16},{pathway:"Inositol phosphate metabolism",pval:.5032,padj:.5591,log2err:.048,ES:-.3697,NES:-1.0029,size:73,leadingEdge:"PLCB3,PIP4K2C,IPPK,IMPA1,ISYNA1,MTMR8,SACM1L,PIP4K2B,INPP5B,MTM1,PI4K2B,INPP4A,PLCB2,PIK3CG,MTMR2,PI4KB,PLCB1,INPP5E,PIK3C2B,MINPP1,ALDH6A1,PLCH2,PLCG1,MTMR4",id:17},{pathway:"Sulfur metabolism",pval:.807,padj:.8495,log2err:.0836,ES:.337,NES:.7439,size:9,leadingEdge:"IMPAD1,MPST,BPNT1,TST,PAPSS2",id:18},{pathway:"Fatty acid biosynthesis",pval:.8773,padj:.8773,log2err:.0324,ES:-.324,NES:-.6885,size:13,leadingEdge:"ACSL3,ACSL6,ACSL5,OXSM",id:19}]}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function is(e){let t,n,a,s;return t=new fe({props:{$$slots:{default:[as]},$$scope:{ctx:e}}}),a=new fe({props:{$$slots:{default:[rs]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function ls(e){let t,n;return t=new ue({props:{$$slots:{content:[is],default:[ts]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function os(e){let t,n,a,s,p,c,m,u,g,h,v,y,S,A,M,w,T,L,E,N,U,R,D,I,B,F,G,H,V,z,O,k,q,K,j,X,Y,W,Q,J,Z,te,ne,ae,se,re,ie,le,oe,pe,ce;return a=new Ca({props:{title:"Oxidative phosphorylation",class:"pipen-report-table-image-descr"}}),p=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Oxidative-phosphorylation.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Oxidative-phosphorylation.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),u=new Ca({props:{title:"Pentose and glucuronate interconversions",class:"pipen-report-table-image-descr"}}),h=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Pentose-and-glucuronate-interconversions.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Pentose-and-glucuronate-interconversions.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),S=new Ca({props:{title:"Amino sugar and nucleotide sugar metabolism",class:"pipen-report-table-image-descr"}}),M=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Amino-sugar-and-nucleotide-sugar-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Amino-sugar-and-nucleotide-sugar-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),L=new Ca({props:{title:"Ascorbate and aldarate metabolism",class:"pipen-report-table-image-descr"}}),N=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Ascorbate-and-aldarate-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Ascorbate-and-aldarate-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),D=new Ca({props:{title:"Starch and sucrose metabolism",class:"pipen-report-table-image-descr"}}),B=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Starch-and-sucrose-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Starch-and-sucrose-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),H=new Ca({props:{title:"Glycolysis / Gluconeogenesis",class:"pipen-report-table-image-descr"}}),z=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Glycolysis-Gluconeogenesis.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Glycolysis-Gluconeogenesis.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),q=new Ca({props:{title:"Pentose phosphate pathway",class:"pipen-report-table-image-descr"}}),j=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Pentose-phosphate-pathway.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Pentose-phosphate-pathway.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),W=new Ca({props:{title:"Fatty acid degradation",class:"pipen-report-table-image-descr"}}),J=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Fatty-acid-degradation.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Fatty-acid-degradation.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new Ca({props:{title:"Pyruvate metabolism",class:"pipen-report-table-image-descr"}}),se=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Pyruvate-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Pyruvate-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),le=new Ca({props:{title:"Galactose metabolism",class:"pipen-report-table-image-descr"}}),pe=new On({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Galactose-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Trm-cytotoxic-T-cells/Colitis_vs_Control/fgsea_Galactose-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=r("div"),n=r("div"),f(a.$$.fragment),s=$(),f(p.$$.fragment),c=$(),m=r("div"),f(u.$$.fragment),g=$(),f(h.$$.fragment),v=$(),y=r("div"),f(S.$$.fragment),A=$(),f(M.$$.fragment),w=$(),T=r("div"),f(L.$$.fragment),E=$(),f(N.$$.fragment),U=$(),R=r("div"),f(D.$$.fragment),I=$(),f(B.$$.fragment),F=$(),G=r("div"),f(H.$$.fragment),V=$(),f(z.$$.fragment),O=$(),k=r("div"),f(q.$$.fragment),K=$(),f(j.$$.fragment),X=$(),Y=r("div"),f(W.$$.fragment),Q=$(),f(J.$$.fragment),Z=$(),te=r("div"),f(ne.$$.fragment),ae=$(),f(se.$$.fragment),re=$(),ie=r("div"),f(le.$$.fragment),oe=$(),f(pe.$$.fragment),i(t,"class","pipen-report-table-of-images"),ee(t,"grid-template-columns","repeat(2, auto)")},m(e,r){l(e,t,r),x(t,n),b(a,n,null),x(n,s),b(p,n,null),x(t,c),x(t,m),b(u,m,null),x(m,g),b(h,m,null),x(t,v),x(t,y),b(S,y,null),x(y,A),b(M,y,null),x(t,w),x(t,T),b(L,T,null),x(T,E),b(N,T,null),x(t,U),x(t,R),b(D,R,null),x(R,I),b(B,R,null),x(t,F),x(t,G),b(H,G,null),x(G,V),b(z,G,null),x(t,O),x(t,k),b(q,k,null),x(k,K),b(j,k,null),x(t,X),x(t,Y),b(W,Y,null),x(Y,Q),b(J,Y,null),x(t,Z),x(t,te),b(ne,te,null),x(te,ae),b(se,te,null),x(t,re),x(t,ie),b(le,ie,null),x(ie,oe),b(pe,ie,null),ce=!0},p:P,i(e){ce||(o(a.$$.fragment,e),o(p.$$.fragment,e),o(u.$$.fragment,e),o(h.$$.fragment,e),o(S.$$.fragment,e),o(M.$$.fragment,e),o(L.$$.fragment,e),o(N.$$.fragment,e),o(D.$$.fragment,e),o(B.$$.fragment,e),o(H.$$.fragment,e),o(z.$$.fragment,e),o(q.$$.fragment,e),o(j.$$.fragment,e),o(W.$$.fragment,e),o(J.$$.fragment,e),o(ne.$$.fragment,e),o(se.$$.fragment,e),o(le.$$.fragment,e),o(pe.$$.fragment,e),ce=!0)},o(e){d(a.$$.fragment,e),d(p.$$.fragment,e),d(u.$$.fragment,e),d(h.$$.fragment,e),d(S.$$.fragment,e),d(M.$$.fragment,e),d(L.$$.fragment,e),d(N.$$.fragment,e),d(D.$$.fragment,e),d(B.$$.fragment,e),d(H.$$.fragment,e),d(z.$$.fragment,e),d(q.$$.fragment,e),d(j.$$.fragment,e),d(W.$$.fragment,e),d(J.$$.fragment,e),d(ne.$$.fragment,e),d(se.$$.fragment,e),d(le.$$.fragment,e),d(pe.$$.fragment,e),ce=!1},d(e){e&&_(t),C(a),C(p),C(u),C(h),C(S),C(M),C(L),C(N),C(D),C(B),C(H),C(z),C(q),C(j),C(W),C(J),C(ne),C(se),C(le),C(pe)}}}function ps(e){let t,n,a,s;return t=new Ue({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[ls]},$$scope:{ctx:e}}}),a=new Ue({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[os]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function cs(e){let t,n,a,s,p,c,m,u,g,h,v,y,S,x,A,M,P,w,T,L,E,N,U,R,D,I,B,F,G,H,V,z,O,k,q,K,j,X,Y,W;return m=new Ne({props:{align:"start",$$slots:{default:[Pa]},$$scope:{ctx:e}}}),A=new Ne({props:{align:"start",$$slots:{default:[Ia]},$$scope:{ctx:e}}}),U=new Ne({props:{align:"start",$$slots:{default:[qa]},$$scope:{ctx:e}}}),V=new Ne({props:{align:"start",$$slots:{default:[es]},$$scope:{ctx:e}}}),Y=new Ne({props:{align:"start",$$slots:{default:[ps]},$$scope:{ctx:e}}}),{c(){t=r("h1"),t.textContent="Cluster_Regulatory T cells.1",n=r("a"),n.innerHTML="",a=$(),s=r("h2"),s.textContent="Colitis_vs_Control",p=r("a"),p.innerHTML="",c=$(),f(m.$$.fragment),u=$(),g=r("h1"),g.textContent="Cluster_Tcm/Naive helper T cells",h=r("a"),h.innerHTML="",v=$(),y=r("h2"),y.textContent="Colitis_vs_Control",S=r("a"),S.innerHTML="",x=$(),f(A.$$.fragment),M=$(),P=r("h1"),P.textContent="Cluster_Tem/Trm cytotoxic T cells",w=r("a"),w.innerHTML="",T=$(),L=r("h2"),L.textContent="Colitis_vs_Control",E=r("a"),E.innerHTML="",N=$(),f(U.$$.fragment),R=$(),D=r("h1"),D.textContent="Cluster_Tem/Trm cytotoxic T cells.1",I=r("a"),I.innerHTML="",B=$(),F=r("h2"),F.textContent="Colitis_vs_Control",G=r("a"),G.innerHTML="",H=$(),f(V.$$.fragment),z=$(),O=r("h1"),O.textContent="Cluster_Trm cytotoxic T cells",k=r("a"),k.innerHTML="",q=$(),K=r("h2"),K.textContent="Colitis_vs_Control",j=r("a"),j.innerHTML="",X=$(),f(Y.$$.fragment),i(n,"id","prt-h1-5-cluster-regulatory-t-cells-1"),i(n,"class","pipen-report-toc-anchor"),i(p,"id","prt-h2-5-colitis-vs-control"),i(p,"class","pipen-report-toc-anchor"),i(h,"id","prt-h1-6-cluster-tcm-naive-helper-t-cells"),i(h,"class","pipen-report-toc-anchor"),i(S,"id","prt-h2-6-colitis-vs-control"),i(S,"class","pipen-report-toc-anchor"),i(w,"id","prt-h1-7-cluster-tem-trm-cytotoxic-t-cells"),i(w,"class","pipen-report-toc-anchor"),i(E,"id","prt-h2-7-colitis-vs-control"),i(E,"class","pipen-report-toc-anchor"),i(I,"id","prt-h1-8-cluster-tem-trm-cytotoxic-t-cells-1"),i(I,"class","pipen-report-toc-anchor"),i(G,"id","prt-h2-8-colitis-vs-control"),i(G,"class","pipen-report-toc-anchor"),i(k,"id","prt-h1-9-cluster-trm-cytotoxic-t-cells"),i(k,"class","pipen-report-toc-anchor"),i(j,"id","prt-h2-9-colitis-vs-control"),i(j,"class","pipen-report-toc-anchor")},m(e,r){l(e,t,r),l(e,n,r),l(e,a,r),l(e,s,r),l(e,p,r),l(e,c,r),b(m,e,r),l(e,u,r),l(e,g,r),l(e,h,r),l(e,v,r),l(e,y,r),l(e,S,r),l(e,x,r),b(A,e,r),l(e,M,r),l(e,P,r),l(e,w,r),l(e,T,r),l(e,L,r),l(e,E,r),l(e,N,r),b(U,e,r),l(e,R,r),l(e,D,r),l(e,I,r),l(e,B,r),l(e,F,r),l(e,G,r),l(e,H,r),b(V,e,r),l(e,z,r),l(e,O,r),l(e,k,r),l(e,q,r),l(e,K,r),l(e,j,r),l(e,X,r),b(Y,e,r),W=!0},p(e,[t]){const n={};1&t&&(n.$$scope={dirty:t,ctx:e}),m.$set(n);const a={};1&t&&(a.$$scope={dirty:t,ctx:e}),A.$set(a);const s={};1&t&&(s.$$scope={dirty:t,ctx:e}),U.$set(s);const r={};1&t&&(r.$$scope={dirty:t,ctx:e}),V.$set(r);const i={};1&t&&(i.$$scope={dirty:t,ctx:e}),Y.$set(i)},i(e){W||(o(m.$$.fragment,e),o(A.$$.fragment,e),o(U.$$.fragment,e),o(V.$$.fragment,e),o(Y.$$.fragment,e),W=!0)},o(e){d(m.$$.fragment,e),d(A.$$.fragment,e),d(U.$$.fragment,e),d(V.$$.fragment,e),d(Y.$$.fragment,e),W=!1},d(e){e&&(_(t),_(n),_(a),_(s),_(p),_(c),_(u),_(g),_(h),_(v),_(y),_(S),_(x),_(M),_(P),_(w),_(T),_(L),_(E),_(N),_(R),_(D),_(I),_(B),_(F),_(G),_(H),_(z),_(O),_(k),_(q),_(K),_(j),_(X)),C(m,e),C(A,e),C(U,e),C(V,e),C(Y,e)}}}class ds extends t{constructor(e){super(),n(this,e,null,cs,a,{})}}var _s=[{slug:"prt-h1-0-cluster-cycling-t-cells",text:"Cluster_Cycling T cells",children:[{slug:"prt-h2-0-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:0}],page:0},{slug:"prt-h1-1-cluster-follicular-helper-t-cells",text:"Cluster_Follicular helper T cells",children:[{slug:"prt-h2-1-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:0}],page:0},{slug:"prt-h1-2-cluster-gamma-delta-t-cells",text:"Cluster_gamma-delta T cells",children:[{slug:"prt-h2-2-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:0}],page:0},{slug:"prt-h1-3-cluster-plasmablasts",text:"Cluster_Plasmablasts",children:[{slug:"prt-h2-3-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:0}],page:0},{slug:"prt-h1-4-cluster-regulatory-t-cells",text:"Cluster_Regulatory T cells",children:[{slug:"prt-h2-4-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:0}],page:0},{slug:"prt-h1-5-cluster-regulatory-t-cells-1",text:"Cluster_Regulatory T cells.1",children:[{slug:"prt-h2-5-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:1}],page:1},{slug:"prt-h1-6-cluster-tcm-naive-helper-t-cells",text:"Cluster_Tcm/Naive helper T cells",children:[{slug:"prt-h2-6-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:1}],page:1},{slug:"prt-h1-7-cluster-tem-trm-cytotoxic-t-cells",text:"Cluster_Tem/Trm cytotoxic T cells",children:[{slug:"prt-h2-7-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:1}],page:1},{slug:"prt-h1-8-cluster-tem-trm-cytotoxic-t-cells-1",text:"Cluster_Tem/Trm cytotoxic T cells.1",children:[{slug:"prt-h2-8-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:1}],page:1},{slug:"prt-h1-9-cluster-trm-cytotoxic-t-cells",text:"Cluster_Trm cytotoxic T cells",children:[{slug:"prt-h2-9-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:1}],page:1},{slug:"prt-h1-10-cluster-type-1-helper-t-cells",text:"Cluster_Type 1 helper T cells",children:[{slug:"prt-h2-10-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:2}],page:2},{slug:"prt-h1-11-cluster-type-1-helper-t-cells-1",text:"Cluster_Type 1 helper T cells.1",children:[{slug:"prt-h2-11-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:2}],page:2}],ms={pipeline:{name:"Immunopipe",desc:"Immunopipe (v1.4.3): Integrative analysis for scRNA-seq and scTCR-seq data"},versions:'\n    <a href="https://github.com/pwwang/pipen" target="_blank">\n        pipen v0.15.7\n    </a> and\n    <a href="https://github.com/pwwang/pipen-report" target="_blank">\n        pipen-report v0.21.4\n    </a>\n',entries:[{name:"SampleInfo",desc:"List sample information and perform statistics",npages:1,report_toc:!0,order:0,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n[1] ggrepel_0.9.6    ggprism_1.0.5    ggplot2_3.5.1    tidyr_1.3.1     \n[5] dplyr_1.1.4      tidyselect_1.2.1 jsonlite_1.8.9   rlang_1.1.5     \n[9] logger_0.4.0    \n\nloaded via a namespace (and not attached):\n [1] vctrs_0.6.5      cli_3.6.4        purrr_1.0.2      generics_0.1.3  \n [5] glue_1.8.0       labeling_0.4.3   colorspace_2.1-1 scales_1.3.0    \n [9] grid_4.3.3       munsell_0.5.1    tibble_3.2.1     lifecycle_1.0.4 \n[13] compiler_4.3.3   Rcpp_1.0.14      pkgconfig_2.0.3  farver_2.1.2    \n[17] digest_0.6.37    R6_2.6.0         pillar_1.10.1    magrittr_2.0.3  \n[21] withr_3.0.2      gtable_0.3.6    \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 2.19\nuser 11.15\nsys 0.11\npercent 513%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3421.613\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"SeuratPreparing",desc:"Load, prepare and apply QC to data, using `Seurat`",npages:1,report_toc:!0,order:2,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats4    stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] scDblFinder_1.16.0          SingleCellExperiment_1.24.0\n [3] SummarizedExperiment_1.32.0 Biobase_2.62.0             \n [5] GenomicRanges_1.54.1        GenomeInfoDb_1.38.1        \n [7] IRanges_2.36.0              S4Vectors_0.40.2           \n [9] BiocGenerics_0.48.1         MatrixGenerics_1.14.0      \n[11] matrixStats_1.1.0           dplyr_1.1.4                \n[13] ggplot2_3.5.1               bracer_1.2.1               \n[15] future_1.34.0               Seurat_5.1.0               \n[17] SeuratObject_5.0.2          sp_2.2-0                   \n[19] digest_0.6.37               jsonlite_1.8.9             \n[21] rlang_1.1.5                 logger_0.4.0               \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22          splines_4.3.3            \n  [3] later_1.4.1               BiocIO_1.12.0            \n  [5] bitops_1.0-9              tibble_3.2.1             \n  [7] R.oo_1.27.0               polyclip_1.10-7          \n  [9] XML_3.99-0.17             fastDummies_1.7.5        \n [11] lifecycle_1.0.4           edgeR_4.0.16             \n [13] globals_0.16.3            lattice_0.22-6           \n [15] MASS_7.3-60.0.1           magrittr_2.0.3           \n [17] limma_3.58.1              plotly_4.10.4            \n [19] yaml_2.3.10               metapod_1.10.0           \n [21] httpuv_1.6.15             sctransform_0.4.1        \n [23] spam_2.11-1               spatstat.sparse_3.1-0    \n [25] reticulate_1.40.0         cowplot_1.1.3            \n [27] pbapply_1.7-2             RColorBrewer_1.1-3       \n [29] abind_1.4-5               zlibbioc_1.48.0          \n [31] Rtsne_0.17                purrr_1.0.2              \n [33] R.utils_2.12.3            RCurl_1.98-1.16          \n [35] GenomeInfoDbData_1.2.11   ggrepel_0.9.6            \n [37] irlba_2.3.5.1             listenv_0.9.1            \n [39] spatstat.utils_3.1-2      goftest_1.2-3            \n [41] RSpectra_0.16-2           dqrng_0.3.2              \n [43] spatstat.random_3.3-2     fitdistrplus_1.2-2       \n [45] parallelly_1.42.0         DelayedMatrixStats_1.24.0\n [47] leiden_0.4.3.1            codetools_0.2-20         \n [49] DelayedArray_0.28.0       scuttle_1.12.0           \n [51] tidyselect_1.2.1          farver_2.1.2             \n [53] viridis_0.6.5             ScaledMatrix_1.10.0      \n [55] spatstat.explore_3.3-4    GenomicAlignments_1.38.0 \n [57] BiocNeighbors_1.20.0      progressr_0.15.1         \n [59] scater_1.30.1             ggridges_0.5.6           \n [61] survival_3.8-3            systemfonts_1.2.1        \n [63] tools_4.3.3               ragg_1.3.3               \n [65] ica_1.0-3                 Rcpp_1.0.14              \n [67] glue_1.8.0                gridExtra_2.3            \n [69] SparseArray_1.2.2         withr_3.0.2              \n [71] fastmap_1.2.0             bluster_1.12.0           \n [73] rsvd_1.0.5                R6_2.6.0                 \n [75] mime_0.12                 textshaping_0.4.0        \n [77] colorspace_2.1-1          scattermore_1.2          \n [79] tensor_1.5                spatstat.data_3.1-4      \n [81] R.methodsS3_1.8.2         RhpcBLASctl_0.23-42      \n [83] tidyr_1.3.1               generics_0.1.3           \n [85] data.table_1.16.4         rtracklayer_1.62.0       \n [87] httr_1.4.7                htmlwidgets_1.6.4        \n [89] S4Arrays_1.2.0            uwot_0.2.2               \n [91] pkgconfig_2.0.3           gtable_0.3.6             \n [93] lmtest_0.9-40             XVector_0.42.0           \n [95] htmltools_0.5.8.1         dotCall64_1.2            \n [97] scales_1.3.0              png_0.1-8                \n [99] harmony_1.2.0             spatstat.univar_3.1-1    \n[101] scran_1.30.0              reshape2_1.4.4           \n[103] rjson_0.2.23              nlme_3.1-167             \n[105] zoo_1.8-12                stringr_1.5.1            \n[107] KernSmooth_2.23-26        vipor_0.4.7              \n[109] parallel_4.3.3            miniUI_0.1.1.1           \n[111] restfulr_0.0.15           pillar_1.10.1            \n[113] grid_4.3.3                vctrs_0.6.5              \n[115] RANN_2.6.2                promises_1.3.2           \n[117] BiocSingular_1.18.0       beachmat_2.18.0          \n[119] xtable_1.8-4              cluster_2.1.8            \n[121] beeswarm_0.4.0            locfit_1.5-9.11          \n[123] cli_3.6.4                 compiler_4.3.3           \n[125] Rsamtools_2.18.0          crayon_1.5.3             \n[127] future.apply_1.11.3       labeling_0.4.3           \n[129] ggbeeswarm_0.7.2          plyr_1.8.9               \n[131] stringi_1.8.4             viridisLite_0.4.2        \n[133] deldir_2.0-4              BiocParallel_1.36.0      \n[135] munsell_0.5.1             Biostrings_2.70.1        \n[137] lazyeval_0.2.2            spatstat.geom_3.3-5      \n[139] Matrix_1.6-5              RcppHNSW_0.6.0           \n[141] patchwork_1.3.0           sparseMatrixStats_1.14.0 \n[143] statmod_1.5.0             shiny_1.10.0             \n[145] ROCR_1.0-11               igraph_2.0.3             \n[147] xgboost_2.1.4.1          \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 209.65\nuser 1454.55\nsys 7.57\npercent 697%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ClusterMarkersOfAllCells",desc:"Markers for clusters of all cells.",npages:3,report_toc:!0,order:4,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] UpSetR_1.4.0        ggVennDiagram_1.2.3 tidyseurat_0.8.0   \n [4] ttservice_0.4.1     future_1.34.0       ggrepel_0.9.6      \n [7] ggprism_1.0.5       ggplot2_3.5.1       enrichR_3.4        \n[10] Seurat_5.1.0        SeuratObject_5.0.2  sp_2.2-0           \n[13] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[16] tidyselect_1.2.1    digest_0.6.37       jsonlite_1.8.9     \n[19] rlang_1.1.5         logger_0.4.0       \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] ellipsis_0.3.2         labeling_0.4.3         promises_1.3.2        \n [19] purrr_1.0.2            WriteXLS_6.7.0         RVenn_1.1.0           \n [22] goftest_1.2-3          later_1.4.1            spatstat.utils_3.1-2  \n [25] irlba_2.3.5.1          parallel_4.3.3         cluster_2.1.8         \n [28] R6_2.6.0               ica_1.0-3              stringi_1.8.4         \n [31] RColorBrewer_1.1-3     spatstat.data_3.1-4    limma_3.58.1          \n [34] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [37] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [40] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [43] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [46] splines_4.3.3          igraph_2.0.3           abind_1.4-5           \n [49] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.3-4 curl_6.0.1             listenv_0.9.1         \n [55] lattice_0.22-6         plyr_1.8.9             withr_3.0.2           \n [58] shiny_1.10.0           ROCR_1.0-11            Rtsne_0.17            \n [61] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n [64] fitdistrplus_1.2-2     pillar_1.10.1          KernSmooth_2.23-26    \n [67] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n [82] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n [85] nlme_3.1-167           patchwork_1.3.0        presto_1.0.0          \n [88] cli_3.6.4              spatstat.sparse_3.1-0  fansi_1.0.6           \n [91] spam_2.11-1            viridisLite_0.4.2      uwot_0.2.2            \n [94] gtable_0.3.6           progressr_0.15.1       rjson_0.2.23          \n [97] htmlwidgets_1.6.4      farver_2.1.2           htmltools_0.5.8.1     \n[100] lifecycle_1.0.4        httr_1.4.7             statmod_1.5.0         \n[103] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 259.76\nuser 186.00\nsys 8.62\npercent 74%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TopExpressingGenesOfAllCells",desc:"Top expressing genes for clusters of all cells.",npages:3,report_toc:!0,order:5,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ggprism_1.0.5      dplyr_1.1.4        enrichR_3.4        tibble_3.2.1      \n [5] Seurat_5.1.0       SeuratObject_5.0.2 sp_2.2-0           jsonlite_1.8.9    \n [9] rlang_1.1.5        logger_0.4.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] labeling_0.4.3         promises_1.3.2         purrr_1.0.2           \n [19] WriteXLS_6.7.0         goftest_1.2-3          later_1.4.1           \n [22] spatstat.utils_3.1-2   irlba_2.3.5.1          parallel_4.3.3        \n [25] cluster_2.1.8          R6_2.6.0               ica_1.0-3             \n [28] stringi_1.8.4          RColorBrewer_1.1-3     spatstat.data_3.1-4   \n [31] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [37] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [43] splines_4.3.3          igraph_2.0.3           tidyselect_1.2.1      \n [46] abind_1.4-5            spatstat.random_3.3-2  codetools_0.2-20      \n [49] miniUI_0.1.1.1         spatstat.explore_3.3-4 curl_6.0.1            \n [52] listenv_0.9.1          lattice_0.22-6         plyr_1.8.9            \n [55] withr_3.0.2            shiny_1.10.0           ROCR_1.0-11           \n [58] Rtsne_0.17             future_1.34.0          fastDummies_1.7.5     \n [61] survival_3.8-3         polyclip_1.10-7        fitdistrplus_1.2-2    \n [64] pillar_1.10.1          KernSmooth_2.23-26     plotly_4.10.4         \n [67] generics_0.1.3         RcppHNSW_0.6.0         ggplot2_3.5.1         \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n [82] cowplot_1.1.3          grid_4.3.3             tidyr_1.3.1           \n [85] colorspace_2.1-1       nlme_3.1-167           patchwork_1.3.0       \n [88] cli_3.6.4              spatstat.sparse_3.1-0  spam_2.11-1           \n [91] viridisLite_0.4.2      uwot_0.2.2             gtable_0.3.6          \n [94] digest_0.6.37          progressr_0.15.1       ggrepel_0.9.6         \n [97] rjson_0.2.23           htmlwidgets_1.6.4      farver_2.1.2          \n[100] htmltools_0.5.8.1      lifecycle_1.0.4        httr_1.4.7            \n[103] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 73.70\nuser 41.45\nsys 2.53\npercent 59%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TCellSelection",desc:"Separate T and non-T cells and select T cells.",npages:1,report_toc:!0,order:6,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] glue_1.8.0         factoextra_1.0.7   patchwork_1.3.0    ggpubr_0.6.0      \n [5] ggrepel_0.9.6      ggprism_1.0.5      ggplot2_3.5.1      tibble_3.2.1      \n [9] tidyr_1.3.1        dplyr_1.1.4        Seurat_5.1.0       SeuratObject_5.0.2\n[13] sp_2.2-0           jsonlite_1.8.9     rlang_1.1.5        logger_0.4.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] backports_1.5.0        labeling_0.4.3         promises_1.3.2        \n [19] purrr_1.0.2            goftest_1.2-3          later_1.4.1           \n [22] spatstat.utils_3.1-2   broom_1.0.7            irlba_2.3.5.1         \n [25] parallel_4.3.3         cluster_2.1.8          R6_2.6.0              \n [28] ica_1.0-3              stringi_1.8.4          RColorBrewer_1.1-3    \n [31] spatstat.data_3.1-4    reticulate_1.40.0      car_3.1-3             \n [34] parallelly_1.42.0      spatstat.univar_3.1-1  lmtest_0.9-40         \n [37] scattermore_1.2        Rcpp_1.0.14            tensor_1.5            \n [40] future.apply_1.11.3    zoo_1.8-12             sctransform_0.4.1     \n [43] httpuv_1.6.15          Matrix_1.6-5           splines_4.3.3         \n [46] igraph_2.0.3           tidyselect_1.2.1       abind_1.4-5           \n [49] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.3-4 listenv_0.9.1          lattice_0.22-6        \n [55] plyr_1.8.9             withr_3.0.2            shiny_1.10.0          \n [58] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n [61] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n [64] fitdistrplus_1.2-2     pillar_1.10.1          carData_3.0-5         \n [67] KernSmooth_2.23-26     plotly_4.10.4          generics_0.1.3        \n [70] RcppHNSW_0.6.0         munsell_0.5.1          scales_1.3.0          \n [73] globals_0.16.3         xtable_1.8-4           lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] ggsignif_0.6.4         RANN_2.6.2             leiden_0.4.3.1        \n [82] dotCall64_1.2          cowplot_1.1.3          grid_4.3.3            \n [85] colorspace_2.1-1       nlme_3.1-167           Formula_1.2-5         \n [88] cli_3.6.4              spatstat.sparse_3.1-0  spam_2.11-1           \n [91] viridisLite_0.4.2      uwot_0.2.2             gtable_0.3.6          \n [94] rstatix_0.7.2          digest_0.6.37          progressr_0.15.1      \n [97] htmlwidgets_1.6.4      farver_2.1.2           htmltools_0.5.8.1     \n[100] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[103] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 41.63\nuser 49.85\nsys 0.77\npercent 121%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ClusterMarkers",desc:"Markers for clusters of all or selected T cells.",npages:2,report_toc:!0,order:10,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] UpSetR_1.4.0        ggVennDiagram_1.2.3 tidyseurat_0.8.0   \n [4] ttservice_0.4.1     future_1.34.0       ggrepel_0.9.6      \n [7] ggprism_1.0.5       ggplot2_3.5.1       enrichR_3.4        \n[10] Seurat_5.1.0        SeuratObject_5.0.2  sp_2.2-0           \n[13] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[16] tidyselect_1.2.1    digest_0.6.37       jsonlite_1.8.9     \n[19] rlang_1.1.5         logger_0.4.0       \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] ellipsis_0.3.2         labeling_0.4.3         promises_1.3.2        \n [19] purrr_1.0.2            WriteXLS_6.7.0         RVenn_1.1.0           \n [22] goftest_1.2-3          later_1.4.1            spatstat.utils_3.1-2  \n [25] irlba_2.3.5.1          parallel_4.3.3         cluster_2.1.8         \n [28] R6_2.6.0               ica_1.0-3              stringi_1.8.4         \n [31] RColorBrewer_1.1-3     spatstat.data_3.1-4    limma_3.58.1          \n [34] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [37] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [40] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [43] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [46] splines_4.3.3          igraph_2.0.3           abind_1.4-5           \n [49] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.3-4 curl_6.0.1             listenv_0.9.1         \n [55] lattice_0.22-6         plyr_1.8.9             withr_3.0.2           \n [58] shiny_1.10.0           ROCR_1.0-11            Rtsne_0.17            \n [61] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n [64] fitdistrplus_1.2-2     pillar_1.10.1          KernSmooth_2.23-26    \n [67] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n [82] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n [85] nlme_3.1-167           patchwork_1.3.0        presto_1.0.0          \n [88] cli_3.6.4              spatstat.sparse_3.1-0  fansi_1.0.6           \n [91] spam_2.11-1            viridisLite_0.4.2      uwot_0.2.2            \n [94] gtable_0.3.6           progressr_0.15.1       rjson_0.2.23          \n [97] htmlwidgets_1.6.4      farver_2.1.2           htmltools_0.5.8.1     \n[100] lifecycle_1.0.4        httr_1.4.7             statmod_1.5.0         \n[103] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 162.57\nuser 119.15\nsys 5.40\npercent 76%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TopExpressingGenes",desc:"Top expressing genes for clusters of all or selected T cells.",npages:2,report_toc:!0,order:11,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ggprism_1.0.5      dplyr_1.1.4        enrichR_3.4        tibble_3.2.1      \n [5] Seurat_5.1.0       SeuratObject_5.0.2 sp_2.2-0           jsonlite_1.8.9    \n [9] rlang_1.1.5        logger_0.4.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] labeling_0.4.3         promises_1.3.2         purrr_1.0.2           \n [19] WriteXLS_6.7.0         goftest_1.2-3          later_1.4.1           \n [22] spatstat.utils_3.1-2   irlba_2.3.5.1          parallel_4.3.3        \n [25] cluster_2.1.8          R6_2.6.0               ica_1.0-3             \n [28] stringi_1.8.4          RColorBrewer_1.1-3     spatstat.data_3.1-4   \n [31] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [37] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [43] splines_4.3.3          igraph_2.0.3           tidyselect_1.2.1      \n [46] abind_1.4-5            spatstat.random_3.3-2  codetools_0.2-20      \n [49] miniUI_0.1.1.1         spatstat.explore_3.3-4 curl_6.0.1            \n [52] listenv_0.9.1          lattice_0.22-6         plyr_1.8.9            \n [55] withr_3.0.2            shiny_1.10.0           ROCR_1.0-11           \n [58] Rtsne_0.17             future_1.34.0          fastDummies_1.7.5     \n [61] survival_3.8-3         polyclip_1.10-7        fitdistrplus_1.2-2    \n [64] pillar_1.10.1          KernSmooth_2.23-26     plotly_4.10.4         \n [67] generics_0.1.3         RcppHNSW_0.6.0         ggplot2_3.5.1         \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n [82] cowplot_1.1.3          grid_4.3.3             tidyr_1.3.1           \n [85] colorspace_2.1-1       nlme_3.1-167           patchwork_1.3.0       \n [88] cli_3.6.4              spatstat.sparse_3.1-0  spam_2.11-1           \n [91] viridisLite_0.4.2      uwot_0.2.2             gtable_0.3.6          \n [94] digest_0.6.37          progressr_0.15.1       ggrepel_0.9.6         \n [97] rjson_0.2.23           htmlwidgets_1.6.4      farver_2.1.2          \n[100] htmltools_0.5.8.1      lifecycle_1.0.4        httr_1.4.7            \n[103] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 45.21\nuser 28.74\nsys 1.92\npercent 67%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TCRClusterStats",desc:"Statistics of TCR clusters, generated by `TCRClustering`.",npages:1,report_toc:!0,order:13,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] grid      stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] ComplexHeatmap_2.18.0 ggprism_1.0.5         immunarch_0.9.1      \n [4] patchwork_1.3.0       data.table_1.16.4     dtplyr_1.3.1         \n [7] dplyr_1.1.4           tidyr_1.3.1           tibble_3.2.1         \n[10] ggplot2_3.5.1         jsonlite_1.8.9        rlang_1.1.5          \n[13] logger_0.4.0         \n\nloaded via a namespace (and not attached):\n  [1] gridExtra_2.3       phangorn_2.12.1     readxl_1.4.3       \n  [4] magrittr_2.0.3      clue_0.3-66         GetoptLong_1.0.5   \n  [7] matrixStats_1.1.0   compiler_4.3.3      flexmix_2.3-19     \n [10] png_0.1-8           vctrs_0.6.5         reshape2_1.4.4     \n [13] ggalluvial_0.12.5   quadprog_1.5-8      stringr_1.5.1      \n [16] crayon_1.5.3        pkgconfig_2.0.3     shape_1.4.6.1      \n [19] fastmap_1.2.0       backports_1.5.0     labeling_0.4.3     \n [22] ggraph_2.2.1        promises_1.3.2      tzdb_0.4.0         \n [25] UpSetR_1.4.0        purrr_1.0.2         modeltools_0.2-23  \n [28] ggseqlogo_0.2       cachem_1.1.0        later_1.4.1        \n [31] uuid_1.2-1          fpc_2.2-13          tweenr_2.0.3       \n [34] broom_1.0.7         parallel_4.3.3      prabclus_2.3-4     \n [37] cluster_2.1.8       R6_2.6.0            stringi_1.8.4      \n [40] RColorBrewer_1.1-3  rlist_0.4.6.2       car_3.1-3          \n [43] diptest_0.77-1      cellranger_1.1.0    Rcpp_1.0.14        \n [46] iterators_1.0.14    IRanges_2.36.0      readr_2.1.5        \n [49] httpuv_1.6.15       Matrix_1.6-5        nnet_7.3-20        \n [52] igraph_2.0.3        tidyselect_1.2.1    stringdist_0.9.15  \n [55] abind_1.4-5         viridis_0.6.5       doParallel_1.0.17  \n [58] codetools_0.2-20    lattice_0.22-6      plyr_1.8.9         \n [61] shiny_1.10.0        withr_3.0.2         polyclip_1.10-7    \n [64] circlize_0.4.16     mclust_6.1.1        kernlab_0.9-33     \n [67] pillar_1.10.1       ggpubr_0.6.0        carData_3.0-5      \n [70] foreach_1.5.2       stats4_4.3.3        generics_0.1.3     \n [73] S4Vectors_0.40.2    hms_1.1.3           munsell_0.5.1      \n [76] scales_1.3.0        xtable_1.8-4        class_7.3-23       \n [79] glue_1.8.0          pheatmap_1.0.12     tools_4.3.3        \n [82] robustbase_0.99-4-1 ggsignif_0.6.4      graphlayouts_1.2.2 \n [85] Cairo_1.6-2         fastmatch_1.1-6     tidygraph_1.3.0    \n [88] ape_5.8-1           colorspace_2.1-1    nlme_3.1-167       \n [91] ggforce_0.4.2       Formula_1.2-5       cli_3.6.4          \n [94] viridisLite_0.4.2   gtable_0.3.6        DEoptimR_1.1-3-1   \n [97] rstatix_0.7.2       digest_0.6.37       BiocGenerics_0.48.1\n[100] ggrepel_0.9.6       rjson_0.2.23        farver_2.1.2       \n[103] memoise_2.0.1       htmltools_0.5.8.1   lifecycle_1.0.4    \n[106] factoextra_1.0.7    mime_0.12           GlobalOptions_0.1.2\n[109] shinythemes_1.2.0   MASS_7.3-60.0.1    \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 7.17\nuser 15.93\nsys 0.32\npercent 226%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3287.293\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TESSA",desc:"Tessa is a Bayesian model to integrate T cell receptor (TCR) sequence profiling with transcriptomes of T cells.",npages:1,report_toc:!0,order:14,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] igraph_2.0.3         Rtsne_0.17           LaplacesDemon_16.1.6\n [4] MASS_7.3-60.0.1      ggprism_1.0.5        Seurat_5.1.0        \n [7] SeuratObject_5.0.2   sp_2.2-0             tibble_3.2.1        \n[10] glue_1.8.0           immunarch_0.9.1      patchwork_1.3.0     \n[13] data.table_1.16.4    dtplyr_1.3.1         ggplot2_3.5.1       \n[16] tidyr_1.3.1          dplyr_1.1.4          jsonlite_1.8.9      \n[19] rlang_1.1.5          logger_0.4.0        \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22       shinythemes_1.2.0      splines_4.3.3         \n  [4] later_1.4.1            cellranger_1.1.0       polyclip_1.10-7       \n  [7] fastDummies_1.7.5      factoextra_1.0.7       lifecycle_1.0.4       \n [10] rstatix_0.7.2          doParallel_1.0.17      globals_0.16.3        \n [13] lattice_0.22-6         prabclus_2.3-4         backports_1.5.0       \n [16] magrittr_2.0.3         plotly_4.10.4          rlist_0.4.6.2         \n [19] httpuv_1.6.15          sctransform_0.4.1      spam_2.11-1           \n [22] flexmix_2.3-19         spatstat.sparse_3.1-0  reticulate_1.40.0     \n [25] cowplot_1.1.3          pbapply_1.7-2          RColorBrewer_1.1-3    \n [28] abind_1.4-5            quadprog_1.5-8         purrr_1.0.2           \n [31] ggraph_2.2.1           nnet_7.3-20            tweenr_2.0.3          \n [34] circlize_0.4.16        ggrepel_0.9.6          irlba_2.3.5.1         \n [37] listenv_0.9.1          spatstat.utils_3.1-2   pheatmap_1.0.12       \n [40] goftest_1.2-3          RSpectra_0.16-2        spatstat.random_3.3-2 \n [43] fitdistrplus_1.2-2     parallelly_1.42.0      leiden_0.4.3.1        \n [46] codetools_0.2-20       ggforce_0.4.2          tidyselect_1.2.1      \n [49] shape_1.4.6.1          ggseqlogo_0.2          farver_2.1.2          \n [52] viridis_0.6.5          matrixStats_1.1.0      stats4_4.3.3          \n [55] spatstat.explore_3.3-4 tidygraph_1.3.0        progressr_0.15.1      \n [58] Formula_1.2-5          ggridges_0.5.6         ggalluvial_0.12.5     \n [61] survival_3.8-3         iterators_1.0.14       foreach_1.5.2         \n [64] tools_4.3.3            stringdist_0.9.15      ica_1.0-3             \n [67] Rcpp_1.0.14            gridExtra_2.3          withr_3.0.2           \n [70] fastmap_1.2.0          digest_0.6.37          R6_2.6.0              \n [73] mime_0.12              colorspace_2.1-1       scattermore_1.2       \n [76] tensor_1.5             spatstat.data_3.1-4    diptest_0.77-1        \n [79] UpSetR_1.4.0           generics_0.1.3         robustbase_0.99-4-1   \n [82] class_7.3-23           graphlayouts_1.2.2     httr_1.4.7            \n [85] htmlwidgets_1.6.4      uwot_0.2.2             pkgconfig_2.0.3       \n [88] gtable_0.3.6           modeltools_0.2-23      lmtest_0.9-40         \n [91] htmltools_0.5.8.1      carData_3.0-5          dotCall64_1.2         \n [94] scales_1.3.0           png_0.1-8              spatstat.univar_3.1-1 \n [97] tzdb_0.4.0             reshape2_1.4.4         uuid_1.2-1            \n[100] nlme_3.1-167           cachem_1.1.0           zoo_1.8-12            \n[103] GlobalOptions_0.1.2    stringr_1.5.1          KernSmooth_2.23-26    \n[106] parallel_4.3.3         miniUI_0.1.1.1         pillar_1.10.1         \n[109] grid_4.3.3             vctrs_0.6.5            RANN_2.6.2            \n[112] promises_1.3.2         ggpubr_0.6.0           car_3.1-3             \n[115] xtable_1.8-4           cluster_2.1.8          readr_2.1.5           \n[118] cli_3.6.4              compiler_4.3.3         future.apply_1.11.3   \n[121] ggsignif_0.6.4         labeling_0.4.3         mclust_6.1.1          \n[124] plyr_1.8.9             stringi_1.8.4          deldir_2.0-4          \n[127] viridisLite_0.4.2      munsell_0.5.1          lazyeval_0.2.2        \n[130] spatstat.geom_3.3-5    Matrix_1.6-5           RcppHNSW_0.6.0        \n[133] hms_1.1.3              future_1.34.0          fpc_2.2-13            \n[136] shiny_1.10.0           kernlab_0.9-33         ROCR_1.0-11           \n[139] broom_1.0.7            memoise_2.0.1          phangorn_2.12.1       \n[142] fastmatch_1.1-6        DEoptimR_1.1-3-1       readxl_1.4.3          \n[145] ape_5.8-1             \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 1679.73\nuser 2269.05\nsys 21.01\npercent 136%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"SeuratClusterStats",desc:"Statistics of the clustering.",npages:1,report_toc:!0,order:17,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] grid      stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] clustree_0.5.1        ggraph_2.2.1          ComplexHeatmap_2.18.0\n [4] circlize_0.4.16       tidyseurat_0.8.0      ttservice_0.4.1      \n [7] ggrepel_0.9.6         ggprism_1.0.5         tibble_3.2.1         \n[10] Seurat_5.1.0          SeuratObject_5.0.2    sp_2.2-0             \n[13] ggplot2_3.5.1         tidyr_1.3.1           dplyr_1.1.4          \n[16] tidyselect_1.2.1      jsonlite_1.8.9        rlang_1.1.5          \n[19] logger_0.4.0         \n\nloaded via a namespace (and not attached):\n  [1] RColorBrewer_1.1-3     shape_1.4.6.1          magrittr_2.0.3        \n  [4] ggbeeswarm_0.7.2       spatstat.utils_3.1-2   farver_2.1.2          \n  [7] GlobalOptions_0.1.2    vctrs_0.6.5            ROCR_1.0-11           \n [10] memoise_2.0.1          spatstat.explore_3.3-4 htmltools_0.5.8.1     \n [13] sctransform_0.4.1      parallelly_1.42.0      KernSmooth_2.23-26    \n [16] htmlwidgets_1.6.4      ica_1.0-3              plyr_1.8.9            \n [19] cachem_1.1.0           plotly_4.10.4          zoo_1.8-12            \n [22] igraph_2.0.3           mime_0.12              lifecycle_1.0.4       \n [25] iterators_1.0.14       pkgconfig_2.0.3        Matrix_1.6-5          \n [28] R6_2.6.0               fastmap_1.2.0          clue_0.3-66           \n [31] fitdistrplus_1.2-2     future_1.34.0          shiny_1.10.0          \n [34] digest_0.6.37          colorspace_2.1-1       patchwork_1.3.0       \n [37] S4Vectors_0.40.2       tensor_1.5             RSpectra_0.16-2       \n [40] irlba_2.3.5.1          labeling_0.4.3         progressr_0.15.1      \n [43] fansi_1.0.6            spatstat.sparse_3.1-0  httr_1.4.7            \n [46] polyclip_1.10-7        abind_1.4-5            compiler_4.3.3        \n [49] withr_3.0.2            doParallel_1.0.17      viridis_0.6.5         \n [52] fastDummies_1.7.5      ggforce_0.4.2          MASS_7.3-60.0.1       \n [55] rjson_0.2.23           tools_4.3.3            vipor_0.4.7           \n [58] lmtest_0.9-40          beeswarm_0.4.0         httpuv_1.6.15         \n [61] future.apply_1.11.3    goftest_1.2-3          glue_1.8.0            \n [64] nlme_3.1-167           promises_1.3.2         Rtsne_0.17            \n [67] cluster_2.1.8          reshape2_1.4.4         generics_0.1.3        \n [70] gtable_0.3.6           spatstat.data_3.1-4    data.table_1.16.4     \n [73] tidygraph_1.3.0        BiocGenerics_0.48.1    spatstat.geom_3.3-5   \n [76] RcppAnnoy_0.0.22       RANN_2.6.2             foreach_1.5.2         \n [79] pillar_1.10.1          stringr_1.5.1          spam_2.11-1           \n [82] RcppHNSW_0.6.0         later_1.4.1            splines_4.3.3         \n [85] tweenr_2.0.3           lattice_0.22-6         survival_3.8-3        \n [88] deldir_2.0-4           miniUI_0.1.1.1         pbapply_1.7-2         \n [91] gridExtra_2.3          IRanges_2.36.0         scattermore_1.2       \n [94] stats4_4.3.3           graphlayouts_1.2.2     matrixStats_1.1.0     \n [97] stringi_1.8.4          lazyeval_0.2.2         codetools_0.2-20      \n[100] cli_3.6.4              uwot_0.2.2             xtable_1.8-4          \n[103] reticulate_1.40.0      munsell_0.5.1          Rcpp_1.0.14           \n[106] globals_0.16.3         spatstat.random_3.3-2  png_0.1-8             \n[109] ggrastr_1.0.2          spatstat.univar_3.1-1  parallel_4.3.3        \n[112] ellipsis_0.3.2         dotCall64_1.2          listenv_0.9.1         \n[115] viridisLite_0.4.2      scales_1.3.0           ggridges_0.5.6        \n[118] leiden_0.4.3.1         purrr_1.0.2            crayon_1.5.3          \n[121] GetoptLong_1.0.5       cowplot_1.1.3         \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 18.76\nuser 27.07\nsys 0.71\npercent 148%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"Immunarch",desc:"Exploration of Single-cell and Bulk T-cell/Antibody Immune Repertoires",npages:3,report_toc:!0,order:18,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] circlize_0.4.16   tibble_3.2.1      glue_1.8.0        ggprism_1.0.5    \n [5] immunarch_0.9.1   patchwork_1.3.0   data.table_1.16.4 dtplyr_1.3.1     \n [9] ggplot2_3.5.1     tidyr_1.3.1       dplyr_1.1.4       jsonlite_1.8.9   \n[13] rlang_1.1.5       logger_0.4.0     \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22       shinythemes_1.2.0      splines_4.3.3         \n  [4] later_1.4.1            cellranger_1.1.0       polyclip_1.10-7       \n  [7] fastDummies_1.7.5      factoextra_1.0.7       lifecycle_1.0.4       \n [10] rstatix_0.7.2          doParallel_1.0.17      globals_0.16.3        \n [13] lattice_0.22-6         prabclus_2.3-4         MASS_7.3-60.0.1       \n [16] backports_1.5.0        magrittr_2.0.3         plotly_4.10.4         \n [19] rlist_0.4.6.2          httpuv_1.6.15          Seurat_5.1.0          \n [22] sctransform_0.4.1      spam_2.11-1            flexmix_2.3-19        \n [25] spatstat.sparse_3.1-0  sp_2.2-0               reticulate_1.40.0     \n [28] cowplot_1.1.3          pbapply_1.7-2          RColorBrewer_1.1-3    \n [31] abind_1.4-5            Rtsne_0.17             quadprog_1.5-8        \n [34] purrr_1.0.2            ggraph_2.2.1           nnet_7.3-20           \n [37] tweenr_2.0.3           ggrepel_0.9.6          irlba_2.3.5.1         \n [40] spatstat.utils_3.1-2   listenv_0.9.1          pheatmap_1.0.12       \n [43] goftest_1.2-3          RSpectra_0.16-2        spatstat.random_3.3-2 \n [46] fitdistrplus_1.2-2     parallelly_1.42.0      leiden_0.4.3.1        \n [49] codetools_0.2-20       ggforce_0.4.2          tidyselect_1.2.1      \n [52] shape_1.4.6.1          ggseqlogo_0.2          farver_2.1.2          \n [55] viridis_0.6.5          matrixStats_1.1.0      stats4_4.3.3          \n [58] spatstat.explore_3.3-4 tidygraph_1.3.0        progressr_0.15.1      \n [61] Formula_1.2-5          ggridges_0.5.6         ggalluvial_0.12.5     \n [64] survival_3.8-3         iterators_1.0.14       foreach_1.5.2         \n [67] tools_4.3.3            stringdist_0.9.15      ica_1.0-3             \n [70] Rcpp_1.0.14            gridExtra_2.3          withr_3.0.2           \n [73] fastmap_1.2.0          digest_0.6.37          R6_2.6.0              \n [76] mime_0.12              colorspace_2.1-1       scattermore_1.2       \n [79] tensor_1.5             spatstat.data_3.1-4    diptest_0.77-1        \n [82] UpSetR_1.4.0           generics_0.1.3         robustbase_0.99-4-1   \n [85] class_7.3-23           graphlayouts_1.2.2     httr_1.4.7            \n [88] htmlwidgets_1.6.4      uwot_0.2.2             pkgconfig_2.0.3       \n [91] gtable_0.3.6           modeltools_0.2-23      lmtest_0.9-40         \n [94] htmltools_0.5.8.1      carData_3.0-5          dotCall64_1.2         \n [97] SeuratObject_5.0.2     scales_1.3.0           png_0.1-8             \n[100] spatstat.univar_3.1-1  tzdb_0.4.0             reshape2_1.4.4        \n[103] uuid_1.2-1             nlme_3.1-167           cachem_1.1.0          \n[106] zoo_1.8-12             GlobalOptions_0.1.2    stringr_1.5.1         \n[109] KernSmooth_2.23-26     parallel_4.3.3         miniUI_0.1.1.1        \n[112] pillar_1.10.1          grid_4.3.3             vctrs_0.6.5           \n[115] RANN_2.6.2             promises_1.3.2         ggpubr_0.6.0          \n[118] car_3.1-3              xtable_1.8-4           cluster_2.1.8         \n[121] readr_2.1.5            cli_3.6.4              compiler_4.3.3        \n[124] future.apply_1.11.3    ggsignif_0.6.4         labeling_0.4.3        \n[127] mclust_6.1.1           plyr_1.8.9             stringi_1.8.4         \n[130] deldir_2.0-4           viridisLite_0.4.2      assertthat_0.2.1      \n[133] munsell_0.5.1          lazyeval_0.2.2         spatstat.geom_3.3-5   \n[136] Matrix_1.6-5           RcppHNSW_0.6.0         hms_1.1.3             \n[139] future_1.34.0          fpc_2.2-13             shiny_1.10.0          \n[142] kernlab_0.9-33         ROCR_1.0-11            igraph_2.0.3          \n[145] broom_1.0.7            memoise_2.0.1          phangorn_2.12.1       \n[148] fastmatch_1.1-6        DEoptimR_1.1-3-1       readxl_1.4.3          \n[151] ape_5.8-1             \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 102.43\nuser 256.71\nsys 5.18\npercent 255%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"RadarPlots",desc:"Radar plots for cell proportion in different clusters.",npages:1,report_toc:!0,order:19,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ggprism_1.0.5      ggradar_0.2        ggplot2_3.5.1      tibble_3.2.1      \n [5] tidyr_1.3.1        dplyr_1.1.4        Seurat_5.1.0       SeuratObject_5.0.2\n [9] sp_2.2-0           jsonlite_1.8.9     rlang_1.1.5        logger_0.4.0      \n\nloaded via a namespace (and not attached):\n [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n[10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n[13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n[16] labeling_0.4.3         promises_1.3.2         purrr_1.0.2           \n[19] goftest_1.2-3          later_1.4.1            spatstat.utils_3.1-2  \n[22] irlba_2.3.5.1          parallel_4.3.3         cluster_2.1.8         \n[25] R6_2.6.0               ica_1.0-3              stringi_1.8.4         \n[28] RColorBrewer_1.1-3     spatstat.data_3.1-4    reticulate_1.40.0     \n[31] parallelly_1.42.0      spatstat.univar_3.1-1  lmtest_0.9-40         \n[34] scattermore_1.2        Rcpp_1.0.14            tensor_1.5            \n[37] future.apply_1.11.3    zoo_1.8-12             sctransform_0.4.1     \n[40] httpuv_1.6.15          Matrix_1.6-5           splines_4.3.3         \n[43] igraph_2.0.3           tidyselect_1.2.1       abind_1.4-5           \n[46] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n[49] spatstat.explore_3.3-4 listenv_0.9.1          lattice_0.22-6        \n[52] plyr_1.8.9             withr_3.0.2            shiny_1.10.0          \n[55] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n[58] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n[61] fitdistrplus_1.2-2     pillar_1.10.1          KernSmooth_2.23-26    \n[64] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n[67] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n[70] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n[73] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n[76] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n[79] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n[82] nlme_3.1-167           patchwork_1.3.0        cli_3.6.4             \n[85] spatstat.sparse_3.1-0  spam_2.11-1            viridisLite_0.4.2     \n[88] uwot_0.2.2             gtable_0.3.6           digest_0.6.37         \n[91] progressr_0.15.1       ggrepel_0.9.6          htmlwidgets_1.6.4     \n[94] farver_2.1.2           htmltools_0.5.8.1      lifecycle_1.0.4       \n[97] httr_1.4.7             mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 8.07\nuser 16.74\nsys 0.38\npercent 212%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"CloneResidency",desc:"Identification of clone residency",npages:1,report_toc:!0,order:20,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ComplexUpset_1.3.3  ggVennDiagram_1.2.3 ggprism_1.0.5      \n [4] ggplot2_3.5.1       ggnewscale_0.5.0    glue_1.8.0         \n [7] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[10] jsonlite_1.8.9      rlang_1.1.5         logger_0.4.0       \n\nloaded via a namespace (and not attached):\n  [1] DBI_1.2.3              deldir_2.0-4           pbapply_1.7-2         \n  [4] gridExtra_2.3          magrittr_2.0.3         RcppAnnoy_0.0.22      \n  [7] e1071_1.7-16           spatstat.geom_3.3-5    matrixStats_1.1.0     \n [10] ggridges_0.5.6         compiler_4.3.3         png_0.1-8             \n [13] vctrs_0.6.5            reshape2_1.4.4         stringr_1.5.1         \n [16] pkgconfig_2.0.3        fastmap_1.2.0          labeling_0.4.3        \n [19] promises_1.3.2         purrr_1.0.2            RVenn_1.1.0           \n [22] goftest_1.2-3          later_1.4.1            spatstat.utils_3.1-2  \n [25] irlba_2.3.5.1          parallel_4.3.3         cluster_2.1.8         \n [28] R6_2.6.0               ica_1.0-3              stringi_1.8.4         \n [31] RColorBrewer_1.1-3     spatstat.data_3.1-4    reticulate_1.40.0     \n [34] parallelly_1.42.0      spatstat.univar_3.1-1  lmtest_0.9-40         \n [37] scattermore_1.2        Rcpp_1.0.14            tensor_1.5            \n [40] future.apply_1.11.3    zoo_1.8-12             sctransform_0.4.1     \n [43] httpuv_1.6.15          Matrix_1.6-5           splines_4.3.3         \n [46] igraph_2.0.3           tidyselect_1.2.1       abind_1.4-5           \n [49] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.3-4 listenv_0.9.1          lattice_0.22-6        \n [55] plyr_1.8.9             shiny_1.10.0           withr_3.0.2           \n [58] ROCR_1.0-11            Rtsne_0.17             sf_1.0-19             \n [61] future_1.34.0          fastDummies_1.7.5      survival_3.8-3        \n [64] units_0.8-5            proxy_0.4-27           polyclip_1.10-7       \n [67] fitdistrplus_1.2-2     pillar_1.10.1          Seurat_5.1.0          \n [70] KernSmooth_2.23-26     plotly_4.10.4          generics_0.1.3        \n [73] RcppHNSW_0.6.0         sp_2.2-0               munsell_0.5.1         \n [76] scales_1.3.0           globals_0.16.3         xtable_1.8-4          \n [79] class_7.3-23           lazyeval_0.2.2         tools_4.3.3           \n [82] data.table_1.16.4      RSpectra_0.16-2        RANN_2.6.2            \n [85] leiden_0.4.3.1         dotCall64_1.2          cowplot_1.1.3         \n [88] grid_4.3.3             colorspace_2.1-1       nlme_3.1-167          \n [91] patchwork_1.3.0        cli_3.6.4              spatstat.sparse_3.1-0 \n [94] spam_2.11-1            viridisLite_0.4.2      uwot_0.2.2            \n [97] gtable_0.3.6           digest_0.6.37          classInt_0.4-11       \n[100] progressr_0.15.1       ggrepel_0.9.6          htmlwidgets_1.6.4     \n[103] SeuratObject_5.0.2     farver_2.1.2           htmltools_0.5.8.1     \n[106] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[109] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 19.89\nuser 28.24\nsys 0.66\npercent 145%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ScFGSEA",desc:"Gene set enrichment analysis for cells in different groups using `fgsea`",npages:2,report_toc:!0,order:21,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] fgsea_1.28.0       data.table_1.16.4  matrixStats_1.1.0  tidyseurat_0.8.0  \n [5] ttservice_0.4.1    Seurat_5.1.0       SeuratObject_5.0.2 sp_2.2-0          \n [9] tidyr_1.3.1        tidyselect_1.2.1   tibble_3.2.1       dplyr_1.1.4       \n[13] ggplot2_3.5.1      jsonlite_1.8.9     rlang_1.1.5        logger_0.4.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       ggridges_0.5.6        \n  [7] compiler_4.3.3         spatstat.geom_3.3-5    png_0.1-8             \n [10] vctrs_0.6.5            reshape2_1.4.4         stringr_1.5.1         \n [13] pkgconfig_2.0.3        fastmap_1.2.0          ellipsis_0.3.2        \n [16] labeling_0.4.3         promises_1.3.2         purrr_1.0.2           \n [19] goftest_1.2-3          later_1.4.1            BiocParallel_1.36.0   \n [22] spatstat.utils_3.1-2   irlba_2.3.5.1          parallel_4.3.3        \n [25] cluster_2.1.8          R6_2.6.0               ica_1.0-3             \n [28] stringi_1.8.4          RColorBrewer_1.1-3     spatstat.data_3.1-4   \n [31] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [37] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [43] splines_4.3.3          igraph_2.0.3           abind_1.4-5           \n [46] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n [49] spatstat.explore_3.3-4 listenv_0.9.1          lattice_0.22-6        \n [52] plyr_1.8.9             shiny_1.10.0           withr_3.0.2           \n [55] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n [58] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n [61] fitdistrplus_1.2-2     pillar_1.10.1          KernSmooth_2.23-26    \n [64] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n [67] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [70] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [73] tools_4.3.3            RSpectra_0.16-2        RANN_2.6.2            \n [76] leiden_0.4.3.1         dotCall64_1.2          fastmatch_1.1-6       \n [79] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n [82] nlme_3.1-167           patchwork_1.3.0        cli_3.6.4             \n [85] spatstat.sparse_3.1-0  fansi_1.0.6            spam_2.11-1           \n [88] viridisLite_0.4.2      uwot_0.2.2             gtable_0.3.6          \n [91] digest_0.6.37          progressr_0.15.1       ggrepel_0.9.6         \n [94] htmlwidgets_1.6.4      farver_2.1.2           htmltools_0.5.8.1     \n [97] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[100] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 128.45\nuser 135.35\nsys 1.52\npercent 106%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"MarkersFinder",desc:"Find markers between different groups of cells",npages:1,report_toc:!0,order:22,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] UpSetR_1.4.0        ggVennDiagram_1.2.3 tidyseurat_0.8.0   \n [4] ttservice_0.4.1     future_1.34.0       ggrepel_0.9.6      \n [7] ggprism_1.0.5       ggplot2_3.5.1       enrichR_3.4        \n[10] Seurat_5.1.0        SeuratObject_5.0.2  sp_2.2-0           \n[13] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[16] tidyselect_1.2.1    digest_0.6.37       jsonlite_1.8.9     \n[19] rlang_1.1.5         logger_0.4.0       \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] ellipsis_0.3.2         labeling_0.4.3         promises_1.3.2        \n [19] purrr_1.0.2            WriteXLS_6.7.0         RVenn_1.1.0           \n [22] goftest_1.2-3          later_1.4.1            spatstat.utils_3.1-2  \n [25] irlba_2.3.5.1          parallel_4.3.3         cluster_2.1.8         \n [28] R6_2.6.0               ica_1.0-3              stringi_1.8.4         \n [31] RColorBrewer_1.1-3     spatstat.data_3.1-4    limma_3.58.1          \n [34] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [37] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [40] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [43] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [46] splines_4.3.3          igraph_2.0.3           abind_1.4-5           \n [49] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.3-4 curl_6.0.1             listenv_0.9.1         \n [55] lattice_0.22-6         plyr_1.8.9             withr_3.0.2           \n [58] shiny_1.10.0           ROCR_1.0-11            Rtsne_0.17            \n [61] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n [64] fitdistrplus_1.2-2     pillar_1.10.1          KernSmooth_2.23-26    \n [67] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n [82] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n [85] nlme_3.1-167           patchwork_1.3.0        presto_1.0.0          \n [88] cli_3.6.4              spatstat.sparse_3.1-0  fansi_1.0.6           \n [91] spam_2.11-1            viridisLite_0.4.2      uwot_0.2.2            \n [94] gtable_0.3.6           progressr_0.15.1       rjson_0.2.23          \n [97] htmlwidgets_1.6.4      farver_2.1.2           htmltools_0.5.8.1     \n[100] lifecycle_1.0.4        httr_1.4.7             statmod_1.5.0         \n[103] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 19.16\nuser 23.15\nsys 0.56\npercent 123%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"MetaMarkers",desc:"Find markers between three or more groups of cells, using one-way ANOVA or Kruskal-Wallis test.",npages:2,report_toc:!0,order:23,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] parallel  stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] tidyseurat_0.8.0   ttservice_0.4.1    ggprism_1.0.5      ggplot2_3.5.1     \n [5] enrichR_3.4        Seurat_5.1.0       SeuratObject_5.0.2 sp_2.2-0          \n [9] Matrix_1.6-5       tibble_3.2.1       tidyr_1.3.1        dplyr_1.1.4       \n[13] tidyselect_1.2.1   jsonlite_1.8.9     rlang_1.1.5        logger_0.4.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] ellipsis_0.3.2         labeling_0.4.3         promises_1.3.2        \n [19] purrr_1.0.2            WriteXLS_6.7.0         goftest_1.2-3         \n [22] later_1.4.1            spatstat.utils_3.1-2   irlba_2.3.5.1         \n [25] cluster_2.1.8          R6_2.6.0               ica_1.0-3             \n [28] stringi_1.8.4          RColorBrewer_1.1-3     spatstat.data_3.1-4   \n [31] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [37] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          splines_4.3.3         \n [43] igraph_2.0.3           abind_1.4-5            spatstat.random_3.3-2 \n [46] codetools_0.2-20       miniUI_0.1.1.1         spatstat.explore_3.3-4\n [49] curl_6.0.1             listenv_0.9.1          lattice_0.22-6        \n [52] plyr_1.8.9             withr_3.0.2            shiny_1.10.0          \n [55] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n [58] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n [61] fitdistrplus_1.2-2     pillar_1.10.1          KernSmooth_2.23-26    \n [64] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n [67] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [70] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [73] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [76] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n [79] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n [82] nlme_3.1-167           patchwork_1.3.0        cli_3.6.4             \n [85] spatstat.sparse_3.1-0  fansi_1.0.6            spam_2.11-1           \n [88] viridisLite_0.4.2      uwot_0.2.2             gtable_0.3.6          \n [91] digest_0.6.37          progressr_0.15.1       ggrepel_0.9.6         \n [94] rjson_0.2.23           htmlwidgets_1.6.4      farver_2.1.2          \n [97] htmltools_0.5.8.1      lifecycle_1.0.4        httr_1.4.7            \n[100] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 344.55\nuser 1593.68\nsys 68.40\npercent 482%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"CellsDistribution",desc:"Distribution of cells (i.e. in a TCR clone) from different groups for each cluster",npages:1,report_toc:!0,order:24,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] grid      stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] ComplexHeatmap_2.18.0 circlize_0.4.16       UpSetR_1.4.0         \n [4] ggVennDiagram_1.2.3   patchwork_1.3.0       ggplot2_3.5.1        \n [7] tibble_3.2.1          Seurat_5.1.0          SeuratObject_5.0.2   \n[10] sp_2.2-0              tidyr_1.3.1           dplyr_1.1.4          \n[13] tidyselect_1.2.1      jsonlite_1.8.9        rlang_1.1.5          \n[16] logger_0.4.0         \n\nloaded via a namespace (and not attached):\n  [1] RColorBrewer_1.1-3     shape_1.4.6.1          magrittr_2.0.3        \n  [4] spatstat.utils_3.1-2   farver_2.1.2           GlobalOptions_0.1.2   \n  [7] vctrs_0.6.5            ROCR_1.0-11            Cairo_1.6-2           \n [10] spatstat.explore_3.3-4 htmltools_0.5.8.1      sctransform_0.4.1     \n [13] parallelly_1.42.0      KernSmooth_2.23-26     htmlwidgets_1.6.4     \n [16] ica_1.0-3              plyr_1.8.9             plotly_4.10.4         \n [19] zoo_1.8-12             igraph_2.0.3           mime_0.12             \n [22] lifecycle_1.0.4        iterators_1.0.14       pkgconfig_2.0.3       \n [25] Matrix_1.6-5           R6_2.6.0               fastmap_1.2.0         \n [28] clue_0.3-66            fitdistrplus_1.2-2     future_1.34.0         \n [31] shiny_1.10.0           digest_0.6.37          colorspace_2.1-1      \n [34] S4Vectors_0.40.2       tensor_1.5             RSpectra_0.16-2       \n [37] irlba_2.3.5.1          labeling_0.4.3         progressr_0.15.1      \n [40] spatstat.sparse_3.1-0  httr_1.4.7             polyclip_1.10-7       \n [43] abind_1.4-5            compiler_4.3.3         withr_3.0.2           \n [46] doParallel_1.0.17      fastDummies_1.7.5      MASS_7.3-60.0.1       \n [49] rjson_0.2.23           tools_4.3.3            lmtest_0.9-40         \n [52] httpuv_1.6.15          future.apply_1.11.3    goftest_1.2-3         \n [55] glue_1.8.0             nlme_3.1-167           promises_1.3.2        \n [58] Rtsne_0.17             cluster_2.1.8          reshape2_1.4.4        \n [61] generics_0.1.3         gtable_0.3.6           spatstat.data_3.1-4   \n [64] data.table_1.16.4      BiocGenerics_0.48.1    spatstat.geom_3.3-5   \n [67] RcppAnnoy_0.0.22       ggrepel_0.9.6          RANN_2.6.2            \n [70] foreach_1.5.2          pillar_1.10.1          stringr_1.5.1         \n [73] spam_2.11-1            RcppHNSW_0.6.0         later_1.4.1           \n [76] splines_4.3.3          lattice_0.22-6         survival_3.8-3        \n [79] deldir_2.0-4           miniUI_0.1.1.1         pbapply_1.7-2         \n [82] gridExtra_2.3          IRanges_2.36.0         scattermore_1.2       \n [85] stats4_4.3.3           matrixStats_1.1.0      stringi_1.8.4         \n [88] lazyeval_0.2.2         codetools_0.2-20       RVenn_1.1.0           \n [91] cli_3.6.4              uwot_0.2.2             xtable_1.8-4          \n [94] reticulate_1.40.0      munsell_0.5.1          Rcpp_1.0.14           \n [97] globals_0.16.3         spatstat.random_3.3-2  png_0.1-8             \n[100] spatstat.univar_3.1-1  parallel_4.3.3         dotCall64_1.2         \n[103] listenv_0.9.1          viridisLite_0.4.2      scales_1.3.0          \n[106] ggridges_0.5.6         leiden_0.4.3.1         purrr_1.0.2           \n[109] crayon_1.5.3           GetoptLong_1.0.5       cowplot_1.1.3         \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 12.22\nuser 20.80\nsys 0.49\npercent 174%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"CDR3AAPhyschem",desc:"CDR3 AA physicochemical feature analysis",npages:1,report_toc:!0,order:25,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] stringr_1.5.1       broom.mixed_0.2.9.6 glmnet_4.1-8       \n [4] Matrix_1.6-5        hash_3.0.1          glue_1.8.0         \n [7] ggridges_0.5.6      ggplot2_3.5.1       tibble_3.2.1       \n[10] tidyr_1.3.1         dplyr_1.1.4         jsonlite_1.8.9     \n[13] rlang_1.1.5         logger_0.4.0       \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         furrr_0.3.1            RcppAnnoy_0.0.22      \n  [7] spatstat.geom_3.3-5    matrixStats_1.1.0      compiler_4.3.3        \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] pkgconfig_2.0.3        shape_1.4.6.1          fastmap_1.2.0         \n [16] backports_1.5.0        labeling_0.4.3         promises_1.3.2        \n [19] purrr_1.0.2            goftest_1.2-3          later_1.4.1           \n [22] spatstat.utils_3.1-2   broom_1.0.7            irlba_2.3.5.1         \n [25] parallel_4.3.3         cluster_2.1.8          R6_2.6.0              \n [28] ica_1.0-3              spatstat.data_3.1-4    stringi_1.8.4         \n [31] RColorBrewer_1.1-3     reticulate_1.40.0      spatstat.univar_3.1-1 \n [34] parallelly_1.42.0      lmtest_0.9-40          scattermore_1.2       \n [37] Rcpp_1.0.14            iterators_1.0.14       tensor_1.5            \n [40] future.apply_1.11.3    zoo_1.8-12             sctransform_0.4.1     \n [43] httpuv_1.6.15          splines_4.3.3          igraph_2.0.3          \n [46] tidyselect_1.2.1       abind_1.4-5            spatstat.random_3.3-2 \n [49] codetools_0.2-20       miniUI_0.1.1.1         spatstat.explore_3.3-4\n [52] listenv_0.9.1          lattice_0.22-6         plyr_1.8.9            \n [55] shiny_1.10.0           withr_3.0.2            ROCR_1.0-11           \n [58] Rtsne_0.17             future_1.34.0          fastDummies_1.7.5     \n [61] survival_3.8-3         polyclip_1.10-7        fitdistrplus_1.2-2    \n [64] pillar_1.10.1          Seurat_5.1.0           KernSmooth_2.23-26    \n [67] foreach_1.5.2          plotly_4.10.4          generics_0.1.3        \n [70] RcppHNSW_0.6.0         sp_2.2-0               munsell_0.5.1         \n [73] scales_1.3.0           globals_0.16.3         xtable_1.8-4          \n [76] lazyeval_0.2.2         tools_4.3.3            data.table_1.16.4     \n [79] RSpectra_0.16-2        forcats_1.0.0          RANN_2.6.2            \n [82] leiden_0.4.3.1         dotCall64_1.2          cowplot_1.1.3         \n [85] grid_4.3.3             colorspace_2.1-1       nlme_3.1-167          \n [88] patchwork_1.3.0        cli_3.6.4              spatstat.sparse_3.1-0 \n [91] spam_2.11-1            viridisLite_0.4.2      uwot_0.2.2            \n [94] gtable_0.3.6           digest_0.6.37          progressr_0.15.1      \n [97] ggrepel_0.9.6          htmlwidgets_1.6.4      SeuratObject_5.0.2    \n[100] farver_2.1.2           htmltools_0.5.8.1      lifecycle_1.0.4       \n[103] httr_1.4.7             mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 11.03\nuser 19.55\nsys 0.54\npercent 182%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ScrnaMetabolicLandscape",desc:"Metabolic landscape analysis for scRNA-seq data",order:28,procs:[{name:"MetabolicPathwayActivity",desc:"This process calculates the pathway activities in different groups and subsets.",npages:1,report_toc:!0,order:28,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n [1] grid      parallel  stats4    stats     graphics  grDevices utils    \n [8] datasets  methods   base     \n\nother attached packages:\n [1] ComplexHeatmap_2.18.0       Seurat_5.1.0               \n [3] SeuratObject_5.0.2          sp_2.2-0                   \n [5] ggprism_1.0.5               RColorBrewer_1.1-3         \n [7] reshape2_1.4.4              scater_1.30.1              \n [9] scuttle_1.12.0              SingleCellExperiment_1.24.0\n[11] SummarizedExperiment_1.32.0 Biobase_2.62.0             \n[13] GenomicRanges_1.54.1        GenomeInfoDb_1.38.1        \n[15] IRanges_2.36.0              S4Vectors_0.40.2           \n[17] BiocGenerics_0.48.1         MatrixGenerics_1.14.0      \n[19] matrixStats_1.1.0           tibble_3.2.1               \n[21] dplyr_1.1.4                 ggplot2_3.5.1              \n[23] jsonlite_1.8.9              rlang_1.1.5                \n[25] logger_0.4.0               \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22          splines_4.3.3            \n  [3] later_1.4.1               bitops_1.0-9             \n  [5] polyclip_1.10-7           fastDummies_1.7.5        \n  [7] lifecycle_1.0.4           doParallel_1.0.17        \n  [9] globals_0.16.3            lattice_0.22-6           \n [11] MASS_7.3-60.0.1           magrittr_2.0.3           \n [13] plotly_4.10.4             httpuv_1.6.15            \n [15] sctransform_0.4.1         spam_2.11-1              \n [17] spatstat.sparse_3.1-0     reticulate_1.40.0        \n [19] cowplot_1.1.3             pbapply_1.7-2            \n [21] abind_1.4-5               zlibbioc_1.48.0          \n [23] Rtsne_0.17                purrr_1.0.2              \n [25] RCurl_1.98-1.16           circlize_0.4.16          \n [27] GenomeInfoDbData_1.2.11   ggrepel_0.9.6            \n [29] irlba_2.3.5.1             listenv_0.9.1            \n [31] spatstat.utils_3.1-2      goftest_1.2-3            \n [33] RSpectra_0.16-2           spatstat.random_3.3-2    \n [35] fitdistrplus_1.2-2        parallelly_1.42.0        \n [37] DelayedMatrixStats_1.24.0 leiden_0.4.3.1           \n [39] codetools_0.2-20          DelayedArray_0.28.0      \n [41] tidyselect_1.2.1          shape_1.4.6.1            \n [43] farver_2.1.2              ScaledMatrix_1.10.0      \n [45] viridis_0.6.5             spatstat.explore_3.3-4   \n [47] GetoptLong_1.0.5          BiocNeighbors_1.20.0     \n [49] progressr_0.15.1          ggridges_0.5.6           \n [51] survival_3.8-3            iterators_1.0.14         \n [53] foreach_1.5.2             tools_4.3.3              \n [55] ica_1.0-3                 Rcpp_1.0.14              \n [57] glue_1.8.0                gridExtra_2.3            \n [59] SparseArray_1.2.2         withr_3.0.2              \n [61] fastmap_1.2.0             digest_0.6.37            \n [63] rsvd_1.0.5                R6_2.6.0                 \n [65] mime_0.12                 colorspace_2.1-1         \n [67] Cairo_1.6-2               scattermore_1.2          \n [69] tensor_1.5                spatstat.data_3.1-4      \n [71] tidyr_1.3.1               generics_0.1.3           \n [73] data.table_1.16.4         httr_1.4.7               \n [75] htmlwidgets_1.6.4         S4Arrays_1.2.0           \n [77] uwot_0.2.2                pkgconfig_2.0.3          \n [79] gtable_0.3.6              lmtest_0.9-40            \n [81] XVector_0.42.0            htmltools_0.5.8.1        \n [83] dotCall64_1.2             clue_0.3-66              \n [85] scales_1.3.0              png_0.1-8                \n [87] spatstat.univar_3.1-1     rjson_0.2.23             \n [89] nlme_3.1-167              zoo_1.8-12               \n [91] GlobalOptions_0.1.2       stringr_1.5.1            \n [93] KernSmooth_2.23-26        miniUI_0.1.1.1           \n [95] vipor_0.4.7               pillar_1.10.1            \n [97] vctrs_0.6.5               RANN_2.6.2               \n [99] promises_1.3.2            BiocSingular_1.18.0      \n[101] beachmat_2.18.0           xtable_1.8-4             \n[103] cluster_2.1.8             beeswarm_0.4.0           \n[105] cli_3.6.4                 compiler_4.3.3           \n[107] crayon_1.5.3              future.apply_1.11.3      \n[109] plyr_1.8.9                ggbeeswarm_0.7.2         \n[111] stringi_1.8.4             viridisLite_0.4.2        \n[113] deldir_2.0-4              BiocParallel_1.36.0      \n[115] munsell_0.5.1             lazyeval_0.2.2           \n[117] spatstat.geom_3.3-5       Matrix_1.6-5             \n[119] RcppHNSW_0.6.0            patchwork_1.3.0          \n[121] sparseMatrixStats_1.14.0  future_1.34.0            \n[123] shiny_1.10.0              ROCR_1.0-11              \n[125] igraph_2.0.3             \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 755.68\nuser 748.76\nsys 14.13\npercent 100%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"MetabolicFeatures",desc:"This process performs enrichment analysis for the metabolic pathways for each group in each subset.",npages:1,report_toc:!0,order:29,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] parallel  stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] matrixStats_1.1.0  fgsea_1.28.0       data.table_1.16.4  Seurat_5.1.0      \n [5] SeuratObject_5.0.2 sp_2.2-0           tibble_3.2.1       dplyr_1.1.4       \n [9] ggplot2_3.5.1      jsonlite_1.8.9     rlang_1.1.5        logger_0.4.0      \n\nloaded via a namespace (and not attached):\n [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n [4] magrittr_2.0.3         RcppAnnoy_0.0.22       ggridges_0.5.6        \n [7] compiler_4.3.3         spatstat.geom_3.3-5    png_0.1-8             \n[10] vctrs_0.6.5            reshape2_1.4.4         stringr_1.5.1         \n[13] pkgconfig_2.0.3        fastmap_1.2.0          labeling_0.4.3        \n[16] promises_1.3.2         purrr_1.0.2            goftest_1.2-3         \n[19] later_1.4.1            BiocParallel_1.36.0    spatstat.utils_3.1-2  \n[22] irlba_2.3.5.1          cluster_2.1.8          R6_2.6.0              \n[25] ica_1.0-3              stringi_1.8.4          RColorBrewer_1.1-3    \n[28] spatstat.data_3.1-4    reticulate_1.40.0      parallelly_1.42.0     \n[31] spatstat.univar_3.1-1  lmtest_0.9-40          scattermore_1.2       \n[34] Rcpp_1.0.14            tensor_1.5             future.apply_1.11.3   \n[37] zoo_1.8-12             sctransform_0.4.1      httpuv_1.6.15         \n[40] Matrix_1.6-5           splines_4.3.3          igraph_2.0.3          \n[43] tidyselect_1.2.1       abind_1.4-5            spatstat.random_3.3-2 \n[46] codetools_0.2-20       miniUI_0.1.1.1         spatstat.explore_3.3-4\n[49] listenv_0.9.1          lattice_0.22-6         plyr_1.8.9            \n[52] shiny_1.10.0           withr_3.0.2            ROCR_1.0-11           \n[55] Rtsne_0.17             future_1.34.0          fastDummies_1.7.5     \n[58] survival_3.8-3         polyclip_1.10-7        fitdistrplus_1.2-2    \n[61] pillar_1.10.1          KernSmooth_2.23-26     plotly_4.10.4         \n[64] generics_0.1.3         RcppHNSW_0.6.0         munsell_0.5.1         \n[67] scales_1.3.0           globals_0.16.3         xtable_1.8-4          \n[70] glue_1.8.0             lazyeval_0.2.2         tools_4.3.3           \n[73] RSpectra_0.16-2        RANN_2.6.2             leiden_0.4.3.1        \n[76] dotCall64_1.2          fastmatch_1.1-6        cowplot_1.1.3         \n[79] grid_4.3.3             tidyr_1.3.1            colorspace_2.1-1      \n[82] nlme_3.1-167           patchwork_1.3.0        cli_3.6.4             \n[85] spatstat.sparse_3.1-0  spam_2.11-1            viridisLite_0.4.2     \n[88] uwot_0.2.2             gtable_0.3.6           digest_0.6.37         \n[91] progressr_0.15.1       ggrepel_0.9.6          htmlwidgets_1.6.4     \n[94] farver_2.1.2           htmltools_0.5.8.1      lifecycle_1.0.4       \n[97] httr_1.4.7             mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 265.61\nuser 265.24\nsys 11.02\npercent 104%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"MetabolicFeaturesIntraSubset",desc:"Intra-subset metabolic features - Enrichment analysis in details",npages:3,report_toc:!0,order:30,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats4    parallel  stats     graphics  grDevices utils     datasets \n[8] methods   base     \n\nother attached packages:\n [1] fgsea_1.28.0                data.table_1.16.4          \n [3] Seurat_5.1.0                SeuratObject_5.0.2         \n [5] sp_2.2-0                    scater_1.30.1              \n [7] scuttle_1.12.0              SingleCellExperiment_1.24.0\n [9] SummarizedExperiment_1.32.0 Biobase_2.62.0             \n[11] GenomicRanges_1.54.1        GenomeInfoDb_1.38.1        \n[13] IRanges_2.36.0              S4Vectors_0.40.2           \n[15] BiocGenerics_0.48.1         MatrixGenerics_1.14.0      \n[17] matrixStats_1.1.0           tibble_3.2.1               \n[19] dplyr_1.1.4                 ggplot2_3.5.1              \n[21] jsonlite_1.8.9              rlang_1.1.5                \n[23] logger_0.4.0               \n\nloaded via a namespace (and not attached):\n  [1] RColorBrewer_1.1-3        magrittr_2.0.3           \n  [3] spatstat.utils_3.1-2      ggbeeswarm_0.7.2         \n  [5] farver_2.1.2              zlibbioc_1.48.0          \n  [7] vctrs_0.6.5               ROCR_1.0-11              \n  [9] spatstat.explore_3.3-4    DelayedMatrixStats_1.24.0\n [11] RCurl_1.98-1.16           htmltools_0.5.8.1        \n [13] S4Arrays_1.2.0            BiocNeighbors_1.20.0     \n [15] SparseArray_1.2.2         sctransform_0.4.1        \n [17] parallelly_1.42.0         KernSmooth_2.23-26       \n [19] htmlwidgets_1.6.4         ica_1.0-3                \n [21] plyr_1.8.9                plotly_4.10.4            \n [23] zoo_1.8-12                igraph_2.0.3             \n [25] mime_0.12                 lifecycle_1.0.4          \n [27] pkgconfig_2.0.3           rsvd_1.0.5               \n [29] Matrix_1.6-5              R6_2.6.0                 \n [31] fastmap_1.2.0             GenomeInfoDbData_1.2.11  \n [33] fitdistrplus_1.2-2        future_1.34.0            \n [35] shiny_1.10.0              digest_0.6.37            \n [37] colorspace_2.1-1          patchwork_1.3.0          \n [39] tensor_1.5                RSpectra_0.16-2          \n [41] irlba_2.3.5.1             beachmat_2.18.0          \n [43] labeling_0.4.3            progressr_0.15.1         \n [45] spatstat.sparse_3.1-0     polyclip_1.10-7          \n [47] httr_1.4.7                abind_1.4-5              \n [49] compiler_4.3.3            withr_3.0.2              \n [51] BiocParallel_1.36.0       viridis_0.6.5            \n [53] fastDummies_1.7.5         MASS_7.3-60.0.1          \n [55] DelayedArray_0.28.0       tools_4.3.3              \n [57] vipor_0.4.7               lmtest_0.9-40            \n [59] beeswarm_0.4.0            httpuv_1.6.15            \n [61] future.apply_1.11.3       goftest_1.2-3            \n [63] glue_1.8.0                nlme_3.1-167             \n [65] promises_1.3.2            grid_4.3.3               \n [67] Rtsne_0.17                cluster_2.1.8            \n [69] reshape2_1.4.4            generics_0.1.3           \n [71] spatstat.data_3.1-4       gtable_0.3.6             \n [73] tidyr_1.3.1               BiocSingular_1.18.0      \n [75] ScaledMatrix_1.10.0       XVector_0.42.0           \n [77] spatstat.geom_3.3-5       RcppAnnoy_0.0.22         \n [79] ggrepel_0.9.6             RANN_2.6.2               \n [81] pillar_1.10.1             stringr_1.5.1            \n [83] spam_2.11-1               RcppHNSW_0.6.0           \n [85] later_1.4.1               splines_4.3.3            \n [87] lattice_0.22-6            deldir_2.0-4             \n [89] survival_3.8-3            tidyselect_1.2.1         \n [91] miniUI_0.1.1.1            pbapply_1.7-2            \n [93] gridExtra_2.3             scattermore_1.2          \n [95] stringi_1.8.4             lazyeval_0.2.2           \n [97] codetools_0.2-20          cli_3.6.4                \n [99] uwot_0.2.2                xtable_1.8-4             \n[101] reticulate_1.40.0         munsell_0.5.1            \n[103] Rcpp_1.0.14               spatstat.random_3.3-2    \n[105] globals_0.16.3            png_0.1-8                \n[107] spatstat.univar_3.1-1     dotCall64_1.2            \n[109] sparseMatrixStats_1.14.0  bitops_1.0-9             \n[111] listenv_0.9.1             viridisLite_0.4.2        \n[113] scales_1.3.0              ggridges_0.5.6           \n[115] leiden_0.4.3.1            purrr_1.0.2              \n[117] crayon_1.5.3              fastmatch_1.1-6          \n[119] cowplot_1.1.3            \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 116.82\nuser 123.29\nsys 2.07\npercent 107%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"MetabolicPathwayHeterogeneity",desc:"Calculate Metabolic Pathway heterogeneity.",npages:1,report_toc:!0,order:31}]}]};function us(e){let t,n;return t=new an({props:{dir:"up"}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function gs(e){let t,n;return t=new an({props:{dir:"down"}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function fs(e){let t,n,a,s,r,i=e[1]>0&&us();n=new ds({});let m=e[1]<e[3].npages-1&&gs();return{c(){i&&i.c(),t=$(),f(n.$$.fragment),a=$(),m&&m.c(),s=g()},m(e,o){i&&i.m(e,o),l(e,t,o),b(n,e,o),l(e,a,o),m&&m.m(e,o),l(e,s,o),r=!0},p(e,n){e[1]>0?i?2&n&&o(i,1):(i=us(),i.c(),o(i,1),i.m(t.parentNode,t)):i&&(p(),d(i,1,1,(()=>{i=null})),c()),e[1]<e[3].npages-1?m?10&n&&o(m,1):(m=gs(),m.c(),o(m,1),m.m(s.parentNode,s)):m&&(p(),d(m,1,1,(()=>{m=null})),c())},i(e){r||(o(i),o(n.$$.fragment,e),o(m),r=!0)},o(e){d(i),d(n.$$.fragment,e),d(m),r=!1},d(e){e&&(_(t),_(a),_(s)),i&&i.d(e),C(n,e),m&&m.d(e)}}}function bs(e){let t,n,a;function s(t){e[4](t)}let r={logo:e[0],logotext:e[3].desc,versions:ms.versions,entries:ms.entries,pipeline_name:ms.pipeline.name,report_toc:e[3].report_toc,runinfo:e[3].runinfo,toc:_s,page:e[1],$$slots:{default:[fs]},$$scope:{ctx:e}};return void 0!==e[2]&&(r.isSideNavOpen=e[2]),t=new Zt({props:r}),h.push((()=>v(t,"isSideNavOpen",s))),{c(){f(t.$$.fragment)},m(e,n){b(t,e,n),a=!0},p(e,[a]){const s={};1&a&&(s.logo=e[0]),8&a&&(s.logotext=e[3].desc),8&a&&(s.report_toc=e[3].report_toc),8&a&&(s.runinfo=e[3].runinfo),2&a&&(s.page=e[1]),42&a&&(s.$$scope={dirty:a,ctx:e}),!n&&4&a&&(n=!0,s.isSideNavOpen=e[2],y((()=>n=!1))),t.$set(s)},i(e){a||(o(t.$$.fragment,e),a=!0)},o(e){d(t.$$.fragment,e),a=!1},d(e){C(t,e)}}}function Cs(e,t,n){let a,{name:s}=t,{page:r=0}=t,i=!1;for(const e of ms.entries)if(e.procs){for(const t of e.procs)if(t.name==s){a=t;break}}else if(e.name==s){a=e;break}return u((()=>{const e=new URL(window.location.href);let t;e.hash.length>1&&(t=document.getElementById(`${e.hash.slice(1)}`)),t&&t.scrollIntoView({block:"start",behavior:"smooth"})})),e.$$set=e=>{"name"in e&&n(0,s=e.name),"page"in e&&n(1,r=e.page)},[s,r,i,a,function(e){i=e,n(2,i)}]}e("default",new class extends t{constructor(e){super(),n(this,e,Cs,bs,a,{name:0,page:1})}}({target:document.body,props:{name:"MetabolicFeaturesIntraSubset",page:1}}))}}}));
