System.register(["./vendor/svelte-C_apNiMW.js","./vendor/carbon-components-svelte-CB8VNYQX.js","./vendor/carbon-icons-svelte-DkM4cpjm.js"],(function(e){"use strict";var t,n,a,s,i,r,l,o,p,c,_,d,m,u,g,f,b,h,v,C,y,$,x,S,M,A,P,L,T,w,R,U,N,E,B,I,D,V,H,G,F,z,k,q,O,K,j,W,Y,X,Q,J,Z,ee,te,ne,ae,se,ie,re,le,oe,pe,ce,_e,de,me,ue,ge,fe,be,he,ve,Ce,ye,$e,xe,Se,Me,Ae,Pe,Le,Te,we,Re,Ue,Ne,Ee,Be,Ie,De,Ve,He,Ge,Fe,ze,ke,qe,Oe,Ke,je,We,Ye,Xe,Qe,Je,Ze;return{setters:[function(e){t=e.S,n=e.i,a=e.s,s=e.a1,i=e.l,r=e.O,l=e.a,o=e.b,p=e.g,c=e.c,_=e.t,d=e.d,m=e.a5,u=e.B,g=e.e,f=e.G,b=e.H,h=e.I,v=e.y,C=e.K,y=e.T,$=e.U,x=e.L,S=e.D,M=e.M,A=e.z,P=e.Q,L=e.k,T=e.u,w=e.n,R=e.o,U=e.h,N=e.p,E=e.a0,B=e.f,I=e.j,D=e.A,V=e.m,H=e.q,G=e.aa,F=e.ab,z=e.x,k=e.W,q=e.ac,O=e.R,K=e.$,j=e.v,W=e.Y,Y=e.Z,X=e.F,Q=e.ad,J=e.w,Z=e.J,ee=e.r},function(e){te=e.S,ne=e.a,ae=e.b,se=e.H,ie=e.c,re=e.T,le=e.d,oe=e.e,pe=e.f,ce=e.g,_e=e.h,de=e.M,me=e.I,ue=e.i,ge=e.j,fe=e.k,be=e.C,he=e.l,ve=e.m,Ce=e.G,ye=e.R,$e=e.n,xe=e.B,Se=e.o,Me=e.A,Ae=e.p,Pe=e.D,Le=e.P,Te=e.q,we=e.r,Re=e.s,Ue=e.t,Ne=e.u},function(e){Ee=e.C,Be=e.a,Ie=e.D,De=e.A,Ve=e.S,He=e.I,Ge=e.b,Fe=e.c,ze=e.V,ke=e.d,qe=e.T,Oe=e.G,Ke=e.J,je=e.P,We=e.e,Ye=e.f,Xe=e.O,Qe=e.g,Je=e.h,Ze=e.i}],execute:function(){function et(e,t,n){const a=e.slice();return a[4]=t[n],a}function tt(e,t,n){const a=e.slice();return a[7]=t[n],a}function nt(e){let t,n;return t=new te({props:{href:`${e[1](e[4].page)}#${e[4].slug}`,class:e[2](e[4].page),title:e[4].text,text:e[4].text,expanded:!0,$$slots:{default:[it]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.href=`${e[1](e[4].page)}#${e[4].slug}`),1&n&&(a.class=e[2](e[4].page)),1&n&&(a.title=e[4].text),1&n&&(a.text=e[4].text),1025&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function at(e){let t,n;return t=new ne({props:{href:`${e[1](e[4].page)}#${e[4].slug}`,class:e[2](e[4].page),title:e[4].text,text:e[4].text}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.href=`${e[1](e[4].page)}#${e[4].slug}`),1&n&&(a.class=e[2](e[4].page)),1&n&&(a.title=e[4].text),1&n&&(a.text=e[4].text),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function st(e){let t,n;return t=new ae({props:{href:`${e[1](e[7].page)}#${e[7].slug}`,class:e[2](e[7].page),title:e[7].text,text:e[7].text}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.href=`${e[1](e[7].page)}#${e[7].slug}`),1&n&&(a.class=e[2](e[7].page)),1&n&&(a.title=e[7].text),1&n&&(a.text=e[7].text),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function it(e){let t,n,a=s(e[4].children),i=[];for(let t=0;t<a.length;t+=1)i[t]=st(tt(e,a,t));const r=e=>_(i[e],1,1,(()=>{i[e]=null}));return{c(){for(let e=0;e<i.length;e+=1)i[e].c();t=v()},m(e,a){for(let t=0;t<i.length;t+=1)i[t]&&i[t].m(e,a);l(e,t,a),n=!0},p(e,n){if(7&n){let l;for(a=s(e[4].children),l=0;l<a.length;l+=1){const s=tt(e,a,l);i[l]?(i[l].p(s,n),o(i[l],1)):(i[l]=st(s),i[l].c(),o(i[l],1),i[l].m(t.parentNode,t))}for(p(),l=a.length;l<i.length;l+=1)r(l);c()}},i(e){if(!n){for(let e=0;e<a.length;e+=1)o(i[e]);n=!0}},o(e){i=i.filter(Boolean);for(let e=0;e<i.length;e+=1)_(i[e]);n=!1},d(e){e&&d(t),m(i,e)}}}function rt(e){let t,n,a,s;const i=[at,nt],r=[];function m(e,t){return 0==e[4].children.length?0:1}return t=m(e),n=r[t]=i[t](e),{c(){n.c(),a=g()},m(e,n){r[t].m(e,n),l(e,a,n),s=!0},p(e,s){let l=t;t=m(e),t===l?r[t].p(e,s):(p(),_(r[l],1,1,(()=>{r[l]=null})),c(),n=r[t],n?n.p(e,s):(n=r[t]=i[t](e),n.c()),o(n,1),n.m(a.parentNode,a))},i(e){s||(o(n),s=!0)},o(e){_(n),s=!1},d(e){e&&d(a),r[t].d(e)}}}function lt(e){let t,n,a=s(e[0]),u=[];for(let t=0;t<a.length;t+=1)u[t]=rt(et(e,a,t));const g=e=>_(u[e],1,1,(()=>{u[e]=null}));return{c(){t=i("ul");for(let e=0;e<u.length;e+=1)u[e].c();r(t,"class","bx--side-nav__items pipen-report-sidenav svelte-12gpq3")},m(e,a){l(e,t,a);for(let e=0;e<u.length;e+=1)u[e]&&u[e].m(t,null);n=!0},p(e,[n]){if(7&n){let i;for(a=s(e[0]),i=0;i<a.length;i+=1){const s=et(e,a,i);u[i]?(u[i].p(s,n),o(u[i],1)):(u[i]=rt(s),u[i].c(),o(u[i],1),u[i].m(t,null))}for(p(),i=a.length;i<u.length;i+=1)g(i);c()}},i(e){if(!n){for(let e=0;e<a.length;e+=1)o(u[e]);n=!0}},o(e){u=u.filter(Boolean);for(let e=0;e<u.length;e+=1)_(u[e]);n=!1},d(e){e&&d(t),m(u,e)}}}function ot(e,t,n){let{toc:a}=t,{page:s}=t;return u((()=>{let e,t=new URL(window.location.href);e=t.hash.length>0?document.querySelector(`[href="${t.hash}"]`):document.querySelector(".toc-current-page"),e.scrollIntoView({behavior:"smooth"})})),e.$$set=e=>{"toc"in e&&n(0,a=e.toc),"page"in e&&n(3,s=e.page)},[a,e=>{if(s===e)return"";let t=new URL(window.location.href);return t.hash="",t.searchParams.set("page",e),t.toString()},e=>e!=s?"toc-noncurrent-page":"toc-current-page",s]}class pt extends t{constructor(e){super(),n(this,e,ot,lt,a,{toc:0,page:3})}}function ct(e,t,n){const a=e.slice();return a[7]=t[n],a}function _t(e,t,n){const a=e.slice();return a[10]=t[n],a}function dt(e){let t,n;return t=new oe({props:{isOpen:Rt,transition:e[5],$$slots:{default:[St]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8197&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function mt(e){let t;return{c(){t=x("Go to home:")},m(e,n){l(e,t,n)},d(e){e&&d(t)}}}function ut(e){let t;return{c(){t=x(e[0])},m(e,n){l(e,t,n)},p(e,n){1&n&&M(t,e[0])},d(e){e&&d(t)}}}function gt(e){let t;return{c(){t=x("Switch to process:")},m(e,n){l(e,t,n)},d(e){e&&d(t)}}}function ft(e){let t,n;return t=new _e({props:{title:e[7].name,href:`?proc=${e[7].name}`,$$slots:{default:[ht]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};4&n&&(a.title=e[7].name),4&n&&(a.href=`?proc=${e[7].name}`),8196&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function bt(e){let t,n,a,i;t=new _e({props:{class:"procgroup",$$slots:{default:[vt]},$$scope:{ctx:e}}});let r=s(e[7].procs),u=[];for(let t=0;t<r.length;t+=1)u[t]=yt(_t(e,r,t));const C=e=>_(u[e],1,1,(()=>{u[e]=null}));return{c(){f(t.$$.fragment),n=v();for(let e=0;e<u.length;e+=1)u[e].c();a=g()},m(e,s){b(t,e,s),l(e,n,s);for(let t=0;t<u.length;t+=1)u[t]&&u[t].m(e,s);l(e,a,s),i=!0},p(e,n){const i={};if(8196&n&&(i.$$scope={dirty:n,ctx:e}),t.$set(i),4&n){let t;for(r=s(e[7].procs),t=0;t<r.length;t+=1){const s=_t(e,r,t);u[t]?(u[t].p(s,n),o(u[t],1)):(u[t]=yt(s),u[t].c(),o(u[t],1),u[t].m(a.parentNode,a))}for(p(),t=r.length;t<u.length;t+=1)C(t);c()}},i(e){if(!i){o(t.$$.fragment,e);for(let e=0;e<r.length;e+=1)o(u[e]);i=!0}},o(e){_(t.$$.fragment,e),u=u.filter(Boolean);for(let e=0;e<u.length;e+=1)_(u[e]);i=!1},d(e){e&&(d(n),d(a)),h(t,e),m(u,e)}}}function ht(e){let t,n,a=e[7].name+"";return{c(){t=x(a),n=v()},m(e,a){l(e,t,a),l(e,n,a)},p(e,n){4&n&&a!==(a=e[7].name+"")&&M(t,a)},d(e){e&&(d(t),d(n))}}}function vt(e){let t,n=e[7].name+"";return{c(){t=x(n)},m(e,n){l(e,t,n)},p(e,a){4&a&&n!==(n=e[7].name+"")&&M(t,n)},d(e){e&&d(t)}}}function Ct(e){let t,n,a,s=e[10].name+"";return{c(){t=x("- "),n=x(s),a=v()},m(e,s){l(e,t,s),l(e,n,s),l(e,a,s)},p(e,t){4&t&&s!==(s=e[10].name+"")&&M(n,s)},d(e){e&&(d(t),d(n),d(a))}}}function yt(e){let t,n;return t=new _e({props:{title:e[10].name,class:"procgroup-proc",href:`?proc=${e[10].name}`,$$slots:{default:[Ct]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};4&n&&(a.title=e[10].name),4&n&&(a.href=`?proc=${e[10].name}`),8196&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function $t(e){let t,n,a,s;const i=[bt,ft],r=[];function m(e,t){return e[7].procs?0:1}return t=m(e),n=r[t]=i[t](e),{c(){n.c(),a=g()},m(e,n){r[t].m(e,n),l(e,a,n),s=!0},p(e,s){let l=t;t=m(e),t===l?r[t].p(e,s):(p(),_(r[l],1,1,(()=>{r[l]=null})),c(),n=r[t],n?n.p(e,s):(n=r[t]=i[t](e),n.c()),o(n,1),n.m(a.parentNode,a))},i(e){s||(o(n),s=!0)},o(e){_(n),s=!1},d(e){e&&d(a),r[t].d(e)}}}function xt(e){let t,n,a,i,r,u,C,y;t=new ce({props:{$$slots:{default:[mt]},$$scope:{ctx:e}}}),a=new _e({props:{href:"index.html",$$slots:{default:[ut]},$$scope:{ctx:e}}}),r=new ce({props:{$$slots:{default:[gt]},$$scope:{ctx:e}}});let $=s(e[2]),x=[];for(let t=0;t<$.length;t+=1)x[t]=$t(ct(e,$,t));const S=e=>_(x[e],1,1,(()=>{x[e]=null}));return{c(){f(t.$$.fragment),n=v(),f(a.$$.fragment),i=v(),f(r.$$.fragment),u=v();for(let e=0;e<x.length;e+=1)x[e].c();C=g()},m(e,s){b(t,e,s),l(e,n,s),b(a,e,s),l(e,i,s),b(r,e,s),l(e,u,s);for(let t=0;t<x.length;t+=1)x[t]&&x[t].m(e,s);l(e,C,s),y=!0},p(e,n){const i={};8192&n&&(i.$$scope={dirty:n,ctx:e}),t.$set(i);const l={};8193&n&&(l.$$scope={dirty:n,ctx:e}),a.$set(l);const _={};if(8192&n&&(_.$$scope={dirty:n,ctx:e}),r.$set(_),4&n){let t;for($=s(e[2]),t=0;t<$.length;t+=1){const a=ct(e,$,t);x[t]?(x[t].p(a,n),o(x[t],1)):(x[t]=$t(a),x[t].c(),o(x[t],1),x[t].m(C.parentNode,C))}for(p(),t=$.length;t<x.length;t+=1)S(t);c()}},i(e){if(!y){o(t.$$.fragment,e),o(a.$$.fragment,e),o(r.$$.fragment,e);for(let e=0;e<$.length;e+=1)o(x[e]);y=!0}},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),_(r.$$.fragment,e),x=x.filter(Boolean);for(let e=0;e<x.length;e+=1)_(x[e]);y=!1},d(e){e&&(d(n),d(i),d(u),d(C)),h(t,e),h(a,e),h(r,e),m(x,e)}}}function St(e){let t,n;return t=new pe({props:{$$slots:{default:[xt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8197&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function Mt(e){let t,n,a,s;t=new le({props:{render:"select",theme:"g10",persist:!0,persistKey:"__pipen-report-theme",select:{themes:["g10","white","g80","g90","g100"],labelText:"Theme",inline:!0}}});let i=e[2].length>0&&dt(e);return{c(){f(t.$$.fragment),n=v(),i&&i.c(),a=g()},m(e,r){b(t,e,r),l(e,n,r),i&&i.m(e,r),l(e,a,r),s=!0},p(e,t){e[2].length>0?i?(i.p(e,t),4&t&&o(i,1)):(i=dt(e),i.c(),o(i,1),i.m(a.parentNode,a)):i&&(p(),_(i,1,1,(()=>{i=null})),c())},i(e){s||(o(t.$$.fragment,e),o(i),s=!0)},o(e){_(t.$$.fragment,e),_(i),s=!1},d(e){e&&(d(n),d(a)),h(t,e),i&&i.d(e)}}}function At(e){let t,n;return t=new ie({props:{class:"header-util",$$slots:{default:[Mt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8197&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function Pt(e){let t;return{c(){t=x(e[4])},m(e,n){l(e,t,n)},p(e,n){16&n&&M(t,e[4])},d(e){e&&d(t)}}}function Lt(e){let t,n,a,s,r=null!==e[4]&&Pt(e);return{c(){t=i("h1"),n=x(e[3]),a=v(),r&&r.c(),s=g()},m(e,i){l(e,t,i),S(t,n),l(e,a,i),r&&r.m(e,i),l(e,s,i)},p(e,t){8&t&&M(n,e[3]),null!==e[4]?r?r.p(e,t):(r=Pt(e),r.c(),r.m(s.parentNode,s)):r&&(r.d(1),r=null)},d(e){e&&(d(t),d(a),d(s)),r&&r.d(e)}}}function Tt(e){let t,n;return t=new re({props:{class:"header-tile",slot:"platform",$$slots:{default:[Lt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8216&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function wt(e){let t,n,a;function s(t){e[6](t)}let i={$$slots:{platform:[Tt],default:[At]},$$scope:{ctx:e}};return void 0!==e[1]&&(i.isSideNavOpen=e[1]),t=new se({props:i}),C.push((()=>y(t,"isSideNavOpen",s))),{c(){f(t.$$.fragment)},m(e,n){b(t,e,n),a=!0},p(e,[a]){const s={};8221&a&&(s.$$scope={dirty:a,ctx:e}),!n&&2&a&&(n=!0,s.isSideNavOpen=e[1],$((()=>n=!1))),t.$set(s)},i(e){a||(o(t.$$.fragment,e),a=!0)},o(e){_(t.$$.fragment,e),a=!1},d(e){h(t,e)}}}let Rt=!1;function Ut(e,t,n){let{entries:a}=t,{logo:s}=t,{logotext:i=null}=t,{pipeline_name:r}=t,{isSideNavOpen:l}=t;return r=r||s,e.$$set=e=>{"entries"in e&&n(2,a=e.entries),"logo"in e&&n(3,s=e.logo),"logotext"in e&&n(4,i=e.logotext),"pipeline_name"in e&&n(0,r=e.pipeline_name),"isSideNavOpen"in e&&n(1,l=e.isSideNavOpen)},[r,l,a,s,i,{duration:200},function(e){l=e,n(1,l)}]}class Nt extends t{constructor(e){super(),n(this,e,Ut,wt,a,{entries:2,logo:3,logotext:4,pipeline_name:0,isSideNavOpen:1})}}function Et(e){let t,n,a;function s(t){e[3](t)}let i={modalHeading:"Running Information",passiveModal:!0,hasScrollingContent:!0,class:"runinfo",$$slots:{default:[Gt]},$$scope:{ctx:e}};return void 0!==e[2]&&(i.open=e[2]),t=new de({props:i}),C.push((()=>y(t,"open",s))),{c(){f(t.$$.fragment)},m(e,n){b(t,e,n),a=!0},p(e,a){const s={};34&a&&(s.$$scope={dirty:a,ctx:e}),!n&&4&a&&(n=!0,s.open=e[2],$((()=>n=!1))),t.$set(s)},i(e){a||(o(t.$$.fragment,e),a=!0)},o(e){_(t.$$.fragment,e),a=!1},d(e){h(t,e)}}}function Bt(e){let t,n,a,s,i,r;return t=new ge({props:{label:"Session Information"}}),a=new ge({props:{label:"Time Spent"}}),i=new ge({props:{label:"Device Information"}}),{c(){f(t.$$.fragment),n=v(),f(a.$$.fragment),s=v(),f(i.$$.fragment)},m(e,o){b(t,e,o),l(e,n,o),b(a,e,o),l(e,s,o),b(i,e,o),r=!0},p:P,i(e){r||(o(t.$$.fragment,e),o(a.$$.fragment,e),o(i.$$.fragment,e),r=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),_(i.$$.fragment,e),r=!1},d(e){e&&(d(n),d(s)),h(t,e),h(a,e),h(i,e)}}}function It(e){let t,n;return t=new be({props:{type:"multi",code:e[1].session}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};2&n&&(a.code=e[1].session),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function Dt(e){let t,n;return t=new be({props:{type:"multi",code:e[1].time}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};2&n&&(a.code=e[1].time),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function Vt(e){let t,n;return t=new be({props:{type:"multi",code:e[1].device}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};2&n&&(a.code=e[1].device),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function Ht(e){let t,n,a,s,i,r;return t=new fe({props:{$$slots:{default:[It]},$$scope:{ctx:e}}}),a=new fe({props:{$$slots:{default:[Dt]},$$scope:{ctx:e}}}),i=new fe({props:{$$slots:{default:[Vt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=v(),f(a.$$.fragment),s=v(),f(i.$$.fragment)},m(e,o){b(t,e,o),l(e,n,o),b(a,e,o),l(e,s,o),b(i,e,o),r=!0},p(e,n){const s={};34&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};34&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r);const l={};34&n&&(l.$$scope={dirty:n,ctx:e}),i.$set(l)},i(e){r||(o(t.$$.fragment,e),o(a.$$.fragment,e),o(i.$$.fragment,e),r=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),_(i.$$.fragment,e),r=!1},d(e){e&&(d(n),d(s)),h(t,e),h(a,e),h(i,e)}}}function Gt(e){let t,n,a,s;return t=new me({props:{kind:"warning",lowContrast:!0,hideCloseButton:!0,title:"Showing the first job only. Check the workdir for information of other jobs if any."}}),a=new ue({props:{$$slots:{content:[Ht],default:[Bt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=v(),f(a.$$.fragment)},m(e,i){b(t,e,i),l(e,n,i),b(a,e,i),s=!0},p(e,t){const n={};34&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),h(t,e),h(a,e)}}}function Ft(e){let t,n,a;return{c(){t=i("a"),t.textContent="> Running Information",r(t,"href","javascript:void(0);")},m(s,i){l(s,t,i),n||(a=A(t,"click",e[4]),n=!0)},p:P,d(e){e&&d(t),n=!1,a()}}}function zt(e){let t,n,a,s,m,u,g,f,b=e[1]&&Et(e),h=e[1]&&Ft(e);return{c(){b&&b.c(),t=v(),n=i("div"),a=i("div"),h&&h.c(),s=v(),m=i("div"),u=x("Powered by "),g=i("span"),r(a,"class","runinfo"),r(g,"class","pipen-report-footer-versions"),r(m,"class","versions"),r(n,"class","pipen-report-footer bx--grid")},m(i,r){b&&b.m(i,r),l(i,t,r),l(i,n,r),S(n,a),h&&h.m(a,null),S(n,s),S(n,m),S(m,u),S(m,g),g.innerHTML=e[0],f=!0},p(e,[n]){e[1]?b?(b.p(e,n),2&n&&o(b,1)):(b=Et(e),b.c(),o(b,1),b.m(t.parentNode,t)):b&&(p(),_(b,1,1,(()=>{b=null})),c()),e[1]?h?h.p(e,n):(h=Ft(e),h.c(),h.m(a,null)):h&&(h.d(1),h=null),(!f||1&n)&&(g.innerHTML=e[0])},i(e){f||(o(b),f=!0)},o(e){_(b),f=!1},d(e){e&&(d(t),d(n)),b&&b.d(e),h&&h.d()}}}function kt(e,t,n){let{versions:a}=t,{runinfo:s}=t,i=!1;return e.$$set=e=>{"versions"in e&&n(0,a=e.versions),"runinfo"in e&&n(1,s=e.runinfo)},[a,s,i,function(e){i=e,n(2,i)},()=>{n(2,i=!0)}]}class qt extends t{constructor(e){super(),n(this,e,kt,zt,a,{versions:0,runinfo:1})}}function Ot(e){let t,n,a;function s(t){e[12](t)}let i={$$slots:{default:[Kt]},$$scope:{ctx:e}};return void 0!==e[0]&&(i.isOpen=e[0]),t=new ve({props:i}),C.push((()=>y(t,"isOpen",s))),{c(){f(t.$$.fragment)},m(e,n){b(t,e,n),a=!0},p(e,a){const s={};8576&a&&(s.$$scope={dirty:a,ctx:e}),!n&&1&a&&(n=!0,s.isOpen=e[0],$((()=>n=!1))),t.$set(s)},i(e){a||(o(t.$$.fragment,e),a=!0)},o(e){_(t.$$.fragment,e),a=!1},d(e){h(t,e)}}}function Kt(e){let t,n;return t=new pt({props:{toc:e[7],page:e[8]}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};128&n&&(a.toc=e[7]),256&n&&(a.page=e[8]),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function jt(e){let t;const n=e[10].default,a=L(n,e,e[13],null);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||8192&s)&&T(a,n,e,e[13],t?R(n,e[13],s,null):w(e[13]),null)},i(e){t||(o(a,e),t=!0)},o(e){_(a,e),t=!1},d(e){a&&a.d(e)}}}function Wt(e){let t,n;return t=new $e({props:{$$slots:{default:[jt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8192&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function Yt(e){let t,n;return t=new ye({props:{$$slots:{default:[Wt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8192&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function Xt(e){let t,n,a,s;return t=new Ce({props:{$$slots:{default:[Yt]},$$scope:{ctx:e}}}),a=new qt({props:{versions:e[3],runinfo:e[9]}}),{c(){f(t.$$.fragment),n=v(),f(a.$$.fragment)},m(e,i){b(t,e,i),l(e,n,i),b(a,e,i),s=!0},p(e,n){const s={};8192&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const i={};8&n&&(i.versions=e[3]),512&n&&(i.runinfo=e[9]),a.$set(i)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),h(t,e),h(a,e)}}}function Qt(e){let t,n,a,s,i,r;function m(t){e[11](t)}let u={logo:e[1],logotext:e[2],entries:e[4],pipeline_name:e[5]};void 0!==e[0]&&(u.isSideNavOpen=e[0]),t=new Nt({props:u}),C.push((()=>y(t,"isSideNavOpen",m)));let g=e[6]&&e[7].length>0&&Ot(e);return i=new he({props:{$$slots:{default:[Xt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),a=v(),g&&g.c(),s=v(),f(i.$$.fragment)},m(e,n){b(t,e,n),l(e,a,n),g&&g.m(e,n),l(e,s,n),b(i,e,n),r=!0},p(e,[a]){const r={};2&a&&(r.logo=e[1]),4&a&&(r.logotext=e[2]),16&a&&(r.entries=e[4]),32&a&&(r.pipeline_name=e[5]),!n&&1&a&&(n=!0,r.isSideNavOpen=e[0],$((()=>n=!1))),t.$set(r),e[6]&&e[7].length>0?g?(g.p(e,a),192&a&&o(g,1)):(g=Ot(e),g.c(),o(g,1),g.m(s.parentNode,s)):g&&(p(),_(g,1,1,(()=>{g=null})),c());const l={};8712&a&&(l.$$scope={dirty:a,ctx:e}),i.$set(l)},i(e){r||(o(t.$$.fragment,e),o(g),o(i.$$.fragment,e),r=!0)},o(e){_(t.$$.fragment,e),_(g),_(i.$$.fragment,e),r=!1},d(e){e&&(d(a),d(s)),h(t,e),g&&g.d(e),h(i,e)}}}function Jt(e,t,n){let a,{$$slots:s={},$$scope:i}=t,{logo:r}=t,{logotext:l=null}=t,{versions:o}=t,{entries:p}=t,{pipeline_name:c}=t,{report_toc:_}=t,{toc:d}=t,{page:m}=t,{isSideNavOpen:g}=t,{runinfo:f}=t;return u((()=>{const e=new URL(window.location.href);let t;t=e.hash.length>0?document.querySelector(`li:has(>[href="${e.hash}"])`):document.querySelector("li:has(>.toc-current-page)"),t&&(t.classList.add("toc-active"),t.scrollIntoView());const n=document.querySelectorAll(".pipen-report-toc-anchor");a=new IntersectionObserver((e=>{e.forEach((e=>{const t=e.target.getAttribute("id"),n=document.querySelector(`li:has(>[href="#${t}"])`);n&&(e.isIntersecting?n.classList.add("toc-active"):n.classList.remove("toc-active"))}))}),{root:null,rootMargin:"0px",threshold:.5}),n.forEach((e=>{a.observe(e)}))})),e.$$set=e=>{"logo"in e&&n(1,r=e.logo),"logotext"in e&&n(2,l=e.logotext),"versions"in e&&n(3,o=e.versions),"entries"in e&&n(4,p=e.entries),"pipeline_name"in e&&n(5,c=e.pipeline_name),"report_toc"in e&&n(6,_=e.report_toc),"toc"in e&&n(7,d=e.toc),"page"in e&&n(8,m=e.page),"isSideNavOpen"in e&&n(0,g=e.isSideNavOpen),"runinfo"in e&&n(9,f=e.runinfo),"$$scope"in e&&n(13,i=e.$$scope)},[g,r,l,o,p,c,_,d,m,f,s,function(e){g=e,n(0,g)},function(e){g=e,n(0,g)},i]}class Zt extends t{constructor(e){super(),n(this,e,Jt,Qt,a,{logo:1,logotext:2,versions:3,entries:4,pipeline_name:5,report_toc:6,toc:7,page:8,isSideNavOpen:0,runinfo:9})}}function en(e){let t;return{c(){t=x(e[2])},m(e,n){l(e,t,n)},p:P,d(e){e&&d(t)}}}function tn(e){let t,n,a,s;return n=new xe({props:{icon:e[1],kind:"tertiary",href:e[3](),$$slots:{default:[en]},$$scope:{ctx:e}}}),{c(){t=i("div"),f(n.$$.fragment),r(t,"class",a=`pagenav ${e[0]}`)},m(e,a){l(e,t,a),b(n,t,null),s=!0},p(e,[i]){const l={};16&i&&(l.$$scope={dirty:i,ctx:e}),n.$set(l),(!s||1&i&&a!==(a=`pagenav ${e[0]}`))&&r(t,"class",a)},i(e){s||(o(n.$$.fragment,e),s=!0)},o(e){_(n.$$.fragment,e),s=!1},d(e){e&&d(t),h(n)}}}function nn(e,t,n){let{dir:a}=t,s="up"===a?Ee:Be,i="up"===a?"Go to previous page":"Go to next page";return e.$$set=e=>{"dir"in e&&n(0,a=e.dir)},[a,s,i,()=>{let e=new URL(window.location.href);e.hash="";const t=parseInt(e.searchParams.get("page")||0),n="up"===a?t-1:t+1;return 0===n?e.searchParams.delete("page"):e.searchParams.set("page",n),e.toString()}]}class an extends t{constructor(e){super(),n(this,e,nn,tn,a,{dir:0})}}function sn(e){let t,n;const a=[{tooltipText:e[0]},{icon:e[1]},{direction:"bottom"},{align:"start"},e[3]];let s={};for(let e=0;e<a.length;e+=1)s=U(s,a[e]);return t=new Se({props:s}),t.$on("click",e[2]),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,[n]){const s=11&n?N(a,[1&n&&{tooltipText:e[0]},2&n&&{icon:e[1]},a[2],a[3],8&n&&E(e[3])]):{};t.$set(s)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function rn(e,t,n){const a=["src","tip","icon"];let s,i=B(t,a),{src:r=""}=t,{tip:l=""}=t,{icon:o="auto"}=t,p=r.split(".").at(-1).toLocaleLowerCase();if("string"==typeof o)switch("auto"===o&&(o=p),o){case"pdf":case"DocumentPdf":s=Ye;break;case"html":case"htm":case"ChartLineData":s=We;break;case"png":case"Png":s=je;break;case"jpg":case"jpeg":case"Jpg":s=Ke;break;case"gif":case"Gif":s=Oe;break;case"tif":case"tiff":case"Tif":s=qe;break;case"eps":case"DatabaseElastic":s=ke;break;case"mp4":case"avi":case"mov":case"mkv":case"webm":case"VideoPlayer":s=ze;break;case"Image":s=Fe;break;case"Data":case"DataVolume":s=Ge;break;case"Code":case"IbmWatsonxCodeAssistantForZRefactor":s=He;break;case"svg":case"Svg":s=Ve;break;case"zip":case"tar":case"gz":case"bz2":case"7z":case"rar":case"xz":case"Archive":s=De;break;default:s=Ie}const c={...i,class:[i.class,"pipen-report-image-download"].filter(Boolean).join(" ")};return e.$$set=e=>{t=U(U({},t),I(e)),n(7,i=B(t,a)),"src"in e&&n(5,r=e.src),"tip"in e&&n(0,l=e.tip),"icon"in e&&n(4,o=e.icon)},[l,s,()=>{const e=document.createElement("a");e.href=r,e.download=r.split("/").pop(),e.click()},c,o,r]}class ln extends t{constructor(e){super(),n(this,e,rn,sn,a,{src:5,tip:0,icon:4})}}const{Boolean:on}=G;function pn(e,t,n){const a=e.slice();return a[5]=t[n].src,a[6]=t[n].tip,a[7]=t[n].icon,a[9]=n,a}function cn(e){let t,n,a,r,u=s(e[0]),g=[];for(let t=0;t<u.length;t+=1)g[t]=_n(pn(e,u,t));const f=e=>_(g[e],1,1,(()=>{g[e]=null}));let b=[e[1],{role:"button"},{tabindex:"0"}],h={};for(let e=0;e<b.length;e+=1)h=U(h,b[e]);return{c(){t=i("div");for(let e=0;e<g.length;e+=1)g[e].c();V(t,h),H(t,"svelte-14lcnln",!0)},m(s,i){l(s,t,i);for(let e=0;e<g.length;e+=1)g[e]&&g[e].m(t,null);n=!0,a||(r=A(t,"mouseenter",e[3]),a=!0)},p(e,n){if(1&n){let a;for(u=s(e[0]),a=0;a<u.length;a+=1){const s=pn(e,u,a);g[a]?(g[a].p(s,n),o(g[a],1)):(g[a]=_n(s),g[a].c(),o(g[a],1),g[a].m(t,null))}for(p(),a=u.length;a<g.length;a+=1)f(a);c()}H(t,"svelte-14lcnln",!0)},i(e){if(!n){for(let e=0;e<u.length;e+=1)o(g[e]);n=!0}},o(e){g=g.filter(on);for(let e=0;e<g.length;e+=1)_(g[e]);n=!1},d(e){e&&d(t),m(g,e),a=!1,r()}}}function _n(e){let t,n;return t=new ln({props:{src:e[5],tip:e[6],icon:e[7]}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.src=e[5]),1&n&&(a.tip=e[6]),1&n&&(a.icon=e[7]),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function dn(e){let t,n,a=e[0].length>0&&cn(e);return{c(){a&&a.c(),t=g()},m(e,s){a&&a.m(e,s),l(e,t,s),n=!0},p(e,[n]){e[0].length>0?a?(a.p(e,n),1&n&&o(a,1)):(a=cn(e),a.c(),o(a,1),a.m(t.parentNode,t)):a&&(p(),_(a,1,1,(()=>{a=null})),c())},i(e){n||(o(a),n=!0)},o(e){_(a),n=!1},d(e){e&&d(t),a&&a.d(e)}}}function mn(e,t,n){const a=["data"];let s=B(t,a),{data:i=[]}=t;Array.isArray(i)||(i=[i]),i=i.map((e=>"string"==typeof e?{src:e,tip:`Download the ${e.split(".").at(-1)} format.`,icon:"auto"}:(e.tip=e.tip||`Download the ${e.src.split(".").at(-1)} format.`,e.icon=e.icon||"auto",e)));const r={...s,class:[s.class,"pipen-report-image-download-list"].filter(Boolean).join(" ")},l=D();return e.$$set=e=>{t=U(U({},t),I(e)),n(4,s=B(t,a)),"data"in e&&n(0,i=e.data)},[i,r,l,()=>l("mouseenter")]}class un extends t{constructor(e){super(),n(this,e,mn,dn,a,{data:0})}}const gn=e=>({}),fn=e=>({}),bn=e=>({}),hn=e=>({}),vn=e=>({}),Cn=e=>({}),yn=e=>({}),$n=e=>({});function xn(e){let t,n;return t=new Me({props:{ratio:e[6],$$slots:{default:[Tn]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};64&n&&(a.ratio=e[6]),33556159&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function Sn(e){let t,n,a,s,i=e[2]&&wn(e),r=e[0]&&Rn(e),m=e[1]&&Nn(e);return{c(){i&&i.c(),t=v(),r&&r.c(),n=v(),m&&m.c(),a=g()},m(e,o){i&&i.m(e,o),l(e,t,o),r&&r.m(e,o),l(e,n,o),m&&m.m(e,o),l(e,a,o),s=!0},p(e,s){e[2]?i?(i.p(e,s),4&s&&o(i,1)):(i=wn(e),i.c(),o(i,1),i.m(t.parentNode,t)):i&&(p(),_(i,1,1,(()=>{i=null})),c()),e[0]?r?(r.p(e,s),1&s&&o(r,1)):(r=Rn(e),r.c(),o(r,1),r.m(n.parentNode,n)):r&&(p(),_(r,1,1,(()=>{r=null})),c()),e[1]?m?(m.p(e,s),2&s&&o(m,1)):(m=Nn(e),m.c(),o(m,1),m.m(a.parentNode,a)):m&&(p(),_(m,1,1,(()=>{m=null})),c())},i(e){s||(o(i),o(r),o(m),s=!0)},o(e){_(i),_(r),_(m),s=!1},d(e){e&&(d(t),d(n),d(a)),i&&i.d(e),r&&r.d(e),m&&m.d(e)}}}function Mn(e){let t;const n=e[15].loading,a=L(n,e,e[25],hn);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||33554432&s)&&T(a,n,e,e[25],t?R(n,e[25],s,bn):w(e[25]),hn)},i(e){t||(o(a,e),t=!0)},o(e){_(a,e),t=!1},d(e){a&&a.d(e)}}}function An(e){let t,n,a,s,r,m,u,f,b=[e[12],{src:n=e[3]},{alt:e[4]},{title:"Click to zoom in ..."}],h={};for(let e=0;e<b.length;e+=1)h=U(h,b[e]);let C=e[9]&&Pn(e);return{c(){t=i("img"),s=v(),C&&C.c(),r=g(),V(t,h)},m(n,a){l(n,t,a),l(n,s,a),C&&C.m(n,a),l(n,r,a),m=!0,u||(f=[A(t,"click",e[11]),A(t,"mouseenter",e[19]),A(t,"mouseleave",e[20])],u=!0)},p(a,s){V(t,h=N(b,[(e=a)[12],(!m||8&s&&!F(t.src,n=e[3]))&&{src:n},(!m||16&s)&&{alt:e[4]},{title:"Click to zoom in ..."}])),e[9]?C?(C.p(e,s),512&s&&o(C,1)):(C=Pn(e),C.c(),o(C,1),C.m(r.parentNode,r)):C&&(p(),_(C,1,1,(()=>{C=null})),c())},i(n){m||(n&&z((()=>{m&&(a||(a=k(t,q,{duration:e[7]?In:0},!0)),a.run(1))})),o(C),m=!0)},o(n){n&&(a||(a=k(t,q,{duration:e[7]?In:0},!1)),a.run(0)),_(C),m=!1},d(e){e&&(d(t),d(s),d(r)),e&&a&&a.end(),C&&C.d(e),u=!1,O(f)}}}function Pn(e){let t,n;return t=new un({props:{data:e[5]}}),t.$on("mouseenter",e[21]),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};32&n&&(a.data=e[5]),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function Ln(e){let t;const n=e[15].error,a=L(n,e,e[25],fn);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||33554432&s)&&T(a,n,e,e[25],t?R(n,e[25],s,gn):w(e[25]),fn)},i(e){t||(o(a,e),t=!0)},o(e){_(a,e),t=!1},d(e){a&&a.d(e)}}}function Tn(e){let t,n,a,s,i=e[2]&&Mn(e),r=e[0]&&An(e),m=e[1]&&Ln(e);return{c(){i&&i.c(),t=v(),r&&r.c(),n=v(),m&&m.c(),a=g()},m(e,o){i&&i.m(e,o),l(e,t,o),r&&r.m(e,o),l(e,n,o),m&&m.m(e,o),l(e,a,o),s=!0},p(e,s){e[2]?i?(i.p(e,s),4&s&&o(i,1)):(i=Mn(e),i.c(),o(i,1),i.m(t.parentNode,t)):i&&(p(),_(i,1,1,(()=>{i=null})),c()),e[0]?r?(r.p(e,s),1&s&&o(r,1)):(r=An(e),r.c(),o(r,1),r.m(n.parentNode,n)):r&&(p(),_(r,1,1,(()=>{r=null})),c()),e[1]?m?(m.p(e,s),2&s&&o(m,1)):(m=Ln(e),m.c(),o(m,1),m.m(a.parentNode,a)):m&&(p(),_(m,1,1,(()=>{m=null})),c())},i(e){s||(o(i),o(r),o(m),s=!0)},o(e){_(i),_(r),_(m),s=!1},d(e){e&&(d(t),d(n),d(a)),i&&i.d(e),r&&r.d(e),m&&m.d(e)}}}function wn(e){let t;const n=e[15].loading,a=L(n,e,e[25],$n);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||33554432&s)&&T(a,n,e,e[25],t?R(n,e[25],s,yn):w(e[25]),$n)},i(e){t||(o(a,e),t=!0)},o(e){_(a,e),t=!1},d(e){a&&a.d(e)}}}function Rn(e){let t,n,a,s,r,m,u,f,b=[e[12],{src:n=e[3]},{alt:e[4]},{title:"Click to zoom in ..."}],h={};for(let e=0;e<b.length;e+=1)h=U(h,b[e]);let C=e[9]&&Un(e);return{c(){t=i("img"),s=v(),C&&C.c(),r=g(),V(t,h)},m(n,a){l(n,t,a),l(n,s,a),C&&C.m(n,a),l(n,r,a),m=!0,u||(f=[A(t,"click",e[11]),A(t,"mouseenter",e[16]),A(t,"mouseleave",e[17])],u=!0)},p(a,s){V(t,h=N(b,[(e=a)[12],(!m||8&s&&!F(t.src,n=e[3]))&&{src:n},(!m||16&s)&&{alt:e[4]},{title:"Click to zoom in ..."}])),e[9]?C?(C.p(e,s),512&s&&o(C,1)):(C=Un(e),C.c(),o(C,1),C.m(r.parentNode,r)):C&&(p(),_(C,1,1,(()=>{C=null})),c())},i(n){m||(n&&z((()=>{m&&(a||(a=k(t,q,{duration:e[7]?In:0},!0)),a.run(1))})),o(C),m=!0)},o(n){n&&(a||(a=k(t,q,{duration:e[7]?In:0},!1)),a.run(0)),_(C),m=!1},d(e){e&&(d(t),d(s),d(r)),e&&a&&a.end(),C&&C.d(e),u=!1,O(f)}}}function Un(e){let t,n;return t=new un({props:{data:e[5]}}),t.$on("mouseenter",e[18]),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};32&n&&(a.data=e[5]),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function Nn(e){let t;const n=e[15].error,a=L(n,e,e[25],Cn);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||33554432&s)&&T(a,n,e,e[25],t?R(n,e[25],s,vn):w(e[25]),Cn)},i(e){t||(o(a,e),t=!0)},o(e){_(a,e),t=!1},d(e){a&&a.d(e)}}}function En(e){let t,n,a,s;return{c(){t=i("img"),r(t,"class","pipen-report-image-lg"),F(t.src,n=e[3])||r(t,"src",n),r(t,"alt",e[4])},m(e,n){l(e,t,n),s=!0},p(a,i){e=a,(!s||8&i&&!F(t.src,n=e[3]))&&r(t,"src",n),(!s||16&i)&&r(t,"alt",e[4])},i(n){s||(n&&z((()=>{s&&(a||(a=k(t,q,{duration:e[7]?In:0},!0)),a.run(1))})),s=!0)},o(n){n&&(a||(a=k(t,q,{duration:e[7]?In:0},!1)),a.run(0)),s=!1},d(e){e&&d(t),e&&a&&a.end()}}}function Bn(e){let t,n,a,s,i,r;const m=[Sn,xn],u=[];function g(e,t){return void 0===e[6]?0:1}function x(t){e[22](t)}t=g(e),n=u[t]=m[t](e);let S={modalHeading:e[4],primaryButtonText:"Open in new window ...",secondaryButtonText:"Close",$$slots:{default:[En]},$$scope:{ctx:e}};return void 0!==e[8]&&(S.open=e[8]),s=new de({props:S}),C.push((()=>y(s,"open",x))),s.$on("click:button--secondary",e[23]),s.$on("click:button--primary",e[24]),{c(){n.c(),a=v(),f(s.$$.fragment)},m(e,n){u[t].m(e,n),l(e,a,n),b(s,e,n),r=!0},p(e,[r]){let l=t;t=g(e),t===l?u[t].p(e,r):(p(),_(u[l],1,1,(()=>{u[l]=null})),c(),n=u[t],n?n.p(e,r):(n=u[t]=m[t](e),n.c()),o(n,1),n.m(a.parentNode,a));const d={};16&r&&(d.modalHeading=e[4]),33554584&r&&(d.$$scope={dirty:r,ctx:e}),!i&&256&r&&(i=!0,d.open=e[8],$((()=>i=!1))),s.$set(d)},i(e){r||(o(n),o(s.$$.fragment,e),r=!0)},o(e){_(n),_(s.$$.fragment,e),r=!1},d(e){e&&d(a),u[t].d(e),h(s,e)}}}const In=110;function Dn(e,t,n){const a=["src","alt","download","ratio","loading","loaded","error","fadeIn","border","loadImage"];let s=B(t,a),{$$slots:i={},$$scope:r}=t,{src:l=""}=t,{alt:o=""}=t,{download:p=[]}=t,{ratio:c}=t,{loading:_=!1}=t,{loaded:d=!1}=t,{error:m=!1}=t,{fadeIn:g=!1}=t,{border:f=!0}=t,b=!1,h=!1;const v=e=>{null!=$&&($=null),n(0,d=!1),n(1,m=!1),$=new Image,$.src=e||l,$.onload=()=>n(0,d=!0),$.onerror=()=>n(1,m=!0)},C=D();let y,$=null;u((()=>()=>$=null));const x={...s,class:[s.class,"pipen-report-image",f?"":"pipen-report-image-noborder"].filter(Boolean).join(" ")};return e.$$set=e=>{t=U(U({},t),I(e)),n(28,s=B(t,a)),"src"in e&&n(3,l=e.src),"alt"in e&&n(4,o=e.alt),"download"in e&&n(5,p=e.download),"ratio"in e&&n(6,c=e.ratio),"loading"in e&&n(2,_=e.loading),"loaded"in e&&n(0,d=e.loaded),"error"in e&&n(1,m=e.error),"fadeIn"in e&&n(7,g=e.fadeIn),"border"in e&&n(13,f=e.border),"$$scope"in e&&n(25,r=e.$$scope)},e.$$.update=()=>{3&e.$$.dirty&&n(2,_=!d&&!m),8&e.$$.dirty&&l&&"undefined"!=typeof window&&v(),1&e.$$.dirty&&d&&C("load"),2&e.$$.dirty&&m&&C("error")},[d,m,_,l,o,p,c,g,b,h,y,()=>{n(8,b=!0)},x,f,v,i,()=>{n(9,h=!0)},()=>{n(10,y=setTimeout((()=>{n(9,h=!1)}),100))},()=>{clearTimeout(y),n(9,h=!0)},()=>{n(9,h=!0)},()=>{n(10,y=setTimeout((()=>{n(9,h=!1)}),100))},()=>{clearTimeout(y),n(9,h=!0)},function(e){b=e,n(8,b)},()=>{n(8,b=!1)},()=>{n(8,b=!1),window.open(l)},r]}class Vn extends t{constructor(e){super(),n(this,e,Dn,Bn,a,{src:3,alt:4,download:5,ratio:6,loading:2,loaded:0,error:1,fadeIn:7,border:13,loadImage:14})}get loadImage(){return this.$$.ctx[14]}}function Hn(e){let t,n,a=[e[2]],s={};for(let e=0;e<a.length;e+=1)s=U(s,a[e]);return{c(){t=i("div"),n=x("Image loading ..."),V(t,s)},m(e,a){l(e,t,a),S(t,n)},p:P,d(e){e&&d(t)}}}function Gn(e){let t,n,a=[e[3]],s={};for(let e=0;e<a.length;e+=1)s=U(s,a[e]);return{c(){t=i("div"),n=x("Image loading error!"),V(t,s)},m(e,a){l(e,t,a),S(t,n)},p:P,d(e){e&&d(t)}}}function Fn(e){let t,n,a;const s=[{src:e[1]},e[4]];let r={$$slots:{error:[Gn],loading:[Hn]},$$scope:{ctx:e}};for(let e=0;e<s.length;e+=1)r=U(r,s[e]);n=new Vn({props:r});let p=[e[0]],c={};for(let e=0;e<p.length;e+=1)c=U(c,p[e]);return{c(){t=i("div"),f(n.$$.fragment),V(t,c)},m(e,s){l(e,t,s),b(n,t,null),a=!0},p(e,[a]){const i=18&a?N(s,[2&a&&{src:e[1]},16&a&&E(e[4])]):{};256&a&&(i.$$scope={dirty:a,ctx:e}),n.$set(i),V(t,c=N(p,[1&a&&e[0]]))},i(e){a||(o(n.$$.fragment,e),a=!0)},o(e){_(n.$$.fragment,e),a=!1},d(e){e&&d(t),h(n)}}}function zn(e,t,n){const a=["frameProps","width","height","src"];let s=B(t,a),{frameProps:i={class:"pipen-report-image-frame"}}=t,{width:r=300}=t,{height:l=200}=t,{src:o}=t;i||(i={"style:position":"relative"});let p=`width: ${r}px; height: ${l}px; aspect-ratio: ${r} / ${l}`,c={...s,class:[s.class,"pipen-report-image-loading"].filter(Boolean).join(" "),style:s.style?`${s.style}; ${p}`:p},_={...c,class:[c.class,"error"].join(" ")};return e.$$set=e=>{t=U(U({},t),I(e)),n(4,s=B(t,a)),"frameProps"in e&&n(0,i=e.frameProps),"width"in e&&n(5,r=e.width),"height"in e&&n(6,l=e.height),"src"in e&&n(1,o=e.src)},[i,o,c,_,s,r,l]}let kn=class extends t{constructor(e){super(),n(this,e,zn,Fn,a,{frameProps:0,width:5,height:6,src:1})}};const qn=e=>({}),On=e=>({});function Kn(e){let t,n;const a=e[25].default,s=L(a,e,e[24],null);return{c(){t=i("ul"),s&&s.c(),r(t,"role","menu"),r(t,"tabindex","-1"),r(t,"aria-label",e[14]),r(t,"data-floating-menu-direction",e[5]),r(t,"class",e[9]),H(t,"bx--overflow-menu-options",!0),H(t,"bx--overflow-menu--flip",e[7]),H(t,"bx--overflow-menu-options--open",e[0]),H(t,"bx--overflow-menu-options--light",e[6]),H(t,"bx--overflow-menu-options--sm","sm"===e[4]),H(t,"bx--overflow-menu-options--xl","xl"===e[4]),H(t,"bx--breadcrumb-menu-options",!!e[15])},m(a,i){l(a,t,i),s&&s.m(t,null),e[32](t),n=!0},p(e,i){s&&s.p&&(!n||16777216&i[0])&&T(s,a,e,e[24],n?R(a,e[24],i,null):w(e[24]),null),(!n||16384&i[0])&&r(t,"aria-label",e[14]),(!n||32&i[0])&&r(t,"data-floating-menu-direction",e[5]),(!n||512&i[0])&&r(t,"class",e[9]),(!n||512&i[0])&&H(t,"bx--overflow-menu-options",!0),(!n||640&i[0])&&H(t,"bx--overflow-menu--flip",e[7]),(!n||513&i[0])&&H(t,"bx--overflow-menu-options--open",e[0]),(!n||576&i[0])&&H(t,"bx--overflow-menu-options--light",e[6]),(!n||528&i[0])&&H(t,"bx--overflow-menu-options--sm","sm"===e[4]),(!n||528&i[0])&&H(t,"bx--overflow-menu-options--xl","xl"===e[4]),(!n||33280&i[0])&&H(t,"bx--breadcrumb-menu-options",!!e[15])},i(e){n||(o(s,e),n=!0)},o(e){_(s,e),n=!1},d(n){n&&d(t),s&&s.d(n),e[32](null)}}}function jn(e){let t,n,a,s,r,m,u,C;const y=e[25].menu,$=L(y,e,e[24],On),x=$||function(e){let t,n,a;var s=e[1];function i(e,t){return{props:{"aria-label":e[11],title:e[11],class:"bx--overflow-menu__icon "+e[10]}}}return s&&(t=X(s,i(e))),{c(){t&&f(t.$$.fragment),n=g()},m(e,s){t&&b(t,e,s),l(e,n,s),a=!0},p(e,a){if(2&a[0]&&s!==(s=e[1])){if(t){p();const e=t;_(e.$$.fragment,1,0,(()=>{h(e,1)})),c()}s?(t=X(s,i(e)),f(t.$$.fragment),o(t.$$.fragment,1),b(t,n.parentNode,n)):t=null}else if(s){const n={};2048&a[0]&&(n["aria-label"]=e[11]),2048&a[0]&&(n.title=e[11]),1024&a[0]&&(n.class="bx--overflow-menu__icon "+e[10]),t.$set(n)}},i(e){a||(t&&o(t.$$.fragment,e),a=!0)},o(e){t&&_(t.$$.fragment,e),a=!1},d(e){e&&d(n),t&&h(t,e)}}}(e);let M=e[0]&&Kn(e),P=[{type:"button"},{"aria-haspopup":""},{"aria-expanded":e[0]},{"aria-label":e[14]},{id:e[12]},e[20]],E={};for(let e=0;e<P.length;e+=1)E=U(E,P[e]);return{c(){t=new Q(!1),n=g(),a=v(),s=i("button"),x.c(),r=v(),M&&M.c(),t.a=n,V(s,E),H(s,"bx--overflow-menu",!0),H(s,"bx--overflow-menu--open",e[0]),H(s,"bx--overflow-menu--light",e[6]),H(s,"bx--overflow-menu--sm","sm"===e[4]),H(s,"bx--overflow-menu--xl","xl"===e[4])},m(i,o){t.m(e[13],document.head),S(document.head,n),l(i,a,o),l(i,s,o),x.m(s,null),S(s,r),M&&M.m(s,null),s.autofocus&&s.focus(),e[33](s),m=!0,u||(C=[A(window,"click",e[31]),A(s,"click",e[26]),A(s,"click",e[34]),A(s,"mouseover",e[27]),A(s,"mouseenter",e[28]),A(s,"mouseleave",e[29]),A(s,"keydown",e[30]),A(s,"keydown",e[35]),A(s,"focusout",e[36])],u=!0)},p(e,n){(!m||8192&n[0])&&t.p(e[13]),$?$.p&&(!m||16777216&n[0])&&T($,y,e,e[24],m?R(y,e[24],n,qn):w(e[24]),On):x.p&&(!m||3074&n[0])&&x.p(e,m?n:[-1,-1]),e[0]?M?(M.p(e,n),1&n[0]&&o(M,1)):(M=Kn(e),M.c(),o(M,1),M.m(s,null)):M&&(p(),_(M,1,1,(()=>{M=null})),c()),V(s,E=N(P,[{type:"button"},{"aria-haspopup":""},(!m||1&n[0])&&{"aria-expanded":e[0]},(!m||16384&n[0])&&{"aria-label":e[14]},(!m||4096&n[0])&&{id:e[12]},1048576&n[0]&&e[20]])),H(s,"bx--overflow-menu",!0),H(s,"bx--overflow-menu--open",e[0]),H(s,"bx--overflow-menu--light",e[6]),H(s,"bx--overflow-menu--sm","sm"===e[4]),H(s,"bx--overflow-menu--xl","xl"===e[4])},i(e){m||(o(x,e),o(M),m=!0)},o(e){_(x,e),_(M),m=!1},d(i){i&&(t.d(),d(a),d(s)),d(n),x.d(i),M&&M.d(),e[33](null),u=!1,O(C)}}}function Wn(e,t,n){let a,s;const i=["size","direction","open","light","flipped","persistentClickItems","menuOptionsClass","icon","iconClass","iconDescription","id","buttonRef","menuRef"];let r,l,o,p=B(t,i),{$$slots:c={},$$scope:_}=t,{size:d}=t,{direction:m="bottom"}=t,{open:u=!1}=t,{light:g=!1}=t,{flipped:f=!1}=t,{persistentClickItems:b=!1}=t,{menuOptionsClass:h}=t,{icon:v=Qe}=t,{iconClass:y}=t,{iconDescription:$="Open and close list of options"}=t,{id:x="ccs-"+Math.random().toString(36)}=t,{buttonRef:S=null}=t,{menuRef:M=null}=t;const A=K("BreadcrumbItem"),P=D(),L=J([]);j(e,L,(e=>n(23,l=e)));const T=J(void 0);j(e,T,(e=>n(38,o=e)));const w=J(void 0),R=J(-1);let N;j(e,R,(e=>n(22,r=e)));let E=!0;W("OverflowMenu",{focusedId:w,add:({id:e,text:t,primaryFocus:n,disabled:a})=>{L.update((s=>(n&&R.set(s.length),[...s,{id:e,text:t,primaryFocus:n,disabled:a,index:s.length}])))},update:e=>{T.set(e)},change:e=>{let t=r+e;t<0?t=l.length-1:t>=l.length&&(t=0);let n=l[t].disabled;for(;n;)t+=e,t<0?t=l.length-1:t>=l.length&&(t=0),n=l[t].disabled;R.set(t)}}),Y((()=>{if(o){const{index:e,text:t}=l.filter((e=>e.id===o))[0];P("close",{index:e,text:t}),n(0,u=!1)}if(u){const{width:e,height:t}=S.getBoundingClientRect();n(21,N=e),!E&&r<0&&M.focus(),f&&(n(3,M.style.left="auto",M),n(3,M.style.right=0,M)),"top"===m?(n(3,M.style.top="auto",M),n(3,M.style.bottom=t+"px",M)):"bottom"===m&&n(3,M.style.top=t+"px",M),A&&(n(3,M.style.top=t+10+"px",M),n(3,M.style.left="-11px",M))}u||(L.set([]),T.set(void 0),R.set(0)),E=!1}));return e.$$set=e=>{n(40,t=U(U({},t),I(e))),n(20,p=B(t,i)),"size"in e&&n(4,d=e.size),"direction"in e&&n(5,m=e.direction),"open"in e&&n(0,u=e.open),"light"in e&&n(6,g=e.light),"flipped"in e&&n(7,f=e.flipped),"persistentClickItems"in e&&n(8,b=e.persistentClickItems),"menuOptionsClass"in e&&n(9,h=e.menuOptionsClass),"icon"in e&&n(1,v=e.icon),"iconClass"in e&&n(10,y=e.iconClass),"iconDescription"in e&&n(11,$=e.iconDescription),"id"in e&&n(12,x=e.id),"buttonRef"in e&&n(2,S=e.buttonRef),"menuRef"in e&&n(3,M=e.menuRef),"$$scope"in e&&n(24,_=e.$$scope)},e.$$.update=()=>{n(14,a=t["aria-label"]||"menu"),12582912&e.$$.dirty[0]&&l[r]&&w.set(l[r].id),2101248&e.$$.dirty[0]&&n(13,s=`<style>\n    #${x} .bx--overflow-menu-options.bx--overflow-menu-options:after {\n      width: ${N?N+"px":"2rem"};\n    }\n  </style>`)},A&&n(1,v=Xe),t=I(t),[u,v,S,M,d,m,g,f,b,h,y,$,x,s,a,A,P,L,T,R,p,N,r,l,_,c,function(t){Z.call(this,e,t)},function(t){Z.call(this,e,t)},function(t){Z.call(this,e,t)},function(t){Z.call(this,e,t)},function(t){Z.call(this,e,t)},({target:e})=>{S&&S.contains(e)||M&&!M.contains(e)&&n(0,u=!1)},function(e){C[e?"unshift":"push"]((()=>{M=e,n(3,M)}))},function(e){C[e?"unshift":"push"]((()=>{S=e,n(2,S)}))},e=>{b&&e.stopPropagation(),M&&M.contains(e.target)||(n(0,u=!u),u||P("close"))},e=>{u&&(["ArrowDown","ArrowLeft","ArrowRight","ArrowUp"].includes(e.key)?e.preventDefault():"Escape"===e.key&&(e.stopPropagation(),P("close"),n(0,u=!1),S.focus()))},e=>{u&&(S.contains(e.relatedTarget)||(P("close"),n(0,u=!1)))}]}class Yn extends t{constructor(e){super(),n(this,e,Wn,jn,a,{size:4,direction:5,open:0,light:6,flipped:7,persistentClickItems:8,menuOptionsClass:9,icon:1,iconClass:10,iconDescription:11,id:12,buttonRef:2,menuRef:3},null,[-1,-1])}}function Xn(e,t,n){const a=e.slice();return a[3]=t[n],a[4]=t,a[5]=n,a}function Qn(e){let t,n,a,s,p;function c(t){e[1](t,e[3])}let m={labelText:e[3].value};return void 0!==e[3].selected&&(m.checked=e[3].selected),n=new Ae({props:m}),C.push((()=>y(n,"checked",c))),n.$on("change",e[2]),{c(){t=i("li"),f(n.$$.fragment),s=v(),r(t,"class","pipen-report-column-selector-item svelte-147jw2r")},m(e,a){l(e,t,a),b(n,t,null),S(t,s),p=!0},p(t,s){e=t;const i={};1&s&&(i.labelText=e[3].value),!a&&1&s&&(a=!0,i.checked=e[3].selected,$((()=>a=!1))),n.$set(i)},i(e){p||(o(n.$$.fragment,e),p=!0)},o(e){_(n.$$.fragment,e),p=!1},d(e){e&&d(t),h(n)}}}function Jn(e){let t,n,a=s(e[0]),i=[];for(let t=0;t<a.length;t+=1)i[t]=Qn(Xn(e,a,t));const r=e=>_(i[e],1,1,(()=>{i[e]=null}));return{c(){for(let e=0;e<i.length;e+=1)i[e].c();t=g()},m(e,a){for(let t=0;t<i.length;t+=1)i[t]&&i[t].m(e,a);l(e,t,a),n=!0},p(e,n){if(1&n){let l;for(a=s(e[0]),l=0;l<a.length;l+=1){const s=Xn(e,a,l);i[l]?(i[l].p(s,n),o(i[l],1)):(i[l]=Qn(s),i[l].c(),o(i[l],1),i[l].m(t.parentNode,t))}for(p(),l=a.length;l<i.length;l+=1)r(l);c()}},i(e){if(!n){for(let e=0;e<a.length;e+=1)o(i[e]);n=!0}},o(e){i=i.filter(Boolean);for(let e=0;e<i.length;e+=1)_(i[e]);n=!1},d(e){e&&d(t),m(i,e)}}}function Zn(e){let t,n;return t=new Yn({props:{class:"pipen-report-overflowmenu",persistentClickItems:!0,size:"sm",icon:Je,iconDescription:"Select visible columns ...",$$slots:{default:[Jn]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,[n]){const a={};65&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function ea(e,t,n){let{allColumns:a}=t;return e.$$set=e=>{"allColumns"in e&&n(0,a=e.allColumns)},[a,function(t,s){e.$$.not_equal(s.selected,t)&&(s.selected=t,n(0,a))},()=>{n(0,a)}]}class ta extends t{constructor(e){super(),n(this,e,ea,Zn,a,{allColumns:0})}}function na(e){let t,n;return t=new xe({props:{iconDescription:"Right click and save as to download the entire data",tooltipPosition:"top",icon:Ze,href:e[2]}}),t.$on("click",e[13]),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};4&n&&(a.href=e[2]),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function aa(e){let t,n,a,s,i,r,m,u,x,S;function M(t){e[17](t)}let A={persistent:!0};function P(t){e[18](t)}void 0!==e[7]&&(A.value=e[7]),t=new Re({props:A}),C.push((()=>y(t,"value",M)));let L={};void 0!==e[6]&&(L.allColumns=e[6]),s=new ta({props:L}),C.push((()=>y(s,"allColumns",P))),m=new xe({props:{iconDescription:`Download current ${e[11].length} records (may be partial of the entire data)`,tooltipPosition:"top",icon:Ie}}),m.$on("click",e[14]);let T=!!e[2]&&na(e);return{c(){f(t.$$.fragment),a=v(),f(s.$$.fragment),r=v(),f(m.$$.fragment),u=v(),T&&T.c(),x=g()},m(e,n){b(t,e,n),l(e,a,n),b(s,e,n),l(e,r,n),b(m,e,n),l(e,u,n),T&&T.m(e,n),l(e,x,n),S=!0},p(e,a){const r={};!n&&128&a&&(n=!0,r.value=e[7],$((()=>n=!1))),t.$set(r);const l={};!i&&64&a&&(i=!0,l.allColumns=e[6],$((()=>i=!1))),s.$set(l);const d={};2048&a&&(d.iconDescription=`Download current ${e[11].length} records (may be partial of the entire data)`),m.$set(d),e[2]?T?(T.p(e,a),4&a&&o(T,1)):(T=na(e),T.c(),o(T,1),T.m(x.parentNode,x)):T&&(p(),_(T,1,1,(()=>{T=null})),c())},i(e){S||(o(t.$$.fragment,e),o(s.$$.fragment,e),o(m.$$.fragment,e),o(T),S=!0)},o(e){_(t.$$.fragment,e),_(s.$$.fragment,e),_(m.$$.fragment,e),_(T),S=!1},d(e){e&&(d(a),d(r),d(u),d(x)),h(t,e),h(s,e),h(m,e),T&&T.d(e)}}}function sa(e){let t,n;return t=new we({props:{$$slots:{default:[aa]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};134219972&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function ia(e){let t,n;return t=new Te({props:{size:e[12],$$slots:{default:[sa]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};134219972&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function ra(e){let t;return{c(){t=i("p"),t.textContent='You must use "Save as" or "Save link as" from the context menu (by right-clicking the button) to download the entire data.'},m(e,n){l(e,t,n)},p:P,d(e){e&&d(t)}}}function la(e){let t,n,a,s,i,r,p,c,m;const u=[{size:e[0]},{zebra:e[3]},{sortable:e[4]},{page:e[8]},{pageSize:e[9]},{rows:e[11]},{headers:e[1]},e[15]];let g={$$slots:{default:[ia]},$$scope:{ctx:e}};for(let e=0;e<u.length;e+=1)g=U(g,u[e]);function x(t){e[19](t)}function S(t){e[20](t)}t=new Pe({props:g});let M={pageSizes:e[5],totalItems:e[11].length};function A(t){e[21](t)}void 0!==e[9]&&(M.pageSize=e[9]),void 0!==e[8]&&(M.page=e[8]),a=new Le({props:M}),C.push((()=>y(a,"pageSize",x))),C.push((()=>y(a,"page",S)));let P={passiveModal:!0,modalHeading:"Downloading entire data",$$slots:{default:[ra]},$$scope:{ctx:e}};return void 0!==e[10]&&(P.open=e[10]),p=new de({props:P}),C.push((()=>y(p,"open",A))),p.$on("open",e[22]),p.$on("close",e[23]),{c(){f(t.$$.fragment),n=v(),f(a.$$.fragment),r=v(),f(p.$$.fragment)},m(e,s){b(t,e,s),l(e,n,s),b(a,e,s),l(e,r,s),b(p,e,s),m=!0},p(e,[n]){const r=35611&n?N(u,[1&n&&{size:e[0]},8&n&&{zebra:e[3]},16&n&&{sortable:e[4]},256&n&&{page:e[8]},512&n&&{pageSize:e[9]},2048&n&&{rows:e[11]},2&n&&{headers:e[1]},32768&n&&E(e[15])]):{};134219972&n&&(r.$$scope={dirty:n,ctx:e}),t.$set(r);const l={};32&n&&(l.pageSizes=e[5]),2048&n&&(l.totalItems=e[11].length),!s&&512&n&&(s=!0,l.pageSize=e[9],$((()=>s=!1))),!i&&256&n&&(i=!0,l.page=e[8],$((()=>i=!1))),a.$set(l);const o={};134217728&n&&(o.$$scope={dirty:n,ctx:e}),!c&&1024&n&&(c=!0,o.open=e[10],$((()=>c=!1))),p.$set(o)},i(e){m||(o(t.$$.fragment,e),o(a.$$.fragment,e),o(p.$$.fragment,e),m=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),_(p.$$.fragment,e),m=!1},d(e){e&&(d(n),d(r)),h(t,e),h(a,e),h(p,e)}}}function oa(e,t,n){const a=["size","src","zebra","sortable","rows","headers","pageSizes"];let s=B(t,a),{size:i="sm"}=t,{src:r}=t;i={sm:"short",small:"short",short:"short",md:"medium",medium:"medium",lg:"tall",large:"tall",tall:"tall",compact:"compact"}[i];let l={short:"sm",compact:"sm",medium:"default",tall:"default"}[i],{zebra:o=!0}=t,{sortable:p=!0}=t,{rows:c=[]}=t,{headers:_=[]}=t,{pageSizes:d=[20,50,75,100]}=t,m=_.map((e=>({...e,selected:!0}))),u=1,g=d[0],f="",b=!1,h=c;return e.$$set=e=>{t=U(U({},t),I(e)),n(15,s=B(t,a)),"size"in e&&n(0,i=e.size),"src"in e&&n(2,r=e.src),"zebra"in e&&n(3,o=e.zebra),"sortable"in e&&n(4,p=e.sortable),"rows"in e&&n(16,c=e.rows),"headers"in e&&n(1,_=e.headers),"pageSizes"in e&&n(5,d=e.pageSizes)},e.$$.update=()=>{64&e.$$.dirty&&n(1,_=m.filter((e=>e.selected))),65664&e.$$.dirty&&(f.length>0?n(11,h=c.filter((e=>Object.values(e).filter((e=>!!e&&e.toString().includes(f))).length>0))):n(11,h=c))},[i,_,r,o,p,d,m,f,u,g,b,h,l,e=>{e.preventDefault(),n(10,b=!0)},()=>{((e,t)=>{const n=document.createElement("a");n.href=e,n.download=t,document.body.appendChild(n),n.click(),document.body.removeChild(n)})("data:application/vnd.ms-excel;charset=utf-8,\ufeff"+encodeURIComponent(function(){const e=_.map((e=>e.value)),t=[];return h.forEach((e=>{const n=_.map((t=>`"${e[t.key]}"`));t.push(n.join(","))})),e.join(",")+"\r\n"+t.join("\r\n")}()),r&&r.length>0?r.split("/").pop().split(".")[0]+".xls":"data-subset.xls")},s,c,function(e){f=e,n(7,f)},function(e){m=e,n(6,m)},function(e){g=e,n(9,g)},function(e){u=e,n(8,u)},function(e){b=e,n(10,b)},function(t){Z.call(this,e,t)},function(t){Z.call(this,e,t)}]}class pa extends t{constructor(e){super(),n(this,e,oa,la,a,{size:0,src:2,zebra:3,sortable:4,rows:16,headers:1,pageSizes:5})}}function ca(e){let t,n,a;const s=[{headers:e[0]},{rows:e[1]},e[3]];let r={};for(let e=0;e<s.length;e+=1)r=U(r,s[e]);n=new pa({props:r});let p=[e[2]],c={};for(let e=0;e<p.length;e+=1)c=U(c,p[e]);return{c(){t=i("div"),f(n.$$.fragment),V(t,c)},m(e,s){l(e,t,s),b(n,t,null),a=!0},p(e,a){const i=11&a?N(s,[1&a&&{headers:e[0]},2&a&&{rows:e[1]},8&a&&E(e[3])]):{};n.$set(i),V(t,c=N(p,[4&a&&e[2]]))},i(e){a||(o(n.$$.fragment,e),a=!0)},o(e){_(n.$$.fragment,e),a=!1},d(e){e&&d(t),h(n)}}}function _a(e){let t,n;const a=[{headers:e[0]},{rows:e[1]},e[3]];let s={};for(let e=0;e<a.length;e+=1)s=U(s,a[e]);return t=new pa({props:s}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const s=11&n?N(a,[1&n&&{headers:e[0]},2&n&&{rows:e[1]},8&n&&E(e[3])]):{};t.$set(s)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function da(e){let t,n,a,s;const i=[_a,ca],r=[];function m(e,t){return e[2]?1:0}return t=m(e),n=r[t]=i[t](e),{c(){n.c(),a=g()},m(e,n){r[t].m(e,n),l(e,a,n),s=!0},p(e,[s]){let l=t;t=m(e),t===l?r[t].p(e,s):(p(),_(r[l],1,1,(()=>{r[l]=null})),c(),n=r[t],n?n.p(e,s):(n=r[t]=i[t](e),n.c()),o(n,1),n.m(a.parentNode,a))},i(e){s||(o(n),s=!0)},o(e){_(n),s=!1},d(e){e&&d(a),r[t].d(e)}}}function ma(e,t,n){const a=["frameProps","data","headers","rows"];let s=B(t,a),{frameProps:i={class:"pipen-report-datatable-frame"}}=t,{data:r={}}=t,{headers:l=[]}=t,{rows:o=[]}=t;if(l.length+o.length>0&&r.length>0)throw"Can only specify `data` or `headers`/`rows` for `DataTable`!";return r.length>0&&(o=r,l=Object.keys(o[0]).filter((e=>"id"!==e)).map((e=>({key:e,value:e})))),e.$$set=e=>{t=U(U({},t),I(e)),n(3,s=B(t,a)),"frameProps"in e&&n(2,i=e.frameProps),"data"in e&&n(4,r=e.data),"headers"in e&&n(0,l=e.headers),"rows"in e&&n(1,o=e.rows)},[l,o,i,s,r]}class ua extends t{constructor(e){super(),n(this,e,ma,da,a,{frameProps:2,data:4,headers:0,rows:1})}}function ga(e){let t,n;return{c(){t=i("div"),n=x(e[0]),r(t,"class","pipen-report-descr-title svelte-1xzxuva")},m(e,a){l(e,t,a),S(t,n)},p(e,t){1&t&&M(n,e[0])},d(e){e&&d(t)}}}function fa(e){let t,n,a,s=e[0]&&ga(e);const r=e[3].default,p=L(r,e,e[2],null);let c=[e[1]],m={};for(let e=0;e<c.length;e+=1)m=U(m,c[e]);return{c(){t=i("div"),s&&s.c(),n=v(),p&&p.c(),V(t,m),H(t,"svelte-1xzxuva",!0)},m(e,i){l(e,t,i),s&&s.m(t,null),S(t,n),p&&p.m(t,null),a=!0},p(e,[i]){e[0]?s?s.p(e,i):(s=ga(e),s.c(),s.m(t,n)):s&&(s.d(1),s=null),p&&p.p&&(!a||4&i)&&T(p,r,e,e[2],a?R(r,e[2],i,null):w(e[2]),null),V(t,m=N(c,[2&i&&e[1]])),H(t,"svelte-1xzxuva",!0)},i(e){a||(o(p,e),a=!0)},o(e){_(p,e),a=!1},d(e){e&&d(t),s&&s.d(),p&&p.d(e)}}}function ba(e,t,n){const a=["title"];let s=B(t,a),{$$slots:i={},$$scope:r}=t,{title:l}=t,{klass:o,...p}=s;return o=o?`pipen-report-descr ${o}`:"pipen-report-descr",p||(p={}),p.class=o,e.$$set=e=>{t=U(U({},t),I(e)),n(5,s=B(t,a)),"title"in e&&n(0,l=e.title),"$$scope"in e&&n(2,r=e.$$scope)},[l,p,r,i]}class ha extends t{constructor(e){super(),n(this,e,ba,fa,a,{title:0})}}function va(e){let t,n,a,s;return t=new ge({props:{label:"Plot"}}),a=new ge({props:{label:"Table"}}),{c(){f(t.$$.fragment),n=v(),f(a.$$.fragment)},m(e,i){b(t,e,i),l(e,n,i),b(a,e,i),s=!0},p:P,i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),h(t,e),h(a,e)}}}function Ca(e){let t;return{c(){t=x("This table presents a comprehensive summary of the top enriched pathways derived from the fgsea. Each row corresponds to a pathway, and the gene ranks are shown based on the ranking metric used in the analysis. The enrichment score, p-value, and adjusted p-value are also provided to assess the significance of the enrichment.")},m(e,n){l(e,t,n)},d(e){e&&d(t)}}}function ya(e){let t,n,a,s;return t=new ha({props:{$$slots:{default:[Ca]},$$scope:{ctx:e}}}),a=new kn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/gsea_table.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/gsea_table.pdf",width:1e3,height:600}}),{c(){f(t.$$.fragment),n=v(),f(a.$$.fragment)},m(e,i){b(t,e,i),l(e,n,i),b(a,e,i),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),h(t,e),h(a,e)}}}function $a(e){let t;return{c(){t=x("This plot represents the GSEA results for a specified gene set, illustrating the distribution and impact of the gene set along the ranked list of genes. The running enrichment score curve shows the cumulative enrichment score as genes from the input list are encountered. Positive peaks on the curve indicate regions where members of the gene set are predominantly found.")},m(e,n){l(e,t,n)},d(e){e&&d(t)}}}function xa(e){let t,n,a,s;return t=new ha({props:{$$slots:{default:[$a]},$$scope:{ctx:e}}}),a=new ua({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/fgsea.txt",data:[{pathway:"Ascorbate and aldarate metabolism",pval:2e-4,padj:.0023,log2err:.5188,ES:-.7316,NES:-1.6163,size:27,leadingEdge:"ALDH9A1,UGDH,ALDH1B1,ALDH3A2",id:0},{pathway:"Nitrogen metabolism",pval:2e-4,padj:.0023,log2err:.5188,ES:.7213,NES:2.3546,size:17,leadingEdge:"GLUL,CA5B,CPS1,GLUD1,CA4",id:1},{pathway:"Fatty acid degradation",pval:.0261,padj:.1574,log2err:.282,ES:-.5682,NES:-1.2968,size:44,leadingEdge:"ACADVL,ADH5,ACAA2,ALDH9A1,CPT1A,ACSL3,ACOX1,ACSL5,ACADSB,HADHA,ECHS1,ALDH1B1,ALDH3A2,CPT2,HADHB,HADH,ACSL6,ACAT2,ACADS",id:2},{pathway:"Amino sugar and nucleotide sugar metabolism",pval:.0331,padj:.1574,log2err:.2489,ES:-.5539,NES:-1.2724,size:48,leadingEdge:"FPGT,HEXA,GNPDA2,HEXB,PGM2,GNPNAT1,GPI,GALT,MPI,RENBP,PMM1,CYB5R3,GNPDA1,CYB5R1,UGDH,CMAS,GNE,PGM1,HK1,GALK1,AMDHD2,GMPPA",id:3},{pathway:"Citrate cycle (TCA cycle)",pval:.0394,padj:.1574,log2err:.228,ES:-.5902,NES:-1.3163,size:30,leadingEdge:"DLST,ACO1,SUCLA2,FH,DLAT,CS,ACLY,SDHC,MDH1,SDHD,MDH2,IDH3B,SUCLG1,PDHA1,ACO2,PCK2,IDH3G,PDHB,OGDH,DLD,PC",id:4},{pathway:"Glycolysis / Gluconeogenesis",pval:.0539,padj:.1798,log2err:.1919,ES:-.5213,NES:-1.2234,size:68,leadingEdge:"ADH5,ALDH9A1,PKM,PGM2,ALDOA,ENO2,PFKL,GPI,DLAT,PGAM1,FBP1,MINPP1,ALDH1B1,ENO1,ALDH3A2,ALDOC,PGM1,HK1,GAPDH",id:5},{pathway:"Pentose and glucuronate interconversions",pval:.131,padj:.338,log2err:.1183,ES:-.5348,NES:-1.2019,size:34,leadingEdge:"AKR1B1,CRYL1,SORD,UGDH,RPE,GUSB",id:6},{pathway:"Pentose phosphate pathway",pval:.1352,padj:.338,log2err:.1162,ES:-.5427,NES:-1.2104,size:30,leadingEdge:"PGM2,PRPS1,ALDOA,PFKL,GPI,PRPS2,FBP1,ALDOC,PGM1,IDNK,G6PD,RPE,TALDO1",id:7},{pathway:"Fatty acid elongation",pval:.1847,padj:.4104,log2err:.0966,ES:-.5256,NES:-1.1721,size:30,leadingEdge:"ACAA2,HSD17B12,HACD2,PPT1,HADHA,ACOT4,ACOT2,HACD4,ECHS1,HACD3,HADHB,HADH",id:8},{pathway:"Propanoate metabolism",pval:.2377,padj:.4329,log2err:.0822,ES:-.5069,NES:-1.1368,size:32,leadingEdge:"MLYCD,MUT,SUCLA2,HADHA,HIBCH,ECHS1,BCKDHB,DBT,SUCLG1,BCKDHA,ECHDC1,PCCA,ACAT2,PCCB",id:9},{pathway:"Glyoxylate and dicarboxylate metabolism",pval:.2381,padj:.4329,log2err:.0824,ES:-.5173,NES:-1.1474,size:28,leadingEdge:"CAT,HYI,GCSH,MUT,ACO1,CS,MDH1,MDH2,PCCA,ACAT2,PCCB,ACO2",id:10},{pathway:"Starch and sucrose metabolism",pval:.3778,padj:.5948,log2err:.0591,ES:-.4726,NES:-1.0649,size:35,leadingEdge:"GAA,GYS1,PGM2,PYGB,AGL,GPI,GYG1,PGM1,HK1,GBE1,GANC",id:11},{pathway:"Oxidative phosphorylation",pval:.3866,padj:.5948,log2err:.0575,ES:-.431,NES:-1.0343,size:120,leadingEdge:"NDUFV3,PPA2,NDUFS1,ATP6V1E1,ATP6V1B2,ATP5PO,ATP6V1E2,ATP5PB,TCIRG1,NDUFA5,NDUFA11,ATP6V1A,NDUFA4,ATP6AP1,UQCRC1,ATP6V1D,ATP5F1C,LHPP,SDHC,NDUFA9,ATP5F1B,SDHD,UQCRC2,NDUFS8,COX6B1,NDUFB7,NDUFB10,COX15,ATP6V0D1,COX8A,ATP6V1H,NDUFS4,ATP5PF,ATP6V1C1,ATP6V1G1,ATP5F1D,NDUFAB1,NDUFC2,NDUFS2,ATP5MC2,ATP5MC3,NDUFS6,CYC1,NDUFS3,ATP5MC1,NDUFA13,ATP6V0E1,COX11,NDUFV1,NDUFA8,ATP6V1F,NDUFB2,COX5A,COX4I1,NDUFB3,COX6A1,ATP6V0A1,UQCR11",id:12},{pathway:"Galactose metabolism",pval:.5091,padj:.7123,log2err:.0453,ES:-.4497,NES:-1.0032,size:31,leadingEdge:"AKR1B1,GAA,PGM2,PFKL,GALT,GLB1,PGM1,HK1,GALK1",id:13},{pathway:"Butanoate metabolism",pval:.5755,padj:.7123,log2err:.0398,ES:-.4384,NES:-.9724,size:28,leadingEdge:"BDH2,HADHA,AACS,ECHS1,HMGCL,HADH,ALDH5A1,ACAT2,ACADS,OXCT1,BDH1,ACSM3,L2HGDH",id:14},{pathway:"Sulfur metabolism",pval:.5802,padj:.7123,log2err:.0438,ES:-.5086,NES:-.9563,size:9,leadingEdge:"PAPSS1,BPNT1",id:15},{pathway:"Pyruvate metabolism",pval:.6054,padj:.7123,log2err:.0371,ES:-.4244,NES:-.9653,size:39,leadingEdge:"ALDH9A1,GLO1,PKM,FH,DLAT,ME2,ALDH1B1,ALDH3A2,MDH1,MDH2,PDHA1,ACAT2,PCK2,PDHB,DLD,PC",id:16},{pathway:"Fructose and mannose metabolism",pval:.6462,padj:.713,log2err:.0342,ES:-.4187,NES:-.9399,size:33,leadingEdge:"FPGT,AKR1B1,ENOSF1,ALDOA,PFKL,SORD,MPI,FBP1,PMM1,ALDOC,HK1,GMPPA",id:17},{pathway:"Inositol phosphate metabolism",pval:.6773,padj:.713,log2err:.0315,ES:-.4016,NES:-.9461,size:73,leadingEdge:"PIP4K2C,PLCB2,IMPA1,PIK3CG,SACM1L,PLCB3,MTMR2,IPPK,INPP5B,MTM1,MINPP1,PIK3C2A,MTMR1,PIK3CD,INPP4A,MTMR6,PLCB1,MTMR8,PIP4K2B,PLCG1,ISYNA1,MTMR4,TPI1,PI4KB,PIP5K1A,CDIPT,PI4K2A,PLCH2,SYNJ1,PIK3CB,ITPKC,INPP5E,PIKFYVE,INPP4B,PIK3C2B,INPP5F,INPPL1",id:18},{pathway:"Fatty acid biosynthesis",pval:.9735,padj:.9735,log2err:.0138,ES:-.2825,NES:-.5709,size:13,leadingEdge:"ACSL3,ACSL5",id:19}]}}),{c(){f(t.$$.fragment),n=v(),f(a.$$.fragment)},m(e,i){b(t,e,i),l(e,n,i),b(a,e,i),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),h(t,e),h(a,e)}}}function Sa(e){let t,n,a,s;return t=new fe({props:{$$slots:{default:[ya]},$$scope:{ctx:e}}}),a=new fe({props:{$$slots:{default:[xa]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=v(),f(a.$$.fragment)},m(e,i){b(t,e,i),l(e,n,i),b(a,e,i),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const i={};1&n&&(i.$$scope={dirty:n,ctx:e}),a.$set(i)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),h(t,e),h(a,e)}}}function Ma(e){let t,n;return t=new ue({props:{$$slots:{content:[Sa],default:[va]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function Aa(e){let t,n,a,s,p,c,m,u,g,C,y,$,x,M,A,L,T,w,R,U,N,E,B,I,D,V,H,G,F,z,k,q,O,K,j,W,Y,X,Q,J,Z,te,ne,ae,se,ie,re,le,oe,pe,ce;return a=new ha({props:{title:"Ascorbate and aldarate metabolism",class:"pipen-report-table-image-descr"}}),p=new kn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/fgsea_Ascorbate-and-aldarate-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/fgsea_Ascorbate-and-aldarate-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),u=new ha({props:{title:"Nitrogen metabolism",class:"pipen-report-table-image-descr"}}),C=new kn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/fgsea_Nitrogen-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/fgsea_Nitrogen-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),x=new ha({props:{title:"Fatty acid degradation",class:"pipen-report-table-image-descr"}}),A=new kn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/fgsea_Fatty-acid-degradation.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/fgsea_Fatty-acid-degradation.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),w=new ha({props:{title:"Amino sugar and nucleotide sugar metabolism",class:"pipen-report-table-image-descr"}}),U=new kn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/fgsea_Amino-sugar-and-nucleotide-sugar-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/fgsea_Amino-sugar-and-nucleotide-sugar-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),B=new ha({props:{title:"Citrate cycle (TCA cycle)",class:"pipen-report-table-image-descr"}}),D=new kn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/fgsea_Citrate-cycle-TCA-cycle-.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/fgsea_Citrate-cycle-TCA-cycle-.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),G=new ha({props:{title:"Glycolysis / Gluconeogenesis",class:"pipen-report-table-image-descr"}}),z=new kn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/fgsea_Glycolysis-Gluconeogenesis.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/fgsea_Glycolysis-Gluconeogenesis.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),O=new ha({props:{title:"Pentose and glucuronate interconversions",class:"pipen-report-table-image-descr"}}),j=new kn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/fgsea_Pentose-and-glucuronate-interconversions.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/fgsea_Pentose-and-glucuronate-interconversions.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),X=new ha({props:{title:"Pentose phosphate pathway",class:"pipen-report-table-image-descr"}}),J=new kn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/fgsea_Pentose-phosphate-pathway.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/fgsea_Pentose-phosphate-pathway.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new ha({props:{title:"Fatty acid elongation",class:"pipen-report-table-image-descr"}}),se=new kn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/fgsea_Fatty-acid-elongation.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/fgsea_Fatty-acid-elongation.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),le=new ha({props:{title:"Propanoate metabolism",class:"pipen-report-table-image-descr"}}),pe=new kn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/fgsea_Propanoate-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/fgsea_Propanoate-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=i("div"),n=i("div"),f(a.$$.fragment),s=v(),f(p.$$.fragment),c=v(),m=i("div"),f(u.$$.fragment),g=v(),f(C.$$.fragment),y=v(),$=i("div"),f(x.$$.fragment),M=v(),f(A.$$.fragment),L=v(),T=i("div"),f(w.$$.fragment),R=v(),f(U.$$.fragment),N=v(),E=i("div"),f(B.$$.fragment),I=v(),f(D.$$.fragment),V=v(),H=i("div"),f(G.$$.fragment),F=v(),f(z.$$.fragment),k=v(),q=i("div"),f(O.$$.fragment),K=v(),f(j.$$.fragment),W=v(),Y=i("div"),f(X.$$.fragment),Q=v(),f(J.$$.fragment),Z=v(),te=i("div"),f(ne.$$.fragment),ae=v(),f(se.$$.fragment),ie=v(),re=i("div"),f(le.$$.fragment),oe=v(),f(pe.$$.fragment),r(t,"class","pipen-report-table-of-images"),ee(t,"grid-template-columns","repeat(2, auto)")},m(e,i){l(e,t,i),S(t,n),b(a,n,null),S(n,s),b(p,n,null),S(t,c),S(t,m),b(u,m,null),S(m,g),b(C,m,null),S(t,y),S(t,$),b(x,$,null),S($,M),b(A,$,null),S(t,L),S(t,T),b(w,T,null),S(T,R),b(U,T,null),S(t,N),S(t,E),b(B,E,null),S(E,I),b(D,E,null),S(t,V),S(t,H),b(G,H,null),S(H,F),b(z,H,null),S(t,k),S(t,q),b(O,q,null),S(q,K),b(j,q,null),S(t,W),S(t,Y),b(X,Y,null),S(Y,Q),b(J,Y,null),S(t,Z),S(t,te),b(ne,te,null),S(te,ae),b(se,te,null),S(t,ie),S(t,re),b(le,re,null),S(re,oe),b(pe,re,null),ce=!0},p:P,i(e){ce||(o(a.$$.fragment,e),o(p.$$.fragment,e),o(u.$$.fragment,e),o(C.$$.fragment,e),o(x.$$.fragment,e),o(A.$$.fragment,e),o(w.$$.fragment,e),o(U.$$.fragment,e),o(B.$$.fragment,e),o(D.$$.fragment,e),o(G.$$.fragment,e),o(z.$$.fragment,e),o(O.$$.fragment,e),o(j.$$.fragment,e),o(X.$$.fragment,e),o(J.$$.fragment,e),o(ne.$$.fragment,e),o(se.$$.fragment,e),o(le.$$.fragment,e),o(pe.$$.fragment,e),ce=!0)},o(e){_(a.$$.fragment,e),_(p.$$.fragment,e),_(u.$$.fragment,e),_(C.$$.fragment,e),_(x.$$.fragment,e),_(A.$$.fragment,e),_(w.$$.fragment,e),_(U.$$.fragment,e),_(B.$$.fragment,e),_(D.$$.fragment,e),_(G.$$.fragment,e),_(z.$$.fragment,e),_(O.$$.fragment,e),_(j.$$.fragment,e),_(X.$$.fragment,e),_(J.$$.fragment,e),_(ne.$$.fragment,e),_(se.$$.fragment,e),_(le.$$.fragment,e),_(pe.$$.fragment,e),ce=!1},d(e){e&&d(t),h(a),h(p),h(u),h(C),h(x),h(A),h(w),h(U),h(B),h(D),h(G),h(z),h(O),h(j),h(X),h(J),h(ne),h(se),h(le),h(pe)}}}function Pa(e){let t,n,a,s;return t=new Ne({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[Ma]},$$scope:{ctx:e}}}),a=new Ne({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[Aa]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=v(),f(a.$$.fragment)},m(e,i){b(t,e,i),l(e,n,i),b(a,e,i),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const i={};1&n&&(i.$$scope={dirty:n,ctx:e}),a.$set(i)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),h(t,e),h(a,e)}}}function La(e){let t,n,a,s;return t=new ge({props:{label:"Plot"}}),a=new ge({props:{label:"Table"}}),{c(){f(t.$$.fragment),n=v(),f(a.$$.fragment)},m(e,i){b(t,e,i),l(e,n,i),b(a,e,i),s=!0},p:P,i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),h(t,e),h(a,e)}}}function Ta(e){let t;return{c(){t=x("This table presents a comprehensive summary of the top enriched pathways derived from the fgsea. Each row corresponds to a pathway, and the gene ranks are shown based on the ranking metric used in the analysis. The enrichment score, p-value, and adjusted p-value are also provided to assess the significance of the enrichment.")},m(e,n){l(e,t,n)},d(e){e&&d(t)}}}function wa(e){let t,n,a,s;return t=new ha({props:{$$slots:{default:[Ta]},$$scope:{ctx:e}}}),a=new kn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/gsea_table.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/gsea_table.pdf",width:1e3,height:600}}),{c(){f(t.$$.fragment),n=v(),f(a.$$.fragment)},m(e,i){b(t,e,i),l(e,n,i),b(a,e,i),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),h(t,e),h(a,e)}}}function Ra(e){let t;return{c(){t=x("This plot represents the GSEA results for a specified gene set, illustrating the distribution and impact of the gene set along the ranked list of genes. The running enrichment score curve shows the cumulative enrichment score as genes from the input list are encountered. Positive peaks on the curve indicate regions where members of the gene set are predominantly found.")},m(e,n){l(e,t,n)},d(e){e&&d(t)}}}function Ua(e){let t,n,a,s;return t=new ha({props:{$$slots:{default:[Ra]},$$scope:{ctx:e}}}),a=new ua({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/fgsea.txt",data:[{pathway:"Nitrogen metabolism",pval:1e-4,padj:.0023,log2err:.5384,ES:.7249,NES:2.2387,size:17,leadingEdge:"GLUL,GLUD1,CA5B,CA3,CA8,CA4,CA2",id:0},{pathway:"Ascorbate and aldarate metabolism",pval:.0096,padj:.0822,log2err:.3807,ES:-.6304,NES:-1.4453,size:27,leadingEdge:"ALDH9A1,ALDH3A2,UGDH,ALDH1B1",id:1},{pathway:"Fatty acid degradation",pval:.0123,padj:.0822,log2err:.3807,ES:-.5804,NES:-1.3856,size:44,leadingEdge:"ADH5,CPT1A,ACADVL,ALDH9A1,CPT2,HADHA,ACAA2,ACSL5,ACSL3,ACOX1,HADH,HADHB,ALDH3A2,ACADSB,ECHS1,ACSL6",id:2},{pathway:"Amino sugar and nucleotide sugar metabolism",pval:.0636,padj:.3179,log2err:.1767,ES:-.5163,NES:-1.2438,size:48,leadingEdge:"GNPDA2,HEXA,FPGT,MPI,HEXB,GALT,PGM2,PMM1,RENBP,CYB5R1,GNPNAT1,AMDHD2,GALK1,CMAS,GPI,FUK,UGDH,GNE,CYB5R3",id:3},{pathway:"Citrate cycle (TCA cycle)",pval:.0808,padj:.3231,log2err:.1563,ES:-.5535,NES:-1.2851,size:30,leadingEdge:"ACO1,DLST,DLAT,SUCLA2,SDHD,SDHC,FH,ACLY,SUCLG1,OGDH,MDH2,PCK2,CS,IDH3G,IDH3B,DLD,PDHA1,IDH1,IDH2,PDHB,MDH1,SDHA",id:4},{pathway:"Pentose phosphate pathway",pval:.0992,padj:.3306,log2err:.1396,ES:-.5426,NES:-1.2598,size:30,leadingEdge:"PGM2,PRPS1,ALDOA,PFKL,PRPS2,GPI,FBP1,TALDO1,RPE,GLYCTK,ALDOC,H6PD,DERA,PGM1,G6PD",id:5},{pathway:"Oxidative phosphorylation",pval:.1249,padj:.3568,log2err:.121,ES:-.4518,NES:-1.1459,size:120,leadingEdge:"NDUFV3,PPA2,ATP6V1B2,ATP6V1E2,NDUFA5,ATP6V1E1,NDUFA4,ATP5PB,ATP5PO,SDHD,ATP6V1A,ATP6V1G1,TCIRG1,NDUFS1,ATP6AP1,NDUFB10,SDHC,NDUFA11,COX6B1,UQCRC2,COX4I1,NDUFA9,LHPP,NDUFS4,COX15,ATP5F1A,COX11,COX7A2L,ATP6V1D,ATP6V0E1,NDUFB3,NDUFA7,ATP5F1B,NDUFS2,NDUFC2,ATP5MC2,UQCRC1,NDUFB7,NDUFA8,ATP5PF,UQCR11,ATP5MG,NDUFA13,NDUFB5,ATP6V0D1,UQCRB,NDUFS8,NDUFA1,ATP6V1C1,ATP5F1D,ATP6V0A1,ATP6V1H,ATP5F1C,NDUFS6,COX8A,SDHA,NDUFB11,ATP5MF,NDUFS3,NDUFB6,ATP5MC3,COX5B,COX5A,UQCR10,NDUFB1,NDUFB9",id:6},{pathway:"Propanoate metabolism",pval:.2313,padj:.4577,log2err:.0846,ES:-.497,NES:-1.1605,size:32,leadingEdge:"MLYCD,HADHA,SUCLA2,HIBCH,MUT,BCKDHA,PCCA,BCKDHB,ECHDC1,SUCLG1,ECHS1,MCEE,DBT,DLD,ACSS2,ACAT2,ACACB,LDHB",id:7},{pathway:"Fatty acid elongation",pval:.2342,padj:.4577,log2err:.0838,ES:-.5002,NES:-1.1613,size:30,leadingEdge:"HACD2,HSD17B12,HADHA,PPT1,ACAA2,ACOT2,ACOT4,HACD4,HADH,HADHB,HACD3,ECHS1",id:8},{pathway:"Pentose and glucuronate interconversions",pval:.24,padj:.4577,log2err:.0824,ES:-.4878,NES:-1.1432,size:34,leadingEdge:"AKR1B1,CRYL1,SORD,UGDH,RPE",id:9},{pathway:"Glycolysis / Gluconeogenesis",pval:.2518,padj:.4577,log2err:.0789,ES:-.4488,NES:-1.1085,size:68,leadingEdge:"ADH5,ALDH9A1,PGM2,ALDOA,ENO2,DLAT,PFKL,MINPP1,ALDH3A2,GPI,PKM,FBP1,ALDOC,HK1,PCK2,PGAM1,PGM1,BPGM,ALDH1B1,DLD,PDHA1,ACSS2,LDHB,G6PC3,PDHB,GAPDH,ENO3,PFKM",id:10},{pathway:"Pyruvate metabolism",pval:.3178,padj:.5296,log2err:.0677,ES:-.4608,NES:-1.095,size:39,leadingEdge:"ALDH9A1,GLO1,DLAT,FH,ME2,ALDH3A2,PKM,GRHPR,MDH2,PCK2,ALDH1B1,ACYP2,DLD,PDHA1,ACSS2,ACAT2,ACACB,LDHB,PDHB,MDH1",id:11},{pathway:"Starch and sucrose metabolism",pval:.4812,padj:.7025,log2err:.0482,ES:-.4354,NES:-1.0267,size:35,leadingEdge:"GAA,PGM2,PYGB,GYS1,AGL,GPI,GANC,GYG1,GBE1,HK1,PGM1",id:12},{pathway:"Glyoxylate and dicarboxylate metabolism",pval:.4918,padj:.7025,log2err:.0477,ES:-.4384,NES:-1.0104,size:28,leadingEdge:"CAT,ACO1,GCSH,HYI,MUT,PCCA,GRHPR,MCEE,MDH2,GLYCTK,CS,DLD,ACAT2",id:13},{pathway:"Galactose metabolism",pval:.5914,padj:.7462,log2err:.039,ES:-.411,NES:-.956,size:31,leadingEdge:"AKR1B1,GAA,GALT,PGM2,PFKL,GALK1,GANC,GLB1,HK1,PGM1",id:14},{pathway:"Fructose and mannose metabolism",pval:.5969,padj:.7462,log2err:.0385,ES:-.4081,NES:-.9561,size:33,leadingEdge:"AKR1B1,FPGT,MPI,ENOSF1,PMM1,ALDOA,PFKL,SORD,FUK,FBP1",id:15},{pathway:"Butanoate metabolism",pval:.6903,padj:.8122,log2err:.032,ES:-.3885,NES:-.8953,size:28,leadingEdge:"BDH2,HADHA,HMGCL,HADH,ACSM3,ECHS1,ALDH5A1,BDH1,ACAT2,ACADS,OXCT1",id:16},{pathway:"Inositol phosphate metabolism",pval:.8547,padj:.9497,log2err:.019,ES:-.3376,NES:-.8377,size:73,leadingEdge:"PIP4K2C,PLCB2,SACM1L,MTMR2,IPPK,PIK3CG,IMPA1,MTM1,INPP5B,MINPP1,MTMR1,MTMR8,PLCB3,PIP4K2B,PIKFYVE,PIK3C2A,PIK3CD,ISYNA1,MTMR6,PLCG1,PI4K2A,PLCH2,MTMR14,SYNJ1,CDIPT,INPP4A,PIK3C2B,MTMR4,PLCB1,INPP5K",id:17},{pathway:"Fatty acid biosynthesis",pval:.9718,padj:.9807,log2err:.0155,ES:-.2674,NES:-.5562,size:13,leadingEdge:"ACSL5,ACSL3,ACSL6",id:18},{pathway:"Sulfur metabolism",pval:.9807,padj:.9807,log2err:.0181,ES:-.2869,NES:-.5526,size:9,leadingEdge:"PAPSS1",id:19}]}}),{c(){f(t.$$.fragment),n=v(),f(a.$$.fragment)},m(e,i){b(t,e,i),l(e,n,i),b(a,e,i),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),h(t,e),h(a,e)}}}function Na(e){let t,n,a,s;return t=new fe({props:{$$slots:{default:[wa]},$$scope:{ctx:e}}}),a=new fe({props:{$$slots:{default:[Ua]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=v(),f(a.$$.fragment)},m(e,i){b(t,e,i),l(e,n,i),b(a,e,i),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const i={};1&n&&(i.$$scope={dirty:n,ctx:e}),a.$set(i)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),h(t,e),h(a,e)}}}function Ea(e){let t,n;return t=new ue({props:{$$slots:{content:[Na],default:[La]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function Ba(e){let t,n,a,s,p,c,m,u,g,C,y,$,x,M,A,L,T,w,R,U,N,E,B,I,D,V,H,G,F,z,k,q,O,K,j,W,Y,X,Q,J,Z,te,ne,ae,se,ie,re,le,oe,pe,ce;return a=new ha({props:{title:"Nitrogen metabolism",class:"pipen-report-table-image-descr"}}),p=new kn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/fgsea_Nitrogen-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/fgsea_Nitrogen-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),u=new ha({props:{title:"Ascorbate and aldarate metabolism",class:"pipen-report-table-image-descr"}}),C=new kn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/fgsea_Ascorbate-and-aldarate-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/fgsea_Ascorbate-and-aldarate-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),x=new ha({props:{title:"Fatty acid degradation",class:"pipen-report-table-image-descr"}}),A=new kn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/fgsea_Fatty-acid-degradation.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/fgsea_Fatty-acid-degradation.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),w=new ha({props:{title:"Amino sugar and nucleotide sugar metabolism",class:"pipen-report-table-image-descr"}}),U=new kn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/fgsea_Amino-sugar-and-nucleotide-sugar-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/fgsea_Amino-sugar-and-nucleotide-sugar-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),B=new ha({props:{title:"Citrate cycle (TCA cycle)",class:"pipen-report-table-image-descr"}}),D=new kn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/fgsea_Citrate-cycle-TCA-cycle-.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/fgsea_Citrate-cycle-TCA-cycle-.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),G=new ha({props:{title:"Pentose phosphate pathway",class:"pipen-report-table-image-descr"}}),z=new kn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/fgsea_Pentose-phosphate-pathway.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/fgsea_Pentose-phosphate-pathway.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),O=new ha({props:{title:"Oxidative phosphorylation",class:"pipen-report-table-image-descr"}}),j=new kn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/fgsea_Oxidative-phosphorylation.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/fgsea_Oxidative-phosphorylation.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),X=new ha({props:{title:"Propanoate metabolism",class:"pipen-report-table-image-descr"}}),J=new kn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/fgsea_Propanoate-metabolism.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/fgsea_Propanoate-metabolism.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new ha({props:{title:"Fatty acid elongation",class:"pipen-report-table-image-descr"}}),se=new kn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/fgsea_Fatty-acid-elongation.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/fgsea_Fatty-acid-elongation.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),le=new ha({props:{title:"Pentose and glucuronate interconversions",class:"pipen-report-table-image-descr"}}),pe=new kn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/fgsea_Pentose-and-glucuronate-interconversions.png",download:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/fgsea_Pentose-and-glucuronate-interconversions.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=i("div"),n=i("div"),f(a.$$.fragment),s=v(),f(p.$$.fragment),c=v(),m=i("div"),f(u.$$.fragment),g=v(),f(C.$$.fragment),y=v(),$=i("div"),f(x.$$.fragment),M=v(),f(A.$$.fragment),L=v(),T=i("div"),f(w.$$.fragment),R=v(),f(U.$$.fragment),N=v(),E=i("div"),f(B.$$.fragment),I=v(),f(D.$$.fragment),V=v(),H=i("div"),f(G.$$.fragment),F=v(),f(z.$$.fragment),k=v(),q=i("div"),f(O.$$.fragment),K=v(),f(j.$$.fragment),W=v(),Y=i("div"),f(X.$$.fragment),Q=v(),f(J.$$.fragment),Z=v(),te=i("div"),f(ne.$$.fragment),ae=v(),f(se.$$.fragment),ie=v(),re=i("div"),f(le.$$.fragment),oe=v(),f(pe.$$.fragment),r(t,"class","pipen-report-table-of-images"),ee(t,"grid-template-columns","repeat(2, auto)")},m(e,i){l(e,t,i),S(t,n),b(a,n,null),S(n,s),b(p,n,null),S(t,c),S(t,m),b(u,m,null),S(m,g),b(C,m,null),S(t,y),S(t,$),b(x,$,null),S($,M),b(A,$,null),S(t,L),S(t,T),b(w,T,null),S(T,R),b(U,T,null),S(t,N),S(t,E),b(B,E,null),S(E,I),b(D,E,null),S(t,V),S(t,H),b(G,H,null),S(H,F),b(z,H,null),S(t,k),S(t,q),b(O,q,null),S(q,K),b(j,q,null),S(t,W),S(t,Y),b(X,Y,null),S(Y,Q),b(J,Y,null),S(t,Z),S(t,te),b(ne,te,null),S(te,ae),b(se,te,null),S(t,ie),S(t,re),b(le,re,null),S(re,oe),b(pe,re,null),ce=!0},p:P,i(e){ce||(o(a.$$.fragment,e),o(p.$$.fragment,e),o(u.$$.fragment,e),o(C.$$.fragment,e),o(x.$$.fragment,e),o(A.$$.fragment,e),o(w.$$.fragment,e),o(U.$$.fragment,e),o(B.$$.fragment,e),o(D.$$.fragment,e),o(G.$$.fragment,e),o(z.$$.fragment,e),o(O.$$.fragment,e),o(j.$$.fragment,e),o(X.$$.fragment,e),o(J.$$.fragment,e),o(ne.$$.fragment,e),o(se.$$.fragment,e),o(le.$$.fragment,e),o(pe.$$.fragment,e),ce=!0)},o(e){_(a.$$.fragment,e),_(p.$$.fragment,e),_(u.$$.fragment,e),_(C.$$.fragment,e),_(x.$$.fragment,e),_(A.$$.fragment,e),_(w.$$.fragment,e),_(U.$$.fragment,e),_(B.$$.fragment,e),_(D.$$.fragment,e),_(G.$$.fragment,e),_(z.$$.fragment,e),_(O.$$.fragment,e),_(j.$$.fragment,e),_(X.$$.fragment,e),_(J.$$.fragment,e),_(ne.$$.fragment,e),_(se.$$.fragment,e),_(le.$$.fragment,e),_(pe.$$.fragment,e),ce=!1},d(e){e&&d(t),h(a),h(p),h(u),h(C),h(x),h(A),h(w),h(U),h(B),h(D),h(G),h(z),h(O),h(j),h(X),h(J),h(ne),h(se),h(le),h(pe)}}}function Ia(e){let t,n,a,s;return t=new Ne({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[Ea]},$$scope:{ctx:e}}}),a=new Ne({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[Ba]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=v(),f(a.$$.fragment)},m(e,i){b(t,e,i),l(e,n,i),b(a,e,i),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const i={};1&n&&(i.$$scope={dirty:n,ctx:e}),a.$set(i)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),h(t,e),h(a,e)}}}function Da(e){let t,n,a,s,p,c,m,u,g,C,y,$,x,S,M,A;return m=new Ue({props:{align:"start",$$slots:{default:[Pa]},$$scope:{ctx:e}}}),M=new Ue({props:{align:"start",$$slots:{default:[Ia]},$$scope:{ctx:e}}}),{c(){t=i("h1"),t.textContent="Cluster_Type 1 helper T cells",n=i("a"),n.innerHTML="",a=v(),s=i("h2"),s.textContent="Colitis_vs_Control",p=i("a"),p.innerHTML="",c=v(),f(m.$$.fragment),u=v(),g=i("h1"),g.textContent="Cluster_Type 1 helper T cells.1",C=i("a"),C.innerHTML="",y=v(),$=i("h2"),$.textContent="Colitis_vs_Control",x=i("a"),x.innerHTML="",S=v(),f(M.$$.fragment),r(n,"id","prt-h1-10-cluster-type-1-helper-t-cells"),r(n,"class","pipen-report-toc-anchor"),r(p,"id","prt-h2-10-colitis-vs-control"),r(p,"class","pipen-report-toc-anchor"),r(C,"id","prt-h1-11-cluster-type-1-helper-t-cells-1"),r(C,"class","pipen-report-toc-anchor"),r(x,"id","prt-h2-11-colitis-vs-control"),r(x,"class","pipen-report-toc-anchor")},m(e,i){l(e,t,i),l(e,n,i),l(e,a,i),l(e,s,i),l(e,p,i),l(e,c,i),b(m,e,i),l(e,u,i),l(e,g,i),l(e,C,i),l(e,y,i),l(e,$,i),l(e,x,i),l(e,S,i),b(M,e,i),A=!0},p(e,[t]){const n={};1&t&&(n.$$scope={dirty:t,ctx:e}),m.$set(n);const a={};1&t&&(a.$$scope={dirty:t,ctx:e}),M.$set(a)},i(e){A||(o(m.$$.fragment,e),o(M.$$.fragment,e),A=!0)},o(e){_(m.$$.fragment,e),_(M.$$.fragment,e),A=!1},d(e){e&&(d(t),d(n),d(a),d(s),d(p),d(c),d(u),d(g),d(C),d(y),d($),d(x),d(S)),h(m,e),h(M,e)}}}class Va extends t{constructor(e){super(),n(this,e,null,Da,a,{})}}var Ha=[{slug:"prt-h1-0-cluster-cycling-t-cells",text:"Cluster_Cycling T cells",children:[{slug:"prt-h2-0-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:0}],page:0},{slug:"prt-h1-1-cluster-follicular-helper-t-cells",text:"Cluster_Follicular helper T cells",children:[{slug:"prt-h2-1-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:0}],page:0},{slug:"prt-h1-2-cluster-gamma-delta-t-cells",text:"Cluster_gamma-delta T cells",children:[{slug:"prt-h2-2-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:0}],page:0},{slug:"prt-h1-3-cluster-plasmablasts",text:"Cluster_Plasmablasts",children:[{slug:"prt-h2-3-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:0}],page:0},{slug:"prt-h1-4-cluster-regulatory-t-cells",text:"Cluster_Regulatory T cells",children:[{slug:"prt-h2-4-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:0}],page:0},{slug:"prt-h1-5-cluster-regulatory-t-cells-1",text:"Cluster_Regulatory T cells.1",children:[{slug:"prt-h2-5-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:1}],page:1},{slug:"prt-h1-6-cluster-tcm-naive-helper-t-cells",text:"Cluster_Tcm/Naive helper T cells",children:[{slug:"prt-h2-6-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:1}],page:1},{slug:"prt-h1-7-cluster-tem-trm-cytotoxic-t-cells",text:"Cluster_Tem/Trm cytotoxic T cells",children:[{slug:"prt-h2-7-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:1}],page:1},{slug:"prt-h1-8-cluster-tem-trm-cytotoxic-t-cells-1",text:"Cluster_Tem/Trm cytotoxic T cells.1",children:[{slug:"prt-h2-8-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:1}],page:1},{slug:"prt-h1-9-cluster-trm-cytotoxic-t-cells",text:"Cluster_Trm cytotoxic T cells",children:[{slug:"prt-h2-9-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:1}],page:1},{slug:"prt-h1-10-cluster-type-1-helper-t-cells",text:"Cluster_Type 1 helper T cells",children:[{slug:"prt-h2-10-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:2}],page:2},{slug:"prt-h1-11-cluster-type-1-helper-t-cells-1",text:"Cluster_Type 1 helper T cells.1",children:[{slug:"prt-h2-11-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:2}],page:2}],Ga={pipeline:{name:"Immunopipe",desc:"Immunopipe (v1.4.3): Integrative analysis for scRNA-seq and scTCR-seq data"},versions:'\n    <a href="https://github.com/pwwang/pipen" target="_blank">\n        pipen v0.15.7\n    </a> and\n    <a href="https://github.com/pwwang/pipen-report" target="_blank">\n        pipen-report v0.21.4\n    </a>\n',entries:[{name:"SampleInfo",desc:"List sample information and perform statistics",npages:1,report_toc:!0,order:0,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n[1] ggrepel_0.9.6    ggprism_1.0.5    ggplot2_3.5.1    tidyr_1.3.1     \n[5] dplyr_1.1.4      tidyselect_1.2.1 jsonlite_1.8.9   rlang_1.1.5     \n[9] logger_0.4.0    \n\nloaded via a namespace (and not attached):\n [1] vctrs_0.6.5      cli_3.6.4        purrr_1.0.2      generics_0.1.3  \n [5] glue_1.8.0       labeling_0.4.3   colorspace_2.1-1 scales_1.3.0    \n [9] grid_4.3.3       munsell_0.5.1    tibble_3.2.1     lifecycle_1.0.4 \n[13] compiler_4.3.3   Rcpp_1.0.14      pkgconfig_2.0.3  farver_2.1.2    \n[17] digest_0.6.37    R6_2.6.0         pillar_1.10.1    magrittr_2.0.3  \n[21] withr_3.0.2      gtable_0.3.6    \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 2.19\nuser 11.15\nsys 0.11\npercent 513%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3421.613\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"SeuratPreparing",desc:"Load, prepare and apply QC to data, using `Seurat`",npages:1,report_toc:!0,order:2,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats4    stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] scDblFinder_1.16.0          SingleCellExperiment_1.24.0\n [3] SummarizedExperiment_1.32.0 Biobase_2.62.0             \n [5] GenomicRanges_1.54.1        GenomeInfoDb_1.38.1        \n [7] IRanges_2.36.0              S4Vectors_0.40.2           \n [9] BiocGenerics_0.48.1         MatrixGenerics_1.14.0      \n[11] matrixStats_1.1.0           dplyr_1.1.4                \n[13] ggplot2_3.5.1               bracer_1.2.1               \n[15] future_1.34.0               Seurat_5.1.0               \n[17] SeuratObject_5.0.2          sp_2.2-0                   \n[19] digest_0.6.37               jsonlite_1.8.9             \n[21] rlang_1.1.5                 logger_0.4.0               \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22          splines_4.3.3            \n  [3] later_1.4.1               BiocIO_1.12.0            \n  [5] bitops_1.0-9              tibble_3.2.1             \n  [7] R.oo_1.27.0               polyclip_1.10-7          \n  [9] XML_3.99-0.17             fastDummies_1.7.5        \n [11] lifecycle_1.0.4           edgeR_4.0.16             \n [13] globals_0.16.3            lattice_0.22-6           \n [15] MASS_7.3-60.0.1           magrittr_2.0.3           \n [17] limma_3.58.1              plotly_4.10.4            \n [19] yaml_2.3.10               metapod_1.10.0           \n [21] httpuv_1.6.15             sctransform_0.4.1        \n [23] spam_2.11-1               spatstat.sparse_3.1-0    \n [25] reticulate_1.40.0         cowplot_1.1.3            \n [27] pbapply_1.7-2             RColorBrewer_1.1-3       \n [29] abind_1.4-5               zlibbioc_1.48.0          \n [31] Rtsne_0.17                purrr_1.0.2              \n [33] R.utils_2.12.3            RCurl_1.98-1.16          \n [35] GenomeInfoDbData_1.2.11   ggrepel_0.9.6            \n [37] irlba_2.3.5.1             listenv_0.9.1            \n [39] spatstat.utils_3.1-2      goftest_1.2-3            \n [41] RSpectra_0.16-2           dqrng_0.3.2              \n [43] spatstat.random_3.3-2     fitdistrplus_1.2-2       \n [45] parallelly_1.42.0         DelayedMatrixStats_1.24.0\n [47] leiden_0.4.3.1            codetools_0.2-20         \n [49] DelayedArray_0.28.0       scuttle_1.12.0           \n [51] tidyselect_1.2.1          farver_2.1.2             \n [53] viridis_0.6.5             ScaledMatrix_1.10.0      \n [55] spatstat.explore_3.3-4    GenomicAlignments_1.38.0 \n [57] BiocNeighbors_1.20.0      progressr_0.15.1         \n [59] scater_1.30.1             ggridges_0.5.6           \n [61] survival_3.8-3            systemfonts_1.2.1        \n [63] tools_4.3.3               ragg_1.3.3               \n [65] ica_1.0-3                 Rcpp_1.0.14              \n [67] glue_1.8.0                gridExtra_2.3            \n [69] SparseArray_1.2.2         withr_3.0.2              \n [71] fastmap_1.2.0             bluster_1.12.0           \n [73] rsvd_1.0.5                R6_2.6.0                 \n [75] mime_0.12                 textshaping_0.4.0        \n [77] colorspace_2.1-1          scattermore_1.2          \n [79] tensor_1.5                spatstat.data_3.1-4      \n [81] R.methodsS3_1.8.2         RhpcBLASctl_0.23-42      \n [83] tidyr_1.3.1               generics_0.1.3           \n [85] data.table_1.16.4         rtracklayer_1.62.0       \n [87] httr_1.4.7                htmlwidgets_1.6.4        \n [89] S4Arrays_1.2.0            uwot_0.2.2               \n [91] pkgconfig_2.0.3           gtable_0.3.6             \n [93] lmtest_0.9-40             XVector_0.42.0           \n [95] htmltools_0.5.8.1         dotCall64_1.2            \n [97] scales_1.3.0              png_0.1-8                \n [99] harmony_1.2.0             spatstat.univar_3.1-1    \n[101] scran_1.30.0              reshape2_1.4.4           \n[103] rjson_0.2.23              nlme_3.1-167             \n[105] zoo_1.8-12                stringr_1.5.1            \n[107] KernSmooth_2.23-26        vipor_0.4.7              \n[109] parallel_4.3.3            miniUI_0.1.1.1           \n[111] restfulr_0.0.15           pillar_1.10.1            \n[113] grid_4.3.3                vctrs_0.6.5              \n[115] RANN_2.6.2                promises_1.3.2           \n[117] BiocSingular_1.18.0       beachmat_2.18.0          \n[119] xtable_1.8-4              cluster_2.1.8            \n[121] beeswarm_0.4.0            locfit_1.5-9.11          \n[123] cli_3.6.4                 compiler_4.3.3           \n[125] Rsamtools_2.18.0          crayon_1.5.3             \n[127] future.apply_1.11.3       labeling_0.4.3           \n[129] ggbeeswarm_0.7.2          plyr_1.8.9               \n[131] stringi_1.8.4             viridisLite_0.4.2        \n[133] deldir_2.0-4              BiocParallel_1.36.0      \n[135] munsell_0.5.1             Biostrings_2.70.1        \n[137] lazyeval_0.2.2            spatstat.geom_3.3-5      \n[139] Matrix_1.6-5              RcppHNSW_0.6.0           \n[141] patchwork_1.3.0           sparseMatrixStats_1.14.0 \n[143] statmod_1.5.0             shiny_1.10.0             \n[145] ROCR_1.0-11               igraph_2.0.3             \n[147] xgboost_2.1.4.1          \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 209.65\nuser 1454.55\nsys 7.57\npercent 697%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ClusterMarkersOfAllCells",desc:"Markers for clusters of all cells.",npages:3,report_toc:!0,order:4,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] UpSetR_1.4.0        ggVennDiagram_1.2.3 tidyseurat_0.8.0   \n [4] ttservice_0.4.1     future_1.34.0       ggrepel_0.9.6      \n [7] ggprism_1.0.5       ggplot2_3.5.1       enrichR_3.4        \n[10] Seurat_5.1.0        SeuratObject_5.0.2  sp_2.2-0           \n[13] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[16] tidyselect_1.2.1    digest_0.6.37       jsonlite_1.8.9     \n[19] rlang_1.1.5         logger_0.4.0       \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] ellipsis_0.3.2         labeling_0.4.3         promises_1.3.2        \n [19] purrr_1.0.2            WriteXLS_6.7.0         RVenn_1.1.0           \n [22] goftest_1.2-3          later_1.4.1            spatstat.utils_3.1-2  \n [25] irlba_2.3.5.1          parallel_4.3.3         cluster_2.1.8         \n [28] R6_2.6.0               ica_1.0-3              stringi_1.8.4         \n [31] RColorBrewer_1.1-3     spatstat.data_3.1-4    limma_3.58.1          \n [34] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [37] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [40] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [43] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [46] splines_4.3.3          igraph_2.0.3           abind_1.4-5           \n [49] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.3-4 curl_6.0.1             listenv_0.9.1         \n [55] lattice_0.22-6         plyr_1.8.9             withr_3.0.2           \n [58] shiny_1.10.0           ROCR_1.0-11            Rtsne_0.17            \n [61] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n [64] fitdistrplus_1.2-2     pillar_1.10.1          KernSmooth_2.23-26    \n [67] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n [82] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n [85] nlme_3.1-167           patchwork_1.3.0        presto_1.0.0          \n [88] cli_3.6.4              spatstat.sparse_3.1-0  fansi_1.0.6           \n [91] spam_2.11-1            viridisLite_0.4.2      uwot_0.2.2            \n [94] gtable_0.3.6           progressr_0.15.1       rjson_0.2.23          \n [97] htmlwidgets_1.6.4      farver_2.1.2           htmltools_0.5.8.1     \n[100] lifecycle_1.0.4        httr_1.4.7             statmod_1.5.0         \n[103] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 259.76\nuser 186.00\nsys 8.62\npercent 74%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TopExpressingGenesOfAllCells",desc:"Top expressing genes for clusters of all cells.",npages:3,report_toc:!0,order:5,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ggprism_1.0.5      dplyr_1.1.4        enrichR_3.4        tibble_3.2.1      \n [5] Seurat_5.1.0       SeuratObject_5.0.2 sp_2.2-0           jsonlite_1.8.9    \n [9] rlang_1.1.5        logger_0.4.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] labeling_0.4.3         promises_1.3.2         purrr_1.0.2           \n [19] WriteXLS_6.7.0         goftest_1.2-3          later_1.4.1           \n [22] spatstat.utils_3.1-2   irlba_2.3.5.1          parallel_4.3.3        \n [25] cluster_2.1.8          R6_2.6.0               ica_1.0-3             \n [28] stringi_1.8.4          RColorBrewer_1.1-3     spatstat.data_3.1-4   \n [31] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [37] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [43] splines_4.3.3          igraph_2.0.3           tidyselect_1.2.1      \n [46] abind_1.4-5            spatstat.random_3.3-2  codetools_0.2-20      \n [49] miniUI_0.1.1.1         spatstat.explore_3.3-4 curl_6.0.1            \n [52] listenv_0.9.1          lattice_0.22-6         plyr_1.8.9            \n [55] withr_3.0.2            shiny_1.10.0           ROCR_1.0-11           \n [58] Rtsne_0.17             future_1.34.0          fastDummies_1.7.5     \n [61] survival_3.8-3         polyclip_1.10-7        fitdistrplus_1.2-2    \n [64] pillar_1.10.1          KernSmooth_2.23-26     plotly_4.10.4         \n [67] generics_0.1.3         RcppHNSW_0.6.0         ggplot2_3.5.1         \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n [82] cowplot_1.1.3          grid_4.3.3             tidyr_1.3.1           \n [85] colorspace_2.1-1       nlme_3.1-167           patchwork_1.3.0       \n [88] cli_3.6.4              spatstat.sparse_3.1-0  spam_2.11-1           \n [91] viridisLite_0.4.2      uwot_0.2.2             gtable_0.3.6          \n [94] digest_0.6.37          progressr_0.15.1       ggrepel_0.9.6         \n [97] rjson_0.2.23           htmlwidgets_1.6.4      farver_2.1.2          \n[100] htmltools_0.5.8.1      lifecycle_1.0.4        httr_1.4.7            \n[103] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 73.70\nuser 41.45\nsys 2.53\npercent 59%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TCellSelection",desc:"Separate T and non-T cells and select T cells.",npages:1,report_toc:!0,order:6,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] glue_1.8.0         factoextra_1.0.7   patchwork_1.3.0    ggpubr_0.6.0      \n [5] ggrepel_0.9.6      ggprism_1.0.5      ggplot2_3.5.1      tibble_3.2.1      \n [9] tidyr_1.3.1        dplyr_1.1.4        Seurat_5.1.0       SeuratObject_5.0.2\n[13] sp_2.2-0           jsonlite_1.8.9     rlang_1.1.5        logger_0.4.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] backports_1.5.0        labeling_0.4.3         promises_1.3.2        \n [19] purrr_1.0.2            goftest_1.2-3          later_1.4.1           \n [22] spatstat.utils_3.1-2   broom_1.0.7            irlba_2.3.5.1         \n [25] parallel_4.3.3         cluster_2.1.8          R6_2.6.0              \n [28] ica_1.0-3              stringi_1.8.4          RColorBrewer_1.1-3    \n [31] spatstat.data_3.1-4    reticulate_1.40.0      car_3.1-3             \n [34] parallelly_1.42.0      spatstat.univar_3.1-1  lmtest_0.9-40         \n [37] scattermore_1.2        Rcpp_1.0.14            tensor_1.5            \n [40] future.apply_1.11.3    zoo_1.8-12             sctransform_0.4.1     \n [43] httpuv_1.6.15          Matrix_1.6-5           splines_4.3.3         \n [46] igraph_2.0.3           tidyselect_1.2.1       abind_1.4-5           \n [49] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.3-4 listenv_0.9.1          lattice_0.22-6        \n [55] plyr_1.8.9             withr_3.0.2            shiny_1.10.0          \n [58] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n [61] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n [64] fitdistrplus_1.2-2     pillar_1.10.1          carData_3.0-5         \n [67] KernSmooth_2.23-26     plotly_4.10.4          generics_0.1.3        \n [70] RcppHNSW_0.6.0         munsell_0.5.1          scales_1.3.0          \n [73] globals_0.16.3         xtable_1.8-4           lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] ggsignif_0.6.4         RANN_2.6.2             leiden_0.4.3.1        \n [82] dotCall64_1.2          cowplot_1.1.3          grid_4.3.3            \n [85] colorspace_2.1-1       nlme_3.1-167           Formula_1.2-5         \n [88] cli_3.6.4              spatstat.sparse_3.1-0  spam_2.11-1           \n [91] viridisLite_0.4.2      uwot_0.2.2             gtable_0.3.6          \n [94] rstatix_0.7.2          digest_0.6.37          progressr_0.15.1      \n [97] htmlwidgets_1.6.4      farver_2.1.2           htmltools_0.5.8.1     \n[100] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[103] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 41.63\nuser 49.85\nsys 0.77\npercent 121%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ClusterMarkers",desc:"Markers for clusters of all or selected T cells.",npages:2,report_toc:!0,order:10,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] UpSetR_1.4.0        ggVennDiagram_1.2.3 tidyseurat_0.8.0   \n [4] ttservice_0.4.1     future_1.34.0       ggrepel_0.9.6      \n [7] ggprism_1.0.5       ggplot2_3.5.1       enrichR_3.4        \n[10] Seurat_5.1.0        SeuratObject_5.0.2  sp_2.2-0           \n[13] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[16] tidyselect_1.2.1    digest_0.6.37       jsonlite_1.8.9     \n[19] rlang_1.1.5         logger_0.4.0       \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] ellipsis_0.3.2         labeling_0.4.3         promises_1.3.2        \n [19] purrr_1.0.2            WriteXLS_6.7.0         RVenn_1.1.0           \n [22] goftest_1.2-3          later_1.4.1            spatstat.utils_3.1-2  \n [25] irlba_2.3.5.1          parallel_4.3.3         cluster_2.1.8         \n [28] R6_2.6.0               ica_1.0-3              stringi_1.8.4         \n [31] RColorBrewer_1.1-3     spatstat.data_3.1-4    limma_3.58.1          \n [34] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [37] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [40] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [43] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [46] splines_4.3.3          igraph_2.0.3           abind_1.4-5           \n [49] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.3-4 curl_6.0.1             listenv_0.9.1         \n [55] lattice_0.22-6         plyr_1.8.9             withr_3.0.2           \n [58] shiny_1.10.0           ROCR_1.0-11            Rtsne_0.17            \n [61] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n [64] fitdistrplus_1.2-2     pillar_1.10.1          KernSmooth_2.23-26    \n [67] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n [82] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n [85] nlme_3.1-167           patchwork_1.3.0        presto_1.0.0          \n [88] cli_3.6.4              spatstat.sparse_3.1-0  fansi_1.0.6           \n [91] spam_2.11-1            viridisLite_0.4.2      uwot_0.2.2            \n [94] gtable_0.3.6           progressr_0.15.1       rjson_0.2.23          \n [97] htmlwidgets_1.6.4      farver_2.1.2           htmltools_0.5.8.1     \n[100] lifecycle_1.0.4        httr_1.4.7             statmod_1.5.0         \n[103] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 162.57\nuser 119.15\nsys 5.40\npercent 76%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TopExpressingGenes",desc:"Top expressing genes for clusters of all or selected T cells.",npages:2,report_toc:!0,order:11,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ggprism_1.0.5      dplyr_1.1.4        enrichR_3.4        tibble_3.2.1      \n [5] Seurat_5.1.0       SeuratObject_5.0.2 sp_2.2-0           jsonlite_1.8.9    \n [9] rlang_1.1.5        logger_0.4.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] labeling_0.4.3         promises_1.3.2         purrr_1.0.2           \n [19] WriteXLS_6.7.0         goftest_1.2-3          later_1.4.1           \n [22] spatstat.utils_3.1-2   irlba_2.3.5.1          parallel_4.3.3        \n [25] cluster_2.1.8          R6_2.6.0               ica_1.0-3             \n [28] stringi_1.8.4          RColorBrewer_1.1-3     spatstat.data_3.1-4   \n [31] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [37] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [43] splines_4.3.3          igraph_2.0.3           tidyselect_1.2.1      \n [46] abind_1.4-5            spatstat.random_3.3-2  codetools_0.2-20      \n [49] miniUI_0.1.1.1         spatstat.explore_3.3-4 curl_6.0.1            \n [52] listenv_0.9.1          lattice_0.22-6         plyr_1.8.9            \n [55] withr_3.0.2            shiny_1.10.0           ROCR_1.0-11           \n [58] Rtsne_0.17             future_1.34.0          fastDummies_1.7.5     \n [61] survival_3.8-3         polyclip_1.10-7        fitdistrplus_1.2-2    \n [64] pillar_1.10.1          KernSmooth_2.23-26     plotly_4.10.4         \n [67] generics_0.1.3         RcppHNSW_0.6.0         ggplot2_3.5.1         \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n [82] cowplot_1.1.3          grid_4.3.3             tidyr_1.3.1           \n [85] colorspace_2.1-1       nlme_3.1-167           patchwork_1.3.0       \n [88] cli_3.6.4              spatstat.sparse_3.1-0  spam_2.11-1           \n [91] viridisLite_0.4.2      uwot_0.2.2             gtable_0.3.6          \n [94] digest_0.6.37          progressr_0.15.1       ggrepel_0.9.6         \n [97] rjson_0.2.23           htmlwidgets_1.6.4      farver_2.1.2          \n[100] htmltools_0.5.8.1      lifecycle_1.0.4        httr_1.4.7            \n[103] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 45.21\nuser 28.74\nsys 1.92\npercent 67%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TCRClusterStats",desc:"Statistics of TCR clusters, generated by `TCRClustering`.",npages:1,report_toc:!0,order:13,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] grid      stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] ComplexHeatmap_2.18.0 ggprism_1.0.5         immunarch_0.9.1      \n [4] patchwork_1.3.0       data.table_1.16.4     dtplyr_1.3.1         \n [7] dplyr_1.1.4           tidyr_1.3.1           tibble_3.2.1         \n[10] ggplot2_3.5.1         jsonlite_1.8.9        rlang_1.1.5          \n[13] logger_0.4.0         \n\nloaded via a namespace (and not attached):\n  [1] gridExtra_2.3       phangorn_2.12.1     readxl_1.4.3       \n  [4] magrittr_2.0.3      clue_0.3-66         GetoptLong_1.0.5   \n  [7] matrixStats_1.1.0   compiler_4.3.3      flexmix_2.3-19     \n [10] png_0.1-8           vctrs_0.6.5         reshape2_1.4.4     \n [13] ggalluvial_0.12.5   quadprog_1.5-8      stringr_1.5.1      \n [16] crayon_1.5.3        pkgconfig_2.0.3     shape_1.4.6.1      \n [19] fastmap_1.2.0       backports_1.5.0     labeling_0.4.3     \n [22] ggraph_2.2.1        promises_1.3.2      tzdb_0.4.0         \n [25] UpSetR_1.4.0        purrr_1.0.2         modeltools_0.2-23  \n [28] ggseqlogo_0.2       cachem_1.1.0        later_1.4.1        \n [31] uuid_1.2-1          fpc_2.2-13          tweenr_2.0.3       \n [34] broom_1.0.7         parallel_4.3.3      prabclus_2.3-4     \n [37] cluster_2.1.8       R6_2.6.0            stringi_1.8.4      \n [40] RColorBrewer_1.1-3  rlist_0.4.6.2       car_3.1-3          \n [43] diptest_0.77-1      cellranger_1.1.0    Rcpp_1.0.14        \n [46] iterators_1.0.14    IRanges_2.36.0      readr_2.1.5        \n [49] httpuv_1.6.15       Matrix_1.6-5        nnet_7.3-20        \n [52] igraph_2.0.3        tidyselect_1.2.1    stringdist_0.9.15  \n [55] abind_1.4-5         viridis_0.6.5       doParallel_1.0.17  \n [58] codetools_0.2-20    lattice_0.22-6      plyr_1.8.9         \n [61] shiny_1.10.0        withr_3.0.2         polyclip_1.10-7    \n [64] circlize_0.4.16     mclust_6.1.1        kernlab_0.9-33     \n [67] pillar_1.10.1       ggpubr_0.6.0        carData_3.0-5      \n [70] foreach_1.5.2       stats4_4.3.3        generics_0.1.3     \n [73] S4Vectors_0.40.2    hms_1.1.3           munsell_0.5.1      \n [76] scales_1.3.0        xtable_1.8-4        class_7.3-23       \n [79] glue_1.8.0          pheatmap_1.0.12     tools_4.3.3        \n [82] robustbase_0.99-4-1 ggsignif_0.6.4      graphlayouts_1.2.2 \n [85] Cairo_1.6-2         fastmatch_1.1-6     tidygraph_1.3.0    \n [88] ape_5.8-1           colorspace_2.1-1    nlme_3.1-167       \n [91] ggforce_0.4.2       Formula_1.2-5       cli_3.6.4          \n [94] viridisLite_0.4.2   gtable_0.3.6        DEoptimR_1.1-3-1   \n [97] rstatix_0.7.2       digest_0.6.37       BiocGenerics_0.48.1\n[100] ggrepel_0.9.6       rjson_0.2.23        farver_2.1.2       \n[103] memoise_2.0.1       htmltools_0.5.8.1   lifecycle_1.0.4    \n[106] factoextra_1.0.7    mime_0.12           GlobalOptions_0.1.2\n[109] shinythemes_1.2.0   MASS_7.3-60.0.1    \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 7.17\nuser 15.93\nsys 0.32\npercent 226%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3287.293\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TESSA",desc:"Tessa is a Bayesian model to integrate T cell receptor (TCR) sequence profiling with transcriptomes of T cells.",npages:1,report_toc:!0,order:14,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] igraph_2.0.3         Rtsne_0.17           LaplacesDemon_16.1.6\n [4] MASS_7.3-60.0.1      ggprism_1.0.5        Seurat_5.1.0        \n [7] SeuratObject_5.0.2   sp_2.2-0             tibble_3.2.1        \n[10] glue_1.8.0           immunarch_0.9.1      patchwork_1.3.0     \n[13] data.table_1.16.4    dtplyr_1.3.1         ggplot2_3.5.1       \n[16] tidyr_1.3.1          dplyr_1.1.4          jsonlite_1.8.9      \n[19] rlang_1.1.5          logger_0.4.0        \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22       shinythemes_1.2.0      splines_4.3.3         \n  [4] later_1.4.1            cellranger_1.1.0       polyclip_1.10-7       \n  [7] fastDummies_1.7.5      factoextra_1.0.7       lifecycle_1.0.4       \n [10] rstatix_0.7.2          doParallel_1.0.17      globals_0.16.3        \n [13] lattice_0.22-6         prabclus_2.3-4         backports_1.5.0       \n [16] magrittr_2.0.3         plotly_4.10.4          rlist_0.4.6.2         \n [19] httpuv_1.6.15          sctransform_0.4.1      spam_2.11-1           \n [22] flexmix_2.3-19         spatstat.sparse_3.1-0  reticulate_1.40.0     \n [25] cowplot_1.1.3          pbapply_1.7-2          RColorBrewer_1.1-3    \n [28] abind_1.4-5            quadprog_1.5-8         purrr_1.0.2           \n [31] ggraph_2.2.1           nnet_7.3-20            tweenr_2.0.3          \n [34] circlize_0.4.16        ggrepel_0.9.6          irlba_2.3.5.1         \n [37] listenv_0.9.1          spatstat.utils_3.1-2   pheatmap_1.0.12       \n [40] goftest_1.2-3          RSpectra_0.16-2        spatstat.random_3.3-2 \n [43] fitdistrplus_1.2-2     parallelly_1.42.0      leiden_0.4.3.1        \n [46] codetools_0.2-20       ggforce_0.4.2          tidyselect_1.2.1      \n [49] shape_1.4.6.1          ggseqlogo_0.2          farver_2.1.2          \n [52] viridis_0.6.5          matrixStats_1.1.0      stats4_4.3.3          \n [55] spatstat.explore_3.3-4 tidygraph_1.3.0        progressr_0.15.1      \n [58] Formula_1.2-5          ggridges_0.5.6         ggalluvial_0.12.5     \n [61] survival_3.8-3         iterators_1.0.14       foreach_1.5.2         \n [64] tools_4.3.3            stringdist_0.9.15      ica_1.0-3             \n [67] Rcpp_1.0.14            gridExtra_2.3          withr_3.0.2           \n [70] fastmap_1.2.0          digest_0.6.37          R6_2.6.0              \n [73] mime_0.12              colorspace_2.1-1       scattermore_1.2       \n [76] tensor_1.5             spatstat.data_3.1-4    diptest_0.77-1        \n [79] UpSetR_1.4.0           generics_0.1.3         robustbase_0.99-4-1   \n [82] class_7.3-23           graphlayouts_1.2.2     httr_1.4.7            \n [85] htmlwidgets_1.6.4      uwot_0.2.2             pkgconfig_2.0.3       \n [88] gtable_0.3.6           modeltools_0.2-23      lmtest_0.9-40         \n [91] htmltools_0.5.8.1      carData_3.0-5          dotCall64_1.2         \n [94] scales_1.3.0           png_0.1-8              spatstat.univar_3.1-1 \n [97] tzdb_0.4.0             reshape2_1.4.4         uuid_1.2-1            \n[100] nlme_3.1-167           cachem_1.1.0           zoo_1.8-12            \n[103] GlobalOptions_0.1.2    stringr_1.5.1          KernSmooth_2.23-26    \n[106] parallel_4.3.3         miniUI_0.1.1.1         pillar_1.10.1         \n[109] grid_4.3.3             vctrs_0.6.5            RANN_2.6.2            \n[112] promises_1.3.2         ggpubr_0.6.0           car_3.1-3             \n[115] xtable_1.8-4           cluster_2.1.8          readr_2.1.5           \n[118] cli_3.6.4              compiler_4.3.3         future.apply_1.11.3   \n[121] ggsignif_0.6.4         labeling_0.4.3         mclust_6.1.1          \n[124] plyr_1.8.9             stringi_1.8.4          deldir_2.0-4          \n[127] viridisLite_0.4.2      munsell_0.5.1          lazyeval_0.2.2        \n[130] spatstat.geom_3.3-5    Matrix_1.6-5           RcppHNSW_0.6.0        \n[133] hms_1.1.3              future_1.34.0          fpc_2.2-13            \n[136] shiny_1.10.0           kernlab_0.9-33         ROCR_1.0-11           \n[139] broom_1.0.7            memoise_2.0.1          phangorn_2.12.1       \n[142] fastmatch_1.1-6        DEoptimR_1.1-3-1       readxl_1.4.3          \n[145] ape_5.8-1             \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 1679.73\nuser 2269.05\nsys 21.01\npercent 136%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"SeuratClusterStats",desc:"Statistics of the clustering.",npages:1,report_toc:!0,order:17,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] grid      stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] clustree_0.5.1        ggraph_2.2.1          ComplexHeatmap_2.18.0\n [4] circlize_0.4.16       tidyseurat_0.8.0      ttservice_0.4.1      \n [7] ggrepel_0.9.6         ggprism_1.0.5         tibble_3.2.1         \n[10] Seurat_5.1.0          SeuratObject_5.0.2    sp_2.2-0             \n[13] ggplot2_3.5.1         tidyr_1.3.1           dplyr_1.1.4          \n[16] tidyselect_1.2.1      jsonlite_1.8.9        rlang_1.1.5          \n[19] logger_0.4.0         \n\nloaded via a namespace (and not attached):\n  [1] RColorBrewer_1.1-3     shape_1.4.6.1          magrittr_2.0.3        \n  [4] ggbeeswarm_0.7.2       spatstat.utils_3.1-2   farver_2.1.2          \n  [7] GlobalOptions_0.1.2    vctrs_0.6.5            ROCR_1.0-11           \n [10] memoise_2.0.1          spatstat.explore_3.3-4 htmltools_0.5.8.1     \n [13] sctransform_0.4.1      parallelly_1.42.0      KernSmooth_2.23-26    \n [16] htmlwidgets_1.6.4      ica_1.0-3              plyr_1.8.9            \n [19] cachem_1.1.0           plotly_4.10.4          zoo_1.8-12            \n [22] igraph_2.0.3           mime_0.12              lifecycle_1.0.4       \n [25] iterators_1.0.14       pkgconfig_2.0.3        Matrix_1.6-5          \n [28] R6_2.6.0               fastmap_1.2.0          clue_0.3-66           \n [31] fitdistrplus_1.2-2     future_1.34.0          shiny_1.10.0          \n [34] digest_0.6.37          colorspace_2.1-1       patchwork_1.3.0       \n [37] S4Vectors_0.40.2       tensor_1.5             RSpectra_0.16-2       \n [40] irlba_2.3.5.1          labeling_0.4.3         progressr_0.15.1      \n [43] fansi_1.0.6            spatstat.sparse_3.1-0  httr_1.4.7            \n [46] polyclip_1.10-7        abind_1.4-5            compiler_4.3.3        \n [49] withr_3.0.2            doParallel_1.0.17      viridis_0.6.5         \n [52] fastDummies_1.7.5      ggforce_0.4.2          MASS_7.3-60.0.1       \n [55] rjson_0.2.23           tools_4.3.3            vipor_0.4.7           \n [58] lmtest_0.9-40          beeswarm_0.4.0         httpuv_1.6.15         \n [61] future.apply_1.11.3    goftest_1.2-3          glue_1.8.0            \n [64] nlme_3.1-167           promises_1.3.2         Rtsne_0.17            \n [67] cluster_2.1.8          reshape2_1.4.4         generics_0.1.3        \n [70] gtable_0.3.6           spatstat.data_3.1-4    data.table_1.16.4     \n [73] tidygraph_1.3.0        BiocGenerics_0.48.1    spatstat.geom_3.3-5   \n [76] RcppAnnoy_0.0.22       RANN_2.6.2             foreach_1.5.2         \n [79] pillar_1.10.1          stringr_1.5.1          spam_2.11-1           \n [82] RcppHNSW_0.6.0         later_1.4.1            splines_4.3.3         \n [85] tweenr_2.0.3           lattice_0.22-6         survival_3.8-3        \n [88] deldir_2.0-4           miniUI_0.1.1.1         pbapply_1.7-2         \n [91] gridExtra_2.3          IRanges_2.36.0         scattermore_1.2       \n [94] stats4_4.3.3           graphlayouts_1.2.2     matrixStats_1.1.0     \n [97] stringi_1.8.4          lazyeval_0.2.2         codetools_0.2-20      \n[100] cli_3.6.4              uwot_0.2.2             xtable_1.8-4          \n[103] reticulate_1.40.0      munsell_0.5.1          Rcpp_1.0.14           \n[106] globals_0.16.3         spatstat.random_3.3-2  png_0.1-8             \n[109] ggrastr_1.0.2          spatstat.univar_3.1-1  parallel_4.3.3        \n[112] ellipsis_0.3.2         dotCall64_1.2          listenv_0.9.1         \n[115] viridisLite_0.4.2      scales_1.3.0           ggridges_0.5.6        \n[118] leiden_0.4.3.1         purrr_1.0.2            crayon_1.5.3          \n[121] GetoptLong_1.0.5       cowplot_1.1.3         \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 18.76\nuser 27.07\nsys 0.71\npercent 148%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"Immunarch",desc:"Exploration of Single-cell and Bulk T-cell/Antibody Immune Repertoires",npages:3,report_toc:!0,order:18,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] circlize_0.4.16   tibble_3.2.1      glue_1.8.0        ggprism_1.0.5    \n [5] immunarch_0.9.1   patchwork_1.3.0   data.table_1.16.4 dtplyr_1.3.1     \n [9] ggplot2_3.5.1     tidyr_1.3.1       dplyr_1.1.4       jsonlite_1.8.9   \n[13] rlang_1.1.5       logger_0.4.0     \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22       shinythemes_1.2.0      splines_4.3.3         \n  [4] later_1.4.1            cellranger_1.1.0       polyclip_1.10-7       \n  [7] fastDummies_1.7.5      factoextra_1.0.7       lifecycle_1.0.4       \n [10] rstatix_0.7.2          doParallel_1.0.17      globals_0.16.3        \n [13] lattice_0.22-6         prabclus_2.3-4         MASS_7.3-60.0.1       \n [16] backports_1.5.0        magrittr_2.0.3         plotly_4.10.4         \n [19] rlist_0.4.6.2          httpuv_1.6.15          Seurat_5.1.0          \n [22] sctransform_0.4.1      spam_2.11-1            flexmix_2.3-19        \n [25] spatstat.sparse_3.1-0  sp_2.2-0               reticulate_1.40.0     \n [28] cowplot_1.1.3          pbapply_1.7-2          RColorBrewer_1.1-3    \n [31] abind_1.4-5            Rtsne_0.17             quadprog_1.5-8        \n [34] purrr_1.0.2            ggraph_2.2.1           nnet_7.3-20           \n [37] tweenr_2.0.3           ggrepel_0.9.6          irlba_2.3.5.1         \n [40] spatstat.utils_3.1-2   listenv_0.9.1          pheatmap_1.0.12       \n [43] goftest_1.2-3          RSpectra_0.16-2        spatstat.random_3.3-2 \n [46] fitdistrplus_1.2-2     parallelly_1.42.0      leiden_0.4.3.1        \n [49] codetools_0.2-20       ggforce_0.4.2          tidyselect_1.2.1      \n [52] shape_1.4.6.1          ggseqlogo_0.2          farver_2.1.2          \n [55] viridis_0.6.5          matrixStats_1.1.0      stats4_4.3.3          \n [58] spatstat.explore_3.3-4 tidygraph_1.3.0        progressr_0.15.1      \n [61] Formula_1.2-5          ggridges_0.5.6         ggalluvial_0.12.5     \n [64] survival_3.8-3         iterators_1.0.14       foreach_1.5.2         \n [67] tools_4.3.3            stringdist_0.9.15      ica_1.0-3             \n [70] Rcpp_1.0.14            gridExtra_2.3          withr_3.0.2           \n [73] fastmap_1.2.0          digest_0.6.37          R6_2.6.0              \n [76] mime_0.12              colorspace_2.1-1       scattermore_1.2       \n [79] tensor_1.5             spatstat.data_3.1-4    diptest_0.77-1        \n [82] UpSetR_1.4.0           generics_0.1.3         robustbase_0.99-4-1   \n [85] class_7.3-23           graphlayouts_1.2.2     httr_1.4.7            \n [88] htmlwidgets_1.6.4      uwot_0.2.2             pkgconfig_2.0.3       \n [91] gtable_0.3.6           modeltools_0.2-23      lmtest_0.9-40         \n [94] htmltools_0.5.8.1      carData_3.0-5          dotCall64_1.2         \n [97] SeuratObject_5.0.2     scales_1.3.0           png_0.1-8             \n[100] spatstat.univar_3.1-1  tzdb_0.4.0             reshape2_1.4.4        \n[103] uuid_1.2-1             nlme_3.1-167           cachem_1.1.0          \n[106] zoo_1.8-12             GlobalOptions_0.1.2    stringr_1.5.1         \n[109] KernSmooth_2.23-26     parallel_4.3.3         miniUI_0.1.1.1        \n[112] pillar_1.10.1          grid_4.3.3             vctrs_0.6.5           \n[115] RANN_2.6.2             promises_1.3.2         ggpubr_0.6.0          \n[118] car_3.1-3              xtable_1.8-4           cluster_2.1.8         \n[121] readr_2.1.5            cli_3.6.4              compiler_4.3.3        \n[124] future.apply_1.11.3    ggsignif_0.6.4         labeling_0.4.3        \n[127] mclust_6.1.1           plyr_1.8.9             stringi_1.8.4         \n[130] deldir_2.0-4           viridisLite_0.4.2      assertthat_0.2.1      \n[133] munsell_0.5.1          lazyeval_0.2.2         spatstat.geom_3.3-5   \n[136] Matrix_1.6-5           RcppHNSW_0.6.0         hms_1.1.3             \n[139] future_1.34.0          fpc_2.2-13             shiny_1.10.0          \n[142] kernlab_0.9-33         ROCR_1.0-11            igraph_2.0.3          \n[145] broom_1.0.7            memoise_2.0.1          phangorn_2.12.1       \n[148] fastmatch_1.1-6        DEoptimR_1.1-3-1       readxl_1.4.3          \n[151] ape_5.8-1             \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 102.43\nuser 256.71\nsys 5.18\npercent 255%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"RadarPlots",desc:"Radar plots for cell proportion in different clusters.",npages:1,report_toc:!0,order:19,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ggprism_1.0.5      ggradar_0.2        ggplot2_3.5.1      tibble_3.2.1      \n [5] tidyr_1.3.1        dplyr_1.1.4        Seurat_5.1.0       SeuratObject_5.0.2\n [9] sp_2.2-0           jsonlite_1.8.9     rlang_1.1.5        logger_0.4.0      \n\nloaded via a namespace (and not attached):\n [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n[10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n[13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n[16] labeling_0.4.3         promises_1.3.2         purrr_1.0.2           \n[19] goftest_1.2-3          later_1.4.1            spatstat.utils_3.1-2  \n[22] irlba_2.3.5.1          parallel_4.3.3         cluster_2.1.8         \n[25] R6_2.6.0               ica_1.0-3              stringi_1.8.4         \n[28] RColorBrewer_1.1-3     spatstat.data_3.1-4    reticulate_1.40.0     \n[31] parallelly_1.42.0      spatstat.univar_3.1-1  lmtest_0.9-40         \n[34] scattermore_1.2        Rcpp_1.0.14            tensor_1.5            \n[37] future.apply_1.11.3    zoo_1.8-12             sctransform_0.4.1     \n[40] httpuv_1.6.15          Matrix_1.6-5           splines_4.3.3         \n[43] igraph_2.0.3           tidyselect_1.2.1       abind_1.4-5           \n[46] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n[49] spatstat.explore_3.3-4 listenv_0.9.1          lattice_0.22-6        \n[52] plyr_1.8.9             withr_3.0.2            shiny_1.10.0          \n[55] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n[58] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n[61] fitdistrplus_1.2-2     pillar_1.10.1          KernSmooth_2.23-26    \n[64] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n[67] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n[70] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n[73] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n[76] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n[79] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n[82] nlme_3.1-167           patchwork_1.3.0        cli_3.6.4             \n[85] spatstat.sparse_3.1-0  spam_2.11-1            viridisLite_0.4.2     \n[88] uwot_0.2.2             gtable_0.3.6           digest_0.6.37         \n[91] progressr_0.15.1       ggrepel_0.9.6          htmlwidgets_1.6.4     \n[94] farver_2.1.2           htmltools_0.5.8.1      lifecycle_1.0.4       \n[97] httr_1.4.7             mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 8.07\nuser 16.74\nsys 0.38\npercent 212%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"CloneResidency",desc:"Identification of clone residency",npages:1,report_toc:!0,order:20,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ComplexUpset_1.3.3  ggVennDiagram_1.2.3 ggprism_1.0.5      \n [4] ggplot2_3.5.1       ggnewscale_0.5.0    glue_1.8.0         \n [7] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[10] jsonlite_1.8.9      rlang_1.1.5         logger_0.4.0       \n\nloaded via a namespace (and not attached):\n  [1] DBI_1.2.3              deldir_2.0-4           pbapply_1.7-2         \n  [4] gridExtra_2.3          magrittr_2.0.3         RcppAnnoy_0.0.22      \n  [7] e1071_1.7-16           spatstat.geom_3.3-5    matrixStats_1.1.0     \n [10] ggridges_0.5.6         compiler_4.3.3         png_0.1-8             \n [13] vctrs_0.6.5            reshape2_1.4.4         stringr_1.5.1         \n [16] pkgconfig_2.0.3        fastmap_1.2.0          labeling_0.4.3        \n [19] promises_1.3.2         purrr_1.0.2            RVenn_1.1.0           \n [22] goftest_1.2-3          later_1.4.1            spatstat.utils_3.1-2  \n [25] irlba_2.3.5.1          parallel_4.3.3         cluster_2.1.8         \n [28] R6_2.6.0               ica_1.0-3              stringi_1.8.4         \n [31] RColorBrewer_1.1-3     spatstat.data_3.1-4    reticulate_1.40.0     \n [34] parallelly_1.42.0      spatstat.univar_3.1-1  lmtest_0.9-40         \n [37] scattermore_1.2        Rcpp_1.0.14            tensor_1.5            \n [40] future.apply_1.11.3    zoo_1.8-12             sctransform_0.4.1     \n [43] httpuv_1.6.15          Matrix_1.6-5           splines_4.3.3         \n [46] igraph_2.0.3           tidyselect_1.2.1       abind_1.4-5           \n [49] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.3-4 listenv_0.9.1          lattice_0.22-6        \n [55] plyr_1.8.9             shiny_1.10.0           withr_3.0.2           \n [58] ROCR_1.0-11            Rtsne_0.17             sf_1.0-19             \n [61] future_1.34.0          fastDummies_1.7.5      survival_3.8-3        \n [64] units_0.8-5            proxy_0.4-27           polyclip_1.10-7       \n [67] fitdistrplus_1.2-2     pillar_1.10.1          Seurat_5.1.0          \n [70] KernSmooth_2.23-26     plotly_4.10.4          generics_0.1.3        \n [73] RcppHNSW_0.6.0         sp_2.2-0               munsell_0.5.1         \n [76] scales_1.3.0           globals_0.16.3         xtable_1.8-4          \n [79] class_7.3-23           lazyeval_0.2.2         tools_4.3.3           \n [82] data.table_1.16.4      RSpectra_0.16-2        RANN_2.6.2            \n [85] leiden_0.4.3.1         dotCall64_1.2          cowplot_1.1.3         \n [88] grid_4.3.3             colorspace_2.1-1       nlme_3.1-167          \n [91] patchwork_1.3.0        cli_3.6.4              spatstat.sparse_3.1-0 \n [94] spam_2.11-1            viridisLite_0.4.2      uwot_0.2.2            \n [97] gtable_0.3.6           digest_0.6.37          classInt_0.4-11       \n[100] progressr_0.15.1       ggrepel_0.9.6          htmlwidgets_1.6.4     \n[103] SeuratObject_5.0.2     farver_2.1.2           htmltools_0.5.8.1     \n[106] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[109] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 19.89\nuser 28.24\nsys 0.66\npercent 145%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ScFGSEA",desc:"Gene set enrichment analysis for cells in different groups using `fgsea`",npages:2,report_toc:!0,order:21,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] fgsea_1.28.0       data.table_1.16.4  matrixStats_1.1.0  tidyseurat_0.8.0  \n [5] ttservice_0.4.1    Seurat_5.1.0       SeuratObject_5.0.2 sp_2.2-0          \n [9] tidyr_1.3.1        tidyselect_1.2.1   tibble_3.2.1       dplyr_1.1.4       \n[13] ggplot2_3.5.1      jsonlite_1.8.9     rlang_1.1.5        logger_0.4.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       ggridges_0.5.6        \n  [7] compiler_4.3.3         spatstat.geom_3.3-5    png_0.1-8             \n [10] vctrs_0.6.5            reshape2_1.4.4         stringr_1.5.1         \n [13] pkgconfig_2.0.3        fastmap_1.2.0          ellipsis_0.3.2        \n [16] labeling_0.4.3         promises_1.3.2         purrr_1.0.2           \n [19] goftest_1.2-3          later_1.4.1            BiocParallel_1.36.0   \n [22] spatstat.utils_3.1-2   irlba_2.3.5.1          parallel_4.3.3        \n [25] cluster_2.1.8          R6_2.6.0               ica_1.0-3             \n [28] stringi_1.8.4          RColorBrewer_1.1-3     spatstat.data_3.1-4   \n [31] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [37] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [43] splines_4.3.3          igraph_2.0.3           abind_1.4-5           \n [46] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n [49] spatstat.explore_3.3-4 listenv_0.9.1          lattice_0.22-6        \n [52] plyr_1.8.9             shiny_1.10.0           withr_3.0.2           \n [55] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n [58] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n [61] fitdistrplus_1.2-2     pillar_1.10.1          KernSmooth_2.23-26    \n [64] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n [67] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [70] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [73] tools_4.3.3            RSpectra_0.16-2        RANN_2.6.2            \n [76] leiden_0.4.3.1         dotCall64_1.2          fastmatch_1.1-6       \n [79] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n [82] nlme_3.1-167           patchwork_1.3.0        cli_3.6.4             \n [85] spatstat.sparse_3.1-0  fansi_1.0.6            spam_2.11-1           \n [88] viridisLite_0.4.2      uwot_0.2.2             gtable_0.3.6          \n [91] digest_0.6.37          progressr_0.15.1       ggrepel_0.9.6         \n [94] htmlwidgets_1.6.4      farver_2.1.2           htmltools_0.5.8.1     \n [97] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[100] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 128.45\nuser 135.35\nsys 1.52\npercent 106%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"MarkersFinder",desc:"Find markers between different groups of cells",npages:1,report_toc:!0,order:22,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] UpSetR_1.4.0        ggVennDiagram_1.2.3 tidyseurat_0.8.0   \n [4] ttservice_0.4.1     future_1.34.0       ggrepel_0.9.6      \n [7] ggprism_1.0.5       ggplot2_3.5.1       enrichR_3.4        \n[10] Seurat_5.1.0        SeuratObject_5.0.2  sp_2.2-0           \n[13] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[16] tidyselect_1.2.1    digest_0.6.37       jsonlite_1.8.9     \n[19] rlang_1.1.5         logger_0.4.0       \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] ellipsis_0.3.2         labeling_0.4.3         promises_1.3.2        \n [19] purrr_1.0.2            WriteXLS_6.7.0         RVenn_1.1.0           \n [22] goftest_1.2-3          later_1.4.1            spatstat.utils_3.1-2  \n [25] irlba_2.3.5.1          parallel_4.3.3         cluster_2.1.8         \n [28] R6_2.6.0               ica_1.0-3              stringi_1.8.4         \n [31] RColorBrewer_1.1-3     spatstat.data_3.1-4    limma_3.58.1          \n [34] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [37] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [40] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [43] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [46] splines_4.3.3          igraph_2.0.3           abind_1.4-5           \n [49] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.3-4 curl_6.0.1             listenv_0.9.1         \n [55] lattice_0.22-6         plyr_1.8.9             withr_3.0.2           \n [58] shiny_1.10.0           ROCR_1.0-11            Rtsne_0.17            \n [61] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n [64] fitdistrplus_1.2-2     pillar_1.10.1          KernSmooth_2.23-26    \n [67] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n [82] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n [85] nlme_3.1-167           patchwork_1.3.0        presto_1.0.0          \n [88] cli_3.6.4              spatstat.sparse_3.1-0  fansi_1.0.6           \n [91] spam_2.11-1            viridisLite_0.4.2      uwot_0.2.2            \n [94] gtable_0.3.6           progressr_0.15.1       rjson_0.2.23          \n [97] htmlwidgets_1.6.4      farver_2.1.2           htmltools_0.5.8.1     \n[100] lifecycle_1.0.4        httr_1.4.7             statmod_1.5.0         \n[103] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 19.16\nuser 23.15\nsys 0.56\npercent 123%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"MetaMarkers",desc:"Find markers between three or more groups of cells, using one-way ANOVA or Kruskal-Wallis test.",npages:2,report_toc:!0,order:23,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] parallel  stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] tidyseurat_0.8.0   ttservice_0.4.1    ggprism_1.0.5      ggplot2_3.5.1     \n [5] enrichR_3.4        Seurat_5.1.0       SeuratObject_5.0.2 sp_2.2-0          \n [9] Matrix_1.6-5       tibble_3.2.1       tidyr_1.3.1        dplyr_1.1.4       \n[13] tidyselect_1.2.1   jsonlite_1.8.9     rlang_1.1.5        logger_0.4.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] ellipsis_0.3.2         labeling_0.4.3         promises_1.3.2        \n [19] purrr_1.0.2            WriteXLS_6.7.0         goftest_1.2-3         \n [22] later_1.4.1            spatstat.utils_3.1-2   irlba_2.3.5.1         \n [25] cluster_2.1.8          R6_2.6.0               ica_1.0-3             \n [28] stringi_1.8.4          RColorBrewer_1.1-3     spatstat.data_3.1-4   \n [31] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [37] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          splines_4.3.3         \n [43] igraph_2.0.3           abind_1.4-5            spatstat.random_3.3-2 \n [46] codetools_0.2-20       miniUI_0.1.1.1         spatstat.explore_3.3-4\n [49] curl_6.0.1             listenv_0.9.1          lattice_0.22-6        \n [52] plyr_1.8.9             withr_3.0.2            shiny_1.10.0          \n [55] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n [58] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n [61] fitdistrplus_1.2-2     pillar_1.10.1          KernSmooth_2.23-26    \n [64] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n [67] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [70] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [73] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [76] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n [79] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n [82] nlme_3.1-167           patchwork_1.3.0        cli_3.6.4             \n [85] spatstat.sparse_3.1-0  fansi_1.0.6            spam_2.11-1           \n [88] viridisLite_0.4.2      uwot_0.2.2             gtable_0.3.6          \n [91] digest_0.6.37          progressr_0.15.1       ggrepel_0.9.6         \n [94] rjson_0.2.23           htmlwidgets_1.6.4      farver_2.1.2          \n [97] htmltools_0.5.8.1      lifecycle_1.0.4        httr_1.4.7            \n[100] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 344.55\nuser 1593.68\nsys 68.40\npercent 482%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"CellsDistribution",desc:"Distribution of cells (i.e. in a TCR clone) from different groups for each cluster",npages:1,report_toc:!0,order:24,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] grid      stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] ComplexHeatmap_2.18.0 circlize_0.4.16       UpSetR_1.4.0         \n [4] ggVennDiagram_1.2.3   patchwork_1.3.0       ggplot2_3.5.1        \n [7] tibble_3.2.1          Seurat_5.1.0          SeuratObject_5.0.2   \n[10] sp_2.2-0              tidyr_1.3.1           dplyr_1.1.4          \n[13] tidyselect_1.2.1      jsonlite_1.8.9        rlang_1.1.5          \n[16] logger_0.4.0         \n\nloaded via a namespace (and not attached):\n  [1] RColorBrewer_1.1-3     shape_1.4.6.1          magrittr_2.0.3        \n  [4] spatstat.utils_3.1-2   farver_2.1.2           GlobalOptions_0.1.2   \n  [7] vctrs_0.6.5            ROCR_1.0-11            Cairo_1.6-2           \n [10] spatstat.explore_3.3-4 htmltools_0.5.8.1      sctransform_0.4.1     \n [13] parallelly_1.42.0      KernSmooth_2.23-26     htmlwidgets_1.6.4     \n [16] ica_1.0-3              plyr_1.8.9             plotly_4.10.4         \n [19] zoo_1.8-12             igraph_2.0.3           mime_0.12             \n [22] lifecycle_1.0.4        iterators_1.0.14       pkgconfig_2.0.3       \n [25] Matrix_1.6-5           R6_2.6.0               fastmap_1.2.0         \n [28] clue_0.3-66            fitdistrplus_1.2-2     future_1.34.0         \n [31] shiny_1.10.0           digest_0.6.37          colorspace_2.1-1      \n [34] S4Vectors_0.40.2       tensor_1.5             RSpectra_0.16-2       \n [37] irlba_2.3.5.1          labeling_0.4.3         progressr_0.15.1      \n [40] spatstat.sparse_3.1-0  httr_1.4.7             polyclip_1.10-7       \n [43] abind_1.4-5            compiler_4.3.3         withr_3.0.2           \n [46] doParallel_1.0.17      fastDummies_1.7.5      MASS_7.3-60.0.1       \n [49] rjson_0.2.23           tools_4.3.3            lmtest_0.9-40         \n [52] httpuv_1.6.15          future.apply_1.11.3    goftest_1.2-3         \n [55] glue_1.8.0             nlme_3.1-167           promises_1.3.2        \n [58] Rtsne_0.17             cluster_2.1.8          reshape2_1.4.4        \n [61] generics_0.1.3         gtable_0.3.6           spatstat.data_3.1-4   \n [64] data.table_1.16.4      BiocGenerics_0.48.1    spatstat.geom_3.3-5   \n [67] RcppAnnoy_0.0.22       ggrepel_0.9.6          RANN_2.6.2            \n [70] foreach_1.5.2          pillar_1.10.1          stringr_1.5.1         \n [73] spam_2.11-1            RcppHNSW_0.6.0         later_1.4.1           \n [76] splines_4.3.3          lattice_0.22-6         survival_3.8-3        \n [79] deldir_2.0-4           miniUI_0.1.1.1         pbapply_1.7-2         \n [82] gridExtra_2.3          IRanges_2.36.0         scattermore_1.2       \n [85] stats4_4.3.3           matrixStats_1.1.0      stringi_1.8.4         \n [88] lazyeval_0.2.2         codetools_0.2-20       RVenn_1.1.0           \n [91] cli_3.6.4              uwot_0.2.2             xtable_1.8-4          \n [94] reticulate_1.40.0      munsell_0.5.1          Rcpp_1.0.14           \n [97] globals_0.16.3         spatstat.random_3.3-2  png_0.1-8             \n[100] spatstat.univar_3.1-1  parallel_4.3.3         dotCall64_1.2         \n[103] listenv_0.9.1          viridisLite_0.4.2      scales_1.3.0          \n[106] ggridges_0.5.6         leiden_0.4.3.1         purrr_1.0.2           \n[109] crayon_1.5.3           GetoptLong_1.0.5       cowplot_1.1.3         \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 12.22\nuser 20.80\nsys 0.49\npercent 174%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"CDR3AAPhyschem",desc:"CDR3 AA physicochemical feature analysis",npages:1,report_toc:!0,order:25,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] stringr_1.5.1       broom.mixed_0.2.9.6 glmnet_4.1-8       \n [4] Matrix_1.6-5        hash_3.0.1          glue_1.8.0         \n [7] ggridges_0.5.6      ggplot2_3.5.1       tibble_3.2.1       \n[10] tidyr_1.3.1         dplyr_1.1.4         jsonlite_1.8.9     \n[13] rlang_1.1.5         logger_0.4.0       \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         furrr_0.3.1            RcppAnnoy_0.0.22      \n  [7] spatstat.geom_3.3-5    matrixStats_1.1.0      compiler_4.3.3        \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] pkgconfig_2.0.3        shape_1.4.6.1          fastmap_1.2.0         \n [16] backports_1.5.0        labeling_0.4.3         promises_1.3.2        \n [19] purrr_1.0.2            goftest_1.2-3          later_1.4.1           \n [22] spatstat.utils_3.1-2   broom_1.0.7            irlba_2.3.5.1         \n [25] parallel_4.3.3         cluster_2.1.8          R6_2.6.0              \n [28] ica_1.0-3              spatstat.data_3.1-4    stringi_1.8.4         \n [31] RColorBrewer_1.1-3     reticulate_1.40.0      spatstat.univar_3.1-1 \n [34] parallelly_1.42.0      lmtest_0.9-40          scattermore_1.2       \n [37] Rcpp_1.0.14            iterators_1.0.14       tensor_1.5            \n [40] future.apply_1.11.3    zoo_1.8-12             sctransform_0.4.1     \n [43] httpuv_1.6.15          splines_4.3.3          igraph_2.0.3          \n [46] tidyselect_1.2.1       abind_1.4-5            spatstat.random_3.3-2 \n [49] codetools_0.2-20       miniUI_0.1.1.1         spatstat.explore_3.3-4\n [52] listenv_0.9.1          lattice_0.22-6         plyr_1.8.9            \n [55] shiny_1.10.0           withr_3.0.2            ROCR_1.0-11           \n [58] Rtsne_0.17             future_1.34.0          fastDummies_1.7.5     \n [61] survival_3.8-3         polyclip_1.10-7        fitdistrplus_1.2-2    \n [64] pillar_1.10.1          Seurat_5.1.0           KernSmooth_2.23-26    \n [67] foreach_1.5.2          plotly_4.10.4          generics_0.1.3        \n [70] RcppHNSW_0.6.0         sp_2.2-0               munsell_0.5.1         \n [73] scales_1.3.0           globals_0.16.3         xtable_1.8-4          \n [76] lazyeval_0.2.2         tools_4.3.3            data.table_1.16.4     \n [79] RSpectra_0.16-2        forcats_1.0.0          RANN_2.6.2            \n [82] leiden_0.4.3.1         dotCall64_1.2          cowplot_1.1.3         \n [85] grid_4.3.3             colorspace_2.1-1       nlme_3.1-167          \n [88] patchwork_1.3.0        cli_3.6.4              spatstat.sparse_3.1-0 \n [91] spam_2.11-1            viridisLite_0.4.2      uwot_0.2.2            \n [94] gtable_0.3.6           digest_0.6.37          progressr_0.15.1      \n [97] ggrepel_0.9.6          htmlwidgets_1.6.4      SeuratObject_5.0.2    \n[100] farver_2.1.2           htmltools_0.5.8.1      lifecycle_1.0.4       \n[103] httr_1.4.7             mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 11.03\nuser 19.55\nsys 0.54\npercent 182%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ScrnaMetabolicLandscape",desc:"Metabolic landscape analysis for scRNA-seq data",order:28,procs:[{name:"MetabolicPathwayActivity",desc:"This process calculates the pathway activities in different groups and subsets.",npages:1,report_toc:!0,order:28,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n [1] grid      parallel  stats4    stats     graphics  grDevices utils    \n [8] datasets  methods   base     \n\nother attached packages:\n [1] ComplexHeatmap_2.18.0       Seurat_5.1.0               \n [3] SeuratObject_5.0.2          sp_2.2-0                   \n [5] ggprism_1.0.5               RColorBrewer_1.1-3         \n [7] reshape2_1.4.4              scater_1.30.1              \n [9] scuttle_1.12.0              SingleCellExperiment_1.24.0\n[11] SummarizedExperiment_1.32.0 Biobase_2.62.0             \n[13] GenomicRanges_1.54.1        GenomeInfoDb_1.38.1        \n[15] IRanges_2.36.0              S4Vectors_0.40.2           \n[17] BiocGenerics_0.48.1         MatrixGenerics_1.14.0      \n[19] matrixStats_1.1.0           tibble_3.2.1               \n[21] dplyr_1.1.4                 ggplot2_3.5.1              \n[23] jsonlite_1.8.9              rlang_1.1.5                \n[25] logger_0.4.0               \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22          splines_4.3.3            \n  [3] later_1.4.1               bitops_1.0-9             \n  [5] polyclip_1.10-7           fastDummies_1.7.5        \n  [7] lifecycle_1.0.4           doParallel_1.0.17        \n  [9] globals_0.16.3            lattice_0.22-6           \n [11] MASS_7.3-60.0.1           magrittr_2.0.3           \n [13] plotly_4.10.4             httpuv_1.6.15            \n [15] sctransform_0.4.1         spam_2.11-1              \n [17] spatstat.sparse_3.1-0     reticulate_1.40.0        \n [19] cowplot_1.1.3             pbapply_1.7-2            \n [21] abind_1.4-5               zlibbioc_1.48.0          \n [23] Rtsne_0.17                purrr_1.0.2              \n [25] RCurl_1.98-1.16           circlize_0.4.16          \n [27] GenomeInfoDbData_1.2.11   ggrepel_0.9.6            \n [29] irlba_2.3.5.1             listenv_0.9.1            \n [31] spatstat.utils_3.1-2      goftest_1.2-3            \n [33] RSpectra_0.16-2           spatstat.random_3.3-2    \n [35] fitdistrplus_1.2-2        parallelly_1.42.0        \n [37] DelayedMatrixStats_1.24.0 leiden_0.4.3.1           \n [39] codetools_0.2-20          DelayedArray_0.28.0      \n [41] tidyselect_1.2.1          shape_1.4.6.1            \n [43] farver_2.1.2              ScaledMatrix_1.10.0      \n [45] viridis_0.6.5             spatstat.explore_3.3-4   \n [47] GetoptLong_1.0.5          BiocNeighbors_1.20.0     \n [49] progressr_0.15.1          ggridges_0.5.6           \n [51] survival_3.8-3            iterators_1.0.14         \n [53] foreach_1.5.2             tools_4.3.3              \n [55] ica_1.0-3                 Rcpp_1.0.14              \n [57] glue_1.8.0                gridExtra_2.3            \n [59] SparseArray_1.2.2         withr_3.0.2              \n [61] fastmap_1.2.0             digest_0.6.37            \n [63] rsvd_1.0.5                R6_2.6.0                 \n [65] mime_0.12                 colorspace_2.1-1         \n [67] Cairo_1.6-2               scattermore_1.2          \n [69] tensor_1.5                spatstat.data_3.1-4      \n [71] tidyr_1.3.1               generics_0.1.3           \n [73] data.table_1.16.4         httr_1.4.7               \n [75] htmlwidgets_1.6.4         S4Arrays_1.2.0           \n [77] uwot_0.2.2                pkgconfig_2.0.3          \n [79] gtable_0.3.6              lmtest_0.9-40            \n [81] XVector_0.42.0            htmltools_0.5.8.1        \n [83] dotCall64_1.2             clue_0.3-66              \n [85] scales_1.3.0              png_0.1-8                \n [87] spatstat.univar_3.1-1     rjson_0.2.23             \n [89] nlme_3.1-167              zoo_1.8-12               \n [91] GlobalOptions_0.1.2       stringr_1.5.1            \n [93] KernSmooth_2.23-26        miniUI_0.1.1.1           \n [95] vipor_0.4.7               pillar_1.10.1            \n [97] vctrs_0.6.5               RANN_2.6.2               \n [99] promises_1.3.2            BiocSingular_1.18.0      \n[101] beachmat_2.18.0           xtable_1.8-4             \n[103] cluster_2.1.8             beeswarm_0.4.0           \n[105] cli_3.6.4                 compiler_4.3.3           \n[107] crayon_1.5.3              future.apply_1.11.3      \n[109] plyr_1.8.9                ggbeeswarm_0.7.2         \n[111] stringi_1.8.4             viridisLite_0.4.2        \n[113] deldir_2.0-4              BiocParallel_1.36.0      \n[115] munsell_0.5.1             lazyeval_0.2.2           \n[117] spatstat.geom_3.3-5       Matrix_1.6-5             \n[119] RcppHNSW_0.6.0            patchwork_1.3.0          \n[121] sparseMatrixStats_1.14.0  future_1.34.0            \n[123] shiny_1.10.0              ROCR_1.0-11              \n[125] igraph_2.0.3             \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 755.68\nuser 748.76\nsys 14.13\npercent 100%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"MetabolicFeatures",desc:"This process performs enrichment analysis for the metabolic pathways for each group in each subset.",npages:1,report_toc:!0,order:29,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] parallel  stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] matrixStats_1.1.0  fgsea_1.28.0       data.table_1.16.4  Seurat_5.1.0      \n [5] SeuratObject_5.0.2 sp_2.2-0           tibble_3.2.1       dplyr_1.1.4       \n [9] ggplot2_3.5.1      jsonlite_1.8.9     rlang_1.1.5        logger_0.4.0      \n\nloaded via a namespace (and not attached):\n [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n [4] magrittr_2.0.3         RcppAnnoy_0.0.22       ggridges_0.5.6        \n [7] compiler_4.3.3         spatstat.geom_3.3-5    png_0.1-8             \n[10] vctrs_0.6.5            reshape2_1.4.4         stringr_1.5.1         \n[13] pkgconfig_2.0.3        fastmap_1.2.0          labeling_0.4.3        \n[16] promises_1.3.2         purrr_1.0.2            goftest_1.2-3         \n[19] later_1.4.1            BiocParallel_1.36.0    spatstat.utils_3.1-2  \n[22] irlba_2.3.5.1          cluster_2.1.8          R6_2.6.0              \n[25] ica_1.0-3              stringi_1.8.4          RColorBrewer_1.1-3    \n[28] spatstat.data_3.1-4    reticulate_1.40.0      parallelly_1.42.0     \n[31] spatstat.univar_3.1-1  lmtest_0.9-40          scattermore_1.2       \n[34] Rcpp_1.0.14            tensor_1.5             future.apply_1.11.3   \n[37] zoo_1.8-12             sctransform_0.4.1      httpuv_1.6.15         \n[40] Matrix_1.6-5           splines_4.3.3          igraph_2.0.3          \n[43] tidyselect_1.2.1       abind_1.4-5            spatstat.random_3.3-2 \n[46] codetools_0.2-20       miniUI_0.1.1.1         spatstat.explore_3.3-4\n[49] listenv_0.9.1          lattice_0.22-6         plyr_1.8.9            \n[52] shiny_1.10.0           withr_3.0.2            ROCR_1.0-11           \n[55] Rtsne_0.17             future_1.34.0          fastDummies_1.7.5     \n[58] survival_3.8-3         polyclip_1.10-7        fitdistrplus_1.2-2    \n[61] pillar_1.10.1          KernSmooth_2.23-26     plotly_4.10.4         \n[64] generics_0.1.3         RcppHNSW_0.6.0         munsell_0.5.1         \n[67] scales_1.3.0           globals_0.16.3         xtable_1.8-4          \n[70] glue_1.8.0             lazyeval_0.2.2         tools_4.3.3           \n[73] RSpectra_0.16-2        RANN_2.6.2             leiden_0.4.3.1        \n[76] dotCall64_1.2          fastmatch_1.1-6        cowplot_1.1.3         \n[79] grid_4.3.3             tidyr_1.3.1            colorspace_2.1-1      \n[82] nlme_3.1-167           patchwork_1.3.0        cli_3.6.4             \n[85] spatstat.sparse_3.1-0  spam_2.11-1            viridisLite_0.4.2     \n[88] uwot_0.2.2             gtable_0.3.6           digest_0.6.37         \n[91] progressr_0.15.1       ggrepel_0.9.6          htmlwidgets_1.6.4     \n[94] farver_2.1.2           htmltools_0.5.8.1      lifecycle_1.0.4       \n[97] httr_1.4.7             mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 265.61\nuser 265.24\nsys 11.02\npercent 104%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"MetabolicFeaturesIntraSubset",desc:"Intra-subset metabolic features - Enrichment analysis in details",npages:3,report_toc:!0,order:30,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats4    parallel  stats     graphics  grDevices utils     datasets \n[8] methods   base     \n\nother attached packages:\n [1] fgsea_1.28.0                data.table_1.16.4          \n [3] Seurat_5.1.0                SeuratObject_5.0.2         \n [5] sp_2.2-0                    scater_1.30.1              \n [7] scuttle_1.12.0              SingleCellExperiment_1.24.0\n [9] SummarizedExperiment_1.32.0 Biobase_2.62.0             \n[11] GenomicRanges_1.54.1        GenomeInfoDb_1.38.1        \n[13] IRanges_2.36.0              S4Vectors_0.40.2           \n[15] BiocGenerics_0.48.1         MatrixGenerics_1.14.0      \n[17] matrixStats_1.1.0           tibble_3.2.1               \n[19] dplyr_1.1.4                 ggplot2_3.5.1              \n[21] jsonlite_1.8.9              rlang_1.1.5                \n[23] logger_0.4.0               \n\nloaded via a namespace (and not attached):\n  [1] RColorBrewer_1.1-3        magrittr_2.0.3           \n  [3] spatstat.utils_3.1-2      ggbeeswarm_0.7.2         \n  [5] farver_2.1.2              zlibbioc_1.48.0          \n  [7] vctrs_0.6.5               ROCR_1.0-11              \n  [9] spatstat.explore_3.3-4    DelayedMatrixStats_1.24.0\n [11] RCurl_1.98-1.16           htmltools_0.5.8.1        \n [13] S4Arrays_1.2.0            BiocNeighbors_1.20.0     \n [15] SparseArray_1.2.2         sctransform_0.4.1        \n [17] parallelly_1.42.0         KernSmooth_2.23-26       \n [19] htmlwidgets_1.6.4         ica_1.0-3                \n [21] plyr_1.8.9                plotly_4.10.4            \n [23] zoo_1.8-12                igraph_2.0.3             \n [25] mime_0.12                 lifecycle_1.0.4          \n [27] pkgconfig_2.0.3           rsvd_1.0.5               \n [29] Matrix_1.6-5              R6_2.6.0                 \n [31] fastmap_1.2.0             GenomeInfoDbData_1.2.11  \n [33] fitdistrplus_1.2-2        future_1.34.0            \n [35] shiny_1.10.0              digest_0.6.37            \n [37] colorspace_2.1-1          patchwork_1.3.0          \n [39] tensor_1.5                RSpectra_0.16-2          \n [41] irlba_2.3.5.1             beachmat_2.18.0          \n [43] labeling_0.4.3            progressr_0.15.1         \n [45] spatstat.sparse_3.1-0     polyclip_1.10-7          \n [47] httr_1.4.7                abind_1.4-5              \n [49] compiler_4.3.3            withr_3.0.2              \n [51] BiocParallel_1.36.0       viridis_0.6.5            \n [53] fastDummies_1.7.5         MASS_7.3-60.0.1          \n [55] DelayedArray_0.28.0       tools_4.3.3              \n [57] vipor_0.4.7               lmtest_0.9-40            \n [59] beeswarm_0.4.0            httpuv_1.6.15            \n [61] future.apply_1.11.3       goftest_1.2-3            \n [63] glue_1.8.0                nlme_3.1-167             \n [65] promises_1.3.2            grid_4.3.3               \n [67] Rtsne_0.17                cluster_2.1.8            \n [69] reshape2_1.4.4            generics_0.1.3           \n [71] spatstat.data_3.1-4       gtable_0.3.6             \n [73] tidyr_1.3.1               BiocSingular_1.18.0      \n [75] ScaledMatrix_1.10.0       XVector_0.42.0           \n [77] spatstat.geom_3.3-5       RcppAnnoy_0.0.22         \n [79] ggrepel_0.9.6             RANN_2.6.2               \n [81] pillar_1.10.1             stringr_1.5.1            \n [83] spam_2.11-1               RcppHNSW_0.6.0           \n [85] later_1.4.1               splines_4.3.3            \n [87] lattice_0.22-6            deldir_2.0-4             \n [89] survival_3.8-3            tidyselect_1.2.1         \n [91] miniUI_0.1.1.1            pbapply_1.7-2            \n [93] gridExtra_2.3             scattermore_1.2          \n [95] stringi_1.8.4             lazyeval_0.2.2           \n [97] codetools_0.2-20          cli_3.6.4                \n [99] uwot_0.2.2                xtable_1.8-4             \n[101] reticulate_1.40.0         munsell_0.5.1            \n[103] Rcpp_1.0.14               spatstat.random_3.3-2    \n[105] globals_0.16.3            png_0.1-8                \n[107] spatstat.univar_3.1-1     dotCall64_1.2            \n[109] sparseMatrixStats_1.14.0  bitops_1.0-9             \n[111] listenv_0.9.1             viridisLite_0.4.2        \n[113] scales_1.3.0              ggridges_0.5.6           \n[115] leiden_0.4.3.1            purrr_1.0.2              \n[117] crayon_1.5.3              fastmatch_1.1-6          \n[119] cowplot_1.1.3            \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 116.82\nuser 123.29\nsys 2.07\npercent 107%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"MetabolicPathwayHeterogeneity",desc:"Calculate Metabolic Pathway heterogeneity.",npages:1,report_toc:!0,order:31}]}]};function Fa(e){let t,n;return t=new an({props:{dir:"up"}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function za(e){let t,n;return t=new an({props:{dir:"down"}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function ka(e){let t,n,a,s,i,r=e[1]>0&&Fa();n=new Va({});let m=e[1]<e[3].npages-1&&za();return{c(){r&&r.c(),t=v(),f(n.$$.fragment),a=v(),m&&m.c(),s=g()},m(e,o){r&&r.m(e,o),l(e,t,o),b(n,e,o),l(e,a,o),m&&m.m(e,o),l(e,s,o),i=!0},p(e,n){e[1]>0?r?2&n&&o(r,1):(r=Fa(),r.c(),o(r,1),r.m(t.parentNode,t)):r&&(p(),_(r,1,1,(()=>{r=null})),c()),e[1]<e[3].npages-1?m?10&n&&o(m,1):(m=za(),m.c(),o(m,1),m.m(s.parentNode,s)):m&&(p(),_(m,1,1,(()=>{m=null})),c())},i(e){i||(o(r),o(n.$$.fragment,e),o(m),i=!0)},o(e){_(r),_(n.$$.fragment,e),_(m),i=!1},d(e){e&&(d(t),d(a),d(s)),r&&r.d(e),h(n,e),m&&m.d(e)}}}function qa(e){let t,n,a;function s(t){e[4](t)}let i={logo:e[0],logotext:e[3].desc,versions:Ga.versions,entries:Ga.entries,pipeline_name:Ga.pipeline.name,report_toc:e[3].report_toc,runinfo:e[3].runinfo,toc:Ha,page:e[1],$$slots:{default:[ka]},$$scope:{ctx:e}};return void 0!==e[2]&&(i.isSideNavOpen=e[2]),t=new Zt({props:i}),C.push((()=>y(t,"isSideNavOpen",s))),{c(){f(t.$$.fragment)},m(e,n){b(t,e,n),a=!0},p(e,[a]){const s={};1&a&&(s.logo=e[0]),8&a&&(s.logotext=e[3].desc),8&a&&(s.report_toc=e[3].report_toc),8&a&&(s.runinfo=e[3].runinfo),2&a&&(s.page=e[1]),42&a&&(s.$$scope={dirty:a,ctx:e}),!n&&4&a&&(n=!0,s.isSideNavOpen=e[2],$((()=>n=!1))),t.$set(s)},i(e){a||(o(t.$$.fragment,e),a=!0)},o(e){_(t.$$.fragment,e),a=!1},d(e){h(t,e)}}}function Oa(e,t,n){let a,{name:s}=t,{page:i=0}=t,r=!1;for(const e of Ga.entries)if(e.procs){for(const t of e.procs)if(t.name==s){a=t;break}}else if(e.name==s){a=e;break}return u((()=>{const e=new URL(window.location.href);let t;e.hash.length>1&&(t=document.getElementById(`${e.hash.slice(1)}`)),t&&t.scrollIntoView({block:"start",behavior:"smooth"})})),e.$$set=e=>{"name"in e&&n(0,s=e.name),"page"in e&&n(1,i=e.page)},[s,i,r,a,function(e){r=e,n(2,r)}]}e("default",new class extends t{constructor(e){super(),n(this,e,Oa,qa,a,{name:0,page:1})}}({target:document.body,props:{name:"MetabolicFeaturesIntraSubset",page:2}}))}}}));
