System.register(["./vendor/svelte-CiOtOeYy.js","./vendor/carbon-components-svelte-BhQO6q4b.js","./vendor/carbon-icons-svelte-DVphRbRG.js"],(function(e){"use strict";var t,n,a,s,i,r,l,o,p,c,_,d,m,u,g,f,b,h,v,C,y,x,$,S,M,A,P,L,U,T,R,E,N,w,B,I,D,V,G,H,F,z,O,q,k,j,K,Y,X,W,Q,J,Z,ee,te,ne,ae,se,ie,re,le,oe,pe,ce,_e,de,me,ue,ge,fe,be,he,ve,Ce,ye,xe,$e,Se,Me,Ae,Pe,Le,Ue,Te,Re,Ee,Ne,we,Be,Ie,De,Ve;return{setters:[function(e){t=e.S,n=e.i,a=e.s,s=e.a1,i=e.l,r=e.O,l=e.a,o=e.b,p=e.g,c=e.c,_=e.t,d=e.d,m=e.a5,u=e.B,g=e.e,f=e.G,b=e.H,h=e.I,v=e.y,C=e.K,y=e.T,x=e.U,$=e.L,S=e.D,M=e.M,A=e.z,P=e.Q,L=e.k,U=e.u,T=e.n,R=e.o,E=e.f,N=e.A,w=e.h,B=e.j,I=e.aa,D=e.x,V=e.W,G=e.ab,H=e.m,F=e.p,z=e.a0,O=e.q,q=e.R,k=e.$,j=e.v,K=e.Y,Y=e.Z,X=e.F,W=e.ac,Q=e.w,J=e.J,Z=e.r},function(e){ee=e.S,te=e.a,ne=e.b,ae=e.H,se=e.c,ie=e.T,re=e.d,le=e.e,oe=e.f,pe=e.g,ce=e.h,_e=e.M,de=e.I,me=e.i,ue=e.j,ge=e.k,fe=e.C,be=e.l,he=e.m,ve=e.G,Ce=e.R,ye=e.n,xe=e.B,$e=e.A,Se=e.o,Me=e.D,Ae=e.P,Pe=e.p,Le=e.q,Ue=e.r,Te=e.s,Re=e.t},function(e){Ee=e.C,Ne=e.a,we=e.O,Be=e.b,Ie=e.c,De=e.D,Ve=e.d}],execute:function(){function Ge(e,t,n){const a=e.slice();return a[4]=t[n],a}function He(e,t,n){const a=e.slice();return a[7]=t[n],a}function Fe(e){let t,n;return t=new ee({props:{href:`${e[1](e[4].page)}#${e[4].slug}`,class:e[2](e[4].page),title:e[4].text,text:e[4].text,expanded:!0,$$slots:{default:[qe]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.href=`${e[1](e[4].page)}#${e[4].slug}`),1&n&&(a.class=e[2](e[4].page)),1&n&&(a.title=e[4].text),1&n&&(a.text=e[4].text),1025&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function ze(e){let t,n;return t=new te({props:{href:`${e[1](e[4].page)}#${e[4].slug}`,class:e[2](e[4].page),title:e[4].text,text:e[4].text}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.href=`${e[1](e[4].page)}#${e[4].slug}`),1&n&&(a.class=e[2](e[4].page)),1&n&&(a.title=e[4].text),1&n&&(a.text=e[4].text),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function Oe(e){let t,n;return t=new ne({props:{href:`${e[1](e[7].page)}#${e[7].slug}`,class:e[2](e[7].page),title:e[7].text,text:e[7].text}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.href=`${e[1](e[7].page)}#${e[7].slug}`),1&n&&(a.class=e[2](e[7].page)),1&n&&(a.title=e[7].text),1&n&&(a.text=e[7].text),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function qe(e){let t,n,a=s(e[4].children),i=[];for(let t=0;t<a.length;t+=1)i[t]=Oe(He(e,a,t));const r=e=>_(i[e],1,1,(()=>{i[e]=null}));return{c(){for(let e=0;e<i.length;e+=1)i[e].c();t=v()},m(e,a){for(let t=0;t<i.length;t+=1)i[t]&&i[t].m(e,a);l(e,t,a),n=!0},p(e,n){if(7&n){let l;for(a=s(e[4].children),l=0;l<a.length;l+=1){const s=He(e,a,l);i[l]?(i[l].p(s,n),o(i[l],1)):(i[l]=Oe(s),i[l].c(),o(i[l],1),i[l].m(t.parentNode,t))}for(p(),l=a.length;l<i.length;l+=1)r(l);c()}},i(e){if(!n){for(let e=0;e<a.length;e+=1)o(i[e]);n=!0}},o(e){i=i.filter(Boolean);for(let e=0;e<i.length;e+=1)_(i[e]);n=!1},d(e){e&&d(t),m(i,e)}}}function ke(e){let t,n,a,s;const i=[ze,Fe],r=[];function m(e,t){return 0==e[4].children.length?0:1}return t=m(e),n=r[t]=i[t](e),{c(){n.c(),a=g()},m(e,n){r[t].m(e,n),l(e,a,n),s=!0},p(e,s){let l=t;t=m(e),t===l?r[t].p(e,s):(p(),_(r[l],1,1,(()=>{r[l]=null})),c(),n=r[t],n?n.p(e,s):(n=r[t]=i[t](e),n.c()),o(n,1),n.m(a.parentNode,a))},i(e){s||(o(n),s=!0)},o(e){_(n),s=!1},d(e){e&&d(a),r[t].d(e)}}}function je(e){let t,n,a=s(e[0]),u=[];for(let t=0;t<a.length;t+=1)u[t]=ke(Ge(e,a,t));const g=e=>_(u[e],1,1,(()=>{u[e]=null}));return{c(){t=i("ul");for(let e=0;e<u.length;e+=1)u[e].c();r(t,"class","bx--side-nav__items pipen-report-sidenav svelte-12gpq3")},m(e,a){l(e,t,a);for(let e=0;e<u.length;e+=1)u[e]&&u[e].m(t,null);n=!0},p(e,[n]){if(7&n){let i;for(a=s(e[0]),i=0;i<a.length;i+=1){const s=Ge(e,a,i);u[i]?(u[i].p(s,n),o(u[i],1)):(u[i]=ke(s),u[i].c(),o(u[i],1),u[i].m(t,null))}for(p(),i=a.length;i<u.length;i+=1)g(i);c()}},i(e){if(!n){for(let e=0;e<a.length;e+=1)o(u[e]);n=!0}},o(e){u=u.filter(Boolean);for(let e=0;e<u.length;e+=1)_(u[e]);n=!1},d(e){e&&d(t),m(u,e)}}}function Ke(e,t,n){let{toc:a}=t,{page:s}=t;return u((()=>{let e,t=new URL(window.location.href);e=t.hash.length>0?document.querySelector(`[href="${t.hash}"]`):document.querySelector(".toc-current-page"),e.scrollIntoView({behavior:"smooth"})})),e.$$set=e=>{"toc"in e&&n(0,a=e.toc),"page"in e&&n(3,s=e.page)},[a,e=>{if(s===e)return"";let t=new URL(window.location.href);return t.hash="",t.searchParams.set("page",e),t.toString()},e=>e!=s?"toc-noncurrent-page":"toc-current-page",s]}class Ye extends t{constructor(e){super(),n(this,e,Ke,je,a,{toc:0,page:3})}}function Xe(e,t,n){const a=e.slice();return a[7]=t[n],a}function We(e,t,n){const a=e.slice();return a[10]=t[n],a}function Qe(e){let t,n;return t=new le({props:{isOpen:ft,transition:e[5],$$slots:{default:[pt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8197&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function Je(e){let t;return{c(){t=$("Go to home:")},m(e,n){l(e,t,n)},d(e){e&&d(t)}}}function Ze(e){let t;return{c(){t=$(e[0])},m(e,n){l(e,t,n)},p(e,n){1&n&&M(t,e[0])},d(e){e&&d(t)}}}function et(e){let t;return{c(){t=$("Switch to process:")},m(e,n){l(e,t,n)},d(e){e&&d(t)}}}function tt(e){let t,n;return t=new ce({props:{title:e[7].name,href:`?proc=${e[7].name}`,$$slots:{default:[at]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};4&n&&(a.title=e[7].name),4&n&&(a.href=`?proc=${e[7].name}`),8196&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function nt(e){let t,n,a,i;t=new ce({props:{class:"procgroup",$$slots:{default:[st]},$$scope:{ctx:e}}});let r=s(e[7].procs),u=[];for(let t=0;t<r.length;t+=1)u[t]=rt(We(e,r,t));const C=e=>_(u[e],1,1,(()=>{u[e]=null}));return{c(){f(t.$$.fragment),n=v();for(let e=0;e<u.length;e+=1)u[e].c();a=g()},m(e,s){b(t,e,s),l(e,n,s);for(let t=0;t<u.length;t+=1)u[t]&&u[t].m(e,s);l(e,a,s),i=!0},p(e,n){const i={};if(8196&n&&(i.$$scope={dirty:n,ctx:e}),t.$set(i),4&n){let t;for(r=s(e[7].procs),t=0;t<r.length;t+=1){const s=We(e,r,t);u[t]?(u[t].p(s,n),o(u[t],1)):(u[t]=rt(s),u[t].c(),o(u[t],1),u[t].m(a.parentNode,a))}for(p(),t=r.length;t<u.length;t+=1)C(t);c()}},i(e){if(!i){o(t.$$.fragment,e);for(let e=0;e<r.length;e+=1)o(u[e]);i=!0}},o(e){_(t.$$.fragment,e),u=u.filter(Boolean);for(let e=0;e<u.length;e+=1)_(u[e]);i=!1},d(e){e&&(d(n),d(a)),h(t,e),m(u,e)}}}function at(e){let t,n,a=e[7].name+"";return{c(){t=$(a),n=v()},m(e,a){l(e,t,a),l(e,n,a)},p(e,n){4&n&&a!==(a=e[7].name+"")&&M(t,a)},d(e){e&&(d(t),d(n))}}}function st(e){let t,n=e[7].name+"";return{c(){t=$(n)},m(e,n){l(e,t,n)},p(e,a){4&a&&n!==(n=e[7].name+"")&&M(t,n)},d(e){e&&d(t)}}}function it(e){let t,n,a,s=e[10].name+"";return{c(){t=$("- "),n=$(s),a=v()},m(e,s){l(e,t,s),l(e,n,s),l(e,a,s)},p(e,t){4&t&&s!==(s=e[10].name+"")&&M(n,s)},d(e){e&&(d(t),d(n),d(a))}}}function rt(e){let t,n;return t=new ce({props:{title:e[10].name,class:"procgroup-proc",href:`?proc=${e[10].name}`,$$slots:{default:[it]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};4&n&&(a.title=e[10].name),4&n&&(a.href=`?proc=${e[10].name}`),8196&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function lt(e){let t,n,a,s;const i=[nt,tt],r=[];function m(e,t){return e[7].procs?0:1}return t=m(e),n=r[t]=i[t](e),{c(){n.c(),a=g()},m(e,n){r[t].m(e,n),l(e,a,n),s=!0},p(e,s){let l=t;t=m(e),t===l?r[t].p(e,s):(p(),_(r[l],1,1,(()=>{r[l]=null})),c(),n=r[t],n?n.p(e,s):(n=r[t]=i[t](e),n.c()),o(n,1),n.m(a.parentNode,a))},i(e){s||(o(n),s=!0)},o(e){_(n),s=!1},d(e){e&&d(a),r[t].d(e)}}}function ot(e){let t,n,a,i,r,u,C,y;t=new pe({props:{$$slots:{default:[Je]},$$scope:{ctx:e}}}),a=new ce({props:{href:"index.html",$$slots:{default:[Ze]},$$scope:{ctx:e}}}),r=new pe({props:{$$slots:{default:[et]},$$scope:{ctx:e}}});let x=s(e[2]),$=[];for(let t=0;t<x.length;t+=1)$[t]=lt(Xe(e,x,t));const S=e=>_($[e],1,1,(()=>{$[e]=null}));return{c(){f(t.$$.fragment),n=v(),f(a.$$.fragment),i=v(),f(r.$$.fragment),u=v();for(let e=0;e<$.length;e+=1)$[e].c();C=g()},m(e,s){b(t,e,s),l(e,n,s),b(a,e,s),l(e,i,s),b(r,e,s),l(e,u,s);for(let t=0;t<$.length;t+=1)$[t]&&$[t].m(e,s);l(e,C,s),y=!0},p(e,n){const i={};8192&n&&(i.$$scope={dirty:n,ctx:e}),t.$set(i);const l={};8193&n&&(l.$$scope={dirty:n,ctx:e}),a.$set(l);const _={};if(8192&n&&(_.$$scope={dirty:n,ctx:e}),r.$set(_),4&n){let t;for(x=s(e[2]),t=0;t<x.length;t+=1){const a=Xe(e,x,t);$[t]?($[t].p(a,n),o($[t],1)):($[t]=lt(a),$[t].c(),o($[t],1),$[t].m(C.parentNode,C))}for(p(),t=x.length;t<$.length;t+=1)S(t);c()}},i(e){if(!y){o(t.$$.fragment,e),o(a.$$.fragment,e),o(r.$$.fragment,e);for(let e=0;e<x.length;e+=1)o($[e]);y=!0}},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),_(r.$$.fragment,e),$=$.filter(Boolean);for(let e=0;e<$.length;e+=1)_($[e]);y=!1},d(e){e&&(d(n),d(i),d(u),d(C)),h(t,e),h(a,e),h(r,e),m($,e)}}}function pt(e){let t,n;return t=new oe({props:{$$slots:{default:[ot]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8197&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function ct(e){let t,n,a,s;t=new re({props:{render:"select",theme:"g10",persist:!0,persistKey:"__pipen-report-theme",select:{themes:["g10","white","g80","g90","g100"],labelText:"Theme",inline:!0}}});let i=e[2].length>0&&Qe(e);return{c(){f(t.$$.fragment),n=v(),i&&i.c(),a=g()},m(e,r){b(t,e,r),l(e,n,r),i&&i.m(e,r),l(e,a,r),s=!0},p(e,t){e[2].length>0?i?(i.p(e,t),4&t&&o(i,1)):(i=Qe(e),i.c(),o(i,1),i.m(a.parentNode,a)):i&&(p(),_(i,1,1,(()=>{i=null})),c())},i(e){s||(o(t.$$.fragment,e),o(i),s=!0)},o(e){_(t.$$.fragment,e),_(i),s=!1},d(e){e&&(d(n),d(a)),h(t,e),i&&i.d(e)}}}function _t(e){let t,n;return t=new se({props:{class:"header-util",$$slots:{default:[ct]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8197&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function dt(e){let t;return{c(){t=$(e[4])},m(e,n){l(e,t,n)},p(e,n){16&n&&M(t,e[4])},d(e){e&&d(t)}}}function mt(e){let t,n,a,s,r=null!==e[4]&&dt(e);return{c(){t=i("h1"),n=$(e[3]),a=v(),r&&r.c(),s=g()},m(e,i){l(e,t,i),S(t,n),l(e,a,i),r&&r.m(e,i),l(e,s,i)},p(e,t){8&t&&M(n,e[3]),null!==e[4]?r?r.p(e,t):(r=dt(e),r.c(),r.m(s.parentNode,s)):r&&(r.d(1),r=null)},d(e){e&&(d(t),d(a),d(s)),r&&r.d(e)}}}function ut(e){let t,n;return t=new ie({props:{class:"header-tile",slot:"platform",$$slots:{default:[mt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8216&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function gt(e){let t,n,a;function s(t){e[6](t)}let i={$$slots:{platform:[ut],default:[_t]},$$scope:{ctx:e}};return void 0!==e[1]&&(i.isSideNavOpen=e[1]),t=new ae({props:i}),C.push((()=>y(t,"isSideNavOpen",s))),{c(){f(t.$$.fragment)},m(e,n){b(t,e,n),a=!0},p(e,[a]){const s={};8221&a&&(s.$$scope={dirty:a,ctx:e}),!n&&2&a&&(n=!0,s.isSideNavOpen=e[1],x((()=>n=!1))),t.$set(s)},i(e){a||(o(t.$$.fragment,e),a=!0)},o(e){_(t.$$.fragment,e),a=!1},d(e){h(t,e)}}}let ft=!1;function bt(e,t,n){let{entries:a}=t,{logo:s}=t,{logotext:i=null}=t,{pipeline_name:r}=t,{isSideNavOpen:l}=t;return r=r||s,e.$$set=e=>{"entries"in e&&n(2,a=e.entries),"logo"in e&&n(3,s=e.logo),"logotext"in e&&n(4,i=e.logotext),"pipeline_name"in e&&n(0,r=e.pipeline_name),"isSideNavOpen"in e&&n(1,l=e.isSideNavOpen)},[r,l,a,s,i,{duration:200},function(e){l=e,n(1,l)}]}class ht extends t{constructor(e){super(),n(this,e,bt,gt,a,{entries:2,logo:3,logotext:4,pipeline_name:0,isSideNavOpen:1})}}function vt(e){let t,n,a;function s(t){e[3](t)}let i={modalHeading:"Running Information",passiveModal:!0,hasScrollingContent:!0,class:"runinfo",$$slots:{default:[Mt]},$$scope:{ctx:e}};return void 0!==e[2]&&(i.open=e[2]),t=new _e({props:i}),C.push((()=>y(t,"open",s))),{c(){f(t.$$.fragment)},m(e,n){b(t,e,n),a=!0},p(e,a){const s={};34&a&&(s.$$scope={dirty:a,ctx:e}),!n&&4&a&&(n=!0,s.open=e[2],x((()=>n=!1))),t.$set(s)},i(e){a||(o(t.$$.fragment,e),a=!0)},o(e){_(t.$$.fragment,e),a=!1},d(e){h(t,e)}}}function Ct(e){let t,n,a,s,i,r;return t=new ue({props:{label:"Session Information"}}),a=new ue({props:{label:"Time Spent"}}),i=new ue({props:{label:"Device Information"}}),{c(){f(t.$$.fragment),n=v(),f(a.$$.fragment),s=v(),f(i.$$.fragment)},m(e,o){b(t,e,o),l(e,n,o),b(a,e,o),l(e,s,o),b(i,e,o),r=!0},p:P,i(e){r||(o(t.$$.fragment,e),o(a.$$.fragment,e),o(i.$$.fragment,e),r=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),_(i.$$.fragment,e),r=!1},d(e){e&&(d(n),d(s)),h(t,e),h(a,e),h(i,e)}}}function yt(e){let t,n;return t=new fe({props:{type:"multi",code:e[1].session}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};2&n&&(a.code=e[1].session),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function xt(e){let t,n;return t=new fe({props:{type:"multi",code:e[1].time}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};2&n&&(a.code=e[1].time),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function $t(e){let t,n;return t=new fe({props:{type:"multi",code:e[1].device}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};2&n&&(a.code=e[1].device),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function St(e){let t,n,a,s,i,r;return t=new ge({props:{$$slots:{default:[yt]},$$scope:{ctx:e}}}),a=new ge({props:{$$slots:{default:[xt]},$$scope:{ctx:e}}}),i=new ge({props:{$$slots:{default:[$t]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=v(),f(a.$$.fragment),s=v(),f(i.$$.fragment)},m(e,o){b(t,e,o),l(e,n,o),b(a,e,o),l(e,s,o),b(i,e,o),r=!0},p(e,n){const s={};34&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};34&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r);const l={};34&n&&(l.$$scope={dirty:n,ctx:e}),i.$set(l)},i(e){r||(o(t.$$.fragment,e),o(a.$$.fragment,e),o(i.$$.fragment,e),r=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),_(i.$$.fragment,e),r=!1},d(e){e&&(d(n),d(s)),h(t,e),h(a,e),h(i,e)}}}function Mt(e){let t,n,a,s;return t=new de({props:{kind:"warning",lowContrast:!0,hideCloseButton:!0,title:"Showing the first job only. Check the workdir for information of other jobs if any."}}),a=new me({props:{$$slots:{content:[St],default:[Ct]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=v(),f(a.$$.fragment)},m(e,i){b(t,e,i),l(e,n,i),b(a,e,i),s=!0},p(e,t){const n={};34&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),h(t,e),h(a,e)}}}function At(e){let t,n,a;return{c(){t=i("a"),t.textContent="> Running Information",r(t,"href","javascript:void(0);")},m(s,i){l(s,t,i),n||(a=A(t,"click",e[4]),n=!0)},p:P,d(e){e&&d(t),n=!1,a()}}}function Pt(e){let t,n,a,s,m,u,g,f,b=e[1]&&vt(e),h=e[1]&&At(e);return{c(){b&&b.c(),t=v(),n=i("div"),a=i("div"),h&&h.c(),s=v(),m=i("div"),u=$("Powered by "),g=i("span"),r(a,"class","runinfo"),r(g,"class","pipen-report-footer-versions"),r(m,"class","versions"),r(n,"class","pipen-report-footer bx--grid")},m(i,r){b&&b.m(i,r),l(i,t,r),l(i,n,r),S(n,a),h&&h.m(a,null),S(n,s),S(n,m),S(m,u),S(m,g),g.innerHTML=e[0],f=!0},p(e,[n]){e[1]?b?(b.p(e,n),2&n&&o(b,1)):(b=vt(e),b.c(),o(b,1),b.m(t.parentNode,t)):b&&(p(),_(b,1,1,(()=>{b=null})),c()),e[1]?h?h.p(e,n):(h=At(e),h.c(),h.m(a,null)):h&&(h.d(1),h=null),(!f||1&n)&&(g.innerHTML=e[0])},i(e){f||(o(b),f=!0)},o(e){_(b),f=!1},d(e){e&&(d(t),d(n)),b&&b.d(e),h&&h.d()}}}function Lt(e,t,n){let{versions:a}=t,{runinfo:s}=t,i=!1;return e.$$set=e=>{"versions"in e&&n(0,a=e.versions),"runinfo"in e&&n(1,s=e.runinfo)},[a,s,i,function(e){i=e,n(2,i)},()=>{n(2,i=!0)}]}class Ut extends t{constructor(e){super(),n(this,e,Lt,Pt,a,{versions:0,runinfo:1})}}function Tt(e){let t,n,a;function s(t){e[12](t)}let i={$$slots:{default:[Rt]},$$scope:{ctx:e}};return void 0!==e[0]&&(i.isOpen=e[0]),t=new he({props:i}),C.push((()=>y(t,"isOpen",s))),{c(){f(t.$$.fragment)},m(e,n){b(t,e,n),a=!0},p(e,a){const s={};8576&a&&(s.$$scope={dirty:a,ctx:e}),!n&&1&a&&(n=!0,s.isOpen=e[0],x((()=>n=!1))),t.$set(s)},i(e){a||(o(t.$$.fragment,e),a=!0)},o(e){_(t.$$.fragment,e),a=!1},d(e){h(t,e)}}}function Rt(e){let t,n;return t=new Ye({props:{toc:e[7],page:e[8]}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};128&n&&(a.toc=e[7]),256&n&&(a.page=e[8]),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function Et(e){let t;const n=e[10].default,a=L(n,e,e[13],null);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||8192&s)&&U(a,n,e,e[13],t?R(n,e[13],s,null):T(e[13]),null)},i(e){t||(o(a,e),t=!0)},o(e){_(a,e),t=!1},d(e){a&&a.d(e)}}}function Nt(e){let t,n;return t=new ye({props:{$$slots:{default:[Et]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8192&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function wt(e){let t,n;return t=new Ce({props:{$$slots:{default:[Nt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8192&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function Bt(e){let t,n,a,s;return t=new ve({props:{$$slots:{default:[wt]},$$scope:{ctx:e}}}),a=new Ut({props:{versions:e[3],runinfo:e[9]}}),{c(){f(t.$$.fragment),n=v(),f(a.$$.fragment)},m(e,i){b(t,e,i),l(e,n,i),b(a,e,i),s=!0},p(e,n){const s={};8192&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const i={};8&n&&(i.versions=e[3]),512&n&&(i.runinfo=e[9]),a.$set(i)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),h(t,e),h(a,e)}}}function It(e){let t,n,a,s,i,r;function m(t){e[11](t)}let u={logo:e[1],logotext:e[2],entries:e[4],pipeline_name:e[5]};void 0!==e[0]&&(u.isSideNavOpen=e[0]),t=new ht({props:u}),C.push((()=>y(t,"isSideNavOpen",m)));let g=e[6]&&e[7].length>0&&Tt(e);return i=new be({props:{$$slots:{default:[Bt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),a=v(),g&&g.c(),s=v(),f(i.$$.fragment)},m(e,n){b(t,e,n),l(e,a,n),g&&g.m(e,n),l(e,s,n),b(i,e,n),r=!0},p(e,[a]){const r={};2&a&&(r.logo=e[1]),4&a&&(r.logotext=e[2]),16&a&&(r.entries=e[4]),32&a&&(r.pipeline_name=e[5]),!n&&1&a&&(n=!0,r.isSideNavOpen=e[0],x((()=>n=!1))),t.$set(r),e[6]&&e[7].length>0?g?(g.p(e,a),192&a&&o(g,1)):(g=Tt(e),g.c(),o(g,1),g.m(s.parentNode,s)):g&&(p(),_(g,1,1,(()=>{g=null})),c());const l={};8712&a&&(l.$$scope={dirty:a,ctx:e}),i.$set(l)},i(e){r||(o(t.$$.fragment,e),o(g),o(i.$$.fragment,e),r=!0)},o(e){_(t.$$.fragment,e),_(g),_(i.$$.fragment,e),r=!1},d(e){e&&(d(a),d(s)),h(t,e),g&&g.d(e),h(i,e)}}}function Dt(e,t,n){let a,{$$slots:s={},$$scope:i}=t,{logo:r}=t,{logotext:l=null}=t,{versions:o}=t,{entries:p}=t,{pipeline_name:c}=t,{report_toc:_}=t,{toc:d}=t,{page:m}=t,{isSideNavOpen:g}=t,{runinfo:f}=t;return u((()=>{const e=new URL(window.location.href);let t;t=e.hash.length>0?document.querySelector(`li:has(>[href="${e.hash}"])`):document.querySelector("li:has(>.toc-current-page)"),t&&(t.classList.add("toc-active"),t.scrollIntoView());const n=document.querySelectorAll(".pipen-report-toc-anchor");a=new IntersectionObserver((e=>{e.forEach((e=>{const t=e.target.getAttribute("id"),n=document.querySelector(`li:has(>[href="#${t}"])`);n&&(e.isIntersecting?n.classList.add("toc-active"):n.classList.remove("toc-active"))}))}),{root:null,rootMargin:"0px",threshold:.5}),n.forEach((e=>{a.observe(e)}))})),e.$$set=e=>{"logo"in e&&n(1,r=e.logo),"logotext"in e&&n(2,l=e.logotext),"versions"in e&&n(3,o=e.versions),"entries"in e&&n(4,p=e.entries),"pipeline_name"in e&&n(5,c=e.pipeline_name),"report_toc"in e&&n(6,_=e.report_toc),"toc"in e&&n(7,d=e.toc),"page"in e&&n(8,m=e.page),"isSideNavOpen"in e&&n(0,g=e.isSideNavOpen),"runinfo"in e&&n(9,f=e.runinfo),"$$scope"in e&&n(13,i=e.$$scope)},[g,r,l,o,p,c,_,d,m,f,s,function(e){g=e,n(0,g)},function(e){g=e,n(0,g)},i]}class Vt extends t{constructor(e){super(),n(this,e,Dt,It,a,{logo:1,logotext:2,versions:3,entries:4,pipeline_name:5,report_toc:6,toc:7,page:8,isSideNavOpen:0,runinfo:9})}}function Gt(e){let t;return{c(){t=$(e[2])},m(e,n){l(e,t,n)},p:P,d(e){e&&d(t)}}}function Ht(e){let t,n,a,s;return n=new xe({props:{icon:e[1],kind:"tertiary",href:e[3](),$$slots:{default:[Gt]},$$scope:{ctx:e}}}),{c(){t=i("div"),f(n.$$.fragment),r(t,"class",a=`pagenav ${e[0]}`)},m(e,a){l(e,t,a),b(n,t,null),s=!0},p(e,[i]){const l={};16&i&&(l.$$scope={dirty:i,ctx:e}),n.$set(l),(!s||1&i&&a!==(a=`pagenav ${e[0]}`))&&r(t,"class",a)},i(e){s||(o(n.$$.fragment,e),s=!0)},o(e){_(n.$$.fragment,e),s=!1},d(e){e&&d(t),h(n)}}}function Ft(e,t,n){let{dir:a}=t,s="up"===a?Ee:Ne,i="up"===a?"Go to previous page":"Go to next page";return e.$$set=e=>{"dir"in e&&n(0,a=e.dir)},[a,s,i,()=>{let e=new URL(window.location.href);e.hash="";const t=parseInt(e.searchParams.get("page")||0),n="up"===a?t-1:t+1;return 0===n?e.searchParams.delete("page"):e.searchParams.set("page",n),e.toString()}]}class zt extends t{constructor(e){super(),n(this,e,Ft,Ht,a,{dir:0})}}const Ot=e=>({}),qt=e=>({}),kt=e=>({}),jt=e=>({}),Kt=e=>({}),Yt=e=>({}),Xt=e=>({}),Wt=e=>({});function Qt(e){let t,n;return t=new $e({props:{ratio:e[5],$$slots:{default:[nn]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};32&n&&(a.ratio=e[5]),65631&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function Jt(e){let t,n,a,s,i=e[2]&&an(e),r=e[0]&&sn(e),m=e[1]&&rn(e);return{c(){i&&i.c(),t=v(),r&&r.c(),n=v(),m&&m.c(),a=g()},m(e,o){i&&i.m(e,o),l(e,t,o),r&&r.m(e,o),l(e,n,o),m&&m.m(e,o),l(e,a,o),s=!0},p(e,s){e[2]?i?(i.p(e,s),4&s&&o(i,1)):(i=an(e),i.c(),o(i,1),i.m(t.parentNode,t)):i&&(p(),_(i,1,1,(()=>{i=null})),c()),e[0]?r?(r.p(e,s),1&s&&o(r,1)):(r=sn(e),r.c(),o(r,1),r.m(n.parentNode,n)):r&&(p(),_(r,1,1,(()=>{r=null})),c()),e[1]?m?(m.p(e,s),2&s&&o(m,1)):(m=rn(e),m.c(),o(m,1),m.m(a.parentNode,a)):m&&(p(),_(m,1,1,(()=>{m=null})),c())},i(e){s||(o(i),o(r),o(m),s=!0)},o(e){_(i),_(r),_(m),s=!1},d(e){e&&(d(t),d(n),d(a)),i&&i.d(e),r&&r.d(e),m&&m.d(e)}}}function Zt(e){let t;const n=e[12].loading,a=L(n,e,e[16],jt);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||65536&s)&&U(a,n,e,e[16],t?R(n,e[16],s,kt):T(e[16]),jt)},i(e){t||(o(a,e),t=!0)},o(e){_(a,e),t=!1},d(e){a&&a.d(e)}}}function en(e){let t,n,a,s,r,o,p=[e[9],{src:n=e[3]},{alt:e[4]},{title:"Click to zoom in ..."}],c={};for(let e=0;e<p.length;e+=1)c=w(c,p[e]);return{c(){t=i("img"),H(t,c)},m(n,a){l(n,t,a),s=!0,r||(o=A(t,"click",e[8]),r=!0)},p(a,i){H(t,c=F(p,[(e=a)[9],(!s||8&i&&!I(t.src,n=e[3]))&&{src:n},(!s||16&i)&&{alt:e[4]},{title:"Click to zoom in ..."}]))},i(n){s||(n&&D((()=>{s&&(a||(a=V(t,G,{duration:e[6]?pn:0},!0)),a.run(1))})),s=!0)},o(n){n&&(a||(a=V(t,G,{duration:e[6]?pn:0},!1)),a.run(0)),s=!1},d(e){e&&d(t),e&&a&&a.end(),r=!1,o()}}}function tn(e){let t;const n=e[12].error,a=L(n,e,e[16],qt);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||65536&s)&&U(a,n,e,e[16],t?R(n,e[16],s,Ot):T(e[16]),qt)},i(e){t||(o(a,e),t=!0)},o(e){_(a,e),t=!1},d(e){a&&a.d(e)}}}function nn(e){let t,n,a,s,i=e[2]&&Zt(e),r=e[0]&&en(e),m=e[1]&&tn(e);return{c(){i&&i.c(),t=v(),r&&r.c(),n=v(),m&&m.c(),a=g()},m(e,o){i&&i.m(e,o),l(e,t,o),r&&r.m(e,o),l(e,n,o),m&&m.m(e,o),l(e,a,o),s=!0},p(e,s){e[2]?i?(i.p(e,s),4&s&&o(i,1)):(i=Zt(e),i.c(),o(i,1),i.m(t.parentNode,t)):i&&(p(),_(i,1,1,(()=>{i=null})),c()),e[0]?r?(r.p(e,s),1&s&&o(r,1)):(r=en(e),r.c(),o(r,1),r.m(n.parentNode,n)):r&&(p(),_(r,1,1,(()=>{r=null})),c()),e[1]?m?(m.p(e,s),2&s&&o(m,1)):(m=tn(e),m.c(),o(m,1),m.m(a.parentNode,a)):m&&(p(),_(m,1,1,(()=>{m=null})),c())},i(e){s||(o(i),o(r),o(m),s=!0)},o(e){_(i),_(r),_(m),s=!1},d(e){e&&(d(t),d(n),d(a)),i&&i.d(e),r&&r.d(e),m&&m.d(e)}}}function an(e){let t;const n=e[12].loading,a=L(n,e,e[16],Wt);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||65536&s)&&U(a,n,e,e[16],t?R(n,e[16],s,Xt):T(e[16]),Wt)},i(e){t||(o(a,e),t=!0)},o(e){_(a,e),t=!1},d(e){a&&a.d(e)}}}function sn(e){let t,n,a,s,r,o,p=[e[9],{src:n=e[3]},{alt:e[4]},{title:"Click to zoom in ..."}],c={};for(let e=0;e<p.length;e+=1)c=w(c,p[e]);return{c(){t=i("img"),H(t,c)},m(n,a){l(n,t,a),s=!0,r||(o=A(t,"click",e[8]),r=!0)},p(a,i){H(t,c=F(p,[(e=a)[9],(!s||8&i&&!I(t.src,n=e[3]))&&{src:n},(!s||16&i)&&{alt:e[4]},{title:"Click to zoom in ..."}]))},i(n){s||(n&&D((()=>{s&&(a||(a=V(t,G,{duration:e[6]?pn:0},!0)),a.run(1))})),s=!0)},o(n){n&&(a||(a=V(t,G,{duration:e[6]?pn:0},!1)),a.run(0)),s=!1},d(e){e&&d(t),e&&a&&a.end(),r=!1,o()}}}function rn(e){let t;const n=e[12].error,a=L(n,e,e[16],Yt);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||65536&s)&&U(a,n,e,e[16],t?R(n,e[16],s,Kt):T(e[16]),Yt)},i(e){t||(o(a,e),t=!0)},o(e){_(a,e),t=!1},d(e){a&&a.d(e)}}}function ln(e){let t,n,a,s;return{c(){t=i("img"),r(t,"class","pipen-report-image-lg"),I(t.src,n=e[3])||r(t,"src",n),r(t,"alt",e[4])},m(e,n){l(e,t,n),s=!0},p(a,i){e=a,(!s||8&i&&!I(t.src,n=e[3]))&&r(t,"src",n),(!s||16&i)&&r(t,"alt",e[4])},i(n){s||(n&&D((()=>{s&&(a||(a=V(t,G,{duration:e[6]?pn:0},!0)),a.run(1))})),s=!0)},o(n){n&&(a||(a=V(t,G,{duration:e[6]?pn:0},!1)),a.run(0)),s=!1},d(e){e&&d(t),e&&a&&a.end()}}}function on(e){let t,n,a,s,i,r;const m=[Jt,Qt],u=[];function g(e,t){return void 0===e[5]?0:1}function $(t){e[13](t)}t=g(e),n=u[t]=m[t](e);let S={modalHeading:e[4],primaryButtonText:"Open in new window ...",secondaryButtonText:"Close",$$slots:{default:[ln]},$$scope:{ctx:e}};return void 0!==e[7]&&(S.open=e[7]),s=new _e({props:S}),C.push((()=>y(s,"open",$))),s.$on("click:button--secondary",e[14]),s.$on("click:button--primary",e[15]),{c(){n.c(),a=v(),f(s.$$.fragment)},m(e,n){u[t].m(e,n),l(e,a,n),b(s,e,n),r=!0},p(e,[r]){let l=t;t=g(e),t===l?u[t].p(e,r):(p(),_(u[l],1,1,(()=>{u[l]=null})),c(),n=u[t],n?n.p(e,r):(n=u[t]=m[t](e),n.c()),o(n,1),n.m(a.parentNode,a));const d={};16&r&&(d.modalHeading=e[4]),65624&r&&(d.$$scope={dirty:r,ctx:e}),!i&&128&r&&(i=!0,d.open=e[7],x((()=>i=!1))),s.$set(d)},i(e){r||(o(n),o(s.$$.fragment,e),r=!0)},o(e){_(n),_(s.$$.fragment,e),r=!1},d(e){e&&d(a),u[t].d(e),h(s,e)}}}const pn=110;function cn(e,t,n){const a=["src","alt","ratio","loading","loaded","error","fadeIn","border","loadImage"];let s=E(t,a),{$$slots:i={},$$scope:r}=t,{src:l=""}=t,{alt:o=""}=t,{ratio:p}=t,{loading:c=!1}=t,{loaded:_=!1}=t,{error:d=!1}=t,{fadeIn:m=!1}=t,{border:g=!0}=t,f=!1;const b=e=>{null!=v&&(v=null),n(0,_=!1),n(1,d=!1),v=new Image,v.src=e||l,v.onload=()=>n(0,_=!0),v.onerror=()=>n(1,d=!0)},h=N();let v=null;u((()=>()=>v=null));const C={...s,class:[s.class,"pipen-report-image",g?"":"pipen-report-image-noborder"].filter(Boolean).join(" ")};return e.$$set=e=>{t=w(w({},t),B(e)),n(19,s=E(t,a)),"src"in e&&n(3,l=e.src),"alt"in e&&n(4,o=e.alt),"ratio"in e&&n(5,p=e.ratio),"loading"in e&&n(2,c=e.loading),"loaded"in e&&n(0,_=e.loaded),"error"in e&&n(1,d=e.error),"fadeIn"in e&&n(6,m=e.fadeIn),"border"in e&&n(10,g=e.border),"$$scope"in e&&n(16,r=e.$$scope)},e.$$.update=()=>{3&e.$$.dirty&&n(2,c=!_&&!d),8&e.$$.dirty&&l&&"undefined"!=typeof window&&b(),1&e.$$.dirty&&_&&h("load"),2&e.$$.dirty&&d&&h("error")},[_,d,c,l,o,p,m,f,()=>{n(7,f=!0)},C,g,b,i,function(e){f=e,n(7,f)},()=>{n(7,f=!1)},()=>{n(7,f=!1),window.open(l)},r]}class _n extends t{constructor(e){super(),n(this,e,cn,on,a,{src:3,alt:4,ratio:5,loading:2,loaded:0,error:1,fadeIn:6,border:10,loadImage:11})}get loadImage(){return this.$$.ctx[11]}}function dn(e){let t,n;const a=[{src:e[1]},e[4]];let s={$$slots:{error:[gn],loading:[un]},$$scope:{ctx:e}};for(let e=0;e<a.length;e+=1)s=w(s,a[e]);return t=new _n({props:s}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const s=18&n?F(a,[2&n&&{src:e[1]},16&n&&z(e[4])]):{};256&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function mn(e){let t,n,a;const s=[{src:e[1]},e[4]];let r={$$slots:{error:[bn],loading:[fn]},$$scope:{ctx:e}};for(let e=0;e<s.length;e+=1)r=w(r,s[e]);n=new _n({props:r});let p=[e[0]],c={};for(let e=0;e<p.length;e+=1)c=w(c,p[e]);return{c(){t=i("div"),f(n.$$.fragment),H(t,c)},m(e,s){l(e,t,s),b(n,t,null),a=!0},p(e,a){const i=18&a?F(s,[2&a&&{src:e[1]},16&a&&z(e[4])]):{};256&a&&(i.$$scope={dirty:a,ctx:e}),n.$set(i),H(t,c=F(p,[1&a&&e[0]]))},i(e){a||(o(n.$$.fragment,e),a=!0)},o(e){_(n.$$.fragment,e),a=!1},d(e){e&&d(t),h(n)}}}function un(e){let t,n,a=[e[2]],s={};for(let e=0;e<a.length;e+=1)s=w(s,a[e]);return{c(){t=i("div"),n=$("Image loading ..."),H(t,s)},m(e,a){l(e,t,a),S(t,n)},p:P,d(e){e&&d(t)}}}function gn(e){let t,n,a=[e[3]],s={};for(let e=0;e<a.length;e+=1)s=w(s,a[e]);return{c(){t=i("div"),n=$("Image loading error!"),H(t,s)},m(e,a){l(e,t,a),S(t,n)},p:P,d(e){e&&d(t)}}}function fn(e){let t,n,a=[e[2]],s={};for(let e=0;e<a.length;e+=1)s=w(s,a[e]);return{c(){t=i("div"),n=$("Image loading ..."),H(t,s)},m(e,a){l(e,t,a),S(t,n)},p:P,d(e){e&&d(t)}}}function bn(e){let t,n,a=[e[3]],s={};for(let e=0;e<a.length;e+=1)s=w(s,a[e]);return{c(){t=i("div"),n=$("Image loading error!"),H(t,s)},m(e,a){l(e,t,a),S(t,n)},p:P,d(e){e&&d(t)}}}function hn(e){let t,n,a,s;const i=[mn,dn],r=[];function m(e,t){return e[0]?0:1}return t=m(e),n=r[t]=i[t](e),{c(){n.c(),a=g()},m(e,n){r[t].m(e,n),l(e,a,n),s=!0},p(e,[s]){let l=t;t=m(e),t===l?r[t].p(e,s):(p(),_(r[l],1,1,(()=>{r[l]=null})),c(),n=r[t],n?n.p(e,s):(n=r[t]=i[t](e),n.c()),o(n,1),n.m(a.parentNode,a))},i(e){s||(o(n),s=!0)},o(e){_(n),s=!1},d(e){e&&d(a),r[t].d(e)}}}function vn(e,t,n){const a=["frameProps","width","height","src"];let s=E(t,a),{frameProps:i={class:"pipen-report-image-frame"}}=t,{width:r=300}=t,{height:l=200}=t,{src:o}=t,p=`width: ${r}px; height: ${l}px; aspect-ratio: ${r} / ${l}`,c={...s,class:[s.class,"pipen-report-image-loading"].filter(Boolean).join(" "),style:s.style?`${s.style}; ${p}`:p},_={...c,class:[c.class,"error"].join(" ")};return e.$$set=e=>{t=w(w({},t),B(e)),n(4,s=E(t,a)),"frameProps"in e&&n(0,i=e.frameProps),"width"in e&&n(5,r=e.width),"height"in e&&n(6,l=e.height),"src"in e&&n(1,o=e.src)},[i,o,c,_,s,r,l]}let Cn=class extends t{constructor(e){super(),n(this,e,vn,hn,a,{frameProps:0,width:5,height:6,src:1})}};const yn=e=>({}),xn=e=>({});function $n(e){let t,n;const a=e[25].default,s=L(a,e,e[24],null);return{c(){t=i("ul"),s&&s.c(),r(t,"role","menu"),r(t,"tabindex","-1"),r(t,"aria-label",e[14]),r(t,"data-floating-menu-direction",e[5]),r(t,"class",e[9]),O(t,"bx--overflow-menu-options",!0),O(t,"bx--overflow-menu--flip",e[7]),O(t,"bx--overflow-menu-options--open",e[0]),O(t,"bx--overflow-menu-options--light",e[6]),O(t,"bx--overflow-menu-options--sm","sm"===e[4]),O(t,"bx--overflow-menu-options--xl","xl"===e[4]),O(t,"bx--breadcrumb-menu-options",!!e[15])},m(a,i){l(a,t,i),s&&s.m(t,null),e[32](t),n=!0},p(e,i){s&&s.p&&(!n||16777216&i[0])&&U(s,a,e,e[24],n?R(a,e[24],i,null):T(e[24]),null),(!n||16384&i[0])&&r(t,"aria-label",e[14]),(!n||32&i[0])&&r(t,"data-floating-menu-direction",e[5]),(!n||512&i[0])&&r(t,"class",e[9]),(!n||512&i[0])&&O(t,"bx--overflow-menu-options",!0),(!n||640&i[0])&&O(t,"bx--overflow-menu--flip",e[7]),(!n||513&i[0])&&O(t,"bx--overflow-menu-options--open",e[0]),(!n||576&i[0])&&O(t,"bx--overflow-menu-options--light",e[6]),(!n||528&i[0])&&O(t,"bx--overflow-menu-options--sm","sm"===e[4]),(!n||528&i[0])&&O(t,"bx--overflow-menu-options--xl","xl"===e[4]),(!n||33280&i[0])&&O(t,"bx--breadcrumb-menu-options",!!e[15])},i(e){n||(o(s,e),n=!0)},o(e){_(s,e),n=!1},d(n){n&&d(t),s&&s.d(n),e[32](null)}}}function Sn(e){let t,n,a,s,r,m,u,C;const y=e[25].menu,x=L(y,e,e[24],xn),$=x||function(e){let t,n,a;var s=e[1];function i(e,t){return{props:{"aria-label":e[11],title:e[11],class:"bx--overflow-menu__icon "+e[10]}}}return s&&(t=X(s,i(e))),{c(){t&&f(t.$$.fragment),n=g()},m(e,s){t&&b(t,e,s),l(e,n,s),a=!0},p(e,a){if(2&a[0]&&s!==(s=e[1])){if(t){p();const e=t;_(e.$$.fragment,1,0,(()=>{h(e,1)})),c()}s?(t=X(s,i(e)),f(t.$$.fragment),o(t.$$.fragment,1),b(t,n.parentNode,n)):t=null}else if(s){const n={};2048&a[0]&&(n["aria-label"]=e[11]),2048&a[0]&&(n.title=e[11]),1024&a[0]&&(n.class="bx--overflow-menu__icon "+e[10]),t.$set(n)}},i(e){a||(t&&o(t.$$.fragment,e),a=!0)},o(e){t&&_(t.$$.fragment,e),a=!1},d(e){e&&d(n),t&&h(t,e)}}}(e);let M=e[0]&&$n(e),P=[{type:"button"},{"aria-haspopup":""},{"aria-expanded":e[0]},{"aria-label":e[14]},{id:e[12]},e[20]],E={};for(let e=0;e<P.length;e+=1)E=w(E,P[e]);return{c(){t=new W(!1),n=g(),a=v(),s=i("button"),$&&$.c(),r=v(),M&&M.c(),t.a=n,H(s,E),O(s,"bx--overflow-menu",!0),O(s,"bx--overflow-menu--open",e[0]),O(s,"bx--overflow-menu--light",e[6]),O(s,"bx--overflow-menu--sm","sm"===e[4]),O(s,"bx--overflow-menu--xl","xl"===e[4])},m(i,o){t.m(e[13],document.head),S(document.head,n),l(i,a,o),l(i,s,o),$&&$.m(s,null),S(s,r),M&&M.m(s,null),s.autofocus&&s.focus(),e[33](s),m=!0,u||(C=[A(window,"click",e[31]),A(s,"click",e[26]),A(s,"click",e[34]),A(s,"mouseover",e[27]),A(s,"mouseenter",e[28]),A(s,"mouseleave",e[29]),A(s,"keydown",e[30]),A(s,"keydown",e[35]),A(s,"focusout",e[36])],u=!0)},p(e,n){(!m||8192&n[0])&&t.p(e[13]),x?x.p&&(!m||16777216&n[0])&&U(x,y,e,e[24],m?R(y,e[24],n,yn):T(e[24]),xn):$&&$.p&&(!m||3074&n[0])&&$.p(e,m?n:[-1,-1]),e[0]?M?(M.p(e,n),1&n[0]&&o(M,1)):(M=$n(e),M.c(),o(M,1),M.m(s,null)):M&&(p(),_(M,1,1,(()=>{M=null})),c()),H(s,E=F(P,[{type:"button"},{"aria-haspopup":""},(!m||1&n[0])&&{"aria-expanded":e[0]},(!m||16384&n[0])&&{"aria-label":e[14]},(!m||4096&n[0])&&{id:e[12]},1048576&n[0]&&e[20]])),O(s,"bx--overflow-menu",!0),O(s,"bx--overflow-menu--open",e[0]),O(s,"bx--overflow-menu--light",e[6]),O(s,"bx--overflow-menu--sm","sm"===e[4]),O(s,"bx--overflow-menu--xl","xl"===e[4])},i(e){m||(o($,e),o(M),m=!0)},o(e){_($,e),_(M),m=!1},d(i){i&&(t.d(),d(a),d(s)),d(n),$&&$.d(i),M&&M.d(),e[33](null),u=!1,q(C)}}}function Mn(e,t,n){let a,s;const i=["size","direction","open","light","flipped","persistentClickItems","menuOptionsClass","icon","iconClass","iconDescription","id","buttonRef","menuRef"];let r,l,o,p=E(t,i),{$$slots:c={},$$scope:_}=t,{size:d}=t,{direction:m="bottom"}=t,{open:u=!1}=t,{light:g=!1}=t,{flipped:f=!1}=t,{persistentClickItems:b=!1}=t,{menuOptionsClass:h}=t,{icon:v=Be}=t,{iconClass:y}=t,{iconDescription:x="Open and close list of options"}=t,{id:$="ccs-"+Math.random().toString(36)}=t,{buttonRef:S=null}=t,{menuRef:M=null}=t;const A=k("BreadcrumbItem"),P=N(),L=Q([]);j(e,L,(e=>n(23,l=e)));const U=Q(void 0);j(e,U,(e=>n(38,o=e)));const T=Q(void 0),R=Q(-1);let I;j(e,R,(e=>n(22,r=e)));let D=!0;K("OverflowMenu",{focusedId:T,add:({id:e,text:t,primaryFocus:n,disabled:a})=>{L.update((s=>(n&&R.set(s.length),[...s,{id:e,text:t,primaryFocus:n,disabled:a,index:s.length}])))},update:e=>{U.set(e)},change:e=>{let t=r+e;t<0?t=l.length-1:t>=l.length&&(t=0);let n=l[t].disabled;for(;n;)t+=e,t<0?t=l.length-1:t>=l.length&&(t=0),n=l[t].disabled;R.set(t)}}),Y((()=>{if(o){const{index:e,text:t}=l.filter((e=>e.id===o))[0];P("close",{index:e,text:t}),n(0,u=!1)}if(u){const{width:e,height:t}=S.getBoundingClientRect();n(21,I=e),!D&&r<0&&M.focus(),f&&(n(3,M.style.left="auto",M),n(3,M.style.right=0,M)),"top"===m?(n(3,M.style.top="auto",M),n(3,M.style.bottom=t+"px",M)):"bottom"===m&&n(3,M.style.top=t+"px",M),A&&(n(3,M.style.top=t+10+"px",M),n(3,M.style.left="-11px",M))}u||(L.set([]),U.set(void 0),R.set(0)),D=!1}));return e.$$set=e=>{n(40,t=w(w({},t),B(e))),n(20,p=E(t,i)),"size"in e&&n(4,d=e.size),"direction"in e&&n(5,m=e.direction),"open"in e&&n(0,u=e.open),"light"in e&&n(6,g=e.light),"flipped"in e&&n(7,f=e.flipped),"persistentClickItems"in e&&n(8,b=e.persistentClickItems),"menuOptionsClass"in e&&n(9,h=e.menuOptionsClass),"icon"in e&&n(1,v=e.icon),"iconClass"in e&&n(10,y=e.iconClass),"iconDescription"in e&&n(11,x=e.iconDescription),"id"in e&&n(12,$=e.id),"buttonRef"in e&&n(2,S=e.buttonRef),"menuRef"in e&&n(3,M=e.menuRef),"$$scope"in e&&n(24,_=e.$$scope)},e.$$.update=()=>{n(14,a=t["aria-label"]||"menu"),12582912&e.$$.dirty[0]&&l[r]&&T.set(l[r].id),2101248&e.$$.dirty[0]&&n(13,s=`<style>\n    #${$} .bx--overflow-menu-options.bx--overflow-menu-options:after {\n      width: ${I?I+"px":"2rem"};\n    }\n  </style>`)},A&&n(1,v=we),t=B(t),[u,v,S,M,d,m,g,f,b,h,y,x,$,s,a,A,P,L,U,R,p,I,r,l,_,c,function(t){J.call(this,e,t)},function(t){J.call(this,e,t)},function(t){J.call(this,e,t)},function(t){J.call(this,e,t)},function(t){J.call(this,e,t)},({target:e})=>{S&&S.contains(e)||M&&!M.contains(e)&&n(0,u=!1)},function(e){C[e?"unshift":"push"]((()=>{M=e,n(3,M)}))},function(e){C[e?"unshift":"push"]((()=>{S=e,n(2,S)}))},e=>{b&&e.stopPropagation(),M&&M.contains(e.target)||(n(0,u=!u),u||P("close"))},e=>{u&&(["ArrowDown","ArrowLeft","ArrowRight","ArrowUp"].includes(e.key)?e.preventDefault():"Escape"===e.key&&(e.stopPropagation(),P("close"),n(0,u=!1),S.focus()))},e=>{u&&(S.contains(e.relatedTarget)||(P("close"),n(0,u=!1)))}]}class An extends t{constructor(e){super(),n(this,e,Mn,Sn,a,{size:4,direction:5,open:0,light:6,flipped:7,persistentClickItems:8,menuOptionsClass:9,icon:1,iconClass:10,iconDescription:11,id:12,buttonRef:2,menuRef:3},null,[-1,-1])}}function Pn(e,t,n){const a=e.slice();return a[3]=t[n],a[4]=t,a[5]=n,a}function Ln(e){let t,n,a,s,p;function c(t){e[1](t,e[3])}let m={labelText:e[3].value};return void 0!==e[3].selected&&(m.checked=e[3].selected),n=new Se({props:m}),C.push((()=>y(n,"checked",c))),n.$on("change",e[2]),{c(){t=i("li"),f(n.$$.fragment),s=v(),r(t,"class","pipen-report-column-selector-item svelte-147jw2r")},m(e,a){l(e,t,a),b(n,t,null),S(t,s),p=!0},p(t,s){e=t;const i={};1&s&&(i.labelText=e[3].value),!a&&1&s&&(a=!0,i.checked=e[3].selected,x((()=>a=!1))),n.$set(i)},i(e){p||(o(n.$$.fragment,e),p=!0)},o(e){_(n.$$.fragment,e),p=!1},d(e){e&&d(t),h(n)}}}function Un(e){let t,n,a=s(e[0]),i=[];for(let t=0;t<a.length;t+=1)i[t]=Ln(Pn(e,a,t));const r=e=>_(i[e],1,1,(()=>{i[e]=null}));return{c(){for(let e=0;e<i.length;e+=1)i[e].c();t=g()},m(e,a){for(let t=0;t<i.length;t+=1)i[t]&&i[t].m(e,a);l(e,t,a),n=!0},p(e,n){if(1&n){let l;for(a=s(e[0]),l=0;l<a.length;l+=1){const s=Pn(e,a,l);i[l]?(i[l].p(s,n),o(i[l],1)):(i[l]=Ln(s),i[l].c(),o(i[l],1),i[l].m(t.parentNode,t))}for(p(),l=a.length;l<i.length;l+=1)r(l);c()}},i(e){if(!n){for(let e=0;e<a.length;e+=1)o(i[e]);n=!0}},o(e){i=i.filter(Boolean);for(let e=0;e<i.length;e+=1)_(i[e]);n=!1},d(e){e&&d(t),m(i,e)}}}function Tn(e){let t,n;return t=new An({props:{class:"pipen-report-overflowmenu",persistentClickItems:!0,size:"sm",icon:Ie,iconDescription:"Select visible columns ...",$$slots:{default:[Un]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,[n]){const a={};65&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function Rn(e,t,n){let{allColumns:a}=t;return e.$$set=e=>{"allColumns"in e&&n(0,a=e.allColumns)},[a,function(t,s){e.$$.not_equal(s.selected,t)&&(s.selected=t,n(0,a))},()=>{n(0,a)}]}class En extends t{constructor(e){super(),n(this,e,Rn,Tn,a,{allColumns:0})}}function Nn(e){let t,n;return t=new xe({props:{iconDescription:"Right click and save as to download the entire data",tooltipPosition:"top",icon:Ve,href:e[2]}}),t.$on("click",e[13]),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};4&n&&(a.href=e[2]),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function wn(e){let t,n,a,s,i,r,m,u,$,S;function M(t){e[17](t)}let A={persistent:!0};function P(t){e[18](t)}void 0!==e[7]&&(A.value=e[7]),t=new Ue({props:A}),C.push((()=>y(t,"value",M)));let L={};void 0!==e[6]&&(L.allColumns=e[6]),s=new En({props:L}),C.push((()=>y(s,"allColumns",P))),m=new xe({props:{iconDescription:`Download current ${e[11].length} records (may be partial of the entire data)`,tooltipPosition:"top",icon:De}}),m.$on("click",e[14]);let U=!!e[2]&&Nn(e);return{c(){f(t.$$.fragment),a=v(),f(s.$$.fragment),r=v(),f(m.$$.fragment),u=v(),U&&U.c(),$=g()},m(e,n){b(t,e,n),l(e,a,n),b(s,e,n),l(e,r,n),b(m,e,n),l(e,u,n),U&&U.m(e,n),l(e,$,n),S=!0},p(e,a){const r={};!n&&128&a&&(n=!0,r.value=e[7],x((()=>n=!1))),t.$set(r);const l={};!i&&64&a&&(i=!0,l.allColumns=e[6],x((()=>i=!1))),s.$set(l);const d={};2048&a&&(d.iconDescription=`Download current ${e[11].length} records (may be partial of the entire data)`),m.$set(d),e[2]?U?(U.p(e,a),4&a&&o(U,1)):(U=Nn(e),U.c(),o(U,1),U.m($.parentNode,$)):U&&(p(),_(U,1,1,(()=>{U=null})),c())},i(e){S||(o(t.$$.fragment,e),o(s.$$.fragment,e),o(m.$$.fragment,e),o(U),S=!0)},o(e){_(t.$$.fragment,e),_(s.$$.fragment,e),_(m.$$.fragment,e),_(U),S=!1},d(e){e&&(d(a),d(r),d(u),d($)),h(t,e),h(s,e),h(m,e),U&&U.d(e)}}}function Bn(e){let t,n;return t=new Le({props:{$$slots:{default:[wn]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};134219972&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function In(e){let t,n;return t=new Pe({props:{size:e[12],$$slots:{default:[Bn]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};134219972&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function Dn(e){let t;return{c(){t=i("p"),t.textContent='You must use "Save as" or "Save link as" from the context menu (by right-clicking the button) to download the entire data.'},m(e,n){l(e,t,n)},p:P,d(e){e&&d(t)}}}function Vn(e){let t,n,a,s,i,r,p,c,m;const u=[{size:e[0]},{zebra:e[3]},{sortable:e[4]},{page:e[8]},{pageSize:e[9]},{rows:e[11]},{headers:e[1]},e[15]];let g={$$slots:{default:[In]},$$scope:{ctx:e}};for(let e=0;e<u.length;e+=1)g=w(g,u[e]);function $(t){e[19](t)}function S(t){e[20](t)}t=new Me({props:g});let M={pageSizes:e[5],totalItems:e[11].length};function A(t){e[21](t)}void 0!==e[9]&&(M.pageSize=e[9]),void 0!==e[8]&&(M.page=e[8]),a=new Ae({props:M}),C.push((()=>y(a,"pageSize",$))),C.push((()=>y(a,"page",S)));let P={passiveModal:!0,modalHeading:"Downloading entire data",$$slots:{default:[Dn]},$$scope:{ctx:e}};return void 0!==e[10]&&(P.open=e[10]),p=new _e({props:P}),C.push((()=>y(p,"open",A))),p.$on("open",e[22]),p.$on("close",e[23]),{c(){f(t.$$.fragment),n=v(),f(a.$$.fragment),r=v(),f(p.$$.fragment)},m(e,s){b(t,e,s),l(e,n,s),b(a,e,s),l(e,r,s),b(p,e,s),m=!0},p(e,[n]){const r=35611&n?F(u,[1&n&&{size:e[0]},8&n&&{zebra:e[3]},16&n&&{sortable:e[4]},256&n&&{page:e[8]},512&n&&{pageSize:e[9]},2048&n&&{rows:e[11]},2&n&&{headers:e[1]},32768&n&&z(e[15])]):{};134219972&n&&(r.$$scope={dirty:n,ctx:e}),t.$set(r);const l={};32&n&&(l.pageSizes=e[5]),2048&n&&(l.totalItems=e[11].length),!s&&512&n&&(s=!0,l.pageSize=e[9],x((()=>s=!1))),!i&&256&n&&(i=!0,l.page=e[8],x((()=>i=!1))),a.$set(l);const o={};134217728&n&&(o.$$scope={dirty:n,ctx:e}),!c&&1024&n&&(c=!0,o.open=e[10],x((()=>c=!1))),p.$set(o)},i(e){m||(o(t.$$.fragment,e),o(a.$$.fragment,e),o(p.$$.fragment,e),m=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),_(p.$$.fragment,e),m=!1},d(e){e&&(d(n),d(r)),h(t,e),h(a,e),h(p,e)}}}function Gn(e,t,n){const a=["size","src","zebra","sortable","rows","headers","pageSizes"];let s=E(t,a),{size:i="sm"}=t,{src:r}=t;i={sm:"short",small:"short",short:"short",md:"medium",medium:"medium",lg:"tall",large:"tall",tall:"tall",compact:"compact"}[i];let l={short:"sm",compact:"sm",medium:"default",tall:"default"}[i],{zebra:o=!0}=t,{sortable:p=!0}=t,{rows:c=[]}=t,{headers:_=[]}=t,{pageSizes:d=[20,50,75,100]}=t,m=_.map((e=>({...e,selected:!0}))),u=1,g=d[0],f="",b=!1,h=c;return e.$$set=e=>{t=w(w({},t),B(e)),n(15,s=E(t,a)),"size"in e&&n(0,i=e.size),"src"in e&&n(2,r=e.src),"zebra"in e&&n(3,o=e.zebra),"sortable"in e&&n(4,p=e.sortable),"rows"in e&&n(16,c=e.rows),"headers"in e&&n(1,_=e.headers),"pageSizes"in e&&n(5,d=e.pageSizes)},e.$$.update=()=>{64&e.$$.dirty&&n(1,_=m.filter((e=>e.selected))),65664&e.$$.dirty&&(f.length>0?n(11,h=c.filter((e=>Object.values(e).filter((e=>!!e&&e.toString().includes(f))).length>0))):n(11,h=c))},[i,_,r,o,p,d,m,f,u,g,b,h,l,e=>{e.preventDefault(),n(10,b=!0)},()=>{((e,t)=>{const n=document.createElement("a");n.href=e,n.download=t,document.body.appendChild(n),n.click(),document.body.removeChild(n)})("data:application/vnd.ms-excel;charset=utf-8,\ufeff"+encodeURIComponent(function(){const e=_.map((e=>e.value)),t=[];return h.forEach((e=>{const n=_.map((t=>`"${e[t.key]}"`));t.push(n.join(","))})),e.join(",")+"\r\n"+t.join("\r\n")}()),r&&r.length>0?r.split("/").pop().split(".")[0]+".xls":"data-subset.xls")},s,c,function(e){f=e,n(7,f)},function(e){m=e,n(6,m)},function(e){g=e,n(9,g)},function(e){u=e,n(8,u)},function(e){b=e,n(10,b)},function(t){J.call(this,e,t)},function(t){J.call(this,e,t)}]}class Hn extends t{constructor(e){super(),n(this,e,Gn,Vn,a,{size:0,src:2,zebra:3,sortable:4,rows:16,headers:1,pageSizes:5})}}function Fn(e){let t,n,a;const s=[{headers:e[0]},{rows:e[1]},e[3]];let r={};for(let e=0;e<s.length;e+=1)r=w(r,s[e]);n=new Hn({props:r});let p=[e[2]],c={};for(let e=0;e<p.length;e+=1)c=w(c,p[e]);return{c(){t=i("div"),f(n.$$.fragment),H(t,c)},m(e,s){l(e,t,s),b(n,t,null),a=!0},p(e,a){const i=11&a?F(s,[1&a&&{headers:e[0]},2&a&&{rows:e[1]},8&a&&z(e[3])]):{};n.$set(i),H(t,c=F(p,[4&a&&e[2]]))},i(e){a||(o(n.$$.fragment,e),a=!0)},o(e){_(n.$$.fragment,e),a=!1},d(e){e&&d(t),h(n)}}}function zn(e){let t,n;const a=[{headers:e[0]},{rows:e[1]},e[3]];let s={};for(let e=0;e<a.length;e+=1)s=w(s,a[e]);return t=new Hn({props:s}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const s=11&n?F(a,[1&n&&{headers:e[0]},2&n&&{rows:e[1]},8&n&&z(e[3])]):{};t.$set(s)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function On(e){let t,n,a,s;const i=[zn,Fn],r=[];function m(e,t){return e[2]?1:0}return t=m(e),n=r[t]=i[t](e),{c(){n.c(),a=g()},m(e,n){r[t].m(e,n),l(e,a,n),s=!0},p(e,[s]){let l=t;t=m(e),t===l?r[t].p(e,s):(p(),_(r[l],1,1,(()=>{r[l]=null})),c(),n=r[t],n?n.p(e,s):(n=r[t]=i[t](e),n.c()),o(n,1),n.m(a.parentNode,a))},i(e){s||(o(n),s=!0)},o(e){_(n),s=!1},d(e){e&&d(a),r[t].d(e)}}}function qn(e,t,n){const a=["frameProps","data","headers","rows"];let s=E(t,a),{frameProps:i={class:"pipen-report-datatable-frame"}}=t,{data:r={}}=t,{headers:l=[]}=t,{rows:o=[]}=t;if(l.length+o.length>0&&r.length>0)throw"Can only specify `data` or `headers`/`rows` for `DataTable`!";return r.length>0&&(o=r,l=Object.keys(o[0]).filter((e=>"id"!==e)).map((e=>({key:e,value:e})))),e.$$set=e=>{t=w(w({},t),B(e)),n(3,s=E(t,a)),"frameProps"in e&&n(2,i=e.frameProps),"data"in e&&n(4,r=e.data),"headers"in e&&n(0,l=e.headers),"rows"in e&&n(1,o=e.rows)},[l,o,i,s,r]}class kn extends t{constructor(e){super(),n(this,e,qn,On,a,{frameProps:2,data:4,headers:0,rows:1})}}function jn(e){let t,n;return{c(){t=i("div"),n=$(e[0]),r(t,"class","pipen-report-descr-title svelte-1xzxuva")},m(e,a){l(e,t,a),S(t,n)},p(e,t){1&t&&M(n,e[0])},d(e){e&&d(t)}}}function Kn(e){let t,n,a,s=e[0]&&jn(e);const r=e[3].default,p=L(r,e,e[2],null);let c=[e[1]],m={};for(let e=0;e<c.length;e+=1)m=w(m,c[e]);return{c(){t=i("div"),s&&s.c(),n=v(),p&&p.c(),H(t,m),O(t,"svelte-1xzxuva",!0)},m(e,i){l(e,t,i),s&&s.m(t,null),S(t,n),p&&p.m(t,null),a=!0},p(e,[i]){e[0]?s?s.p(e,i):(s=jn(e),s.c(),s.m(t,n)):s&&(s.d(1),s=null),p&&p.p&&(!a||4&i)&&U(p,r,e,e[2],a?R(r,e[2],i,null):T(e[2]),null),H(t,m=F(c,[2&i&&e[1]])),O(t,"svelte-1xzxuva",!0)},i(e){a||(o(p,e),a=!0)},o(e){_(p,e),a=!1},d(e){e&&d(t),s&&s.d(),p&&p.d(e)}}}function Yn(e,t,n){const a=["title"];let s=E(t,a),{$$slots:i={},$$scope:r}=t,{title:l}=t,{klass:o,...p}=s;return o=o?`pipen-report-descr ${o}`:"pipen-report-descr",p||(p={}),p.class=o,e.$$set=e=>{t=w(w({},t),B(e)),n(5,s=E(t,a)),"title"in e&&n(0,l=e.title),"$$scope"in e&&n(2,r=e.$$scope)},[l,p,r,i]}class Xn extends t{constructor(e){super(),n(this,e,Yn,Kn,a,{title:0})}}function Wn(e){let t,n,a,s;return t=new ue({props:{label:"Plot"}}),a=new ue({props:{label:"Table"}}),{c(){f(t.$$.fragment),n=v(),f(a.$$.fragment)},m(e,i){b(t,e,i),l(e,n,i),b(a,e,i),s=!0},p:P,i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),h(t,e),h(a,e)}}}function Qn(e){let t,n;return t=new Cn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/gsea_table.png",width:1e3,height:600}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p:P,i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function Jn(e){let t,n;return t=new kn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/fgsea.txt",data:[{pathway:"Ascorbate and aldarate metabolism",pval:0,padj:1e-4,log2err:.6105,ES:-.7778,NES:-1.718,size:27,leadingEdge:"ALDH9A1,ALDH1B1,ALDH3A2,UGDH",id:0},{pathway:"Nitrogen metabolism",pval:.0034,padj:.0338,log2err:.4317,ES:.6142,NES:1.9795,size:17,leadingEdge:"GLUL,CA5B,GLUD1,CPS1,CA4",id:1},{pathway:"Glycolysis / Gluconeogenesis",pval:.0107,padj:.0712,log2err:.3807,ES:-.5453,NES:-1.2797,size:68,leadingEdge:"ADH5,PKM,ALDH9A1,PGM2,ENO2,GPI,ALDOA,PGAM1,MINPP1,FBP1,ALDH1B1,DLAT,HK1,ALDH3A2,PFKL,ALDOC,ENO1,GAPDH",id:2},{pathway:"Fatty acid degradation",pval:.0234,padj:.117,log2err:.3525,ES:-.5697,NES:-1.2992,size:44,leadingEdge:"ACADVL,ACAA2,CPT1A,ADH5,ALDH9A1,ACOX1,HADHA,ACSL3,ACSL5,ACADSB,ALDH1B1,ALDH3A2,ECHS1,ACADS,CPT2,HADHB,HADH,ACSL4,ACAT2",id:3},{pathway:"Citrate cycle (TCA cycle)",pval:.0656,padj:.2624,log2err:.1737,ES:-.5706,NES:-1.2729,size:30,leadingEdge:"DLST,ACO1,FH,MDH2,SUCLA2,DLAT,CS,IDH3B,SDHC,SUCLG1,MDH1,SDHD,IDH3G,ACLY,PDHB,OGDH,PCK2,PDHA1,SUCLG2,PC,ACO2",id:4},{pathway:"Pentose phosphate pathway",pval:.1564,padj:.4421,log2err:.1067,ES:-.5337,NES:-1.1906,size:30,leadingEdge:"PRPS1,PGM2,GPI,ALDOA,PRPS2,FBP1,PFKL,ALDOC,PGM1,RPE,G6PD,IDNK,TALDO1",id:5},{pathway:"Amino sugar and nucleotide sugar metabolism",pval:.1616,padj:.4421,log2err:.1043,ES:-.501,NES:-1.1506,size:48,leadingEdge:"FPGT,HEXA,HEXB,PGM2,GNPDA2,GPI,GALT,GNPNAT1,MPI,HK1,GNE,AMDHD2,UGDH,GNPDA1,PGM1,PMM1,GALK1,RENBP,CYB5R1,GMPPA,CYB5R3",id:6},{pathway:"Oxidative phosphorylation",pval:.1768,padj:.4421,log2err:.0985,ES:-.4598,NES:-1.1036,size:120,leadingEdge:"NDUFV3,ATP6V1E1,PPA2,NDUFS1,ATP5PO,ATP6V1B2,LHPP,NDUFA5,ATP5F1C,ATP6V1E2,ATP5PB,UQCRC1,NDUFA11,NDUFS8,NDUFA9,COX8A,ATP6V0D1,ATP5PF,UQCRC2,ATP6V1A,SDHC,NDUFV1,TCIRG1,ATP6V1D,COX15,NDUFA4,SDHD,ATP5F1B,NDUFS6,NDUFB7,NDUFS3,NDUFC2,ATP6AP1,ATP6V1G1,NDUFA13,NDUFB2,NDUFS2,NDUFA7,COX6B1,ATP5MC1,NDUFAB1,COX11,CYC1,NDUFS4,ATP5MC3,ATP6V1H,NDUFS7,ATP6V0E2,NDUFA8,COX6A1,COX4I1,NDUFB10,COX5A,ATP6V0E1,ATP5MC2,ATP6V1F,ATP5F1A",id:7},{pathway:"Pentose and glucuronate interconversions",pval:.2281,padj:.5068,log2err:.0846,ES:-.5065,NES:-1.1379,size:34,leadingEdge:"AKR1B1,CRYL1,SORD,UGDH,RPE",id:8},{pathway:"Starch and sucrose metabolism",pval:.2758,padj:.5515,log2err:.0746,ES:-.4919,NES:-1.1085,size:35,leadingEdge:"GAA,GYS1,PGM2,GPI,AGL,PYGB,HK1,GBE1,PGM1",id:9},{pathway:"Glyoxylate and dicarboxylate metabolism",pval:.3519,padj:.6399,log2err:.0627,ES:-.4893,NES:-1.085,size:28,leadingEdge:"CAT,HYI,MUT,GCSH,ACO1,MDH2,CS,PCCB,MDH1",id:10},{pathway:"Propanoate metabolism",pval:.3952,padj:.652,log2err:.0569,ES:-.473,NES:-1.0606,size:32,leadingEdge:"MLYCD,MUT,HADHA,HIBCH,SUCLA2,ECHS1,BCKDHB,PCCB,SUCLG1,BCKDHA,DBT,ACAT2,ECHDC1,SUCLG2,ACACA,PCCA,DLD,ACSS1,ACACB",id:11},{pathway:"Fatty acid elongation",pval:.4238,padj:.652,log2err:.0537,ES:-.4715,NES:-1.052,size:30,leadingEdge:"ACAA2,HSD17B12,HACD2,HADHA,PPT1,ACOT4,ACOT2,ECHS1,HACD4,HADHB,HACD3,HADH",id:12},{pathway:"Butanoate metabolism",pval:.6004,padj:.8045,log2err:.0379,ES:-.4311,NES:-.956,size:28,leadingEdge:"BDH2,HADHA,AACS,ECHS1,ACADS,HMGCL,HADH,ACAT2,ALDH5A1",id:13},{pathway:"Fructose and mannose metabolism",pval:.6323,padj:.8045,log2err:.0353,ES:-.4223,NES:-.9473,size:33,leadingEdge:"FPGT,AKR1B1,ENOSF1,SORD,ALDOA,FBP1,MPI,HK1,PFKL,ALDOC",id:14},{pathway:"Pyruvate metabolism",pval:.6436,padj:.8045,log2err:.0342,ES:-.4176,NES:-.9496,size:39,leadingEdge:"PKM,GLO1,ALDH9A1,FH,MDH2,ALDH1B1,DLAT,ALDH3A2,ME2,MDH1,ACAT2,PDHB,PCK2,PDHA1,PC,ACACA,DLD,ACSS1,ACYP2,ACACB",id:15},{pathway:"Inositol phosphate metabolism",pval:.7373,padj:.8297,log2err:.0272,ES:-.3914,NES:-.9223,size:73,leadingEdge:"PIK3CG,PIP4K2C,IMPA1,PLCB2,IPPK,SACM1L,PLCB3,INPP5B,MTMR2,MTM1,MINPP1,PIK3C2A,MTMR4,INPP4A,MTMR1,PIK3CD,PI4K2A,SYNJ1,MTMR6,PIP5K1A,PLCB1,PLCG1,CDIPT,ISYNA1,TPI1,PIK3C2B,INPP5E,PIKFYVE,INPPL1,PIP4K2B,INPP5F,INPP4B,MTMR14,PIK3CB,PLCH2",id:16},{pathway:"Galactose metabolism",pval:.7467,padj:.8297,log2err:.0271,ES:-.3936,NES:-.8788,size:31,leadingEdge:"GAA,AKR1B1,PGM2,GALT,HK1,PFKL,PGM1,GALK1",id:17},{pathway:"Sulfur metabolism",pval:.8654,padj:.911,log2err:.0244,ES:-.4051,NES:-.7613,size:9,leadingEdge:"PAPSS1,BPNT1",id:18},{pathway:"Fatty acid biosynthesis",pval:.9798,padj:.9798,log2err:.0134,ES:-.2742,NES:-.5535,size:13,leadingEdge:"ACSL3,ACSL5",id:19}]}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p:P,i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function Zn(e){let t,n,a,s;return t=new ge({props:{$$slots:{default:[Qn]},$$scope:{ctx:e}}}),a=new ge({props:{$$slots:{default:[Jn]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=v(),f(a.$$.fragment)},m(e,i){b(t,e,i),l(e,n,i),b(a,e,i),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const i={};1&n&&(i.$$scope={dirty:n,ctx:e}),a.$set(i)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),h(t,e),h(a,e)}}}function ea(e){let t,n;return t=new me({props:{$$slots:{content:[Zn],default:[Wn]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function ta(e){let t,n,a,s,p,c,m,u,g,C,y,x,$,M,A,L,U,T,R,E,N,w,B,I,D,V,G,H,F,z,O,q,k,j,K,Y,X,W,Q,J,ee,te,ne,ae,se,ie,re,le,oe,pe,ce;return a=new Xn({props:{title:"Ascorbate and aldarate metabolism",class:"pipen-report-table-image-descr"}}),p=new Cn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/fgsea_Ascorbate-and-aldarate-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),u=new Xn({props:{title:"Nitrogen metabolism",class:"pipen-report-table-image-descr"}}),C=new Cn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/fgsea_Nitrogen-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),$=new Xn({props:{title:"Glycolysis / Gluconeogenesis",class:"pipen-report-table-image-descr"}}),A=new Cn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/fgsea_Glycolysis-Gluconeogenesis.png",class:"pipen-report-table-image",width:1e3,height:800}}),T=new Xn({props:{title:"Fatty acid degradation",class:"pipen-report-table-image-descr"}}),E=new Cn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/fgsea_Fatty-acid-degradation.png",class:"pipen-report-table-image",width:1e3,height:800}}),B=new Xn({props:{title:"Citrate cycle (TCA cycle)",class:"pipen-report-table-image-descr"}}),D=new Cn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/fgsea_Citrate-cycle-TCA-cycle-.png",class:"pipen-report-table-image",width:1e3,height:800}}),H=new Xn({props:{title:"Pentose phosphate pathway",class:"pipen-report-table-image-descr"}}),z=new Cn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/fgsea_Pentose-phosphate-pathway.png",class:"pipen-report-table-image",width:1e3,height:800}}),k=new Xn({props:{title:"Amino sugar and nucleotide sugar metabolism",class:"pipen-report-table-image-descr"}}),K=new Cn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/fgsea_Amino-sugar-and-nucleotide-sugar-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),W=new Xn({props:{title:"Oxidative phosphorylation",class:"pipen-report-table-image-descr"}}),J=new Cn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/fgsea_Oxidative-phosphorylation.png",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new Xn({props:{title:"Pentose and glucuronate interconversions",class:"pipen-report-table-image-descr"}}),se=new Cn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/fgsea_Pentose-and-glucuronate-interconversions.png",class:"pipen-report-table-image",width:1e3,height:800}}),le=new Xn({props:{title:"Starch and sucrose metabolism",class:"pipen-report-table-image-descr"}}),pe=new Cn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells/Colitis_vs_Control/fgsea_Starch-and-sucrose-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=i("div"),n=i("div"),f(a.$$.fragment),s=v(),f(p.$$.fragment),c=v(),m=i("div"),f(u.$$.fragment),g=v(),f(C.$$.fragment),y=v(),x=i("div"),f($.$$.fragment),M=v(),f(A.$$.fragment),L=v(),U=i("div"),f(T.$$.fragment),R=v(),f(E.$$.fragment),N=v(),w=i("div"),f(B.$$.fragment),I=v(),f(D.$$.fragment),V=v(),G=i("div"),f(H.$$.fragment),F=v(),f(z.$$.fragment),O=v(),q=i("div"),f(k.$$.fragment),j=v(),f(K.$$.fragment),Y=v(),X=i("div"),f(W.$$.fragment),Q=v(),f(J.$$.fragment),ee=v(),te=i("div"),f(ne.$$.fragment),ae=v(),f(se.$$.fragment),ie=v(),re=i("div"),f(le.$$.fragment),oe=v(),f(pe.$$.fragment),r(t,"class","pipen-report-table-of-images"),Z(t,"grid-template-columns","repeat(2, auto)")},m(e,i){l(e,t,i),S(t,n),b(a,n,null),S(n,s),b(p,n,null),S(t,c),S(t,m),b(u,m,null),S(m,g),b(C,m,null),S(t,y),S(t,x),b($,x,null),S(x,M),b(A,x,null),S(t,L),S(t,U),b(T,U,null),S(U,R),b(E,U,null),S(t,N),S(t,w),b(B,w,null),S(w,I),b(D,w,null),S(t,V),S(t,G),b(H,G,null),S(G,F),b(z,G,null),S(t,O),S(t,q),b(k,q,null),S(q,j),b(K,q,null),S(t,Y),S(t,X),b(W,X,null),S(X,Q),b(J,X,null),S(t,ee),S(t,te),b(ne,te,null),S(te,ae),b(se,te,null),S(t,ie),S(t,re),b(le,re,null),S(re,oe),b(pe,re,null),ce=!0},p:P,i(e){ce||(o(a.$$.fragment,e),o(p.$$.fragment,e),o(u.$$.fragment,e),o(C.$$.fragment,e),o($.$$.fragment,e),o(A.$$.fragment,e),o(T.$$.fragment,e),o(E.$$.fragment,e),o(B.$$.fragment,e),o(D.$$.fragment,e),o(H.$$.fragment,e),o(z.$$.fragment,e),o(k.$$.fragment,e),o(K.$$.fragment,e),o(W.$$.fragment,e),o(J.$$.fragment,e),o(ne.$$.fragment,e),o(se.$$.fragment,e),o(le.$$.fragment,e),o(pe.$$.fragment,e),ce=!0)},o(e){_(a.$$.fragment,e),_(p.$$.fragment,e),_(u.$$.fragment,e),_(C.$$.fragment,e),_($.$$.fragment,e),_(A.$$.fragment,e),_(T.$$.fragment,e),_(E.$$.fragment,e),_(B.$$.fragment,e),_(D.$$.fragment,e),_(H.$$.fragment,e),_(z.$$.fragment,e),_(k.$$.fragment,e),_(K.$$.fragment,e),_(W.$$.fragment,e),_(J.$$.fragment,e),_(ne.$$.fragment,e),_(se.$$.fragment,e),_(le.$$.fragment,e),_(pe.$$.fragment,e),ce=!1},d(e){e&&d(t),h(a),h(p),h(u),h(C),h($),h(A),h(T),h(E),h(B),h(D),h(H),h(z),h(k),h(K),h(W),h(J),h(ne),h(se),h(le),h(pe)}}}function na(e){let t,n,a,s;return t=new Re({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[ea]},$$scope:{ctx:e}}}),a=new Re({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[ta]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=v(),f(a.$$.fragment)},m(e,i){b(t,e,i),l(e,n,i),b(a,e,i),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const i={};1&n&&(i.$$scope={dirty:n,ctx:e}),a.$set(i)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),h(t,e),h(a,e)}}}function aa(e){let t,n,a,s;return t=new ue({props:{label:"Plot"}}),a=new ue({props:{label:"Table"}}),{c(){f(t.$$.fragment),n=v(),f(a.$$.fragment)},m(e,i){b(t,e,i),l(e,n,i),b(a,e,i),s=!0},p:P,i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),h(t,e),h(a,e)}}}function sa(e){let t,n;return t=new Cn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/gsea_table.png",width:1e3,height:600}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p:P,i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function ia(e){let t,n;return t=new kn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/fgsea.txt",data:[{pathway:"Nitrogen metabolism",pval:.0012,padj:.0234,log2err:.4551,ES:.6446,NES:2.0259,size:17,leadingEdge:"GLUL,GLUD1,CA5B,CA3,CA2",id:0},{pathway:"Ascorbate and aldarate metabolism",pval:.0207,padj:.1898,log2err:.3525,ES:-.6172,NES:-1.4114,size:27,leadingEdge:"ALDH9A1,ALDH3A2,UGDH,ALDH1B1",id:1},{pathway:"Fatty acid degradation",pval:.0303,padj:.1898,log2err:.2617,ES:-.5478,NES:-1.305,size:44,leadingEdge:"ADH5,CPT1A,ACADVL,CPT2,ACAA2,HADHA,ALDH9A1,ACSL3,ACSL5,ACOX1,HADH,ALDH3A2,HADHB,ECHS1,ACADSB,ACADS,ECI2,ALDH1B1,ACSL6,ACAA1,ACAT2",id:2},{pathway:"Oxidative phosphorylation",pval:.038,padj:.1898,log2err:.2311,ES:-.4801,NES:-1.2149,size:120,leadingEdge:"NDUFV3,ATP6V1B2,PPA2,ATP6V1E1,ATP6V1E2,NDUFA5,SDHD,ATP5PO,ATP6V1G1,SDHC,ATP5PB,NDUFA4,UQCRC2,LHPP,COX4I1,ATP6V1A,NDUFS1,COX6B1,NDUFA9,NDUFA11,COX15,NDUFS8,TCIRG1,NDUFB10,COX11,ATP5F1A,COX8A,ATP5PF,NDUFA7,UQCRC1,ATP6AP1,NDUFS4,ATP6V0D1,NDUFS2,ATP6V0E1,NDUFC2,NDUFS6,ATP5F1C,NDUFB7,UQCR11,NDUFA8,ATP6V1D,NDUFA13,NDUFS3,COX7A2L,NDUFB3,ATP5F1B,ATP6V0E2,NDUFB5,NDUFA1,ATP5MC2,ATP5MG,ATP5MF,COX5B,NDUFB6,NDUFV1,NDUFS7,NDUFB1,NDUFA2,UQCRB,ATP6V1H,NDUFB2,ATP5MC3,COX5A,NDUFB11,ATP6V1G2,ATP6V1F,COX6C,UQCR10,NDUFAB1,UQCRQ,ATP5ME,SDHA,CYC1",id:3},{pathway:"Pentose phosphate pathway",pval:.0983,padj:.3629,log2err:.1404,ES:-.5423,NES:-1.2568,size:30,leadingEdge:"PRPS1,PGM2,PRPS2,ALDOA,PFKL,GPI,FBP1,TALDO1,RPE,ALDOC,GLYCTK,H6PD,PGM1,G6PD,DERA,PGLS,IDNK",id:4},{pathway:"Amino sugar and nucleotide sugar metabolism",pval:.1089,padj:.3629,log2err:.1315,ES:-.5008,NES:-1.2045,size:48,leadingEdge:"GNPDA2,HEXA,FPGT,HEXB,MPI,GALT,PGM2,AMDHD2,CYB5R1,PMM1,RENBP,GALK1,GNE,GPI,GNPNAT1",id:5},{pathway:"Citrate cycle (TCA cycle)",pval:.1382,padj:.3948,log2err:.1157,ES:-.5299,NES:-1.2279,size:30,leadingEdge:"DLST,ACO1,SDHD,SDHC,MDH2,FH,DLAT,OGDH,SUCLA2,SUCLG1,IDH3G,IDH3B,ACLY,IDH1,PDHB,PCK2,CS,SUCLG2,MDH1,IDH2,DLD,PDHA1",id:6},{pathway:"Pentose and glucuronate interconversions",pval:.2084,padj:.5209,log2err:.0903,ES:-.4963,NES:-1.161,size:34,leadingEdge:"AKR1B1,CRYL1,SORD,UGDH,RPE",id:7},{pathway:"Fatty acid elongation",pval:.3695,padj:.7668,log2err:.0608,ES:-.4661,NES:-1.08,size:30,leadingEdge:"HACD2,HSD17B12,ACAA2,HADHA,ACOT4,PPT1,ACOT2,HADH,HADHB,HACD4,ECHS1,HACD3",id:8},{pathway:"Starch and sucrose metabolism",pval:.4254,padj:.7668,log2err:.0538,ES:-.4463,NES:-1.0508,size:35,leadingEdge:"GAA,PGM2,GYS1,PYGB,GPI,AGL,GBE1,HK1,GYG1,PGM1,GANC",id:9},{pathway:"Glyoxylate and dicarboxylate metabolism",pval:.5428,padj:.7668,log2err:.0434,ES:-.4278,NES:-.9829,size:28,leadingEdge:"CAT,HYI,MUT,ACO1,GCSH,MDH2,GRHPR,GLYCTK,ACAT2,MCEE,PCCA,CS,MDH1,PCCB,DLD",id:10},{pathway:"Butanoate metabolism",pval:.5501,padj:.7668,log2err:.0428,ES:-.4263,NES:-.9794,size:28,leadingEdge:"BDH2,HADHA,HMGCL,ACSM3,HADH,ECHS1,AACS,ACADS,BDH1,ALDH5A1,ACAT2",id:11},{pathway:"Glycolysis / Gluconeogenesis",pval:.5501,padj:.7668,log2err:.0414,ES:-.3988,NES:-.9832,size:68,leadingEdge:"ADH5,PGM2,ENO2,ALDH9A1,ALDOA,PFKL,PKM,MINPP1,GPI,DLAT,ALDH3A2,FBP1,ALDOC,BPGM,HK1,PGAM1,PGM1,ALDH1B1,GAPDH,PDHB,PCK2,PFKM,ENO1,DLD,PDHA1,LDHB,ENO3",id:12},{pathway:"Propanoate metabolism",pval:.5675,padj:.7668,log2err:.0409,ES:-.4191,NES:-.9774,size:32,leadingEdge:"MLYCD,HADHA,HIBCH,MUT,BCKDHB,BCKDHA,SUCLA2,SUCLG1,ECHS1,ECHDC1,DBT,ACAT2,MCEE,PCCA,SUCLG2,PCCB,DLD,LDHB,ACACB,ACSS1,ALDH6A1",id:13},{pathway:"Galactose metabolism",pval:.5751,padj:.7668,log2err:.0402,ES:-.415,NES:-.9638,size:31,leadingEdge:"AKR1B1,GAA,GALT,PGM2,PFKL,GALK1",id:14},{pathway:"Pyruvate metabolism",pval:.667,padj:.8338,log2err:.033,ES:-.3909,NES:-.9271,size:39,leadingEdge:"GLO1,ALDH9A1,PKM,MDH2,FH,DLAT,ME2,ALDH3A2,GRHPR,ALDH1B1,ACAT2,PDHB,PCK2,ACYP2,MDH1,DLD,PDHA1,LDHB,ACACB,ACSS1,ACACA,ACOT12",id:15},{pathway:"Fructose and mannose metabolism",pval:.714,padj:.84,log2err:.03,ES:-.3819,NES:-.8927,size:33,leadingEdge:"AKR1B1,FPGT,ENOSF1,MPI,ALDOA,PFKL,PMM1,SORD,FBP1,ALDOC",id:16},{pathway:"Inositol phosphate metabolism",pval:.7998,padj:.8887,log2err:.0229,ES:-.3521,NES:-.8722,size:73,leadingEdge:"PIP4K2C,PLCB2,IPPK,SACM1L,IMPA1,MTMR2,PIK3CG,INPP5B,MTMR1,MINPP1,MTMR4,PIK3C2A,MTM1,PLCB3,SYNJ1,PI4K2A,PIK3CD,PIKFYVE,MTMR6,PIK3C2B,ISYNA1,PLCG1,INPP4A,PIP4K2B,MTMR14,CDIPT,PIP5K1A",id:17},{pathway:"Sulfur metabolism",pval:.9727,padj:.973,log2err:.0189,ES:-.3001,NES:-.5758,size:9,leadingEdge:"PAPSS1,BPNT1,SQOR,TST,ETHE1",id:18},{pathway:"Fatty acid biosynthesis",pval:.973,padj:.973,log2err:.0153,ES:-.2611,NES:-.5431,size:13,leadingEdge:"ACSL3,ACSL5",id:19}]}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p:P,i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function ra(e){let t,n,a,s;return t=new ge({props:{$$slots:{default:[sa]},$$scope:{ctx:e}}}),a=new ge({props:{$$slots:{default:[ia]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=v(),f(a.$$.fragment)},m(e,i){b(t,e,i),l(e,n,i),b(a,e,i),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const i={};1&n&&(i.$$scope={dirty:n,ctx:e}),a.$set(i)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),h(t,e),h(a,e)}}}function la(e){let t,n;return t=new me({props:{$$slots:{content:[ra],default:[aa]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function oa(e){let t,n,a,s,p,c,m,u,g,C,y,x,$,M,A,L,U,T,R,E,N,w,B,I,D,V,G,H,F,z,O,q,k,j,K,Y,X,W,Q,J,ee,te,ne,ae,se,ie,re,le,oe,pe,ce;return a=new Xn({props:{title:"Nitrogen metabolism",class:"pipen-report-table-image-descr"}}),p=new Cn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/fgsea_Nitrogen-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),u=new Xn({props:{title:"Ascorbate and aldarate metabolism",class:"pipen-report-table-image-descr"}}),C=new Cn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/fgsea_Ascorbate-and-aldarate-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),$=new Xn({props:{title:"Fatty acid degradation",class:"pipen-report-table-image-descr"}}),A=new Cn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/fgsea_Fatty-acid-degradation.png",class:"pipen-report-table-image",width:1e3,height:800}}),T=new Xn({props:{title:"Oxidative phosphorylation",class:"pipen-report-table-image-descr"}}),E=new Cn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/fgsea_Oxidative-phosphorylation.png",class:"pipen-report-table-image",width:1e3,height:800}}),B=new Xn({props:{title:"Pentose phosphate pathway",class:"pipen-report-table-image-descr"}}),D=new Cn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/fgsea_Pentose-phosphate-pathway.png",class:"pipen-report-table-image",width:1e3,height:800}}),H=new Xn({props:{title:"Amino sugar and nucleotide sugar metabolism",class:"pipen-report-table-image-descr"}}),z=new Cn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/fgsea_Amino-sugar-and-nucleotide-sugar-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),k=new Xn({props:{title:"Citrate cycle (TCA cycle)",class:"pipen-report-table-image-descr"}}),K=new Cn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/fgsea_Citrate-cycle-TCA-cycle-.png",class:"pipen-report-table-image",width:1e3,height:800}}),W=new Xn({props:{title:"Pentose and glucuronate interconversions",class:"pipen-report-table-image-descr"}}),J=new Cn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/fgsea_Pentose-and-glucuronate-interconversions.png",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new Xn({props:{title:"Fatty acid elongation",class:"pipen-report-table-image-descr"}}),se=new Cn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/fgsea_Fatty-acid-elongation.png",class:"pipen-report-table-image",width:1e3,height:800}}),le=new Xn({props:{title:"Starch and sucrose metabolism",class:"pipen-report-table-image-descr"}}),pe=new Cn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Type-1-helper-T-cells-1/Colitis_vs_Control/fgsea_Starch-and-sucrose-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=i("div"),n=i("div"),f(a.$$.fragment),s=v(),f(p.$$.fragment),c=v(),m=i("div"),f(u.$$.fragment),g=v(),f(C.$$.fragment),y=v(),x=i("div"),f($.$$.fragment),M=v(),f(A.$$.fragment),L=v(),U=i("div"),f(T.$$.fragment),R=v(),f(E.$$.fragment),N=v(),w=i("div"),f(B.$$.fragment),I=v(),f(D.$$.fragment),V=v(),G=i("div"),f(H.$$.fragment),F=v(),f(z.$$.fragment),O=v(),q=i("div"),f(k.$$.fragment),j=v(),f(K.$$.fragment),Y=v(),X=i("div"),f(W.$$.fragment),Q=v(),f(J.$$.fragment),ee=v(),te=i("div"),f(ne.$$.fragment),ae=v(),f(se.$$.fragment),ie=v(),re=i("div"),f(le.$$.fragment),oe=v(),f(pe.$$.fragment),r(t,"class","pipen-report-table-of-images"),Z(t,"grid-template-columns","repeat(2, auto)")},m(e,i){l(e,t,i),S(t,n),b(a,n,null),S(n,s),b(p,n,null),S(t,c),S(t,m),b(u,m,null),S(m,g),b(C,m,null),S(t,y),S(t,x),b($,x,null),S(x,M),b(A,x,null),S(t,L),S(t,U),b(T,U,null),S(U,R),b(E,U,null),S(t,N),S(t,w),b(B,w,null),S(w,I),b(D,w,null),S(t,V),S(t,G),b(H,G,null),S(G,F),b(z,G,null),S(t,O),S(t,q),b(k,q,null),S(q,j),b(K,q,null),S(t,Y),S(t,X),b(W,X,null),S(X,Q),b(J,X,null),S(t,ee),S(t,te),b(ne,te,null),S(te,ae),b(se,te,null),S(t,ie),S(t,re),b(le,re,null),S(re,oe),b(pe,re,null),ce=!0},p:P,i(e){ce||(o(a.$$.fragment,e),o(p.$$.fragment,e),o(u.$$.fragment,e),o(C.$$.fragment,e),o($.$$.fragment,e),o(A.$$.fragment,e),o(T.$$.fragment,e),o(E.$$.fragment,e),o(B.$$.fragment,e),o(D.$$.fragment,e),o(H.$$.fragment,e),o(z.$$.fragment,e),o(k.$$.fragment,e),o(K.$$.fragment,e),o(W.$$.fragment,e),o(J.$$.fragment,e),o(ne.$$.fragment,e),o(se.$$.fragment,e),o(le.$$.fragment,e),o(pe.$$.fragment,e),ce=!0)},o(e){_(a.$$.fragment,e),_(p.$$.fragment,e),_(u.$$.fragment,e),_(C.$$.fragment,e),_($.$$.fragment,e),_(A.$$.fragment,e),_(T.$$.fragment,e),_(E.$$.fragment,e),_(B.$$.fragment,e),_(D.$$.fragment,e),_(H.$$.fragment,e),_(z.$$.fragment,e),_(k.$$.fragment,e),_(K.$$.fragment,e),_(W.$$.fragment,e),_(J.$$.fragment,e),_(ne.$$.fragment,e),_(se.$$.fragment,e),_(le.$$.fragment,e),_(pe.$$.fragment,e),ce=!1},d(e){e&&d(t),h(a),h(p),h(u),h(C),h($),h(A),h(T),h(E),h(B),h(D),h(H),h(z),h(k),h(K),h(W),h(J),h(ne),h(se),h(le),h(pe)}}}function pa(e){let t,n,a,s;return t=new Re({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[la]},$$scope:{ctx:e}}}),a=new Re({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[oa]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=v(),f(a.$$.fragment)},m(e,i){b(t,e,i),l(e,n,i),b(a,e,i),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const i={};1&n&&(i.$$scope={dirty:n,ctx:e}),a.$set(i)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),h(t,e),h(a,e)}}}function ca(e){let t,n,a,s,p,c,m,u,g,C,y,x,$,S,M,A;return m=new Te({props:{align:"start",$$slots:{default:[na]},$$scope:{ctx:e}}}),M=new Te({props:{align:"start",$$slots:{default:[pa]},$$scope:{ctx:e}}}),{c(){t=i("h1"),t.textContent="Cluster_Type 1 helper T cells",n=i("a"),n.innerHTML="",a=v(),s=i("h2"),s.textContent="Colitis_vs_Control",p=i("a"),p.innerHTML="",c=v(),f(m.$$.fragment),u=v(),g=i("h1"),g.textContent="Cluster_Type 1 helper T cells.1",C=i("a"),C.innerHTML="",y=v(),x=i("h2"),x.textContent="Colitis_vs_Control",$=i("a"),$.innerHTML="",S=v(),f(M.$$.fragment),r(n,"id","prt-h1-10-cluster-type-1-helper-t-cells"),r(n,"class","pipen-report-toc-anchor"),r(p,"id","prt-h2-10-colitis-vs-control"),r(p,"class","pipen-report-toc-anchor"),r(C,"id","prt-h1-11-cluster-type-1-helper-t-cells-1"),r(C,"class","pipen-report-toc-anchor"),r($,"id","prt-h2-11-colitis-vs-control"),r($,"class","pipen-report-toc-anchor")},m(e,i){l(e,t,i),l(e,n,i),l(e,a,i),l(e,s,i),l(e,p,i),l(e,c,i),b(m,e,i),l(e,u,i),l(e,g,i),l(e,C,i),l(e,y,i),l(e,x,i),l(e,$,i),l(e,S,i),b(M,e,i),A=!0},p(e,[t]){const n={};1&t&&(n.$$scope={dirty:t,ctx:e}),m.$set(n);const a={};1&t&&(a.$$scope={dirty:t,ctx:e}),M.$set(a)},i(e){A||(o(m.$$.fragment,e),o(M.$$.fragment,e),A=!0)},o(e){_(m.$$.fragment,e),_(M.$$.fragment,e),A=!1},d(e){e&&(d(t),d(n),d(a),d(s),d(p),d(c),d(u),d(g),d(C),d(y),d(x),d($),d(S)),h(m,e),h(M,e)}}}class _a extends t{constructor(e){super(),n(this,e,null,ca,a,{})}}var da=[{slug:"prt-h1-0-cluster-cycling-t-cells",text:"Cluster_Cycling T cells",children:[{slug:"prt-h2-0-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:0}],page:0},{slug:"prt-h1-1-cluster-follicular-helper-t-cells",text:"Cluster_Follicular helper T cells",children:[{slug:"prt-h2-1-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:0}],page:0},{slug:"prt-h1-2-cluster-gamma-delta-t-cells",text:"Cluster_gamma-delta T cells",children:[{slug:"prt-h2-2-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:0}],page:0},{slug:"prt-h1-3-cluster-plasmablasts",text:"Cluster_Plasmablasts",children:[{slug:"prt-h2-3-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:0}],page:0},{slug:"prt-h1-4-cluster-regulatory-t-cells",text:"Cluster_Regulatory T cells",children:[{slug:"prt-h2-4-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:0}],page:0},{slug:"prt-h1-5-cluster-regulatory-t-cells-1",text:"Cluster_Regulatory T cells.1",children:[{slug:"prt-h2-5-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:1}],page:1},{slug:"prt-h1-6-cluster-tcm-naive-helper-t-cells",text:"Cluster_Tcm/Naive helper T cells",children:[{slug:"prt-h2-6-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:1}],page:1},{slug:"prt-h1-7-cluster-tem-trm-cytotoxic-t-cells",text:"Cluster_Tem/Trm cytotoxic T cells",children:[{slug:"prt-h2-7-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:1}],page:1},{slug:"prt-h1-8-cluster-tem-trm-cytotoxic-t-cells-1",text:"Cluster_Tem/Trm cytotoxic T cells.1",children:[{slug:"prt-h2-8-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:1}],page:1},{slug:"prt-h1-9-cluster-trm-cytotoxic-t-cells",text:"Cluster_Trm cytotoxic T cells",children:[{slug:"prt-h2-9-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:1}],page:1},{slug:"prt-h1-10-cluster-type-1-helper-t-cells",text:"Cluster_Type 1 helper T cells",children:[{slug:"prt-h2-10-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:2}],page:2},{slug:"prt-h1-11-cluster-type-1-helper-t-cells-1",text:"Cluster_Type 1 helper T cells.1",children:[{slug:"prt-h2-11-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:2}],page:2}],ma={pipeline:{name:"Immunopipe",desc:"Immunopipe (v1.4.2): Integrative analysis for scRNA-seq and scTCR-seq data"},versions:'\n    <a href="https://github.com/pwwang/pipen" target="_blank">\n        pipen v0.15.2\n    </a> and\n    <a href="https://github.com/pwwang/pipen-report" target="_blank">\n        pipen-report v0.20.1\n    </a>\n',entries:[{name:"SampleInfo",desc:"List sample information and perform statistics",npages:1,report_toc:!0,order:0,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n[1] ggrepel_0.9.5    ggprism_1.0.5    ggplot2_3.5.1    tidyr_1.3.1     \n[5] dplyr_1.1.4      tidyselect_1.2.1 jsonlite_1.8.8   rlang_1.1.4     \n[9] logger_0.3.0    \n\nloaded via a namespace (and not attached):\n [1] vctrs_0.6.5      cli_3.6.3        purrr_1.0.2      generics_0.1.3  \n [5] labeling_0.4.3   glue_1.7.0       colorspace_2.1-1 scales_1.3.0    \n [9] fansi_1.0.6      grid_4.3.3       munsell_0.5.1    tibble_3.2.1    \n[13] lifecycle_1.0.4  compiler_4.3.3   Rcpp_1.0.13      pkgconfig_2.0.3 \n[17] farver_2.1.2     digest_0.6.36    R6_2.5.1         utf8_1.2.4      \n[21] pillar_1.9.0     magrittr_2.0.3   withr_3.0.1      gtable_0.3.5    \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 1.73\nuser 10.62\nsys 0.10\npercent 617%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"SeuratPreparing",desc:"Load, prepare and apply QC to data, using `Seurat`",npages:1,report_toc:!0,order:2,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats4    stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] scDblFinder_1.16.0          SingleCellExperiment_1.24.0\n [3] SummarizedExperiment_1.32.0 Biobase_2.62.0             \n [5] GenomicRanges_1.54.1        GenomeInfoDb_1.38.1        \n [7] IRanges_2.36.0              S4Vectors_0.40.2           \n [9] BiocGenerics_0.48.1         MatrixGenerics_1.14.0      \n[11] matrixStats_1.1.0           dplyr_1.1.4                \n[13] ggplot2_3.5.1               bracer_1.2.1               \n[15] future_1.34.0               Seurat_5.1.0               \n[17] SeuratObject_5.0.2          sp_2.1-4                   \n[19] digest_0.6.36               jsonlite_1.8.8             \n[21] rlang_1.1.4                 logger_0.3.0               \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22          splines_4.3.3            \n  [3] later_1.3.2               BiocIO_1.12.0            \n  [5] bitops_1.0-8              tibble_3.2.1             \n  [7] R.oo_1.26.0               polyclip_1.10-7          \n  [9] XML_3.99-0.17             fastDummies_1.7.3        \n [11] lifecycle_1.0.4           edgeR_4.0.16             \n [13] globals_0.16.3            lattice_0.22-6           \n [15] MASS_7.3-60.0.1           magrittr_2.0.3           \n [17] limma_3.58.1              plotly_4.10.4            \n [19] yaml_2.3.10               metapod_1.10.0           \n [21] httpuv_1.6.15             sctransform_0.4.1        \n [23] spam_2.10-0               spatstat.sparse_3.1-0    \n [25] reticulate_1.38.0         cowplot_1.1.3            \n [27] pbapply_1.7-2             RColorBrewer_1.1-3       \n [29] abind_1.4-5               zlibbioc_1.48.0          \n [31] Rtsne_0.17                purrr_1.0.2              \n [33] R.utils_2.12.3            RCurl_1.98-1.16          \n [35] GenomeInfoDbData_1.2.11   ggrepel_0.9.5            \n [37] irlba_2.3.5.1             listenv_0.9.1            \n [39] spatstat.utils_3.0-5      goftest_1.2-3            \n [41] RSpectra_0.16-1           dqrng_0.3.2              \n [43] spatstat.random_3.2-3     fitdistrplus_1.1-11      \n [45] parallelly_1.38.0         DelayedMatrixStats_1.24.0\n [47] leiden_0.4.3.1            codetools_0.2-20         \n [49] DelayedArray_0.28.0       scuttle_1.12.0           \n [51] tidyselect_1.2.1          farver_2.1.2             \n [53] viridis_0.6.5             ScaledMatrix_1.10.0      \n [55] spatstat.explore_3.2-6    GenomicAlignments_1.38.0 \n [57] BiocNeighbors_1.20.0      progressr_0.14.0         \n [59] scater_1.30.1             ggridges_0.5.6           \n [61] survival_3.7-0            systemfonts_1.1.0        \n [63] tools_4.3.3               ragg_1.3.2               \n [65] ica_1.0-3                 Rcpp_1.0.13              \n [67] glue_1.7.0                gridExtra_2.3            \n [69] SparseArray_1.2.2         withr_3.0.1              \n [71] fastmap_1.2.0             bluster_1.12.0           \n [73] fansi_1.0.6               rsvd_1.0.5               \n [75] R6_2.5.1                  mime_0.12                \n [77] textshaping_0.4.0         colorspace_2.1-1         \n [79] scattermore_1.2           tensor_1.5               \n [81] spatstat.data_3.1-2       R.methodsS3_1.8.2        \n [83] RhpcBLASctl_0.23-42       utf8_1.2.4               \n [85] tidyr_1.3.1               generics_0.1.3           \n [87] data.table_1.15.4         rtracklayer_1.62.0       \n [89] httr_1.4.7                htmlwidgets_1.6.4        \n [91] S4Arrays_1.2.0            uwot_0.1.16              \n [93] pkgconfig_2.0.3           gtable_0.3.5             \n [95] lmtest_0.9-40             XVector_0.42.0           \n [97] htmltools_0.5.8.1         dotCall64_1.1-1          \n [99] scales_1.3.0              png_0.1-8                \n[101] harmony_1.2.0             scran_1.30.0             \n[103] reshape2_1.4.4            rjson_0.2.21             \n[105] nlme_3.1-165              zoo_1.8-12               \n[107] stringr_1.5.1             KernSmooth_2.23-24       \n[109] vipor_0.4.7               parallel_4.3.3           \n[111] miniUI_0.1.1.1            restfulr_0.0.15          \n[113] pillar_1.9.0              grid_4.3.3               \n[115] vctrs_0.6.5               RANN_2.6.1               \n[117] promises_1.3.0            BiocSingular_1.18.0      \n[119] beachmat_2.18.0           xtable_1.8-4             \n[121] cluster_2.1.6             beeswarm_0.4.0           \n[123] locfit_1.5-9.9            cli_3.6.3                \n[125] compiler_4.3.3            Rsamtools_2.18.0         \n[127] crayon_1.5.3              future.apply_1.11.2      \n[129] labeling_0.4.3            ggbeeswarm_0.7.2         \n[131] plyr_1.8.9                stringi_1.8.4            \n[133] viridisLite_0.4.2         deldir_2.0-4             \n[135] BiocParallel_1.36.0       munsell_0.5.1            \n[137] Biostrings_2.70.1         lazyeval_0.2.2           \n[139] spatstat.geom_3.2-9       Matrix_1.6-5             \n[141] RcppHNSW_0.6.0            patchwork_1.2.0          \n[143] sparseMatrixStats_1.14.0  statmod_1.5.0            \n[145] shiny_1.9.1               ROCR_1.0-11              \n[147] igraph_2.0.3              xgboost_2.1.1.1          \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 202.98\nuser 1582.98\nsys 7.06\npercent 783%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ClusterMarkersOfAllCells",desc:"Markers for clusters of all cells.",npages:3,report_toc:!0,order:4,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] UpSetR_1.4.0        ggVennDiagram_1.2.3 tidyseurat_0.8.0   \n [4] ttservice_0.4.1     future_1.34.0       ggrepel_0.9.5      \n [7] ggprism_1.0.5       ggplot2_3.5.1       enrichR_3.2        \n[10] Seurat_5.1.0        SeuratObject_5.0.2  sp_2.1-4           \n[13] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[16] tidyselect_1.2.1    digest_0.6.36       jsonlite_1.8.8     \n[19] rlang_1.1.4         logger_0.3.0       \n\nloaded via a namespace (and not attached):\n  [1] RColorBrewer_1.1-3     magrittr_2.0.3         spatstat.utils_3.0-5  \n  [4] farver_2.1.2           vctrs_0.6.5            ROCR_1.0-11           \n  [7] spatstat.explore_3.2-6 htmltools_0.5.8.1      curl_5.2.1            \n [10] sctransform_0.4.1      parallelly_1.38.0      KernSmooth_2.23-24    \n [13] htmlwidgets_1.6.4      ica_1.0-3              plyr_1.8.9            \n [16] plotly_4.10.4          zoo_1.8-12             igraph_2.0.3          \n [19] mime_0.12              lifecycle_1.0.4        pkgconfig_2.0.3       \n [22] Matrix_1.6-5           R6_2.5.1               fastmap_1.2.0         \n [25] fitdistrplus_1.1-11    shiny_1.9.1            colorspace_2.1-1      \n [28] patchwork_1.2.0        tensor_1.5             RSpectra_0.16-1       \n [31] irlba_2.3.5.1          WriteXLS_6.7.0         labeling_0.4.3        \n [34] progressr_0.14.0       fansi_1.0.6            spatstat.sparse_3.1-0 \n [37] httr_1.4.7             polyclip_1.10-7        abind_1.4-5           \n [40] compiler_4.3.3         withr_3.0.1            fastDummies_1.7.3     \n [43] MASS_7.3-60.0.1        rjson_0.2.21           tools_4.3.3           \n [46] lmtest_0.9-40          httpuv_1.6.15          future.apply_1.11.2   \n [49] goftest_1.2-3          glue_1.7.0             nlme_3.1-165          \n [52] promises_1.3.0         grid_4.3.3             Rtsne_0.17            \n [55] cluster_2.1.6          reshape2_1.4.4         generics_0.1.3        \n [58] gtable_0.3.5           spatstat.data_3.1-2    data.table_1.15.4     \n [61] utf8_1.2.4             spatstat.geom_3.2-9    RcppAnnoy_0.0.22      \n [64] RANN_2.6.1             pillar_1.9.0           stringr_1.5.1         \n [67] spam_2.10-0            RcppHNSW_0.6.0         limma_3.58.1          \n [70] later_1.3.2            splines_4.3.3          lattice_0.22-6        \n [73] survival_3.7-0         deldir_2.0-4           miniUI_0.1.1.1        \n [76] pbapply_1.7-2          gridExtra_2.3          scattermore_1.2       \n [79] statmod_1.5.0          matrixStats_1.1.0      stringi_1.8.4         \n [82] lazyeval_0.2.2         codetools_0.2-20       RVenn_1.1.0           \n [85] cli_3.6.3              uwot_0.1.16            xtable_1.8-4          \n [88] reticulate_1.38.0      munsell_0.5.1          Rcpp_1.0.13           \n [91] globals_0.16.3         spatstat.random_3.2-3  png_0.1-8             \n [94] parallel_4.3.3         ellipsis_0.3.2         presto_1.0.0          \n [97] dotCall64_1.1-1        listenv_0.9.1          viridisLite_0.4.2     \n[100] scales_1.3.0           ggridges_0.5.6         leiden_0.4.3.1        \n[103] purrr_1.0.2            cowplot_1.1.3         \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 216.89\nuser 139.48\nsys 7.23\npercent 67%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TopExpressingGenesOfAllCells",desc:"Top expressing genes for clusters of all cells.",npages:3,report_toc:!0,order:5,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ggprism_1.0.5      dplyr_1.1.4        enrichR_3.2        tibble_3.2.1      \n [5] Seurat_5.1.0       SeuratObject_5.0.2 sp_2.1-4           jsonlite_1.8.8    \n [9] rlang_1.1.4        logger_0.3.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.2-9   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] labeling_0.4.3         utf8_1.2.4             promises_1.3.0        \n [19] purrr_1.0.2            WriteXLS_6.7.0         goftest_1.2-3         \n [22] later_1.3.2            spatstat.utils_3.0-5   irlba_2.3.5.1         \n [25] parallel_4.3.3         cluster_2.1.6          R6_2.5.1              \n [28] ica_1.0-3              stringi_1.8.4          RColorBrewer_1.1-3    \n [31] spatstat.data_3.1-2    reticulate_1.38.0      parallelly_1.38.0     \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.13           \n [37] tensor_1.5             future.apply_1.11.2    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [43] splines_4.3.3          igraph_2.0.3           tidyselect_1.2.1      \n [46] abind_1.4-5            spatstat.random_3.2-3  codetools_0.2-20      \n [49] miniUI_0.1.1.1         spatstat.explore_3.2-6 curl_5.2.1            \n [52] listenv_0.9.1          lattice_0.22-6         plyr_1.8.9            \n [55] withr_3.0.1            shiny_1.9.1            ROCR_1.0-11           \n [58] Rtsne_0.17             future_1.34.0          fastDummies_1.7.3     \n [61] survival_3.7-0         polyclip_1.10-7        fitdistrplus_1.1-11   \n [64] pillar_1.9.0           KernSmooth_2.23-24     plotly_4.10.4         \n [67] generics_0.1.3         RcppHNSW_0.6.0         ggplot2_3.5.1         \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.7.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.15.4      RSpectra_0.16-1       \n [79] RANN_2.6.1             leiden_0.4.3.1         dotCall64_1.1-1       \n [82] cowplot_1.1.3          grid_4.3.3             tidyr_1.3.1           \n [85] colorspace_2.1-1       nlme_3.1-165           patchwork_1.2.0       \n [88] cli_3.6.3              spatstat.sparse_3.1-0  spam_2.10-0           \n [91] fansi_1.0.6            viridisLite_0.4.2      uwot_0.1.16           \n [94] gtable_0.3.5           digest_0.6.36          progressr_0.14.0      \n [97] ggrepel_0.9.5          farver_2.1.2           rjson_0.2.21          \n[100] htmlwidgets_1.6.4      htmltools_0.5.8.1      lifecycle_1.0.4       \n[103] httr_1.4.7             mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 52.61\nuser 38.55\nsys 2.30\npercent 77%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TCellSelection",desc:"Separate T and non-T cells and select T cells.",npages:1,report_toc:!0,order:6,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] glue_1.7.0         factoextra_1.0.7   patchwork_1.2.0    ggpubr_0.6.0      \n [5] ggrepel_0.9.5      ggprism_1.0.5      ggplot2_3.5.1      tibble_3.2.1      \n [9] tidyr_1.3.1        dplyr_1.1.4        Seurat_5.1.0       SeuratObject_5.0.2\n[13] sp_2.1-4           jsonlite_1.8.8     rlang_1.1.4        logger_0.3.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.2-9   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] backports_1.5.0        labeling_0.4.3         utf8_1.2.4            \n [19] promises_1.3.0         purrr_1.0.2            goftest_1.2-3         \n [22] later_1.3.2            spatstat.utils_3.0-5   broom_1.0.6           \n [25] irlba_2.3.5.1          parallel_4.3.3         cluster_2.1.6         \n [28] R6_2.5.1               ica_1.0-3              stringi_1.8.4         \n [31] RColorBrewer_1.1-3     spatstat.data_3.1-2    reticulate_1.38.0     \n [34] car_3.1-2              parallelly_1.38.0      lmtest_0.9-40         \n [37] scattermore_1.2        Rcpp_1.0.13            tensor_1.5            \n [40] future.apply_1.11.2    zoo_1.8-12             sctransform_0.4.1     \n [43] httpuv_1.6.15          Matrix_1.6-5           splines_4.3.3         \n [46] igraph_2.0.3           tidyselect_1.2.1       abind_1.4-5           \n [49] spatstat.random_3.2-3  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.2-6 listenv_0.9.1          lattice_0.22-6        \n [55] plyr_1.8.9             withr_3.0.1            shiny_1.9.1           \n [58] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n [61] fastDummies_1.7.3      survival_3.7-0         polyclip_1.10-7       \n [64] fitdistrplus_1.1-11    pillar_1.9.0           carData_3.0-5         \n [67] KernSmooth_2.23-24     plotly_4.10.4          generics_0.1.3        \n [70] RcppHNSW_0.6.0         munsell_0.5.1          scales_1.3.0          \n [73] globals_0.16.3         xtable_1.8-4           lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.15.4      RSpectra_0.16-1       \n [79] ggsignif_0.6.4         RANN_2.6.1             leiden_0.4.3.1        \n [82] dotCall64_1.1-1        cowplot_1.1.3          grid_4.3.3            \n [85] colorspace_2.1-1       nlme_3.1-165           cli_3.6.3             \n [88] spatstat.sparse_3.1-0  spam_2.10-0            fansi_1.0.6           \n [91] viridisLite_0.4.2      uwot_0.1.16            gtable_0.3.5          \n [94] rstatix_0.7.2          digest_0.6.36          progressr_0.14.0      \n [97] farver_2.1.2           htmlwidgets_1.6.4      htmltools_0.5.8.1     \n[100] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[103] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 38.03\nuser 45.75\nsys 0.83\npercent 122%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3435.798\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ClusterMarkers",desc:"Markers for clusters of all or selected T cells.",npages:2,report_toc:!0,order:10,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] UpSetR_1.4.0        ggVennDiagram_1.2.3 tidyseurat_0.8.0   \n [4] ttservice_0.4.1     future_1.34.0       ggrepel_0.9.5      \n [7] ggprism_1.0.5       ggplot2_3.5.1       enrichR_3.2        \n[10] Seurat_5.1.0        SeuratObject_5.0.2  sp_2.1-4           \n[13] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[16] tidyselect_1.2.1    digest_0.6.36       jsonlite_1.8.8     \n[19] rlang_1.1.4         logger_0.3.0       \n\nloaded via a namespace (and not attached):\n  [1] RColorBrewer_1.1-3     magrittr_2.0.3         spatstat.utils_3.0-5  \n  [4] farver_2.1.2           vctrs_0.6.5            ROCR_1.0-11           \n  [7] spatstat.explore_3.2-6 htmltools_0.5.8.1      curl_5.2.1            \n [10] sctransform_0.4.1      parallelly_1.38.0      KernSmooth_2.23-24    \n [13] htmlwidgets_1.6.4      ica_1.0-3              plyr_1.8.9            \n [16] plotly_4.10.4          zoo_1.8-12             igraph_2.0.3          \n [19] mime_0.12              lifecycle_1.0.4        pkgconfig_2.0.3       \n [22] Matrix_1.6-5           R6_2.5.1               fastmap_1.2.0         \n [25] fitdistrplus_1.1-11    shiny_1.9.1            colorspace_2.1-1      \n [28] patchwork_1.2.0        tensor_1.5             RSpectra_0.16-1       \n [31] irlba_2.3.5.1          WriteXLS_6.7.0         labeling_0.4.3        \n [34] progressr_0.14.0       fansi_1.0.6            spatstat.sparse_3.1-0 \n [37] httr_1.4.7             polyclip_1.10-7        abind_1.4-5           \n [40] compiler_4.3.3         withr_3.0.1            fastDummies_1.7.3     \n [43] MASS_7.3-60.0.1        rjson_0.2.21           tools_4.3.3           \n [46] lmtest_0.9-40          httpuv_1.6.15          future.apply_1.11.2   \n [49] goftest_1.2-3          glue_1.7.0             nlme_3.1-165          \n [52] promises_1.3.0         grid_4.3.3             Rtsne_0.17            \n [55] cluster_2.1.6          reshape2_1.4.4         generics_0.1.3        \n [58] gtable_0.3.5           spatstat.data_3.1-2    data.table_1.15.4     \n [61] utf8_1.2.4             spatstat.geom_3.2-9    RcppAnnoy_0.0.22      \n [64] RANN_2.6.1             pillar_1.9.0           stringr_1.5.1         \n [67] spam_2.10-0            RcppHNSW_0.6.0         limma_3.58.1          \n [70] later_1.3.2            splines_4.3.3          lattice_0.22-6        \n [73] survival_3.7-0         deldir_2.0-4           miniUI_0.1.1.1        \n [76] pbapply_1.7-2          gridExtra_2.3          scattermore_1.2       \n [79] statmod_1.5.0          matrixStats_1.1.0      stringi_1.8.4         \n [82] lazyeval_0.2.2         codetools_0.2-20       RVenn_1.1.0           \n [85] cli_3.6.3              uwot_0.1.16            xtable_1.8-4          \n [88] reticulate_1.38.0      munsell_0.5.1          Rcpp_1.0.13           \n [91] globals_0.16.3         spatstat.random_3.2-3  png_0.1-8             \n [94] parallel_4.3.3         ellipsis_0.3.2         presto_1.0.0          \n [97] dotCall64_1.1-1        listenv_0.9.1          viridisLite_0.4.2     \n[100] scales_1.3.0           ggridges_0.5.6         leiden_0.4.3.1        \n[103] purrr_1.0.2            cowplot_1.1.3         \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 136.36\nuser 90.28\nsys 4.48\npercent 69%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TopExpressingGenes",desc:"Top expressing genes for clusters of all or selected T cells.",npages:2,report_toc:!0,order:11,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ggprism_1.0.5      dplyr_1.1.4        enrichR_3.2        tibble_3.2.1      \n [5] Seurat_5.1.0       SeuratObject_5.0.2 sp_2.1-4           jsonlite_1.8.8    \n [9] rlang_1.1.4        logger_0.3.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.2-9   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] labeling_0.4.3         utf8_1.2.4             promises_1.3.0        \n [19] purrr_1.0.2            WriteXLS_6.7.0         goftest_1.2-3         \n [22] later_1.3.2            spatstat.utils_3.0-5   irlba_2.3.5.1         \n [25] parallel_4.3.3         cluster_2.1.6          R6_2.5.1              \n [28] ica_1.0-3              stringi_1.8.4          RColorBrewer_1.1-3    \n [31] spatstat.data_3.1-2    reticulate_1.38.0      parallelly_1.38.0     \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.13           \n [37] tensor_1.5             future.apply_1.11.2    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [43] splines_4.3.3          igraph_2.0.3           tidyselect_1.2.1      \n [46] abind_1.4-5            spatstat.random_3.2-3  codetools_0.2-20      \n [49] miniUI_0.1.1.1         spatstat.explore_3.2-6 curl_5.2.1            \n [52] listenv_0.9.1          lattice_0.22-6         plyr_1.8.9            \n [55] withr_3.0.1            shiny_1.9.1            ROCR_1.0-11           \n [58] Rtsne_0.17             future_1.34.0          fastDummies_1.7.3     \n [61] survival_3.7-0         polyclip_1.10-7        fitdistrplus_1.1-11   \n [64] pillar_1.9.0           KernSmooth_2.23-24     plotly_4.10.4         \n [67] generics_0.1.3         RcppHNSW_0.6.0         ggplot2_3.5.1         \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.7.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.15.4      RSpectra_0.16-1       \n [79] RANN_2.6.1             leiden_0.4.3.1         dotCall64_1.1-1       \n [82] cowplot_1.1.3          grid_4.3.3             tidyr_1.3.1           \n [85] colorspace_2.1-1       nlme_3.1-165           patchwork_1.2.0       \n [88] cli_3.6.3              spatstat.sparse_3.1-0  spam_2.10-0           \n [91] fansi_1.0.6            viridisLite_0.4.2      uwot_0.1.16           \n [94] gtable_0.3.5           digest_0.6.36          progressr_0.14.0      \n [97] ggrepel_0.9.5          farver_2.1.2           rjson_0.2.21          \n[100] htmlwidgets_1.6.4      htmltools_0.5.8.1      lifecycle_1.0.4       \n[103] httr_1.4.7             mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 31.30\nuser 27.76\nsys 1.57\npercent 93%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TCRClusterStats",desc:"Statistics of TCR clusters, generated by `TCRClustering`.",npages:1,report_toc:!0,order:13,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] grid      stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] ComplexHeatmap_2.18.0 ggprism_1.0.5         immunarch_0.9.1      \n [4] patchwork_1.2.0       data.table_1.15.4     dtplyr_1.3.1         \n [7] dplyr_1.1.4           tidyr_1.3.1           tibble_3.2.1         \n[10] ggplot2_3.5.1         jsonlite_1.8.8        rlang_1.1.4          \n[13] logger_0.3.0         \n\nloaded via a namespace (and not attached):\n  [1] gridExtra_2.3       phangorn_2.11.1     readxl_1.4.3       \n  [4] magrittr_2.0.3      clue_0.3-65         GetoptLong_1.0.5   \n  [7] matrixStats_1.1.0   compiler_4.3.3      flexmix_2.3-19     \n [10] png_0.1-8           vctrs_0.6.5         reshape2_1.4.4     \n [13] ggalluvial_0.12.5   quadprog_1.5-8      stringr_1.5.1      \n [16] crayon_1.5.3        pkgconfig_2.0.3     shape_1.4.6.1      \n [19] fastmap_1.2.0       backports_1.5.0     labeling_0.4.3     \n [22] ggraph_2.1.0        utf8_1.2.4          promises_1.3.0     \n [25] tzdb_0.4.0          UpSetR_1.4.0        purrr_1.0.2        \n [28] modeltools_0.2-23   ggseqlogo_0.2       later_1.3.2        \n [31] uuid_1.2-1          fpc_2.2-12          tweenr_2.0.3       \n [34] broom_1.0.6         parallel_4.3.3      prabclus_2.3-3     \n [37] cluster_2.1.6       R6_2.5.1            stringi_1.8.4      \n [40] RColorBrewer_1.1-3  rlist_0.4.6.2       car_3.1-2          \n [43] diptest_0.77-1      cellranger_1.1.0    Rcpp_1.0.13        \n [46] iterators_1.0.14    IRanges_2.36.0      readr_2.1.5        \n [49] httpuv_1.6.15       Matrix_1.6-5        nnet_7.3-19        \n [52] igraph_2.0.3        tidyselect_1.2.1    stringdist_0.9.12  \n [55] abind_1.4-5         viridis_0.6.5       doParallel_1.0.17  \n [58] codetools_0.2-20    lattice_0.22-6      plyr_1.8.9         \n [61] shiny_1.9.1         withr_3.0.1         polyclip_1.10-7    \n [64] circlize_0.4.16     mclust_6.1.1        kernlab_0.9-32     \n [67] pillar_1.9.0        ggpubr_0.6.0        carData_3.0-5      \n [70] foreach_1.5.2       stats4_4.3.3        generics_0.1.3     \n [73] S4Vectors_0.40.2    hms_1.1.3           munsell_0.5.1      \n [76] scales_1.3.0        xtable_1.8-4        class_7.3-22       \n [79] glue_1.7.0          pheatmap_1.0.12     tools_4.3.3        \n [82] robustbase_0.99-3   ggsignif_0.6.4      graphlayouts_1.1.0 \n [85] Cairo_1.6-2         fastmatch_1.1-4     tidygraph_1.3.0    \n [88] ape_5.8             colorspace_2.1-1    nlme_3.1-165       \n [91] ggforce_0.4.2       cli_3.6.3           fansi_1.0.6        \n [94] viridisLite_0.4.2   gtable_0.3.5        DEoptimR_1.1-3     \n [97] rstatix_0.7.2       digest_0.6.36       BiocGenerics_0.48.1\n[100] ggrepel_0.9.5       rjson_0.2.21        farver_2.1.2       \n[103] htmltools_0.5.8.1   lifecycle_1.0.4     factoextra_1.0.7   \n[106] GlobalOptions_0.1.2 mime_0.12           shinythemes_1.2.0  \n[109] MASS_7.3-60.0.1    \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 5.36\nuser 14.04\nsys 0.30\npercent 267%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TESSA",desc:"Tessa is a Bayesian model to integrate T cell receptor (TCR) sequence profiling with transcriptomes of T cells.",npages:1,report_toc:!0,order:14,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] igraph_2.0.3         Rtsne_0.17           LaplacesDemon_16.1.6\n [4] MASS_7.3-60.0.1      ggprism_1.0.5        Seurat_5.1.0        \n [7] SeuratObject_5.0.2   sp_2.1-4             tibble_3.2.1        \n[10] glue_1.7.0           immunarch_0.9.1      patchwork_1.2.0     \n[13] data.table_1.15.4    dtplyr_1.3.1         ggplot2_3.5.1       \n[16] tidyr_1.3.1          dplyr_1.1.4          jsonlite_1.8.8      \n[19] rlang_1.1.4          logger_0.3.0        \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22       shinythemes_1.2.0      splines_4.3.3         \n  [4] later_1.3.2            cellranger_1.1.0       polyclip_1.10-7       \n  [7] fastDummies_1.7.3      factoextra_1.0.7       lifecycle_1.0.4       \n [10] rstatix_0.7.2          doParallel_1.0.17      globals_0.16.3        \n [13] lattice_0.22-6         prabclus_2.3-3         backports_1.5.0       \n [16] magrittr_2.0.3         plotly_4.10.4          rlist_0.4.6.2         \n [19] httpuv_1.6.15          sctransform_0.4.1      spam_2.10-0           \n [22] flexmix_2.3-19         spatstat.sparse_3.1-0  reticulate_1.38.0     \n [25] cowplot_1.1.3          pbapply_1.7-2          RColorBrewer_1.1-3    \n [28] abind_1.4-5            quadprog_1.5-8         purrr_1.0.2           \n [31] ggraph_2.1.0           nnet_7.3-19            tweenr_2.0.3          \n [34] circlize_0.4.16        ggrepel_0.9.5          irlba_2.3.5.1         \n [37] listenv_0.9.1          spatstat.utils_3.0-5   pheatmap_1.0.12       \n [40] goftest_1.2-3          RSpectra_0.16-1        spatstat.random_3.2-3 \n [43] fitdistrplus_1.1-11    parallelly_1.38.0      leiden_0.4.3.1        \n [46] codetools_0.2-20       ggforce_0.4.2          tidyselect_1.2.1      \n [49] shape_1.4.6.1          ggseqlogo_0.2          farver_2.1.2          \n [52] viridis_0.6.5          matrixStats_1.1.0      stats4_4.3.3          \n [55] spatstat.explore_3.2-6 tidygraph_1.3.0        progressr_0.14.0      \n [58] ggridges_0.5.6         ggalluvial_0.12.5      survival_3.7-0        \n [61] iterators_1.0.14       foreach_1.5.2          tools_4.3.3           \n [64] stringdist_0.9.12      ica_1.0-3              Rcpp_1.0.13           \n [67] gridExtra_2.3          withr_3.0.1            fastmap_1.2.0         \n [70] fansi_1.0.6            digest_0.6.36          R6_2.5.1              \n [73] mime_0.12              colorspace_2.1-1       scattermore_1.2       \n [76] tensor_1.5             spatstat.data_3.1-2    diptest_0.77-1        \n [79] UpSetR_1.4.0           utf8_1.2.4             generics_0.1.3        \n [82] robustbase_0.99-3      class_7.3-22           graphlayouts_1.1.0    \n [85] httr_1.4.7             htmlwidgets_1.6.4      uwot_0.1.16           \n [88] pkgconfig_2.0.3        gtable_0.3.5           modeltools_0.2-23     \n [91] lmtest_0.9-40          htmltools_0.5.8.1      carData_3.0-5         \n [94] dotCall64_1.1-1        scales_1.3.0           png_0.1-8             \n [97] tzdb_0.4.0             reshape2_1.4.4         uuid_1.2-1            \n[100] nlme_3.1-165           zoo_1.8-12             GlobalOptions_0.1.2   \n[103] stringr_1.5.1          KernSmooth_2.23-24     parallel_4.3.3        \n[106] miniUI_0.1.1.1         pillar_1.9.0           grid_4.3.3            \n[109] vctrs_0.6.5            RANN_2.6.1             promises_1.3.0        \n[112] ggpubr_0.6.0           car_3.1-2              xtable_1.8-4          \n[115] cluster_2.1.6          readr_2.1.5            cli_3.6.3             \n[118] compiler_4.3.3         future.apply_1.11.2    ggsignif_0.6.4        \n[121] labeling_0.4.3         mclust_6.1.1           plyr_1.8.9            \n[124] stringi_1.8.4          deldir_2.0-4           viridisLite_0.4.2     \n[127] munsell_0.5.1          lazyeval_0.2.2         spatstat.geom_3.2-9   \n[130] Matrix_1.6-5           RcppHNSW_0.6.0         hms_1.1.3             \n[133] future_1.34.0          fpc_2.2-12             shiny_1.9.1           \n[136] kernlab_0.9-32         ROCR_1.0-11            broom_1.0.6           \n[139] phangorn_2.11.1        fastmatch_1.1-4        DEoptimR_1.1-3        \n[142] readxl_1.4.3           ape_5.8               \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 1774.00\nuser 8784.42\nsys 20.96\npercent 496%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"SeuratClusterStats",desc:"Statistics of the clustering.",npages:1,report_toc:!0,order:17,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] grid      stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] clustree_0.5.1        ggraph_2.1.0          ComplexHeatmap_2.18.0\n [4] circlize_0.4.16       tidyseurat_0.8.0      ttservice_0.4.1      \n [7] ggrepel_0.9.5         ggprism_1.0.5         tibble_3.2.1         \n[10] Seurat_5.1.0          SeuratObject_5.0.2    sp_2.1-4             \n[13] ggplot2_3.5.1         tidyr_1.3.1           dplyr_1.1.4          \n[16] tidyselect_1.2.1      jsonlite_1.8.8        rlang_1.1.4          \n[19] logger_0.3.0         \n\nloaded via a namespace (and not attached):\n  [1] RColorBrewer_1.1-3     shape_1.4.6.1          magrittr_2.0.3        \n  [4] ggbeeswarm_0.7.2       spatstat.utils_3.0-5   farver_2.1.2          \n  [7] GlobalOptions_0.1.2    vctrs_0.6.5            ROCR_1.0-11           \n [10] spatstat.explore_3.2-6 htmltools_0.5.8.1      sctransform_0.4.1     \n [13] parallelly_1.38.0      KernSmooth_2.23-24     htmlwidgets_1.6.4     \n [16] ica_1.0-3              plyr_1.8.9             plotly_4.10.4         \n [19] zoo_1.8-12             igraph_2.0.3           mime_0.12             \n [22] lifecycle_1.0.4        iterators_1.0.14       pkgconfig_2.0.3       \n [25] Matrix_1.6-5           R6_2.5.1               fastmap_1.2.0         \n [28] clue_0.3-65            fitdistrplus_1.1-11    future_1.34.0         \n [31] shiny_1.9.1            digest_0.6.36          colorspace_2.1-1      \n [34] patchwork_1.2.0        S4Vectors_0.40.2       tensor_1.5            \n [37] RSpectra_0.16-1        irlba_2.3.5.1          labeling_0.4.3        \n [40] progressr_0.14.0       fansi_1.0.6            spatstat.sparse_3.1-0 \n [43] httr_1.4.7             polyclip_1.10-7        abind_1.4-5           \n [46] compiler_4.3.3         withr_3.0.1            doParallel_1.0.17     \n [49] viridis_0.6.5          fastDummies_1.7.3      ggforce_0.4.2         \n [52] MASS_7.3-60.0.1        rjson_0.2.21           tools_4.3.3           \n [55] vipor_0.4.7            lmtest_0.9-40          beeswarm_0.4.0        \n [58] httpuv_1.6.15          future.apply_1.11.2    goftest_1.2-3         \n [61] glue_1.7.0             nlme_3.1-165           promises_1.3.0        \n [64] Rtsne_0.17             cluster_2.1.6          reshape2_1.4.4        \n [67] generics_0.1.3         gtable_0.3.5           spatstat.data_3.1-2   \n [70] data.table_1.15.4      tidygraph_1.3.0        utf8_1.2.4            \n [73] BiocGenerics_0.48.1    spatstat.geom_3.2-9    RcppAnnoy_0.0.22      \n [76] RANN_2.6.1             foreach_1.5.2          pillar_1.9.0          \n [79] stringr_1.5.1          spam_2.10-0            RcppHNSW_0.6.0        \n [82] later_1.3.2            splines_4.3.3          tweenr_2.0.3          \n [85] lattice_0.22-6         survival_3.7-0         deldir_2.0-4          \n [88] miniUI_0.1.1.1         pbapply_1.7-2          gridExtra_2.3         \n [91] IRanges_2.36.0         scattermore_1.2        stats4_4.3.3          \n [94] graphlayouts_1.1.0     matrixStats_1.1.0      stringi_1.8.4         \n [97] lazyeval_0.2.2         codetools_0.2-20       cli_3.6.3             \n[100] uwot_0.1.16            xtable_1.8-4           reticulate_1.38.0     \n[103] munsell_0.5.1          Rcpp_1.0.13            globals_0.16.3        \n[106] spatstat.random_3.2-3  png_0.1-8              ggrastr_1.0.2         \n[109] parallel_4.3.3         ellipsis_0.3.2         dotCall64_1.1-1       \n[112] listenv_0.9.1          viridisLite_0.4.2      scales_1.3.0          \n[115] ggridges_0.5.6         leiden_0.4.3.1         purrr_1.0.2           \n[118] crayon_1.5.3           GetoptLong_1.0.5       cowplot_1.1.3         \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 15.10\nuser 23.39\nsys 0.59\npercent 158%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"Immunarch",desc:"Exploration of Single-cell and Bulk T-cell/Antibody Immune Repertoires",npages:3,report_toc:!0,order:18,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] circlize_0.4.16   tibble_3.2.1      glue_1.7.0        ggprism_1.0.5    \n [5] immunarch_0.9.1   patchwork_1.2.0   data.table_1.15.4 dtplyr_1.3.1     \n [9] ggplot2_3.5.1     tidyr_1.3.1       dplyr_1.1.4       jsonlite_1.8.8   \n[13] rlang_1.1.4       logger_0.3.0     \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22       shinythemes_1.2.0      splines_4.3.3         \n  [4] later_1.3.2            cellranger_1.1.0       polyclip_1.10-7       \n  [7] fastDummies_1.7.3      factoextra_1.0.7       lifecycle_1.0.4       \n [10] rstatix_0.7.2          doParallel_1.0.17      globals_0.16.3        \n [13] lattice_0.22-6         prabclus_2.3-3         MASS_7.3-60.0.1       \n [16] backports_1.5.0        magrittr_2.0.3         plotly_4.10.4         \n [19] rlist_0.4.6.2          httpuv_1.6.15          Seurat_5.1.0          \n [22] sctransform_0.4.1      spam_2.10-0            flexmix_2.3-19        \n [25] spatstat.sparse_3.1-0  sp_2.1-4               reticulate_1.38.0     \n [28] cowplot_1.1.3          pbapply_1.7-2          RColorBrewer_1.1-3    \n [31] abind_1.4-5            Rtsne_0.17             quadprog_1.5-8        \n [34] purrr_1.0.2            ggraph_2.1.0           nnet_7.3-19           \n [37] tweenr_2.0.3           ggrepel_0.9.5          irlba_2.3.5.1         \n [40] listenv_0.9.1          spatstat.utils_3.0-5   pheatmap_1.0.12       \n [43] goftest_1.2-3          RSpectra_0.16-1        spatstat.random_3.2-3 \n [46] fitdistrplus_1.1-11    parallelly_1.38.0      leiden_0.4.3.1        \n [49] codetools_0.2-20       ggforce_0.4.2          tidyselect_1.2.1      \n [52] shape_1.4.6.1          ggseqlogo_0.2          farver_2.1.2          \n [55] viridis_0.6.5          matrixStats_1.1.0      stats4_4.3.3          \n [58] spatstat.explore_3.2-6 tidygraph_1.3.0        progressr_0.14.0      \n [61] ggridges_0.5.6         ggalluvial_0.12.5      survival_3.7-0        \n [64] iterators_1.0.14       foreach_1.5.2          tools_4.3.3           \n [67] stringdist_0.9.12      ica_1.0-3              Rcpp_1.0.13           \n [70] gridExtra_2.3          withr_3.0.1            fastmap_1.2.0         \n [73] fansi_1.0.6            digest_0.6.36          R6_2.5.1              \n [76] mime_0.12              colorspace_2.1-1       scattermore_1.2       \n [79] tensor_1.5             spatstat.data_3.1-2    diptest_0.77-1        \n [82] UpSetR_1.4.0           utf8_1.2.4             generics_0.1.3        \n [85] robustbase_0.99-3      class_7.3-22           graphlayouts_1.1.0    \n [88] httr_1.4.7             htmlwidgets_1.6.4      uwot_0.1.16           \n [91] pkgconfig_2.0.3        gtable_0.3.5           modeltools_0.2-23     \n [94] lmtest_0.9-40          htmltools_0.5.8.1      carData_3.0-5         \n [97] dotCall64_1.1-1        SeuratObject_5.0.2     scales_1.3.0          \n[100] png_0.1-8              tzdb_0.4.0             reshape2_1.4.4        \n[103] uuid_1.2-1             nlme_3.1-165           zoo_1.8-12            \n[106] GlobalOptions_0.1.2    stringr_1.5.1          KernSmooth_2.23-24    \n[109] parallel_4.3.3         miniUI_0.1.1.1         pillar_1.9.0          \n[112] grid_4.3.3             vctrs_0.6.5            RANN_2.6.1            \n[115] promises_1.3.0         ggpubr_0.6.0           car_3.1-2             \n[118] xtable_1.8-4           cluster_2.1.6          readr_2.1.5           \n[121] cli_3.6.3              compiler_4.3.3         future.apply_1.11.2   \n[124] ggsignif_0.6.4         labeling_0.4.3         mclust_6.1.1          \n[127] plyr_1.8.9             stringi_1.8.4          deldir_2.0-4          \n[130] viridisLite_0.4.2      assertthat_0.2.1       munsell_0.5.1         \n[133] lazyeval_0.2.2         spatstat.geom_3.2-9    Matrix_1.6-5          \n[136] RcppHNSW_0.6.0         hms_1.1.3              future_1.34.0         \n[139] fpc_2.2-12             shiny_1.9.1            kernlab_0.9-32        \n[142] ROCR_1.0-11            igraph_2.0.3           broom_1.0.6           \n[145] phangorn_2.11.1        fastmatch_1.1-4        DEoptimR_1.1-3        \n[148] readxl_1.4.3           ape_5.8               \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 91.14\nuser 113.47\nsys 1.29\npercent 125%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"RadarPlots",desc:"Radar plots for cell proportion in different clusters.",npages:1,report_toc:!0,order:19,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ggprism_1.0.5      ggradar_0.2        ggplot2_3.5.1      tibble_3.2.1      \n [5] tidyr_1.3.1        dplyr_1.1.4        Seurat_5.1.0       SeuratObject_5.0.2\n [9] sp_2.1-4           jsonlite_1.8.8     rlang_1.1.4        logger_0.3.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.2-9   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] labeling_0.4.3         utf8_1.2.4             promises_1.3.0        \n [19] purrr_1.0.2            goftest_1.2-3          later_1.3.2           \n [22] spatstat.utils_3.0-5   irlba_2.3.5.1          parallel_4.3.3        \n [25] cluster_2.1.6          R6_2.5.1               ica_1.0-3             \n [28] stringi_1.8.4          RColorBrewer_1.1-3     spatstat.data_3.1-2   \n [31] reticulate_1.38.0      parallelly_1.38.0      lmtest_0.9-40         \n [34] scattermore_1.2        Rcpp_1.0.13            tensor_1.5            \n [37] future.apply_1.11.2    zoo_1.8-12             sctransform_0.4.1     \n [40] httpuv_1.6.15          Matrix_1.6-5           splines_4.3.3         \n [43] igraph_2.0.3           tidyselect_1.2.1       abind_1.4-5           \n [46] spatstat.random_3.2-3  codetools_0.2-20       miniUI_0.1.1.1        \n [49] spatstat.explore_3.2-6 listenv_0.9.1          lattice_0.22-6        \n [52] plyr_1.8.9             withr_3.0.1            shiny_1.9.1           \n [55] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n [58] fastDummies_1.7.3      survival_3.7-0         polyclip_1.10-7       \n [61] fitdistrplus_1.1-11    pillar_1.9.0           KernSmooth_2.23-24    \n [64] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n [67] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [70] xtable_1.8-4           glue_1.7.0             lazyeval_0.2.2        \n [73] tools_4.3.3            data.table_1.15.4      RSpectra_0.16-1       \n [76] RANN_2.6.1             leiden_0.4.3.1         dotCall64_1.1-1       \n [79] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n [82] nlme_3.1-165           patchwork_1.2.0        cli_3.6.3             \n [85] spatstat.sparse_3.1-0  spam_2.10-0            fansi_1.0.6           \n [88] viridisLite_0.4.2      uwot_0.1.16            gtable_0.3.5          \n [91] digest_0.6.36          progressr_0.14.0       ggrepel_0.9.5         \n [94] farver_2.1.2           htmlwidgets_1.6.4      htmltools_0.5.8.1     \n [97] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[100] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 7.79\nuser 16.36\nsys 0.40\npercent 215%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"CloneResidency",desc:"Identification of clone residency",npages:1,report_toc:!0,order:20,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ComplexUpset_1.3.3  ggVennDiagram_1.2.3 ggprism_1.0.5      \n [4] ggplot2_3.5.1       ggnewscale_0.5.0    glue_1.7.0         \n [7] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[10] jsonlite_1.8.8      rlang_1.1.4         logger_0.3.0       \n\nloaded via a namespace (and not attached):\n  [1] DBI_1.2.3              deldir_2.0-4           pbapply_1.7-2         \n  [4] gridExtra_2.3          magrittr_2.0.3         RcppAnnoy_0.0.22      \n  [7] e1071_1.7-14           spatstat.geom_3.2-9    matrixStats_1.1.0     \n [10] ggridges_0.5.6         compiler_4.3.3         png_0.1-8             \n [13] vctrs_0.6.5            reshape2_1.4.4         stringr_1.5.1         \n [16] pkgconfig_2.0.3        fastmap_1.2.0          labeling_0.4.3        \n [19] utf8_1.2.4             promises_1.3.0         purrr_1.0.2           \n [22] RVenn_1.1.0            goftest_1.2-3          later_1.3.2           \n [25] spatstat.utils_3.0-5   irlba_2.3.5.1          parallel_4.3.3        \n [28] cluster_2.1.6          R6_2.5.1               ica_1.0-3             \n [31] stringi_1.8.4          RColorBrewer_1.1-3     spatstat.data_3.1-2   \n [34] reticulate_1.38.0      parallelly_1.38.0      lmtest_0.9-40         \n [37] scattermore_1.2        Rcpp_1.0.13            tensor_1.5            \n [40] future.apply_1.11.2    zoo_1.8-12             sctransform_0.4.1     \n [43] httpuv_1.6.15          Matrix_1.6-5           splines_4.3.3         \n [46] igraph_2.0.3           tidyselect_1.2.1       abind_1.4-5           \n [49] spatstat.random_3.2-3  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.2-6 listenv_0.9.1          lattice_0.22-6        \n [55] plyr_1.8.9             shiny_1.9.1            withr_3.0.1           \n [58] ROCR_1.0-11            Rtsne_0.17             sf_1.0-16             \n [61] future_1.34.0          fastDummies_1.7.3      survival_3.7-0        \n [64] units_0.8-5            proxy_0.4-27           polyclip_1.10-7       \n [67] fitdistrplus_1.1-11    pillar_1.9.0           Seurat_5.1.0          \n [70] KernSmooth_2.23-24     plotly_4.10.4          generics_0.1.3        \n [73] RcppHNSW_0.6.0         sp_2.1-4               munsell_0.5.1         \n [76] scales_1.3.0           globals_0.16.3         xtable_1.8-4          \n [79] class_7.3-22           lazyeval_0.2.2         tools_4.3.3           \n [82] data.table_1.15.4      RSpectra_0.16-1        RANN_2.6.1            \n [85] leiden_0.4.3.1         dotCall64_1.1-1        cowplot_1.1.3         \n [88] grid_4.3.3             colorspace_2.1-1       nlme_3.1-165          \n [91] patchwork_1.2.0        cli_3.6.3              spatstat.sparse_3.1-0 \n [94] spam_2.10-0            fansi_1.0.6            viridisLite_0.4.2     \n [97] uwot_0.1.16            gtable_0.3.5           digest_0.6.36         \n[100] classInt_0.4-10        progressr_0.14.0       ggrepel_0.9.5         \n[103] farver_2.1.2           htmlwidgets_1.6.4      SeuratObject_5.0.2    \n[106] htmltools_0.5.8.1      lifecycle_1.0.4        httr_1.4.7            \n[109] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 16.51\nuser 24.80\nsys 0.51\npercent 153%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ScFGSEA",desc:"Gene set enrichment analysis for cells in different groups using `fgsea`",npages:2,report_toc:!0,order:21,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] fgsea_1.28.0       data.table_1.15.4  matrixStats_1.1.0  tidyseurat_0.8.0  \n [5] ttservice_0.4.1    Seurat_5.1.0       SeuratObject_5.0.2 sp_2.1-4          \n [9] tidyr_1.3.1        tidyselect_1.2.1   tibble_3.2.1       dplyr_1.1.4       \n[13] ggplot2_3.5.1      jsonlite_1.8.8     rlang_1.1.4        logger_0.3.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       ggridges_0.5.6        \n  [7] compiler_4.3.3         spatstat.geom_3.2-9    png_0.1-8             \n [10] vctrs_0.6.5            reshape2_1.4.4         stringr_1.5.1         \n [13] pkgconfig_2.0.3        fastmap_1.2.0          ellipsis_0.3.2        \n [16] labeling_0.4.3         utf8_1.2.4             promises_1.3.0        \n [19] purrr_1.0.2            goftest_1.2-3          later_1.3.2           \n [22] BiocParallel_1.36.0    spatstat.utils_3.0-5   irlba_2.3.5.1         \n [25] parallel_4.3.3         cluster_2.1.6          R6_2.5.1              \n [28] ica_1.0-3              stringi_1.8.4          RColorBrewer_1.1-3    \n [31] spatstat.data_3.1-2    reticulate_1.38.0      parallelly_1.38.0     \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.13           \n [37] tensor_1.5             future.apply_1.11.2    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [43] splines_4.3.3          igraph_2.0.3           abind_1.4-5           \n [46] spatstat.random_3.2-3  codetools_0.2-20       miniUI_0.1.1.1        \n [49] spatstat.explore_3.2-6 listenv_0.9.1          lattice_0.22-6        \n [52] plyr_1.8.9             shiny_1.9.1            withr_3.0.1           \n [55] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n [58] fastDummies_1.7.3      survival_3.7-0         polyclip_1.10-7       \n [61] fitdistrplus_1.1-11    pillar_1.9.0           KernSmooth_2.23-24    \n [64] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n [67] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [70] xtable_1.8-4           glue_1.7.0             lazyeval_0.2.2        \n [73] tools_4.3.3            RSpectra_0.16-1        RANN_2.6.1            \n [76] leiden_0.4.3.1         dotCall64_1.1-1        fastmatch_1.1-4       \n [79] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n [82] nlme_3.1-165           patchwork_1.2.0        cli_3.6.3             \n [85] spatstat.sparse_3.1-0  spam_2.10-0            fansi_1.0.6           \n [88] viridisLite_0.4.2      uwot_0.1.16            gtable_0.3.5          \n [91] digest_0.6.36          progressr_0.14.0       ggrepel_0.9.5         \n [94] farver_2.1.2           htmlwidgets_1.6.4      htmltools_0.5.8.1     \n [97] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[100] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 96.79\nuser 103.64\nsys 1.43\npercent 108%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"MarkersFinder",desc:"Find markers between different groups of cells",npages:1,report_toc:!0,order:22,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] UpSetR_1.4.0        ggVennDiagram_1.2.3 tidyseurat_0.8.0   \n [4] ttservice_0.4.1     future_1.34.0       ggrepel_0.9.5      \n [7] ggprism_1.0.5       ggplot2_3.5.1       enrichR_3.2        \n[10] Seurat_5.1.0        SeuratObject_5.0.2  sp_2.1-4           \n[13] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[16] tidyselect_1.2.1    digest_0.6.36       jsonlite_1.8.8     \n[19] rlang_1.1.4         logger_0.3.0       \n\nloaded via a namespace (and not attached):\n  [1] RColorBrewer_1.1-3     magrittr_2.0.3         spatstat.utils_3.0-5  \n  [4] farver_2.1.2           vctrs_0.6.5            ROCR_1.0-11           \n  [7] spatstat.explore_3.2-6 htmltools_0.5.8.1      curl_5.2.1            \n [10] sctransform_0.4.1      parallelly_1.38.0      KernSmooth_2.23-24    \n [13] htmlwidgets_1.6.4      ica_1.0-3              plyr_1.8.9            \n [16] plotly_4.10.4          zoo_1.8-12             igraph_2.0.3          \n [19] mime_0.12              lifecycle_1.0.4        pkgconfig_2.0.3       \n [22] Matrix_1.6-5           R6_2.5.1               fastmap_1.2.0         \n [25] fitdistrplus_1.1-11    shiny_1.9.1            colorspace_2.1-1      \n [28] patchwork_1.2.0        tensor_1.5             RSpectra_0.16-1       \n [31] irlba_2.3.5.1          WriteXLS_6.7.0         labeling_0.4.3        \n [34] progressr_0.14.0       fansi_1.0.6            spatstat.sparse_3.1-0 \n [37] httr_1.4.7             polyclip_1.10-7        abind_1.4-5           \n [40] compiler_4.3.3         withr_3.0.1            fastDummies_1.7.3     \n [43] MASS_7.3-60.0.1        rjson_0.2.21           tools_4.3.3           \n [46] lmtest_0.9-40          httpuv_1.6.15          future.apply_1.11.2   \n [49] goftest_1.2-3          glue_1.7.0             nlme_3.1-165          \n [52] promises_1.3.0         grid_4.3.3             Rtsne_0.17            \n [55] cluster_2.1.6          reshape2_1.4.4         generics_0.1.3        \n [58] gtable_0.3.5           spatstat.data_3.1-2    data.table_1.15.4     \n [61] utf8_1.2.4             spatstat.geom_3.2-9    RcppAnnoy_0.0.22      \n [64] RANN_2.6.1             pillar_1.9.0           stringr_1.5.1         \n [67] spam_2.10-0            RcppHNSW_0.6.0         limma_3.58.1          \n [70] later_1.3.2            splines_4.3.3          lattice_0.22-6        \n [73] survival_3.7-0         deldir_2.0-4           miniUI_0.1.1.1        \n [76] pbapply_1.7-2          gridExtra_2.3          scattermore_1.2       \n [79] statmod_1.5.0          matrixStats_1.1.0      stringi_1.8.4         \n [82] lazyeval_0.2.2         codetools_0.2-20       RVenn_1.1.0           \n [85] cli_3.6.3              uwot_0.1.16            xtable_1.8-4          \n [88] reticulate_1.38.0      munsell_0.5.1          Rcpp_1.0.13           \n [91] globals_0.16.3         spatstat.random_3.2-3  png_0.1-8             \n [94] parallel_4.3.3         ellipsis_0.3.2         presto_1.0.0          \n [97] dotCall64_1.1-1        listenv_0.9.1          viridisLite_0.4.2     \n[100] scales_1.3.0           ggridges_0.5.6         leiden_0.4.3.1        \n[103] purrr_1.0.2            cowplot_1.1.3         \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 15.26\nuser 20.73\nsys 0.60\npercent 139%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"MetaMarkers",desc:"Find markers between three or more groups of cells, using one-way ANOVA or Kruskal-Wallis test.",npages:2,report_toc:!0,order:23,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] parallel  stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] tidyseurat_0.8.0   ttservice_0.4.1    ggprism_1.0.5      ggplot2_3.5.1     \n [5] enrichR_3.2        Seurat_5.1.0       SeuratObject_5.0.2 sp_2.1-4          \n [9] Matrix_1.6-5       tibble_3.2.1       tidyr_1.3.1        dplyr_1.1.4       \n[13] tidyselect_1.2.1   jsonlite_1.8.8     rlang_1.1.4        logger_0.3.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.2-9   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] ellipsis_0.3.2         labeling_0.4.3         utf8_1.2.4            \n [19] promises_1.3.0         purrr_1.0.2            WriteXLS_6.7.0        \n [22] goftest_1.2-3          later_1.3.2            spatstat.utils_3.0-5  \n [25] irlba_2.3.5.1          cluster_2.1.6          R6_2.5.1              \n [28] ica_1.0-3              stringi_1.8.4          RColorBrewer_1.1-3    \n [31] spatstat.data_3.1-2    reticulate_1.38.0      parallelly_1.38.0     \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.13           \n [37] tensor_1.5             future.apply_1.11.2    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          splines_4.3.3         \n [43] igraph_2.0.3           abind_1.4-5            spatstat.random_3.2-3 \n [46] codetools_0.2-20       miniUI_0.1.1.1         spatstat.explore_3.2-6\n [49] curl_5.2.1             listenv_0.9.1          lattice_0.22-6        \n [52] plyr_1.8.9             withr_3.0.1            shiny_1.9.1           \n [55] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n [58] fastDummies_1.7.3      survival_3.7-0         polyclip_1.10-7       \n [61] fitdistrplus_1.1-11    pillar_1.9.0           KernSmooth_2.23-24    \n [64] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n [67] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [70] xtable_1.8-4           glue_1.7.0             lazyeval_0.2.2        \n [73] tools_4.3.3            data.table_1.15.4      RSpectra_0.16-1       \n [76] RANN_2.6.1             leiden_0.4.3.1         dotCall64_1.1-1       \n [79] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n [82] nlme_3.1-165           patchwork_1.2.0        cli_3.6.3             \n [85] spatstat.sparse_3.1-0  spam_2.10-0            fansi_1.0.6           \n [88] viridisLite_0.4.2      uwot_0.1.16            gtable_0.3.5          \n [91] digest_0.6.36          progressr_0.14.0       ggrepel_0.9.5         \n [94] farver_2.1.2           rjson_0.2.21           htmlwidgets_1.6.4     \n [97] htmltools_0.5.8.1      lifecycle_1.0.4        httr_1.4.7            \n[100] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 335.98\nuser 1633.65\nsys 90.08\npercent 513%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"CellsDistribution",desc:"Distribution of cells (i.e. in a TCR clone) from different groups for each cluster",npages:1,report_toc:!0,order:24,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] grid      stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] ComplexHeatmap_2.18.0 circlize_0.4.16       UpSetR_1.4.0         \n [4] ggVennDiagram_1.2.3   patchwork_1.2.0       ggplot2_3.5.1        \n [7] tibble_3.2.1          Seurat_5.1.0          SeuratObject_5.0.2   \n[10] sp_2.1-4              tidyr_1.3.1           dplyr_1.1.4          \n[13] tidyselect_1.2.1      jsonlite_1.8.8        rlang_1.1.4          \n[16] logger_0.3.0         \n\nloaded via a namespace (and not attached):\n  [1] RColorBrewer_1.1-3     shape_1.4.6.1          magrittr_2.0.3        \n  [4] spatstat.utils_3.0-5   farver_2.1.2           GlobalOptions_0.1.2   \n  [7] vctrs_0.6.5            ROCR_1.0-11            Cairo_1.6-2           \n [10] spatstat.explore_3.2-6 htmltools_0.5.8.1      sctransform_0.4.1     \n [13] parallelly_1.38.0      KernSmooth_2.23-24     htmlwidgets_1.6.4     \n [16] ica_1.0-3              plyr_1.8.9             plotly_4.10.4         \n [19] zoo_1.8-12             igraph_2.0.3           mime_0.12             \n [22] lifecycle_1.0.4        iterators_1.0.14       pkgconfig_2.0.3       \n [25] Matrix_1.6-5           R6_2.5.1               fastmap_1.2.0         \n [28] clue_0.3-65            fitdistrplus_1.1-11    future_1.34.0         \n [31] shiny_1.9.1            digest_0.6.36          colorspace_2.1-1      \n [34] S4Vectors_0.40.2       tensor_1.5             RSpectra_0.16-1       \n [37] irlba_2.3.5.1          labeling_0.4.3         progressr_0.14.0      \n [40] fansi_1.0.6            spatstat.sparse_3.1-0  httr_1.4.7            \n [43] polyclip_1.10-7        abind_1.4-5            compiler_4.3.3        \n [46] withr_3.0.1            doParallel_1.0.17      fastDummies_1.7.3     \n [49] MASS_7.3-60.0.1        rjson_0.2.21           tools_4.3.3           \n [52] lmtest_0.9-40          httpuv_1.6.15          future.apply_1.11.2   \n [55] goftest_1.2-3          glue_1.7.0             nlme_3.1-165          \n [58] promises_1.3.0         Rtsne_0.17             cluster_2.1.6         \n [61] reshape2_1.4.4         generics_0.1.3         gtable_0.3.5          \n [64] spatstat.data_3.1-2    data.table_1.15.4      utf8_1.2.4            \n [67] BiocGenerics_0.48.1    spatstat.geom_3.2-9    RcppAnnoy_0.0.22      \n [70] ggrepel_0.9.5          RANN_2.6.1             foreach_1.5.2         \n [73] pillar_1.9.0           stringr_1.5.1          spam_2.10-0           \n [76] RcppHNSW_0.6.0         later_1.3.2            splines_4.3.3         \n [79] lattice_0.22-6         survival_3.7-0         deldir_2.0-4          \n [82] miniUI_0.1.1.1         pbapply_1.7-2          gridExtra_2.3         \n [85] IRanges_2.36.0         scattermore_1.2        stats4_4.3.3          \n [88] matrixStats_1.1.0      stringi_1.8.4          lazyeval_0.2.2        \n [91] codetools_0.2-20       RVenn_1.1.0            cli_3.6.3             \n [94] uwot_0.1.16            xtable_1.8-4           reticulate_1.38.0     \n [97] munsell_0.5.1          Rcpp_1.0.13            globals_0.16.3        \n[100] spatstat.random_3.2-3  png_0.1-8              parallel_4.3.3        \n[103] dotCall64_1.1-1        listenv_0.9.1          viridisLite_0.4.2     \n[106] scales_1.3.0           ggridges_0.5.6         leiden_0.4.3.1        \n[109] purrr_1.0.2            crayon_1.5.3           GetoptLong_1.0.5      \n[112] cowplot_1.1.3         \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 10.15\nuser 18.49\nsys 0.52\npercent 187%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"CDR3AAPhyschem",desc:"CDR3 AA physicochemical feature analysis",npages:1,report_toc:!0,order:25,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] stringr_1.5.1       broom.mixed_0.2.9.5 glmnet_4.1-8       \n [4] Matrix_1.6-5        hash_3.0.1          glue_1.7.0         \n [7] ggridges_0.5.6      ggplot2_3.5.1       tibble_3.2.1       \n[10] tidyr_1.3.1         dplyr_1.1.4         jsonlite_1.8.8     \n[13] rlang_1.1.4         logger_0.3.0       \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         furrr_0.3.1            RcppAnnoy_0.0.22      \n  [7] spatstat.geom_3.2-9    matrixStats_1.1.0      compiler_4.3.3        \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] pkgconfig_2.0.3        shape_1.4.6.1          fastmap_1.2.0         \n [16] backports_1.5.0        labeling_0.4.3         utf8_1.2.4            \n [19] promises_1.3.0         purrr_1.0.2            goftest_1.2-3         \n [22] later_1.3.2            spatstat.utils_3.0-5   broom_1.0.6           \n [25] irlba_2.3.5.1          parallel_4.3.3         cluster_2.1.6         \n [28] R6_2.5.1               ica_1.0-3              spatstat.data_3.1-2   \n [31] stringi_1.8.4          RColorBrewer_1.1-3     reticulate_1.38.0     \n [34] parallelly_1.38.0      lmtest_0.9-40          scattermore_1.2       \n [37] Rcpp_1.0.13            iterators_1.0.14       tensor_1.5            \n [40] future.apply_1.11.2    zoo_1.8-12             sctransform_0.4.1     \n [43] httpuv_1.6.15          splines_4.3.3          igraph_2.0.3          \n [46] tidyselect_1.2.1       abind_1.4-5            spatstat.random_3.2-3 \n [49] codetools_0.2-20       miniUI_0.1.1.1         spatstat.explore_3.2-6\n [52] listenv_0.9.1          lattice_0.22-6         plyr_1.8.9            \n [55] shiny_1.9.1            withr_3.0.1            ROCR_1.0-11           \n [58] Rtsne_0.17             future_1.34.0          fastDummies_1.7.3     \n [61] survival_3.7-0         polyclip_1.10-7        fitdistrplus_1.1-11   \n [64] pillar_1.9.0           Seurat_5.1.0           KernSmooth_2.23-24    \n [67] foreach_1.5.2          plotly_4.10.4          generics_0.1.3        \n [70] RcppHNSW_0.6.0         sp_2.1-4               munsell_0.5.1         \n [73] scales_1.3.0           globals_0.16.3         xtable_1.8-4          \n [76] lazyeval_0.2.2         tools_4.3.3            data.table_1.15.4     \n [79] RSpectra_0.16-1        forcats_1.0.0          RANN_2.6.1            \n [82] leiden_0.4.3.1         dotCall64_1.1-1        cowplot_1.1.3         \n [85] grid_4.3.3             colorspace_2.1-1       nlme_3.1-165          \n [88] patchwork_1.2.0        cli_3.6.3              spatstat.sparse_3.1-0 \n [91] spam_2.10-0            fansi_1.0.6            viridisLite_0.4.2     \n [94] uwot_0.1.16            gtable_0.3.5           digest_0.6.36         \n [97] progressr_0.14.0       ggrepel_0.9.5          farver_2.1.2          \n[100] htmlwidgets_1.6.4      SeuratObject_5.0.2     htmltools_0.5.8.1     \n[103] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[106] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 10.75\nuser 19.09\nsys 0.55\npercent 182%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ScrnaMetabolicLandscape",desc:"Metabolic landscape analysis for scRNA-seq data",order:28,procs:[{name:"MetabolicPathwayActivity",desc:"This process calculates the pathway activities in different groups and subsets.",npages:1,report_toc:!0,order:28,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n [1] grid      parallel  stats4    stats     graphics  grDevices utils    \n [8] datasets  methods   base     \n\nother attached packages:\n [1] ComplexHeatmap_2.18.0       Seurat_5.1.0               \n [3] SeuratObject_5.0.2          sp_2.1-4                   \n [5] ggprism_1.0.5               RColorBrewer_1.1-3         \n [7] reshape2_1.4.4              scater_1.30.1              \n [9] scuttle_1.12.0              SingleCellExperiment_1.24.0\n[11] SummarizedExperiment_1.32.0 Biobase_2.62.0             \n[13] GenomicRanges_1.54.1        GenomeInfoDb_1.38.1        \n[15] IRanges_2.36.0              S4Vectors_0.40.2           \n[17] BiocGenerics_0.48.1         MatrixGenerics_1.14.0      \n[19] matrixStats_1.1.0           tibble_3.2.1               \n[21] dplyr_1.1.4                 ggplot2_3.5.1              \n[23] jsonlite_1.8.8              rlang_1.1.4                \n[25] logger_0.3.0               \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22          splines_4.3.3            \n  [3] later_1.3.2               bitops_1.0-8             \n  [5] polyclip_1.10-7           fastDummies_1.7.3        \n  [7] lifecycle_1.0.4           doParallel_1.0.17        \n  [9] globals_0.16.3            lattice_0.22-6           \n [11] MASS_7.3-60.0.1           magrittr_2.0.3           \n [13] plotly_4.10.4             httpuv_1.6.15            \n [15] sctransform_0.4.1         spam_2.10-0              \n [17] spatstat.sparse_3.1-0     reticulate_1.38.0        \n [19] cowplot_1.1.3             pbapply_1.7-2            \n [21] abind_1.4-5               zlibbioc_1.48.0          \n [23] Rtsne_0.17                purrr_1.0.2              \n [25] RCurl_1.98-1.16           circlize_0.4.16          \n [27] GenomeInfoDbData_1.2.11   ggrepel_0.9.5            \n [29] irlba_2.3.5.1             listenv_0.9.1            \n [31] spatstat.utils_3.0-5      goftest_1.2-3            \n [33] RSpectra_0.16-1           spatstat.random_3.2-3    \n [35] fitdistrplus_1.1-11       parallelly_1.38.0        \n [37] DelayedMatrixStats_1.24.0 leiden_0.4.3.1           \n [39] codetools_0.2-20          DelayedArray_0.28.0      \n [41] shape_1.4.6.1             tidyselect_1.2.1         \n [43] farver_2.1.2              ScaledMatrix_1.10.0      \n [45] viridis_0.6.5             spatstat.explore_3.2-6   \n [47] GetoptLong_1.0.5          BiocNeighbors_1.20.0     \n [49] progressr_0.14.0          ggridges_0.5.6           \n [51] survival_3.7-0            iterators_1.0.14         \n [53] foreach_1.5.2             tools_4.3.3              \n [55] ica_1.0-3                 Rcpp_1.0.13              \n [57] glue_1.7.0                gridExtra_2.3            \n [59] SparseArray_1.2.2         withr_3.0.1              \n [61] fastmap_1.2.0             fansi_1.0.6              \n [63] digest_0.6.36             rsvd_1.0.5               \n [65] R6_2.5.1                  mime_0.12                \n [67] colorspace_2.1-1          Cairo_1.6-2              \n [69] scattermore_1.2           tensor_1.5               \n [71] spatstat.data_3.1-2       utf8_1.2.4               \n [73] tidyr_1.3.1               generics_0.1.3           \n [75] data.table_1.15.4         httr_1.4.7               \n [77] htmlwidgets_1.6.4         S4Arrays_1.2.0           \n [79] uwot_0.1.16               pkgconfig_2.0.3          \n [81] gtable_0.3.5              lmtest_0.9-40            \n [83] XVector_0.42.0            htmltools_0.5.8.1        \n [85] dotCall64_1.1-1           clue_0.3-65              \n [87] scales_1.3.0              png_0.1-8                \n [89] rjson_0.2.21              nlme_3.1-165             \n [91] zoo_1.8-12                GlobalOptions_0.1.2      \n [93] stringr_1.5.1             KernSmooth_2.23-24       \n [95] miniUI_0.1.1.1            vipor_0.4.7              \n [97] pillar_1.9.0              vctrs_0.6.5              \n [99] RANN_2.6.1                promises_1.3.0           \n[101] BiocSingular_1.18.0       beachmat_2.18.0          \n[103] xtable_1.8-4              cluster_2.1.6            \n[105] beeswarm_0.4.0            cli_3.6.3                \n[107] compiler_4.3.3            crayon_1.5.3             \n[109] future.apply_1.11.2       plyr_1.8.9               \n[111] ggbeeswarm_0.7.2          stringi_1.8.4            \n[113] viridisLite_0.4.2         deldir_2.0-4             \n[115] BiocParallel_1.36.0       munsell_0.5.1            \n[117] lazyeval_0.2.2            spatstat.geom_3.2-9      \n[119] Matrix_1.6-5              RcppHNSW_0.6.0           \n[121] patchwork_1.2.0           sparseMatrixStats_1.14.0 \n[123] future_1.34.0             shiny_1.9.1              \n[125] ROCR_1.0-11               igraph_2.0.3             \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 763.83\nuser 756.90\nsys 13.77\npercent 100%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3459.029\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"MetabolicFeatures",desc:"This process performs enrichment analysis for the metabolic pathways for each group in each subset.",npages:1,report_toc:!0,order:29,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] parallel  stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] matrixStats_1.1.0  fgsea_1.28.0       data.table_1.15.4  Seurat_5.1.0      \n [5] SeuratObject_5.0.2 sp_2.1-4           tibble_3.2.1       dplyr_1.1.4       \n [9] ggplot2_3.5.1      jsonlite_1.8.8     rlang_1.1.4        logger_0.3.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       ggridges_0.5.6        \n  [7] compiler_4.3.3         spatstat.geom_3.2-9    png_0.1-8             \n [10] vctrs_0.6.5            reshape2_1.4.4         stringr_1.5.1         \n [13] pkgconfig_2.0.3        fastmap_1.2.0          labeling_0.4.3        \n [16] utf8_1.2.4             promises_1.3.0         purrr_1.0.2           \n [19] goftest_1.2-3          later_1.3.2            BiocParallel_1.36.0   \n [22] spatstat.utils_3.0-5   irlba_2.3.5.1          cluster_2.1.6         \n [25] R6_2.5.1               ica_1.0-3              stringi_1.8.4         \n [28] RColorBrewer_1.1-3     spatstat.data_3.1-2    reticulate_1.38.0     \n [31] parallelly_1.38.0      lmtest_0.9-40          scattermore_1.2       \n [34] Rcpp_1.0.13            tensor_1.5             future.apply_1.11.2   \n [37] zoo_1.8-12             sctransform_0.4.1      httpuv_1.6.15         \n [40] Matrix_1.6-5           splines_4.3.3          igraph_2.0.3          \n [43] tidyselect_1.2.1       abind_1.4-5            spatstat.random_3.2-3 \n [46] codetools_0.2-20       miniUI_0.1.1.1         spatstat.explore_3.2-6\n [49] listenv_0.9.1          lattice_0.22-6         plyr_1.8.9            \n [52] shiny_1.9.1            withr_3.0.1            ROCR_1.0-11           \n [55] Rtsne_0.17             future_1.34.0          fastDummies_1.7.3     \n [58] survival_3.7-0         polyclip_1.10-7        fitdistrplus_1.1-11   \n [61] pillar_1.9.0           KernSmooth_2.23-24     plotly_4.10.4         \n [64] generics_0.1.3         RcppHNSW_0.6.0         munsell_0.5.1         \n [67] scales_1.3.0           globals_0.16.3         xtable_1.8-4          \n [70] glue_1.7.0             lazyeval_0.2.2         tools_4.3.3           \n [73] RSpectra_0.16-1        RANN_2.6.1             leiden_0.4.3.1        \n [76] dotCall64_1.1-1        fastmatch_1.1-4        cowplot_1.1.3         \n [79] grid_4.3.3             tidyr_1.3.1            colorspace_2.1-1      \n [82] nlme_3.1-165           patchwork_1.2.0        cli_3.6.3             \n [85] spatstat.sparse_3.1-0  spam_2.10-0            fansi_1.0.6           \n [88] viridisLite_0.4.2      uwot_0.1.16            gtable_0.3.5          \n [91] digest_0.6.36          progressr_0.14.0       ggrepel_0.9.5         \n [94] farver_2.1.2           htmlwidgets_1.6.4      htmltools_0.5.8.1     \n [97] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[100] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 188.82\nuser 192.45\nsys 7.25\npercent 105%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"MetabolicFeaturesIntraSubset",desc:"Intra-subset metabolic features - Enrichment analysis in details",npages:3,report_toc:!0,order:30,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats4    parallel  stats     graphics  grDevices utils     datasets \n[8] methods   base     \n\nother attached packages:\n [1] fgsea_1.28.0                data.table_1.15.4          \n [3] Seurat_5.1.0                SeuratObject_5.0.2         \n [5] sp_2.1-4                    scater_1.30.1              \n [7] scuttle_1.12.0              SingleCellExperiment_1.24.0\n [9] SummarizedExperiment_1.32.0 Biobase_2.62.0             \n[11] GenomicRanges_1.54.1        GenomeInfoDb_1.38.1        \n[13] IRanges_2.36.0              S4Vectors_0.40.2           \n[15] BiocGenerics_0.48.1         MatrixGenerics_1.14.0      \n[17] matrixStats_1.1.0           tibble_3.2.1               \n[19] dplyr_1.1.4                 ggplot2_3.5.1              \n[21] jsonlite_1.8.8              rlang_1.1.4                \n[23] logger_0.3.0               \n\nloaded via a namespace (and not attached):\n  [1] RColorBrewer_1.1-3        magrittr_2.0.3           \n  [3] spatstat.utils_3.0-5      ggbeeswarm_0.7.2         \n  [5] farver_2.1.2              zlibbioc_1.48.0          \n  [7] vctrs_0.6.5               ROCR_1.0-11              \n  [9] spatstat.explore_3.2-6    DelayedMatrixStats_1.24.0\n [11] RCurl_1.98-1.16           htmltools_0.5.8.1        \n [13] S4Arrays_1.2.0            BiocNeighbors_1.20.0     \n [15] SparseArray_1.2.2         sctransform_0.4.1        \n [17] parallelly_1.38.0         KernSmooth_2.23-24       \n [19] htmlwidgets_1.6.4         ica_1.0-3                \n [21] plyr_1.8.9                plotly_4.10.4            \n [23] zoo_1.8-12                igraph_2.0.3             \n [25] mime_0.12                 lifecycle_1.0.4          \n [27] pkgconfig_2.0.3           rsvd_1.0.5               \n [29] Matrix_1.6-5              R6_2.5.1                 \n [31] fastmap_1.2.0             GenomeInfoDbData_1.2.11  \n [33] fitdistrplus_1.1-11       future_1.34.0            \n [35] shiny_1.9.1               digest_0.6.36            \n [37] colorspace_2.1-1          patchwork_1.2.0          \n [39] tensor_1.5                RSpectra_0.16-1          \n [41] irlba_2.3.5.1             beachmat_2.18.0          \n [43] labeling_0.4.3            progressr_0.14.0         \n [45] spatstat.sparse_3.1-0     fansi_1.0.6              \n [47] polyclip_1.10-7           httr_1.4.7               \n [49] abind_1.4-5               compiler_4.3.3           \n [51] withr_3.0.1               BiocParallel_1.36.0      \n [53] viridis_0.6.5             fastDummies_1.7.3        \n [55] MASS_7.3-60.0.1           DelayedArray_0.28.0      \n [57] tools_4.3.3               vipor_0.4.7              \n [59] lmtest_0.9-40             beeswarm_0.4.0           \n [61] httpuv_1.6.15             future.apply_1.11.2      \n [63] goftest_1.2-3             glue_1.7.0               \n [65] nlme_3.1-165              promises_1.3.0           \n [67] grid_4.3.3                Rtsne_0.17               \n [69] cluster_2.1.6             reshape2_1.4.4           \n [71] generics_0.1.3            spatstat.data_3.1-2      \n [73] gtable_0.3.5              tidyr_1.3.1              \n [75] BiocSingular_1.18.0       ScaledMatrix_1.10.0      \n [77] utf8_1.2.4                XVector_0.42.0           \n [79] spatstat.geom_3.2-9       RcppAnnoy_0.0.22         \n [81] stringr_1.5.1             ggrepel_0.9.5            \n [83] RANN_2.6.1                pillar_1.9.0             \n [85] spam_2.10-0               RcppHNSW_0.6.0           \n [87] later_1.3.2               splines_4.3.3            \n [89] lattice_0.22-6            deldir_2.0-4             \n [91] survival_3.7-0            tidyselect_1.2.1         \n [93] miniUI_0.1.1.1            pbapply_1.7-2            \n [95] gridExtra_2.3             scattermore_1.2          \n [97] stringi_1.8.4             lazyeval_0.2.2           \n [99] codetools_0.2-20          cli_3.6.3                \n[101] uwot_0.1.16               xtable_1.8-4             \n[103] reticulate_1.38.0         munsell_0.5.1            \n[105] Rcpp_1.0.13               spatstat.random_3.2-3    \n[107] globals_0.16.3            png_0.1-8                \n[109] dotCall64_1.1-1           sparseMatrixStats_1.14.0 \n[111] bitops_1.0-8              listenv_0.9.1            \n[113] viridisLite_0.4.2         scales_1.3.0             \n[115] ggridges_0.5.6            leiden_0.4.3.1           \n[117] purrr_1.0.2               crayon_1.5.3             \n[119] fastmatch_1.1-4           cowplot_1.1.3            \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 99.35\nuser 104.68\nsys 3.22\npercent 108%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"MetabolicPathwayHeterogeneity",desc:"Calculate Metabolic Pathway heterogeneity.",npages:1,report_toc:!0,order:31}]}]};function ua(e){let t,n;return t=new zt({props:{dir:"up"}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function ga(e){let t,n;return t=new zt({props:{dir:"down"}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){h(t,e)}}}function fa(e){let t,n,a,s,i,r=e[1]>0&&ua();n=new _a({});let m=e[1]<e[3].npages-1&&ga();return{c(){r&&r.c(),t=v(),f(n.$$.fragment),a=v(),m&&m.c(),s=g()},m(e,o){r&&r.m(e,o),l(e,t,o),b(n,e,o),l(e,a,o),m&&m.m(e,o),l(e,s,o),i=!0},p(e,n){e[1]>0?r?2&n&&o(r,1):(r=ua(),r.c(),o(r,1),r.m(t.parentNode,t)):r&&(p(),_(r,1,1,(()=>{r=null})),c()),e[1]<e[3].npages-1?m?10&n&&o(m,1):(m=ga(),m.c(),o(m,1),m.m(s.parentNode,s)):m&&(p(),_(m,1,1,(()=>{m=null})),c())},i(e){i||(o(r),o(n.$$.fragment,e),o(m),i=!0)},o(e){_(r),_(n.$$.fragment,e),_(m),i=!1},d(e){e&&(d(t),d(a),d(s)),r&&r.d(e),h(n,e),m&&m.d(e)}}}function ba(e){let t,n,a;function s(t){e[4](t)}let i={logo:e[0],logotext:e[3].desc,versions:ma.versions,entries:ma.entries,pipeline_name:ma.pipeline.name,report_toc:e[3].report_toc,runinfo:e[3].runinfo,toc:da,page:e[1],$$slots:{default:[fa]},$$scope:{ctx:e}};return void 0!==e[2]&&(i.isSideNavOpen=e[2]),t=new Vt({props:i}),C.push((()=>y(t,"isSideNavOpen",s))),{c(){f(t.$$.fragment)},m(e,n){b(t,e,n),a=!0},p(e,[a]){const s={};1&a&&(s.logo=e[0]),8&a&&(s.logotext=e[3].desc),8&a&&(s.report_toc=e[3].report_toc),8&a&&(s.runinfo=e[3].runinfo),2&a&&(s.page=e[1]),42&a&&(s.$$scope={dirty:a,ctx:e}),!n&&4&a&&(n=!0,s.isSideNavOpen=e[2],x((()=>n=!1))),t.$set(s)},i(e){a||(o(t.$$.fragment,e),a=!0)},o(e){_(t.$$.fragment,e),a=!1},d(e){h(t,e)}}}function ha(e,t,n){let a,{name:s}=t,{page:i=0}=t,r=!1;for(const e of ma.entries)if(e.procs){for(const t of e.procs)if(t.name==s){a=t;break}}else if(e.name==s){a=e;break}return u((()=>{const e=new URL(window.location.href);let t;e.hash.length>1&&(t=document.getElementById(`${e.hash.slice(1)}`)),t&&t.scrollIntoView({block:"start",behavior:"smooth"})})),e.$$set=e=>{"name"in e&&n(0,s=e.name),"page"in e&&n(1,i=e.page)},[s,i,r,a,function(e){r=e,n(2,r)}]}e("default",new class extends t{constructor(e){super(),n(this,e,ha,ba,a,{name:0,page:1})}}({target:document.body,props:{name:"MetabolicFeaturesIntraSubset",page:2}}))}}}));
