System.register(["./vendor/svelte-CiOtOeYy.js","./vendor/carbon-components-svelte-BhQO6q4b.js","./vendor/carbon-icons-svelte-DVphRbRG.js"],(function(e){"use strict";var t,n,a,s,r,i,l,o,p,c,d,_,m,g,u,f,b,C,$,h,v,y,A,x,S,P,M,L,E,w,T,U,N,D,R,B,I,H,G,F,V,z,O,q,k,K,j,X,Y,W,Q,J,Z,ee,te,ne,ae,se,re,ie,le,oe,pe,ce,de,_e,me,ge,ue,fe,be,Ce,$e,he,ve,ye,Ae,xe,Se,Pe,Me,Le,Ee,we,Te,Ue,Ne,De,Re,Be,Ie,He;return{setters:[function(e){t=e.S,n=e.i,a=e.s,s=e.a1,r=e.l,i=e.O,l=e.a,o=e.b,p=e.g,c=e.c,d=e.t,_=e.d,m=e.a5,g=e.B,u=e.e,f=e.G,b=e.H,C=e.I,$=e.y,h=e.K,v=e.T,y=e.U,A=e.L,x=e.D,S=e.M,P=e.z,M=e.Q,L=e.k,E=e.u,w=e.n,T=e.o,U=e.f,N=e.A,D=e.h,R=e.j,B=e.aa,I=e.x,H=e.W,G=e.ab,F=e.m,V=e.p,z=e.a0,O=e.q,q=e.R,k=e.$,K=e.v,j=e.Y,X=e.Z,Y=e.F,W=e.ac,Q=e.w,J=e.J,Z=e.r},function(e){ee=e.S,te=e.a,ne=e.b,ae=e.H,se=e.c,re=e.T,ie=e.d,le=e.e,oe=e.f,pe=e.g,ce=e.h,de=e.M,_e=e.I,me=e.i,ge=e.j,ue=e.k,fe=e.C,be=e.l,Ce=e.m,$e=e.G,he=e.R,ve=e.n,ye=e.B,Ae=e.A,xe=e.o,Se=e.D,Pe=e.P,Me=e.p,Le=e.q,Ee=e.r,we=e.s,Te=e.t},function(e){Ue=e.C,Ne=e.a,De=e.O,Re=e.b,Be=e.c,Ie=e.D,He=e.d}],execute:function(){function Ge(e,t,n){const a=e.slice();return a[4]=t[n],a}function Fe(e,t,n){const a=e.slice();return a[7]=t[n],a}function Ve(e){let t,n;return t=new ee({props:{href:`${e[1](e[4].page)}#${e[4].slug}`,class:e[2](e[4].page),title:e[4].text,text:e[4].text,expanded:!0,$$slots:{default:[qe]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.href=`${e[1](e[4].page)}#${e[4].slug}`),1&n&&(a.class=e[2](e[4].page)),1&n&&(a.title=e[4].text),1&n&&(a.text=e[4].text),1025&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function ze(e){let t,n;return t=new te({props:{href:`${e[1](e[4].page)}#${e[4].slug}`,class:e[2](e[4].page),title:e[4].text,text:e[4].text}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.href=`${e[1](e[4].page)}#${e[4].slug}`),1&n&&(a.class=e[2](e[4].page)),1&n&&(a.title=e[4].text),1&n&&(a.text=e[4].text),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function Oe(e){let t,n;return t=new ne({props:{href:`${e[1](e[7].page)}#${e[7].slug}`,class:e[2](e[7].page),title:e[7].text,text:e[7].text}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.href=`${e[1](e[7].page)}#${e[7].slug}`),1&n&&(a.class=e[2](e[7].page)),1&n&&(a.title=e[7].text),1&n&&(a.text=e[7].text),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function qe(e){let t,n,a=s(e[4].children),r=[];for(let t=0;t<a.length;t+=1)r[t]=Oe(Fe(e,a,t));const i=e=>d(r[e],1,1,(()=>{r[e]=null}));return{c(){for(let e=0;e<r.length;e+=1)r[e].c();t=$()},m(e,a){for(let t=0;t<r.length;t+=1)r[t]&&r[t].m(e,a);l(e,t,a),n=!0},p(e,n){if(7&n){let l;for(a=s(e[4].children),l=0;l<a.length;l+=1){const s=Fe(e,a,l);r[l]?(r[l].p(s,n),o(r[l],1)):(r[l]=Oe(s),r[l].c(),o(r[l],1),r[l].m(t.parentNode,t))}for(p(),l=a.length;l<r.length;l+=1)i(l);c()}},i(e){if(!n){for(let e=0;e<a.length;e+=1)o(r[e]);n=!0}},o(e){r=r.filter(Boolean);for(let e=0;e<r.length;e+=1)d(r[e]);n=!1},d(e){e&&_(t),m(r,e)}}}function ke(e){let t,n,a,s;const r=[ze,Ve],i=[];function m(e,t){return 0==e[4].children.length?0:1}return t=m(e),n=i[t]=r[t](e),{c(){n.c(),a=u()},m(e,n){i[t].m(e,n),l(e,a,n),s=!0},p(e,s){let l=t;t=m(e),t===l?i[t].p(e,s):(p(),d(i[l],1,1,(()=>{i[l]=null})),c(),n=i[t],n?n.p(e,s):(n=i[t]=r[t](e),n.c()),o(n,1),n.m(a.parentNode,a))},i(e){s||(o(n),s=!0)},o(e){d(n),s=!1},d(e){e&&_(a),i[t].d(e)}}}function Ke(e){let t,n,a=s(e[0]),g=[];for(let t=0;t<a.length;t+=1)g[t]=ke(Ge(e,a,t));const u=e=>d(g[e],1,1,(()=>{g[e]=null}));return{c(){t=r("ul");for(let e=0;e<g.length;e+=1)g[e].c();i(t,"class","bx--side-nav__items pipen-report-sidenav svelte-12gpq3")},m(e,a){l(e,t,a);for(let e=0;e<g.length;e+=1)g[e]&&g[e].m(t,null);n=!0},p(e,[n]){if(7&n){let r;for(a=s(e[0]),r=0;r<a.length;r+=1){const s=Ge(e,a,r);g[r]?(g[r].p(s,n),o(g[r],1)):(g[r]=ke(s),g[r].c(),o(g[r],1),g[r].m(t,null))}for(p(),r=a.length;r<g.length;r+=1)u(r);c()}},i(e){if(!n){for(let e=0;e<a.length;e+=1)o(g[e]);n=!0}},o(e){g=g.filter(Boolean);for(let e=0;e<g.length;e+=1)d(g[e]);n=!1},d(e){e&&_(t),m(g,e)}}}function je(e,t,n){let{toc:a}=t,{page:s}=t;return g((()=>{let e,t=new URL(window.location.href);e=t.hash.length>0?document.querySelector(`[href="${t.hash}"]`):document.querySelector(".toc-current-page"),e.scrollIntoView({behavior:"smooth"})})),e.$$set=e=>{"toc"in e&&n(0,a=e.toc),"page"in e&&n(3,s=e.page)},[a,e=>{if(s===e)return"";let t=new URL(window.location.href);return t.hash="",t.searchParams.set("page",e),t.toString()},e=>e!=s?"toc-noncurrent-page":"toc-current-page",s]}class Xe extends t{constructor(e){super(),n(this,e,je,Ke,a,{toc:0,page:3})}}function Ye(e,t,n){const a=e.slice();return a[7]=t[n],a}function We(e,t,n){const a=e.slice();return a[10]=t[n],a}function Qe(e){let t,n;return t=new le({props:{isOpen:ft,transition:e[5],$$slots:{default:[pt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8197&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function Je(e){let t;return{c(){t=A("Go to home:")},m(e,n){l(e,t,n)},d(e){e&&_(t)}}}function Ze(e){let t;return{c(){t=A(e[0])},m(e,n){l(e,t,n)},p(e,n){1&n&&S(t,e[0])},d(e){e&&_(t)}}}function et(e){let t;return{c(){t=A("Switch to process:")},m(e,n){l(e,t,n)},d(e){e&&_(t)}}}function tt(e){let t,n;return t=new ce({props:{title:e[7].name,href:`?proc=${e[7].name}`,$$slots:{default:[at]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};4&n&&(a.title=e[7].name),4&n&&(a.href=`?proc=${e[7].name}`),8196&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function nt(e){let t,n,a,r;t=new ce({props:{class:"procgroup",$$slots:{default:[st]},$$scope:{ctx:e}}});let i=s(e[7].procs),g=[];for(let t=0;t<i.length;t+=1)g[t]=it(We(e,i,t));const h=e=>d(g[e],1,1,(()=>{g[e]=null}));return{c(){f(t.$$.fragment),n=$();for(let e=0;e<g.length;e+=1)g[e].c();a=u()},m(e,s){b(t,e,s),l(e,n,s);for(let t=0;t<g.length;t+=1)g[t]&&g[t].m(e,s);l(e,a,s),r=!0},p(e,n){const r={};if(8196&n&&(r.$$scope={dirty:n,ctx:e}),t.$set(r),4&n){let t;for(i=s(e[7].procs),t=0;t<i.length;t+=1){const s=We(e,i,t);g[t]?(g[t].p(s,n),o(g[t],1)):(g[t]=it(s),g[t].c(),o(g[t],1),g[t].m(a.parentNode,a))}for(p(),t=i.length;t<g.length;t+=1)h(t);c()}},i(e){if(!r){o(t.$$.fragment,e);for(let e=0;e<i.length;e+=1)o(g[e]);r=!0}},o(e){d(t.$$.fragment,e),g=g.filter(Boolean);for(let e=0;e<g.length;e+=1)d(g[e]);r=!1},d(e){e&&(_(n),_(a)),C(t,e),m(g,e)}}}function at(e){let t,n,a=e[7].name+"";return{c(){t=A(a),n=$()},m(e,a){l(e,t,a),l(e,n,a)},p(e,n){4&n&&a!==(a=e[7].name+"")&&S(t,a)},d(e){e&&(_(t),_(n))}}}function st(e){let t,n=e[7].name+"";return{c(){t=A(n)},m(e,n){l(e,t,n)},p(e,a){4&a&&n!==(n=e[7].name+"")&&S(t,n)},d(e){e&&_(t)}}}function rt(e){let t,n,a,s=e[10].name+"";return{c(){t=A("- "),n=A(s),a=$()},m(e,s){l(e,t,s),l(e,n,s),l(e,a,s)},p(e,t){4&t&&s!==(s=e[10].name+"")&&S(n,s)},d(e){e&&(_(t),_(n),_(a))}}}function it(e){let t,n;return t=new ce({props:{title:e[10].name,class:"procgroup-proc",href:`?proc=${e[10].name}`,$$slots:{default:[rt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};4&n&&(a.title=e[10].name),4&n&&(a.href=`?proc=${e[10].name}`),8196&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function lt(e){let t,n,a,s;const r=[nt,tt],i=[];function m(e,t){return e[7].procs?0:1}return t=m(e),n=i[t]=r[t](e),{c(){n.c(),a=u()},m(e,n){i[t].m(e,n),l(e,a,n),s=!0},p(e,s){let l=t;t=m(e),t===l?i[t].p(e,s):(p(),d(i[l],1,1,(()=>{i[l]=null})),c(),n=i[t],n?n.p(e,s):(n=i[t]=r[t](e),n.c()),o(n,1),n.m(a.parentNode,a))},i(e){s||(o(n),s=!0)},o(e){d(n),s=!1},d(e){e&&_(a),i[t].d(e)}}}function ot(e){let t,n,a,r,i,g,h,v;t=new pe({props:{$$slots:{default:[Je]},$$scope:{ctx:e}}}),a=new ce({props:{href:"index.html",$$slots:{default:[Ze]},$$scope:{ctx:e}}}),i=new pe({props:{$$slots:{default:[et]},$$scope:{ctx:e}}});let y=s(e[2]),A=[];for(let t=0;t<y.length;t+=1)A[t]=lt(Ye(e,y,t));const x=e=>d(A[e],1,1,(()=>{A[e]=null}));return{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment),r=$(),f(i.$$.fragment),g=$();for(let e=0;e<A.length;e+=1)A[e].c();h=u()},m(e,s){b(t,e,s),l(e,n,s),b(a,e,s),l(e,r,s),b(i,e,s),l(e,g,s);for(let t=0;t<A.length;t+=1)A[t]&&A[t].m(e,s);l(e,h,s),v=!0},p(e,n){const r={};8192&n&&(r.$$scope={dirty:n,ctx:e}),t.$set(r);const l={};8193&n&&(l.$$scope={dirty:n,ctx:e}),a.$set(l);const d={};if(8192&n&&(d.$$scope={dirty:n,ctx:e}),i.$set(d),4&n){let t;for(y=s(e[2]),t=0;t<y.length;t+=1){const a=Ye(e,y,t);A[t]?(A[t].p(a,n),o(A[t],1)):(A[t]=lt(a),A[t].c(),o(A[t],1),A[t].m(h.parentNode,h))}for(p(),t=y.length;t<A.length;t+=1)x(t);c()}},i(e){if(!v){o(t.$$.fragment,e),o(a.$$.fragment,e),o(i.$$.fragment,e);for(let e=0;e<y.length;e+=1)o(A[e]);v=!0}},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),d(i.$$.fragment,e),A=A.filter(Boolean);for(let e=0;e<A.length;e+=1)d(A[e]);v=!1},d(e){e&&(_(n),_(r),_(g),_(h)),C(t,e),C(a,e),C(i,e),m(A,e)}}}function pt(e){let t,n;return t=new oe({props:{$$slots:{default:[ot]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8197&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function ct(e){let t,n,a,s;t=new ie({props:{render:"select",theme:"g10",persist:!0,persistKey:"__pipen-report-theme",select:{themes:["g10","white","g80","g90","g100"],labelText:"Theme",inline:!0}}});let r=e[2].length>0&&Qe(e);return{c(){f(t.$$.fragment),n=$(),r&&r.c(),a=u()},m(e,i){b(t,e,i),l(e,n,i),r&&r.m(e,i),l(e,a,i),s=!0},p(e,t){e[2].length>0?r?(r.p(e,t),4&t&&o(r,1)):(r=Qe(e),r.c(),o(r,1),r.m(a.parentNode,a)):r&&(p(),d(r,1,1,(()=>{r=null})),c())},i(e){s||(o(t.$$.fragment,e),o(r),s=!0)},o(e){d(t.$$.fragment,e),d(r),s=!1},d(e){e&&(_(n),_(a)),C(t,e),r&&r.d(e)}}}function dt(e){let t,n;return t=new se({props:{class:"header-util",$$slots:{default:[ct]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8197&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function _t(e){let t;return{c(){t=A(e[4])},m(e,n){l(e,t,n)},p(e,n){16&n&&S(t,e[4])},d(e){e&&_(t)}}}function mt(e){let t,n,a,s,i=null!==e[4]&&_t(e);return{c(){t=r("h1"),n=A(e[3]),a=$(),i&&i.c(),s=u()},m(e,r){l(e,t,r),x(t,n),l(e,a,r),i&&i.m(e,r),l(e,s,r)},p(e,t){8&t&&S(n,e[3]),null!==e[4]?i?i.p(e,t):(i=_t(e),i.c(),i.m(s.parentNode,s)):i&&(i.d(1),i=null)},d(e){e&&(_(t),_(a),_(s)),i&&i.d(e)}}}function gt(e){let t,n;return t=new re({props:{class:"header-tile",slot:"platform",$$slots:{default:[mt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8216&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function ut(e){let t,n,a;function s(t){e[6](t)}let r={$$slots:{platform:[gt],default:[dt]},$$scope:{ctx:e}};return void 0!==e[1]&&(r.isSideNavOpen=e[1]),t=new ae({props:r}),h.push((()=>v(t,"isSideNavOpen",s))),{c(){f(t.$$.fragment)},m(e,n){b(t,e,n),a=!0},p(e,[a]){const s={};8221&a&&(s.$$scope={dirty:a,ctx:e}),!n&&2&a&&(n=!0,s.isSideNavOpen=e[1],y((()=>n=!1))),t.$set(s)},i(e){a||(o(t.$$.fragment,e),a=!0)},o(e){d(t.$$.fragment,e),a=!1},d(e){C(t,e)}}}let ft=!1;function bt(e,t,n){let{entries:a}=t,{logo:s}=t,{logotext:r=null}=t,{pipeline_name:i}=t,{isSideNavOpen:l}=t;return i=i||s,e.$$set=e=>{"entries"in e&&n(2,a=e.entries),"logo"in e&&n(3,s=e.logo),"logotext"in e&&n(4,r=e.logotext),"pipeline_name"in e&&n(0,i=e.pipeline_name),"isSideNavOpen"in e&&n(1,l=e.isSideNavOpen)},[i,l,a,s,r,{duration:200},function(e){l=e,n(1,l)}]}class Ct extends t{constructor(e){super(),n(this,e,bt,ut,a,{entries:2,logo:3,logotext:4,pipeline_name:0,isSideNavOpen:1})}}function $t(e){let t,n,a;function s(t){e[3](t)}let r={modalHeading:"Running Information",passiveModal:!0,hasScrollingContent:!0,class:"runinfo",$$slots:{default:[St]},$$scope:{ctx:e}};return void 0!==e[2]&&(r.open=e[2]),t=new de({props:r}),h.push((()=>v(t,"open",s))),{c(){f(t.$$.fragment)},m(e,n){b(t,e,n),a=!0},p(e,a){const s={};34&a&&(s.$$scope={dirty:a,ctx:e}),!n&&4&a&&(n=!0,s.open=e[2],y((()=>n=!1))),t.$set(s)},i(e){a||(o(t.$$.fragment,e),a=!0)},o(e){d(t.$$.fragment,e),a=!1},d(e){C(t,e)}}}function ht(e){let t,n,a,s,r,i;return t=new ge({props:{label:"Session Information"}}),a=new ge({props:{label:"Time Spent"}}),r=new ge({props:{label:"Device Information"}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment),s=$(),f(r.$$.fragment)},m(e,o){b(t,e,o),l(e,n,o),b(a,e,o),l(e,s,o),b(r,e,o),i=!0},p:M,i(e){i||(o(t.$$.fragment,e),o(a.$$.fragment,e),o(r.$$.fragment,e),i=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),d(r.$$.fragment,e),i=!1},d(e){e&&(_(n),_(s)),C(t,e),C(a,e),C(r,e)}}}function vt(e){let t,n;return t=new fe({props:{type:"multi",code:e[1].session}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};2&n&&(a.code=e[1].session),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function yt(e){let t,n;return t=new fe({props:{type:"multi",code:e[1].time}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};2&n&&(a.code=e[1].time),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function At(e){let t,n;return t=new fe({props:{type:"multi",code:e[1].device}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};2&n&&(a.code=e[1].device),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function xt(e){let t,n,a,s,r,i;return t=new ue({props:{$$slots:{default:[vt]},$$scope:{ctx:e}}}),a=new ue({props:{$$slots:{default:[yt]},$$scope:{ctx:e}}}),r=new ue({props:{$$slots:{default:[At]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment),s=$(),f(r.$$.fragment)},m(e,o){b(t,e,o),l(e,n,o),b(a,e,o),l(e,s,o),b(r,e,o),i=!0},p(e,n){const s={};34&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const i={};34&n&&(i.$$scope={dirty:n,ctx:e}),a.$set(i);const l={};34&n&&(l.$$scope={dirty:n,ctx:e}),r.$set(l)},i(e){i||(o(t.$$.fragment,e),o(a.$$.fragment,e),o(r.$$.fragment,e),i=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),d(r.$$.fragment,e),i=!1},d(e){e&&(_(n),_(s)),C(t,e),C(a,e),C(r,e)}}}function St(e){let t,n,a,s;return t=new _e({props:{kind:"warning",lowContrast:!0,hideCloseButton:!0,title:"Showing the first job only. Check the workdir for information of other jobs if any."}}),a=new me({props:{$$slots:{content:[xt],default:[ht]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,t){const n={};34&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function Pt(e){let t,n,a;return{c(){t=r("a"),t.textContent="> Running Information",i(t,"href","javascript:void(0);")},m(s,r){l(s,t,r),n||(a=P(t,"click",e[4]),n=!0)},p:M,d(e){e&&_(t),n=!1,a()}}}function Mt(e){let t,n,a,s,m,g,u,f,b=e[1]&&$t(e),C=e[1]&&Pt(e);return{c(){b&&b.c(),t=$(),n=r("div"),a=r("div"),C&&C.c(),s=$(),m=r("div"),g=A("Powered by "),u=r("span"),i(a,"class","runinfo"),i(u,"class","pipen-report-footer-versions"),i(m,"class","versions"),i(n,"class","pipen-report-footer bx--grid")},m(r,i){b&&b.m(r,i),l(r,t,i),l(r,n,i),x(n,a),C&&C.m(a,null),x(n,s),x(n,m),x(m,g),x(m,u),u.innerHTML=e[0],f=!0},p(e,[n]){e[1]?b?(b.p(e,n),2&n&&o(b,1)):(b=$t(e),b.c(),o(b,1),b.m(t.parentNode,t)):b&&(p(),d(b,1,1,(()=>{b=null})),c()),e[1]?C?C.p(e,n):(C=Pt(e),C.c(),C.m(a,null)):C&&(C.d(1),C=null),(!f||1&n)&&(u.innerHTML=e[0])},i(e){f||(o(b),f=!0)},o(e){d(b),f=!1},d(e){e&&(_(t),_(n)),b&&b.d(e),C&&C.d()}}}function Lt(e,t,n){let{versions:a}=t,{runinfo:s}=t,r=!1;return e.$$set=e=>{"versions"in e&&n(0,a=e.versions),"runinfo"in e&&n(1,s=e.runinfo)},[a,s,r,function(e){r=e,n(2,r)},()=>{n(2,r=!0)}]}class Et extends t{constructor(e){super(),n(this,e,Lt,Mt,a,{versions:0,runinfo:1})}}function wt(e){let t,n,a;function s(t){e[12](t)}let r={$$slots:{default:[Tt]},$$scope:{ctx:e}};return void 0!==e[0]&&(r.isOpen=e[0]),t=new Ce({props:r}),h.push((()=>v(t,"isOpen",s))),{c(){f(t.$$.fragment)},m(e,n){b(t,e,n),a=!0},p(e,a){const s={};8576&a&&(s.$$scope={dirty:a,ctx:e}),!n&&1&a&&(n=!0,s.isOpen=e[0],y((()=>n=!1))),t.$set(s)},i(e){a||(o(t.$$.fragment,e),a=!0)},o(e){d(t.$$.fragment,e),a=!1},d(e){C(t,e)}}}function Tt(e){let t,n;return t=new Xe({props:{toc:e[7],page:e[8]}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};128&n&&(a.toc=e[7]),256&n&&(a.page=e[8]),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function Ut(e){let t;const n=e[10].default,a=L(n,e,e[13],null);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||8192&s)&&E(a,n,e,e[13],t?T(n,e[13],s,null):w(e[13]),null)},i(e){t||(o(a,e),t=!0)},o(e){d(a,e),t=!1},d(e){a&&a.d(e)}}}function Nt(e){let t,n;return t=new ve({props:{$$slots:{default:[Ut]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8192&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function Dt(e){let t,n;return t=new he({props:{$$slots:{default:[Nt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8192&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function Rt(e){let t,n,a,s;return t=new $e({props:{$$slots:{default:[Dt]},$$scope:{ctx:e}}}),a=new Et({props:{versions:e[3],runinfo:e[9]}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const s={};8192&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};8&n&&(r.versions=e[3]),512&n&&(r.runinfo=e[9]),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function Bt(e){let t,n,a,s,r,i;function m(t){e[11](t)}let g={logo:e[1],logotext:e[2],entries:e[4],pipeline_name:e[5]};void 0!==e[0]&&(g.isSideNavOpen=e[0]),t=new Ct({props:g}),h.push((()=>v(t,"isSideNavOpen",m)));let u=e[6]&&e[7].length>0&&wt(e);return r=new be({props:{$$slots:{default:[Rt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),a=$(),u&&u.c(),s=$(),f(r.$$.fragment)},m(e,n){b(t,e,n),l(e,a,n),u&&u.m(e,n),l(e,s,n),b(r,e,n),i=!0},p(e,[a]){const i={};2&a&&(i.logo=e[1]),4&a&&(i.logotext=e[2]),16&a&&(i.entries=e[4]),32&a&&(i.pipeline_name=e[5]),!n&&1&a&&(n=!0,i.isSideNavOpen=e[0],y((()=>n=!1))),t.$set(i),e[6]&&e[7].length>0?u?(u.p(e,a),192&a&&o(u,1)):(u=wt(e),u.c(),o(u,1),u.m(s.parentNode,s)):u&&(p(),d(u,1,1,(()=>{u=null})),c());const l={};8712&a&&(l.$$scope={dirty:a,ctx:e}),r.$set(l)},i(e){i||(o(t.$$.fragment,e),o(u),o(r.$$.fragment,e),i=!0)},o(e){d(t.$$.fragment,e),d(u),d(r.$$.fragment,e),i=!1},d(e){e&&(_(a),_(s)),C(t,e),u&&u.d(e),C(r,e)}}}function It(e,t,n){let a,{$$slots:s={},$$scope:r}=t,{logo:i}=t,{logotext:l=null}=t,{versions:o}=t,{entries:p}=t,{pipeline_name:c}=t,{report_toc:d}=t,{toc:_}=t,{page:m}=t,{isSideNavOpen:u}=t,{runinfo:f}=t;return g((()=>{const e=new URL(window.location.href);let t;t=e.hash.length>0?document.querySelector(`li:has(>[href="${e.hash}"])`):document.querySelector("li:has(>.toc-current-page)"),t&&(t.classList.add("toc-active"),t.scrollIntoView());const n=document.querySelectorAll(".pipen-report-toc-anchor");a=new IntersectionObserver((e=>{e.forEach((e=>{const t=e.target.getAttribute("id"),n=document.querySelector(`li:has(>[href="#${t}"])`);n&&(e.isIntersecting?n.classList.add("toc-active"):n.classList.remove("toc-active"))}))}),{root:null,rootMargin:"0px",threshold:.5}),n.forEach((e=>{a.observe(e)}))})),e.$$set=e=>{"logo"in e&&n(1,i=e.logo),"logotext"in e&&n(2,l=e.logotext),"versions"in e&&n(3,o=e.versions),"entries"in e&&n(4,p=e.entries),"pipeline_name"in e&&n(5,c=e.pipeline_name),"report_toc"in e&&n(6,d=e.report_toc),"toc"in e&&n(7,_=e.toc),"page"in e&&n(8,m=e.page),"isSideNavOpen"in e&&n(0,u=e.isSideNavOpen),"runinfo"in e&&n(9,f=e.runinfo),"$$scope"in e&&n(13,r=e.$$scope)},[u,i,l,o,p,c,d,_,m,f,s,function(e){u=e,n(0,u)},function(e){u=e,n(0,u)},r]}class Ht extends t{constructor(e){super(),n(this,e,It,Bt,a,{logo:1,logotext:2,versions:3,entries:4,pipeline_name:5,report_toc:6,toc:7,page:8,isSideNavOpen:0,runinfo:9})}}function Gt(e){let t;return{c(){t=A(e[2])},m(e,n){l(e,t,n)},p:M,d(e){e&&_(t)}}}function Ft(e){let t,n,a,s;return n=new ye({props:{icon:e[1],kind:"tertiary",href:e[3](),$$slots:{default:[Gt]},$$scope:{ctx:e}}}),{c(){t=r("div"),f(n.$$.fragment),i(t,"class",a=`pagenav ${e[0]}`)},m(e,a){l(e,t,a),b(n,t,null),s=!0},p(e,[r]){const l={};16&r&&(l.$$scope={dirty:r,ctx:e}),n.$set(l),(!s||1&r&&a!==(a=`pagenav ${e[0]}`))&&i(t,"class",a)},i(e){s||(o(n.$$.fragment,e),s=!0)},o(e){d(n.$$.fragment,e),s=!1},d(e){e&&_(t),C(n)}}}function Vt(e,t,n){let{dir:a}=t,s="up"===a?Ue:Ne,r="up"===a?"Go to previous page":"Go to next page";return e.$$set=e=>{"dir"in e&&n(0,a=e.dir)},[a,s,r,()=>{let e=new URL(window.location.href);e.hash="";const t=parseInt(e.searchParams.get("page")||0),n="up"===a?t-1:t+1;return 0===n?e.searchParams.delete("page"):e.searchParams.set("page",n),e.toString()}]}class zt extends t{constructor(e){super(),n(this,e,Vt,Ft,a,{dir:0})}}const Ot=e=>({}),qt=e=>({}),kt=e=>({}),Kt=e=>({}),jt=e=>({}),Xt=e=>({}),Yt=e=>({}),Wt=e=>({});function Qt(e){let t,n;return t=new Ae({props:{ratio:e[5],$$slots:{default:[nn]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};32&n&&(a.ratio=e[5]),65631&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function Jt(e){let t,n,a,s,r=e[2]&&an(e),i=e[0]&&sn(e),m=e[1]&&rn(e);return{c(){r&&r.c(),t=$(),i&&i.c(),n=$(),m&&m.c(),a=u()},m(e,o){r&&r.m(e,o),l(e,t,o),i&&i.m(e,o),l(e,n,o),m&&m.m(e,o),l(e,a,o),s=!0},p(e,s){e[2]?r?(r.p(e,s),4&s&&o(r,1)):(r=an(e),r.c(),o(r,1),r.m(t.parentNode,t)):r&&(p(),d(r,1,1,(()=>{r=null})),c()),e[0]?i?(i.p(e,s),1&s&&o(i,1)):(i=sn(e),i.c(),o(i,1),i.m(n.parentNode,n)):i&&(p(),d(i,1,1,(()=>{i=null})),c()),e[1]?m?(m.p(e,s),2&s&&o(m,1)):(m=rn(e),m.c(),o(m,1),m.m(a.parentNode,a)):m&&(p(),d(m,1,1,(()=>{m=null})),c())},i(e){s||(o(r),o(i),o(m),s=!0)},o(e){d(r),d(i),d(m),s=!1},d(e){e&&(_(t),_(n),_(a)),r&&r.d(e),i&&i.d(e),m&&m.d(e)}}}function Zt(e){let t;const n=e[12].loading,a=L(n,e,e[16],Kt);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||65536&s)&&E(a,n,e,e[16],t?T(n,e[16],s,kt):w(e[16]),Kt)},i(e){t||(o(a,e),t=!0)},o(e){d(a,e),t=!1},d(e){a&&a.d(e)}}}function en(e){let t,n,a,s,i,o,p=[e[9],{src:n=e[3]},{alt:e[4]},{title:"Click to zoom in ..."}],c={};for(let e=0;e<p.length;e+=1)c=D(c,p[e]);return{c(){t=r("img"),F(t,c)},m(n,a){l(n,t,a),s=!0,i||(o=P(t,"click",e[8]),i=!0)},p(a,r){F(t,c=V(p,[(e=a)[9],(!s||8&r&&!B(t.src,n=e[3]))&&{src:n},(!s||16&r)&&{alt:e[4]},{title:"Click to zoom in ..."}]))},i(n){s||(n&&I((()=>{s&&(a||(a=H(t,G,{duration:e[6]?pn:0},!0)),a.run(1))})),s=!0)},o(n){n&&(a||(a=H(t,G,{duration:e[6]?pn:0},!1)),a.run(0)),s=!1},d(e){e&&_(t),e&&a&&a.end(),i=!1,o()}}}function tn(e){let t;const n=e[12].error,a=L(n,e,e[16],qt);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||65536&s)&&E(a,n,e,e[16],t?T(n,e[16],s,Ot):w(e[16]),qt)},i(e){t||(o(a,e),t=!0)},o(e){d(a,e),t=!1},d(e){a&&a.d(e)}}}function nn(e){let t,n,a,s,r=e[2]&&Zt(e),i=e[0]&&en(e),m=e[1]&&tn(e);return{c(){r&&r.c(),t=$(),i&&i.c(),n=$(),m&&m.c(),a=u()},m(e,o){r&&r.m(e,o),l(e,t,o),i&&i.m(e,o),l(e,n,o),m&&m.m(e,o),l(e,a,o),s=!0},p(e,s){e[2]?r?(r.p(e,s),4&s&&o(r,1)):(r=Zt(e),r.c(),o(r,1),r.m(t.parentNode,t)):r&&(p(),d(r,1,1,(()=>{r=null})),c()),e[0]?i?(i.p(e,s),1&s&&o(i,1)):(i=en(e),i.c(),o(i,1),i.m(n.parentNode,n)):i&&(p(),d(i,1,1,(()=>{i=null})),c()),e[1]?m?(m.p(e,s),2&s&&o(m,1)):(m=tn(e),m.c(),o(m,1),m.m(a.parentNode,a)):m&&(p(),d(m,1,1,(()=>{m=null})),c())},i(e){s||(o(r),o(i),o(m),s=!0)},o(e){d(r),d(i),d(m),s=!1},d(e){e&&(_(t),_(n),_(a)),r&&r.d(e),i&&i.d(e),m&&m.d(e)}}}function an(e){let t;const n=e[12].loading,a=L(n,e,e[16],Wt);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||65536&s)&&E(a,n,e,e[16],t?T(n,e[16],s,Yt):w(e[16]),Wt)},i(e){t||(o(a,e),t=!0)},o(e){d(a,e),t=!1},d(e){a&&a.d(e)}}}function sn(e){let t,n,a,s,i,o,p=[e[9],{src:n=e[3]},{alt:e[4]},{title:"Click to zoom in ..."}],c={};for(let e=0;e<p.length;e+=1)c=D(c,p[e]);return{c(){t=r("img"),F(t,c)},m(n,a){l(n,t,a),s=!0,i||(o=P(t,"click",e[8]),i=!0)},p(a,r){F(t,c=V(p,[(e=a)[9],(!s||8&r&&!B(t.src,n=e[3]))&&{src:n},(!s||16&r)&&{alt:e[4]},{title:"Click to zoom in ..."}]))},i(n){s||(n&&I((()=>{s&&(a||(a=H(t,G,{duration:e[6]?pn:0},!0)),a.run(1))})),s=!0)},o(n){n&&(a||(a=H(t,G,{duration:e[6]?pn:0},!1)),a.run(0)),s=!1},d(e){e&&_(t),e&&a&&a.end(),i=!1,o()}}}function rn(e){let t;const n=e[12].error,a=L(n,e,e[16],Xt);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||65536&s)&&E(a,n,e,e[16],t?T(n,e[16],s,jt):w(e[16]),Xt)},i(e){t||(o(a,e),t=!0)},o(e){d(a,e),t=!1},d(e){a&&a.d(e)}}}function ln(e){let t,n,a,s;return{c(){t=r("img"),i(t,"class","pipen-report-image-lg"),B(t.src,n=e[3])||i(t,"src",n),i(t,"alt",e[4])},m(e,n){l(e,t,n),s=!0},p(a,r){e=a,(!s||8&r&&!B(t.src,n=e[3]))&&i(t,"src",n),(!s||16&r)&&i(t,"alt",e[4])},i(n){s||(n&&I((()=>{s&&(a||(a=H(t,G,{duration:e[6]?pn:0},!0)),a.run(1))})),s=!0)},o(n){n&&(a||(a=H(t,G,{duration:e[6]?pn:0},!1)),a.run(0)),s=!1},d(e){e&&_(t),e&&a&&a.end()}}}function on(e){let t,n,a,s,r,i;const m=[Jt,Qt],g=[];function u(e,t){return void 0===e[5]?0:1}function A(t){e[13](t)}t=u(e),n=g[t]=m[t](e);let x={modalHeading:e[4],primaryButtonText:"Open in new window ...",secondaryButtonText:"Close",$$slots:{default:[ln]},$$scope:{ctx:e}};return void 0!==e[7]&&(x.open=e[7]),s=new de({props:x}),h.push((()=>v(s,"open",A))),s.$on("click:button--secondary",e[14]),s.$on("click:button--primary",e[15]),{c(){n.c(),a=$(),f(s.$$.fragment)},m(e,n){g[t].m(e,n),l(e,a,n),b(s,e,n),i=!0},p(e,[i]){let l=t;t=u(e),t===l?g[t].p(e,i):(p(),d(g[l],1,1,(()=>{g[l]=null})),c(),n=g[t],n?n.p(e,i):(n=g[t]=m[t](e),n.c()),o(n,1),n.m(a.parentNode,a));const _={};16&i&&(_.modalHeading=e[4]),65624&i&&(_.$$scope={dirty:i,ctx:e}),!r&&128&i&&(r=!0,_.open=e[7],y((()=>r=!1))),s.$set(_)},i(e){i||(o(n),o(s.$$.fragment,e),i=!0)},o(e){d(n),d(s.$$.fragment,e),i=!1},d(e){e&&_(a),g[t].d(e),C(s,e)}}}const pn=110;function cn(e,t,n){const a=["src","alt","ratio","loading","loaded","error","fadeIn","border","loadImage"];let s=U(t,a),{$$slots:r={},$$scope:i}=t,{src:l=""}=t,{alt:o=""}=t,{ratio:p}=t,{loading:c=!1}=t,{loaded:d=!1}=t,{error:_=!1}=t,{fadeIn:m=!1}=t,{border:u=!0}=t,f=!1;const b=e=>{null!=$&&($=null),n(0,d=!1),n(1,_=!1),$=new Image,$.src=e||l,$.onload=()=>n(0,d=!0),$.onerror=()=>n(1,_=!0)},C=N();let $=null;g((()=>()=>$=null));const h={...s,class:[s.class,"pipen-report-image",u?"":"pipen-report-image-noborder"].filter(Boolean).join(" ")};return e.$$set=e=>{t=D(D({},t),R(e)),n(19,s=U(t,a)),"src"in e&&n(3,l=e.src),"alt"in e&&n(4,o=e.alt),"ratio"in e&&n(5,p=e.ratio),"loading"in e&&n(2,c=e.loading),"loaded"in e&&n(0,d=e.loaded),"error"in e&&n(1,_=e.error),"fadeIn"in e&&n(6,m=e.fadeIn),"border"in e&&n(10,u=e.border),"$$scope"in e&&n(16,i=e.$$scope)},e.$$.update=()=>{3&e.$$.dirty&&n(2,c=!d&&!_),8&e.$$.dirty&&l&&"undefined"!=typeof window&&b(),1&e.$$.dirty&&d&&C("load"),2&e.$$.dirty&&_&&C("error")},[d,_,c,l,o,p,m,f,()=>{n(7,f=!0)},h,u,b,r,function(e){f=e,n(7,f)},()=>{n(7,f=!1)},()=>{n(7,f=!1),window.open(l)},i]}class dn extends t{constructor(e){super(),n(this,e,cn,on,a,{src:3,alt:4,ratio:5,loading:2,loaded:0,error:1,fadeIn:6,border:10,loadImage:11})}get loadImage(){return this.$$.ctx[11]}}function _n(e){let t,n;const a=[{src:e[1]},e[4]];let s={$$slots:{error:[un],loading:[gn]},$$scope:{ctx:e}};for(let e=0;e<a.length;e+=1)s=D(s,a[e]);return t=new dn({props:s}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const s=18&n?V(a,[2&n&&{src:e[1]},16&n&&z(e[4])]):{};256&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function mn(e){let t,n,a;const s=[{src:e[1]},e[4]];let i={$$slots:{error:[bn],loading:[fn]},$$scope:{ctx:e}};for(let e=0;e<s.length;e+=1)i=D(i,s[e]);n=new dn({props:i});let p=[e[0]],c={};for(let e=0;e<p.length;e+=1)c=D(c,p[e]);return{c(){t=r("div"),f(n.$$.fragment),F(t,c)},m(e,s){l(e,t,s),b(n,t,null),a=!0},p(e,a){const r=18&a?V(s,[2&a&&{src:e[1]},16&a&&z(e[4])]):{};256&a&&(r.$$scope={dirty:a,ctx:e}),n.$set(r),F(t,c=V(p,[1&a&&e[0]]))},i(e){a||(o(n.$$.fragment,e),a=!0)},o(e){d(n.$$.fragment,e),a=!1},d(e){e&&_(t),C(n)}}}function gn(e){let t,n,a=[e[2]],s={};for(let e=0;e<a.length;e+=1)s=D(s,a[e]);return{c(){t=r("div"),n=A("Image loading ..."),F(t,s)},m(e,a){l(e,t,a),x(t,n)},p:M,d(e){e&&_(t)}}}function un(e){let t,n,a=[e[3]],s={};for(let e=0;e<a.length;e+=1)s=D(s,a[e]);return{c(){t=r("div"),n=A("Image loading error!"),F(t,s)},m(e,a){l(e,t,a),x(t,n)},p:M,d(e){e&&_(t)}}}function fn(e){let t,n,a=[e[2]],s={};for(let e=0;e<a.length;e+=1)s=D(s,a[e]);return{c(){t=r("div"),n=A("Image loading ..."),F(t,s)},m(e,a){l(e,t,a),x(t,n)},p:M,d(e){e&&_(t)}}}function bn(e){let t,n,a=[e[3]],s={};for(let e=0;e<a.length;e+=1)s=D(s,a[e]);return{c(){t=r("div"),n=A("Image loading error!"),F(t,s)},m(e,a){l(e,t,a),x(t,n)},p:M,d(e){e&&_(t)}}}function Cn(e){let t,n,a,s;const r=[mn,_n],i=[];function m(e,t){return e[0]?0:1}return t=m(e),n=i[t]=r[t](e),{c(){n.c(),a=u()},m(e,n){i[t].m(e,n),l(e,a,n),s=!0},p(e,[s]){let l=t;t=m(e),t===l?i[t].p(e,s):(p(),d(i[l],1,1,(()=>{i[l]=null})),c(),n=i[t],n?n.p(e,s):(n=i[t]=r[t](e),n.c()),o(n,1),n.m(a.parentNode,a))},i(e){s||(o(n),s=!0)},o(e){d(n),s=!1},d(e){e&&_(a),i[t].d(e)}}}function $n(e,t,n){const a=["frameProps","width","height","src"];let s=U(t,a),{frameProps:r={class:"pipen-report-image-frame"}}=t,{width:i=300}=t,{height:l=200}=t,{src:o}=t,p=`width: ${i}px; height: ${l}px; aspect-ratio: ${i} / ${l}`,c={...s,class:[s.class,"pipen-report-image-loading"].filter(Boolean).join(" "),style:s.style?`${s.style}; ${p}`:p},d={...c,class:[c.class,"error"].join(" ")};return e.$$set=e=>{t=D(D({},t),R(e)),n(4,s=U(t,a)),"frameProps"in e&&n(0,r=e.frameProps),"width"in e&&n(5,i=e.width),"height"in e&&n(6,l=e.height),"src"in e&&n(1,o=e.src)},[r,o,c,d,s,i,l]}let hn=class extends t{constructor(e){super(),n(this,e,$n,Cn,a,{frameProps:0,width:5,height:6,src:1})}};const vn=e=>({}),yn=e=>({});function An(e){let t,n;const a=e[25].default,s=L(a,e,e[24],null);return{c(){t=r("ul"),s&&s.c(),i(t,"role","menu"),i(t,"tabindex","-1"),i(t,"aria-label",e[14]),i(t,"data-floating-menu-direction",e[5]),i(t,"class",e[9]),O(t,"bx--overflow-menu-options",!0),O(t,"bx--overflow-menu--flip",e[7]),O(t,"bx--overflow-menu-options--open",e[0]),O(t,"bx--overflow-menu-options--light",e[6]),O(t,"bx--overflow-menu-options--sm","sm"===e[4]),O(t,"bx--overflow-menu-options--xl","xl"===e[4]),O(t,"bx--breadcrumb-menu-options",!!e[15])},m(a,r){l(a,t,r),s&&s.m(t,null),e[32](t),n=!0},p(e,r){s&&s.p&&(!n||16777216&r[0])&&E(s,a,e,e[24],n?T(a,e[24],r,null):w(e[24]),null),(!n||16384&r[0])&&i(t,"aria-label",e[14]),(!n||32&r[0])&&i(t,"data-floating-menu-direction",e[5]),(!n||512&r[0])&&i(t,"class",e[9]),(!n||512&r[0])&&O(t,"bx--overflow-menu-options",!0),(!n||640&r[0])&&O(t,"bx--overflow-menu--flip",e[7]),(!n||513&r[0])&&O(t,"bx--overflow-menu-options--open",e[0]),(!n||576&r[0])&&O(t,"bx--overflow-menu-options--light",e[6]),(!n||528&r[0])&&O(t,"bx--overflow-menu-options--sm","sm"===e[4]),(!n||528&r[0])&&O(t,"bx--overflow-menu-options--xl","xl"===e[4]),(!n||33280&r[0])&&O(t,"bx--breadcrumb-menu-options",!!e[15])},i(e){n||(o(s,e),n=!0)},o(e){d(s,e),n=!1},d(n){n&&_(t),s&&s.d(n),e[32](null)}}}function xn(e){let t,n,a,s,i,m,g,h;const v=e[25].menu,y=L(v,e,e[24],yn),A=y||function(e){let t,n,a;var s=e[1];function r(e,t){return{props:{"aria-label":e[11],title:e[11],class:"bx--overflow-menu__icon "+e[10]}}}return s&&(t=Y(s,r(e))),{c(){t&&f(t.$$.fragment),n=u()},m(e,s){t&&b(t,e,s),l(e,n,s),a=!0},p(e,a){if(2&a[0]&&s!==(s=e[1])){if(t){p();const e=t;d(e.$$.fragment,1,0,(()=>{C(e,1)})),c()}s?(t=Y(s,r(e)),f(t.$$.fragment),o(t.$$.fragment,1),b(t,n.parentNode,n)):t=null}else if(s){const n={};2048&a[0]&&(n["aria-label"]=e[11]),2048&a[0]&&(n.title=e[11]),1024&a[0]&&(n.class="bx--overflow-menu__icon "+e[10]),t.$set(n)}},i(e){a||(t&&o(t.$$.fragment,e),a=!0)},o(e){t&&d(t.$$.fragment,e),a=!1},d(e){e&&_(n),t&&C(t,e)}}}(e);let S=e[0]&&An(e),M=[{type:"button"},{"aria-haspopup":""},{"aria-expanded":e[0]},{"aria-label":e[14]},{id:e[12]},e[20]],U={};for(let e=0;e<M.length;e+=1)U=D(U,M[e]);return{c(){t=new W(!1),n=u(),a=$(),s=r("button"),A&&A.c(),i=$(),S&&S.c(),t.a=n,F(s,U),O(s,"bx--overflow-menu",!0),O(s,"bx--overflow-menu--open",e[0]),O(s,"bx--overflow-menu--light",e[6]),O(s,"bx--overflow-menu--sm","sm"===e[4]),O(s,"bx--overflow-menu--xl","xl"===e[4])},m(r,o){t.m(e[13],document.head),x(document.head,n),l(r,a,o),l(r,s,o),A&&A.m(s,null),x(s,i),S&&S.m(s,null),s.autofocus&&s.focus(),e[33](s),m=!0,g||(h=[P(window,"click",e[31]),P(s,"click",e[26]),P(s,"click",e[34]),P(s,"mouseover",e[27]),P(s,"mouseenter",e[28]),P(s,"mouseleave",e[29]),P(s,"keydown",e[30]),P(s,"keydown",e[35]),P(s,"focusout",e[36])],g=!0)},p(e,n){(!m||8192&n[0])&&t.p(e[13]),y?y.p&&(!m||16777216&n[0])&&E(y,v,e,e[24],m?T(v,e[24],n,vn):w(e[24]),yn):A&&A.p&&(!m||3074&n[0])&&A.p(e,m?n:[-1,-1]),e[0]?S?(S.p(e,n),1&n[0]&&o(S,1)):(S=An(e),S.c(),o(S,1),S.m(s,null)):S&&(p(),d(S,1,1,(()=>{S=null})),c()),F(s,U=V(M,[{type:"button"},{"aria-haspopup":""},(!m||1&n[0])&&{"aria-expanded":e[0]},(!m||16384&n[0])&&{"aria-label":e[14]},(!m||4096&n[0])&&{id:e[12]},1048576&n[0]&&e[20]])),O(s,"bx--overflow-menu",!0),O(s,"bx--overflow-menu--open",e[0]),O(s,"bx--overflow-menu--light",e[6]),O(s,"bx--overflow-menu--sm","sm"===e[4]),O(s,"bx--overflow-menu--xl","xl"===e[4])},i(e){m||(o(A,e),o(S),m=!0)},o(e){d(A,e),d(S),m=!1},d(r){r&&(t.d(),_(a),_(s)),_(n),A&&A.d(r),S&&S.d(),e[33](null),g=!1,q(h)}}}function Sn(e,t,n){let a,s;const r=["size","direction","open","light","flipped","persistentClickItems","menuOptionsClass","icon","iconClass","iconDescription","id","buttonRef","menuRef"];let i,l,o,p=U(t,r),{$$slots:c={},$$scope:d}=t,{size:_}=t,{direction:m="bottom"}=t,{open:g=!1}=t,{light:u=!1}=t,{flipped:f=!1}=t,{persistentClickItems:b=!1}=t,{menuOptionsClass:C}=t,{icon:$=Re}=t,{iconClass:v}=t,{iconDescription:y="Open and close list of options"}=t,{id:A="ccs-"+Math.random().toString(36)}=t,{buttonRef:x=null}=t,{menuRef:S=null}=t;const P=k("BreadcrumbItem"),M=N(),L=Q([]);K(e,L,(e=>n(23,l=e)));const E=Q(void 0);K(e,E,(e=>n(38,o=e)));const w=Q(void 0),T=Q(-1);let B;K(e,T,(e=>n(22,i=e)));let I=!0;j("OverflowMenu",{focusedId:w,add:({id:e,text:t,primaryFocus:n,disabled:a})=>{L.update((s=>(n&&T.set(s.length),[...s,{id:e,text:t,primaryFocus:n,disabled:a,index:s.length}])))},update:e=>{E.set(e)},change:e=>{let t=i+e;t<0?t=l.length-1:t>=l.length&&(t=0);let n=l[t].disabled;for(;n;)t+=e,t<0?t=l.length-1:t>=l.length&&(t=0),n=l[t].disabled;T.set(t)}}),X((()=>{if(o){const{index:e,text:t}=l.filter((e=>e.id===o))[0];M("close",{index:e,text:t}),n(0,g=!1)}if(g){const{width:e,height:t}=x.getBoundingClientRect();n(21,B=e),!I&&i<0&&S.focus(),f&&(n(3,S.style.left="auto",S),n(3,S.style.right=0,S)),"top"===m?(n(3,S.style.top="auto",S),n(3,S.style.bottom=t+"px",S)):"bottom"===m&&n(3,S.style.top=t+"px",S),P&&(n(3,S.style.top=t+10+"px",S),n(3,S.style.left="-11px",S))}g||(L.set([]),E.set(void 0),T.set(0)),I=!1}));return e.$$set=e=>{n(40,t=D(D({},t),R(e))),n(20,p=U(t,r)),"size"in e&&n(4,_=e.size),"direction"in e&&n(5,m=e.direction),"open"in e&&n(0,g=e.open),"light"in e&&n(6,u=e.light),"flipped"in e&&n(7,f=e.flipped),"persistentClickItems"in e&&n(8,b=e.persistentClickItems),"menuOptionsClass"in e&&n(9,C=e.menuOptionsClass),"icon"in e&&n(1,$=e.icon),"iconClass"in e&&n(10,v=e.iconClass),"iconDescription"in e&&n(11,y=e.iconDescription),"id"in e&&n(12,A=e.id),"buttonRef"in e&&n(2,x=e.buttonRef),"menuRef"in e&&n(3,S=e.menuRef),"$$scope"in e&&n(24,d=e.$$scope)},e.$$.update=()=>{n(14,a=t["aria-label"]||"menu"),12582912&e.$$.dirty[0]&&l[i]&&w.set(l[i].id),2101248&e.$$.dirty[0]&&n(13,s=`<style>\n    #${A} .bx--overflow-menu-options.bx--overflow-menu-options:after {\n      width: ${B?B+"px":"2rem"};\n    }\n  </style>`)},P&&n(1,$=De),t=R(t),[g,$,x,S,_,m,u,f,b,C,v,y,A,s,a,P,M,L,E,T,p,B,i,l,d,c,function(t){J.call(this,e,t)},function(t){J.call(this,e,t)},function(t){J.call(this,e,t)},function(t){J.call(this,e,t)},function(t){J.call(this,e,t)},({target:e})=>{x&&x.contains(e)||S&&!S.contains(e)&&n(0,g=!1)},function(e){h[e?"unshift":"push"]((()=>{S=e,n(3,S)}))},function(e){h[e?"unshift":"push"]((()=>{x=e,n(2,x)}))},e=>{b&&e.stopPropagation(),S&&S.contains(e.target)||(n(0,g=!g),g||M("close"))},e=>{g&&(["ArrowDown","ArrowLeft","ArrowRight","ArrowUp"].includes(e.key)?e.preventDefault():"Escape"===e.key&&(e.stopPropagation(),M("close"),n(0,g=!1),x.focus()))},e=>{g&&(x.contains(e.relatedTarget)||(M("close"),n(0,g=!1)))}]}class Pn extends t{constructor(e){super(),n(this,e,Sn,xn,a,{size:4,direction:5,open:0,light:6,flipped:7,persistentClickItems:8,menuOptionsClass:9,icon:1,iconClass:10,iconDescription:11,id:12,buttonRef:2,menuRef:3},null,[-1,-1])}}function Mn(e,t,n){const a=e.slice();return a[3]=t[n],a[4]=t,a[5]=n,a}function Ln(e){let t,n,a,s,p;function c(t){e[1](t,e[3])}let m={labelText:e[3].value};return void 0!==e[3].selected&&(m.checked=e[3].selected),n=new xe({props:m}),h.push((()=>v(n,"checked",c))),n.$on("change",e[2]),{c(){t=r("li"),f(n.$$.fragment),s=$(),i(t,"class","pipen-report-column-selector-item svelte-147jw2r")},m(e,a){l(e,t,a),b(n,t,null),x(t,s),p=!0},p(t,s){e=t;const r={};1&s&&(r.labelText=e[3].value),!a&&1&s&&(a=!0,r.checked=e[3].selected,y((()=>a=!1))),n.$set(r)},i(e){p||(o(n.$$.fragment,e),p=!0)},o(e){d(n.$$.fragment,e),p=!1},d(e){e&&_(t),C(n)}}}function En(e){let t,n,a=s(e[0]),r=[];for(let t=0;t<a.length;t+=1)r[t]=Ln(Mn(e,a,t));const i=e=>d(r[e],1,1,(()=>{r[e]=null}));return{c(){for(let e=0;e<r.length;e+=1)r[e].c();t=u()},m(e,a){for(let t=0;t<r.length;t+=1)r[t]&&r[t].m(e,a);l(e,t,a),n=!0},p(e,n){if(1&n){let l;for(a=s(e[0]),l=0;l<a.length;l+=1){const s=Mn(e,a,l);r[l]?(r[l].p(s,n),o(r[l],1)):(r[l]=Ln(s),r[l].c(),o(r[l],1),r[l].m(t.parentNode,t))}for(p(),l=a.length;l<r.length;l+=1)i(l);c()}},i(e){if(!n){for(let e=0;e<a.length;e+=1)o(r[e]);n=!0}},o(e){r=r.filter(Boolean);for(let e=0;e<r.length;e+=1)d(r[e]);n=!1},d(e){e&&_(t),m(r,e)}}}function wn(e){let t,n;return t=new Pn({props:{class:"pipen-report-overflowmenu",persistentClickItems:!0,size:"sm",icon:Be,iconDescription:"Select visible columns ...",$$slots:{default:[En]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,[n]){const a={};65&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function Tn(e,t,n){let{allColumns:a}=t;return e.$$set=e=>{"allColumns"in e&&n(0,a=e.allColumns)},[a,function(t,s){e.$$.not_equal(s.selected,t)&&(s.selected=t,n(0,a))},()=>{n(0,a)}]}class Un extends t{constructor(e){super(),n(this,e,Tn,wn,a,{allColumns:0})}}function Nn(e){let t,n;return t=new ye({props:{iconDescription:"Right click and save as to download the entire data",tooltipPosition:"top",icon:He,href:e[2]}}),t.$on("click",e[13]),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};4&n&&(a.href=e[2]),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function Dn(e){let t,n,a,s,r,i,m,g,A,x;function S(t){e[17](t)}let P={persistent:!0};function M(t){e[18](t)}void 0!==e[7]&&(P.value=e[7]),t=new Ee({props:P}),h.push((()=>v(t,"value",S)));let L={};void 0!==e[6]&&(L.allColumns=e[6]),s=new Un({props:L}),h.push((()=>v(s,"allColumns",M))),m=new ye({props:{iconDescription:`Download current ${e[11].length} records (may be partial of the entire data)`,tooltipPosition:"top",icon:Ie}}),m.$on("click",e[14]);let E=!!e[2]&&Nn(e);return{c(){f(t.$$.fragment),a=$(),f(s.$$.fragment),i=$(),f(m.$$.fragment),g=$(),E&&E.c(),A=u()},m(e,n){b(t,e,n),l(e,a,n),b(s,e,n),l(e,i,n),b(m,e,n),l(e,g,n),E&&E.m(e,n),l(e,A,n),x=!0},p(e,a){const i={};!n&&128&a&&(n=!0,i.value=e[7],y((()=>n=!1))),t.$set(i);const l={};!r&&64&a&&(r=!0,l.allColumns=e[6],y((()=>r=!1))),s.$set(l);const _={};2048&a&&(_.iconDescription=`Download current ${e[11].length} records (may be partial of the entire data)`),m.$set(_),e[2]?E?(E.p(e,a),4&a&&o(E,1)):(E=Nn(e),E.c(),o(E,1),E.m(A.parentNode,A)):E&&(p(),d(E,1,1,(()=>{E=null})),c())},i(e){x||(o(t.$$.fragment,e),o(s.$$.fragment,e),o(m.$$.fragment,e),o(E),x=!0)},o(e){d(t.$$.fragment,e),d(s.$$.fragment,e),d(m.$$.fragment,e),d(E),x=!1},d(e){e&&(_(a),_(i),_(g),_(A)),C(t,e),C(s,e),C(m,e),E&&E.d(e)}}}function Rn(e){let t,n;return t=new Le({props:{$$slots:{default:[Dn]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};134219972&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function Bn(e){let t,n;return t=new Me({props:{size:e[12],$$slots:{default:[Rn]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};134219972&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function In(e){let t;return{c(){t=r("p"),t.textContent='You must use "Save as" or "Save link as" from the context menu (by right-clicking the button) to download the entire data.'},m(e,n){l(e,t,n)},p:M,d(e){e&&_(t)}}}function Hn(e){let t,n,a,s,r,i,p,c,m;const g=[{size:e[0]},{zebra:e[3]},{sortable:e[4]},{page:e[8]},{pageSize:e[9]},{rows:e[11]},{headers:e[1]},e[15]];let u={$$slots:{default:[Bn]},$$scope:{ctx:e}};for(let e=0;e<g.length;e+=1)u=D(u,g[e]);function A(t){e[19](t)}function x(t){e[20](t)}t=new Se({props:u});let S={pageSizes:e[5],totalItems:e[11].length};function P(t){e[21](t)}void 0!==e[9]&&(S.pageSize=e[9]),void 0!==e[8]&&(S.page=e[8]),a=new Pe({props:S}),h.push((()=>v(a,"pageSize",A))),h.push((()=>v(a,"page",x)));let M={passiveModal:!0,modalHeading:"Downloading entire data",$$slots:{default:[In]},$$scope:{ctx:e}};return void 0!==e[10]&&(M.open=e[10]),p=new de({props:M}),h.push((()=>v(p,"open",P))),p.$on("open",e[22]),p.$on("close",e[23]),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment),i=$(),f(p.$$.fragment)},m(e,s){b(t,e,s),l(e,n,s),b(a,e,s),l(e,i,s),b(p,e,s),m=!0},p(e,[n]){const i=35611&n?V(g,[1&n&&{size:e[0]},8&n&&{zebra:e[3]},16&n&&{sortable:e[4]},256&n&&{page:e[8]},512&n&&{pageSize:e[9]},2048&n&&{rows:e[11]},2&n&&{headers:e[1]},32768&n&&z(e[15])]):{};134219972&n&&(i.$$scope={dirty:n,ctx:e}),t.$set(i);const l={};32&n&&(l.pageSizes=e[5]),2048&n&&(l.totalItems=e[11].length),!s&&512&n&&(s=!0,l.pageSize=e[9],y((()=>s=!1))),!r&&256&n&&(r=!0,l.page=e[8],y((()=>r=!1))),a.$set(l);const o={};134217728&n&&(o.$$scope={dirty:n,ctx:e}),!c&&1024&n&&(c=!0,o.open=e[10],y((()=>c=!1))),p.$set(o)},i(e){m||(o(t.$$.fragment,e),o(a.$$.fragment,e),o(p.$$.fragment,e),m=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),d(p.$$.fragment,e),m=!1},d(e){e&&(_(n),_(i)),C(t,e),C(a,e),C(p,e)}}}function Gn(e,t,n){const a=["size","src","zebra","sortable","rows","headers","pageSizes"];let s=U(t,a),{size:r="sm"}=t,{src:i}=t;r={sm:"short",small:"short",short:"short",md:"medium",medium:"medium",lg:"tall",large:"tall",tall:"tall",compact:"compact"}[r];let l={short:"sm",compact:"sm",medium:"default",tall:"default"}[r],{zebra:o=!0}=t,{sortable:p=!0}=t,{rows:c=[]}=t,{headers:d=[]}=t,{pageSizes:_=[20,50,75,100]}=t,m=d.map((e=>({...e,selected:!0}))),g=1,u=_[0],f="",b=!1,C=c;return e.$$set=e=>{t=D(D({},t),R(e)),n(15,s=U(t,a)),"size"in e&&n(0,r=e.size),"src"in e&&n(2,i=e.src),"zebra"in e&&n(3,o=e.zebra),"sortable"in e&&n(4,p=e.sortable),"rows"in e&&n(16,c=e.rows),"headers"in e&&n(1,d=e.headers),"pageSizes"in e&&n(5,_=e.pageSizes)},e.$$.update=()=>{64&e.$$.dirty&&n(1,d=m.filter((e=>e.selected))),65664&e.$$.dirty&&(f.length>0?n(11,C=c.filter((e=>Object.values(e).filter((e=>!!e&&e.toString().includes(f))).length>0))):n(11,C=c))},[r,d,i,o,p,_,m,f,g,u,b,C,l,e=>{e.preventDefault(),n(10,b=!0)},()=>{((e,t)=>{const n=document.createElement("a");n.href=e,n.download=t,document.body.appendChild(n),n.click(),document.body.removeChild(n)})("data:application/vnd.ms-excel;charset=utf-8,\ufeff"+encodeURIComponent(function(){const e=d.map((e=>e.value)),t=[];return C.forEach((e=>{const n=d.map((t=>`"${e[t.key]}"`));t.push(n.join(","))})),e.join(",")+"\r\n"+t.join("\r\n")}()),i&&i.length>0?i.split("/").pop().split(".")[0]+".xls":"data-subset.xls")},s,c,function(e){f=e,n(7,f)},function(e){m=e,n(6,m)},function(e){u=e,n(9,u)},function(e){g=e,n(8,g)},function(e){b=e,n(10,b)},function(t){J.call(this,e,t)},function(t){J.call(this,e,t)}]}class Fn extends t{constructor(e){super(),n(this,e,Gn,Hn,a,{size:0,src:2,zebra:3,sortable:4,rows:16,headers:1,pageSizes:5})}}function Vn(e){let t,n,a;const s=[{headers:e[0]},{rows:e[1]},e[3]];let i={};for(let e=0;e<s.length;e+=1)i=D(i,s[e]);n=new Fn({props:i});let p=[e[2]],c={};for(let e=0;e<p.length;e+=1)c=D(c,p[e]);return{c(){t=r("div"),f(n.$$.fragment),F(t,c)},m(e,s){l(e,t,s),b(n,t,null),a=!0},p(e,a){const r=11&a?V(s,[1&a&&{headers:e[0]},2&a&&{rows:e[1]},8&a&&z(e[3])]):{};n.$set(r),F(t,c=V(p,[4&a&&e[2]]))},i(e){a||(o(n.$$.fragment,e),a=!0)},o(e){d(n.$$.fragment,e),a=!1},d(e){e&&_(t),C(n)}}}function zn(e){let t,n;const a=[{headers:e[0]},{rows:e[1]},e[3]];let s={};for(let e=0;e<a.length;e+=1)s=D(s,a[e]);return t=new Fn({props:s}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const s=11&n?V(a,[1&n&&{headers:e[0]},2&n&&{rows:e[1]},8&n&&z(e[3])]):{};t.$set(s)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function On(e){let t,n,a,s;const r=[zn,Vn],i=[];function m(e,t){return e[2]?1:0}return t=m(e),n=i[t]=r[t](e),{c(){n.c(),a=u()},m(e,n){i[t].m(e,n),l(e,a,n),s=!0},p(e,[s]){let l=t;t=m(e),t===l?i[t].p(e,s):(p(),d(i[l],1,1,(()=>{i[l]=null})),c(),n=i[t],n?n.p(e,s):(n=i[t]=r[t](e),n.c()),o(n,1),n.m(a.parentNode,a))},i(e){s||(o(n),s=!0)},o(e){d(n),s=!1},d(e){e&&_(a),i[t].d(e)}}}function qn(e,t,n){const a=["frameProps","data","headers","rows"];let s=U(t,a),{frameProps:r={class:"pipen-report-datatable-frame"}}=t,{data:i={}}=t,{headers:l=[]}=t,{rows:o=[]}=t;if(l.length+o.length>0&&i.length>0)throw"Can only specify `data` or `headers`/`rows` for `DataTable`!";return i.length>0&&(o=i,l=Object.keys(o[0]).filter((e=>"id"!==e)).map((e=>({key:e,value:e})))),e.$$set=e=>{t=D(D({},t),R(e)),n(3,s=U(t,a)),"frameProps"in e&&n(2,r=e.frameProps),"data"in e&&n(4,i=e.data),"headers"in e&&n(0,l=e.headers),"rows"in e&&n(1,o=e.rows)},[l,o,r,s,i]}class kn extends t{constructor(e){super(),n(this,e,qn,On,a,{frameProps:2,data:4,headers:0,rows:1})}}function Kn(e){let t,n;return{c(){t=r("div"),n=A(e[0]),i(t,"class","pipen-report-descr-title svelte-1xzxuva")},m(e,a){l(e,t,a),x(t,n)},p(e,t){1&t&&S(n,e[0])},d(e){e&&_(t)}}}function jn(e){let t,n,a,s=e[0]&&Kn(e);const i=e[3].default,p=L(i,e,e[2],null);let c=[e[1]],m={};for(let e=0;e<c.length;e+=1)m=D(m,c[e]);return{c(){t=r("div"),s&&s.c(),n=$(),p&&p.c(),F(t,m),O(t,"svelte-1xzxuva",!0)},m(e,r){l(e,t,r),s&&s.m(t,null),x(t,n),p&&p.m(t,null),a=!0},p(e,[r]){e[0]?s?s.p(e,r):(s=Kn(e),s.c(),s.m(t,n)):s&&(s.d(1),s=null),p&&p.p&&(!a||4&r)&&E(p,i,e,e[2],a?T(i,e[2],r,null):w(e[2]),null),F(t,m=V(c,[2&r&&e[1]])),O(t,"svelte-1xzxuva",!0)},i(e){a||(o(p,e),a=!0)},o(e){d(p,e),a=!1},d(e){e&&_(t),s&&s.d(),p&&p.d(e)}}}function Xn(e,t,n){const a=["title"];let s=U(t,a),{$$slots:r={},$$scope:i}=t,{title:l}=t,{klass:o,...p}=s;return o=o?`pipen-report-descr ${o}`:"pipen-report-descr",p||(p={}),p.class=o,e.$$set=e=>{t=D(D({},t),R(e)),n(5,s=U(t,a)),"title"in e&&n(0,l=e.title),"$$scope"in e&&n(2,i=e.$$scope)},[l,p,i,r]}class Yn extends t{constructor(e){super(),n(this,e,Xn,jn,a,{title:0})}}function Wn(e){let t,n,a,s;return t=new ge({props:{label:"Plot"}}),a=new ge({props:{label:"Table"}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p:M,i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function Qn(e){let t,n;return t=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Cycling-T-cells/Colitis_vs_Control/gsea_table.png",width:1e3,height:600}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p:M,i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function Jn(e){let t,n;return t=new kn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Cycling-T-cells/Colitis_vs_Control/fgsea.txt",data:[{pathway:"Oxidative phosphorylation",pval:5207e-20,padj:0,log2err:1.0672,ES:.7312,NES:2.0407,size:120,leadingEdge:"UQCRFS1,NDUFB10,ATP5F1C,NDUFB5,NDUFB3,NDUFA12,ATP5MC3,ATP5MC2,COX6A1,ATP5MG,NDUFA7,COX5A,NDUFA2,NDUFB4,NDUFB9,UQCRH,NDUFV2,SDHB,COX7A2,NDUFS4,NDUFB11,ATP6V1D,ATP5F1D,NDUFS7,NDUFA6,ATP6V0B,COX8A,NDUFS3,NDUFA8,ATP5PD,CYC1,COX7B,ATP6V0D1,NDUFAB1,ATP5PB,UQCRQ,ATP6V1A,SDHC,UQCRC2,ATP5MF,COX6B1,COX7A2L,NDUFS6,UQCRC1,NDUFA1,COX5B,ATP6V1C1,NDUFS8,ATP5F1A,NDUFB8,UQCR10,COX10,COX6C,UQCRHL,NDUFB7,NDUFC1,NDUFB6,NDUFA5,NDUFS2,NDUFA9,ATP5F1B,ATP6V1B2,ATP5PF,PPA1,NDUFA10,COX15,ATP6V1H,ATP5ME,NDUFS5,ATP6V0C,SDHD,NDUFB1,ATP6V1E1,NDUFA11,NDUFC2",id:0},{pathway:"Ascorbate and aldarate metabolism",pval:5e-4,padj:.0047,log2err:.4985,ES:-.5857,NES:-1.9824,size:27,leadingEdge:"ALDH3A2,UGDH,ALDH7A1,ALDH1B1,ALDH9A1",id:1},{pathway:"Starch and sucrose metabolism",pval:.0027,padj:.0183,log2err:.4317,ES:-.4763,NES:-1.6713,size:35,leadingEdge:"PGM1,GANC,PGM2L1,GAA,AGL,UGP2",id:2},{pathway:"Citrate cycle (TCA cycle)",pval:.0591,padj:.2954,log2err:.1919,ES:.5405,NES:1.3364,size:30,leadingEdge:"IDH2,SUCLG1,SDHB,IDH3G,SDHC,IDH3B,MDH2,PDHB,DLD,PCK2,SDHD,ACO2,FH,CS,MDH1,IDH1,SDHA,OGDH,ACLY,OGDHL,IDH3A,DLAT",id:3},{pathway:"Amino sugar and nucleotide sugar metabolism",pval:.0848,padj:.3391,log2err:.2878,ES:-.3183,NES:-1.2001,size:48,leadingEdge:"CYB5R1,PGM1,UGP2,CYB5R3,GNPDA2,GNE,GALT,FUK,NPL,NAGK,UGDH,GALE,GNPDA1,CHIT1",id:4},{pathway:"Galactose metabolism",pval:.1087,padj:.3624,log2err:.2878,ES:-.3476,NES:-1.2171,size:31,leadingEdge:"PGM1,GANC,GAA,UGP2,GALT,GLA,GALE",id:5},{pathway:"Propanoate metabolism",pval:.2435,padj:.6957,log2err:.0848,ES:.4619,NES:1.1574,size:32,leadingEdge:"SUCLG1,EHHADH,ACAT1,DLD,ABAT,LDHB,PCCB,BCKDHA,LDHA,ACSS1,ACAT2,ACACA,MCEE,DBT",id:6},{pathway:"Glyoxylate and dicarboxylate metabolism",pval:.2834,padj:.7084,log2err:.0773,ES:.4602,NES:1.1325,size:28,leadingEdge:"ACAT1,GLDC,SHMT2,MDH2,GLYCTK,DLD,PCCB,ACO2,ACAT2,CS,MCEE,MDH1,GRHPR,AFMID",id:7},{pathway:"Inositol phosphate metabolism",pval:.3935,padj:.8,log2err:.2035,ES:-.234,NES:-.9607,size:73,leadingEdge:"PLCB3,ISYNA1,PLCB1,PLCH2,MTM1,MTMR8,PIP4K2C,PIK3C2B,PIK3CB,ITPKB,IMPA1,ALDH6A1,SACM1L,INPP5B,PLCB2,PIK3CG,PLCG2,INPP5A,PIK3CA,INPP5E,MTMR3,IPMK,MTMR6,PLCG1",id:8},{pathway:"Pentose and glucuronate interconversions",pval:.4,padj:.8,log2err:.2713,ES:-.3027,NES:-1.0757,size:34,leadingEdge:"CRYL1,UGP2,UGDH,DHDH,DCXR",id:9},{pathway:"Sulfur metabolism",pval:.4781,padj:.8333,log2err:.0581,ES:.5079,NES:1.0082,size:9,leadingEdge:"SQOR,IMPAD1,MPST,TST,PAPSS2,BPNT1",id:10},{pathway:"Pentose phosphate pathway",pval:.5,padj:.8333,log2err:.2139,ES:-.2791,NES:-.969,size:30,leadingEdge:"PGM1,FBP1,ALDOC,RBKS,H6PD",id:11},{pathway:"Pyruvate metabolism",pval:.6023,padj:.9267,log2err:.0401,ES:.3699,NES:.9509,size:39,leadingEdge:"ACAT1,MDH2,PDHB,DLD,PCK2,LDHB,ACYP2,LDHA,ACSS1,ACAT2,FH,ACACA,MDH1,GRHPR",id:12},{pathway:"Fatty acid elongation",pval:.837,padj:.9989,log2err:.0253,ES:.3098,NES:.766,size:30,leadingEdge:"ACOT2,ABHD17B,ACAA2,ACOT7,ELOVL5,HADHB,ELOVL1,HADH,ABHD17C,HACD1,MECR",id:13},{pathway:"Butanoate metabolism",pval:.9136,padj:.9989,log2err:.0203,ES:.2759,NES:.679,size:28,leadingEdge:"EHHADH,ACAT1,HMGCS1,ABAT,ACAT2,L2HGDH,HADH,OXCT2,BDH1,OXCT1,HMGCL",id:14},{pathway:"Nitrogen metabolism",pval:.925,padj:.9989,log2err:.1294,ES:-.2552,NES:-.7458,size:17,leadingEdge:"CA3,CPS1,CA13,CA4,CA5A,CA9,CA12,CA7,CA6,GLUD2,GLUL,CA2,CA8,GLUD1,CA5B,CA1,CA14",id:15},{pathway:"Fatty acid biosynthesis",pval:.9316,padj:.9989,log2err:.0246,ES:.2992,NES:.6418,size:13,leadingEdge:"ACSL4,MCAT,ACACA,ACSL5,FASN",id:16},{pathway:"Fructose and mannose metabolism",pval:.9741,padj:.9989,log2err:.0149,ES:.2435,NES:.6122,size:33,leadingEdge:"TIGAR,TKFC,GMPPA,PMM2,MPI,HKDC1,GMDS,PFKP,HK2,TSTA3,SORD,PFKFB3,HK1,GMPPB,ALDOA",id:17},{pathway:"Glycolysis / Gluconeogenesis",pval:.9795,padj:.9989,log2err:.0101,ES:.2429,NES:.657,size:68,leadingEdge:"GALM,GAPDH,BPGM,PDHB,G6PC3,DLD,PCK2,LDHB,AKR1A1,LDHA,PGAM1,ACSS1,HKDC1,ADH5,PGM2,PFKP,HK2,ENO3,ENO1,HK1,DLAT,ADPGK,MINPP1,ALDOA,PDHA1",id:18},{pathway:"Fatty acid degradation",pval:.9989,padj:.9989,log2err:.0112,ES:.2098,NES:.549,size:44,leadingEdge:"ACAA1,EHHADH,ACAT1,ACSL4,ACAA2,ACAT2,HADHB,ADH5,ACSL5,HADH,GCDH",id:19}]}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p:M,i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function Zn(e){let t,n,a,s;return t=new ue({props:{$$slots:{default:[Qn]},$$scope:{ctx:e}}}),a=new ue({props:{$$slots:{default:[Jn]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function ea(e){let t,n;return t=new me({props:{$$slots:{content:[Zn],default:[Wn]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function ta(e){let t,n,a,s,p,c,m,g,u,h,v,y,A,S,P,L,E,w,T,U,N,D,R,B,I,H,G,F,V,z,O,q,k,K,j,X,Y,W,Q,J,ee,te,ne,ae,se,re,ie,le,oe,pe,ce;return a=new Yn({props:{title:"Oxidative phosphorylation",class:"pipen-report-table-image-descr"}}),p=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Cycling-T-cells/Colitis_vs_Control/fgsea_Oxidative-phosphorylation.png",class:"pipen-report-table-image",width:1e3,height:800}}),g=new Yn({props:{title:"Ascorbate and aldarate metabolism",class:"pipen-report-table-image-descr"}}),h=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Cycling-T-cells/Colitis_vs_Control/fgsea_Ascorbate-and-aldarate-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),A=new Yn({props:{title:"Starch and sucrose metabolism",class:"pipen-report-table-image-descr"}}),P=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Cycling-T-cells/Colitis_vs_Control/fgsea_Starch-and-sucrose-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),w=new Yn({props:{title:"Citrate cycle (TCA cycle)",class:"pipen-report-table-image-descr"}}),U=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Cycling-T-cells/Colitis_vs_Control/fgsea_Citrate-cycle-TCA-cycle-.png",class:"pipen-report-table-image",width:1e3,height:800}}),R=new Yn({props:{title:"Amino sugar and nucleotide sugar metabolism",class:"pipen-report-table-image-descr"}}),I=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Cycling-T-cells/Colitis_vs_Control/fgsea_Amino-sugar-and-nucleotide-sugar-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),F=new Yn({props:{title:"Galactose metabolism",class:"pipen-report-table-image-descr"}}),z=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Cycling-T-cells/Colitis_vs_Control/fgsea_Galactose-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),k=new Yn({props:{title:"Propanoate metabolism",class:"pipen-report-table-image-descr"}}),j=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Cycling-T-cells/Colitis_vs_Control/fgsea_Propanoate-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),W=new Yn({props:{title:"Glyoxylate and dicarboxylate metabolism",class:"pipen-report-table-image-descr"}}),J=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Cycling-T-cells/Colitis_vs_Control/fgsea_Glyoxylate-and-dicarboxylate-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new Yn({props:{title:"Inositol phosphate metabolism",class:"pipen-report-table-image-descr"}}),se=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Cycling-T-cells/Colitis_vs_Control/fgsea_Inositol-phosphate-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),le=new Yn({props:{title:"Pentose and glucuronate interconversions",class:"pipen-report-table-image-descr"}}),pe=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Cycling-T-cells/Colitis_vs_Control/fgsea_Pentose-and-glucuronate-interconversions.png",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=r("div"),n=r("div"),f(a.$$.fragment),s=$(),f(p.$$.fragment),c=$(),m=r("div"),f(g.$$.fragment),u=$(),f(h.$$.fragment),v=$(),y=r("div"),f(A.$$.fragment),S=$(),f(P.$$.fragment),L=$(),E=r("div"),f(w.$$.fragment),T=$(),f(U.$$.fragment),N=$(),D=r("div"),f(R.$$.fragment),B=$(),f(I.$$.fragment),H=$(),G=r("div"),f(F.$$.fragment),V=$(),f(z.$$.fragment),O=$(),q=r("div"),f(k.$$.fragment),K=$(),f(j.$$.fragment),X=$(),Y=r("div"),f(W.$$.fragment),Q=$(),f(J.$$.fragment),ee=$(),te=r("div"),f(ne.$$.fragment),ae=$(),f(se.$$.fragment),re=$(),ie=r("div"),f(le.$$.fragment),oe=$(),f(pe.$$.fragment),i(t,"class","pipen-report-table-of-images"),Z(t,"grid-template-columns","repeat(2, auto)")},m(e,r){l(e,t,r),x(t,n),b(a,n,null),x(n,s),b(p,n,null),x(t,c),x(t,m),b(g,m,null),x(m,u),b(h,m,null),x(t,v),x(t,y),b(A,y,null),x(y,S),b(P,y,null),x(t,L),x(t,E),b(w,E,null),x(E,T),b(U,E,null),x(t,N),x(t,D),b(R,D,null),x(D,B),b(I,D,null),x(t,H),x(t,G),b(F,G,null),x(G,V),b(z,G,null),x(t,O),x(t,q),b(k,q,null),x(q,K),b(j,q,null),x(t,X),x(t,Y),b(W,Y,null),x(Y,Q),b(J,Y,null),x(t,ee),x(t,te),b(ne,te,null),x(te,ae),b(se,te,null),x(t,re),x(t,ie),b(le,ie,null),x(ie,oe),b(pe,ie,null),ce=!0},p:M,i(e){ce||(o(a.$$.fragment,e),o(p.$$.fragment,e),o(g.$$.fragment,e),o(h.$$.fragment,e),o(A.$$.fragment,e),o(P.$$.fragment,e),o(w.$$.fragment,e),o(U.$$.fragment,e),o(R.$$.fragment,e),o(I.$$.fragment,e),o(F.$$.fragment,e),o(z.$$.fragment,e),o(k.$$.fragment,e),o(j.$$.fragment,e),o(W.$$.fragment,e),o(J.$$.fragment,e),o(ne.$$.fragment,e),o(se.$$.fragment,e),o(le.$$.fragment,e),o(pe.$$.fragment,e),ce=!0)},o(e){d(a.$$.fragment,e),d(p.$$.fragment,e),d(g.$$.fragment,e),d(h.$$.fragment,e),d(A.$$.fragment,e),d(P.$$.fragment,e),d(w.$$.fragment,e),d(U.$$.fragment,e),d(R.$$.fragment,e),d(I.$$.fragment,e),d(F.$$.fragment,e),d(z.$$.fragment,e),d(k.$$.fragment,e),d(j.$$.fragment,e),d(W.$$.fragment,e),d(J.$$.fragment,e),d(ne.$$.fragment,e),d(se.$$.fragment,e),d(le.$$.fragment,e),d(pe.$$.fragment,e),ce=!1},d(e){e&&_(t),C(a),C(p),C(g),C(h),C(A),C(P),C(w),C(U),C(R),C(I),C(F),C(z),C(k),C(j),C(W),C(J),C(ne),C(se),C(le),C(pe)}}}function na(e){let t,n,a,s;return t=new Te({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[ea]},$$scope:{ctx:e}}}),a=new Te({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[ta]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function aa(e){let t,n,a,s;return t=new ge({props:{label:"Plot"}}),a=new ge({props:{label:"Table"}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p:M,i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function sa(e){let t,n;return t=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Follicular-helper-T-cells/Colitis_vs_Control/gsea_table.png",width:1e3,height:600}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p:M,i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function ra(e){let t,n;return t=new kn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Follicular-helper-T-cells/Colitis_vs_Control/fgsea.txt",data:[{pathway:"Oxidative phosphorylation",pval:0,padj:0,log2err:.6901,ES:.4637,NES:1.8417,size:120,leadingEdge:"ATP5PD,NDUFS5,ATP5MG,ATP5F1E,COX7C,ATP5F1D,ATP5ME,UQCRH,NDUFB11,COX7A2,NDUFA12,NDUFB5,UQCRQ,ATP6V0B,COX7B,UQCR10,NDUFA1,COX5B,NDUFA3,NDUFB10,COX5A,NDUFB3,NDUFA6,NDUFA2,COX6A1,PPA1,ATP5MF,NDUFB1,ATP6V1F,NDUFB6,NDUFB8,NDUFB9,UQCRB,NDUFB4,COX7A2L,ATP5F1A,COX17,UQCR11,ATP6V0C,COX10,LHPP,COX4I1,UQCRFS1,ATP5MC3",id:0},{pathway:"Nitrogen metabolism",pval:8e-4,padj:.0075,log2err:.4773,ES:.7371,NES:2.0668,size:17,leadingEdge:"GLUL,CA5B,GLUD1,CA13,CA3,CPS1,CA14,GLUD2,CA6,CA4,CA9",id:1},{pathway:"Inositol phosphate metabolism",pval:.0012,padj:.0079,log2err:.4551,ES:-.5527,NES:-1.5906,size:73,leadingEdge:"INPP5B,PIK3CG,IPPK,MTMR4,PIP4K2C,MTM1,PIK3CD,INPP4B,MTMR2,PIK3CB,ISYNA1,PLCB2,IMPA1,MINPP1,PIK3CA,SACM1L,MTMR6,INPP5E,PIKFYVE,PI4K2A,MTMR8,PIK3C2A,OCRL,CDIPT,INPP5F,ALDH6A1,PLCG1,IMPAD1,PLCH2,MTMR1,SYNJ1,PIP4K2B,PLCB3,IPMK,ITPKB",id:2},{pathway:"Citrate cycle (TCA cycle)",pval:.0033,padj:.0163,log2err:.4317,ES:-.6274,NES:-1.604,size:30,leadingEdge:"DLAT,ACO1,FH,SUCLA2,DLST,MDH1,DLD,CS,SDHD,ACLY,SDHC,IDH3A,IDH3B,PDHB,PDHA1,SUCLG2,PC,IDH3G,ACO2",id:3},{pathway:"Ascorbate and aldarate metabolism",pval:.0057,padj:.023,log2err:.407,ES:-.6374,NES:-1.6101,size:27,leadingEdge:"ALDH9A1,ALDH3A2",id:4},{pathway:"Fatty acid degradation",pval:.0174,padj:.058,log2err:.3525,ES:-.535,NES:-1.4462,size:44,leadingEdge:"ACSL3,ALDH9A1,ACADVL,CPT1A,ACADSB,ALDH3A2,ADH5,ACSL5,ACOX1,ACSL1,ACSL6,CPT2,ACOX3,GCDH,ACAA2,ECHS1,ACAT2,HADH,CPT1C,ALDH1B1,ACADS,HADHB",id:5},{pathway:"Amino sugar and nucleotide sugar metabolism",pval:.08,padj:.2284,log2err:.1657,ES:-.4779,NES:-1.3164,size:48,leadingEdge:"GNPNAT1,FPGT,GNPDA2,AMDHD2,PGM3,GNE,PMM2,CYB5R1,GALT,GMPPA,GPI,CMAS,NAGK,GFPT1,GALK1,HEXA,PGM2,CHIT1",id:6},{pathway:"Propanoate metabolism",pval:.1027,padj:.2548,log2err:.1502,ES:-.5035,NES:-1.3064,size:32,leadingEdge:"MLYCD,HIBCH,MUT,BCKDHB,SUCLA2,ACSS2,DLD,DBT,ALDH6A1,PCCB,ACACB,ECHS1,ACAT2,ECHDC1,SUCLG2,HADHA,ACACA,ACSS1",id:7},{pathway:"Glycolysis / Gluconeogenesis",pval:.1147,padj:.2548,log2err:.1343,ES:-.432,NES:-1.2341,size:68,leadingEdge:"DLAT,ALDH9A1,ENO2,FBP1,ALDH3A2,ADH5,MINPP1,G6PC3,ACSS2,PKM,ALDOC,DLD,GPI,ADPGK",id:8},{pathway:"Sulfur metabolism",pval:.194,padj:.3851,log2err:.1847,ES:.5478,NES:1.2489,size:9,leadingEdge:"ETHE1,SQOR,MPST,PAPSS1,TST",id:9},{pathway:"Pyruvate metabolism",pval:.2118,padj:.3851,log2err:.0966,ES:-.4416,NES:-1.1865,size:39,leadingEdge:"DLAT,ALDH9A1,ALDH3A2,FH,ACSS2,MDH1,PKM,DLD,ME2,ACACB,GLO1,ACAT2,ALDH1B1,ME3,PDHB,PDHA1,PC,ACACA,ACSS1,LDHD",id:10},{pathway:"Pentose and glucuronate interconversions",pval:.3073,padj:.5122,log2err:.0771,ES:-.4286,NES:-1.1262,size:34,leadingEdge:"CRYL1,AKR1B1,DCXR,GUSB,SORD,UGDH,RPE",id:11},{pathway:"Starch and sucrose metabolism",pval:.3388,padj:.5213,log2err:.072,ES:-.4171,NES:-1.1037,size:35,leadingEdge:"AGL,G6PC3,GYS1,AMY2B,GAA,GPI,GANC,PGM2,PYGB,GBE1",id:12},{pathway:"Butanoate metabolism",pval:.4015,padj:.5736,log2err:.0663,ES:-.4144,NES:-1.0466,size:28,leadingEdge:"ALDH5A1,AACS,HMGCL,OXCT1,L2HGDH,ECHS1,ACAT2,HADH,ACADS,BDH1,HADHA",id:13},{pathway:"Pentose phosphate pathway",pval:.4846,padj:.6449,log2err:.0568,ES:-.3904,NES:-.9983,size:30,leadingEdge:"PRPS1,PRPS2,FBP1,H6PD,IDNK,RPIA,ALDOC,GPI,PGM2",id:14},{pathway:"Glyoxylate and dicarboxylate metabolism",pval:.5374,padj:.6449,log2err:.0525,ES:-.3786,NES:-.9561,size:28,leadingEdge:"HYI,ACO1,MUT,CAT,MDH1,DLD,CS,GCSH,PCCB,ACAT2",id:15},{pathway:"Fructose and mannose metabolism",pval:.5776,padj:.6449,log2err:.0472,ES:-.3604,NES:-.9432,size:33,leadingEdge:"ENOSF1,FPGT,FBP1,PMM2,AKR1B1,ALDOC,GMPPA",id:16},{pathway:"Fatty acid biosynthesis",pval:.5804,padj:.6449,log2err:.0523,ES:-.4213,NES:-.9289,size:13,leadingEdge:"ACSL3,ACSL5,ACSL1,ACSL6,OXSM,ACACB",id:17},{pathway:"Fatty acid elongation",pval:.7078,padj:.745,log2err:.0394,ES:-.3334,NES:-.8525,size:30,leadingEdge:"HSD17B12,MECR,HACD3,ACOT4,HACD2,PPT2,ACAA2,ACOT2,PPT1,ECHS1,HADH,ABHD17B,HADHB,HADHA,TECR",id:18},{pathway:"Galactose metabolism",pval:.9415,padj:.9415,log2err:.0248,ES:-.2572,NES:-.6617,size:31,leadingEdge:"G6PC3,GALT,GAA,AKR1B1,GANC,GALK1,PGM2,GLA,PFKL",id:19}]}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p:M,i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function ia(e){let t,n,a,s;return t=new ue({props:{$$slots:{default:[sa]},$$scope:{ctx:e}}}),a=new ue({props:{$$slots:{default:[ra]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function la(e){let t,n;return t=new me({props:{$$slots:{content:[ia],default:[aa]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function oa(e){let t,n,a,s,p,c,m,g,u,h,v,y,A,S,P,L,E,w,T,U,N,D,R,B,I,H,G,F,V,z,O,q,k,K,j,X,Y,W,Q,J,ee,te,ne,ae,se,re,ie,le,oe,pe,ce;return a=new Yn({props:{title:"Oxidative phosphorylation",class:"pipen-report-table-image-descr"}}),p=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Follicular-helper-T-cells/Colitis_vs_Control/fgsea_Oxidative-phosphorylation.png",class:"pipen-report-table-image",width:1e3,height:800}}),g=new Yn({props:{title:"Nitrogen metabolism",class:"pipen-report-table-image-descr"}}),h=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Follicular-helper-T-cells/Colitis_vs_Control/fgsea_Nitrogen-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),A=new Yn({props:{title:"Inositol phosphate metabolism",class:"pipen-report-table-image-descr"}}),P=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Follicular-helper-T-cells/Colitis_vs_Control/fgsea_Inositol-phosphate-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),w=new Yn({props:{title:"Citrate cycle (TCA cycle)",class:"pipen-report-table-image-descr"}}),U=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Follicular-helper-T-cells/Colitis_vs_Control/fgsea_Citrate-cycle-TCA-cycle-.png",class:"pipen-report-table-image",width:1e3,height:800}}),R=new Yn({props:{title:"Ascorbate and aldarate metabolism",class:"pipen-report-table-image-descr"}}),I=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Follicular-helper-T-cells/Colitis_vs_Control/fgsea_Ascorbate-and-aldarate-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),F=new Yn({props:{title:"Fatty acid degradation",class:"pipen-report-table-image-descr"}}),z=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Follicular-helper-T-cells/Colitis_vs_Control/fgsea_Fatty-acid-degradation.png",class:"pipen-report-table-image",width:1e3,height:800}}),k=new Yn({props:{title:"Amino sugar and nucleotide sugar metabolism",class:"pipen-report-table-image-descr"}}),j=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Follicular-helper-T-cells/Colitis_vs_Control/fgsea_Amino-sugar-and-nucleotide-sugar-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),W=new Yn({props:{title:"Propanoate metabolism",class:"pipen-report-table-image-descr"}}),J=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Follicular-helper-T-cells/Colitis_vs_Control/fgsea_Propanoate-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new Yn({props:{title:"Glycolysis / Gluconeogenesis",class:"pipen-report-table-image-descr"}}),se=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Follicular-helper-T-cells/Colitis_vs_Control/fgsea_Glycolysis-Gluconeogenesis.png",class:"pipen-report-table-image",width:1e3,height:800}}),le=new Yn({props:{title:"Sulfur metabolism",class:"pipen-report-table-image-descr"}}),pe=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Follicular-helper-T-cells/Colitis_vs_Control/fgsea_Sulfur-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=r("div"),n=r("div"),f(a.$$.fragment),s=$(),f(p.$$.fragment),c=$(),m=r("div"),f(g.$$.fragment),u=$(),f(h.$$.fragment),v=$(),y=r("div"),f(A.$$.fragment),S=$(),f(P.$$.fragment),L=$(),E=r("div"),f(w.$$.fragment),T=$(),f(U.$$.fragment),N=$(),D=r("div"),f(R.$$.fragment),B=$(),f(I.$$.fragment),H=$(),G=r("div"),f(F.$$.fragment),V=$(),f(z.$$.fragment),O=$(),q=r("div"),f(k.$$.fragment),K=$(),f(j.$$.fragment),X=$(),Y=r("div"),f(W.$$.fragment),Q=$(),f(J.$$.fragment),ee=$(),te=r("div"),f(ne.$$.fragment),ae=$(),f(se.$$.fragment),re=$(),ie=r("div"),f(le.$$.fragment),oe=$(),f(pe.$$.fragment),i(t,"class","pipen-report-table-of-images"),Z(t,"grid-template-columns","repeat(2, auto)")},m(e,r){l(e,t,r),x(t,n),b(a,n,null),x(n,s),b(p,n,null),x(t,c),x(t,m),b(g,m,null),x(m,u),b(h,m,null),x(t,v),x(t,y),b(A,y,null),x(y,S),b(P,y,null),x(t,L),x(t,E),b(w,E,null),x(E,T),b(U,E,null),x(t,N),x(t,D),b(R,D,null),x(D,B),b(I,D,null),x(t,H),x(t,G),b(F,G,null),x(G,V),b(z,G,null),x(t,O),x(t,q),b(k,q,null),x(q,K),b(j,q,null),x(t,X),x(t,Y),b(W,Y,null),x(Y,Q),b(J,Y,null),x(t,ee),x(t,te),b(ne,te,null),x(te,ae),b(se,te,null),x(t,re),x(t,ie),b(le,ie,null),x(ie,oe),b(pe,ie,null),ce=!0},p:M,i(e){ce||(o(a.$$.fragment,e),o(p.$$.fragment,e),o(g.$$.fragment,e),o(h.$$.fragment,e),o(A.$$.fragment,e),o(P.$$.fragment,e),o(w.$$.fragment,e),o(U.$$.fragment,e),o(R.$$.fragment,e),o(I.$$.fragment,e),o(F.$$.fragment,e),o(z.$$.fragment,e),o(k.$$.fragment,e),o(j.$$.fragment,e),o(W.$$.fragment,e),o(J.$$.fragment,e),o(ne.$$.fragment,e),o(se.$$.fragment,e),o(le.$$.fragment,e),o(pe.$$.fragment,e),ce=!0)},o(e){d(a.$$.fragment,e),d(p.$$.fragment,e),d(g.$$.fragment,e),d(h.$$.fragment,e),d(A.$$.fragment,e),d(P.$$.fragment,e),d(w.$$.fragment,e),d(U.$$.fragment,e),d(R.$$.fragment,e),d(I.$$.fragment,e),d(F.$$.fragment,e),d(z.$$.fragment,e),d(k.$$.fragment,e),d(j.$$.fragment,e),d(W.$$.fragment,e),d(J.$$.fragment,e),d(ne.$$.fragment,e),d(se.$$.fragment,e),d(le.$$.fragment,e),d(pe.$$.fragment,e),ce=!1},d(e){e&&_(t),C(a),C(p),C(g),C(h),C(A),C(P),C(w),C(U),C(R),C(I),C(F),C(z),C(k),C(j),C(W),C(J),C(ne),C(se),C(le),C(pe)}}}function pa(e){let t,n,a,s;return t=new Te({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[la]},$$scope:{ctx:e}}}),a=new Te({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[oa]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function ca(e){let t,n,a,s;return t=new ge({props:{label:"Plot"}}),a=new ge({props:{label:"Table"}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p:M,i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function da(e){let t,n;return t=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_gamma-delta-T-cells/Colitis_vs_Control/gsea_table.png",width:1e3,height:600}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p:M,i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function _a(e){let t,n;return t=new kn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_gamma-delta-T-cells/Colitis_vs_Control/fgsea.txt",data:[{pathway:"Ascorbate and aldarate metabolism",pval:1e-4,padj:.0022,log2err:.5384,ES:-.7489,NES:-1.7466,size:27,leadingEdge:"UGDH,ALDH1B1,ALDH9A1",id:0},{pathway:"Citrate cycle (TCA cycle)",pval:.0276,padj:.23,log2err:.3525,ES:-.609,NES:-1.4482,size:30,leadingEdge:"DLAT,FH,ACO1,SUCLA2,ACLY,ACO2,IDH3G,OGDH,SDHD,PDHA1,SDHA",id:1},{pathway:"Amino sugar and nucleotide sugar metabolism",pval:.0385,padj:.23,log2err:.2344,ES:-.5423,NES:-1.3418,size:48,leadingEdge:"FPGT,PGM3,MPI,UGDH,PGM2,GNPDA2,CYB5R3,GALT,CYB5R1,NAGK,NPL,HEXB,UXS1,GNPNAT1,UAP1,AMDHD2,GMPPA,GPI,GALE,GNPDA1,FUK,CMAS,PGM1,GFPT1,GNE",id:2},{pathway:"Pentose and glucuronate interconversions",pval:.0612,padj:.23,log2err:.1864,ES:-.557,NES:-1.3338,size:34,leadingEdge:"CRYL1,UGDH,AKR1B1,SORD",id:3},{pathway:"Fatty acid degradation",pval:.065,padj:.23,log2err:.1782,ES:-.5306,NES:-1.2982,size:44,leadingEdge:"ADH5,ACSL3,ACADVL,CPT1A,ACAA2,HADHB,ACADS,ACSL5,ALDH1B1,HADHA,ALDH9A1,ACOX3,ACSL6,ACADSB,ECHS1,ACSL4,ACSL1,CPT1C,CPT1B,CPT2,GCDH,ACAT2",id:4},{pathway:"Nitrogen metabolism",pval:.069,padj:.23,log2err:.2878,ES:.4536,NES:1.2859,size:17,leadingEdge:"GLUD1,GLUL,GLUD2,CA14,CA8,CA2,CA4,CPS1,CA9",id:5},{pathway:"Pyruvate metabolism",pval:.176,padj:.503,log2err:.1024,ES:-.4935,NES:-1.2011,size:39,leadingEdge:"GLO1,DLAT,FH,ACACA,ALDH1B1,PKM,ALDH9A1,PDHA1,ME2,ACYP1,GRHPR,MDH1",id:6},{pathway:"Propanoate metabolism",pval:.3185,padj:.7011,log2err:.0708,ES:-.4718,NES:-1.123,size:32,leadingEdge:"BCKDHB,MLYCD,SUCLA2,PCCB,ACACA,ECHDC1,HADHA,MCEE,ECHS1,BCKDHA,SUCLG1,MUT",id:7},{pathway:"Butanoate metabolism",pval:.3505,padj:.7011,log2err:.0667,ES:-.4675,NES:-1.0956,size:28,leadingEdge:"ACSM3,ACADS,HADHA,BDH2,HMGCL,OXCT1,ALDH5A1,ECHS1,AACS,L2HGDH,ACAT2,HMGCS2,ACAT1,BDH1",id:8},{pathway:"Glyoxylate and dicarboxylate metabolism",pval:.3505,padj:.7011,log2err:.0667,ES:-.4674,NES:-1.0952,size:28,leadingEdge:"ACO1,HYI,CAT,PCCB,ACO2,MCEE,GCSH,SHMT1,GRHPR,MDH1",id:9},{pathway:"Sulfur metabolism",pval:.4434,padj:.7543,log2err:.1421,ES:.4257,NES:.9657,size:9,leadingEdge:"MPST,IMPAD1,TST",id:10},{pathway:"Fatty acid elongation",pval:.4941,padj:.7543,log2err:.0495,ES:-.4228,NES:-1.0056,size:30,leadingEdge:"PPT1,HSD17B12,HACD2,ACAA2,HADHB,HADHA,ELOVL1,MECR,HACD3,ECHS1,ACOT4",id:11},{pathway:"Glycolysis / Gluconeogenesis",pval:.5339,padj:.7543,log2err:.0439,ES:-.3883,NES:-.9896,size:68,leadingEdge:"ENO2,DLAT,ADH5,PGM2,PFKL,ALDH1B1,PKM,ALDH9A1,PDHA1,ALDOA,GPI,FBP1",id:12},{pathway:"Pentose phosphate pathway",pval:.5553,padj:.7543,log2err:.0442,ES:-.4082,NES:-.9709,size:30,leadingEdge:"PGM2,PRPS2,PFKL,ALDOA,PGD,GPI,FBP1,TALDO1,ALDOC,G6PD,DERA,PGM1",id:13},{pathway:"Fatty acid biosynthesis",pval:.5841,padj:.7543,log2err:.047,ES:-.4525,NES:-.9382,size:13,leadingEdge:"ACSL3,ACSL5,ACACA,ACSL6,ACSL4,ACSL1",id:14},{pathway:"Fructose and mannose metabolism",pval:.6289,padj:.7543,log2err:.0386,ES:-.3829,NES:-.9145,size:33,leadingEdge:"FPGT,MPI,KHK,PFKL,AKR1B1,SORD,ALDOA,GMPPA,FBP1,ALDOC,FUK",id:15},{pathway:"Galactose metabolism",pval:.6412,padj:.7543,log2err:.0377,ES:-.3815,NES:-.9062,size:31,leadingEdge:"PGM2,GALT,PFKL,GLB1,AKR1B1,GAA,GALE,GLA,PGM1,GANC",id:16},{pathway:"Starch and sucrose metabolism",pval:.6838,padj:.7597,log2err:.0343,ES:-.3661,NES:-.883,size:35,leadingEdge:"PGM2,AGL,GAA,GYG1,GPI,GBE1,PGM1,GANC,PYGB",id:17},{pathway:"Inositol phosphate metabolism",pval:.7566,padj:.7965,log2err:.0271,ES:-.3437,NES:-.8798,size:73,leadingEdge:"IPPK,IMPA1,SACM1L,PIP4K2B,PLCB2,PIP4K2C,PLCB3,ISYNA1,INPP5E,MTMR1,MTMR4,INPP5B,MTM1,PIKFYVE,PIK3CG,MTMR8,PIK3C2B",id:18},{pathway:"Oxidative phosphorylation",pval:.8157,padj:.8157,log2err:.0222,ES:-.3312,NES:-.8721,size:120,leadingEdge:"ATP6V1E1,PPA2,NDUFV3,ATP5MC1,LHPP,ATP5PB,NDUFC2,NDUFS1,ATP6V0E1,SDHD,ATP5PO,NDUFA5,ATP5PF,SDHA,NDUFA9,COX11,NDUFS6,UQCRC2,NDUFB9,NDUFA4,NDUFA11,NDUFS4,NDUFB7,ATP5F1A,COX6A2,NDUFB2,ATP6V1A,NDUFA8,ATP6V1B2,NDUFB6,NDUFC1,UQCRC1,ATP6V1G1,NDUFAB1,NDUFA2,ATP6V1F,NDUFB10,SDHC,NDUFA13,NDUFB8,ATP5F1B,NDUFB5,ATP6V1G2,ATP6V1H,NDUFA7,NDUFV1,ATP6V0E2,NDUFA3",id:19}]}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p:M,i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function ma(e){let t,n,a,s;return t=new ue({props:{$$slots:{default:[da]},$$scope:{ctx:e}}}),a=new ue({props:{$$slots:{default:[_a]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function ga(e){let t,n;return t=new me({props:{$$slots:{content:[ma],default:[ca]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function ua(e){let t,n,a,s,p,c,m,g,u,h,v,y,A,S,P,L,E,w,T,U,N,D,R,B,I,H,G,F,V,z,O,q,k,K,j,X,Y,W,Q,J,ee,te,ne,ae,se,re,ie,le,oe,pe,ce;return a=new Yn({props:{title:"Ascorbate and aldarate metabolism",class:"pipen-report-table-image-descr"}}),p=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_gamma-delta-T-cells/Colitis_vs_Control/fgsea_Ascorbate-and-aldarate-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),g=new Yn({props:{title:"Citrate cycle (TCA cycle)",class:"pipen-report-table-image-descr"}}),h=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_gamma-delta-T-cells/Colitis_vs_Control/fgsea_Citrate-cycle-TCA-cycle-.png",class:"pipen-report-table-image",width:1e3,height:800}}),A=new Yn({props:{title:"Amino sugar and nucleotide sugar metabolism",class:"pipen-report-table-image-descr"}}),P=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_gamma-delta-T-cells/Colitis_vs_Control/fgsea_Amino-sugar-and-nucleotide-sugar-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),w=new Yn({props:{title:"Pentose and glucuronate interconversions",class:"pipen-report-table-image-descr"}}),U=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_gamma-delta-T-cells/Colitis_vs_Control/fgsea_Pentose-and-glucuronate-interconversions.png",class:"pipen-report-table-image",width:1e3,height:800}}),R=new Yn({props:{title:"Fatty acid degradation",class:"pipen-report-table-image-descr"}}),I=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_gamma-delta-T-cells/Colitis_vs_Control/fgsea_Fatty-acid-degradation.png",class:"pipen-report-table-image",width:1e3,height:800}}),F=new Yn({props:{title:"Nitrogen metabolism",class:"pipen-report-table-image-descr"}}),z=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_gamma-delta-T-cells/Colitis_vs_Control/fgsea_Nitrogen-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),k=new Yn({props:{title:"Pyruvate metabolism",class:"pipen-report-table-image-descr"}}),j=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_gamma-delta-T-cells/Colitis_vs_Control/fgsea_Pyruvate-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),W=new Yn({props:{title:"Propanoate metabolism",class:"pipen-report-table-image-descr"}}),J=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_gamma-delta-T-cells/Colitis_vs_Control/fgsea_Propanoate-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new Yn({props:{title:"Butanoate metabolism",class:"pipen-report-table-image-descr"}}),se=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_gamma-delta-T-cells/Colitis_vs_Control/fgsea_Butanoate-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),le=new Yn({props:{title:"Glyoxylate and dicarboxylate metabolism",class:"pipen-report-table-image-descr"}}),pe=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_gamma-delta-T-cells/Colitis_vs_Control/fgsea_Glyoxylate-and-dicarboxylate-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=r("div"),n=r("div"),f(a.$$.fragment),s=$(),f(p.$$.fragment),c=$(),m=r("div"),f(g.$$.fragment),u=$(),f(h.$$.fragment),v=$(),y=r("div"),f(A.$$.fragment),S=$(),f(P.$$.fragment),L=$(),E=r("div"),f(w.$$.fragment),T=$(),f(U.$$.fragment),N=$(),D=r("div"),f(R.$$.fragment),B=$(),f(I.$$.fragment),H=$(),G=r("div"),f(F.$$.fragment),V=$(),f(z.$$.fragment),O=$(),q=r("div"),f(k.$$.fragment),K=$(),f(j.$$.fragment),X=$(),Y=r("div"),f(W.$$.fragment),Q=$(),f(J.$$.fragment),ee=$(),te=r("div"),f(ne.$$.fragment),ae=$(),f(se.$$.fragment),re=$(),ie=r("div"),f(le.$$.fragment),oe=$(),f(pe.$$.fragment),i(t,"class","pipen-report-table-of-images"),Z(t,"grid-template-columns","repeat(2, auto)")},m(e,r){l(e,t,r),x(t,n),b(a,n,null),x(n,s),b(p,n,null),x(t,c),x(t,m),b(g,m,null),x(m,u),b(h,m,null),x(t,v),x(t,y),b(A,y,null),x(y,S),b(P,y,null),x(t,L),x(t,E),b(w,E,null),x(E,T),b(U,E,null),x(t,N),x(t,D),b(R,D,null),x(D,B),b(I,D,null),x(t,H),x(t,G),b(F,G,null),x(G,V),b(z,G,null),x(t,O),x(t,q),b(k,q,null),x(q,K),b(j,q,null),x(t,X),x(t,Y),b(W,Y,null),x(Y,Q),b(J,Y,null),x(t,ee),x(t,te),b(ne,te,null),x(te,ae),b(se,te,null),x(t,re),x(t,ie),b(le,ie,null),x(ie,oe),b(pe,ie,null),ce=!0},p:M,i(e){ce||(o(a.$$.fragment,e),o(p.$$.fragment,e),o(g.$$.fragment,e),o(h.$$.fragment,e),o(A.$$.fragment,e),o(P.$$.fragment,e),o(w.$$.fragment,e),o(U.$$.fragment,e),o(R.$$.fragment,e),o(I.$$.fragment,e),o(F.$$.fragment,e),o(z.$$.fragment,e),o(k.$$.fragment,e),o(j.$$.fragment,e),o(W.$$.fragment,e),o(J.$$.fragment,e),o(ne.$$.fragment,e),o(se.$$.fragment,e),o(le.$$.fragment,e),o(pe.$$.fragment,e),ce=!0)},o(e){d(a.$$.fragment,e),d(p.$$.fragment,e),d(g.$$.fragment,e),d(h.$$.fragment,e),d(A.$$.fragment,e),d(P.$$.fragment,e),d(w.$$.fragment,e),d(U.$$.fragment,e),d(R.$$.fragment,e),d(I.$$.fragment,e),d(F.$$.fragment,e),d(z.$$.fragment,e),d(k.$$.fragment,e),d(j.$$.fragment,e),d(W.$$.fragment,e),d(J.$$.fragment,e),d(ne.$$.fragment,e),d(se.$$.fragment,e),d(le.$$.fragment,e),d(pe.$$.fragment,e),ce=!1},d(e){e&&_(t),C(a),C(p),C(g),C(h),C(A),C(P),C(w),C(U),C(R),C(I),C(F),C(z),C(k),C(j),C(W),C(J),C(ne),C(se),C(le),C(pe)}}}function fa(e){let t,n,a,s;return t=new Te({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[ga]},$$scope:{ctx:e}}}),a=new Te({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[ua]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function ba(e){let t,n,a,s;return t=new ge({props:{label:"Plot"}}),a=new ge({props:{label:"Table"}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p:M,i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function Ca(e){let t,n;return t=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Plasmablasts/Colitis_vs_Control/gsea_table.png",width:1e3,height:600}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p:M,i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function $a(e){let t,n;return t=new kn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Plasmablasts/Colitis_vs_Control/fgsea.txt",data:[{pathway:"Oxidative phosphorylation",pval:303e-30,padj:606e-29,log2err:1.3803,ES:.8052,NES:3.0158,size:120,leadingEdge:"ATP5MG,COX7A2,UQCR11,UQCRB,ATP6V1G1,UQCRFS1,COX7B,COX5B,ATP5MC2,UQCR10,ATP5F1D,NDUFS7,NDUFB10,COX4I1,NDUFA1,NDUFB4,NDUFB3,COX7C,ATP6V0C,ATP5MC3,COX6C,NDUFB11,ATP5F1E,ATP5MF,NDUFA3,NDUFS5,NDUFA7,NDUFA12,COX6B1,NDUFB1,ATP5ME,NDUFA6,COX6A1,UQCRQ,UQCRH,NDUFA2,ATP6V0B,COX7A2L,ATP5PD,SDHB,ATP5F1C,NDUFA5,SDHC,NDUFB5,NDUFA8,NDUFS3,ATP5PF,COX8A,NDUFV2,UQCRHL,NDUFAB1,NDUFC1,NDUFA13,PPA1,COX5A,ATP6V0D1,NDUFS6,NDUFS8,NDUFA11,NDUFB6,ATP6V1F,NDUFB2,NDUFB9,ATP6V1E1,UQCRC2,NDUFB7,ATP5F1A,NDUFB8,ATP5PB,ATP6V1A,NDUFA4,ATP6AP1",id:0},{pathway:"Amino sugar and nucleotide sugar metabolism",pval:0,padj:4e-4,log2err:.5573,ES:-.6308,NES:-1.947,size:48,leadingEdge:"FUK,CYB5R3,PGM1,UGP2,NPL,UAP1L1,UGDH,UAP1,GALT,PGM2,CYB5RL,PGM3,NAGK,FPGT,GMPPB,GNE,GFPT1,HK1,GALE,HEXB,PMM1,CYB5R1,CYB5R2,GNPDA2",id:1},{pathway:"Inositol phosphate metabolism",pval:1e-4,padj:6e-4,log2err:.5384,ES:-.5378,NES:-1.7883,size:73,leadingEdge:"ISYNA1,PLCH2,PLCB3,INPP5B,MTMR8,MTM1,ALDH6A1,PLCG2,PIP5K1C,MTMR6,PIP4K2C,ITPKB,PIK3CA,PIK3CG,PIK3C2B,INPP5E,IPMK,PIP4K2B,INPP4A,MTMR4,MTMR3,PLCB2,MTMR2,SACM1L,PI4K2B,IMPA1,SYNJ1,PLCB1,PIK3CB,PLCB4,PLCH1,PLCG1,INPP5F,ITPK1,PTEN,PLCD3,PIK3CD,PIKFYVE,INPP5A",id:2},{pathway:"Ascorbate and aldarate metabolism",pval:.0014,padj:.0071,log2err:.4551,ES:-.6448,NES:-1.7699,size:27,leadingEdge:"ALDH3A2,UGDH,ALDH7A1,ALDH9A1,ALDH2,ALDH1B1",id:3},{pathway:"Starch and sucrose metabolism",pval:.0037,padj:.0126,log2err:.4317,ES:-.5764,NES:-1.6667,size:35,leadingEdge:"PGM2L1,PGM1,AGL,UGP2,PGM2,GAA,HK1,GANC,PYGM",id:4},{pathway:"Fatty acid degradation",pval:.0038,padj:.0126,log2err:.4317,ES:-.543,NES:-1.6515,size:44,leadingEdge:"ACSL3,CPT1A,ALDH3A2,ACADSB,ECI2,ACADVL,HADHB,ACOX1,ACSL1,HADHA,ALDH7A1,ACOX3,CPT1C,ADH5,ACADM,ALDH9A1,CPT1B,ECHS1,ECI1,ACADS,ACAT2,ALDH2,ALDH1B1,ACSL6,ACSL5,HADH,CPT2",id:5},{pathway:"Pentose and glucuronate interconversions",pval:.0073,padj:.0188,log2err:.407,ES:-.5548,NES:-1.5987,size:34,leadingEdge:"UGP2,UGDH,CRYL1,DHDH,GUSB,KL",id:6},{pathway:"Galactose metabolism",pval:.0075,padj:.0188,log2err:.407,ES:-.5687,NES:-1.603,size:31,leadingEdge:"PGM1,UGP2,LCT,GALT,PGM2,GAA,PFKL,GLA,HK1,GALE,GANC,GLB1",id:7},{pathway:"Fatty acid elongation",pval:.0295,padj:.0655,log2err:.3525,ES:-.5245,NES:-1.467,size:30,leadingEdge:"HSD17B12,HACD2,ELOVL2,ELOVL5,HACD4,HADHB,HADHA,ELOVL4,ELOVL1,ECHS1,HACD1,HACD3,ACOT1,ELOVL6,HADH,PPT2",id:8},{pathway:"Butanoate metabolism",pval:.0329,padj:.0658,log2err:.3218,ES:-.5312,NES:-1.4674,size:28,leadingEdge:"AACS,ALDH5A1,HMGCS2,HADHA,ACSM3,OXCT1,ECHS1,BDH2,ACADS,ACAT2,OXCT2,BDH1,HADH",id:9},{pathway:"Pentose phosphate pathway",pval:.1431,padj:.2602,log2err:.1511,ES:-.4466,NES:-1.2492,size:30,leadingEdge:"PGM1,FBP1,H6PD,PGM2,PFKL,RBKS",id:10},{pathway:"Fructose and mannose metabolism",pval:.1734,padj:.289,log2err:.135,ES:-.4213,NES:-1.2078,size:33,leadingEdge:"FUK,FBP1,FPGT,PFKL,GMPPB,HK1,ENOSF1,PMM1,PFKFB4",id:11},{pathway:"Glyoxylate and dicarboxylate metabolism",pval:.1931,padj:.2971,log2err:.1288,ES:-.4352,NES:-1.2024,size:28,leadingEdge:"ACO1,HYI,CAT,MUT,PCCA,CS,ACO2,GCSH",id:12},{pathway:"Fatty acid biosynthesis",pval:.4043,padj:.5124,log2err:.085,ES:-.4553,NES:-1.0506,size:13,leadingEdge:"ACSL3,ACACB,ACSL1",id:13},{pathway:"Propanoate metabolism",pval:.4064,padj:.5124,log2err:.0806,ES:-.3631,NES:-1.0344,size:32,leadingEdge:"SUCLA2,ALDH6A1,MLYCD,ACACB,MUT,HADHA,PCCA,ACADM,BCKDHB,ECHS1,SUCLG2,ECHDC1,HIBCH",id:14},{pathway:"Nitrogen metabolism",pval:.4099,padj:.5124,log2err:.0831,ES:-.4116,NES:-1.0255,size:17,leadingEdge:"GLUD1,CA3,CA8",id:15},{pathway:"Pyruvate metabolism",pval:.4859,padj:.5717,log2err:.071,ES:-.3277,NES:-.9742,size:39,leadingEdge:"ALDH3A2,ME2,GLO1,PC,ACACB,ALDH7A1,HAGH,ALDH9A1,ME1,PKM,ACYP1,PDHA1,ME3,ACAT2,ALDH2,ALDH1B1",id:16},{pathway:"Glycolysis / Gluconeogenesis",pval:.7111,padj:.7901,log2err:.0507,ES:-.2717,NES:-.8874,size:68,leadingEdge:"PGM1,FBP1,ENO2,ALDH3A2,PGM2,PFKL,ALDH7A1,HK1,ADH5,PGAM2,ALDH9A1",id:17},{pathway:"Citrate cycle (TCA cycle)",pval:.7819,padj:.8231,log2err:.0496,ES:-.2879,NES:-.8054,size:30,leadingEdge:"ACO1,SUCLA2,DLST,PC",id:18},{pathway:"Sulfur metabolism",pval:.8372,padj:.8372,log2err:.0602,ES:.3342,NES:.7221,size:9,leadingEdge:"SQOR,BPNT1,IMPAD1",id:19}]}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p:M,i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function ha(e){let t,n,a,s;return t=new ue({props:{$$slots:{default:[Ca]},$$scope:{ctx:e}}}),a=new ue({props:{$$slots:{default:[$a]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function va(e){let t,n;return t=new me({props:{$$slots:{content:[ha],default:[ba]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function ya(e){let t,n,a,s,p,c,m,g,u,h,v,y,A,S,P,L,E,w,T,U,N,D,R,B,I,H,G,F,V,z,O,q,k,K,j,X,Y,W,Q,J,ee,te,ne,ae,se,re,ie,le,oe,pe,ce;return a=new Yn({props:{title:"Oxidative phosphorylation",class:"pipen-report-table-image-descr"}}),p=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Plasmablasts/Colitis_vs_Control/fgsea_Oxidative-phosphorylation.png",class:"pipen-report-table-image",width:1e3,height:800}}),g=new Yn({props:{title:"Amino sugar and nucleotide sugar metabolism",class:"pipen-report-table-image-descr"}}),h=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Plasmablasts/Colitis_vs_Control/fgsea_Amino-sugar-and-nucleotide-sugar-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),A=new Yn({props:{title:"Inositol phosphate metabolism",class:"pipen-report-table-image-descr"}}),P=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Plasmablasts/Colitis_vs_Control/fgsea_Inositol-phosphate-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),w=new Yn({props:{title:"Ascorbate and aldarate metabolism",class:"pipen-report-table-image-descr"}}),U=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Plasmablasts/Colitis_vs_Control/fgsea_Ascorbate-and-aldarate-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),R=new Yn({props:{title:"Starch and sucrose metabolism",class:"pipen-report-table-image-descr"}}),I=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Plasmablasts/Colitis_vs_Control/fgsea_Starch-and-sucrose-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),F=new Yn({props:{title:"Fatty acid degradation",class:"pipen-report-table-image-descr"}}),z=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Plasmablasts/Colitis_vs_Control/fgsea_Fatty-acid-degradation.png",class:"pipen-report-table-image",width:1e3,height:800}}),k=new Yn({props:{title:"Pentose and glucuronate interconversions",class:"pipen-report-table-image-descr"}}),j=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Plasmablasts/Colitis_vs_Control/fgsea_Pentose-and-glucuronate-interconversions.png",class:"pipen-report-table-image",width:1e3,height:800}}),W=new Yn({props:{title:"Galactose metabolism",class:"pipen-report-table-image-descr"}}),J=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Plasmablasts/Colitis_vs_Control/fgsea_Galactose-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new Yn({props:{title:"Fatty acid elongation",class:"pipen-report-table-image-descr"}}),se=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Plasmablasts/Colitis_vs_Control/fgsea_Fatty-acid-elongation.png",class:"pipen-report-table-image",width:1e3,height:800}}),le=new Yn({props:{title:"Butanoate metabolism",class:"pipen-report-table-image-descr"}}),pe=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Plasmablasts/Colitis_vs_Control/fgsea_Butanoate-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=r("div"),n=r("div"),f(a.$$.fragment),s=$(),f(p.$$.fragment),c=$(),m=r("div"),f(g.$$.fragment),u=$(),f(h.$$.fragment),v=$(),y=r("div"),f(A.$$.fragment),S=$(),f(P.$$.fragment),L=$(),E=r("div"),f(w.$$.fragment),T=$(),f(U.$$.fragment),N=$(),D=r("div"),f(R.$$.fragment),B=$(),f(I.$$.fragment),H=$(),G=r("div"),f(F.$$.fragment),V=$(),f(z.$$.fragment),O=$(),q=r("div"),f(k.$$.fragment),K=$(),f(j.$$.fragment),X=$(),Y=r("div"),f(W.$$.fragment),Q=$(),f(J.$$.fragment),ee=$(),te=r("div"),f(ne.$$.fragment),ae=$(),f(se.$$.fragment),re=$(),ie=r("div"),f(le.$$.fragment),oe=$(),f(pe.$$.fragment),i(t,"class","pipen-report-table-of-images"),Z(t,"grid-template-columns","repeat(2, auto)")},m(e,r){l(e,t,r),x(t,n),b(a,n,null),x(n,s),b(p,n,null),x(t,c),x(t,m),b(g,m,null),x(m,u),b(h,m,null),x(t,v),x(t,y),b(A,y,null),x(y,S),b(P,y,null),x(t,L),x(t,E),b(w,E,null),x(E,T),b(U,E,null),x(t,N),x(t,D),b(R,D,null),x(D,B),b(I,D,null),x(t,H),x(t,G),b(F,G,null),x(G,V),b(z,G,null),x(t,O),x(t,q),b(k,q,null),x(q,K),b(j,q,null),x(t,X),x(t,Y),b(W,Y,null),x(Y,Q),b(J,Y,null),x(t,ee),x(t,te),b(ne,te,null),x(te,ae),b(se,te,null),x(t,re),x(t,ie),b(le,ie,null),x(ie,oe),b(pe,ie,null),ce=!0},p:M,i(e){ce||(o(a.$$.fragment,e),o(p.$$.fragment,e),o(g.$$.fragment,e),o(h.$$.fragment,e),o(A.$$.fragment,e),o(P.$$.fragment,e),o(w.$$.fragment,e),o(U.$$.fragment,e),o(R.$$.fragment,e),o(I.$$.fragment,e),o(F.$$.fragment,e),o(z.$$.fragment,e),o(k.$$.fragment,e),o(j.$$.fragment,e),o(W.$$.fragment,e),o(J.$$.fragment,e),o(ne.$$.fragment,e),o(se.$$.fragment,e),o(le.$$.fragment,e),o(pe.$$.fragment,e),ce=!0)},o(e){d(a.$$.fragment,e),d(p.$$.fragment,e),d(g.$$.fragment,e),d(h.$$.fragment,e),d(A.$$.fragment,e),d(P.$$.fragment,e),d(w.$$.fragment,e),d(U.$$.fragment,e),d(R.$$.fragment,e),d(I.$$.fragment,e),d(F.$$.fragment,e),d(z.$$.fragment,e),d(k.$$.fragment,e),d(j.$$.fragment,e),d(W.$$.fragment,e),d(J.$$.fragment,e),d(ne.$$.fragment,e),d(se.$$.fragment,e),d(le.$$.fragment,e),d(pe.$$.fragment,e),ce=!1},d(e){e&&_(t),C(a),C(p),C(g),C(h),C(A),C(P),C(w),C(U),C(R),C(I),C(F),C(z),C(k),C(j),C(W),C(J),C(ne),C(se),C(le),C(pe)}}}function Aa(e){let t,n,a,s;return t=new Te({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[va]},$$scope:{ctx:e}}}),a=new Te({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[ya]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function xa(e){let t,n,a,s;return t=new ge({props:{label:"Plot"}}),a=new ge({props:{label:"Table"}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p:M,i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function Sa(e){let t,n;return t=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells/Colitis_vs_Control/gsea_table.png",width:1e3,height:600}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p:M,i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function Pa(e){let t,n;return t=new kn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells/Colitis_vs_Control/fgsea.txt",data:[{pathway:"Oxidative phosphorylation",pval:0,padj:0,log2err:.7749,ES:.6159,NES:1.8845,size:120,leadingEdge:"ATP5ME,NDUFS5,NDUFA6,UQCRFS1,UQCR10,ATP6V0B,COX7B,COX7A2,UQCRQ,NDUFB4,NDUFA12,COX5A,ATP5MC3,ATP5PD,NDUFC2,NDUFA2,PPA1,COX6C,NDUFA7,NDUFA1,ATP5F1E,NDUFB11,NDUFB8,NDUFC1,ATP5MF,COX6A1,ATP5F1C,NDUFB9,CYC1,UQCR11,COX8A,ATP6V0D1,TCIRG1,NDUFB3,SDHB,NDUFB1,NDUFAB1,ATP6V0C,NDUFS2,NDUFA9,UQCRH,NDUFA8,NDUFA3,ATP6V1A,ATP5MG,UQCRHL,COX10,COX5B,NDUFB5,COX7C,COX7A2L,ATP5F1D,NDUFA13,ATP5PF,NDUFA11,SDHD,ATP6V1F,ATP5F1A,NDUFB6,COX6B1,LHPP,NDUFS4,NDUFS7,ATP6V1C2,NDUFS8,NDUFB10,NDUFB2,COX17,ATP6V1G1",id:0},{pathway:"Ascorbate and aldarate metabolism",pval:0,padj:0,log2err:.6901,ES:-.787,NES:-2.613,size:27,leadingEdge:"UGDH,ALDH9A1",id:1},{pathway:"Amino sugar and nucleotide sugar metabolism",pval:.0516,padj:.2719,log2err:.3218,ES:-.3537,NES:-1.2875,size:48,leadingEdge:"FPGT,AMDHD2,GNPDA2,GNPNAT1,CYB5R1,UGDH,HEXB,CHIT1,GNE,UGP2",id:2},{pathway:"Sulfur metabolism",pval:.0544,padj:.2719,log2err:.2221,ES:.6876,NES:1.4017,size:9,leadingEdge:"SQOR,ETHE1,TST,MPST,BPNT1,IMPAD1",id:3},{pathway:"Starch and sucrose metabolism",pval:.2742,padj:.8769,log2err:.245,ES:-.3193,NES:-1.1142,size:35,leadingEdge:"PYGB,AMY2B,GAA,GBE1,AGL,UGP2,GANC,GPI",id:4},{pathway:"Nitrogen metabolism",pval:.3587,padj:.8769,log2err:.1723,ES:-.3554,NES:-1.0521,size:17,leadingEdge:"CA9,CA8,CA13,CA5A,CA12,CA7,CA6,CA4,CA14,CA1,CA3,CA2,GLUD2,CA5B,GLUL,GLUD1,CPS1",id:5},{pathway:"Pentose and glucuronate interconversions",pval:.391,padj:.8769,log2err:.1958,ES:-.2906,NES:-1.0196,size:34,leadingEdge:"UGDH,UGP2",id:6},{pathway:"Glyoxylate and dicarboxylate metabolism",pval:.3965,padj:.8769,log2err:.0641,ES:.4066,NES:1.0511,size:28,leadingEdge:"GLDC,GRHPR,SHMT2,GLYCTK,MDH2,PGP,DLD,ACO2,PCCB,ACAT1,AFMID,MCEE,AMT",id:7},{pathway:"Citrate cycle (TCA cycle)",pval:.3998,padj:.8769,log2err:.0629,ES:.4067,NES:1.0632,size:30,leadingEdge:"IDH2,IDH3G,SDHB,SUCLG1,MDH2,IDH3A,PDHB,SDHD,DLD,ACO2,PDHA1,IDH1,SDHA",id:8},{pathway:"Glycolysis / Gluconeogenesis",pval:.4384,padj:.8769,log2err:.0545,ES:.348,NES:1.0199,size:68,leadingEdge:"GALM,ENO3,GAPDH,LDHA,HK1,ADPGK,G6PC3,AKR1A1,TPI1,HKDC1,PDHB,PFKP,DLD,PGM2,PDHA1,PGM1,HK2,PGK1",id:9},{pathway:"Inositol phosphate metabolism",pval:.5536,padj:.9235,log2err:.2572,ES:-.2498,NES:-.9755,size:73,leadingEdge:"ISYNA1,MTM1,PLCH1,PIP4K2C,INPP5B,INPP5F,ALDH6A1,PLCB3,PLCH2,PLCB2,PI4K2A,OCRL,SYNJ1,MTMR8,PIK3CB,PIK3CG,INPP4B,IPMK,PIP5KL1,INPP1,MTMR4,PIP4K2B",id:10},{pathway:"Pyruvate metabolism",pval:.5971,padj:.9235,log2err:.0428,ES:.346,NES:.9477,size:39,leadingEdge:"ACYP2,GRHPR,LDHA,MDH2,HAGH,PDHB,DLD,ACAT1,ME2,PDHA1",id:11},{pathway:"Fatty acid biosynthesis",pval:.6003,padj:.9235,log2err:.0488,ES:.4205,NES:.9311,size:13,leadingEdge:"FASN,MCAT,ACSL4,OXSM,ACSL5,ACACB",id:12},{pathway:"Galactose metabolism",pval:.6636,padj:.948,log2err:.0392,ES:.3385,NES:.8924,size:31,leadingEdge:"GALM,HK1,GALK1,G6PC3,HKDC1,PFKP,PGM2,B4GALT1,PGM1,HK2,B4GALT2,PFKM",id:13},{pathway:"Pentose phosphate pathway",pval:.73,padj:.9733,log2err:.035,ES:.3238,NES:.8465,size:30,leadingEdge:"TKT,GLYCTK,DERA,PGLS,PFKP,TALDO1,PGM2,PGM1,H6PD,RPE,TKTL1,RBKS,PFKM",id:14},{pathway:"Fatty acid elongation",pval:.898,padj:.9783,log2err:.0246,ES:.2688,NES:.7027,size:30,leadingEdge:"ACOT7,ELOVL5,ABHD17A,ELOVL1,HACD4,ELOVL6,HADHB,ABHD17C,ACOT1,PPT2,ACOT2,HADH",id:15},{pathway:"Fatty acid degradation",pval:.9238,padj:.9783,log2err:.1396,ES:-.2191,NES:-.7836,size:44,leadingEdge:"ACADSB,ACSL3,ALDH9A1,ECI2,ACOX1",id:16},{pathway:"Butanoate metabolism",pval:.9408,padj:.9783,log2err:.1119,ES:-.2205,NES:-.7385,size:28,leadingEdge:"ACSM3,BDH2,AACS,L2HGDH,HMGCL,ACAT2,HADHA",id:17},{pathway:"Propanoate metabolism",pval:.9735,padj:.9783,log2err:.0196,ES:.229,NES:.6069,size:32,leadingEdge:"LDHA,ACADM,SUCLG1,ABAT,DLD,PCCB,ACAT1,MCEE,ECHDC1,ACACB,BCKDHA",id:18},{pathway:"Fructose and mannose metabolism",pval:.9783,padj:.9783,log2err:.0187,ES:.223,NES:.5945,size:33,leadingEdge:"PFKFB3,HK1,TSTA3,TPI1,HKDC1,TKFC,PFKP,GMDS,TIGAR,HK2",id:19}]}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p:M,i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function Ma(e){let t,n,a,s;return t=new ue({props:{$$slots:{default:[Sa]},$$scope:{ctx:e}}}),a=new ue({props:{$$slots:{default:[Pa]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function La(e){let t,n;return t=new me({props:{$$slots:{content:[Ma],default:[xa]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function Ea(e){let t,n,a,s,p,c,m,g,u,h,v,y,A,S,P,L,E,w,T,U,N,D,R,B,I,H,G,F,V,z,O,q,k,K,j,X,Y,W,Q,J,ee,te,ne,ae,se,re,ie,le,oe,pe,ce;return a=new Yn({props:{title:"Oxidative phosphorylation",class:"pipen-report-table-image-descr"}}),p=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells/Colitis_vs_Control/fgsea_Oxidative-phosphorylation.png",class:"pipen-report-table-image",width:1e3,height:800}}),g=new Yn({props:{title:"Ascorbate and aldarate metabolism",class:"pipen-report-table-image-descr"}}),h=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells/Colitis_vs_Control/fgsea_Ascorbate-and-aldarate-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),A=new Yn({props:{title:"Amino sugar and nucleotide sugar metabolism",class:"pipen-report-table-image-descr"}}),P=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells/Colitis_vs_Control/fgsea_Amino-sugar-and-nucleotide-sugar-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),w=new Yn({props:{title:"Sulfur metabolism",class:"pipen-report-table-image-descr"}}),U=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells/Colitis_vs_Control/fgsea_Sulfur-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),R=new Yn({props:{title:"Starch and sucrose metabolism",class:"pipen-report-table-image-descr"}}),I=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells/Colitis_vs_Control/fgsea_Starch-and-sucrose-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),F=new Yn({props:{title:"Nitrogen metabolism",class:"pipen-report-table-image-descr"}}),z=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells/Colitis_vs_Control/fgsea_Nitrogen-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),k=new Yn({props:{title:"Pentose and glucuronate interconversions",class:"pipen-report-table-image-descr"}}),j=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells/Colitis_vs_Control/fgsea_Pentose-and-glucuronate-interconversions.png",class:"pipen-report-table-image",width:1e3,height:800}}),W=new Yn({props:{title:"Glyoxylate and dicarboxylate metabolism",class:"pipen-report-table-image-descr"}}),J=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells/Colitis_vs_Control/fgsea_Glyoxylate-and-dicarboxylate-metabolism.png",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new Yn({props:{title:"Citrate cycle (TCA cycle)",class:"pipen-report-table-image-descr"}}),se=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells/Colitis_vs_Control/fgsea_Citrate-cycle-TCA-cycle-.png",class:"pipen-report-table-image",width:1e3,height:800}}),le=new Yn({props:{title:"Glycolysis / Gluconeogenesis",class:"pipen-report-table-image-descr"}}),pe=new hn({props:{src:"../MetabolicFeaturesIntraSubset/samples.seurat.annotated.tessa.imputed.intra-subset-pathwayfeatures/Cluster_Regulatory-T-cells/Colitis_vs_Control/fgsea_Glycolysis-Gluconeogenesis.png",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=r("div"),n=r("div"),f(a.$$.fragment),s=$(),f(p.$$.fragment),c=$(),m=r("div"),f(g.$$.fragment),u=$(),f(h.$$.fragment),v=$(),y=r("div"),f(A.$$.fragment),S=$(),f(P.$$.fragment),L=$(),E=r("div"),f(w.$$.fragment),T=$(),f(U.$$.fragment),N=$(),D=r("div"),f(R.$$.fragment),B=$(),f(I.$$.fragment),H=$(),G=r("div"),f(F.$$.fragment),V=$(),f(z.$$.fragment),O=$(),q=r("div"),f(k.$$.fragment),K=$(),f(j.$$.fragment),X=$(),Y=r("div"),f(W.$$.fragment),Q=$(),f(J.$$.fragment),ee=$(),te=r("div"),f(ne.$$.fragment),ae=$(),f(se.$$.fragment),re=$(),ie=r("div"),f(le.$$.fragment),oe=$(),f(pe.$$.fragment),i(t,"class","pipen-report-table-of-images"),Z(t,"grid-template-columns","repeat(2, auto)")},m(e,r){l(e,t,r),x(t,n),b(a,n,null),x(n,s),b(p,n,null),x(t,c),x(t,m),b(g,m,null),x(m,u),b(h,m,null),x(t,v),x(t,y),b(A,y,null),x(y,S),b(P,y,null),x(t,L),x(t,E),b(w,E,null),x(E,T),b(U,E,null),x(t,N),x(t,D),b(R,D,null),x(D,B),b(I,D,null),x(t,H),x(t,G),b(F,G,null),x(G,V),b(z,G,null),x(t,O),x(t,q),b(k,q,null),x(q,K),b(j,q,null),x(t,X),x(t,Y),b(W,Y,null),x(Y,Q),b(J,Y,null),x(t,ee),x(t,te),b(ne,te,null),x(te,ae),b(se,te,null),x(t,re),x(t,ie),b(le,ie,null),x(ie,oe),b(pe,ie,null),ce=!0},p:M,i(e){ce||(o(a.$$.fragment,e),o(p.$$.fragment,e),o(g.$$.fragment,e),o(h.$$.fragment,e),o(A.$$.fragment,e),o(P.$$.fragment,e),o(w.$$.fragment,e),o(U.$$.fragment,e),o(R.$$.fragment,e),o(I.$$.fragment,e),o(F.$$.fragment,e),o(z.$$.fragment,e),o(k.$$.fragment,e),o(j.$$.fragment,e),o(W.$$.fragment,e),o(J.$$.fragment,e),o(ne.$$.fragment,e),o(se.$$.fragment,e),o(le.$$.fragment,e),o(pe.$$.fragment,e),ce=!0)},o(e){d(a.$$.fragment,e),d(p.$$.fragment,e),d(g.$$.fragment,e),d(h.$$.fragment,e),d(A.$$.fragment,e),d(P.$$.fragment,e),d(w.$$.fragment,e),d(U.$$.fragment,e),d(R.$$.fragment,e),d(I.$$.fragment,e),d(F.$$.fragment,e),d(z.$$.fragment,e),d(k.$$.fragment,e),d(j.$$.fragment,e),d(W.$$.fragment,e),d(J.$$.fragment,e),d(ne.$$.fragment,e),d(se.$$.fragment,e),d(le.$$.fragment,e),d(pe.$$.fragment,e),ce=!1},d(e){e&&_(t),C(a),C(p),C(g),C(h),C(A),C(P),C(w),C(U),C(R),C(I),C(F),C(z),C(k),C(j),C(W),C(J),C(ne),C(se),C(le),C(pe)}}}function wa(e){let t,n,a,s;return t=new Te({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[La]},$$scope:{ctx:e}}}),a=new Te({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[Ea]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=$(),f(a.$$.fragment)},m(e,r){b(t,e,r),l(e,n,r),b(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),C(t,e),C(a,e)}}}function Ta(e){let t,n,a,s,p,c,m,g,u,h,v,y,A,x,S,P,M,L,E,w,T,U,N,D,R,B,I,H,G,F,V,z,O,q,k,K,j,X,Y,W;return m=new we({props:{align:"start",$$slots:{default:[na]},$$scope:{ctx:e}}}),S=new we({props:{align:"start",$$slots:{default:[pa]},$$scope:{ctx:e}}}),N=new we({props:{align:"start",$$slots:{default:[fa]},$$scope:{ctx:e}}}),V=new we({props:{align:"start",$$slots:{default:[Aa]},$$scope:{ctx:e}}}),Y=new we({props:{align:"start",$$slots:{default:[wa]},$$scope:{ctx:e}}}),{c(){t=r("h1"),t.textContent="Cluster_Cycling T cells",n=r("a"),n.innerHTML="",a=$(),s=r("h2"),s.textContent="Colitis_vs_Control",p=r("a"),p.innerHTML="",c=$(),f(m.$$.fragment),g=$(),u=r("h1"),u.textContent="Cluster_Follicular helper T cells",h=r("a"),h.innerHTML="",v=$(),y=r("h2"),y.textContent="Colitis_vs_Control",A=r("a"),A.innerHTML="",x=$(),f(S.$$.fragment),P=$(),M=r("h1"),M.textContent="Cluster_gamma-delta T cells",L=r("a"),L.innerHTML="",E=$(),w=r("h2"),w.textContent="Colitis_vs_Control",T=r("a"),T.innerHTML="",U=$(),f(N.$$.fragment),D=$(),R=r("h1"),R.textContent="Cluster_Plasmablasts",B=r("a"),B.innerHTML="",I=$(),H=r("h2"),H.textContent="Colitis_vs_Control",G=r("a"),G.innerHTML="",F=$(),f(V.$$.fragment),z=$(),O=r("h1"),O.textContent="Cluster_Regulatory T cells",q=r("a"),q.innerHTML="",k=$(),K=r("h2"),K.textContent="Colitis_vs_Control",j=r("a"),j.innerHTML="",X=$(),f(Y.$$.fragment),i(n,"id","prt-h1-0-cluster-cycling-t-cells"),i(n,"class","pipen-report-toc-anchor"),i(p,"id","prt-h2-0-colitis-vs-control"),i(p,"class","pipen-report-toc-anchor"),i(h,"id","prt-h1-1-cluster-follicular-helper-t-cells"),i(h,"class","pipen-report-toc-anchor"),i(A,"id","prt-h2-1-colitis-vs-control"),i(A,"class","pipen-report-toc-anchor"),i(L,"id","prt-h1-2-cluster-gamma-delta-t-cells"),i(L,"class","pipen-report-toc-anchor"),i(T,"id","prt-h2-2-colitis-vs-control"),i(T,"class","pipen-report-toc-anchor"),i(B,"id","prt-h1-3-cluster-plasmablasts"),i(B,"class","pipen-report-toc-anchor"),i(G,"id","prt-h2-3-colitis-vs-control"),i(G,"class","pipen-report-toc-anchor"),i(q,"id","prt-h1-4-cluster-regulatory-t-cells"),i(q,"class","pipen-report-toc-anchor"),i(j,"id","prt-h2-4-colitis-vs-control"),i(j,"class","pipen-report-toc-anchor")},m(e,r){l(e,t,r),l(e,n,r),l(e,a,r),l(e,s,r),l(e,p,r),l(e,c,r),b(m,e,r),l(e,g,r),l(e,u,r),l(e,h,r),l(e,v,r),l(e,y,r),l(e,A,r),l(e,x,r),b(S,e,r),l(e,P,r),l(e,M,r),l(e,L,r),l(e,E,r),l(e,w,r),l(e,T,r),l(e,U,r),b(N,e,r),l(e,D,r),l(e,R,r),l(e,B,r),l(e,I,r),l(e,H,r),l(e,G,r),l(e,F,r),b(V,e,r),l(e,z,r),l(e,O,r),l(e,q,r),l(e,k,r),l(e,K,r),l(e,j,r),l(e,X,r),b(Y,e,r),W=!0},p(e,[t]){const n={};1&t&&(n.$$scope={dirty:t,ctx:e}),m.$set(n);const a={};1&t&&(a.$$scope={dirty:t,ctx:e}),S.$set(a);const s={};1&t&&(s.$$scope={dirty:t,ctx:e}),N.$set(s);const r={};1&t&&(r.$$scope={dirty:t,ctx:e}),V.$set(r);const i={};1&t&&(i.$$scope={dirty:t,ctx:e}),Y.$set(i)},i(e){W||(o(m.$$.fragment,e),o(S.$$.fragment,e),o(N.$$.fragment,e),o(V.$$.fragment,e),o(Y.$$.fragment,e),W=!0)},o(e){d(m.$$.fragment,e),d(S.$$.fragment,e),d(N.$$.fragment,e),d(V.$$.fragment,e),d(Y.$$.fragment,e),W=!1},d(e){e&&(_(t),_(n),_(a),_(s),_(p),_(c),_(g),_(u),_(h),_(v),_(y),_(A),_(x),_(P),_(M),_(L),_(E),_(w),_(T),_(U),_(D),_(R),_(B),_(I),_(H),_(G),_(F),_(z),_(O),_(q),_(k),_(K),_(j),_(X)),C(m,e),C(S,e),C(N,e),C(V,e),C(Y,e)}}}class Ua extends t{constructor(e){super(),n(this,e,null,Ta,a,{})}}var Na=[{slug:"prt-h1-0-cluster-cycling-t-cells",text:"Cluster_Cycling T cells",children:[{slug:"prt-h2-0-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:0}],page:0},{slug:"prt-h1-1-cluster-follicular-helper-t-cells",text:"Cluster_Follicular helper T cells",children:[{slug:"prt-h2-1-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:0}],page:0},{slug:"prt-h1-2-cluster-gamma-delta-t-cells",text:"Cluster_gamma-delta T cells",children:[{slug:"prt-h2-2-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:0}],page:0},{slug:"prt-h1-3-cluster-plasmablasts",text:"Cluster_Plasmablasts",children:[{slug:"prt-h2-3-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:0}],page:0},{slug:"prt-h1-4-cluster-regulatory-t-cells",text:"Cluster_Regulatory T cells",children:[{slug:"prt-h2-4-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:0}],page:0},{slug:"prt-h1-5-cluster-regulatory-t-cells-1",text:"Cluster_Regulatory T cells.1",children:[{slug:"prt-h2-5-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:1}],page:1},{slug:"prt-h1-6-cluster-tcm-naive-helper-t-cells",text:"Cluster_Tcm/Naive helper T cells",children:[{slug:"prt-h2-6-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:1}],page:1},{slug:"prt-h1-7-cluster-tem-trm-cytotoxic-t-cells",text:"Cluster_Tem/Trm cytotoxic T cells",children:[{slug:"prt-h2-7-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:1}],page:1},{slug:"prt-h1-8-cluster-tem-trm-cytotoxic-t-cells-1",text:"Cluster_Tem/Trm cytotoxic T cells.1",children:[{slug:"prt-h2-8-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:1}],page:1},{slug:"prt-h1-9-cluster-trm-cytotoxic-t-cells",text:"Cluster_Trm cytotoxic T cells",children:[{slug:"prt-h2-9-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:1}],page:1},{slug:"prt-h1-10-cluster-type-1-helper-t-cells",text:"Cluster_Type 1 helper T cells",children:[{slug:"prt-h2-10-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:2}],page:2},{slug:"prt-h1-11-cluster-type-1-helper-t-cells-1",text:"Cluster_Type 1 helper T cells.1",children:[{slug:"prt-h2-11-colitis-vs-control",text:"Colitis_vs_Control",children:[],page:2}],page:2}],Da={pipeline:{name:"Immunopipe",desc:"Immunopipe (v1.4.2): Integrative analysis for scRNA-seq and scTCR-seq data"},versions:'\n    <a href="https://github.com/pwwang/pipen" target="_blank">\n        pipen v0.15.2\n    </a> and\n    <a href="https://github.com/pwwang/pipen-report" target="_blank">\n        pipen-report v0.20.1\n    </a>\n',entries:[{name:"SampleInfo",desc:"List sample information and perform statistics",npages:1,report_toc:!0,order:0,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n[1] ggrepel_0.9.5    ggprism_1.0.5    ggplot2_3.5.1    tidyr_1.3.1     \n[5] dplyr_1.1.4      tidyselect_1.2.1 jsonlite_1.8.8   rlang_1.1.4     \n[9] logger_0.3.0    \n\nloaded via a namespace (and not attached):\n [1] vctrs_0.6.5      cli_3.6.3        purrr_1.0.2      generics_0.1.3  \n [5] labeling_0.4.3   glue_1.7.0       colorspace_2.1-1 scales_1.3.0    \n [9] fansi_1.0.6      grid_4.3.3       munsell_0.5.1    tibble_3.2.1    \n[13] lifecycle_1.0.4  compiler_4.3.3   Rcpp_1.0.13      pkgconfig_2.0.3 \n[17] farver_2.1.2     digest_0.6.36    R6_2.5.1         utf8_1.2.4      \n[21] pillar_1.9.0     magrittr_2.0.3   withr_3.0.1      gtable_0.3.5    \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 1.73\nuser 10.62\nsys 0.10\npercent 617%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"SeuratPreparing",desc:"Load, prepare and apply QC to data, using `Seurat`",npages:1,report_toc:!0,order:2,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats4    stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] scDblFinder_1.16.0          SingleCellExperiment_1.24.0\n [3] SummarizedExperiment_1.32.0 Biobase_2.62.0             \n [5] GenomicRanges_1.54.1        GenomeInfoDb_1.38.1        \n [7] IRanges_2.36.0              S4Vectors_0.40.2           \n [9] BiocGenerics_0.48.1         MatrixGenerics_1.14.0      \n[11] matrixStats_1.1.0           dplyr_1.1.4                \n[13] ggplot2_3.5.1               bracer_1.2.1               \n[15] future_1.34.0               Seurat_5.1.0               \n[17] SeuratObject_5.0.2          sp_2.1-4                   \n[19] digest_0.6.36               jsonlite_1.8.8             \n[21] rlang_1.1.4                 logger_0.3.0               \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22          splines_4.3.3            \n  [3] later_1.3.2               BiocIO_1.12.0            \n  [5] bitops_1.0-8              tibble_3.2.1             \n  [7] R.oo_1.26.0               polyclip_1.10-7          \n  [9] XML_3.99-0.17             fastDummies_1.7.3        \n [11] lifecycle_1.0.4           edgeR_4.0.16             \n [13] globals_0.16.3            lattice_0.22-6           \n [15] MASS_7.3-60.0.1           magrittr_2.0.3           \n [17] limma_3.58.1              plotly_4.10.4            \n [19] yaml_2.3.10               metapod_1.10.0           \n [21] httpuv_1.6.15             sctransform_0.4.1        \n [23] spam_2.10-0               spatstat.sparse_3.1-0    \n [25] reticulate_1.38.0         cowplot_1.1.3            \n [27] pbapply_1.7-2             RColorBrewer_1.1-3       \n [29] abind_1.4-5               zlibbioc_1.48.0          \n [31] Rtsne_0.17                purrr_1.0.2              \n [33] R.utils_2.12.3            RCurl_1.98-1.16          \n [35] GenomeInfoDbData_1.2.11   ggrepel_0.9.5            \n [37] irlba_2.3.5.1             listenv_0.9.1            \n [39] spatstat.utils_3.0-5      goftest_1.2-3            \n [41] RSpectra_0.16-1           dqrng_0.3.2              \n [43] spatstat.random_3.2-3     fitdistrplus_1.1-11      \n [45] parallelly_1.38.0         DelayedMatrixStats_1.24.0\n [47] leiden_0.4.3.1            codetools_0.2-20         \n [49] DelayedArray_0.28.0       scuttle_1.12.0           \n [51] tidyselect_1.2.1          farver_2.1.2             \n [53] viridis_0.6.5             ScaledMatrix_1.10.0      \n [55] spatstat.explore_3.2-6    GenomicAlignments_1.38.0 \n [57] BiocNeighbors_1.20.0      progressr_0.14.0         \n [59] scater_1.30.1             ggridges_0.5.6           \n [61] survival_3.7-0            systemfonts_1.1.0        \n [63] tools_4.3.3               ragg_1.3.2               \n [65] ica_1.0-3                 Rcpp_1.0.13              \n [67] glue_1.7.0                gridExtra_2.3            \n [69] SparseArray_1.2.2         withr_3.0.1              \n [71] fastmap_1.2.0             bluster_1.12.0           \n [73] fansi_1.0.6               rsvd_1.0.5               \n [75] R6_2.5.1                  mime_0.12                \n [77] textshaping_0.4.0         colorspace_2.1-1         \n [79] scattermore_1.2           tensor_1.5               \n [81] spatstat.data_3.1-2       R.methodsS3_1.8.2        \n [83] RhpcBLASctl_0.23-42       utf8_1.2.4               \n [85] tidyr_1.3.1               generics_0.1.3           \n [87] data.table_1.15.4         rtracklayer_1.62.0       \n [89] httr_1.4.7                htmlwidgets_1.6.4        \n [91] S4Arrays_1.2.0            uwot_0.1.16              \n [93] pkgconfig_2.0.3           gtable_0.3.5             \n [95] lmtest_0.9-40             XVector_0.42.0           \n [97] htmltools_0.5.8.1         dotCall64_1.1-1          \n [99] scales_1.3.0              png_0.1-8                \n[101] harmony_1.2.0             scran_1.30.0             \n[103] reshape2_1.4.4            rjson_0.2.21             \n[105] nlme_3.1-165              zoo_1.8-12               \n[107] stringr_1.5.1             KernSmooth_2.23-24       \n[109] vipor_0.4.7               parallel_4.3.3           \n[111] miniUI_0.1.1.1            restfulr_0.0.15          \n[113] pillar_1.9.0              grid_4.3.3               \n[115] vctrs_0.6.5               RANN_2.6.1               \n[117] promises_1.3.0            BiocSingular_1.18.0      \n[119] beachmat_2.18.0           xtable_1.8-4             \n[121] cluster_2.1.6             beeswarm_0.4.0           \n[123] locfit_1.5-9.9            cli_3.6.3                \n[125] compiler_4.3.3            Rsamtools_2.18.0         \n[127] crayon_1.5.3              future.apply_1.11.2      \n[129] labeling_0.4.3            ggbeeswarm_0.7.2         \n[131] plyr_1.8.9                stringi_1.8.4            \n[133] viridisLite_0.4.2         deldir_2.0-4             \n[135] BiocParallel_1.36.0       munsell_0.5.1            \n[137] Biostrings_2.70.1         lazyeval_0.2.2           \n[139] spatstat.geom_3.2-9       Matrix_1.6-5             \n[141] RcppHNSW_0.6.0            patchwork_1.2.0          \n[143] sparseMatrixStats_1.14.0  statmod_1.5.0            \n[145] shiny_1.9.1               ROCR_1.0-11              \n[147] igraph_2.0.3              xgboost_2.1.1.1          \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 202.98\nuser 1582.98\nsys 7.06\npercent 783%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ClusterMarkersOfAllCells",desc:"Markers for clusters of all cells.",npages:3,report_toc:!0,order:4,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] UpSetR_1.4.0        ggVennDiagram_1.2.3 tidyseurat_0.8.0   \n [4] ttservice_0.4.1     future_1.34.0       ggrepel_0.9.5      \n [7] ggprism_1.0.5       ggplot2_3.5.1       enrichR_3.2        \n[10] Seurat_5.1.0        SeuratObject_5.0.2  sp_2.1-4           \n[13] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[16] tidyselect_1.2.1    digest_0.6.36       jsonlite_1.8.8     \n[19] rlang_1.1.4         logger_0.3.0       \n\nloaded via a namespace (and not attached):\n  [1] RColorBrewer_1.1-3     magrittr_2.0.3         spatstat.utils_3.0-5  \n  [4] farver_2.1.2           vctrs_0.6.5            ROCR_1.0-11           \n  [7] spatstat.explore_3.2-6 htmltools_0.5.8.1      curl_5.2.1            \n [10] sctransform_0.4.1      parallelly_1.38.0      KernSmooth_2.23-24    \n [13] htmlwidgets_1.6.4      ica_1.0-3              plyr_1.8.9            \n [16] plotly_4.10.4          zoo_1.8-12             igraph_2.0.3          \n [19] mime_0.12              lifecycle_1.0.4        pkgconfig_2.0.3       \n [22] Matrix_1.6-5           R6_2.5.1               fastmap_1.2.0         \n [25] fitdistrplus_1.1-11    shiny_1.9.1            colorspace_2.1-1      \n [28] patchwork_1.2.0        tensor_1.5             RSpectra_0.16-1       \n [31] irlba_2.3.5.1          WriteXLS_6.7.0         labeling_0.4.3        \n [34] progressr_0.14.0       fansi_1.0.6            spatstat.sparse_3.1-0 \n [37] httr_1.4.7             polyclip_1.10-7        abind_1.4-5           \n [40] compiler_4.3.3         withr_3.0.1            fastDummies_1.7.3     \n [43] MASS_7.3-60.0.1        rjson_0.2.21           tools_4.3.3           \n [46] lmtest_0.9-40          httpuv_1.6.15          future.apply_1.11.2   \n [49] goftest_1.2-3          glue_1.7.0             nlme_3.1-165          \n [52] promises_1.3.0         grid_4.3.3             Rtsne_0.17            \n [55] cluster_2.1.6          reshape2_1.4.4         generics_0.1.3        \n [58] gtable_0.3.5           spatstat.data_3.1-2    data.table_1.15.4     \n [61] utf8_1.2.4             spatstat.geom_3.2-9    RcppAnnoy_0.0.22      \n [64] RANN_2.6.1             pillar_1.9.0           stringr_1.5.1         \n [67] spam_2.10-0            RcppHNSW_0.6.0         limma_3.58.1          \n [70] later_1.3.2            splines_4.3.3          lattice_0.22-6        \n [73] survival_3.7-0         deldir_2.0-4           miniUI_0.1.1.1        \n [76] pbapply_1.7-2          gridExtra_2.3          scattermore_1.2       \n [79] statmod_1.5.0          matrixStats_1.1.0      stringi_1.8.4         \n [82] lazyeval_0.2.2         codetools_0.2-20       RVenn_1.1.0           \n [85] cli_3.6.3              uwot_0.1.16            xtable_1.8-4          \n [88] reticulate_1.38.0      munsell_0.5.1          Rcpp_1.0.13           \n [91] globals_0.16.3         spatstat.random_3.2-3  png_0.1-8             \n [94] parallel_4.3.3         ellipsis_0.3.2         presto_1.0.0          \n [97] dotCall64_1.1-1        listenv_0.9.1          viridisLite_0.4.2     \n[100] scales_1.3.0           ggridges_0.5.6         leiden_0.4.3.1        \n[103] purrr_1.0.2            cowplot_1.1.3         \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 216.89\nuser 139.48\nsys 7.23\npercent 67%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TopExpressingGenesOfAllCells",desc:"Top expressing genes for clusters of all cells.",npages:3,report_toc:!0,order:5,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ggprism_1.0.5      dplyr_1.1.4        enrichR_3.2        tibble_3.2.1      \n [5] Seurat_5.1.0       SeuratObject_5.0.2 sp_2.1-4           jsonlite_1.8.8    \n [9] rlang_1.1.4        logger_0.3.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.2-9   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] labeling_0.4.3         utf8_1.2.4             promises_1.3.0        \n [19] purrr_1.0.2            WriteXLS_6.7.0         goftest_1.2-3         \n [22] later_1.3.2            spatstat.utils_3.0-5   irlba_2.3.5.1         \n [25] parallel_4.3.3         cluster_2.1.6          R6_2.5.1              \n [28] ica_1.0-3              stringi_1.8.4          RColorBrewer_1.1-3    \n [31] spatstat.data_3.1-2    reticulate_1.38.0      parallelly_1.38.0     \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.13           \n [37] tensor_1.5             future.apply_1.11.2    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [43] splines_4.3.3          igraph_2.0.3           tidyselect_1.2.1      \n [46] abind_1.4-5            spatstat.random_3.2-3  codetools_0.2-20      \n [49] miniUI_0.1.1.1         spatstat.explore_3.2-6 curl_5.2.1            \n [52] listenv_0.9.1          lattice_0.22-6         plyr_1.8.9            \n [55] withr_3.0.1            shiny_1.9.1            ROCR_1.0-11           \n [58] Rtsne_0.17             future_1.34.0          fastDummies_1.7.3     \n [61] survival_3.7-0         polyclip_1.10-7        fitdistrplus_1.1-11   \n [64] pillar_1.9.0           KernSmooth_2.23-24     plotly_4.10.4         \n [67] generics_0.1.3         RcppHNSW_0.6.0         ggplot2_3.5.1         \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.7.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.15.4      RSpectra_0.16-1       \n [79] RANN_2.6.1             leiden_0.4.3.1         dotCall64_1.1-1       \n [82] cowplot_1.1.3          grid_4.3.3             tidyr_1.3.1           \n [85] colorspace_2.1-1       nlme_3.1-165           patchwork_1.2.0       \n [88] cli_3.6.3              spatstat.sparse_3.1-0  spam_2.10-0           \n [91] fansi_1.0.6            viridisLite_0.4.2      uwot_0.1.16           \n [94] gtable_0.3.5           digest_0.6.36          progressr_0.14.0      \n [97] ggrepel_0.9.5          farver_2.1.2           rjson_0.2.21          \n[100] htmlwidgets_1.6.4      htmltools_0.5.8.1      lifecycle_1.0.4       \n[103] httr_1.4.7             mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 52.61\nuser 38.55\nsys 2.30\npercent 77%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TCellSelection",desc:"Separate T and non-T cells and select T cells.",npages:1,report_toc:!0,order:6,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] glue_1.7.0         factoextra_1.0.7   patchwork_1.2.0    ggpubr_0.6.0      \n [5] ggrepel_0.9.5      ggprism_1.0.5      ggplot2_3.5.1      tibble_3.2.1      \n [9] tidyr_1.3.1        dplyr_1.1.4        Seurat_5.1.0       SeuratObject_5.0.2\n[13] sp_2.1-4           jsonlite_1.8.8     rlang_1.1.4        logger_0.3.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.2-9   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] backports_1.5.0        labeling_0.4.3         utf8_1.2.4            \n [19] promises_1.3.0         purrr_1.0.2            goftest_1.2-3         \n [22] later_1.3.2            spatstat.utils_3.0-5   broom_1.0.6           \n [25] irlba_2.3.5.1          parallel_4.3.3         cluster_2.1.6         \n [28] R6_2.5.1               ica_1.0-3              stringi_1.8.4         \n [31] RColorBrewer_1.1-3     spatstat.data_3.1-2    reticulate_1.38.0     \n [34] car_3.1-2              parallelly_1.38.0      lmtest_0.9-40         \n [37] scattermore_1.2        Rcpp_1.0.13            tensor_1.5            \n [40] future.apply_1.11.2    zoo_1.8-12             sctransform_0.4.1     \n [43] httpuv_1.6.15          Matrix_1.6-5           splines_4.3.3         \n [46] igraph_2.0.3           tidyselect_1.2.1       abind_1.4-5           \n [49] spatstat.random_3.2-3  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.2-6 listenv_0.9.1          lattice_0.22-6        \n [55] plyr_1.8.9             withr_3.0.1            shiny_1.9.1           \n [58] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n [61] fastDummies_1.7.3      survival_3.7-0         polyclip_1.10-7       \n [64] fitdistrplus_1.1-11    pillar_1.9.0           carData_3.0-5         \n [67] KernSmooth_2.23-24     plotly_4.10.4          generics_0.1.3        \n [70] RcppHNSW_0.6.0         munsell_0.5.1          scales_1.3.0          \n [73] globals_0.16.3         xtable_1.8-4           lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.15.4      RSpectra_0.16-1       \n [79] ggsignif_0.6.4         RANN_2.6.1             leiden_0.4.3.1        \n [82] dotCall64_1.1-1        cowplot_1.1.3          grid_4.3.3            \n [85] colorspace_2.1-1       nlme_3.1-165           cli_3.6.3             \n [88] spatstat.sparse_3.1-0  spam_2.10-0            fansi_1.0.6           \n [91] viridisLite_0.4.2      uwot_0.1.16            gtable_0.3.5          \n [94] rstatix_0.7.2          digest_0.6.36          progressr_0.14.0      \n [97] farver_2.1.2           htmlwidgets_1.6.4      htmltools_0.5.8.1     \n[100] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[103] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 38.03\nuser 45.75\nsys 0.83\npercent 122%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3435.798\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ClusterMarkers",desc:"Markers for clusters of all or selected T cells.",npages:2,report_toc:!0,order:10,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] UpSetR_1.4.0        ggVennDiagram_1.2.3 tidyseurat_0.8.0   \n [4] ttservice_0.4.1     future_1.34.0       ggrepel_0.9.5      \n [7] ggprism_1.0.5       ggplot2_3.5.1       enrichR_3.2        \n[10] Seurat_5.1.0        SeuratObject_5.0.2  sp_2.1-4           \n[13] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[16] tidyselect_1.2.1    digest_0.6.36       jsonlite_1.8.8     \n[19] rlang_1.1.4         logger_0.3.0       \n\nloaded via a namespace (and not attached):\n  [1] RColorBrewer_1.1-3     magrittr_2.0.3         spatstat.utils_3.0-5  \n  [4] farver_2.1.2           vctrs_0.6.5            ROCR_1.0-11           \n  [7] spatstat.explore_3.2-6 htmltools_0.5.8.1      curl_5.2.1            \n [10] sctransform_0.4.1      parallelly_1.38.0      KernSmooth_2.23-24    \n [13] htmlwidgets_1.6.4      ica_1.0-3              plyr_1.8.9            \n [16] plotly_4.10.4          zoo_1.8-12             igraph_2.0.3          \n [19] mime_0.12              lifecycle_1.0.4        pkgconfig_2.0.3       \n [22] Matrix_1.6-5           R6_2.5.1               fastmap_1.2.0         \n [25] fitdistrplus_1.1-11    shiny_1.9.1            colorspace_2.1-1      \n [28] patchwork_1.2.0        tensor_1.5             RSpectra_0.16-1       \n [31] irlba_2.3.5.1          WriteXLS_6.7.0         labeling_0.4.3        \n [34] progressr_0.14.0       fansi_1.0.6            spatstat.sparse_3.1-0 \n [37] httr_1.4.7             polyclip_1.10-7        abind_1.4-5           \n [40] compiler_4.3.3         withr_3.0.1            fastDummies_1.7.3     \n [43] MASS_7.3-60.0.1        rjson_0.2.21           tools_4.3.3           \n [46] lmtest_0.9-40          httpuv_1.6.15          future.apply_1.11.2   \n [49] goftest_1.2-3          glue_1.7.0             nlme_3.1-165          \n [52] promises_1.3.0         grid_4.3.3             Rtsne_0.17            \n [55] cluster_2.1.6          reshape2_1.4.4         generics_0.1.3        \n [58] gtable_0.3.5           spatstat.data_3.1-2    data.table_1.15.4     \n [61] utf8_1.2.4             spatstat.geom_3.2-9    RcppAnnoy_0.0.22      \n [64] RANN_2.6.1             pillar_1.9.0           stringr_1.5.1         \n [67] spam_2.10-0            RcppHNSW_0.6.0         limma_3.58.1          \n [70] later_1.3.2            splines_4.3.3          lattice_0.22-6        \n [73] survival_3.7-0         deldir_2.0-4           miniUI_0.1.1.1        \n [76] pbapply_1.7-2          gridExtra_2.3          scattermore_1.2       \n [79] statmod_1.5.0          matrixStats_1.1.0      stringi_1.8.4         \n [82] lazyeval_0.2.2         codetools_0.2-20       RVenn_1.1.0           \n [85] cli_3.6.3              uwot_0.1.16            xtable_1.8-4          \n [88] reticulate_1.38.0      munsell_0.5.1          Rcpp_1.0.13           \n [91] globals_0.16.3         spatstat.random_3.2-3  png_0.1-8             \n [94] parallel_4.3.3         ellipsis_0.3.2         presto_1.0.0          \n [97] dotCall64_1.1-1        listenv_0.9.1          viridisLite_0.4.2     \n[100] scales_1.3.0           ggridges_0.5.6         leiden_0.4.3.1        \n[103] purrr_1.0.2            cowplot_1.1.3         \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 136.36\nuser 90.28\nsys 4.48\npercent 69%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TopExpressingGenes",desc:"Top expressing genes for clusters of all or selected T cells.",npages:2,report_toc:!0,order:11,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ggprism_1.0.5      dplyr_1.1.4        enrichR_3.2        tibble_3.2.1      \n [5] Seurat_5.1.0       SeuratObject_5.0.2 sp_2.1-4           jsonlite_1.8.8    \n [9] rlang_1.1.4        logger_0.3.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.2-9   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] labeling_0.4.3         utf8_1.2.4             promises_1.3.0        \n [19] purrr_1.0.2            WriteXLS_6.7.0         goftest_1.2-3         \n [22] later_1.3.2            spatstat.utils_3.0-5   irlba_2.3.5.1         \n [25] parallel_4.3.3         cluster_2.1.6          R6_2.5.1              \n [28] ica_1.0-3              stringi_1.8.4          RColorBrewer_1.1-3    \n [31] spatstat.data_3.1-2    reticulate_1.38.0      parallelly_1.38.0     \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.13           \n [37] tensor_1.5             future.apply_1.11.2    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [43] splines_4.3.3          igraph_2.0.3           tidyselect_1.2.1      \n [46] abind_1.4-5            spatstat.random_3.2-3  codetools_0.2-20      \n [49] miniUI_0.1.1.1         spatstat.explore_3.2-6 curl_5.2.1            \n [52] listenv_0.9.1          lattice_0.22-6         plyr_1.8.9            \n [55] withr_3.0.1            shiny_1.9.1            ROCR_1.0-11           \n [58] Rtsne_0.17             future_1.34.0          fastDummies_1.7.3     \n [61] survival_3.7-0         polyclip_1.10-7        fitdistrplus_1.1-11   \n [64] pillar_1.9.0           KernSmooth_2.23-24     plotly_4.10.4         \n [67] generics_0.1.3         RcppHNSW_0.6.0         ggplot2_3.5.1         \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.7.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.15.4      RSpectra_0.16-1       \n [79] RANN_2.6.1             leiden_0.4.3.1         dotCall64_1.1-1       \n [82] cowplot_1.1.3          grid_4.3.3             tidyr_1.3.1           \n [85] colorspace_2.1-1       nlme_3.1-165           patchwork_1.2.0       \n [88] cli_3.6.3              spatstat.sparse_3.1-0  spam_2.10-0           \n [91] fansi_1.0.6            viridisLite_0.4.2      uwot_0.1.16           \n [94] gtable_0.3.5           digest_0.6.36          progressr_0.14.0      \n [97] ggrepel_0.9.5          farver_2.1.2           rjson_0.2.21          \n[100] htmlwidgets_1.6.4      htmltools_0.5.8.1      lifecycle_1.0.4       \n[103] httr_1.4.7             mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 31.30\nuser 27.76\nsys 1.57\npercent 93%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TCRClusterStats",desc:"Statistics of TCR clusters, generated by `TCRClustering`.",npages:1,report_toc:!0,order:13,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] grid      stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] ComplexHeatmap_2.18.0 ggprism_1.0.5         immunarch_0.9.1      \n [4] patchwork_1.2.0       data.table_1.15.4     dtplyr_1.3.1         \n [7] dplyr_1.1.4           tidyr_1.3.1           tibble_3.2.1         \n[10] ggplot2_3.5.1         jsonlite_1.8.8        rlang_1.1.4          \n[13] logger_0.3.0         \n\nloaded via a namespace (and not attached):\n  [1] gridExtra_2.3       phangorn_2.11.1     readxl_1.4.3       \n  [4] magrittr_2.0.3      clue_0.3-65         GetoptLong_1.0.5   \n  [7] matrixStats_1.1.0   compiler_4.3.3      flexmix_2.3-19     \n [10] png_0.1-8           vctrs_0.6.5         reshape2_1.4.4     \n [13] ggalluvial_0.12.5   quadprog_1.5-8      stringr_1.5.1      \n [16] crayon_1.5.3        pkgconfig_2.0.3     shape_1.4.6.1      \n [19] fastmap_1.2.0       backports_1.5.0     labeling_0.4.3     \n [22] ggraph_2.1.0        utf8_1.2.4          promises_1.3.0     \n [25] tzdb_0.4.0          UpSetR_1.4.0        purrr_1.0.2        \n [28] modeltools_0.2-23   ggseqlogo_0.2       later_1.3.2        \n [31] uuid_1.2-1          fpc_2.2-12          tweenr_2.0.3       \n [34] broom_1.0.6         parallel_4.3.3      prabclus_2.3-3     \n [37] cluster_2.1.6       R6_2.5.1            stringi_1.8.4      \n [40] RColorBrewer_1.1-3  rlist_0.4.6.2       car_3.1-2          \n [43] diptest_0.77-1      cellranger_1.1.0    Rcpp_1.0.13        \n [46] iterators_1.0.14    IRanges_2.36.0      readr_2.1.5        \n [49] httpuv_1.6.15       Matrix_1.6-5        nnet_7.3-19        \n [52] igraph_2.0.3        tidyselect_1.2.1    stringdist_0.9.12  \n [55] abind_1.4-5         viridis_0.6.5       doParallel_1.0.17  \n [58] codetools_0.2-20    lattice_0.22-6      plyr_1.8.9         \n [61] shiny_1.9.1         withr_3.0.1         polyclip_1.10-7    \n [64] circlize_0.4.16     mclust_6.1.1        kernlab_0.9-32     \n [67] pillar_1.9.0        ggpubr_0.6.0        carData_3.0-5      \n [70] foreach_1.5.2       stats4_4.3.3        generics_0.1.3     \n [73] S4Vectors_0.40.2    hms_1.1.3           munsell_0.5.1      \n [76] scales_1.3.0        xtable_1.8-4        class_7.3-22       \n [79] glue_1.7.0          pheatmap_1.0.12     tools_4.3.3        \n [82] robustbase_0.99-3   ggsignif_0.6.4      graphlayouts_1.1.0 \n [85] Cairo_1.6-2         fastmatch_1.1-4     tidygraph_1.3.0    \n [88] ape_5.8             colorspace_2.1-1    nlme_3.1-165       \n [91] ggforce_0.4.2       cli_3.6.3           fansi_1.0.6        \n [94] viridisLite_0.4.2   gtable_0.3.5        DEoptimR_1.1-3     \n [97] rstatix_0.7.2       digest_0.6.36       BiocGenerics_0.48.1\n[100] ggrepel_0.9.5       rjson_0.2.21        farver_2.1.2       \n[103] htmltools_0.5.8.1   lifecycle_1.0.4     factoextra_1.0.7   \n[106] GlobalOptions_0.1.2 mime_0.12           shinythemes_1.2.0  \n[109] MASS_7.3-60.0.1    \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 5.36\nuser 14.04\nsys 0.30\npercent 267%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TESSA",desc:"Tessa is a Bayesian model to integrate T cell receptor (TCR) sequence profiling with transcriptomes of T cells.",npages:1,report_toc:!0,order:14,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] igraph_2.0.3         Rtsne_0.17           LaplacesDemon_16.1.6\n [4] MASS_7.3-60.0.1      ggprism_1.0.5        Seurat_5.1.0        \n [7] SeuratObject_5.0.2   sp_2.1-4             tibble_3.2.1        \n[10] glue_1.7.0           immunarch_0.9.1      patchwork_1.2.0     \n[13] data.table_1.15.4    dtplyr_1.3.1         ggplot2_3.5.1       \n[16] tidyr_1.3.1          dplyr_1.1.4          jsonlite_1.8.8      \n[19] rlang_1.1.4          logger_0.3.0        \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22       shinythemes_1.2.0      splines_4.3.3         \n  [4] later_1.3.2            cellranger_1.1.0       polyclip_1.10-7       \n  [7] fastDummies_1.7.3      factoextra_1.0.7       lifecycle_1.0.4       \n [10] rstatix_0.7.2          doParallel_1.0.17      globals_0.16.3        \n [13] lattice_0.22-6         prabclus_2.3-3         backports_1.5.0       \n [16] magrittr_2.0.3         plotly_4.10.4          rlist_0.4.6.2         \n [19] httpuv_1.6.15          sctransform_0.4.1      spam_2.10-0           \n [22] flexmix_2.3-19         spatstat.sparse_3.1-0  reticulate_1.38.0     \n [25] cowplot_1.1.3          pbapply_1.7-2          RColorBrewer_1.1-3    \n [28] abind_1.4-5            quadprog_1.5-8         purrr_1.0.2           \n [31] ggraph_2.1.0           nnet_7.3-19            tweenr_2.0.3          \n [34] circlize_0.4.16        ggrepel_0.9.5          irlba_2.3.5.1         \n [37] listenv_0.9.1          spatstat.utils_3.0-5   pheatmap_1.0.12       \n [40] goftest_1.2-3          RSpectra_0.16-1        spatstat.random_3.2-3 \n [43] fitdistrplus_1.1-11    parallelly_1.38.0      leiden_0.4.3.1        \n [46] codetools_0.2-20       ggforce_0.4.2          tidyselect_1.2.1      \n [49] shape_1.4.6.1          ggseqlogo_0.2          farver_2.1.2          \n [52] viridis_0.6.5          matrixStats_1.1.0      stats4_4.3.3          \n [55] spatstat.explore_3.2-6 tidygraph_1.3.0        progressr_0.14.0      \n [58] ggridges_0.5.6         ggalluvial_0.12.5      survival_3.7-0        \n [61] iterators_1.0.14       foreach_1.5.2          tools_4.3.3           \n [64] stringdist_0.9.12      ica_1.0-3              Rcpp_1.0.13           \n [67] gridExtra_2.3          withr_3.0.1            fastmap_1.2.0         \n [70] fansi_1.0.6            digest_0.6.36          R6_2.5.1              \n [73] mime_0.12              colorspace_2.1-1       scattermore_1.2       \n [76] tensor_1.5             spatstat.data_3.1-2    diptest_0.77-1        \n [79] UpSetR_1.4.0           utf8_1.2.4             generics_0.1.3        \n [82] robustbase_0.99-3      class_7.3-22           graphlayouts_1.1.0    \n [85] httr_1.4.7             htmlwidgets_1.6.4      uwot_0.1.16           \n [88] pkgconfig_2.0.3        gtable_0.3.5           modeltools_0.2-23     \n [91] lmtest_0.9-40          htmltools_0.5.8.1      carData_3.0-5         \n [94] dotCall64_1.1-1        scales_1.3.0           png_0.1-8             \n [97] tzdb_0.4.0             reshape2_1.4.4         uuid_1.2-1            \n[100] nlme_3.1-165           zoo_1.8-12             GlobalOptions_0.1.2   \n[103] stringr_1.5.1          KernSmooth_2.23-24     parallel_4.3.3        \n[106] miniUI_0.1.1.1         pillar_1.9.0           grid_4.3.3            \n[109] vctrs_0.6.5            RANN_2.6.1             promises_1.3.0        \n[112] ggpubr_0.6.0           car_3.1-2              xtable_1.8-4          \n[115] cluster_2.1.6          readr_2.1.5            cli_3.6.3             \n[118] compiler_4.3.3         future.apply_1.11.2    ggsignif_0.6.4        \n[121] labeling_0.4.3         mclust_6.1.1           plyr_1.8.9            \n[124] stringi_1.8.4          deldir_2.0-4           viridisLite_0.4.2     \n[127] munsell_0.5.1          lazyeval_0.2.2         spatstat.geom_3.2-9   \n[130] Matrix_1.6-5           RcppHNSW_0.6.0         hms_1.1.3             \n[133] future_1.34.0          fpc_2.2-12             shiny_1.9.1           \n[136] kernlab_0.9-32         ROCR_1.0-11            broom_1.0.6           \n[139] phangorn_2.11.1        fastmatch_1.1-4        DEoptimR_1.1-3        \n[142] readxl_1.4.3           ape_5.8               \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 1774.00\nuser 8784.42\nsys 20.96\npercent 496%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"SeuratClusterStats",desc:"Statistics of the clustering.",npages:1,report_toc:!0,order:17,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] grid      stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] clustree_0.5.1        ggraph_2.1.0          ComplexHeatmap_2.18.0\n [4] circlize_0.4.16       tidyseurat_0.8.0      ttservice_0.4.1      \n [7] ggrepel_0.9.5         ggprism_1.0.5         tibble_3.2.1         \n[10] Seurat_5.1.0          SeuratObject_5.0.2    sp_2.1-4             \n[13] ggplot2_3.5.1         tidyr_1.3.1           dplyr_1.1.4          \n[16] tidyselect_1.2.1      jsonlite_1.8.8        rlang_1.1.4          \n[19] logger_0.3.0         \n\nloaded via a namespace (and not attached):\n  [1] RColorBrewer_1.1-3     shape_1.4.6.1          magrittr_2.0.3        \n  [4] ggbeeswarm_0.7.2       spatstat.utils_3.0-5   farver_2.1.2          \n  [7] GlobalOptions_0.1.2    vctrs_0.6.5            ROCR_1.0-11           \n [10] spatstat.explore_3.2-6 htmltools_0.5.8.1      sctransform_0.4.1     \n [13] parallelly_1.38.0      KernSmooth_2.23-24     htmlwidgets_1.6.4     \n [16] ica_1.0-3              plyr_1.8.9             plotly_4.10.4         \n [19] zoo_1.8-12             igraph_2.0.3           mime_0.12             \n [22] lifecycle_1.0.4        iterators_1.0.14       pkgconfig_2.0.3       \n [25] Matrix_1.6-5           R6_2.5.1               fastmap_1.2.0         \n [28] clue_0.3-65            fitdistrplus_1.1-11    future_1.34.0         \n [31] shiny_1.9.1            digest_0.6.36          colorspace_2.1-1      \n [34] patchwork_1.2.0        S4Vectors_0.40.2       tensor_1.5            \n [37] RSpectra_0.16-1        irlba_2.3.5.1          labeling_0.4.3        \n [40] progressr_0.14.0       fansi_1.0.6            spatstat.sparse_3.1-0 \n [43] httr_1.4.7             polyclip_1.10-7        abind_1.4-5           \n [46] compiler_4.3.3         withr_3.0.1            doParallel_1.0.17     \n [49] viridis_0.6.5          fastDummies_1.7.3      ggforce_0.4.2         \n [52] MASS_7.3-60.0.1        rjson_0.2.21           tools_4.3.3           \n [55] vipor_0.4.7            lmtest_0.9-40          beeswarm_0.4.0        \n [58] httpuv_1.6.15          future.apply_1.11.2    goftest_1.2-3         \n [61] glue_1.7.0             nlme_3.1-165           promises_1.3.0        \n [64] Rtsne_0.17             cluster_2.1.6          reshape2_1.4.4        \n [67] generics_0.1.3         gtable_0.3.5           spatstat.data_3.1-2   \n [70] data.table_1.15.4      tidygraph_1.3.0        utf8_1.2.4            \n [73] BiocGenerics_0.48.1    spatstat.geom_3.2-9    RcppAnnoy_0.0.22      \n [76] RANN_2.6.1             foreach_1.5.2          pillar_1.9.0          \n [79] stringr_1.5.1          spam_2.10-0            RcppHNSW_0.6.0        \n [82] later_1.3.2            splines_4.3.3          tweenr_2.0.3          \n [85] lattice_0.22-6         survival_3.7-0         deldir_2.0-4          \n [88] miniUI_0.1.1.1         pbapply_1.7-2          gridExtra_2.3         \n [91] IRanges_2.36.0         scattermore_1.2        stats4_4.3.3          \n [94] graphlayouts_1.1.0     matrixStats_1.1.0      stringi_1.8.4         \n [97] lazyeval_0.2.2         codetools_0.2-20       cli_3.6.3             \n[100] uwot_0.1.16            xtable_1.8-4           reticulate_1.38.0     \n[103] munsell_0.5.1          Rcpp_1.0.13            globals_0.16.3        \n[106] spatstat.random_3.2-3  png_0.1-8              ggrastr_1.0.2         \n[109] parallel_4.3.3         ellipsis_0.3.2         dotCall64_1.1-1       \n[112] listenv_0.9.1          viridisLite_0.4.2      scales_1.3.0          \n[115] ggridges_0.5.6         leiden_0.4.3.1         purrr_1.0.2           \n[118] crayon_1.5.3           GetoptLong_1.0.5       cowplot_1.1.3         \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 15.10\nuser 23.39\nsys 0.59\npercent 158%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"Immunarch",desc:"Exploration of Single-cell and Bulk T-cell/Antibody Immune Repertoires",npages:3,report_toc:!0,order:18,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] circlize_0.4.16   tibble_3.2.1      glue_1.7.0        ggprism_1.0.5    \n [5] immunarch_0.9.1   patchwork_1.2.0   data.table_1.15.4 dtplyr_1.3.1     \n [9] ggplot2_3.5.1     tidyr_1.3.1       dplyr_1.1.4       jsonlite_1.8.8   \n[13] rlang_1.1.4       logger_0.3.0     \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22       shinythemes_1.2.0      splines_4.3.3         \n  [4] later_1.3.2            cellranger_1.1.0       polyclip_1.10-7       \n  [7] fastDummies_1.7.3      factoextra_1.0.7       lifecycle_1.0.4       \n [10] rstatix_0.7.2          doParallel_1.0.17      globals_0.16.3        \n [13] lattice_0.22-6         prabclus_2.3-3         MASS_7.3-60.0.1       \n [16] backports_1.5.0        magrittr_2.0.3         plotly_4.10.4         \n [19] rlist_0.4.6.2          httpuv_1.6.15          Seurat_5.1.0          \n [22] sctransform_0.4.1      spam_2.10-0            flexmix_2.3-19        \n [25] spatstat.sparse_3.1-0  sp_2.1-4               reticulate_1.38.0     \n [28] cowplot_1.1.3          pbapply_1.7-2          RColorBrewer_1.1-3    \n [31] abind_1.4-5            Rtsne_0.17             quadprog_1.5-8        \n [34] purrr_1.0.2            ggraph_2.1.0           nnet_7.3-19           \n [37] tweenr_2.0.3           ggrepel_0.9.5          irlba_2.3.5.1         \n [40] listenv_0.9.1          spatstat.utils_3.0-5   pheatmap_1.0.12       \n [43] goftest_1.2-3          RSpectra_0.16-1        spatstat.random_3.2-3 \n [46] fitdistrplus_1.1-11    parallelly_1.38.0      leiden_0.4.3.1        \n [49] codetools_0.2-20       ggforce_0.4.2          tidyselect_1.2.1      \n [52] shape_1.4.6.1          ggseqlogo_0.2          farver_2.1.2          \n [55] viridis_0.6.5          matrixStats_1.1.0      stats4_4.3.3          \n [58] spatstat.explore_3.2-6 tidygraph_1.3.0        progressr_0.14.0      \n [61] ggridges_0.5.6         ggalluvial_0.12.5      survival_3.7-0        \n [64] iterators_1.0.14       foreach_1.5.2          tools_4.3.3           \n [67] stringdist_0.9.12      ica_1.0-3              Rcpp_1.0.13           \n [70] gridExtra_2.3          withr_3.0.1            fastmap_1.2.0         \n [73] fansi_1.0.6            digest_0.6.36          R6_2.5.1              \n [76] mime_0.12              colorspace_2.1-1       scattermore_1.2       \n [79] tensor_1.5             spatstat.data_3.1-2    diptest_0.77-1        \n [82] UpSetR_1.4.0           utf8_1.2.4             generics_0.1.3        \n [85] robustbase_0.99-3      class_7.3-22           graphlayouts_1.1.0    \n [88] httr_1.4.7             htmlwidgets_1.6.4      uwot_0.1.16           \n [91] pkgconfig_2.0.3        gtable_0.3.5           modeltools_0.2-23     \n [94] lmtest_0.9-40          htmltools_0.5.8.1      carData_3.0-5         \n [97] dotCall64_1.1-1        SeuratObject_5.0.2     scales_1.3.0          \n[100] png_0.1-8              tzdb_0.4.0             reshape2_1.4.4        \n[103] uuid_1.2-1             nlme_3.1-165           zoo_1.8-12            \n[106] GlobalOptions_0.1.2    stringr_1.5.1          KernSmooth_2.23-24    \n[109] parallel_4.3.3         miniUI_0.1.1.1         pillar_1.9.0          \n[112] grid_4.3.3             vctrs_0.6.5            RANN_2.6.1            \n[115] promises_1.3.0         ggpubr_0.6.0           car_3.1-2             \n[118] xtable_1.8-4           cluster_2.1.6          readr_2.1.5           \n[121] cli_3.6.3              compiler_4.3.3         future.apply_1.11.2   \n[124] ggsignif_0.6.4         labeling_0.4.3         mclust_6.1.1          \n[127] plyr_1.8.9             stringi_1.8.4          deldir_2.0-4          \n[130] viridisLite_0.4.2      assertthat_0.2.1       munsell_0.5.1         \n[133] lazyeval_0.2.2         spatstat.geom_3.2-9    Matrix_1.6-5          \n[136] RcppHNSW_0.6.0         hms_1.1.3              future_1.34.0         \n[139] fpc_2.2-12             shiny_1.9.1            kernlab_0.9-32        \n[142] ROCR_1.0-11            igraph_2.0.3           broom_1.0.6           \n[145] phangorn_2.11.1        fastmatch_1.1-4        DEoptimR_1.1-3        \n[148] readxl_1.4.3           ape_5.8               \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 91.14\nuser 113.47\nsys 1.29\npercent 125%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"RadarPlots",desc:"Radar plots for cell proportion in different clusters.",npages:1,report_toc:!0,order:19,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ggprism_1.0.5      ggradar_0.2        ggplot2_3.5.1      tibble_3.2.1      \n [5] tidyr_1.3.1        dplyr_1.1.4        Seurat_5.1.0       SeuratObject_5.0.2\n [9] sp_2.1-4           jsonlite_1.8.8     rlang_1.1.4        logger_0.3.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.2-9   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] labeling_0.4.3         utf8_1.2.4             promises_1.3.0        \n [19] purrr_1.0.2            goftest_1.2-3          later_1.3.2           \n [22] spatstat.utils_3.0-5   irlba_2.3.5.1          parallel_4.3.3        \n [25] cluster_2.1.6          R6_2.5.1               ica_1.0-3             \n [28] stringi_1.8.4          RColorBrewer_1.1-3     spatstat.data_3.1-2   \n [31] reticulate_1.38.0      parallelly_1.38.0      lmtest_0.9-40         \n [34] scattermore_1.2        Rcpp_1.0.13            tensor_1.5            \n [37] future.apply_1.11.2    zoo_1.8-12             sctransform_0.4.1     \n [40] httpuv_1.6.15          Matrix_1.6-5           splines_4.3.3         \n [43] igraph_2.0.3           tidyselect_1.2.1       abind_1.4-5           \n [46] spatstat.random_3.2-3  codetools_0.2-20       miniUI_0.1.1.1        \n [49] spatstat.explore_3.2-6 listenv_0.9.1          lattice_0.22-6        \n [52] plyr_1.8.9             withr_3.0.1            shiny_1.9.1           \n [55] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n [58] fastDummies_1.7.3      survival_3.7-0         polyclip_1.10-7       \n [61] fitdistrplus_1.1-11    pillar_1.9.0           KernSmooth_2.23-24    \n [64] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n [67] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [70] xtable_1.8-4           glue_1.7.0             lazyeval_0.2.2        \n [73] tools_4.3.3            data.table_1.15.4      RSpectra_0.16-1       \n [76] RANN_2.6.1             leiden_0.4.3.1         dotCall64_1.1-1       \n [79] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n [82] nlme_3.1-165           patchwork_1.2.0        cli_3.6.3             \n [85] spatstat.sparse_3.1-0  spam_2.10-0            fansi_1.0.6           \n [88] viridisLite_0.4.2      uwot_0.1.16            gtable_0.3.5          \n [91] digest_0.6.36          progressr_0.14.0       ggrepel_0.9.5         \n [94] farver_2.1.2           htmlwidgets_1.6.4      htmltools_0.5.8.1     \n [97] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[100] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 7.79\nuser 16.36\nsys 0.40\npercent 215%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"CloneResidency",desc:"Identification of clone residency",npages:1,report_toc:!0,order:20,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ComplexUpset_1.3.3  ggVennDiagram_1.2.3 ggprism_1.0.5      \n [4] ggplot2_3.5.1       ggnewscale_0.5.0    glue_1.7.0         \n [7] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[10] jsonlite_1.8.8      rlang_1.1.4         logger_0.3.0       \n\nloaded via a namespace (and not attached):\n  [1] DBI_1.2.3              deldir_2.0-4           pbapply_1.7-2         \n  [4] gridExtra_2.3          magrittr_2.0.3         RcppAnnoy_0.0.22      \n  [7] e1071_1.7-14           spatstat.geom_3.2-9    matrixStats_1.1.0     \n [10] ggridges_0.5.6         compiler_4.3.3         png_0.1-8             \n [13] vctrs_0.6.5            reshape2_1.4.4         stringr_1.5.1         \n [16] pkgconfig_2.0.3        fastmap_1.2.0          labeling_0.4.3        \n [19] utf8_1.2.4             promises_1.3.0         purrr_1.0.2           \n [22] RVenn_1.1.0            goftest_1.2-3          later_1.3.2           \n [25] spatstat.utils_3.0-5   irlba_2.3.5.1          parallel_4.3.3        \n [28] cluster_2.1.6          R6_2.5.1               ica_1.0-3             \n [31] stringi_1.8.4          RColorBrewer_1.1-3     spatstat.data_3.1-2   \n [34] reticulate_1.38.0      parallelly_1.38.0      lmtest_0.9-40         \n [37] scattermore_1.2        Rcpp_1.0.13            tensor_1.5            \n [40] future.apply_1.11.2    zoo_1.8-12             sctransform_0.4.1     \n [43] httpuv_1.6.15          Matrix_1.6-5           splines_4.3.3         \n [46] igraph_2.0.3           tidyselect_1.2.1       abind_1.4-5           \n [49] spatstat.random_3.2-3  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.2-6 listenv_0.9.1          lattice_0.22-6        \n [55] plyr_1.8.9             shiny_1.9.1            withr_3.0.1           \n [58] ROCR_1.0-11            Rtsne_0.17             sf_1.0-16             \n [61] future_1.34.0          fastDummies_1.7.3      survival_3.7-0        \n [64] units_0.8-5            proxy_0.4-27           polyclip_1.10-7       \n [67] fitdistrplus_1.1-11    pillar_1.9.0           Seurat_5.1.0          \n [70] KernSmooth_2.23-24     plotly_4.10.4          generics_0.1.3        \n [73] RcppHNSW_0.6.0         sp_2.1-4               munsell_0.5.1         \n [76] scales_1.3.0           globals_0.16.3         xtable_1.8-4          \n [79] class_7.3-22           lazyeval_0.2.2         tools_4.3.3           \n [82] data.table_1.15.4      RSpectra_0.16-1        RANN_2.6.1            \n [85] leiden_0.4.3.1         dotCall64_1.1-1        cowplot_1.1.3         \n [88] grid_4.3.3             colorspace_2.1-1       nlme_3.1-165          \n [91] patchwork_1.2.0        cli_3.6.3              spatstat.sparse_3.1-0 \n [94] spam_2.10-0            fansi_1.0.6            viridisLite_0.4.2     \n [97] uwot_0.1.16            gtable_0.3.5           digest_0.6.36         \n[100] classInt_0.4-10        progressr_0.14.0       ggrepel_0.9.5         \n[103] farver_2.1.2           htmlwidgets_1.6.4      SeuratObject_5.0.2    \n[106] htmltools_0.5.8.1      lifecycle_1.0.4        httr_1.4.7            \n[109] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 16.51\nuser 24.80\nsys 0.51\npercent 153%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ScFGSEA",desc:"Gene set enrichment analysis for cells in different groups using `fgsea`",npages:2,report_toc:!0,order:21,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] fgsea_1.28.0       data.table_1.15.4  matrixStats_1.1.0  tidyseurat_0.8.0  \n [5] ttservice_0.4.1    Seurat_5.1.0       SeuratObject_5.0.2 sp_2.1-4          \n [9] tidyr_1.3.1        tidyselect_1.2.1   tibble_3.2.1       dplyr_1.1.4       \n[13] ggplot2_3.5.1      jsonlite_1.8.8     rlang_1.1.4        logger_0.3.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       ggridges_0.5.6        \n  [7] compiler_4.3.3         spatstat.geom_3.2-9    png_0.1-8             \n [10] vctrs_0.6.5            reshape2_1.4.4         stringr_1.5.1         \n [13] pkgconfig_2.0.3        fastmap_1.2.0          ellipsis_0.3.2        \n [16] labeling_0.4.3         utf8_1.2.4             promises_1.3.0        \n [19] purrr_1.0.2            goftest_1.2-3          later_1.3.2           \n [22] BiocParallel_1.36.0    spatstat.utils_3.0-5   irlba_2.3.5.1         \n [25] parallel_4.3.3         cluster_2.1.6          R6_2.5.1              \n [28] ica_1.0-3              stringi_1.8.4          RColorBrewer_1.1-3    \n [31] spatstat.data_3.1-2    reticulate_1.38.0      parallelly_1.38.0     \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.13           \n [37] tensor_1.5             future.apply_1.11.2    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [43] splines_4.3.3          igraph_2.0.3           abind_1.4-5           \n [46] spatstat.random_3.2-3  codetools_0.2-20       miniUI_0.1.1.1        \n [49] spatstat.explore_3.2-6 listenv_0.9.1          lattice_0.22-6        \n [52] plyr_1.8.9             shiny_1.9.1            withr_3.0.1           \n [55] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n [58] fastDummies_1.7.3      survival_3.7-0         polyclip_1.10-7       \n [61] fitdistrplus_1.1-11    pillar_1.9.0           KernSmooth_2.23-24    \n [64] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n [67] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [70] xtable_1.8-4           glue_1.7.0             lazyeval_0.2.2        \n [73] tools_4.3.3            RSpectra_0.16-1        RANN_2.6.1            \n [76] leiden_0.4.3.1         dotCall64_1.1-1        fastmatch_1.1-4       \n [79] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n [82] nlme_3.1-165           patchwork_1.2.0        cli_3.6.3             \n [85] spatstat.sparse_3.1-0  spam_2.10-0            fansi_1.0.6           \n [88] viridisLite_0.4.2      uwot_0.1.16            gtable_0.3.5          \n [91] digest_0.6.36          progressr_0.14.0       ggrepel_0.9.5         \n [94] farver_2.1.2           htmlwidgets_1.6.4      htmltools_0.5.8.1     \n [97] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[100] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 96.79\nuser 103.64\nsys 1.43\npercent 108%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"MarkersFinder",desc:"Find markers between different groups of cells",npages:1,report_toc:!0,order:22,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] UpSetR_1.4.0        ggVennDiagram_1.2.3 tidyseurat_0.8.0   \n [4] ttservice_0.4.1     future_1.34.0       ggrepel_0.9.5      \n [7] ggprism_1.0.5       ggplot2_3.5.1       enrichR_3.2        \n[10] Seurat_5.1.0        SeuratObject_5.0.2  sp_2.1-4           \n[13] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[16] tidyselect_1.2.1    digest_0.6.36       jsonlite_1.8.8     \n[19] rlang_1.1.4         logger_0.3.0       \n\nloaded via a namespace (and not attached):\n  [1] RColorBrewer_1.1-3     magrittr_2.0.3         spatstat.utils_3.0-5  \n  [4] farver_2.1.2           vctrs_0.6.5            ROCR_1.0-11           \n  [7] spatstat.explore_3.2-6 htmltools_0.5.8.1      curl_5.2.1            \n [10] sctransform_0.4.1      parallelly_1.38.0      KernSmooth_2.23-24    \n [13] htmlwidgets_1.6.4      ica_1.0-3              plyr_1.8.9            \n [16] plotly_4.10.4          zoo_1.8-12             igraph_2.0.3          \n [19] mime_0.12              lifecycle_1.0.4        pkgconfig_2.0.3       \n [22] Matrix_1.6-5           R6_2.5.1               fastmap_1.2.0         \n [25] fitdistrplus_1.1-11    shiny_1.9.1            colorspace_2.1-1      \n [28] patchwork_1.2.0        tensor_1.5             RSpectra_0.16-1       \n [31] irlba_2.3.5.1          WriteXLS_6.7.0         labeling_0.4.3        \n [34] progressr_0.14.0       fansi_1.0.6            spatstat.sparse_3.1-0 \n [37] httr_1.4.7             polyclip_1.10-7        abind_1.4-5           \n [40] compiler_4.3.3         withr_3.0.1            fastDummies_1.7.3     \n [43] MASS_7.3-60.0.1        rjson_0.2.21           tools_4.3.3           \n [46] lmtest_0.9-40          httpuv_1.6.15          future.apply_1.11.2   \n [49] goftest_1.2-3          glue_1.7.0             nlme_3.1-165          \n [52] promises_1.3.0         grid_4.3.3             Rtsne_0.17            \n [55] cluster_2.1.6          reshape2_1.4.4         generics_0.1.3        \n [58] gtable_0.3.5           spatstat.data_3.1-2    data.table_1.15.4     \n [61] utf8_1.2.4             spatstat.geom_3.2-9    RcppAnnoy_0.0.22      \n [64] RANN_2.6.1             pillar_1.9.0           stringr_1.5.1         \n [67] spam_2.10-0            RcppHNSW_0.6.0         limma_3.58.1          \n [70] later_1.3.2            splines_4.3.3          lattice_0.22-6        \n [73] survival_3.7-0         deldir_2.0-4           miniUI_0.1.1.1        \n [76] pbapply_1.7-2          gridExtra_2.3          scattermore_1.2       \n [79] statmod_1.5.0          matrixStats_1.1.0      stringi_1.8.4         \n [82] lazyeval_0.2.2         codetools_0.2-20       RVenn_1.1.0           \n [85] cli_3.6.3              uwot_0.1.16            xtable_1.8-4          \n [88] reticulate_1.38.0      munsell_0.5.1          Rcpp_1.0.13           \n [91] globals_0.16.3         spatstat.random_3.2-3  png_0.1-8             \n [94] parallel_4.3.3         ellipsis_0.3.2         presto_1.0.0          \n [97] dotCall64_1.1-1        listenv_0.9.1          viridisLite_0.4.2     \n[100] scales_1.3.0           ggridges_0.5.6         leiden_0.4.3.1        \n[103] purrr_1.0.2            cowplot_1.1.3         \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 15.26\nuser 20.73\nsys 0.60\npercent 139%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"MetaMarkers",desc:"Find markers between three or more groups of cells, using one-way ANOVA or Kruskal-Wallis test.",npages:2,report_toc:!0,order:23,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] parallel  stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] tidyseurat_0.8.0   ttservice_0.4.1    ggprism_1.0.5      ggplot2_3.5.1     \n [5] enrichR_3.2        Seurat_5.1.0       SeuratObject_5.0.2 sp_2.1-4          \n [9] Matrix_1.6-5       tibble_3.2.1       tidyr_1.3.1        dplyr_1.1.4       \n[13] tidyselect_1.2.1   jsonlite_1.8.8     rlang_1.1.4        logger_0.3.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.2-9   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] ellipsis_0.3.2         labeling_0.4.3         utf8_1.2.4            \n [19] promises_1.3.0         purrr_1.0.2            WriteXLS_6.7.0        \n [22] goftest_1.2-3          later_1.3.2            spatstat.utils_3.0-5  \n [25] irlba_2.3.5.1          cluster_2.1.6          R6_2.5.1              \n [28] ica_1.0-3              stringi_1.8.4          RColorBrewer_1.1-3    \n [31] spatstat.data_3.1-2    reticulate_1.38.0      parallelly_1.38.0     \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.13           \n [37] tensor_1.5             future.apply_1.11.2    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          splines_4.3.3         \n [43] igraph_2.0.3           abind_1.4-5            spatstat.random_3.2-3 \n [46] codetools_0.2-20       miniUI_0.1.1.1         spatstat.explore_3.2-6\n [49] curl_5.2.1             listenv_0.9.1          lattice_0.22-6        \n [52] plyr_1.8.9             withr_3.0.1            shiny_1.9.1           \n [55] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n [58] fastDummies_1.7.3      survival_3.7-0         polyclip_1.10-7       \n [61] fitdistrplus_1.1-11    pillar_1.9.0           KernSmooth_2.23-24    \n [64] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n [67] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [70] xtable_1.8-4           glue_1.7.0             lazyeval_0.2.2        \n [73] tools_4.3.3            data.table_1.15.4      RSpectra_0.16-1       \n [76] RANN_2.6.1             leiden_0.4.3.1         dotCall64_1.1-1       \n [79] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n [82] nlme_3.1-165           patchwork_1.2.0        cli_3.6.3             \n [85] spatstat.sparse_3.1-0  spam_2.10-0            fansi_1.0.6           \n [88] viridisLite_0.4.2      uwot_0.1.16            gtable_0.3.5          \n [91] digest_0.6.36          progressr_0.14.0       ggrepel_0.9.5         \n [94] farver_2.1.2           rjson_0.2.21           htmlwidgets_1.6.4     \n [97] htmltools_0.5.8.1      lifecycle_1.0.4        httr_1.4.7            \n[100] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 335.98\nuser 1633.65\nsys 90.08\npercent 513%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"CellsDistribution",desc:"Distribution of cells (i.e. in a TCR clone) from different groups for each cluster",npages:1,report_toc:!0,order:24,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] grid      stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] ComplexHeatmap_2.18.0 circlize_0.4.16       UpSetR_1.4.0         \n [4] ggVennDiagram_1.2.3   patchwork_1.2.0       ggplot2_3.5.1        \n [7] tibble_3.2.1          Seurat_5.1.0          SeuratObject_5.0.2   \n[10] sp_2.1-4              tidyr_1.3.1           dplyr_1.1.4          \n[13] tidyselect_1.2.1      jsonlite_1.8.8        rlang_1.1.4          \n[16] logger_0.3.0         \n\nloaded via a namespace (and not attached):\n  [1] RColorBrewer_1.1-3     shape_1.4.6.1          magrittr_2.0.3        \n  [4] spatstat.utils_3.0-5   farver_2.1.2           GlobalOptions_0.1.2   \n  [7] vctrs_0.6.5            ROCR_1.0-11            Cairo_1.6-2           \n [10] spatstat.explore_3.2-6 htmltools_0.5.8.1      sctransform_0.4.1     \n [13] parallelly_1.38.0      KernSmooth_2.23-24     htmlwidgets_1.6.4     \n [16] ica_1.0-3              plyr_1.8.9             plotly_4.10.4         \n [19] zoo_1.8-12             igraph_2.0.3           mime_0.12             \n [22] lifecycle_1.0.4        iterators_1.0.14       pkgconfig_2.0.3       \n [25] Matrix_1.6-5           R6_2.5.1               fastmap_1.2.0         \n [28] clue_0.3-65            fitdistrplus_1.1-11    future_1.34.0         \n [31] shiny_1.9.1            digest_0.6.36          colorspace_2.1-1      \n [34] S4Vectors_0.40.2       tensor_1.5             RSpectra_0.16-1       \n [37] irlba_2.3.5.1          labeling_0.4.3         progressr_0.14.0      \n [40] fansi_1.0.6            spatstat.sparse_3.1-0  httr_1.4.7            \n [43] polyclip_1.10-7        abind_1.4-5            compiler_4.3.3        \n [46] withr_3.0.1            doParallel_1.0.17      fastDummies_1.7.3     \n [49] MASS_7.3-60.0.1        rjson_0.2.21           tools_4.3.3           \n [52] lmtest_0.9-40          httpuv_1.6.15          future.apply_1.11.2   \n [55] goftest_1.2-3          glue_1.7.0             nlme_3.1-165          \n [58] promises_1.3.0         Rtsne_0.17             cluster_2.1.6         \n [61] reshape2_1.4.4         generics_0.1.3         gtable_0.3.5          \n [64] spatstat.data_3.1-2    data.table_1.15.4      utf8_1.2.4            \n [67] BiocGenerics_0.48.1    spatstat.geom_3.2-9    RcppAnnoy_0.0.22      \n [70] ggrepel_0.9.5          RANN_2.6.1             foreach_1.5.2         \n [73] pillar_1.9.0           stringr_1.5.1          spam_2.10-0           \n [76] RcppHNSW_0.6.0         later_1.3.2            splines_4.3.3         \n [79] lattice_0.22-6         survival_3.7-0         deldir_2.0-4          \n [82] miniUI_0.1.1.1         pbapply_1.7-2          gridExtra_2.3         \n [85] IRanges_2.36.0         scattermore_1.2        stats4_4.3.3          \n [88] matrixStats_1.1.0      stringi_1.8.4          lazyeval_0.2.2        \n [91] codetools_0.2-20       RVenn_1.1.0            cli_3.6.3             \n [94] uwot_0.1.16            xtable_1.8-4           reticulate_1.38.0     \n [97] munsell_0.5.1          Rcpp_1.0.13            globals_0.16.3        \n[100] spatstat.random_3.2-3  png_0.1-8              parallel_4.3.3        \n[103] dotCall64_1.1-1        listenv_0.9.1          viridisLite_0.4.2     \n[106] scales_1.3.0           ggridges_0.5.6         leiden_0.4.3.1        \n[109] purrr_1.0.2            crayon_1.5.3           GetoptLong_1.0.5      \n[112] cowplot_1.1.3         \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 10.15\nuser 18.49\nsys 0.52\npercent 187%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"CDR3AAPhyschem",desc:"CDR3 AA physicochemical feature analysis",npages:1,report_toc:!0,order:25,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] stringr_1.5.1       broom.mixed_0.2.9.5 glmnet_4.1-8       \n [4] Matrix_1.6-5        hash_3.0.1          glue_1.7.0         \n [7] ggridges_0.5.6      ggplot2_3.5.1       tibble_3.2.1       \n[10] tidyr_1.3.1         dplyr_1.1.4         jsonlite_1.8.8     \n[13] rlang_1.1.4         logger_0.3.0       \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         furrr_0.3.1            RcppAnnoy_0.0.22      \n  [7] spatstat.geom_3.2-9    matrixStats_1.1.0      compiler_4.3.3        \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] pkgconfig_2.0.3        shape_1.4.6.1          fastmap_1.2.0         \n [16] backports_1.5.0        labeling_0.4.3         utf8_1.2.4            \n [19] promises_1.3.0         purrr_1.0.2            goftest_1.2-3         \n [22] later_1.3.2            spatstat.utils_3.0-5   broom_1.0.6           \n [25] irlba_2.3.5.1          parallel_4.3.3         cluster_2.1.6         \n [28] R6_2.5.1               ica_1.0-3              spatstat.data_3.1-2   \n [31] stringi_1.8.4          RColorBrewer_1.1-3     reticulate_1.38.0     \n [34] parallelly_1.38.0      lmtest_0.9-40          scattermore_1.2       \n [37] Rcpp_1.0.13            iterators_1.0.14       tensor_1.5            \n [40] future.apply_1.11.2    zoo_1.8-12             sctransform_0.4.1     \n [43] httpuv_1.6.15          splines_4.3.3          igraph_2.0.3          \n [46] tidyselect_1.2.1       abind_1.4-5            spatstat.random_3.2-3 \n [49] codetools_0.2-20       miniUI_0.1.1.1         spatstat.explore_3.2-6\n [52] listenv_0.9.1          lattice_0.22-6         plyr_1.8.9            \n [55] shiny_1.9.1            withr_3.0.1            ROCR_1.0-11           \n [58] Rtsne_0.17             future_1.34.0          fastDummies_1.7.3     \n [61] survival_3.7-0         polyclip_1.10-7        fitdistrplus_1.1-11   \n [64] pillar_1.9.0           Seurat_5.1.0           KernSmooth_2.23-24    \n [67] foreach_1.5.2          plotly_4.10.4          generics_0.1.3        \n [70] RcppHNSW_0.6.0         sp_2.1-4               munsell_0.5.1         \n [73] scales_1.3.0           globals_0.16.3         xtable_1.8-4          \n [76] lazyeval_0.2.2         tools_4.3.3            data.table_1.15.4     \n [79] RSpectra_0.16-1        forcats_1.0.0          RANN_2.6.1            \n [82] leiden_0.4.3.1         dotCall64_1.1-1        cowplot_1.1.3         \n [85] grid_4.3.3             colorspace_2.1-1       nlme_3.1-165          \n [88] patchwork_1.2.0        cli_3.6.3              spatstat.sparse_3.1-0 \n [91] spam_2.10-0            fansi_1.0.6            viridisLite_0.4.2     \n [94] uwot_0.1.16            gtable_0.3.5           digest_0.6.36         \n [97] progressr_0.14.0       ggrepel_0.9.5          farver_2.1.2          \n[100] htmlwidgets_1.6.4      SeuratObject_5.0.2     htmltools_0.5.8.1     \n[103] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[106] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 10.75\nuser 19.09\nsys 0.55\npercent 182%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ScrnaMetabolicLandscape",desc:"Metabolic landscape analysis for scRNA-seq data",order:28,procs:[{name:"MetabolicPathwayActivity",desc:"This process calculates the pathway activities in different groups and subsets.",npages:1,report_toc:!0,order:28,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n [1] grid      parallel  stats4    stats     graphics  grDevices utils    \n [8] datasets  methods   base     \n\nother attached packages:\n [1] ComplexHeatmap_2.18.0       Seurat_5.1.0               \n [3] SeuratObject_5.0.2          sp_2.1-4                   \n [5] ggprism_1.0.5               RColorBrewer_1.1-3         \n [7] reshape2_1.4.4              scater_1.30.1              \n [9] scuttle_1.12.0              SingleCellExperiment_1.24.0\n[11] SummarizedExperiment_1.32.0 Biobase_2.62.0             \n[13] GenomicRanges_1.54.1        GenomeInfoDb_1.38.1        \n[15] IRanges_2.36.0              S4Vectors_0.40.2           \n[17] BiocGenerics_0.48.1         MatrixGenerics_1.14.0      \n[19] matrixStats_1.1.0           tibble_3.2.1               \n[21] dplyr_1.1.4                 ggplot2_3.5.1              \n[23] jsonlite_1.8.8              rlang_1.1.4                \n[25] logger_0.3.0               \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22          splines_4.3.3            \n  [3] later_1.3.2               bitops_1.0-8             \n  [5] polyclip_1.10-7           fastDummies_1.7.3        \n  [7] lifecycle_1.0.4           doParallel_1.0.17        \n  [9] globals_0.16.3            lattice_0.22-6           \n [11] MASS_7.3-60.0.1           magrittr_2.0.3           \n [13] plotly_4.10.4             httpuv_1.6.15            \n [15] sctransform_0.4.1         spam_2.10-0              \n [17] spatstat.sparse_3.1-0     reticulate_1.38.0        \n [19] cowplot_1.1.3             pbapply_1.7-2            \n [21] abind_1.4-5               zlibbioc_1.48.0          \n [23] Rtsne_0.17                purrr_1.0.2              \n [25] RCurl_1.98-1.16           circlize_0.4.16          \n [27] GenomeInfoDbData_1.2.11   ggrepel_0.9.5            \n [29] irlba_2.3.5.1             listenv_0.9.1            \n [31] spatstat.utils_3.0-5      goftest_1.2-3            \n [33] RSpectra_0.16-1           spatstat.random_3.2-3    \n [35] fitdistrplus_1.1-11       parallelly_1.38.0        \n [37] DelayedMatrixStats_1.24.0 leiden_0.4.3.1           \n [39] codetools_0.2-20          DelayedArray_0.28.0      \n [41] shape_1.4.6.1             tidyselect_1.2.1         \n [43] farver_2.1.2              ScaledMatrix_1.10.0      \n [45] viridis_0.6.5             spatstat.explore_3.2-6   \n [47] GetoptLong_1.0.5          BiocNeighbors_1.20.0     \n [49] progressr_0.14.0          ggridges_0.5.6           \n [51] survival_3.7-0            iterators_1.0.14         \n [53] foreach_1.5.2             tools_4.3.3              \n [55] ica_1.0-3                 Rcpp_1.0.13              \n [57] glue_1.7.0                gridExtra_2.3            \n [59] SparseArray_1.2.2         withr_3.0.1              \n [61] fastmap_1.2.0             fansi_1.0.6              \n [63] digest_0.6.36             rsvd_1.0.5               \n [65] R6_2.5.1                  mime_0.12                \n [67] colorspace_2.1-1          Cairo_1.6-2              \n [69] scattermore_1.2           tensor_1.5               \n [71] spatstat.data_3.1-2       utf8_1.2.4               \n [73] tidyr_1.3.1               generics_0.1.3           \n [75] data.table_1.15.4         httr_1.4.7               \n [77] htmlwidgets_1.6.4         S4Arrays_1.2.0           \n [79] uwot_0.1.16               pkgconfig_2.0.3          \n [81] gtable_0.3.5              lmtest_0.9-40            \n [83] XVector_0.42.0            htmltools_0.5.8.1        \n [85] dotCall64_1.1-1           clue_0.3-65              \n [87] scales_1.3.0              png_0.1-8                \n [89] rjson_0.2.21              nlme_3.1-165             \n [91] zoo_1.8-12                GlobalOptions_0.1.2      \n [93] stringr_1.5.1             KernSmooth_2.23-24       \n [95] miniUI_0.1.1.1            vipor_0.4.7              \n [97] pillar_1.9.0              vctrs_0.6.5              \n [99] RANN_2.6.1                promises_1.3.0           \n[101] BiocSingular_1.18.0       beachmat_2.18.0          \n[103] xtable_1.8-4              cluster_2.1.6            \n[105] beeswarm_0.4.0            cli_3.6.3                \n[107] compiler_4.3.3            crayon_1.5.3             \n[109] future.apply_1.11.2       plyr_1.8.9               \n[111] ggbeeswarm_0.7.2          stringi_1.8.4            \n[113] viridisLite_0.4.2         deldir_2.0-4             \n[115] BiocParallel_1.36.0       munsell_0.5.1            \n[117] lazyeval_0.2.2            spatstat.geom_3.2-9      \n[119] Matrix_1.6-5              RcppHNSW_0.6.0           \n[121] patchwork_1.2.0           sparseMatrixStats_1.14.0 \n[123] future_1.34.0             shiny_1.9.1              \n[125] ROCR_1.0-11               igraph_2.0.3             \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 763.83\nuser 756.90\nsys 13.77\npercent 100%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3459.029\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"MetabolicFeatures",desc:"This process performs enrichment analysis for the metabolic pathways for each group in each subset.",npages:1,report_toc:!0,order:29,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] parallel  stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] matrixStats_1.1.0  fgsea_1.28.0       data.table_1.15.4  Seurat_5.1.0      \n [5] SeuratObject_5.0.2 sp_2.1-4           tibble_3.2.1       dplyr_1.1.4       \n [9] ggplot2_3.5.1      jsonlite_1.8.8     rlang_1.1.4        logger_0.3.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       ggridges_0.5.6        \n  [7] compiler_4.3.3         spatstat.geom_3.2-9    png_0.1-8             \n [10] vctrs_0.6.5            reshape2_1.4.4         stringr_1.5.1         \n [13] pkgconfig_2.0.3        fastmap_1.2.0          labeling_0.4.3        \n [16] utf8_1.2.4             promises_1.3.0         purrr_1.0.2           \n [19] goftest_1.2-3          later_1.3.2            BiocParallel_1.36.0   \n [22] spatstat.utils_3.0-5   irlba_2.3.5.1          cluster_2.1.6         \n [25] R6_2.5.1               ica_1.0-3              stringi_1.8.4         \n [28] RColorBrewer_1.1-3     spatstat.data_3.1-2    reticulate_1.38.0     \n [31] parallelly_1.38.0      lmtest_0.9-40          scattermore_1.2       \n [34] Rcpp_1.0.13            tensor_1.5             future.apply_1.11.2   \n [37] zoo_1.8-12             sctransform_0.4.1      httpuv_1.6.15         \n [40] Matrix_1.6-5           splines_4.3.3          igraph_2.0.3          \n [43] tidyselect_1.2.1       abind_1.4-5            spatstat.random_3.2-3 \n [46] codetools_0.2-20       miniUI_0.1.1.1         spatstat.explore_3.2-6\n [49] listenv_0.9.1          lattice_0.22-6         plyr_1.8.9            \n [52] shiny_1.9.1            withr_3.0.1            ROCR_1.0-11           \n [55] Rtsne_0.17             future_1.34.0          fastDummies_1.7.3     \n [58] survival_3.7-0         polyclip_1.10-7        fitdistrplus_1.1-11   \n [61] pillar_1.9.0           KernSmooth_2.23-24     plotly_4.10.4         \n [64] generics_0.1.3         RcppHNSW_0.6.0         munsell_0.5.1         \n [67] scales_1.3.0           globals_0.16.3         xtable_1.8-4          \n [70] glue_1.7.0             lazyeval_0.2.2         tools_4.3.3           \n [73] RSpectra_0.16-1        RANN_2.6.1             leiden_0.4.3.1        \n [76] dotCall64_1.1-1        fastmatch_1.1-4        cowplot_1.1.3         \n [79] grid_4.3.3             tidyr_1.3.1            colorspace_2.1-1      \n [82] nlme_3.1-165           patchwork_1.2.0        cli_3.6.3             \n [85] spatstat.sparse_3.1-0  spam_2.10-0            fansi_1.0.6           \n [88] viridisLite_0.4.2      uwot_0.1.16            gtable_0.3.5          \n [91] digest_0.6.36          progressr_0.14.0       ggrepel_0.9.5         \n [94] farver_2.1.2           htmlwidgets_1.6.4      htmltools_0.5.8.1     \n [97] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[100] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 188.82\nuser 192.45\nsys 7.25\npercent 105%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"MetabolicFeaturesIntraSubset",desc:"Intra-subset metabolic features - Enrichment analysis in details",npages:3,report_toc:!0,order:30,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats4    parallel  stats     graphics  grDevices utils     datasets \n[8] methods   base     \n\nother attached packages:\n [1] fgsea_1.28.0                data.table_1.15.4          \n [3] Seurat_5.1.0                SeuratObject_5.0.2         \n [5] sp_2.1-4                    scater_1.30.1              \n [7] scuttle_1.12.0              SingleCellExperiment_1.24.0\n [9] SummarizedExperiment_1.32.0 Biobase_2.62.0             \n[11] GenomicRanges_1.54.1        GenomeInfoDb_1.38.1        \n[13] IRanges_2.36.0              S4Vectors_0.40.2           \n[15] BiocGenerics_0.48.1         MatrixGenerics_1.14.0      \n[17] matrixStats_1.1.0           tibble_3.2.1               \n[19] dplyr_1.1.4                 ggplot2_3.5.1              \n[21] jsonlite_1.8.8              rlang_1.1.4                \n[23] logger_0.3.0               \n\nloaded via a namespace (and not attached):\n  [1] RColorBrewer_1.1-3        magrittr_2.0.3           \n  [3] spatstat.utils_3.0-5      ggbeeswarm_0.7.2         \n  [5] farver_2.1.2              zlibbioc_1.48.0          \n  [7] vctrs_0.6.5               ROCR_1.0-11              \n  [9] spatstat.explore_3.2-6    DelayedMatrixStats_1.24.0\n [11] RCurl_1.98-1.16           htmltools_0.5.8.1        \n [13] S4Arrays_1.2.0            BiocNeighbors_1.20.0     \n [15] SparseArray_1.2.2         sctransform_0.4.1        \n [17] parallelly_1.38.0         KernSmooth_2.23-24       \n [19] htmlwidgets_1.6.4         ica_1.0-3                \n [21] plyr_1.8.9                plotly_4.10.4            \n [23] zoo_1.8-12                igraph_2.0.3             \n [25] mime_0.12                 lifecycle_1.0.4          \n [27] pkgconfig_2.0.3           rsvd_1.0.5               \n [29] Matrix_1.6-5              R6_2.5.1                 \n [31] fastmap_1.2.0             GenomeInfoDbData_1.2.11  \n [33] fitdistrplus_1.1-11       future_1.34.0            \n [35] shiny_1.9.1               digest_0.6.36            \n [37] colorspace_2.1-1          patchwork_1.2.0          \n [39] tensor_1.5                RSpectra_0.16-1          \n [41] irlba_2.3.5.1             beachmat_2.18.0          \n [43] labeling_0.4.3            progressr_0.14.0         \n [45] spatstat.sparse_3.1-0     fansi_1.0.6              \n [47] polyclip_1.10-7           httr_1.4.7               \n [49] abind_1.4-5               compiler_4.3.3           \n [51] withr_3.0.1               BiocParallel_1.36.0      \n [53] viridis_0.6.5             fastDummies_1.7.3        \n [55] MASS_7.3-60.0.1           DelayedArray_0.28.0      \n [57] tools_4.3.3               vipor_0.4.7              \n [59] lmtest_0.9-40             beeswarm_0.4.0           \n [61] httpuv_1.6.15             future.apply_1.11.2      \n [63] goftest_1.2-3             glue_1.7.0               \n [65] nlme_3.1-165              promises_1.3.0           \n [67] grid_4.3.3                Rtsne_0.17               \n [69] cluster_2.1.6             reshape2_1.4.4           \n [71] generics_0.1.3            spatstat.data_3.1-2      \n [73] gtable_0.3.5              tidyr_1.3.1              \n [75] BiocSingular_1.18.0       ScaledMatrix_1.10.0      \n [77] utf8_1.2.4                XVector_0.42.0           \n [79] spatstat.geom_3.2-9       RcppAnnoy_0.0.22         \n [81] stringr_1.5.1             ggrepel_0.9.5            \n [83] RANN_2.6.1                pillar_1.9.0             \n [85] spam_2.10-0               RcppHNSW_0.6.0           \n [87] later_1.3.2               splines_4.3.3            \n [89] lattice_0.22-6            deldir_2.0-4             \n [91] survival_3.7-0            tidyselect_1.2.1         \n [93] miniUI_0.1.1.1            pbapply_1.7-2            \n [95] gridExtra_2.3             scattermore_1.2          \n [97] stringi_1.8.4             lazyeval_0.2.2           \n [99] codetools_0.2-20          cli_3.6.3                \n[101] uwot_0.1.16               xtable_1.8-4             \n[103] reticulate_1.38.0         munsell_0.5.1            \n[105] Rcpp_1.0.13               spatstat.random_3.2-3    \n[107] globals_0.16.3            png_0.1-8                \n[109] dotCall64_1.1-1           sparseMatrixStats_1.14.0 \n[111] bitops_1.0-8              listenv_0.9.1            \n[113] viridisLite_0.4.2         scales_1.3.0             \n[115] ggridges_0.5.6            leiden_0.4.3.1           \n[117] purrr_1.0.2               crayon_1.5.3             \n[119] fastmatch_1.1-4           cowplot_1.1.3            \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 99.35\nuser 104.68\nsys 3.22\npercent 108%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"MetabolicPathwayHeterogeneity",desc:"Calculate Metabolic Pathway heterogeneity.",npages:1,report_toc:!0,order:31}]}]};function Ra(e){let t,n;return t=new zt({props:{dir:"up"}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function Ba(e){let t,n;return t=new zt({props:{dir:"down"}}),{c(){f(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){C(t,e)}}}function Ia(e){let t,n,a,s,r,i=e[1]>0&&Ra();n=new Ua({});let m=e[1]<e[3].npages-1&&Ba();return{c(){i&&i.c(),t=$(),f(n.$$.fragment),a=$(),m&&m.c(),s=u()},m(e,o){i&&i.m(e,o),l(e,t,o),b(n,e,o),l(e,a,o),m&&m.m(e,o),l(e,s,o),r=!0},p(e,n){e[1]>0?i?2&n&&o(i,1):(i=Ra(),i.c(),o(i,1),i.m(t.parentNode,t)):i&&(p(),d(i,1,1,(()=>{i=null})),c()),e[1]<e[3].npages-1?m?10&n&&o(m,1):(m=Ba(),m.c(),o(m,1),m.m(s.parentNode,s)):m&&(p(),d(m,1,1,(()=>{m=null})),c())},i(e){r||(o(i),o(n.$$.fragment,e),o(m),r=!0)},o(e){d(i),d(n.$$.fragment,e),d(m),r=!1},d(e){e&&(_(t),_(a),_(s)),i&&i.d(e),C(n,e),m&&m.d(e)}}}function Ha(e){let t,n,a;function s(t){e[4](t)}let r={logo:e[0],logotext:e[3].desc,versions:Da.versions,entries:Da.entries,pipeline_name:Da.pipeline.name,report_toc:e[3].report_toc,runinfo:e[3].runinfo,toc:Na,page:e[1],$$slots:{default:[Ia]},$$scope:{ctx:e}};return void 0!==e[2]&&(r.isSideNavOpen=e[2]),t=new Ht({props:r}),h.push((()=>v(t,"isSideNavOpen",s))),{c(){f(t.$$.fragment)},m(e,n){b(t,e,n),a=!0},p(e,[a]){const s={};1&a&&(s.logo=e[0]),8&a&&(s.logotext=e[3].desc),8&a&&(s.report_toc=e[3].report_toc),8&a&&(s.runinfo=e[3].runinfo),2&a&&(s.page=e[1]),42&a&&(s.$$scope={dirty:a,ctx:e}),!n&&4&a&&(n=!0,s.isSideNavOpen=e[2],y((()=>n=!1))),t.$set(s)},i(e){a||(o(t.$$.fragment,e),a=!0)},o(e){d(t.$$.fragment,e),a=!1},d(e){C(t,e)}}}function Ga(e,t,n){let a,{name:s}=t,{page:r=0}=t,i=!1;for(const e of Da.entries)if(e.procs){for(const t of e.procs)if(t.name==s){a=t;break}}else if(e.name==s){a=e;break}return g((()=>{const e=new URL(window.location.href);let t;e.hash.length>1&&(t=document.getElementById(`${e.hash.slice(1)}`)),t&&t.scrollIntoView({block:"start",behavior:"smooth"})})),e.$$set=e=>{"name"in e&&n(0,s=e.name),"page"in e&&n(1,r=e.page)},[s,r,i,a,function(e){i=e,n(2,i)}]}e("default",new class extends t{constructor(e){super(),n(this,e,Ga,Ha,a,{name:0,page:1})}}({target:document.body,props:{name:"MetabolicFeaturesIntraSubset",page:0}}))}}}));
