System.register(["./vendor/svelte-C_apNiMW.js","./vendor/carbon-components-svelte-CB8VNYQX.js","./vendor/carbon-icons-svelte-DkM4cpjm.js"],(function(e){"use strict";var t,n,a,s,r,l,i,o,c,p,_,d,m,g,f,u,$,A,S,b,h,C,L,E,v,R,T,y,x,N,M,P,I,w,G,B,F,H,U,O,D,K,V,z,k,q,j,X,W,Y,J,Z,Q,ee,te,ne,ae,se,re,le,ie,oe,ce,pe,_e,de,me,ge,fe,ue,$e,Ae,Se,be,he,Ce,Le,Ee,ve,Re,Te,ye,xe,Ne,Me,Pe,Ie,we,Ge,Be,Fe,He,Ue,Oe,De,Ke,Ve,ze,ke,qe,je,Xe,We,Ye,Je,Ze,Qe;return{setters:[function(e){t=e.S,n=e.i,a=e.s,s=e.a1,r=e.l,l=e.O,i=e.a,o=e.b,c=e.g,p=e.c,_=e.t,d=e.d,m=e.a5,g=e.B,f=e.e,u=e.G,$=e.H,A=e.I,S=e.y,b=e.K,h=e.T,C=e.U,L=e.L,E=e.D,v=e.M,R=e.z,T=e.Q,y=e.k,x=e.u,N=e.n,M=e.o,P=e.h,I=e.p,w=e.a0,G=e.f,B=e.j,F=e.A,H=e.m,U=e.q,O=e.aa,D=e.ab,K=e.x,V=e.W,z=e.ac,k=e.R,q=e.$,j=e.v,X=e.Y,W=e.Z,Y=e.F,J=e.ad,Z=e.w,Q=e.J,ee=e.r},function(e){te=e.S,ne=e.a,ae=e.b,se=e.H,re=e.c,le=e.T,ie=e.d,oe=e.e,ce=e.f,pe=e.g,_e=e.h,de=e.M,me=e.I,ge=e.i,fe=e.j,ue=e.k,$e=e.C,Ae=e.l,Se=e.m,be=e.G,he=e.R,Ce=e.n,Le=e.B,Ee=e.o,ve=e.A,Re=e.p,Te=e.D,ye=e.P,xe=e.q,Ne=e.r,Me=e.s,Pe=e.t,Ie=e.u},function(e){we=e.C,Ge=e.a,Be=e.D,Fe=e.A,He=e.S,Ue=e.I,Oe=e.b,De=e.c,Ke=e.V,Ve=e.d,ze=e.T,ke=e.G,qe=e.J,je=e.P,Xe=e.e,We=e.f,Ye=e.O,Je=e.g,Ze=e.h,Qe=e.i}],execute:function(){function et(e,t,n){const a=e.slice();return a[4]=t[n],a}function tt(e,t,n){const a=e.slice();return a[7]=t[n],a}function nt(e){let t,n;return t=new te({props:{href:`${e[1](e[4].page)}#${e[4].slug}`,class:e[2](e[4].page),title:e[4].text,text:e[4].text,expanded:!0,$$slots:{default:[rt]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){$(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.href=`${e[1](e[4].page)}#${e[4].slug}`),1&n&&(a.class=e[2](e[4].page)),1&n&&(a.title=e[4].text),1&n&&(a.text=e[4].text),1025&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){A(t,e)}}}function at(e){let t,n;return t=new ne({props:{href:`${e[1](e[4].page)}#${e[4].slug}`,class:e[2](e[4].page),title:e[4].text,text:e[4].text}}),{c(){u(t.$$.fragment)},m(e,a){$(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.href=`${e[1](e[4].page)}#${e[4].slug}`),1&n&&(a.class=e[2](e[4].page)),1&n&&(a.title=e[4].text),1&n&&(a.text=e[4].text),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){A(t,e)}}}function st(e){let t,n;return t=new ae({props:{href:`${e[1](e[7].page)}#${e[7].slug}`,class:e[2](e[7].page),title:e[7].text,text:e[7].text}}),{c(){u(t.$$.fragment)},m(e,a){$(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.href=`${e[1](e[7].page)}#${e[7].slug}`),1&n&&(a.class=e[2](e[7].page)),1&n&&(a.title=e[7].text),1&n&&(a.text=e[7].text),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){A(t,e)}}}function rt(e){let t,n,a=s(e[4].children),r=[];for(let t=0;t<a.length;t+=1)r[t]=st(tt(e,a,t));const l=e=>_(r[e],1,1,(()=>{r[e]=null}));return{c(){for(let e=0;e<r.length;e+=1)r[e].c();t=S()},m(e,a){for(let t=0;t<r.length;t+=1)r[t]&&r[t].m(e,a);i(e,t,a),n=!0},p(e,n){if(7&n){let i;for(a=s(e[4].children),i=0;i<a.length;i+=1){const s=tt(e,a,i);r[i]?(r[i].p(s,n),o(r[i],1)):(r[i]=st(s),r[i].c(),o(r[i],1),r[i].m(t.parentNode,t))}for(c(),i=a.length;i<r.length;i+=1)l(i);p()}},i(e){if(!n){for(let e=0;e<a.length;e+=1)o(r[e]);n=!0}},o(e){r=r.filter(Boolean);for(let e=0;e<r.length;e+=1)_(r[e]);n=!1},d(e){e&&d(t),m(r,e)}}}function lt(e){let t,n,a,s;const r=[at,nt],l=[];function m(e,t){return 0==e[4].children.length?0:1}return t=m(e),n=l[t]=r[t](e),{c(){n.c(),a=f()},m(e,n){l[t].m(e,n),i(e,a,n),s=!0},p(e,s){let i=t;t=m(e),t===i?l[t].p(e,s):(c(),_(l[i],1,1,(()=>{l[i]=null})),p(),n=l[t],n?n.p(e,s):(n=l[t]=r[t](e),n.c()),o(n,1),n.m(a.parentNode,a))},i(e){s||(o(n),s=!0)},o(e){_(n),s=!1},d(e){e&&d(a),l[t].d(e)}}}function it(e){let t,n,a=s(e[0]),g=[];for(let t=0;t<a.length;t+=1)g[t]=lt(et(e,a,t));const f=e=>_(g[e],1,1,(()=>{g[e]=null}));return{c(){t=r("ul");for(let e=0;e<g.length;e+=1)g[e].c();l(t,"class","bx--side-nav__items pipen-report-sidenav svelte-12gpq3")},m(e,a){i(e,t,a);for(let e=0;e<g.length;e+=1)g[e]&&g[e].m(t,null);n=!0},p(e,[n]){if(7&n){let r;for(a=s(e[0]),r=0;r<a.length;r+=1){const s=et(e,a,r);g[r]?(g[r].p(s,n),o(g[r],1)):(g[r]=lt(s),g[r].c(),o(g[r],1),g[r].m(t,null))}for(c(),r=a.length;r<g.length;r+=1)f(r);p()}},i(e){if(!n){for(let e=0;e<a.length;e+=1)o(g[e]);n=!0}},o(e){g=g.filter(Boolean);for(let e=0;e<g.length;e+=1)_(g[e]);n=!1},d(e){e&&d(t),m(g,e)}}}function ot(e,t,n){let{toc:a}=t,{page:s}=t;return g((()=>{let e,t=new URL(window.location.href);e=t.hash.length>0?document.querySelector(`[href="${t.hash}"]`):document.querySelector(".toc-current-page"),e.scrollIntoView({behavior:"smooth"})})),e.$$set=e=>{"toc"in e&&n(0,a=e.toc),"page"in e&&n(3,s=e.page)},[a,e=>{if(s===e)return"";let t=new URL(window.location.href);return t.hash="",t.searchParams.set("page",e),t.toString()},e=>e!=s?"toc-noncurrent-page":"toc-current-page",s]}class ct extends t{constructor(e){super(),n(this,e,ot,it,a,{toc:0,page:3})}}function pt(e,t,n){const a=e.slice();return a[7]=t[n],a}function _t(e,t,n){const a=e.slice();return a[10]=t[n],a}function dt(e){let t,n;return t=new oe({props:{isOpen:Mt,transition:e[5],$$slots:{default:[Et]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){$(t,e,a),n=!0},p(e,n){const a={};8197&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){A(t,e)}}}function mt(e){let t;return{c(){t=L("Go to home:")},m(e,n){i(e,t,n)},d(e){e&&d(t)}}}function gt(e){let t;return{c(){t=L(e[0])},m(e,n){i(e,t,n)},p(e,n){1&n&&v(t,e[0])},d(e){e&&d(t)}}}function ft(e){let t;return{c(){t=L("Switch to process:")},m(e,n){i(e,t,n)},d(e){e&&d(t)}}}function ut(e){let t,n;return t=new _e({props:{title:e[7].name,href:`?proc=${e[7].name}`,$$slots:{default:[At]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){$(t,e,a),n=!0},p(e,n){const a={};4&n&&(a.title=e[7].name),4&n&&(a.href=`?proc=${e[7].name}`),8196&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){A(t,e)}}}function $t(e){let t,n,a,r;t=new _e({props:{class:"procgroup",$$slots:{default:[St]},$$scope:{ctx:e}}});let l=s(e[7].procs),g=[];for(let t=0;t<l.length;t+=1)g[t]=ht(_t(e,l,t));const b=e=>_(g[e],1,1,(()=>{g[e]=null}));return{c(){u(t.$$.fragment),n=S();for(let e=0;e<g.length;e+=1)g[e].c();a=f()},m(e,s){$(t,e,s),i(e,n,s);for(let t=0;t<g.length;t+=1)g[t]&&g[t].m(e,s);i(e,a,s),r=!0},p(e,n){const r={};if(8196&n&&(r.$$scope={dirty:n,ctx:e}),t.$set(r),4&n){let t;for(l=s(e[7].procs),t=0;t<l.length;t+=1){const s=_t(e,l,t);g[t]?(g[t].p(s,n),o(g[t],1)):(g[t]=ht(s),g[t].c(),o(g[t],1),g[t].m(a.parentNode,a))}for(c(),t=l.length;t<g.length;t+=1)b(t);p()}},i(e){if(!r){o(t.$$.fragment,e);for(let e=0;e<l.length;e+=1)o(g[e]);r=!0}},o(e){_(t.$$.fragment,e),g=g.filter(Boolean);for(let e=0;e<g.length;e+=1)_(g[e]);r=!1},d(e){e&&(d(n),d(a)),A(t,e),m(g,e)}}}function At(e){let t,n,a=e[7].name+"";return{c(){t=L(a),n=S()},m(e,a){i(e,t,a),i(e,n,a)},p(e,n){4&n&&a!==(a=e[7].name+"")&&v(t,a)},d(e){e&&(d(t),d(n))}}}function St(e){let t,n=e[7].name+"";return{c(){t=L(n)},m(e,n){i(e,t,n)},p(e,a){4&a&&n!==(n=e[7].name+"")&&v(t,n)},d(e){e&&d(t)}}}function bt(e){let t,n,a,s=e[10].name+"";return{c(){t=L("- "),n=L(s),a=S()},m(e,s){i(e,t,s),i(e,n,s),i(e,a,s)},p(e,t){4&t&&s!==(s=e[10].name+"")&&v(n,s)},d(e){e&&(d(t),d(n),d(a))}}}function ht(e){let t,n;return t=new _e({props:{title:e[10].name,class:"procgroup-proc",href:`?proc=${e[10].name}`,$$slots:{default:[bt]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){$(t,e,a),n=!0},p(e,n){const a={};4&n&&(a.title=e[10].name),4&n&&(a.href=`?proc=${e[10].name}`),8196&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){A(t,e)}}}function Ct(e){let t,n,a,s;const r=[$t,ut],l=[];function m(e,t){return e[7].procs?0:1}return t=m(e),n=l[t]=r[t](e),{c(){n.c(),a=f()},m(e,n){l[t].m(e,n),i(e,a,n),s=!0},p(e,s){let i=t;t=m(e),t===i?l[t].p(e,s):(c(),_(l[i],1,1,(()=>{l[i]=null})),p(),n=l[t],n?n.p(e,s):(n=l[t]=r[t](e),n.c()),o(n,1),n.m(a.parentNode,a))},i(e){s||(o(n),s=!0)},o(e){_(n),s=!1},d(e){e&&d(a),l[t].d(e)}}}function Lt(e){let t,n,a,r,l,g,b,h;t=new pe({props:{$$slots:{default:[mt]},$$scope:{ctx:e}}}),a=new _e({props:{href:"index.html",$$slots:{default:[gt]},$$scope:{ctx:e}}}),l=new pe({props:{$$slots:{default:[ft]},$$scope:{ctx:e}}});let C=s(e[2]),L=[];for(let t=0;t<C.length;t+=1)L[t]=Ct(pt(e,C,t));const E=e=>_(L[e],1,1,(()=>{L[e]=null}));return{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment),r=S(),u(l.$$.fragment),g=S();for(let e=0;e<L.length;e+=1)L[e].c();b=f()},m(e,s){$(t,e,s),i(e,n,s),$(a,e,s),i(e,r,s),$(l,e,s),i(e,g,s);for(let t=0;t<L.length;t+=1)L[t]&&L[t].m(e,s);i(e,b,s),h=!0},p(e,n){const r={};8192&n&&(r.$$scope={dirty:n,ctx:e}),t.$set(r);const i={};8193&n&&(i.$$scope={dirty:n,ctx:e}),a.$set(i);const _={};if(8192&n&&(_.$$scope={dirty:n,ctx:e}),l.$set(_),4&n){let t;for(C=s(e[2]),t=0;t<C.length;t+=1){const a=pt(e,C,t);L[t]?(L[t].p(a,n),o(L[t],1)):(L[t]=Ct(a),L[t].c(),o(L[t],1),L[t].m(b.parentNode,b))}for(c(),t=C.length;t<L.length;t+=1)E(t);p()}},i(e){if(!h){o(t.$$.fragment,e),o(a.$$.fragment,e),o(l.$$.fragment,e);for(let e=0;e<C.length;e+=1)o(L[e]);h=!0}},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),_(l.$$.fragment,e),L=L.filter(Boolean);for(let e=0;e<L.length;e+=1)_(L[e]);h=!1},d(e){e&&(d(n),d(r),d(g),d(b)),A(t,e),A(a,e),A(l,e),m(L,e)}}}function Et(e){let t,n;return t=new ce({props:{$$slots:{default:[Lt]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){$(t,e,a),n=!0},p(e,n){const a={};8197&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){A(t,e)}}}function vt(e){let t,n,a,s;t=new ie({props:{render:"select",theme:"g10",persist:!0,persistKey:"__pipen-report-theme",select:{themes:["g10","white","g80","g90","g100"],labelText:"Theme",inline:!0}}});let r=e[2].length>0&&dt(e);return{c(){u(t.$$.fragment),n=S(),r&&r.c(),a=f()},m(e,l){$(t,e,l),i(e,n,l),r&&r.m(e,l),i(e,a,l),s=!0},p(e,t){e[2].length>0?r?(r.p(e,t),4&t&&o(r,1)):(r=dt(e),r.c(),o(r,1),r.m(a.parentNode,a)):r&&(c(),_(r,1,1,(()=>{r=null})),p())},i(e){s||(o(t.$$.fragment,e),o(r),s=!0)},o(e){_(t.$$.fragment,e),_(r),s=!1},d(e){e&&(d(n),d(a)),A(t,e),r&&r.d(e)}}}function Rt(e){let t,n;return t=new re({props:{class:"header-util",$$slots:{default:[vt]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){$(t,e,a),n=!0},p(e,n){const a={};8197&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){A(t,e)}}}function Tt(e){let t;return{c(){t=L(e[4])},m(e,n){i(e,t,n)},p(e,n){16&n&&v(t,e[4])},d(e){e&&d(t)}}}function yt(e){let t,n,a,s,l=null!==e[4]&&Tt(e);return{c(){t=r("h1"),n=L(e[3]),a=S(),l&&l.c(),s=f()},m(e,r){i(e,t,r),E(t,n),i(e,a,r),l&&l.m(e,r),i(e,s,r)},p(e,t){8&t&&v(n,e[3]),null!==e[4]?l?l.p(e,t):(l=Tt(e),l.c(),l.m(s.parentNode,s)):l&&(l.d(1),l=null)},d(e){e&&(d(t),d(a),d(s)),l&&l.d(e)}}}function xt(e){let t,n;return t=new le({props:{class:"header-tile",slot:"platform",$$slots:{default:[yt]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){$(t,e,a),n=!0},p(e,n){const a={};8216&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){A(t,e)}}}function Nt(e){let t,n,a;function s(t){e[6](t)}let r={$$slots:{platform:[xt],default:[Rt]},$$scope:{ctx:e}};return void 0!==e[1]&&(r.isSideNavOpen=e[1]),t=new se({props:r}),b.push((()=>h(t,"isSideNavOpen",s))),{c(){u(t.$$.fragment)},m(e,n){$(t,e,n),a=!0},p(e,[a]){const s={};8221&a&&(s.$$scope={dirty:a,ctx:e}),!n&&2&a&&(n=!0,s.isSideNavOpen=e[1],C((()=>n=!1))),t.$set(s)},i(e){a||(o(t.$$.fragment,e),a=!0)},o(e){_(t.$$.fragment,e),a=!1},d(e){A(t,e)}}}let Mt=!1;function Pt(e,t,n){let{entries:a}=t,{logo:s}=t,{logotext:r=null}=t,{pipeline_name:l}=t,{isSideNavOpen:i}=t;return l=l||s,e.$$set=e=>{"entries"in e&&n(2,a=e.entries),"logo"in e&&n(3,s=e.logo),"logotext"in e&&n(4,r=e.logotext),"pipeline_name"in e&&n(0,l=e.pipeline_name),"isSideNavOpen"in e&&n(1,i=e.isSideNavOpen)},[l,i,a,s,r,{duration:200},function(e){i=e,n(1,i)}]}class It extends t{constructor(e){super(),n(this,e,Pt,Nt,a,{entries:2,logo:3,logotext:4,pipeline_name:0,isSideNavOpen:1})}}function wt(e){let t,n,a;function s(t){e[3](t)}let r={modalHeading:"Running Information",passiveModal:!0,hasScrollingContent:!0,class:"runinfo",$$slots:{default:[Ot]},$$scope:{ctx:e}};return void 0!==e[2]&&(r.open=e[2]),t=new de({props:r}),b.push((()=>h(t,"open",s))),{c(){u(t.$$.fragment)},m(e,n){$(t,e,n),a=!0},p(e,a){const s={};34&a&&(s.$$scope={dirty:a,ctx:e}),!n&&4&a&&(n=!0,s.open=e[2],C((()=>n=!1))),t.$set(s)},i(e){a||(o(t.$$.fragment,e),a=!0)},o(e){_(t.$$.fragment,e),a=!1},d(e){A(t,e)}}}function Gt(e){let t,n,a,s,r,l;return t=new fe({props:{label:"Session Information"}}),a=new fe({props:{label:"Time Spent"}}),r=new fe({props:{label:"Device Information"}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment),s=S(),u(r.$$.fragment)},m(e,o){$(t,e,o),i(e,n,o),$(a,e,o),i(e,s,o),$(r,e,o),l=!0},p:T,i(e){l||(o(t.$$.fragment,e),o(a.$$.fragment,e),o(r.$$.fragment,e),l=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),_(r.$$.fragment,e),l=!1},d(e){e&&(d(n),d(s)),A(t,e),A(a,e),A(r,e)}}}function Bt(e){let t,n;return t=new $e({props:{type:"multi",code:e[1].session}}),{c(){u(t.$$.fragment)},m(e,a){$(t,e,a),n=!0},p(e,n){const a={};2&n&&(a.code=e[1].session),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){A(t,e)}}}function Ft(e){let t,n;return t=new $e({props:{type:"multi",code:e[1].time}}),{c(){u(t.$$.fragment)},m(e,a){$(t,e,a),n=!0},p(e,n){const a={};2&n&&(a.code=e[1].time),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){A(t,e)}}}function Ht(e){let t,n;return t=new $e({props:{type:"multi",code:e[1].device}}),{c(){u(t.$$.fragment)},m(e,a){$(t,e,a),n=!0},p(e,n){const a={};2&n&&(a.code=e[1].device),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){A(t,e)}}}function Ut(e){let t,n,a,s,r,l;return t=new ue({props:{$$slots:{default:[Bt]},$$scope:{ctx:e}}}),a=new ue({props:{$$slots:{default:[Ft]},$$scope:{ctx:e}}}),r=new ue({props:{$$slots:{default:[Ht]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment),s=S(),u(r.$$.fragment)},m(e,o){$(t,e,o),i(e,n,o),$(a,e,o),i(e,s,o),$(r,e,o),l=!0},p(e,n){const s={};34&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const l={};34&n&&(l.$$scope={dirty:n,ctx:e}),a.$set(l);const i={};34&n&&(i.$$scope={dirty:n,ctx:e}),r.$set(i)},i(e){l||(o(t.$$.fragment,e),o(a.$$.fragment,e),o(r.$$.fragment,e),l=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),_(r.$$.fragment,e),l=!1},d(e){e&&(d(n),d(s)),A(t,e),A(a,e),A(r,e)}}}function Ot(e){let t,n,a,s;return t=new me({props:{kind:"warning",lowContrast:!0,hideCloseButton:!0,title:"Showing the first job only. Check the workdir for information of other jobs if any."}}),a=new ge({props:{$$slots:{content:[Ut],default:[Gt]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){$(t,e,r),i(e,n,r),$(a,e,r),s=!0},p(e,t){const n={};34&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),A(t,e),A(a,e)}}}function Dt(e){let t,n,a;return{c(){t=r("a"),t.textContent="> Running Information",l(t,"href","javascript:void(0);")},m(s,r){i(s,t,r),n||(a=R(t,"click",e[4]),n=!0)},p:T,d(e){e&&d(t),n=!1,a()}}}function Kt(e){let t,n,a,s,m,g,f,u,$=e[1]&&wt(e),A=e[1]&&Dt(e);return{c(){$&&$.c(),t=S(),n=r("div"),a=r("div"),A&&A.c(),s=S(),m=r("div"),g=L("Powered by "),f=r("span"),l(a,"class","runinfo"),l(f,"class","pipen-report-footer-versions"),l(m,"class","versions"),l(n,"class","pipen-report-footer bx--grid")},m(r,l){$&&$.m(r,l),i(r,t,l),i(r,n,l),E(n,a),A&&A.m(a,null),E(n,s),E(n,m),E(m,g),E(m,f),f.innerHTML=e[0],u=!0},p(e,[n]){e[1]?$?($.p(e,n),2&n&&o($,1)):($=wt(e),$.c(),o($,1),$.m(t.parentNode,t)):$&&(c(),_($,1,1,(()=>{$=null})),p()),e[1]?A?A.p(e,n):(A=Dt(e),A.c(),A.m(a,null)):A&&(A.d(1),A=null),(!u||1&n)&&(f.innerHTML=e[0])},i(e){u||(o($),u=!0)},o(e){_($),u=!1},d(e){e&&(d(t),d(n)),$&&$.d(e),A&&A.d()}}}function Vt(e,t,n){let{versions:a}=t,{runinfo:s}=t,r=!1;return e.$$set=e=>{"versions"in e&&n(0,a=e.versions),"runinfo"in e&&n(1,s=e.runinfo)},[a,s,r,function(e){r=e,n(2,r)},()=>{n(2,r=!0)}]}class zt extends t{constructor(e){super(),n(this,e,Vt,Kt,a,{versions:0,runinfo:1})}}function kt(e){let t,n,a;function s(t){e[12](t)}let r={$$slots:{default:[qt]},$$scope:{ctx:e}};return void 0!==e[0]&&(r.isOpen=e[0]),t=new Se({props:r}),b.push((()=>h(t,"isOpen",s))),{c(){u(t.$$.fragment)},m(e,n){$(t,e,n),a=!0},p(e,a){const s={};8576&a&&(s.$$scope={dirty:a,ctx:e}),!n&&1&a&&(n=!0,s.isOpen=e[0],C((()=>n=!1))),t.$set(s)},i(e){a||(o(t.$$.fragment,e),a=!0)},o(e){_(t.$$.fragment,e),a=!1},d(e){A(t,e)}}}function qt(e){let t,n;return t=new ct({props:{toc:e[7],page:e[8]}}),{c(){u(t.$$.fragment)},m(e,a){$(t,e,a),n=!0},p(e,n){const a={};128&n&&(a.toc=e[7]),256&n&&(a.page=e[8]),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){A(t,e)}}}function jt(e){let t;const n=e[10].default,a=y(n,e,e[13],null);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||8192&s)&&x(a,n,e,e[13],t?M(n,e[13],s,null):N(e[13]),null)},i(e){t||(o(a,e),t=!0)},o(e){_(a,e),t=!1},d(e){a&&a.d(e)}}}function Xt(e){let t,n;return t=new Ce({props:{$$slots:{default:[jt]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){$(t,e,a),n=!0},p(e,n){const a={};8192&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){A(t,e)}}}function Wt(e){let t,n;return t=new he({props:{$$slots:{default:[Xt]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){$(t,e,a),n=!0},p(e,n){const a={};8192&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){A(t,e)}}}function Yt(e){let t,n,a,s;return t=new be({props:{$$slots:{default:[Wt]},$$scope:{ctx:e}}}),a=new zt({props:{versions:e[3],runinfo:e[9]}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){$(t,e,r),i(e,n,r),$(a,e,r),s=!0},p(e,n){const s={};8192&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};8&n&&(r.versions=e[3]),512&n&&(r.runinfo=e[9]),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),A(t,e),A(a,e)}}}function Jt(e){let t,n,a,s,r,l;function m(t){e[11](t)}let g={logo:e[1],logotext:e[2],entries:e[4],pipeline_name:e[5]};void 0!==e[0]&&(g.isSideNavOpen=e[0]),t=new It({props:g}),b.push((()=>h(t,"isSideNavOpen",m)));let f=e[6]&&e[7].length>0&&kt(e);return r=new Ae({props:{$$slots:{default:[Yt]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment),a=S(),f&&f.c(),s=S(),u(r.$$.fragment)},m(e,n){$(t,e,n),i(e,a,n),f&&f.m(e,n),i(e,s,n),$(r,e,n),l=!0},p(e,[a]){const l={};2&a&&(l.logo=e[1]),4&a&&(l.logotext=e[2]),16&a&&(l.entries=e[4]),32&a&&(l.pipeline_name=e[5]),!n&&1&a&&(n=!0,l.isSideNavOpen=e[0],C((()=>n=!1))),t.$set(l),e[6]&&e[7].length>0?f?(f.p(e,a),192&a&&o(f,1)):(f=kt(e),f.c(),o(f,1),f.m(s.parentNode,s)):f&&(c(),_(f,1,1,(()=>{f=null})),p());const i={};8712&a&&(i.$$scope={dirty:a,ctx:e}),r.$set(i)},i(e){l||(o(t.$$.fragment,e),o(f),o(r.$$.fragment,e),l=!0)},o(e){_(t.$$.fragment,e),_(f),_(r.$$.fragment,e),l=!1},d(e){e&&(d(a),d(s)),A(t,e),f&&f.d(e),A(r,e)}}}function Zt(e,t,n){let a,{$$slots:s={},$$scope:r}=t,{logo:l}=t,{logotext:i=null}=t,{versions:o}=t,{entries:c}=t,{pipeline_name:p}=t,{report_toc:_}=t,{toc:d}=t,{page:m}=t,{isSideNavOpen:f}=t,{runinfo:u}=t;return g((()=>{const e=new URL(window.location.href);let t;t=e.hash.length>0?document.querySelector(`li:has(>[href="${e.hash}"])`):document.querySelector("li:has(>.toc-current-page)"),t&&(t.classList.add("toc-active"),t.scrollIntoView());const n=document.querySelectorAll(".pipen-report-toc-anchor");a=new IntersectionObserver((e=>{e.forEach((e=>{const t=e.target.getAttribute("id"),n=document.querySelector(`li:has(>[href="#${t}"])`);n&&(e.isIntersecting?n.classList.add("toc-active"):n.classList.remove("toc-active"))}))}),{root:null,rootMargin:"0px",threshold:.5}),n.forEach((e=>{a.observe(e)}))})),e.$$set=e=>{"logo"in e&&n(1,l=e.logo),"logotext"in e&&n(2,i=e.logotext),"versions"in e&&n(3,o=e.versions),"entries"in e&&n(4,c=e.entries),"pipeline_name"in e&&n(5,p=e.pipeline_name),"report_toc"in e&&n(6,_=e.report_toc),"toc"in e&&n(7,d=e.toc),"page"in e&&n(8,m=e.page),"isSideNavOpen"in e&&n(0,f=e.isSideNavOpen),"runinfo"in e&&n(9,u=e.runinfo),"$$scope"in e&&n(13,r=e.$$scope)},[f,l,i,o,c,p,_,d,m,u,s,function(e){f=e,n(0,f)},function(e){f=e,n(0,f)},r]}class Qt extends t{constructor(e){super(),n(this,e,Zt,Jt,a,{logo:1,logotext:2,versions:3,entries:4,pipeline_name:5,report_toc:6,toc:7,page:8,isSideNavOpen:0,runinfo:9})}}function en(e){let t;return{c(){t=L(e[2])},m(e,n){i(e,t,n)},p:T,d(e){e&&d(t)}}}function tn(e){let t,n,a,s;return n=new Le({props:{icon:e[1],kind:"tertiary",href:e[3](),$$slots:{default:[en]},$$scope:{ctx:e}}}),{c(){t=r("div"),u(n.$$.fragment),l(t,"class",a=`pagenav ${e[0]}`)},m(e,a){i(e,t,a),$(n,t,null),s=!0},p(e,[r]){const i={};16&r&&(i.$$scope={dirty:r,ctx:e}),n.$set(i),(!s||1&r&&a!==(a=`pagenav ${e[0]}`))&&l(t,"class",a)},i(e){s||(o(n.$$.fragment,e),s=!0)},o(e){_(n.$$.fragment,e),s=!1},d(e){e&&d(t),A(n)}}}function nn(e,t,n){let{dir:a}=t,s="up"===a?we:Ge,r="up"===a?"Go to previous page":"Go to next page";return e.$$set=e=>{"dir"in e&&n(0,a=e.dir)},[a,s,r,()=>{let e=new URL(window.location.href);e.hash="";const t=parseInt(e.searchParams.get("page")||0),n="up"===a?t-1:t+1;return 0===n?e.searchParams.delete("page"):e.searchParams.set("page",n),e.toString()}]}class an extends t{constructor(e){super(),n(this,e,nn,tn,a,{dir:0})}}function sn(e){let t,n;const a=[{tooltipText:e[0]},{icon:e[1]},{direction:"bottom"},{align:"start"},e[3]];let s={};for(let e=0;e<a.length;e+=1)s=P(s,a[e]);return t=new Ee({props:s}),t.$on("click",e[2]),{c(){u(t.$$.fragment)},m(e,a){$(t,e,a),n=!0},p(e,[n]){const s=11&n?I(a,[1&n&&{tooltipText:e[0]},2&n&&{icon:e[1]},a[2],a[3],8&n&&w(e[3])]):{};t.$set(s)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){A(t,e)}}}function rn(e,t,n){const a=["src","tip","icon"];let s,r=G(t,a),{src:l=""}=t,{tip:i=""}=t,{icon:o="auto"}=t,c=l.split(".").at(-1).toLocaleLowerCase();if("string"==typeof o)switch("auto"===o&&(o=c),o){case"pdf":case"DocumentPdf":s=We;break;case"html":case"htm":case"ChartLineData":s=Xe;break;case"png":case"Png":s=je;break;case"jpg":case"jpeg":case"Jpg":s=qe;break;case"gif":case"Gif":s=ke;break;case"tif":case"tiff":case"Tif":s=ze;break;case"eps":case"DatabaseElastic":s=Ve;break;case"mp4":case"avi":case"mov":case"mkv":case"webm":case"VideoPlayer":s=Ke;break;case"Image":s=De;break;case"Data":case"DataVolume":s=Oe;break;case"Code":case"IbmWatsonxCodeAssistantForZRefactor":s=Ue;break;case"svg":case"Svg":s=He;break;case"zip":case"tar":case"gz":case"bz2":case"7z":case"rar":case"xz":case"Archive":s=Fe;break;default:s=Be}const p={...r,class:[r.class,"pipen-report-image-download"].filter(Boolean).join(" ")};return e.$$set=e=>{t=P(P({},t),B(e)),n(7,r=G(t,a)),"src"in e&&n(5,l=e.src),"tip"in e&&n(0,i=e.tip),"icon"in e&&n(4,o=e.icon)},[i,s,()=>{const e=document.createElement("a");e.href=l,e.download=l.split("/").pop(),e.click()},p,o,l]}class ln extends t{constructor(e){super(),n(this,e,rn,sn,a,{src:5,tip:0,icon:4})}}const{Boolean:on}=O;function cn(e,t,n){const a=e.slice();return a[5]=t[n].src,a[6]=t[n].tip,a[7]=t[n].icon,a[9]=n,a}function pn(e){let t,n,a,l,g=s(e[0]),f=[];for(let t=0;t<g.length;t+=1)f[t]=_n(cn(e,g,t));const u=e=>_(f[e],1,1,(()=>{f[e]=null}));let $=[e[1],{role:"button"},{tabindex:"0"}],A={};for(let e=0;e<$.length;e+=1)A=P(A,$[e]);return{c(){t=r("div");for(let e=0;e<f.length;e+=1)f[e].c();H(t,A),U(t,"svelte-14lcnln",!0)},m(s,r){i(s,t,r);for(let e=0;e<f.length;e+=1)f[e]&&f[e].m(t,null);n=!0,a||(l=R(t,"mouseenter",e[3]),a=!0)},p(e,n){if(1&n){let a;for(g=s(e[0]),a=0;a<g.length;a+=1){const s=cn(e,g,a);f[a]?(f[a].p(s,n),o(f[a],1)):(f[a]=_n(s),f[a].c(),o(f[a],1),f[a].m(t,null))}for(c(),a=g.length;a<f.length;a+=1)u(a);p()}U(t,"svelte-14lcnln",!0)},i(e){if(!n){for(let e=0;e<g.length;e+=1)o(f[e]);n=!0}},o(e){f=f.filter(on);for(let e=0;e<f.length;e+=1)_(f[e]);n=!1},d(e){e&&d(t),m(f,e),a=!1,l()}}}function _n(e){let t,n;return t=new ln({props:{src:e[5],tip:e[6],icon:e[7]}}),{c(){u(t.$$.fragment)},m(e,a){$(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.src=e[5]),1&n&&(a.tip=e[6]),1&n&&(a.icon=e[7]),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){A(t,e)}}}function dn(e){let t,n,a=e[0].length>0&&pn(e);return{c(){a&&a.c(),t=f()},m(e,s){a&&a.m(e,s),i(e,t,s),n=!0},p(e,[n]){e[0].length>0?a?(a.p(e,n),1&n&&o(a,1)):(a=pn(e),a.c(),o(a,1),a.m(t.parentNode,t)):a&&(c(),_(a,1,1,(()=>{a=null})),p())},i(e){n||(o(a),n=!0)},o(e){_(a),n=!1},d(e){e&&d(t),a&&a.d(e)}}}function mn(e,t,n){const a=["data"];let s=G(t,a),{data:r=[]}=t;Array.isArray(r)||(r=[r]),r=r.map((e=>"string"==typeof e?{src:e,tip:`Download the ${e.split(".").at(-1)} format.`,icon:"auto"}:(e.tip=e.tip||`Download the ${e.src.split(".").at(-1)} format.`,e.icon=e.icon||"auto",e)));const l={...s,class:[s.class,"pipen-report-image-download-list"].filter(Boolean).join(" ")},i=F();return e.$$set=e=>{t=P(P({},t),B(e)),n(4,s=G(t,a)),"data"in e&&n(0,r=e.data)},[r,l,i,()=>i("mouseenter")]}class gn extends t{constructor(e){super(),n(this,e,mn,dn,a,{data:0})}}const fn=e=>({}),un=e=>({}),$n=e=>({}),An=e=>({}),Sn=e=>({}),bn=e=>({}),hn=e=>({}),Cn=e=>({});function Ln(e){let t,n;return t=new ve({props:{ratio:e[6],$$slots:{default:[xn]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){$(t,e,a),n=!0},p(e,n){const a={};64&n&&(a.ratio=e[6]),33556159&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){A(t,e)}}}function En(e){let t,n,a,s,r=e[2]&&Nn(e),l=e[0]&&Mn(e),m=e[1]&&In(e);return{c(){r&&r.c(),t=S(),l&&l.c(),n=S(),m&&m.c(),a=f()},m(e,o){r&&r.m(e,o),i(e,t,o),l&&l.m(e,o),i(e,n,o),m&&m.m(e,o),i(e,a,o),s=!0},p(e,s){e[2]?r?(r.p(e,s),4&s&&o(r,1)):(r=Nn(e),r.c(),o(r,1),r.m(t.parentNode,t)):r&&(c(),_(r,1,1,(()=>{r=null})),p()),e[0]?l?(l.p(e,s),1&s&&o(l,1)):(l=Mn(e),l.c(),o(l,1),l.m(n.parentNode,n)):l&&(c(),_(l,1,1,(()=>{l=null})),p()),e[1]?m?(m.p(e,s),2&s&&o(m,1)):(m=In(e),m.c(),o(m,1),m.m(a.parentNode,a)):m&&(c(),_(m,1,1,(()=>{m=null})),p())},i(e){s||(o(r),o(l),o(m),s=!0)},o(e){_(r),_(l),_(m),s=!1},d(e){e&&(d(t),d(n),d(a)),r&&r.d(e),l&&l.d(e),m&&m.d(e)}}}function vn(e){let t;const n=e[15].loading,a=y(n,e,e[25],An);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||33554432&s)&&x(a,n,e,e[25],t?M(n,e[25],s,$n):N(e[25]),An)},i(e){t||(o(a,e),t=!0)},o(e){_(a,e),t=!1},d(e){a&&a.d(e)}}}function Rn(e){let t,n,a,s,l,m,g,u,$=[e[12],{src:n=e[3]},{alt:e[4]},{title:"Click to zoom in ..."}],A={};for(let e=0;e<$.length;e+=1)A=P(A,$[e]);let b=e[9]&&Tn(e);return{c(){t=r("img"),s=S(),b&&b.c(),l=f(),H(t,A)},m(n,a){i(n,t,a),i(n,s,a),b&&b.m(n,a),i(n,l,a),m=!0,g||(u=[R(t,"click",e[11]),R(t,"mouseenter",e[19]),R(t,"mouseleave",e[20])],g=!0)},p(a,s){H(t,A=I($,[(e=a)[12],(!m||8&s&&!D(t.src,n=e[3]))&&{src:n},(!m||16&s)&&{alt:e[4]},{title:"Click to zoom in ..."}])),e[9]?b?(b.p(e,s),512&s&&o(b,1)):(b=Tn(e),b.c(),o(b,1),b.m(l.parentNode,l)):b&&(c(),_(b,1,1,(()=>{b=null})),p())},i(n){m||(n&&K((()=>{m&&(a||(a=V(t,z,{duration:e[7]?Bn:0},!0)),a.run(1))})),o(b),m=!0)},o(n){n&&(a||(a=V(t,z,{duration:e[7]?Bn:0},!1)),a.run(0)),_(b),m=!1},d(e){e&&(d(t),d(s),d(l)),e&&a&&a.end(),b&&b.d(e),g=!1,k(u)}}}function Tn(e){let t,n;return t=new gn({props:{data:e[5]}}),t.$on("mouseenter",e[21]),{c(){u(t.$$.fragment)},m(e,a){$(t,e,a),n=!0},p(e,n){const a={};32&n&&(a.data=e[5]),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){A(t,e)}}}function yn(e){let t;const n=e[15].error,a=y(n,e,e[25],un);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||33554432&s)&&x(a,n,e,e[25],t?M(n,e[25],s,fn):N(e[25]),un)},i(e){t||(o(a,e),t=!0)},o(e){_(a,e),t=!1},d(e){a&&a.d(e)}}}function xn(e){let t,n,a,s,r=e[2]&&vn(e),l=e[0]&&Rn(e),m=e[1]&&yn(e);return{c(){r&&r.c(),t=S(),l&&l.c(),n=S(),m&&m.c(),a=f()},m(e,o){r&&r.m(e,o),i(e,t,o),l&&l.m(e,o),i(e,n,o),m&&m.m(e,o),i(e,a,o),s=!0},p(e,s){e[2]?r?(r.p(e,s),4&s&&o(r,1)):(r=vn(e),r.c(),o(r,1),r.m(t.parentNode,t)):r&&(c(),_(r,1,1,(()=>{r=null})),p()),e[0]?l?(l.p(e,s),1&s&&o(l,1)):(l=Rn(e),l.c(),o(l,1),l.m(n.parentNode,n)):l&&(c(),_(l,1,1,(()=>{l=null})),p()),e[1]?m?(m.p(e,s),2&s&&o(m,1)):(m=yn(e),m.c(),o(m,1),m.m(a.parentNode,a)):m&&(c(),_(m,1,1,(()=>{m=null})),p())},i(e){s||(o(r),o(l),o(m),s=!0)},o(e){_(r),_(l),_(m),s=!1},d(e){e&&(d(t),d(n),d(a)),r&&r.d(e),l&&l.d(e),m&&m.d(e)}}}function Nn(e){let t;const n=e[15].loading,a=y(n,e,e[25],Cn);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||33554432&s)&&x(a,n,e,e[25],t?M(n,e[25],s,hn):N(e[25]),Cn)},i(e){t||(o(a,e),t=!0)},o(e){_(a,e),t=!1},d(e){a&&a.d(e)}}}function Mn(e){let t,n,a,s,l,m,g,u,$=[e[12],{src:n=e[3]},{alt:e[4]},{title:"Click to zoom in ..."}],A={};for(let e=0;e<$.length;e+=1)A=P(A,$[e]);let b=e[9]&&Pn(e);return{c(){t=r("img"),s=S(),b&&b.c(),l=f(),H(t,A)},m(n,a){i(n,t,a),i(n,s,a),b&&b.m(n,a),i(n,l,a),m=!0,g||(u=[R(t,"click",e[11]),R(t,"mouseenter",e[16]),R(t,"mouseleave",e[17])],g=!0)},p(a,s){H(t,A=I($,[(e=a)[12],(!m||8&s&&!D(t.src,n=e[3]))&&{src:n},(!m||16&s)&&{alt:e[4]},{title:"Click to zoom in ..."}])),e[9]?b?(b.p(e,s),512&s&&o(b,1)):(b=Pn(e),b.c(),o(b,1),b.m(l.parentNode,l)):b&&(c(),_(b,1,1,(()=>{b=null})),p())},i(n){m||(n&&K((()=>{m&&(a||(a=V(t,z,{duration:e[7]?Bn:0},!0)),a.run(1))})),o(b),m=!0)},o(n){n&&(a||(a=V(t,z,{duration:e[7]?Bn:0},!1)),a.run(0)),_(b),m=!1},d(e){e&&(d(t),d(s),d(l)),e&&a&&a.end(),b&&b.d(e),g=!1,k(u)}}}function Pn(e){let t,n;return t=new gn({props:{data:e[5]}}),t.$on("mouseenter",e[18]),{c(){u(t.$$.fragment)},m(e,a){$(t,e,a),n=!0},p(e,n){const a={};32&n&&(a.data=e[5]),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){A(t,e)}}}function In(e){let t;const n=e[15].error,a=y(n,e,e[25],bn);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||33554432&s)&&x(a,n,e,e[25],t?M(n,e[25],s,Sn):N(e[25]),bn)},i(e){t||(o(a,e),t=!0)},o(e){_(a,e),t=!1},d(e){a&&a.d(e)}}}function wn(e){let t,n,a,s;return{c(){t=r("img"),l(t,"class","pipen-report-image-lg"),D(t.src,n=e[3])||l(t,"src",n),l(t,"alt",e[4])},m(e,n){i(e,t,n),s=!0},p(a,r){e=a,(!s||8&r&&!D(t.src,n=e[3]))&&l(t,"src",n),(!s||16&r)&&l(t,"alt",e[4])},i(n){s||(n&&K((()=>{s&&(a||(a=V(t,z,{duration:e[7]?Bn:0},!0)),a.run(1))})),s=!0)},o(n){n&&(a||(a=V(t,z,{duration:e[7]?Bn:0},!1)),a.run(0)),s=!1},d(e){e&&d(t),e&&a&&a.end()}}}function Gn(e){let t,n,a,s,r,l;const m=[En,Ln],g=[];function f(e,t){return void 0===e[6]?0:1}function L(t){e[22](t)}t=f(e),n=g[t]=m[t](e);let E={modalHeading:e[4],primaryButtonText:"Open in new window ...",secondaryButtonText:"Close",$$slots:{default:[wn]},$$scope:{ctx:e}};return void 0!==e[8]&&(E.open=e[8]),s=new de({props:E}),b.push((()=>h(s,"open",L))),s.$on("click:button--secondary",e[23]),s.$on("click:button--primary",e[24]),{c(){n.c(),a=S(),u(s.$$.fragment)},m(e,n){g[t].m(e,n),i(e,a,n),$(s,e,n),l=!0},p(e,[l]){let i=t;t=f(e),t===i?g[t].p(e,l):(c(),_(g[i],1,1,(()=>{g[i]=null})),p(),n=g[t],n?n.p(e,l):(n=g[t]=m[t](e),n.c()),o(n,1),n.m(a.parentNode,a));const d={};16&l&&(d.modalHeading=e[4]),33554584&l&&(d.$$scope={dirty:l,ctx:e}),!r&&256&l&&(r=!0,d.open=e[8],C((()=>r=!1))),s.$set(d)},i(e){l||(o(n),o(s.$$.fragment,e),l=!0)},o(e){_(n),_(s.$$.fragment,e),l=!1},d(e){e&&d(a),g[t].d(e),A(s,e)}}}const Bn=110;function Fn(e,t,n){const a=["src","alt","download","ratio","loading","loaded","error","fadeIn","border","loadImage"];let s=G(t,a),{$$slots:r={},$$scope:l}=t,{src:i=""}=t,{alt:o=""}=t,{download:c=[]}=t,{ratio:p}=t,{loading:_=!1}=t,{loaded:d=!1}=t,{error:m=!1}=t,{fadeIn:f=!1}=t,{border:u=!0}=t,$=!1,A=!1;const S=e=>{null!=C&&(C=null),n(0,d=!1),n(1,m=!1),C=new Image,C.src=e||i,C.onload=()=>n(0,d=!0),C.onerror=()=>n(1,m=!0)},b=F();let h,C=null;g((()=>()=>C=null));const L={...s,class:[s.class,"pipen-report-image",u?"":"pipen-report-image-noborder"].filter(Boolean).join(" ")};return e.$$set=e=>{t=P(P({},t),B(e)),n(28,s=G(t,a)),"src"in e&&n(3,i=e.src),"alt"in e&&n(4,o=e.alt),"download"in e&&n(5,c=e.download),"ratio"in e&&n(6,p=e.ratio),"loading"in e&&n(2,_=e.loading),"loaded"in e&&n(0,d=e.loaded),"error"in e&&n(1,m=e.error),"fadeIn"in e&&n(7,f=e.fadeIn),"border"in e&&n(13,u=e.border),"$$scope"in e&&n(25,l=e.$$scope)},e.$$.update=()=>{3&e.$$.dirty&&n(2,_=!d&&!m),8&e.$$.dirty&&i&&"undefined"!=typeof window&&S(),1&e.$$.dirty&&d&&b("load"),2&e.$$.dirty&&m&&b("error")},[d,m,_,i,o,c,p,f,$,A,h,()=>{n(8,$=!0)},L,u,S,r,()=>{n(9,A=!0)},()=>{n(10,h=setTimeout((()=>{n(9,A=!1)}),100))},()=>{clearTimeout(h),n(9,A=!0)},()=>{n(9,A=!0)},()=>{n(10,h=setTimeout((()=>{n(9,A=!1)}),100))},()=>{clearTimeout(h),n(9,A=!0)},function(e){$=e,n(8,$)},()=>{n(8,$=!1)},()=>{n(8,$=!1),window.open(i)},l]}class Hn extends t{constructor(e){super(),n(this,e,Fn,Gn,a,{src:3,alt:4,download:5,ratio:6,loading:2,loaded:0,error:1,fadeIn:7,border:13,loadImage:14})}get loadImage(){return this.$$.ctx[14]}}function Un(e){let t,n,a=[e[2]],s={};for(let e=0;e<a.length;e+=1)s=P(s,a[e]);return{c(){t=r("div"),n=L("Image loading ..."),H(t,s)},m(e,a){i(e,t,a),E(t,n)},p:T,d(e){e&&d(t)}}}function On(e){let t,n,a=[e[3]],s={};for(let e=0;e<a.length;e+=1)s=P(s,a[e]);return{c(){t=r("div"),n=L("Image loading error!"),H(t,s)},m(e,a){i(e,t,a),E(t,n)},p:T,d(e){e&&d(t)}}}function Dn(e){let t,n,a;const s=[{src:e[1]},e[4]];let l={$$slots:{error:[On],loading:[Un]},$$scope:{ctx:e}};for(let e=0;e<s.length;e+=1)l=P(l,s[e]);n=new Hn({props:l});let c=[e[0]],p={};for(let e=0;e<c.length;e+=1)p=P(p,c[e]);return{c(){t=r("div"),u(n.$$.fragment),H(t,p)},m(e,s){i(e,t,s),$(n,t,null),a=!0},p(e,[a]){const r=18&a?I(s,[2&a&&{src:e[1]},16&a&&w(e[4])]):{};256&a&&(r.$$scope={dirty:a,ctx:e}),n.$set(r),H(t,p=I(c,[1&a&&e[0]]))},i(e){a||(o(n.$$.fragment,e),a=!0)},o(e){_(n.$$.fragment,e),a=!1},d(e){e&&d(t),A(n)}}}function Kn(e,t,n){const a=["frameProps","width","height","src"];let s=G(t,a),{frameProps:r={class:"pipen-report-image-frame"}}=t,{width:l=300}=t,{height:i=200}=t,{src:o}=t;r||(r={"style:position":"relative"});let c=`width: ${l}px; height: ${i}px; aspect-ratio: ${l} / ${i}`,p={...s,class:[s.class,"pipen-report-image-loading"].filter(Boolean).join(" "),style:s.style?`${s.style}; ${c}`:c},_={...p,class:[p.class,"error"].join(" ")};return e.$$set=e=>{t=P(P({},t),B(e)),n(4,s=G(t,a)),"frameProps"in e&&n(0,r=e.frameProps),"width"in e&&n(5,l=e.width),"height"in e&&n(6,i=e.height),"src"in e&&n(1,o=e.src)},[r,o,p,_,s,l,i]}let Vn=class extends t{constructor(e){super(),n(this,e,Kn,Dn,a,{frameProps:0,width:5,height:6,src:1})}};const zn=e=>({}),kn=e=>({});function qn(e){let t,n;const a=e[25].default,s=y(a,e,e[24],null);return{c(){t=r("ul"),s&&s.c(),l(t,"role","menu"),l(t,"tabindex","-1"),l(t,"aria-label",e[14]),l(t,"data-floating-menu-direction",e[5]),l(t,"class",e[9]),U(t,"bx--overflow-menu-options",!0),U(t,"bx--overflow-menu--flip",e[7]),U(t,"bx--overflow-menu-options--open",e[0]),U(t,"bx--overflow-menu-options--light",e[6]),U(t,"bx--overflow-menu-options--sm","sm"===e[4]),U(t,"bx--overflow-menu-options--xl","xl"===e[4]),U(t,"bx--breadcrumb-menu-options",!!e[15])},m(a,r){i(a,t,r),s&&s.m(t,null),e[32](t),n=!0},p(e,r){s&&s.p&&(!n||16777216&r[0])&&x(s,a,e,e[24],n?M(a,e[24],r,null):N(e[24]),null),(!n||16384&r[0])&&l(t,"aria-label",e[14]),(!n||32&r[0])&&l(t,"data-floating-menu-direction",e[5]),(!n||512&r[0])&&l(t,"class",e[9]),(!n||512&r[0])&&U(t,"bx--overflow-menu-options",!0),(!n||640&r[0])&&U(t,"bx--overflow-menu--flip",e[7]),(!n||513&r[0])&&U(t,"bx--overflow-menu-options--open",e[0]),(!n||576&r[0])&&U(t,"bx--overflow-menu-options--light",e[6]),(!n||528&r[0])&&U(t,"bx--overflow-menu-options--sm","sm"===e[4]),(!n||528&r[0])&&U(t,"bx--overflow-menu-options--xl","xl"===e[4]),(!n||33280&r[0])&&U(t,"bx--breadcrumb-menu-options",!!e[15])},i(e){n||(o(s,e),n=!0)},o(e){_(s,e),n=!1},d(n){n&&d(t),s&&s.d(n),e[32](null)}}}function jn(e){let t,n,a,s,l,m,g,b;const h=e[25].menu,C=y(h,e,e[24],kn),L=C||function(e){let t,n,a;var s=e[1];function r(e,t){return{props:{"aria-label":e[11],title:e[11],class:"bx--overflow-menu__icon "+e[10]}}}return s&&(t=Y(s,r(e))),{c(){t&&u(t.$$.fragment),n=f()},m(e,s){t&&$(t,e,s),i(e,n,s),a=!0},p(e,a){if(2&a[0]&&s!==(s=e[1])){if(t){c();const e=t;_(e.$$.fragment,1,0,(()=>{A(e,1)})),p()}s?(t=Y(s,r(e)),u(t.$$.fragment),o(t.$$.fragment,1),$(t,n.parentNode,n)):t=null}else if(s){const n={};2048&a[0]&&(n["aria-label"]=e[11]),2048&a[0]&&(n.title=e[11]),1024&a[0]&&(n.class="bx--overflow-menu__icon "+e[10]),t.$set(n)}},i(e){a||(t&&o(t.$$.fragment,e),a=!0)},o(e){t&&_(t.$$.fragment,e),a=!1},d(e){e&&d(n),t&&A(t,e)}}}(e);let v=e[0]&&qn(e),T=[{type:"button"},{"aria-haspopup":""},{"aria-expanded":e[0]},{"aria-label":e[14]},{id:e[12]},e[20]],w={};for(let e=0;e<T.length;e+=1)w=P(w,T[e]);return{c(){t=new J(!1),n=f(),a=S(),s=r("button"),L.c(),l=S(),v&&v.c(),t.a=n,H(s,w),U(s,"bx--overflow-menu",!0),U(s,"bx--overflow-menu--open",e[0]),U(s,"bx--overflow-menu--light",e[6]),U(s,"bx--overflow-menu--sm","sm"===e[4]),U(s,"bx--overflow-menu--xl","xl"===e[4])},m(r,o){t.m(e[13],document.head),E(document.head,n),i(r,a,o),i(r,s,o),L.m(s,null),E(s,l),v&&v.m(s,null),s.autofocus&&s.focus(),e[33](s),m=!0,g||(b=[R(window,"click",e[31]),R(s,"click",e[26]),R(s,"click",e[34]),R(s,"mouseover",e[27]),R(s,"mouseenter",e[28]),R(s,"mouseleave",e[29]),R(s,"keydown",e[30]),R(s,"keydown",e[35]),R(s,"focusout",e[36])],g=!0)},p(e,n){(!m||8192&n[0])&&t.p(e[13]),C?C.p&&(!m||16777216&n[0])&&x(C,h,e,e[24],m?M(h,e[24],n,zn):N(e[24]),kn):L.p&&(!m||3074&n[0])&&L.p(e,m?n:[-1,-1]),e[0]?v?(v.p(e,n),1&n[0]&&o(v,1)):(v=qn(e),v.c(),o(v,1),v.m(s,null)):v&&(c(),_(v,1,1,(()=>{v=null})),p()),H(s,w=I(T,[{type:"button"},{"aria-haspopup":""},(!m||1&n[0])&&{"aria-expanded":e[0]},(!m||16384&n[0])&&{"aria-label":e[14]},(!m||4096&n[0])&&{id:e[12]},1048576&n[0]&&e[20]])),U(s,"bx--overflow-menu",!0),U(s,"bx--overflow-menu--open",e[0]),U(s,"bx--overflow-menu--light",e[6]),U(s,"bx--overflow-menu--sm","sm"===e[4]),U(s,"bx--overflow-menu--xl","xl"===e[4])},i(e){m||(o(L,e),o(v),m=!0)},o(e){_(L,e),_(v),m=!1},d(r){r&&(t.d(),d(a),d(s)),d(n),L.d(r),v&&v.d(),e[33](null),g=!1,k(b)}}}function Xn(e,t,n){let a,s;const r=["size","direction","open","light","flipped","persistentClickItems","menuOptionsClass","icon","iconClass","iconDescription","id","buttonRef","menuRef"];let l,i,o,c=G(t,r),{$$slots:p={},$$scope:_}=t,{size:d}=t,{direction:m="bottom"}=t,{open:g=!1}=t,{light:f=!1}=t,{flipped:u=!1}=t,{persistentClickItems:$=!1}=t,{menuOptionsClass:A}=t,{icon:S=Je}=t,{iconClass:h}=t,{iconDescription:C="Open and close list of options"}=t,{id:L="ccs-"+Math.random().toString(36)}=t,{buttonRef:E=null}=t,{menuRef:v=null}=t;const R=q("BreadcrumbItem"),T=F(),y=Z([]);j(e,y,(e=>n(23,i=e)));const x=Z(void 0);j(e,x,(e=>n(38,o=e)));const N=Z(void 0),M=Z(-1);let I;j(e,M,(e=>n(22,l=e)));let w=!0;X("OverflowMenu",{focusedId:N,add:({id:e,text:t,primaryFocus:n,disabled:a})=>{y.update((s=>(n&&M.set(s.length),[...s,{id:e,text:t,primaryFocus:n,disabled:a,index:s.length}])))},update:e=>{x.set(e)},change:e=>{let t=l+e;t<0?t=i.length-1:t>=i.length&&(t=0);let n=i[t].disabled;for(;n;)t+=e,t<0?t=i.length-1:t>=i.length&&(t=0),n=i[t].disabled;M.set(t)}}),W((()=>{if(o){const{index:e,text:t}=i.filter((e=>e.id===o))[0];T("close",{index:e,text:t}),n(0,g=!1)}if(g){const{width:e,height:t}=E.getBoundingClientRect();n(21,I=e),!w&&l<0&&v.focus(),u&&(n(3,v.style.left="auto",v),n(3,v.style.right=0,v)),"top"===m?(n(3,v.style.top="auto",v),n(3,v.style.bottom=t+"px",v)):"bottom"===m&&n(3,v.style.top=t+"px",v),R&&(n(3,v.style.top=t+10+"px",v),n(3,v.style.left="-11px",v))}g||(y.set([]),x.set(void 0),M.set(0)),w=!1}));return e.$$set=e=>{n(40,t=P(P({},t),B(e))),n(20,c=G(t,r)),"size"in e&&n(4,d=e.size),"direction"in e&&n(5,m=e.direction),"open"in e&&n(0,g=e.open),"light"in e&&n(6,f=e.light),"flipped"in e&&n(7,u=e.flipped),"persistentClickItems"in e&&n(8,$=e.persistentClickItems),"menuOptionsClass"in e&&n(9,A=e.menuOptionsClass),"icon"in e&&n(1,S=e.icon),"iconClass"in e&&n(10,h=e.iconClass),"iconDescription"in e&&n(11,C=e.iconDescription),"id"in e&&n(12,L=e.id),"buttonRef"in e&&n(2,E=e.buttonRef),"menuRef"in e&&n(3,v=e.menuRef),"$$scope"in e&&n(24,_=e.$$scope)},e.$$.update=()=>{n(14,a=t["aria-label"]||"menu"),12582912&e.$$.dirty[0]&&i[l]&&N.set(i[l].id),2101248&e.$$.dirty[0]&&n(13,s=`<style>\n    #${L} .bx--overflow-menu-options.bx--overflow-menu-options:after {\n      width: ${I?I+"px":"2rem"};\n    }\n  </style>`)},R&&n(1,S=Ye),t=B(t),[g,S,E,v,d,m,f,u,$,A,h,C,L,s,a,R,T,y,x,M,c,I,l,i,_,p,function(t){Q.call(this,e,t)},function(t){Q.call(this,e,t)},function(t){Q.call(this,e,t)},function(t){Q.call(this,e,t)},function(t){Q.call(this,e,t)},({target:e})=>{E&&E.contains(e)||v&&!v.contains(e)&&n(0,g=!1)},function(e){b[e?"unshift":"push"]((()=>{v=e,n(3,v)}))},function(e){b[e?"unshift":"push"]((()=>{E=e,n(2,E)}))},e=>{$&&e.stopPropagation(),v&&v.contains(e.target)||(n(0,g=!g),g||T("close"))},e=>{g&&(["ArrowDown","ArrowLeft","ArrowRight","ArrowUp"].includes(e.key)?e.preventDefault():"Escape"===e.key&&(e.stopPropagation(),T("close"),n(0,g=!1),E.focus()))},e=>{g&&(E.contains(e.relatedTarget)||(T("close"),n(0,g=!1)))}]}class Wn extends t{constructor(e){super(),n(this,e,Xn,jn,a,{size:4,direction:5,open:0,light:6,flipped:7,persistentClickItems:8,menuOptionsClass:9,icon:1,iconClass:10,iconDescription:11,id:12,buttonRef:2,menuRef:3},null,[-1,-1])}}function Yn(e,t,n){const a=e.slice();return a[3]=t[n],a[4]=t,a[5]=n,a}function Jn(e){let t,n,a,s,c;function p(t){e[1](t,e[3])}let m={labelText:e[3].value};return void 0!==e[3].selected&&(m.checked=e[3].selected),n=new Re({props:m}),b.push((()=>h(n,"checked",p))),n.$on("change",e[2]),{c(){t=r("li"),u(n.$$.fragment),s=S(),l(t,"class","pipen-report-column-selector-item svelte-147jw2r")},m(e,a){i(e,t,a),$(n,t,null),E(t,s),c=!0},p(t,s){e=t;const r={};1&s&&(r.labelText=e[3].value),!a&&1&s&&(a=!0,r.checked=e[3].selected,C((()=>a=!1))),n.$set(r)},i(e){c||(o(n.$$.fragment,e),c=!0)},o(e){_(n.$$.fragment,e),c=!1},d(e){e&&d(t),A(n)}}}function Zn(e){let t,n,a=s(e[0]),r=[];for(let t=0;t<a.length;t+=1)r[t]=Jn(Yn(e,a,t));const l=e=>_(r[e],1,1,(()=>{r[e]=null}));return{c(){for(let e=0;e<r.length;e+=1)r[e].c();t=f()},m(e,a){for(let t=0;t<r.length;t+=1)r[t]&&r[t].m(e,a);i(e,t,a),n=!0},p(e,n){if(1&n){let i;for(a=s(e[0]),i=0;i<a.length;i+=1){const s=Yn(e,a,i);r[i]?(r[i].p(s,n),o(r[i],1)):(r[i]=Jn(s),r[i].c(),o(r[i],1),r[i].m(t.parentNode,t))}for(c(),i=a.length;i<r.length;i+=1)l(i);p()}},i(e){if(!n){for(let e=0;e<a.length;e+=1)o(r[e]);n=!0}},o(e){r=r.filter(Boolean);for(let e=0;e<r.length;e+=1)_(r[e]);n=!1},d(e){e&&d(t),m(r,e)}}}function Qn(e){let t,n;return t=new Wn({props:{class:"pipen-report-overflowmenu",persistentClickItems:!0,size:"sm",icon:Ze,iconDescription:"Select visible columns ...",$$slots:{default:[Zn]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){$(t,e,a),n=!0},p(e,[n]){const a={};65&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){A(t,e)}}}function ea(e,t,n){let{allColumns:a}=t;return e.$$set=e=>{"allColumns"in e&&n(0,a=e.allColumns)},[a,function(t,s){e.$$.not_equal(s.selected,t)&&(s.selected=t,n(0,a))},()=>{n(0,a)}]}class ta extends t{constructor(e){super(),n(this,e,ea,Qn,a,{allColumns:0})}}function na(e){let t,n;return t=new Le({props:{iconDescription:"Right click and save as to download the entire data",tooltipPosition:"top",icon:Qe,href:e[2]}}),t.$on("click",e[13]),{c(){u(t.$$.fragment)},m(e,a){$(t,e,a),n=!0},p(e,n){const a={};4&n&&(a.href=e[2]),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){A(t,e)}}}function aa(e){let t,n,a,s,r,l,m,g,L,E;function v(t){e[17](t)}let R={persistent:!0};function T(t){e[18](t)}void 0!==e[7]&&(R.value=e[7]),t=new Me({props:R}),b.push((()=>h(t,"value",v)));let y={};void 0!==e[6]&&(y.allColumns=e[6]),s=new ta({props:y}),b.push((()=>h(s,"allColumns",T))),m=new Le({props:{iconDescription:`Download current ${e[11].length} records (may be partial of the entire data)`,tooltipPosition:"top",icon:Be}}),m.$on("click",e[14]);let x=!!e[2]&&na(e);return{c(){u(t.$$.fragment),a=S(),u(s.$$.fragment),l=S(),u(m.$$.fragment),g=S(),x&&x.c(),L=f()},m(e,n){$(t,e,n),i(e,a,n),$(s,e,n),i(e,l,n),$(m,e,n),i(e,g,n),x&&x.m(e,n),i(e,L,n),E=!0},p(e,a){const l={};!n&&128&a&&(n=!0,l.value=e[7],C((()=>n=!1))),t.$set(l);const i={};!r&&64&a&&(r=!0,i.allColumns=e[6],C((()=>r=!1))),s.$set(i);const d={};2048&a&&(d.iconDescription=`Download current ${e[11].length} records (may be partial of the entire data)`),m.$set(d),e[2]?x?(x.p(e,a),4&a&&o(x,1)):(x=na(e),x.c(),o(x,1),x.m(L.parentNode,L)):x&&(c(),_(x,1,1,(()=>{x=null})),p())},i(e){E||(o(t.$$.fragment,e),o(s.$$.fragment,e),o(m.$$.fragment,e),o(x),E=!0)},o(e){_(t.$$.fragment,e),_(s.$$.fragment,e),_(m.$$.fragment,e),_(x),E=!1},d(e){e&&(d(a),d(l),d(g),d(L)),A(t,e),A(s,e),A(m,e),x&&x.d(e)}}}function sa(e){let t,n;return t=new Ne({props:{$$slots:{default:[aa]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){$(t,e,a),n=!0},p(e,n){const a={};134219972&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){A(t,e)}}}function ra(e){let t,n;return t=new xe({props:{size:e[12],$$slots:{default:[sa]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){$(t,e,a),n=!0},p(e,n){const a={};134219972&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){A(t,e)}}}function la(e){let t;return{c(){t=r("p"),t.textContent='You must use "Save as" or "Save link as" from the context menu (by right-clicking the button) to download the entire data.'},m(e,n){i(e,t,n)},p:T,d(e){e&&d(t)}}}function ia(e){let t,n,a,s,r,l,c,p,m;const g=[{size:e[0]},{zebra:e[3]},{sortable:e[4]},{page:e[8]},{pageSize:e[9]},{rows:e[11]},{headers:e[1]},e[15]];let f={$$slots:{default:[ra]},$$scope:{ctx:e}};for(let e=0;e<g.length;e+=1)f=P(f,g[e]);function L(t){e[19](t)}function E(t){e[20](t)}t=new Te({props:f});let v={pageSizes:e[5],totalItems:e[11].length};function R(t){e[21](t)}void 0!==e[9]&&(v.pageSize=e[9]),void 0!==e[8]&&(v.page=e[8]),a=new ye({props:v}),b.push((()=>h(a,"pageSize",L))),b.push((()=>h(a,"page",E)));let T={passiveModal:!0,modalHeading:"Downloading entire data",$$slots:{default:[la]},$$scope:{ctx:e}};return void 0!==e[10]&&(T.open=e[10]),c=new de({props:T}),b.push((()=>h(c,"open",R))),c.$on("open",e[22]),c.$on("close",e[23]),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment),l=S(),u(c.$$.fragment)},m(e,s){$(t,e,s),i(e,n,s),$(a,e,s),i(e,l,s),$(c,e,s),m=!0},p(e,[n]){const l=35611&n?I(g,[1&n&&{size:e[0]},8&n&&{zebra:e[3]},16&n&&{sortable:e[4]},256&n&&{page:e[8]},512&n&&{pageSize:e[9]},2048&n&&{rows:e[11]},2&n&&{headers:e[1]},32768&n&&w(e[15])]):{};134219972&n&&(l.$$scope={dirty:n,ctx:e}),t.$set(l);const i={};32&n&&(i.pageSizes=e[5]),2048&n&&(i.totalItems=e[11].length),!s&&512&n&&(s=!0,i.pageSize=e[9],C((()=>s=!1))),!r&&256&n&&(r=!0,i.page=e[8],C((()=>r=!1))),a.$set(i);const o={};134217728&n&&(o.$$scope={dirty:n,ctx:e}),!p&&1024&n&&(p=!0,o.open=e[10],C((()=>p=!1))),c.$set(o)},i(e){m||(o(t.$$.fragment,e),o(a.$$.fragment,e),o(c.$$.fragment,e),m=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),_(c.$$.fragment,e),m=!1},d(e){e&&(d(n),d(l)),A(t,e),A(a,e),A(c,e)}}}function oa(e,t,n){const a=["size","src","zebra","sortable","rows","headers","pageSizes"];let s=G(t,a),{size:r="sm"}=t,{src:l}=t;r={sm:"short",small:"short",short:"short",md:"medium",medium:"medium",lg:"tall",large:"tall",tall:"tall",compact:"compact"}[r];let i={short:"sm",compact:"sm",medium:"default",tall:"default"}[r],{zebra:o=!0}=t,{sortable:c=!0}=t,{rows:p=[]}=t,{headers:_=[]}=t,{pageSizes:d=[20,50,75,100]}=t,m=_.map((e=>({...e,selected:!0}))),g=1,f=d[0],u="",$=!1,A=p;return e.$$set=e=>{t=P(P({},t),B(e)),n(15,s=G(t,a)),"size"in e&&n(0,r=e.size),"src"in e&&n(2,l=e.src),"zebra"in e&&n(3,o=e.zebra),"sortable"in e&&n(4,c=e.sortable),"rows"in e&&n(16,p=e.rows),"headers"in e&&n(1,_=e.headers),"pageSizes"in e&&n(5,d=e.pageSizes)},e.$$.update=()=>{64&e.$$.dirty&&n(1,_=m.filter((e=>e.selected))),65664&e.$$.dirty&&(u.length>0?n(11,A=p.filter((e=>Object.values(e).filter((e=>!!e&&e.toString().includes(u))).length>0))):n(11,A=p))},[r,_,l,o,c,d,m,u,g,f,$,A,i,e=>{e.preventDefault(),n(10,$=!0)},()=>{((e,t)=>{const n=document.createElement("a");n.href=e,n.download=t,document.body.appendChild(n),n.click(),document.body.removeChild(n)})("data:application/vnd.ms-excel;charset=utf-8,\ufeff"+encodeURIComponent(function(){const e=_.map((e=>e.value)),t=[];return A.forEach((e=>{const n=_.map((t=>`"${e[t.key]}"`));t.push(n.join(","))})),e.join(",")+"\r\n"+t.join("\r\n")}()),l&&l.length>0?l.split("/").pop().split(".")[0]+".xls":"data-subset.xls")},s,p,function(e){u=e,n(7,u)},function(e){m=e,n(6,m)},function(e){f=e,n(9,f)},function(e){g=e,n(8,g)},function(e){$=e,n(10,$)},function(t){Q.call(this,e,t)},function(t){Q.call(this,e,t)}]}class ca extends t{constructor(e){super(),n(this,e,oa,ia,a,{size:0,src:2,zebra:3,sortable:4,rows:16,headers:1,pageSizes:5})}}function pa(e){let t,n,a;const s=[{headers:e[0]},{rows:e[1]},e[3]];let l={};for(let e=0;e<s.length;e+=1)l=P(l,s[e]);n=new ca({props:l});let c=[e[2]],p={};for(let e=0;e<c.length;e+=1)p=P(p,c[e]);return{c(){t=r("div"),u(n.$$.fragment),H(t,p)},m(e,s){i(e,t,s),$(n,t,null),a=!0},p(e,a){const r=11&a?I(s,[1&a&&{headers:e[0]},2&a&&{rows:e[1]},8&a&&w(e[3])]):{};n.$set(r),H(t,p=I(c,[4&a&&e[2]]))},i(e){a||(o(n.$$.fragment,e),a=!0)},o(e){_(n.$$.fragment,e),a=!1},d(e){e&&d(t),A(n)}}}function _a(e){let t,n;const a=[{headers:e[0]},{rows:e[1]},e[3]];let s={};for(let e=0;e<a.length;e+=1)s=P(s,a[e]);return t=new ca({props:s}),{c(){u(t.$$.fragment)},m(e,a){$(t,e,a),n=!0},p(e,n){const s=11&n?I(a,[1&n&&{headers:e[0]},2&n&&{rows:e[1]},8&n&&w(e[3])]):{};t.$set(s)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){A(t,e)}}}function da(e){let t,n,a,s;const r=[_a,pa],l=[];function m(e,t){return e[2]?1:0}return t=m(e),n=l[t]=r[t](e),{c(){n.c(),a=f()},m(e,n){l[t].m(e,n),i(e,a,n),s=!0},p(e,[s]){let i=t;t=m(e),t===i?l[t].p(e,s):(c(),_(l[i],1,1,(()=>{l[i]=null})),p(),n=l[t],n?n.p(e,s):(n=l[t]=r[t](e),n.c()),o(n,1),n.m(a.parentNode,a))},i(e){s||(o(n),s=!0)},o(e){_(n),s=!1},d(e){e&&d(a),l[t].d(e)}}}function ma(e,t,n){const a=["frameProps","data","headers","rows"];let s=G(t,a),{frameProps:r={class:"pipen-report-datatable-frame"}}=t,{data:l={}}=t,{headers:i=[]}=t,{rows:o=[]}=t;if(i.length+o.length>0&&l.length>0)throw"Can only specify `data` or `headers`/`rows` for `DataTable`!";return l.length>0&&(o=l,i=Object.keys(o[0]).filter((e=>"id"!==e)).map((e=>({key:e,value:e})))),e.$$set=e=>{t=P(P({},t),B(e)),n(3,s=G(t,a)),"frameProps"in e&&n(2,r=e.frameProps),"data"in e&&n(4,l=e.data),"headers"in e&&n(0,i=e.headers),"rows"in e&&n(1,o=e.rows)},[i,o,r,s,l]}class ga extends t{constructor(e){super(),n(this,e,ma,da,a,{frameProps:2,data:4,headers:0,rows:1})}}function fa(e){let t,n;return{c(){t=r("div"),n=L(e[0]),l(t,"class","pipen-report-descr-title svelte-1xzxuva")},m(e,a){i(e,t,a),E(t,n)},p(e,t){1&t&&v(n,e[0])},d(e){e&&d(t)}}}function ua(e){let t,n,a,s=e[0]&&fa(e);const l=e[3].default,c=y(l,e,e[2],null);let p=[e[1]],m={};for(let e=0;e<p.length;e+=1)m=P(m,p[e]);return{c(){t=r("div"),s&&s.c(),n=S(),c&&c.c(),H(t,m),U(t,"svelte-1xzxuva",!0)},m(e,r){i(e,t,r),s&&s.m(t,null),E(t,n),c&&c.m(t,null),a=!0},p(e,[r]){e[0]?s?s.p(e,r):(s=fa(e),s.c(),s.m(t,n)):s&&(s.d(1),s=null),c&&c.p&&(!a||4&r)&&x(c,l,e,e[2],a?M(l,e[2],r,null):N(e[2]),null),H(t,m=I(p,[2&r&&e[1]])),U(t,"svelte-1xzxuva",!0)},i(e){a||(o(c,e),a=!0)},o(e){_(c,e),a=!1},d(e){e&&d(t),s&&s.d(),c&&c.d(e)}}}function $a(e,t,n){const a=["title"];let s=G(t,a),{$$slots:r={},$$scope:l}=t,{title:i}=t,{klass:o,...c}=s;return o=o?`pipen-report-descr ${o}`:"pipen-report-descr",c||(c={}),c.class=o,e.$$set=e=>{t=P(P({},t),B(e)),n(5,s=G(t,a)),"title"in e&&n(0,i=e.title),"$$scope"in e&&n(2,l=e.$$scope)},[i,c,l,r]}class Aa extends t{constructor(e){super(),n(this,e,$a,ua,a,{title:0})}}function Sa(e){let t,n,a,s;return t=new fe({props:{label:"Plot"}}),a=new fe({props:{label:"Table"}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){$(t,e,r),i(e,n,r),$(a,e,r),s=!0},p:T,i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),A(t,e),A(a,e)}}}function ba(e){let t;return{c(){t=L("This table presents a comprehensive summary of the top enriched pathways derived from the fgsea. Each row corresponds to a pathway, and the gene ranks are shown based on the ranking metric used in the analysis. The enrichment score, p-value, and adjusted p-value are also provided to assess the significance of the enrichment.")},m(e,n){i(e,t,n)},d(e){e&&d(t)}}}function ha(e){let t,n,a,s;return t=new Aa({props:{$$slots:{default:[ba]},$$scope:{ctx:e}}}),a=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells-1/gsea_table.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells-1/gsea_table.pdf",width:1e3,height:760}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){$(t,e,r),i(e,n,r),$(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),A(t,e),A(a,e)}}}function Ca(e){let t;return{c(){t=L("This plot represents the GSEA results for a specified gene set, illustrating the distribution and impact of the gene set along the ranked list of genes. The running enrichment score curve shows the cumulative enrichment score as genes from the input list are encountered. Positive peaks on the curve indicate regions where members of the gene set are predominantly found.")},m(e,n){i(e,t,n)},d(e){e&&d(t)}}}function La(e){let t,n,a,s;return t=new Aa({props:{$$slots:{default:[Ca]},$$scope:{ctx:e}}}),a=new ga({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells-1/fgsea.txt",data:[{pathway:"HALLMARK_INTERFERON_ALPHA_RESPONSE",pval:0,padj:0,log2err:.6273,ES:.7644,NES:1.7278,size:96,leadingEdge:"ISG20,IFITM2,UBE2L6,IFITM3,LY6E,GBP2,TAP1,GBP4,PSME2,SELL,PSMB9,PSME1,PSMB8,CD74,EPSTI1,PARP9,IFITM1,SP110,BST2,IRF7,IFI35,NCOA7,ISG15,CD47,RTP4,ELF1,IRF1,HELZ2,SAMD9L,IFI27,IRF9,PARP12,CNP,LAMP3,TMEM140,OAS1,PLSCR1,IL7,IFI44L,RSAD2,TRIM14,LAP3,CMPK2,STAT2,USP18,ADAR,MOV10,SAMD9,EIF2AK2,NUB1,TRAFD1,MVB12A,CASP1,IFIT3,PARP14,TRIM21",id:0},{pathway:"HALLMARK_IL6_JAK_STAT3_SIGNALING",pval:8e-4,padj:.0058,log2err:.4773,ES:.6933,NES:1.5495,size:87,leadingEdge:"CD38,STAT1,SOCS3,TNFRSF1B,BAK1,IL12RB1,CSF2,CCR1,PIM1,PIK3R5,IL18R1,CD44,IRF1,TGFB1,SOCS1,IRF9,IL15RA,ITGA4,IL7,MYD88,IL2RG,STAT2,IL6ST,MAP3K8,CBL,LEPR,IL10RB,CXCL3,PTPN2,CRLF2",id:1},{pathway:"HALLMARK_MYC_TARGETS_V2",pval:.0207,padj:.0966,log2err:.3525,ES:-.5866,NES:-1.414,size:58,leadingEdge:"NOLC1,WDR74,UTP20,SLC29A2,DCTPP1,MYBBP1A,HSPE1,PPAN,BYSL,DDX18,EXOSC5,SORD,IMP4,MRTO4,CDK4,RRP9,MAP3K6,GNL3,WDR43,LAS1L,MPHOSPH10,NDUFAF4,PPRC1,NIP7,NOP16,RRP12,IPO4,PRMT3,GRWD1,PES1,RABEPK,FARSA,TMEM97,NOP2,PLK4,TBRG4",id:2},{pathway:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",pval:.0394,padj:.124,log2err:.2617,ES:.6212,NES:1.3636,size:73,leadingEdge:"FDFT1,TP53INP1,ANXA5,S100A11,LDLR,MVD,ATF3,FBXO6,ABCA2,SC5D,GLDC,PLSCR1,GSTM2,ETHE1,LGALS3,JAG1,SCD,FADS2,PLAUR,STX5,FDPS,HSD17B7,PMVK,ATXN2,HMGCR,GUSB",id:3},{pathway:"HALLMARK_PANCREAS_BETA_CELLS",pval:.0443,padj:.124,log2err:.2572,ES:.6807,NES:1.3777,size:40,leadingEdge:"DPP4,VDR,SRP14,FOXO1,SEC11A,SPCS1,SRP9,STXBP1,GCG,ELP4",id:4},{pathway:"HALLMARK_WNT_BETA_CATENIN_SIGNALING",pval:.3209,padj:.621,log2err:.1412,ES:-.466,NES:-1.0625,size:42,leadingEdge:"CCND2,TCF7,PTCH1,CTNNB1,NCSTN,HDAC2,DVL2,JAG2,PPARD,HDAC11,NCOR2,MAML1,KAT2A,RBPJ,NKD1,SKP2",id:5},{pathway:"HALLMARK_APICAL_SURFACE",pval:.3505,padj:.621,log2err:.1308,ES:-.4439,NES:-1.0342,size:44,leadingEdge:"CD160,HSPB1,GSTM3,CROCC,AKAP7,GAS1,LYPD3",id:6},{pathway:"HALLMARK_TGF_BETA_SIGNALING",pval:.3808,padj:.621,log2err:.0746,ES:.4966,NES:1.0415,size:54,leadingEdge:"ID2,FURIN,UBE2D3,FKBP1A,ARID4B,SPTBN1,SERPINE1,TGFB1,PPP1CA,SMURF2,SMAD7,CDKN1C,WWTR1,CDK9,TRIM33,ENG",id:7},{pathway:"HALLMARK_ANDROGEN_RESPONSE",pval:.3992,padj:.621,log2err:.0674,ES:.4575,NES:1.0367,size:99,leadingEdge:"FKBP5,SAT1,CCND3,HPGD,INSIG1,ARID5B,DBI,B4GALT1,IQGAP2,SRF,SCD,RAB4A,TMEM50A,DHCR24,RPS6KA3,SRP19,CENPN,UBE2I,HOMER2,GPD1L,ELL2,ZMIZ1,HERC3,HMGCR,NCOA4,FADS1,MAF,UBE2J1,PTK2B,LMAN1,STK39,IDI1,VAPA,ADRM1",id:8},{pathway:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",pval:.4868,padj:.6815,log2err:.0624,ES:.4726,NES:.9843,size:49,leadingEdge:"GLRX,NDUFB4,LSP1,PRDX2,NDUFA6,TXN,FES,ABCC1,SBNO2,SRXN1,HMOX2,PRDX6,GLRX2,NDUFS2,LAMTOR5,G6PD,NQO1,STK25",id:9},{pathway:"HALLMARK_NOTCH_SIGNALING",pval:.6657,padj:.8264,log2err:.0497,ES:.4446,NES:.8719,size:32,leadingEdge:"LFNG,FBXW11,JAG1,NOTCH1,RBX1,PRKCA,SAP30,DTX2",id:10},{pathway:"HALLMARK_HEDGEHOG_SIGNALING",pval:.7083,padj:.8264,log2err:.0858,ES:-.3888,NES:-.8654,size:36,leadingEdge:"ETS2,PTCH1,L1CAM,VLDLR,LDB1,DPYSL2,CDK6,CDK5R1",id:11},{pathway:"HALLMARK_PROTEIN_SECRETION",pval:.8095,padj:.8718,log2err:.101,ES:-.3394,NES:-.8762,size:95,leadingEdge:"ATP1A1,PAM,ANP32E,AP3S1,SCAMP1,TMED10,AP1G1,SEC31A,ZW10,SCAMP3,SCRN1,VPS45,CD63,VPS4B,COG2,KRT18,PPT1,ATP7A,RAB14,ICA1,ARFGAP3,AP2B1,BNIP3,RAB2A,M6PR,RAB9A,LAMP2,SEC24D,ARFGEF1",id:12},{pathway:"HALLMARK_ANGIOGENESIS",pval:.8855,padj:.8855,log2err:.0365,ES:.3767,NES:.7512,size:36,leadingEdge:"S100A4,TIMP1,JAG1,PRG2,VAV2,TNFRSF21,MSX1,FGFR1,APP",id:13}]}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){$(t,e,r),i(e,n,r),$(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),A(t,e),A(a,e)}}}function Ea(e){let t,n,a,s;return t=new ue({props:{$$slots:{default:[ha]},$$scope:{ctx:e}}}),a=new ue({props:{$$slots:{default:[La]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){$(t,e,r),i(e,n,r),$(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),A(t,e),A(a,e)}}}function va(e){let t,n;return t=new ge({props:{$$slots:{content:[Ea],default:[Sa]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){$(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){A(t,e)}}}function Ra(e){let t,n,a,s,c,p,m,g,f,b,h,C,L,v,R,y,x,N,M,P,I,w,G,B,F,H,U,O,D,K,V,z,k,q,j,X,W,Y,J,Z,Q,te,ne,ae,se,re,le,ie,oe,ce,pe;return a=new Aa({props:{title:"HALLMARK_INTERFERON_ALPHA_RESPONSE",class:"pipen-report-table-image-descr"}}),c=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells-1/fgsea_HALLMARK_INTERFERON_ALPHA_RESPONSE.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells-1/fgsea_HALLMARK_INTERFERON_ALPHA_RESPONSE.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),g=new Aa({props:{title:"HALLMARK_IL6_JAK_STAT3_SIGNALING",class:"pipen-report-table-image-descr"}}),b=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells-1/fgsea_HALLMARK_IL6_JAK_STAT3_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells-1/fgsea_HALLMARK_IL6_JAK_STAT3_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),L=new Aa({props:{title:"HALLMARK_MYC_TARGETS_V2",class:"pipen-report-table-image-descr"}}),R=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells-1/fgsea_HALLMARK_MYC_TARGETS_V2.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells-1/fgsea_HALLMARK_MYC_TARGETS_V2.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),N=new Aa({props:{title:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",class:"pipen-report-table-image-descr"}}),P=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells-1/fgsea_HALLMARK_CHOLESTEROL_HOMEOSTASIS.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells-1/fgsea_HALLMARK_CHOLESTEROL_HOMEOSTASIS.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),G=new Aa({props:{title:"HALLMARK_PANCREAS_BETA_CELLS",class:"pipen-report-table-image-descr"}}),F=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells-1/fgsea_HALLMARK_PANCREAS_BETA_CELLS.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells-1/fgsea_HALLMARK_PANCREAS_BETA_CELLS.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),O=new Aa({props:{title:"HALLMARK_WNT_BETA_CATENIN_SIGNALING",class:"pipen-report-table-image-descr"}}),K=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells-1/fgsea_HALLMARK_WNT_BETA_CATENIN_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells-1/fgsea_HALLMARK_WNT_BETA_CATENIN_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),k=new Aa({props:{title:"HALLMARK_APICAL_SURFACE",class:"pipen-report-table-image-descr"}}),j=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells-1/fgsea_HALLMARK_APICAL_SURFACE.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells-1/fgsea_HALLMARK_APICAL_SURFACE.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),Y=new Aa({props:{title:"HALLMARK_TGF_BETA_SIGNALING",class:"pipen-report-table-image-descr"}}),Z=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells-1/fgsea_HALLMARK_TGF_BETA_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells-1/fgsea_HALLMARK_TGF_BETA_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new Aa({props:{title:"HALLMARK_ANDROGEN_RESPONSE",class:"pipen-report-table-image-descr"}}),se=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells-1/fgsea_HALLMARK_ANDROGEN_RESPONSE.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells-1/fgsea_HALLMARK_ANDROGEN_RESPONSE.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),ie=new Aa({props:{title:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",class:"pipen-report-table-image-descr"}}),ce=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells-1/fgsea_HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells-1/fgsea_HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=r("div"),n=r("div"),u(a.$$.fragment),s=S(),u(c.$$.fragment),p=S(),m=r("div"),u(g.$$.fragment),f=S(),u(b.$$.fragment),h=S(),C=r("div"),u(L.$$.fragment),v=S(),u(R.$$.fragment),y=S(),x=r("div"),u(N.$$.fragment),M=S(),u(P.$$.fragment),I=S(),w=r("div"),u(G.$$.fragment),B=S(),u(F.$$.fragment),H=S(),U=r("div"),u(O.$$.fragment),D=S(),u(K.$$.fragment),V=S(),z=r("div"),u(k.$$.fragment),q=S(),u(j.$$.fragment),X=S(),W=r("div"),u(Y.$$.fragment),J=S(),u(Z.$$.fragment),Q=S(),te=r("div"),u(ne.$$.fragment),ae=S(),u(se.$$.fragment),re=S(),le=r("div"),u(ie.$$.fragment),oe=S(),u(ce.$$.fragment),l(t,"class","pipen-report-table-of-images"),ee(t,"grid-template-columns","repeat(2, auto)")},m(e,r){i(e,t,r),E(t,n),$(a,n,null),E(n,s),$(c,n,null),E(t,p),E(t,m),$(g,m,null),E(m,f),$(b,m,null),E(t,h),E(t,C),$(L,C,null),E(C,v),$(R,C,null),E(t,y),E(t,x),$(N,x,null),E(x,M),$(P,x,null),E(t,I),E(t,w),$(G,w,null),E(w,B),$(F,w,null),E(t,H),E(t,U),$(O,U,null),E(U,D),$(K,U,null),E(t,V),E(t,z),$(k,z,null),E(z,q),$(j,z,null),E(t,X),E(t,W),$(Y,W,null),E(W,J),$(Z,W,null),E(t,Q),E(t,te),$(ne,te,null),E(te,ae),$(se,te,null),E(t,re),E(t,le),$(ie,le,null),E(le,oe),$(ce,le,null),pe=!0},p:T,i(e){pe||(o(a.$$.fragment,e),o(c.$$.fragment,e),o(g.$$.fragment,e),o(b.$$.fragment,e),o(L.$$.fragment,e),o(R.$$.fragment,e),o(N.$$.fragment,e),o(P.$$.fragment,e),o(G.$$.fragment,e),o(F.$$.fragment,e),o(O.$$.fragment,e),o(K.$$.fragment,e),o(k.$$.fragment,e),o(j.$$.fragment,e),o(Y.$$.fragment,e),o(Z.$$.fragment,e),o(ne.$$.fragment,e),o(se.$$.fragment,e),o(ie.$$.fragment,e),o(ce.$$.fragment,e),pe=!0)},o(e){_(a.$$.fragment,e),_(c.$$.fragment,e),_(g.$$.fragment,e),_(b.$$.fragment,e),_(L.$$.fragment,e),_(R.$$.fragment,e),_(N.$$.fragment,e),_(P.$$.fragment,e),_(G.$$.fragment,e),_(F.$$.fragment,e),_(O.$$.fragment,e),_(K.$$.fragment,e),_(k.$$.fragment,e),_(j.$$.fragment,e),_(Y.$$.fragment,e),_(Z.$$.fragment,e),_(ne.$$.fragment,e),_(se.$$.fragment,e),_(ie.$$.fragment,e),_(ce.$$.fragment,e),pe=!1},d(e){e&&d(t),A(a),A(c),A(g),A(b),A(L),A(R),A(N),A(P),A(G),A(F),A(O),A(K),A(k),A(j),A(Y),A(Z),A(ne),A(se),A(ie),A(ce)}}}function Ta(e){let t,n,a,s;return t=new Ie({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[va]},$$scope:{ctx:e}}}),a=new Ie({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[Ra]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){$(t,e,r),i(e,n,r),$(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),A(t,e),A(a,e)}}}function ya(e){let t,n,a,s;return t=new fe({props:{label:"Plot"}}),a=new fe({props:{label:"Table"}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){$(t,e,r),i(e,n,r),$(a,e,r),s=!0},p:T,i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),A(t,e),A(a,e)}}}function xa(e){let t;return{c(){t=L("This table presents a comprehensive summary of the top enriched pathways derived from the fgsea. Each row corresponds to a pathway, and the gene ranks are shown based on the ranking metric used in the analysis. The enrichment score, p-value, and adjusted p-value are also provided to assess the significance of the enrichment.")},m(e,n){i(e,t,n)},d(e){e&&d(t)}}}function Na(e){let t,n,a,s;return t=new Aa({props:{$$slots:{default:[xa]},$$scope:{ctx:e}}}),a=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Trm-cytotoxic-T-cells/gsea_table.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Trm-cytotoxic-T-cells/gsea_table.pdf",width:1e3,height:760}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){$(t,e,r),i(e,n,r),$(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),A(t,e),A(a,e)}}}function Ma(e){let t;return{c(){t=L("This plot represents the GSEA results for a specified gene set, illustrating the distribution and impact of the gene set along the ranked list of genes. The running enrichment score curve shows the cumulative enrichment score as genes from the input list are encountered. Positive peaks on the curve indicate regions where members of the gene set are predominantly found.")},m(e,n){i(e,t,n)},d(e){e&&d(t)}}}function Pa(e){let t,n,a,s;return t=new Aa({props:{$$slots:{default:[Ma]},$$scope:{ctx:e}}}),a=new ga({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Trm-cytotoxic-T-cells/fgsea.txt",data:[{pathway:"HALLMARK_INTERFERON_ALPHA_RESPONSE",pval:2e-4,padj:.0021,log2err:.5188,ES:.6469,NES:1.6654,size:96,leadingEdge:"ISG20,B2M,CD74,UBE2L6,GBP2,PSMB9,LY6E,TAP1,ISG15,PSMB8,BST2,IRF7,GBP4,PARP9,PARP12,IRF1,PSME2,PLSCR1,IFITM3,CNP,STAT2,ELF1,CASP8,TMEM140,IFITM1,ADAR,IRF9,RSAD2,NMI,CSF1,LGALS3BP,NUB1,MOV10,EPSTI1,USP18",id:0},{pathway:"HALLMARK_ANGIOGENESIS",pval:5e-4,padj:.0035,log2err:.4773,ES:-.8394,NES:-1.6161,size:36,leadingEdge:"S100A4,CCND2,TIMP1",id:1},{pathway:"HALLMARK_PANCREAS_BETA_CELLS",pval:.0288,padj:.1345,log2err:.3525,ES:.6365,NES:1.4083,size:40,leadingEdge:"FOXO1,VDR,SPCS1,DPP4,SRPRB,SRP14,SEC11A,STXBP1",id:2},{pathway:"HALLMARK_PROTEIN_SECRETION",pval:.1268,padj:.444,log2err:.135,ES:-.5401,NES:-1.1857,size:95,leadingEdge:"ATP1A1,VPS45,TMED10,CD63,COPB2,PPT1,ABCA1,M6PR,ARFGEF1,GNAS,KRT18,GLA,SEC31A,USO1,ARCN1,STX12,RAB22A,RAB14,BNIP3,DNM1L,ADAM10,RAB5A,VAMP3,MON2,KIF1B,MAPK1,CLN5,TMED2,ANP32E,ATP6V1H,CLTA,DST,YKT6,VAMP7,LAMP2,ARFIP1,NAPG,ARFGAP3,SEC22B,ATP7A,AP2B1,RAB2A,SNX2,GOSR2,AP2M1,VPS4B,SCAMP3,CTSC,RAB9A",id:3},{pathway:"HALLMARK_APICAL_SURFACE",pval:.1946,padj:.545,log2err:.111,ES:-.5895,NES:-1.1751,size:44,leadingEdge:"CD160,TMEM8B,PLAUR,MAL,FLOT2,AKAP7,CRYBG1,NCOA6,HSPB1,ADAM10,MDGA1,RTN4RL1,GHRL,GAS1,DCBLD2,ADIPOR2,GSTM3,APP",id:4},{pathway:"HALLMARK_TGF_BETA_SIGNALING",pval:.2415,padj:.5635,log2err:.0959,ES:-.5519,NES:-1.1348,size:54,leadingEdge:"PPP1R15A,KLF10,SKIL,CDK9,TGFBR1,TGFB1,ID1,IFNGR2,ENG,BMPR2,ID3,APC,CTNNB1,RHOA,HDAC1,CDKN1C,SERPINE1,SLC20A1,FNTA,BMPR1A,NCOR2,MAP3K7,XIAP",id:5},{pathway:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",pval:.2972,padj:.5943,log2err:.1767,ES:.4253,NES:1.0444,size:73,leadingEdge:"ATF3,TP53INP1,LDLR,MVD,PLSCR1,FABP5,GSTM2,SC5D,ABCA2,ECH1,SREBF2,CPEB2,CXCL16,FDFT1,PNRC1,FADS2,FBXO6,PMVK,TM7SF2",id:6},{pathway:"HALLMARK_HEDGEHOG_SIGNALING",pval:.3434,padj:.5995,log2err:.0793,ES:-.5539,NES:-1.0664,size:36,leadingEdge:"CDK6,PTCH1,DPYSL2,L1CAM,CELSR1,PML,VLDLR,LDB1,AMOT,GLI1,ACHE,PLG",id:7},{pathway:"HALLMARK_IL6_JAK_STAT3_SIGNALING",pval:.3854,padj:.5995,log2err:.162,ES:.3993,NES:1.0122,size:87,leadingEdge:"STAT1,SOCS3,CD38,IRF1,HMOX1,STAT3,IL6ST,TYK2,TNFRSF1B,STAT2,IRF9,CSF1,SOCS1,CBL,IL18R1,IFNGR1,PTPN1,IL4R,GRB2,PIK3R5,FAS",id:8},{pathway:"HALLMARK_ANDROGEN_RESPONSE",pval:.625,padj:.8342,log2err:.1269,ES:.3644,NES:.9433,size:99,leadingEdge:"FKBP5,B2M,SAT1,ZBTB10,INPP4B,ARID5B,UBE2J1,RPS6KA3,NKX3-1,B4GALT1,CCND3,UBE2I,STK39,HPGD,CCND1,SORD,PLPP1,AKT1,ABHD2",id:9},{pathway:"HALLMARK_NOTCH_SIGNALING",pval:.6797,padj:.8342,log2err:.0941,ES:.4118,NES:.8823,size:32,leadingEdge:"NOTCH1,LFNG,APH1A,CCND1,SAP30,MAML2,FBXW11,NOTCH2,ARRB1,PSENEN,DTX1,RBX1",id:10},{pathway:"HALLMARK_WNT_BETA_CATENIN_SIGNALING",pval:.7151,padj:.8342,log2err:.0428,ES:-.4365,NES:-.8682,size:42,leadingEdge:"CCND2,PTCH1,SKP2,DLL1,HDAC2,CTNNB1,NKD1,AXIN2,NCOR2,NUMB,PSEN2,NCSTN",id:11},{pathway:"HALLMARK_MYC_TARGETS_V2",pval:.9382,padj:.9814,log2err:.0279,ES:-.3521,NES:-.7313,size:58,leadingEdge:"SLC29A2,RRP12,BYSL,MRTO4,CDK4,MAP3K6,PRMT3,AIMP2,IMP4,MPHOSPH10,LAS1L,PLK4,RCL1,MCM4,WDR43,PLK1,UTP20,HK2,PA2G4,MYBBP1A,GNL3,PPAN,NDUFAF4,CBX3,TFB2M,GRWD1,PES1,NOLC1",id:12},{pathway:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",pval:.9814,padj:.9814,log2err:.0272,ES:-.3458,NES:-.701,size:49,leadingEdge:"GCLC,CAT,LSP1,GPX4,PRNP,MBP,HHEX,GCLM,ATOX1,TXNRD2,TXNRD1,GPX3,PRDX4,GSR,FTL,EGLN2,SOD2,ERCC2,GLRX2",id:13}]}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){$(t,e,r),i(e,n,r),$(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),A(t,e),A(a,e)}}}function Ia(e){let t,n,a,s;return t=new ue({props:{$$slots:{default:[Na]},$$scope:{ctx:e}}}),a=new ue({props:{$$slots:{default:[Pa]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){$(t,e,r),i(e,n,r),$(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),A(t,e),A(a,e)}}}function wa(e){let t,n;return t=new ge({props:{$$slots:{content:[Ia],default:[ya]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){$(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){A(t,e)}}}function Ga(e){let t,n,a,s,c,p,m,g,f,b,h,C,L,v,R,y,x,N,M,P,I,w,G,B,F,H,U,O,D,K,V,z,k,q,j,X,W,Y,J,Z,Q,te,ne,ae,se,re,le,ie,oe,ce,pe;return a=new Aa({props:{title:"HALLMARK_INTERFERON_ALPHA_RESPONSE",class:"pipen-report-table-image-descr"}}),c=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Trm-cytotoxic-T-cells/fgsea_HALLMARK_INTERFERON_ALPHA_RESPONSE.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Trm-cytotoxic-T-cells/fgsea_HALLMARK_INTERFERON_ALPHA_RESPONSE.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),g=new Aa({props:{title:"HALLMARK_ANGIOGENESIS",class:"pipen-report-table-image-descr"}}),b=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Trm-cytotoxic-T-cells/fgsea_HALLMARK_ANGIOGENESIS.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Trm-cytotoxic-T-cells/fgsea_HALLMARK_ANGIOGENESIS.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),L=new Aa({props:{title:"HALLMARK_PANCREAS_BETA_CELLS",class:"pipen-report-table-image-descr"}}),R=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Trm-cytotoxic-T-cells/fgsea_HALLMARK_PANCREAS_BETA_CELLS.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Trm-cytotoxic-T-cells/fgsea_HALLMARK_PANCREAS_BETA_CELLS.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),N=new Aa({props:{title:"HALLMARK_PROTEIN_SECRETION",class:"pipen-report-table-image-descr"}}),P=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Trm-cytotoxic-T-cells/fgsea_HALLMARK_PROTEIN_SECRETION.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Trm-cytotoxic-T-cells/fgsea_HALLMARK_PROTEIN_SECRETION.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),G=new Aa({props:{title:"HALLMARK_APICAL_SURFACE",class:"pipen-report-table-image-descr"}}),F=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Trm-cytotoxic-T-cells/fgsea_HALLMARK_APICAL_SURFACE.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Trm-cytotoxic-T-cells/fgsea_HALLMARK_APICAL_SURFACE.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),O=new Aa({props:{title:"HALLMARK_TGF_BETA_SIGNALING",class:"pipen-report-table-image-descr"}}),K=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Trm-cytotoxic-T-cells/fgsea_HALLMARK_TGF_BETA_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Trm-cytotoxic-T-cells/fgsea_HALLMARK_TGF_BETA_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),k=new Aa({props:{title:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",class:"pipen-report-table-image-descr"}}),j=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Trm-cytotoxic-T-cells/fgsea_HALLMARK_CHOLESTEROL_HOMEOSTASIS.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Trm-cytotoxic-T-cells/fgsea_HALLMARK_CHOLESTEROL_HOMEOSTASIS.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),Y=new Aa({props:{title:"HALLMARK_HEDGEHOG_SIGNALING",class:"pipen-report-table-image-descr"}}),Z=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Trm-cytotoxic-T-cells/fgsea_HALLMARK_HEDGEHOG_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Trm-cytotoxic-T-cells/fgsea_HALLMARK_HEDGEHOG_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new Aa({props:{title:"HALLMARK_IL6_JAK_STAT3_SIGNALING",class:"pipen-report-table-image-descr"}}),se=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Trm-cytotoxic-T-cells/fgsea_HALLMARK_IL6_JAK_STAT3_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Trm-cytotoxic-T-cells/fgsea_HALLMARK_IL6_JAK_STAT3_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),ie=new Aa({props:{title:"HALLMARK_ANDROGEN_RESPONSE",class:"pipen-report-table-image-descr"}}),ce=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Trm-cytotoxic-T-cells/fgsea_HALLMARK_ANDROGEN_RESPONSE.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Trm-cytotoxic-T-cells/fgsea_HALLMARK_ANDROGEN_RESPONSE.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=r("div"),n=r("div"),u(a.$$.fragment),s=S(),u(c.$$.fragment),p=S(),m=r("div"),u(g.$$.fragment),f=S(),u(b.$$.fragment),h=S(),C=r("div"),u(L.$$.fragment),v=S(),u(R.$$.fragment),y=S(),x=r("div"),u(N.$$.fragment),M=S(),u(P.$$.fragment),I=S(),w=r("div"),u(G.$$.fragment),B=S(),u(F.$$.fragment),H=S(),U=r("div"),u(O.$$.fragment),D=S(),u(K.$$.fragment),V=S(),z=r("div"),u(k.$$.fragment),q=S(),u(j.$$.fragment),X=S(),W=r("div"),u(Y.$$.fragment),J=S(),u(Z.$$.fragment),Q=S(),te=r("div"),u(ne.$$.fragment),ae=S(),u(se.$$.fragment),re=S(),le=r("div"),u(ie.$$.fragment),oe=S(),u(ce.$$.fragment),l(t,"class","pipen-report-table-of-images"),ee(t,"grid-template-columns","repeat(2, auto)")},m(e,r){i(e,t,r),E(t,n),$(a,n,null),E(n,s),$(c,n,null),E(t,p),E(t,m),$(g,m,null),E(m,f),$(b,m,null),E(t,h),E(t,C),$(L,C,null),E(C,v),$(R,C,null),E(t,y),E(t,x),$(N,x,null),E(x,M),$(P,x,null),E(t,I),E(t,w),$(G,w,null),E(w,B),$(F,w,null),E(t,H),E(t,U),$(O,U,null),E(U,D),$(K,U,null),E(t,V),E(t,z),$(k,z,null),E(z,q),$(j,z,null),E(t,X),E(t,W),$(Y,W,null),E(W,J),$(Z,W,null),E(t,Q),E(t,te),$(ne,te,null),E(te,ae),$(se,te,null),E(t,re),E(t,le),$(ie,le,null),E(le,oe),$(ce,le,null),pe=!0},p:T,i(e){pe||(o(a.$$.fragment,e),o(c.$$.fragment,e),o(g.$$.fragment,e),o(b.$$.fragment,e),o(L.$$.fragment,e),o(R.$$.fragment,e),o(N.$$.fragment,e),o(P.$$.fragment,e),o(G.$$.fragment,e),o(F.$$.fragment,e),o(O.$$.fragment,e),o(K.$$.fragment,e),o(k.$$.fragment,e),o(j.$$.fragment,e),o(Y.$$.fragment,e),o(Z.$$.fragment,e),o(ne.$$.fragment,e),o(se.$$.fragment,e),o(ie.$$.fragment,e),o(ce.$$.fragment,e),pe=!0)},o(e){_(a.$$.fragment,e),_(c.$$.fragment,e),_(g.$$.fragment,e),_(b.$$.fragment,e),_(L.$$.fragment,e),_(R.$$.fragment,e),_(N.$$.fragment,e),_(P.$$.fragment,e),_(G.$$.fragment,e),_(F.$$.fragment,e),_(O.$$.fragment,e),_(K.$$.fragment,e),_(k.$$.fragment,e),_(j.$$.fragment,e),_(Y.$$.fragment,e),_(Z.$$.fragment,e),_(ne.$$.fragment,e),_(se.$$.fragment,e),_(ie.$$.fragment,e),_(ce.$$.fragment,e),pe=!1},d(e){e&&d(t),A(a),A(c),A(g),A(b),A(L),A(R),A(N),A(P),A(G),A(F),A(O),A(K),A(k),A(j),A(Y),A(Z),A(ne),A(se),A(ie),A(ce)}}}function Ba(e){let t,n,a,s;return t=new Ie({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[wa]},$$scope:{ctx:e}}}),a=new Ie({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[Ga]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){$(t,e,r),i(e,n,r),$(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),A(t,e),A(a,e)}}}function Fa(e){let t,n,a,s;return t=new fe({props:{label:"Plot"}}),a=new fe({props:{label:"Table"}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){$(t,e,r),i(e,n,r),$(a,e,r),s=!0},p:T,i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),A(t,e),A(a,e)}}}function Ha(e){let t;return{c(){t=L("This table presents a comprehensive summary of the top enriched pathways derived from the fgsea. Each row corresponds to a pathway, and the gene ranks are shown based on the ranking metric used in the analysis. The enrichment score, p-value, and adjusted p-value are also provided to assess the significance of the enrichment.")},m(e,n){i(e,t,n)},d(e){e&&d(t)}}}function Ua(e){let t,n,a,s;return t=new Aa({props:{$$slots:{default:[Ha]},$$scope:{ctx:e}}}),a=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells/gsea_table.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells/gsea_table.pdf",width:1e3,height:760}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){$(t,e,r),i(e,n,r),$(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),A(t,e),A(a,e)}}}function Oa(e){let t;return{c(){t=L("This plot represents the GSEA results for a specified gene set, illustrating the distribution and impact of the gene set along the ranked list of genes. The running enrichment score curve shows the cumulative enrichment score as genes from the input list are encountered. Positive peaks on the curve indicate regions where members of the gene set are predominantly found.")},m(e,n){i(e,t,n)},d(e){e&&d(t)}}}function Da(e){let t,n,a,s;return t=new Aa({props:{$$slots:{default:[Oa]},$$scope:{ctx:e}}}),a=new ga({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells/fgsea.txt",data:[{pathway:"HALLMARK_INTERFERON_ALPHA_RESPONSE",pval:2e-4,padj:.0027,log2err:.5188,ES:.6071,NES:1.6279,size:96,leadingEdge:"ISG20,PSMB9,B2M,IFI44L,SELL,TAP1,UBE2L6,RIPK2,GBP4,IFITM3,PSMB8,GBP2,PSME1,CD74,SAMD9L,PSME2,IRF1,ADAR,LY6E,IFIT3,TMEM140,IFI27,IRF7,IFI44,LPAR6,EPSTI1,ISG15,UBA7,SP110,RSAD2,ELF1,IFI35,IFITM2,IFIH1,TENT5A,IL7,PLSCR1,NCOA7,LAMP3,CNP,TRIM14,PARP14,PARP9,HELZ2,PARP12,IFIT2,TDRD7,TRIM21,OASL,NMI,OGFR",id:0},{pathway:"HALLMARK_PROTEIN_SECRETION",pval:.0138,padj:.0969,log2err:.3807,ES:-.6545,NES:-1.3729,size:95,leadingEdge:"CD63,ANP32E,CTSC,TMED2,ABCA1,ARFIP1,COPE,TMED10,M6PR,CLTA,ATP1A1,ARCN1,ARFGEF2,SNAP23,RER1,VPS4B,SEC31A,LAMP2,YIPF6,ARFGEF1,AP2S1,GLA,VAMP7,IGF2R,RAB14,ZW10,COG2,PAM,GNAS,ARFGAP3,SEC24D,TMX1,AP2B1,SCAMP3,VPS45,RAB2A,NAPG,CLN5,CLTC,CLCN3,COPB1,USO1,SCRN1,AP3B1,STX12,OCRL,LMAN1,ERGIC3,RAB5A,VAMP3,ARF1,COPB2",id:1},{pathway:"HALLMARK_ANGIOGENESIS",pval:.0789,padj:.368,log2err:.1723,ES:-.6874,NES:-1.2945,size:36,leadingEdge:"S100A4,CCND2,APP,PTK2,JAG1",id:2},{pathway:"HALLMARK_IL6_JAK_STAT3_SIGNALING",pval:.1879,padj:.6578,log2err:.2492,ES:.3986,NES:1.0543,size:87,leadingEdge:"STAT1,IL6ST,IL18R1,SOCS3,SOCS1,IRF1,IL10RB,IL15RA,FAS,IL3RA,REG1A,IL1R1,IL1R2,CD38,STAT3,IFNGR2,IL2RA,CD44,IL7,MAP3K8,LTB,PIK3R5,HAX1,LTBR,CSF2,TNFRSF1B,IL12RB1,MYD88",id:3},{pathway:"HALLMARK_APICAL_SURFACE",pval:.3244,padj:.8657,log2err:.0734,ES:-.5651,NES:-1.0939,size:44,leadingEdge:"IL2RG,HSPB1,THY1,GSTM3,APP,ATP8B1,GATA3,DCBLD2,MDGA1,SCUBE1,CD160,FLOT2,ADIPOR2,BRCA1,TMEM8B,PLAUR",id:4},{pathway:"HALLMARK_ANDROGEN_RESPONSE",pval:.443,padj:.8657,log2err:.2413,ES:.3643,NES:.9895,size:99,leadingEdge:"FKBP5,ARID5B,B2M,CCND3,INSIG1,SAT1,IQGAP2,KRT19,LIFR,MAK,HMGCS1,ELL2,UBE2J1,RPS6KA3,B4GALT1,ABHD2,SGK1,ELK4,PMEPA1,SRP19,RAB4A,DNAJB9,ZBTB10,NKX3-1,FADS1,KRT8",id:5},{pathway:"HALLMARK_TGF_BETA_SIGNALING",pval:.491,padj:.8657,log2err:.0522,ES:-.5057,NES:-1.0043,size:54,leadingEdge:"KLF10,PPP1CA,TGFBR1,FKBP1A,HDAC1,ENG,FNTA,SLC20A1,TGFB1,RHOA,ID3,BMPR1A,SMURF1,PPP1R15A,NCOR2,APC,CTNNB1,SMAD1,SMURF2",id:6},{pathway:"HALLMARK_WNT_BETA_CATENIN_SIGNALING",pval:.4947,padj:.8657,log2err:.0537,ES:-.5201,NES:-1.0017,size:42,leadingEdge:"CCND2,NCSTN,HDAC2,DLL1,PSEN2,WNT1,NCOR2,ADAM17,HDAC11,RBPJ,SKP2,CTNNB1,NKD1,CUL1,NUMB,KAT2A,JAG1,WNT5B,WNT6",id:7},{pathway:"HALLMARK_PANCREAS_BETA_CELLS",pval:.6532,padj:.9023,log2err:.0413,ES:-.4795,NES:-.9135,size:40,leadingEdge:"SPCS1,SRP9,MAFB,GCK,SRP14,SRPRB,PAX6",id:8},{pathway:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",pval:.6693,padj:.9023,log2err:.0382,ES:-.4576,NES:-.9024,size:49,leadingEdge:"LSP1,GPX4,CAT,SCAF4,MSRA,G6PD,HHEX,PRNP,GCLM,MBP,LAMTOR5,PRDX2,CDKN2D,PRDX4,PRDX6,FTL,EGLN2,SRXN1,PTPA,GCLC,GSR,TXNRD1",id:9},{pathway:"HALLMARK_MYC_TARGETS_V2",pval:.8557,padj:.9023,log2err:.0258,ES:-.3949,NES:-.7885,size:58,leadingEdge:"PA2G4,TMEM97,PRMT3,PPAN,CBX3,PLK4,AIMP2,PES1,MPHOSPH10,MYBBP1A,SORD,SLC19A1,NIP7,SLC29A2,TFB2M,PLK1,RRP9,NDUFAF4,BYSL,RABEPK,RCL1",id:10},{pathway:"HALLMARK_HEDGEHOG_SIGNALING",pval:.8829,padj:.9023,log2err:.0272,ES:-.3962,NES:-.746,size:36,leadingEdge:"CDK6,THY1,CELSR1,LDB1,L1CAM,AMOT,TLE3",id:11},{pathway:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",pval:.8841,padj:.9023,log2err:.0234,ES:-.3814,NES:-.7775,size:73,leadingEdge:"ACTG1,CLU,EBP,TMEM97,CD9,ANXA5,ALDOC,ANTXR2,ECH1,ATXN2,LGALS3,TNFRSF12A,CPEB2,LSS,HMGCR,CYP51A1,STX5,CTNNB1,ACAT2,TM7SF2,NSDHL,ETHE1,PLAUR,FDPS,CHKA,SQLE,JAG1",id:12},{pathway:"HALLMARK_NOTCH_SIGNALING",pval:.9023,padj:.9023,log2err:.0265,ES:-.3957,NES:-.7327,size:32,leadingEdge:"ARRB1,FBXW11,APH1A,DLL1,PSEN2,PSENEN,RBX1,CUL1,NOTCH2,KAT2A,JAG1,HEYL,SKP1,PPARD,DTX2,LFNG",id:13}]}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){$(t,e,r),i(e,n,r),$(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),A(t,e),A(a,e)}}}function Ka(e){let t,n,a,s;return t=new ue({props:{$$slots:{default:[Ua]},$$scope:{ctx:e}}}),a=new ue({props:{$$slots:{default:[Da]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){$(t,e,r),i(e,n,r),$(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),A(t,e),A(a,e)}}}function Va(e){let t,n;return t=new ge({props:{$$slots:{content:[Ka],default:[Fa]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){$(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){A(t,e)}}}function za(e){let t,n,a,s,c,p,m,g,f,b,h,C,L,v,R,y,x,N,M,P,I,w,G,B,F,H,U,O,D,K,V,z,k,q,j,X,W,Y,J,Z,Q,te,ne,ae,se,re,le,ie,oe,ce,pe;return a=new Aa({props:{title:"HALLMARK_INTERFERON_ALPHA_RESPONSE",class:"pipen-report-table-image-descr"}}),c=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells/fgsea_HALLMARK_INTERFERON_ALPHA_RESPONSE.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells/fgsea_HALLMARK_INTERFERON_ALPHA_RESPONSE.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),g=new Aa({props:{title:"HALLMARK_PROTEIN_SECRETION",class:"pipen-report-table-image-descr"}}),b=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells/fgsea_HALLMARK_PROTEIN_SECRETION.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells/fgsea_HALLMARK_PROTEIN_SECRETION.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),L=new Aa({props:{title:"HALLMARK_ANGIOGENESIS",class:"pipen-report-table-image-descr"}}),R=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells/fgsea_HALLMARK_ANGIOGENESIS.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells/fgsea_HALLMARK_ANGIOGENESIS.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),N=new Aa({props:{title:"HALLMARK_IL6_JAK_STAT3_SIGNALING",class:"pipen-report-table-image-descr"}}),P=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells/fgsea_HALLMARK_IL6_JAK_STAT3_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells/fgsea_HALLMARK_IL6_JAK_STAT3_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),G=new Aa({props:{title:"HALLMARK_APICAL_SURFACE",class:"pipen-report-table-image-descr"}}),F=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells/fgsea_HALLMARK_APICAL_SURFACE.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells/fgsea_HALLMARK_APICAL_SURFACE.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),O=new Aa({props:{title:"HALLMARK_ANDROGEN_RESPONSE",class:"pipen-report-table-image-descr"}}),K=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells/fgsea_HALLMARK_ANDROGEN_RESPONSE.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells/fgsea_HALLMARK_ANDROGEN_RESPONSE.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),k=new Aa({props:{title:"HALLMARK_TGF_BETA_SIGNALING",class:"pipen-report-table-image-descr"}}),j=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells/fgsea_HALLMARK_TGF_BETA_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells/fgsea_HALLMARK_TGF_BETA_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),Y=new Aa({props:{title:"HALLMARK_WNT_BETA_CATENIN_SIGNALING",class:"pipen-report-table-image-descr"}}),Z=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells/fgsea_HALLMARK_WNT_BETA_CATENIN_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells/fgsea_HALLMARK_WNT_BETA_CATENIN_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new Aa({props:{title:"HALLMARK_PANCREAS_BETA_CELLS",class:"pipen-report-table-image-descr"}}),se=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells/fgsea_HALLMARK_PANCREAS_BETA_CELLS.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells/fgsea_HALLMARK_PANCREAS_BETA_CELLS.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),ie=new Aa({props:{title:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",class:"pipen-report-table-image-descr"}}),ce=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells/fgsea_HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells/fgsea_HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=r("div"),n=r("div"),u(a.$$.fragment),s=S(),u(c.$$.fragment),p=S(),m=r("div"),u(g.$$.fragment),f=S(),u(b.$$.fragment),h=S(),C=r("div"),u(L.$$.fragment),v=S(),u(R.$$.fragment),y=S(),x=r("div"),u(N.$$.fragment),M=S(),u(P.$$.fragment),I=S(),w=r("div"),u(G.$$.fragment),B=S(),u(F.$$.fragment),H=S(),U=r("div"),u(O.$$.fragment),D=S(),u(K.$$.fragment),V=S(),z=r("div"),u(k.$$.fragment),q=S(),u(j.$$.fragment),X=S(),W=r("div"),u(Y.$$.fragment),J=S(),u(Z.$$.fragment),Q=S(),te=r("div"),u(ne.$$.fragment),ae=S(),u(se.$$.fragment),re=S(),le=r("div"),u(ie.$$.fragment),oe=S(),u(ce.$$.fragment),l(t,"class","pipen-report-table-of-images"),ee(t,"grid-template-columns","repeat(2, auto)")},m(e,r){i(e,t,r),E(t,n),$(a,n,null),E(n,s),$(c,n,null),E(t,p),E(t,m),$(g,m,null),E(m,f),$(b,m,null),E(t,h),E(t,C),$(L,C,null),E(C,v),$(R,C,null),E(t,y),E(t,x),$(N,x,null),E(x,M),$(P,x,null),E(t,I),E(t,w),$(G,w,null),E(w,B),$(F,w,null),E(t,H),E(t,U),$(O,U,null),E(U,D),$(K,U,null),E(t,V),E(t,z),$(k,z,null),E(z,q),$(j,z,null),E(t,X),E(t,W),$(Y,W,null),E(W,J),$(Z,W,null),E(t,Q),E(t,te),$(ne,te,null),E(te,ae),$(se,te,null),E(t,re),E(t,le),$(ie,le,null),E(le,oe),$(ce,le,null),pe=!0},p:T,i(e){pe||(o(a.$$.fragment,e),o(c.$$.fragment,e),o(g.$$.fragment,e),o(b.$$.fragment,e),o(L.$$.fragment,e),o(R.$$.fragment,e),o(N.$$.fragment,e),o(P.$$.fragment,e),o(G.$$.fragment,e),o(F.$$.fragment,e),o(O.$$.fragment,e),o(K.$$.fragment,e),o(k.$$.fragment,e),o(j.$$.fragment,e),o(Y.$$.fragment,e),o(Z.$$.fragment,e),o(ne.$$.fragment,e),o(se.$$.fragment,e),o(ie.$$.fragment,e),o(ce.$$.fragment,e),pe=!0)},o(e){_(a.$$.fragment,e),_(c.$$.fragment,e),_(g.$$.fragment,e),_(b.$$.fragment,e),_(L.$$.fragment,e),_(R.$$.fragment,e),_(N.$$.fragment,e),_(P.$$.fragment,e),_(G.$$.fragment,e),_(F.$$.fragment,e),_(O.$$.fragment,e),_(K.$$.fragment,e),_(k.$$.fragment,e),_(j.$$.fragment,e),_(Y.$$.fragment,e),_(Z.$$.fragment,e),_(ne.$$.fragment,e),_(se.$$.fragment,e),_(ie.$$.fragment,e),_(ce.$$.fragment,e),pe=!1},d(e){e&&d(t),A(a),A(c),A(g),A(b),A(L),A(R),A(N),A(P),A(G),A(F),A(O),A(K),A(k),A(j),A(Y),A(Z),A(ne),A(se),A(ie),A(ce)}}}function ka(e){let t,n,a,s;return t=new Ie({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[Va]},$$scope:{ctx:e}}}),a=new Ie({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[za]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){$(t,e,r),i(e,n,r),$(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),A(t,e),A(a,e)}}}function qa(e){let t,n,a,s;return t=new fe({props:{label:"Plot"}}),a=new fe({props:{label:"Table"}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){$(t,e,r),i(e,n,r),$(a,e,r),s=!0},p:T,i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),A(t,e),A(a,e)}}}function ja(e){let t;return{c(){t=L("This table presents a comprehensive summary of the top enriched pathways derived from the fgsea. Each row corresponds to a pathway, and the gene ranks are shown based on the ranking metric used in the analysis. The enrichment score, p-value, and adjusted p-value are also provided to assess the significance of the enrichment.")},m(e,n){i(e,t,n)},d(e){e&&d(t)}}}function Xa(e){let t,n,a,s;return t=new Aa({props:{$$slots:{default:[ja]},$$scope:{ctx:e}}}),a=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells-1/gsea_table.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells-1/gsea_table.pdf",width:1e3,height:760}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){$(t,e,r),i(e,n,r),$(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),A(t,e),A(a,e)}}}function Wa(e){let t;return{c(){t=L("This plot represents the GSEA results for a specified gene set, illustrating the distribution and impact of the gene set along the ranked list of genes. The running enrichment score curve shows the cumulative enrichment score as genes from the input list are encountered. Positive peaks on the curve indicate regions where members of the gene set are predominantly found.")},m(e,n){i(e,t,n)},d(e){e&&d(t)}}}function Ya(e){let t,n,a,s;return t=new Aa({props:{$$slots:{default:[Wa]},$$scope:{ctx:e}}}),a=new ga({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells-1/fgsea.txt",data:[{pathway:"HALLMARK_INTERFERON_ALPHA_RESPONSE",pval:.0062,padj:.0643,log2err:.407,ES:.5354,NES:1.4183,size:96,leadingEdge:"ISG20,B2M,IRF1,TAP1,UBE2L6,PSMB9,IFITM3,PARP9,IFI44L,CD74,IRF7,GBP2,PSME2,CSF1,IFI44,TMEM140,IFITM2,IFIH1,LY6E,EPSTI1,IFITM1,GBP4,SELL,IL15,SP110,PARP12,USP18,RSAD2,PSME1,LAMP3,TRIM25,IFI35,PNPT1,EIF2AK2,CCRL2,ELF1",id:0},{pathway:"HALLMARK_ANGIOGENESIS",pval:.0092,padj:.0643,log2err:.3807,ES:-.7639,NES:-1.4042,size:36,leadingEdge:"S100A4,CCND2,TIMP1,APP",id:1},{pathway:"HALLMARK_IL6_JAK_STAT3_SIGNALING",pval:.0731,padj:.3411,log2err:.2878,ES:.453,NES:1.1745,size:87,leadingEdge:"IRF1,IL18R1,STAT1,IL6ST,IL2RA,STAT3,MAP3K8,CSF1,CSF2,MYD88,HAX1,EBI3,IL13RA1,HMOX1,IFNGR1,FAS,IL17RB,IFNGR2,CD44,CD38,SOCS1,TNFRSF1B,SOCS3,PTPN2,IL15RA,PTPN1",id:2},{pathway:"HALLMARK_ANDROGEN_RESPONSE",pval:.1206,padj:.3762,log2err:.2878,ES:.4157,NES:1.112,size:99,leadingEdge:"FKBP5,B2M,ARID5B,SAT1,ZBTB10,ELL2,ELK4,UBE2J1,NKX3-1,RPS6KA3,SRF,AKT1,DNAJB9,ABCC4,TPD52,SCD,CCND3,KRT8,PLPP1,MAK,LMAN1,GPD1L,HPGD,SRP19,SGK1,DHCR24,B4GALT1,HERC3,ABHD2",id:3},{pathway:"HALLMARK_PROTEIN_SECRETION",pval:.1343,padj:.3762,log2err:.1215,ES:-.5775,NES:-1.1812,size:95,leadingEdge:"CD63,M6PR,ANP32E,VPS4B,CLTA,ATP1A1,SEC22B,AP2M1,SCAMP1,AP2S1,CLTC,ZW10,RER1,COPB2,SNX2,VPS45,YIPF6,RAB5A,TMED10,GOSR2,CTSC,ARFIP1,AP2B1,KRT18,LAMP2,PPT1,OCRL,SEC31A,AP1G1,RAB14,TMED2,SCAMP3,CLCN3,NAPG,STX16,AP3B1,ARFGEF1,RAB9A,DST,COPE,ARFGAP3,SNAP23,COG2,COPB1,GALC,GOLGA4,GNAS,SEC24D,CLN5,ARFGEF2,STX7,BNIP3,VAMP4,IGF2R,ATP7A,ARF1,PAM",id:4},{pathway:"HALLMARK_APICAL_SURFACE",pval:.23,padj:.4844,log2err:.092,ES:-.6068,NES:-1.1464,size:44,leadingEdge:"THY1,AKAP7,LYN,CD160,SRPX,IL2RG,GSTM3,MAL,ATP8B1,APP,ADIPOR2,GAS1,SULF2,CROCC,FLOT2,SCUBE1,DCBLD2",id:5},{pathway:"HALLMARK_MYC_TARGETS_V2",pval:.2432,padj:.4844,log2err:.2765,ES:.4586,NES:1.1128,size:58,leadingEdge:"DUSP2,HSPD1,HSPE1,PUS1,NOP16,SRM,DCTPP1,WDR43,MYC,NOLC1,PLK1,GNL3,IPO4,WDR74,NDUFAF4,PHB,EXOSC5,MYBBP1A,FARSA,RRP9,PRMT3,PPRC1,NOP56,HK2,IMP4,SLC19A1,SUPV3L1,RCL1",id:6},{pathway:"HALLMARK_TGF_BETA_SIGNALING",pval:.2768,padj:.4844,log2err:.2572,ES:.4643,NES:1.1026,size:54,leadingEdge:"SKIL,PPP1R15A,FURIN,ID2,TGIF1,JUNB,THBS1,SMURF2,ARID4B,XIAP,SERPINE1,ENG,SMAD7,IFNGR2,ACVR1,ID1",id:7},{pathway:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",pval:.4038,padj:.6282,log2err:.2193,ES:.3955,NES:1.009,size:73,leadingEdge:"ATF3,LDLR,TP53INP1,PNRC1,SREBF2,LGMN,FBXO6,STX5,ABCA2,STARD4,CYP51A1,ERRFI1,SCD,ACAT2,FDPS,TM7SF2,MVK,FASN,CHKA,SC5D,LGALS3,FDFT1,DHCR7",id:8},{pathway:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",pval:.4704,padj:.6349,log2err:.0545,ES:-.5305,NES:-1.0193,size:49,leadingEdge:"LSP1,CAT,PRNP,FTL,GPX4,MSRA,EGLN2,PRDX2,TXNRD1,SCAF4,GCLM,HHEX,NDUFB4,PFKP,NDUFS2,CDKN2D,PRDX1,GLRX2,LAMTOR5,G6PD",id:9},{pathway:"HALLMARK_PANCREAS_BETA_CELLS",pval:.5065,padj:.6349,log2err:.053,ES:-.5344,NES:-.9927,size:40,leadingEdge:"SRP14,SPCS1,SRP9,DPP4,ELP4,MAFB,SEC11A",id:10},{pathway:"HALLMARK_WNT_BETA_CATENIN_SIGNALING",pval:.5442,padj:.6349,log2err:.0493,ES:-.5178,NES:-.9729,size:42,leadingEdge:"CCND2,DVL2,LEF1,WNT1,HDAC5,NCSTN,AXIN2,NOTCH4,PTCH1,HDAC11,AXIN1,PPARD,CTNNB1,HDAC2,ADAM17,PSEN2",id:11},{pathway:"HALLMARK_HEDGEHOG_SIGNALING",pval:.6996,padj:.7535,log2err:.0383,ES:-.4846,NES:-.8907,size:36,leadingEdge:"THY1,CDK6,CELSR1,CDK5R1,OPHN1,AMOT,L1CAM,PLG,PTCH1,SLIT1,VEGFA,ACHE,VLDLR,DPYSL2,NF1",id:12},{pathway:"HALLMARK_NOTCH_SIGNALING",pval:.8538,padj:.8538,log2err:.1106,ES:.3684,NES:.8207,size:32,leadingEdge:"SAP30,DTX1,NOTCH1,PRKCA,NOTCH2,TCF7L2",id:13}]}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){$(t,e,r),i(e,n,r),$(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),A(t,e),A(a,e)}}}function Ja(e){let t,n,a,s;return t=new ue({props:{$$slots:{default:[Xa]},$$scope:{ctx:e}}}),a=new ue({props:{$$slots:{default:[Ya]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){$(t,e,r),i(e,n,r),$(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),A(t,e),A(a,e)}}}function Za(e){let t,n;return t=new ge({props:{$$slots:{content:[Ja],default:[qa]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){$(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){A(t,e)}}}function Qa(e){let t,n,a,s,c,p,m,g,f,b,h,C,L,v,R,y,x,N,M,P,I,w,G,B,F,H,U,O,D,K,V,z,k,q,j,X,W,Y,J,Z,Q,te,ne,ae,se,re,le,ie,oe,ce,pe;return a=new Aa({props:{title:"HALLMARK_INTERFERON_ALPHA_RESPONSE",class:"pipen-report-table-image-descr"}}),c=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells-1/fgsea_HALLMARK_INTERFERON_ALPHA_RESPONSE.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells-1/fgsea_HALLMARK_INTERFERON_ALPHA_RESPONSE.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),g=new Aa({props:{title:"HALLMARK_ANGIOGENESIS",class:"pipen-report-table-image-descr"}}),b=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells-1/fgsea_HALLMARK_ANGIOGENESIS.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells-1/fgsea_HALLMARK_ANGIOGENESIS.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),L=new Aa({props:{title:"HALLMARK_IL6_JAK_STAT3_SIGNALING",class:"pipen-report-table-image-descr"}}),R=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells-1/fgsea_HALLMARK_IL6_JAK_STAT3_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells-1/fgsea_HALLMARK_IL6_JAK_STAT3_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),N=new Aa({props:{title:"HALLMARK_ANDROGEN_RESPONSE",class:"pipen-report-table-image-descr"}}),P=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells-1/fgsea_HALLMARK_ANDROGEN_RESPONSE.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells-1/fgsea_HALLMARK_ANDROGEN_RESPONSE.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),G=new Aa({props:{title:"HALLMARK_PROTEIN_SECRETION",class:"pipen-report-table-image-descr"}}),F=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells-1/fgsea_HALLMARK_PROTEIN_SECRETION.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells-1/fgsea_HALLMARK_PROTEIN_SECRETION.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),O=new Aa({props:{title:"HALLMARK_APICAL_SURFACE",class:"pipen-report-table-image-descr"}}),K=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells-1/fgsea_HALLMARK_APICAL_SURFACE.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells-1/fgsea_HALLMARK_APICAL_SURFACE.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),k=new Aa({props:{title:"HALLMARK_MYC_TARGETS_V2",class:"pipen-report-table-image-descr"}}),j=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells-1/fgsea_HALLMARK_MYC_TARGETS_V2.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells-1/fgsea_HALLMARK_MYC_TARGETS_V2.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),Y=new Aa({props:{title:"HALLMARK_TGF_BETA_SIGNALING",class:"pipen-report-table-image-descr"}}),Z=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells-1/fgsea_HALLMARK_TGF_BETA_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells-1/fgsea_HALLMARK_TGF_BETA_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new Aa({props:{title:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",class:"pipen-report-table-image-descr"}}),se=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells-1/fgsea_HALLMARK_CHOLESTEROL_HOMEOSTASIS.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells-1/fgsea_HALLMARK_CHOLESTEROL_HOMEOSTASIS.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),ie=new Aa({props:{title:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",class:"pipen-report-table-image-descr"}}),ce=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells-1/fgsea_HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Type-1-helper-T-cells-1/fgsea_HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=r("div"),n=r("div"),u(a.$$.fragment),s=S(),u(c.$$.fragment),p=S(),m=r("div"),u(g.$$.fragment),f=S(),u(b.$$.fragment),h=S(),C=r("div"),u(L.$$.fragment),v=S(),u(R.$$.fragment),y=S(),x=r("div"),u(N.$$.fragment),M=S(),u(P.$$.fragment),I=S(),w=r("div"),u(G.$$.fragment),B=S(),u(F.$$.fragment),H=S(),U=r("div"),u(O.$$.fragment),D=S(),u(K.$$.fragment),V=S(),z=r("div"),u(k.$$.fragment),q=S(),u(j.$$.fragment),X=S(),W=r("div"),u(Y.$$.fragment),J=S(),u(Z.$$.fragment),Q=S(),te=r("div"),u(ne.$$.fragment),ae=S(),u(se.$$.fragment),re=S(),le=r("div"),u(ie.$$.fragment),oe=S(),u(ce.$$.fragment),l(t,"class","pipen-report-table-of-images"),ee(t,"grid-template-columns","repeat(2, auto)")},m(e,r){i(e,t,r),E(t,n),$(a,n,null),E(n,s),$(c,n,null),E(t,p),E(t,m),$(g,m,null),E(m,f),$(b,m,null),E(t,h),E(t,C),$(L,C,null),E(C,v),$(R,C,null),E(t,y),E(t,x),$(N,x,null),E(x,M),$(P,x,null),E(t,I),E(t,w),$(G,w,null),E(w,B),$(F,w,null),E(t,H),E(t,U),$(O,U,null),E(U,D),$(K,U,null),E(t,V),E(t,z),$(k,z,null),E(z,q),$(j,z,null),E(t,X),E(t,W),$(Y,W,null),E(W,J),$(Z,W,null),E(t,Q),E(t,te),$(ne,te,null),E(te,ae),$(se,te,null),E(t,re),E(t,le),$(ie,le,null),E(le,oe),$(ce,le,null),pe=!0},p:T,i(e){pe||(o(a.$$.fragment,e),o(c.$$.fragment,e),o(g.$$.fragment,e),o(b.$$.fragment,e),o(L.$$.fragment,e),o(R.$$.fragment,e),o(N.$$.fragment,e),o(P.$$.fragment,e),o(G.$$.fragment,e),o(F.$$.fragment,e),o(O.$$.fragment,e),o(K.$$.fragment,e),o(k.$$.fragment,e),o(j.$$.fragment,e),o(Y.$$.fragment,e),o(Z.$$.fragment,e),o(ne.$$.fragment,e),o(se.$$.fragment,e),o(ie.$$.fragment,e),o(ce.$$.fragment,e),pe=!0)},o(e){_(a.$$.fragment,e),_(c.$$.fragment,e),_(g.$$.fragment,e),_(b.$$.fragment,e),_(L.$$.fragment,e),_(R.$$.fragment,e),_(N.$$.fragment,e),_(P.$$.fragment,e),_(G.$$.fragment,e),_(F.$$.fragment,e),_(O.$$.fragment,e),_(K.$$.fragment,e),_(k.$$.fragment,e),_(j.$$.fragment,e),_(Y.$$.fragment,e),_(Z.$$.fragment,e),_(ne.$$.fragment,e),_(se.$$.fragment,e),_(ie.$$.fragment,e),_(ce.$$.fragment,e),pe=!1},d(e){e&&d(t),A(a),A(c),A(g),A(b),A(L),A(R),A(N),A(P),A(G),A(F),A(O),A(K),A(k),A(j),A(Y),A(Z),A(ne),A(se),A(ie),A(ce)}}}function es(e){let t,n,a,s;return t=new Ie({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[Za]},$$scope:{ctx:e}}}),a=new Ie({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[Qa]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){$(t,e,r),i(e,n,r),$(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),A(t,e),A(a,e)}}}function ts(e){let t,n,a,s,c,p,m,g,f,b,h,C,L,E,v,R,T,y,x,N;return s=new Pe({props:{align:"start",$$slots:{default:[Ta]},$$scope:{ctx:e}}}),f=new Pe({props:{align:"start",$$slots:{default:[Ba]},$$scope:{ctx:e}}}),E=new Pe({props:{align:"start",$$slots:{default:[ka]},$$scope:{ctx:e}}}),x=new Pe({props:{align:"start",$$slots:{default:[es]},$$scope:{ctx:e}}}),{c(){t=r("h1"),t.textContent="seurat_clusters - Tem/Trm cytotoxic T cells.1",n=r("a"),n.innerHTML="",a=S(),u(s.$$.fragment),c=S(),p=r("h1"),p.textContent="seurat_clusters - Trm cytotoxic T cells",m=r("a"),m.innerHTML="",g=S(),u(f.$$.fragment),b=S(),h=r("h1"),h.textContent="seurat_clusters - Type 1 helper T cells",C=r("a"),C.innerHTML="",L=S(),u(E.$$.fragment),v=S(),R=r("h1"),R.textContent="seurat_clusters - Type 1 helper T cells.1",T=r("a"),T.innerHTML="",y=S(),u(x.$$.fragment),l(n,"id","prt-h1-8-seurat-clusters-tem-trm-cytotoxic-t-cells-1"),l(n,"class","pipen-report-toc-anchor"),l(m,"id","prt-h1-9-seurat-clusters-trm-cytotoxic-t-cells"),l(m,"class","pipen-report-toc-anchor"),l(C,"id","prt-h1-10-seurat-clusters-type-1-helper-t-cells"),l(C,"class","pipen-report-toc-anchor"),l(T,"id","prt-h1-11-seurat-clusters-type-1-helper-t-cells-1"),l(T,"class","pipen-report-toc-anchor")},m(e,r){i(e,t,r),i(e,n,r),i(e,a,r),$(s,e,r),i(e,c,r),i(e,p,r),i(e,m,r),i(e,g,r),$(f,e,r),i(e,b,r),i(e,h,r),i(e,C,r),i(e,L,r),$(E,e,r),i(e,v,r),i(e,R,r),i(e,T,r),i(e,y,r),$(x,e,r),N=!0},p(e,[t]){const n={};1&t&&(n.$$scope={dirty:t,ctx:e}),s.$set(n);const a={};1&t&&(a.$$scope={dirty:t,ctx:e}),f.$set(a);const r={};1&t&&(r.$$scope={dirty:t,ctx:e}),E.$set(r);const l={};1&t&&(l.$$scope={dirty:t,ctx:e}),x.$set(l)},i(e){N||(o(s.$$.fragment,e),o(f.$$.fragment,e),o(E.$$.fragment,e),o(x.$$.fragment,e),N=!0)},o(e){_(s.$$.fragment,e),_(f.$$.fragment,e),_(E.$$.fragment,e),_(x.$$.fragment,e),N=!1},d(e){e&&(d(t),d(n),d(a),d(c),d(p),d(m),d(g),d(b),d(h),d(C),d(L),d(v),d(R),d(T),d(y)),A(s,e),A(f,e),A(E,e),A(x,e)}}}class ns extends t{constructor(e){super(),n(this,e,null,ts,a,{})}}var as=[{slug:"prt-h1-0-seurat-clusters-cycling-t-cells",text:"seurat_clusters - Cycling T cells",children:[],page:0},{slug:"prt-h1-1-seurat-clusters-follicular-helper-t-cells",text:"seurat_clusters - Follicular helper T cells",children:[],page:0},{slug:"prt-h1-2-seurat-clusters-gamma-delta-t-cells",text:"seurat_clusters - gamma-delta T cells",children:[],page:0},{slug:"prt-h1-3-seurat-clusters-plasmablasts",text:"seurat_clusters - Plasmablasts",children:[],page:0},{slug:"prt-h1-4-seurat-clusters-regulatory-t-cells",text:"seurat_clusters - Regulatory T cells",children:[],page:0},{slug:"prt-h1-5-seurat-clusters-regulatory-t-cells-1",text:"seurat_clusters - Regulatory T cells.1",children:[],page:0},{slug:"prt-h1-6-seurat-clusters-tcm-naive-helper-t-cells",text:"seurat_clusters - Tcm/Naive helper T cells",children:[],page:0},{slug:"prt-h1-7-seurat-clusters-tem-trm-cytotoxic-t-cells",text:"seurat_clusters - Tem/Trm cytotoxic T cells",children:[],page:0},{slug:"prt-h1-8-seurat-clusters-tem-trm-cytotoxic-t-cells-1",text:"seurat_clusters - Tem/Trm cytotoxic T cells.1",children:[],page:1},{slug:"prt-h1-9-seurat-clusters-trm-cytotoxic-t-cells",text:"seurat_clusters - Trm cytotoxic T cells",children:[],page:1},{slug:"prt-h1-10-seurat-clusters-type-1-helper-t-cells",text:"seurat_clusters - Type 1 helper T cells",children:[],page:1},{slug:"prt-h1-11-seurat-clusters-type-1-helper-t-cells-1",text:"seurat_clusters - Type 1 helper T cells.1",children:[],page:1}],ss={pipeline:{name:"Immunopipe",desc:"Immunopipe (v1.4.3): Integrative analysis for scRNA-seq and scTCR-seq data"},versions:'\n    <a href="https://github.com/pwwang/pipen" target="_blank">\n        pipen v0.15.7\n    </a> and\n    <a href="https://github.com/pwwang/pipen-report" target="_blank">\n        pipen-report v0.21.4\n    </a>\n',entries:[{name:"SampleInfo",desc:"List sample information and perform statistics",npages:1,report_toc:!0,order:0,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n[1] ggrepel_0.9.6    ggprism_1.0.5    ggplot2_3.5.1    tidyr_1.3.1     \n[5] dplyr_1.1.4      tidyselect_1.2.1 jsonlite_1.8.9   rlang_1.1.5     \n[9] logger_0.4.0    \n\nloaded via a namespace (and not attached):\n [1] vctrs_0.6.5      cli_3.6.4        purrr_1.0.2      generics_0.1.3  \n [5] glue_1.8.0       labeling_0.4.3   colorspace_2.1-1 scales_1.3.0    \n [9] grid_4.3.3       munsell_0.5.1    tibble_3.2.1     lifecycle_1.0.4 \n[13] compiler_4.3.3   Rcpp_1.0.14      pkgconfig_2.0.3  farver_2.1.2    \n[17] digest_0.6.37    R6_2.6.0         pillar_1.10.1    magrittr_2.0.3  \n[21] withr_3.0.2      gtable_0.3.6    \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 2.19\nuser 11.15\nsys 0.11\npercent 513%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3421.613\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"SeuratPreparing",desc:"Load, prepare and apply QC to data, using `Seurat`",npages:1,report_toc:!0,order:2,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats4    stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] scDblFinder_1.16.0          SingleCellExperiment_1.24.0\n [3] SummarizedExperiment_1.32.0 Biobase_2.62.0             \n [5] GenomicRanges_1.54.1        GenomeInfoDb_1.38.1        \n [7] IRanges_2.36.0              S4Vectors_0.40.2           \n [9] BiocGenerics_0.48.1         MatrixGenerics_1.14.0      \n[11] matrixStats_1.1.0           dplyr_1.1.4                \n[13] ggplot2_3.5.1               bracer_1.2.1               \n[15] future_1.34.0               Seurat_5.1.0               \n[17] SeuratObject_5.0.2          sp_2.2-0                   \n[19] digest_0.6.37               jsonlite_1.8.9             \n[21] rlang_1.1.5                 logger_0.4.0               \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22          splines_4.3.3            \n  [3] later_1.4.1               BiocIO_1.12.0            \n  [5] bitops_1.0-9              tibble_3.2.1             \n  [7] R.oo_1.27.0               polyclip_1.10-7          \n  [9] XML_3.99-0.17             fastDummies_1.7.5        \n [11] lifecycle_1.0.4           edgeR_4.0.16             \n [13] globals_0.16.3            lattice_0.22-6           \n [15] MASS_7.3-60.0.1           magrittr_2.0.3           \n [17] limma_3.58.1              plotly_4.10.4            \n [19] yaml_2.3.10               metapod_1.10.0           \n [21] httpuv_1.6.15             sctransform_0.4.1        \n [23] spam_2.11-1               spatstat.sparse_3.1-0    \n [25] reticulate_1.40.0         cowplot_1.1.3            \n [27] pbapply_1.7-2             RColorBrewer_1.1-3       \n [29] abind_1.4-5               zlibbioc_1.48.0          \n [31] Rtsne_0.17                purrr_1.0.2              \n [33] R.utils_2.12.3            RCurl_1.98-1.16          \n [35] GenomeInfoDbData_1.2.11   ggrepel_0.9.6            \n [37] irlba_2.3.5.1             listenv_0.9.1            \n [39] spatstat.utils_3.1-2      goftest_1.2-3            \n [41] RSpectra_0.16-2           dqrng_0.3.2              \n [43] spatstat.random_3.3-2     fitdistrplus_1.2-2       \n [45] parallelly_1.42.0         DelayedMatrixStats_1.24.0\n [47] leiden_0.4.3.1            codetools_0.2-20         \n [49] DelayedArray_0.28.0       scuttle_1.12.0           \n [51] tidyselect_1.2.1          farver_2.1.2             \n [53] viridis_0.6.5             ScaledMatrix_1.10.0      \n [55] spatstat.explore_3.3-4    GenomicAlignments_1.38.0 \n [57] BiocNeighbors_1.20.0      progressr_0.15.1         \n [59] scater_1.30.1             ggridges_0.5.6           \n [61] survival_3.8-3            systemfonts_1.2.1        \n [63] tools_4.3.3               ragg_1.3.3               \n [65] ica_1.0-3                 Rcpp_1.0.14              \n [67] glue_1.8.0                gridExtra_2.3            \n [69] SparseArray_1.2.2         withr_3.0.2              \n [71] fastmap_1.2.0             bluster_1.12.0           \n [73] rsvd_1.0.5                R6_2.6.0                 \n [75] mime_0.12                 textshaping_0.4.0        \n [77] colorspace_2.1-1          scattermore_1.2          \n [79] tensor_1.5                spatstat.data_3.1-4      \n [81] R.methodsS3_1.8.2         RhpcBLASctl_0.23-42      \n [83] tidyr_1.3.1               generics_0.1.3           \n [85] data.table_1.16.4         rtracklayer_1.62.0       \n [87] httr_1.4.7                htmlwidgets_1.6.4        \n [89] S4Arrays_1.2.0            uwot_0.2.2               \n [91] pkgconfig_2.0.3           gtable_0.3.6             \n [93] lmtest_0.9-40             XVector_0.42.0           \n [95] htmltools_0.5.8.1         dotCall64_1.2            \n [97] scales_1.3.0              png_0.1-8                \n [99] harmony_1.2.0             spatstat.univar_3.1-1    \n[101] scran_1.30.0              reshape2_1.4.4           \n[103] rjson_0.2.23              nlme_3.1-167             \n[105] zoo_1.8-12                stringr_1.5.1            \n[107] KernSmooth_2.23-26        vipor_0.4.7              \n[109] parallel_4.3.3            miniUI_0.1.1.1           \n[111] restfulr_0.0.15           pillar_1.10.1            \n[113] grid_4.3.3                vctrs_0.6.5              \n[115] RANN_2.6.2                promises_1.3.2           \n[117] BiocSingular_1.18.0       beachmat_2.18.0          \n[119] xtable_1.8-4              cluster_2.1.8            \n[121] beeswarm_0.4.0            locfit_1.5-9.11          \n[123] cli_3.6.4                 compiler_4.3.3           \n[125] Rsamtools_2.18.0          crayon_1.5.3             \n[127] future.apply_1.11.3       labeling_0.4.3           \n[129] ggbeeswarm_0.7.2          plyr_1.8.9               \n[131] stringi_1.8.4             viridisLite_0.4.2        \n[133] deldir_2.0-4              BiocParallel_1.36.0      \n[135] munsell_0.5.1             Biostrings_2.70.1        \n[137] lazyeval_0.2.2            spatstat.geom_3.3-5      \n[139] Matrix_1.6-5              RcppHNSW_0.6.0           \n[141] patchwork_1.3.0           sparseMatrixStats_1.14.0 \n[143] statmod_1.5.0             shiny_1.10.0             \n[145] ROCR_1.0-11               igraph_2.0.3             \n[147] xgboost_2.1.4.1          \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 209.65\nuser 1454.55\nsys 7.57\npercent 697%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ClusterMarkersOfAllCells",desc:"Markers for clusters of all cells.",npages:3,report_toc:!0,order:4,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] UpSetR_1.4.0        ggVennDiagram_1.2.3 tidyseurat_0.8.0   \n [4] ttservice_0.4.1     future_1.34.0       ggrepel_0.9.6      \n [7] ggprism_1.0.5       ggplot2_3.5.1       enrichR_3.4        \n[10] Seurat_5.1.0        SeuratObject_5.0.2  sp_2.2-0           \n[13] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[16] tidyselect_1.2.1    digest_0.6.37       jsonlite_1.8.9     \n[19] rlang_1.1.5         logger_0.4.0       \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] ellipsis_0.3.2         labeling_0.4.3         promises_1.3.2        \n [19] purrr_1.0.2            WriteXLS_6.7.0         RVenn_1.1.0           \n [22] goftest_1.2-3          later_1.4.1            spatstat.utils_3.1-2  \n [25] irlba_2.3.5.1          parallel_4.3.3         cluster_2.1.8         \n [28] R6_2.6.0               ica_1.0-3              stringi_1.8.4         \n [31] RColorBrewer_1.1-3     spatstat.data_3.1-4    limma_3.58.1          \n [34] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [37] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [40] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [43] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [46] splines_4.3.3          igraph_2.0.3           abind_1.4-5           \n [49] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.3-4 curl_6.0.1             listenv_0.9.1         \n [55] lattice_0.22-6         plyr_1.8.9             withr_3.0.2           \n [58] shiny_1.10.0           ROCR_1.0-11            Rtsne_0.17            \n [61] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n [64] fitdistrplus_1.2-2     pillar_1.10.1          KernSmooth_2.23-26    \n [67] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n [82] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n [85] nlme_3.1-167           patchwork_1.3.0        presto_1.0.0          \n [88] cli_3.6.4              spatstat.sparse_3.1-0  fansi_1.0.6           \n [91] spam_2.11-1            viridisLite_0.4.2      uwot_0.2.2            \n [94] gtable_0.3.6           progressr_0.15.1       rjson_0.2.23          \n [97] htmlwidgets_1.6.4      farver_2.1.2           htmltools_0.5.8.1     \n[100] lifecycle_1.0.4        httr_1.4.7             statmod_1.5.0         \n[103] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 259.76\nuser 186.00\nsys 8.62\npercent 74%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TopExpressingGenesOfAllCells",desc:"Top expressing genes for clusters of all cells.",npages:3,report_toc:!0,order:5,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ggprism_1.0.5      dplyr_1.1.4        enrichR_3.4        tibble_3.2.1      \n [5] Seurat_5.1.0       SeuratObject_5.0.2 sp_2.2-0           jsonlite_1.8.9    \n [9] rlang_1.1.5        logger_0.4.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] labeling_0.4.3         promises_1.3.2         purrr_1.0.2           \n [19] WriteXLS_6.7.0         goftest_1.2-3          later_1.4.1           \n [22] spatstat.utils_3.1-2   irlba_2.3.5.1          parallel_4.3.3        \n [25] cluster_2.1.8          R6_2.6.0               ica_1.0-3             \n [28] stringi_1.8.4          RColorBrewer_1.1-3     spatstat.data_3.1-4   \n [31] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [37] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [43] splines_4.3.3          igraph_2.0.3           tidyselect_1.2.1      \n [46] abind_1.4-5            spatstat.random_3.3-2  codetools_0.2-20      \n [49] miniUI_0.1.1.1         spatstat.explore_3.3-4 curl_6.0.1            \n [52] listenv_0.9.1          lattice_0.22-6         plyr_1.8.9            \n [55] withr_3.0.2            shiny_1.10.0           ROCR_1.0-11           \n [58] Rtsne_0.17             future_1.34.0          fastDummies_1.7.5     \n [61] survival_3.8-3         polyclip_1.10-7        fitdistrplus_1.2-2    \n [64] pillar_1.10.1          KernSmooth_2.23-26     plotly_4.10.4         \n [67] generics_0.1.3         RcppHNSW_0.6.0         ggplot2_3.5.1         \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n [82] cowplot_1.1.3          grid_4.3.3             tidyr_1.3.1           \n [85] colorspace_2.1-1       nlme_3.1-167           patchwork_1.3.0       \n [88] cli_3.6.4              spatstat.sparse_3.1-0  spam_2.11-1           \n [91] viridisLite_0.4.2      uwot_0.2.2             gtable_0.3.6          \n [94] digest_0.6.37          progressr_0.15.1       ggrepel_0.9.6         \n [97] rjson_0.2.23           htmlwidgets_1.6.4      farver_2.1.2          \n[100] htmltools_0.5.8.1      lifecycle_1.0.4        httr_1.4.7            \n[103] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 73.70\nuser 41.45\nsys 2.53\npercent 59%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TCellSelection",desc:"Separate T and non-T cells and select T cells.",npages:1,report_toc:!0,order:6,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] glue_1.8.0         factoextra_1.0.7   patchwork_1.3.0    ggpubr_0.6.0      \n [5] ggrepel_0.9.6      ggprism_1.0.5      ggplot2_3.5.1      tibble_3.2.1      \n [9] tidyr_1.3.1        dplyr_1.1.4        Seurat_5.1.0       SeuratObject_5.0.2\n[13] sp_2.2-0           jsonlite_1.8.9     rlang_1.1.5        logger_0.4.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] backports_1.5.0        labeling_0.4.3         promises_1.3.2        \n [19] purrr_1.0.2            goftest_1.2-3          later_1.4.1           \n [22] spatstat.utils_3.1-2   broom_1.0.7            irlba_2.3.5.1         \n [25] parallel_4.3.3         cluster_2.1.8          R6_2.6.0              \n [28] ica_1.0-3              stringi_1.8.4          RColorBrewer_1.1-3    \n [31] spatstat.data_3.1-4    reticulate_1.40.0      car_3.1-3             \n [34] parallelly_1.42.0      spatstat.univar_3.1-1  lmtest_0.9-40         \n [37] scattermore_1.2        Rcpp_1.0.14            tensor_1.5            \n [40] future.apply_1.11.3    zoo_1.8-12             sctransform_0.4.1     \n [43] httpuv_1.6.15          Matrix_1.6-5           splines_4.3.3         \n [46] igraph_2.0.3           tidyselect_1.2.1       abind_1.4-5           \n [49] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.3-4 listenv_0.9.1          lattice_0.22-6        \n [55] plyr_1.8.9             withr_3.0.2            shiny_1.10.0          \n [58] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n [61] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n [64] fitdistrplus_1.2-2     pillar_1.10.1          carData_3.0-5         \n [67] KernSmooth_2.23-26     plotly_4.10.4          generics_0.1.3        \n [70] RcppHNSW_0.6.0         munsell_0.5.1          scales_1.3.0          \n [73] globals_0.16.3         xtable_1.8-4           lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] ggsignif_0.6.4         RANN_2.6.2             leiden_0.4.3.1        \n [82] dotCall64_1.2          cowplot_1.1.3          grid_4.3.3            \n [85] colorspace_2.1-1       nlme_3.1-167           Formula_1.2-5         \n [88] cli_3.6.4              spatstat.sparse_3.1-0  spam_2.11-1           \n [91] viridisLite_0.4.2      uwot_0.2.2             gtable_0.3.6          \n [94] rstatix_0.7.2          digest_0.6.37          progressr_0.15.1      \n [97] htmlwidgets_1.6.4      farver_2.1.2           htmltools_0.5.8.1     \n[100] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[103] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 41.63\nuser 49.85\nsys 0.77\npercent 121%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ClusterMarkers",desc:"Markers for clusters of all or selected T cells.",npages:2,report_toc:!0,order:10,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] UpSetR_1.4.0        ggVennDiagram_1.2.3 tidyseurat_0.8.0   \n [4] ttservice_0.4.1     future_1.34.0       ggrepel_0.9.6      \n [7] ggprism_1.0.5       ggplot2_3.5.1       enrichR_3.4        \n[10] Seurat_5.1.0        SeuratObject_5.0.2  sp_2.2-0           \n[13] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[16] tidyselect_1.2.1    digest_0.6.37       jsonlite_1.8.9     \n[19] rlang_1.1.5         logger_0.4.0       \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] ellipsis_0.3.2         labeling_0.4.3         promises_1.3.2        \n [19] purrr_1.0.2            WriteXLS_6.7.0         RVenn_1.1.0           \n [22] goftest_1.2-3          later_1.4.1            spatstat.utils_3.1-2  \n [25] irlba_2.3.5.1          parallel_4.3.3         cluster_2.1.8         \n [28] R6_2.6.0               ica_1.0-3              stringi_1.8.4         \n [31] RColorBrewer_1.1-3     spatstat.data_3.1-4    limma_3.58.1          \n [34] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [37] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [40] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [43] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [46] splines_4.3.3          igraph_2.0.3           abind_1.4-5           \n [49] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.3-4 curl_6.0.1             listenv_0.9.1         \n [55] lattice_0.22-6         plyr_1.8.9             withr_3.0.2           \n [58] shiny_1.10.0           ROCR_1.0-11            Rtsne_0.17            \n [61] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n [64] fitdistrplus_1.2-2     pillar_1.10.1          KernSmooth_2.23-26    \n [67] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n [82] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n [85] nlme_3.1-167           patchwork_1.3.0        presto_1.0.0          \n [88] cli_3.6.4              spatstat.sparse_3.1-0  fansi_1.0.6           \n [91] spam_2.11-1            viridisLite_0.4.2      uwot_0.2.2            \n [94] gtable_0.3.6           progressr_0.15.1       rjson_0.2.23          \n [97] htmlwidgets_1.6.4      farver_2.1.2           htmltools_0.5.8.1     \n[100] lifecycle_1.0.4        httr_1.4.7             statmod_1.5.0         \n[103] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 162.57\nuser 119.15\nsys 5.40\npercent 76%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TopExpressingGenes",desc:"Top expressing genes for clusters of all or selected T cells.",npages:2,report_toc:!0,order:11,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ggprism_1.0.5      dplyr_1.1.4        enrichR_3.4        tibble_3.2.1      \n [5] Seurat_5.1.0       SeuratObject_5.0.2 sp_2.2-0           jsonlite_1.8.9    \n [9] rlang_1.1.5        logger_0.4.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] labeling_0.4.3         promises_1.3.2         purrr_1.0.2           \n [19] WriteXLS_6.7.0         goftest_1.2-3          later_1.4.1           \n [22] spatstat.utils_3.1-2   irlba_2.3.5.1          parallel_4.3.3        \n [25] cluster_2.1.8          R6_2.6.0               ica_1.0-3             \n [28] stringi_1.8.4          RColorBrewer_1.1-3     spatstat.data_3.1-4   \n [31] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [37] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [43] splines_4.3.3          igraph_2.0.3           tidyselect_1.2.1      \n [46] abind_1.4-5            spatstat.random_3.3-2  codetools_0.2-20      \n [49] miniUI_0.1.1.1         spatstat.explore_3.3-4 curl_6.0.1            \n [52] listenv_0.9.1          lattice_0.22-6         plyr_1.8.9            \n [55] withr_3.0.2            shiny_1.10.0           ROCR_1.0-11           \n [58] Rtsne_0.17             future_1.34.0          fastDummies_1.7.5     \n [61] survival_3.8-3         polyclip_1.10-7        fitdistrplus_1.2-2    \n [64] pillar_1.10.1          KernSmooth_2.23-26     plotly_4.10.4         \n [67] generics_0.1.3         RcppHNSW_0.6.0         ggplot2_3.5.1         \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n [82] cowplot_1.1.3          grid_4.3.3             tidyr_1.3.1           \n [85] colorspace_2.1-1       nlme_3.1-167           patchwork_1.3.0       \n [88] cli_3.6.4              spatstat.sparse_3.1-0  spam_2.11-1           \n [91] viridisLite_0.4.2      uwot_0.2.2             gtable_0.3.6          \n [94] digest_0.6.37          progressr_0.15.1       ggrepel_0.9.6         \n [97] rjson_0.2.23           htmlwidgets_1.6.4      farver_2.1.2          \n[100] htmltools_0.5.8.1      lifecycle_1.0.4        httr_1.4.7            \n[103] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 45.21\nuser 28.74\nsys 1.92\npercent 67%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TCRClusterStats",desc:"Statistics of TCR clusters, generated by `TCRClustering`.",npages:1,report_toc:!0,order:13,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] grid      stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] ComplexHeatmap_2.18.0 ggprism_1.0.5         immunarch_0.9.1      \n [4] patchwork_1.3.0       data.table_1.16.4     dtplyr_1.3.1         \n [7] dplyr_1.1.4           tidyr_1.3.1           tibble_3.2.1         \n[10] ggplot2_3.5.1         jsonlite_1.8.9        rlang_1.1.5          \n[13] logger_0.4.0         \n\nloaded via a namespace (and not attached):\n  [1] gridExtra_2.3       phangorn_2.12.1     readxl_1.4.3       \n  [4] magrittr_2.0.3      clue_0.3-66         GetoptLong_1.0.5   \n  [7] matrixStats_1.1.0   compiler_4.3.3      flexmix_2.3-19     \n [10] png_0.1-8           vctrs_0.6.5         reshape2_1.4.4     \n [13] ggalluvial_0.12.5   quadprog_1.5-8      stringr_1.5.1      \n [16] crayon_1.5.3        pkgconfig_2.0.3     shape_1.4.6.1      \n [19] fastmap_1.2.0       backports_1.5.0     labeling_0.4.3     \n [22] ggraph_2.2.1        promises_1.3.2      tzdb_0.4.0         \n [25] UpSetR_1.4.0        purrr_1.0.2         modeltools_0.2-23  \n [28] ggseqlogo_0.2       cachem_1.1.0        later_1.4.1        \n [31] uuid_1.2-1          fpc_2.2-13          tweenr_2.0.3       \n [34] broom_1.0.7         parallel_4.3.3      prabclus_2.3-4     \n [37] cluster_2.1.8       R6_2.6.0            stringi_1.8.4      \n [40] RColorBrewer_1.1-3  rlist_0.4.6.2       car_3.1-3          \n [43] diptest_0.77-1      cellranger_1.1.0    Rcpp_1.0.14        \n [46] iterators_1.0.14    IRanges_2.36.0      readr_2.1.5        \n [49] httpuv_1.6.15       Matrix_1.6-5        nnet_7.3-20        \n [52] igraph_2.0.3        tidyselect_1.2.1    stringdist_0.9.15  \n [55] abind_1.4-5         viridis_0.6.5       doParallel_1.0.17  \n [58] codetools_0.2-20    lattice_0.22-6      plyr_1.8.9         \n [61] shiny_1.10.0        withr_3.0.2         polyclip_1.10-7    \n [64] circlize_0.4.16     mclust_6.1.1        kernlab_0.9-33     \n [67] pillar_1.10.1       ggpubr_0.6.0        carData_3.0-5      \n [70] foreach_1.5.2       stats4_4.3.3        generics_0.1.3     \n [73] S4Vectors_0.40.2    hms_1.1.3           munsell_0.5.1      \n [76] scales_1.3.0        xtable_1.8-4        class_7.3-23       \n [79] glue_1.8.0          pheatmap_1.0.12     tools_4.3.3        \n [82] robustbase_0.99-4-1 ggsignif_0.6.4      graphlayouts_1.2.2 \n [85] Cairo_1.6-2         fastmatch_1.1-6     tidygraph_1.3.0    \n [88] ape_5.8-1           colorspace_2.1-1    nlme_3.1-167       \n [91] ggforce_0.4.2       Formula_1.2-5       cli_3.6.4          \n [94] viridisLite_0.4.2   gtable_0.3.6        DEoptimR_1.1-3-1   \n [97] rstatix_0.7.2       digest_0.6.37       BiocGenerics_0.48.1\n[100] ggrepel_0.9.6       rjson_0.2.23        farver_2.1.2       \n[103] memoise_2.0.1       htmltools_0.5.8.1   lifecycle_1.0.4    \n[106] factoextra_1.0.7    mime_0.12           GlobalOptions_0.1.2\n[109] shinythemes_1.2.0   MASS_7.3-60.0.1    \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 7.17\nuser 15.93\nsys 0.32\npercent 226%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3287.293\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TESSA",desc:"Tessa is a Bayesian model to integrate T cell receptor (TCR) sequence profiling with transcriptomes of T cells.",npages:1,report_toc:!0,order:14,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] igraph_2.0.3         Rtsne_0.17           LaplacesDemon_16.1.6\n [4] MASS_7.3-60.0.1      ggprism_1.0.5        Seurat_5.1.0        \n [7] SeuratObject_5.0.2   sp_2.2-0             tibble_3.2.1        \n[10] glue_1.8.0           immunarch_0.9.1      patchwork_1.3.0     \n[13] data.table_1.16.4    dtplyr_1.3.1         ggplot2_3.5.1       \n[16] tidyr_1.3.1          dplyr_1.1.4          jsonlite_1.8.9      \n[19] rlang_1.1.5          logger_0.4.0        \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22       shinythemes_1.2.0      splines_4.3.3         \n  [4] later_1.4.1            cellranger_1.1.0       polyclip_1.10-7       \n  [7] fastDummies_1.7.5      factoextra_1.0.7       lifecycle_1.0.4       \n [10] rstatix_0.7.2          doParallel_1.0.17      globals_0.16.3        \n [13] lattice_0.22-6         prabclus_2.3-4         backports_1.5.0       \n [16] magrittr_2.0.3         plotly_4.10.4          rlist_0.4.6.2         \n [19] httpuv_1.6.15          sctransform_0.4.1      spam_2.11-1           \n [22] flexmix_2.3-19         spatstat.sparse_3.1-0  reticulate_1.40.0     \n [25] cowplot_1.1.3          pbapply_1.7-2          RColorBrewer_1.1-3    \n [28] abind_1.4-5            quadprog_1.5-8         purrr_1.0.2           \n [31] ggraph_2.2.1           nnet_7.3-20            tweenr_2.0.3          \n [34] circlize_0.4.16        ggrepel_0.9.6          irlba_2.3.5.1         \n [37] listenv_0.9.1          spatstat.utils_3.1-2   pheatmap_1.0.12       \n [40] goftest_1.2-3          RSpectra_0.16-2        spatstat.random_3.3-2 \n [43] fitdistrplus_1.2-2     parallelly_1.42.0      leiden_0.4.3.1        \n [46] codetools_0.2-20       ggforce_0.4.2          tidyselect_1.2.1      \n [49] shape_1.4.6.1          ggseqlogo_0.2          farver_2.1.2          \n [52] viridis_0.6.5          matrixStats_1.1.0      stats4_4.3.3          \n [55] spatstat.explore_3.3-4 tidygraph_1.3.0        progressr_0.15.1      \n [58] Formula_1.2-5          ggridges_0.5.6         ggalluvial_0.12.5     \n [61] survival_3.8-3         iterators_1.0.14       foreach_1.5.2         \n [64] tools_4.3.3            stringdist_0.9.15      ica_1.0-3             \n [67] Rcpp_1.0.14            gridExtra_2.3          withr_3.0.2           \n [70] fastmap_1.2.0          digest_0.6.37          R6_2.6.0              \n [73] mime_0.12              colorspace_2.1-1       scattermore_1.2       \n [76] tensor_1.5             spatstat.data_3.1-4    diptest_0.77-1        \n [79] UpSetR_1.4.0           generics_0.1.3         robustbase_0.99-4-1   \n [82] class_7.3-23           graphlayouts_1.2.2     httr_1.4.7            \n [85] htmlwidgets_1.6.4      uwot_0.2.2             pkgconfig_2.0.3       \n [88] gtable_0.3.6           modeltools_0.2-23      lmtest_0.9-40         \n [91] htmltools_0.5.8.1      carData_3.0-5          dotCall64_1.2         \n [94] scales_1.3.0           png_0.1-8              spatstat.univar_3.1-1 \n [97] tzdb_0.4.0             reshape2_1.4.4         uuid_1.2-1            \n[100] nlme_3.1-167           cachem_1.1.0           zoo_1.8-12            \n[103] GlobalOptions_0.1.2    stringr_1.5.1          KernSmooth_2.23-26    \n[106] parallel_4.3.3         miniUI_0.1.1.1         pillar_1.10.1         \n[109] grid_4.3.3             vctrs_0.6.5            RANN_2.6.2            \n[112] promises_1.3.2         ggpubr_0.6.0           car_3.1-3             \n[115] xtable_1.8-4           cluster_2.1.8          readr_2.1.5           \n[118] cli_3.6.4              compiler_4.3.3         future.apply_1.11.3   \n[121] ggsignif_0.6.4         labeling_0.4.3         mclust_6.1.1          \n[124] plyr_1.8.9             stringi_1.8.4          deldir_2.0-4          \n[127] viridisLite_0.4.2      munsell_0.5.1          lazyeval_0.2.2        \n[130] spatstat.geom_3.3-5    Matrix_1.6-5           RcppHNSW_0.6.0        \n[133] hms_1.1.3              future_1.34.0          fpc_2.2-13            \n[136] shiny_1.10.0           kernlab_0.9-33         ROCR_1.0-11           \n[139] broom_1.0.7            memoise_2.0.1          phangorn_2.12.1       \n[142] fastmatch_1.1-6        DEoptimR_1.1-3-1       readxl_1.4.3          \n[145] ape_5.8-1             \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 1679.73\nuser 2269.05\nsys 21.01\npercent 136%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"SeuratClusterStats",desc:"Statistics of the clustering.",npages:1,report_toc:!0,order:17,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] grid      stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] clustree_0.5.1        ggraph_2.2.1          ComplexHeatmap_2.18.0\n [4] circlize_0.4.16       tidyseurat_0.8.0      ttservice_0.4.1      \n [7] ggrepel_0.9.6         ggprism_1.0.5         tibble_3.2.1         \n[10] Seurat_5.1.0          SeuratObject_5.0.2    sp_2.2-0             \n[13] ggplot2_3.5.1         tidyr_1.3.1           dplyr_1.1.4          \n[16] tidyselect_1.2.1      jsonlite_1.8.9        rlang_1.1.5          \n[19] logger_0.4.0         \n\nloaded via a namespace (and not attached):\n  [1] RColorBrewer_1.1-3     shape_1.4.6.1          magrittr_2.0.3        \n  [4] ggbeeswarm_0.7.2       spatstat.utils_3.1-2   farver_2.1.2          \n  [7] GlobalOptions_0.1.2    vctrs_0.6.5            ROCR_1.0-11           \n [10] memoise_2.0.1          spatstat.explore_3.3-4 htmltools_0.5.8.1     \n [13] sctransform_0.4.1      parallelly_1.42.0      KernSmooth_2.23-26    \n [16] htmlwidgets_1.6.4      ica_1.0-3              plyr_1.8.9            \n [19] cachem_1.1.0           plotly_4.10.4          zoo_1.8-12            \n [22] igraph_2.0.3           mime_0.12              lifecycle_1.0.4       \n [25] iterators_1.0.14       pkgconfig_2.0.3        Matrix_1.6-5          \n [28] R6_2.6.0               fastmap_1.2.0          clue_0.3-66           \n [31] fitdistrplus_1.2-2     future_1.34.0          shiny_1.10.0          \n [34] digest_0.6.37          colorspace_2.1-1       patchwork_1.3.0       \n [37] S4Vectors_0.40.2       tensor_1.5             RSpectra_0.16-2       \n [40] irlba_2.3.5.1          labeling_0.4.3         progressr_0.15.1      \n [43] fansi_1.0.6            spatstat.sparse_3.1-0  httr_1.4.7            \n [46] polyclip_1.10-7        abind_1.4-5            compiler_4.3.3        \n [49] withr_3.0.2            doParallel_1.0.17      viridis_0.6.5         \n [52] fastDummies_1.7.5      ggforce_0.4.2          MASS_7.3-60.0.1       \n [55] rjson_0.2.23           tools_4.3.3            vipor_0.4.7           \n [58] lmtest_0.9-40          beeswarm_0.4.0         httpuv_1.6.15         \n [61] future.apply_1.11.3    goftest_1.2-3          glue_1.8.0            \n [64] nlme_3.1-167           promises_1.3.2         Rtsne_0.17            \n [67] cluster_2.1.8          reshape2_1.4.4         generics_0.1.3        \n [70] gtable_0.3.6           spatstat.data_3.1-4    data.table_1.16.4     \n [73] tidygraph_1.3.0        BiocGenerics_0.48.1    spatstat.geom_3.3-5   \n [76] RcppAnnoy_0.0.22       RANN_2.6.2             foreach_1.5.2         \n [79] pillar_1.10.1          stringr_1.5.1          spam_2.11-1           \n [82] RcppHNSW_0.6.0         later_1.4.1            splines_4.3.3         \n [85] tweenr_2.0.3           lattice_0.22-6         survival_3.8-3        \n [88] deldir_2.0-4           miniUI_0.1.1.1         pbapply_1.7-2         \n [91] gridExtra_2.3          IRanges_2.36.0         scattermore_1.2       \n [94] stats4_4.3.3           graphlayouts_1.2.2     matrixStats_1.1.0     \n [97] stringi_1.8.4          lazyeval_0.2.2         codetools_0.2-20      \n[100] cli_3.6.4              uwot_0.2.2             xtable_1.8-4          \n[103] reticulate_1.40.0      munsell_0.5.1          Rcpp_1.0.14           \n[106] globals_0.16.3         spatstat.random_3.3-2  png_0.1-8             \n[109] ggrastr_1.0.2          spatstat.univar_3.1-1  parallel_4.3.3        \n[112] ellipsis_0.3.2         dotCall64_1.2          listenv_0.9.1         \n[115] viridisLite_0.4.2      scales_1.3.0           ggridges_0.5.6        \n[118] leiden_0.4.3.1         purrr_1.0.2            crayon_1.5.3          \n[121] GetoptLong_1.0.5       cowplot_1.1.3         \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 18.76\nuser 27.07\nsys 0.71\npercent 148%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"Immunarch",desc:"Exploration of Single-cell and Bulk T-cell/Antibody Immune Repertoires",npages:3,report_toc:!0,order:18,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] circlize_0.4.16   tibble_3.2.1      glue_1.8.0        ggprism_1.0.5    \n [5] immunarch_0.9.1   patchwork_1.3.0   data.table_1.16.4 dtplyr_1.3.1     \n [9] ggplot2_3.5.1     tidyr_1.3.1       dplyr_1.1.4       jsonlite_1.8.9   \n[13] rlang_1.1.5       logger_0.4.0     \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22       shinythemes_1.2.0      splines_4.3.3         \n  [4] later_1.4.1            cellranger_1.1.0       polyclip_1.10-7       \n  [7] fastDummies_1.7.5      factoextra_1.0.7       lifecycle_1.0.4       \n [10] rstatix_0.7.2          doParallel_1.0.17      globals_0.16.3        \n [13] lattice_0.22-6         prabclus_2.3-4         MASS_7.3-60.0.1       \n [16] backports_1.5.0        magrittr_2.0.3         plotly_4.10.4         \n [19] rlist_0.4.6.2          httpuv_1.6.15          Seurat_5.1.0          \n [22] sctransform_0.4.1      spam_2.11-1            flexmix_2.3-19        \n [25] spatstat.sparse_3.1-0  sp_2.2-0               reticulate_1.40.0     \n [28] cowplot_1.1.3          pbapply_1.7-2          RColorBrewer_1.1-3    \n [31] abind_1.4-5            Rtsne_0.17             quadprog_1.5-8        \n [34] purrr_1.0.2            ggraph_2.2.1           nnet_7.3-20           \n [37] tweenr_2.0.3           ggrepel_0.9.6          irlba_2.3.5.1         \n [40] spatstat.utils_3.1-2   listenv_0.9.1          pheatmap_1.0.12       \n [43] goftest_1.2-3          RSpectra_0.16-2        spatstat.random_3.3-2 \n [46] fitdistrplus_1.2-2     parallelly_1.42.0      leiden_0.4.3.1        \n [49] codetools_0.2-20       ggforce_0.4.2          tidyselect_1.2.1      \n [52] shape_1.4.6.1          ggseqlogo_0.2          farver_2.1.2          \n [55] viridis_0.6.5          matrixStats_1.1.0      stats4_4.3.3          \n [58] spatstat.explore_3.3-4 tidygraph_1.3.0        progressr_0.15.1      \n [61] Formula_1.2-5          ggridges_0.5.6         ggalluvial_0.12.5     \n [64] survival_3.8-3         iterators_1.0.14       foreach_1.5.2         \n [67] tools_4.3.3            stringdist_0.9.15      ica_1.0-3             \n [70] Rcpp_1.0.14            gridExtra_2.3          withr_3.0.2           \n [73] fastmap_1.2.0          digest_0.6.37          R6_2.6.0              \n [76] mime_0.12              colorspace_2.1-1       scattermore_1.2       \n [79] tensor_1.5             spatstat.data_3.1-4    diptest_0.77-1        \n [82] UpSetR_1.4.0           generics_0.1.3         robustbase_0.99-4-1   \n [85] class_7.3-23           graphlayouts_1.2.2     httr_1.4.7            \n [88] htmlwidgets_1.6.4      uwot_0.2.2             pkgconfig_2.0.3       \n [91] gtable_0.3.6           modeltools_0.2-23      lmtest_0.9-40         \n [94] htmltools_0.5.8.1      carData_3.0-5          dotCall64_1.2         \n [97] SeuratObject_5.0.2     scales_1.3.0           png_0.1-8             \n[100] spatstat.univar_3.1-1  tzdb_0.4.0             reshape2_1.4.4        \n[103] uuid_1.2-1             nlme_3.1-167           cachem_1.1.0          \n[106] zoo_1.8-12             GlobalOptions_0.1.2    stringr_1.5.1         \n[109] KernSmooth_2.23-26     parallel_4.3.3         miniUI_0.1.1.1        \n[112] pillar_1.10.1          grid_4.3.3             vctrs_0.6.5           \n[115] RANN_2.6.2             promises_1.3.2         ggpubr_0.6.0          \n[118] car_3.1-3              xtable_1.8-4           cluster_2.1.8         \n[121] readr_2.1.5            cli_3.6.4              compiler_4.3.3        \n[124] future.apply_1.11.3    ggsignif_0.6.4         labeling_0.4.3        \n[127] mclust_6.1.1           plyr_1.8.9             stringi_1.8.4         \n[130] deldir_2.0-4           viridisLite_0.4.2      assertthat_0.2.1      \n[133] munsell_0.5.1          lazyeval_0.2.2         spatstat.geom_3.3-5   \n[136] Matrix_1.6-5           RcppHNSW_0.6.0         hms_1.1.3             \n[139] future_1.34.0          fpc_2.2-13             shiny_1.10.0          \n[142] kernlab_0.9-33         ROCR_1.0-11            igraph_2.0.3          \n[145] broom_1.0.7            memoise_2.0.1          phangorn_2.12.1       \n[148] fastmatch_1.1-6        DEoptimR_1.1-3-1       readxl_1.4.3          \n[151] ape_5.8-1             \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 102.43\nuser 256.71\nsys 5.18\npercent 255%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"RadarPlots",desc:"Radar plots for cell proportion in different clusters.",npages:1,report_toc:!0,order:19,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ggprism_1.0.5      ggradar_0.2        ggplot2_3.5.1      tibble_3.2.1      \n [5] tidyr_1.3.1        dplyr_1.1.4        Seurat_5.1.0       SeuratObject_5.0.2\n [9] sp_2.2-0           jsonlite_1.8.9     rlang_1.1.5        logger_0.4.0      \n\nloaded via a namespace (and not attached):\n [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n[10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n[13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n[16] labeling_0.4.3         promises_1.3.2         purrr_1.0.2           \n[19] goftest_1.2-3          later_1.4.1            spatstat.utils_3.1-2  \n[22] irlba_2.3.5.1          parallel_4.3.3         cluster_2.1.8         \n[25] R6_2.6.0               ica_1.0-3              stringi_1.8.4         \n[28] RColorBrewer_1.1-3     spatstat.data_3.1-4    reticulate_1.40.0     \n[31] parallelly_1.42.0      spatstat.univar_3.1-1  lmtest_0.9-40         \n[34] scattermore_1.2        Rcpp_1.0.14            tensor_1.5            \n[37] future.apply_1.11.3    zoo_1.8-12             sctransform_0.4.1     \n[40] httpuv_1.6.15          Matrix_1.6-5           splines_4.3.3         \n[43] igraph_2.0.3           tidyselect_1.2.1       abind_1.4-5           \n[46] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n[49] spatstat.explore_3.3-4 listenv_0.9.1          lattice_0.22-6        \n[52] plyr_1.8.9             withr_3.0.2            shiny_1.10.0          \n[55] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n[58] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n[61] fitdistrplus_1.2-2     pillar_1.10.1          KernSmooth_2.23-26    \n[64] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n[67] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n[70] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n[73] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n[76] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n[79] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n[82] nlme_3.1-167           patchwork_1.3.0        cli_3.6.4             \n[85] spatstat.sparse_3.1-0  spam_2.11-1            viridisLite_0.4.2     \n[88] uwot_0.2.2             gtable_0.3.6           digest_0.6.37         \n[91] progressr_0.15.1       ggrepel_0.9.6          htmlwidgets_1.6.4     \n[94] farver_2.1.2           htmltools_0.5.8.1      lifecycle_1.0.4       \n[97] httr_1.4.7             mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 8.07\nuser 16.74\nsys 0.38\npercent 212%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"CloneResidency",desc:"Identification of clone residency",npages:1,report_toc:!0,order:20,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ComplexUpset_1.3.3  ggVennDiagram_1.2.3 ggprism_1.0.5      \n [4] ggplot2_3.5.1       ggnewscale_0.5.0    glue_1.8.0         \n [7] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[10] jsonlite_1.8.9      rlang_1.1.5         logger_0.4.0       \n\nloaded via a namespace (and not attached):\n  [1] DBI_1.2.3              deldir_2.0-4           pbapply_1.7-2         \n  [4] gridExtra_2.3          magrittr_2.0.3         RcppAnnoy_0.0.22      \n  [7] e1071_1.7-16           spatstat.geom_3.3-5    matrixStats_1.1.0     \n [10] ggridges_0.5.6         compiler_4.3.3         png_0.1-8             \n [13] vctrs_0.6.5            reshape2_1.4.4         stringr_1.5.1         \n [16] pkgconfig_2.0.3        fastmap_1.2.0          labeling_0.4.3        \n [19] promises_1.3.2         purrr_1.0.2            RVenn_1.1.0           \n [22] goftest_1.2-3          later_1.4.1            spatstat.utils_3.1-2  \n [25] irlba_2.3.5.1          parallel_4.3.3         cluster_2.1.8         \n [28] R6_2.6.0               ica_1.0-3              stringi_1.8.4         \n [31] RColorBrewer_1.1-3     spatstat.data_3.1-4    reticulate_1.40.0     \n [34] parallelly_1.42.0      spatstat.univar_3.1-1  lmtest_0.9-40         \n [37] scattermore_1.2        Rcpp_1.0.14            tensor_1.5            \n [40] future.apply_1.11.3    zoo_1.8-12             sctransform_0.4.1     \n [43] httpuv_1.6.15          Matrix_1.6-5           splines_4.3.3         \n [46] igraph_2.0.3           tidyselect_1.2.1       abind_1.4-5           \n [49] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.3-4 listenv_0.9.1          lattice_0.22-6        \n [55] plyr_1.8.9             shiny_1.10.0           withr_3.0.2           \n [58] ROCR_1.0-11            Rtsne_0.17             sf_1.0-19             \n [61] future_1.34.0          fastDummies_1.7.5      survival_3.8-3        \n [64] units_0.8-5            proxy_0.4-27           polyclip_1.10-7       \n [67] fitdistrplus_1.2-2     pillar_1.10.1          Seurat_5.1.0          \n [70] KernSmooth_2.23-26     plotly_4.10.4          generics_0.1.3        \n [73] RcppHNSW_0.6.0         sp_2.2-0               munsell_0.5.1         \n [76] scales_1.3.0           globals_0.16.3         xtable_1.8-4          \n [79] class_7.3-23           lazyeval_0.2.2         tools_4.3.3           \n [82] data.table_1.16.4      RSpectra_0.16-2        RANN_2.6.2            \n [85] leiden_0.4.3.1         dotCall64_1.2          cowplot_1.1.3         \n [88] grid_4.3.3             colorspace_2.1-1       nlme_3.1-167          \n [91] patchwork_1.3.0        cli_3.6.4              spatstat.sparse_3.1-0 \n [94] spam_2.11-1            viridisLite_0.4.2      uwot_0.2.2            \n [97] gtable_0.3.6           digest_0.6.37          classInt_0.4-11       \n[100] progressr_0.15.1       ggrepel_0.9.6          htmlwidgets_1.6.4     \n[103] SeuratObject_5.0.2     farver_2.1.2           htmltools_0.5.8.1     \n[106] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[109] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 19.89\nuser 28.24\nsys 0.66\npercent 145%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ScFGSEA",desc:"Gene set enrichment analysis for cells in different groups using `fgsea`",npages:2,report_toc:!0,order:21,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] fgsea_1.28.0       data.table_1.16.4  matrixStats_1.1.0  tidyseurat_0.8.0  \n [5] ttservice_0.4.1    Seurat_5.1.0       SeuratObject_5.0.2 sp_2.2-0          \n [9] tidyr_1.3.1        tidyselect_1.2.1   tibble_3.2.1       dplyr_1.1.4       \n[13] ggplot2_3.5.1      jsonlite_1.8.9     rlang_1.1.5        logger_0.4.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       ggridges_0.5.6        \n  [7] compiler_4.3.3         spatstat.geom_3.3-5    png_0.1-8             \n [10] vctrs_0.6.5            reshape2_1.4.4         stringr_1.5.1         \n [13] pkgconfig_2.0.3        fastmap_1.2.0          ellipsis_0.3.2        \n [16] labeling_0.4.3         promises_1.3.2         purrr_1.0.2           \n [19] goftest_1.2-3          later_1.4.1            BiocParallel_1.36.0   \n [22] spatstat.utils_3.1-2   irlba_2.3.5.1          parallel_4.3.3        \n [25] cluster_2.1.8          R6_2.6.0               ica_1.0-3             \n [28] stringi_1.8.4          RColorBrewer_1.1-3     spatstat.data_3.1-4   \n [31] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [37] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [43] splines_4.3.3          igraph_2.0.3           abind_1.4-5           \n [46] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n [49] spatstat.explore_3.3-4 listenv_0.9.1          lattice_0.22-6        \n [52] plyr_1.8.9             shiny_1.10.0           withr_3.0.2           \n [55] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n [58] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n [61] fitdistrplus_1.2-2     pillar_1.10.1          KernSmooth_2.23-26    \n [64] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n [67] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [70] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [73] tools_4.3.3            RSpectra_0.16-2        RANN_2.6.2            \n [76] leiden_0.4.3.1         dotCall64_1.2          fastmatch_1.1-6       \n [79] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n [82] nlme_3.1-167           patchwork_1.3.0        cli_3.6.4             \n [85] spatstat.sparse_3.1-0  fansi_1.0.6            spam_2.11-1           \n [88] viridisLite_0.4.2      uwot_0.2.2             gtable_0.3.6          \n [91] digest_0.6.37          progressr_0.15.1       ggrepel_0.9.6         \n [94] htmlwidgets_1.6.4      farver_2.1.2           htmltools_0.5.8.1     \n [97] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[100] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 128.45\nuser 135.35\nsys 1.52\npercent 106%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"MarkersFinder",desc:"Find markers between different groups of cells",npages:1,report_toc:!0,order:22},{name:"MetaMarkers",desc:"Find markers between three or more groups of cells, using one-way ANOVA or Kruskal-Wallis test.",npages:1,report_toc:!0,order:23},{name:"CellsDistribution",desc:"Distribution of cells (i.e. in a TCR clone) from different groups for each cluster",npages:1,report_toc:!0,order:24},{name:"CDR3AAPhyschem",desc:"CDR3 AA physicochemical feature analysis",npages:1,report_toc:!0,order:25},{name:"ScrnaMetabolicLandscape",desc:"Metabolic landscape analysis for scRNA-seq data",order:28,procs:[{name:"MetabolicPathwayActivity",desc:"This process calculates the pathway activities in different groups and subsets.",npages:1,report_toc:!0,order:28},{name:"MetabolicFeatures",desc:"This process performs enrichment analysis for the metabolic pathways for each group in each subset.",npages:1,report_toc:!0,order:29},{name:"MetabolicFeaturesIntraSubset",desc:"Intra-subset metabolic features - Enrichment analysis in details",npages:1,report_toc:!0,order:30},{name:"MetabolicPathwayHeterogeneity",desc:"Calculate Metabolic Pathway heterogeneity.",npages:1,report_toc:!0,order:31}]}]};function rs(e){let t,n;return t=new an({props:{dir:"up"}}),{c(){u(t.$$.fragment)},m(e,a){$(t,e,a),n=!0},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){A(t,e)}}}function ls(e){let t,n;return t=new an({props:{dir:"down"}}),{c(){u(t.$$.fragment)},m(e,a){$(t,e,a),n=!0},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){A(t,e)}}}function is(e){let t,n,a,s,r,l=e[1]>0&&rs();n=new ns({});let m=e[1]<e[3].npages-1&&ls();return{c(){l&&l.c(),t=S(),u(n.$$.fragment),a=S(),m&&m.c(),s=f()},m(e,o){l&&l.m(e,o),i(e,t,o),$(n,e,o),i(e,a,o),m&&m.m(e,o),i(e,s,o),r=!0},p(e,n){e[1]>0?l?2&n&&o(l,1):(l=rs(),l.c(),o(l,1),l.m(t.parentNode,t)):l&&(c(),_(l,1,1,(()=>{l=null})),p()),e[1]<e[3].npages-1?m?10&n&&o(m,1):(m=ls(),m.c(),o(m,1),m.m(s.parentNode,s)):m&&(c(),_(m,1,1,(()=>{m=null})),p())},i(e){r||(o(l),o(n.$$.fragment,e),o(m),r=!0)},o(e){_(l),_(n.$$.fragment,e),_(m),r=!1},d(e){e&&(d(t),d(a),d(s)),l&&l.d(e),A(n,e),m&&m.d(e)}}}function os(e){let t,n,a;function s(t){e[4](t)}let r={logo:e[0],logotext:e[3].desc,versions:ss.versions,entries:ss.entries,pipeline_name:ss.pipeline.name,report_toc:e[3].report_toc,runinfo:e[3].runinfo,toc:as,page:e[1],$$slots:{default:[is]},$$scope:{ctx:e}};return void 0!==e[2]&&(r.isSideNavOpen=e[2]),t=new Qt({props:r}),b.push((()=>h(t,"isSideNavOpen",s))),{c(){u(t.$$.fragment)},m(e,n){$(t,e,n),a=!0},p(e,[a]){const s={};1&a&&(s.logo=e[0]),8&a&&(s.logotext=e[3].desc),8&a&&(s.report_toc=e[3].report_toc),8&a&&(s.runinfo=e[3].runinfo),2&a&&(s.page=e[1]),42&a&&(s.$$scope={dirty:a,ctx:e}),!n&&4&a&&(n=!0,s.isSideNavOpen=e[2],C((()=>n=!1))),t.$set(s)},i(e){a||(o(t.$$.fragment,e),a=!0)},o(e){_(t.$$.fragment,e),a=!1},d(e){A(t,e)}}}function cs(e,t,n){let a,{name:s}=t,{page:r=0}=t,l=!1;for(const e of ss.entries)if(e.procs){for(const t of e.procs)if(t.name==s){a=t;break}}else if(e.name==s){a=e;break}return g((()=>{const e=new URL(window.location.href);let t;e.hash.length>1&&(t=document.getElementById(`${e.hash.slice(1)}`)),t&&t.scrollIntoView({block:"start",behavior:"smooth"})})),e.$$set=e=>{"name"in e&&n(0,s=e.name),"page"in e&&n(1,r=e.page)},[s,r,l,a,function(e){l=e,n(2,l)}]}e("default",new class extends t{constructor(e){super(),n(this,e,cs,os,a,{name:0,page:1})}}({target:document.body,props:{name:"ScFGSEA",page:1}}))}}}));
