System.register(["./vendor/svelte-CiOtOeYy.js","./vendor/carbon-components-svelte-BhQO6q4b.js","./vendor/carbon-icons-svelte-DVphRbRG.js"],(function(e){"use strict";var t,n,a,s,r,l,i,p,o,c,_,d,g,m,$,f,A,u,S,L,E,R,C,b,P,T,h,N,M,v,I,y,x,G,w,B,H,F,O,D,K,U,V,z,X,k,q,j,Y,W,J,Z,Q,ee,te,ne,ae,se,re,le,ie,pe,oe,ce,_e,de,ge,me,$e,fe,Ae,ue,Se,Le,Ee,Re,Ce,be,Pe,Te,he,Ne,Me,ve,Ie,ye,xe,Ge,we,Be,He,Fe;return{setters:[function(e){t=e.S,n=e.i,a=e.s,s=e.a1,r=e.l,l=e.O,i=e.a,p=e.b,o=e.g,c=e.c,_=e.t,d=e.d,g=e.a5,m=e.B,$=e.e,f=e.G,A=e.H,u=e.I,S=e.y,L=e.K,E=e.T,R=e.U,C=e.L,b=e.D,P=e.M,T=e.z,h=e.Q,N=e.k,M=e.u,v=e.n,I=e.o,y=e.f,x=e.A,G=e.h,w=e.j,B=e.aa,H=e.x,F=e.W,O=e.ab,D=e.m,K=e.p,U=e.a0,V=e.q,z=e.R,X=e.$,k=e.v,q=e.Y,j=e.Z,Y=e.F,W=e.ac,J=e.w,Z=e.J,Q=e.r},function(e){ee=e.S,te=e.a,ne=e.b,ae=e.H,se=e.c,re=e.T,le=e.d,ie=e.e,pe=e.f,oe=e.g,ce=e.h,_e=e.M,de=e.I,ge=e.i,me=e.j,$e=e.k,fe=e.C,Ae=e.l,ue=e.m,Se=e.G,Le=e.R,Ee=e.n,Re=e.B,Ce=e.A,be=e.o,Pe=e.D,Te=e.P,he=e.p,Ne=e.q,Me=e.r,ve=e.s,Ie=e.t},function(e){ye=e.C,xe=e.a,Ge=e.O,we=e.b,Be=e.c,He=e.D,Fe=e.d}],execute:function(){function Oe(e,t,n){const a=e.slice();return a[4]=t[n],a}function De(e,t,n){const a=e.slice();return a[7]=t[n],a}function Ke(e){let t,n;return t=new ee({props:{href:`${e[1](e[4].page)}#${e[4].slug}`,class:e[2](e[4].page),title:e[4].text,text:e[4].text,expanded:!0,$$slots:{default:[ze]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.href=`${e[1](e[4].page)}#${e[4].slug}`),1&n&&(a.class=e[2](e[4].page)),1&n&&(a.title=e[4].text),1&n&&(a.text=e[4].text),1025&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function Ue(e){let t,n;return t=new te({props:{href:`${e[1](e[4].page)}#${e[4].slug}`,class:e[2](e[4].page),title:e[4].text,text:e[4].text}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.href=`${e[1](e[4].page)}#${e[4].slug}`),1&n&&(a.class=e[2](e[4].page)),1&n&&(a.title=e[4].text),1&n&&(a.text=e[4].text),t.$set(a)},i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function Ve(e){let t,n;return t=new ne({props:{href:`${e[1](e[7].page)}#${e[7].slug}`,class:e[2](e[7].page),title:e[7].text,text:e[7].text}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.href=`${e[1](e[7].page)}#${e[7].slug}`),1&n&&(a.class=e[2](e[7].page)),1&n&&(a.title=e[7].text),1&n&&(a.text=e[7].text),t.$set(a)},i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function ze(e){let t,n,a=s(e[4].children),r=[];for(let t=0;t<a.length;t+=1)r[t]=Ve(De(e,a,t));const l=e=>_(r[e],1,1,(()=>{r[e]=null}));return{c(){for(let e=0;e<r.length;e+=1)r[e].c();t=S()},m(e,a){for(let t=0;t<r.length;t+=1)r[t]&&r[t].m(e,a);i(e,t,a),n=!0},p(e,n){if(7&n){let i;for(a=s(e[4].children),i=0;i<a.length;i+=1){const s=De(e,a,i);r[i]?(r[i].p(s,n),p(r[i],1)):(r[i]=Ve(s),r[i].c(),p(r[i],1),r[i].m(t.parentNode,t))}for(o(),i=a.length;i<r.length;i+=1)l(i);c()}},i(e){if(!n){for(let e=0;e<a.length;e+=1)p(r[e]);n=!0}},o(e){r=r.filter(Boolean);for(let e=0;e<r.length;e+=1)_(r[e]);n=!1},d(e){e&&d(t),g(r,e)}}}function Xe(e){let t,n,a,s;const r=[Ue,Ke],l=[];function g(e,t){return 0==e[4].children.length?0:1}return t=g(e),n=l[t]=r[t](e),{c(){n.c(),a=$()},m(e,n){l[t].m(e,n),i(e,a,n),s=!0},p(e,s){let i=t;t=g(e),t===i?l[t].p(e,s):(o(),_(l[i],1,1,(()=>{l[i]=null})),c(),n=l[t],n?n.p(e,s):(n=l[t]=r[t](e),n.c()),p(n,1),n.m(a.parentNode,a))},i(e){s||(p(n),s=!0)},o(e){_(n),s=!1},d(e){e&&d(a),l[t].d(e)}}}function ke(e){let t,n,a=s(e[0]),m=[];for(let t=0;t<a.length;t+=1)m[t]=Xe(Oe(e,a,t));const $=e=>_(m[e],1,1,(()=>{m[e]=null}));return{c(){t=r("ul");for(let e=0;e<m.length;e+=1)m[e].c();l(t,"class","bx--side-nav__items pipen-report-sidenav svelte-12gpq3")},m(e,a){i(e,t,a);for(let e=0;e<m.length;e+=1)m[e]&&m[e].m(t,null);n=!0},p(e,[n]){if(7&n){let r;for(a=s(e[0]),r=0;r<a.length;r+=1){const s=Oe(e,a,r);m[r]?(m[r].p(s,n),p(m[r],1)):(m[r]=Xe(s),m[r].c(),p(m[r],1),m[r].m(t,null))}for(o(),r=a.length;r<m.length;r+=1)$(r);c()}},i(e){if(!n){for(let e=0;e<a.length;e+=1)p(m[e]);n=!0}},o(e){m=m.filter(Boolean);for(let e=0;e<m.length;e+=1)_(m[e]);n=!1},d(e){e&&d(t),g(m,e)}}}function qe(e,t,n){let{toc:a}=t,{page:s}=t;return m((()=>{let e,t=new URL(window.location.href);e=t.hash.length>0?document.querySelector(`[href="${t.hash}"]`):document.querySelector(".toc-current-page"),e.scrollIntoView({behavior:"smooth"})})),e.$$set=e=>{"toc"in e&&n(0,a=e.toc),"page"in e&&n(3,s=e.page)},[a,e=>{if(s===e)return"";let t=new URL(window.location.href);return t.hash="",t.searchParams.set("page",e),t.toString()},e=>e!=s?"toc-noncurrent-page":"toc-current-page",s]}class je extends t{constructor(e){super(),n(this,e,qe,ke,a,{toc:0,page:3})}}function Ye(e,t,n){const a=e.slice();return a[7]=t[n],a}function We(e,t,n){const a=e.slice();return a[10]=t[n],a}function Je(e){let t,n;return t=new ie({props:{isOpen:ft,transition:e[5],$$slots:{default:[ot]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};8197&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function Ze(e){let t;return{c(){t=C("Go to home:")},m(e,n){i(e,t,n)},d(e){e&&d(t)}}}function Qe(e){let t;return{c(){t=C(e[0])},m(e,n){i(e,t,n)},p(e,n){1&n&&P(t,e[0])},d(e){e&&d(t)}}}function et(e){let t;return{c(){t=C("Switch to process:")},m(e,n){i(e,t,n)},d(e){e&&d(t)}}}function tt(e){let t,n;return t=new ce({props:{title:e[7].name,href:`?proc=${e[7].name}`,$$slots:{default:[at]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};4&n&&(a.title=e[7].name),4&n&&(a.href=`?proc=${e[7].name}`),8196&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function nt(e){let t,n,a,r;t=new ce({props:{class:"procgroup",$$slots:{default:[st]},$$scope:{ctx:e}}});let l=s(e[7].procs),m=[];for(let t=0;t<l.length;t+=1)m[t]=lt(We(e,l,t));const L=e=>_(m[e],1,1,(()=>{m[e]=null}));return{c(){f(t.$$.fragment),n=S();for(let e=0;e<m.length;e+=1)m[e].c();a=$()},m(e,s){A(t,e,s),i(e,n,s);for(let t=0;t<m.length;t+=1)m[t]&&m[t].m(e,s);i(e,a,s),r=!0},p(e,n){const r={};if(8196&n&&(r.$$scope={dirty:n,ctx:e}),t.$set(r),4&n){let t;for(l=s(e[7].procs),t=0;t<l.length;t+=1){const s=We(e,l,t);m[t]?(m[t].p(s,n),p(m[t],1)):(m[t]=lt(s),m[t].c(),p(m[t],1),m[t].m(a.parentNode,a))}for(o(),t=l.length;t<m.length;t+=1)L(t);c()}},i(e){if(!r){p(t.$$.fragment,e);for(let e=0;e<l.length;e+=1)p(m[e]);r=!0}},o(e){_(t.$$.fragment,e),m=m.filter(Boolean);for(let e=0;e<m.length;e+=1)_(m[e]);r=!1},d(e){e&&(d(n),d(a)),u(t,e),g(m,e)}}}function at(e){let t,n,a=e[7].name+"";return{c(){t=C(a),n=S()},m(e,a){i(e,t,a),i(e,n,a)},p(e,n){4&n&&a!==(a=e[7].name+"")&&P(t,a)},d(e){e&&(d(t),d(n))}}}function st(e){let t,n=e[7].name+"";return{c(){t=C(n)},m(e,n){i(e,t,n)},p(e,a){4&a&&n!==(n=e[7].name+"")&&P(t,n)},d(e){e&&d(t)}}}function rt(e){let t,n,a,s=e[10].name+"";return{c(){t=C("- "),n=C(s),a=S()},m(e,s){i(e,t,s),i(e,n,s),i(e,a,s)},p(e,t){4&t&&s!==(s=e[10].name+"")&&P(n,s)},d(e){e&&(d(t),d(n),d(a))}}}function lt(e){let t,n;return t=new ce({props:{title:e[10].name,class:"procgroup-proc",href:`?proc=${e[10].name}`,$$slots:{default:[rt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};4&n&&(a.title=e[10].name),4&n&&(a.href=`?proc=${e[10].name}`),8196&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function it(e){let t,n,a,s;const r=[nt,tt],l=[];function g(e,t){return e[7].procs?0:1}return t=g(e),n=l[t]=r[t](e),{c(){n.c(),a=$()},m(e,n){l[t].m(e,n),i(e,a,n),s=!0},p(e,s){let i=t;t=g(e),t===i?l[t].p(e,s):(o(),_(l[i],1,1,(()=>{l[i]=null})),c(),n=l[t],n?n.p(e,s):(n=l[t]=r[t](e),n.c()),p(n,1),n.m(a.parentNode,a))},i(e){s||(p(n),s=!0)},o(e){_(n),s=!1},d(e){e&&d(a),l[t].d(e)}}}function pt(e){let t,n,a,r,l,m,L,E;t=new oe({props:{$$slots:{default:[Ze]},$$scope:{ctx:e}}}),a=new ce({props:{href:"index.html",$$slots:{default:[Qe]},$$scope:{ctx:e}}}),l=new oe({props:{$$slots:{default:[et]},$$scope:{ctx:e}}});let R=s(e[2]),C=[];for(let t=0;t<R.length;t+=1)C[t]=it(Ye(e,R,t));const b=e=>_(C[e],1,1,(()=>{C[e]=null}));return{c(){f(t.$$.fragment),n=S(),f(a.$$.fragment),r=S(),f(l.$$.fragment),m=S();for(let e=0;e<C.length;e+=1)C[e].c();L=$()},m(e,s){A(t,e,s),i(e,n,s),A(a,e,s),i(e,r,s),A(l,e,s),i(e,m,s);for(let t=0;t<C.length;t+=1)C[t]&&C[t].m(e,s);i(e,L,s),E=!0},p(e,n){const r={};8192&n&&(r.$$scope={dirty:n,ctx:e}),t.$set(r);const i={};8193&n&&(i.$$scope={dirty:n,ctx:e}),a.$set(i);const _={};if(8192&n&&(_.$$scope={dirty:n,ctx:e}),l.$set(_),4&n){let t;for(R=s(e[2]),t=0;t<R.length;t+=1){const a=Ye(e,R,t);C[t]?(C[t].p(a,n),p(C[t],1)):(C[t]=it(a),C[t].c(),p(C[t],1),C[t].m(L.parentNode,L))}for(o(),t=R.length;t<C.length;t+=1)b(t);c()}},i(e){if(!E){p(t.$$.fragment,e),p(a.$$.fragment,e),p(l.$$.fragment,e);for(let e=0;e<R.length;e+=1)p(C[e]);E=!0}},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),_(l.$$.fragment,e),C=C.filter(Boolean);for(let e=0;e<C.length;e+=1)_(C[e]);E=!1},d(e){e&&(d(n),d(r),d(m),d(L)),u(t,e),u(a,e),u(l,e),g(C,e)}}}function ot(e){let t,n;return t=new pe({props:{$$slots:{default:[pt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};8197&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function ct(e){let t,n,a,s;t=new le({props:{render:"select",theme:"g10",persist:!0,persistKey:"__pipen-report-theme",select:{themes:["g10","white","g80","g90","g100"],labelText:"Theme",inline:!0}}});let r=e[2].length>0&&Je(e);return{c(){f(t.$$.fragment),n=S(),r&&r.c(),a=$()},m(e,l){A(t,e,l),i(e,n,l),r&&r.m(e,l),i(e,a,l),s=!0},p(e,t){e[2].length>0?r?(r.p(e,t),4&t&&p(r,1)):(r=Je(e),r.c(),p(r,1),r.m(a.parentNode,a)):r&&(o(),_(r,1,1,(()=>{r=null})),c())},i(e){s||(p(t.$$.fragment,e),p(r),s=!0)},o(e){_(t.$$.fragment,e),_(r),s=!1},d(e){e&&(d(n),d(a)),u(t,e),r&&r.d(e)}}}function _t(e){let t,n;return t=new se({props:{class:"header-util",$$slots:{default:[ct]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};8197&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function dt(e){let t;return{c(){t=C(e[4])},m(e,n){i(e,t,n)},p(e,n){16&n&&P(t,e[4])},d(e){e&&d(t)}}}function gt(e){let t,n,a,s,l=null!==e[4]&&dt(e);return{c(){t=r("h1"),n=C(e[3]),a=S(),l&&l.c(),s=$()},m(e,r){i(e,t,r),b(t,n),i(e,a,r),l&&l.m(e,r),i(e,s,r)},p(e,t){8&t&&P(n,e[3]),null!==e[4]?l?l.p(e,t):(l=dt(e),l.c(),l.m(s.parentNode,s)):l&&(l.d(1),l=null)},d(e){e&&(d(t),d(a),d(s)),l&&l.d(e)}}}function mt(e){let t,n;return t=new re({props:{class:"header-tile",slot:"platform",$$slots:{default:[gt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};8216&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function $t(e){let t,n,a;function s(t){e[6](t)}let r={$$slots:{platform:[mt],default:[_t]},$$scope:{ctx:e}};return void 0!==e[1]&&(r.isSideNavOpen=e[1]),t=new ae({props:r}),L.push((()=>E(t,"isSideNavOpen",s))),{c(){f(t.$$.fragment)},m(e,n){A(t,e,n),a=!0},p(e,[a]){const s={};8221&a&&(s.$$scope={dirty:a,ctx:e}),!n&&2&a&&(n=!0,s.isSideNavOpen=e[1],R((()=>n=!1))),t.$set(s)},i(e){a||(p(t.$$.fragment,e),a=!0)},o(e){_(t.$$.fragment,e),a=!1},d(e){u(t,e)}}}let ft=!1;function At(e,t,n){let{entries:a}=t,{logo:s}=t,{logotext:r=null}=t,{pipeline_name:l}=t,{isSideNavOpen:i}=t;return l=l||s,e.$$set=e=>{"entries"in e&&n(2,a=e.entries),"logo"in e&&n(3,s=e.logo),"logotext"in e&&n(4,r=e.logotext),"pipeline_name"in e&&n(0,l=e.pipeline_name),"isSideNavOpen"in e&&n(1,i=e.isSideNavOpen)},[l,i,a,s,r,{duration:200},function(e){i=e,n(1,i)}]}class ut extends t{constructor(e){super(),n(this,e,At,$t,a,{entries:2,logo:3,logotext:4,pipeline_name:0,isSideNavOpen:1})}}function St(e){let t,n,a;function s(t){e[3](t)}let r={modalHeading:"Running Information",passiveModal:!0,hasScrollingContent:!0,class:"runinfo",$$slots:{default:[Pt]},$$scope:{ctx:e}};return void 0!==e[2]&&(r.open=e[2]),t=new _e({props:r}),L.push((()=>E(t,"open",s))),{c(){f(t.$$.fragment)},m(e,n){A(t,e,n),a=!0},p(e,a){const s={};34&a&&(s.$$scope={dirty:a,ctx:e}),!n&&4&a&&(n=!0,s.open=e[2],R((()=>n=!1))),t.$set(s)},i(e){a||(p(t.$$.fragment,e),a=!0)},o(e){_(t.$$.fragment,e),a=!1},d(e){u(t,e)}}}function Lt(e){let t,n,a,s,r,l;return t=new me({props:{label:"Session Information"}}),a=new me({props:{label:"Time Spent"}}),r=new me({props:{label:"Device Information"}}),{c(){f(t.$$.fragment),n=S(),f(a.$$.fragment),s=S(),f(r.$$.fragment)},m(e,p){A(t,e,p),i(e,n,p),A(a,e,p),i(e,s,p),A(r,e,p),l=!0},p:h,i(e){l||(p(t.$$.fragment,e),p(a.$$.fragment,e),p(r.$$.fragment,e),l=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),_(r.$$.fragment,e),l=!1},d(e){e&&(d(n),d(s)),u(t,e),u(a,e),u(r,e)}}}function Et(e){let t,n;return t=new fe({props:{type:"multi",code:e[1].session}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};2&n&&(a.code=e[1].session),t.$set(a)},i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function Rt(e){let t,n;return t=new fe({props:{type:"multi",code:e[1].time}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};2&n&&(a.code=e[1].time),t.$set(a)},i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function Ct(e){let t,n;return t=new fe({props:{type:"multi",code:e[1].device}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};2&n&&(a.code=e[1].device),t.$set(a)},i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function bt(e){let t,n,a,s,r,l;return t=new $e({props:{$$slots:{default:[Et]},$$scope:{ctx:e}}}),a=new $e({props:{$$slots:{default:[Rt]},$$scope:{ctx:e}}}),r=new $e({props:{$$slots:{default:[Ct]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=S(),f(a.$$.fragment),s=S(),f(r.$$.fragment)},m(e,p){A(t,e,p),i(e,n,p),A(a,e,p),i(e,s,p),A(r,e,p),l=!0},p(e,n){const s={};34&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const l={};34&n&&(l.$$scope={dirty:n,ctx:e}),a.$set(l);const i={};34&n&&(i.$$scope={dirty:n,ctx:e}),r.$set(i)},i(e){l||(p(t.$$.fragment,e),p(a.$$.fragment,e),p(r.$$.fragment,e),l=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),_(r.$$.fragment,e),l=!1},d(e){e&&(d(n),d(s)),u(t,e),u(a,e),u(r,e)}}}function Pt(e){let t,n,a,s;return t=new de({props:{kind:"warning",lowContrast:!0,hideCloseButton:!0,title:"Showing the first job only. Check the workdir for information of other jobs if any."}}),a=new ge({props:{$$slots:{content:[bt],default:[Lt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=S(),f(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,t){const n={};34&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){s||(p(t.$$.fragment,e),p(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),u(t,e),u(a,e)}}}function Tt(e){let t,n,a;return{c(){t=r("a"),t.textContent="> Running Information",l(t,"href","javascript:void(0);")},m(s,r){i(s,t,r),n||(a=T(t,"click",e[4]),n=!0)},p:h,d(e){e&&d(t),n=!1,a()}}}function ht(e){let t,n,a,s,g,m,$,f,A=e[1]&&St(e),u=e[1]&&Tt(e);return{c(){A&&A.c(),t=S(),n=r("div"),a=r("div"),u&&u.c(),s=S(),g=r("div"),m=C("Powered by "),$=r("span"),l(a,"class","runinfo"),l($,"class","pipen-report-footer-versions"),l(g,"class","versions"),l(n,"class","pipen-report-footer bx--grid")},m(r,l){A&&A.m(r,l),i(r,t,l),i(r,n,l),b(n,a),u&&u.m(a,null),b(n,s),b(n,g),b(g,m),b(g,$),$.innerHTML=e[0],f=!0},p(e,[n]){e[1]?A?(A.p(e,n),2&n&&p(A,1)):(A=St(e),A.c(),p(A,1),A.m(t.parentNode,t)):A&&(o(),_(A,1,1,(()=>{A=null})),c()),e[1]?u?u.p(e,n):(u=Tt(e),u.c(),u.m(a,null)):u&&(u.d(1),u=null),(!f||1&n)&&($.innerHTML=e[0])},i(e){f||(p(A),f=!0)},o(e){_(A),f=!1},d(e){e&&(d(t),d(n)),A&&A.d(e),u&&u.d()}}}function Nt(e,t,n){let{versions:a}=t,{runinfo:s}=t,r=!1;return e.$$set=e=>{"versions"in e&&n(0,a=e.versions),"runinfo"in e&&n(1,s=e.runinfo)},[a,s,r,function(e){r=e,n(2,r)},()=>{n(2,r=!0)}]}class Mt extends t{constructor(e){super(),n(this,e,Nt,ht,a,{versions:0,runinfo:1})}}function vt(e){let t,n,a;function s(t){e[12](t)}let r={$$slots:{default:[It]},$$scope:{ctx:e}};return void 0!==e[0]&&(r.isOpen=e[0]),t=new ue({props:r}),L.push((()=>E(t,"isOpen",s))),{c(){f(t.$$.fragment)},m(e,n){A(t,e,n),a=!0},p(e,a){const s={};8576&a&&(s.$$scope={dirty:a,ctx:e}),!n&&1&a&&(n=!0,s.isOpen=e[0],R((()=>n=!1))),t.$set(s)},i(e){a||(p(t.$$.fragment,e),a=!0)},o(e){_(t.$$.fragment,e),a=!1},d(e){u(t,e)}}}function It(e){let t,n;return t=new je({props:{toc:e[7],page:e[8]}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};128&n&&(a.toc=e[7]),256&n&&(a.page=e[8]),t.$set(a)},i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function yt(e){let t;const n=e[10].default,a=N(n,e,e[13],null);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||8192&s)&&M(a,n,e,e[13],t?I(n,e[13],s,null):v(e[13]),null)},i(e){t||(p(a,e),t=!0)},o(e){_(a,e),t=!1},d(e){a&&a.d(e)}}}function xt(e){let t,n;return t=new Ee({props:{$$slots:{default:[yt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};8192&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function Gt(e){let t,n;return t=new Le({props:{$$slots:{default:[xt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};8192&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function wt(e){let t,n,a,s;return t=new Se({props:{$$slots:{default:[Gt]},$$scope:{ctx:e}}}),a=new Mt({props:{versions:e[3],runinfo:e[9]}}),{c(){f(t.$$.fragment),n=S(),f(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};8192&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};8&n&&(r.versions=e[3]),512&n&&(r.runinfo=e[9]),a.$set(r)},i(e){s||(p(t.$$.fragment,e),p(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),u(t,e),u(a,e)}}}function Bt(e){let t,n,a,s,r,l;function g(t){e[11](t)}let m={logo:e[1],logotext:e[2],entries:e[4],pipeline_name:e[5]};void 0!==e[0]&&(m.isSideNavOpen=e[0]),t=new ut({props:m}),L.push((()=>E(t,"isSideNavOpen",g)));let $=e[6]&&e[7].length>0&&vt(e);return r=new Ae({props:{$$slots:{default:[wt]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),a=S(),$&&$.c(),s=S(),f(r.$$.fragment)},m(e,n){A(t,e,n),i(e,a,n),$&&$.m(e,n),i(e,s,n),A(r,e,n),l=!0},p(e,[a]){const l={};2&a&&(l.logo=e[1]),4&a&&(l.logotext=e[2]),16&a&&(l.entries=e[4]),32&a&&(l.pipeline_name=e[5]),!n&&1&a&&(n=!0,l.isSideNavOpen=e[0],R((()=>n=!1))),t.$set(l),e[6]&&e[7].length>0?$?($.p(e,a),192&a&&p($,1)):($=vt(e),$.c(),p($,1),$.m(s.parentNode,s)):$&&(o(),_($,1,1,(()=>{$=null})),c());const i={};8712&a&&(i.$$scope={dirty:a,ctx:e}),r.$set(i)},i(e){l||(p(t.$$.fragment,e),p($),p(r.$$.fragment,e),l=!0)},o(e){_(t.$$.fragment,e),_($),_(r.$$.fragment,e),l=!1},d(e){e&&(d(a),d(s)),u(t,e),$&&$.d(e),u(r,e)}}}function Ht(e,t,n){let a,{$$slots:s={},$$scope:r}=t,{logo:l}=t,{logotext:i=null}=t,{versions:p}=t,{entries:o}=t,{pipeline_name:c}=t,{report_toc:_}=t,{toc:d}=t,{page:g}=t,{isSideNavOpen:$}=t,{runinfo:f}=t;return m((()=>{const e=new URL(window.location.href);let t;t=e.hash.length>0?document.querySelector(`li:has(>[href="${e.hash}"])`):document.querySelector("li:has(>.toc-current-page)"),t&&(t.classList.add("toc-active"),t.scrollIntoView());const n=document.querySelectorAll(".pipen-report-toc-anchor");a=new IntersectionObserver((e=>{e.forEach((e=>{const t=e.target.getAttribute("id"),n=document.querySelector(`li:has(>[href="#${t}"])`);n&&(e.isIntersecting?n.classList.add("toc-active"):n.classList.remove("toc-active"))}))}),{root:null,rootMargin:"0px",threshold:.5}),n.forEach((e=>{a.observe(e)}))})),e.$$set=e=>{"logo"in e&&n(1,l=e.logo),"logotext"in e&&n(2,i=e.logotext),"versions"in e&&n(3,p=e.versions),"entries"in e&&n(4,o=e.entries),"pipeline_name"in e&&n(5,c=e.pipeline_name),"report_toc"in e&&n(6,_=e.report_toc),"toc"in e&&n(7,d=e.toc),"page"in e&&n(8,g=e.page),"isSideNavOpen"in e&&n(0,$=e.isSideNavOpen),"runinfo"in e&&n(9,f=e.runinfo),"$$scope"in e&&n(13,r=e.$$scope)},[$,l,i,p,o,c,_,d,g,f,s,function(e){$=e,n(0,$)},function(e){$=e,n(0,$)},r]}class Ft extends t{constructor(e){super(),n(this,e,Ht,Bt,a,{logo:1,logotext:2,versions:3,entries:4,pipeline_name:5,report_toc:6,toc:7,page:8,isSideNavOpen:0,runinfo:9})}}function Ot(e){let t;return{c(){t=C(e[2])},m(e,n){i(e,t,n)},p:h,d(e){e&&d(t)}}}function Dt(e){let t,n,a,s;return n=new Re({props:{icon:e[1],kind:"tertiary",href:e[3](),$$slots:{default:[Ot]},$$scope:{ctx:e}}}),{c(){t=r("div"),f(n.$$.fragment),l(t,"class",a=`pagenav ${e[0]}`)},m(e,a){i(e,t,a),A(n,t,null),s=!0},p(e,[r]){const i={};16&r&&(i.$$scope={dirty:r,ctx:e}),n.$set(i),(!s||1&r&&a!==(a=`pagenav ${e[0]}`))&&l(t,"class",a)},i(e){s||(p(n.$$.fragment,e),s=!0)},o(e){_(n.$$.fragment,e),s=!1},d(e){e&&d(t),u(n)}}}function Kt(e,t,n){let{dir:a}=t,s="up"===a?ye:xe,r="up"===a?"Go to previous page":"Go to next page";return e.$$set=e=>{"dir"in e&&n(0,a=e.dir)},[a,s,r,()=>{let e=new URL(window.location.href);e.hash="";const t=parseInt(e.searchParams.get("page")||0),n="up"===a?t-1:t+1;return 0===n?e.searchParams.delete("page"):e.searchParams.set("page",n),e.toString()}]}class Ut extends t{constructor(e){super(),n(this,e,Kt,Dt,a,{dir:0})}}const Vt=e=>({}),zt=e=>({}),Xt=e=>({}),kt=e=>({}),qt=e=>({}),jt=e=>({}),Yt=e=>({}),Wt=e=>({});function Jt(e){let t,n;return t=new Ce({props:{ratio:e[5],$$slots:{default:[nn]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};32&n&&(a.ratio=e[5]),65631&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function Zt(e){let t,n,a,s,r=e[2]&&an(e),l=e[0]&&sn(e),g=e[1]&&rn(e);return{c(){r&&r.c(),t=S(),l&&l.c(),n=S(),g&&g.c(),a=$()},m(e,p){r&&r.m(e,p),i(e,t,p),l&&l.m(e,p),i(e,n,p),g&&g.m(e,p),i(e,a,p),s=!0},p(e,s){e[2]?r?(r.p(e,s),4&s&&p(r,1)):(r=an(e),r.c(),p(r,1),r.m(t.parentNode,t)):r&&(o(),_(r,1,1,(()=>{r=null})),c()),e[0]?l?(l.p(e,s),1&s&&p(l,1)):(l=sn(e),l.c(),p(l,1),l.m(n.parentNode,n)):l&&(o(),_(l,1,1,(()=>{l=null})),c()),e[1]?g?(g.p(e,s),2&s&&p(g,1)):(g=rn(e),g.c(),p(g,1),g.m(a.parentNode,a)):g&&(o(),_(g,1,1,(()=>{g=null})),c())},i(e){s||(p(r),p(l),p(g),s=!0)},o(e){_(r),_(l),_(g),s=!1},d(e){e&&(d(t),d(n),d(a)),r&&r.d(e),l&&l.d(e),g&&g.d(e)}}}function Qt(e){let t;const n=e[12].loading,a=N(n,e,e[16],kt);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||65536&s)&&M(a,n,e,e[16],t?I(n,e[16],s,Xt):v(e[16]),kt)},i(e){t||(p(a,e),t=!0)},o(e){_(a,e),t=!1},d(e){a&&a.d(e)}}}function en(e){let t,n,a,s,l,p,o=[e[9],{src:n=e[3]},{alt:e[4]},{title:"Click to zoom in ..."}],c={};for(let e=0;e<o.length;e+=1)c=G(c,o[e]);return{c(){t=r("img"),D(t,c)},m(n,a){i(n,t,a),s=!0,l||(p=T(t,"click",e[8]),l=!0)},p(a,r){D(t,c=K(o,[(e=a)[9],(!s||8&r&&!B(t.src,n=e[3]))&&{src:n},(!s||16&r)&&{alt:e[4]},{title:"Click to zoom in ..."}]))},i(n){s||(n&&H((()=>{s&&(a||(a=F(t,O,{duration:e[6]?on:0},!0)),a.run(1))})),s=!0)},o(n){n&&(a||(a=F(t,O,{duration:e[6]?on:0},!1)),a.run(0)),s=!1},d(e){e&&d(t),e&&a&&a.end(),l=!1,p()}}}function tn(e){let t;const n=e[12].error,a=N(n,e,e[16],zt);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||65536&s)&&M(a,n,e,e[16],t?I(n,e[16],s,Vt):v(e[16]),zt)},i(e){t||(p(a,e),t=!0)},o(e){_(a,e),t=!1},d(e){a&&a.d(e)}}}function nn(e){let t,n,a,s,r=e[2]&&Qt(e),l=e[0]&&en(e),g=e[1]&&tn(e);return{c(){r&&r.c(),t=S(),l&&l.c(),n=S(),g&&g.c(),a=$()},m(e,p){r&&r.m(e,p),i(e,t,p),l&&l.m(e,p),i(e,n,p),g&&g.m(e,p),i(e,a,p),s=!0},p(e,s){e[2]?r?(r.p(e,s),4&s&&p(r,1)):(r=Qt(e),r.c(),p(r,1),r.m(t.parentNode,t)):r&&(o(),_(r,1,1,(()=>{r=null})),c()),e[0]?l?(l.p(e,s),1&s&&p(l,1)):(l=en(e),l.c(),p(l,1),l.m(n.parentNode,n)):l&&(o(),_(l,1,1,(()=>{l=null})),c()),e[1]?g?(g.p(e,s),2&s&&p(g,1)):(g=tn(e),g.c(),p(g,1),g.m(a.parentNode,a)):g&&(o(),_(g,1,1,(()=>{g=null})),c())},i(e){s||(p(r),p(l),p(g),s=!0)},o(e){_(r),_(l),_(g),s=!1},d(e){e&&(d(t),d(n),d(a)),r&&r.d(e),l&&l.d(e),g&&g.d(e)}}}function an(e){let t;const n=e[12].loading,a=N(n,e,e[16],Wt);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||65536&s)&&M(a,n,e,e[16],t?I(n,e[16],s,Yt):v(e[16]),Wt)},i(e){t||(p(a,e),t=!0)},o(e){_(a,e),t=!1},d(e){a&&a.d(e)}}}function sn(e){let t,n,a,s,l,p,o=[e[9],{src:n=e[3]},{alt:e[4]},{title:"Click to zoom in ..."}],c={};for(let e=0;e<o.length;e+=1)c=G(c,o[e]);return{c(){t=r("img"),D(t,c)},m(n,a){i(n,t,a),s=!0,l||(p=T(t,"click",e[8]),l=!0)},p(a,r){D(t,c=K(o,[(e=a)[9],(!s||8&r&&!B(t.src,n=e[3]))&&{src:n},(!s||16&r)&&{alt:e[4]},{title:"Click to zoom in ..."}]))},i(n){s||(n&&H((()=>{s&&(a||(a=F(t,O,{duration:e[6]?on:0},!0)),a.run(1))})),s=!0)},o(n){n&&(a||(a=F(t,O,{duration:e[6]?on:0},!1)),a.run(0)),s=!1},d(e){e&&d(t),e&&a&&a.end(),l=!1,p()}}}function rn(e){let t;const n=e[12].error,a=N(n,e,e[16],jt);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||65536&s)&&M(a,n,e,e[16],t?I(n,e[16],s,qt):v(e[16]),jt)},i(e){t||(p(a,e),t=!0)},o(e){_(a,e),t=!1},d(e){a&&a.d(e)}}}function ln(e){let t,n,a,s;return{c(){t=r("img"),l(t,"class","pipen-report-image-lg"),B(t.src,n=e[3])||l(t,"src",n),l(t,"alt",e[4])},m(e,n){i(e,t,n),s=!0},p(a,r){e=a,(!s||8&r&&!B(t.src,n=e[3]))&&l(t,"src",n),(!s||16&r)&&l(t,"alt",e[4])},i(n){s||(n&&H((()=>{s&&(a||(a=F(t,O,{duration:e[6]?on:0},!0)),a.run(1))})),s=!0)},o(n){n&&(a||(a=F(t,O,{duration:e[6]?on:0},!1)),a.run(0)),s=!1},d(e){e&&d(t),e&&a&&a.end()}}}function pn(e){let t,n,a,s,r,l;const g=[Zt,Jt],m=[];function $(e,t){return void 0===e[5]?0:1}function C(t){e[13](t)}t=$(e),n=m[t]=g[t](e);let b={modalHeading:e[4],primaryButtonText:"Open in new window ...",secondaryButtonText:"Close",$$slots:{default:[ln]},$$scope:{ctx:e}};return void 0!==e[7]&&(b.open=e[7]),s=new _e({props:b}),L.push((()=>E(s,"open",C))),s.$on("click:button--secondary",e[14]),s.$on("click:button--primary",e[15]),{c(){n.c(),a=S(),f(s.$$.fragment)},m(e,n){m[t].m(e,n),i(e,a,n),A(s,e,n),l=!0},p(e,[l]){let i=t;t=$(e),t===i?m[t].p(e,l):(o(),_(m[i],1,1,(()=>{m[i]=null})),c(),n=m[t],n?n.p(e,l):(n=m[t]=g[t](e),n.c()),p(n,1),n.m(a.parentNode,a));const d={};16&l&&(d.modalHeading=e[4]),65624&l&&(d.$$scope={dirty:l,ctx:e}),!r&&128&l&&(r=!0,d.open=e[7],R((()=>r=!1))),s.$set(d)},i(e){l||(p(n),p(s.$$.fragment,e),l=!0)},o(e){_(n),_(s.$$.fragment,e),l=!1},d(e){e&&d(a),m[t].d(e),u(s,e)}}}const on=110;function cn(e,t,n){const a=["src","alt","ratio","loading","loaded","error","fadeIn","border","loadImage"];let s=y(t,a),{$$slots:r={},$$scope:l}=t,{src:i=""}=t,{alt:p=""}=t,{ratio:o}=t,{loading:c=!1}=t,{loaded:_=!1}=t,{error:d=!1}=t,{fadeIn:g=!1}=t,{border:$=!0}=t,f=!1;const A=e=>{null!=S&&(S=null),n(0,_=!1),n(1,d=!1),S=new Image,S.src=e||i,S.onload=()=>n(0,_=!0),S.onerror=()=>n(1,d=!0)},u=x();let S=null;m((()=>()=>S=null));const L={...s,class:[s.class,"pipen-report-image",$?"":"pipen-report-image-noborder"].filter(Boolean).join(" ")};return e.$$set=e=>{t=G(G({},t),w(e)),n(19,s=y(t,a)),"src"in e&&n(3,i=e.src),"alt"in e&&n(4,p=e.alt),"ratio"in e&&n(5,o=e.ratio),"loading"in e&&n(2,c=e.loading),"loaded"in e&&n(0,_=e.loaded),"error"in e&&n(1,d=e.error),"fadeIn"in e&&n(6,g=e.fadeIn),"border"in e&&n(10,$=e.border),"$$scope"in e&&n(16,l=e.$$scope)},e.$$.update=()=>{3&e.$$.dirty&&n(2,c=!_&&!d),8&e.$$.dirty&&i&&"undefined"!=typeof window&&A(),1&e.$$.dirty&&_&&u("load"),2&e.$$.dirty&&d&&u("error")},[_,d,c,i,p,o,g,f,()=>{n(7,f=!0)},L,$,A,r,function(e){f=e,n(7,f)},()=>{n(7,f=!1)},()=>{n(7,f=!1),window.open(i)},l]}class _n extends t{constructor(e){super(),n(this,e,cn,pn,a,{src:3,alt:4,ratio:5,loading:2,loaded:0,error:1,fadeIn:6,border:10,loadImage:11})}get loadImage(){return this.$$.ctx[11]}}function dn(e){let t,n;const a=[{src:e[1]},e[4]];let s={$$slots:{error:[$n],loading:[mn]},$$scope:{ctx:e}};for(let e=0;e<a.length;e+=1)s=G(s,a[e]);return t=new _n({props:s}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const s=18&n?K(a,[2&n&&{src:e[1]},16&n&&U(e[4])]):{};256&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s)},i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function gn(e){let t,n,a;const s=[{src:e[1]},e[4]];let l={$$slots:{error:[An],loading:[fn]},$$scope:{ctx:e}};for(let e=0;e<s.length;e+=1)l=G(l,s[e]);n=new _n({props:l});let o=[e[0]],c={};for(let e=0;e<o.length;e+=1)c=G(c,o[e]);return{c(){t=r("div"),f(n.$$.fragment),D(t,c)},m(e,s){i(e,t,s),A(n,t,null),a=!0},p(e,a){const r=18&a?K(s,[2&a&&{src:e[1]},16&a&&U(e[4])]):{};256&a&&(r.$$scope={dirty:a,ctx:e}),n.$set(r),D(t,c=K(o,[1&a&&e[0]]))},i(e){a||(p(n.$$.fragment,e),a=!0)},o(e){_(n.$$.fragment,e),a=!1},d(e){e&&d(t),u(n)}}}function mn(e){let t,n,a=[e[2]],s={};for(let e=0;e<a.length;e+=1)s=G(s,a[e]);return{c(){t=r("div"),n=C("Image loading ..."),D(t,s)},m(e,a){i(e,t,a),b(t,n)},p:h,d(e){e&&d(t)}}}function $n(e){let t,n,a=[e[3]],s={};for(let e=0;e<a.length;e+=1)s=G(s,a[e]);return{c(){t=r("div"),n=C("Image loading error!"),D(t,s)},m(e,a){i(e,t,a),b(t,n)},p:h,d(e){e&&d(t)}}}function fn(e){let t,n,a=[e[2]],s={};for(let e=0;e<a.length;e+=1)s=G(s,a[e]);return{c(){t=r("div"),n=C("Image loading ..."),D(t,s)},m(e,a){i(e,t,a),b(t,n)},p:h,d(e){e&&d(t)}}}function An(e){let t,n,a=[e[3]],s={};for(let e=0;e<a.length;e+=1)s=G(s,a[e]);return{c(){t=r("div"),n=C("Image loading error!"),D(t,s)},m(e,a){i(e,t,a),b(t,n)},p:h,d(e){e&&d(t)}}}function un(e){let t,n,a,s;const r=[gn,dn],l=[];function g(e,t){return e[0]?0:1}return t=g(e),n=l[t]=r[t](e),{c(){n.c(),a=$()},m(e,n){l[t].m(e,n),i(e,a,n),s=!0},p(e,[s]){let i=t;t=g(e),t===i?l[t].p(e,s):(o(),_(l[i],1,1,(()=>{l[i]=null})),c(),n=l[t],n?n.p(e,s):(n=l[t]=r[t](e),n.c()),p(n,1),n.m(a.parentNode,a))},i(e){s||(p(n),s=!0)},o(e){_(n),s=!1},d(e){e&&d(a),l[t].d(e)}}}function Sn(e,t,n){const a=["frameProps","width","height","src"];let s=y(t,a),{frameProps:r={class:"pipen-report-image-frame"}}=t,{width:l=300}=t,{height:i=200}=t,{src:p}=t,o=`width: ${l}px; height: ${i}px; aspect-ratio: ${l} / ${i}`,c={...s,class:[s.class,"pipen-report-image-loading"].filter(Boolean).join(" "),style:s.style?`${s.style}; ${o}`:o},_={...c,class:[c.class,"error"].join(" ")};return e.$$set=e=>{t=G(G({},t),w(e)),n(4,s=y(t,a)),"frameProps"in e&&n(0,r=e.frameProps),"width"in e&&n(5,l=e.width),"height"in e&&n(6,i=e.height),"src"in e&&n(1,p=e.src)},[r,p,c,_,s,l,i]}let Ln=class extends t{constructor(e){super(),n(this,e,Sn,un,a,{frameProps:0,width:5,height:6,src:1})}};const En=e=>({}),Rn=e=>({});function Cn(e){let t,n;const a=e[25].default,s=N(a,e,e[24],null);return{c(){t=r("ul"),s&&s.c(),l(t,"role","menu"),l(t,"tabindex","-1"),l(t,"aria-label",e[14]),l(t,"data-floating-menu-direction",e[5]),l(t,"class",e[9]),V(t,"bx--overflow-menu-options",!0),V(t,"bx--overflow-menu--flip",e[7]),V(t,"bx--overflow-menu-options--open",e[0]),V(t,"bx--overflow-menu-options--light",e[6]),V(t,"bx--overflow-menu-options--sm","sm"===e[4]),V(t,"bx--overflow-menu-options--xl","xl"===e[4]),V(t,"bx--breadcrumb-menu-options",!!e[15])},m(a,r){i(a,t,r),s&&s.m(t,null),e[32](t),n=!0},p(e,r){s&&s.p&&(!n||16777216&r[0])&&M(s,a,e,e[24],n?I(a,e[24],r,null):v(e[24]),null),(!n||16384&r[0])&&l(t,"aria-label",e[14]),(!n||32&r[0])&&l(t,"data-floating-menu-direction",e[5]),(!n||512&r[0])&&l(t,"class",e[9]),(!n||512&r[0])&&V(t,"bx--overflow-menu-options",!0),(!n||640&r[0])&&V(t,"bx--overflow-menu--flip",e[7]),(!n||513&r[0])&&V(t,"bx--overflow-menu-options--open",e[0]),(!n||576&r[0])&&V(t,"bx--overflow-menu-options--light",e[6]),(!n||528&r[0])&&V(t,"bx--overflow-menu-options--sm","sm"===e[4]),(!n||528&r[0])&&V(t,"bx--overflow-menu-options--xl","xl"===e[4]),(!n||33280&r[0])&&V(t,"bx--breadcrumb-menu-options",!!e[15])},i(e){n||(p(s,e),n=!0)},o(e){_(s,e),n=!1},d(n){n&&d(t),s&&s.d(n),e[32](null)}}}function bn(e){let t,n,a,s,l,g,m,L;const E=e[25].menu,R=N(E,e,e[24],Rn),C=R||function(e){let t,n,a;var s=e[1];function r(e,t){return{props:{"aria-label":e[11],title:e[11],class:"bx--overflow-menu__icon "+e[10]}}}return s&&(t=Y(s,r(e))),{c(){t&&f(t.$$.fragment),n=$()},m(e,s){t&&A(t,e,s),i(e,n,s),a=!0},p(e,a){if(2&a[0]&&s!==(s=e[1])){if(t){o();const e=t;_(e.$$.fragment,1,0,(()=>{u(e,1)})),c()}s?(t=Y(s,r(e)),f(t.$$.fragment),p(t.$$.fragment,1),A(t,n.parentNode,n)):t=null}else if(s){const n={};2048&a[0]&&(n["aria-label"]=e[11]),2048&a[0]&&(n.title=e[11]),1024&a[0]&&(n.class="bx--overflow-menu__icon "+e[10]),t.$set(n)}},i(e){a||(t&&p(t.$$.fragment,e),a=!0)},o(e){t&&_(t.$$.fragment,e),a=!1},d(e){e&&d(n),t&&u(t,e)}}}(e);let P=e[0]&&Cn(e),h=[{type:"button"},{"aria-haspopup":""},{"aria-expanded":e[0]},{"aria-label":e[14]},{id:e[12]},e[20]],y={};for(let e=0;e<h.length;e+=1)y=G(y,h[e]);return{c(){t=new W(!1),n=$(),a=S(),s=r("button"),C&&C.c(),l=S(),P&&P.c(),t.a=n,D(s,y),V(s,"bx--overflow-menu",!0),V(s,"bx--overflow-menu--open",e[0]),V(s,"bx--overflow-menu--light",e[6]),V(s,"bx--overflow-menu--sm","sm"===e[4]),V(s,"bx--overflow-menu--xl","xl"===e[4])},m(r,p){t.m(e[13],document.head),b(document.head,n),i(r,a,p),i(r,s,p),C&&C.m(s,null),b(s,l),P&&P.m(s,null),s.autofocus&&s.focus(),e[33](s),g=!0,m||(L=[T(window,"click",e[31]),T(s,"click",e[26]),T(s,"click",e[34]),T(s,"mouseover",e[27]),T(s,"mouseenter",e[28]),T(s,"mouseleave",e[29]),T(s,"keydown",e[30]),T(s,"keydown",e[35]),T(s,"focusout",e[36])],m=!0)},p(e,n){(!g||8192&n[0])&&t.p(e[13]),R?R.p&&(!g||16777216&n[0])&&M(R,E,e,e[24],g?I(E,e[24],n,En):v(e[24]),Rn):C&&C.p&&(!g||3074&n[0])&&C.p(e,g?n:[-1,-1]),e[0]?P?(P.p(e,n),1&n[0]&&p(P,1)):(P=Cn(e),P.c(),p(P,1),P.m(s,null)):P&&(o(),_(P,1,1,(()=>{P=null})),c()),D(s,y=K(h,[{type:"button"},{"aria-haspopup":""},(!g||1&n[0])&&{"aria-expanded":e[0]},(!g||16384&n[0])&&{"aria-label":e[14]},(!g||4096&n[0])&&{id:e[12]},1048576&n[0]&&e[20]])),V(s,"bx--overflow-menu",!0),V(s,"bx--overflow-menu--open",e[0]),V(s,"bx--overflow-menu--light",e[6]),V(s,"bx--overflow-menu--sm","sm"===e[4]),V(s,"bx--overflow-menu--xl","xl"===e[4])},i(e){g||(p(C,e),p(P),g=!0)},o(e){_(C,e),_(P),g=!1},d(r){r&&(t.d(),d(a),d(s)),d(n),C&&C.d(r),P&&P.d(),e[33](null),m=!1,z(L)}}}function Pn(e,t,n){let a,s;const r=["size","direction","open","light","flipped","persistentClickItems","menuOptionsClass","icon","iconClass","iconDescription","id","buttonRef","menuRef"];let l,i,p,o=y(t,r),{$$slots:c={},$$scope:_}=t,{size:d}=t,{direction:g="bottom"}=t,{open:m=!1}=t,{light:$=!1}=t,{flipped:f=!1}=t,{persistentClickItems:A=!1}=t,{menuOptionsClass:u}=t,{icon:S=we}=t,{iconClass:E}=t,{iconDescription:R="Open and close list of options"}=t,{id:C="ccs-"+Math.random().toString(36)}=t,{buttonRef:b=null}=t,{menuRef:P=null}=t;const T=X("BreadcrumbItem"),h=x(),N=J([]);k(e,N,(e=>n(23,i=e)));const M=J(void 0);k(e,M,(e=>n(38,p=e)));const v=J(void 0),I=J(-1);let B;k(e,I,(e=>n(22,l=e)));let H=!0;q("OverflowMenu",{focusedId:v,add:({id:e,text:t,primaryFocus:n,disabled:a})=>{N.update((s=>(n&&I.set(s.length),[...s,{id:e,text:t,primaryFocus:n,disabled:a,index:s.length}])))},update:e=>{M.set(e)},change:e=>{let t=l+e;t<0?t=i.length-1:t>=i.length&&(t=0);let n=i[t].disabled;for(;n;)t+=e,t<0?t=i.length-1:t>=i.length&&(t=0),n=i[t].disabled;I.set(t)}}),j((()=>{if(p){const{index:e,text:t}=i.filter((e=>e.id===p))[0];h("close",{index:e,text:t}),n(0,m=!1)}if(m){const{width:e,height:t}=b.getBoundingClientRect();n(21,B=e),!H&&l<0&&P.focus(),f&&(n(3,P.style.left="auto",P),n(3,P.style.right=0,P)),"top"===g?(n(3,P.style.top="auto",P),n(3,P.style.bottom=t+"px",P)):"bottom"===g&&n(3,P.style.top=t+"px",P),T&&(n(3,P.style.top=t+10+"px",P),n(3,P.style.left="-11px",P))}m||(N.set([]),M.set(void 0),I.set(0)),H=!1}));return e.$$set=e=>{n(40,t=G(G({},t),w(e))),n(20,o=y(t,r)),"size"in e&&n(4,d=e.size),"direction"in e&&n(5,g=e.direction),"open"in e&&n(0,m=e.open),"light"in e&&n(6,$=e.light),"flipped"in e&&n(7,f=e.flipped),"persistentClickItems"in e&&n(8,A=e.persistentClickItems),"menuOptionsClass"in e&&n(9,u=e.menuOptionsClass),"icon"in e&&n(1,S=e.icon),"iconClass"in e&&n(10,E=e.iconClass),"iconDescription"in e&&n(11,R=e.iconDescription),"id"in e&&n(12,C=e.id),"buttonRef"in e&&n(2,b=e.buttonRef),"menuRef"in e&&n(3,P=e.menuRef),"$$scope"in e&&n(24,_=e.$$scope)},e.$$.update=()=>{n(14,a=t["aria-label"]||"menu"),12582912&e.$$.dirty[0]&&i[l]&&v.set(i[l].id),2101248&e.$$.dirty[0]&&n(13,s=`<style>\n    #${C} .bx--overflow-menu-options.bx--overflow-menu-options:after {\n      width: ${B?B+"px":"2rem"};\n    }\n  </style>`)},T&&n(1,S=Ge),t=w(t),[m,S,b,P,d,g,$,f,A,u,E,R,C,s,a,T,h,N,M,I,o,B,l,i,_,c,function(t){Z.call(this,e,t)},function(t){Z.call(this,e,t)},function(t){Z.call(this,e,t)},function(t){Z.call(this,e,t)},function(t){Z.call(this,e,t)},({target:e})=>{b&&b.contains(e)||P&&!P.contains(e)&&n(0,m=!1)},function(e){L[e?"unshift":"push"]((()=>{P=e,n(3,P)}))},function(e){L[e?"unshift":"push"]((()=>{b=e,n(2,b)}))},e=>{A&&e.stopPropagation(),P&&P.contains(e.target)||(n(0,m=!m),m||h("close"))},e=>{m&&(["ArrowDown","ArrowLeft","ArrowRight","ArrowUp"].includes(e.key)?e.preventDefault():"Escape"===e.key&&(e.stopPropagation(),h("close"),n(0,m=!1),b.focus()))},e=>{m&&(b.contains(e.relatedTarget)||(h("close"),n(0,m=!1)))}]}class Tn extends t{constructor(e){super(),n(this,e,Pn,bn,a,{size:4,direction:5,open:0,light:6,flipped:7,persistentClickItems:8,menuOptionsClass:9,icon:1,iconClass:10,iconDescription:11,id:12,buttonRef:2,menuRef:3},null,[-1,-1])}}function hn(e,t,n){const a=e.slice();return a[3]=t[n],a[4]=t,a[5]=n,a}function Nn(e){let t,n,a,s,o;function c(t){e[1](t,e[3])}let g={labelText:e[3].value};return void 0!==e[3].selected&&(g.checked=e[3].selected),n=new be({props:g}),L.push((()=>E(n,"checked",c))),n.$on("change",e[2]),{c(){t=r("li"),f(n.$$.fragment),s=S(),l(t,"class","pipen-report-column-selector-item svelte-147jw2r")},m(e,a){i(e,t,a),A(n,t,null),b(t,s),o=!0},p(t,s){e=t;const r={};1&s&&(r.labelText=e[3].value),!a&&1&s&&(a=!0,r.checked=e[3].selected,R((()=>a=!1))),n.$set(r)},i(e){o||(p(n.$$.fragment,e),o=!0)},o(e){_(n.$$.fragment,e),o=!1},d(e){e&&d(t),u(n)}}}function Mn(e){let t,n,a=s(e[0]),r=[];for(let t=0;t<a.length;t+=1)r[t]=Nn(hn(e,a,t));const l=e=>_(r[e],1,1,(()=>{r[e]=null}));return{c(){for(let e=0;e<r.length;e+=1)r[e].c();t=$()},m(e,a){for(let t=0;t<r.length;t+=1)r[t]&&r[t].m(e,a);i(e,t,a),n=!0},p(e,n){if(1&n){let i;for(a=s(e[0]),i=0;i<a.length;i+=1){const s=hn(e,a,i);r[i]?(r[i].p(s,n),p(r[i],1)):(r[i]=Nn(s),r[i].c(),p(r[i],1),r[i].m(t.parentNode,t))}for(o(),i=a.length;i<r.length;i+=1)l(i);c()}},i(e){if(!n){for(let e=0;e<a.length;e+=1)p(r[e]);n=!0}},o(e){r=r.filter(Boolean);for(let e=0;e<r.length;e+=1)_(r[e]);n=!1},d(e){e&&d(t),g(r,e)}}}function vn(e){let t,n;return t=new Tn({props:{class:"pipen-report-overflowmenu",persistentClickItems:!0,size:"sm",icon:Be,iconDescription:"Select visible columns ...",$$slots:{default:[Mn]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,[n]){const a={};65&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function In(e,t,n){let{allColumns:a}=t;return e.$$set=e=>{"allColumns"in e&&n(0,a=e.allColumns)},[a,function(t,s){e.$$.not_equal(s.selected,t)&&(s.selected=t,n(0,a))},()=>{n(0,a)}]}class yn extends t{constructor(e){super(),n(this,e,In,vn,a,{allColumns:0})}}function xn(e){let t,n;return t=new Re({props:{iconDescription:"Right click and save as to download the entire data",tooltipPosition:"top",icon:Fe,href:e[2]}}),t.$on("click",e[13]),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};4&n&&(a.href=e[2]),t.$set(a)},i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function Gn(e){let t,n,a,s,r,l,g,m,C,b;function P(t){e[17](t)}let T={persistent:!0};function h(t){e[18](t)}void 0!==e[7]&&(T.value=e[7]),t=new Me({props:T}),L.push((()=>E(t,"value",P)));let N={};void 0!==e[6]&&(N.allColumns=e[6]),s=new yn({props:N}),L.push((()=>E(s,"allColumns",h))),g=new Re({props:{iconDescription:`Download current ${e[11].length} records (may be partial of the entire data)`,tooltipPosition:"top",icon:He}}),g.$on("click",e[14]);let M=!!e[2]&&xn(e);return{c(){f(t.$$.fragment),a=S(),f(s.$$.fragment),l=S(),f(g.$$.fragment),m=S(),M&&M.c(),C=$()},m(e,n){A(t,e,n),i(e,a,n),A(s,e,n),i(e,l,n),A(g,e,n),i(e,m,n),M&&M.m(e,n),i(e,C,n),b=!0},p(e,a){const l={};!n&&128&a&&(n=!0,l.value=e[7],R((()=>n=!1))),t.$set(l);const i={};!r&&64&a&&(r=!0,i.allColumns=e[6],R((()=>r=!1))),s.$set(i);const d={};2048&a&&(d.iconDescription=`Download current ${e[11].length} records (may be partial of the entire data)`),g.$set(d),e[2]?M?(M.p(e,a),4&a&&p(M,1)):(M=xn(e),M.c(),p(M,1),M.m(C.parentNode,C)):M&&(o(),_(M,1,1,(()=>{M=null})),c())},i(e){b||(p(t.$$.fragment,e),p(s.$$.fragment,e),p(g.$$.fragment,e),p(M),b=!0)},o(e){_(t.$$.fragment,e),_(s.$$.fragment,e),_(g.$$.fragment,e),_(M),b=!1},d(e){e&&(d(a),d(l),d(m),d(C)),u(t,e),u(s,e),u(g,e),M&&M.d(e)}}}function wn(e){let t,n;return t=new Ne({props:{$$slots:{default:[Gn]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};134219972&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function Bn(e){let t,n;return t=new he({props:{size:e[12],$$slots:{default:[wn]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};134219972&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function Hn(e){let t;return{c(){t=r("p"),t.textContent='You must use "Save as" or "Save link as" from the context menu (by right-clicking the button) to download the entire data.'},m(e,n){i(e,t,n)},p:h,d(e){e&&d(t)}}}function Fn(e){let t,n,a,s,r,l,o,c,g;const m=[{size:e[0]},{zebra:e[3]},{sortable:e[4]},{page:e[8]},{pageSize:e[9]},{rows:e[11]},{headers:e[1]},e[15]];let $={$$slots:{default:[Bn]},$$scope:{ctx:e}};for(let e=0;e<m.length;e+=1)$=G($,m[e]);function C(t){e[19](t)}function b(t){e[20](t)}t=new Pe({props:$});let P={pageSizes:e[5],totalItems:e[11].length};function T(t){e[21](t)}void 0!==e[9]&&(P.pageSize=e[9]),void 0!==e[8]&&(P.page=e[8]),a=new Te({props:P}),L.push((()=>E(a,"pageSize",C))),L.push((()=>E(a,"page",b)));let h={passiveModal:!0,modalHeading:"Downloading entire data",$$slots:{default:[Hn]},$$scope:{ctx:e}};return void 0!==e[10]&&(h.open=e[10]),o=new _e({props:h}),L.push((()=>E(o,"open",T))),o.$on("open",e[22]),o.$on("close",e[23]),{c(){f(t.$$.fragment),n=S(),f(a.$$.fragment),l=S(),f(o.$$.fragment)},m(e,s){A(t,e,s),i(e,n,s),A(a,e,s),i(e,l,s),A(o,e,s),g=!0},p(e,[n]){const l=35611&n?K(m,[1&n&&{size:e[0]},8&n&&{zebra:e[3]},16&n&&{sortable:e[4]},256&n&&{page:e[8]},512&n&&{pageSize:e[9]},2048&n&&{rows:e[11]},2&n&&{headers:e[1]},32768&n&&U(e[15])]):{};134219972&n&&(l.$$scope={dirty:n,ctx:e}),t.$set(l);const i={};32&n&&(i.pageSizes=e[5]),2048&n&&(i.totalItems=e[11].length),!s&&512&n&&(s=!0,i.pageSize=e[9],R((()=>s=!1))),!r&&256&n&&(r=!0,i.page=e[8],R((()=>r=!1))),a.$set(i);const p={};134217728&n&&(p.$$scope={dirty:n,ctx:e}),!c&&1024&n&&(c=!0,p.open=e[10],R((()=>c=!1))),o.$set(p)},i(e){g||(p(t.$$.fragment,e),p(a.$$.fragment,e),p(o.$$.fragment,e),g=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),_(o.$$.fragment,e),g=!1},d(e){e&&(d(n),d(l)),u(t,e),u(a,e),u(o,e)}}}function On(e,t,n){const a=["size","src","zebra","sortable","rows","headers","pageSizes"];let s=y(t,a),{size:r="sm"}=t,{src:l}=t;r={sm:"short",small:"short",short:"short",md:"medium",medium:"medium",lg:"tall",large:"tall",tall:"tall",compact:"compact"}[r];let i={short:"sm",compact:"sm",medium:"default",tall:"default"}[r],{zebra:p=!0}=t,{sortable:o=!0}=t,{rows:c=[]}=t,{headers:_=[]}=t,{pageSizes:d=[20,50,75,100]}=t,g=_.map((e=>({...e,selected:!0}))),m=1,$=d[0],f="",A=!1,u=c;return e.$$set=e=>{t=G(G({},t),w(e)),n(15,s=y(t,a)),"size"in e&&n(0,r=e.size),"src"in e&&n(2,l=e.src),"zebra"in e&&n(3,p=e.zebra),"sortable"in e&&n(4,o=e.sortable),"rows"in e&&n(16,c=e.rows),"headers"in e&&n(1,_=e.headers),"pageSizes"in e&&n(5,d=e.pageSizes)},e.$$.update=()=>{64&e.$$.dirty&&n(1,_=g.filter((e=>e.selected))),65664&e.$$.dirty&&(f.length>0?n(11,u=c.filter((e=>Object.values(e).filter((e=>!!e&&e.toString().includes(f))).length>0))):n(11,u=c))},[r,_,l,p,o,d,g,f,m,$,A,u,i,e=>{e.preventDefault(),n(10,A=!0)},()=>{((e,t)=>{const n=document.createElement("a");n.href=e,n.download=t,document.body.appendChild(n),n.click(),document.body.removeChild(n)})("data:application/vnd.ms-excel;charset=utf-8,\ufeff"+encodeURIComponent(function(){const e=_.map((e=>e.value)),t=[];return u.forEach((e=>{const n=_.map((t=>`"${e[t.key]}"`));t.push(n.join(","))})),e.join(",")+"\r\n"+t.join("\r\n")}()),l&&l.length>0?l.split("/").pop().split(".")[0]+".xls":"data-subset.xls")},s,c,function(e){f=e,n(7,f)},function(e){g=e,n(6,g)},function(e){$=e,n(9,$)},function(e){m=e,n(8,m)},function(e){A=e,n(10,A)},function(t){Z.call(this,e,t)},function(t){Z.call(this,e,t)}]}class Dn extends t{constructor(e){super(),n(this,e,On,Fn,a,{size:0,src:2,zebra:3,sortable:4,rows:16,headers:1,pageSizes:5})}}function Kn(e){let t,n,a;const s=[{headers:e[0]},{rows:e[1]},e[3]];let l={};for(let e=0;e<s.length;e+=1)l=G(l,s[e]);n=new Dn({props:l});let o=[e[2]],c={};for(let e=0;e<o.length;e+=1)c=G(c,o[e]);return{c(){t=r("div"),f(n.$$.fragment),D(t,c)},m(e,s){i(e,t,s),A(n,t,null),a=!0},p(e,a){const r=11&a?K(s,[1&a&&{headers:e[0]},2&a&&{rows:e[1]},8&a&&U(e[3])]):{};n.$set(r),D(t,c=K(o,[4&a&&e[2]]))},i(e){a||(p(n.$$.fragment,e),a=!0)},o(e){_(n.$$.fragment,e),a=!1},d(e){e&&d(t),u(n)}}}function Un(e){let t,n;const a=[{headers:e[0]},{rows:e[1]},e[3]];let s={};for(let e=0;e<a.length;e+=1)s=G(s,a[e]);return t=new Dn({props:s}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const s=11&n?K(a,[1&n&&{headers:e[0]},2&n&&{rows:e[1]},8&n&&U(e[3])]):{};t.$set(s)},i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function Vn(e){let t,n,a,s;const r=[Un,Kn],l=[];function g(e,t){return e[2]?1:0}return t=g(e),n=l[t]=r[t](e),{c(){n.c(),a=$()},m(e,n){l[t].m(e,n),i(e,a,n),s=!0},p(e,[s]){let i=t;t=g(e),t===i?l[t].p(e,s):(o(),_(l[i],1,1,(()=>{l[i]=null})),c(),n=l[t],n?n.p(e,s):(n=l[t]=r[t](e),n.c()),p(n,1),n.m(a.parentNode,a))},i(e){s||(p(n),s=!0)},o(e){_(n),s=!1},d(e){e&&d(a),l[t].d(e)}}}function zn(e,t,n){const a=["frameProps","data","headers","rows"];let s=y(t,a),{frameProps:r={class:"pipen-report-datatable-frame"}}=t,{data:l={}}=t,{headers:i=[]}=t,{rows:p=[]}=t;if(i.length+p.length>0&&l.length>0)throw"Can only specify `data` or `headers`/`rows` for `DataTable`!";return l.length>0&&(p=l,i=Object.keys(p[0]).filter((e=>"id"!==e)).map((e=>({key:e,value:e})))),e.$$set=e=>{t=G(G({},t),w(e)),n(3,s=y(t,a)),"frameProps"in e&&n(2,r=e.frameProps),"data"in e&&n(4,l=e.data),"headers"in e&&n(0,i=e.headers),"rows"in e&&n(1,p=e.rows)},[i,p,r,s,l]}class Xn extends t{constructor(e){super(),n(this,e,zn,Vn,a,{frameProps:2,data:4,headers:0,rows:1})}}function kn(e){let t,n;return{c(){t=r("div"),n=C(e[0]),l(t,"class","pipen-report-descr-title svelte-1xzxuva")},m(e,a){i(e,t,a),b(t,n)},p(e,t){1&t&&P(n,e[0])},d(e){e&&d(t)}}}function qn(e){let t,n,a,s=e[0]&&kn(e);const l=e[3].default,o=N(l,e,e[2],null);let c=[e[1]],g={};for(let e=0;e<c.length;e+=1)g=G(g,c[e]);return{c(){t=r("div"),s&&s.c(),n=S(),o&&o.c(),D(t,g),V(t,"svelte-1xzxuva",!0)},m(e,r){i(e,t,r),s&&s.m(t,null),b(t,n),o&&o.m(t,null),a=!0},p(e,[r]){e[0]?s?s.p(e,r):(s=kn(e),s.c(),s.m(t,n)):s&&(s.d(1),s=null),o&&o.p&&(!a||4&r)&&M(o,l,e,e[2],a?I(l,e[2],r,null):v(e[2]),null),D(t,g=K(c,[2&r&&e[1]])),V(t,"svelte-1xzxuva",!0)},i(e){a||(p(o,e),a=!0)},o(e){_(o,e),a=!1},d(e){e&&d(t),s&&s.d(),o&&o.d(e)}}}function jn(e,t,n){const a=["title"];let s=y(t,a),{$$slots:r={},$$scope:l}=t,{title:i}=t,{klass:p,...o}=s;return p=p?`pipen-report-descr ${p}`:"pipen-report-descr",o||(o={}),o.class=p,e.$$set=e=>{t=G(G({},t),w(e)),n(5,s=y(t,a)),"title"in e&&n(0,i=e.title),"$$scope"in e&&n(2,l=e.$$scope)},[i,o,l,r]}class Yn extends t{constructor(e){super(),n(this,e,jn,qn,a,{title:0})}}function Wn(e){let t,n,a,s;return t=new me({props:{label:"Plot"}}),a=new me({props:{label:"Table"}}),{c(){f(t.$$.fragment),n=S(),f(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p:h,i(e){s||(p(t.$$.fragment,e),p(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),u(t,e),u(a,e)}}}function Jn(e){let t,n;return t=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/gsea_table.png",width:1e3,height:760}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p:h,i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function Zn(e){let t,n;return t=new Xn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/fgsea.txt",data:[{pathway:"HALLMARK_INTERFERON_ALPHA_RESPONSE",pval:3e-4,padj:.0047,log2err:.4985,ES:.6982,NES:1.4923,size:96,leadingEdge:"CD74,PSMB8,ISG20,GBP4,GBP2,UBE2L6,IFITM3,NCOA7,PSME2,PSMA3,PSMB9,USP18,TAP1,RTP4,PLSCR1,IRF1,LY6E,HELZ2,EPSTI1,PSME1,IFIH1,IFI35,ISG15,SP110,SELL,SLC25A28,OAS1,HERC6,ADAR,OASL,MVB12A,NMI,IL15,TMEM140,PARP9,B2M,TRIM25,MOV10,IFI27,LAMP3,BST2,TDRD7,PARP12,IRF2,LAP3,CD47,CXCL10,OGFR,STAT2,SAMD9L,DHX58,TRAFD1,IRF9,PNPT1,TRIM21,CNP,IRF7,ELF1",id:0},{pathway:"HALLMARK_APICAL_SURFACE",pval:.0108,padj:.0754,log2err:.3807,ES:-.6423,NES:-1.5497,size:44,leadingEdge:"CD160,HSPB1,LYN,FLOT2,CROCC,APP,EPHB4,LYPD3,TMEM8B,MAL,ATP8B1,GSTM3,SCUBE1,AKAP7,B4GALT1,PLAUR,BRCA1",id:1},{pathway:"HALLMARK_ANGIOGENESIS",pval:.0355,padj:.1655,log2err:.3218,ES:-.6139,NES:-1.4101,size:36,leadingEdge:"TIMP1,CCND2,APP,VAV2,ITGAV",id:2},{pathway:"HALLMARK_MYC_TARGETS_V2",pval:.1224,padj:.4283,log2err:.1336,ES:.6072,NES:1.2351,size:58,leadingEdge:"DUSP2,CBX3,HSPD1,PHB,WDR74,MCM4,DCTPP1,NOP16,NOP56,MRTO4,HSPE1,NPM1,NOC4L,TCOF1,PLK1,NIP7,CDK4,PPAN,PLK4,MYC,SUPV3L1,LAS1L,AIMP2,SORD,PRMT3,RABEPK,PES1,NOP2,GNL3,FARSA,EXOSC5,RCL1,PA2G4,UNG,RRP9,MYBBP1A,IMP4,BYSL,DDX18,UTP20,HK2,TBRG4,NOLC1,SLC19A1,PUS1",id:3},{pathway:"HALLMARK_PANCREAS_BETA_CELLS",pval:.2713,padj:.6394,log2err:.086,ES:.5816,NES:1.1318,size:40,leadingEdge:"SRP9,SEC11A,FOXO1,SRPRB,SRP14,SPCS1,DPP4,VDR",id:4},{pathway:"HALLMARK_IL6_JAK_STAT3_SIGNALING",pval:.274,padj:.6394,log2err:.0789,ES:.5225,NES:1.1091,size:87,leadingEdge:"BAK1,CD38,STAM2,MYD88,ITGA4,IRF1,PTPN11,CSF2,TNFRSF1B,PIM1,IL15RA,TNFRSF1A,STAT1,IL2RG,GRB2,FAS,LEPR,EBI3,CXCL10,IL10RB,CXCL13,STAT2,IRF9,PIK3R5,TYK2,IL18R1,LTBR,ACVR1B",id:5},{pathway:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",pval:.4017,padj:.7161,log2err:.0644,ES:.5228,NES:1.0418,size:49,leadingEdge:"TXN,GLRX,NDUFA6,PRDX1,NDUFB4,GLRX2,PRDX6,TXNRD2,PFKP,G6PD,NDUFS2,LAMTOR5,ATOX1,SOD1,FTL,PRDX2,HMOX2",id:6},{pathway:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",pval:.42,padj:.7161,log2err:.0596,ES:.4923,NES:1.0252,size:73,leadingEdge:"FABP5,ACTG1,PLSCR1,HSD17B7,STX5,ANXA5,ANTXR2,PMVK,FBXO6,GLDC,FASN,FADS2,ALCAM,SREBF2,JAG1,STARD4,MVD,FDPS,ERRFI1,ETHE1,CYP51A1,DHCR7,SQLE,TP53INP1,PPARG,ECH1,IDI1,LDLR,ACAT2,GUSB,ABCA2,PCYT2,NFIL3",id:7},{pathway:"HALLMARK_WNT_BETA_CATENIN_SIGNALING",pval:.5137,padj:.7161,log2err:.1421,ES:-.4037,NES:-.9605,size:42,leadingEdge:"TCF7,CCND2,DVL2,CUL1,DLL1,PSEN2,PTCH1,NKD1,WNT1,HDAC2,HDAC5,CTNNB1",id:8},{pathway:"HALLMARK_ANDROGEN_RESPONSE",pval:.5613,padj:.7161,log2err:.0442,ES:.4487,NES:.9629,size:99,leadingEdge:"SRP19,ARID5B,FKBP5,SPCS3,CENPN,DBI,DHCR24,NGLY1,INSIG1,SAT1,UBE2J1,ADRM1,STK39,MAF,ELOVL5,UBE2I,B2M,TMEM50A,HOMER2,CCND3,RAB4A,CDK6,SRF,XRCC6,ELL2,ABHD2,SMS,PMEPA1,SORD,INPP4B,ABCC4,CDC14B",id:9},{pathway:"HALLMARK_TGF_BETA_SIGNALING",pval:.6115,padj:.7161,log2err:.1404,ES:-.3835,NES:-.9516,size:54,leadingEdge:"JUNB,PPP1R15A,TGFBR1,ID3,ID1,IFNGR2,SMURF2,XIAP,SKI,THBS1,SLC20A1,SMAD3,SERPINE1,ENG,FNTA,WWTR1,HIPK2,TRIM33,PPM1A,KLF10,BMPR1A,SMURF1,CTNNB1",id:10},{pathway:"HALLMARK_PROTEIN_SECRETION",pval:.6138,padj:.7161,log2err:.0404,ES:.4392,NES:.9359,size:95,leadingEdge:"CLTA,STX7,ICA1,CLTC,COPB2,SEC22B,BET1,COPE,NAPA,AP1G1,VAMP7,ANP32E,AP2M1,DNM1L,AP2S1,RER1,SOD1,GOSR2,CLCN3,STX12,TMX1,CLN5,GOLGA4,GNAS,ARFGEF1,ARCN1,RAB5A,STAM,ADAM10,YKT6,MON2,RPS6KA3,NAPG,TOM1L1,CTSC,LMAN1,RAB9A,SGMS1,ATP6V1H,ARF1,PPT1,TMED2",id:11},{pathway:"HALLMARK_HEDGEHOG_SIGNALING",pval:.8177,padj:.8501,log2err:.1059,ES:-.3646,NES:-.8375,size:36,leadingEdge:"ADGRG1,CELSR1,AMOT,PTCH1,TLE3",id:12},{pathway:"HALLMARK_NOTCH_SIGNALING",pval:.8501,padj:.8501,log2err:.0317,ES:.4113,NES:.777,size:32,leadingEdge:"SAP30,RBX1,LFNG,SKP1,JAG1,DTX2,FBXW11,HES1,APH1A,CCND1",id:13}]}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p:h,i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function Qn(e){let t,n,a,s;return t=new $e({props:{$$slots:{default:[Jn]},$$scope:{ctx:e}}}),a=new $e({props:{$$slots:{default:[Zn]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=S(),f(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(p(t.$$.fragment,e),p(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),u(t,e),u(a,e)}}}function ea(e){let t,n;return t=new ge({props:{$$slots:{content:[Qn],default:[Wn]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function ta(e){let t,n,a,s,o,c,g,m,$,L,E,R,C,P,T,N,M,v,I,y,x,G,w,B,H,F,O,D,K,U,V,z,X,k,q,j,Y,W,J,Z,ee,te,ne,ae,se,re,le,ie,pe,oe,ce;return a=new Yn({props:{title:"HALLMARK_INTERFERON_ALPHA_RESPONSE",class:"pipen-report-table-image-descr"}}),o=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/fgsea_HALLMARK_INTERFERON_ALPHA_RESPONSE.png",class:"pipen-report-table-image",width:1e3,height:800}}),m=new Yn({props:{title:"HALLMARK_APICAL_SURFACE",class:"pipen-report-table-image-descr"}}),L=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/fgsea_HALLMARK_APICAL_SURFACE.png",class:"pipen-report-table-image",width:1e3,height:800}}),C=new Yn({props:{title:"HALLMARK_ANGIOGENESIS",class:"pipen-report-table-image-descr"}}),T=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/fgsea_HALLMARK_ANGIOGENESIS.png",class:"pipen-report-table-image",width:1e3,height:800}}),v=new Yn({props:{title:"HALLMARK_MYC_TARGETS_V2",class:"pipen-report-table-image-descr"}}),y=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/fgsea_HALLMARK_MYC_TARGETS_V2.png",class:"pipen-report-table-image",width:1e3,height:800}}),w=new Yn({props:{title:"HALLMARK_PANCREAS_BETA_CELLS",class:"pipen-report-table-image-descr"}}),H=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/fgsea_HALLMARK_PANCREAS_BETA_CELLS.png",class:"pipen-report-table-image",width:1e3,height:800}}),D=new Yn({props:{title:"HALLMARK_IL6_JAK_STAT3_SIGNALING",class:"pipen-report-table-image-descr"}}),U=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/fgsea_HALLMARK_IL6_JAK_STAT3_SIGNALING.png",class:"pipen-report-table-image",width:1e3,height:800}}),X=new Yn({props:{title:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",class:"pipen-report-table-image-descr"}}),q=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/fgsea_HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY.png",class:"pipen-report-table-image",width:1e3,height:800}}),W=new Yn({props:{title:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",class:"pipen-report-table-image-descr"}}),Z=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/fgsea_HALLMARK_CHOLESTEROL_HOMEOSTASIS.png",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new Yn({props:{title:"HALLMARK_WNT_BETA_CATENIN_SIGNALING",class:"pipen-report-table-image-descr"}}),se=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/fgsea_HALLMARK_WNT_BETA_CATENIN_SIGNALING.png",class:"pipen-report-table-image",width:1e3,height:800}}),ie=new Yn({props:{title:"HALLMARK_ANDROGEN_RESPONSE",class:"pipen-report-table-image-descr"}}),oe=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/fgsea_HALLMARK_ANDROGEN_RESPONSE.png",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=r("div"),n=r("div"),f(a.$$.fragment),s=S(),f(o.$$.fragment),c=S(),g=r("div"),f(m.$$.fragment),$=S(),f(L.$$.fragment),E=S(),R=r("div"),f(C.$$.fragment),P=S(),f(T.$$.fragment),N=S(),M=r("div"),f(v.$$.fragment),I=S(),f(y.$$.fragment),x=S(),G=r("div"),f(w.$$.fragment),B=S(),f(H.$$.fragment),F=S(),O=r("div"),f(D.$$.fragment),K=S(),f(U.$$.fragment),V=S(),z=r("div"),f(X.$$.fragment),k=S(),f(q.$$.fragment),j=S(),Y=r("div"),f(W.$$.fragment),J=S(),f(Z.$$.fragment),ee=S(),te=r("div"),f(ne.$$.fragment),ae=S(),f(se.$$.fragment),re=S(),le=r("div"),f(ie.$$.fragment),pe=S(),f(oe.$$.fragment),l(t,"class","pipen-report-table-of-images"),Q(t,"grid-template-columns","repeat(2, auto)")},m(e,r){i(e,t,r),b(t,n),A(a,n,null),b(n,s),A(o,n,null),b(t,c),b(t,g),A(m,g,null),b(g,$),A(L,g,null),b(t,E),b(t,R),A(C,R,null),b(R,P),A(T,R,null),b(t,N),b(t,M),A(v,M,null),b(M,I),A(y,M,null),b(t,x),b(t,G),A(w,G,null),b(G,B),A(H,G,null),b(t,F),b(t,O),A(D,O,null),b(O,K),A(U,O,null),b(t,V),b(t,z),A(X,z,null),b(z,k),A(q,z,null),b(t,j),b(t,Y),A(W,Y,null),b(Y,J),A(Z,Y,null),b(t,ee),b(t,te),A(ne,te,null),b(te,ae),A(se,te,null),b(t,re),b(t,le),A(ie,le,null),b(le,pe),A(oe,le,null),ce=!0},p:h,i(e){ce||(p(a.$$.fragment,e),p(o.$$.fragment,e),p(m.$$.fragment,e),p(L.$$.fragment,e),p(C.$$.fragment,e),p(T.$$.fragment,e),p(v.$$.fragment,e),p(y.$$.fragment,e),p(w.$$.fragment,e),p(H.$$.fragment,e),p(D.$$.fragment,e),p(U.$$.fragment,e),p(X.$$.fragment,e),p(q.$$.fragment,e),p(W.$$.fragment,e),p(Z.$$.fragment,e),p(ne.$$.fragment,e),p(se.$$.fragment,e),p(ie.$$.fragment,e),p(oe.$$.fragment,e),ce=!0)},o(e){_(a.$$.fragment,e),_(o.$$.fragment,e),_(m.$$.fragment,e),_(L.$$.fragment,e),_(C.$$.fragment,e),_(T.$$.fragment,e),_(v.$$.fragment,e),_(y.$$.fragment,e),_(w.$$.fragment,e),_(H.$$.fragment,e),_(D.$$.fragment,e),_(U.$$.fragment,e),_(X.$$.fragment,e),_(q.$$.fragment,e),_(W.$$.fragment,e),_(Z.$$.fragment,e),_(ne.$$.fragment,e),_(se.$$.fragment,e),_(ie.$$.fragment,e),_(oe.$$.fragment,e),ce=!1},d(e){e&&d(t),u(a),u(o),u(m),u(L),u(C),u(T),u(v),u(y),u(w),u(H),u(D),u(U),u(X),u(q),u(W),u(Z),u(ne),u(se),u(ie),u(oe)}}}function na(e){let t,n,a,s;return t=new Ie({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[ea]},$$scope:{ctx:e}}}),a=new Ie({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[ta]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=S(),f(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(p(t.$$.fragment,e),p(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),u(t,e),u(a,e)}}}function aa(e){let t,n,a,s;return t=new me({props:{label:"Plot"}}),a=new me({props:{label:"Table"}}),{c(){f(t.$$.fragment),n=S(),f(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p:h,i(e){s||(p(t.$$.fragment,e),p(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),u(t,e),u(a,e)}}}function sa(e){let t,n;return t=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/gsea_table.png",width:1e3,height:760}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p:h,i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function ra(e){let t,n;return t=new Xn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/fgsea.txt",data:[{pathway:"HALLMARK_INTERFERON_ALPHA_RESPONSE",pval:1e-4,padj:8e-4,log2err:.5573,ES:.6595,NES:1.7229,size:96,leadingEdge:"ISG20,SELL,B2M,IFITM2,PSMB9,LY6E,RIPK2,UBE2L6,PSME1,GBP2,PSMB8,PSME2,IFITM3,ISG15,IRF7,CD74,TAP1,GBP4,IFI35,IL7,OASL,IFI27,PARP9,IFI30,LGALS3BP,TMEM140,PARP14,RTP4,IRF1,CASP1,IL15,CMPK2,IRF9,IFI44L,SLC25A28,HERC6,HELZ2,IL4R,PARP12,MOV10,USP18,EPSTI1,LAP3",id:0},{pathway:"HALLMARK_ANGIOGENESIS",pval:.0179,padj:.1256,log2err:.3525,ES:.6647,NES:1.4724,size:36,leadingEdge:"S100A4,TIMP1,PDGFA,FGFR1,STC1,JAG1,VEGFA,COL5A2",id:1},{pathway:"HALLMARK_IL6_JAK_STAT3_SIGNALING",pval:.0967,padj:.4513,log2err:.2878,ES:.4644,NES:1.1935,size:87,leadingEdge:"SOCS1,TNFRSF1B,STAT1,IL18R1,CXCL13,TGFB1,IL7,IL12RB1,IL10RB,IL1R2,IL9R,IRF1,IRF9,IL1R1,CSF2,IL13RA1,PLA2G2A,IL4R,MAP3K8,LEPR,IL15RA,LTB,PIK3R5",id:2},{pathway:"HALLMARK_PROTEIN_SECRETION",pval:.1296,padj:.4536,log2err:.1357,ES:-.5301,NES:-1.1878,size:95,leadingEdge:"KRT18,TMED10,ANP32E,COPB1,COG2,SGMS1,SNX2,RAB14,SCAMP1,ICA1,TMED2,RER1,SCAMP3,GOSR2,STX16,ARFGEF1,M6PR,ARCN1,SEC24D,CLTC,CLCN3,GALC,PPT1,ATP6V1H,ZW10,COPB2,RAB5A,TOM1L1,TSPAN8,GOLGA4,CLTA,SNAP23,AP3B1,GLA,RAB22A,ARFIP1,VAMP3,TSG101,GBF1,SEC31A,CD63,OCRL,RPS6KA3,AP1G1,TMX1,CTSC,PAM,VPS45,ATP7A,SEC22B,MON2,NAPA",id:3},{pathway:"HALLMARK_WNT_BETA_CATENIN_SIGNALING",pval:.2391,padj:.5917,log2err:.0989,ES:-.5646,NES:-1.1458,size:42,leadingEdge:"TCF7,DLL1,CTNNB1,NCOR2,LEF1,NCSTN,NUMB,TP53,RBPJ,PSEN2,CUL1,HDAC2,SKP2,DVL2,NOTCH1,WNT1,HDAC11,PTCH1,NKD1,AXIN2",id:4},{pathway:"HALLMARK_MYC_TARGETS_V2",pval:.2536,padj:.5917,log2err:.0931,ES:-.5347,NES:-1.134,size:58,leadingEdge:"CBX3,TFB2M,PPAN,MCM5,NDUFAF4,SUPV3L1,HSPE1,MPHOSPH10,CDK4,PES1,RRP9,RRP12,MYBBP1A,IMP4,TMEM97,TCOF1,GNL3,PRMT3,WDR43,RABEPK,NOP2,FARSA,SRM,SLC19A1,DDX18,PHB,GRWD1,IPO4,NOC4L,RCL1,UNG,PA2G4,PPRC1,NOP16,MRTO4,MYC,DCTPP1,SLC29A2,HSPD1,MAP3K6,PLK4",id:5},{pathway:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",pval:.3805,padj:.6949,log2err:.1492,ES:.4093,NES:1.0179,size:73,leadingEdge:"S100A11,ANXA5,ATF3,LDLR,SC5D,LGALS3,ETHE1,FDFT1,FADS2,ABCA2,JAG1,PPARG,MAL2,FASN,TP53INP1,ALCAM,STX5,PLAUR,MVD,EBP,ERRFI1,SCD,STARD4,FDPS,ATF5,GNAI1,CYP51A1,PCYT2,CXCL16,HSD17B7",id:6},{pathway:"HALLMARK_ANDROGEN_RESPONSE",pval:.3971,padj:.6949,log2err:.1542,ES:.3884,NES:1.0143,size:99,leadingEdge:"FKBP5,SAT1,B2M,B4GALT1,CCND3,ARID5B,FADS1,RAB4A,IQGAP2,HPGD,INSIG1,MAF,DBI,CDC14B,KRT19,ZMIZ1,LMAN1,MYL12A,DNAJB9,SRF,SGK1,CAMKK2,TMEM50A,SCD,SORD,APPBP2,SRP19,HOMER2,MERTK,NKX3-1,CCND1",id:7},{pathway:"HALLMARK_PANCREAS_BETA_CELLS",pval:.5432,padj:.8353,log2err:.0557,ES:-.4765,NES:-.958,size:40,leadingEdge:"SRP9,ELP4,SRPRB,MAFB,NKX2-2,GCK,SRP14",id:8},{pathway:"HALLMARK_NOTCH_SIGNALING",pval:.6466,padj:.8353,log2err:.0492,ES:-.4605,NES:-.8902,size:32,leadingEdge:"MAML2,DLL1,HES1,SKP1,DTX2,FBXW11,PSEN2,TCF7L2,CUL1,PRKCA",id:9},{pathway:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",pval:.6719,padj:.8353,log2err:.1003,ES:.382,NES:.9017,size:49,leadingEdge:"SOD2,TXN,GLRX,SOD1,SBNO2,CDKN2D,ABCC1,PFKP,SRXN1,HMOX2,LAMTOR5,SELENOS,PRDX6,NDUFB4,GCLC,PRDX4,NQO1,ATOX1,PRDX1",id:10},{pathway:"HALLMARK_APICAL_SURFACE",pval:.7401,padj:.8353,log2err:.0418,ES:-.4207,NES:-.8554,size:44,leadingEdge:"APP,GSTM3,HSPB1,BRCA1,ADIPOR2,GAS1,AFAP1L2,LYPD3,AKAP7,LYN,SULF2,ATP8B1",id:11},{pathway:"HALLMARK_HEDGEHOG_SIGNALING",pval:.7756,padj:.8353,log2err:.0405,ES:-.4209,NES:-.8277,size:36,leadingEdge:"CDK5R1,ETS2,CELSR1,TLE3,NF1,NRCAM,CRMP1",id:12},{pathway:"HALLMARK_TGF_BETA_SIGNALING",pval:.8999,padj:.8999,log2err:.0318,ES:-.3627,NES:-.7593,size:54,leadingEdge:"ID3,JUNB,TGFBR1,KLF10,CTNNB1,NCOR2,FNTA,CDK9,SMAD1,PPM1A,XIAP,SLC20A1,MAP3K7,THBS1",id:13}]}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p:h,i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function la(e){let t,n,a,s;return t=new $e({props:{$$slots:{default:[sa]},$$scope:{ctx:e}}}),a=new $e({props:{$$slots:{default:[ra]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=S(),f(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(p(t.$$.fragment,e),p(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),u(t,e),u(a,e)}}}function ia(e){let t,n;return t=new ge({props:{$$slots:{content:[la],default:[aa]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function pa(e){let t,n,a,s,o,c,g,m,$,L,E,R,C,P,T,N,M,v,I,y,x,G,w,B,H,F,O,D,K,U,V,z,X,k,q,j,Y,W,J,Z,ee,te,ne,ae,se,re,le,ie,pe,oe,ce;return a=new Yn({props:{title:"HALLMARK_INTERFERON_ALPHA_RESPONSE",class:"pipen-report-table-image-descr"}}),o=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/fgsea_HALLMARK_INTERFERON_ALPHA_RESPONSE.png",class:"pipen-report-table-image",width:1e3,height:800}}),m=new Yn({props:{title:"HALLMARK_ANGIOGENESIS",class:"pipen-report-table-image-descr"}}),L=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/fgsea_HALLMARK_ANGIOGENESIS.png",class:"pipen-report-table-image",width:1e3,height:800}}),C=new Yn({props:{title:"HALLMARK_IL6_JAK_STAT3_SIGNALING",class:"pipen-report-table-image-descr"}}),T=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/fgsea_HALLMARK_IL6_JAK_STAT3_SIGNALING.png",class:"pipen-report-table-image",width:1e3,height:800}}),v=new Yn({props:{title:"HALLMARK_PROTEIN_SECRETION",class:"pipen-report-table-image-descr"}}),y=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/fgsea_HALLMARK_PROTEIN_SECRETION.png",class:"pipen-report-table-image",width:1e3,height:800}}),w=new Yn({props:{title:"HALLMARK_WNT_BETA_CATENIN_SIGNALING",class:"pipen-report-table-image-descr"}}),H=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/fgsea_HALLMARK_WNT_BETA_CATENIN_SIGNALING.png",class:"pipen-report-table-image",width:1e3,height:800}}),D=new Yn({props:{title:"HALLMARK_MYC_TARGETS_V2",class:"pipen-report-table-image-descr"}}),U=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/fgsea_HALLMARK_MYC_TARGETS_V2.png",class:"pipen-report-table-image",width:1e3,height:800}}),X=new Yn({props:{title:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",class:"pipen-report-table-image-descr"}}),q=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/fgsea_HALLMARK_CHOLESTEROL_HOMEOSTASIS.png",class:"pipen-report-table-image",width:1e3,height:800}}),W=new Yn({props:{title:"HALLMARK_ANDROGEN_RESPONSE",class:"pipen-report-table-image-descr"}}),Z=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/fgsea_HALLMARK_ANDROGEN_RESPONSE.png",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new Yn({props:{title:"HALLMARK_PANCREAS_BETA_CELLS",class:"pipen-report-table-image-descr"}}),se=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/fgsea_HALLMARK_PANCREAS_BETA_CELLS.png",class:"pipen-report-table-image",width:1e3,height:800}}),ie=new Yn({props:{title:"HALLMARK_NOTCH_SIGNALING",class:"pipen-report-table-image-descr"}}),oe=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/fgsea_HALLMARK_NOTCH_SIGNALING.png",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=r("div"),n=r("div"),f(a.$$.fragment),s=S(),f(o.$$.fragment),c=S(),g=r("div"),f(m.$$.fragment),$=S(),f(L.$$.fragment),E=S(),R=r("div"),f(C.$$.fragment),P=S(),f(T.$$.fragment),N=S(),M=r("div"),f(v.$$.fragment),I=S(),f(y.$$.fragment),x=S(),G=r("div"),f(w.$$.fragment),B=S(),f(H.$$.fragment),F=S(),O=r("div"),f(D.$$.fragment),K=S(),f(U.$$.fragment),V=S(),z=r("div"),f(X.$$.fragment),k=S(),f(q.$$.fragment),j=S(),Y=r("div"),f(W.$$.fragment),J=S(),f(Z.$$.fragment),ee=S(),te=r("div"),f(ne.$$.fragment),ae=S(),f(se.$$.fragment),re=S(),le=r("div"),f(ie.$$.fragment),pe=S(),f(oe.$$.fragment),l(t,"class","pipen-report-table-of-images"),Q(t,"grid-template-columns","repeat(2, auto)")},m(e,r){i(e,t,r),b(t,n),A(a,n,null),b(n,s),A(o,n,null),b(t,c),b(t,g),A(m,g,null),b(g,$),A(L,g,null),b(t,E),b(t,R),A(C,R,null),b(R,P),A(T,R,null),b(t,N),b(t,M),A(v,M,null),b(M,I),A(y,M,null),b(t,x),b(t,G),A(w,G,null),b(G,B),A(H,G,null),b(t,F),b(t,O),A(D,O,null),b(O,K),A(U,O,null),b(t,V),b(t,z),A(X,z,null),b(z,k),A(q,z,null),b(t,j),b(t,Y),A(W,Y,null),b(Y,J),A(Z,Y,null),b(t,ee),b(t,te),A(ne,te,null),b(te,ae),A(se,te,null),b(t,re),b(t,le),A(ie,le,null),b(le,pe),A(oe,le,null),ce=!0},p:h,i(e){ce||(p(a.$$.fragment,e),p(o.$$.fragment,e),p(m.$$.fragment,e),p(L.$$.fragment,e),p(C.$$.fragment,e),p(T.$$.fragment,e),p(v.$$.fragment,e),p(y.$$.fragment,e),p(w.$$.fragment,e),p(H.$$.fragment,e),p(D.$$.fragment,e),p(U.$$.fragment,e),p(X.$$.fragment,e),p(q.$$.fragment,e),p(W.$$.fragment,e),p(Z.$$.fragment,e),p(ne.$$.fragment,e),p(se.$$.fragment,e),p(ie.$$.fragment,e),p(oe.$$.fragment,e),ce=!0)},o(e){_(a.$$.fragment,e),_(o.$$.fragment,e),_(m.$$.fragment,e),_(L.$$.fragment,e),_(C.$$.fragment,e),_(T.$$.fragment,e),_(v.$$.fragment,e),_(y.$$.fragment,e),_(w.$$.fragment,e),_(H.$$.fragment,e),_(D.$$.fragment,e),_(U.$$.fragment,e),_(X.$$.fragment,e),_(q.$$.fragment,e),_(W.$$.fragment,e),_(Z.$$.fragment,e),_(ne.$$.fragment,e),_(se.$$.fragment,e),_(ie.$$.fragment,e),_(oe.$$.fragment,e),ce=!1},d(e){e&&d(t),u(a),u(o),u(m),u(L),u(C),u(T),u(v),u(y),u(w),u(H),u(D),u(U),u(X),u(q),u(W),u(Z),u(ne),u(se),u(ie),u(oe)}}}function oa(e){let t,n,a,s;return t=new Ie({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[ia]},$$scope:{ctx:e}}}),a=new Ie({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[pa]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=S(),f(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(p(t.$$.fragment,e),p(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),u(t,e),u(a,e)}}}function ca(e){let t,n,a,s;return t=new me({props:{label:"Plot"}}),a=new me({props:{label:"Table"}}),{c(){f(t.$$.fragment),n=S(),f(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p:h,i(e){s||(p(t.$$.fragment,e),p(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),u(t,e),u(a,e)}}}function _a(e){let t,n;return t=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/gsea_table.png",width:1e3,height:760}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p:h,i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function da(e){let t,n;return t=new Xn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/fgsea.txt",data:[{pathway:"HALLMARK_INTERFERON_ALPHA_RESPONSE",pval:.008,padj:.1125,log2err:.3807,ES:.5734,NES:1.4385,size:96,leadingEdge:"ISG20,CSF1,GBP2,IFITM3,B2M,TAP1,PSMB9,IRF1,TMEM140,UBE2L6,RIPK2,ELF1,PLSCR1,CMPK2,IRF7,CCRL2,TRIM25,LY6E,CASP8,IL4R,PARP9,LAMP3,EPSTI1,IFIT3,HERC6,SLC25A28,CMTR1,LPAR6,HELZ2,SP110,NCOA7,PSMB8,BST2,IFITM2",id:0},{pathway:"HALLMARK_ANGIOGENESIS",pval:.0344,padj:.2405,log2err:.3218,ES:-.7476,NES:-1.4124,size:36,leadingEdge:"S100A4,TIMP1",id:1},{pathway:"HALLMARK_IL6_JAK_STAT3_SIGNALING",pval:.0701,padj:.3273,log2err:.2878,ES:.4833,NES:1.1953,size:87,leadingEdge:"CSF1,PIM1,SOCS3,TNFRSF1B,CD38,MAP3K8,IRF1,STAT3,STAT1,IL6ST,IL17RA,HMOX1,IL4R,LTBR,PTPN11",id:2},{pathway:"HALLMARK_ANDROGEN_RESPONSE",pval:.194,padj:.66,log2err:.228,ES:.4361,NES:1.0999,size:99,leadingEdge:"FKBP5,ARID5B,PLPP1,B2M,SAT1,HPGD,SGK1,STK39,SCD,ZBTB10,INSIG1,MAF,KRT8,ELK4,ZMIZ1,DHCR24,SRF,SPCS3,IQGAP2,CCND3,SEC24D,SLC38A2,PTK2B,VAPA,INPP4B,CDK6,HMGCR,GSR",id:3},{pathway:"HALLMARK_NOTCH_SIGNALING",pval:.3137,padj:.66,log2err:.085,ES:-.5943,NES:-1.103,size:32,leadingEdge:"DLL1,RBX1,PRKCA,HES1,CUL1,PSENEN,FZD7,KAT2A,SKP1,MAML2,APH1A,PSEN2,ARRB1,NOTCH2",id:4},{pathway:"HALLMARK_WNT_BETA_CATENIN_SIGNALING",pval:.316,padj:.66,log2err:.0834,ES:-.5625,NES:-1.0944,size:42,leadingEdge:"DLL1,TCF7,LEF1,HDAC2,HEY1,CUL1,AXIN1,NKD1,FZD8,KAT2A,PSEN2,DVL2,MAML1,NCSTN,TP53,HDAC5",id:5},{pathway:"HALLMARK_APICAL_SURFACE",pval:.3833,padj:.66,log2err:.0725,ES:-.5357,NES:-1.0487,size:44,leadingEdge:"CD160,FLOT2,LYPD3,HSPB1,ADIPOR2,GSTM3,IL2RG,PLAUR,SCUBE1",id:6},{pathway:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",pval:.3903,padj:.66,log2err:.0715,ES:-.5274,NES:-1.0473,size:49,leadingEdge:"FTL,PRNP,PRDX1,JUNB,CAT,CDKN2D,HHEX,OXSR1,GPX4,PTPA,EGLN2,TXNRD1,PDLIM1,SOD2,G6PD,TXN,GLRX,PRDX4,GCLM,GCLC",id:7},{pathway:"HALLMARK_PROTEIN_SECRETION",pval:.4666,padj:.66,log2err:.0596,ES:-.4658,NES:-.9997,size:95,leadingEdge:"ABCA1,ATP1A1,CD63,GLA,PPT1,TMED2,SEC31A,LMAN1,TSG101,RAB14,GOSR2,BNIP3,RAB22A,AP2B1,COPE,KIF1B,ATP7A,AP2M1,TMED10,SCAMP3,VPS45,MAPK1,M6PR,SNX2,OCRL,ARCN1,COPB2,STX16,SGMS1,ARFIP1,DNM1L,COG2,GOLGA4,TPD52,SEC22B,PAM",id:8},{pathway:"HALLMARK_HEDGEHOG_SIGNALING",pval:.4714,padj:.66,log2err:.1172,ES:.4516,NES:.9655,size:36,leadingEdge:"TLE1,ADGRG1,ACHE,MYH9,DPYSL2,ETS2,CDK6,PTCH1",id:9},{pathway:"HALLMARK_MYC_TARGETS_V2",pval:.7243,padj:.9015,log2err:.0407,ES:-.4268,NES:-.869,size:58,leadingEdge:"PA2G4,MAP3K6,PES1,MYBBP1A,GRWD1,PLK4,CBX3,NIP7,LAS1L,MCM5,RRP12,SLC19A1,MRTO4,TMEM97,RABEPK,TCOF1,TBRG4,DUSP2,NOLC1,MPHOSPH10,MCM4,AIMP2,NOP2,FARSA,SRM,SUPV3L1,IMP4",id:10},{pathway:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",pval:.7727,padj:.9015,log2err:.101,ES:.3732,NES:.8894,size:73,leadingEdge:"TP53INP1,ATF3,LDLR,STX5,PLSCR1,SCD,GSTM2,FASN,NFIL3,PNRC1,FADS2,STARD4,FABP5,PDK3,ETHE1,SREBF2,GUSB,FDFT1,HMGCR,NSDHL,ALCAM,CTNNB1",id:11},{pathway:"HALLMARK_PANCREAS_BETA_CELLS",pval:.8432,padj:.9081,log2err:.0355,ES:-.4081,NES:-.7879,size:40,leadingEdge:"STXBP1,AKT3,NEUROG3,ELP4",id:12},{pathway:"HALLMARK_TGF_BETA_SIGNALING",pval:.9348,padj:.9348,log2err:.0298,ES:-.3713,NES:-.7461,size:54,leadingEdge:"JUNB,HDAC1,PPP1CA,WWTR1,ENG,KLF10,APC,TGFBR1,BMPR2,SMAD3,FNTA,NOG,BMP2,IFNGR2,BCAR3,CDK9,SERPINE1,SMAD7,MAP3K7,SMURF2,RHOA,SPTBN1,SMURF1,SKIL",id:13}]}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p:h,i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function ga(e){let t,n,a,s;return t=new $e({props:{$$slots:{default:[_a]},$$scope:{ctx:e}}}),a=new $e({props:{$$slots:{default:[da]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=S(),f(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(p(t.$$.fragment,e),p(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),u(t,e),u(a,e)}}}function ma(e){let t,n;return t=new ge({props:{$$slots:{content:[ga],default:[ca]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function $a(e){let t,n,a,s,o,c,g,m,$,L,E,R,C,P,T,N,M,v,I,y,x,G,w,B,H,F,O,D,K,U,V,z,X,k,q,j,Y,W,J,Z,ee,te,ne,ae,se,re,le,ie,pe,oe,ce;return a=new Yn({props:{title:"HALLMARK_INTERFERON_ALPHA_RESPONSE",class:"pipen-report-table-image-descr"}}),o=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/fgsea_HALLMARK_INTERFERON_ALPHA_RESPONSE.png",class:"pipen-report-table-image",width:1e3,height:800}}),m=new Yn({props:{title:"HALLMARK_ANGIOGENESIS",class:"pipen-report-table-image-descr"}}),L=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/fgsea_HALLMARK_ANGIOGENESIS.png",class:"pipen-report-table-image",width:1e3,height:800}}),C=new Yn({props:{title:"HALLMARK_IL6_JAK_STAT3_SIGNALING",class:"pipen-report-table-image-descr"}}),T=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/fgsea_HALLMARK_IL6_JAK_STAT3_SIGNALING.png",class:"pipen-report-table-image",width:1e3,height:800}}),v=new Yn({props:{title:"HALLMARK_ANDROGEN_RESPONSE",class:"pipen-report-table-image-descr"}}),y=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/fgsea_HALLMARK_ANDROGEN_RESPONSE.png",class:"pipen-report-table-image",width:1e3,height:800}}),w=new Yn({props:{title:"HALLMARK_NOTCH_SIGNALING",class:"pipen-report-table-image-descr"}}),H=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/fgsea_HALLMARK_NOTCH_SIGNALING.png",class:"pipen-report-table-image",width:1e3,height:800}}),D=new Yn({props:{title:"HALLMARK_WNT_BETA_CATENIN_SIGNALING",class:"pipen-report-table-image-descr"}}),U=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/fgsea_HALLMARK_WNT_BETA_CATENIN_SIGNALING.png",class:"pipen-report-table-image",width:1e3,height:800}}),X=new Yn({props:{title:"HALLMARK_APICAL_SURFACE",class:"pipen-report-table-image-descr"}}),q=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/fgsea_HALLMARK_APICAL_SURFACE.png",class:"pipen-report-table-image",width:1e3,height:800}}),W=new Yn({props:{title:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",class:"pipen-report-table-image-descr"}}),Z=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/fgsea_HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY.png",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new Yn({props:{title:"HALLMARK_PROTEIN_SECRETION",class:"pipen-report-table-image-descr"}}),se=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/fgsea_HALLMARK_PROTEIN_SECRETION.png",class:"pipen-report-table-image",width:1e3,height:800}}),ie=new Yn({props:{title:"HALLMARK_HEDGEHOG_SIGNALING",class:"pipen-report-table-image-descr"}}),oe=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/fgsea_HALLMARK_HEDGEHOG_SIGNALING.png",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=r("div"),n=r("div"),f(a.$$.fragment),s=S(),f(o.$$.fragment),c=S(),g=r("div"),f(m.$$.fragment),$=S(),f(L.$$.fragment),E=S(),R=r("div"),f(C.$$.fragment),P=S(),f(T.$$.fragment),N=S(),M=r("div"),f(v.$$.fragment),I=S(),f(y.$$.fragment),x=S(),G=r("div"),f(w.$$.fragment),B=S(),f(H.$$.fragment),F=S(),O=r("div"),f(D.$$.fragment),K=S(),f(U.$$.fragment),V=S(),z=r("div"),f(X.$$.fragment),k=S(),f(q.$$.fragment),j=S(),Y=r("div"),f(W.$$.fragment),J=S(),f(Z.$$.fragment),ee=S(),te=r("div"),f(ne.$$.fragment),ae=S(),f(se.$$.fragment),re=S(),le=r("div"),f(ie.$$.fragment),pe=S(),f(oe.$$.fragment),l(t,"class","pipen-report-table-of-images"),Q(t,"grid-template-columns","repeat(2, auto)")},m(e,r){i(e,t,r),b(t,n),A(a,n,null),b(n,s),A(o,n,null),b(t,c),b(t,g),A(m,g,null),b(g,$),A(L,g,null),b(t,E),b(t,R),A(C,R,null),b(R,P),A(T,R,null),b(t,N),b(t,M),A(v,M,null),b(M,I),A(y,M,null),b(t,x),b(t,G),A(w,G,null),b(G,B),A(H,G,null),b(t,F),b(t,O),A(D,O,null),b(O,K),A(U,O,null),b(t,V),b(t,z),A(X,z,null),b(z,k),A(q,z,null),b(t,j),b(t,Y),A(W,Y,null),b(Y,J),A(Z,Y,null),b(t,ee),b(t,te),A(ne,te,null),b(te,ae),A(se,te,null),b(t,re),b(t,le),A(ie,le,null),b(le,pe),A(oe,le,null),ce=!0},p:h,i(e){ce||(p(a.$$.fragment,e),p(o.$$.fragment,e),p(m.$$.fragment,e),p(L.$$.fragment,e),p(C.$$.fragment,e),p(T.$$.fragment,e),p(v.$$.fragment,e),p(y.$$.fragment,e),p(w.$$.fragment,e),p(H.$$.fragment,e),p(D.$$.fragment,e),p(U.$$.fragment,e),p(X.$$.fragment,e),p(q.$$.fragment,e),p(W.$$.fragment,e),p(Z.$$.fragment,e),p(ne.$$.fragment,e),p(se.$$.fragment,e),p(ie.$$.fragment,e),p(oe.$$.fragment,e),ce=!0)},o(e){_(a.$$.fragment,e),_(o.$$.fragment,e),_(m.$$.fragment,e),_(L.$$.fragment,e),_(C.$$.fragment,e),_(T.$$.fragment,e),_(v.$$.fragment,e),_(y.$$.fragment,e),_(w.$$.fragment,e),_(H.$$.fragment,e),_(D.$$.fragment,e),_(U.$$.fragment,e),_(X.$$.fragment,e),_(q.$$.fragment,e),_(W.$$.fragment,e),_(Z.$$.fragment,e),_(ne.$$.fragment,e),_(se.$$.fragment,e),_(ie.$$.fragment,e),_(oe.$$.fragment,e),ce=!1},d(e){e&&d(t),u(a),u(o),u(m),u(L),u(C),u(T),u(v),u(y),u(w),u(H),u(D),u(U),u(X),u(q),u(W),u(Z),u(ne),u(se),u(ie),u(oe)}}}function fa(e){let t,n,a,s;return t=new Ie({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[ma]},$$scope:{ctx:e}}}),a=new Ie({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[$a]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=S(),f(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(p(t.$$.fragment,e),p(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),u(t,e),u(a,e)}}}function Aa(e){let t,n,a,s;return t=new me({props:{label:"Plot"}}),a=new me({props:{label:"Table"}}),{c(){f(t.$$.fragment),n=S(),f(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p:h,i(e){s||(p(t.$$.fragment,e),p(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),u(t,e),u(a,e)}}}function ua(e){let t,n;return t=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/gsea_table.png",width:1e3,height:760}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p:h,i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function Sa(e){let t,n;return t=new Xn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/fgsea.txt",data:[{pathway:"HALLMARK_INTERFERON_ALPHA_RESPONSE",pval:0,padj:2e-4,log2err:.5933,ES:.6907,NES:1.7933,size:96,leadingEdge:"PSMB9,PSMB8,PSME1,IFITM2,UBE2L6,LY6E,GBP2,ISG20,CD74,PSME2,USP18,OASL,B2M,GBP4,TAP1,LAP3,IRF7,SLC25A28,PSMA3,CASP8,UBA7,IFI35,MX1,IRF2,MOV10,RSAD2,PLSCR1,BST2,TMEM140,SP110,IL15,SELL,PARP9,IRF1,OAS1,IFITM1,IL7,TRIM25,IFI27,DHX58,HELZ2,IFITM3,ADAR,NMI,IFI30,NCOA7,TENT5A",id:0},{pathway:"HALLMARK_APICAL_SURFACE",pval:.0628,padj:.4395,log2err:.2878,ES:-.6043,NES:-1.3526,size:44,leadingEdge:"LYN,HSPB1,BRCA1,GSTM3,FLOT2,CRYBG1,CROCC,NCOA6,ADAM10,PLAUR,DCBLD2,LYPD3,TMEM8B,AFAP1L2,MDGA1,MAL,AKAP7,CD160",id:1},{pathway:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",pval:.1343,padj:.504,log2err:.1882,ES:.5135,NES:1.2095,size:49,leadingEdge:"FTL,SOD1,JUNB,SELENOS,NDUFA6,TXN,GLRX2,NDUFB4,PRDX6,PFKP,GLRX,PRDX2,G6PD,PRDX1,HMOX2,LAMTOR5",id:2},{pathway:"HALLMARK_PROTEIN_SECRETION",pval:.144,padj:.504,log2err:.1464,ES:-.4729,NES:-1.179,size:95,leadingEdge:"DNM1L,SCAMP1,ZW10,SGMS1,ANP32E,VAMP3,GNAS,AP3S1,SNX2,ARFGAP3,SH3GL2,ERGIC3,ADAM10,VPS45,CD63,ATP1A1,VAMP7,TPD52,ARFIP1,BNIP3,ARFGEF2,LMAN1,ARFGEF1,STAM,IGF2R,MON2,GALC,MAPK1,ATP6V1H,SCAMP3,SEC24D,ATP7A,COG2,CLN5,AP1G1,STX12,SEC22B,COPB2,LAMP2,VPS4B,YIPF6,COPB1,ARCN1,TMED10,RER1,RAB22A,USO1,KIF1B,CLTC,CLTA,STX16",id:3},{pathway:"HALLMARK_PANCREAS_BETA_CELLS",pval:.2409,padj:.6745,log2err:.138,ES:.5088,NES:1.1428,size:40,leadingEdge:"DPP4,SRP14,SPCS1,FOXO1,PAK3,GCK,VDR,SRPRB",id:4},{pathway:"HALLMARK_MYC_TARGETS_V2",pval:.3493,padj:.7682,log2err:.1106,ES:.4278,NES:1.0361,size:58,leadingEdge:"DUSP2,NOP16,SLC19A1,IPO4,WDR43,HK2,NOP56,PPRC1,WDR74,GNL3,PUS1,NPM1,SORD,HSPD1,PHB,HSPE1,PA2G4,DDX18,IMP4,SLC29A2,SRM",id:5},{pathway:"HALLMARK_TGF_BETA_SIGNALING",pval:.4777,padj:.7682,log2err:.0736,ES:-.4316,NES:-.9881,size:54,leadingEdge:"ENG,ID3,ID1,CTNNB1,WWTR1,MAP3K7,IFNGR2,SLC20A1,PPP1R15A,HDAC1,SMURF2,RHOA,TRIM33,APC,BMPR2,TGFBR1,SKIL,SPTBN1,FNTA,PMEPA1,SMURF1,NCOR2,PPM1A,KLF10,HIPK2,PPP1CA,ARID4B",id:6},{pathway:"HALLMARK_ANDROGEN_RESPONSE",pval:.5355,padj:.7682,log2err:.0648,ES:-.3855,NES:-.9622,size:99,leadingEdge:"GNAI3,RRP12,ANKH,APPBP2,TSC22D1,ITGAV,ACTN1,HMGCS1,CDK6,NCOA4,ABCC4,VAPA,ELK4,SMS,NGLY1,PLPP1,ACSL3,PTK2B,IDI1,KRT8,TPD52,NDRG1,LMAN1,MAF,TNFAIP8,HMGCR,XRCC6,AKT1,CAMKK2,CENPN,SLC38A2,SEC24D,XRCC5,ZBTB10,PMEPA1,PIAS1,SLC26A2,SGK1",id:7},{pathway:"HALLMARK_IL6_JAK_STAT3_SIGNALING",pval:.5855,padj:.7682,log2err:.0605,ES:-.3799,NES:-.9419,size:87,leadingEdge:"TNF,PIM1,IL17RA,FAS,JUN,IFNGR2,CRLF2,LTBR,ACVR1B,TLR2,IL4R,CD9,IL1R2,IFNAR1,IL1R1,CBL,EBI3,IL2RA,TNFRSF12A,GRB2,IL10RB,STAM2,IL9R,STAT2,MAP3K8,IL6ST,IRF9",id:8},{pathway:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",pval:.6122,padj:.7682,log2err:.0813,ES:.3705,NES:.9217,size:73,leadingEdge:"FABP5,FASN,FBXO6,NFIL3,STARD4,GLDC,PLSCR1,ATF5,ERRFI1,ANXA5,STX5,CHKA,MVD,FADS2,JAG1,ACTG1,CPEB2",id:9},{pathway:"HALLMARK_ANGIOGENESIS",pval:.6326,padj:.7682,log2err:.0602,ES:-.4199,NES:-.9037,size:36,leadingEdge:"TIMP1,ITGAV,VAV2,CCND2",id:10},{pathway:"HALLMARK_HEDGEHOG_SIGNALING",pval:.6742,padj:.7682,log2err:.0572,ES:-.4133,NES:-.8895,size:36,leadingEdge:"DPYSL2,L1CAM,CDK6,ADGRG1",id:11},{pathway:"HALLMARK_NOTCH_SIGNALING",pval:.7133,padj:.7682,log2err:.0545,ES:-.4057,NES:-.8526,size:32,leadingEdge:"HES1,TCF7L2,PSENEN,DTX2,KAT2A,NOTCH2,LFNG",id:12},{pathway:"HALLMARK_WNT_BETA_CATENIN_SIGNALING",pval:.7896,padj:.7896,log2err:.0486,ES:-.3682,NES:-.823,size:42,leadingEdge:"HDAC5,CTNNB1,HDAC11,SKP2,TCF7,CCND2,KAT2A",id:13}]}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p:h,i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function La(e){let t,n,a,s;return t=new $e({props:{$$slots:{default:[ua]},$$scope:{ctx:e}}}),a=new $e({props:{$$slots:{default:[Sa]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=S(),f(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(p(t.$$.fragment,e),p(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),u(t,e),u(a,e)}}}function Ea(e){let t,n;return t=new ge({props:{$$slots:{content:[La],default:[Aa]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function Ra(e){let t,n,a,s,o,c,g,m,$,L,E,R,C,P,T,N,M,v,I,y,x,G,w,B,H,F,O,D,K,U,V,z,X,k,q,j,Y,W,J,Z,ee,te,ne,ae,se,re,le,ie,pe,oe,ce;return a=new Yn({props:{title:"HALLMARK_INTERFERON_ALPHA_RESPONSE",class:"pipen-report-table-image-descr"}}),o=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/fgsea_HALLMARK_INTERFERON_ALPHA_RESPONSE.png",class:"pipen-report-table-image",width:1e3,height:800}}),m=new Yn({props:{title:"HALLMARK_APICAL_SURFACE",class:"pipen-report-table-image-descr"}}),L=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/fgsea_HALLMARK_APICAL_SURFACE.png",class:"pipen-report-table-image",width:1e3,height:800}}),C=new Yn({props:{title:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",class:"pipen-report-table-image-descr"}}),T=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/fgsea_HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY.png",class:"pipen-report-table-image",width:1e3,height:800}}),v=new Yn({props:{title:"HALLMARK_PROTEIN_SECRETION",class:"pipen-report-table-image-descr"}}),y=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/fgsea_HALLMARK_PROTEIN_SECRETION.png",class:"pipen-report-table-image",width:1e3,height:800}}),w=new Yn({props:{title:"HALLMARK_PANCREAS_BETA_CELLS",class:"pipen-report-table-image-descr"}}),H=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/fgsea_HALLMARK_PANCREAS_BETA_CELLS.png",class:"pipen-report-table-image",width:1e3,height:800}}),D=new Yn({props:{title:"HALLMARK_MYC_TARGETS_V2",class:"pipen-report-table-image-descr"}}),U=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/fgsea_HALLMARK_MYC_TARGETS_V2.png",class:"pipen-report-table-image",width:1e3,height:800}}),X=new Yn({props:{title:"HALLMARK_TGF_BETA_SIGNALING",class:"pipen-report-table-image-descr"}}),q=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/fgsea_HALLMARK_TGF_BETA_SIGNALING.png",class:"pipen-report-table-image",width:1e3,height:800}}),W=new Yn({props:{title:"HALLMARK_ANDROGEN_RESPONSE",class:"pipen-report-table-image-descr"}}),Z=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/fgsea_HALLMARK_ANDROGEN_RESPONSE.png",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new Yn({props:{title:"HALLMARK_IL6_JAK_STAT3_SIGNALING",class:"pipen-report-table-image-descr"}}),se=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/fgsea_HALLMARK_IL6_JAK_STAT3_SIGNALING.png",class:"pipen-report-table-image",width:1e3,height:800}}),ie=new Yn({props:{title:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",class:"pipen-report-table-image-descr"}}),oe=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/fgsea_HALLMARK_CHOLESTEROL_HOMEOSTASIS.png",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=r("div"),n=r("div"),f(a.$$.fragment),s=S(),f(o.$$.fragment),c=S(),g=r("div"),f(m.$$.fragment),$=S(),f(L.$$.fragment),E=S(),R=r("div"),f(C.$$.fragment),P=S(),f(T.$$.fragment),N=S(),M=r("div"),f(v.$$.fragment),I=S(),f(y.$$.fragment),x=S(),G=r("div"),f(w.$$.fragment),B=S(),f(H.$$.fragment),F=S(),O=r("div"),f(D.$$.fragment),K=S(),f(U.$$.fragment),V=S(),z=r("div"),f(X.$$.fragment),k=S(),f(q.$$.fragment),j=S(),Y=r("div"),f(W.$$.fragment),J=S(),f(Z.$$.fragment),ee=S(),te=r("div"),f(ne.$$.fragment),ae=S(),f(se.$$.fragment),re=S(),le=r("div"),f(ie.$$.fragment),pe=S(),f(oe.$$.fragment),l(t,"class","pipen-report-table-of-images"),Q(t,"grid-template-columns","repeat(2, auto)")},m(e,r){i(e,t,r),b(t,n),A(a,n,null),b(n,s),A(o,n,null),b(t,c),b(t,g),A(m,g,null),b(g,$),A(L,g,null),b(t,E),b(t,R),A(C,R,null),b(R,P),A(T,R,null),b(t,N),b(t,M),A(v,M,null),b(M,I),A(y,M,null),b(t,x),b(t,G),A(w,G,null),b(G,B),A(H,G,null),b(t,F),b(t,O),A(D,O,null),b(O,K),A(U,O,null),b(t,V),b(t,z),A(X,z,null),b(z,k),A(q,z,null),b(t,j),b(t,Y),A(W,Y,null),b(Y,J),A(Z,Y,null),b(t,ee),b(t,te),A(ne,te,null),b(te,ae),A(se,te,null),b(t,re),b(t,le),A(ie,le,null),b(le,pe),A(oe,le,null),ce=!0},p:h,i(e){ce||(p(a.$$.fragment,e),p(o.$$.fragment,e),p(m.$$.fragment,e),p(L.$$.fragment,e),p(C.$$.fragment,e),p(T.$$.fragment,e),p(v.$$.fragment,e),p(y.$$.fragment,e),p(w.$$.fragment,e),p(H.$$.fragment,e),p(D.$$.fragment,e),p(U.$$.fragment,e),p(X.$$.fragment,e),p(q.$$.fragment,e),p(W.$$.fragment,e),p(Z.$$.fragment,e),p(ne.$$.fragment,e),p(se.$$.fragment,e),p(ie.$$.fragment,e),p(oe.$$.fragment,e),ce=!0)},o(e){_(a.$$.fragment,e),_(o.$$.fragment,e),_(m.$$.fragment,e),_(L.$$.fragment,e),_(C.$$.fragment,e),_(T.$$.fragment,e),_(v.$$.fragment,e),_(y.$$.fragment,e),_(w.$$.fragment,e),_(H.$$.fragment,e),_(D.$$.fragment,e),_(U.$$.fragment,e),_(X.$$.fragment,e),_(q.$$.fragment,e),_(W.$$.fragment,e),_(Z.$$.fragment,e),_(ne.$$.fragment,e),_(se.$$.fragment,e),_(ie.$$.fragment,e),_(oe.$$.fragment,e),ce=!1},d(e){e&&d(t),u(a),u(o),u(m),u(L),u(C),u(T),u(v),u(y),u(w),u(H),u(D),u(U),u(X),u(q),u(W),u(Z),u(ne),u(se),u(ie),u(oe)}}}function Ca(e){let t,n,a,s;return t=new Ie({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[Ea]},$$scope:{ctx:e}}}),a=new Ie({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[Ra]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=S(),f(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(p(t.$$.fragment,e),p(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),u(t,e),u(a,e)}}}function ba(e){let t,n,a,s;return t=new me({props:{label:"Plot"}}),a=new me({props:{label:"Table"}}),{c(){f(t.$$.fragment),n=S(),f(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p:h,i(e){s||(p(t.$$.fragment,e),p(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),u(t,e),u(a,e)}}}function Pa(e){let t,n;return t=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/gsea_table.png",width:1e3,height:760}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p:h,i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function Ta(e){let t,n;return t=new Xn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/fgsea.txt",data:[{pathway:"HALLMARK_INTERFERON_ALPHA_RESPONSE",pval:0,padj:0,log2err:.705,ES:.8055,NES:1.8307,size:96,leadingEdge:"ISG20,PSMB9,IFITM2,TAP1,UBE2L6,B2M,CD74,LY6E,PSME2,LGALS3BP,GBP4,IRF1,PSMB8,IFITM1,IFI35,PSME1,IFITM3,OASL,SAMD9L,ISG15,PARP9,NMI,IRF7,CMPK2,HELZ2,SELL,BST2,PLSCR1,IFI30,TMEM140,IL15,GBP2,CNP,IFIT3,OAS1,RTP4,IFI44L,USP18,EIF2AK2,ADAR,MOV10,UBA7,LAMP3,RSAD2,IFI27,PARP14,IL4R,CXCL10,SAMD9,ELF1,LAP3,EPSTI1,DHX58,CMTR1,CD47,HERC6,TRIM5,PSMA3,PARP12,IRF2,SLC25A28",id:0},{pathway:"HALLMARK_IL6_JAK_STAT3_SIGNALING",pval:.0141,padj:.0985,log2err:.3807,ES:.6159,NES:1.3854,size:87,leadingEdge:"MAP3K8,STAT1,CD38,CXCL13,IRF1,SOCS1,PIM1,HMOX1,CSF2,IL18R1,SOCS3,TGFB1,IL12RB1,BAK1,IL2RG,CCR1,IL1R2,STAT3,ITGA4,EBI3,PIK3R5,IL4R,CXCL10,TNFRSF1B,GRB2,IL2RA,IL15RA,CXCL9,IL3RA,STAT2,TYK2,IL7,FAS,CBL",id:1},{pathway:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",pval:.0483,padj:.2252,log2err:.2378,ES:.6069,NES:1.345,size:73,leadingEdge:"TP53INP1,S100A11,GLDC,FASN,PLSCR1,FBXO6,MVD,LDLR,FADS2,ETHE1,CXCL16,STX5,FDFT1,FABP5,ANXA5,STARD4,FDPS,SC5D,MVK,ABCA2,PLAUR,GNAI1,CLU,ATF3,GSTM2,SREBF2,CYP51A1,CPEB2,ATF5,HMGCS1,JAG1,PPARG,LGALS3,ECH1,DHCR7,ALCAM,SCD,PMVK,IDI1,TMEM97",id:2},{pathway:"HALLMARK_MYC_TARGETS_V2",pval:.0852,padj:.2982,log2err:.1798,ES:.597,NES:1.2785,size:58,leadingEdge:"DUSP2,HSPE1,HSPD1,SLC19A1,WDR74,HK2,AIMP2,RABEPK,DCTPP1,NOC4L,RCL1,WDR43,CBX3,PA2G4,NOP16,GRWD1,EXOSC5,NOP56,NPM1,MRTO4,PHB,PUS1,NOP2,NIP7,GNL3,TMEM97,PES1,NOLC1,SRM,TFB2M,UTP20",id:3},{pathway:"HALLMARK_ANDROGEN_RESPONSE",pval:.175,padj:.49,log2err:.1167,ES:.5087,NES:1.1588,size:99,leadingEdge:"SAT1,B2M,FKBP5,ARID5B,LMAN1,CCND3,PLPP1,SRF,RPS6KA3,INSIG1,DHCR24,ABHD2,ADRM1,ELOVL5,MAF,CDK6,ELL2,SPCS3,PA2G4,HMGCS1,NKX3-1,KLK2,MAK,UBE2I,TMEM50A,HOMER2,B4GALT1,FADS1,DBI,VAPA,SCD,SEC24D,SRP19,IDI1,UAP1,UBE2J1,ELK4,MERTK,HPGD,CCND1,PIAS1,NCOA4,SMS,ZBTB10,GPD1L,HMGCR,MYL12A,PGM3",id:4},{pathway:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",pval:.2371,padj:.4971,log2err:.1024,ES:.5427,NES:1.1389,size:49,leadingEdge:"SOD1,TXN,LAMTOR5,SELENOS,MBP,NDUFA6,NDUFB4,ABCC1,PRDX6,LSP1,HMOX2,PRDX2,SBNO2,NDUFS2,ATOX1,GLRX,EGLN2,TXNRD1",id:5},{pathway:"HALLMARK_PANCREAS_BETA_CELLS",pval:.2485,padj:.4971,log2err:.1006,ES:.5596,NES:1.1417,size:40,leadingEdge:"VDR,SEC11A,SRP14,SPCS1,SRPRB,LMO2,DPP4,SRP9",id:6},{pathway:"HALLMARK_ANGIOGENESIS",pval:.3075,padj:.538,log2err:.138,ES:-.4824,NES:-1.0653,size:36,leadingEdge:"ITGAV,CCND2,PTK2,APOH,JAG2,NRP1,MSX1",id:7},{pathway:"HALLMARK_APICAL_SURFACE",pval:.5276,padj:.8102,log2err:.1003,ES:-.4164,NES:-.967,size:44,leadingEdge:"LYPD3,PKHD1,HSPB1,AKAP7,TMEM8B,FLOT2,ADAM10,CROCC,BRCA1,GSTM3,LYN",id:8},{pathway:"HALLMARK_TGF_BETA_SIGNALING",pval:.5787,padj:.8102,log2err:.055,ES:.4444,NES:.9398,size:54,leadingEdge:"FKBP1A,FURIN,ID2,TGIF1,UBE2D3,ARID4B,TGFB1,PPP1CA,HIPK2,SMAD3,CDH1,THBS1,ACVR1,WWTR1,RHOA,SPTBN1,SKIL,ID1,TRIM33,SMAD7,PPM1A,SMURF1,BMPR1A,SLC20A1,ENG,XIAP",id:9},{pathway:"HALLMARK_PROTEIN_SECRETION",pval:.708,padj:.8777,log2err:.0419,ES:.3935,NES:.8929,size:95,leadingEdge:"SOD1,LMAN1,AP2S1,RPS6KA3,RAB9A,YKT6,RER1,COPE,IGF2R,VAMP4,VAMP7,GNAS,BET1,GOSR2,AP2M1,ARFGAP3,ARFGEF1,NAPG,GALC,DNM1L,ARFGEF2,KIF1B,NAPA,ERGIC3,SEC22B,SCAMP1,STX7,SEC24D,SCAMP3,YIPF6,BNIP3,USO1,ARFIP1,ARF1,CLTA,ANP32E,TMX1,PAM,VPS4B,SNAP23,ARCN1,GBF1,CLTC",id:10},{pathway:"HALLMARK_NOTCH_SIGNALING",pval:.7523,padj:.8777,log2err:.0806,ES:-.3982,NES:-.8445,size:32,leadingEdge:"SKP1,MAML2,FBXW11,HES1,ST3GAL6,DTX4,HEYL,APH1A,DTX2,ARRB1,PSENEN",id:11},{pathway:"HALLMARK_HEDGEHOG_SIGNALING",pval:.9658,padj:.9807,log2err:.068,ES:-.3416,NES:-.7543,size:36,leadingEdge:"VLDLR,CRMP1,CELSR1,OPHN1,NRP1,CDK5R1,NF1,HEY1",id:12},{pathway:"HALLMARK_WNT_BETA_CATENIN_SIGNALING",pval:.9807,padj:.9807,log2err:.069,ES:-.326,NES:-.7417,size:42,leadingEdge:"NKD1,NCOR2,CTNNB1,CCND2,WNT1",id:13}]}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p:h,i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function ha(e){let t,n,a,s;return t=new $e({props:{$$slots:{default:[Pa]},$$scope:{ctx:e}}}),a=new $e({props:{$$slots:{default:[Ta]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=S(),f(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(p(t.$$.fragment,e),p(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),u(t,e),u(a,e)}}}function Na(e){let t,n;return t=new ge({props:{$$slots:{content:[ha],default:[ba]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function Ma(e){let t,n,a,s,o,c,g,m,$,L,E,R,C,P,T,N,M,v,I,y,x,G,w,B,H,F,O,D,K,U,V,z,X,k,q,j,Y,W,J,Z,ee,te,ne,ae,se,re,le,ie,pe,oe,ce;return a=new Yn({props:{title:"HALLMARK_INTERFERON_ALPHA_RESPONSE",class:"pipen-report-table-image-descr"}}),o=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/fgsea_HALLMARK_INTERFERON_ALPHA_RESPONSE.png",class:"pipen-report-table-image",width:1e3,height:800}}),m=new Yn({props:{title:"HALLMARK_IL6_JAK_STAT3_SIGNALING",class:"pipen-report-table-image-descr"}}),L=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/fgsea_HALLMARK_IL6_JAK_STAT3_SIGNALING.png",class:"pipen-report-table-image",width:1e3,height:800}}),C=new Yn({props:{title:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",class:"pipen-report-table-image-descr"}}),T=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/fgsea_HALLMARK_CHOLESTEROL_HOMEOSTASIS.png",class:"pipen-report-table-image",width:1e3,height:800}}),v=new Yn({props:{title:"HALLMARK_MYC_TARGETS_V2",class:"pipen-report-table-image-descr"}}),y=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/fgsea_HALLMARK_MYC_TARGETS_V2.png",class:"pipen-report-table-image",width:1e3,height:800}}),w=new Yn({props:{title:"HALLMARK_ANDROGEN_RESPONSE",class:"pipen-report-table-image-descr"}}),H=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/fgsea_HALLMARK_ANDROGEN_RESPONSE.png",class:"pipen-report-table-image",width:1e3,height:800}}),D=new Yn({props:{title:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",class:"pipen-report-table-image-descr"}}),U=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/fgsea_HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY.png",class:"pipen-report-table-image",width:1e3,height:800}}),X=new Yn({props:{title:"HALLMARK_PANCREAS_BETA_CELLS",class:"pipen-report-table-image-descr"}}),q=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/fgsea_HALLMARK_PANCREAS_BETA_CELLS.png",class:"pipen-report-table-image",width:1e3,height:800}}),W=new Yn({props:{title:"HALLMARK_ANGIOGENESIS",class:"pipen-report-table-image-descr"}}),Z=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/fgsea_HALLMARK_ANGIOGENESIS.png",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new Yn({props:{title:"HALLMARK_APICAL_SURFACE",class:"pipen-report-table-image-descr"}}),se=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/fgsea_HALLMARK_APICAL_SURFACE.png",class:"pipen-report-table-image",width:1e3,height:800}}),ie=new Yn({props:{title:"HALLMARK_TGF_BETA_SIGNALING",class:"pipen-report-table-image-descr"}}),oe=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/fgsea_HALLMARK_TGF_BETA_SIGNALING.png",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=r("div"),n=r("div"),f(a.$$.fragment),s=S(),f(o.$$.fragment),c=S(),g=r("div"),f(m.$$.fragment),$=S(),f(L.$$.fragment),E=S(),R=r("div"),f(C.$$.fragment),P=S(),f(T.$$.fragment),N=S(),M=r("div"),f(v.$$.fragment),I=S(),f(y.$$.fragment),x=S(),G=r("div"),f(w.$$.fragment),B=S(),f(H.$$.fragment),F=S(),O=r("div"),f(D.$$.fragment),K=S(),f(U.$$.fragment),V=S(),z=r("div"),f(X.$$.fragment),k=S(),f(q.$$.fragment),j=S(),Y=r("div"),f(W.$$.fragment),J=S(),f(Z.$$.fragment),ee=S(),te=r("div"),f(ne.$$.fragment),ae=S(),f(se.$$.fragment),re=S(),le=r("div"),f(ie.$$.fragment),pe=S(),f(oe.$$.fragment),l(t,"class","pipen-report-table-of-images"),Q(t,"grid-template-columns","repeat(2, auto)")},m(e,r){i(e,t,r),b(t,n),A(a,n,null),b(n,s),A(o,n,null),b(t,c),b(t,g),A(m,g,null),b(g,$),A(L,g,null),b(t,E),b(t,R),A(C,R,null),b(R,P),A(T,R,null),b(t,N),b(t,M),A(v,M,null),b(M,I),A(y,M,null),b(t,x),b(t,G),A(w,G,null),b(G,B),A(H,G,null),b(t,F),b(t,O),A(D,O,null),b(O,K),A(U,O,null),b(t,V),b(t,z),A(X,z,null),b(z,k),A(q,z,null),b(t,j),b(t,Y),A(W,Y,null),b(Y,J),A(Z,Y,null),b(t,ee),b(t,te),A(ne,te,null),b(te,ae),A(se,te,null),b(t,re),b(t,le),A(ie,le,null),b(le,pe),A(oe,le,null),ce=!0},p:h,i(e){ce||(p(a.$$.fragment,e),p(o.$$.fragment,e),p(m.$$.fragment,e),p(L.$$.fragment,e),p(C.$$.fragment,e),p(T.$$.fragment,e),p(v.$$.fragment,e),p(y.$$.fragment,e),p(w.$$.fragment,e),p(H.$$.fragment,e),p(D.$$.fragment,e),p(U.$$.fragment,e),p(X.$$.fragment,e),p(q.$$.fragment,e),p(W.$$.fragment,e),p(Z.$$.fragment,e),p(ne.$$.fragment,e),p(se.$$.fragment,e),p(ie.$$.fragment,e),p(oe.$$.fragment,e),ce=!0)},o(e){_(a.$$.fragment,e),_(o.$$.fragment,e),_(m.$$.fragment,e),_(L.$$.fragment,e),_(C.$$.fragment,e),_(T.$$.fragment,e),_(v.$$.fragment,e),_(y.$$.fragment,e),_(w.$$.fragment,e),_(H.$$.fragment,e),_(D.$$.fragment,e),_(U.$$.fragment,e),_(X.$$.fragment,e),_(q.$$.fragment,e),_(W.$$.fragment,e),_(Z.$$.fragment,e),_(ne.$$.fragment,e),_(se.$$.fragment,e),_(ie.$$.fragment,e),_(oe.$$.fragment,e),ce=!1},d(e){e&&d(t),u(a),u(o),u(m),u(L),u(C),u(T),u(v),u(y),u(w),u(H),u(D),u(U),u(X),u(q),u(W),u(Z),u(ne),u(se),u(ie),u(oe)}}}function va(e){let t,n,a,s;return t=new Ie({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[Na]},$$scope:{ctx:e}}}),a=new Ie({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[Ma]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=S(),f(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(p(t.$$.fragment,e),p(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),u(t,e),u(a,e)}}}function Ia(e){let t,n,a,s;return t=new me({props:{label:"Plot"}}),a=new me({props:{label:"Table"}}),{c(){f(t.$$.fragment),n=S(),f(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p:h,i(e){s||(p(t.$$.fragment,e),p(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),u(t,e),u(a,e)}}}function ya(e){let t,n;return t=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/gsea_table.png",width:1e3,height:760}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p:h,i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function xa(e){let t,n;return t=new Xn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/fgsea.txt",data:[{pathway:"HALLMARK_INTERFERON_ALPHA_RESPONSE",pval:0,padj:3e-4,log2err:.5756,ES:.7611,NES:1.6724,size:96,leadingEdge:"ISG20,IFITM2,CD74,UBE2L6,PSMB9,TAP1,IFITM1,LY6E,B2M,PSME1,ISG15,PSMB8,GBP2,PARP9,IRF1,IFI35,PSME2,GBP4,IL7,SAMD9L,USP18,NMI,IFITM3,PLSCR1,EPSTI1,IRF7,MOV10,ELF1,RSAD2,UBA7,NCOA7,IFI27,OGFR,TRAFD1,DHX58,SP110,IL4R,BST2,ADAR,TMEM140,CNP,OASL,SELL,LGALS3BP,SAMD9,HELZ2,IFIT2,IFIT3,SLC25A28,TENT5A,PARP12,IFIH1,TRIM5,IRF2,LAP3,HERC6",id:0},{pathway:"HALLMARK_PANCREAS_BETA_CELLS",pval:.0055,padj:.0383,log2err:.407,ES:.7729,NES:1.5221,size:40,leadingEdge:"VDR,SRP9,SPCS1,SEC11A,STXBP1,AKT3,SRP14,DPP4,ELP4",id:1},{pathway:"HALLMARK_IL6_JAK_STAT3_SIGNALING",pval:.1005,padj:.469,log2err:.1552,ES:.5655,NES:1.2324,size:87,leadingEdge:"CD38,STAT1,IL18R1,MAP3K8,SOCS3,IRF1,IL7,MYD88,IL12RB1,PIK3R5,CSF2,GRB2,TYK2,IL4R,PIM1,PTPN1,TGFB1,CXCL13,STAT3,SOCS1,CD9,CCR1,IL2RG,EBI3,TNFRSF1A,BAK1,PTPN2,IL1R2,CXCL10,CD14,LTBR,ACVR1B",id:2},{pathway:"HALLMARK_MYC_TARGETS_V2",pval:.1863,padj:.6038,log2err:.2021,ES:-.4899,NES:-1.1381,size:58,leadingEdge:"MCM4,TBRG4,PHB,IMP4,PRMT3,WDR74,NDUFAF4,PPAN,TFB2M,MYBBP1A,MPHOSPH10,UNG,CDK4,SORD,PA2G4,WDR43,NPM1,DCTPP1,MCM5,HSPE1,UTP20,NOP56,NIP7,IPO4,RRP9,TCOF1,PLK4,RRP12,NOP2,PPRC1,NOC4L,NOP16,PES1",id:3},{pathway:"HALLMARK_ANGIOGENESIS",pval:.2264,padj:.6038,log2err:.1063,ES:.6064,NES:1.1749,size:36,leadingEdge:"TIMP1,S100A4,JAG1,APP,PDGFA,FGFR1",id:4},{pathway:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",pval:.2769,padj:.6038,log2err:.0884,ES:.513,NES:1.0933,size:73,leadingEdge:"TP53INP1,GSTM2,ETHE1,FBXO6,PLSCR1,CXCL16,ALCAM,GLDC,LDLR,FDFT1,S100A11,SREBF2,LGMN,JAG1,ANXA5,SC5D,FADS2,SCD,CD9,ATXN2,ACTG1,STARD4,TM7SF2,ATF3,IDI1,CPEB2,CLU,MVD,HSD17B7,HMGCS1,ATF5,PDK3",id:5},{pathway:"HALLMARK_HEDGEHOG_SIGNALING",pval:.3019,padj:.6038,log2err:.0892,ES:.5684,NES:1.1013,size:36,leadingEdge:"CDK5R1,ADGRG1,TLE1,PML,CDK6,RASA1,L1CAM,NF1,MYH9,LDB1",id:6},{pathway:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",pval:.4473,padj:.7827,log2err:.0669,ES:.4967,NES:1.0024,size:49,leadingEdge:"NDUFA6,LSP1,PRDX6,SOD1,SBNO2,NDUFB4,HMOX2,ATOX1,FES,GLRX,PRDX2,GLRX2,STK25,ABCC1,MBP,EGLN2,CAT",id:7},{pathway:"HALLMARK_ANDROGEN_RESPONSE",pval:.5553,padj:.8267,log2err:.0518,ES:.4365,NES:.96,size:99,leadingEdge:"FKBP5,ARID5B,SAT1,B2M,RPS6KA3,ELOVL5,ABCC4,ELL2,CCND3,CDK6,HOMER2,DHCR24,TNFAIP8,PLPP1,ZMIZ1,PDLIM5,SCD,MAK,RAB4A,INSIG1,DNAJB9,KRT19,KRT8,NKX3-1,MERTK,IDI1,TMEM50A,UAP1,ABHD2,AKT1,SLC38A2",id:8},{pathway:"HALLMARK_PROTEIN_SECRETION",pval:.5905,padj:.8267,log2err:.1215,ES:-.3726,NES:-.9326,size:95,leadingEdge:"CD63,ANP32E,ATP1A1,TPD52,AP3B1,RAB5A,ARCN1,SCAMP3,TMED2,CTSC,AP3S1,RAB14,BET1,PPT1,PAM,TMED10,VAMP3,SGMS1,STAM,SEC24D,GLA,CLTA,COPB1,ICA1,GOLGA4,STX16,TSG101,BNIP3",id:9},{pathway:"HALLMARK_APICAL_SURFACE",pval:.8874,padj:.9294,log2err:.0755,ES:-.366,NES:-.823,size:44,leadingEdge:"GSTM3,HSPB1,CD160,AFAP1L2,PLAUR,FLOT2,SCUBE1,LYPD3,ADAM10,B4GALT1,CROCC",id:10},{pathway:"HALLMARK_NOTCH_SIGNALING",pval:.9108,padj:.9294,log2err:.0722,ES:-.3719,NES:-.7733,size:32,leadingEdge:"SKP1,KAT2A,ARRB1,DTX2",id:11},{pathway:"HALLMARK_WNT_BETA_CATENIN_SIGNALING",pval:.9141,padj:.9294,log2err:.0336,ES:.3771,NES:.7469,size:42,leadingEdge:"NOTCH1,NKD1,JAG1,WNT1,FRAT1,AXIN2,HDAC5,SKP2,NUMB,DVL2,LEF1,NOTCH4,MYC,PPARD,CUL1,ADAM17,RBPJ,TCF7",id:12},{pathway:"HALLMARK_TGF_BETA_SIGNALING",pval:.9294,padj:.9294,log2err:.032,ES:.3685,NES:.7509,size:54,leadingEdge:"ID2,FURIN,SKIL,SPTBN1,ACVR1,UBE2D3,THBS1,TGFB1,ID1,ARID4B,SERPINE1,BCAR3,SMURF1,PPP1CA,SMAD7,FKBP1A,WWTR1,RHOA,CDK9,HDAC1,BMPR2",id:13}]}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p:h,i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function Ga(e){let t,n,a,s;return t=new $e({props:{$$slots:{default:[ya]},$$scope:{ctx:e}}}),a=new $e({props:{$$slots:{default:[xa]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=S(),f(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(p(t.$$.fragment,e),p(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),u(t,e),u(a,e)}}}function wa(e){let t,n;return t=new ge({props:{$$slots:{content:[Ga],default:[Ia]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function Ba(e){let t,n,a,s,o,c,g,m,$,L,E,R,C,P,T,N,M,v,I,y,x,G,w,B,H,F,O,D,K,U,V,z,X,k,q,j,Y,W,J,Z,ee,te,ne,ae,se,re,le,ie,pe,oe,ce;return a=new Yn({props:{title:"HALLMARK_INTERFERON_ALPHA_RESPONSE",class:"pipen-report-table-image-descr"}}),o=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/fgsea_HALLMARK_INTERFERON_ALPHA_RESPONSE.png",class:"pipen-report-table-image",width:1e3,height:800}}),m=new Yn({props:{title:"HALLMARK_PANCREAS_BETA_CELLS",class:"pipen-report-table-image-descr"}}),L=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/fgsea_HALLMARK_PANCREAS_BETA_CELLS.png",class:"pipen-report-table-image",width:1e3,height:800}}),C=new Yn({props:{title:"HALLMARK_IL6_JAK_STAT3_SIGNALING",class:"pipen-report-table-image-descr"}}),T=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/fgsea_HALLMARK_IL6_JAK_STAT3_SIGNALING.png",class:"pipen-report-table-image",width:1e3,height:800}}),v=new Yn({props:{title:"HALLMARK_MYC_TARGETS_V2",class:"pipen-report-table-image-descr"}}),y=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/fgsea_HALLMARK_MYC_TARGETS_V2.png",class:"pipen-report-table-image",width:1e3,height:800}}),w=new Yn({props:{title:"HALLMARK_ANGIOGENESIS",class:"pipen-report-table-image-descr"}}),H=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/fgsea_HALLMARK_ANGIOGENESIS.png",class:"pipen-report-table-image",width:1e3,height:800}}),D=new Yn({props:{title:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",class:"pipen-report-table-image-descr"}}),U=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/fgsea_HALLMARK_CHOLESTEROL_HOMEOSTASIS.png",class:"pipen-report-table-image",width:1e3,height:800}}),X=new Yn({props:{title:"HALLMARK_HEDGEHOG_SIGNALING",class:"pipen-report-table-image-descr"}}),q=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/fgsea_HALLMARK_HEDGEHOG_SIGNALING.png",class:"pipen-report-table-image",width:1e3,height:800}}),W=new Yn({props:{title:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",class:"pipen-report-table-image-descr"}}),Z=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/fgsea_HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY.png",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new Yn({props:{title:"HALLMARK_ANDROGEN_RESPONSE",class:"pipen-report-table-image-descr"}}),se=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/fgsea_HALLMARK_ANDROGEN_RESPONSE.png",class:"pipen-report-table-image",width:1e3,height:800}}),ie=new Yn({props:{title:"HALLMARK_PROTEIN_SECRETION",class:"pipen-report-table-image-descr"}}),oe=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/fgsea_HALLMARK_PROTEIN_SECRETION.png",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=r("div"),n=r("div"),f(a.$$.fragment),s=S(),f(o.$$.fragment),c=S(),g=r("div"),f(m.$$.fragment),$=S(),f(L.$$.fragment),E=S(),R=r("div"),f(C.$$.fragment),P=S(),f(T.$$.fragment),N=S(),M=r("div"),f(v.$$.fragment),I=S(),f(y.$$.fragment),x=S(),G=r("div"),f(w.$$.fragment),B=S(),f(H.$$.fragment),F=S(),O=r("div"),f(D.$$.fragment),K=S(),f(U.$$.fragment),V=S(),z=r("div"),f(X.$$.fragment),k=S(),f(q.$$.fragment),j=S(),Y=r("div"),f(W.$$.fragment),J=S(),f(Z.$$.fragment),ee=S(),te=r("div"),f(ne.$$.fragment),ae=S(),f(se.$$.fragment),re=S(),le=r("div"),f(ie.$$.fragment),pe=S(),f(oe.$$.fragment),l(t,"class","pipen-report-table-of-images"),Q(t,"grid-template-columns","repeat(2, auto)")},m(e,r){i(e,t,r),b(t,n),A(a,n,null),b(n,s),A(o,n,null),b(t,c),b(t,g),A(m,g,null),b(g,$),A(L,g,null),b(t,E),b(t,R),A(C,R,null),b(R,P),A(T,R,null),b(t,N),b(t,M),A(v,M,null),b(M,I),A(y,M,null),b(t,x),b(t,G),A(w,G,null),b(G,B),A(H,G,null),b(t,F),b(t,O),A(D,O,null),b(O,K),A(U,O,null),b(t,V),b(t,z),A(X,z,null),b(z,k),A(q,z,null),b(t,j),b(t,Y),A(W,Y,null),b(Y,J),A(Z,Y,null),b(t,ee),b(t,te),A(ne,te,null),b(te,ae),A(se,te,null),b(t,re),b(t,le),A(ie,le,null),b(le,pe),A(oe,le,null),ce=!0},p:h,i(e){ce||(p(a.$$.fragment,e),p(o.$$.fragment,e),p(m.$$.fragment,e),p(L.$$.fragment,e),p(C.$$.fragment,e),p(T.$$.fragment,e),p(v.$$.fragment,e),p(y.$$.fragment,e),p(w.$$.fragment,e),p(H.$$.fragment,e),p(D.$$.fragment,e),p(U.$$.fragment,e),p(X.$$.fragment,e),p(q.$$.fragment,e),p(W.$$.fragment,e),p(Z.$$.fragment,e),p(ne.$$.fragment,e),p(se.$$.fragment,e),p(ie.$$.fragment,e),p(oe.$$.fragment,e),ce=!0)},o(e){_(a.$$.fragment,e),_(o.$$.fragment,e),_(m.$$.fragment,e),_(L.$$.fragment,e),_(C.$$.fragment,e),_(T.$$.fragment,e),_(v.$$.fragment,e),_(y.$$.fragment,e),_(w.$$.fragment,e),_(H.$$.fragment,e),_(D.$$.fragment,e),_(U.$$.fragment,e),_(X.$$.fragment,e),_(q.$$.fragment,e),_(W.$$.fragment,e),_(Z.$$.fragment,e),_(ne.$$.fragment,e),_(se.$$.fragment,e),_(ie.$$.fragment,e),_(oe.$$.fragment,e),ce=!1},d(e){e&&d(t),u(a),u(o),u(m),u(L),u(C),u(T),u(v),u(y),u(w),u(H),u(D),u(U),u(X),u(q),u(W),u(Z),u(ne),u(se),u(ie),u(oe)}}}function Ha(e){let t,n,a,s;return t=new Ie({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[wa]},$$scope:{ctx:e}}}),a=new Ie({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[Ba]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=S(),f(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(p(t.$$.fragment,e),p(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),u(t,e),u(a,e)}}}function Fa(e){let t,n,a,s;return t=new me({props:{label:"Plot"}}),a=new me({props:{label:"Table"}}),{c(){f(t.$$.fragment),n=S(),f(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p:h,i(e){s||(p(t.$$.fragment,e),p(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),u(t,e),u(a,e)}}}function Oa(e){let t,n;return t=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/gsea_table.png",width:1e3,height:760}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p:h,i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function Da(e){let t,n;return t=new Xn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/fgsea.txt",data:[{pathway:"HALLMARK_INTERFERON_ALPHA_RESPONSE",pval:0,padj:2e-4,log2err:.5756,ES:.7006,NES:1.7921,size:96,leadingEdge:"ISG20,PSMB9,PSME2,UBE2L6,IFI44L,IFITM1,LY6E,TAP1,PSME1,IFITM2,ISG15,USP18,GBP4,RIPK2,IFIT3,IRF7,GBP2,PLSCR1,PSMB8,PARP9,IRF1,CD74,RSAD2,LAMP3,SAMD9L,IL4R,HERC6,PARP14,ADAR,IFI27,CMPK2,NUB1,IFI35,SP110,OASL,TXNIP,SELL,EIF2AK2,TRIM21,TRIM5,PROCR,B2M,IFITM3,CASP1,MX1,CMTR1,TRIM14,PARP12,TMEM140,BST2,OAS1,IFIT2,IFI30,MVB12A,RTP4,TDRD7",id:0},{pathway:"HALLMARK_PANCREAS_BETA_CELLS",pval:.136,padj:.6957,log2err:.2572,ES:.5384,NES:1.1935,size:40,leadingEdge:"PAK3,SEC11A,SPCS1,VDR,SRP14,FOXO1,DPP4",id:1},{pathway:"HALLMARK_WNT_BETA_CATENIN_SIGNALING",pval:.1493,padj:.6957,log2err:.1263,ES:-.6297,NES:-1.2153,size:42,leadingEdge:"TCF7,AXIN1,HDAC5,NCSTN,NOTCH4,DVL2,ADAM17,WNT1,CTNNB1,RBPJ,HEY2,HEY1,FZD1,FZD8,NUMB,HDAC2,LEF1,AXIN2,SKP2,CSNK1E,NCOR2",id:2},{pathway:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",pval:.2147,padj:.6957,log2err:.2311,ES:.4348,NES:1.0763,size:73,leadingEdge:"S100A11,PLSCR1,ATF3,HSD17B7,SC5D,FADS2,IDI1,LDLR,GNAI1,LGALS3,CXCL16,FASN,EBP,GSTM2,TMEM97,NSDHL,TRIB3,LGMN,STARD4,GUSB,TP53INP1,STX5,ERRFI1,CLU,ACSS2,MVD,CHKA,PDK3,FBXO6",id:3},{pathway:"HALLMARK_ANGIOGENESIS",pval:.2623,padj:.6957,log2err:.1752,ES:.5017,NES:1.1039,size:36,leadingEdge:"LRPAP1,FGFR1,TNFRSF21,TIMP1,COL5A2,ITGAV",id:4},{pathway:"HALLMARK_PROTEIN_SECRETION",pval:.2981,padj:.6957,log2err:.0777,ES:-.5126,NES:-1.0866,size:95,leadingEdge:"CTSC,PPT1,M6PR,NAPA,MAPK1,ATP1A1,DNM1L,AP1G1,ATP6V1H,STX12,RAB22A,RAB5A,LAMP2,SNX2,ANP32E,MON2,GALC,ARFIP1,YKT6,COPB1,ADAM10,ATP7A,VAMP7,IGF2R,SEC31A,USO1,VAMP4,ARCN1,GOLGA4,CLCN3,ERGIC3,SGMS1,GLA,COG2,CLTC,SEC24D,TPD52,NAPG,GOSR2,ARF1,GBF1,STX7,BET1,SNAP23,SEC22B,ICA1,AP2M1,RAB14,KIF1B,ARFGEF1,YIPF6,AP2B1,SCRN1,CLTA,CD63,CLN5",id:5},{pathway:"HALLMARK_HEDGEHOG_SIGNALING",pval:.3852,padj:.7423,log2err:.0711,ES:-.5599,NES:-1.0483,size:36,leadingEdge:"TLE3,CELSR1,NRCAM,NF1,MYH9,HEY2,VLDLR,HEY1,RASA1,CDK5R1",id:6},{pathway:"HALLMARK_TGF_BETA_SIGNALING",pval:.427,padj:.7423,log2err:.0625,ES:-.5143,NES:-1.0286,size:54,leadingEdge:"ID3,SPTBN1,SLC20A1,PPP1R15A,CTNNB1,KLF10,CDK9,CDKN1C,SMAD6,CDH1,SMAD1,TGFBR1,FKBP1A,TRIM33,IFNGR2,APC,UBE2D3,HDAC1,TGFB1,NCOR2,SMAD3",id:7},{pathway:"HALLMARK_ANDROGEN_RESPONSE",pval:.5173,padj:.7423,log2err:.0506,ES:-.4651,NES:-.9873,size:99,leadingEdge:"TNFAIP8,ACTN1,XRCC6,HSD17B14,ANKH,XRCC5,SCD,HPGD,MAP7,PDLIM5,NCOA4,PIAS1,SORD,PLPP1,PTK2B,ACSL3,SMS,DBI,SLC38A2,ZMIZ1,ABCC4,CAMKK2,NGLY1,ELL2,HMGCR,ELK4,APPBP2,GPD1L,RAB4A,SEC24D,UAP1,FADS1,TPD52,SGK1,TSC22D1,SLC26A2,STK39,RRP12,GSR,CDC14B,ZBTB10,PA2G4",id:8},{pathway:"HALLMARK_IL6_JAK_STAT3_SIGNALING",pval:.5302,padj:.7423,log2err:.1563,ES:.3823,NES:.9689,size:87,leadingEdge:"STAT1,SOCS1,IL18R1,IL15RA,IRF1,FAS,TNFRSF1B,IL4R,SOCS3,TNFRSF21,PTPN1,MAP3K8,PIM1,JUN,LTB,CD44,IL12RB1",id:9},{pathway:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",pval:.7531,padj:.8607,log2err:.0372,ES:-.4291,NES:-.8461,size:49,leadingEdge:"CAT,NQO1,ABCC1,TXNRD1,SCAF4,PRDX1,PFKP,PRDX6,ATOX1,SRXN1,PDLIM1,GPX4,MSRA,PRNP,LAMTOR5,PRDX2,LSP1",id:10},{pathway:"HALLMARK_NOTCH_SIGNALING",pval:.7659,padj:.8607,log2err:.039,ES:-.4489,NES:-.8285,size:32,leadingEdge:"DTX2,MAML2,ARRB1,FZD1,SAP30,TCF7L2,DTX1,SKP1,PSENEN,PPARD,PRKCA,KAT2A,FBXW11",id:11},{pathway:"HALLMARK_APICAL_SURFACE",pval:.7992,padj:.8607,log2err:.0351,ES:-.4262,NES:-.8266,size:44,leadingEdge:"HSPB1,AFAP1L2,GATA3,ADIPOR2,APP,LYPD3,ADAM10,RTN4RL1,CD160,GSTM3,PLAUR,FLOT2",id:12},{pathway:"HALLMARK_MYC_TARGETS_V2",pval:.9891,padj:.9891,log2err:.0219,ES:-.3395,NES:-.6844,size:58,leadingEdge:"CBX3,TFB2M,BYSL,SRM,CDK4,NDUFAF4,SORD,MYBBP1A,NOLC1,UTP20,FARSA,DDX18,NOP2,IMP4,WDR43,TBRG4,GNL3,NIP7,NOP16,HSPE1,PLK4,MPHOSPH10,RRP12,PA2G4,LAS1L,MAP3K6",id:13}]}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p:h,i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function Ka(e){let t,n,a,s;return t=new $e({props:{$$slots:{default:[Oa]},$$scope:{ctx:e}}}),a=new $e({props:{$$slots:{default:[Da]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=S(),f(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(p(t.$$.fragment,e),p(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),u(t,e),u(a,e)}}}function Ua(e){let t,n;return t=new ge({props:{$$slots:{content:[Ka],default:[Fa]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function Va(e){let t,n,a,s,o,c,g,m,$,L,E,R,C,P,T,N,M,v,I,y,x,G,w,B,H,F,O,D,K,U,V,z,X,k,q,j,Y,W,J,Z,ee,te,ne,ae,se,re,le,ie,pe,oe,ce;return a=new Yn({props:{title:"HALLMARK_INTERFERON_ALPHA_RESPONSE",class:"pipen-report-table-image-descr"}}),o=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/fgsea_HALLMARK_INTERFERON_ALPHA_RESPONSE.png",class:"pipen-report-table-image",width:1e3,height:800}}),m=new Yn({props:{title:"HALLMARK_PANCREAS_BETA_CELLS",class:"pipen-report-table-image-descr"}}),L=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/fgsea_HALLMARK_PANCREAS_BETA_CELLS.png",class:"pipen-report-table-image",width:1e3,height:800}}),C=new Yn({props:{title:"HALLMARK_WNT_BETA_CATENIN_SIGNALING",class:"pipen-report-table-image-descr"}}),T=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/fgsea_HALLMARK_WNT_BETA_CATENIN_SIGNALING.png",class:"pipen-report-table-image",width:1e3,height:800}}),v=new Yn({props:{title:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",class:"pipen-report-table-image-descr"}}),y=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/fgsea_HALLMARK_CHOLESTEROL_HOMEOSTASIS.png",class:"pipen-report-table-image",width:1e3,height:800}}),w=new Yn({props:{title:"HALLMARK_ANGIOGENESIS",class:"pipen-report-table-image-descr"}}),H=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/fgsea_HALLMARK_ANGIOGENESIS.png",class:"pipen-report-table-image",width:1e3,height:800}}),D=new Yn({props:{title:"HALLMARK_PROTEIN_SECRETION",class:"pipen-report-table-image-descr"}}),U=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/fgsea_HALLMARK_PROTEIN_SECRETION.png",class:"pipen-report-table-image",width:1e3,height:800}}),X=new Yn({props:{title:"HALLMARK_HEDGEHOG_SIGNALING",class:"pipen-report-table-image-descr"}}),q=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/fgsea_HALLMARK_HEDGEHOG_SIGNALING.png",class:"pipen-report-table-image",width:1e3,height:800}}),W=new Yn({props:{title:"HALLMARK_TGF_BETA_SIGNALING",class:"pipen-report-table-image-descr"}}),Z=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/fgsea_HALLMARK_TGF_BETA_SIGNALING.png",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new Yn({props:{title:"HALLMARK_ANDROGEN_RESPONSE",class:"pipen-report-table-image-descr"}}),se=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/fgsea_HALLMARK_ANDROGEN_RESPONSE.png",class:"pipen-report-table-image",width:1e3,height:800}}),ie=new Yn({props:{title:"HALLMARK_IL6_JAK_STAT3_SIGNALING",class:"pipen-report-table-image-descr"}}),oe=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/fgsea_HALLMARK_IL6_JAK_STAT3_SIGNALING.png",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=r("div"),n=r("div"),f(a.$$.fragment),s=S(),f(o.$$.fragment),c=S(),g=r("div"),f(m.$$.fragment),$=S(),f(L.$$.fragment),E=S(),R=r("div"),f(C.$$.fragment),P=S(),f(T.$$.fragment),N=S(),M=r("div"),f(v.$$.fragment),I=S(),f(y.$$.fragment),x=S(),G=r("div"),f(w.$$.fragment),B=S(),f(H.$$.fragment),F=S(),O=r("div"),f(D.$$.fragment),K=S(),f(U.$$.fragment),V=S(),z=r("div"),f(X.$$.fragment),k=S(),f(q.$$.fragment),j=S(),Y=r("div"),f(W.$$.fragment),J=S(),f(Z.$$.fragment),ee=S(),te=r("div"),f(ne.$$.fragment),ae=S(),f(se.$$.fragment),re=S(),le=r("div"),f(ie.$$.fragment),pe=S(),f(oe.$$.fragment),l(t,"class","pipen-report-table-of-images"),Q(t,"grid-template-columns","repeat(2, auto)")},m(e,r){i(e,t,r),b(t,n),A(a,n,null),b(n,s),A(o,n,null),b(t,c),b(t,g),A(m,g,null),b(g,$),A(L,g,null),b(t,E),b(t,R),A(C,R,null),b(R,P),A(T,R,null),b(t,N),b(t,M),A(v,M,null),b(M,I),A(y,M,null),b(t,x),b(t,G),A(w,G,null),b(G,B),A(H,G,null),b(t,F),b(t,O),A(D,O,null),b(O,K),A(U,O,null),b(t,V),b(t,z),A(X,z,null),b(z,k),A(q,z,null),b(t,j),b(t,Y),A(W,Y,null),b(Y,J),A(Z,Y,null),b(t,ee),b(t,te),A(ne,te,null),b(te,ae),A(se,te,null),b(t,re),b(t,le),A(ie,le,null),b(le,pe),A(oe,le,null),ce=!0},p:h,i(e){ce||(p(a.$$.fragment,e),p(o.$$.fragment,e),p(m.$$.fragment,e),p(L.$$.fragment,e),p(C.$$.fragment,e),p(T.$$.fragment,e),p(v.$$.fragment,e),p(y.$$.fragment,e),p(w.$$.fragment,e),p(H.$$.fragment,e),p(D.$$.fragment,e),p(U.$$.fragment,e),p(X.$$.fragment,e),p(q.$$.fragment,e),p(W.$$.fragment,e),p(Z.$$.fragment,e),p(ne.$$.fragment,e),p(se.$$.fragment,e),p(ie.$$.fragment,e),p(oe.$$.fragment,e),ce=!0)},o(e){_(a.$$.fragment,e),_(o.$$.fragment,e),_(m.$$.fragment,e),_(L.$$.fragment,e),_(C.$$.fragment,e),_(T.$$.fragment,e),_(v.$$.fragment,e),_(y.$$.fragment,e),_(w.$$.fragment,e),_(H.$$.fragment,e),_(D.$$.fragment,e),_(U.$$.fragment,e),_(X.$$.fragment,e),_(q.$$.fragment,e),_(W.$$.fragment,e),_(Z.$$.fragment,e),_(ne.$$.fragment,e),_(se.$$.fragment,e),_(ie.$$.fragment,e),_(oe.$$.fragment,e),ce=!1},d(e){e&&d(t),u(a),u(o),u(m),u(L),u(C),u(T),u(v),u(y),u(w),u(H),u(D),u(U),u(X),u(q),u(W),u(Z),u(ne),u(se),u(ie),u(oe)}}}function za(e){let t,n,a,s;return t=new Ie({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[Ua]},$$scope:{ctx:e}}}),a=new Ie({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[Va]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=S(),f(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(p(t.$$.fragment,e),p(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),u(t,e),u(a,e)}}}function Xa(e){let t,n,a,s;return t=new me({props:{label:"Plot"}}),a=new me({props:{label:"Table"}}),{c(){f(t.$$.fragment),n=S(),f(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p:h,i(e){s||(p(t.$$.fragment,e),p(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),u(t,e),u(a,e)}}}function ka(e){let t,n;return t=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/gsea_table.png",width:1e3,height:760}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p:h,i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function qa(e){let t,n;return t=new Xn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/fgsea.txt",data:[{pathway:"HALLMARK_INTERFERON_ALPHA_RESPONSE",pval:.0289,padj:.3547,log2err:.282,ES:.5944,NES:1.3288,size:96,leadingEdge:"ISG20,CD74,TAP1,PLSCR1,PSMB8,IFITM3,GBP2,PSMB9,RTP4,ISG15,SELL,LGALS3BP,NCOA7,UBA7,UBE2L6,TRAFD1,HELZ2,PARP9,NMI,IL4R,GBP4,SP110,PSME2,MVB12A,EPSTI1,LPAR6,CSF1,IRF7,PSME1,IRF1,IRF2,STAT2,IL7,BST2,SAMD9L,IFIT3,MOV10,PARP14,ELF1,NUB1,IFI27,DHX58,IFITM2,TMEM140,PNPT1,OAS1,IFI44,LAMP3,LY6E,IFITM1,LAP3",id:0},{pathway:"HALLMARK_IL6_JAK_STAT3_SIGNALING",pval:.0642,padj:.3547,log2err:.1864,ES:.5847,NES:1.2921,size:87,leadingEdge:"CD38,SOCS3,STAT1,IL2RG,BAK1,HMOX1,MYD88,CBL,TYK2,TNFRSF1B,STAM2,IL4R,IL6ST,CSF1,IRF1,CD9,STAT2,IL7,ITGA4,STAT3,IL18R1,TGFB1,CSF2,IL17RA",id:1},{pathway:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",pval:.0882,padj:.3547,log2err:.1596,ES:.5828,NES:1.2655,size:73,leadingEdge:"GSTM2,TP53INP1,PLSCR1,ATF5,CYP51A1,NSDHL,ANXA5,GLDC,ACTG1,ABCA2,SCD,HSD17B7,CXCL16,ERRFI1,HMGCS1,ETHE1,CD9,ATXN2,STX5,CHKA,GUSB,DHCR7,ECH1,JAG1,CTNNB1,FADS2,PLAUR,SREBF2,TM7SF2,STARD4,SC5D,PDK3,PMVK,MVD",id:2},{pathway:"HALLMARK_TGF_BETA_SIGNALING",pval:.1013,padj:.3547,log2err:.1521,ES:.607,NES:1.2647,size:54,leadingEdge:"ID2,FURIN,FKBP1A,SPTBN1,PMEPA1,ACVR1,SMURF1,TGIF1,ARID4B,BMPR1A,HIPK2,THBS1,UBE2D3,TGFB1,SKIL,TRIM33,SMAD1,CTNNB1,SERPINE1",id:3},{pathway:"HALLMARK_PANCREAS_BETA_CELLS",pval:.1467,padj:.4107,log2err:.1269,ES:.6197,NES:1.2441,size:40,leadingEdge:"VDR,DPP4,SRP14,SRPRB,FOXO1,ELP4,SEC11A,MAFB,SPCS1,AKT3",id:4},{pathway:"HALLMARK_NOTCH_SIGNALING",pval:.2185,padj:.4522,log2err:.1017,ES:.6046,NES:1.1775,size:32,leadingEdge:"PPARD,ARRB1,DTX1,NOTCH1,NOTCH2,LFNG,DTX2,JAG1,RBX1,MAML2,KAT2A",id:5},{pathway:"HALLMARK_HEDGEHOG_SIGNALING",pval:.2372,padj:.4522,log2err:.1978,ES:-.484,NES:-1.1203,size:36,leadingEdge:"L1CAM,RASA1,CDK5R1,PTCH1,LDB1,TLE3,DPYSL2",id:6},{pathway:"HALLMARK_MYC_TARGETS_V2",pval:.2584,padj:.4522,log2err:.209,ES:-.4272,NES:-1.077,size:58,leadingEdge:"TFB2M,TMEM97,NOC4L,SLC29A2,MPHOSPH10,MCM5,PPAN,PHB,IMP4,PRMT3,MCM4,RRP9,NDUFAF4,PLK4,NOP16,PA2G4,NOLC1,MYBBP1A,WDR74,SUPV3L1,UTP20,RABEPK,PPRC1,MYC,BYSL,LAS1L,TBRG4,SRM,UNG",id:7},{pathway:"HALLMARK_ANGIOGENESIS",pval:.4917,padj:.7649,log2err:.0575,ES:.4953,NES:.9856,size:36,leadingEdge:"APP,FGFR1,LRPAP1,PDGFA,JAG1,JAG2,VAV2",id:8},{pathway:"HALLMARK_ANDROGEN_RESPONSE",pval:.6493,padj:.909,log2err:.0385,ES:.4085,NES:.916,size:99,leadingEdge:"FKBP5,SAT1,HPGD,ARID5B,RPS6KA3,ELOVL5,DHCR24,PMEPA1,PLPP1,PTK2B,SCD,GPD1L,HMGCS1,SRP19,NGLY1,UAP1,DBI,HOMER2",id:9},{pathway:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",pval:.8,padj:.9358,log2err:.034,ES:.397,NES:.8181,size:49,leadingEdge:"PRDX2,PRDX6,SELENOS,NDUFS2,FES,LSP1,NQO1,SBNO2,ABCC1,NDUFA6,SRXN1,NDUFB4,MBP,TXNRD1,PTPA,EGLN2,STK25",id:10},{pathway:"HALLMARK_PROTEIN_SECRETION",pval:.8181,padj:.9358,log2err:.0276,ES:.3685,NES:.8206,size:95,leadingEdge:"ERGIC3,GNAS,RPS6KA3,RER1,ICA1,AP2S1,COPB2,KIF1B,ARFIP1,ARF1,TSG101,STAM,VAMP4,GOLGA4,VAMP3,DNM1L,GALC,KRT18,ARCN1,TMED10,RAB2A,NAPG,IGF2R,YIPF6,SCAMP3,USO1,LMAN1,BNIP3,SGMS1,AP2M1,VAMP7,SNAP23,NAPA,ATP7A,ABCA1,COPE,AP1G1,BET1,ANP32E,TMED2,SCAMP1",id:11},{pathway:"HALLMARK_APICAL_SURFACE",pval:.9356,padj:.9358,log2err:.027,ES:.3511,NES:.7134,size:44,leadingEdge:"IL2RG,GATA3,APP,MAL,SCUBE1,TMEM8B,PLAUR,MDGA1,IL2RB,ATP8B1",id:12},{pathway:"HALLMARK_WNT_BETA_CATENIN_SIGNALING",pval:.9358,padj:.9358,log2err:.0269,ES:.3578,NES:.722,size:42,leadingEdge:"HDAC5,PPARD,RBPJ,NOTCH1,WNT1,JAG1,CTNNB1,JAG2,FRAT1,KAT2A,NOTCH4,AXIN1,HDAC2,TP53,LEF1,HEY1",id:13}]}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p:h,i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function ja(e){let t,n,a,s;return t=new $e({props:{$$slots:{default:[ka]},$$scope:{ctx:e}}}),a=new $e({props:{$$slots:{default:[qa]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=S(),f(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(p(t.$$.fragment,e),p(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),u(t,e),u(a,e)}}}function Ya(e){let t,n;return t=new ge({props:{$$slots:{content:[ja],default:[Xa]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function Wa(e){let t,n,a,s,o,c,g,m,$,L,E,R,C,P,T,N,M,v,I,y,x,G,w,B,H,F,O,D,K,U,V,z,X,k,q,j,Y,W,J,Z,ee,te,ne,ae,se,re,le,ie,pe,oe,ce;return a=new Yn({props:{title:"HALLMARK_INTERFERON_ALPHA_RESPONSE",class:"pipen-report-table-image-descr"}}),o=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/fgsea_HALLMARK_INTERFERON_ALPHA_RESPONSE.png",class:"pipen-report-table-image",width:1e3,height:800}}),m=new Yn({props:{title:"HALLMARK_IL6_JAK_STAT3_SIGNALING",class:"pipen-report-table-image-descr"}}),L=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/fgsea_HALLMARK_IL6_JAK_STAT3_SIGNALING.png",class:"pipen-report-table-image",width:1e3,height:800}}),C=new Yn({props:{title:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",class:"pipen-report-table-image-descr"}}),T=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/fgsea_HALLMARK_CHOLESTEROL_HOMEOSTASIS.png",class:"pipen-report-table-image",width:1e3,height:800}}),v=new Yn({props:{title:"HALLMARK_TGF_BETA_SIGNALING",class:"pipen-report-table-image-descr"}}),y=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/fgsea_HALLMARK_TGF_BETA_SIGNALING.png",class:"pipen-report-table-image",width:1e3,height:800}}),w=new Yn({props:{title:"HALLMARK_PANCREAS_BETA_CELLS",class:"pipen-report-table-image-descr"}}),H=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/fgsea_HALLMARK_PANCREAS_BETA_CELLS.png",class:"pipen-report-table-image",width:1e3,height:800}}),D=new Yn({props:{title:"HALLMARK_NOTCH_SIGNALING",class:"pipen-report-table-image-descr"}}),U=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/fgsea_HALLMARK_NOTCH_SIGNALING.png",class:"pipen-report-table-image",width:1e3,height:800}}),X=new Yn({props:{title:"HALLMARK_HEDGEHOG_SIGNALING",class:"pipen-report-table-image-descr"}}),q=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/fgsea_HALLMARK_HEDGEHOG_SIGNALING.png",class:"pipen-report-table-image",width:1e3,height:800}}),W=new Yn({props:{title:"HALLMARK_MYC_TARGETS_V2",class:"pipen-report-table-image-descr"}}),Z=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/fgsea_HALLMARK_MYC_TARGETS_V2.png",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new Yn({props:{title:"HALLMARK_ANGIOGENESIS",class:"pipen-report-table-image-descr"}}),se=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/fgsea_HALLMARK_ANGIOGENESIS.png",class:"pipen-report-table-image",width:1e3,height:800}}),ie=new Yn({props:{title:"HALLMARK_ANDROGEN_RESPONSE",class:"pipen-report-table-image-descr"}}),oe=new Ln({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/fgsea_HALLMARK_ANDROGEN_RESPONSE.png",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=r("div"),n=r("div"),f(a.$$.fragment),s=S(),f(o.$$.fragment),c=S(),g=r("div"),f(m.$$.fragment),$=S(),f(L.$$.fragment),E=S(),R=r("div"),f(C.$$.fragment),P=S(),f(T.$$.fragment),N=S(),M=r("div"),f(v.$$.fragment),I=S(),f(y.$$.fragment),x=S(),G=r("div"),f(w.$$.fragment),B=S(),f(H.$$.fragment),F=S(),O=r("div"),f(D.$$.fragment),K=S(),f(U.$$.fragment),V=S(),z=r("div"),f(X.$$.fragment),k=S(),f(q.$$.fragment),j=S(),Y=r("div"),f(W.$$.fragment),J=S(),f(Z.$$.fragment),ee=S(),te=r("div"),f(ne.$$.fragment),ae=S(),f(se.$$.fragment),re=S(),le=r("div"),f(ie.$$.fragment),pe=S(),f(oe.$$.fragment),l(t,"class","pipen-report-table-of-images"),Q(t,"grid-template-columns","repeat(2, auto)")},m(e,r){i(e,t,r),b(t,n),A(a,n,null),b(n,s),A(o,n,null),b(t,c),b(t,g),A(m,g,null),b(g,$),A(L,g,null),b(t,E),b(t,R),A(C,R,null),b(R,P),A(T,R,null),b(t,N),b(t,M),A(v,M,null),b(M,I),A(y,M,null),b(t,x),b(t,G),A(w,G,null),b(G,B),A(H,G,null),b(t,F),b(t,O),A(D,O,null),b(O,K),A(U,O,null),b(t,V),b(t,z),A(X,z,null),b(z,k),A(q,z,null),b(t,j),b(t,Y),A(W,Y,null),b(Y,J),A(Z,Y,null),b(t,ee),b(t,te),A(ne,te,null),b(te,ae),A(se,te,null),b(t,re),b(t,le),A(ie,le,null),b(le,pe),A(oe,le,null),ce=!0},p:h,i(e){ce||(p(a.$$.fragment,e),p(o.$$.fragment,e),p(m.$$.fragment,e),p(L.$$.fragment,e),p(C.$$.fragment,e),p(T.$$.fragment,e),p(v.$$.fragment,e),p(y.$$.fragment,e),p(w.$$.fragment,e),p(H.$$.fragment,e),p(D.$$.fragment,e),p(U.$$.fragment,e),p(X.$$.fragment,e),p(q.$$.fragment,e),p(W.$$.fragment,e),p(Z.$$.fragment,e),p(ne.$$.fragment,e),p(se.$$.fragment,e),p(ie.$$.fragment,e),p(oe.$$.fragment,e),ce=!0)},o(e){_(a.$$.fragment,e),_(o.$$.fragment,e),_(m.$$.fragment,e),_(L.$$.fragment,e),_(C.$$.fragment,e),_(T.$$.fragment,e),_(v.$$.fragment,e),_(y.$$.fragment,e),_(w.$$.fragment,e),_(H.$$.fragment,e),_(D.$$.fragment,e),_(U.$$.fragment,e),_(X.$$.fragment,e),_(q.$$.fragment,e),_(W.$$.fragment,e),_(Z.$$.fragment,e),_(ne.$$.fragment,e),_(se.$$.fragment,e),_(ie.$$.fragment,e),_(oe.$$.fragment,e),ce=!1},d(e){e&&d(t),u(a),u(o),u(m),u(L),u(C),u(T),u(v),u(y),u(w),u(H),u(D),u(U),u(X),u(q),u(W),u(Z),u(ne),u(se),u(ie),u(oe)}}}function Ja(e){let t,n,a,s;return t=new Ie({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[Ya]},$$scope:{ctx:e}}}),a=new Ie({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[Wa]},$$scope:{ctx:e}}}),{c(){f(t.$$.fragment),n=S(),f(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(p(t.$$.fragment,e),p(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),u(t,e),u(a,e)}}}function Za(e){let t,n,a,s,o,c,g,m,$,L,E,R,C,b,P,T,h,N,M,v,I,y,x,G,w,B,H,F,O,D,K,U,V,z,X,k,q,j,Y,W;return s=new ve({props:{align:"start",$$slots:{default:[na]},$$scope:{ctx:e}}}),$=new ve({props:{align:"start",$$slots:{default:[oa]},$$scope:{ctx:e}}}),b=new ve({props:{align:"start",$$slots:{default:[fa]},$$scope:{ctx:e}}}),M=new ve({props:{align:"start",$$slots:{default:[Ca]},$$scope:{ctx:e}}}),G=new ve({props:{align:"start",$$slots:{default:[va]},$$scope:{ctx:e}}}),O=new ve({props:{align:"start",$$slots:{default:[Ha]},$$scope:{ctx:e}}}),z=new ve({props:{align:"start",$$slots:{default:[za]},$$scope:{ctx:e}}}),Y=new ve({props:{align:"start",$$slots:{default:[Ja]},$$scope:{ctx:e}}}),{c(){t=r("h1"),t.textContent="seurat_clusters - Cycling T cells",n=r("a"),n.innerHTML="",a=S(),f(s.$$.fragment),o=S(),c=r("h1"),c.textContent="seurat_clusters - Follicular helper T cells",g=r("a"),g.innerHTML="",m=S(),f($.$$.fragment),L=S(),E=r("h1"),E.textContent="seurat_clusters - gamma-delta T cells",R=r("a"),R.innerHTML="",C=S(),f(b.$$.fragment),P=S(),T=r("h1"),T.textContent="seurat_clusters - Plasmablasts",h=r("a"),h.innerHTML="",N=S(),f(M.$$.fragment),v=S(),I=r("h1"),I.textContent="seurat_clusters - Regulatory T cells",y=r("a"),y.innerHTML="",x=S(),f(G.$$.fragment),w=S(),B=r("h1"),B.textContent="seurat_clusters - Regulatory T cells.1",H=r("a"),H.innerHTML="",F=S(),f(O.$$.fragment),D=S(),K=r("h1"),K.textContent="seurat_clusters - Tcm/Naive helper T cells",U=r("a"),U.innerHTML="",V=S(),f(z.$$.fragment),X=S(),k=r("h1"),k.textContent="seurat_clusters - Tem/Trm cytotoxic T cells",q=r("a"),q.innerHTML="",j=S(),f(Y.$$.fragment),l(n,"id","prt-h1-0-seurat-clusters-cycling-t-cells"),l(n,"class","pipen-report-toc-anchor"),l(g,"id","prt-h1-1-seurat-clusters-follicular-helper-t-cells"),l(g,"class","pipen-report-toc-anchor"),l(R,"id","prt-h1-2-seurat-clusters-gamma-delta-t-cells"),l(R,"class","pipen-report-toc-anchor"),l(h,"id","prt-h1-3-seurat-clusters-plasmablasts"),l(h,"class","pipen-report-toc-anchor"),l(y,"id","prt-h1-4-seurat-clusters-regulatory-t-cells"),l(y,"class","pipen-report-toc-anchor"),l(H,"id","prt-h1-5-seurat-clusters-regulatory-t-cells-1"),l(H,"class","pipen-report-toc-anchor"),l(U,"id","prt-h1-6-seurat-clusters-tcm-naive-helper-t-cells"),l(U,"class","pipen-report-toc-anchor"),l(q,"id","prt-h1-7-seurat-clusters-tem-trm-cytotoxic-t-cells"),l(q,"class","pipen-report-toc-anchor")},m(e,r){i(e,t,r),i(e,n,r),i(e,a,r),A(s,e,r),i(e,o,r),i(e,c,r),i(e,g,r),i(e,m,r),A($,e,r),i(e,L,r),i(e,E,r),i(e,R,r),i(e,C,r),A(b,e,r),i(e,P,r),i(e,T,r),i(e,h,r),i(e,N,r),A(M,e,r),i(e,v,r),i(e,I,r),i(e,y,r),i(e,x,r),A(G,e,r),i(e,w,r),i(e,B,r),i(e,H,r),i(e,F,r),A(O,e,r),i(e,D,r),i(e,K,r),i(e,U,r),i(e,V,r),A(z,e,r),i(e,X,r),i(e,k,r),i(e,q,r),i(e,j,r),A(Y,e,r),W=!0},p(e,[t]){const n={};1&t&&(n.$$scope={dirty:t,ctx:e}),s.$set(n);const a={};1&t&&(a.$$scope={dirty:t,ctx:e}),$.$set(a);const r={};1&t&&(r.$$scope={dirty:t,ctx:e}),b.$set(r);const l={};1&t&&(l.$$scope={dirty:t,ctx:e}),M.$set(l);const i={};1&t&&(i.$$scope={dirty:t,ctx:e}),G.$set(i);const p={};1&t&&(p.$$scope={dirty:t,ctx:e}),O.$set(p);const o={};1&t&&(o.$$scope={dirty:t,ctx:e}),z.$set(o);const c={};1&t&&(c.$$scope={dirty:t,ctx:e}),Y.$set(c)},i(e){W||(p(s.$$.fragment,e),p($.$$.fragment,e),p(b.$$.fragment,e),p(M.$$.fragment,e),p(G.$$.fragment,e),p(O.$$.fragment,e),p(z.$$.fragment,e),p(Y.$$.fragment,e),W=!0)},o(e){_(s.$$.fragment,e),_($.$$.fragment,e),_(b.$$.fragment,e),_(M.$$.fragment,e),_(G.$$.fragment,e),_(O.$$.fragment,e),_(z.$$.fragment,e),_(Y.$$.fragment,e),W=!1},d(e){e&&(d(t),d(n),d(a),d(o),d(c),d(g),d(m),d(L),d(E),d(R),d(C),d(P),d(T),d(h),d(N),d(v),d(I),d(y),d(x),d(w),d(B),d(H),d(F),d(D),d(K),d(U),d(V),d(X),d(k),d(q),d(j)),u(s,e),u($,e),u(b,e),u(M,e),u(G,e),u(O,e),u(z,e),u(Y,e)}}}class Qa extends t{constructor(e){super(),n(this,e,null,Za,a,{})}}var es=[{slug:"prt-h1-0-seurat-clusters-cycling-t-cells",text:"seurat_clusters - Cycling T cells",children:[],page:0},{slug:"prt-h1-1-seurat-clusters-follicular-helper-t-cells",text:"seurat_clusters - Follicular helper T cells",children:[],page:0},{slug:"prt-h1-2-seurat-clusters-gamma-delta-t-cells",text:"seurat_clusters - gamma-delta T cells",children:[],page:0},{slug:"prt-h1-3-seurat-clusters-plasmablasts",text:"seurat_clusters - Plasmablasts",children:[],page:0},{slug:"prt-h1-4-seurat-clusters-regulatory-t-cells",text:"seurat_clusters - Regulatory T cells",children:[],page:0},{slug:"prt-h1-5-seurat-clusters-regulatory-t-cells-1",text:"seurat_clusters - Regulatory T cells.1",children:[],page:0},{slug:"prt-h1-6-seurat-clusters-tcm-naive-helper-t-cells",text:"seurat_clusters - Tcm/Naive helper T cells",children:[],page:0},{slug:"prt-h1-7-seurat-clusters-tem-trm-cytotoxic-t-cells",text:"seurat_clusters - Tem/Trm cytotoxic T cells",children:[],page:0},{slug:"prt-h1-8-seurat-clusters-tem-trm-cytotoxic-t-cells-1",text:"seurat_clusters - Tem/Trm cytotoxic T cells.1",children:[],page:1},{slug:"prt-h1-9-seurat-clusters-trm-cytotoxic-t-cells",text:"seurat_clusters - Trm cytotoxic T cells",children:[],page:1},{slug:"prt-h1-10-seurat-clusters-type-1-helper-t-cells",text:"seurat_clusters - Type 1 helper T cells",children:[],page:1},{slug:"prt-h1-11-seurat-clusters-type-1-helper-t-cells-1",text:"seurat_clusters - Type 1 helper T cells.1",children:[],page:1}],ts={pipeline:{name:"Immunopipe",desc:"Immunopipe (v1.4.2): Integrative analysis for scRNA-seq and scTCR-seq data"},versions:'\n    <a href="https://github.com/pwwang/pipen" target="_blank">\n        pipen v0.15.2\n    </a> and\n    <a href="https://github.com/pwwang/pipen-report" target="_blank">\n        pipen-report v0.20.1\n    </a>\n',entries:[{name:"SampleInfo",desc:"List sample information and perform statistics",npages:1,report_toc:!0,order:0,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n[1] ggrepel_0.9.5    ggprism_1.0.5    ggplot2_3.5.1    tidyr_1.3.1     \n[5] dplyr_1.1.4      tidyselect_1.2.1 jsonlite_1.8.8   rlang_1.1.4     \n[9] logger_0.3.0    \n\nloaded via a namespace (and not attached):\n [1] vctrs_0.6.5      cli_3.6.3        purrr_1.0.2      generics_0.1.3  \n [5] labeling_0.4.3   glue_1.7.0       colorspace_2.1-1 scales_1.3.0    \n [9] fansi_1.0.6      grid_4.3.3       munsell_0.5.1    tibble_3.2.1    \n[13] lifecycle_1.0.4  compiler_4.3.3   Rcpp_1.0.13      pkgconfig_2.0.3 \n[17] farver_2.1.2     digest_0.6.36    R6_2.5.1         utf8_1.2.4      \n[21] pillar_1.9.0     magrittr_2.0.3   withr_3.0.1      gtable_0.3.5    \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 1.73\nuser 10.62\nsys 0.10\npercent 617%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"SeuratPreparing",desc:"Load, prepare and apply QC to data, using `Seurat`",npages:1,report_toc:!0,order:2,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats4    stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] scDblFinder_1.16.0          SingleCellExperiment_1.24.0\n [3] SummarizedExperiment_1.32.0 Biobase_2.62.0             \n [5] GenomicRanges_1.54.1        GenomeInfoDb_1.38.1        \n [7] IRanges_2.36.0              S4Vectors_0.40.2           \n [9] BiocGenerics_0.48.1         MatrixGenerics_1.14.0      \n[11] matrixStats_1.1.0           dplyr_1.1.4                \n[13] ggplot2_3.5.1               bracer_1.2.1               \n[15] future_1.34.0               Seurat_5.1.0               \n[17] SeuratObject_5.0.2          sp_2.1-4                   \n[19] digest_0.6.36               jsonlite_1.8.8             \n[21] rlang_1.1.4                 logger_0.3.0               \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22          splines_4.3.3            \n  [3] later_1.3.2               BiocIO_1.12.0            \n  [5] bitops_1.0-8              tibble_3.2.1             \n  [7] R.oo_1.26.0               polyclip_1.10-7          \n  [9] XML_3.99-0.17             fastDummies_1.7.3        \n [11] lifecycle_1.0.4           edgeR_4.0.16             \n [13] globals_0.16.3            lattice_0.22-6           \n [15] MASS_7.3-60.0.1           magrittr_2.0.3           \n [17] limma_3.58.1              plotly_4.10.4            \n [19] yaml_2.3.10               metapod_1.10.0           \n [21] httpuv_1.6.15             sctransform_0.4.1        \n [23] spam_2.10-0               spatstat.sparse_3.1-0    \n [25] reticulate_1.38.0         cowplot_1.1.3            \n [27] pbapply_1.7-2             RColorBrewer_1.1-3       \n [29] abind_1.4-5               zlibbioc_1.48.0          \n [31] Rtsne_0.17                purrr_1.0.2              \n [33] R.utils_2.12.3            RCurl_1.98-1.16          \n [35] GenomeInfoDbData_1.2.11   ggrepel_0.9.5            \n [37] irlba_2.3.5.1             listenv_0.9.1            \n [39] spatstat.utils_3.0-5      goftest_1.2-3            \n [41] RSpectra_0.16-1           dqrng_0.3.2              \n [43] spatstat.random_3.2-3     fitdistrplus_1.1-11      \n [45] parallelly_1.38.0         DelayedMatrixStats_1.24.0\n [47] leiden_0.4.3.1            codetools_0.2-20         \n [49] DelayedArray_0.28.0       scuttle_1.12.0           \n [51] tidyselect_1.2.1          farver_2.1.2             \n [53] viridis_0.6.5             ScaledMatrix_1.10.0      \n [55] spatstat.explore_3.2-6    GenomicAlignments_1.38.0 \n [57] BiocNeighbors_1.20.0      progressr_0.14.0         \n [59] scater_1.30.1             ggridges_0.5.6           \n [61] survival_3.7-0            systemfonts_1.1.0        \n [63] tools_4.3.3               ragg_1.3.2               \n [65] ica_1.0-3                 Rcpp_1.0.13              \n [67] glue_1.7.0                gridExtra_2.3            \n [69] SparseArray_1.2.2         withr_3.0.1              \n [71] fastmap_1.2.0             bluster_1.12.0           \n [73] fansi_1.0.6               rsvd_1.0.5               \n [75] R6_2.5.1                  mime_0.12                \n [77] textshaping_0.4.0         colorspace_2.1-1         \n [79] scattermore_1.2           tensor_1.5               \n [81] spatstat.data_3.1-2       R.methodsS3_1.8.2        \n [83] RhpcBLASctl_0.23-42       utf8_1.2.4               \n [85] tidyr_1.3.1               generics_0.1.3           \n [87] data.table_1.15.4         rtracklayer_1.62.0       \n [89] httr_1.4.7                htmlwidgets_1.6.4        \n [91] S4Arrays_1.2.0            uwot_0.1.16              \n [93] pkgconfig_2.0.3           gtable_0.3.5             \n [95] lmtest_0.9-40             XVector_0.42.0           \n [97] htmltools_0.5.8.1         dotCall64_1.1-1          \n [99] scales_1.3.0              png_0.1-8                \n[101] harmony_1.2.0             scran_1.30.0             \n[103] reshape2_1.4.4            rjson_0.2.21             \n[105] nlme_3.1-165              zoo_1.8-12               \n[107] stringr_1.5.1             KernSmooth_2.23-24       \n[109] vipor_0.4.7               parallel_4.3.3           \n[111] miniUI_0.1.1.1            restfulr_0.0.15          \n[113] pillar_1.9.0              grid_4.3.3               \n[115] vctrs_0.6.5               RANN_2.6.1               \n[117] promises_1.3.0            BiocSingular_1.18.0      \n[119] beachmat_2.18.0           xtable_1.8-4             \n[121] cluster_2.1.6             beeswarm_0.4.0           \n[123] locfit_1.5-9.9            cli_3.6.3                \n[125] compiler_4.3.3            Rsamtools_2.18.0         \n[127] crayon_1.5.3              future.apply_1.11.2      \n[129] labeling_0.4.3            ggbeeswarm_0.7.2         \n[131] plyr_1.8.9                stringi_1.8.4            \n[133] viridisLite_0.4.2         deldir_2.0-4             \n[135] BiocParallel_1.36.0       munsell_0.5.1            \n[137] Biostrings_2.70.1         lazyeval_0.2.2           \n[139] spatstat.geom_3.2-9       Matrix_1.6-5             \n[141] RcppHNSW_0.6.0            patchwork_1.2.0          \n[143] sparseMatrixStats_1.14.0  statmod_1.5.0            \n[145] shiny_1.9.1               ROCR_1.0-11              \n[147] igraph_2.0.3              xgboost_2.1.1.1          \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 202.98\nuser 1582.98\nsys 7.06\npercent 783%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ClusterMarkersOfAllCells",desc:"Markers for clusters of all cells.",npages:3,report_toc:!0,order:4,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] UpSetR_1.4.0        ggVennDiagram_1.2.3 tidyseurat_0.8.0   \n [4] ttservice_0.4.1     future_1.34.0       ggrepel_0.9.5      \n [7] ggprism_1.0.5       ggplot2_3.5.1       enrichR_3.2        \n[10] Seurat_5.1.0        SeuratObject_5.0.2  sp_2.1-4           \n[13] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[16] tidyselect_1.2.1    digest_0.6.36       jsonlite_1.8.8     \n[19] rlang_1.1.4         logger_0.3.0       \n\nloaded via a namespace (and not attached):\n  [1] RColorBrewer_1.1-3     magrittr_2.0.3         spatstat.utils_3.0-5  \n  [4] farver_2.1.2           vctrs_0.6.5            ROCR_1.0-11           \n  [7] spatstat.explore_3.2-6 htmltools_0.5.8.1      curl_5.2.1            \n [10] sctransform_0.4.1      parallelly_1.38.0      KernSmooth_2.23-24    \n [13] htmlwidgets_1.6.4      ica_1.0-3              plyr_1.8.9            \n [16] plotly_4.10.4          zoo_1.8-12             igraph_2.0.3          \n [19] mime_0.12              lifecycle_1.0.4        pkgconfig_2.0.3       \n [22] Matrix_1.6-5           R6_2.5.1               fastmap_1.2.0         \n [25] fitdistrplus_1.1-11    shiny_1.9.1            colorspace_2.1-1      \n [28] patchwork_1.2.0        tensor_1.5             RSpectra_0.16-1       \n [31] irlba_2.3.5.1          WriteXLS_6.7.0         labeling_0.4.3        \n [34] progressr_0.14.0       fansi_1.0.6            spatstat.sparse_3.1-0 \n [37] httr_1.4.7             polyclip_1.10-7        abind_1.4-5           \n [40] compiler_4.3.3         withr_3.0.1            fastDummies_1.7.3     \n [43] MASS_7.3-60.0.1        rjson_0.2.21           tools_4.3.3           \n [46] lmtest_0.9-40          httpuv_1.6.15          future.apply_1.11.2   \n [49] goftest_1.2-3          glue_1.7.0             nlme_3.1-165          \n [52] promises_1.3.0         grid_4.3.3             Rtsne_0.17            \n [55] cluster_2.1.6          reshape2_1.4.4         generics_0.1.3        \n [58] gtable_0.3.5           spatstat.data_3.1-2    data.table_1.15.4     \n [61] utf8_1.2.4             spatstat.geom_3.2-9    RcppAnnoy_0.0.22      \n [64] RANN_2.6.1             pillar_1.9.0           stringr_1.5.1         \n [67] spam_2.10-0            RcppHNSW_0.6.0         limma_3.58.1          \n [70] later_1.3.2            splines_4.3.3          lattice_0.22-6        \n [73] survival_3.7-0         deldir_2.0-4           miniUI_0.1.1.1        \n [76] pbapply_1.7-2          gridExtra_2.3          scattermore_1.2       \n [79] statmod_1.5.0          matrixStats_1.1.0      stringi_1.8.4         \n [82] lazyeval_0.2.2         codetools_0.2-20       RVenn_1.1.0           \n [85] cli_3.6.3              uwot_0.1.16            xtable_1.8-4          \n [88] reticulate_1.38.0      munsell_0.5.1          Rcpp_1.0.13           \n [91] globals_0.16.3         spatstat.random_3.2-3  png_0.1-8             \n [94] parallel_4.3.3         ellipsis_0.3.2         presto_1.0.0          \n [97] dotCall64_1.1-1        listenv_0.9.1          viridisLite_0.4.2     \n[100] scales_1.3.0           ggridges_0.5.6         leiden_0.4.3.1        \n[103] purrr_1.0.2            cowplot_1.1.3         \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 216.89\nuser 139.48\nsys 7.23\npercent 67%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TopExpressingGenesOfAllCells",desc:"Top expressing genes for clusters of all cells.",npages:3,report_toc:!0,order:5,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ggprism_1.0.5      dplyr_1.1.4        enrichR_3.2        tibble_3.2.1      \n [5] Seurat_5.1.0       SeuratObject_5.0.2 sp_2.1-4           jsonlite_1.8.8    \n [9] rlang_1.1.4        logger_0.3.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.2-9   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] labeling_0.4.3         utf8_1.2.4             promises_1.3.0        \n [19] purrr_1.0.2            WriteXLS_6.7.0         goftest_1.2-3         \n [22] later_1.3.2            spatstat.utils_3.0-5   irlba_2.3.5.1         \n [25] parallel_4.3.3         cluster_2.1.6          R6_2.5.1              \n [28] ica_1.0-3              stringi_1.8.4          RColorBrewer_1.1-3    \n [31] spatstat.data_3.1-2    reticulate_1.38.0      parallelly_1.38.0     \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.13           \n [37] tensor_1.5             future.apply_1.11.2    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [43] splines_4.3.3          igraph_2.0.3           tidyselect_1.2.1      \n [46] abind_1.4-5            spatstat.random_3.2-3  codetools_0.2-20      \n [49] miniUI_0.1.1.1         spatstat.explore_3.2-6 curl_5.2.1            \n [52] listenv_0.9.1          lattice_0.22-6         plyr_1.8.9            \n [55] withr_3.0.1            shiny_1.9.1            ROCR_1.0-11           \n [58] Rtsne_0.17             future_1.34.0          fastDummies_1.7.3     \n [61] survival_3.7-0         polyclip_1.10-7        fitdistrplus_1.1-11   \n [64] pillar_1.9.0           KernSmooth_2.23-24     plotly_4.10.4         \n [67] generics_0.1.3         RcppHNSW_0.6.0         ggplot2_3.5.1         \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.7.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.15.4      RSpectra_0.16-1       \n [79] RANN_2.6.1             leiden_0.4.3.1         dotCall64_1.1-1       \n [82] cowplot_1.1.3          grid_4.3.3             tidyr_1.3.1           \n [85] colorspace_2.1-1       nlme_3.1-165           patchwork_1.2.0       \n [88] cli_3.6.3              spatstat.sparse_3.1-0  spam_2.10-0           \n [91] fansi_1.0.6            viridisLite_0.4.2      uwot_0.1.16           \n [94] gtable_0.3.5           digest_0.6.36          progressr_0.14.0      \n [97] ggrepel_0.9.5          farver_2.1.2           rjson_0.2.21          \n[100] htmlwidgets_1.6.4      htmltools_0.5.8.1      lifecycle_1.0.4       \n[103] httr_1.4.7             mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 52.61\nuser 38.55\nsys 2.30\npercent 77%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TCellSelection",desc:"Separate T and non-T cells and select T cells.",npages:1,report_toc:!0,order:6,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] glue_1.7.0         factoextra_1.0.7   patchwork_1.2.0    ggpubr_0.6.0      \n [5] ggrepel_0.9.5      ggprism_1.0.5      ggplot2_3.5.1      tibble_3.2.1      \n [9] tidyr_1.3.1        dplyr_1.1.4        Seurat_5.1.0       SeuratObject_5.0.2\n[13] sp_2.1-4           jsonlite_1.8.8     rlang_1.1.4        logger_0.3.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.2-9   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] backports_1.5.0        labeling_0.4.3         utf8_1.2.4            \n [19] promises_1.3.0         purrr_1.0.2            goftest_1.2-3         \n [22] later_1.3.2            spatstat.utils_3.0-5   broom_1.0.6           \n [25] irlba_2.3.5.1          parallel_4.3.3         cluster_2.1.6         \n [28] R6_2.5.1               ica_1.0-3              stringi_1.8.4         \n [31] RColorBrewer_1.1-3     spatstat.data_3.1-2    reticulate_1.38.0     \n [34] car_3.1-2              parallelly_1.38.0      lmtest_0.9-40         \n [37] scattermore_1.2        Rcpp_1.0.13            tensor_1.5            \n [40] future.apply_1.11.2    zoo_1.8-12             sctransform_0.4.1     \n [43] httpuv_1.6.15          Matrix_1.6-5           splines_4.3.3         \n [46] igraph_2.0.3           tidyselect_1.2.1       abind_1.4-5           \n [49] spatstat.random_3.2-3  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.2-6 listenv_0.9.1          lattice_0.22-6        \n [55] plyr_1.8.9             withr_3.0.1            shiny_1.9.1           \n [58] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n [61] fastDummies_1.7.3      survival_3.7-0         polyclip_1.10-7       \n [64] fitdistrplus_1.1-11    pillar_1.9.0           carData_3.0-5         \n [67] KernSmooth_2.23-24     plotly_4.10.4          generics_0.1.3        \n [70] RcppHNSW_0.6.0         munsell_0.5.1          scales_1.3.0          \n [73] globals_0.16.3         xtable_1.8-4           lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.15.4      RSpectra_0.16-1       \n [79] ggsignif_0.6.4         RANN_2.6.1             leiden_0.4.3.1        \n [82] dotCall64_1.1-1        cowplot_1.1.3          grid_4.3.3            \n [85] colorspace_2.1-1       nlme_3.1-165           cli_3.6.3             \n [88] spatstat.sparse_3.1-0  spam_2.10-0            fansi_1.0.6           \n [91] viridisLite_0.4.2      uwot_0.1.16            gtable_0.3.5          \n [94] rstatix_0.7.2          digest_0.6.36          progressr_0.14.0      \n [97] farver_2.1.2           htmlwidgets_1.6.4      htmltools_0.5.8.1     \n[100] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[103] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 38.03\nuser 45.75\nsys 0.83\npercent 122%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3435.798\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ClusterMarkers",desc:"Markers for clusters of all or selected T cells.",npages:2,report_toc:!0,order:10,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] UpSetR_1.4.0        ggVennDiagram_1.2.3 tidyseurat_0.8.0   \n [4] ttservice_0.4.1     future_1.34.0       ggrepel_0.9.5      \n [7] ggprism_1.0.5       ggplot2_3.5.1       enrichR_3.2        \n[10] Seurat_5.1.0        SeuratObject_5.0.2  sp_2.1-4           \n[13] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[16] tidyselect_1.2.1    digest_0.6.36       jsonlite_1.8.8     \n[19] rlang_1.1.4         logger_0.3.0       \n\nloaded via a namespace (and not attached):\n  [1] RColorBrewer_1.1-3     magrittr_2.0.3         spatstat.utils_3.0-5  \n  [4] farver_2.1.2           vctrs_0.6.5            ROCR_1.0-11           \n  [7] spatstat.explore_3.2-6 htmltools_0.5.8.1      curl_5.2.1            \n [10] sctransform_0.4.1      parallelly_1.38.0      KernSmooth_2.23-24    \n [13] htmlwidgets_1.6.4      ica_1.0-3              plyr_1.8.9            \n [16] plotly_4.10.4          zoo_1.8-12             igraph_2.0.3          \n [19] mime_0.12              lifecycle_1.0.4        pkgconfig_2.0.3       \n [22] Matrix_1.6-5           R6_2.5.1               fastmap_1.2.0         \n [25] fitdistrplus_1.1-11    shiny_1.9.1            colorspace_2.1-1      \n [28] patchwork_1.2.0        tensor_1.5             RSpectra_0.16-1       \n [31] irlba_2.3.5.1          WriteXLS_6.7.0         labeling_0.4.3        \n [34] progressr_0.14.0       fansi_1.0.6            spatstat.sparse_3.1-0 \n [37] httr_1.4.7             polyclip_1.10-7        abind_1.4-5           \n [40] compiler_4.3.3         withr_3.0.1            fastDummies_1.7.3     \n [43] MASS_7.3-60.0.1        rjson_0.2.21           tools_4.3.3           \n [46] lmtest_0.9-40          httpuv_1.6.15          future.apply_1.11.2   \n [49] goftest_1.2-3          glue_1.7.0             nlme_3.1-165          \n [52] promises_1.3.0         grid_4.3.3             Rtsne_0.17            \n [55] cluster_2.1.6          reshape2_1.4.4         generics_0.1.3        \n [58] gtable_0.3.5           spatstat.data_3.1-2    data.table_1.15.4     \n [61] utf8_1.2.4             spatstat.geom_3.2-9    RcppAnnoy_0.0.22      \n [64] RANN_2.6.1             pillar_1.9.0           stringr_1.5.1         \n [67] spam_2.10-0            RcppHNSW_0.6.0         limma_3.58.1          \n [70] later_1.3.2            splines_4.3.3          lattice_0.22-6        \n [73] survival_3.7-0         deldir_2.0-4           miniUI_0.1.1.1        \n [76] pbapply_1.7-2          gridExtra_2.3          scattermore_1.2       \n [79] statmod_1.5.0          matrixStats_1.1.0      stringi_1.8.4         \n [82] lazyeval_0.2.2         codetools_0.2-20       RVenn_1.1.0           \n [85] cli_3.6.3              uwot_0.1.16            xtable_1.8-4          \n [88] reticulate_1.38.0      munsell_0.5.1          Rcpp_1.0.13           \n [91] globals_0.16.3         spatstat.random_3.2-3  png_0.1-8             \n [94] parallel_4.3.3         ellipsis_0.3.2         presto_1.0.0          \n [97] dotCall64_1.1-1        listenv_0.9.1          viridisLite_0.4.2     \n[100] scales_1.3.0           ggridges_0.5.6         leiden_0.4.3.1        \n[103] purrr_1.0.2            cowplot_1.1.3         \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 136.36\nuser 90.28\nsys 4.48\npercent 69%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TopExpressingGenes",desc:"Top expressing genes for clusters of all or selected T cells.",npages:2,report_toc:!0,order:11,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ggprism_1.0.5      dplyr_1.1.4        enrichR_3.2        tibble_3.2.1      \n [5] Seurat_5.1.0       SeuratObject_5.0.2 sp_2.1-4           jsonlite_1.8.8    \n [9] rlang_1.1.4        logger_0.3.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.2-9   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] labeling_0.4.3         utf8_1.2.4             promises_1.3.0        \n [19] purrr_1.0.2            WriteXLS_6.7.0         goftest_1.2-3         \n [22] later_1.3.2            spatstat.utils_3.0-5   irlba_2.3.5.1         \n [25] parallel_4.3.3         cluster_2.1.6          R6_2.5.1              \n [28] ica_1.0-3              stringi_1.8.4          RColorBrewer_1.1-3    \n [31] spatstat.data_3.1-2    reticulate_1.38.0      parallelly_1.38.0     \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.13           \n [37] tensor_1.5             future.apply_1.11.2    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [43] splines_4.3.3          igraph_2.0.3           tidyselect_1.2.1      \n [46] abind_1.4-5            spatstat.random_3.2-3  codetools_0.2-20      \n [49] miniUI_0.1.1.1         spatstat.explore_3.2-6 curl_5.2.1            \n [52] listenv_0.9.1          lattice_0.22-6         plyr_1.8.9            \n [55] withr_3.0.1            shiny_1.9.1            ROCR_1.0-11           \n [58] Rtsne_0.17             future_1.34.0          fastDummies_1.7.3     \n [61] survival_3.7-0         polyclip_1.10-7        fitdistrplus_1.1-11   \n [64] pillar_1.9.0           KernSmooth_2.23-24     plotly_4.10.4         \n [67] generics_0.1.3         RcppHNSW_0.6.0         ggplot2_3.5.1         \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.7.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.15.4      RSpectra_0.16-1       \n [79] RANN_2.6.1             leiden_0.4.3.1         dotCall64_1.1-1       \n [82] cowplot_1.1.3          grid_4.3.3             tidyr_1.3.1           \n [85] colorspace_2.1-1       nlme_3.1-165           patchwork_1.2.0       \n [88] cli_3.6.3              spatstat.sparse_3.1-0  spam_2.10-0           \n [91] fansi_1.0.6            viridisLite_0.4.2      uwot_0.1.16           \n [94] gtable_0.3.5           digest_0.6.36          progressr_0.14.0      \n [97] ggrepel_0.9.5          farver_2.1.2           rjson_0.2.21          \n[100] htmlwidgets_1.6.4      htmltools_0.5.8.1      lifecycle_1.0.4       \n[103] httr_1.4.7             mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 31.30\nuser 27.76\nsys 1.57\npercent 93%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TCRClusterStats",desc:"Statistics of TCR clusters, generated by `TCRClustering`.",npages:1,report_toc:!0,order:13,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] grid      stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] ComplexHeatmap_2.18.0 ggprism_1.0.5         immunarch_0.9.1      \n [4] patchwork_1.2.0       data.table_1.15.4     dtplyr_1.3.1         \n [7] dplyr_1.1.4           tidyr_1.3.1           tibble_3.2.1         \n[10] ggplot2_3.5.1         jsonlite_1.8.8        rlang_1.1.4          \n[13] logger_0.3.0         \n\nloaded via a namespace (and not attached):\n  [1] gridExtra_2.3       phangorn_2.11.1     readxl_1.4.3       \n  [4] magrittr_2.0.3      clue_0.3-65         GetoptLong_1.0.5   \n  [7] matrixStats_1.1.0   compiler_4.3.3      flexmix_2.3-19     \n [10] png_0.1-8           vctrs_0.6.5         reshape2_1.4.4     \n [13] ggalluvial_0.12.5   quadprog_1.5-8      stringr_1.5.1      \n [16] crayon_1.5.3        pkgconfig_2.0.3     shape_1.4.6.1      \n [19] fastmap_1.2.0       backports_1.5.0     labeling_0.4.3     \n [22] ggraph_2.1.0        utf8_1.2.4          promises_1.3.0     \n [25] tzdb_0.4.0          UpSetR_1.4.0        purrr_1.0.2        \n [28] modeltools_0.2-23   ggseqlogo_0.2       later_1.3.2        \n [31] uuid_1.2-1          fpc_2.2-12          tweenr_2.0.3       \n [34] broom_1.0.6         parallel_4.3.3      prabclus_2.3-3     \n [37] cluster_2.1.6       R6_2.5.1            stringi_1.8.4      \n [40] RColorBrewer_1.1-3  rlist_0.4.6.2       car_3.1-2          \n [43] diptest_0.77-1      cellranger_1.1.0    Rcpp_1.0.13        \n [46] iterators_1.0.14    IRanges_2.36.0      readr_2.1.5        \n [49] httpuv_1.6.15       Matrix_1.6-5        nnet_7.3-19        \n [52] igraph_2.0.3        tidyselect_1.2.1    stringdist_0.9.12  \n [55] abind_1.4-5         viridis_0.6.5       doParallel_1.0.17  \n [58] codetools_0.2-20    lattice_0.22-6      plyr_1.8.9         \n [61] shiny_1.9.1         withr_3.0.1         polyclip_1.10-7    \n [64] circlize_0.4.16     mclust_6.1.1        kernlab_0.9-32     \n [67] pillar_1.9.0        ggpubr_0.6.0        carData_3.0-5      \n [70] foreach_1.5.2       stats4_4.3.3        generics_0.1.3     \n [73] S4Vectors_0.40.2    hms_1.1.3           munsell_0.5.1      \n [76] scales_1.3.0        xtable_1.8-4        class_7.3-22       \n [79] glue_1.7.0          pheatmap_1.0.12     tools_4.3.3        \n [82] robustbase_0.99-3   ggsignif_0.6.4      graphlayouts_1.1.0 \n [85] Cairo_1.6-2         fastmatch_1.1-4     tidygraph_1.3.0    \n [88] ape_5.8             colorspace_2.1-1    nlme_3.1-165       \n [91] ggforce_0.4.2       cli_3.6.3           fansi_1.0.6        \n [94] viridisLite_0.4.2   gtable_0.3.5        DEoptimR_1.1-3     \n [97] rstatix_0.7.2       digest_0.6.36       BiocGenerics_0.48.1\n[100] ggrepel_0.9.5       rjson_0.2.21        farver_2.1.2       \n[103] htmltools_0.5.8.1   lifecycle_1.0.4     factoextra_1.0.7   \n[106] GlobalOptions_0.1.2 mime_0.12           shinythemes_1.2.0  \n[109] MASS_7.3-60.0.1    \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 5.36\nuser 14.04\nsys 0.30\npercent 267%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TESSA",desc:"Tessa is a Bayesian model to integrate T cell receptor (TCR) sequence profiling with transcriptomes of T cells.",npages:1,report_toc:!0,order:14,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] igraph_2.0.3         Rtsne_0.17           LaplacesDemon_16.1.6\n [4] MASS_7.3-60.0.1      ggprism_1.0.5        Seurat_5.1.0        \n [7] SeuratObject_5.0.2   sp_2.1-4             tibble_3.2.1        \n[10] glue_1.7.0           immunarch_0.9.1      patchwork_1.2.0     \n[13] data.table_1.15.4    dtplyr_1.3.1         ggplot2_3.5.1       \n[16] tidyr_1.3.1          dplyr_1.1.4          jsonlite_1.8.8      \n[19] rlang_1.1.4          logger_0.3.0        \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22       shinythemes_1.2.0      splines_4.3.3         \n  [4] later_1.3.2            cellranger_1.1.0       polyclip_1.10-7       \n  [7] fastDummies_1.7.3      factoextra_1.0.7       lifecycle_1.0.4       \n [10] rstatix_0.7.2          doParallel_1.0.17      globals_0.16.3        \n [13] lattice_0.22-6         prabclus_2.3-3         backports_1.5.0       \n [16] magrittr_2.0.3         plotly_4.10.4          rlist_0.4.6.2         \n [19] httpuv_1.6.15          sctransform_0.4.1      spam_2.10-0           \n [22] flexmix_2.3-19         spatstat.sparse_3.1-0  reticulate_1.38.0     \n [25] cowplot_1.1.3          pbapply_1.7-2          RColorBrewer_1.1-3    \n [28] abind_1.4-5            quadprog_1.5-8         purrr_1.0.2           \n [31] ggraph_2.1.0           nnet_7.3-19            tweenr_2.0.3          \n [34] circlize_0.4.16        ggrepel_0.9.5          irlba_2.3.5.1         \n [37] listenv_0.9.1          spatstat.utils_3.0-5   pheatmap_1.0.12       \n [40] goftest_1.2-3          RSpectra_0.16-1        spatstat.random_3.2-3 \n [43] fitdistrplus_1.1-11    parallelly_1.38.0      leiden_0.4.3.1        \n [46] codetools_0.2-20       ggforce_0.4.2          tidyselect_1.2.1      \n [49] shape_1.4.6.1          ggseqlogo_0.2          farver_2.1.2          \n [52] viridis_0.6.5          matrixStats_1.1.0      stats4_4.3.3          \n [55] spatstat.explore_3.2-6 tidygraph_1.3.0        progressr_0.14.0      \n [58] ggridges_0.5.6         ggalluvial_0.12.5      survival_3.7-0        \n [61] iterators_1.0.14       foreach_1.5.2          tools_4.3.3           \n [64] stringdist_0.9.12      ica_1.0-3              Rcpp_1.0.13           \n [67] gridExtra_2.3          withr_3.0.1            fastmap_1.2.0         \n [70] fansi_1.0.6            digest_0.6.36          R6_2.5.1              \n [73] mime_0.12              colorspace_2.1-1       scattermore_1.2       \n [76] tensor_1.5             spatstat.data_3.1-2    diptest_0.77-1        \n [79] UpSetR_1.4.0           utf8_1.2.4             generics_0.1.3        \n [82] robustbase_0.99-3      class_7.3-22           graphlayouts_1.1.0    \n [85] httr_1.4.7             htmlwidgets_1.6.4      uwot_0.1.16           \n [88] pkgconfig_2.0.3        gtable_0.3.5           modeltools_0.2-23     \n [91] lmtest_0.9-40          htmltools_0.5.8.1      carData_3.0-5         \n [94] dotCall64_1.1-1        scales_1.3.0           png_0.1-8             \n [97] tzdb_0.4.0             reshape2_1.4.4         uuid_1.2-1            \n[100] nlme_3.1-165           zoo_1.8-12             GlobalOptions_0.1.2   \n[103] stringr_1.5.1          KernSmooth_2.23-24     parallel_4.3.3        \n[106] miniUI_0.1.1.1         pillar_1.9.0           grid_4.3.3            \n[109] vctrs_0.6.5            RANN_2.6.1             promises_1.3.0        \n[112] ggpubr_0.6.0           car_3.1-2              xtable_1.8-4          \n[115] cluster_2.1.6          readr_2.1.5            cli_3.6.3             \n[118] compiler_4.3.3         future.apply_1.11.2    ggsignif_0.6.4        \n[121] labeling_0.4.3         mclust_6.1.1           plyr_1.8.9            \n[124] stringi_1.8.4          deldir_2.0-4           viridisLite_0.4.2     \n[127] munsell_0.5.1          lazyeval_0.2.2         spatstat.geom_3.2-9   \n[130] Matrix_1.6-5           RcppHNSW_0.6.0         hms_1.1.3             \n[133] future_1.34.0          fpc_2.2-12             shiny_1.9.1           \n[136] kernlab_0.9-32         ROCR_1.0-11            broom_1.0.6           \n[139] phangorn_2.11.1        fastmatch_1.1-4        DEoptimR_1.1-3        \n[142] readxl_1.4.3           ape_5.8               \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 1774.00\nuser 8784.42\nsys 20.96\npercent 496%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"SeuratClusterStats",desc:"Statistics of the clustering.",npages:1,report_toc:!0,order:17,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] grid      stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] clustree_0.5.1        ggraph_2.1.0          ComplexHeatmap_2.18.0\n [4] circlize_0.4.16       tidyseurat_0.8.0      ttservice_0.4.1      \n [7] ggrepel_0.9.5         ggprism_1.0.5         tibble_3.2.1         \n[10] Seurat_5.1.0          SeuratObject_5.0.2    sp_2.1-4             \n[13] ggplot2_3.5.1         tidyr_1.3.1           dplyr_1.1.4          \n[16] tidyselect_1.2.1      jsonlite_1.8.8        rlang_1.1.4          \n[19] logger_0.3.0         \n\nloaded via a namespace (and not attached):\n  [1] RColorBrewer_1.1-3     shape_1.4.6.1          magrittr_2.0.3        \n  [4] ggbeeswarm_0.7.2       spatstat.utils_3.0-5   farver_2.1.2          \n  [7] GlobalOptions_0.1.2    vctrs_0.6.5            ROCR_1.0-11           \n [10] spatstat.explore_3.2-6 htmltools_0.5.8.1      sctransform_0.4.1     \n [13] parallelly_1.38.0      KernSmooth_2.23-24     htmlwidgets_1.6.4     \n [16] ica_1.0-3              plyr_1.8.9             plotly_4.10.4         \n [19] zoo_1.8-12             igraph_2.0.3           mime_0.12             \n [22] lifecycle_1.0.4        iterators_1.0.14       pkgconfig_2.0.3       \n [25] Matrix_1.6-5           R6_2.5.1               fastmap_1.2.0         \n [28] clue_0.3-65            fitdistrplus_1.1-11    future_1.34.0         \n [31] shiny_1.9.1            digest_0.6.36          colorspace_2.1-1      \n [34] patchwork_1.2.0        S4Vectors_0.40.2       tensor_1.5            \n [37] RSpectra_0.16-1        irlba_2.3.5.1          labeling_0.4.3        \n [40] progressr_0.14.0       fansi_1.0.6            spatstat.sparse_3.1-0 \n [43] httr_1.4.7             polyclip_1.10-7        abind_1.4-5           \n [46] compiler_4.3.3         withr_3.0.1            doParallel_1.0.17     \n [49] viridis_0.6.5          fastDummies_1.7.3      ggforce_0.4.2         \n [52] MASS_7.3-60.0.1        rjson_0.2.21           tools_4.3.3           \n [55] vipor_0.4.7            lmtest_0.9-40          beeswarm_0.4.0        \n [58] httpuv_1.6.15          future.apply_1.11.2    goftest_1.2-3         \n [61] glue_1.7.0             nlme_3.1-165           promises_1.3.0        \n [64] Rtsne_0.17             cluster_2.1.6          reshape2_1.4.4        \n [67] generics_0.1.3         gtable_0.3.5           spatstat.data_3.1-2   \n [70] data.table_1.15.4      tidygraph_1.3.0        utf8_1.2.4            \n [73] BiocGenerics_0.48.1    spatstat.geom_3.2-9    RcppAnnoy_0.0.22      \n [76] RANN_2.6.1             foreach_1.5.2          pillar_1.9.0          \n [79] stringr_1.5.1          spam_2.10-0            RcppHNSW_0.6.0        \n [82] later_1.3.2            splines_4.3.3          tweenr_2.0.3          \n [85] lattice_0.22-6         survival_3.7-0         deldir_2.0-4          \n [88] miniUI_0.1.1.1         pbapply_1.7-2          gridExtra_2.3         \n [91] IRanges_2.36.0         scattermore_1.2        stats4_4.3.3          \n [94] graphlayouts_1.1.0     matrixStats_1.1.0      stringi_1.8.4         \n [97] lazyeval_0.2.2         codetools_0.2-20       cli_3.6.3             \n[100] uwot_0.1.16            xtable_1.8-4           reticulate_1.38.0     \n[103] munsell_0.5.1          Rcpp_1.0.13            globals_0.16.3        \n[106] spatstat.random_3.2-3  png_0.1-8              ggrastr_1.0.2         \n[109] parallel_4.3.3         ellipsis_0.3.2         dotCall64_1.1-1       \n[112] listenv_0.9.1          viridisLite_0.4.2      scales_1.3.0          \n[115] ggridges_0.5.6         leiden_0.4.3.1         purrr_1.0.2           \n[118] crayon_1.5.3           GetoptLong_1.0.5       cowplot_1.1.3         \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 15.10\nuser 23.39\nsys 0.59\npercent 158%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"Immunarch",desc:"Exploration of Single-cell and Bulk T-cell/Antibody Immune Repertoires",npages:3,report_toc:!0,order:18,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] circlize_0.4.16   tibble_3.2.1      glue_1.7.0        ggprism_1.0.5    \n [5] immunarch_0.9.1   patchwork_1.2.0   data.table_1.15.4 dtplyr_1.3.1     \n [9] ggplot2_3.5.1     tidyr_1.3.1       dplyr_1.1.4       jsonlite_1.8.8   \n[13] rlang_1.1.4       logger_0.3.0     \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22       shinythemes_1.2.0      splines_4.3.3         \n  [4] later_1.3.2            cellranger_1.1.0       polyclip_1.10-7       \n  [7] fastDummies_1.7.3      factoextra_1.0.7       lifecycle_1.0.4       \n [10] rstatix_0.7.2          doParallel_1.0.17      globals_0.16.3        \n [13] lattice_0.22-6         prabclus_2.3-3         MASS_7.3-60.0.1       \n [16] backports_1.5.0        magrittr_2.0.3         plotly_4.10.4         \n [19] rlist_0.4.6.2          httpuv_1.6.15          Seurat_5.1.0          \n [22] sctransform_0.4.1      spam_2.10-0            flexmix_2.3-19        \n [25] spatstat.sparse_3.1-0  sp_2.1-4               reticulate_1.38.0     \n [28] cowplot_1.1.3          pbapply_1.7-2          RColorBrewer_1.1-3    \n [31] abind_1.4-5            Rtsne_0.17             quadprog_1.5-8        \n [34] purrr_1.0.2            ggraph_2.1.0           nnet_7.3-19           \n [37] tweenr_2.0.3           ggrepel_0.9.5          irlba_2.3.5.1         \n [40] listenv_0.9.1          spatstat.utils_3.0-5   pheatmap_1.0.12       \n [43] goftest_1.2-3          RSpectra_0.16-1        spatstat.random_3.2-3 \n [46] fitdistrplus_1.1-11    parallelly_1.38.0      leiden_0.4.3.1        \n [49] codetools_0.2-20       ggforce_0.4.2          tidyselect_1.2.1      \n [52] shape_1.4.6.1          ggseqlogo_0.2          farver_2.1.2          \n [55] viridis_0.6.5          matrixStats_1.1.0      stats4_4.3.3          \n [58] spatstat.explore_3.2-6 tidygraph_1.3.0        progressr_0.14.0      \n [61] ggridges_0.5.6         ggalluvial_0.12.5      survival_3.7-0        \n [64] iterators_1.0.14       foreach_1.5.2          tools_4.3.3           \n [67] stringdist_0.9.12      ica_1.0-3              Rcpp_1.0.13           \n [70] gridExtra_2.3          withr_3.0.1            fastmap_1.2.0         \n [73] fansi_1.0.6            digest_0.6.36          R6_2.5.1              \n [76] mime_0.12              colorspace_2.1-1       scattermore_1.2       \n [79] tensor_1.5             spatstat.data_3.1-2    diptest_0.77-1        \n [82] UpSetR_1.4.0           utf8_1.2.4             generics_0.1.3        \n [85] robustbase_0.99-3      class_7.3-22           graphlayouts_1.1.0    \n [88] httr_1.4.7             htmlwidgets_1.6.4      uwot_0.1.16           \n [91] pkgconfig_2.0.3        gtable_0.3.5           modeltools_0.2-23     \n [94] lmtest_0.9-40          htmltools_0.5.8.1      carData_3.0-5         \n [97] dotCall64_1.1-1        SeuratObject_5.0.2     scales_1.3.0          \n[100] png_0.1-8              tzdb_0.4.0             reshape2_1.4.4        \n[103] uuid_1.2-1             nlme_3.1-165           zoo_1.8-12            \n[106] GlobalOptions_0.1.2    stringr_1.5.1          KernSmooth_2.23-24    \n[109] parallel_4.3.3         miniUI_0.1.1.1         pillar_1.9.0          \n[112] grid_4.3.3             vctrs_0.6.5            RANN_2.6.1            \n[115] promises_1.3.0         ggpubr_0.6.0           car_3.1-2             \n[118] xtable_1.8-4           cluster_2.1.6          readr_2.1.5           \n[121] cli_3.6.3              compiler_4.3.3         future.apply_1.11.2   \n[124] ggsignif_0.6.4         labeling_0.4.3         mclust_6.1.1          \n[127] plyr_1.8.9             stringi_1.8.4          deldir_2.0-4          \n[130] viridisLite_0.4.2      assertthat_0.2.1       munsell_0.5.1         \n[133] lazyeval_0.2.2         spatstat.geom_3.2-9    Matrix_1.6-5          \n[136] RcppHNSW_0.6.0         hms_1.1.3              future_1.34.0         \n[139] fpc_2.2-12             shiny_1.9.1            kernlab_0.9-32        \n[142] ROCR_1.0-11            igraph_2.0.3           broom_1.0.6           \n[145] phangorn_2.11.1        fastmatch_1.1-4        DEoptimR_1.1-3        \n[148] readxl_1.4.3           ape_5.8               \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 91.14\nuser 113.47\nsys 1.29\npercent 125%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"RadarPlots",desc:"Radar plots for cell proportion in different clusters.",npages:1,report_toc:!0,order:19,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ggprism_1.0.5      ggradar_0.2        ggplot2_3.5.1      tibble_3.2.1      \n [5] tidyr_1.3.1        dplyr_1.1.4        Seurat_5.1.0       SeuratObject_5.0.2\n [9] sp_2.1-4           jsonlite_1.8.8     rlang_1.1.4        logger_0.3.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.2-9   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] labeling_0.4.3         utf8_1.2.4             promises_1.3.0        \n [19] purrr_1.0.2            goftest_1.2-3          later_1.3.2           \n [22] spatstat.utils_3.0-5   irlba_2.3.5.1          parallel_4.3.3        \n [25] cluster_2.1.6          R6_2.5.1               ica_1.0-3             \n [28] stringi_1.8.4          RColorBrewer_1.1-3     spatstat.data_3.1-2   \n [31] reticulate_1.38.0      parallelly_1.38.0      lmtest_0.9-40         \n [34] scattermore_1.2        Rcpp_1.0.13            tensor_1.5            \n [37] future.apply_1.11.2    zoo_1.8-12             sctransform_0.4.1     \n [40] httpuv_1.6.15          Matrix_1.6-5           splines_4.3.3         \n [43] igraph_2.0.3           tidyselect_1.2.1       abind_1.4-5           \n [46] spatstat.random_3.2-3  codetools_0.2-20       miniUI_0.1.1.1        \n [49] spatstat.explore_3.2-6 listenv_0.9.1          lattice_0.22-6        \n [52] plyr_1.8.9             withr_3.0.1            shiny_1.9.1           \n [55] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n [58] fastDummies_1.7.3      survival_3.7-0         polyclip_1.10-7       \n [61] fitdistrplus_1.1-11    pillar_1.9.0           KernSmooth_2.23-24    \n [64] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n [67] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [70] xtable_1.8-4           glue_1.7.0             lazyeval_0.2.2        \n [73] tools_4.3.3            data.table_1.15.4      RSpectra_0.16-1       \n [76] RANN_2.6.1             leiden_0.4.3.1         dotCall64_1.1-1       \n [79] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n [82] nlme_3.1-165           patchwork_1.2.0        cli_3.6.3             \n [85] spatstat.sparse_3.1-0  spam_2.10-0            fansi_1.0.6           \n [88] viridisLite_0.4.2      uwot_0.1.16            gtable_0.3.5          \n [91] digest_0.6.36          progressr_0.14.0       ggrepel_0.9.5         \n [94] farver_2.1.2           htmlwidgets_1.6.4      htmltools_0.5.8.1     \n [97] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[100] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 7.79\nuser 16.36\nsys 0.40\npercent 215%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"CloneResidency",desc:"Identification of clone residency",npages:1,report_toc:!0,order:20,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ComplexUpset_1.3.3  ggVennDiagram_1.2.3 ggprism_1.0.5      \n [4] ggplot2_3.5.1       ggnewscale_0.5.0    glue_1.7.0         \n [7] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[10] jsonlite_1.8.8      rlang_1.1.4         logger_0.3.0       \n\nloaded via a namespace (and not attached):\n  [1] DBI_1.2.3              deldir_2.0-4           pbapply_1.7-2         \n  [4] gridExtra_2.3          magrittr_2.0.3         RcppAnnoy_0.0.22      \n  [7] e1071_1.7-14           spatstat.geom_3.2-9    matrixStats_1.1.0     \n [10] ggridges_0.5.6         compiler_4.3.3         png_0.1-8             \n [13] vctrs_0.6.5            reshape2_1.4.4         stringr_1.5.1         \n [16] pkgconfig_2.0.3        fastmap_1.2.0          labeling_0.4.3        \n [19] utf8_1.2.4             promises_1.3.0         purrr_1.0.2           \n [22] RVenn_1.1.0            goftest_1.2-3          later_1.3.2           \n [25] spatstat.utils_3.0-5   irlba_2.3.5.1          parallel_4.3.3        \n [28] cluster_2.1.6          R6_2.5.1               ica_1.0-3             \n [31] stringi_1.8.4          RColorBrewer_1.1-3     spatstat.data_3.1-2   \n [34] reticulate_1.38.0      parallelly_1.38.0      lmtest_0.9-40         \n [37] scattermore_1.2        Rcpp_1.0.13            tensor_1.5            \n [40] future.apply_1.11.2    zoo_1.8-12             sctransform_0.4.1     \n [43] httpuv_1.6.15          Matrix_1.6-5           splines_4.3.3         \n [46] igraph_2.0.3           tidyselect_1.2.1       abind_1.4-5           \n [49] spatstat.random_3.2-3  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.2-6 listenv_0.9.1          lattice_0.22-6        \n [55] plyr_1.8.9             shiny_1.9.1            withr_3.0.1           \n [58] ROCR_1.0-11            Rtsne_0.17             sf_1.0-16             \n [61] future_1.34.0          fastDummies_1.7.3      survival_3.7-0        \n [64] units_0.8-5            proxy_0.4-27           polyclip_1.10-7       \n [67] fitdistrplus_1.1-11    pillar_1.9.0           Seurat_5.1.0          \n [70] KernSmooth_2.23-24     plotly_4.10.4          generics_0.1.3        \n [73] RcppHNSW_0.6.0         sp_2.1-4               munsell_0.5.1         \n [76] scales_1.3.0           globals_0.16.3         xtable_1.8-4          \n [79] class_7.3-22           lazyeval_0.2.2         tools_4.3.3           \n [82] data.table_1.15.4      RSpectra_0.16-1        RANN_2.6.1            \n [85] leiden_0.4.3.1         dotCall64_1.1-1        cowplot_1.1.3         \n [88] grid_4.3.3             colorspace_2.1-1       nlme_3.1-165          \n [91] patchwork_1.2.0        cli_3.6.3              spatstat.sparse_3.1-0 \n [94] spam_2.10-0            fansi_1.0.6            viridisLite_0.4.2     \n [97] uwot_0.1.16            gtable_0.3.5           digest_0.6.36         \n[100] classInt_0.4-10        progressr_0.14.0       ggrepel_0.9.5         \n[103] farver_2.1.2           htmlwidgets_1.6.4      SeuratObject_5.0.2    \n[106] htmltools_0.5.8.1      lifecycle_1.0.4        httr_1.4.7            \n[109] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 16.51\nuser 24.80\nsys 0.51\npercent 153%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ScFGSEA",desc:"Gene set enrichment analysis for cells in different groups using `fgsea`",npages:2,report_toc:!0,order:21,runinfo:{session:"# Generated by pipen_runinfo v0.8.0\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.27.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] fgsea_1.28.0       data.table_1.15.4  matrixStats_1.1.0  tidyseurat_0.8.0  \n [5] ttservice_0.4.1    Seurat_5.1.0       SeuratObject_5.0.2 sp_2.1-4          \n [9] tidyr_1.3.1        tidyselect_1.2.1   tibble_3.2.1       dplyr_1.1.4       \n[13] ggplot2_3.5.1      jsonlite_1.8.8     rlang_1.1.4        logger_0.3.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       ggridges_0.5.6        \n  [7] compiler_4.3.3         spatstat.geom_3.2-9    png_0.1-8             \n [10] vctrs_0.6.5            reshape2_1.4.4         stringr_1.5.1         \n [13] pkgconfig_2.0.3        fastmap_1.2.0          ellipsis_0.3.2        \n [16] labeling_0.4.3         utf8_1.2.4             promises_1.3.0        \n [19] purrr_1.0.2            goftest_1.2-3          later_1.3.2           \n [22] BiocParallel_1.36.0    spatstat.utils_3.0-5   irlba_2.3.5.1         \n [25] parallel_4.3.3         cluster_2.1.6          R6_2.5.1              \n [28] ica_1.0-3              stringi_1.8.4          RColorBrewer_1.1-3    \n [31] spatstat.data_3.1-2    reticulate_1.38.0      parallelly_1.38.0     \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.13           \n [37] tensor_1.5             future.apply_1.11.2    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [43] splines_4.3.3          igraph_2.0.3           abind_1.4-5           \n [46] spatstat.random_3.2-3  codetools_0.2-20       miniUI_0.1.1.1        \n [49] spatstat.explore_3.2-6 listenv_0.9.1          lattice_0.22-6        \n [52] plyr_1.8.9             shiny_1.9.1            withr_3.0.1           \n [55] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n [58] fastDummies_1.7.3      survival_3.7-0         polyclip_1.10-7       \n [61] fitdistrplus_1.1-11    pillar_1.9.0           KernSmooth_2.23-24    \n [64] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n [67] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [70] xtable_1.8-4           glue_1.7.0             lazyeval_0.2.2        \n [73] tools_4.3.3            RSpectra_0.16-1        RANN_2.6.1            \n [76] leiden_0.4.3.1         dotCall64_1.1-1        fastmatch_1.1-4       \n [79] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n [82] nlme_3.1-165           patchwork_1.2.0        cli_3.6.3             \n [85] spatstat.sparse_3.1-0  spam_2.10-0            fansi_1.0.6           \n [88] viridisLite_0.4.2      uwot_0.1.16            gtable_0.3.5          \n [91] digest_0.6.36          progressr_0.14.0       ggrepel_0.9.5         \n [94] farver_2.1.2           htmlwidgets_1.6.4      htmltools_0.5.8.1     \n [97] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[100] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.0\n\nreal 96.79\nuser 103.64\nsys 1.43\npercent 108%\n",device:"# Generated by pipen-runinfo v0.8.0\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                       x86_64\nCPU op-mode(s):                     32-bit, 64-bit\nByte Order:                         Little Endian\nAddress sizes:                      46 bits physical, 57 bits virtual\nCPU(s):                             96\nOn-line CPU(s) list:                0-95\nThread(s) per core:                 2\nCore(s) per socket:                 24\nSocket(s):                          2\nNUMA node(s):                       2\nVendor ID:                          GenuineIntel\nCPU family:                         6\nModel:                              106\nModel name:                         Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                           6\nCPU MHz:                            3500.000\nCPU max MHz:                        3500.0000\nCPU min MHz:                        800.0000\nBogoMIPS:                           5600.00\nVirtualization:                     VT-x\nL1d cache:                          2.3 MiB\nL1i cache:                          1.5 MiB\nL2 cache:                           60 MiB\nL3 cache:                           72 MiB\nNUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling: Mitigation; Microcode\nVulnerability Itlb multihit:        Not affected\nVulnerability L1tf:                 Not affected\nVulnerability Mds:                  Not affected\nVulnerability Meltdown:             Not affected\nVulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Retbleed:             Not affected\nVulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence\nVulnerability Srbds:                Not affected\nVulnerability Tsx async abort:      Not affected\nFlags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"MarkersFinder",desc:"Find markers between different groups of cells",npages:1,report_toc:!0,order:22},{name:"MetaMarkers",desc:"Find markers between three or more groups of cells, using one-way ANOVA or Kruskal-Wallis test.",npages:1,report_toc:!0,order:23},{name:"CellsDistribution",desc:"Distribution of cells (i.e. in a TCR clone) from different groups for each cluster",npages:1,report_toc:!0,order:24},{name:"CDR3AAPhyschem",desc:"CDR3 AA physicochemical feature analysis",npages:1,report_toc:!0,order:25},{name:"ScrnaMetabolicLandscape",desc:"Metabolic landscape analysis for scRNA-seq data",order:28,procs:[{name:"MetabolicPathwayActivity",desc:"This process calculates the pathway activities in different groups and subsets.",npages:1,report_toc:!0,order:28},{name:"MetabolicFeatures",desc:"This process performs enrichment analysis for the metabolic pathways for each group in each subset.",npages:1,report_toc:!0,order:29},{name:"MetabolicFeaturesIntraSubset",desc:"Intra-subset metabolic features - Enrichment analysis in details",npages:1,report_toc:!0,order:30},{name:"MetabolicPathwayHeterogeneity",desc:"Calculate Metabolic Pathway heterogeneity.",npages:1,report_toc:!0,order:31}]}]};function ns(e){let t,n;return t=new Ut({props:{dir:"up"}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function as(e){let t,n;return t=new Ut({props:{dir:"down"}}),{c(){f(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},i(e){n||(p(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){u(t,e)}}}function ss(e){let t,n,a,s,r,l=e[1]>0&&ns();n=new Qa({});let g=e[1]<e[3].npages-1&&as();return{c(){l&&l.c(),t=S(),f(n.$$.fragment),a=S(),g&&g.c(),s=$()},m(e,p){l&&l.m(e,p),i(e,t,p),A(n,e,p),i(e,a,p),g&&g.m(e,p),i(e,s,p),r=!0},p(e,n){e[1]>0?l?2&n&&p(l,1):(l=ns(),l.c(),p(l,1),l.m(t.parentNode,t)):l&&(o(),_(l,1,1,(()=>{l=null})),c()),e[1]<e[3].npages-1?g?10&n&&p(g,1):(g=as(),g.c(),p(g,1),g.m(s.parentNode,s)):g&&(o(),_(g,1,1,(()=>{g=null})),c())},i(e){r||(p(l),p(n.$$.fragment,e),p(g),r=!0)},o(e){_(l),_(n.$$.fragment,e),_(g),r=!1},d(e){e&&(d(t),d(a),d(s)),l&&l.d(e),u(n,e),g&&g.d(e)}}}function rs(e){let t,n,a;function s(t){e[4](t)}let r={logo:e[0],logotext:e[3].desc,versions:ts.versions,entries:ts.entries,pipeline_name:ts.pipeline.name,report_toc:e[3].report_toc,runinfo:e[3].runinfo,toc:es,page:e[1],$$slots:{default:[ss]},$$scope:{ctx:e}};return void 0!==e[2]&&(r.isSideNavOpen=e[2]),t=new Ft({props:r}),L.push((()=>E(t,"isSideNavOpen",s))),{c(){f(t.$$.fragment)},m(e,n){A(t,e,n),a=!0},p(e,[a]){const s={};1&a&&(s.logo=e[0]),8&a&&(s.logotext=e[3].desc),8&a&&(s.report_toc=e[3].report_toc),8&a&&(s.runinfo=e[3].runinfo),2&a&&(s.page=e[1]),42&a&&(s.$$scope={dirty:a,ctx:e}),!n&&4&a&&(n=!0,s.isSideNavOpen=e[2],R((()=>n=!1))),t.$set(s)},i(e){a||(p(t.$$.fragment,e),a=!0)},o(e){_(t.$$.fragment,e),a=!1},d(e){u(t,e)}}}function ls(e,t,n){let a,{name:s}=t,{page:r=0}=t,l=!1;for(const e of ts.entries)if(e.procs){for(const t of e.procs)if(t.name==s){a=t;break}}else if(e.name==s){a=e;break}return m((()=>{const e=new URL(window.location.href);let t;e.hash.length>1&&(t=document.getElementById(`${e.hash.slice(1)}`)),t&&t.scrollIntoView({block:"start",behavior:"smooth"})})),e.$$set=e=>{"name"in e&&n(0,s=e.name),"page"in e&&n(1,r=e.page)},[s,r,l,a,function(e){l=e,n(2,l)}]}e("default",new class extends t{constructor(e){super(),n(this,e,ls,rs,a,{name:0,page:1})}}({target:document.body,props:{name:"ScFGSEA",page:0}}))}}}));
