System.register(["./vendor/svelte-C_apNiMW.js","./vendor/carbon-components-svelte-CB8VNYQX.js","./vendor/carbon-icons-svelte-DkM4cpjm.js"],(function(e){"use strict";var t,n,a,s,r,l,i,o,p,c,_,d,g,m,f,u,A,$,S,E,L,R,h,C,T,N,b,P,M,v,I,y,G,x,w,H,F,B,O,D,K,U,V,z,k,X,q,j,Y,W,J,Z,Q,ee,te,ne,ae,se,re,le,ie,oe,pe,ce,_e,de,ge,me,fe,ue,Ae,$e,Se,Ee,Le,Re,he,Ce,Te,Ne,be,Pe,Me,ve,Ie,ye,Ge,xe,we,He,Fe,Be,Oe,De,Ke,Ue,Ve,ze,ke,Xe,qe,je,Ye,We,Je,Ze,Qe;return{setters:[function(e){t=e.S,n=e.i,a=e.s,s=e.a1,r=e.l,l=e.O,i=e.a,o=e.b,p=e.g,c=e.c,_=e.t,d=e.d,g=e.a5,m=e.B,f=e.e,u=e.G,A=e.H,$=e.I,S=e.y,E=e.K,L=e.T,R=e.U,h=e.L,C=e.D,T=e.M,N=e.z,b=e.Q,P=e.k,M=e.u,v=e.n,I=e.o,y=e.h,G=e.p,x=e.a0,w=e.f,H=e.j,F=e.A,B=e.m,O=e.q,D=e.aa,K=e.ab,U=e.x,V=e.W,z=e.ac,k=e.R,X=e.$,q=e.v,j=e.Y,Y=e.Z,W=e.F,J=e.ad,Z=e.w,Q=e.J,ee=e.r},function(e){te=e.S,ne=e.a,ae=e.b,se=e.H,re=e.c,le=e.T,ie=e.d,oe=e.e,pe=e.f,ce=e.g,_e=e.h,de=e.M,ge=e.I,me=e.i,fe=e.j,ue=e.k,Ae=e.C,$e=e.l,Se=e.m,Ee=e.G,Le=e.R,Re=e.n,he=e.B,Ce=e.o,Te=e.A,Ne=e.p,be=e.D,Pe=e.P,Me=e.q,ve=e.r,Ie=e.s,ye=e.t,Ge=e.u},function(e){xe=e.C,we=e.a,He=e.D,Fe=e.A,Be=e.S,Oe=e.I,De=e.b,Ke=e.c,Ue=e.V,Ve=e.d,ze=e.T,ke=e.G,Xe=e.J,qe=e.P,je=e.e,Ye=e.f,We=e.O,Je=e.g,Ze=e.h,Qe=e.i}],execute:function(){function et(e,t,n){const a=e.slice();return a[4]=t[n],a}function tt(e,t,n){const a=e.slice();return a[7]=t[n],a}function nt(e){let t,n;return t=new te({props:{href:`${e[1](e[4].page)}#${e[4].slug}`,class:e[2](e[4].page),title:e[4].text,text:e[4].text,expanded:!0,$$slots:{default:[rt]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.href=`${e[1](e[4].page)}#${e[4].slug}`),1&n&&(a.class=e[2](e[4].page)),1&n&&(a.title=e[4].text),1&n&&(a.text=e[4].text),1025&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function at(e){let t,n;return t=new ne({props:{href:`${e[1](e[4].page)}#${e[4].slug}`,class:e[2](e[4].page),title:e[4].text,text:e[4].text}}),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.href=`${e[1](e[4].page)}#${e[4].slug}`),1&n&&(a.class=e[2](e[4].page)),1&n&&(a.title=e[4].text),1&n&&(a.text=e[4].text),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function st(e){let t,n;return t=new ae({props:{href:`${e[1](e[7].page)}#${e[7].slug}`,class:e[2](e[7].page),title:e[7].text,text:e[7].text}}),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.href=`${e[1](e[7].page)}#${e[7].slug}`),1&n&&(a.class=e[2](e[7].page)),1&n&&(a.title=e[7].text),1&n&&(a.text=e[7].text),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function rt(e){let t,n,a=s(e[4].children),r=[];for(let t=0;t<a.length;t+=1)r[t]=st(tt(e,a,t));const l=e=>_(r[e],1,1,(()=>{r[e]=null}));return{c(){for(let e=0;e<r.length;e+=1)r[e].c();t=S()},m(e,a){for(let t=0;t<r.length;t+=1)r[t]&&r[t].m(e,a);i(e,t,a),n=!0},p(e,n){if(7&n){let i;for(a=s(e[4].children),i=0;i<a.length;i+=1){const s=tt(e,a,i);r[i]?(r[i].p(s,n),o(r[i],1)):(r[i]=st(s),r[i].c(),o(r[i],1),r[i].m(t.parentNode,t))}for(p(),i=a.length;i<r.length;i+=1)l(i);c()}},i(e){if(!n){for(let e=0;e<a.length;e+=1)o(r[e]);n=!0}},o(e){r=r.filter(Boolean);for(let e=0;e<r.length;e+=1)_(r[e]);n=!1},d(e){e&&d(t),g(r,e)}}}function lt(e){let t,n,a,s;const r=[at,nt],l=[];function g(e,t){return 0==e[4].children.length?0:1}return t=g(e),n=l[t]=r[t](e),{c(){n.c(),a=f()},m(e,n){l[t].m(e,n),i(e,a,n),s=!0},p(e,s){let i=t;t=g(e),t===i?l[t].p(e,s):(p(),_(l[i],1,1,(()=>{l[i]=null})),c(),n=l[t],n?n.p(e,s):(n=l[t]=r[t](e),n.c()),o(n,1),n.m(a.parentNode,a))},i(e){s||(o(n),s=!0)},o(e){_(n),s=!1},d(e){e&&d(a),l[t].d(e)}}}function it(e){let t,n,a=s(e[0]),m=[];for(let t=0;t<a.length;t+=1)m[t]=lt(et(e,a,t));const f=e=>_(m[e],1,1,(()=>{m[e]=null}));return{c(){t=r("ul");for(let e=0;e<m.length;e+=1)m[e].c();l(t,"class","bx--side-nav__items pipen-report-sidenav svelte-12gpq3")},m(e,a){i(e,t,a);for(let e=0;e<m.length;e+=1)m[e]&&m[e].m(t,null);n=!0},p(e,[n]){if(7&n){let r;for(a=s(e[0]),r=0;r<a.length;r+=1){const s=et(e,a,r);m[r]?(m[r].p(s,n),o(m[r],1)):(m[r]=lt(s),m[r].c(),o(m[r],1),m[r].m(t,null))}for(p(),r=a.length;r<m.length;r+=1)f(r);c()}},i(e){if(!n){for(let e=0;e<a.length;e+=1)o(m[e]);n=!0}},o(e){m=m.filter(Boolean);for(let e=0;e<m.length;e+=1)_(m[e]);n=!1},d(e){e&&d(t),g(m,e)}}}function ot(e,t,n){let{toc:a}=t,{page:s}=t;return m((()=>{let e,t=new URL(window.location.href);e=t.hash.length>0?document.querySelector(`[href="${t.hash}"]`):document.querySelector(".toc-current-page"),e.scrollIntoView({behavior:"smooth"})})),e.$$set=e=>{"toc"in e&&n(0,a=e.toc),"page"in e&&n(3,s=e.page)},[a,e=>{if(s===e)return"";let t=new URL(window.location.href);return t.hash="",t.searchParams.set("page",e),t.toString()},e=>e!=s?"toc-noncurrent-page":"toc-current-page",s]}class pt extends t{constructor(e){super(),n(this,e,ot,it,a,{toc:0,page:3})}}function ct(e,t,n){const a=e.slice();return a[7]=t[n],a}function _t(e,t,n){const a=e.slice();return a[10]=t[n],a}function dt(e){let t,n;return t=new oe({props:{isOpen:It,transition:e[5],$$slots:{default:[Ct]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};8197&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function gt(e){let t;return{c(){t=h("Go to home:")},m(e,n){i(e,t,n)},d(e){e&&d(t)}}}function mt(e){let t;return{c(){t=h(e[0])},m(e,n){i(e,t,n)},p(e,n){1&n&&T(t,e[0])},d(e){e&&d(t)}}}function ft(e){let t;return{c(){t=h("Switch to process:")},m(e,n){i(e,t,n)},d(e){e&&d(t)}}}function ut(e){let t,n;return t=new _e({props:{title:e[7].name,href:`?proc=${e[7].name}`,$$slots:{default:[$t]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};4&n&&(a.title=e[7].name),4&n&&(a.href=`?proc=${e[7].name}`),8196&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function At(e){let t,n,a,r;t=new _e({props:{class:"procgroup",$$slots:{default:[St]},$$scope:{ctx:e}}});let l=s(e[7].procs),m=[];for(let t=0;t<l.length;t+=1)m[t]=Lt(_t(e,l,t));const E=e=>_(m[e],1,1,(()=>{m[e]=null}));return{c(){u(t.$$.fragment),n=S();for(let e=0;e<m.length;e+=1)m[e].c();a=f()},m(e,s){A(t,e,s),i(e,n,s);for(let t=0;t<m.length;t+=1)m[t]&&m[t].m(e,s);i(e,a,s),r=!0},p(e,n){const r={};if(8196&n&&(r.$$scope={dirty:n,ctx:e}),t.$set(r),4&n){let t;for(l=s(e[7].procs),t=0;t<l.length;t+=1){const s=_t(e,l,t);m[t]?(m[t].p(s,n),o(m[t],1)):(m[t]=Lt(s),m[t].c(),o(m[t],1),m[t].m(a.parentNode,a))}for(p(),t=l.length;t<m.length;t+=1)E(t);c()}},i(e){if(!r){o(t.$$.fragment,e);for(let e=0;e<l.length;e+=1)o(m[e]);r=!0}},o(e){_(t.$$.fragment,e),m=m.filter(Boolean);for(let e=0;e<m.length;e+=1)_(m[e]);r=!1},d(e){e&&(d(n),d(a)),$(t,e),g(m,e)}}}function $t(e){let t,n,a=e[7].name+"";return{c(){t=h(a),n=S()},m(e,a){i(e,t,a),i(e,n,a)},p(e,n){4&n&&a!==(a=e[7].name+"")&&T(t,a)},d(e){e&&(d(t),d(n))}}}function St(e){let t,n=e[7].name+"";return{c(){t=h(n)},m(e,n){i(e,t,n)},p(e,a){4&a&&n!==(n=e[7].name+"")&&T(t,n)},d(e){e&&d(t)}}}function Et(e){let t,n,a,s=e[10].name+"";return{c(){t=h("- "),n=h(s),a=S()},m(e,s){i(e,t,s),i(e,n,s),i(e,a,s)},p(e,t){4&t&&s!==(s=e[10].name+"")&&T(n,s)},d(e){e&&(d(t),d(n),d(a))}}}function Lt(e){let t,n;return t=new _e({props:{title:e[10].name,class:"procgroup-proc",href:`?proc=${e[10].name}`,$$slots:{default:[Et]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};4&n&&(a.title=e[10].name),4&n&&(a.href=`?proc=${e[10].name}`),8196&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Rt(e){let t,n,a,s;const r=[At,ut],l=[];function g(e,t){return e[7].procs?0:1}return t=g(e),n=l[t]=r[t](e),{c(){n.c(),a=f()},m(e,n){l[t].m(e,n),i(e,a,n),s=!0},p(e,s){let i=t;t=g(e),t===i?l[t].p(e,s):(p(),_(l[i],1,1,(()=>{l[i]=null})),c(),n=l[t],n?n.p(e,s):(n=l[t]=r[t](e),n.c()),o(n,1),n.m(a.parentNode,a))},i(e){s||(o(n),s=!0)},o(e){_(n),s=!1},d(e){e&&d(a),l[t].d(e)}}}function ht(e){let t,n,a,r,l,m,E,L;t=new ce({props:{$$slots:{default:[gt]},$$scope:{ctx:e}}}),a=new _e({props:{href:"index.html",$$slots:{default:[mt]},$$scope:{ctx:e}}}),l=new ce({props:{$$slots:{default:[ft]},$$scope:{ctx:e}}});let R=s(e[2]),h=[];for(let t=0;t<R.length;t+=1)h[t]=Rt(ct(e,R,t));const C=e=>_(h[e],1,1,(()=>{h[e]=null}));return{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment),r=S(),u(l.$$.fragment),m=S();for(let e=0;e<h.length;e+=1)h[e].c();E=f()},m(e,s){A(t,e,s),i(e,n,s),A(a,e,s),i(e,r,s),A(l,e,s),i(e,m,s);for(let t=0;t<h.length;t+=1)h[t]&&h[t].m(e,s);i(e,E,s),L=!0},p(e,n){const r={};8192&n&&(r.$$scope={dirty:n,ctx:e}),t.$set(r);const i={};8193&n&&(i.$$scope={dirty:n,ctx:e}),a.$set(i);const _={};if(8192&n&&(_.$$scope={dirty:n,ctx:e}),l.$set(_),4&n){let t;for(R=s(e[2]),t=0;t<R.length;t+=1){const a=ct(e,R,t);h[t]?(h[t].p(a,n),o(h[t],1)):(h[t]=Rt(a),h[t].c(),o(h[t],1),h[t].m(E.parentNode,E))}for(p(),t=R.length;t<h.length;t+=1)C(t);c()}},i(e){if(!L){o(t.$$.fragment,e),o(a.$$.fragment,e),o(l.$$.fragment,e);for(let e=0;e<R.length;e+=1)o(h[e]);L=!0}},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),_(l.$$.fragment,e),h=h.filter(Boolean);for(let e=0;e<h.length;e+=1)_(h[e]);L=!1},d(e){e&&(d(n),d(r),d(m),d(E)),$(t,e),$(a,e),$(l,e),g(h,e)}}}function Ct(e){let t,n;return t=new pe({props:{$$slots:{default:[ht]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};8197&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Tt(e){let t,n,a,s;t=new ie({props:{render:"select",theme:"g10",persist:!0,persistKey:"__pipen-report-theme",select:{themes:["g10","white","g80","g90","g100"],labelText:"Theme",inline:!0}}});let r=e[2].length>0&&dt(e);return{c(){u(t.$$.fragment),n=S(),r&&r.c(),a=f()},m(e,l){A(t,e,l),i(e,n,l),r&&r.m(e,l),i(e,a,l),s=!0},p(e,t){e[2].length>0?r?(r.p(e,t),4&t&&o(r,1)):(r=dt(e),r.c(),o(r,1),r.m(a.parentNode,a)):r&&(p(),_(r,1,1,(()=>{r=null})),c())},i(e){s||(o(t.$$.fragment,e),o(r),s=!0)},o(e){_(t.$$.fragment,e),_(r),s=!1},d(e){e&&(d(n),d(a)),$(t,e),r&&r.d(e)}}}function Nt(e){let t,n;return t=new re({props:{class:"header-util",$$slots:{default:[Tt]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};8197&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function bt(e){let t;return{c(){t=h(e[4])},m(e,n){i(e,t,n)},p(e,n){16&n&&T(t,e[4])},d(e){e&&d(t)}}}function Pt(e){let t,n,a,s,l=null!==e[4]&&bt(e);return{c(){t=r("h1"),n=h(e[3]),a=S(),l&&l.c(),s=f()},m(e,r){i(e,t,r),C(t,n),i(e,a,r),l&&l.m(e,r),i(e,s,r)},p(e,t){8&t&&T(n,e[3]),null!==e[4]?l?l.p(e,t):(l=bt(e),l.c(),l.m(s.parentNode,s)):l&&(l.d(1),l=null)},d(e){e&&(d(t),d(a),d(s)),l&&l.d(e)}}}function Mt(e){let t,n;return t=new le({props:{class:"header-tile",slot:"platform",$$slots:{default:[Pt]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};8216&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function vt(e){let t,n,a;function s(t){e[6](t)}let r={$$slots:{platform:[Mt],default:[Nt]},$$scope:{ctx:e}};return void 0!==e[1]&&(r.isSideNavOpen=e[1]),t=new se({props:r}),E.push((()=>L(t,"isSideNavOpen",s))),{c(){u(t.$$.fragment)},m(e,n){A(t,e,n),a=!0},p(e,[a]){const s={};8221&a&&(s.$$scope={dirty:a,ctx:e}),!n&&2&a&&(n=!0,s.isSideNavOpen=e[1],R((()=>n=!1))),t.$set(s)},i(e){a||(o(t.$$.fragment,e),a=!0)},o(e){_(t.$$.fragment,e),a=!1},d(e){$(t,e)}}}let It=!1;function yt(e,t,n){let{entries:a}=t,{logo:s}=t,{logotext:r=null}=t,{pipeline_name:l}=t,{isSideNavOpen:i}=t;return l=l||s,e.$$set=e=>{"entries"in e&&n(2,a=e.entries),"logo"in e&&n(3,s=e.logo),"logotext"in e&&n(4,r=e.logotext),"pipeline_name"in e&&n(0,l=e.pipeline_name),"isSideNavOpen"in e&&n(1,i=e.isSideNavOpen)},[l,i,a,s,r,{duration:200},function(e){i=e,n(1,i)}]}class Gt extends t{constructor(e){super(),n(this,e,yt,vt,a,{entries:2,logo:3,logotext:4,pipeline_name:0,isSideNavOpen:1})}}function xt(e){let t,n,a;function s(t){e[3](t)}let r={modalHeading:"Running Information",passiveModal:!0,hasScrollingContent:!0,class:"runinfo",$$slots:{default:[Dt]},$$scope:{ctx:e}};return void 0!==e[2]&&(r.open=e[2]),t=new de({props:r}),E.push((()=>L(t,"open",s))),{c(){u(t.$$.fragment)},m(e,n){A(t,e,n),a=!0},p(e,a){const s={};34&a&&(s.$$scope={dirty:a,ctx:e}),!n&&4&a&&(n=!0,s.open=e[2],R((()=>n=!1))),t.$set(s)},i(e){a||(o(t.$$.fragment,e),a=!0)},o(e){_(t.$$.fragment,e),a=!1},d(e){$(t,e)}}}function wt(e){let t,n,a,s,r,l;return t=new fe({props:{label:"Session Information"}}),a=new fe({props:{label:"Time Spent"}}),r=new fe({props:{label:"Device Information"}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment),s=S(),u(r.$$.fragment)},m(e,o){A(t,e,o),i(e,n,o),A(a,e,o),i(e,s,o),A(r,e,o),l=!0},p:b,i(e){l||(o(t.$$.fragment,e),o(a.$$.fragment,e),o(r.$$.fragment,e),l=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),_(r.$$.fragment,e),l=!1},d(e){e&&(d(n),d(s)),$(t,e),$(a,e),$(r,e)}}}function Ht(e){let t,n;return t=new Ae({props:{type:"multi",code:e[1].session}}),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};2&n&&(a.code=e[1].session),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Ft(e){let t,n;return t=new Ae({props:{type:"multi",code:e[1].time}}),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};2&n&&(a.code=e[1].time),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Bt(e){let t,n;return t=new Ae({props:{type:"multi",code:e[1].device}}),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};2&n&&(a.code=e[1].device),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Ot(e){let t,n,a,s,r,l;return t=new ue({props:{$$slots:{default:[Ht]},$$scope:{ctx:e}}}),a=new ue({props:{$$slots:{default:[Ft]},$$scope:{ctx:e}}}),r=new ue({props:{$$slots:{default:[Bt]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment),s=S(),u(r.$$.fragment)},m(e,o){A(t,e,o),i(e,n,o),A(a,e,o),i(e,s,o),A(r,e,o),l=!0},p(e,n){const s={};34&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const l={};34&n&&(l.$$scope={dirty:n,ctx:e}),a.$set(l);const i={};34&n&&(i.$$scope={dirty:n,ctx:e}),r.$set(i)},i(e){l||(o(t.$$.fragment,e),o(a.$$.fragment,e),o(r.$$.fragment,e),l=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),_(r.$$.fragment,e),l=!1},d(e){e&&(d(n),d(s)),$(t,e),$(a,e),$(r,e)}}}function Dt(e){let t,n,a,s;return t=new ge({props:{kind:"warning",lowContrast:!0,hideCloseButton:!0,title:"Showing the first job only. Check the workdir for information of other jobs if any."}}),a=new me({props:{$$slots:{content:[Ot],default:[wt]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,t){const n={};34&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function Kt(e){let t,n,a;return{c(){t=r("a"),t.textContent="> Running Information",l(t,"href","javascript:void(0);")},m(s,r){i(s,t,r),n||(a=N(t,"click",e[4]),n=!0)},p:b,d(e){e&&d(t),n=!1,a()}}}function Ut(e){let t,n,a,s,g,m,f,u,A=e[1]&&xt(e),$=e[1]&&Kt(e);return{c(){A&&A.c(),t=S(),n=r("div"),a=r("div"),$&&$.c(),s=S(),g=r("div"),m=h("Powered by "),f=r("span"),l(a,"class","runinfo"),l(f,"class","pipen-report-footer-versions"),l(g,"class","versions"),l(n,"class","pipen-report-footer bx--grid")},m(r,l){A&&A.m(r,l),i(r,t,l),i(r,n,l),C(n,a),$&&$.m(a,null),C(n,s),C(n,g),C(g,m),C(g,f),f.innerHTML=e[0],u=!0},p(e,[n]){e[1]?A?(A.p(e,n),2&n&&o(A,1)):(A=xt(e),A.c(),o(A,1),A.m(t.parentNode,t)):A&&(p(),_(A,1,1,(()=>{A=null})),c()),e[1]?$?$.p(e,n):($=Kt(e),$.c(),$.m(a,null)):$&&($.d(1),$=null),(!u||1&n)&&(f.innerHTML=e[0])},i(e){u||(o(A),u=!0)},o(e){_(A),u=!1},d(e){e&&(d(t),d(n)),A&&A.d(e),$&&$.d()}}}function Vt(e,t,n){let{versions:a}=t,{runinfo:s}=t,r=!1;return e.$$set=e=>{"versions"in e&&n(0,a=e.versions),"runinfo"in e&&n(1,s=e.runinfo)},[a,s,r,function(e){r=e,n(2,r)},()=>{n(2,r=!0)}]}class zt extends t{constructor(e){super(),n(this,e,Vt,Ut,a,{versions:0,runinfo:1})}}function kt(e){let t,n,a;function s(t){e[12](t)}let r={$$slots:{default:[Xt]},$$scope:{ctx:e}};return void 0!==e[0]&&(r.isOpen=e[0]),t=new Se({props:r}),E.push((()=>L(t,"isOpen",s))),{c(){u(t.$$.fragment)},m(e,n){A(t,e,n),a=!0},p(e,a){const s={};8576&a&&(s.$$scope={dirty:a,ctx:e}),!n&&1&a&&(n=!0,s.isOpen=e[0],R((()=>n=!1))),t.$set(s)},i(e){a||(o(t.$$.fragment,e),a=!0)},o(e){_(t.$$.fragment,e),a=!1},d(e){$(t,e)}}}function Xt(e){let t,n;return t=new pt({props:{toc:e[7],page:e[8]}}),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};128&n&&(a.toc=e[7]),256&n&&(a.page=e[8]),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function qt(e){let t;const n=e[10].default,a=P(n,e,e[13],null);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||8192&s)&&M(a,n,e,e[13],t?I(n,e[13],s,null):v(e[13]),null)},i(e){t||(o(a,e),t=!0)},o(e){_(a,e),t=!1},d(e){a&&a.d(e)}}}function jt(e){let t,n;return t=new Re({props:{$$slots:{default:[qt]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};8192&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Yt(e){let t,n;return t=new Le({props:{$$slots:{default:[jt]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};8192&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Wt(e){let t,n,a,s;return t=new Ee({props:{$$slots:{default:[Yt]},$$scope:{ctx:e}}}),a=new zt({props:{versions:e[3],runinfo:e[9]}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};8192&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};8&n&&(r.versions=e[3]),512&n&&(r.runinfo=e[9]),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function Jt(e){let t,n,a,s,r,l;function g(t){e[11](t)}let m={logo:e[1],logotext:e[2],entries:e[4],pipeline_name:e[5]};void 0!==e[0]&&(m.isSideNavOpen=e[0]),t=new Gt({props:m}),E.push((()=>L(t,"isSideNavOpen",g)));let f=e[6]&&e[7].length>0&&kt(e);return r=new $e({props:{$$slots:{default:[Wt]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment),a=S(),f&&f.c(),s=S(),u(r.$$.fragment)},m(e,n){A(t,e,n),i(e,a,n),f&&f.m(e,n),i(e,s,n),A(r,e,n),l=!0},p(e,[a]){const l={};2&a&&(l.logo=e[1]),4&a&&(l.logotext=e[2]),16&a&&(l.entries=e[4]),32&a&&(l.pipeline_name=e[5]),!n&&1&a&&(n=!0,l.isSideNavOpen=e[0],R((()=>n=!1))),t.$set(l),e[6]&&e[7].length>0?f?(f.p(e,a),192&a&&o(f,1)):(f=kt(e),f.c(),o(f,1),f.m(s.parentNode,s)):f&&(p(),_(f,1,1,(()=>{f=null})),c());const i={};8712&a&&(i.$$scope={dirty:a,ctx:e}),r.$set(i)},i(e){l||(o(t.$$.fragment,e),o(f),o(r.$$.fragment,e),l=!0)},o(e){_(t.$$.fragment,e),_(f),_(r.$$.fragment,e),l=!1},d(e){e&&(d(a),d(s)),$(t,e),f&&f.d(e),$(r,e)}}}function Zt(e,t,n){let a,{$$slots:s={},$$scope:r}=t,{logo:l}=t,{logotext:i=null}=t,{versions:o}=t,{entries:p}=t,{pipeline_name:c}=t,{report_toc:_}=t,{toc:d}=t,{page:g}=t,{isSideNavOpen:f}=t,{runinfo:u}=t;return m((()=>{const e=new URL(window.location.href);let t;t=e.hash.length>0?document.querySelector(`li:has(>[href="${e.hash}"])`):document.querySelector("li:has(>.toc-current-page)"),t&&(t.classList.add("toc-active"),t.scrollIntoView());const n=document.querySelectorAll(".pipen-report-toc-anchor");a=new IntersectionObserver((e=>{e.forEach((e=>{const t=e.target.getAttribute("id"),n=document.querySelector(`li:has(>[href="#${t}"])`);n&&(e.isIntersecting?n.classList.add("toc-active"):n.classList.remove("toc-active"))}))}),{root:null,rootMargin:"0px",threshold:.5}),n.forEach((e=>{a.observe(e)}))})),e.$$set=e=>{"logo"in e&&n(1,l=e.logo),"logotext"in e&&n(2,i=e.logotext),"versions"in e&&n(3,o=e.versions),"entries"in e&&n(4,p=e.entries),"pipeline_name"in e&&n(5,c=e.pipeline_name),"report_toc"in e&&n(6,_=e.report_toc),"toc"in e&&n(7,d=e.toc),"page"in e&&n(8,g=e.page),"isSideNavOpen"in e&&n(0,f=e.isSideNavOpen),"runinfo"in e&&n(9,u=e.runinfo),"$$scope"in e&&n(13,r=e.$$scope)},[f,l,i,o,p,c,_,d,g,u,s,function(e){f=e,n(0,f)},function(e){f=e,n(0,f)},r]}class Qt extends t{constructor(e){super(),n(this,e,Zt,Jt,a,{logo:1,logotext:2,versions:3,entries:4,pipeline_name:5,report_toc:6,toc:7,page:8,isSideNavOpen:0,runinfo:9})}}function en(e){let t;return{c(){t=h(e[2])},m(e,n){i(e,t,n)},p:b,d(e){e&&d(t)}}}function tn(e){let t,n,a,s;return n=new he({props:{icon:e[1],kind:"tertiary",href:e[3](),$$slots:{default:[en]},$$scope:{ctx:e}}}),{c(){t=r("div"),u(n.$$.fragment),l(t,"class",a=`pagenav ${e[0]}`)},m(e,a){i(e,t,a),A(n,t,null),s=!0},p(e,[r]){const i={};16&r&&(i.$$scope={dirty:r,ctx:e}),n.$set(i),(!s||1&r&&a!==(a=`pagenav ${e[0]}`))&&l(t,"class",a)},i(e){s||(o(n.$$.fragment,e),s=!0)},o(e){_(n.$$.fragment,e),s=!1},d(e){e&&d(t),$(n)}}}function nn(e,t,n){let{dir:a}=t,s="up"===a?xe:we,r="up"===a?"Go to previous page":"Go to next page";return e.$$set=e=>{"dir"in e&&n(0,a=e.dir)},[a,s,r,()=>{let e=new URL(window.location.href);e.hash="";const t=parseInt(e.searchParams.get("page")||0),n="up"===a?t-1:t+1;return 0===n?e.searchParams.delete("page"):e.searchParams.set("page",n),e.toString()}]}class an extends t{constructor(e){super(),n(this,e,nn,tn,a,{dir:0})}}function sn(e){let t,n;const a=[{tooltipText:e[0]},{icon:e[1]},{direction:"bottom"},{align:"start"},e[3]];let s={};for(let e=0;e<a.length;e+=1)s=y(s,a[e]);return t=new Ce({props:s}),t.$on("click",e[2]),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,[n]){const s=11&n?G(a,[1&n&&{tooltipText:e[0]},2&n&&{icon:e[1]},a[2],a[3],8&n&&x(e[3])]):{};t.$set(s)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function rn(e,t,n){const a=["src","tip","icon"];let s,r=w(t,a),{src:l=""}=t,{tip:i=""}=t,{icon:o="auto"}=t,p=l.split(".").at(-1).toLocaleLowerCase();if("string"==typeof o)switch("auto"===o&&(o=p),o){case"pdf":case"DocumentPdf":s=Ye;break;case"html":case"htm":case"ChartLineData":s=je;break;case"png":case"Png":s=qe;break;case"jpg":case"jpeg":case"Jpg":s=Xe;break;case"gif":case"Gif":s=ke;break;case"tif":case"tiff":case"Tif":s=ze;break;case"eps":case"DatabaseElastic":s=Ve;break;case"mp4":case"avi":case"mov":case"mkv":case"webm":case"VideoPlayer":s=Ue;break;case"Image":s=Ke;break;case"Data":case"DataVolume":s=De;break;case"Code":case"IbmWatsonxCodeAssistantForZRefactor":s=Oe;break;case"svg":case"Svg":s=Be;break;case"zip":case"tar":case"gz":case"bz2":case"7z":case"rar":case"xz":case"Archive":s=Fe;break;default:s=He}const c={...r,class:[r.class,"pipen-report-image-download"].filter(Boolean).join(" ")};return e.$$set=e=>{t=y(y({},t),H(e)),n(7,r=w(t,a)),"src"in e&&n(5,l=e.src),"tip"in e&&n(0,i=e.tip),"icon"in e&&n(4,o=e.icon)},[i,s,()=>{const e=document.createElement("a");e.href=l,e.download=l.split("/").pop(),e.click()},c,o,l]}class ln extends t{constructor(e){super(),n(this,e,rn,sn,a,{src:5,tip:0,icon:4})}}const{Boolean:on}=D;function pn(e,t,n){const a=e.slice();return a[5]=t[n].src,a[6]=t[n].tip,a[7]=t[n].icon,a[9]=n,a}function cn(e){let t,n,a,l,m=s(e[0]),f=[];for(let t=0;t<m.length;t+=1)f[t]=_n(pn(e,m,t));const u=e=>_(f[e],1,1,(()=>{f[e]=null}));let A=[e[1],{role:"button"},{tabindex:"0"}],$={};for(let e=0;e<A.length;e+=1)$=y($,A[e]);return{c(){t=r("div");for(let e=0;e<f.length;e+=1)f[e].c();B(t,$),O(t,"svelte-14lcnln",!0)},m(s,r){i(s,t,r);for(let e=0;e<f.length;e+=1)f[e]&&f[e].m(t,null);n=!0,a||(l=N(t,"mouseenter",e[3]),a=!0)},p(e,n){if(1&n){let a;for(m=s(e[0]),a=0;a<m.length;a+=1){const s=pn(e,m,a);f[a]?(f[a].p(s,n),o(f[a],1)):(f[a]=_n(s),f[a].c(),o(f[a],1),f[a].m(t,null))}for(p(),a=m.length;a<f.length;a+=1)u(a);c()}O(t,"svelte-14lcnln",!0)},i(e){if(!n){for(let e=0;e<m.length;e+=1)o(f[e]);n=!0}},o(e){f=f.filter(on);for(let e=0;e<f.length;e+=1)_(f[e]);n=!1},d(e){e&&d(t),g(f,e),a=!1,l()}}}function _n(e){let t,n;return t=new ln({props:{src:e[5],tip:e[6],icon:e[7]}}),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.src=e[5]),1&n&&(a.tip=e[6]),1&n&&(a.icon=e[7]),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function dn(e){let t,n,a=e[0].length>0&&cn(e);return{c(){a&&a.c(),t=f()},m(e,s){a&&a.m(e,s),i(e,t,s),n=!0},p(e,[n]){e[0].length>0?a?(a.p(e,n),1&n&&o(a,1)):(a=cn(e),a.c(),o(a,1),a.m(t.parentNode,t)):a&&(p(),_(a,1,1,(()=>{a=null})),c())},i(e){n||(o(a),n=!0)},o(e){_(a),n=!1},d(e){e&&d(t),a&&a.d(e)}}}function gn(e,t,n){const a=["data"];let s=w(t,a),{data:r=[]}=t;Array.isArray(r)||(r=[r]),r=r.map((e=>"string"==typeof e?{src:e,tip:`Download the ${e.split(".").at(-1)} format.`,icon:"auto"}:(e.tip=e.tip||`Download the ${e.src.split(".").at(-1)} format.`,e.icon=e.icon||"auto",e)));const l={...s,class:[s.class,"pipen-report-image-download-list"].filter(Boolean).join(" ")},i=F();return e.$$set=e=>{t=y(y({},t),H(e)),n(4,s=w(t,a)),"data"in e&&n(0,r=e.data)},[r,l,i,()=>i("mouseenter")]}class mn extends t{constructor(e){super(),n(this,e,gn,dn,a,{data:0})}}const fn=e=>({}),un=e=>({}),An=e=>({}),$n=e=>({}),Sn=e=>({}),En=e=>({}),Ln=e=>({}),Rn=e=>({});function hn(e){let t,n;return t=new Te({props:{ratio:e[6],$$slots:{default:[Mn]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};64&n&&(a.ratio=e[6]),33556159&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Cn(e){let t,n,a,s,r=e[2]&&vn(e),l=e[0]&&In(e),g=e[1]&&Gn(e);return{c(){r&&r.c(),t=S(),l&&l.c(),n=S(),g&&g.c(),a=f()},m(e,o){r&&r.m(e,o),i(e,t,o),l&&l.m(e,o),i(e,n,o),g&&g.m(e,o),i(e,a,o),s=!0},p(e,s){e[2]?r?(r.p(e,s),4&s&&o(r,1)):(r=vn(e),r.c(),o(r,1),r.m(t.parentNode,t)):r&&(p(),_(r,1,1,(()=>{r=null})),c()),e[0]?l?(l.p(e,s),1&s&&o(l,1)):(l=In(e),l.c(),o(l,1),l.m(n.parentNode,n)):l&&(p(),_(l,1,1,(()=>{l=null})),c()),e[1]?g?(g.p(e,s),2&s&&o(g,1)):(g=Gn(e),g.c(),o(g,1),g.m(a.parentNode,a)):g&&(p(),_(g,1,1,(()=>{g=null})),c())},i(e){s||(o(r),o(l),o(g),s=!0)},o(e){_(r),_(l),_(g),s=!1},d(e){e&&(d(t),d(n),d(a)),r&&r.d(e),l&&l.d(e),g&&g.d(e)}}}function Tn(e){let t;const n=e[15].loading,a=P(n,e,e[25],$n);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||33554432&s)&&M(a,n,e,e[25],t?I(n,e[25],s,An):v(e[25]),$n)},i(e){t||(o(a,e),t=!0)},o(e){_(a,e),t=!1},d(e){a&&a.d(e)}}}function Nn(e){let t,n,a,s,l,g,m,u,A=[e[12],{src:n=e[3]},{alt:e[4]},{title:"Click to zoom in ..."}],$={};for(let e=0;e<A.length;e+=1)$=y($,A[e]);let E=e[9]&&bn(e);return{c(){t=r("img"),s=S(),E&&E.c(),l=f(),B(t,$)},m(n,a){i(n,t,a),i(n,s,a),E&&E.m(n,a),i(n,l,a),g=!0,m||(u=[N(t,"click",e[11]),N(t,"mouseenter",e[19]),N(t,"mouseleave",e[20])],m=!0)},p(a,s){B(t,$=G(A,[(e=a)[12],(!g||8&s&&!K(t.src,n=e[3]))&&{src:n},(!g||16&s)&&{alt:e[4]},{title:"Click to zoom in ..."}])),e[9]?E?(E.p(e,s),512&s&&o(E,1)):(E=bn(e),E.c(),o(E,1),E.m(l.parentNode,l)):E&&(p(),_(E,1,1,(()=>{E=null})),c())},i(n){g||(n&&U((()=>{g&&(a||(a=V(t,z,{duration:e[7]?Hn:0},!0)),a.run(1))})),o(E),g=!0)},o(n){n&&(a||(a=V(t,z,{duration:e[7]?Hn:0},!1)),a.run(0)),_(E),g=!1},d(e){e&&(d(t),d(s),d(l)),e&&a&&a.end(),E&&E.d(e),m=!1,k(u)}}}function bn(e){let t,n;return t=new mn({props:{data:e[5]}}),t.$on("mouseenter",e[21]),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};32&n&&(a.data=e[5]),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Pn(e){let t;const n=e[15].error,a=P(n,e,e[25],un);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||33554432&s)&&M(a,n,e,e[25],t?I(n,e[25],s,fn):v(e[25]),un)},i(e){t||(o(a,e),t=!0)},o(e){_(a,e),t=!1},d(e){a&&a.d(e)}}}function Mn(e){let t,n,a,s,r=e[2]&&Tn(e),l=e[0]&&Nn(e),g=e[1]&&Pn(e);return{c(){r&&r.c(),t=S(),l&&l.c(),n=S(),g&&g.c(),a=f()},m(e,o){r&&r.m(e,o),i(e,t,o),l&&l.m(e,o),i(e,n,o),g&&g.m(e,o),i(e,a,o),s=!0},p(e,s){e[2]?r?(r.p(e,s),4&s&&o(r,1)):(r=Tn(e),r.c(),o(r,1),r.m(t.parentNode,t)):r&&(p(),_(r,1,1,(()=>{r=null})),c()),e[0]?l?(l.p(e,s),1&s&&o(l,1)):(l=Nn(e),l.c(),o(l,1),l.m(n.parentNode,n)):l&&(p(),_(l,1,1,(()=>{l=null})),c()),e[1]?g?(g.p(e,s),2&s&&o(g,1)):(g=Pn(e),g.c(),o(g,1),g.m(a.parentNode,a)):g&&(p(),_(g,1,1,(()=>{g=null})),c())},i(e){s||(o(r),o(l),o(g),s=!0)},o(e){_(r),_(l),_(g),s=!1},d(e){e&&(d(t),d(n),d(a)),r&&r.d(e),l&&l.d(e),g&&g.d(e)}}}function vn(e){let t;const n=e[15].loading,a=P(n,e,e[25],Rn);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||33554432&s)&&M(a,n,e,e[25],t?I(n,e[25],s,Ln):v(e[25]),Rn)},i(e){t||(o(a,e),t=!0)},o(e){_(a,e),t=!1},d(e){a&&a.d(e)}}}function In(e){let t,n,a,s,l,g,m,u,A=[e[12],{src:n=e[3]},{alt:e[4]},{title:"Click to zoom in ..."}],$={};for(let e=0;e<A.length;e+=1)$=y($,A[e]);let E=e[9]&&yn(e);return{c(){t=r("img"),s=S(),E&&E.c(),l=f(),B(t,$)},m(n,a){i(n,t,a),i(n,s,a),E&&E.m(n,a),i(n,l,a),g=!0,m||(u=[N(t,"click",e[11]),N(t,"mouseenter",e[16]),N(t,"mouseleave",e[17])],m=!0)},p(a,s){B(t,$=G(A,[(e=a)[12],(!g||8&s&&!K(t.src,n=e[3]))&&{src:n},(!g||16&s)&&{alt:e[4]},{title:"Click to zoom in ..."}])),e[9]?E?(E.p(e,s),512&s&&o(E,1)):(E=yn(e),E.c(),o(E,1),E.m(l.parentNode,l)):E&&(p(),_(E,1,1,(()=>{E=null})),c())},i(n){g||(n&&U((()=>{g&&(a||(a=V(t,z,{duration:e[7]?Hn:0},!0)),a.run(1))})),o(E),g=!0)},o(n){n&&(a||(a=V(t,z,{duration:e[7]?Hn:0},!1)),a.run(0)),_(E),g=!1},d(e){e&&(d(t),d(s),d(l)),e&&a&&a.end(),E&&E.d(e),m=!1,k(u)}}}function yn(e){let t,n;return t=new mn({props:{data:e[5]}}),t.$on("mouseenter",e[18]),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};32&n&&(a.data=e[5]),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Gn(e){let t;const n=e[15].error,a=P(n,e,e[25],En);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||33554432&s)&&M(a,n,e,e[25],t?I(n,e[25],s,Sn):v(e[25]),En)},i(e){t||(o(a,e),t=!0)},o(e){_(a,e),t=!1},d(e){a&&a.d(e)}}}function xn(e){let t,n,a,s;return{c(){t=r("img"),l(t,"class","pipen-report-image-lg"),K(t.src,n=e[3])||l(t,"src",n),l(t,"alt",e[4])},m(e,n){i(e,t,n),s=!0},p(a,r){e=a,(!s||8&r&&!K(t.src,n=e[3]))&&l(t,"src",n),(!s||16&r)&&l(t,"alt",e[4])},i(n){s||(n&&U((()=>{s&&(a||(a=V(t,z,{duration:e[7]?Hn:0},!0)),a.run(1))})),s=!0)},o(n){n&&(a||(a=V(t,z,{duration:e[7]?Hn:0},!1)),a.run(0)),s=!1},d(e){e&&d(t),e&&a&&a.end()}}}function wn(e){let t,n,a,s,r,l;const g=[Cn,hn],m=[];function f(e,t){return void 0===e[6]?0:1}function h(t){e[22](t)}t=f(e),n=m[t]=g[t](e);let C={modalHeading:e[4],primaryButtonText:"Open in new window ...",secondaryButtonText:"Close",$$slots:{default:[xn]},$$scope:{ctx:e}};return void 0!==e[8]&&(C.open=e[8]),s=new de({props:C}),E.push((()=>L(s,"open",h))),s.$on("click:button--secondary",e[23]),s.$on("click:button--primary",e[24]),{c(){n.c(),a=S(),u(s.$$.fragment)},m(e,n){m[t].m(e,n),i(e,a,n),A(s,e,n),l=!0},p(e,[l]){let i=t;t=f(e),t===i?m[t].p(e,l):(p(),_(m[i],1,1,(()=>{m[i]=null})),c(),n=m[t],n?n.p(e,l):(n=m[t]=g[t](e),n.c()),o(n,1),n.m(a.parentNode,a));const d={};16&l&&(d.modalHeading=e[4]),33554584&l&&(d.$$scope={dirty:l,ctx:e}),!r&&256&l&&(r=!0,d.open=e[8],R((()=>r=!1))),s.$set(d)},i(e){l||(o(n),o(s.$$.fragment,e),l=!0)},o(e){_(n),_(s.$$.fragment,e),l=!1},d(e){e&&d(a),m[t].d(e),$(s,e)}}}const Hn=110;function Fn(e,t,n){const a=["src","alt","download","ratio","loading","loaded","error","fadeIn","border","loadImage"];let s=w(t,a),{$$slots:r={},$$scope:l}=t,{src:i=""}=t,{alt:o=""}=t,{download:p=[]}=t,{ratio:c}=t,{loading:_=!1}=t,{loaded:d=!1}=t,{error:g=!1}=t,{fadeIn:f=!1}=t,{border:u=!0}=t,A=!1,$=!1;const S=e=>{null!=R&&(R=null),n(0,d=!1),n(1,g=!1),R=new Image,R.src=e||i,R.onload=()=>n(0,d=!0),R.onerror=()=>n(1,g=!0)},E=F();let L,R=null;m((()=>()=>R=null));const h={...s,class:[s.class,"pipen-report-image",u?"":"pipen-report-image-noborder"].filter(Boolean).join(" ")};return e.$$set=e=>{t=y(y({},t),H(e)),n(28,s=w(t,a)),"src"in e&&n(3,i=e.src),"alt"in e&&n(4,o=e.alt),"download"in e&&n(5,p=e.download),"ratio"in e&&n(6,c=e.ratio),"loading"in e&&n(2,_=e.loading),"loaded"in e&&n(0,d=e.loaded),"error"in e&&n(1,g=e.error),"fadeIn"in e&&n(7,f=e.fadeIn),"border"in e&&n(13,u=e.border),"$$scope"in e&&n(25,l=e.$$scope)},e.$$.update=()=>{3&e.$$.dirty&&n(2,_=!d&&!g),8&e.$$.dirty&&i&&"undefined"!=typeof window&&S(),1&e.$$.dirty&&d&&E("load"),2&e.$$.dirty&&g&&E("error")},[d,g,_,i,o,p,c,f,A,$,L,()=>{n(8,A=!0)},h,u,S,r,()=>{n(9,$=!0)},()=>{n(10,L=setTimeout((()=>{n(9,$=!1)}),100))},()=>{clearTimeout(L),n(9,$=!0)},()=>{n(9,$=!0)},()=>{n(10,L=setTimeout((()=>{n(9,$=!1)}),100))},()=>{clearTimeout(L),n(9,$=!0)},function(e){A=e,n(8,A)},()=>{n(8,A=!1)},()=>{n(8,A=!1),window.open(i)},l]}class Bn extends t{constructor(e){super(),n(this,e,Fn,wn,a,{src:3,alt:4,download:5,ratio:6,loading:2,loaded:0,error:1,fadeIn:7,border:13,loadImage:14})}get loadImage(){return this.$$.ctx[14]}}function On(e){let t,n,a=[e[2]],s={};for(let e=0;e<a.length;e+=1)s=y(s,a[e]);return{c(){t=r("div"),n=h("Image loading ..."),B(t,s)},m(e,a){i(e,t,a),C(t,n)},p:b,d(e){e&&d(t)}}}function Dn(e){let t,n,a=[e[3]],s={};for(let e=0;e<a.length;e+=1)s=y(s,a[e]);return{c(){t=r("div"),n=h("Image loading error!"),B(t,s)},m(e,a){i(e,t,a),C(t,n)},p:b,d(e){e&&d(t)}}}function Kn(e){let t,n,a;const s=[{src:e[1]},e[4]];let l={$$slots:{error:[Dn],loading:[On]},$$scope:{ctx:e}};for(let e=0;e<s.length;e+=1)l=y(l,s[e]);n=new Bn({props:l});let p=[e[0]],c={};for(let e=0;e<p.length;e+=1)c=y(c,p[e]);return{c(){t=r("div"),u(n.$$.fragment),B(t,c)},m(e,s){i(e,t,s),A(n,t,null),a=!0},p(e,[a]){const r=18&a?G(s,[2&a&&{src:e[1]},16&a&&x(e[4])]):{};256&a&&(r.$$scope={dirty:a,ctx:e}),n.$set(r),B(t,c=G(p,[1&a&&e[0]]))},i(e){a||(o(n.$$.fragment,e),a=!0)},o(e){_(n.$$.fragment,e),a=!1},d(e){e&&d(t),$(n)}}}function Un(e,t,n){const a=["frameProps","width","height","src"];let s=w(t,a),{frameProps:r={class:"pipen-report-image-frame"}}=t,{width:l=300}=t,{height:i=200}=t,{src:o}=t;r||(r={"style:position":"relative"});let p=`width: ${l}px; height: ${i}px; aspect-ratio: ${l} / ${i}`,c={...s,class:[s.class,"pipen-report-image-loading"].filter(Boolean).join(" "),style:s.style?`${s.style}; ${p}`:p},_={...c,class:[c.class,"error"].join(" ")};return e.$$set=e=>{t=y(y({},t),H(e)),n(4,s=w(t,a)),"frameProps"in e&&n(0,r=e.frameProps),"width"in e&&n(5,l=e.width),"height"in e&&n(6,i=e.height),"src"in e&&n(1,o=e.src)},[r,o,c,_,s,l,i]}let Vn=class extends t{constructor(e){super(),n(this,e,Un,Kn,a,{frameProps:0,width:5,height:6,src:1})}};const zn=e=>({}),kn=e=>({});function Xn(e){let t,n;const a=e[25].default,s=P(a,e,e[24],null);return{c(){t=r("ul"),s&&s.c(),l(t,"role","menu"),l(t,"tabindex","-1"),l(t,"aria-label",e[14]),l(t,"data-floating-menu-direction",e[5]),l(t,"class",e[9]),O(t,"bx--overflow-menu-options",!0),O(t,"bx--overflow-menu--flip",e[7]),O(t,"bx--overflow-menu-options--open",e[0]),O(t,"bx--overflow-menu-options--light",e[6]),O(t,"bx--overflow-menu-options--sm","sm"===e[4]),O(t,"bx--overflow-menu-options--xl","xl"===e[4]),O(t,"bx--breadcrumb-menu-options",!!e[15])},m(a,r){i(a,t,r),s&&s.m(t,null),e[32](t),n=!0},p(e,r){s&&s.p&&(!n||16777216&r[0])&&M(s,a,e,e[24],n?I(a,e[24],r,null):v(e[24]),null),(!n||16384&r[0])&&l(t,"aria-label",e[14]),(!n||32&r[0])&&l(t,"data-floating-menu-direction",e[5]),(!n||512&r[0])&&l(t,"class",e[9]),(!n||512&r[0])&&O(t,"bx--overflow-menu-options",!0),(!n||640&r[0])&&O(t,"bx--overflow-menu--flip",e[7]),(!n||513&r[0])&&O(t,"bx--overflow-menu-options--open",e[0]),(!n||576&r[0])&&O(t,"bx--overflow-menu-options--light",e[6]),(!n||528&r[0])&&O(t,"bx--overflow-menu-options--sm","sm"===e[4]),(!n||528&r[0])&&O(t,"bx--overflow-menu-options--xl","xl"===e[4]),(!n||33280&r[0])&&O(t,"bx--breadcrumb-menu-options",!!e[15])},i(e){n||(o(s,e),n=!0)},o(e){_(s,e),n=!1},d(n){n&&d(t),s&&s.d(n),e[32](null)}}}function qn(e){let t,n,a,s,l,g,m,E;const L=e[25].menu,R=P(L,e,e[24],kn),h=R||function(e){let t,n,a;var s=e[1];function r(e,t){return{props:{"aria-label":e[11],title:e[11],class:"bx--overflow-menu__icon "+e[10]}}}return s&&(t=W(s,r(e))),{c(){t&&u(t.$$.fragment),n=f()},m(e,s){t&&A(t,e,s),i(e,n,s),a=!0},p(e,a){if(2&a[0]&&s!==(s=e[1])){if(t){p();const e=t;_(e.$$.fragment,1,0,(()=>{$(e,1)})),c()}s?(t=W(s,r(e)),u(t.$$.fragment),o(t.$$.fragment,1),A(t,n.parentNode,n)):t=null}else if(s){const n={};2048&a[0]&&(n["aria-label"]=e[11]),2048&a[0]&&(n.title=e[11]),1024&a[0]&&(n.class="bx--overflow-menu__icon "+e[10]),t.$set(n)}},i(e){a||(t&&o(t.$$.fragment,e),a=!0)},o(e){t&&_(t.$$.fragment,e),a=!1},d(e){e&&d(n),t&&$(t,e)}}}(e);let T=e[0]&&Xn(e),b=[{type:"button"},{"aria-haspopup":""},{"aria-expanded":e[0]},{"aria-label":e[14]},{id:e[12]},e[20]],x={};for(let e=0;e<b.length;e+=1)x=y(x,b[e]);return{c(){t=new J(!1),n=f(),a=S(),s=r("button"),h.c(),l=S(),T&&T.c(),t.a=n,B(s,x),O(s,"bx--overflow-menu",!0),O(s,"bx--overflow-menu--open",e[0]),O(s,"bx--overflow-menu--light",e[6]),O(s,"bx--overflow-menu--sm","sm"===e[4]),O(s,"bx--overflow-menu--xl","xl"===e[4])},m(r,o){t.m(e[13],document.head),C(document.head,n),i(r,a,o),i(r,s,o),h.m(s,null),C(s,l),T&&T.m(s,null),s.autofocus&&s.focus(),e[33](s),g=!0,m||(E=[N(window,"click",e[31]),N(s,"click",e[26]),N(s,"click",e[34]),N(s,"mouseover",e[27]),N(s,"mouseenter",e[28]),N(s,"mouseleave",e[29]),N(s,"keydown",e[30]),N(s,"keydown",e[35]),N(s,"focusout",e[36])],m=!0)},p(e,n){(!g||8192&n[0])&&t.p(e[13]),R?R.p&&(!g||16777216&n[0])&&M(R,L,e,e[24],g?I(L,e[24],n,zn):v(e[24]),kn):h.p&&(!g||3074&n[0])&&h.p(e,g?n:[-1,-1]),e[0]?T?(T.p(e,n),1&n[0]&&o(T,1)):(T=Xn(e),T.c(),o(T,1),T.m(s,null)):T&&(p(),_(T,1,1,(()=>{T=null})),c()),B(s,x=G(b,[{type:"button"},{"aria-haspopup":""},(!g||1&n[0])&&{"aria-expanded":e[0]},(!g||16384&n[0])&&{"aria-label":e[14]},(!g||4096&n[0])&&{id:e[12]},1048576&n[0]&&e[20]])),O(s,"bx--overflow-menu",!0),O(s,"bx--overflow-menu--open",e[0]),O(s,"bx--overflow-menu--light",e[6]),O(s,"bx--overflow-menu--sm","sm"===e[4]),O(s,"bx--overflow-menu--xl","xl"===e[4])},i(e){g||(o(h,e),o(T),g=!0)},o(e){_(h,e),_(T),g=!1},d(r){r&&(t.d(),d(a),d(s)),d(n),h.d(r),T&&T.d(),e[33](null),m=!1,k(E)}}}function jn(e,t,n){let a,s;const r=["size","direction","open","light","flipped","persistentClickItems","menuOptionsClass","icon","iconClass","iconDescription","id","buttonRef","menuRef"];let l,i,o,p=w(t,r),{$$slots:c={},$$scope:_}=t,{size:d}=t,{direction:g="bottom"}=t,{open:m=!1}=t,{light:f=!1}=t,{flipped:u=!1}=t,{persistentClickItems:A=!1}=t,{menuOptionsClass:$}=t,{icon:S=Je}=t,{iconClass:L}=t,{iconDescription:R="Open and close list of options"}=t,{id:h="ccs-"+Math.random().toString(36)}=t,{buttonRef:C=null}=t,{menuRef:T=null}=t;const N=X("BreadcrumbItem"),b=F(),P=Z([]);q(e,P,(e=>n(23,i=e)));const M=Z(void 0);q(e,M,(e=>n(38,o=e)));const v=Z(void 0),I=Z(-1);let G;q(e,I,(e=>n(22,l=e)));let x=!0;j("OverflowMenu",{focusedId:v,add:({id:e,text:t,primaryFocus:n,disabled:a})=>{P.update((s=>(n&&I.set(s.length),[...s,{id:e,text:t,primaryFocus:n,disabled:a,index:s.length}])))},update:e=>{M.set(e)},change:e=>{let t=l+e;t<0?t=i.length-1:t>=i.length&&(t=0);let n=i[t].disabled;for(;n;)t+=e,t<0?t=i.length-1:t>=i.length&&(t=0),n=i[t].disabled;I.set(t)}}),Y((()=>{if(o){const{index:e,text:t}=i.filter((e=>e.id===o))[0];b("close",{index:e,text:t}),n(0,m=!1)}if(m){const{width:e,height:t}=C.getBoundingClientRect();n(21,G=e),!x&&l<0&&T.focus(),u&&(n(3,T.style.left="auto",T),n(3,T.style.right=0,T)),"top"===g?(n(3,T.style.top="auto",T),n(3,T.style.bottom=t+"px",T)):"bottom"===g&&n(3,T.style.top=t+"px",T),N&&(n(3,T.style.top=t+10+"px",T),n(3,T.style.left="-11px",T))}m||(P.set([]),M.set(void 0),I.set(0)),x=!1}));return e.$$set=e=>{n(40,t=y(y({},t),H(e))),n(20,p=w(t,r)),"size"in e&&n(4,d=e.size),"direction"in e&&n(5,g=e.direction),"open"in e&&n(0,m=e.open),"light"in e&&n(6,f=e.light),"flipped"in e&&n(7,u=e.flipped),"persistentClickItems"in e&&n(8,A=e.persistentClickItems),"menuOptionsClass"in e&&n(9,$=e.menuOptionsClass),"icon"in e&&n(1,S=e.icon),"iconClass"in e&&n(10,L=e.iconClass),"iconDescription"in e&&n(11,R=e.iconDescription),"id"in e&&n(12,h=e.id),"buttonRef"in e&&n(2,C=e.buttonRef),"menuRef"in e&&n(3,T=e.menuRef),"$$scope"in e&&n(24,_=e.$$scope)},e.$$.update=()=>{n(14,a=t["aria-label"]||"menu"),12582912&e.$$.dirty[0]&&i[l]&&v.set(i[l].id),2101248&e.$$.dirty[0]&&n(13,s=`<style>\n    #${h} .bx--overflow-menu-options.bx--overflow-menu-options:after {\n      width: ${G?G+"px":"2rem"};\n    }\n  </style>`)},N&&n(1,S=We),t=H(t),[m,S,C,T,d,g,f,u,A,$,L,R,h,s,a,N,b,P,M,I,p,G,l,i,_,c,function(t){Q.call(this,e,t)},function(t){Q.call(this,e,t)},function(t){Q.call(this,e,t)},function(t){Q.call(this,e,t)},function(t){Q.call(this,e,t)},({target:e})=>{C&&C.contains(e)||T&&!T.contains(e)&&n(0,m=!1)},function(e){E[e?"unshift":"push"]((()=>{T=e,n(3,T)}))},function(e){E[e?"unshift":"push"]((()=>{C=e,n(2,C)}))},e=>{A&&e.stopPropagation(),T&&T.contains(e.target)||(n(0,m=!m),m||b("close"))},e=>{m&&(["ArrowDown","ArrowLeft","ArrowRight","ArrowUp"].includes(e.key)?e.preventDefault():"Escape"===e.key&&(e.stopPropagation(),b("close"),n(0,m=!1),C.focus()))},e=>{m&&(C.contains(e.relatedTarget)||(b("close"),n(0,m=!1)))}]}class Yn extends t{constructor(e){super(),n(this,e,jn,qn,a,{size:4,direction:5,open:0,light:6,flipped:7,persistentClickItems:8,menuOptionsClass:9,icon:1,iconClass:10,iconDescription:11,id:12,buttonRef:2,menuRef:3},null,[-1,-1])}}function Wn(e,t,n){const a=e.slice();return a[3]=t[n],a[4]=t,a[5]=n,a}function Jn(e){let t,n,a,s,p;function c(t){e[1](t,e[3])}let g={labelText:e[3].value};return void 0!==e[3].selected&&(g.checked=e[3].selected),n=new Ne({props:g}),E.push((()=>L(n,"checked",c))),n.$on("change",e[2]),{c(){t=r("li"),u(n.$$.fragment),s=S(),l(t,"class","pipen-report-column-selector-item svelte-147jw2r")},m(e,a){i(e,t,a),A(n,t,null),C(t,s),p=!0},p(t,s){e=t;const r={};1&s&&(r.labelText=e[3].value),!a&&1&s&&(a=!0,r.checked=e[3].selected,R((()=>a=!1))),n.$set(r)},i(e){p||(o(n.$$.fragment,e),p=!0)},o(e){_(n.$$.fragment,e),p=!1},d(e){e&&d(t),$(n)}}}function Zn(e){let t,n,a=s(e[0]),r=[];for(let t=0;t<a.length;t+=1)r[t]=Jn(Wn(e,a,t));const l=e=>_(r[e],1,1,(()=>{r[e]=null}));return{c(){for(let e=0;e<r.length;e+=1)r[e].c();t=f()},m(e,a){for(let t=0;t<r.length;t+=1)r[t]&&r[t].m(e,a);i(e,t,a),n=!0},p(e,n){if(1&n){let i;for(a=s(e[0]),i=0;i<a.length;i+=1){const s=Wn(e,a,i);r[i]?(r[i].p(s,n),o(r[i],1)):(r[i]=Jn(s),r[i].c(),o(r[i],1),r[i].m(t.parentNode,t))}for(p(),i=a.length;i<r.length;i+=1)l(i);c()}},i(e){if(!n){for(let e=0;e<a.length;e+=1)o(r[e]);n=!0}},o(e){r=r.filter(Boolean);for(let e=0;e<r.length;e+=1)_(r[e]);n=!1},d(e){e&&d(t),g(r,e)}}}function Qn(e){let t,n;return t=new Yn({props:{class:"pipen-report-overflowmenu",persistentClickItems:!0,size:"sm",icon:Ze,iconDescription:"Select visible columns ...",$$slots:{default:[Zn]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,[n]){const a={};65&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function ea(e,t,n){let{allColumns:a}=t;return e.$$set=e=>{"allColumns"in e&&n(0,a=e.allColumns)},[a,function(t,s){e.$$.not_equal(s.selected,t)&&(s.selected=t,n(0,a))},()=>{n(0,a)}]}class ta extends t{constructor(e){super(),n(this,e,ea,Qn,a,{allColumns:0})}}function na(e){let t,n;return t=new he({props:{iconDescription:"Right click and save as to download the entire data",tooltipPosition:"top",icon:Qe,href:e[2]}}),t.$on("click",e[13]),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};4&n&&(a.href=e[2]),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function aa(e){let t,n,a,s,r,l,g,m,h,C;function T(t){e[17](t)}let N={persistent:!0};function b(t){e[18](t)}void 0!==e[7]&&(N.value=e[7]),t=new Ie({props:N}),E.push((()=>L(t,"value",T)));let P={};void 0!==e[6]&&(P.allColumns=e[6]),s=new ta({props:P}),E.push((()=>L(s,"allColumns",b))),g=new he({props:{iconDescription:`Download current ${e[11].length} records (may be partial of the entire data)`,tooltipPosition:"top",icon:He}}),g.$on("click",e[14]);let M=!!e[2]&&na(e);return{c(){u(t.$$.fragment),a=S(),u(s.$$.fragment),l=S(),u(g.$$.fragment),m=S(),M&&M.c(),h=f()},m(e,n){A(t,e,n),i(e,a,n),A(s,e,n),i(e,l,n),A(g,e,n),i(e,m,n),M&&M.m(e,n),i(e,h,n),C=!0},p(e,a){const l={};!n&&128&a&&(n=!0,l.value=e[7],R((()=>n=!1))),t.$set(l);const i={};!r&&64&a&&(r=!0,i.allColumns=e[6],R((()=>r=!1))),s.$set(i);const d={};2048&a&&(d.iconDescription=`Download current ${e[11].length} records (may be partial of the entire data)`),g.$set(d),e[2]?M?(M.p(e,a),4&a&&o(M,1)):(M=na(e),M.c(),o(M,1),M.m(h.parentNode,h)):M&&(p(),_(M,1,1,(()=>{M=null})),c())},i(e){C||(o(t.$$.fragment,e),o(s.$$.fragment,e),o(g.$$.fragment,e),o(M),C=!0)},o(e){_(t.$$.fragment,e),_(s.$$.fragment,e),_(g.$$.fragment,e),_(M),C=!1},d(e){e&&(d(a),d(l),d(m),d(h)),$(t,e),$(s,e),$(g,e),M&&M.d(e)}}}function sa(e){let t,n;return t=new ve({props:{$$slots:{default:[aa]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};134219972&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function ra(e){let t,n;return t=new Me({props:{size:e[12],$$slots:{default:[sa]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};134219972&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function la(e){let t;return{c(){t=r("p"),t.textContent='You must use "Save as" or "Save link as" from the context menu (by right-clicking the button) to download the entire data.'},m(e,n){i(e,t,n)},p:b,d(e){e&&d(t)}}}function ia(e){let t,n,a,s,r,l,p,c,g;const m=[{size:e[0]},{zebra:e[3]},{sortable:e[4]},{page:e[8]},{pageSize:e[9]},{rows:e[11]},{headers:e[1]},e[15]];let f={$$slots:{default:[ra]},$$scope:{ctx:e}};for(let e=0;e<m.length;e+=1)f=y(f,m[e]);function h(t){e[19](t)}function C(t){e[20](t)}t=new be({props:f});let T={pageSizes:e[5],totalItems:e[11].length};function N(t){e[21](t)}void 0!==e[9]&&(T.pageSize=e[9]),void 0!==e[8]&&(T.page=e[8]),a=new Pe({props:T}),E.push((()=>L(a,"pageSize",h))),E.push((()=>L(a,"page",C)));let b={passiveModal:!0,modalHeading:"Downloading entire data",$$slots:{default:[la]},$$scope:{ctx:e}};return void 0!==e[10]&&(b.open=e[10]),p=new de({props:b}),E.push((()=>L(p,"open",N))),p.$on("open",e[22]),p.$on("close",e[23]),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment),l=S(),u(p.$$.fragment)},m(e,s){A(t,e,s),i(e,n,s),A(a,e,s),i(e,l,s),A(p,e,s),g=!0},p(e,[n]){const l=35611&n?G(m,[1&n&&{size:e[0]},8&n&&{zebra:e[3]},16&n&&{sortable:e[4]},256&n&&{page:e[8]},512&n&&{pageSize:e[9]},2048&n&&{rows:e[11]},2&n&&{headers:e[1]},32768&n&&x(e[15])]):{};134219972&n&&(l.$$scope={dirty:n,ctx:e}),t.$set(l);const i={};32&n&&(i.pageSizes=e[5]),2048&n&&(i.totalItems=e[11].length),!s&&512&n&&(s=!0,i.pageSize=e[9],R((()=>s=!1))),!r&&256&n&&(r=!0,i.page=e[8],R((()=>r=!1))),a.$set(i);const o={};134217728&n&&(o.$$scope={dirty:n,ctx:e}),!c&&1024&n&&(c=!0,o.open=e[10],R((()=>c=!1))),p.$set(o)},i(e){g||(o(t.$$.fragment,e),o(a.$$.fragment,e),o(p.$$.fragment,e),g=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),_(p.$$.fragment,e),g=!1},d(e){e&&(d(n),d(l)),$(t,e),$(a,e),$(p,e)}}}function oa(e,t,n){const a=["size","src","zebra","sortable","rows","headers","pageSizes"];let s=w(t,a),{size:r="sm"}=t,{src:l}=t;r={sm:"short",small:"short",short:"short",md:"medium",medium:"medium",lg:"tall",large:"tall",tall:"tall",compact:"compact"}[r];let i={short:"sm",compact:"sm",medium:"default",tall:"default"}[r],{zebra:o=!0}=t,{sortable:p=!0}=t,{rows:c=[]}=t,{headers:_=[]}=t,{pageSizes:d=[20,50,75,100]}=t,g=_.map((e=>({...e,selected:!0}))),m=1,f=d[0],u="",A=!1,$=c;return e.$$set=e=>{t=y(y({},t),H(e)),n(15,s=w(t,a)),"size"in e&&n(0,r=e.size),"src"in e&&n(2,l=e.src),"zebra"in e&&n(3,o=e.zebra),"sortable"in e&&n(4,p=e.sortable),"rows"in e&&n(16,c=e.rows),"headers"in e&&n(1,_=e.headers),"pageSizes"in e&&n(5,d=e.pageSizes)},e.$$.update=()=>{64&e.$$.dirty&&n(1,_=g.filter((e=>e.selected))),65664&e.$$.dirty&&(u.length>0?n(11,$=c.filter((e=>Object.values(e).filter((e=>!!e&&e.toString().includes(u))).length>0))):n(11,$=c))},[r,_,l,o,p,d,g,u,m,f,A,$,i,e=>{e.preventDefault(),n(10,A=!0)},()=>{((e,t)=>{const n=document.createElement("a");n.href=e,n.download=t,document.body.appendChild(n),n.click(),document.body.removeChild(n)})("data:application/vnd.ms-excel;charset=utf-8,\ufeff"+encodeURIComponent(function(){const e=_.map((e=>e.value)),t=[];return $.forEach((e=>{const n=_.map((t=>`"${e[t.key]}"`));t.push(n.join(","))})),e.join(",")+"\r\n"+t.join("\r\n")}()),l&&l.length>0?l.split("/").pop().split(".")[0]+".xls":"data-subset.xls")},s,c,function(e){u=e,n(7,u)},function(e){g=e,n(6,g)},function(e){f=e,n(9,f)},function(e){m=e,n(8,m)},function(e){A=e,n(10,A)},function(t){Q.call(this,e,t)},function(t){Q.call(this,e,t)}]}class pa extends t{constructor(e){super(),n(this,e,oa,ia,a,{size:0,src:2,zebra:3,sortable:4,rows:16,headers:1,pageSizes:5})}}function ca(e){let t,n,a;const s=[{headers:e[0]},{rows:e[1]},e[3]];let l={};for(let e=0;e<s.length;e+=1)l=y(l,s[e]);n=new pa({props:l});let p=[e[2]],c={};for(let e=0;e<p.length;e+=1)c=y(c,p[e]);return{c(){t=r("div"),u(n.$$.fragment),B(t,c)},m(e,s){i(e,t,s),A(n,t,null),a=!0},p(e,a){const r=11&a?G(s,[1&a&&{headers:e[0]},2&a&&{rows:e[1]},8&a&&x(e[3])]):{};n.$set(r),B(t,c=G(p,[4&a&&e[2]]))},i(e){a||(o(n.$$.fragment,e),a=!0)},o(e){_(n.$$.fragment,e),a=!1},d(e){e&&d(t),$(n)}}}function _a(e){let t,n;const a=[{headers:e[0]},{rows:e[1]},e[3]];let s={};for(let e=0;e<a.length;e+=1)s=y(s,a[e]);return t=new pa({props:s}),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const s=11&n?G(a,[1&n&&{headers:e[0]},2&n&&{rows:e[1]},8&n&&x(e[3])]):{};t.$set(s)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function da(e){let t,n,a,s;const r=[_a,ca],l=[];function g(e,t){return e[2]?1:0}return t=g(e),n=l[t]=r[t](e),{c(){n.c(),a=f()},m(e,n){l[t].m(e,n),i(e,a,n),s=!0},p(e,[s]){let i=t;t=g(e),t===i?l[t].p(e,s):(p(),_(l[i],1,1,(()=>{l[i]=null})),c(),n=l[t],n?n.p(e,s):(n=l[t]=r[t](e),n.c()),o(n,1),n.m(a.parentNode,a))},i(e){s||(o(n),s=!0)},o(e){_(n),s=!1},d(e){e&&d(a),l[t].d(e)}}}function ga(e,t,n){const a=["frameProps","data","headers","rows"];let s=w(t,a),{frameProps:r={class:"pipen-report-datatable-frame"}}=t,{data:l={}}=t,{headers:i=[]}=t,{rows:o=[]}=t;if(i.length+o.length>0&&l.length>0)throw"Can only specify `data` or `headers`/`rows` for `DataTable`!";return l.length>0&&(o=l,i=Object.keys(o[0]).filter((e=>"id"!==e)).map((e=>({key:e,value:e})))),e.$$set=e=>{t=y(y({},t),H(e)),n(3,s=w(t,a)),"frameProps"in e&&n(2,r=e.frameProps),"data"in e&&n(4,l=e.data),"headers"in e&&n(0,i=e.headers),"rows"in e&&n(1,o=e.rows)},[i,o,r,s,l]}class ma extends t{constructor(e){super(),n(this,e,ga,da,a,{frameProps:2,data:4,headers:0,rows:1})}}function fa(e){let t,n;return{c(){t=r("div"),n=h(e[0]),l(t,"class","pipen-report-descr-title svelte-1xzxuva")},m(e,a){i(e,t,a),C(t,n)},p(e,t){1&t&&T(n,e[0])},d(e){e&&d(t)}}}function ua(e){let t,n,a,s=e[0]&&fa(e);const l=e[3].default,p=P(l,e,e[2],null);let c=[e[1]],g={};for(let e=0;e<c.length;e+=1)g=y(g,c[e]);return{c(){t=r("div"),s&&s.c(),n=S(),p&&p.c(),B(t,g),O(t,"svelte-1xzxuva",!0)},m(e,r){i(e,t,r),s&&s.m(t,null),C(t,n),p&&p.m(t,null),a=!0},p(e,[r]){e[0]?s?s.p(e,r):(s=fa(e),s.c(),s.m(t,n)):s&&(s.d(1),s=null),p&&p.p&&(!a||4&r)&&M(p,l,e,e[2],a?I(l,e[2],r,null):v(e[2]),null),B(t,g=G(c,[2&r&&e[1]])),O(t,"svelte-1xzxuva",!0)},i(e){a||(o(p,e),a=!0)},o(e){_(p,e),a=!1},d(e){e&&d(t),s&&s.d(),p&&p.d(e)}}}function Aa(e,t,n){const a=["title"];let s=w(t,a),{$$slots:r={},$$scope:l}=t,{title:i}=t,{klass:o,...p}=s;return o=o?`pipen-report-descr ${o}`:"pipen-report-descr",p||(p={}),p.class=o,e.$$set=e=>{t=y(y({},t),H(e)),n(5,s=w(t,a)),"title"in e&&n(0,i=e.title),"$$scope"in e&&n(2,l=e.$$scope)},[i,p,l,r]}class $a extends t{constructor(e){super(),n(this,e,Aa,ua,a,{title:0})}}function Sa(e){let t,n,a,s;return t=new fe({props:{label:"Plot"}}),a=new fe({props:{label:"Table"}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p:b,i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function Ea(e){let t;return{c(){t=h("This table presents a comprehensive summary of the top enriched pathways derived from the fgsea. Each row corresponds to a pathway, and the gene ranks are shown based on the ranking metric used in the analysis. The enrichment score, p-value, and adjusted p-value are also provided to assess the significance of the enrichment.")},m(e,n){i(e,t,n)},d(e){e&&d(t)}}}function La(e){let t,n,a,s;return t=new $a({props:{$$slots:{default:[Ea]},$$scope:{ctx:e}}}),a=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/gsea_table.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/gsea_table.pdf",width:1e3,height:760}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function Ra(e){let t;return{c(){t=h("This plot represents the GSEA results for a specified gene set, illustrating the distribution and impact of the gene set along the ranked list of genes. The running enrichment score curve shows the cumulative enrichment score as genes from the input list are encountered. Positive peaks on the curve indicate regions where members of the gene set are predominantly found.")},m(e,n){i(e,t,n)},d(e){e&&d(t)}}}function ha(e){let t,n,a,s;return t=new $a({props:{$$slots:{default:[Ra]},$$scope:{ctx:e}}}),a=new ma({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/fgsea.txt",data:[{pathway:"HALLMARK_INTERFERON_ALPHA_RESPONSE",pval:3e-4,padj:.0047,log2err:.4985,ES:.6982,NES:1.4923,size:96,leadingEdge:"CD74,PSMB8,ISG20,GBP4,GBP2,UBE2L6,IFITM3,NCOA7,PSME2,PSMA3,PSMB9,USP18,TAP1,RTP4,PLSCR1,IRF1,LY6E,HELZ2,EPSTI1,PSME1,IFIH1,IFI35,ISG15,SP110,SELL,SLC25A28,OAS1,HERC6,ADAR,OASL,MVB12A,NMI,IL15,TMEM140,PARP9,B2M,TRIM25,MOV10,IFI27,LAMP3,BST2,TDRD7,PARP12,IRF2,LAP3,CD47,CXCL10,OGFR,STAT2,SAMD9L,DHX58,TRAFD1,IRF9,PNPT1,TRIM21,CNP,IRF7,ELF1",id:0},{pathway:"HALLMARK_APICAL_SURFACE",pval:.0108,padj:.0754,log2err:.3807,ES:-.6423,NES:-1.5497,size:44,leadingEdge:"CD160,HSPB1,LYN,FLOT2,CROCC,APP,EPHB4,LYPD3,TMEM8B,MAL,ATP8B1,GSTM3,SCUBE1,AKAP7,B4GALT1,PLAUR,BRCA1",id:1},{pathway:"HALLMARK_ANGIOGENESIS",pval:.0355,padj:.1655,log2err:.3218,ES:-.6139,NES:-1.4101,size:36,leadingEdge:"TIMP1,CCND2,APP,VAV2,ITGAV",id:2},{pathway:"HALLMARK_MYC_TARGETS_V2",pval:.1224,padj:.4283,log2err:.1336,ES:.6072,NES:1.2351,size:58,leadingEdge:"DUSP2,CBX3,HSPD1,PHB,WDR74,MCM4,DCTPP1,NOP16,NOP56,MRTO4,HSPE1,NPM1,NOC4L,TCOF1,PLK1,NIP7,CDK4,PPAN,PLK4,MYC,SUPV3L1,LAS1L,AIMP2,SORD,PRMT3,RABEPK,PES1,NOP2,GNL3,FARSA,EXOSC5,RCL1,PA2G4,UNG,RRP9,MYBBP1A,IMP4,BYSL,DDX18,UTP20,HK2,TBRG4,NOLC1,SLC19A1,PUS1",id:3},{pathway:"HALLMARK_PANCREAS_BETA_CELLS",pval:.2713,padj:.6394,log2err:.086,ES:.5816,NES:1.1318,size:40,leadingEdge:"SRP9,SEC11A,FOXO1,SRPRB,SRP14,SPCS1,DPP4,VDR",id:4},{pathway:"HALLMARK_IL6_JAK_STAT3_SIGNALING",pval:.274,padj:.6394,log2err:.0789,ES:.5225,NES:1.1091,size:87,leadingEdge:"BAK1,CD38,STAM2,MYD88,ITGA4,IRF1,PTPN11,CSF2,TNFRSF1B,PIM1,IL15RA,TNFRSF1A,STAT1,IL2RG,GRB2,FAS,LEPR,EBI3,CXCL10,IL10RB,CXCL13,STAT2,IRF9,PIK3R5,TYK2,IL18R1,LTBR,ACVR1B",id:5},{pathway:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",pval:.4017,padj:.7161,log2err:.0644,ES:.5228,NES:1.0418,size:49,leadingEdge:"TXN,GLRX,NDUFA6,PRDX1,NDUFB4,GLRX2,PRDX6,TXNRD2,PFKP,G6PD,NDUFS2,LAMTOR5,ATOX1,SOD1,FTL,PRDX2,HMOX2",id:6},{pathway:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",pval:.42,padj:.7161,log2err:.0596,ES:.4923,NES:1.0252,size:73,leadingEdge:"FABP5,ACTG1,PLSCR1,HSD17B7,STX5,ANXA5,ANTXR2,PMVK,FBXO6,GLDC,FASN,FADS2,ALCAM,SREBF2,JAG1,STARD4,MVD,FDPS,ERRFI1,ETHE1,CYP51A1,DHCR7,SQLE,TP53INP1,PPARG,ECH1,IDI1,LDLR,ACAT2,GUSB,ABCA2,PCYT2,NFIL3",id:7},{pathway:"HALLMARK_WNT_BETA_CATENIN_SIGNALING",pval:.5137,padj:.7161,log2err:.1421,ES:-.4037,NES:-.9605,size:42,leadingEdge:"TCF7,CCND2,DVL2,CUL1,DLL1,PSEN2,PTCH1,NKD1,WNT1,HDAC2,HDAC5,CTNNB1",id:8},{pathway:"HALLMARK_ANDROGEN_RESPONSE",pval:.5613,padj:.7161,log2err:.0442,ES:.4487,NES:.9629,size:99,leadingEdge:"SRP19,ARID5B,FKBP5,SPCS3,CENPN,DBI,DHCR24,NGLY1,INSIG1,SAT1,UBE2J1,ADRM1,STK39,MAF,ELOVL5,UBE2I,B2M,TMEM50A,HOMER2,CCND3,RAB4A,CDK6,SRF,XRCC6,ELL2,ABHD2,SMS,PMEPA1,SORD,INPP4B,ABCC4,CDC14B",id:9},{pathway:"HALLMARK_TGF_BETA_SIGNALING",pval:.6115,padj:.7161,log2err:.1404,ES:-.3835,NES:-.9516,size:54,leadingEdge:"JUNB,PPP1R15A,TGFBR1,ID3,ID1,IFNGR2,SMURF2,XIAP,SKI,THBS1,SLC20A1,SMAD3,SERPINE1,ENG,FNTA,WWTR1,HIPK2,TRIM33,PPM1A,KLF10,BMPR1A,SMURF1,CTNNB1",id:10},{pathway:"HALLMARK_PROTEIN_SECRETION",pval:.6138,padj:.7161,log2err:.0404,ES:.4392,NES:.9359,size:95,leadingEdge:"CLTA,STX7,ICA1,CLTC,COPB2,SEC22B,BET1,COPE,NAPA,AP1G1,VAMP7,ANP32E,AP2M1,DNM1L,AP2S1,RER1,SOD1,GOSR2,CLCN3,STX12,TMX1,CLN5,GOLGA4,GNAS,ARFGEF1,ARCN1,RAB5A,STAM,ADAM10,YKT6,MON2,RPS6KA3,NAPG,TOM1L1,CTSC,LMAN1,RAB9A,SGMS1,ATP6V1H,ARF1,PPT1,TMED2",id:11},{pathway:"HALLMARK_HEDGEHOG_SIGNALING",pval:.8177,padj:.8501,log2err:.1059,ES:-.3646,NES:-.8375,size:36,leadingEdge:"ADGRG1,CELSR1,AMOT,PTCH1,TLE3",id:12},{pathway:"HALLMARK_NOTCH_SIGNALING",pval:.8501,padj:.8501,log2err:.0317,ES:.4113,NES:.777,size:32,leadingEdge:"SAP30,RBX1,LFNG,SKP1,JAG1,DTX2,FBXW11,HES1,APH1A,CCND1",id:13}]}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function Ca(e){let t,n,a,s;return t=new ue({props:{$$slots:{default:[La]},$$scope:{ctx:e}}}),a=new ue({props:{$$slots:{default:[ha]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function Ta(e){let t,n;return t=new me({props:{$$slots:{content:[Ca],default:[Sa]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Na(e){let t,n,a,s,p,c,g,m,f,E,L,R,h,T,N,P,M,v,I,y,G,x,w,H,F,B,O,D,K,U,V,z,k,X,q,j,Y,W,J,Z,Q,te,ne,ae,se,re,le,ie,oe,pe,ce;return a=new $a({props:{title:"HALLMARK_INTERFERON_ALPHA_RESPONSE",class:"pipen-report-table-image-descr"}}),p=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/fgsea_HALLMARK_INTERFERON_ALPHA_RESPONSE.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/fgsea_HALLMARK_INTERFERON_ALPHA_RESPONSE.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),m=new $a({props:{title:"HALLMARK_APICAL_SURFACE",class:"pipen-report-table-image-descr"}}),E=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/fgsea_HALLMARK_APICAL_SURFACE.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/fgsea_HALLMARK_APICAL_SURFACE.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),h=new $a({props:{title:"HALLMARK_ANGIOGENESIS",class:"pipen-report-table-image-descr"}}),N=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/fgsea_HALLMARK_ANGIOGENESIS.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/fgsea_HALLMARK_ANGIOGENESIS.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),v=new $a({props:{title:"HALLMARK_MYC_TARGETS_V2",class:"pipen-report-table-image-descr"}}),y=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/fgsea_HALLMARK_MYC_TARGETS_V2.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/fgsea_HALLMARK_MYC_TARGETS_V2.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),w=new $a({props:{title:"HALLMARK_PANCREAS_BETA_CELLS",class:"pipen-report-table-image-descr"}}),F=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/fgsea_HALLMARK_PANCREAS_BETA_CELLS.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/fgsea_HALLMARK_PANCREAS_BETA_CELLS.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),D=new $a({props:{title:"HALLMARK_IL6_JAK_STAT3_SIGNALING",class:"pipen-report-table-image-descr"}}),U=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/fgsea_HALLMARK_IL6_JAK_STAT3_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/fgsea_HALLMARK_IL6_JAK_STAT3_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),k=new $a({props:{title:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",class:"pipen-report-table-image-descr"}}),q=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/fgsea_HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/fgsea_HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),W=new $a({props:{title:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",class:"pipen-report-table-image-descr"}}),Z=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/fgsea_HALLMARK_CHOLESTEROL_HOMEOSTASIS.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/fgsea_HALLMARK_CHOLESTEROL_HOMEOSTASIS.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new $a({props:{title:"HALLMARK_WNT_BETA_CATENIN_SIGNALING",class:"pipen-report-table-image-descr"}}),se=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/fgsea_HALLMARK_WNT_BETA_CATENIN_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/fgsea_HALLMARK_WNT_BETA_CATENIN_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),ie=new $a({props:{title:"HALLMARK_ANDROGEN_RESPONSE",class:"pipen-report-table-image-descr"}}),pe=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/fgsea_HALLMARK_ANDROGEN_RESPONSE.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Cycling-T-cells/fgsea_HALLMARK_ANDROGEN_RESPONSE.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=r("div"),n=r("div"),u(a.$$.fragment),s=S(),u(p.$$.fragment),c=S(),g=r("div"),u(m.$$.fragment),f=S(),u(E.$$.fragment),L=S(),R=r("div"),u(h.$$.fragment),T=S(),u(N.$$.fragment),P=S(),M=r("div"),u(v.$$.fragment),I=S(),u(y.$$.fragment),G=S(),x=r("div"),u(w.$$.fragment),H=S(),u(F.$$.fragment),B=S(),O=r("div"),u(D.$$.fragment),K=S(),u(U.$$.fragment),V=S(),z=r("div"),u(k.$$.fragment),X=S(),u(q.$$.fragment),j=S(),Y=r("div"),u(W.$$.fragment),J=S(),u(Z.$$.fragment),Q=S(),te=r("div"),u(ne.$$.fragment),ae=S(),u(se.$$.fragment),re=S(),le=r("div"),u(ie.$$.fragment),oe=S(),u(pe.$$.fragment),l(t,"class","pipen-report-table-of-images"),ee(t,"grid-template-columns","repeat(2, auto)")},m(e,r){i(e,t,r),C(t,n),A(a,n,null),C(n,s),A(p,n,null),C(t,c),C(t,g),A(m,g,null),C(g,f),A(E,g,null),C(t,L),C(t,R),A(h,R,null),C(R,T),A(N,R,null),C(t,P),C(t,M),A(v,M,null),C(M,I),A(y,M,null),C(t,G),C(t,x),A(w,x,null),C(x,H),A(F,x,null),C(t,B),C(t,O),A(D,O,null),C(O,K),A(U,O,null),C(t,V),C(t,z),A(k,z,null),C(z,X),A(q,z,null),C(t,j),C(t,Y),A(W,Y,null),C(Y,J),A(Z,Y,null),C(t,Q),C(t,te),A(ne,te,null),C(te,ae),A(se,te,null),C(t,re),C(t,le),A(ie,le,null),C(le,oe),A(pe,le,null),ce=!0},p:b,i(e){ce||(o(a.$$.fragment,e),o(p.$$.fragment,e),o(m.$$.fragment,e),o(E.$$.fragment,e),o(h.$$.fragment,e),o(N.$$.fragment,e),o(v.$$.fragment,e),o(y.$$.fragment,e),o(w.$$.fragment,e),o(F.$$.fragment,e),o(D.$$.fragment,e),o(U.$$.fragment,e),o(k.$$.fragment,e),o(q.$$.fragment,e),o(W.$$.fragment,e),o(Z.$$.fragment,e),o(ne.$$.fragment,e),o(se.$$.fragment,e),o(ie.$$.fragment,e),o(pe.$$.fragment,e),ce=!0)},o(e){_(a.$$.fragment,e),_(p.$$.fragment,e),_(m.$$.fragment,e),_(E.$$.fragment,e),_(h.$$.fragment,e),_(N.$$.fragment,e),_(v.$$.fragment,e),_(y.$$.fragment,e),_(w.$$.fragment,e),_(F.$$.fragment,e),_(D.$$.fragment,e),_(U.$$.fragment,e),_(k.$$.fragment,e),_(q.$$.fragment,e),_(W.$$.fragment,e),_(Z.$$.fragment,e),_(ne.$$.fragment,e),_(se.$$.fragment,e),_(ie.$$.fragment,e),_(pe.$$.fragment,e),ce=!1},d(e){e&&d(t),$(a),$(p),$(m),$(E),$(h),$(N),$(v),$(y),$(w),$(F),$(D),$(U),$(k),$(q),$(W),$(Z),$(ne),$(se),$(ie),$(pe)}}}function ba(e){let t,n,a,s;return t=new Ge({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[Ta]},$$scope:{ctx:e}}}),a=new Ge({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[Na]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function Pa(e){let t,n,a,s;return t=new fe({props:{label:"Plot"}}),a=new fe({props:{label:"Table"}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p:b,i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function Ma(e){let t;return{c(){t=h("This table presents a comprehensive summary of the top enriched pathways derived from the fgsea. Each row corresponds to a pathway, and the gene ranks are shown based on the ranking metric used in the analysis. The enrichment score, p-value, and adjusted p-value are also provided to assess the significance of the enrichment.")},m(e,n){i(e,t,n)},d(e){e&&d(t)}}}function va(e){let t,n,a,s;return t=new $a({props:{$$slots:{default:[Ma]},$$scope:{ctx:e}}}),a=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/gsea_table.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/gsea_table.pdf",width:1e3,height:760}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function Ia(e){let t;return{c(){t=h("This plot represents the GSEA results for a specified gene set, illustrating the distribution and impact of the gene set along the ranked list of genes. The running enrichment score curve shows the cumulative enrichment score as genes from the input list are encountered. Positive peaks on the curve indicate regions where members of the gene set are predominantly found.")},m(e,n){i(e,t,n)},d(e){e&&d(t)}}}function ya(e){let t,n,a,s;return t=new $a({props:{$$slots:{default:[Ia]},$$scope:{ctx:e}}}),a=new ma({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/fgsea.txt",data:[{pathway:"HALLMARK_INTERFERON_ALPHA_RESPONSE",pval:1e-4,padj:8e-4,log2err:.5573,ES:.6595,NES:1.7229,size:96,leadingEdge:"ISG20,SELL,B2M,IFITM2,PSMB9,LY6E,RIPK2,UBE2L6,PSME1,GBP2,PSMB8,PSME2,IFITM3,ISG15,IRF7,CD74,TAP1,GBP4,IFI35,IL7,OASL,IFI27,PARP9,IFI30,LGALS3BP,TMEM140,PARP14,RTP4,IRF1,CASP1,IL15,CMPK2,IRF9,IFI44L,SLC25A28,HERC6,HELZ2,IL4R,PARP12,MOV10,USP18,EPSTI1,LAP3",id:0},{pathway:"HALLMARK_ANGIOGENESIS",pval:.0179,padj:.1256,log2err:.3525,ES:.6647,NES:1.4724,size:36,leadingEdge:"S100A4,TIMP1,PDGFA,FGFR1,STC1,JAG1,VEGFA,COL5A2",id:1},{pathway:"HALLMARK_IL6_JAK_STAT3_SIGNALING",pval:.0967,padj:.4513,log2err:.2878,ES:.4644,NES:1.1935,size:87,leadingEdge:"SOCS1,TNFRSF1B,STAT1,IL18R1,CXCL13,TGFB1,IL7,IL12RB1,IL10RB,IL1R2,IL9R,IRF1,IRF9,IL1R1,CSF2,IL13RA1,PLA2G2A,IL4R,MAP3K8,LEPR,IL15RA,LTB,PIK3R5",id:2},{pathway:"HALLMARK_PROTEIN_SECRETION",pval:.1296,padj:.4536,log2err:.1357,ES:-.5301,NES:-1.1878,size:95,leadingEdge:"KRT18,TMED10,ANP32E,COPB1,COG2,SGMS1,SNX2,RAB14,SCAMP1,ICA1,TMED2,RER1,SCAMP3,GOSR2,STX16,ARFGEF1,M6PR,ARCN1,SEC24D,CLTC,CLCN3,GALC,PPT1,ATP6V1H,ZW10,COPB2,RAB5A,TOM1L1,TSPAN8,GOLGA4,CLTA,SNAP23,AP3B1,GLA,RAB22A,ARFIP1,VAMP3,TSG101,GBF1,SEC31A,CD63,OCRL,RPS6KA3,AP1G1,TMX1,CTSC,PAM,VPS45,ATP7A,SEC22B,MON2,NAPA",id:3},{pathway:"HALLMARK_WNT_BETA_CATENIN_SIGNALING",pval:.2391,padj:.5917,log2err:.0989,ES:-.5646,NES:-1.1458,size:42,leadingEdge:"TCF7,DLL1,CTNNB1,NCOR2,LEF1,NCSTN,NUMB,TP53,RBPJ,PSEN2,CUL1,HDAC2,SKP2,DVL2,NOTCH1,WNT1,HDAC11,PTCH1,NKD1,AXIN2",id:4},{pathway:"HALLMARK_MYC_TARGETS_V2",pval:.2536,padj:.5917,log2err:.0931,ES:-.5347,NES:-1.134,size:58,leadingEdge:"CBX3,TFB2M,PPAN,MCM5,NDUFAF4,SUPV3L1,HSPE1,MPHOSPH10,CDK4,PES1,RRP9,RRP12,MYBBP1A,IMP4,TMEM97,TCOF1,GNL3,PRMT3,WDR43,RABEPK,NOP2,FARSA,SRM,SLC19A1,DDX18,PHB,GRWD1,IPO4,NOC4L,RCL1,UNG,PA2G4,PPRC1,NOP16,MRTO4,MYC,DCTPP1,SLC29A2,HSPD1,MAP3K6,PLK4",id:5},{pathway:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",pval:.3805,padj:.6949,log2err:.1492,ES:.4093,NES:1.0179,size:73,leadingEdge:"S100A11,ANXA5,ATF3,LDLR,SC5D,LGALS3,ETHE1,FDFT1,FADS2,ABCA2,JAG1,PPARG,MAL2,FASN,TP53INP1,ALCAM,STX5,PLAUR,MVD,EBP,ERRFI1,SCD,STARD4,FDPS,ATF5,GNAI1,CYP51A1,PCYT2,CXCL16,HSD17B7",id:6},{pathway:"HALLMARK_ANDROGEN_RESPONSE",pval:.3971,padj:.6949,log2err:.1542,ES:.3884,NES:1.0143,size:99,leadingEdge:"FKBP5,SAT1,B2M,B4GALT1,CCND3,ARID5B,FADS1,RAB4A,IQGAP2,HPGD,INSIG1,MAF,DBI,CDC14B,KRT19,ZMIZ1,LMAN1,MYL12A,DNAJB9,SRF,SGK1,CAMKK2,TMEM50A,SCD,SORD,APPBP2,SRP19,HOMER2,MERTK,NKX3-1,CCND1",id:7},{pathway:"HALLMARK_PANCREAS_BETA_CELLS",pval:.5432,padj:.8353,log2err:.0557,ES:-.4765,NES:-.958,size:40,leadingEdge:"SRP9,ELP4,SRPRB,MAFB,NKX2-2,GCK,SRP14",id:8},{pathway:"HALLMARK_NOTCH_SIGNALING",pval:.6466,padj:.8353,log2err:.0492,ES:-.4605,NES:-.8902,size:32,leadingEdge:"MAML2,DLL1,HES1,SKP1,DTX2,FBXW11,PSEN2,TCF7L2,CUL1,PRKCA",id:9},{pathway:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",pval:.6719,padj:.8353,log2err:.1003,ES:.382,NES:.9017,size:49,leadingEdge:"SOD2,TXN,GLRX,SOD1,SBNO2,CDKN2D,ABCC1,PFKP,SRXN1,HMOX2,LAMTOR5,SELENOS,PRDX6,NDUFB4,GCLC,PRDX4,NQO1,ATOX1,PRDX1",id:10},{pathway:"HALLMARK_APICAL_SURFACE",pval:.7401,padj:.8353,log2err:.0418,ES:-.4207,NES:-.8554,size:44,leadingEdge:"APP,GSTM3,HSPB1,BRCA1,ADIPOR2,GAS1,AFAP1L2,LYPD3,AKAP7,LYN,SULF2,ATP8B1",id:11},{pathway:"HALLMARK_HEDGEHOG_SIGNALING",pval:.7756,padj:.8353,log2err:.0405,ES:-.4209,NES:-.8277,size:36,leadingEdge:"CDK5R1,ETS2,CELSR1,TLE3,NF1,NRCAM,CRMP1",id:12},{pathway:"HALLMARK_TGF_BETA_SIGNALING",pval:.8999,padj:.8999,log2err:.0318,ES:-.3627,NES:-.7593,size:54,leadingEdge:"ID3,JUNB,TGFBR1,KLF10,CTNNB1,NCOR2,FNTA,CDK9,SMAD1,PPM1A,XIAP,SLC20A1,MAP3K7,THBS1",id:13}]}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function Ga(e){let t,n,a,s;return t=new ue({props:{$$slots:{default:[va]},$$scope:{ctx:e}}}),a=new ue({props:{$$slots:{default:[ya]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function xa(e){let t,n;return t=new me({props:{$$slots:{content:[Ga],default:[Pa]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function wa(e){let t,n,a,s,p,c,g,m,f,E,L,R,h,T,N,P,M,v,I,y,G,x,w,H,F,B,O,D,K,U,V,z,k,X,q,j,Y,W,J,Z,Q,te,ne,ae,se,re,le,ie,oe,pe,ce;return a=new $a({props:{title:"HALLMARK_INTERFERON_ALPHA_RESPONSE",class:"pipen-report-table-image-descr"}}),p=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/fgsea_HALLMARK_INTERFERON_ALPHA_RESPONSE.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/fgsea_HALLMARK_INTERFERON_ALPHA_RESPONSE.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),m=new $a({props:{title:"HALLMARK_ANGIOGENESIS",class:"pipen-report-table-image-descr"}}),E=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/fgsea_HALLMARK_ANGIOGENESIS.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/fgsea_HALLMARK_ANGIOGENESIS.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),h=new $a({props:{title:"HALLMARK_IL6_JAK_STAT3_SIGNALING",class:"pipen-report-table-image-descr"}}),N=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/fgsea_HALLMARK_IL6_JAK_STAT3_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/fgsea_HALLMARK_IL6_JAK_STAT3_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),v=new $a({props:{title:"HALLMARK_PROTEIN_SECRETION",class:"pipen-report-table-image-descr"}}),y=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/fgsea_HALLMARK_PROTEIN_SECRETION.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/fgsea_HALLMARK_PROTEIN_SECRETION.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),w=new $a({props:{title:"HALLMARK_WNT_BETA_CATENIN_SIGNALING",class:"pipen-report-table-image-descr"}}),F=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/fgsea_HALLMARK_WNT_BETA_CATENIN_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/fgsea_HALLMARK_WNT_BETA_CATENIN_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),D=new $a({props:{title:"HALLMARK_MYC_TARGETS_V2",class:"pipen-report-table-image-descr"}}),U=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/fgsea_HALLMARK_MYC_TARGETS_V2.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/fgsea_HALLMARK_MYC_TARGETS_V2.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),k=new $a({props:{title:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",class:"pipen-report-table-image-descr"}}),q=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/fgsea_HALLMARK_CHOLESTEROL_HOMEOSTASIS.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/fgsea_HALLMARK_CHOLESTEROL_HOMEOSTASIS.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),W=new $a({props:{title:"HALLMARK_ANDROGEN_RESPONSE",class:"pipen-report-table-image-descr"}}),Z=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/fgsea_HALLMARK_ANDROGEN_RESPONSE.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/fgsea_HALLMARK_ANDROGEN_RESPONSE.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new $a({props:{title:"HALLMARK_PANCREAS_BETA_CELLS",class:"pipen-report-table-image-descr"}}),se=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/fgsea_HALLMARK_PANCREAS_BETA_CELLS.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/fgsea_HALLMARK_PANCREAS_BETA_CELLS.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),ie=new $a({props:{title:"HALLMARK_NOTCH_SIGNALING",class:"pipen-report-table-image-descr"}}),pe=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/fgsea_HALLMARK_NOTCH_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Follicular-helper-T-cells/fgsea_HALLMARK_NOTCH_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=r("div"),n=r("div"),u(a.$$.fragment),s=S(),u(p.$$.fragment),c=S(),g=r("div"),u(m.$$.fragment),f=S(),u(E.$$.fragment),L=S(),R=r("div"),u(h.$$.fragment),T=S(),u(N.$$.fragment),P=S(),M=r("div"),u(v.$$.fragment),I=S(),u(y.$$.fragment),G=S(),x=r("div"),u(w.$$.fragment),H=S(),u(F.$$.fragment),B=S(),O=r("div"),u(D.$$.fragment),K=S(),u(U.$$.fragment),V=S(),z=r("div"),u(k.$$.fragment),X=S(),u(q.$$.fragment),j=S(),Y=r("div"),u(W.$$.fragment),J=S(),u(Z.$$.fragment),Q=S(),te=r("div"),u(ne.$$.fragment),ae=S(),u(se.$$.fragment),re=S(),le=r("div"),u(ie.$$.fragment),oe=S(),u(pe.$$.fragment),l(t,"class","pipen-report-table-of-images"),ee(t,"grid-template-columns","repeat(2, auto)")},m(e,r){i(e,t,r),C(t,n),A(a,n,null),C(n,s),A(p,n,null),C(t,c),C(t,g),A(m,g,null),C(g,f),A(E,g,null),C(t,L),C(t,R),A(h,R,null),C(R,T),A(N,R,null),C(t,P),C(t,M),A(v,M,null),C(M,I),A(y,M,null),C(t,G),C(t,x),A(w,x,null),C(x,H),A(F,x,null),C(t,B),C(t,O),A(D,O,null),C(O,K),A(U,O,null),C(t,V),C(t,z),A(k,z,null),C(z,X),A(q,z,null),C(t,j),C(t,Y),A(W,Y,null),C(Y,J),A(Z,Y,null),C(t,Q),C(t,te),A(ne,te,null),C(te,ae),A(se,te,null),C(t,re),C(t,le),A(ie,le,null),C(le,oe),A(pe,le,null),ce=!0},p:b,i(e){ce||(o(a.$$.fragment,e),o(p.$$.fragment,e),o(m.$$.fragment,e),o(E.$$.fragment,e),o(h.$$.fragment,e),o(N.$$.fragment,e),o(v.$$.fragment,e),o(y.$$.fragment,e),o(w.$$.fragment,e),o(F.$$.fragment,e),o(D.$$.fragment,e),o(U.$$.fragment,e),o(k.$$.fragment,e),o(q.$$.fragment,e),o(W.$$.fragment,e),o(Z.$$.fragment,e),o(ne.$$.fragment,e),o(se.$$.fragment,e),o(ie.$$.fragment,e),o(pe.$$.fragment,e),ce=!0)},o(e){_(a.$$.fragment,e),_(p.$$.fragment,e),_(m.$$.fragment,e),_(E.$$.fragment,e),_(h.$$.fragment,e),_(N.$$.fragment,e),_(v.$$.fragment,e),_(y.$$.fragment,e),_(w.$$.fragment,e),_(F.$$.fragment,e),_(D.$$.fragment,e),_(U.$$.fragment,e),_(k.$$.fragment,e),_(q.$$.fragment,e),_(W.$$.fragment,e),_(Z.$$.fragment,e),_(ne.$$.fragment,e),_(se.$$.fragment,e),_(ie.$$.fragment,e),_(pe.$$.fragment,e),ce=!1},d(e){e&&d(t),$(a),$(p),$(m),$(E),$(h),$(N),$(v),$(y),$(w),$(F),$(D),$(U),$(k),$(q),$(W),$(Z),$(ne),$(se),$(ie),$(pe)}}}function Ha(e){let t,n,a,s;return t=new Ge({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[xa]},$$scope:{ctx:e}}}),a=new Ge({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[wa]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function Fa(e){let t,n,a,s;return t=new fe({props:{label:"Plot"}}),a=new fe({props:{label:"Table"}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p:b,i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function Ba(e){let t;return{c(){t=h("This table presents a comprehensive summary of the top enriched pathways derived from the fgsea. Each row corresponds to a pathway, and the gene ranks are shown based on the ranking metric used in the analysis. The enrichment score, p-value, and adjusted p-value are also provided to assess the significance of the enrichment.")},m(e,n){i(e,t,n)},d(e){e&&d(t)}}}function Oa(e){let t,n,a,s;return t=new $a({props:{$$slots:{default:[Ba]},$$scope:{ctx:e}}}),a=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/gsea_table.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/gsea_table.pdf",width:1e3,height:760}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function Da(e){let t;return{c(){t=h("This plot represents the GSEA results for a specified gene set, illustrating the distribution and impact of the gene set along the ranked list of genes. The running enrichment score curve shows the cumulative enrichment score as genes from the input list are encountered. Positive peaks on the curve indicate regions where members of the gene set are predominantly found.")},m(e,n){i(e,t,n)},d(e){e&&d(t)}}}function Ka(e){let t,n,a,s;return t=new $a({props:{$$slots:{default:[Da]},$$scope:{ctx:e}}}),a=new ma({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/fgsea.txt",data:[{pathway:"HALLMARK_INTERFERON_ALPHA_RESPONSE",pval:.008,padj:.1125,log2err:.3807,ES:.5734,NES:1.4385,size:96,leadingEdge:"ISG20,CSF1,GBP2,IFITM3,B2M,TAP1,PSMB9,IRF1,TMEM140,UBE2L6,RIPK2,ELF1,PLSCR1,CMPK2,IRF7,CCRL2,TRIM25,LY6E,CASP8,IL4R,PARP9,LAMP3,EPSTI1,IFIT3,HERC6,SLC25A28,CMTR1,LPAR6,HELZ2,SP110,NCOA7,PSMB8,BST2,IFITM2",id:0},{pathway:"HALLMARK_ANGIOGENESIS",pval:.0344,padj:.2405,log2err:.3218,ES:-.7476,NES:-1.4124,size:36,leadingEdge:"S100A4,TIMP1",id:1},{pathway:"HALLMARK_IL6_JAK_STAT3_SIGNALING",pval:.0701,padj:.3273,log2err:.2878,ES:.4833,NES:1.1953,size:87,leadingEdge:"CSF1,PIM1,SOCS3,TNFRSF1B,CD38,MAP3K8,IRF1,STAT3,STAT1,IL6ST,IL17RA,HMOX1,IL4R,LTBR,PTPN11",id:2},{pathway:"HALLMARK_ANDROGEN_RESPONSE",pval:.194,padj:.66,log2err:.228,ES:.4361,NES:1.0999,size:99,leadingEdge:"FKBP5,ARID5B,PLPP1,B2M,SAT1,HPGD,SGK1,STK39,SCD,ZBTB10,INSIG1,MAF,KRT8,ELK4,ZMIZ1,DHCR24,SRF,SPCS3,IQGAP2,CCND3,SEC24D,SLC38A2,PTK2B,VAPA,INPP4B,CDK6,HMGCR,GSR",id:3},{pathway:"HALLMARK_NOTCH_SIGNALING",pval:.3137,padj:.66,log2err:.085,ES:-.5943,NES:-1.103,size:32,leadingEdge:"DLL1,RBX1,PRKCA,HES1,CUL1,PSENEN,FZD7,KAT2A,SKP1,MAML2,APH1A,PSEN2,ARRB1,NOTCH2",id:4},{pathway:"HALLMARK_WNT_BETA_CATENIN_SIGNALING",pval:.316,padj:.66,log2err:.0834,ES:-.5625,NES:-1.0944,size:42,leadingEdge:"DLL1,TCF7,LEF1,HDAC2,HEY1,CUL1,AXIN1,NKD1,FZD8,KAT2A,PSEN2,DVL2,MAML1,NCSTN,TP53,HDAC5",id:5},{pathway:"HALLMARK_APICAL_SURFACE",pval:.3833,padj:.66,log2err:.0725,ES:-.5357,NES:-1.0487,size:44,leadingEdge:"CD160,FLOT2,LYPD3,HSPB1,ADIPOR2,GSTM3,IL2RG,PLAUR,SCUBE1",id:6},{pathway:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",pval:.3903,padj:.66,log2err:.0715,ES:-.5274,NES:-1.0473,size:49,leadingEdge:"FTL,PRNP,PRDX1,JUNB,CAT,CDKN2D,HHEX,OXSR1,GPX4,PTPA,EGLN2,TXNRD1,PDLIM1,SOD2,G6PD,TXN,GLRX,PRDX4,GCLM,GCLC",id:7},{pathway:"HALLMARK_PROTEIN_SECRETION",pval:.4666,padj:.66,log2err:.0596,ES:-.4658,NES:-.9997,size:95,leadingEdge:"ABCA1,ATP1A1,CD63,GLA,PPT1,TMED2,SEC31A,LMAN1,TSG101,RAB14,GOSR2,BNIP3,RAB22A,AP2B1,COPE,KIF1B,ATP7A,AP2M1,TMED10,SCAMP3,VPS45,MAPK1,M6PR,SNX2,OCRL,ARCN1,COPB2,STX16,SGMS1,ARFIP1,DNM1L,COG2,GOLGA4,TPD52,SEC22B,PAM",id:8},{pathway:"HALLMARK_HEDGEHOG_SIGNALING",pval:.4714,padj:.66,log2err:.1172,ES:.4516,NES:.9655,size:36,leadingEdge:"TLE1,ADGRG1,ACHE,MYH9,DPYSL2,ETS2,CDK6,PTCH1",id:9},{pathway:"HALLMARK_MYC_TARGETS_V2",pval:.7243,padj:.9015,log2err:.0407,ES:-.4268,NES:-.869,size:58,leadingEdge:"PA2G4,MAP3K6,PES1,MYBBP1A,GRWD1,PLK4,CBX3,NIP7,LAS1L,MCM5,RRP12,SLC19A1,MRTO4,TMEM97,RABEPK,TCOF1,TBRG4,DUSP2,NOLC1,MPHOSPH10,MCM4,AIMP2,NOP2,FARSA,SRM,SUPV3L1,IMP4",id:10},{pathway:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",pval:.7727,padj:.9015,log2err:.101,ES:.3732,NES:.8894,size:73,leadingEdge:"TP53INP1,ATF3,LDLR,STX5,PLSCR1,SCD,GSTM2,FASN,NFIL3,PNRC1,FADS2,STARD4,FABP5,PDK3,ETHE1,SREBF2,GUSB,FDFT1,HMGCR,NSDHL,ALCAM,CTNNB1",id:11},{pathway:"HALLMARK_PANCREAS_BETA_CELLS",pval:.8432,padj:.9081,log2err:.0355,ES:-.4081,NES:-.7879,size:40,leadingEdge:"STXBP1,AKT3,NEUROG3,ELP4",id:12},{pathway:"HALLMARK_TGF_BETA_SIGNALING",pval:.9348,padj:.9348,log2err:.0298,ES:-.3713,NES:-.7461,size:54,leadingEdge:"JUNB,HDAC1,PPP1CA,WWTR1,ENG,KLF10,APC,TGFBR1,BMPR2,SMAD3,FNTA,NOG,BMP2,IFNGR2,BCAR3,CDK9,SERPINE1,SMAD7,MAP3K7,SMURF2,RHOA,SPTBN1,SMURF1,SKIL",id:13}]}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function Ua(e){let t,n,a,s;return t=new ue({props:{$$slots:{default:[Oa]},$$scope:{ctx:e}}}),a=new ue({props:{$$slots:{default:[Ka]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function Va(e){let t,n;return t=new me({props:{$$slots:{content:[Ua],default:[Fa]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function za(e){let t,n,a,s,p,c,g,m,f,E,L,R,h,T,N,P,M,v,I,y,G,x,w,H,F,B,O,D,K,U,V,z,k,X,q,j,Y,W,J,Z,Q,te,ne,ae,se,re,le,ie,oe,pe,ce;return a=new $a({props:{title:"HALLMARK_INTERFERON_ALPHA_RESPONSE",class:"pipen-report-table-image-descr"}}),p=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/fgsea_HALLMARK_INTERFERON_ALPHA_RESPONSE.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/fgsea_HALLMARK_INTERFERON_ALPHA_RESPONSE.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),m=new $a({props:{title:"HALLMARK_ANGIOGENESIS",class:"pipen-report-table-image-descr"}}),E=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/fgsea_HALLMARK_ANGIOGENESIS.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/fgsea_HALLMARK_ANGIOGENESIS.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),h=new $a({props:{title:"HALLMARK_IL6_JAK_STAT3_SIGNALING",class:"pipen-report-table-image-descr"}}),N=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/fgsea_HALLMARK_IL6_JAK_STAT3_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/fgsea_HALLMARK_IL6_JAK_STAT3_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),v=new $a({props:{title:"HALLMARK_ANDROGEN_RESPONSE",class:"pipen-report-table-image-descr"}}),y=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/fgsea_HALLMARK_ANDROGEN_RESPONSE.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/fgsea_HALLMARK_ANDROGEN_RESPONSE.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),w=new $a({props:{title:"HALLMARK_NOTCH_SIGNALING",class:"pipen-report-table-image-descr"}}),F=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/fgsea_HALLMARK_NOTCH_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/fgsea_HALLMARK_NOTCH_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),D=new $a({props:{title:"HALLMARK_WNT_BETA_CATENIN_SIGNALING",class:"pipen-report-table-image-descr"}}),U=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/fgsea_HALLMARK_WNT_BETA_CATENIN_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/fgsea_HALLMARK_WNT_BETA_CATENIN_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),k=new $a({props:{title:"HALLMARK_APICAL_SURFACE",class:"pipen-report-table-image-descr"}}),q=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/fgsea_HALLMARK_APICAL_SURFACE.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/fgsea_HALLMARK_APICAL_SURFACE.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),W=new $a({props:{title:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",class:"pipen-report-table-image-descr"}}),Z=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/fgsea_HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/fgsea_HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new $a({props:{title:"HALLMARK_PROTEIN_SECRETION",class:"pipen-report-table-image-descr"}}),se=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/fgsea_HALLMARK_PROTEIN_SECRETION.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/fgsea_HALLMARK_PROTEIN_SECRETION.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),ie=new $a({props:{title:"HALLMARK_HEDGEHOG_SIGNALING",class:"pipen-report-table-image-descr"}}),pe=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/fgsea_HALLMARK_HEDGEHOG_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-gamma-delta-T-cells/fgsea_HALLMARK_HEDGEHOG_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=r("div"),n=r("div"),u(a.$$.fragment),s=S(),u(p.$$.fragment),c=S(),g=r("div"),u(m.$$.fragment),f=S(),u(E.$$.fragment),L=S(),R=r("div"),u(h.$$.fragment),T=S(),u(N.$$.fragment),P=S(),M=r("div"),u(v.$$.fragment),I=S(),u(y.$$.fragment),G=S(),x=r("div"),u(w.$$.fragment),H=S(),u(F.$$.fragment),B=S(),O=r("div"),u(D.$$.fragment),K=S(),u(U.$$.fragment),V=S(),z=r("div"),u(k.$$.fragment),X=S(),u(q.$$.fragment),j=S(),Y=r("div"),u(W.$$.fragment),J=S(),u(Z.$$.fragment),Q=S(),te=r("div"),u(ne.$$.fragment),ae=S(),u(se.$$.fragment),re=S(),le=r("div"),u(ie.$$.fragment),oe=S(),u(pe.$$.fragment),l(t,"class","pipen-report-table-of-images"),ee(t,"grid-template-columns","repeat(2, auto)")},m(e,r){i(e,t,r),C(t,n),A(a,n,null),C(n,s),A(p,n,null),C(t,c),C(t,g),A(m,g,null),C(g,f),A(E,g,null),C(t,L),C(t,R),A(h,R,null),C(R,T),A(N,R,null),C(t,P),C(t,M),A(v,M,null),C(M,I),A(y,M,null),C(t,G),C(t,x),A(w,x,null),C(x,H),A(F,x,null),C(t,B),C(t,O),A(D,O,null),C(O,K),A(U,O,null),C(t,V),C(t,z),A(k,z,null),C(z,X),A(q,z,null),C(t,j),C(t,Y),A(W,Y,null),C(Y,J),A(Z,Y,null),C(t,Q),C(t,te),A(ne,te,null),C(te,ae),A(se,te,null),C(t,re),C(t,le),A(ie,le,null),C(le,oe),A(pe,le,null),ce=!0},p:b,i(e){ce||(o(a.$$.fragment,e),o(p.$$.fragment,e),o(m.$$.fragment,e),o(E.$$.fragment,e),o(h.$$.fragment,e),o(N.$$.fragment,e),o(v.$$.fragment,e),o(y.$$.fragment,e),o(w.$$.fragment,e),o(F.$$.fragment,e),o(D.$$.fragment,e),o(U.$$.fragment,e),o(k.$$.fragment,e),o(q.$$.fragment,e),o(W.$$.fragment,e),o(Z.$$.fragment,e),o(ne.$$.fragment,e),o(se.$$.fragment,e),o(ie.$$.fragment,e),o(pe.$$.fragment,e),ce=!0)},o(e){_(a.$$.fragment,e),_(p.$$.fragment,e),_(m.$$.fragment,e),_(E.$$.fragment,e),_(h.$$.fragment,e),_(N.$$.fragment,e),_(v.$$.fragment,e),_(y.$$.fragment,e),_(w.$$.fragment,e),_(F.$$.fragment,e),_(D.$$.fragment,e),_(U.$$.fragment,e),_(k.$$.fragment,e),_(q.$$.fragment,e),_(W.$$.fragment,e),_(Z.$$.fragment,e),_(ne.$$.fragment,e),_(se.$$.fragment,e),_(ie.$$.fragment,e),_(pe.$$.fragment,e),ce=!1},d(e){e&&d(t),$(a),$(p),$(m),$(E),$(h),$(N),$(v),$(y),$(w),$(F),$(D),$(U),$(k),$(q),$(W),$(Z),$(ne),$(se),$(ie),$(pe)}}}function ka(e){let t,n,a,s;return t=new Ge({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[Va]},$$scope:{ctx:e}}}),a=new Ge({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[za]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function Xa(e){let t,n,a,s;return t=new fe({props:{label:"Plot"}}),a=new fe({props:{label:"Table"}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p:b,i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function qa(e){let t;return{c(){t=h("This table presents a comprehensive summary of the top enriched pathways derived from the fgsea. Each row corresponds to a pathway, and the gene ranks are shown based on the ranking metric used in the analysis. The enrichment score, p-value, and adjusted p-value are also provided to assess the significance of the enrichment.")},m(e,n){i(e,t,n)},d(e){e&&d(t)}}}function ja(e){let t,n,a,s;return t=new $a({props:{$$slots:{default:[qa]},$$scope:{ctx:e}}}),a=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/gsea_table.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/gsea_table.pdf",width:1e3,height:760}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function Ya(e){let t;return{c(){t=h("This plot represents the GSEA results for a specified gene set, illustrating the distribution and impact of the gene set along the ranked list of genes. The running enrichment score curve shows the cumulative enrichment score as genes from the input list are encountered. Positive peaks on the curve indicate regions where members of the gene set are predominantly found.")},m(e,n){i(e,t,n)},d(e){e&&d(t)}}}function Wa(e){let t,n,a,s;return t=new $a({props:{$$slots:{default:[Ya]},$$scope:{ctx:e}}}),a=new ma({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/fgsea.txt",data:[{pathway:"HALLMARK_INTERFERON_ALPHA_RESPONSE",pval:0,padj:2e-4,log2err:.5933,ES:.6907,NES:1.7933,size:96,leadingEdge:"PSMB9,PSMB8,PSME1,IFITM2,UBE2L6,LY6E,GBP2,ISG20,CD74,PSME2,USP18,OASL,B2M,GBP4,TAP1,LAP3,IRF7,SLC25A28,PSMA3,CASP8,UBA7,IFI35,MX1,IRF2,MOV10,RSAD2,PLSCR1,BST2,TMEM140,SP110,IL15,SELL,PARP9,IRF1,OAS1,IFITM1,IL7,TRIM25,IFI27,DHX58,HELZ2,IFITM3,ADAR,NMI,IFI30,NCOA7,TENT5A",id:0},{pathway:"HALLMARK_APICAL_SURFACE",pval:.0628,padj:.4395,log2err:.2878,ES:-.6043,NES:-1.3526,size:44,leadingEdge:"LYN,HSPB1,BRCA1,GSTM3,FLOT2,CRYBG1,CROCC,NCOA6,ADAM10,PLAUR,DCBLD2,LYPD3,TMEM8B,AFAP1L2,MDGA1,MAL,AKAP7,CD160",id:1},{pathway:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",pval:.1343,padj:.504,log2err:.1882,ES:.5135,NES:1.2095,size:49,leadingEdge:"FTL,SOD1,JUNB,SELENOS,NDUFA6,TXN,GLRX2,NDUFB4,PRDX6,PFKP,GLRX,PRDX2,G6PD,PRDX1,HMOX2,LAMTOR5",id:2},{pathway:"HALLMARK_PROTEIN_SECRETION",pval:.144,padj:.504,log2err:.1464,ES:-.4729,NES:-1.179,size:95,leadingEdge:"DNM1L,SCAMP1,ZW10,SGMS1,ANP32E,VAMP3,GNAS,AP3S1,SNX2,ARFGAP3,SH3GL2,ERGIC3,ADAM10,VPS45,CD63,ATP1A1,VAMP7,TPD52,ARFIP1,BNIP3,ARFGEF2,LMAN1,ARFGEF1,STAM,IGF2R,MON2,GALC,MAPK1,ATP6V1H,SCAMP3,SEC24D,ATP7A,COG2,CLN5,AP1G1,STX12,SEC22B,COPB2,LAMP2,VPS4B,YIPF6,COPB1,ARCN1,TMED10,RER1,RAB22A,USO1,KIF1B,CLTC,CLTA,STX16",id:3},{pathway:"HALLMARK_PANCREAS_BETA_CELLS",pval:.2409,padj:.6745,log2err:.138,ES:.5088,NES:1.1428,size:40,leadingEdge:"DPP4,SRP14,SPCS1,FOXO1,PAK3,GCK,VDR,SRPRB",id:4},{pathway:"HALLMARK_MYC_TARGETS_V2",pval:.3493,padj:.7682,log2err:.1106,ES:.4278,NES:1.0361,size:58,leadingEdge:"DUSP2,NOP16,SLC19A1,IPO4,WDR43,HK2,NOP56,PPRC1,WDR74,GNL3,PUS1,NPM1,SORD,HSPD1,PHB,HSPE1,PA2G4,DDX18,IMP4,SLC29A2,SRM",id:5},{pathway:"HALLMARK_TGF_BETA_SIGNALING",pval:.4777,padj:.7682,log2err:.0736,ES:-.4316,NES:-.9881,size:54,leadingEdge:"ENG,ID3,ID1,CTNNB1,WWTR1,MAP3K7,IFNGR2,SLC20A1,PPP1R15A,HDAC1,SMURF2,RHOA,TRIM33,APC,BMPR2,TGFBR1,SKIL,SPTBN1,FNTA,PMEPA1,SMURF1,NCOR2,PPM1A,KLF10,HIPK2,PPP1CA,ARID4B",id:6},{pathway:"HALLMARK_ANDROGEN_RESPONSE",pval:.5355,padj:.7682,log2err:.0648,ES:-.3855,NES:-.9622,size:99,leadingEdge:"GNAI3,RRP12,ANKH,APPBP2,TSC22D1,ITGAV,ACTN1,HMGCS1,CDK6,NCOA4,ABCC4,VAPA,ELK4,SMS,NGLY1,PLPP1,ACSL3,PTK2B,IDI1,KRT8,TPD52,NDRG1,LMAN1,MAF,TNFAIP8,HMGCR,XRCC6,AKT1,CAMKK2,CENPN,SLC38A2,SEC24D,XRCC5,ZBTB10,PMEPA1,PIAS1,SLC26A2,SGK1",id:7},{pathway:"HALLMARK_IL6_JAK_STAT3_SIGNALING",pval:.5855,padj:.7682,log2err:.0605,ES:-.3799,NES:-.9419,size:87,leadingEdge:"TNF,PIM1,IL17RA,FAS,JUN,IFNGR2,CRLF2,LTBR,ACVR1B,TLR2,IL4R,CD9,IL1R2,IFNAR1,IL1R1,CBL,EBI3,IL2RA,TNFRSF12A,GRB2,IL10RB,STAM2,IL9R,STAT2,MAP3K8,IL6ST,IRF9",id:8},{pathway:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",pval:.6122,padj:.7682,log2err:.0813,ES:.3705,NES:.9217,size:73,leadingEdge:"FABP5,FASN,FBXO6,NFIL3,STARD4,GLDC,PLSCR1,ATF5,ERRFI1,ANXA5,STX5,CHKA,MVD,FADS2,JAG1,ACTG1,CPEB2",id:9},{pathway:"HALLMARK_ANGIOGENESIS",pval:.6326,padj:.7682,log2err:.0602,ES:-.4199,NES:-.9037,size:36,leadingEdge:"TIMP1,ITGAV,VAV2,CCND2",id:10},{pathway:"HALLMARK_HEDGEHOG_SIGNALING",pval:.6742,padj:.7682,log2err:.0572,ES:-.4133,NES:-.8895,size:36,leadingEdge:"DPYSL2,L1CAM,CDK6,ADGRG1",id:11},{pathway:"HALLMARK_NOTCH_SIGNALING",pval:.7133,padj:.7682,log2err:.0545,ES:-.4057,NES:-.8526,size:32,leadingEdge:"HES1,TCF7L2,PSENEN,DTX2,KAT2A,NOTCH2,LFNG",id:12},{pathway:"HALLMARK_WNT_BETA_CATENIN_SIGNALING",pval:.7896,padj:.7896,log2err:.0486,ES:-.3682,NES:-.823,size:42,leadingEdge:"HDAC5,CTNNB1,HDAC11,SKP2,TCF7,CCND2,KAT2A",id:13}]}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function Ja(e){let t,n,a,s;return t=new ue({props:{$$slots:{default:[ja]},$$scope:{ctx:e}}}),a=new ue({props:{$$slots:{default:[Wa]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function Za(e){let t,n;return t=new me({props:{$$slots:{content:[Ja],default:[Xa]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Qa(e){let t,n,a,s,p,c,g,m,f,E,L,R,h,T,N,P,M,v,I,y,G,x,w,H,F,B,O,D,K,U,V,z,k,X,q,j,Y,W,J,Z,Q,te,ne,ae,se,re,le,ie,oe,pe,ce;return a=new $a({props:{title:"HALLMARK_INTERFERON_ALPHA_RESPONSE",class:"pipen-report-table-image-descr"}}),p=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/fgsea_HALLMARK_INTERFERON_ALPHA_RESPONSE.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/fgsea_HALLMARK_INTERFERON_ALPHA_RESPONSE.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),m=new $a({props:{title:"HALLMARK_APICAL_SURFACE",class:"pipen-report-table-image-descr"}}),E=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/fgsea_HALLMARK_APICAL_SURFACE.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/fgsea_HALLMARK_APICAL_SURFACE.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),h=new $a({props:{title:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",class:"pipen-report-table-image-descr"}}),N=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/fgsea_HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/fgsea_HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),v=new $a({props:{title:"HALLMARK_PROTEIN_SECRETION",class:"pipen-report-table-image-descr"}}),y=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/fgsea_HALLMARK_PROTEIN_SECRETION.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/fgsea_HALLMARK_PROTEIN_SECRETION.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),w=new $a({props:{title:"HALLMARK_PANCREAS_BETA_CELLS",class:"pipen-report-table-image-descr"}}),F=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/fgsea_HALLMARK_PANCREAS_BETA_CELLS.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/fgsea_HALLMARK_PANCREAS_BETA_CELLS.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),D=new $a({props:{title:"HALLMARK_MYC_TARGETS_V2",class:"pipen-report-table-image-descr"}}),U=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/fgsea_HALLMARK_MYC_TARGETS_V2.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/fgsea_HALLMARK_MYC_TARGETS_V2.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),k=new $a({props:{title:"HALLMARK_TGF_BETA_SIGNALING",class:"pipen-report-table-image-descr"}}),q=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/fgsea_HALLMARK_TGF_BETA_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/fgsea_HALLMARK_TGF_BETA_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),W=new $a({props:{title:"HALLMARK_ANDROGEN_RESPONSE",class:"pipen-report-table-image-descr"}}),Z=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/fgsea_HALLMARK_ANDROGEN_RESPONSE.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/fgsea_HALLMARK_ANDROGEN_RESPONSE.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new $a({props:{title:"HALLMARK_IL6_JAK_STAT3_SIGNALING",class:"pipen-report-table-image-descr"}}),se=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/fgsea_HALLMARK_IL6_JAK_STAT3_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/fgsea_HALLMARK_IL6_JAK_STAT3_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),ie=new $a({props:{title:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",class:"pipen-report-table-image-descr"}}),pe=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/fgsea_HALLMARK_CHOLESTEROL_HOMEOSTASIS.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Plasmablasts/fgsea_HALLMARK_CHOLESTEROL_HOMEOSTASIS.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=r("div"),n=r("div"),u(a.$$.fragment),s=S(),u(p.$$.fragment),c=S(),g=r("div"),u(m.$$.fragment),f=S(),u(E.$$.fragment),L=S(),R=r("div"),u(h.$$.fragment),T=S(),u(N.$$.fragment),P=S(),M=r("div"),u(v.$$.fragment),I=S(),u(y.$$.fragment),G=S(),x=r("div"),u(w.$$.fragment),H=S(),u(F.$$.fragment),B=S(),O=r("div"),u(D.$$.fragment),K=S(),u(U.$$.fragment),V=S(),z=r("div"),u(k.$$.fragment),X=S(),u(q.$$.fragment),j=S(),Y=r("div"),u(W.$$.fragment),J=S(),u(Z.$$.fragment),Q=S(),te=r("div"),u(ne.$$.fragment),ae=S(),u(se.$$.fragment),re=S(),le=r("div"),u(ie.$$.fragment),oe=S(),u(pe.$$.fragment),l(t,"class","pipen-report-table-of-images"),ee(t,"grid-template-columns","repeat(2, auto)")},m(e,r){i(e,t,r),C(t,n),A(a,n,null),C(n,s),A(p,n,null),C(t,c),C(t,g),A(m,g,null),C(g,f),A(E,g,null),C(t,L),C(t,R),A(h,R,null),C(R,T),A(N,R,null),C(t,P),C(t,M),A(v,M,null),C(M,I),A(y,M,null),C(t,G),C(t,x),A(w,x,null),C(x,H),A(F,x,null),C(t,B),C(t,O),A(D,O,null),C(O,K),A(U,O,null),C(t,V),C(t,z),A(k,z,null),C(z,X),A(q,z,null),C(t,j),C(t,Y),A(W,Y,null),C(Y,J),A(Z,Y,null),C(t,Q),C(t,te),A(ne,te,null),C(te,ae),A(se,te,null),C(t,re),C(t,le),A(ie,le,null),C(le,oe),A(pe,le,null),ce=!0},p:b,i(e){ce||(o(a.$$.fragment,e),o(p.$$.fragment,e),o(m.$$.fragment,e),o(E.$$.fragment,e),o(h.$$.fragment,e),o(N.$$.fragment,e),o(v.$$.fragment,e),o(y.$$.fragment,e),o(w.$$.fragment,e),o(F.$$.fragment,e),o(D.$$.fragment,e),o(U.$$.fragment,e),o(k.$$.fragment,e),o(q.$$.fragment,e),o(W.$$.fragment,e),o(Z.$$.fragment,e),o(ne.$$.fragment,e),o(se.$$.fragment,e),o(ie.$$.fragment,e),o(pe.$$.fragment,e),ce=!0)},o(e){_(a.$$.fragment,e),_(p.$$.fragment,e),_(m.$$.fragment,e),_(E.$$.fragment,e),_(h.$$.fragment,e),_(N.$$.fragment,e),_(v.$$.fragment,e),_(y.$$.fragment,e),_(w.$$.fragment,e),_(F.$$.fragment,e),_(D.$$.fragment,e),_(U.$$.fragment,e),_(k.$$.fragment,e),_(q.$$.fragment,e),_(W.$$.fragment,e),_(Z.$$.fragment,e),_(ne.$$.fragment,e),_(se.$$.fragment,e),_(ie.$$.fragment,e),_(pe.$$.fragment,e),ce=!1},d(e){e&&d(t),$(a),$(p),$(m),$(E),$(h),$(N),$(v),$(y),$(w),$(F),$(D),$(U),$(k),$(q),$(W),$(Z),$(ne),$(se),$(ie),$(pe)}}}function es(e){let t,n,a,s;return t=new Ge({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[Za]},$$scope:{ctx:e}}}),a=new Ge({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[Qa]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function ts(e){let t,n,a,s;return t=new fe({props:{label:"Plot"}}),a=new fe({props:{label:"Table"}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p:b,i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function ns(e){let t;return{c(){t=h("This table presents a comprehensive summary of the top enriched pathways derived from the fgsea. Each row corresponds to a pathway, and the gene ranks are shown based on the ranking metric used in the analysis. The enrichment score, p-value, and adjusted p-value are also provided to assess the significance of the enrichment.")},m(e,n){i(e,t,n)},d(e){e&&d(t)}}}function as(e){let t,n,a,s;return t=new $a({props:{$$slots:{default:[ns]},$$scope:{ctx:e}}}),a=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/gsea_table.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/gsea_table.pdf",width:1e3,height:760}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function ss(e){let t;return{c(){t=h("This plot represents the GSEA results for a specified gene set, illustrating the distribution and impact of the gene set along the ranked list of genes. The running enrichment score curve shows the cumulative enrichment score as genes from the input list are encountered. Positive peaks on the curve indicate regions where members of the gene set are predominantly found.")},m(e,n){i(e,t,n)},d(e){e&&d(t)}}}function rs(e){let t,n,a,s;return t=new $a({props:{$$slots:{default:[ss]},$$scope:{ctx:e}}}),a=new ma({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/fgsea.txt",data:[{pathway:"HALLMARK_INTERFERON_ALPHA_RESPONSE",pval:0,padj:0,log2err:.705,ES:.8055,NES:1.8307,size:96,leadingEdge:"ISG20,PSMB9,IFITM2,TAP1,UBE2L6,B2M,CD74,LY6E,PSME2,LGALS3BP,GBP4,IRF1,PSMB8,IFITM1,IFI35,PSME1,IFITM3,OASL,SAMD9L,ISG15,PARP9,NMI,IRF7,CMPK2,HELZ2,SELL,BST2,PLSCR1,IFI30,TMEM140,IL15,GBP2,CNP,IFIT3,OAS1,RTP4,IFI44L,USP18,EIF2AK2,ADAR,MOV10,UBA7,LAMP3,RSAD2,IFI27,PARP14,IL4R,CXCL10,SAMD9,ELF1,LAP3,EPSTI1,DHX58,CMTR1,CD47,HERC6,TRIM5,PSMA3,PARP12,IRF2,SLC25A28",id:0},{pathway:"HALLMARK_IL6_JAK_STAT3_SIGNALING",pval:.0141,padj:.0985,log2err:.3807,ES:.6159,NES:1.3854,size:87,leadingEdge:"MAP3K8,STAT1,CD38,CXCL13,IRF1,SOCS1,PIM1,HMOX1,CSF2,IL18R1,SOCS3,TGFB1,IL12RB1,BAK1,IL2RG,CCR1,IL1R2,STAT3,ITGA4,EBI3,PIK3R5,IL4R,CXCL10,TNFRSF1B,GRB2,IL2RA,IL15RA,CXCL9,IL3RA,STAT2,TYK2,IL7,FAS,CBL",id:1},{pathway:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",pval:.0483,padj:.2252,log2err:.2378,ES:.6069,NES:1.345,size:73,leadingEdge:"TP53INP1,S100A11,GLDC,FASN,PLSCR1,FBXO6,MVD,LDLR,FADS2,ETHE1,CXCL16,STX5,FDFT1,FABP5,ANXA5,STARD4,FDPS,SC5D,MVK,ABCA2,PLAUR,GNAI1,CLU,ATF3,GSTM2,SREBF2,CYP51A1,CPEB2,ATF5,HMGCS1,JAG1,PPARG,LGALS3,ECH1,DHCR7,ALCAM,SCD,PMVK,IDI1,TMEM97",id:2},{pathway:"HALLMARK_MYC_TARGETS_V2",pval:.0852,padj:.2982,log2err:.1798,ES:.597,NES:1.2785,size:58,leadingEdge:"DUSP2,HSPE1,HSPD1,SLC19A1,WDR74,HK2,AIMP2,RABEPK,DCTPP1,NOC4L,RCL1,WDR43,CBX3,PA2G4,NOP16,GRWD1,EXOSC5,NOP56,NPM1,MRTO4,PHB,PUS1,NOP2,NIP7,GNL3,TMEM97,PES1,NOLC1,SRM,TFB2M,UTP20",id:3},{pathway:"HALLMARK_ANDROGEN_RESPONSE",pval:.175,padj:.49,log2err:.1167,ES:.5087,NES:1.1588,size:99,leadingEdge:"SAT1,B2M,FKBP5,ARID5B,LMAN1,CCND3,PLPP1,SRF,RPS6KA3,INSIG1,DHCR24,ABHD2,ADRM1,ELOVL5,MAF,CDK6,ELL2,SPCS3,PA2G4,HMGCS1,NKX3-1,KLK2,MAK,UBE2I,TMEM50A,HOMER2,B4GALT1,FADS1,DBI,VAPA,SCD,SEC24D,SRP19,IDI1,UAP1,UBE2J1,ELK4,MERTK,HPGD,CCND1,PIAS1,NCOA4,SMS,ZBTB10,GPD1L,HMGCR,MYL12A,PGM3",id:4},{pathway:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",pval:.2371,padj:.4971,log2err:.1024,ES:.5427,NES:1.1389,size:49,leadingEdge:"SOD1,TXN,LAMTOR5,SELENOS,MBP,NDUFA6,NDUFB4,ABCC1,PRDX6,LSP1,HMOX2,PRDX2,SBNO2,NDUFS2,ATOX1,GLRX,EGLN2,TXNRD1",id:5},{pathway:"HALLMARK_PANCREAS_BETA_CELLS",pval:.2485,padj:.4971,log2err:.1006,ES:.5596,NES:1.1417,size:40,leadingEdge:"VDR,SEC11A,SRP14,SPCS1,SRPRB,LMO2,DPP4,SRP9",id:6},{pathway:"HALLMARK_ANGIOGENESIS",pval:.3075,padj:.538,log2err:.138,ES:-.4824,NES:-1.0653,size:36,leadingEdge:"ITGAV,CCND2,PTK2,APOH,JAG2,NRP1,MSX1",id:7},{pathway:"HALLMARK_APICAL_SURFACE",pval:.5276,padj:.8102,log2err:.1003,ES:-.4164,NES:-.967,size:44,leadingEdge:"LYPD3,PKHD1,HSPB1,AKAP7,TMEM8B,FLOT2,ADAM10,CROCC,BRCA1,GSTM3,LYN",id:8},{pathway:"HALLMARK_TGF_BETA_SIGNALING",pval:.5787,padj:.8102,log2err:.055,ES:.4444,NES:.9398,size:54,leadingEdge:"FKBP1A,FURIN,ID2,TGIF1,UBE2D3,ARID4B,TGFB1,PPP1CA,HIPK2,SMAD3,CDH1,THBS1,ACVR1,WWTR1,RHOA,SPTBN1,SKIL,ID1,TRIM33,SMAD7,PPM1A,SMURF1,BMPR1A,SLC20A1,ENG,XIAP",id:9},{pathway:"HALLMARK_PROTEIN_SECRETION",pval:.708,padj:.8777,log2err:.0419,ES:.3935,NES:.8929,size:95,leadingEdge:"SOD1,LMAN1,AP2S1,RPS6KA3,RAB9A,YKT6,RER1,COPE,IGF2R,VAMP4,VAMP7,GNAS,BET1,GOSR2,AP2M1,ARFGAP3,ARFGEF1,NAPG,GALC,DNM1L,ARFGEF2,KIF1B,NAPA,ERGIC3,SEC22B,SCAMP1,STX7,SEC24D,SCAMP3,YIPF6,BNIP3,USO1,ARFIP1,ARF1,CLTA,ANP32E,TMX1,PAM,VPS4B,SNAP23,ARCN1,GBF1,CLTC",id:10},{pathway:"HALLMARK_NOTCH_SIGNALING",pval:.7523,padj:.8777,log2err:.0806,ES:-.3982,NES:-.8445,size:32,leadingEdge:"SKP1,MAML2,FBXW11,HES1,ST3GAL6,DTX4,HEYL,APH1A,DTX2,ARRB1,PSENEN",id:11},{pathway:"HALLMARK_HEDGEHOG_SIGNALING",pval:.9658,padj:.9807,log2err:.068,ES:-.3416,NES:-.7543,size:36,leadingEdge:"VLDLR,CRMP1,CELSR1,OPHN1,NRP1,CDK5R1,NF1,HEY1",id:12},{pathway:"HALLMARK_WNT_BETA_CATENIN_SIGNALING",pval:.9807,padj:.9807,log2err:.069,ES:-.326,NES:-.7417,size:42,leadingEdge:"NKD1,NCOR2,CTNNB1,CCND2,WNT1",id:13}]}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function ls(e){let t,n,a,s;return t=new ue({props:{$$slots:{default:[as]},$$scope:{ctx:e}}}),a=new ue({props:{$$slots:{default:[rs]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function is(e){let t,n;return t=new me({props:{$$slots:{content:[ls],default:[ts]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function os(e){let t,n,a,s,p,c,g,m,f,E,L,R,h,T,N,P,M,v,I,y,G,x,w,H,F,B,O,D,K,U,V,z,k,X,q,j,Y,W,J,Z,Q,te,ne,ae,se,re,le,ie,oe,pe,ce;return a=new $a({props:{title:"HALLMARK_INTERFERON_ALPHA_RESPONSE",class:"pipen-report-table-image-descr"}}),p=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/fgsea_HALLMARK_INTERFERON_ALPHA_RESPONSE.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/fgsea_HALLMARK_INTERFERON_ALPHA_RESPONSE.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),m=new $a({props:{title:"HALLMARK_IL6_JAK_STAT3_SIGNALING",class:"pipen-report-table-image-descr"}}),E=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/fgsea_HALLMARK_IL6_JAK_STAT3_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/fgsea_HALLMARK_IL6_JAK_STAT3_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),h=new $a({props:{title:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",class:"pipen-report-table-image-descr"}}),N=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/fgsea_HALLMARK_CHOLESTEROL_HOMEOSTASIS.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/fgsea_HALLMARK_CHOLESTEROL_HOMEOSTASIS.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),v=new $a({props:{title:"HALLMARK_MYC_TARGETS_V2",class:"pipen-report-table-image-descr"}}),y=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/fgsea_HALLMARK_MYC_TARGETS_V2.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/fgsea_HALLMARK_MYC_TARGETS_V2.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),w=new $a({props:{title:"HALLMARK_ANDROGEN_RESPONSE",class:"pipen-report-table-image-descr"}}),F=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/fgsea_HALLMARK_ANDROGEN_RESPONSE.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/fgsea_HALLMARK_ANDROGEN_RESPONSE.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),D=new $a({props:{title:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",class:"pipen-report-table-image-descr"}}),U=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/fgsea_HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/fgsea_HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),k=new $a({props:{title:"HALLMARK_PANCREAS_BETA_CELLS",class:"pipen-report-table-image-descr"}}),q=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/fgsea_HALLMARK_PANCREAS_BETA_CELLS.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/fgsea_HALLMARK_PANCREAS_BETA_CELLS.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),W=new $a({props:{title:"HALLMARK_ANGIOGENESIS",class:"pipen-report-table-image-descr"}}),Z=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/fgsea_HALLMARK_ANGIOGENESIS.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/fgsea_HALLMARK_ANGIOGENESIS.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new $a({props:{title:"HALLMARK_APICAL_SURFACE",class:"pipen-report-table-image-descr"}}),se=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/fgsea_HALLMARK_APICAL_SURFACE.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/fgsea_HALLMARK_APICAL_SURFACE.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),ie=new $a({props:{title:"HALLMARK_TGF_BETA_SIGNALING",class:"pipen-report-table-image-descr"}}),pe=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/fgsea_HALLMARK_TGF_BETA_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells/fgsea_HALLMARK_TGF_BETA_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=r("div"),n=r("div"),u(a.$$.fragment),s=S(),u(p.$$.fragment),c=S(),g=r("div"),u(m.$$.fragment),f=S(),u(E.$$.fragment),L=S(),R=r("div"),u(h.$$.fragment),T=S(),u(N.$$.fragment),P=S(),M=r("div"),u(v.$$.fragment),I=S(),u(y.$$.fragment),G=S(),x=r("div"),u(w.$$.fragment),H=S(),u(F.$$.fragment),B=S(),O=r("div"),u(D.$$.fragment),K=S(),u(U.$$.fragment),V=S(),z=r("div"),u(k.$$.fragment),X=S(),u(q.$$.fragment),j=S(),Y=r("div"),u(W.$$.fragment),J=S(),u(Z.$$.fragment),Q=S(),te=r("div"),u(ne.$$.fragment),ae=S(),u(se.$$.fragment),re=S(),le=r("div"),u(ie.$$.fragment),oe=S(),u(pe.$$.fragment),l(t,"class","pipen-report-table-of-images"),ee(t,"grid-template-columns","repeat(2, auto)")},m(e,r){i(e,t,r),C(t,n),A(a,n,null),C(n,s),A(p,n,null),C(t,c),C(t,g),A(m,g,null),C(g,f),A(E,g,null),C(t,L),C(t,R),A(h,R,null),C(R,T),A(N,R,null),C(t,P),C(t,M),A(v,M,null),C(M,I),A(y,M,null),C(t,G),C(t,x),A(w,x,null),C(x,H),A(F,x,null),C(t,B),C(t,O),A(D,O,null),C(O,K),A(U,O,null),C(t,V),C(t,z),A(k,z,null),C(z,X),A(q,z,null),C(t,j),C(t,Y),A(W,Y,null),C(Y,J),A(Z,Y,null),C(t,Q),C(t,te),A(ne,te,null),C(te,ae),A(se,te,null),C(t,re),C(t,le),A(ie,le,null),C(le,oe),A(pe,le,null),ce=!0},p:b,i(e){ce||(o(a.$$.fragment,e),o(p.$$.fragment,e),o(m.$$.fragment,e),o(E.$$.fragment,e),o(h.$$.fragment,e),o(N.$$.fragment,e),o(v.$$.fragment,e),o(y.$$.fragment,e),o(w.$$.fragment,e),o(F.$$.fragment,e),o(D.$$.fragment,e),o(U.$$.fragment,e),o(k.$$.fragment,e),o(q.$$.fragment,e),o(W.$$.fragment,e),o(Z.$$.fragment,e),o(ne.$$.fragment,e),o(se.$$.fragment,e),o(ie.$$.fragment,e),o(pe.$$.fragment,e),ce=!0)},o(e){_(a.$$.fragment,e),_(p.$$.fragment,e),_(m.$$.fragment,e),_(E.$$.fragment,e),_(h.$$.fragment,e),_(N.$$.fragment,e),_(v.$$.fragment,e),_(y.$$.fragment,e),_(w.$$.fragment,e),_(F.$$.fragment,e),_(D.$$.fragment,e),_(U.$$.fragment,e),_(k.$$.fragment,e),_(q.$$.fragment,e),_(W.$$.fragment,e),_(Z.$$.fragment,e),_(ne.$$.fragment,e),_(se.$$.fragment,e),_(ie.$$.fragment,e),_(pe.$$.fragment,e),ce=!1},d(e){e&&d(t),$(a),$(p),$(m),$(E),$(h),$(N),$(v),$(y),$(w),$(F),$(D),$(U),$(k),$(q),$(W),$(Z),$(ne),$(se),$(ie),$(pe)}}}function ps(e){let t,n,a,s;return t=new Ge({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[is]},$$scope:{ctx:e}}}),a=new Ge({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[os]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function cs(e){let t,n,a,s;return t=new fe({props:{label:"Plot"}}),a=new fe({props:{label:"Table"}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p:b,i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function _s(e){let t;return{c(){t=h("This table presents a comprehensive summary of the top enriched pathways derived from the fgsea. Each row corresponds to a pathway, and the gene ranks are shown based on the ranking metric used in the analysis. The enrichment score, p-value, and adjusted p-value are also provided to assess the significance of the enrichment.")},m(e,n){i(e,t,n)},d(e){e&&d(t)}}}function ds(e){let t,n,a,s;return t=new $a({props:{$$slots:{default:[_s]},$$scope:{ctx:e}}}),a=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/gsea_table.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/gsea_table.pdf",width:1e3,height:760}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function gs(e){let t;return{c(){t=h("This plot represents the GSEA results for a specified gene set, illustrating the distribution and impact of the gene set along the ranked list of genes. The running enrichment score curve shows the cumulative enrichment score as genes from the input list are encountered. Positive peaks on the curve indicate regions where members of the gene set are predominantly found.")},m(e,n){i(e,t,n)},d(e){e&&d(t)}}}function ms(e){let t,n,a,s;return t=new $a({props:{$$slots:{default:[gs]},$$scope:{ctx:e}}}),a=new ma({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/fgsea.txt",data:[{pathway:"HALLMARK_INTERFERON_ALPHA_RESPONSE",pval:0,padj:3e-4,log2err:.5756,ES:.7611,NES:1.6724,size:96,leadingEdge:"ISG20,IFITM2,CD74,UBE2L6,PSMB9,TAP1,IFITM1,LY6E,B2M,PSME1,ISG15,PSMB8,GBP2,PARP9,IRF1,IFI35,PSME2,GBP4,IL7,SAMD9L,USP18,NMI,IFITM3,PLSCR1,EPSTI1,IRF7,MOV10,ELF1,RSAD2,UBA7,NCOA7,IFI27,OGFR,TRAFD1,DHX58,SP110,IL4R,BST2,ADAR,TMEM140,CNP,OASL,SELL,LGALS3BP,SAMD9,HELZ2,IFIT2,IFIT3,SLC25A28,TENT5A,PARP12,IFIH1,TRIM5,IRF2,LAP3,HERC6",id:0},{pathway:"HALLMARK_PANCREAS_BETA_CELLS",pval:.0055,padj:.0383,log2err:.407,ES:.7729,NES:1.5221,size:40,leadingEdge:"VDR,SRP9,SPCS1,SEC11A,STXBP1,AKT3,SRP14,DPP4,ELP4",id:1},{pathway:"HALLMARK_IL6_JAK_STAT3_SIGNALING",pval:.1005,padj:.469,log2err:.1552,ES:.5655,NES:1.2324,size:87,leadingEdge:"CD38,STAT1,IL18R1,MAP3K8,SOCS3,IRF1,IL7,MYD88,IL12RB1,PIK3R5,CSF2,GRB2,TYK2,IL4R,PIM1,PTPN1,TGFB1,CXCL13,STAT3,SOCS1,CD9,CCR1,IL2RG,EBI3,TNFRSF1A,BAK1,PTPN2,IL1R2,CXCL10,CD14,LTBR,ACVR1B",id:2},{pathway:"HALLMARK_MYC_TARGETS_V2",pval:.1863,padj:.6038,log2err:.2021,ES:-.4899,NES:-1.1381,size:58,leadingEdge:"MCM4,TBRG4,PHB,IMP4,PRMT3,WDR74,NDUFAF4,PPAN,TFB2M,MYBBP1A,MPHOSPH10,UNG,CDK4,SORD,PA2G4,WDR43,NPM1,DCTPP1,MCM5,HSPE1,UTP20,NOP56,NIP7,IPO4,RRP9,TCOF1,PLK4,RRP12,NOP2,PPRC1,NOC4L,NOP16,PES1",id:3},{pathway:"HALLMARK_ANGIOGENESIS",pval:.2264,padj:.6038,log2err:.1063,ES:.6064,NES:1.1749,size:36,leadingEdge:"TIMP1,S100A4,JAG1,APP,PDGFA,FGFR1",id:4},{pathway:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",pval:.2769,padj:.6038,log2err:.0884,ES:.513,NES:1.0933,size:73,leadingEdge:"TP53INP1,GSTM2,ETHE1,FBXO6,PLSCR1,CXCL16,ALCAM,GLDC,LDLR,FDFT1,S100A11,SREBF2,LGMN,JAG1,ANXA5,SC5D,FADS2,SCD,CD9,ATXN2,ACTG1,STARD4,TM7SF2,ATF3,IDI1,CPEB2,CLU,MVD,HSD17B7,HMGCS1,ATF5,PDK3",id:5},{pathway:"HALLMARK_HEDGEHOG_SIGNALING",pval:.3019,padj:.6038,log2err:.0892,ES:.5684,NES:1.1013,size:36,leadingEdge:"CDK5R1,ADGRG1,TLE1,PML,CDK6,RASA1,L1CAM,NF1,MYH9,LDB1",id:6},{pathway:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",pval:.4473,padj:.7827,log2err:.0669,ES:.4967,NES:1.0024,size:49,leadingEdge:"NDUFA6,LSP1,PRDX6,SOD1,SBNO2,NDUFB4,HMOX2,ATOX1,FES,GLRX,PRDX2,GLRX2,STK25,ABCC1,MBP,EGLN2,CAT",id:7},{pathway:"HALLMARK_ANDROGEN_RESPONSE",pval:.5553,padj:.8267,log2err:.0518,ES:.4365,NES:.96,size:99,leadingEdge:"FKBP5,ARID5B,SAT1,B2M,RPS6KA3,ELOVL5,ABCC4,ELL2,CCND3,CDK6,HOMER2,DHCR24,TNFAIP8,PLPP1,ZMIZ1,PDLIM5,SCD,MAK,RAB4A,INSIG1,DNAJB9,KRT19,KRT8,NKX3-1,MERTK,IDI1,TMEM50A,UAP1,ABHD2,AKT1,SLC38A2",id:8},{pathway:"HALLMARK_PROTEIN_SECRETION",pval:.5905,padj:.8267,log2err:.1215,ES:-.3726,NES:-.9326,size:95,leadingEdge:"CD63,ANP32E,ATP1A1,TPD52,AP3B1,RAB5A,ARCN1,SCAMP3,TMED2,CTSC,AP3S1,RAB14,BET1,PPT1,PAM,TMED10,VAMP3,SGMS1,STAM,SEC24D,GLA,CLTA,COPB1,ICA1,GOLGA4,STX16,TSG101,BNIP3",id:9},{pathway:"HALLMARK_APICAL_SURFACE",pval:.8874,padj:.9294,log2err:.0755,ES:-.366,NES:-.823,size:44,leadingEdge:"GSTM3,HSPB1,CD160,AFAP1L2,PLAUR,FLOT2,SCUBE1,LYPD3,ADAM10,B4GALT1,CROCC",id:10},{pathway:"HALLMARK_NOTCH_SIGNALING",pval:.9108,padj:.9294,log2err:.0722,ES:-.3719,NES:-.7733,size:32,leadingEdge:"SKP1,KAT2A,ARRB1,DTX2",id:11},{pathway:"HALLMARK_WNT_BETA_CATENIN_SIGNALING",pval:.9141,padj:.9294,log2err:.0336,ES:.3771,NES:.7469,size:42,leadingEdge:"NOTCH1,NKD1,JAG1,WNT1,FRAT1,AXIN2,HDAC5,SKP2,NUMB,DVL2,LEF1,NOTCH4,MYC,PPARD,CUL1,ADAM17,RBPJ,TCF7",id:12},{pathway:"HALLMARK_TGF_BETA_SIGNALING",pval:.9294,padj:.9294,log2err:.032,ES:.3685,NES:.7509,size:54,leadingEdge:"ID2,FURIN,SKIL,SPTBN1,ACVR1,UBE2D3,THBS1,TGFB1,ID1,ARID4B,SERPINE1,BCAR3,SMURF1,PPP1CA,SMAD7,FKBP1A,WWTR1,RHOA,CDK9,HDAC1,BMPR2",id:13}]}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function fs(e){let t,n,a,s;return t=new ue({props:{$$slots:{default:[ds]},$$scope:{ctx:e}}}),a=new ue({props:{$$slots:{default:[ms]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function us(e){let t,n;return t=new me({props:{$$slots:{content:[fs],default:[cs]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function As(e){let t,n,a,s,p,c,g,m,f,E,L,R,h,T,N,P,M,v,I,y,G,x,w,H,F,B,O,D,K,U,V,z,k,X,q,j,Y,W,J,Z,Q,te,ne,ae,se,re,le,ie,oe,pe,ce;return a=new $a({props:{title:"HALLMARK_INTERFERON_ALPHA_RESPONSE",class:"pipen-report-table-image-descr"}}),p=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/fgsea_HALLMARK_INTERFERON_ALPHA_RESPONSE.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/fgsea_HALLMARK_INTERFERON_ALPHA_RESPONSE.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),m=new $a({props:{title:"HALLMARK_PANCREAS_BETA_CELLS",class:"pipen-report-table-image-descr"}}),E=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/fgsea_HALLMARK_PANCREAS_BETA_CELLS.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/fgsea_HALLMARK_PANCREAS_BETA_CELLS.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),h=new $a({props:{title:"HALLMARK_IL6_JAK_STAT3_SIGNALING",class:"pipen-report-table-image-descr"}}),N=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/fgsea_HALLMARK_IL6_JAK_STAT3_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/fgsea_HALLMARK_IL6_JAK_STAT3_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),v=new $a({props:{title:"HALLMARK_MYC_TARGETS_V2",class:"pipen-report-table-image-descr"}}),y=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/fgsea_HALLMARK_MYC_TARGETS_V2.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/fgsea_HALLMARK_MYC_TARGETS_V2.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),w=new $a({props:{title:"HALLMARK_ANGIOGENESIS",class:"pipen-report-table-image-descr"}}),F=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/fgsea_HALLMARK_ANGIOGENESIS.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/fgsea_HALLMARK_ANGIOGENESIS.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),D=new $a({props:{title:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",class:"pipen-report-table-image-descr"}}),U=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/fgsea_HALLMARK_CHOLESTEROL_HOMEOSTASIS.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/fgsea_HALLMARK_CHOLESTEROL_HOMEOSTASIS.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),k=new $a({props:{title:"HALLMARK_HEDGEHOG_SIGNALING",class:"pipen-report-table-image-descr"}}),q=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/fgsea_HALLMARK_HEDGEHOG_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/fgsea_HALLMARK_HEDGEHOG_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),W=new $a({props:{title:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",class:"pipen-report-table-image-descr"}}),Z=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/fgsea_HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/fgsea_HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new $a({props:{title:"HALLMARK_ANDROGEN_RESPONSE",class:"pipen-report-table-image-descr"}}),se=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/fgsea_HALLMARK_ANDROGEN_RESPONSE.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/fgsea_HALLMARK_ANDROGEN_RESPONSE.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),ie=new $a({props:{title:"HALLMARK_PROTEIN_SECRETION",class:"pipen-report-table-image-descr"}}),pe=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/fgsea_HALLMARK_PROTEIN_SECRETION.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Regulatory-T-cells-1/fgsea_HALLMARK_PROTEIN_SECRETION.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=r("div"),n=r("div"),u(a.$$.fragment),s=S(),u(p.$$.fragment),c=S(),g=r("div"),u(m.$$.fragment),f=S(),u(E.$$.fragment),L=S(),R=r("div"),u(h.$$.fragment),T=S(),u(N.$$.fragment),P=S(),M=r("div"),u(v.$$.fragment),I=S(),u(y.$$.fragment),G=S(),x=r("div"),u(w.$$.fragment),H=S(),u(F.$$.fragment),B=S(),O=r("div"),u(D.$$.fragment),K=S(),u(U.$$.fragment),V=S(),z=r("div"),u(k.$$.fragment),X=S(),u(q.$$.fragment),j=S(),Y=r("div"),u(W.$$.fragment),J=S(),u(Z.$$.fragment),Q=S(),te=r("div"),u(ne.$$.fragment),ae=S(),u(se.$$.fragment),re=S(),le=r("div"),u(ie.$$.fragment),oe=S(),u(pe.$$.fragment),l(t,"class","pipen-report-table-of-images"),ee(t,"grid-template-columns","repeat(2, auto)")},m(e,r){i(e,t,r),C(t,n),A(a,n,null),C(n,s),A(p,n,null),C(t,c),C(t,g),A(m,g,null),C(g,f),A(E,g,null),C(t,L),C(t,R),A(h,R,null),C(R,T),A(N,R,null),C(t,P),C(t,M),A(v,M,null),C(M,I),A(y,M,null),C(t,G),C(t,x),A(w,x,null),C(x,H),A(F,x,null),C(t,B),C(t,O),A(D,O,null),C(O,K),A(U,O,null),C(t,V),C(t,z),A(k,z,null),C(z,X),A(q,z,null),C(t,j),C(t,Y),A(W,Y,null),C(Y,J),A(Z,Y,null),C(t,Q),C(t,te),A(ne,te,null),C(te,ae),A(se,te,null),C(t,re),C(t,le),A(ie,le,null),C(le,oe),A(pe,le,null),ce=!0},p:b,i(e){ce||(o(a.$$.fragment,e),o(p.$$.fragment,e),o(m.$$.fragment,e),o(E.$$.fragment,e),o(h.$$.fragment,e),o(N.$$.fragment,e),o(v.$$.fragment,e),o(y.$$.fragment,e),o(w.$$.fragment,e),o(F.$$.fragment,e),o(D.$$.fragment,e),o(U.$$.fragment,e),o(k.$$.fragment,e),o(q.$$.fragment,e),o(W.$$.fragment,e),o(Z.$$.fragment,e),o(ne.$$.fragment,e),o(se.$$.fragment,e),o(ie.$$.fragment,e),o(pe.$$.fragment,e),ce=!0)},o(e){_(a.$$.fragment,e),_(p.$$.fragment,e),_(m.$$.fragment,e),_(E.$$.fragment,e),_(h.$$.fragment,e),_(N.$$.fragment,e),_(v.$$.fragment,e),_(y.$$.fragment,e),_(w.$$.fragment,e),_(F.$$.fragment,e),_(D.$$.fragment,e),_(U.$$.fragment,e),_(k.$$.fragment,e),_(q.$$.fragment,e),_(W.$$.fragment,e),_(Z.$$.fragment,e),_(ne.$$.fragment,e),_(se.$$.fragment,e),_(ie.$$.fragment,e),_(pe.$$.fragment,e),ce=!1},d(e){e&&d(t),$(a),$(p),$(m),$(E),$(h),$(N),$(v),$(y),$(w),$(F),$(D),$(U),$(k),$(q),$(W),$(Z),$(ne),$(se),$(ie),$(pe)}}}function $s(e){let t,n,a,s;return t=new Ge({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[us]},$$scope:{ctx:e}}}),a=new Ge({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[As]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function Ss(e){let t,n,a,s;return t=new fe({props:{label:"Plot"}}),a=new fe({props:{label:"Table"}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p:b,i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function Es(e){let t;return{c(){t=h("This table presents a comprehensive summary of the top enriched pathways derived from the fgsea. Each row corresponds to a pathway, and the gene ranks are shown based on the ranking metric used in the analysis. The enrichment score, p-value, and adjusted p-value are also provided to assess the significance of the enrichment.")},m(e,n){i(e,t,n)},d(e){e&&d(t)}}}function Ls(e){let t,n,a,s;return t=new $a({props:{$$slots:{default:[Es]},$$scope:{ctx:e}}}),a=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/gsea_table.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/gsea_table.pdf",width:1e3,height:760}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function Rs(e){let t;return{c(){t=h("This plot represents the GSEA results for a specified gene set, illustrating the distribution and impact of the gene set along the ranked list of genes. The running enrichment score curve shows the cumulative enrichment score as genes from the input list are encountered. Positive peaks on the curve indicate regions where members of the gene set are predominantly found.")},m(e,n){i(e,t,n)},d(e){e&&d(t)}}}function hs(e){let t,n,a,s;return t=new $a({props:{$$slots:{default:[Rs]},$$scope:{ctx:e}}}),a=new ma({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/fgsea.txt",data:[{pathway:"HALLMARK_INTERFERON_ALPHA_RESPONSE",pval:0,padj:2e-4,log2err:.5756,ES:.7006,NES:1.7921,size:96,leadingEdge:"ISG20,PSMB9,PSME2,UBE2L6,IFI44L,IFITM1,LY6E,TAP1,PSME1,IFITM2,ISG15,USP18,GBP4,RIPK2,IFIT3,IRF7,GBP2,PLSCR1,PSMB8,PARP9,IRF1,CD74,RSAD2,LAMP3,SAMD9L,IL4R,HERC6,PARP14,ADAR,IFI27,CMPK2,NUB1,IFI35,SP110,OASL,TXNIP,SELL,EIF2AK2,TRIM21,TRIM5,PROCR,B2M,IFITM3,CASP1,MX1,CMTR1,TRIM14,PARP12,TMEM140,BST2,OAS1,IFIT2,IFI30,MVB12A,RTP4,TDRD7",id:0},{pathway:"HALLMARK_PANCREAS_BETA_CELLS",pval:.136,padj:.6957,log2err:.2572,ES:.5384,NES:1.1935,size:40,leadingEdge:"PAK3,SEC11A,SPCS1,VDR,SRP14,FOXO1,DPP4",id:1},{pathway:"HALLMARK_WNT_BETA_CATENIN_SIGNALING",pval:.1493,padj:.6957,log2err:.1263,ES:-.6297,NES:-1.2153,size:42,leadingEdge:"TCF7,AXIN1,HDAC5,NCSTN,NOTCH4,DVL2,ADAM17,WNT1,CTNNB1,RBPJ,HEY2,HEY1,FZD1,FZD8,NUMB,HDAC2,LEF1,AXIN2,SKP2,CSNK1E,NCOR2",id:2},{pathway:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",pval:.2147,padj:.6957,log2err:.2311,ES:.4348,NES:1.0763,size:73,leadingEdge:"S100A11,PLSCR1,ATF3,HSD17B7,SC5D,FADS2,IDI1,LDLR,GNAI1,LGALS3,CXCL16,FASN,EBP,GSTM2,TMEM97,NSDHL,TRIB3,LGMN,STARD4,GUSB,TP53INP1,STX5,ERRFI1,CLU,ACSS2,MVD,CHKA,PDK3,FBXO6",id:3},{pathway:"HALLMARK_ANGIOGENESIS",pval:.2623,padj:.6957,log2err:.1752,ES:.5017,NES:1.1039,size:36,leadingEdge:"LRPAP1,FGFR1,TNFRSF21,TIMP1,COL5A2,ITGAV",id:4},{pathway:"HALLMARK_PROTEIN_SECRETION",pval:.2981,padj:.6957,log2err:.0777,ES:-.5126,NES:-1.0866,size:95,leadingEdge:"CTSC,PPT1,M6PR,NAPA,MAPK1,ATP1A1,DNM1L,AP1G1,ATP6V1H,STX12,RAB22A,RAB5A,LAMP2,SNX2,ANP32E,MON2,GALC,ARFIP1,YKT6,COPB1,ADAM10,ATP7A,VAMP7,IGF2R,SEC31A,USO1,VAMP4,ARCN1,GOLGA4,CLCN3,ERGIC3,SGMS1,GLA,COG2,CLTC,SEC24D,TPD52,NAPG,GOSR2,ARF1,GBF1,STX7,BET1,SNAP23,SEC22B,ICA1,AP2M1,RAB14,KIF1B,ARFGEF1,YIPF6,AP2B1,SCRN1,CLTA,CD63,CLN5",id:5},{pathway:"HALLMARK_HEDGEHOG_SIGNALING",pval:.3852,padj:.7423,log2err:.0711,ES:-.5599,NES:-1.0483,size:36,leadingEdge:"TLE3,CELSR1,NRCAM,NF1,MYH9,HEY2,VLDLR,HEY1,RASA1,CDK5R1",id:6},{pathway:"HALLMARK_TGF_BETA_SIGNALING",pval:.427,padj:.7423,log2err:.0625,ES:-.5143,NES:-1.0286,size:54,leadingEdge:"ID3,SPTBN1,SLC20A1,PPP1R15A,CTNNB1,KLF10,CDK9,CDKN1C,SMAD6,CDH1,SMAD1,TGFBR1,FKBP1A,TRIM33,IFNGR2,APC,UBE2D3,HDAC1,TGFB1,NCOR2,SMAD3",id:7},{pathway:"HALLMARK_ANDROGEN_RESPONSE",pval:.5173,padj:.7423,log2err:.0506,ES:-.4651,NES:-.9873,size:99,leadingEdge:"TNFAIP8,ACTN1,XRCC6,HSD17B14,ANKH,XRCC5,SCD,HPGD,MAP7,PDLIM5,NCOA4,PIAS1,SORD,PLPP1,PTK2B,ACSL3,SMS,DBI,SLC38A2,ZMIZ1,ABCC4,CAMKK2,NGLY1,ELL2,HMGCR,ELK4,APPBP2,GPD1L,RAB4A,SEC24D,UAP1,FADS1,TPD52,SGK1,TSC22D1,SLC26A2,STK39,RRP12,GSR,CDC14B,ZBTB10,PA2G4",id:8},{pathway:"HALLMARK_IL6_JAK_STAT3_SIGNALING",pval:.5302,padj:.7423,log2err:.1563,ES:.3823,NES:.9689,size:87,leadingEdge:"STAT1,SOCS1,IL18R1,IL15RA,IRF1,FAS,TNFRSF1B,IL4R,SOCS3,TNFRSF21,PTPN1,MAP3K8,PIM1,JUN,LTB,CD44,IL12RB1",id:9},{pathway:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",pval:.7531,padj:.8607,log2err:.0372,ES:-.4291,NES:-.8461,size:49,leadingEdge:"CAT,NQO1,ABCC1,TXNRD1,SCAF4,PRDX1,PFKP,PRDX6,ATOX1,SRXN1,PDLIM1,GPX4,MSRA,PRNP,LAMTOR5,PRDX2,LSP1",id:10},{pathway:"HALLMARK_NOTCH_SIGNALING",pval:.7659,padj:.8607,log2err:.039,ES:-.4489,NES:-.8285,size:32,leadingEdge:"DTX2,MAML2,ARRB1,FZD1,SAP30,TCF7L2,DTX1,SKP1,PSENEN,PPARD,PRKCA,KAT2A,FBXW11",id:11},{pathway:"HALLMARK_APICAL_SURFACE",pval:.7992,padj:.8607,log2err:.0351,ES:-.4262,NES:-.8266,size:44,leadingEdge:"HSPB1,AFAP1L2,GATA3,ADIPOR2,APP,LYPD3,ADAM10,RTN4RL1,CD160,GSTM3,PLAUR,FLOT2",id:12},{pathway:"HALLMARK_MYC_TARGETS_V2",pval:.9891,padj:.9891,log2err:.0219,ES:-.3395,NES:-.6844,size:58,leadingEdge:"CBX3,TFB2M,BYSL,SRM,CDK4,NDUFAF4,SORD,MYBBP1A,NOLC1,UTP20,FARSA,DDX18,NOP2,IMP4,WDR43,TBRG4,GNL3,NIP7,NOP16,HSPE1,PLK4,MPHOSPH10,RRP12,PA2G4,LAS1L,MAP3K6",id:13}]}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function Cs(e){let t,n,a,s;return t=new ue({props:{$$slots:{default:[Ls]},$$scope:{ctx:e}}}),a=new ue({props:{$$slots:{default:[hs]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function Ts(e){let t,n;return t=new me({props:{$$slots:{content:[Cs],default:[Ss]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Ns(e){let t,n,a,s,p,c,g,m,f,E,L,R,h,T,N,P,M,v,I,y,G,x,w,H,F,B,O,D,K,U,V,z,k,X,q,j,Y,W,J,Z,Q,te,ne,ae,se,re,le,ie,oe,pe,ce;return a=new $a({props:{title:"HALLMARK_INTERFERON_ALPHA_RESPONSE",class:"pipen-report-table-image-descr"}}),p=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/fgsea_HALLMARK_INTERFERON_ALPHA_RESPONSE.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/fgsea_HALLMARK_INTERFERON_ALPHA_RESPONSE.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),m=new $a({props:{title:"HALLMARK_PANCREAS_BETA_CELLS",class:"pipen-report-table-image-descr"}}),E=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/fgsea_HALLMARK_PANCREAS_BETA_CELLS.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/fgsea_HALLMARK_PANCREAS_BETA_CELLS.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),h=new $a({props:{title:"HALLMARK_WNT_BETA_CATENIN_SIGNALING",class:"pipen-report-table-image-descr"}}),N=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/fgsea_HALLMARK_WNT_BETA_CATENIN_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/fgsea_HALLMARK_WNT_BETA_CATENIN_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),v=new $a({props:{title:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",class:"pipen-report-table-image-descr"}}),y=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/fgsea_HALLMARK_CHOLESTEROL_HOMEOSTASIS.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/fgsea_HALLMARK_CHOLESTEROL_HOMEOSTASIS.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),w=new $a({props:{title:"HALLMARK_ANGIOGENESIS",class:"pipen-report-table-image-descr"}}),F=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/fgsea_HALLMARK_ANGIOGENESIS.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/fgsea_HALLMARK_ANGIOGENESIS.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),D=new $a({props:{title:"HALLMARK_PROTEIN_SECRETION",class:"pipen-report-table-image-descr"}}),U=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/fgsea_HALLMARK_PROTEIN_SECRETION.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/fgsea_HALLMARK_PROTEIN_SECRETION.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),k=new $a({props:{title:"HALLMARK_HEDGEHOG_SIGNALING",class:"pipen-report-table-image-descr"}}),q=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/fgsea_HALLMARK_HEDGEHOG_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/fgsea_HALLMARK_HEDGEHOG_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),W=new $a({props:{title:"HALLMARK_TGF_BETA_SIGNALING",class:"pipen-report-table-image-descr"}}),Z=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/fgsea_HALLMARK_TGF_BETA_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/fgsea_HALLMARK_TGF_BETA_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new $a({props:{title:"HALLMARK_ANDROGEN_RESPONSE",class:"pipen-report-table-image-descr"}}),se=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/fgsea_HALLMARK_ANDROGEN_RESPONSE.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/fgsea_HALLMARK_ANDROGEN_RESPONSE.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),ie=new $a({props:{title:"HALLMARK_IL6_JAK_STAT3_SIGNALING",class:"pipen-report-table-image-descr"}}),pe=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/fgsea_HALLMARK_IL6_JAK_STAT3_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tcm-Naive-helper-T-cells/fgsea_HALLMARK_IL6_JAK_STAT3_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=r("div"),n=r("div"),u(a.$$.fragment),s=S(),u(p.$$.fragment),c=S(),g=r("div"),u(m.$$.fragment),f=S(),u(E.$$.fragment),L=S(),R=r("div"),u(h.$$.fragment),T=S(),u(N.$$.fragment),P=S(),M=r("div"),u(v.$$.fragment),I=S(),u(y.$$.fragment),G=S(),x=r("div"),u(w.$$.fragment),H=S(),u(F.$$.fragment),B=S(),O=r("div"),u(D.$$.fragment),K=S(),u(U.$$.fragment),V=S(),z=r("div"),u(k.$$.fragment),X=S(),u(q.$$.fragment),j=S(),Y=r("div"),u(W.$$.fragment),J=S(),u(Z.$$.fragment),Q=S(),te=r("div"),u(ne.$$.fragment),ae=S(),u(se.$$.fragment),re=S(),le=r("div"),u(ie.$$.fragment),oe=S(),u(pe.$$.fragment),l(t,"class","pipen-report-table-of-images"),ee(t,"grid-template-columns","repeat(2, auto)")},m(e,r){i(e,t,r),C(t,n),A(a,n,null),C(n,s),A(p,n,null),C(t,c),C(t,g),A(m,g,null),C(g,f),A(E,g,null),C(t,L),C(t,R),A(h,R,null),C(R,T),A(N,R,null),C(t,P),C(t,M),A(v,M,null),C(M,I),A(y,M,null),C(t,G),C(t,x),A(w,x,null),C(x,H),A(F,x,null),C(t,B),C(t,O),A(D,O,null),C(O,K),A(U,O,null),C(t,V),C(t,z),A(k,z,null),C(z,X),A(q,z,null),C(t,j),C(t,Y),A(W,Y,null),C(Y,J),A(Z,Y,null),C(t,Q),C(t,te),A(ne,te,null),C(te,ae),A(se,te,null),C(t,re),C(t,le),A(ie,le,null),C(le,oe),A(pe,le,null),ce=!0},p:b,i(e){ce||(o(a.$$.fragment,e),o(p.$$.fragment,e),o(m.$$.fragment,e),o(E.$$.fragment,e),o(h.$$.fragment,e),o(N.$$.fragment,e),o(v.$$.fragment,e),o(y.$$.fragment,e),o(w.$$.fragment,e),o(F.$$.fragment,e),o(D.$$.fragment,e),o(U.$$.fragment,e),o(k.$$.fragment,e),o(q.$$.fragment,e),o(W.$$.fragment,e),o(Z.$$.fragment,e),o(ne.$$.fragment,e),o(se.$$.fragment,e),o(ie.$$.fragment,e),o(pe.$$.fragment,e),ce=!0)},o(e){_(a.$$.fragment,e),_(p.$$.fragment,e),_(m.$$.fragment,e),_(E.$$.fragment,e),_(h.$$.fragment,e),_(N.$$.fragment,e),_(v.$$.fragment,e),_(y.$$.fragment,e),_(w.$$.fragment,e),_(F.$$.fragment,e),_(D.$$.fragment,e),_(U.$$.fragment,e),_(k.$$.fragment,e),_(q.$$.fragment,e),_(W.$$.fragment,e),_(Z.$$.fragment,e),_(ne.$$.fragment,e),_(se.$$.fragment,e),_(ie.$$.fragment,e),_(pe.$$.fragment,e),ce=!1},d(e){e&&d(t),$(a),$(p),$(m),$(E),$(h),$(N),$(v),$(y),$(w),$(F),$(D),$(U),$(k),$(q),$(W),$(Z),$(ne),$(se),$(ie),$(pe)}}}function bs(e){let t,n,a,s;return t=new Ge({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[Ts]},$$scope:{ctx:e}}}),a=new Ge({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[Ns]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function Ps(e){let t,n,a,s;return t=new fe({props:{label:"Plot"}}),a=new fe({props:{label:"Table"}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p:b,i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function Ms(e){let t;return{c(){t=h("This table presents a comprehensive summary of the top enriched pathways derived from the fgsea. Each row corresponds to a pathway, and the gene ranks are shown based on the ranking metric used in the analysis. The enrichment score, p-value, and adjusted p-value are also provided to assess the significance of the enrichment.")},m(e,n){i(e,t,n)},d(e){e&&d(t)}}}function vs(e){let t,n,a,s;return t=new $a({props:{$$slots:{default:[Ms]},$$scope:{ctx:e}}}),a=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/gsea_table.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/gsea_table.pdf",width:1e3,height:760}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function Is(e){let t;return{c(){t=h("This plot represents the GSEA results for a specified gene set, illustrating the distribution and impact of the gene set along the ranked list of genes. The running enrichment score curve shows the cumulative enrichment score as genes from the input list are encountered. Positive peaks on the curve indicate regions where members of the gene set are predominantly found.")},m(e,n){i(e,t,n)},d(e){e&&d(t)}}}function ys(e){let t,n,a,s;return t=new $a({props:{$$slots:{default:[Is]},$$scope:{ctx:e}}}),a=new ma({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/fgsea.txt",data:[{pathway:"HALLMARK_INTERFERON_ALPHA_RESPONSE",pval:.0289,padj:.3547,log2err:.282,ES:.5944,NES:1.3288,size:96,leadingEdge:"ISG20,CD74,TAP1,PLSCR1,PSMB8,IFITM3,GBP2,PSMB9,RTP4,ISG15,SELL,LGALS3BP,NCOA7,UBA7,UBE2L6,TRAFD1,HELZ2,PARP9,NMI,IL4R,GBP4,SP110,PSME2,MVB12A,EPSTI1,LPAR6,CSF1,IRF7,PSME1,IRF1,IRF2,STAT2,IL7,BST2,SAMD9L,IFIT3,MOV10,PARP14,ELF1,NUB1,IFI27,DHX58,IFITM2,TMEM140,PNPT1,OAS1,IFI44,LAMP3,LY6E,IFITM1,LAP3",id:0},{pathway:"HALLMARK_IL6_JAK_STAT3_SIGNALING",pval:.0642,padj:.3547,log2err:.1864,ES:.5847,NES:1.2921,size:87,leadingEdge:"CD38,SOCS3,STAT1,IL2RG,BAK1,HMOX1,MYD88,CBL,TYK2,TNFRSF1B,STAM2,IL4R,IL6ST,CSF1,IRF1,CD9,STAT2,IL7,ITGA4,STAT3,IL18R1,TGFB1,CSF2,IL17RA",id:1},{pathway:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",pval:.0882,padj:.3547,log2err:.1596,ES:.5828,NES:1.2655,size:73,leadingEdge:"GSTM2,TP53INP1,PLSCR1,ATF5,CYP51A1,NSDHL,ANXA5,GLDC,ACTG1,ABCA2,SCD,HSD17B7,CXCL16,ERRFI1,HMGCS1,ETHE1,CD9,ATXN2,STX5,CHKA,GUSB,DHCR7,ECH1,JAG1,CTNNB1,FADS2,PLAUR,SREBF2,TM7SF2,STARD4,SC5D,PDK3,PMVK,MVD",id:2},{pathway:"HALLMARK_TGF_BETA_SIGNALING",pval:.1013,padj:.3547,log2err:.1521,ES:.607,NES:1.2647,size:54,leadingEdge:"ID2,FURIN,FKBP1A,SPTBN1,PMEPA1,ACVR1,SMURF1,TGIF1,ARID4B,BMPR1A,HIPK2,THBS1,UBE2D3,TGFB1,SKIL,TRIM33,SMAD1,CTNNB1,SERPINE1",id:3},{pathway:"HALLMARK_PANCREAS_BETA_CELLS",pval:.1467,padj:.4107,log2err:.1269,ES:.6197,NES:1.2441,size:40,leadingEdge:"VDR,DPP4,SRP14,SRPRB,FOXO1,ELP4,SEC11A,MAFB,SPCS1,AKT3",id:4},{pathway:"HALLMARK_NOTCH_SIGNALING",pval:.2185,padj:.4522,log2err:.1017,ES:.6046,NES:1.1775,size:32,leadingEdge:"PPARD,ARRB1,DTX1,NOTCH1,NOTCH2,LFNG,DTX2,JAG1,RBX1,MAML2,KAT2A",id:5},{pathway:"HALLMARK_HEDGEHOG_SIGNALING",pval:.2372,padj:.4522,log2err:.1978,ES:-.484,NES:-1.1203,size:36,leadingEdge:"L1CAM,RASA1,CDK5R1,PTCH1,LDB1,TLE3,DPYSL2",id:6},{pathway:"HALLMARK_MYC_TARGETS_V2",pval:.2584,padj:.4522,log2err:.209,ES:-.4272,NES:-1.077,size:58,leadingEdge:"TFB2M,TMEM97,NOC4L,SLC29A2,MPHOSPH10,MCM5,PPAN,PHB,IMP4,PRMT3,MCM4,RRP9,NDUFAF4,PLK4,NOP16,PA2G4,NOLC1,MYBBP1A,WDR74,SUPV3L1,UTP20,RABEPK,PPRC1,MYC,BYSL,LAS1L,TBRG4,SRM,UNG",id:7},{pathway:"HALLMARK_ANGIOGENESIS",pval:.4917,padj:.7649,log2err:.0575,ES:.4953,NES:.9856,size:36,leadingEdge:"APP,FGFR1,LRPAP1,PDGFA,JAG1,JAG2,VAV2",id:8},{pathway:"HALLMARK_ANDROGEN_RESPONSE",pval:.6493,padj:.909,log2err:.0385,ES:.4085,NES:.916,size:99,leadingEdge:"FKBP5,SAT1,HPGD,ARID5B,RPS6KA3,ELOVL5,DHCR24,PMEPA1,PLPP1,PTK2B,SCD,GPD1L,HMGCS1,SRP19,NGLY1,UAP1,DBI,HOMER2",id:9},{pathway:"HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",pval:.8,padj:.9358,log2err:.034,ES:.397,NES:.8181,size:49,leadingEdge:"PRDX2,PRDX6,SELENOS,NDUFS2,FES,LSP1,NQO1,SBNO2,ABCC1,NDUFA6,SRXN1,NDUFB4,MBP,TXNRD1,PTPA,EGLN2,STK25",id:10},{pathway:"HALLMARK_PROTEIN_SECRETION",pval:.8181,padj:.9358,log2err:.0276,ES:.3685,NES:.8206,size:95,leadingEdge:"ERGIC3,GNAS,RPS6KA3,RER1,ICA1,AP2S1,COPB2,KIF1B,ARFIP1,ARF1,TSG101,STAM,VAMP4,GOLGA4,VAMP3,DNM1L,GALC,KRT18,ARCN1,TMED10,RAB2A,NAPG,IGF2R,YIPF6,SCAMP3,USO1,LMAN1,BNIP3,SGMS1,AP2M1,VAMP7,SNAP23,NAPA,ATP7A,ABCA1,COPE,AP1G1,BET1,ANP32E,TMED2,SCAMP1",id:11},{pathway:"HALLMARK_APICAL_SURFACE",pval:.9356,padj:.9358,log2err:.027,ES:.3511,NES:.7134,size:44,leadingEdge:"IL2RG,GATA3,APP,MAL,SCUBE1,TMEM8B,PLAUR,MDGA1,IL2RB,ATP8B1",id:12},{pathway:"HALLMARK_WNT_BETA_CATENIN_SIGNALING",pval:.9358,padj:.9358,log2err:.0269,ES:.3578,NES:.722,size:42,leadingEdge:"HDAC5,PPARD,RBPJ,NOTCH1,WNT1,JAG1,CTNNB1,JAG2,FRAT1,KAT2A,NOTCH4,AXIN1,HDAC2,TP53,LEF1,HEY1",id:13}]}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function Gs(e){let t,n,a,s;return t=new ue({props:{$$slots:{default:[vs]},$$scope:{ctx:e}}}),a=new ue({props:{$$slots:{default:[ys]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function xs(e){let t,n;return t=new me({props:{$$slots:{content:[Gs],default:[Ps]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function ws(e){let t,n,a,s,p,c,g,m,f,E,L,R,h,T,N,P,M,v,I,y,G,x,w,H,F,B,O,D,K,U,V,z,k,X,q,j,Y,W,J,Z,Q,te,ne,ae,se,re,le,ie,oe,pe,ce;return a=new $a({props:{title:"HALLMARK_INTERFERON_ALPHA_RESPONSE",class:"pipen-report-table-image-descr"}}),p=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/fgsea_HALLMARK_INTERFERON_ALPHA_RESPONSE.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/fgsea_HALLMARK_INTERFERON_ALPHA_RESPONSE.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),m=new $a({props:{title:"HALLMARK_IL6_JAK_STAT3_SIGNALING",class:"pipen-report-table-image-descr"}}),E=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/fgsea_HALLMARK_IL6_JAK_STAT3_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/fgsea_HALLMARK_IL6_JAK_STAT3_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),h=new $a({props:{title:"HALLMARK_CHOLESTEROL_HOMEOSTASIS",class:"pipen-report-table-image-descr"}}),N=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/fgsea_HALLMARK_CHOLESTEROL_HOMEOSTASIS.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/fgsea_HALLMARK_CHOLESTEROL_HOMEOSTASIS.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),v=new $a({props:{title:"HALLMARK_TGF_BETA_SIGNALING",class:"pipen-report-table-image-descr"}}),y=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/fgsea_HALLMARK_TGF_BETA_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/fgsea_HALLMARK_TGF_BETA_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),w=new $a({props:{title:"HALLMARK_PANCREAS_BETA_CELLS",class:"pipen-report-table-image-descr"}}),F=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/fgsea_HALLMARK_PANCREAS_BETA_CELLS.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/fgsea_HALLMARK_PANCREAS_BETA_CELLS.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),D=new $a({props:{title:"HALLMARK_NOTCH_SIGNALING",class:"pipen-report-table-image-descr"}}),U=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/fgsea_HALLMARK_NOTCH_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/fgsea_HALLMARK_NOTCH_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),k=new $a({props:{title:"HALLMARK_HEDGEHOG_SIGNALING",class:"pipen-report-table-image-descr"}}),q=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/fgsea_HALLMARK_HEDGEHOG_SIGNALING.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/fgsea_HALLMARK_HEDGEHOG_SIGNALING.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),W=new $a({props:{title:"HALLMARK_MYC_TARGETS_V2",class:"pipen-report-table-image-descr"}}),Z=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/fgsea_HALLMARK_MYC_TARGETS_V2.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/fgsea_HALLMARK_MYC_TARGETS_V2.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),ne=new $a({props:{title:"HALLMARK_ANGIOGENESIS",class:"pipen-report-table-image-descr"}}),se=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/fgsea_HALLMARK_ANGIOGENESIS.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/fgsea_HALLMARK_ANGIOGENESIS.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),ie=new $a({props:{title:"HALLMARK_ANDROGEN_RESPONSE",class:"pipen-report-table-image-descr"}}),pe=new Vn({props:{src:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/fgsea_HALLMARK_ANDROGEN_RESPONSE.png",download:"../ScFGSEA/samples.fgsea/seurat_clusters-Tem-Trm-cytotoxic-T-cells/fgsea_HALLMARK_ANDROGEN_RESPONSE.pdf",class:"pipen-report-table-image",width:1e3,height:800}}),{c(){t=r("div"),n=r("div"),u(a.$$.fragment),s=S(),u(p.$$.fragment),c=S(),g=r("div"),u(m.$$.fragment),f=S(),u(E.$$.fragment),L=S(),R=r("div"),u(h.$$.fragment),T=S(),u(N.$$.fragment),P=S(),M=r("div"),u(v.$$.fragment),I=S(),u(y.$$.fragment),G=S(),x=r("div"),u(w.$$.fragment),H=S(),u(F.$$.fragment),B=S(),O=r("div"),u(D.$$.fragment),K=S(),u(U.$$.fragment),V=S(),z=r("div"),u(k.$$.fragment),X=S(),u(q.$$.fragment),j=S(),Y=r("div"),u(W.$$.fragment),J=S(),u(Z.$$.fragment),Q=S(),te=r("div"),u(ne.$$.fragment),ae=S(),u(se.$$.fragment),re=S(),le=r("div"),u(ie.$$.fragment),oe=S(),u(pe.$$.fragment),l(t,"class","pipen-report-table-of-images"),ee(t,"grid-template-columns","repeat(2, auto)")},m(e,r){i(e,t,r),C(t,n),A(a,n,null),C(n,s),A(p,n,null),C(t,c),C(t,g),A(m,g,null),C(g,f),A(E,g,null),C(t,L),C(t,R),A(h,R,null),C(R,T),A(N,R,null),C(t,P),C(t,M),A(v,M,null),C(M,I),A(y,M,null),C(t,G),C(t,x),A(w,x,null),C(x,H),A(F,x,null),C(t,B),C(t,O),A(D,O,null),C(O,K),A(U,O,null),C(t,V),C(t,z),A(k,z,null),C(z,X),A(q,z,null),C(t,j),C(t,Y),A(W,Y,null),C(Y,J),A(Z,Y,null),C(t,Q),C(t,te),A(ne,te,null),C(te,ae),A(se,te,null),C(t,re),C(t,le),A(ie,le,null),C(le,oe),A(pe,le,null),ce=!0},p:b,i(e){ce||(o(a.$$.fragment,e),o(p.$$.fragment,e),o(m.$$.fragment,e),o(E.$$.fragment,e),o(h.$$.fragment,e),o(N.$$.fragment,e),o(v.$$.fragment,e),o(y.$$.fragment,e),o(w.$$.fragment,e),o(F.$$.fragment,e),o(D.$$.fragment,e),o(U.$$.fragment,e),o(k.$$.fragment,e),o(q.$$.fragment,e),o(W.$$.fragment,e),o(Z.$$.fragment,e),o(ne.$$.fragment,e),o(se.$$.fragment,e),o(ie.$$.fragment,e),o(pe.$$.fragment,e),ce=!0)},o(e){_(a.$$.fragment,e),_(p.$$.fragment,e),_(m.$$.fragment,e),_(E.$$.fragment,e),_(h.$$.fragment,e),_(N.$$.fragment,e),_(v.$$.fragment,e),_(y.$$.fragment,e),_(w.$$.fragment,e),_(F.$$.fragment,e),_(D.$$.fragment,e),_(U.$$.fragment,e),_(k.$$.fragment,e),_(q.$$.fragment,e),_(W.$$.fragment,e),_(Z.$$.fragment,e),_(ne.$$.fragment,e),_(se.$$.fragment,e),_(ie.$$.fragment,e),_(pe.$$.fragment,e),ce=!1},d(e){e&&d(t),$(a),$(p),$(m),$(E),$(h),$(N),$(v),$(y),$(w),$(F),$(D),$(U),$(k),$(q),$(W),$(Z),$(ne),$(se),$(ie),$(pe)}}}function Hs(e){let t,n,a,s;return t=new Ge({props:{title:"Enrichment Analysis Summary",open:!0,$$slots:{default:[xs]},$$scope:{ctx:e}}}),a=new Ge({props:{title:"Enriched Pathways (Top 10)",$$slots:{default:[ws]},$$scope:{ctx:e}}}),{c(){u(t.$$.fragment),n=S(),u(a.$$.fragment)},m(e,r){A(t,e,r),i(e,n,r),A(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(o(t.$$.fragment,e),o(a.$$.fragment,e),s=!0)},o(e){_(t.$$.fragment,e),_(a.$$.fragment,e),s=!1},d(e){e&&d(n),$(t,e),$(a,e)}}}function Fs(e){let t,n,a,s,p,c,g,m,f,E,L,R,h,C,T,N,b,P,M,v,I,y,G,x,w,H,F,B,O,D,K,U,V,z,k,X,q,j,Y,W;return s=new ye({props:{align:"start",$$slots:{default:[ba]},$$scope:{ctx:e}}}),f=new ye({props:{align:"start",$$slots:{default:[Ha]},$$scope:{ctx:e}}}),C=new ye({props:{align:"start",$$slots:{default:[ka]},$$scope:{ctx:e}}}),M=new ye({props:{align:"start",$$slots:{default:[es]},$$scope:{ctx:e}}}),x=new ye({props:{align:"start",$$slots:{default:[ps]},$$scope:{ctx:e}}}),O=new ye({props:{align:"start",$$slots:{default:[$s]},$$scope:{ctx:e}}}),z=new ye({props:{align:"start",$$slots:{default:[bs]},$$scope:{ctx:e}}}),Y=new ye({props:{align:"start",$$slots:{default:[Hs]},$$scope:{ctx:e}}}),{c(){t=r("h1"),t.textContent="seurat_clusters - Cycling T cells",n=r("a"),n.innerHTML="",a=S(),u(s.$$.fragment),p=S(),c=r("h1"),c.textContent="seurat_clusters - Follicular helper T cells",g=r("a"),g.innerHTML="",m=S(),u(f.$$.fragment),E=S(),L=r("h1"),L.textContent="seurat_clusters - gamma-delta T cells",R=r("a"),R.innerHTML="",h=S(),u(C.$$.fragment),T=S(),N=r("h1"),N.textContent="seurat_clusters - Plasmablasts",b=r("a"),b.innerHTML="",P=S(),u(M.$$.fragment),v=S(),I=r("h1"),I.textContent="seurat_clusters - Regulatory T cells",y=r("a"),y.innerHTML="",G=S(),u(x.$$.fragment),w=S(),H=r("h1"),H.textContent="seurat_clusters - Regulatory T cells.1",F=r("a"),F.innerHTML="",B=S(),u(O.$$.fragment),D=S(),K=r("h1"),K.textContent="seurat_clusters - Tcm/Naive helper T cells",U=r("a"),U.innerHTML="",V=S(),u(z.$$.fragment),k=S(),X=r("h1"),X.textContent="seurat_clusters - Tem/Trm cytotoxic T cells",q=r("a"),q.innerHTML="",j=S(),u(Y.$$.fragment),l(n,"id","prt-h1-0-seurat-clusters-cycling-t-cells"),l(n,"class","pipen-report-toc-anchor"),l(g,"id","prt-h1-1-seurat-clusters-follicular-helper-t-cells"),l(g,"class","pipen-report-toc-anchor"),l(R,"id","prt-h1-2-seurat-clusters-gamma-delta-t-cells"),l(R,"class","pipen-report-toc-anchor"),l(b,"id","prt-h1-3-seurat-clusters-plasmablasts"),l(b,"class","pipen-report-toc-anchor"),l(y,"id","prt-h1-4-seurat-clusters-regulatory-t-cells"),l(y,"class","pipen-report-toc-anchor"),l(F,"id","prt-h1-5-seurat-clusters-regulatory-t-cells-1"),l(F,"class","pipen-report-toc-anchor"),l(U,"id","prt-h1-6-seurat-clusters-tcm-naive-helper-t-cells"),l(U,"class","pipen-report-toc-anchor"),l(q,"id","prt-h1-7-seurat-clusters-tem-trm-cytotoxic-t-cells"),l(q,"class","pipen-report-toc-anchor")},m(e,r){i(e,t,r),i(e,n,r),i(e,a,r),A(s,e,r),i(e,p,r),i(e,c,r),i(e,g,r),i(e,m,r),A(f,e,r),i(e,E,r),i(e,L,r),i(e,R,r),i(e,h,r),A(C,e,r),i(e,T,r),i(e,N,r),i(e,b,r),i(e,P,r),A(M,e,r),i(e,v,r),i(e,I,r),i(e,y,r),i(e,G,r),A(x,e,r),i(e,w,r),i(e,H,r),i(e,F,r),i(e,B,r),A(O,e,r),i(e,D,r),i(e,K,r),i(e,U,r),i(e,V,r),A(z,e,r),i(e,k,r),i(e,X,r),i(e,q,r),i(e,j,r),A(Y,e,r),W=!0},p(e,[t]){const n={};1&t&&(n.$$scope={dirty:t,ctx:e}),s.$set(n);const a={};1&t&&(a.$$scope={dirty:t,ctx:e}),f.$set(a);const r={};1&t&&(r.$$scope={dirty:t,ctx:e}),C.$set(r);const l={};1&t&&(l.$$scope={dirty:t,ctx:e}),M.$set(l);const i={};1&t&&(i.$$scope={dirty:t,ctx:e}),x.$set(i);const o={};1&t&&(o.$$scope={dirty:t,ctx:e}),O.$set(o);const p={};1&t&&(p.$$scope={dirty:t,ctx:e}),z.$set(p);const c={};1&t&&(c.$$scope={dirty:t,ctx:e}),Y.$set(c)},i(e){W||(o(s.$$.fragment,e),o(f.$$.fragment,e),o(C.$$.fragment,e),o(M.$$.fragment,e),o(x.$$.fragment,e),o(O.$$.fragment,e),o(z.$$.fragment,e),o(Y.$$.fragment,e),W=!0)},o(e){_(s.$$.fragment,e),_(f.$$.fragment,e),_(C.$$.fragment,e),_(M.$$.fragment,e),_(x.$$.fragment,e),_(O.$$.fragment,e),_(z.$$.fragment,e),_(Y.$$.fragment,e),W=!1},d(e){e&&(d(t),d(n),d(a),d(p),d(c),d(g),d(m),d(E),d(L),d(R),d(h),d(T),d(N),d(b),d(P),d(v),d(I),d(y),d(G),d(w),d(H),d(F),d(B),d(D),d(K),d(U),d(V),d(k),d(X),d(q),d(j)),$(s,e),$(f,e),$(C,e),$(M,e),$(x,e),$(O,e),$(z,e),$(Y,e)}}}class Bs extends t{constructor(e){super(),n(this,e,null,Fs,a,{})}}var Os=[{slug:"prt-h1-0-seurat-clusters-cycling-t-cells",text:"seurat_clusters - Cycling T cells",children:[],page:0},{slug:"prt-h1-1-seurat-clusters-follicular-helper-t-cells",text:"seurat_clusters - Follicular helper T cells",children:[],page:0},{slug:"prt-h1-2-seurat-clusters-gamma-delta-t-cells",text:"seurat_clusters - gamma-delta T cells",children:[],page:0},{slug:"prt-h1-3-seurat-clusters-plasmablasts",text:"seurat_clusters - Plasmablasts",children:[],page:0},{slug:"prt-h1-4-seurat-clusters-regulatory-t-cells",text:"seurat_clusters - Regulatory T cells",children:[],page:0},{slug:"prt-h1-5-seurat-clusters-regulatory-t-cells-1",text:"seurat_clusters - Regulatory T cells.1",children:[],page:0},{slug:"prt-h1-6-seurat-clusters-tcm-naive-helper-t-cells",text:"seurat_clusters - Tcm/Naive helper T cells",children:[],page:0},{slug:"prt-h1-7-seurat-clusters-tem-trm-cytotoxic-t-cells",text:"seurat_clusters - Tem/Trm cytotoxic T cells",children:[],page:0},{slug:"prt-h1-8-seurat-clusters-tem-trm-cytotoxic-t-cells-1",text:"seurat_clusters - Tem/Trm cytotoxic T cells.1",children:[],page:1},{slug:"prt-h1-9-seurat-clusters-trm-cytotoxic-t-cells",text:"seurat_clusters - Trm cytotoxic T cells",children:[],page:1},{slug:"prt-h1-10-seurat-clusters-type-1-helper-t-cells",text:"seurat_clusters - Type 1 helper T cells",children:[],page:1},{slug:"prt-h1-11-seurat-clusters-type-1-helper-t-cells-1",text:"seurat_clusters - Type 1 helper T cells.1",children:[],page:1}],Ds={pipeline:{name:"Immunopipe",desc:"Immunopipe (v1.4.3): Integrative analysis for scRNA-seq and scTCR-seq data"},versions:'\n    <a href="https://github.com/pwwang/pipen" target="_blank">\n        pipen v0.15.7\n    </a> and\n    <a href="https://github.com/pwwang/pipen-report" target="_blank">\n        pipen-report v0.21.4\n    </a>\n',entries:[{name:"SampleInfo",desc:"List sample information and perform statistics",npages:1,report_toc:!0,order:0,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n[1] ggrepel_0.9.6    ggprism_1.0.5    ggplot2_3.5.1    tidyr_1.3.1     \n[5] dplyr_1.1.4      tidyselect_1.2.1 jsonlite_1.8.9   rlang_1.1.5     \n[9] logger_0.4.0    \n\nloaded via a namespace (and not attached):\n [1] vctrs_0.6.5      cli_3.6.4        purrr_1.0.2      generics_0.1.3  \n [5] glue_1.8.0       labeling_0.4.3   colorspace_2.1-1 scales_1.3.0    \n [9] grid_4.3.3       munsell_0.5.1    tibble_3.2.1     lifecycle_1.0.4 \n[13] compiler_4.3.3   Rcpp_1.0.14      pkgconfig_2.0.3  farver_2.1.2    \n[17] digest_0.6.37    R6_2.6.0         pillar_1.10.1    magrittr_2.0.3  \n[21] withr_3.0.2      gtable_0.3.6    \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 2.19\nuser 11.15\nsys 0.11\npercent 513%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3421.613\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"SeuratPreparing",desc:"Load, prepare and apply QC to data, using `Seurat`",npages:1,report_toc:!0,order:2,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats4    stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] scDblFinder_1.16.0          SingleCellExperiment_1.24.0\n [3] SummarizedExperiment_1.32.0 Biobase_2.62.0             \n [5] GenomicRanges_1.54.1        GenomeInfoDb_1.38.1        \n [7] IRanges_2.36.0              S4Vectors_0.40.2           \n [9] BiocGenerics_0.48.1         MatrixGenerics_1.14.0      \n[11] matrixStats_1.1.0           dplyr_1.1.4                \n[13] ggplot2_3.5.1               bracer_1.2.1               \n[15] future_1.34.0               Seurat_5.1.0               \n[17] SeuratObject_5.0.2          sp_2.2-0                   \n[19] digest_0.6.37               jsonlite_1.8.9             \n[21] rlang_1.1.5                 logger_0.4.0               \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22          splines_4.3.3            \n  [3] later_1.4.1               BiocIO_1.12.0            \n  [5] bitops_1.0-9              tibble_3.2.1             \n  [7] R.oo_1.27.0               polyclip_1.10-7          \n  [9] XML_3.99-0.17             fastDummies_1.7.5        \n [11] lifecycle_1.0.4           edgeR_4.0.16             \n [13] globals_0.16.3            lattice_0.22-6           \n [15] MASS_7.3-60.0.1           magrittr_2.0.3           \n [17] limma_3.58.1              plotly_4.10.4            \n [19] yaml_2.3.10               metapod_1.10.0           \n [21] httpuv_1.6.15             sctransform_0.4.1        \n [23] spam_2.11-1               spatstat.sparse_3.1-0    \n [25] reticulate_1.40.0         cowplot_1.1.3            \n [27] pbapply_1.7-2             RColorBrewer_1.1-3       \n [29] abind_1.4-5               zlibbioc_1.48.0          \n [31] Rtsne_0.17                purrr_1.0.2              \n [33] R.utils_2.12.3            RCurl_1.98-1.16          \n [35] GenomeInfoDbData_1.2.11   ggrepel_0.9.6            \n [37] irlba_2.3.5.1             listenv_0.9.1            \n [39] spatstat.utils_3.1-2      goftest_1.2-3            \n [41] RSpectra_0.16-2           dqrng_0.3.2              \n [43] spatstat.random_3.3-2     fitdistrplus_1.2-2       \n [45] parallelly_1.42.0         DelayedMatrixStats_1.24.0\n [47] leiden_0.4.3.1            codetools_0.2-20         \n [49] DelayedArray_0.28.0       scuttle_1.12.0           \n [51] tidyselect_1.2.1          farver_2.1.2             \n [53] viridis_0.6.5             ScaledMatrix_1.10.0      \n [55] spatstat.explore_3.3-4    GenomicAlignments_1.38.0 \n [57] BiocNeighbors_1.20.0      progressr_0.15.1         \n [59] scater_1.30.1             ggridges_0.5.6           \n [61] survival_3.8-3            systemfonts_1.2.1        \n [63] tools_4.3.3               ragg_1.3.3               \n [65] ica_1.0-3                 Rcpp_1.0.14              \n [67] glue_1.8.0                gridExtra_2.3            \n [69] SparseArray_1.2.2         withr_3.0.2              \n [71] fastmap_1.2.0             bluster_1.12.0           \n [73] rsvd_1.0.5                R6_2.6.0                 \n [75] mime_0.12                 textshaping_0.4.0        \n [77] colorspace_2.1-1          scattermore_1.2          \n [79] tensor_1.5                spatstat.data_3.1-4      \n [81] R.methodsS3_1.8.2         RhpcBLASctl_0.23-42      \n [83] tidyr_1.3.1               generics_0.1.3           \n [85] data.table_1.16.4         rtracklayer_1.62.0       \n [87] httr_1.4.7                htmlwidgets_1.6.4        \n [89] S4Arrays_1.2.0            uwot_0.2.2               \n [91] pkgconfig_2.0.3           gtable_0.3.6             \n [93] lmtest_0.9-40             XVector_0.42.0           \n [95] htmltools_0.5.8.1         dotCall64_1.2            \n [97] scales_1.3.0              png_0.1-8                \n [99] harmony_1.2.0             spatstat.univar_3.1-1    \n[101] scran_1.30.0              reshape2_1.4.4           \n[103] rjson_0.2.23              nlme_3.1-167             \n[105] zoo_1.8-12                stringr_1.5.1            \n[107] KernSmooth_2.23-26        vipor_0.4.7              \n[109] parallel_4.3.3            miniUI_0.1.1.1           \n[111] restfulr_0.0.15           pillar_1.10.1            \n[113] grid_4.3.3                vctrs_0.6.5              \n[115] RANN_2.6.2                promises_1.3.2           \n[117] BiocSingular_1.18.0       beachmat_2.18.0          \n[119] xtable_1.8-4              cluster_2.1.8            \n[121] beeswarm_0.4.0            locfit_1.5-9.11          \n[123] cli_3.6.4                 compiler_4.3.3           \n[125] Rsamtools_2.18.0          crayon_1.5.3             \n[127] future.apply_1.11.3       labeling_0.4.3           \n[129] ggbeeswarm_0.7.2          plyr_1.8.9               \n[131] stringi_1.8.4             viridisLite_0.4.2        \n[133] deldir_2.0-4              BiocParallel_1.36.0      \n[135] munsell_0.5.1             Biostrings_2.70.1        \n[137] lazyeval_0.2.2            spatstat.geom_3.3-5      \n[139] Matrix_1.6-5              RcppHNSW_0.6.0           \n[141] patchwork_1.3.0           sparseMatrixStats_1.14.0 \n[143] statmod_1.5.0             shiny_1.10.0             \n[145] ROCR_1.0-11               igraph_2.0.3             \n[147] xgboost_2.1.4.1          \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 209.65\nuser 1454.55\nsys 7.57\npercent 697%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ClusterMarkersOfAllCells",desc:"Markers for clusters of all cells.",npages:3,report_toc:!0,order:4,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] UpSetR_1.4.0        ggVennDiagram_1.2.3 tidyseurat_0.8.0   \n [4] ttservice_0.4.1     future_1.34.0       ggrepel_0.9.6      \n [7] ggprism_1.0.5       ggplot2_3.5.1       enrichR_3.4        \n[10] Seurat_5.1.0        SeuratObject_5.0.2  sp_2.2-0           \n[13] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[16] tidyselect_1.2.1    digest_0.6.37       jsonlite_1.8.9     \n[19] rlang_1.1.5         logger_0.4.0       \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] ellipsis_0.3.2         labeling_0.4.3         promises_1.3.2        \n [19] purrr_1.0.2            WriteXLS_6.7.0         RVenn_1.1.0           \n [22] goftest_1.2-3          later_1.4.1            spatstat.utils_3.1-2  \n [25] irlba_2.3.5.1          parallel_4.3.3         cluster_2.1.8         \n [28] R6_2.6.0               ica_1.0-3              stringi_1.8.4         \n [31] RColorBrewer_1.1-3     spatstat.data_3.1-4    limma_3.58.1          \n [34] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [37] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [40] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [43] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [46] splines_4.3.3          igraph_2.0.3           abind_1.4-5           \n [49] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.3-4 curl_6.0.1             listenv_0.9.1         \n [55] lattice_0.22-6         plyr_1.8.9             withr_3.0.2           \n [58] shiny_1.10.0           ROCR_1.0-11            Rtsne_0.17            \n [61] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n [64] fitdistrplus_1.2-2     pillar_1.10.1          KernSmooth_2.23-26    \n [67] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n [82] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n [85] nlme_3.1-167           patchwork_1.3.0        presto_1.0.0          \n [88] cli_3.6.4              spatstat.sparse_3.1-0  fansi_1.0.6           \n [91] spam_2.11-1            viridisLite_0.4.2      uwot_0.2.2            \n [94] gtable_0.3.6           progressr_0.15.1       rjson_0.2.23          \n [97] htmlwidgets_1.6.4      farver_2.1.2           htmltools_0.5.8.1     \n[100] lifecycle_1.0.4        httr_1.4.7             statmod_1.5.0         \n[103] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 259.76\nuser 186.00\nsys 8.62\npercent 74%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TopExpressingGenesOfAllCells",desc:"Top expressing genes for clusters of all cells.",npages:3,report_toc:!0,order:5,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ggprism_1.0.5      dplyr_1.1.4        enrichR_3.4        tibble_3.2.1      \n [5] Seurat_5.1.0       SeuratObject_5.0.2 sp_2.2-0           jsonlite_1.8.9    \n [9] rlang_1.1.5        logger_0.4.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] labeling_0.4.3         promises_1.3.2         purrr_1.0.2           \n [19] WriteXLS_6.7.0         goftest_1.2-3          later_1.4.1           \n [22] spatstat.utils_3.1-2   irlba_2.3.5.1          parallel_4.3.3        \n [25] cluster_2.1.8          R6_2.6.0               ica_1.0-3             \n [28] stringi_1.8.4          RColorBrewer_1.1-3     spatstat.data_3.1-4   \n [31] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [37] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [43] splines_4.3.3          igraph_2.0.3           tidyselect_1.2.1      \n [46] abind_1.4-5            spatstat.random_3.3-2  codetools_0.2-20      \n [49] miniUI_0.1.1.1         spatstat.explore_3.3-4 curl_6.0.1            \n [52] listenv_0.9.1          lattice_0.22-6         plyr_1.8.9            \n [55] withr_3.0.2            shiny_1.10.0           ROCR_1.0-11           \n [58] Rtsne_0.17             future_1.34.0          fastDummies_1.7.5     \n [61] survival_3.8-3         polyclip_1.10-7        fitdistrplus_1.2-2    \n [64] pillar_1.10.1          KernSmooth_2.23-26     plotly_4.10.4         \n [67] generics_0.1.3         RcppHNSW_0.6.0         ggplot2_3.5.1         \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n [82] cowplot_1.1.3          grid_4.3.3             tidyr_1.3.1           \n [85] colorspace_2.1-1       nlme_3.1-167           patchwork_1.3.0       \n [88] cli_3.6.4              spatstat.sparse_3.1-0  spam_2.11-1           \n [91] viridisLite_0.4.2      uwot_0.2.2             gtable_0.3.6          \n [94] digest_0.6.37          progressr_0.15.1       ggrepel_0.9.6         \n [97] rjson_0.2.23           htmlwidgets_1.6.4      farver_2.1.2          \n[100] htmltools_0.5.8.1      lifecycle_1.0.4        httr_1.4.7            \n[103] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 73.70\nuser 41.45\nsys 2.53\npercent 59%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TCellSelection",desc:"Separate T and non-T cells and select T cells.",npages:1,report_toc:!0,order:6,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] glue_1.8.0         factoextra_1.0.7   patchwork_1.3.0    ggpubr_0.6.0      \n [5] ggrepel_0.9.6      ggprism_1.0.5      ggplot2_3.5.1      tibble_3.2.1      \n [9] tidyr_1.3.1        dplyr_1.1.4        Seurat_5.1.0       SeuratObject_5.0.2\n[13] sp_2.2-0           jsonlite_1.8.9     rlang_1.1.5        logger_0.4.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] backports_1.5.0        labeling_0.4.3         promises_1.3.2        \n [19] purrr_1.0.2            goftest_1.2-3          later_1.4.1           \n [22] spatstat.utils_3.1-2   broom_1.0.7            irlba_2.3.5.1         \n [25] parallel_4.3.3         cluster_2.1.8          R6_2.6.0              \n [28] ica_1.0-3              stringi_1.8.4          RColorBrewer_1.1-3    \n [31] spatstat.data_3.1-4    reticulate_1.40.0      car_3.1-3             \n [34] parallelly_1.42.0      spatstat.univar_3.1-1  lmtest_0.9-40         \n [37] scattermore_1.2        Rcpp_1.0.14            tensor_1.5            \n [40] future.apply_1.11.3    zoo_1.8-12             sctransform_0.4.1     \n [43] httpuv_1.6.15          Matrix_1.6-5           splines_4.3.3         \n [46] igraph_2.0.3           tidyselect_1.2.1       abind_1.4-5           \n [49] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.3-4 listenv_0.9.1          lattice_0.22-6        \n [55] plyr_1.8.9             withr_3.0.2            shiny_1.10.0          \n [58] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n [61] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n [64] fitdistrplus_1.2-2     pillar_1.10.1          carData_3.0-5         \n [67] KernSmooth_2.23-26     plotly_4.10.4          generics_0.1.3        \n [70] RcppHNSW_0.6.0         munsell_0.5.1          scales_1.3.0          \n [73] globals_0.16.3         xtable_1.8-4           lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] ggsignif_0.6.4         RANN_2.6.2             leiden_0.4.3.1        \n [82] dotCall64_1.2          cowplot_1.1.3          grid_4.3.3            \n [85] colorspace_2.1-1       nlme_3.1-167           Formula_1.2-5         \n [88] cli_3.6.4              spatstat.sparse_3.1-0  spam_2.11-1           \n [91] viridisLite_0.4.2      uwot_0.2.2             gtable_0.3.6          \n [94] rstatix_0.7.2          digest_0.6.37          progressr_0.15.1      \n [97] htmlwidgets_1.6.4      farver_2.1.2           htmltools_0.5.8.1     \n[100] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[103] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 41.63\nuser 49.85\nsys 0.77\npercent 121%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ClusterMarkers",desc:"Markers for clusters of all or selected T cells.",npages:2,report_toc:!0,order:10,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] UpSetR_1.4.0        ggVennDiagram_1.2.3 tidyseurat_0.8.0   \n [4] ttservice_0.4.1     future_1.34.0       ggrepel_0.9.6      \n [7] ggprism_1.0.5       ggplot2_3.5.1       enrichR_3.4        \n[10] Seurat_5.1.0        SeuratObject_5.0.2  sp_2.2-0           \n[13] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[16] tidyselect_1.2.1    digest_0.6.37       jsonlite_1.8.9     \n[19] rlang_1.1.5         logger_0.4.0       \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] ellipsis_0.3.2         labeling_0.4.3         promises_1.3.2        \n [19] purrr_1.0.2            WriteXLS_6.7.0         RVenn_1.1.0           \n [22] goftest_1.2-3          later_1.4.1            spatstat.utils_3.1-2  \n [25] irlba_2.3.5.1          parallel_4.3.3         cluster_2.1.8         \n [28] R6_2.6.0               ica_1.0-3              stringi_1.8.4         \n [31] RColorBrewer_1.1-3     spatstat.data_3.1-4    limma_3.58.1          \n [34] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [37] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [40] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [43] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [46] splines_4.3.3          igraph_2.0.3           abind_1.4-5           \n [49] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.3-4 curl_6.0.1             listenv_0.9.1         \n [55] lattice_0.22-6         plyr_1.8.9             withr_3.0.2           \n [58] shiny_1.10.0           ROCR_1.0-11            Rtsne_0.17            \n [61] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n [64] fitdistrplus_1.2-2     pillar_1.10.1          KernSmooth_2.23-26    \n [67] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n [82] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n [85] nlme_3.1-167           patchwork_1.3.0        presto_1.0.0          \n [88] cli_3.6.4              spatstat.sparse_3.1-0  fansi_1.0.6           \n [91] spam_2.11-1            viridisLite_0.4.2      uwot_0.2.2            \n [94] gtable_0.3.6           progressr_0.15.1       rjson_0.2.23          \n [97] htmlwidgets_1.6.4      farver_2.1.2           htmltools_0.5.8.1     \n[100] lifecycle_1.0.4        httr_1.4.7             statmod_1.5.0         \n[103] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 162.57\nuser 119.15\nsys 5.40\npercent 76%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TopExpressingGenes",desc:"Top expressing genes for clusters of all or selected T cells.",npages:2,report_toc:!0,order:11,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ggprism_1.0.5      dplyr_1.1.4        enrichR_3.4        tibble_3.2.1      \n [5] Seurat_5.1.0       SeuratObject_5.0.2 sp_2.2-0           jsonlite_1.8.9    \n [9] rlang_1.1.5        logger_0.4.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] labeling_0.4.3         promises_1.3.2         purrr_1.0.2           \n [19] WriteXLS_6.7.0         goftest_1.2-3          later_1.4.1           \n [22] spatstat.utils_3.1-2   irlba_2.3.5.1          parallel_4.3.3        \n [25] cluster_2.1.8          R6_2.6.0               ica_1.0-3             \n [28] stringi_1.8.4          RColorBrewer_1.1-3     spatstat.data_3.1-4   \n [31] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [37] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [43] splines_4.3.3          igraph_2.0.3           tidyselect_1.2.1      \n [46] abind_1.4-5            spatstat.random_3.3-2  codetools_0.2-20      \n [49] miniUI_0.1.1.1         spatstat.explore_3.3-4 curl_6.0.1            \n [52] listenv_0.9.1          lattice_0.22-6         plyr_1.8.9            \n [55] withr_3.0.2            shiny_1.10.0           ROCR_1.0-11           \n [58] Rtsne_0.17             future_1.34.0          fastDummies_1.7.5     \n [61] survival_3.8-3         polyclip_1.10-7        fitdistrplus_1.2-2    \n [64] pillar_1.10.1          KernSmooth_2.23-26     plotly_4.10.4         \n [67] generics_0.1.3         RcppHNSW_0.6.0         ggplot2_3.5.1         \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n [82] cowplot_1.1.3          grid_4.3.3             tidyr_1.3.1           \n [85] colorspace_2.1-1       nlme_3.1-167           patchwork_1.3.0       \n [88] cli_3.6.4              spatstat.sparse_3.1-0  spam_2.11-1           \n [91] viridisLite_0.4.2      uwot_0.2.2             gtable_0.3.6          \n [94] digest_0.6.37          progressr_0.15.1       ggrepel_0.9.6         \n [97] rjson_0.2.23           htmlwidgets_1.6.4      farver_2.1.2          \n[100] htmltools_0.5.8.1      lifecycle_1.0.4        httr_1.4.7            \n[103] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 45.21\nuser 28.74\nsys 1.92\npercent 67%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TCRClusterStats",desc:"Statistics of TCR clusters, generated by `TCRClustering`.",npages:1,report_toc:!0,order:13,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] grid      stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] ComplexHeatmap_2.18.0 ggprism_1.0.5         immunarch_0.9.1      \n [4] patchwork_1.3.0       data.table_1.16.4     dtplyr_1.3.1         \n [7] dplyr_1.1.4           tidyr_1.3.1           tibble_3.2.1         \n[10] ggplot2_3.5.1         jsonlite_1.8.9        rlang_1.1.5          \n[13] logger_0.4.0         \n\nloaded via a namespace (and not attached):\n  [1] gridExtra_2.3       phangorn_2.12.1     readxl_1.4.3       \n  [4] magrittr_2.0.3      clue_0.3-66         GetoptLong_1.0.5   \n  [7] matrixStats_1.1.0   compiler_4.3.3      flexmix_2.3-19     \n [10] png_0.1-8           vctrs_0.6.5         reshape2_1.4.4     \n [13] ggalluvial_0.12.5   quadprog_1.5-8      stringr_1.5.1      \n [16] crayon_1.5.3        pkgconfig_2.0.3     shape_1.4.6.1      \n [19] fastmap_1.2.0       backports_1.5.0     labeling_0.4.3     \n [22] ggraph_2.2.1        promises_1.3.2      tzdb_0.4.0         \n [25] UpSetR_1.4.0        purrr_1.0.2         modeltools_0.2-23  \n [28] ggseqlogo_0.2       cachem_1.1.0        later_1.4.1        \n [31] uuid_1.2-1          fpc_2.2-13          tweenr_2.0.3       \n [34] broom_1.0.7         parallel_4.3.3      prabclus_2.3-4     \n [37] cluster_2.1.8       R6_2.6.0            stringi_1.8.4      \n [40] RColorBrewer_1.1-3  rlist_0.4.6.2       car_3.1-3          \n [43] diptest_0.77-1      cellranger_1.1.0    Rcpp_1.0.14        \n [46] iterators_1.0.14    IRanges_2.36.0      readr_2.1.5        \n [49] httpuv_1.6.15       Matrix_1.6-5        nnet_7.3-20        \n [52] igraph_2.0.3        tidyselect_1.2.1    stringdist_0.9.15  \n [55] abind_1.4-5         viridis_0.6.5       doParallel_1.0.17  \n [58] codetools_0.2-20    lattice_0.22-6      plyr_1.8.9         \n [61] shiny_1.10.0        withr_3.0.2         polyclip_1.10-7    \n [64] circlize_0.4.16     mclust_6.1.1        kernlab_0.9-33     \n [67] pillar_1.10.1       ggpubr_0.6.0        carData_3.0-5      \n [70] foreach_1.5.2       stats4_4.3.3        generics_0.1.3     \n [73] S4Vectors_0.40.2    hms_1.1.3           munsell_0.5.1      \n [76] scales_1.3.0        xtable_1.8-4        class_7.3-23       \n [79] glue_1.8.0          pheatmap_1.0.12     tools_4.3.3        \n [82] robustbase_0.99-4-1 ggsignif_0.6.4      graphlayouts_1.2.2 \n [85] Cairo_1.6-2         fastmatch_1.1-6     tidygraph_1.3.0    \n [88] ape_5.8-1           colorspace_2.1-1    nlme_3.1-167       \n [91] ggforce_0.4.2       Formula_1.2-5       cli_3.6.4          \n [94] viridisLite_0.4.2   gtable_0.3.6        DEoptimR_1.1-3-1   \n [97] rstatix_0.7.2       digest_0.6.37       BiocGenerics_0.48.1\n[100] ggrepel_0.9.6       rjson_0.2.23        farver_2.1.2       \n[103] memoise_2.0.1       htmltools_0.5.8.1   lifecycle_1.0.4    \n[106] factoextra_1.0.7    mime_0.12           GlobalOptions_0.1.2\n[109] shinythemes_1.2.0   MASS_7.3-60.0.1    \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 7.17\nuser 15.93\nsys 0.32\npercent 226%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3287.293\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TESSA",desc:"Tessa is a Bayesian model to integrate T cell receptor (TCR) sequence profiling with transcriptomes of T cells.",npages:1,report_toc:!0,order:14,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] igraph_2.0.3         Rtsne_0.17           LaplacesDemon_16.1.6\n [4] MASS_7.3-60.0.1      ggprism_1.0.5        Seurat_5.1.0        \n [7] SeuratObject_5.0.2   sp_2.2-0             tibble_3.2.1        \n[10] glue_1.8.0           immunarch_0.9.1      patchwork_1.3.0     \n[13] data.table_1.16.4    dtplyr_1.3.1         ggplot2_3.5.1       \n[16] tidyr_1.3.1          dplyr_1.1.4          jsonlite_1.8.9      \n[19] rlang_1.1.5          logger_0.4.0        \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22       shinythemes_1.2.0      splines_4.3.3         \n  [4] later_1.4.1            cellranger_1.1.0       polyclip_1.10-7       \n  [7] fastDummies_1.7.5      factoextra_1.0.7       lifecycle_1.0.4       \n [10] rstatix_0.7.2          doParallel_1.0.17      globals_0.16.3        \n [13] lattice_0.22-6         prabclus_2.3-4         backports_1.5.0       \n [16] magrittr_2.0.3         plotly_4.10.4          rlist_0.4.6.2         \n [19] httpuv_1.6.15          sctransform_0.4.1      spam_2.11-1           \n [22] flexmix_2.3-19         spatstat.sparse_3.1-0  reticulate_1.40.0     \n [25] cowplot_1.1.3          pbapply_1.7-2          RColorBrewer_1.1-3    \n [28] abind_1.4-5            quadprog_1.5-8         purrr_1.0.2           \n [31] ggraph_2.2.1           nnet_7.3-20            tweenr_2.0.3          \n [34] circlize_0.4.16        ggrepel_0.9.6          irlba_2.3.5.1         \n [37] listenv_0.9.1          spatstat.utils_3.1-2   pheatmap_1.0.12       \n [40] goftest_1.2-3          RSpectra_0.16-2        spatstat.random_3.3-2 \n [43] fitdistrplus_1.2-2     parallelly_1.42.0      leiden_0.4.3.1        \n [46] codetools_0.2-20       ggforce_0.4.2          tidyselect_1.2.1      \n [49] shape_1.4.6.1          ggseqlogo_0.2          farver_2.1.2          \n [52] viridis_0.6.5          matrixStats_1.1.0      stats4_4.3.3          \n [55] spatstat.explore_3.3-4 tidygraph_1.3.0        progressr_0.15.1      \n [58] Formula_1.2-5          ggridges_0.5.6         ggalluvial_0.12.5     \n [61] survival_3.8-3         iterators_1.0.14       foreach_1.5.2         \n [64] tools_4.3.3            stringdist_0.9.15      ica_1.0-3             \n [67] Rcpp_1.0.14            gridExtra_2.3          withr_3.0.2           \n [70] fastmap_1.2.0          digest_0.6.37          R6_2.6.0              \n [73] mime_0.12              colorspace_2.1-1       scattermore_1.2       \n [76] tensor_1.5             spatstat.data_3.1-4    diptest_0.77-1        \n [79] UpSetR_1.4.0           generics_0.1.3         robustbase_0.99-4-1   \n [82] class_7.3-23           graphlayouts_1.2.2     httr_1.4.7            \n [85] htmlwidgets_1.6.4      uwot_0.2.2             pkgconfig_2.0.3       \n [88] gtable_0.3.6           modeltools_0.2-23      lmtest_0.9-40         \n [91] htmltools_0.5.8.1      carData_3.0-5          dotCall64_1.2         \n [94] scales_1.3.0           png_0.1-8              spatstat.univar_3.1-1 \n [97] tzdb_0.4.0             reshape2_1.4.4         uuid_1.2-1            \n[100] nlme_3.1-167           cachem_1.1.0           zoo_1.8-12            \n[103] GlobalOptions_0.1.2    stringr_1.5.1          KernSmooth_2.23-26    \n[106] parallel_4.3.3         miniUI_0.1.1.1         pillar_1.10.1         \n[109] grid_4.3.3             vctrs_0.6.5            RANN_2.6.2            \n[112] promises_1.3.2         ggpubr_0.6.0           car_3.1-3             \n[115] xtable_1.8-4           cluster_2.1.8          readr_2.1.5           \n[118] cli_3.6.4              compiler_4.3.3         future.apply_1.11.3   \n[121] ggsignif_0.6.4         labeling_0.4.3         mclust_6.1.1          \n[124] plyr_1.8.9             stringi_1.8.4          deldir_2.0-4          \n[127] viridisLite_0.4.2      munsell_0.5.1          lazyeval_0.2.2        \n[130] spatstat.geom_3.3-5    Matrix_1.6-5           RcppHNSW_0.6.0        \n[133] hms_1.1.3              future_1.34.0          fpc_2.2-13            \n[136] shiny_1.10.0           kernlab_0.9-33         ROCR_1.0-11           \n[139] broom_1.0.7            memoise_2.0.1          phangorn_2.12.1       \n[142] fastmatch_1.1-6        DEoptimR_1.1-3-1       readxl_1.4.3          \n[145] ape_5.8-1             \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 1679.73\nuser 2269.05\nsys 21.01\npercent 136%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"SeuratClusterStats",desc:"Statistics of the clustering.",npages:1,report_toc:!0,order:17,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] grid      stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] clustree_0.5.1        ggraph_2.2.1          ComplexHeatmap_2.18.0\n [4] circlize_0.4.16       tidyseurat_0.8.0      ttservice_0.4.1      \n [7] ggrepel_0.9.6         ggprism_1.0.5         tibble_3.2.1         \n[10] Seurat_5.1.0          SeuratObject_5.0.2    sp_2.2-0             \n[13] ggplot2_3.5.1         tidyr_1.3.1           dplyr_1.1.4          \n[16] tidyselect_1.2.1      jsonlite_1.8.9        rlang_1.1.5          \n[19] logger_0.4.0         \n\nloaded via a namespace (and not attached):\n  [1] RColorBrewer_1.1-3     shape_1.4.6.1          magrittr_2.0.3        \n  [4] ggbeeswarm_0.7.2       spatstat.utils_3.1-2   farver_2.1.2          \n  [7] GlobalOptions_0.1.2    vctrs_0.6.5            ROCR_1.0-11           \n [10] memoise_2.0.1          spatstat.explore_3.3-4 htmltools_0.5.8.1     \n [13] sctransform_0.4.1      parallelly_1.42.0      KernSmooth_2.23-26    \n [16] htmlwidgets_1.6.4      ica_1.0-3              plyr_1.8.9            \n [19] cachem_1.1.0           plotly_4.10.4          zoo_1.8-12            \n [22] igraph_2.0.3           mime_0.12              lifecycle_1.0.4       \n [25] iterators_1.0.14       pkgconfig_2.0.3        Matrix_1.6-5          \n [28] R6_2.6.0               fastmap_1.2.0          clue_0.3-66           \n [31] fitdistrplus_1.2-2     future_1.34.0          shiny_1.10.0          \n [34] digest_0.6.37          colorspace_2.1-1       patchwork_1.3.0       \n [37] S4Vectors_0.40.2       tensor_1.5             RSpectra_0.16-2       \n [40] irlba_2.3.5.1          labeling_0.4.3         progressr_0.15.1      \n [43] fansi_1.0.6            spatstat.sparse_3.1-0  httr_1.4.7            \n [46] polyclip_1.10-7        abind_1.4-5            compiler_4.3.3        \n [49] withr_3.0.2            doParallel_1.0.17      viridis_0.6.5         \n [52] fastDummies_1.7.5      ggforce_0.4.2          MASS_7.3-60.0.1       \n [55] rjson_0.2.23           tools_4.3.3            vipor_0.4.7           \n [58] lmtest_0.9-40          beeswarm_0.4.0         httpuv_1.6.15         \n [61] future.apply_1.11.3    goftest_1.2-3          glue_1.8.0            \n [64] nlme_3.1-167           promises_1.3.2         Rtsne_0.17            \n [67] cluster_2.1.8          reshape2_1.4.4         generics_0.1.3        \n [70] gtable_0.3.6           spatstat.data_3.1-4    data.table_1.16.4     \n [73] tidygraph_1.3.0        BiocGenerics_0.48.1    spatstat.geom_3.3-5   \n [76] RcppAnnoy_0.0.22       RANN_2.6.2             foreach_1.5.2         \n [79] pillar_1.10.1          stringr_1.5.1          spam_2.11-1           \n [82] RcppHNSW_0.6.0         later_1.4.1            splines_4.3.3         \n [85] tweenr_2.0.3           lattice_0.22-6         survival_3.8-3        \n [88] deldir_2.0-4           miniUI_0.1.1.1         pbapply_1.7-2         \n [91] gridExtra_2.3          IRanges_2.36.0         scattermore_1.2       \n [94] stats4_4.3.3           graphlayouts_1.2.2     matrixStats_1.1.0     \n [97] stringi_1.8.4          lazyeval_0.2.2         codetools_0.2-20      \n[100] cli_3.6.4              uwot_0.2.2             xtable_1.8-4          \n[103] reticulate_1.40.0      munsell_0.5.1          Rcpp_1.0.14           \n[106] globals_0.16.3         spatstat.random_3.3-2  png_0.1-8             \n[109] ggrastr_1.0.2          spatstat.univar_3.1-1  parallel_4.3.3        \n[112] ellipsis_0.3.2         dotCall64_1.2          listenv_0.9.1         \n[115] viridisLite_0.4.2      scales_1.3.0           ggridges_0.5.6        \n[118] leiden_0.4.3.1         purrr_1.0.2            crayon_1.5.3          \n[121] GetoptLong_1.0.5       cowplot_1.1.3         \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 18.76\nuser 27.07\nsys 0.71\npercent 148%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"Immunarch",desc:"Exploration of Single-cell and Bulk T-cell/Antibody Immune Repertoires",npages:3,report_toc:!0,order:18,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] circlize_0.4.16   tibble_3.2.1      glue_1.8.0        ggprism_1.0.5    \n [5] immunarch_0.9.1   patchwork_1.3.0   data.table_1.16.4 dtplyr_1.3.1     \n [9] ggplot2_3.5.1     tidyr_1.3.1       dplyr_1.1.4       jsonlite_1.8.9   \n[13] rlang_1.1.5       logger_0.4.0     \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22       shinythemes_1.2.0      splines_4.3.3         \n  [4] later_1.4.1            cellranger_1.1.0       polyclip_1.10-7       \n  [7] fastDummies_1.7.5      factoextra_1.0.7       lifecycle_1.0.4       \n [10] rstatix_0.7.2          doParallel_1.0.17      globals_0.16.3        \n [13] lattice_0.22-6         prabclus_2.3-4         MASS_7.3-60.0.1       \n [16] backports_1.5.0        magrittr_2.0.3         plotly_4.10.4         \n [19] rlist_0.4.6.2          httpuv_1.6.15          Seurat_5.1.0          \n [22] sctransform_0.4.1      spam_2.11-1            flexmix_2.3-19        \n [25] spatstat.sparse_3.1-0  sp_2.2-0               reticulate_1.40.0     \n [28] cowplot_1.1.3          pbapply_1.7-2          RColorBrewer_1.1-3    \n [31] abind_1.4-5            Rtsne_0.17             quadprog_1.5-8        \n [34] purrr_1.0.2            ggraph_2.2.1           nnet_7.3-20           \n [37] tweenr_2.0.3           ggrepel_0.9.6          irlba_2.3.5.1         \n [40] spatstat.utils_3.1-2   listenv_0.9.1          pheatmap_1.0.12       \n [43] goftest_1.2-3          RSpectra_0.16-2        spatstat.random_3.3-2 \n [46] fitdistrplus_1.2-2     parallelly_1.42.0      leiden_0.4.3.1        \n [49] codetools_0.2-20       ggforce_0.4.2          tidyselect_1.2.1      \n [52] shape_1.4.6.1          ggseqlogo_0.2          farver_2.1.2          \n [55] viridis_0.6.5          matrixStats_1.1.0      stats4_4.3.3          \n [58] spatstat.explore_3.3-4 tidygraph_1.3.0        progressr_0.15.1      \n [61] Formula_1.2-5          ggridges_0.5.6         ggalluvial_0.12.5     \n [64] survival_3.8-3         iterators_1.0.14       foreach_1.5.2         \n [67] tools_4.3.3            stringdist_0.9.15      ica_1.0-3             \n [70] Rcpp_1.0.14            gridExtra_2.3          withr_3.0.2           \n [73] fastmap_1.2.0          digest_0.6.37          R6_2.6.0              \n [76] mime_0.12              colorspace_2.1-1       scattermore_1.2       \n [79] tensor_1.5             spatstat.data_3.1-4    diptest_0.77-1        \n [82] UpSetR_1.4.0           generics_0.1.3         robustbase_0.99-4-1   \n [85] class_7.3-23           graphlayouts_1.2.2     httr_1.4.7            \n [88] htmlwidgets_1.6.4      uwot_0.2.2             pkgconfig_2.0.3       \n [91] gtable_0.3.6           modeltools_0.2-23      lmtest_0.9-40         \n [94] htmltools_0.5.8.1      carData_3.0-5          dotCall64_1.2         \n [97] SeuratObject_5.0.2     scales_1.3.0           png_0.1-8             \n[100] spatstat.univar_3.1-1  tzdb_0.4.0             reshape2_1.4.4        \n[103] uuid_1.2-1             nlme_3.1-167           cachem_1.1.0          \n[106] zoo_1.8-12             GlobalOptions_0.1.2    stringr_1.5.1         \n[109] KernSmooth_2.23-26     parallel_4.3.3         miniUI_0.1.1.1        \n[112] pillar_1.10.1          grid_4.3.3             vctrs_0.6.5           \n[115] RANN_2.6.2             promises_1.3.2         ggpubr_0.6.0          \n[118] car_3.1-3              xtable_1.8-4           cluster_2.1.8         \n[121] readr_2.1.5            cli_3.6.4              compiler_4.3.3        \n[124] future.apply_1.11.3    ggsignif_0.6.4         labeling_0.4.3        \n[127] mclust_6.1.1           plyr_1.8.9             stringi_1.8.4         \n[130] deldir_2.0-4           viridisLite_0.4.2      assertthat_0.2.1      \n[133] munsell_0.5.1          lazyeval_0.2.2         spatstat.geom_3.3-5   \n[136] Matrix_1.6-5           RcppHNSW_0.6.0         hms_1.1.3             \n[139] future_1.34.0          fpc_2.2-13             shiny_1.10.0          \n[142] kernlab_0.9-33         ROCR_1.0-11            igraph_2.0.3          \n[145] broom_1.0.7            memoise_2.0.1          phangorn_2.12.1       \n[148] fastmatch_1.1-6        DEoptimR_1.1-3-1       readxl_1.4.3          \n[151] ape_5.8-1             \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 102.43\nuser 256.71\nsys 5.18\npercent 255%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"RadarPlots",desc:"Radar plots for cell proportion in different clusters.",npages:1,report_toc:!0,order:19,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ggprism_1.0.5      ggradar_0.2        ggplot2_3.5.1      tibble_3.2.1      \n [5] tidyr_1.3.1        dplyr_1.1.4        Seurat_5.1.0       SeuratObject_5.0.2\n [9] sp_2.2-0           jsonlite_1.8.9     rlang_1.1.5        logger_0.4.0      \n\nloaded via a namespace (and not attached):\n [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n[10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n[13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n[16] labeling_0.4.3         promises_1.3.2         purrr_1.0.2           \n[19] goftest_1.2-3          later_1.4.1            spatstat.utils_3.1-2  \n[22] irlba_2.3.5.1          parallel_4.3.3         cluster_2.1.8         \n[25] R6_2.6.0               ica_1.0-3              stringi_1.8.4         \n[28] RColorBrewer_1.1-3     spatstat.data_3.1-4    reticulate_1.40.0     \n[31] parallelly_1.42.0      spatstat.univar_3.1-1  lmtest_0.9-40         \n[34] scattermore_1.2        Rcpp_1.0.14            tensor_1.5            \n[37] future.apply_1.11.3    zoo_1.8-12             sctransform_0.4.1     \n[40] httpuv_1.6.15          Matrix_1.6-5           splines_4.3.3         \n[43] igraph_2.0.3           tidyselect_1.2.1       abind_1.4-5           \n[46] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n[49] spatstat.explore_3.3-4 listenv_0.9.1          lattice_0.22-6        \n[52] plyr_1.8.9             withr_3.0.2            shiny_1.10.0          \n[55] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n[58] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n[61] fitdistrplus_1.2-2     pillar_1.10.1          KernSmooth_2.23-26    \n[64] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n[67] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n[70] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n[73] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n[76] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n[79] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n[82] nlme_3.1-167           patchwork_1.3.0        cli_3.6.4             \n[85] spatstat.sparse_3.1-0  spam_2.11-1            viridisLite_0.4.2     \n[88] uwot_0.2.2             gtable_0.3.6           digest_0.6.37         \n[91] progressr_0.15.1       ggrepel_0.9.6          htmlwidgets_1.6.4     \n[94] farver_2.1.2           htmltools_0.5.8.1      lifecycle_1.0.4       \n[97] httr_1.4.7             mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 8.07\nuser 16.74\nsys 0.38\npercent 212%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"CloneResidency",desc:"Identification of clone residency",npages:1,report_toc:!0,order:20,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ComplexUpset_1.3.3  ggVennDiagram_1.2.3 ggprism_1.0.5      \n [4] ggplot2_3.5.1       ggnewscale_0.5.0    glue_1.8.0         \n [7] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[10] jsonlite_1.8.9      rlang_1.1.5         logger_0.4.0       \n\nloaded via a namespace (and not attached):\n  [1] DBI_1.2.3              deldir_2.0-4           pbapply_1.7-2         \n  [4] gridExtra_2.3          magrittr_2.0.3         RcppAnnoy_0.0.22      \n  [7] e1071_1.7-16           spatstat.geom_3.3-5    matrixStats_1.1.0     \n [10] ggridges_0.5.6         compiler_4.3.3         png_0.1-8             \n [13] vctrs_0.6.5            reshape2_1.4.4         stringr_1.5.1         \n [16] pkgconfig_2.0.3        fastmap_1.2.0          labeling_0.4.3        \n [19] promises_1.3.2         purrr_1.0.2            RVenn_1.1.0           \n [22] goftest_1.2-3          later_1.4.1            spatstat.utils_3.1-2  \n [25] irlba_2.3.5.1          parallel_4.3.3         cluster_2.1.8         \n [28] R6_2.6.0               ica_1.0-3              stringi_1.8.4         \n [31] RColorBrewer_1.1-3     spatstat.data_3.1-4    reticulate_1.40.0     \n [34] parallelly_1.42.0      spatstat.univar_3.1-1  lmtest_0.9-40         \n [37] scattermore_1.2        Rcpp_1.0.14            tensor_1.5            \n [40] future.apply_1.11.3    zoo_1.8-12             sctransform_0.4.1     \n [43] httpuv_1.6.15          Matrix_1.6-5           splines_4.3.3         \n [46] igraph_2.0.3           tidyselect_1.2.1       abind_1.4-5           \n [49] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.3-4 listenv_0.9.1          lattice_0.22-6        \n [55] plyr_1.8.9             shiny_1.10.0           withr_3.0.2           \n [58] ROCR_1.0-11            Rtsne_0.17             sf_1.0-19             \n [61] future_1.34.0          fastDummies_1.7.5      survival_3.8-3        \n [64] units_0.8-5            proxy_0.4-27           polyclip_1.10-7       \n [67] fitdistrplus_1.2-2     pillar_1.10.1          Seurat_5.1.0          \n [70] KernSmooth_2.23-26     plotly_4.10.4          generics_0.1.3        \n [73] RcppHNSW_0.6.0         sp_2.2-0               munsell_0.5.1         \n [76] scales_1.3.0           globals_0.16.3         xtable_1.8-4          \n [79] class_7.3-23           lazyeval_0.2.2         tools_4.3.3           \n [82] data.table_1.16.4      RSpectra_0.16-2        RANN_2.6.2            \n [85] leiden_0.4.3.1         dotCall64_1.2          cowplot_1.1.3         \n [88] grid_4.3.3             colorspace_2.1-1       nlme_3.1-167          \n [91] patchwork_1.3.0        cli_3.6.4              spatstat.sparse_3.1-0 \n [94] spam_2.11-1            viridisLite_0.4.2      uwot_0.2.2            \n [97] gtable_0.3.6           digest_0.6.37          classInt_0.4-11       \n[100] progressr_0.15.1       ggrepel_0.9.6          htmlwidgets_1.6.4     \n[103] SeuratObject_5.0.2     farver_2.1.2           htmltools_0.5.8.1     \n[106] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[109] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 19.89\nuser 28.24\nsys 0.66\npercent 145%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ScFGSEA",desc:"Gene set enrichment analysis for cells in different groups using `fgsea`",npages:2,report_toc:!0,order:21,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] fgsea_1.28.0       data.table_1.16.4  matrixStats_1.1.0  tidyseurat_0.8.0  \n [5] ttservice_0.4.1    Seurat_5.1.0       SeuratObject_5.0.2 sp_2.2-0          \n [9] tidyr_1.3.1        tidyselect_1.2.1   tibble_3.2.1       dplyr_1.1.4       \n[13] ggplot2_3.5.1      jsonlite_1.8.9     rlang_1.1.5        logger_0.4.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       ggridges_0.5.6        \n  [7] compiler_4.3.3         spatstat.geom_3.3-5    png_0.1-8             \n [10] vctrs_0.6.5            reshape2_1.4.4         stringr_1.5.1         \n [13] pkgconfig_2.0.3        fastmap_1.2.0          ellipsis_0.3.2        \n [16] labeling_0.4.3         promises_1.3.2         purrr_1.0.2           \n [19] goftest_1.2-3          later_1.4.1            BiocParallel_1.36.0   \n [22] spatstat.utils_3.1-2   irlba_2.3.5.1          parallel_4.3.3        \n [25] cluster_2.1.8          R6_2.6.0               ica_1.0-3             \n [28] stringi_1.8.4          RColorBrewer_1.1-3     spatstat.data_3.1-4   \n [31] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [37] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [43] splines_4.3.3          igraph_2.0.3           abind_1.4-5           \n [46] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n [49] spatstat.explore_3.3-4 listenv_0.9.1          lattice_0.22-6        \n [52] plyr_1.8.9             shiny_1.10.0           withr_3.0.2           \n [55] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n [58] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n [61] fitdistrplus_1.2-2     pillar_1.10.1          KernSmooth_2.23-26    \n [64] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n [67] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [70] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [73] tools_4.3.3            RSpectra_0.16-2        RANN_2.6.2            \n [76] leiden_0.4.3.1         dotCall64_1.2          fastmatch_1.1-6       \n [79] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n [82] nlme_3.1-167           patchwork_1.3.0        cli_3.6.4             \n [85] spatstat.sparse_3.1-0  fansi_1.0.6            spam_2.11-1           \n [88] viridisLite_0.4.2      uwot_0.2.2             gtable_0.3.6          \n [91] digest_0.6.37          progressr_0.15.1       ggrepel_0.9.6         \n [94] htmlwidgets_1.6.4      farver_2.1.2           htmltools_0.5.8.1     \n [97] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[100] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 128.45\nuser 135.35\nsys 1.52\npercent 106%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"MarkersFinder",desc:"Find markers between different groups of cells",npages:1,report_toc:!0,order:22},{name:"MetaMarkers",desc:"Find markers between three or more groups of cells, using one-way ANOVA or Kruskal-Wallis test.",npages:1,report_toc:!0,order:23},{name:"CellsDistribution",desc:"Distribution of cells (i.e. in a TCR clone) from different groups for each cluster",npages:1,report_toc:!0,order:24},{name:"CDR3AAPhyschem",desc:"CDR3 AA physicochemical feature analysis",npages:1,report_toc:!0,order:25},{name:"ScrnaMetabolicLandscape",desc:"Metabolic landscape analysis for scRNA-seq data",order:28,procs:[{name:"MetabolicPathwayActivity",desc:"This process calculates the pathway activities in different groups and subsets.",npages:1,report_toc:!0,order:28},{name:"MetabolicFeatures",desc:"This process performs enrichment analysis for the metabolic pathways for each group in each subset.",npages:1,report_toc:!0,order:29},{name:"MetabolicFeaturesIntraSubset",desc:"Intra-subset metabolic features - Enrichment analysis in details",npages:1,report_toc:!0,order:30},{name:"MetabolicPathwayHeterogeneity",desc:"Calculate Metabolic Pathway heterogeneity.",npages:1,report_toc:!0,order:31}]}]};function Ks(e){let t,n;return t=new an({props:{dir:"up"}}),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Us(e){let t,n;return t=new an({props:{dir:"down"}}),{c(){u(t.$$.fragment)},m(e,a){A(t,e,a),n=!0},i(e){n||(o(t.$$.fragment,e),n=!0)},o(e){_(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Vs(e){let t,n,a,s,r,l=e[1]>0&&Ks();n=new Bs({});let g=e[1]<e[3].npages-1&&Us();return{c(){l&&l.c(),t=S(),u(n.$$.fragment),a=S(),g&&g.c(),s=f()},m(e,o){l&&l.m(e,o),i(e,t,o),A(n,e,o),i(e,a,o),g&&g.m(e,o),i(e,s,o),r=!0},p(e,n){e[1]>0?l?2&n&&o(l,1):(l=Ks(),l.c(),o(l,1),l.m(t.parentNode,t)):l&&(p(),_(l,1,1,(()=>{l=null})),c()),e[1]<e[3].npages-1?g?10&n&&o(g,1):(g=Us(),g.c(),o(g,1),g.m(s.parentNode,s)):g&&(p(),_(g,1,1,(()=>{g=null})),c())},i(e){r||(o(l),o(n.$$.fragment,e),o(g),r=!0)},o(e){_(l),_(n.$$.fragment,e),_(g),r=!1},d(e){e&&(d(t),d(a),d(s)),l&&l.d(e),$(n,e),g&&g.d(e)}}}function zs(e){let t,n,a;function s(t){e[4](t)}let r={logo:e[0],logotext:e[3].desc,versions:Ds.versions,entries:Ds.entries,pipeline_name:Ds.pipeline.name,report_toc:e[3].report_toc,runinfo:e[3].runinfo,toc:Os,page:e[1],$$slots:{default:[Vs]},$$scope:{ctx:e}};return void 0!==e[2]&&(r.isSideNavOpen=e[2]),t=new Qt({props:r}),E.push((()=>L(t,"isSideNavOpen",s))),{c(){u(t.$$.fragment)},m(e,n){A(t,e,n),a=!0},p(e,[a]){const s={};1&a&&(s.logo=e[0]),8&a&&(s.logotext=e[3].desc),8&a&&(s.report_toc=e[3].report_toc),8&a&&(s.runinfo=e[3].runinfo),2&a&&(s.page=e[1]),42&a&&(s.$$scope={dirty:a,ctx:e}),!n&&4&a&&(n=!0,s.isSideNavOpen=e[2],R((()=>n=!1))),t.$set(s)},i(e){a||(o(t.$$.fragment,e),a=!0)},o(e){_(t.$$.fragment,e),a=!1},d(e){$(t,e)}}}function ks(e,t,n){let a,{name:s}=t,{page:r=0}=t,l=!1;for(const e of Ds.entries)if(e.procs){for(const t of e.procs)if(t.name==s){a=t;break}}else if(e.name==s){a=e;break}return m((()=>{const e=new URL(window.location.href);let t;e.hash.length>1&&(t=document.getElementById(`${e.hash.slice(1)}`)),t&&t.scrollIntoView({block:"start",behavior:"smooth"})})),e.$$set=e=>{"name"in e&&n(0,s=e.name),"page"in e&&n(1,r=e.page)},[s,r,l,a,function(e){l=e,n(2,l)}]}e("default",new class extends t{constructor(e){super(),n(this,e,ks,zs,a,{name:0,page:1})}}({target:document.body,props:{name:"ScFGSEA",page:0}}))}}}));
