System.register(["./vendor/svelte-C6SaTv4n.js","./vendor/carbon-components-svelte-APCRN8D9.js","./vendor/carbon-icons-svelte-BI8VTlA4.js"],(function(e){"use strict";var t,n,a,s,r,i,o,l,c,p,d,_,m,u,f,g,b,$,h,v,x,C,y,S,M,w,R,L,U,T,A,N,P,E,I,V,B,k,z,q,G,O,F,D,H,j,W,K,X,Y,J,Q,Z,ee,te,ne,ae,se,re,ie,oe,le,ce,pe,de,_e,me,ue,fe,ge,be,$e,he,ve,xe,Ce,ye,Se,Me,we,Re,Le,Ue,Te,Ae,Ne,Pe;return{setters:[function(e){t=e.S,n=e.i,a=e.s,s=e.a1,r=e.l,i=e.O,o=e.a,l=e.b,c=e.g,p=e.c,d=e.t,_=e.d,m=e.a5,u=e.B,f=e.e,g=e.G,b=e.H,$=e.I,h=e.y,v=e.K,x=e.T,C=e.U,y=e.L,S=e.D,M=e.M,w=e.z,R=e.Q,L=e.k,U=e.u,T=e.n,A=e.o,N=e.h,P=e.p,E=e.a0,I=e.f,V=e.j,B=e.A,k=e.m,z=e.q,q=e.a9,G=e.aa,O=e.x,F=e.W,D=e.ab,H=e.R,j=e.r},function(e){W=e.S,K=e.a,X=e.b,Y=e.H,J=e.c,Q=e.T,Z=e.d,ee=e.e,te=e.f,ne=e.g,ae=e.h,se=e.M,re=e.I,ie=e.i,oe=e.j,le=e.k,ce=e.C,pe=e.l,de=e.m,_e=e.G,me=e.R,ue=e.n,fe=e.B,ge=e.o,be=e.A},function(e){$e=e.C,he=e.a,ve=e.D,xe=e.A,Ce=e.S,ye=e.I,Se=e.b,Me=e.c,we=e.V,Re=e.d,Le=e.T,Ue=e.G,Te=e.J,Ae=e.P,Ne=e.e,Pe=e.f}],execute:function(){function Ee(e,t,n){const a=e.slice();return a[4]=t[n],a}function Ie(e,t,n){const a=e.slice();return a[7]=t[n],a}function Ve(e){let t,n;return t=new W({props:{href:`${e[1](e[4].page)}#${e[4].slug}`,class:e[2](e[4].page),title:e[4].text,text:e[4].text,expanded:!0,$$slots:{default:[ze]},$$scope:{ctx:e}}}),{c(){g(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.href=`${e[1](e[4].page)}#${e[4].slug}`),1&n&&(a.class=e[2](e[4].page)),1&n&&(a.title=e[4].text),1&n&&(a.text=e[4].text),1025&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Be(e){let t,n;return t=new K({props:{href:`${e[1](e[4].page)}#${e[4].slug}`,class:e[2](e[4].page),title:e[4].text,text:e[4].text}}),{c(){g(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.href=`${e[1](e[4].page)}#${e[4].slug}`),1&n&&(a.class=e[2](e[4].page)),1&n&&(a.title=e[4].text),1&n&&(a.text=e[4].text),t.$set(a)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function ke(e){let t,n;return t=new X({props:{href:`${e[1](e[7].page)}#${e[7].slug}`,class:e[2](e[7].page),title:e[7].text,text:e[7].text}}),{c(){g(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.href=`${e[1](e[7].page)}#${e[7].slug}`),1&n&&(a.class=e[2](e[7].page)),1&n&&(a.title=e[7].text),1&n&&(a.text=e[7].text),t.$set(a)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function ze(e){let t,n,a=s(e[4].children),r=[];for(let t=0;t<a.length;t+=1)r[t]=ke(Ie(e,a,t));const i=e=>d(r[e],1,1,(()=>{r[e]=null}));return{c(){for(let e=0;e<r.length;e+=1)r[e].c();t=h()},m(e,a){for(let t=0;t<r.length;t+=1)r[t]&&r[t].m(e,a);o(e,t,a),n=!0},p(e,n){if(7&n){let o;for(a=s(e[4].children),o=0;o<a.length;o+=1){const s=Ie(e,a,o);r[o]?(r[o].p(s,n),l(r[o],1)):(r[o]=ke(s),r[o].c(),l(r[o],1),r[o].m(t.parentNode,t))}for(c(),o=a.length;o<r.length;o+=1)i(o);p()}},i(e){if(!n){for(let e=0;e<a.length;e+=1)l(r[e]);n=!0}},o(e){r=r.filter(Boolean);for(let e=0;e<r.length;e+=1)d(r[e]);n=!1},d(e){e&&_(t),m(r,e)}}}function qe(e){let t,n,a,s;const r=[Be,Ve],i=[];function m(e,t){return 0==e[4].children.length?0:1}return t=m(e),n=i[t]=r[t](e),{c(){n.c(),a=f()},m(e,n){i[t].m(e,n),o(e,a,n),s=!0},p(e,s){let o=t;t=m(e),t===o?i[t].p(e,s):(c(),d(i[o],1,1,(()=>{i[o]=null})),p(),n=i[t],n?n.p(e,s):(n=i[t]=r[t](e),n.c()),l(n,1),n.m(a.parentNode,a))},i(e){s||(l(n),s=!0)},o(e){d(n),s=!1},d(e){e&&_(a),i[t].d(e)}}}function Ge(e){let t,n,a=s(e[0]),u=[];for(let t=0;t<a.length;t+=1)u[t]=qe(Ee(e,a,t));const f=e=>d(u[e],1,1,(()=>{u[e]=null}));return{c(){t=r("ul");for(let e=0;e<u.length;e+=1)u[e].c();i(t,"class","bx--side-nav__items pipen-report-sidenav svelte-12gpq3")},m(e,a){o(e,t,a);for(let e=0;e<u.length;e+=1)u[e]&&u[e].m(t,null);n=!0},p(e,[n]){if(7&n){let r;for(a=s(e[0]),r=0;r<a.length;r+=1){const s=Ee(e,a,r);u[r]?(u[r].p(s,n),l(u[r],1)):(u[r]=qe(s),u[r].c(),l(u[r],1),u[r].m(t,null))}for(c(),r=a.length;r<u.length;r+=1)f(r);p()}},i(e){if(!n){for(let e=0;e<a.length;e+=1)l(u[e]);n=!0}},o(e){u=u.filter(Boolean);for(let e=0;e<u.length;e+=1)d(u[e]);n=!1},d(e){e&&_(t),m(u,e)}}}function Oe(e,t,n){let{toc:a}=t,{page:s}=t;return u((()=>{let e,t=new URL(window.location.href);e=t.hash.length>0?document.querySelector(`[href="${t.hash}"]`):document.querySelector(".toc-current-page"),e.scrollIntoView({behavior:"smooth"})})),e.$$set=e=>{"toc"in e&&n(0,a=e.toc),"page"in e&&n(3,s=e.page)},[a,e=>{if(s===e)return"";let t=new URL(window.location.href);return t.hash="",t.searchParams.set("page",e),t.toString()},e=>e!=s?"toc-noncurrent-page":"toc-current-page",s]}class Fe extends t{constructor(e){super(),n(this,e,Oe,Ge,a,{toc:0,page:3})}}function De(e,t,n){const a=e.slice();return a[7]=t[n],a}function He(e,t,n){const a=e.slice();return a[10]=t[n],a}function je(e){let t,n;return t=new ee({props:{isOpen:dt,transition:e[5],$$slots:{default:[st]},$$scope:{ctx:e}}}),{c(){g(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8197&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function We(e){let t;return{c(){t=y("Go to home:")},m(e,n){o(e,t,n)},d(e){e&&_(t)}}}function Ke(e){let t;return{c(){t=y(e[0])},m(e,n){o(e,t,n)},p(e,n){1&n&&M(t,e[0])},d(e){e&&_(t)}}}function Xe(e){let t;return{c(){t=y("Switch to process:")},m(e,n){o(e,t,n)},d(e){e&&_(t)}}}function Ye(e){let t,n;return t=new ae({props:{title:e[7].name,href:`?proc=${e[7].name}`,$$slots:{default:[Qe]},$$scope:{ctx:e}}}),{c(){g(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};4&n&&(a.title=e[7].name),4&n&&(a.href=`?proc=${e[7].name}`),8196&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Je(e){let t,n,a,r;t=new ae({props:{class:"procgroup",$$slots:{default:[Ze]},$$scope:{ctx:e}}});let i=s(e[7].procs),u=[];for(let t=0;t<i.length;t+=1)u[t]=tt(He(e,i,t));const v=e=>d(u[e],1,1,(()=>{u[e]=null}));return{c(){g(t.$$.fragment),n=h();for(let e=0;e<u.length;e+=1)u[e].c();a=f()},m(e,s){b(t,e,s),o(e,n,s);for(let t=0;t<u.length;t+=1)u[t]&&u[t].m(e,s);o(e,a,s),r=!0},p(e,n){const r={};if(8196&n&&(r.$$scope={dirty:n,ctx:e}),t.$set(r),4&n){let t;for(i=s(e[7].procs),t=0;t<i.length;t+=1){const s=He(e,i,t);u[t]?(u[t].p(s,n),l(u[t],1)):(u[t]=tt(s),u[t].c(),l(u[t],1),u[t].m(a.parentNode,a))}for(c(),t=i.length;t<u.length;t+=1)v(t);p()}},i(e){if(!r){l(t.$$.fragment,e);for(let e=0;e<i.length;e+=1)l(u[e]);r=!0}},o(e){d(t.$$.fragment,e),u=u.filter(Boolean);for(let e=0;e<u.length;e+=1)d(u[e]);r=!1},d(e){e&&(_(n),_(a)),$(t,e),m(u,e)}}}function Qe(e){let t,n,a=e[7].name+"";return{c(){t=y(a),n=h()},m(e,a){o(e,t,a),o(e,n,a)},p(e,n){4&n&&a!==(a=e[7].name+"")&&M(t,a)},d(e){e&&(_(t),_(n))}}}function Ze(e){let t,n=e[7].name+"";return{c(){t=y(n)},m(e,n){o(e,t,n)},p(e,a){4&a&&n!==(n=e[7].name+"")&&M(t,n)},d(e){e&&_(t)}}}function et(e){let t,n,a,s=e[10].name+"";return{c(){t=y("- "),n=y(s),a=h()},m(e,s){o(e,t,s),o(e,n,s),o(e,a,s)},p(e,t){4&t&&s!==(s=e[10].name+"")&&M(n,s)},d(e){e&&(_(t),_(n),_(a))}}}function tt(e){let t,n;return t=new ae({props:{title:e[10].name,class:"procgroup-proc",href:`?proc=${e[10].name}`,$$slots:{default:[et]},$$scope:{ctx:e}}}),{c(){g(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};4&n&&(a.title=e[10].name),4&n&&(a.href=`?proc=${e[10].name}`),8196&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function nt(e){let t,n,a,s;const r=[Je,Ye],i=[];function m(e,t){return e[7].procs?0:1}return t=m(e),n=i[t]=r[t](e),{c(){n.c(),a=f()},m(e,n){i[t].m(e,n),o(e,a,n),s=!0},p(e,s){let o=t;t=m(e),t===o?i[t].p(e,s):(c(),d(i[o],1,1,(()=>{i[o]=null})),p(),n=i[t],n?n.p(e,s):(n=i[t]=r[t](e),n.c()),l(n,1),n.m(a.parentNode,a))},i(e){s||(l(n),s=!0)},o(e){d(n),s=!1},d(e){e&&_(a),i[t].d(e)}}}function at(e){let t,n,a,r,i,u,v,x;t=new ne({props:{$$slots:{default:[We]},$$scope:{ctx:e}}}),a=new ae({props:{href:"index.html",$$slots:{default:[Ke]},$$scope:{ctx:e}}}),i=new ne({props:{$$slots:{default:[Xe]},$$scope:{ctx:e}}});let C=s(e[2]),y=[];for(let t=0;t<C.length;t+=1)y[t]=nt(De(e,C,t));const S=e=>d(y[e],1,1,(()=>{y[e]=null}));return{c(){g(t.$$.fragment),n=h(),g(a.$$.fragment),r=h(),g(i.$$.fragment),u=h();for(let e=0;e<y.length;e+=1)y[e].c();v=f()},m(e,s){b(t,e,s),o(e,n,s),b(a,e,s),o(e,r,s),b(i,e,s),o(e,u,s);for(let t=0;t<y.length;t+=1)y[t]&&y[t].m(e,s);o(e,v,s),x=!0},p(e,n){const r={};8192&n&&(r.$$scope={dirty:n,ctx:e}),t.$set(r);const o={};8193&n&&(o.$$scope={dirty:n,ctx:e}),a.$set(o);const d={};if(8192&n&&(d.$$scope={dirty:n,ctx:e}),i.$set(d),4&n){let t;for(C=s(e[2]),t=0;t<C.length;t+=1){const a=De(e,C,t);y[t]?(y[t].p(a,n),l(y[t],1)):(y[t]=nt(a),y[t].c(),l(y[t],1),y[t].m(v.parentNode,v))}for(c(),t=C.length;t<y.length;t+=1)S(t);p()}},i(e){if(!x){l(t.$$.fragment,e),l(a.$$.fragment,e),l(i.$$.fragment,e);for(let e=0;e<C.length;e+=1)l(y[e]);x=!0}},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),d(i.$$.fragment,e),y=y.filter(Boolean);for(let e=0;e<y.length;e+=1)d(y[e]);x=!1},d(e){e&&(_(n),_(r),_(u),_(v)),$(t,e),$(a,e),$(i,e),m(y,e)}}}function st(e){let t,n;return t=new te({props:{$$slots:{default:[at]},$$scope:{ctx:e}}}),{c(){g(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8197&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function rt(e){let t,n,a,s;t=new Z({props:{render:"select",theme:"g10",persist:!0,persistKey:"__pipen-report-theme",select:{themes:["g10","white","g80","g90","g100"],labelText:"Theme",inline:!0}}});let r=e[2].length>0&&je(e);return{c(){g(t.$$.fragment),n=h(),r&&r.c(),a=f()},m(e,i){b(t,e,i),o(e,n,i),r&&r.m(e,i),o(e,a,i),s=!0},p(e,t){e[2].length>0?r?(r.p(e,t),4&t&&l(r,1)):(r=je(e),r.c(),l(r,1),r.m(a.parentNode,a)):r&&(c(),d(r,1,1,(()=>{r=null})),p())},i(e){s||(l(t.$$.fragment,e),l(r),s=!0)},o(e){d(t.$$.fragment,e),d(r),s=!1},d(e){e&&(_(n),_(a)),$(t,e),r&&r.d(e)}}}function it(e){let t,n;return t=new J({props:{class:"header-util",$$slots:{default:[rt]},$$scope:{ctx:e}}}),{c(){g(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8197&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function ot(e){let t;return{c(){t=y(e[4])},m(e,n){o(e,t,n)},p(e,n){16&n&&M(t,e[4])},d(e){e&&_(t)}}}function lt(e){let t,n,a,s,i=null!==e[4]&&ot(e);return{c(){t=r("h1"),n=y(e[3]),a=h(),i&&i.c(),s=f()},m(e,r){o(e,t,r),S(t,n),o(e,a,r),i&&i.m(e,r),o(e,s,r)},p(e,t){8&t&&M(n,e[3]),null!==e[4]?i?i.p(e,t):(i=ot(e),i.c(),i.m(s.parentNode,s)):i&&(i.d(1),i=null)},d(e){e&&(_(t),_(a),_(s)),i&&i.d(e)}}}function ct(e){let t,n;return t=new Q({props:{class:"header-tile",slot:"platform",$$slots:{default:[lt]},$$scope:{ctx:e}}}),{c(){g(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8216&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function pt(e){let t,n,a;function s(t){e[6](t)}let r={$$slots:{platform:[ct],default:[it]},$$scope:{ctx:e}};return void 0!==e[1]&&(r.isSideNavOpen=e[1]),t=new Y({props:r}),v.push((()=>x(t,"isSideNavOpen",s))),{c(){g(t.$$.fragment)},m(e,n){b(t,e,n),a=!0},p(e,[a]){const s={};8221&a&&(s.$$scope={dirty:a,ctx:e}),!n&&2&a&&(n=!0,s.isSideNavOpen=e[1],C((()=>n=!1))),t.$set(s)},i(e){a||(l(t.$$.fragment,e),a=!0)},o(e){d(t.$$.fragment,e),a=!1},d(e){$(t,e)}}}let dt=!1;function _t(e,t,n){let{entries:a}=t,{logo:s}=t,{logotext:r=null}=t,{pipeline_name:i}=t,{isSideNavOpen:o}=t;return i=i||s,e.$$set=e=>{"entries"in e&&n(2,a=e.entries),"logo"in e&&n(3,s=e.logo),"logotext"in e&&n(4,r=e.logotext),"pipeline_name"in e&&n(0,i=e.pipeline_name),"isSideNavOpen"in e&&n(1,o=e.isSideNavOpen)},[i,o,a,s,r,{duration:200},function(e){o=e,n(1,o)}]}class mt extends t{constructor(e){super(),n(this,e,_t,pt,a,{entries:2,logo:3,logotext:4,pipeline_name:0,isSideNavOpen:1})}}function ut(e){let t,n,a;function s(t){e[3](t)}let r={modalHeading:"Running Information",passiveModal:!0,hasScrollingContent:!0,class:"runinfo",$$slots:{default:[vt]},$$scope:{ctx:e}};return void 0!==e[2]&&(r.open=e[2]),t=new se({props:r}),v.push((()=>x(t,"open",s))),{c(){g(t.$$.fragment)},m(e,n){b(t,e,n),a=!0},p(e,a){const s={};34&a&&(s.$$scope={dirty:a,ctx:e}),!n&&4&a&&(n=!0,s.open=e[2],C((()=>n=!1))),t.$set(s)},i(e){a||(l(t.$$.fragment,e),a=!0)},o(e){d(t.$$.fragment,e),a=!1},d(e){$(t,e)}}}function ft(e){let t,n,a,s,r,i;return t=new oe({props:{label:"Session Information"}}),a=new oe({props:{label:"Time Spent"}}),r=new oe({props:{label:"Device Information"}}),{c(){g(t.$$.fragment),n=h(),g(a.$$.fragment),s=h(),g(r.$$.fragment)},m(e,l){b(t,e,l),o(e,n,l),b(a,e,l),o(e,s,l),b(r,e,l),i=!0},p:R,i(e){i||(l(t.$$.fragment,e),l(a.$$.fragment,e),l(r.$$.fragment,e),i=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),d(r.$$.fragment,e),i=!1},d(e){e&&(_(n),_(s)),$(t,e),$(a,e),$(r,e)}}}function gt(e){let t,n;return t=new ce({props:{type:"multi",code:e[1].session}}),{c(){g(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};2&n&&(a.code=e[1].session),t.$set(a)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function bt(e){let t,n;return t=new ce({props:{type:"multi",code:e[1].time}}),{c(){g(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};2&n&&(a.code=e[1].time),t.$set(a)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function $t(e){let t,n;return t=new ce({props:{type:"multi",code:e[1].device}}),{c(){g(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};2&n&&(a.code=e[1].device),t.$set(a)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function ht(e){let t,n,a,s,r,i;return t=new le({props:{$$slots:{default:[gt]},$$scope:{ctx:e}}}),a=new le({props:{$$slots:{default:[bt]},$$scope:{ctx:e}}}),r=new le({props:{$$slots:{default:[$t]},$$scope:{ctx:e}}}),{c(){g(t.$$.fragment),n=h(),g(a.$$.fragment),s=h(),g(r.$$.fragment)},m(e,l){b(t,e,l),o(e,n,l),b(a,e,l),o(e,s,l),b(r,e,l),i=!0},p(e,n){const s={};34&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const i={};34&n&&(i.$$scope={dirty:n,ctx:e}),a.$set(i);const o={};34&n&&(o.$$scope={dirty:n,ctx:e}),r.$set(o)},i(e){i||(l(t.$$.fragment,e),l(a.$$.fragment,e),l(r.$$.fragment,e),i=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),d(r.$$.fragment,e),i=!1},d(e){e&&(_(n),_(s)),$(t,e),$(a,e),$(r,e)}}}function vt(e){let t,n,a,s;return t=new re({props:{kind:"warning",lowContrast:!0,hideCloseButton:!0,title:"Showing the first job only. Check the workdir for information of other jobs if any."}}),a=new ie({props:{$$slots:{content:[ht],default:[ft]},$$scope:{ctx:e}}}),{c(){g(t.$$.fragment),n=h(),g(a.$$.fragment)},m(e,r){b(t,e,r),o(e,n,r),b(a,e,r),s=!0},p(e,t){const n={};34&t&&(n.$$scope={dirty:t,ctx:e}),a.$set(n)},i(e){s||(l(t.$$.fragment,e),l(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),$(t,e),$(a,e)}}}function xt(e){let t,n,a;return{c(){t=r("a"),t.textContent="> Running Information",i(t,"href","javascript:void(0);")},m(s,r){o(s,t,r),n||(a=w(t,"click",e[4]),n=!0)},p:R,d(e){e&&_(t),n=!1,a()}}}function Ct(e){let t,n,a,s,m,u,f,g,b=e[1]&&ut(e),$=e[1]&&xt(e);return{c(){b&&b.c(),t=h(),n=r("div"),a=r("div"),$&&$.c(),s=h(),m=r("div"),u=y("Powered by "),f=r("span"),i(a,"class","runinfo"),i(f,"class","pipen-report-footer-versions"),i(m,"class","versions"),i(n,"class","pipen-report-footer bx--grid")},m(r,i){b&&b.m(r,i),o(r,t,i),o(r,n,i),S(n,a),$&&$.m(a,null),S(n,s),S(n,m),S(m,u),S(m,f),f.innerHTML=e[0],g=!0},p(e,[n]){e[1]?b?(b.p(e,n),2&n&&l(b,1)):(b=ut(e),b.c(),l(b,1),b.m(t.parentNode,t)):b&&(c(),d(b,1,1,(()=>{b=null})),p()),e[1]?$?$.p(e,n):($=xt(e),$.c(),$.m(a,null)):$&&($.d(1),$=null),(!g||1&n)&&(f.innerHTML=e[0])},i(e){g||(l(b),g=!0)},o(e){d(b),g=!1},d(e){e&&(_(t),_(n)),b&&b.d(e),$&&$.d()}}}function yt(e,t,n){let{versions:a}=t,{runinfo:s}=t,r=!1;return e.$$set=e=>{"versions"in e&&n(0,a=e.versions),"runinfo"in e&&n(1,s=e.runinfo)},[a,s,r,function(e){r=e,n(2,r)},()=>{n(2,r=!0)}]}class St extends t{constructor(e){super(),n(this,e,yt,Ct,a,{versions:0,runinfo:1})}}function Mt(e){let t,n,a;function s(t){e[12](t)}let r={$$slots:{default:[wt]},$$scope:{ctx:e}};return void 0!==e[0]&&(r.isOpen=e[0]),t=new de({props:r}),v.push((()=>x(t,"isOpen",s))),{c(){g(t.$$.fragment)},m(e,n){b(t,e,n),a=!0},p(e,a){const s={};8576&a&&(s.$$scope={dirty:a,ctx:e}),!n&&1&a&&(n=!0,s.isOpen=e[0],C((()=>n=!1))),t.$set(s)},i(e){a||(l(t.$$.fragment,e),a=!0)},o(e){d(t.$$.fragment,e),a=!1},d(e){$(t,e)}}}function wt(e){let t,n;return t=new Fe({props:{toc:e[7],page:e[8]}}),{c(){g(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};128&n&&(a.toc=e[7]),256&n&&(a.page=e[8]),t.$set(a)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Rt(e){let t;const n=e[10].default,a=L(n,e,e[13],null);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||8192&s)&&U(a,n,e,e[13],t?A(n,e[13],s,null):T(e[13]),null)},i(e){t||(l(a,e),t=!0)},o(e){d(a,e),t=!1},d(e){a&&a.d(e)}}}function Lt(e){let t,n;return t=new ue({props:{$$slots:{default:[Rt]},$$scope:{ctx:e}}}),{c(){g(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8192&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Ut(e){let t,n;return t=new me({props:{$$slots:{default:[Lt]},$$scope:{ctx:e}}}),{c(){g(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};8192&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Tt(e){let t,n,a,s;return t=new _e({props:{$$slots:{default:[Ut]},$$scope:{ctx:e}}}),a=new St({props:{versions:e[3],runinfo:e[9]}}),{c(){g(t.$$.fragment),n=h(),g(a.$$.fragment)},m(e,r){b(t,e,r),o(e,n,r),b(a,e,r),s=!0},p(e,n){const s={};8192&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};8&n&&(r.versions=e[3]),512&n&&(r.runinfo=e[9]),a.$set(r)},i(e){s||(l(t.$$.fragment,e),l(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),$(t,e),$(a,e)}}}function At(e){let t,n,a,s,r,i;function m(t){e[11](t)}let u={logo:e[1],logotext:e[2],entries:e[4],pipeline_name:e[5]};void 0!==e[0]&&(u.isSideNavOpen=e[0]),t=new mt({props:u}),v.push((()=>x(t,"isSideNavOpen",m)));let f=e[6]&&e[7].length>0&&Mt(e);return r=new pe({props:{$$slots:{default:[Tt]},$$scope:{ctx:e}}}),{c(){g(t.$$.fragment),a=h(),f&&f.c(),s=h(),g(r.$$.fragment)},m(e,n){b(t,e,n),o(e,a,n),f&&f.m(e,n),o(e,s,n),b(r,e,n),i=!0},p(e,[a]){const i={};2&a&&(i.logo=e[1]),4&a&&(i.logotext=e[2]),16&a&&(i.entries=e[4]),32&a&&(i.pipeline_name=e[5]),!n&&1&a&&(n=!0,i.isSideNavOpen=e[0],C((()=>n=!1))),t.$set(i),e[6]&&e[7].length>0?f?(f.p(e,a),192&a&&l(f,1)):(f=Mt(e),f.c(),l(f,1),f.m(s.parentNode,s)):f&&(c(),d(f,1,1,(()=>{f=null})),p());const o={};8712&a&&(o.$$scope={dirty:a,ctx:e}),r.$set(o)},i(e){i||(l(t.$$.fragment,e),l(f),l(r.$$.fragment,e),i=!0)},o(e){d(t.$$.fragment,e),d(f),d(r.$$.fragment,e),i=!1},d(e){e&&(_(a),_(s)),$(t,e),f&&f.d(e),$(r,e)}}}function Nt(e,t,n){let a,{$$slots:s={},$$scope:r}=t,{logo:i}=t,{logotext:o=null}=t,{versions:l}=t,{entries:c}=t,{pipeline_name:p}=t,{report_toc:d}=t,{toc:_}=t,{page:m}=t,{isSideNavOpen:f}=t,{runinfo:g}=t;return u((()=>{const e=new URL(window.location.href);let t;t=e.hash.length>0?document.querySelector(`li:has(>[href="${e.hash}"])`):document.querySelector("li:has(>.toc-current-page)"),t&&(t.classList.add("toc-active"),t.scrollIntoView());const n=document.querySelectorAll(".pipen-report-toc-anchor");a=new IntersectionObserver((e=>{e.forEach((e=>{const t=e.target.getAttribute("id"),n=document.querySelector(`li:has(>[href="#${t}"])`);n&&(e.isIntersecting?n.classList.add("toc-active"):n.classList.remove("toc-active"))}))}),{root:null,rootMargin:"0px",threshold:.5}),n.forEach((e=>{a.observe(e)}))})),e.$$set=e=>{"logo"in e&&n(1,i=e.logo),"logotext"in e&&n(2,o=e.logotext),"versions"in e&&n(3,l=e.versions),"entries"in e&&n(4,c=e.entries),"pipeline_name"in e&&n(5,p=e.pipeline_name),"report_toc"in e&&n(6,d=e.report_toc),"toc"in e&&n(7,_=e.toc),"page"in e&&n(8,m=e.page),"isSideNavOpen"in e&&n(0,f=e.isSideNavOpen),"runinfo"in e&&n(9,g=e.runinfo),"$$scope"in e&&n(13,r=e.$$scope)},[f,i,o,l,c,p,d,_,m,g,s,function(e){f=e,n(0,f)},function(e){f=e,n(0,f)},r]}class Pt extends t{constructor(e){super(),n(this,e,Nt,At,a,{logo:1,logotext:2,versions:3,entries:4,pipeline_name:5,report_toc:6,toc:7,page:8,isSideNavOpen:0,runinfo:9})}}function Et(e){let t;return{c(){t=y(e[2])},m(e,n){o(e,t,n)},p:R,d(e){e&&_(t)}}}function It(e){let t,n,a,s;return n=new fe({props:{icon:e[1],kind:"tertiary",href:e[3](),$$slots:{default:[Et]},$$scope:{ctx:e}}}),{c(){t=r("div"),g(n.$$.fragment),i(t,"class",a=`pagenav ${e[0]}`)},m(e,a){o(e,t,a),b(n,t,null),s=!0},p(e,[r]){const o={};16&r&&(o.$$scope={dirty:r,ctx:e}),n.$set(o),(!s||1&r&&a!==(a=`pagenav ${e[0]}`))&&i(t,"class",a)},i(e){s||(l(n.$$.fragment,e),s=!0)},o(e){d(n.$$.fragment,e),s=!1},d(e){e&&_(t),$(n)}}}function Vt(e,t,n){let{dir:a}=t,s="up"===a?$e:he,r="up"===a?"Go to previous page":"Go to next page";return e.$$set=e=>{"dir"in e&&n(0,a=e.dir)},[a,s,r,()=>{let e=new URL(window.location.href);e.hash="";const t=parseInt(e.searchParams.get("page")||0),n="up"===a?t-1:t+1;return 0===n?e.searchParams.delete("page"):e.searchParams.set("page",n),e.toString()}]}class Bt extends t{constructor(e){super(),n(this,e,Vt,It,a,{dir:0})}}function kt(e){let t,n;const a=[{tooltipText:e[0]},{icon:e[1]},{direction:"bottom"},{align:"start"},e[3]];let s={};for(let e=0;e<a.length;e+=1)s=N(s,a[e]);return t=new ge({props:s}),t.$on("click",e[2]),{c(){g(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,[n]){const s=11&n?P(a,[1&n&&{tooltipText:e[0]},2&n&&{icon:e[1]},a[2],a[3],8&n&&E(e[3])]):{};t.$set(s)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function zt(e,t,n){const a=["src","tip","icon"];let s,r=I(t,a),{src:i=""}=t,{tip:o=""}=t,{icon:l="auto"}=t,c=i.split(".").at(-1).toLocaleLowerCase();if("string"==typeof l)switch("auto"===l&&(l=c),l){case"pdf":case"DocumentPdf":s=Pe;break;case"html":case"htm":case"ChartLineData":s=Ne;break;case"png":case"Png":s=Ae;break;case"jpg":case"jpeg":case"Jpg":s=Te;break;case"gif":case"Gif":s=Ue;break;case"tif":case"tiff":case"Tif":s=Le;break;case"eps":case"DatabaseElastic":s=Re;break;case"mp4":case"avi":case"mov":case"mkv":case"webm":case"VideoPlayer":s=we;break;case"Image":s=Me;break;case"Data":case"DataVolume":s=Se;break;case"Code":case"IbmWatsonxCodeAssistantForZRefactor":s=ye;break;case"svg":case"Svg":s=Ce;break;case"zip":case"tar":case"gz":case"bz2":case"7z":case"rar":case"xz":case"Archive":s=xe;break;default:s=ve}const p={...r,class:[r.class,"pipen-report-image-download"].filter(Boolean).join(" ")};return e.$$set=e=>{t=N(N({},t),V(e)),n(7,r=I(t,a)),"src"in e&&n(5,i=e.src),"tip"in e&&n(0,o=e.tip),"icon"in e&&n(4,l=e.icon)},[o,s,()=>{const e=document.createElement("a");e.href=i,e.download=i.split("/").pop(),e.click()},p,l,i]}class qt extends t{constructor(e){super(),n(this,e,zt,kt,a,{src:5,tip:0,icon:4})}}const{Boolean:Gt}=q;function Ot(e,t,n){const a=e.slice();return a[5]=t[n].src,a[6]=t[n].tip,a[7]=t[n].icon,a[9]=n,a}function Ft(e){let t,n,a,i,u=s(e[0]),f=[];for(let t=0;t<u.length;t+=1)f[t]=Dt(Ot(e,u,t));const g=e=>d(f[e],1,1,(()=>{f[e]=null}));let b=[e[1],{role:"button"},{tabindex:"0"}],$={};for(let e=0;e<b.length;e+=1)$=N($,b[e]);return{c(){t=r("div");for(let e=0;e<f.length;e+=1)f[e].c();k(t,$),z(t,"svelte-14lcnln",!0)},m(s,r){o(s,t,r);for(let e=0;e<f.length;e+=1)f[e]&&f[e].m(t,null);n=!0,a||(i=w(t,"mouseenter",e[3]),a=!0)},p(e,n){if(1&n){let a;for(u=s(e[0]),a=0;a<u.length;a+=1){const s=Ot(e,u,a);f[a]?(f[a].p(s,n),l(f[a],1)):(f[a]=Dt(s),f[a].c(),l(f[a],1),f[a].m(t,null))}for(c(),a=u.length;a<f.length;a+=1)g(a);p()}z(t,"svelte-14lcnln",!0)},i(e){if(!n){for(let e=0;e<u.length;e+=1)l(f[e]);n=!0}},o(e){f=f.filter(Gt);for(let e=0;e<f.length;e+=1)d(f[e]);n=!1},d(e){e&&_(t),m(f,e),a=!1,i()}}}function Dt(e){let t,n;return t=new qt({props:{src:e[5],tip:e[6],icon:e[7]}}),{c(){g(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};1&n&&(a.src=e[5]),1&n&&(a.tip=e[6]),1&n&&(a.icon=e[7]),t.$set(a)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Ht(e){let t,n,a=e[0].length>0&&Ft(e);return{c(){a&&a.c(),t=f()},m(e,s){a&&a.m(e,s),o(e,t,s),n=!0},p(e,[n]){e[0].length>0?a?(a.p(e,n),1&n&&l(a,1)):(a=Ft(e),a.c(),l(a,1),a.m(t.parentNode,t)):a&&(c(),d(a,1,1,(()=>{a=null})),p())},i(e){n||(l(a),n=!0)},o(e){d(a),n=!1},d(e){e&&_(t),a&&a.d(e)}}}function jt(e,t,n){const a=["data"];let s=I(t,a),{data:r=[]}=t;Array.isArray(r)||(r=[r]),r=r.map((e=>"string"==typeof e?{src:e,tip:`Download the ${e.split(".").at(-1)} format.`,icon:"auto"}:(e.tip=e.tip||`Download the ${e.src.split(".").at(-1)} format.`,e.icon=e.icon||"auto",e)));const i={...s,class:[s.class,"pipen-report-image-download-list"].filter(Boolean).join(" ")},o=B();return e.$$set=e=>{t=N(N({},t),V(e)),n(4,s=I(t,a)),"data"in e&&n(0,r=e.data)},[r,i,o,()=>o("mouseenter")]}class Wt extends t{constructor(e){super(),n(this,e,jt,Ht,a,{data:0})}}const Kt=e=>({}),Xt=e=>({}),Yt=e=>({}),Jt=e=>({}),Qt=e=>({}),Zt=e=>({}),en=e=>({}),tn=e=>({});function nn(e){let t,n;return t=new be({props:{ratio:e[6],$$slots:{default:[cn]},$$scope:{ctx:e}}}),{c(){g(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};64&n&&(a.ratio=e[6]),33556159&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function an(e){let t,n,a,s,r=e[2]&&pn(e),i=e[0]&&dn(e),m=e[1]&&mn(e);return{c(){r&&r.c(),t=h(),i&&i.c(),n=h(),m&&m.c(),a=f()},m(e,l){r&&r.m(e,l),o(e,t,l),i&&i.m(e,l),o(e,n,l),m&&m.m(e,l),o(e,a,l),s=!0},p(e,s){e[2]?r?(r.p(e,s),4&s&&l(r,1)):(r=pn(e),r.c(),l(r,1),r.m(t.parentNode,t)):r&&(c(),d(r,1,1,(()=>{r=null})),p()),e[0]?i?(i.p(e,s),1&s&&l(i,1)):(i=dn(e),i.c(),l(i,1),i.m(n.parentNode,n)):i&&(c(),d(i,1,1,(()=>{i=null})),p()),e[1]?m?(m.p(e,s),2&s&&l(m,1)):(m=mn(e),m.c(),l(m,1),m.m(a.parentNode,a)):m&&(c(),d(m,1,1,(()=>{m=null})),p())},i(e){s||(l(r),l(i),l(m),s=!0)},o(e){d(r),d(i),d(m),s=!1},d(e){e&&(_(t),_(n),_(a)),r&&r.d(e),i&&i.d(e),m&&m.d(e)}}}function sn(e){let t;const n=e[15].loading,a=L(n,e,e[25],Jt);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||33554432&s)&&U(a,n,e,e[25],t?A(n,e[25],s,Yt):T(e[25]),Jt)},i(e){t||(l(a,e),t=!0)},o(e){d(a,e),t=!1},d(e){a&&a.d(e)}}}function rn(e){let t,n,a,s,i,m,u,g,b=[e[12],{src:n=e[3]},{alt:e[4]},{title:"Click to zoom in ..."}],$={};for(let e=0;e<b.length;e+=1)$=N($,b[e]);let v=e[9]&&on(e);return{c(){t=r("img"),s=h(),v&&v.c(),i=f(),k(t,$)},m(n,a){o(n,t,a),o(n,s,a),v&&v.m(n,a),o(n,i,a),m=!0,u||(g=[w(t,"click",e[11]),w(t,"mouseenter",e[19]),w(t,"mouseleave",e[20])],u=!0)},p(a,s){k(t,$=P(b,[(e=a)[12],(!m||8&s&&!G(t.src,n=e[3]))&&{src:n},(!m||16&s)&&{alt:e[4]},{title:"Click to zoom in ..."}])),e[9]?v?(v.p(e,s),512&s&&l(v,1)):(v=on(e),v.c(),l(v,1),v.m(i.parentNode,i)):v&&(c(),d(v,1,1,(()=>{v=null})),p())},i(n){m||(n&&O((()=>{m&&(a||(a=F(t,D,{duration:e[7]?gn:0},!0)),a.run(1))})),l(v),m=!0)},o(n){n&&(a||(a=F(t,D,{duration:e[7]?gn:0},!1)),a.run(0)),d(v),m=!1},d(e){e&&(_(t),_(s),_(i)),e&&a&&a.end(),v&&v.d(e),u=!1,H(g)}}}function on(e){let t,n;return t=new Wt({props:{data:e[5]}}),t.$on("mouseenter",e[21]),{c(){g(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};32&n&&(a.data=e[5]),t.$set(a)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function ln(e){let t;const n=e[15].error,a=L(n,e,e[25],Xt);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||33554432&s)&&U(a,n,e,e[25],t?A(n,e[25],s,Kt):T(e[25]),Xt)},i(e){t||(l(a,e),t=!0)},o(e){d(a,e),t=!1},d(e){a&&a.d(e)}}}function cn(e){let t,n,a,s,r=e[2]&&sn(e),i=e[0]&&rn(e),m=e[1]&&ln(e);return{c(){r&&r.c(),t=h(),i&&i.c(),n=h(),m&&m.c(),a=f()},m(e,l){r&&r.m(e,l),o(e,t,l),i&&i.m(e,l),o(e,n,l),m&&m.m(e,l),o(e,a,l),s=!0},p(e,s){e[2]?r?(r.p(e,s),4&s&&l(r,1)):(r=sn(e),r.c(),l(r,1),r.m(t.parentNode,t)):r&&(c(),d(r,1,1,(()=>{r=null})),p()),e[0]?i?(i.p(e,s),1&s&&l(i,1)):(i=rn(e),i.c(),l(i,1),i.m(n.parentNode,n)):i&&(c(),d(i,1,1,(()=>{i=null})),p()),e[1]?m?(m.p(e,s),2&s&&l(m,1)):(m=ln(e),m.c(),l(m,1),m.m(a.parentNode,a)):m&&(c(),d(m,1,1,(()=>{m=null})),p())},i(e){s||(l(r),l(i),l(m),s=!0)},o(e){d(r),d(i),d(m),s=!1},d(e){e&&(_(t),_(n),_(a)),r&&r.d(e),i&&i.d(e),m&&m.d(e)}}}function pn(e){let t;const n=e[15].loading,a=L(n,e,e[25],tn);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||33554432&s)&&U(a,n,e,e[25],t?A(n,e[25],s,en):T(e[25]),tn)},i(e){t||(l(a,e),t=!0)},o(e){d(a,e),t=!1},d(e){a&&a.d(e)}}}function dn(e){let t,n,a,s,i,m,u,g,b=[e[12],{src:n=e[3]},{alt:e[4]},{title:"Click to zoom in ..."}],$={};for(let e=0;e<b.length;e+=1)$=N($,b[e]);let v=e[9]&&_n(e);return{c(){t=r("img"),s=h(),v&&v.c(),i=f(),k(t,$)},m(n,a){o(n,t,a),o(n,s,a),v&&v.m(n,a),o(n,i,a),m=!0,u||(g=[w(t,"click",e[11]),w(t,"mouseenter",e[16]),w(t,"mouseleave",e[17])],u=!0)},p(a,s){k(t,$=P(b,[(e=a)[12],(!m||8&s&&!G(t.src,n=e[3]))&&{src:n},(!m||16&s)&&{alt:e[4]},{title:"Click to zoom in ..."}])),e[9]?v?(v.p(e,s),512&s&&l(v,1)):(v=_n(e),v.c(),l(v,1),v.m(i.parentNode,i)):v&&(c(),d(v,1,1,(()=>{v=null})),p())},i(n){m||(n&&O((()=>{m&&(a||(a=F(t,D,{duration:e[7]?gn:0},!0)),a.run(1))})),l(v),m=!0)},o(n){n&&(a||(a=F(t,D,{duration:e[7]?gn:0},!1)),a.run(0)),d(v),m=!1},d(e){e&&(_(t),_(s),_(i)),e&&a&&a.end(),v&&v.d(e),u=!1,H(g)}}}function _n(e){let t,n;return t=new Wt({props:{data:e[5]}}),t.$on("mouseenter",e[18]),{c(){g(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p(e,n){const a={};32&n&&(a.data=e[5]),t.$set(a)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function mn(e){let t;const n=e[15].error,a=L(n,e,e[25],Zt);return{c(){a&&a.c()},m(e,n){a&&a.m(e,n),t=!0},p(e,s){a&&a.p&&(!t||33554432&s)&&U(a,n,e,e[25],t?A(n,e[25],s,Qt):T(e[25]),Zt)},i(e){t||(l(a,e),t=!0)},o(e){d(a,e),t=!1},d(e){a&&a.d(e)}}}function un(e){let t,n,a,s;return{c(){t=r("img"),i(t,"class","pipen-report-image-lg"),G(t.src,n=e[3])||i(t,"src",n),i(t,"alt",e[4])},m(e,n){o(e,t,n),s=!0},p(a,r){e=a,(!s||8&r&&!G(t.src,n=e[3]))&&i(t,"src",n),(!s||16&r)&&i(t,"alt",e[4])},i(n){s||(n&&O((()=>{s&&(a||(a=F(t,D,{duration:e[7]?gn:0},!0)),a.run(1))})),s=!0)},o(n){n&&(a||(a=F(t,D,{duration:e[7]?gn:0},!1)),a.run(0)),s=!1},d(e){e&&_(t),e&&a&&a.end()}}}function fn(e){let t,n,a,s,r,i;const m=[an,nn],u=[];function f(e,t){return void 0===e[6]?0:1}function y(t){e[22](t)}t=f(e),n=u[t]=m[t](e);let S={modalHeading:e[4],primaryButtonText:"Open in new window ...",secondaryButtonText:"Close",$$slots:{default:[un]},$$scope:{ctx:e}};return void 0!==e[8]&&(S.open=e[8]),s=new se({props:S}),v.push((()=>x(s,"open",y))),s.$on("click:button--secondary",e[23]),s.$on("click:button--primary",e[24]),{c(){n.c(),a=h(),g(s.$$.fragment)},m(e,n){u[t].m(e,n),o(e,a,n),b(s,e,n),i=!0},p(e,[i]){let o=t;t=f(e),t===o?u[t].p(e,i):(c(),d(u[o],1,1,(()=>{u[o]=null})),p(),n=u[t],n?n.p(e,i):(n=u[t]=m[t](e),n.c()),l(n,1),n.m(a.parentNode,a));const _={};16&i&&(_.modalHeading=e[4]),33554584&i&&(_.$$scope={dirty:i,ctx:e}),!r&&256&i&&(r=!0,_.open=e[8],C((()=>r=!1))),s.$set(_)},i(e){i||(l(n),l(s.$$.fragment,e),i=!0)},o(e){d(n),d(s.$$.fragment,e),i=!1},d(e){e&&_(a),u[t].d(e),$(s,e)}}}const gn=110;function bn(e,t,n){const a=["src","alt","download","ratio","loading","loaded","error","fadeIn","border","loadImage"];let s=I(t,a),{$$slots:r={},$$scope:i}=t,{src:o=""}=t,{alt:l=""}=t,{download:c=[]}=t,{ratio:p}=t,{loading:d=!1}=t,{loaded:_=!1}=t,{error:m=!1}=t,{fadeIn:f=!1}=t,{border:g=!0}=t,b=!1,$=!1;const h=e=>{null!=C&&(C=null),n(0,_=!1),n(1,m=!1),C=new Image,C.src=e||o,C.onload=()=>n(0,_=!0),C.onerror=()=>n(1,m=!0)},v=B();let x,C=null;u((()=>()=>C=null));const y={...s,class:[s.class,"pipen-report-image",g?"":"pipen-report-image-noborder"].filter(Boolean).join(" ")};return e.$$set=e=>{t=N(N({},t),V(e)),n(28,s=I(t,a)),"src"in e&&n(3,o=e.src),"alt"in e&&n(4,l=e.alt),"download"in e&&n(5,c=e.download),"ratio"in e&&n(6,p=e.ratio),"loading"in e&&n(2,d=e.loading),"loaded"in e&&n(0,_=e.loaded),"error"in e&&n(1,m=e.error),"fadeIn"in e&&n(7,f=e.fadeIn),"border"in e&&n(13,g=e.border),"$$scope"in e&&n(25,i=e.$$scope)},e.$$.update=()=>{3&e.$$.dirty&&n(2,d=!_&&!m),8&e.$$.dirty&&o&&"undefined"!=typeof window&&h(),1&e.$$.dirty&&_&&v("load"),2&e.$$.dirty&&m&&v("error")},[_,m,d,o,l,c,p,f,b,$,x,()=>{n(8,b=!0)},y,g,h,r,()=>{n(9,$=!0)},()=>{n(10,x=setTimeout((()=>{n(9,$=!1)}),100))},()=>{clearTimeout(x),n(9,$=!0)},()=>{n(9,$=!0)},()=>{n(10,x=setTimeout((()=>{n(9,$=!1)}),100))},()=>{clearTimeout(x),n(9,$=!0)},function(e){b=e,n(8,b)},()=>{n(8,b=!1)},()=>{n(8,b=!1),window.open(o)},i]}class $n extends t{constructor(e){super(),n(this,e,bn,fn,a,{src:3,alt:4,download:5,ratio:6,loading:2,loaded:0,error:1,fadeIn:7,border:13,loadImage:14})}get loadImage(){return this.$$.ctx[14]}}function hn(e){let t,n,a=[e[2]],s={};for(let e=0;e<a.length;e+=1)s=N(s,a[e]);return{c(){t=r("div"),n=y("Image loading ..."),k(t,s)},m(e,a){o(e,t,a),S(t,n)},p:R,d(e){e&&_(t)}}}function vn(e){let t,n,a=[e[3]],s={};for(let e=0;e<a.length;e+=1)s=N(s,a[e]);return{c(){t=r("div"),n=y("Image loading error!"),k(t,s)},m(e,a){o(e,t,a),S(t,n)},p:R,d(e){e&&_(t)}}}function xn(e){let t,n,a;const s=[{src:e[1]},e[4]];let i={$$slots:{error:[vn],loading:[hn]},$$scope:{ctx:e}};for(let e=0;e<s.length;e+=1)i=N(i,s[e]);n=new $n({props:i});let c=[e[0]],p={};for(let e=0;e<c.length;e+=1)p=N(p,c[e]);return{c(){t=r("div"),g(n.$$.fragment),k(t,p)},m(e,s){o(e,t,s),b(n,t,null),a=!0},p(e,[a]){const r=18&a?P(s,[2&a&&{src:e[1]},16&a&&E(e[4])]):{};256&a&&(r.$$scope={dirty:a,ctx:e}),n.$set(r),k(t,p=P(c,[1&a&&e[0]]))},i(e){a||(l(n.$$.fragment,e),a=!0)},o(e){d(n.$$.fragment,e),a=!1},d(e){e&&_(t),$(n)}}}function Cn(e,t,n){const a=["frameProps","width","height","src"];let s=I(t,a),{frameProps:r={class:"pipen-report-image-frame"}}=t,{width:i=300}=t,{height:o=200}=t,{src:l}=t;r||(r={"style:position":"relative"});let c=`width: ${i}px; height: ${o}px; aspect-ratio: ${i} / ${o}`,p={...s,class:[s.class,"pipen-report-image-loading"].filter(Boolean).join(" "),style:s.style?`${s.style}; ${c}`:c},d={...p,class:[p.class,"error"].join(" ")};return e.$$set=e=>{t=N(N({},t),V(e)),n(4,s=I(t,a)),"frameProps"in e&&n(0,r=e.frameProps),"width"in e&&n(5,i=e.width),"height"in e&&n(6,o=e.height),"src"in e&&n(1,l=e.src)},[r,l,p,d,s,i,o]}let yn=class extends t{constructor(e){super(),n(this,e,Cn,xn,a,{frameProps:0,width:5,height:6,src:1})}};function Sn(e){let t,n;return{c(){t=r("div"),n=y(e[0]),i(t,"class","pipen-report-descr-title svelte-1xzxuva")},m(e,a){o(e,t,a),S(t,n)},p(e,t){1&t&&M(n,e[0])},d(e){e&&_(t)}}}function Mn(e){let t,n,a,s=e[0]&&Sn(e);const i=e[3].default,c=L(i,e,e[2],null);let p=[e[1]],m={};for(let e=0;e<p.length;e+=1)m=N(m,p[e]);return{c(){t=r("div"),s&&s.c(),n=h(),c&&c.c(),k(t,m),z(t,"svelte-1xzxuva",!0)},m(e,r){o(e,t,r),s&&s.m(t,null),S(t,n),c&&c.m(t,null),a=!0},p(e,[r]){e[0]?s?s.p(e,r):(s=Sn(e),s.c(),s.m(t,n)):s&&(s.d(1),s=null),c&&c.p&&(!a||4&r)&&U(c,i,e,e[2],a?A(i,e[2],r,null):T(e[2]),null),k(t,m=P(p,[2&r&&e[1]])),z(t,"svelte-1xzxuva",!0)},i(e){a||(l(c,e),a=!0)},o(e){d(c,e),a=!1},d(e){e&&_(t),s&&s.d(),c&&c.d(e)}}}function wn(e,t,n){const a=["title"];let s=I(t,a),{$$slots:r={},$$scope:i}=t,{title:o}=t,{klass:l,...c}=s;return l=l?`pipen-report-descr ${l}`:"pipen-report-descr",c||(c={}),c.class=l,e.$$set=e=>{t=N(N({},t),V(e)),n(5,s=I(t,a)),"title"in e&&n(0,o=e.title),"$$scope"in e&&n(2,i=e.$$scope)},[o,c,i,r]}class Rn extends t{constructor(e){super(),n(this,e,wn,Mn,a,{title:0})}}function Ln(e){let t;return{c(){t=y("Plots showing the number of cells per cluster, by Sex")},m(e,n){o(e,t,n)},d(e){e&&_(t)}}}function Un(e){let t,n,a,s;return t=new oe({props:{label:"Bar Plot"}}),a=new oe({props:{label:"Circos plot"}}),{c(){g(t.$$.fragment),n=h(),g(a.$$.fragment)},m(e,r){b(t,e,r),o(e,n,r),b(a,e,r),s=!0},p:R,i(e){s||(l(t.$$.fragment,e),l(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),$(t,e),$(a,e)}}}function Tn(e){let t,n;return t=new yn({props:{src:"../SeuratClusterStats/samples.seurat.annotated.tessa.cluster_stats/stats/Composition-of-cell-types-in-Colitis-samples.bar.png",download:"../SeuratClusterStats/samples.seurat.annotated.tessa.cluster_stats/stats/Composition-of-cell-types-in-Colitis-samples.bar.pdf",width:800,height:600}}),{c(){g(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p:R,i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function An(e){let t,n;return t=new yn({props:{src:"../SeuratClusterStats/samples.seurat.annotated.tessa.cluster_stats/stats/Composition-of-cell-types-in-Colitis-samples.circos.png",width:600,height:600}}),{c(){g(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},p:R,i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Nn(e){let t,n,a,s;return t=new le({props:{$$slots:{default:[Tn]},$$scope:{ctx:e}}}),a=new le({props:{$$slots:{default:[An]},$$scope:{ctx:e}}}),{c(){g(t.$$.fragment),n=h(),g(a.$$.fragment)},m(e,r){b(t,e,r),o(e,n,r),b(a,e,r),s=!0},p(e,n){const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const r={};1&n&&(r.$$scope={dirty:n,ctx:e}),a.$set(r)},i(e){s||(l(t.$$.fragment,e),l(a.$$.fragment,e),s=!0)},o(e){d(t.$$.fragment,e),d(a.$$.fragment,e),s=!1},d(e){e&&_(n),$(t,e),$(a,e)}}}function Pn(e){let t,n;return{c(){t=y("Histogram showing number of cells along "),n=r("code"),n.textContent="Activation"},m(e,a){o(e,t,a),o(e,n,a)},p:R,d(e){e&&(_(t),_(n))}}}function En(e){let t,n,a,s,i,l;return{c(){t=y("Histogram showing number of cells by "),n=r("code"),n.textContent="CDR3.aa",a=y(" in each "),s=r("code"),s.textContent="Diagnosis",i=y(" along "),l=r("code"),l.textContent="Activation"},m(e,r){o(e,t,r),o(e,n,r),o(e,a,r),o(e,s,r),o(e,i,r),o(e,l,r)},p:R,d(e){e&&(_(t),_(n),_(a),_(s),_(i),_(l))}}}function In(e){let t;return{c(){t=y("NoColitis")},m(e,n){o(e,t,n)},d(e){e&&_(t)}}}function Vn(e){let t;return{c(){t=y("Colitis")},m(e,n){o(e,t,n)},d(e){e&&_(t)}}}function Bn(e){let t;return{c(){t=y("Control")},m(e,n){o(e,t,n)},d(e){e&&_(t)}}}function kn(e){let t;return{c(){t=y("Plots showing the number of genes expressed in each seurat_clusters")},m(e,n){o(e,t,n)},d(e){e&&_(t)}}}function zn(e){let t;return{c(){t=y("violinplots for selected features, by seurat_clusters")},m(e,n){o(e,t,n)},d(e){e&&_(t)}}}function qn(e){let t;return{c(){t=y("violinplots for selected features, by seurat_clusters")},m(e,n){o(e,t,n)},d(e){e&&_(t)}}}function Gn(e){let t;return{c(){t=y("featureplots for selected features, by seurat_clusters")},m(e,n){o(e,t,n)},d(e){e&&_(t)}}}function On(e){let t;return{c(){t=y("featureplots for selected features, by seurat_clusters")},m(e,n){o(e,t,n)},d(e){e&&_(t)}}}function Fn(e){let t;return{c(){t=y("Dimensionality reduction plot for seurat_clusters")},m(e,n){o(e,t,n)},d(e){e&&_(t)}}}function Dn(e){let t;return{c(){t=y("Dimensionality reduction plot for TCR_Presence")},m(e,n){o(e,t,n)},d(e){e&&_(t)}}}function Hn(e){let t,n,a,s,c,p,m,u,f,v,x,C,y,M,w,R,L,U,T,A,N,P,E,I,V,B,k,z,q,G,O,F,D,H,W,K,X,Y,J,Q,Z,ee,te,ne,ae,se,re,oe,le,ce,pe,de,_e,me,ue,fe,ge,be,$e,he,ve,xe,Ce,ye,Se,Me,we,Re,Le,Ue,Te,Ae,Ne,Pe,Ee,Ie,Ve,Be,ke,ze,qe,Ge,Oe,Fe;return s=new Rn({props:{$$slots:{default:[Ln]},$$scope:{ctx:e}}}),p=new ie({props:{$$slots:{content:[Nn],default:[Un]},$$scope:{ctx:e}}}),x=new Rn({props:{$$slots:{default:[Pn]},$$scope:{ctx:e}}}),y=new yn({props:{src:"../SeuratClusterStats/samples.seurat.annotated.tessa.cluster_stats/hists/All-Clonotypes-along-Activation/All-Clonotypes-along-Activation.png",download:"../SeuratClusterStats/samples.seurat.annotated.tessa.cluster_stats/hists/All-Clonotypes-along-Activation/All-Clonotypes-along-Activation.pdf",width:600,height:300}}),U=new Rn({props:{$$slots:{default:[En]},$$scope:{ctx:e}}}),P=new Rn({props:{class:"pipen-report-table-image-descr",$$slots:{default:[In]},$$scope:{ctx:e}}}),I=new yn({props:{src:"../SeuratClusterStats/samples.seurat.annotated.tessa.cluster_stats/hists/Top-10-Clonotypes-along-Activation/Top-10-Clonotypes-along-Activation-NoColitis.png",download:"../SeuratClusterStats/samples.seurat.annotated.tessa.cluster_stats/hists/Top-10-Clonotypes-along-Activation/Top-10-Clonotypes-along-Activation-NoColitis.pdf",class:"pipen-report-table-image",width:600,height:600}}),k=new Rn({props:{class:"pipen-report-table-image-descr",$$slots:{default:[Vn]},$$scope:{ctx:e}}}),q=new yn({props:{src:"../SeuratClusterStats/samples.seurat.annotated.tessa.cluster_stats/hists/Top-10-Clonotypes-along-Activation/Top-10-Clonotypes-along-Activation-Colitis.png",download:"../SeuratClusterStats/samples.seurat.annotated.tessa.cluster_stats/hists/Top-10-Clonotypes-along-Activation/Top-10-Clonotypes-along-Activation-Colitis.pdf",class:"pipen-report-table-image",width:600,height:600}}),F=new Rn({props:{class:"pipen-report-table-image-descr",$$slots:{default:[Bn]},$$scope:{ctx:e}}}),H=new yn({props:{src:"../SeuratClusterStats/samples.seurat.annotated.tessa.cluster_stats/hists/Top-10-Clonotypes-along-Activation/Top-10-Clonotypes-along-Activation-Control.png",download:"../SeuratClusterStats/samples.seurat.annotated.tessa.cluster_stats/hists/Top-10-Clonotypes-along-Activation/Top-10-Clonotypes-along-Activation-Control.pdf",class:"pipen-report-table-image",width:600,height:600}}),J=new Rn({props:{$$slots:{default:[kn]},$$scope:{ctx:e}}}),Z=new yn({props:{src:"../SeuratClusterStats/samples.seurat.annotated.tessa.cluster_stats/ngenes/Number-of-genes-expressed-in-each-cluster.boxplot.png",download:"../SeuratClusterStats/samples.seurat.annotated.tessa.cluster_stats/ngenes/Number-of-genes-expressed-in-each-cluster.boxplot.pdf",width:1e3,height:800}}),se=new Rn({props:{$$slots:{default:[zn]},$$scope:{ctx:e}}}),oe=new yn({props:{src:"../SeuratClusterStats/samples.seurat.annotated.tessa.cluster_stats/features/Violin-Plots.violin.png",download:"../SeuratClusterStats/samples.seurat.annotated.tessa.cluster_stats/features/Violin-Plots.violin.pdf",width:800,height:500}}),_e=new Rn({props:{$$slots:{default:[qn]},$$scope:{ctx:e}}}),ue=new yn({props:{src:"../SeuratClusterStats/samples.seurat.annotated.tessa.cluster_stats/features/Violin-Plots-FOXP3-IL2RA-.violin.png",download:"../SeuratClusterStats/samples.seurat.annotated.tessa.cluster_stats/features/Violin-Plots-FOXP3-IL2RA-.violin.pdf",width:800,height:500}}),he=new Rn({props:{$$slots:{default:[Gn]},$$scope:{ctx:e}}}),xe=new yn({props:{src:"../SeuratClusterStats/samples.seurat.annotated.tessa.cluster_stats/features/TCR-Clone-Size.feature.png",download:"../SeuratClusterStats/samples.seurat.annotated.tessa.cluster_stats/features/TCR-Clone-Size.feature.pdf",width:400,height:300}}),we=new Rn({props:{$$slots:{default:[On]},$$scope:{ctx:e}}}),Le=new yn({props:{src:"../SeuratClusterStats/samples.seurat.annotated.tessa.cluster_stats/features/TCR-Cluster-Size.feature.png",download:"../SeuratClusterStats/samples.seurat.annotated.tessa.cluster_stats/features/TCR-Cluster-Size.feature.pdf",width:400,height:300}}),Pe=new Rn({props:{$$slots:{default:[Fn]},$$scope:{ctx:e}}}),Ie=new yn({props:{src:"../SeuratClusterStats/samples.seurat.annotated.tessa.cluster_stats/dimplots/Dimensional-reduction-plot.dim.png",download:"../SeuratClusterStats/samples.seurat.annotated.tessa.cluster_stats/dimplots/Dimensional-reduction-plot.dim.pdf",width:1e3,height:800}}),qe=new Rn({props:{$$slots:{default:[Dn]},$$scope:{ctx:e}}}),Oe=new yn({props:{src:"../SeuratClusterStats/samples.seurat.annotated.tessa.cluster_stats/dimplots/TCR-presence.dim.png",download:"../SeuratClusterStats/samples.seurat.annotated.tessa.cluster_stats/dimplots/TCR-presence.dim.pdf",width:1e3,height:800}}),{c(){t=r("h1"),t.textContent="Composition of cell types in Colitis samples",n=r("a"),n.innerHTML="",a=h(),g(s.$$.fragment),c=h(),g(p.$$.fragment),m=h(),u=r("h1"),u.textContent="All Clonotypes along Activation",f=r("a"),f.innerHTML="",v=h(),g(x.$$.fragment),C=h(),g(y.$$.fragment),M=h(),w=r("h1"),w.textContent="Top 10 Clonotypes along Activation",R=r("a"),R.innerHTML="",L=h(),g(U.$$.fragment),T=h(),A=r("div"),N=r("div"),g(P.$$.fragment),E=h(),g(I.$$.fragment),V=h(),B=r("div"),g(k.$$.fragment),z=h(),g(q.$$.fragment),G=h(),O=r("div"),g(F.$$.fragment),D=h(),g(H.$$.fragment),W=h(),K=r("h1"),K.textContent="Number of genes expressed in each cluster",X=r("a"),X.innerHTML="",Y=h(),g(J.$$.fragment),Q=h(),g(Z.$$.fragment),ee=h(),te=r("h1"),te.textContent="Violin Plots",ne=r("a"),ne.innerHTML="",ae=h(),g(se.$$.fragment),re=h(),g(oe.$$.fragment),le=h(),ce=r("h1"),ce.textContent="Violin Plots (FOXP3, IL2RA)",pe=r("a"),pe.innerHTML="",de=h(),g(_e.$$.fragment),me=h(),g(ue.$$.fragment),fe=h(),ge=r("h1"),ge.textContent="TCR Clone Size",be=r("a"),be.innerHTML="",$e=h(),g(he.$$.fragment),ve=h(),g(xe.$$.fragment),Ce=h(),ye=r("h1"),ye.textContent="TCR Cluster Size",Se=r("a"),Se.innerHTML="",Me=h(),g(we.$$.fragment),Re=h(),g(Le.$$.fragment),Ue=h(),Te=r("h1"),Te.textContent="Dimensional reduction plot",Ae=r("a"),Ae.innerHTML="",Ne=h(),g(Pe.$$.fragment),Ee=h(),g(Ie.$$.fragment),Ve=h(),Be=r("h1"),Be.textContent="TCR presence",ke=r("a"),ke.innerHTML="",ze=h(),g(qe.$$.fragment),Ge=h(),g(Oe.$$.fragment),i(n,"id","prt-h1-0-composition-of-cell-types-in-colitis-samples"),i(n,"class","pipen-report-toc-anchor"),i(f,"id","prt-h1-1-all-clonotypes-along-activation"),i(f,"class","pipen-report-toc-anchor"),i(R,"id","prt-h1-2-top-10-clonotypes-along-activation"),i(R,"class","pipen-report-toc-anchor"),i(A,"class","pipen-report-table-of-images"),j(A,"grid-template-columns","repeat(2, auto)"),i(X,"id","prt-h1-3-number-of-genes-expressed-in-each-cluster"),i(X,"class","pipen-report-toc-anchor"),i(ne,"id","prt-h1-4-violin-plots"),i(ne,"class","pipen-report-toc-anchor"),i(pe,"id","prt-h1-5-violin-plots-foxp3-il2ra"),i(pe,"class","pipen-report-toc-anchor"),i(be,"id","prt-h1-6-tcr-clone-size"),i(be,"class","pipen-report-toc-anchor"),i(Se,"id","prt-h1-7-tcr-cluster-size"),i(Se,"class","pipen-report-toc-anchor"),i(Ae,"id","prt-h1-8-dimensional-reduction-plot"),i(Ae,"class","pipen-report-toc-anchor"),i(ke,"id","prt-h1-9-tcr-presence"),i(ke,"class","pipen-report-toc-anchor")},m(e,r){o(e,t,r),o(e,n,r),o(e,a,r),b(s,e,r),o(e,c,r),b(p,e,r),o(e,m,r),o(e,u,r),o(e,f,r),o(e,v,r),b(x,e,r),o(e,C,r),b(y,e,r),o(e,M,r),o(e,w,r),o(e,R,r),o(e,L,r),b(U,e,r),o(e,T,r),o(e,A,r),S(A,N),b(P,N,null),S(N,E),b(I,N,null),S(A,V),S(A,B),b(k,B,null),S(B,z),b(q,B,null),S(A,G),S(A,O),b(F,O,null),S(O,D),b(H,O,null),o(e,W,r),o(e,K,r),o(e,X,r),o(e,Y,r),b(J,e,r),o(e,Q,r),b(Z,e,r),o(e,ee,r),o(e,te,r),o(e,ne,r),o(e,ae,r),b(se,e,r),o(e,re,r),b(oe,e,r),o(e,le,r),o(e,ce,r),o(e,pe,r),o(e,de,r),b(_e,e,r),o(e,me,r),b(ue,e,r),o(e,fe,r),o(e,ge,r),o(e,be,r),o(e,$e,r),b(he,e,r),o(e,ve,r),b(xe,e,r),o(e,Ce,r),o(e,ye,r),o(e,Se,r),o(e,Me,r),b(we,e,r),o(e,Re,r),b(Le,e,r),o(e,Ue,r),o(e,Te,r),o(e,Ae,r),o(e,Ne,r),b(Pe,e,r),o(e,Ee,r),b(Ie,e,r),o(e,Ve,r),o(e,Be,r),o(e,ke,r),o(e,ze,r),b(qe,e,r),o(e,Ge,r),b(Oe,e,r),Fe=!0},p(e,[t]){const n={};1&t&&(n.$$scope={dirty:t,ctx:e}),s.$set(n);const a={};1&t&&(a.$$scope={dirty:t,ctx:e}),p.$set(a);const r={};1&t&&(r.$$scope={dirty:t,ctx:e}),x.$set(r);const i={};1&t&&(i.$$scope={dirty:t,ctx:e}),U.$set(i);const o={};1&t&&(o.$$scope={dirty:t,ctx:e}),P.$set(o);const l={};1&t&&(l.$$scope={dirty:t,ctx:e}),k.$set(l);const c={};1&t&&(c.$$scope={dirty:t,ctx:e}),F.$set(c);const d={};1&t&&(d.$$scope={dirty:t,ctx:e}),J.$set(d);const _={};1&t&&(_.$$scope={dirty:t,ctx:e}),se.$set(_);const m={};1&t&&(m.$$scope={dirty:t,ctx:e}),_e.$set(m);const u={};1&t&&(u.$$scope={dirty:t,ctx:e}),he.$set(u);const f={};1&t&&(f.$$scope={dirty:t,ctx:e}),we.$set(f);const g={};1&t&&(g.$$scope={dirty:t,ctx:e}),Pe.$set(g);const b={};1&t&&(b.$$scope={dirty:t,ctx:e}),qe.$set(b)},i(e){Fe||(l(s.$$.fragment,e),l(p.$$.fragment,e),l(x.$$.fragment,e),l(y.$$.fragment,e),l(U.$$.fragment,e),l(P.$$.fragment,e),l(I.$$.fragment,e),l(k.$$.fragment,e),l(q.$$.fragment,e),l(F.$$.fragment,e),l(H.$$.fragment,e),l(J.$$.fragment,e),l(Z.$$.fragment,e),l(se.$$.fragment,e),l(oe.$$.fragment,e),l(_e.$$.fragment,e),l(ue.$$.fragment,e),l(he.$$.fragment,e),l(xe.$$.fragment,e),l(we.$$.fragment,e),l(Le.$$.fragment,e),l(Pe.$$.fragment,e),l(Ie.$$.fragment,e),l(qe.$$.fragment,e),l(Oe.$$.fragment,e),Fe=!0)},o(e){d(s.$$.fragment,e),d(p.$$.fragment,e),d(x.$$.fragment,e),d(y.$$.fragment,e),d(U.$$.fragment,e),d(P.$$.fragment,e),d(I.$$.fragment,e),d(k.$$.fragment,e),d(q.$$.fragment,e),d(F.$$.fragment,e),d(H.$$.fragment,e),d(J.$$.fragment,e),d(Z.$$.fragment,e),d(se.$$.fragment,e),d(oe.$$.fragment,e),d(_e.$$.fragment,e),d(ue.$$.fragment,e),d(he.$$.fragment,e),d(xe.$$.fragment,e),d(we.$$.fragment,e),d(Le.$$.fragment,e),d(Pe.$$.fragment,e),d(Ie.$$.fragment,e),d(qe.$$.fragment,e),d(Oe.$$.fragment,e),Fe=!1},d(e){e&&(_(t),_(n),_(a),_(c),_(m),_(u),_(f),_(v),_(C),_(M),_(w),_(R),_(L),_(T),_(A),_(W),_(K),_(X),_(Y),_(Q),_(ee),_(te),_(ne),_(ae),_(re),_(le),_(ce),_(pe),_(de),_(me),_(fe),_(ge),_(be),_($e),_(ve),_(Ce),_(ye),_(Se),_(Me),_(Re),_(Ue),_(Te),_(Ae),_(Ne),_(Ee),_(Ve),_(Be),_(ke),_(ze),_(Ge)),$(s,e),$(p,e),$(x,e),$(y,e),$(U,e),$(P),$(I),$(k),$(q),$(F),$(H),$(J,e),$(Z,e),$(se,e),$(oe,e),$(_e,e),$(ue,e),$(he,e),$(xe,e),$(we,e),$(Le,e),$(Pe,e),$(Ie,e),$(qe,e),$(Oe,e)}}}class jn extends t{constructor(e){super(),n(this,e,null,Hn,a,{})}}var Wn=[{slug:"prt-h1-0-composition-of-cell-types-in-colitis-samples",text:"Composition of cell types in Colitis samples",children:[],page:0},{slug:"prt-h1-1-all-clonotypes-along-activation",text:"All Clonotypes along Activation",children:[],page:0},{slug:"prt-h1-2-top-10-clonotypes-along-activation",text:"Top 10 Clonotypes along Activation",children:[],page:0},{slug:"prt-h1-3-number-of-genes-expressed-in-each-cluster",text:"Number of genes expressed in each cluster",children:[],page:0},{slug:"prt-h1-4-violin-plots",text:"Violin Plots",children:[],page:0},{slug:"prt-h1-5-violin-plots-foxp3-il2ra",text:"Violin Plots (FOXP3, IL2RA)",children:[],page:0},{slug:"prt-h1-6-tcr-clone-size",text:"TCR Clone Size",children:[],page:0},{slug:"prt-h1-7-tcr-cluster-size",text:"TCR Cluster Size",children:[],page:0},{slug:"prt-h1-8-dimensional-reduction-plot",text:"Dimensional reduction plot",children:[],page:0},{slug:"prt-h1-9-tcr-presence",text:"TCR presence",children:[],page:0}],Kn={pipeline:{name:"Immunopipe",desc:"Immunopipe (v1.4.3): Integrative analysis for scRNA-seq and scTCR-seq data"},versions:'\n    <a href="https://github.com/pwwang/pipen" target="_blank">\n        pipen v0.15.7\n    </a> and\n    <a href="https://github.com/pwwang/pipen-report" target="_blank">\n        pipen-report v0.21.4\n    </a>\n',entries:[{name:"SampleInfo",desc:"List sample information and perform statistics",npages:1,report_toc:!0,order:0,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n[1] ggrepel_0.9.6    ggprism_1.0.5    ggplot2_3.5.1    tidyr_1.3.1     \n[5] dplyr_1.1.4      tidyselect_1.2.1 jsonlite_1.8.9   rlang_1.1.5     \n[9] logger_0.4.0    \n\nloaded via a namespace (and not attached):\n [1] vctrs_0.6.5      cli_3.6.4        purrr_1.0.2      generics_0.1.3  \n [5] glue_1.8.0       labeling_0.4.3   colorspace_2.1-1 scales_1.3.0    \n [9] grid_4.3.3       munsell_0.5.1    tibble_3.2.1     lifecycle_1.0.4 \n[13] compiler_4.3.3   Rcpp_1.0.14      pkgconfig_2.0.3  farver_2.1.2    \n[17] digest_0.6.37    R6_2.6.0         pillar_1.10.1    magrittr_2.0.3  \n[21] withr_3.0.2      gtable_0.3.6    \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 2.19\nuser 11.15\nsys 0.11\npercent 513%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3421.613\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"SeuratPreparing",desc:"Load, prepare and apply QC to data, using `Seurat`",npages:1,report_toc:!0,order:2,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats4    stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] scDblFinder_1.16.0          SingleCellExperiment_1.24.0\n [3] SummarizedExperiment_1.32.0 Biobase_2.62.0             \n [5] GenomicRanges_1.54.1        GenomeInfoDb_1.38.1        \n [7] IRanges_2.36.0              S4Vectors_0.40.2           \n [9] BiocGenerics_0.48.1         MatrixGenerics_1.14.0      \n[11] matrixStats_1.1.0           dplyr_1.1.4                \n[13] ggplot2_3.5.1               bracer_1.2.1               \n[15] future_1.34.0               Seurat_5.1.0               \n[17] SeuratObject_5.0.2          sp_2.2-0                   \n[19] digest_0.6.37               jsonlite_1.8.9             \n[21] rlang_1.1.5                 logger_0.4.0               \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22          splines_4.3.3            \n  [3] later_1.4.1               BiocIO_1.12.0            \n  [5] bitops_1.0-9              tibble_3.2.1             \n  [7] R.oo_1.27.0               polyclip_1.10-7          \n  [9] XML_3.99-0.17             fastDummies_1.7.5        \n [11] lifecycle_1.0.4           edgeR_4.0.16             \n [13] globals_0.16.3            lattice_0.22-6           \n [15] MASS_7.3-60.0.1           magrittr_2.0.3           \n [17] limma_3.58.1              plotly_4.10.4            \n [19] yaml_2.3.10               metapod_1.10.0           \n [21] httpuv_1.6.15             sctransform_0.4.1        \n [23] spam_2.11-1               spatstat.sparse_3.1-0    \n [25] reticulate_1.40.0         cowplot_1.1.3            \n [27] pbapply_1.7-2             RColorBrewer_1.1-3       \n [29] abind_1.4-5               zlibbioc_1.48.0          \n [31] Rtsne_0.17                purrr_1.0.2              \n [33] R.utils_2.12.3            RCurl_1.98-1.16          \n [35] GenomeInfoDbData_1.2.11   ggrepel_0.9.6            \n [37] irlba_2.3.5.1             listenv_0.9.1            \n [39] spatstat.utils_3.1-2      goftest_1.2-3            \n [41] RSpectra_0.16-2           dqrng_0.3.2              \n [43] spatstat.random_3.3-2     fitdistrplus_1.2-2       \n [45] parallelly_1.42.0         DelayedMatrixStats_1.24.0\n [47] leiden_0.4.3.1            codetools_0.2-20         \n [49] DelayedArray_0.28.0       scuttle_1.12.0           \n [51] tidyselect_1.2.1          farver_2.1.2             \n [53] viridis_0.6.5             ScaledMatrix_1.10.0      \n [55] spatstat.explore_3.3-4    GenomicAlignments_1.38.0 \n [57] BiocNeighbors_1.20.0      progressr_0.15.1         \n [59] scater_1.30.1             ggridges_0.5.6           \n [61] survival_3.8-3            systemfonts_1.2.1        \n [63] tools_4.3.3               ragg_1.3.3               \n [65] ica_1.0-3                 Rcpp_1.0.14              \n [67] glue_1.8.0                gridExtra_2.3            \n [69] SparseArray_1.2.2         withr_3.0.2              \n [71] fastmap_1.2.0             bluster_1.12.0           \n [73] rsvd_1.0.5                R6_2.6.0                 \n [75] mime_0.12                 textshaping_0.4.0        \n [77] colorspace_2.1-1          scattermore_1.2          \n [79] tensor_1.5                spatstat.data_3.1-4      \n [81] R.methodsS3_1.8.2         RhpcBLASctl_0.23-42      \n [83] tidyr_1.3.1               generics_0.1.3           \n [85] data.table_1.16.4         rtracklayer_1.62.0       \n [87] httr_1.4.7                htmlwidgets_1.6.4        \n [89] S4Arrays_1.2.0            uwot_0.2.2               \n [91] pkgconfig_2.0.3           gtable_0.3.6             \n [93] lmtest_0.9-40             XVector_0.42.0           \n [95] htmltools_0.5.8.1         dotCall64_1.2            \n [97] scales_1.3.0              png_0.1-8                \n [99] harmony_1.2.0             spatstat.univar_3.1-1    \n[101] scran_1.30.0              reshape2_1.4.4           \n[103] rjson_0.2.23              nlme_3.1-167             \n[105] zoo_1.8-12                stringr_1.5.1            \n[107] KernSmooth_2.23-26        vipor_0.4.7              \n[109] parallel_4.3.3            miniUI_0.1.1.1           \n[111] restfulr_0.0.15           pillar_1.10.1            \n[113] grid_4.3.3                vctrs_0.6.5              \n[115] RANN_2.6.2                promises_1.3.2           \n[117] BiocSingular_1.18.0       beachmat_2.18.0          \n[119] xtable_1.8-4              cluster_2.1.8            \n[121] beeswarm_0.4.0            locfit_1.5-9.11          \n[123] cli_3.6.4                 compiler_4.3.3           \n[125] Rsamtools_2.18.0          crayon_1.5.3             \n[127] future.apply_1.11.3       labeling_0.4.3           \n[129] ggbeeswarm_0.7.2          plyr_1.8.9               \n[131] stringi_1.8.4             viridisLite_0.4.2        \n[133] deldir_2.0-4              BiocParallel_1.36.0      \n[135] munsell_0.5.1             Biostrings_2.70.1        \n[137] lazyeval_0.2.2            spatstat.geom_3.3-5      \n[139] Matrix_1.6-5              RcppHNSW_0.6.0           \n[141] patchwork_1.3.0           sparseMatrixStats_1.14.0 \n[143] statmod_1.5.0             shiny_1.10.0             \n[145] ROCR_1.0-11               igraph_2.0.3             \n[147] xgboost_2.1.4.1          \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 209.65\nuser 1454.55\nsys 7.57\npercent 697%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ClusterMarkersOfAllCells",desc:"Markers for clusters of all cells.",npages:3,report_toc:!0,order:4,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] UpSetR_1.4.0        ggVennDiagram_1.2.3 tidyseurat_0.8.0   \n [4] ttservice_0.4.1     future_1.34.0       ggrepel_0.9.6      \n [7] ggprism_1.0.5       ggplot2_3.5.1       enrichR_3.4        \n[10] Seurat_5.1.0        SeuratObject_5.0.2  sp_2.2-0           \n[13] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[16] tidyselect_1.2.1    digest_0.6.37       jsonlite_1.8.9     \n[19] rlang_1.1.5         logger_0.4.0       \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] ellipsis_0.3.2         labeling_0.4.3         promises_1.3.2        \n [19] purrr_1.0.2            WriteXLS_6.7.0         RVenn_1.1.0           \n [22] goftest_1.2-3          later_1.4.1            spatstat.utils_3.1-2  \n [25] irlba_2.3.5.1          parallel_4.3.3         cluster_2.1.8         \n [28] R6_2.6.0               ica_1.0-3              stringi_1.8.4         \n [31] RColorBrewer_1.1-3     spatstat.data_3.1-4    limma_3.58.1          \n [34] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [37] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [40] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [43] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [46] splines_4.3.3          igraph_2.0.3           abind_1.4-5           \n [49] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.3-4 curl_6.0.1             listenv_0.9.1         \n [55] lattice_0.22-6         plyr_1.8.9             withr_3.0.2           \n [58] shiny_1.10.0           ROCR_1.0-11            Rtsne_0.17            \n [61] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n [64] fitdistrplus_1.2-2     pillar_1.10.1          KernSmooth_2.23-26    \n [67] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n [82] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n [85] nlme_3.1-167           patchwork_1.3.0        presto_1.0.0          \n [88] cli_3.6.4              spatstat.sparse_3.1-0  fansi_1.0.6           \n [91] spam_2.11-1            viridisLite_0.4.2      uwot_0.2.2            \n [94] gtable_0.3.6           progressr_0.15.1       rjson_0.2.23          \n [97] htmlwidgets_1.6.4      farver_2.1.2           htmltools_0.5.8.1     \n[100] lifecycle_1.0.4        httr_1.4.7             statmod_1.5.0         \n[103] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 259.76\nuser 186.00\nsys 8.62\npercent 74%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TopExpressingGenesOfAllCells",desc:"Top expressing genes for clusters of all cells.",npages:3,report_toc:!0,order:5,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ggprism_1.0.5      dplyr_1.1.4        enrichR_3.4        tibble_3.2.1      \n [5] Seurat_5.1.0       SeuratObject_5.0.2 sp_2.2-0           jsonlite_1.8.9    \n [9] rlang_1.1.5        logger_0.4.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] labeling_0.4.3         promises_1.3.2         purrr_1.0.2           \n [19] WriteXLS_6.7.0         goftest_1.2-3          later_1.4.1           \n [22] spatstat.utils_3.1-2   irlba_2.3.5.1          parallel_4.3.3        \n [25] cluster_2.1.8          R6_2.6.0               ica_1.0-3             \n [28] stringi_1.8.4          RColorBrewer_1.1-3     spatstat.data_3.1-4   \n [31] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [37] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [43] splines_4.3.3          igraph_2.0.3           tidyselect_1.2.1      \n [46] abind_1.4-5            spatstat.random_3.3-2  codetools_0.2-20      \n [49] miniUI_0.1.1.1         spatstat.explore_3.3-4 curl_6.0.1            \n [52] listenv_0.9.1          lattice_0.22-6         plyr_1.8.9            \n [55] withr_3.0.2            shiny_1.10.0           ROCR_1.0-11           \n [58] Rtsne_0.17             future_1.34.0          fastDummies_1.7.5     \n [61] survival_3.8-3         polyclip_1.10-7        fitdistrplus_1.2-2    \n [64] pillar_1.10.1          KernSmooth_2.23-26     plotly_4.10.4         \n [67] generics_0.1.3         RcppHNSW_0.6.0         ggplot2_3.5.1         \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n [82] cowplot_1.1.3          grid_4.3.3             tidyr_1.3.1           \n [85] colorspace_2.1-1       nlme_3.1-167           patchwork_1.3.0       \n [88] cli_3.6.4              spatstat.sparse_3.1-0  spam_2.11-1           \n [91] viridisLite_0.4.2      uwot_0.2.2             gtable_0.3.6          \n [94] digest_0.6.37          progressr_0.15.1       ggrepel_0.9.6         \n [97] rjson_0.2.23           htmlwidgets_1.6.4      farver_2.1.2          \n[100] htmltools_0.5.8.1      lifecycle_1.0.4        httr_1.4.7            \n[103] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 73.70\nuser 41.45\nsys 2.53\npercent 59%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TCellSelection",desc:"Separate T and non-T cells and select T cells.",npages:1,report_toc:!0,order:6,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] glue_1.8.0         factoextra_1.0.7   patchwork_1.3.0    ggpubr_0.6.0      \n [5] ggrepel_0.9.6      ggprism_1.0.5      ggplot2_3.5.1      tibble_3.2.1      \n [9] tidyr_1.3.1        dplyr_1.1.4        Seurat_5.1.0       SeuratObject_5.0.2\n[13] sp_2.2-0           jsonlite_1.8.9     rlang_1.1.5        logger_0.4.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] backports_1.5.0        labeling_0.4.3         promises_1.3.2        \n [19] purrr_1.0.2            goftest_1.2-3          later_1.4.1           \n [22] spatstat.utils_3.1-2   broom_1.0.7            irlba_2.3.5.1         \n [25] parallel_4.3.3         cluster_2.1.8          R6_2.6.0              \n [28] ica_1.0-3              stringi_1.8.4          RColorBrewer_1.1-3    \n [31] spatstat.data_3.1-4    reticulate_1.40.0      car_3.1-3             \n [34] parallelly_1.42.0      spatstat.univar_3.1-1  lmtest_0.9-40         \n [37] scattermore_1.2        Rcpp_1.0.14            tensor_1.5            \n [40] future.apply_1.11.3    zoo_1.8-12             sctransform_0.4.1     \n [43] httpuv_1.6.15          Matrix_1.6-5           splines_4.3.3         \n [46] igraph_2.0.3           tidyselect_1.2.1       abind_1.4-5           \n [49] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.3-4 listenv_0.9.1          lattice_0.22-6        \n [55] plyr_1.8.9             withr_3.0.2            shiny_1.10.0          \n [58] ROCR_1.0-11            Rtsne_0.17             future_1.34.0         \n [61] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n [64] fitdistrplus_1.2-2     pillar_1.10.1          carData_3.0-5         \n [67] KernSmooth_2.23-26     plotly_4.10.4          generics_0.1.3        \n [70] RcppHNSW_0.6.0         munsell_0.5.1          scales_1.3.0          \n [73] globals_0.16.3         xtable_1.8-4           lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] ggsignif_0.6.4         RANN_2.6.2             leiden_0.4.3.1        \n [82] dotCall64_1.2          cowplot_1.1.3          grid_4.3.3            \n [85] colorspace_2.1-1       nlme_3.1-167           Formula_1.2-5         \n [88] cli_3.6.4              spatstat.sparse_3.1-0  spam_2.11-1           \n [91] viridisLite_0.4.2      uwot_0.2.2             gtable_0.3.6          \n [94] rstatix_0.7.2          digest_0.6.37          progressr_0.15.1      \n [97] htmlwidgets_1.6.4      farver_2.1.2           htmltools_0.5.8.1     \n[100] lifecycle_1.0.4        httr_1.4.7             mime_0.12             \n[103] MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 41.63\nuser 49.85\nsys 0.77\npercent 121%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"ClusterMarkers",desc:"Markers for clusters of all or selected T cells.",npages:2,report_toc:!0,order:10,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] UpSetR_1.4.0        ggVennDiagram_1.2.3 tidyseurat_0.8.0   \n [4] ttservice_0.4.1     future_1.34.0       ggrepel_0.9.6      \n [7] ggprism_1.0.5       ggplot2_3.5.1       enrichR_3.4        \n[10] Seurat_5.1.0        SeuratObject_5.0.2  sp_2.2-0           \n[13] tibble_3.2.1        tidyr_1.3.1         dplyr_1.1.4        \n[16] tidyselect_1.2.1    digest_0.6.37       jsonlite_1.8.9     \n[19] rlang_1.1.5         logger_0.4.0       \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] ellipsis_0.3.2         labeling_0.4.3         promises_1.3.2        \n [19] purrr_1.0.2            WriteXLS_6.7.0         RVenn_1.1.0           \n [22] goftest_1.2-3          later_1.4.1            spatstat.utils_3.1-2  \n [25] irlba_2.3.5.1          parallel_4.3.3         cluster_2.1.8         \n [28] R6_2.6.0               ica_1.0-3              stringi_1.8.4         \n [31] RColorBrewer_1.1-3     spatstat.data_3.1-4    limma_3.58.1          \n [34] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [37] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [40] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [43] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [46] splines_4.3.3          igraph_2.0.3           abind_1.4-5           \n [49] spatstat.random_3.3-2  codetools_0.2-20       miniUI_0.1.1.1        \n [52] spatstat.explore_3.3-4 curl_6.0.1             listenv_0.9.1         \n [55] lattice_0.22-6         plyr_1.8.9             withr_3.0.2           \n [58] shiny_1.10.0           ROCR_1.0-11            Rtsne_0.17            \n [61] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       \n [64] fitdistrplus_1.2-2     pillar_1.10.1          KernSmooth_2.23-26    \n [67] plotly_4.10.4          generics_0.1.3         RcppHNSW_0.6.0        \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n [82] cowplot_1.1.3          grid_4.3.3             colorspace_2.1-1      \n [85] nlme_3.1-167           patchwork_1.3.0        presto_1.0.0          \n [88] cli_3.6.4              spatstat.sparse_3.1-0  fansi_1.0.6           \n [91] spam_2.11-1            viridisLite_0.4.2      uwot_0.2.2            \n [94] gtable_0.3.6           progressr_0.15.1       rjson_0.2.23          \n [97] htmlwidgets_1.6.4      farver_2.1.2           htmltools_0.5.8.1     \n[100] lifecycle_1.0.4        httr_1.4.7             statmod_1.5.0         \n[103] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 162.57\nuser 119.15\nsys 5.40\npercent 76%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TopExpressingGenes",desc:"Top expressing genes for clusters of all or selected T cells.",npages:2,report_toc:!0,order:11,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] ggprism_1.0.5      dplyr_1.1.4        enrichR_3.4        tibble_3.2.1      \n [5] Seurat_5.1.0       SeuratObject_5.0.2 sp_2.2-0           jsonlite_1.8.9    \n [9] rlang_1.1.5        logger_0.4.0      \n\nloaded via a namespace (and not attached):\n  [1] deldir_2.0-4           pbapply_1.7-2          gridExtra_2.3         \n  [4] magrittr_2.0.3         RcppAnnoy_0.0.22       matrixStats_1.1.0     \n  [7] ggridges_0.5.6         compiler_4.3.3         spatstat.geom_3.3-5   \n [10] png_0.1-8              vctrs_0.6.5            reshape2_1.4.4        \n [13] stringr_1.5.1          pkgconfig_2.0.3        fastmap_1.2.0         \n [16] labeling_0.4.3         promises_1.3.2         purrr_1.0.2           \n [19] WriteXLS_6.7.0         goftest_1.2-3          later_1.4.1           \n [22] spatstat.utils_3.1-2   irlba_2.3.5.1          parallel_4.3.3        \n [25] cluster_2.1.8          R6_2.6.0               ica_1.0-3             \n [28] stringi_1.8.4          RColorBrewer_1.1-3     spatstat.data_3.1-4   \n [31] reticulate_1.40.0      parallelly_1.42.0      spatstat.univar_3.1-1 \n [34] lmtest_0.9-40          scattermore_1.2        Rcpp_1.0.14           \n [37] tensor_1.5             future.apply_1.11.3    zoo_1.8-12            \n [40] sctransform_0.4.1      httpuv_1.6.15          Matrix_1.6-5          \n [43] splines_4.3.3          igraph_2.0.3           tidyselect_1.2.1      \n [46] abind_1.4-5            spatstat.random_3.3-2  codetools_0.2-20      \n [49] miniUI_0.1.1.1         spatstat.explore_3.3-4 curl_6.0.1            \n [52] listenv_0.9.1          lattice_0.22-6         plyr_1.8.9            \n [55] withr_3.0.2            shiny_1.10.0           ROCR_1.0-11           \n [58] Rtsne_0.17             future_1.34.0          fastDummies_1.7.5     \n [61] survival_3.8-3         polyclip_1.10-7        fitdistrplus_1.2-2    \n [64] pillar_1.10.1          KernSmooth_2.23-26     plotly_4.10.4         \n [67] generics_0.1.3         RcppHNSW_0.6.0         ggplot2_3.5.1         \n [70] munsell_0.5.1          scales_1.3.0           globals_0.16.3        \n [73] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        \n [76] tools_4.3.3            data.table_1.16.4      RSpectra_0.16-2       \n [79] RANN_2.6.2             leiden_0.4.3.1         dotCall64_1.2         \n [82] cowplot_1.1.3          grid_4.3.3             tidyr_1.3.1           \n [85] colorspace_2.1-1       nlme_3.1-167           patchwork_1.3.0       \n [88] cli_3.6.4              spatstat.sparse_3.1-0  spam_2.11-1           \n [91] viridisLite_0.4.2      uwot_0.2.2             gtable_0.3.6          \n [94] digest_0.6.37          progressr_0.15.1       ggrepel_0.9.6         \n [97] rjson_0.2.23           htmlwidgets_1.6.4      farver_2.1.2          \n[100] htmltools_0.5.8.1      lifecycle_1.0.4        httr_1.4.7            \n[103] mime_0.12              MASS_7.3-60.0.1       \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 45.21\nuser 28.74\nsys 1.92\npercent 67%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TCRClusterStats",desc:"Statistics of TCR clusters, generated by `TCRClustering`.",npages:1,report_toc:!0,order:13,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] grid      stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] ComplexHeatmap_2.18.0 ggprism_1.0.5         immunarch_0.9.1      \n [4] patchwork_1.3.0       data.table_1.16.4     dtplyr_1.3.1         \n [7] dplyr_1.1.4           tidyr_1.3.1           tibble_3.2.1         \n[10] ggplot2_3.5.1         jsonlite_1.8.9        rlang_1.1.5          \n[13] logger_0.4.0         \n\nloaded via a namespace (and not attached):\n  [1] gridExtra_2.3       phangorn_2.12.1     readxl_1.4.3       \n  [4] magrittr_2.0.3      clue_0.3-66         GetoptLong_1.0.5   \n  [7] matrixStats_1.1.0   compiler_4.3.3      flexmix_2.3-19     \n [10] png_0.1-8           vctrs_0.6.5         reshape2_1.4.4     \n [13] ggalluvial_0.12.5   quadprog_1.5-8      stringr_1.5.1      \n [16] crayon_1.5.3        pkgconfig_2.0.3     shape_1.4.6.1      \n [19] fastmap_1.2.0       backports_1.5.0     labeling_0.4.3     \n [22] ggraph_2.2.1        promises_1.3.2      tzdb_0.4.0         \n [25] UpSetR_1.4.0        purrr_1.0.2         modeltools_0.2-23  \n [28] ggseqlogo_0.2       cachem_1.1.0        later_1.4.1        \n [31] uuid_1.2-1          fpc_2.2-13          tweenr_2.0.3       \n [34] broom_1.0.7         parallel_4.3.3      prabclus_2.3-4     \n [37] cluster_2.1.8       R6_2.6.0            stringi_1.8.4      \n [40] RColorBrewer_1.1-3  rlist_0.4.6.2       car_3.1-3          \n [43] diptest_0.77-1      cellranger_1.1.0    Rcpp_1.0.14        \n [46] iterators_1.0.14    IRanges_2.36.0      readr_2.1.5        \n [49] httpuv_1.6.15       Matrix_1.6-5        nnet_7.3-20        \n [52] igraph_2.0.3        tidyselect_1.2.1    stringdist_0.9.15  \n [55] abind_1.4-5         viridis_0.6.5       doParallel_1.0.17  \n [58] codetools_0.2-20    lattice_0.22-6      plyr_1.8.9         \n [61] shiny_1.10.0        withr_3.0.2         polyclip_1.10-7    \n [64] circlize_0.4.16     mclust_6.1.1        kernlab_0.9-33     \n [67] pillar_1.10.1       ggpubr_0.6.0        carData_3.0-5      \n [70] foreach_1.5.2       stats4_4.3.3        generics_0.1.3     \n [73] S4Vectors_0.40.2    hms_1.1.3           munsell_0.5.1      \n [76] scales_1.3.0        xtable_1.8-4        class_7.3-23       \n [79] glue_1.8.0          pheatmap_1.0.12     tools_4.3.3        \n [82] robustbase_0.99-4-1 ggsignif_0.6.4      graphlayouts_1.2.2 \n [85] Cairo_1.6-2         fastmatch_1.1-6     tidygraph_1.3.0    \n [88] ape_5.8-1           colorspace_2.1-1    nlme_3.1-167       \n [91] ggforce_0.4.2       Formula_1.2-5       cli_3.6.4          \n [94] viridisLite_0.4.2   gtable_0.3.6        DEoptimR_1.1-3-1   \n [97] rstatix_0.7.2       digest_0.6.37       BiocGenerics_0.48.1\n[100] ggrepel_0.9.6       rjson_0.2.23        farver_2.1.2       \n[103] memoise_2.0.1       htmltools_0.5.8.1   lifecycle_1.0.4    \n[106] factoextra_1.0.7    mime_0.12           GlobalOptions_0.1.2\n[109] shinythemes_1.2.0   MASS_7.3-60.0.1    \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 7.17\nuser 15.93\nsys 0.32\npercent 226%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3287.293\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"TESSA",desc:"Tessa is a Bayesian model to integrate T cell receptor (TCR) sequence profiling with transcriptomes of T cells.",npages:1,report_toc:!0,order:14,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] igraph_2.0.3         Rtsne_0.17           LaplacesDemon_16.1.6\n [4] MASS_7.3-60.0.1      ggprism_1.0.5        Seurat_5.1.0        \n [7] SeuratObject_5.0.2   sp_2.2-0             tibble_3.2.1        \n[10] glue_1.8.0           immunarch_0.9.1      patchwork_1.3.0     \n[13] data.table_1.16.4    dtplyr_1.3.1         ggplot2_3.5.1       \n[16] tidyr_1.3.1          dplyr_1.1.4          jsonlite_1.8.9      \n[19] rlang_1.1.5          logger_0.4.0        \n\nloaded via a namespace (and not attached):\n  [1] RcppAnnoy_0.0.22       shinythemes_1.2.0      splines_4.3.3         \n  [4] later_1.4.1            cellranger_1.1.0       polyclip_1.10-7       \n  [7] fastDummies_1.7.5      factoextra_1.0.7       lifecycle_1.0.4       \n [10] rstatix_0.7.2          doParallel_1.0.17      globals_0.16.3        \n [13] lattice_0.22-6         prabclus_2.3-4         backports_1.5.0       \n [16] magrittr_2.0.3         plotly_4.10.4          rlist_0.4.6.2         \n [19] httpuv_1.6.15          sctransform_0.4.1      spam_2.11-1           \n [22] flexmix_2.3-19         spatstat.sparse_3.1-0  reticulate_1.40.0     \n [25] cowplot_1.1.3          pbapply_1.7-2          RColorBrewer_1.1-3    \n [28] abind_1.4-5            quadprog_1.5-8         purrr_1.0.2           \n [31] ggraph_2.2.1           nnet_7.3-20            tweenr_2.0.3          \n [34] circlize_0.4.16        ggrepel_0.9.6          irlba_2.3.5.1         \n [37] listenv_0.9.1          spatstat.utils_3.1-2   pheatmap_1.0.12       \n [40] goftest_1.2-3          RSpectra_0.16-2        spatstat.random_3.3-2 \n [43] fitdistrplus_1.2-2     parallelly_1.42.0      leiden_0.4.3.1        \n [46] codetools_0.2-20       ggforce_0.4.2          tidyselect_1.2.1      \n [49] shape_1.4.6.1          ggseqlogo_0.2          farver_2.1.2          \n [52] viridis_0.6.5          matrixStats_1.1.0      stats4_4.3.3          \n [55] spatstat.explore_3.3-4 tidygraph_1.3.0        progressr_0.15.1      \n [58] Formula_1.2-5          ggridges_0.5.6         ggalluvial_0.12.5     \n [61] survival_3.8-3         iterators_1.0.14       foreach_1.5.2         \n [64] tools_4.3.3            stringdist_0.9.15      ica_1.0-3             \n [67] Rcpp_1.0.14            gridExtra_2.3          withr_3.0.2           \n [70] fastmap_1.2.0          digest_0.6.37          R6_2.6.0              \n [73] mime_0.12              colorspace_2.1-1       scattermore_1.2       \n [76] tensor_1.5             spatstat.data_3.1-4    diptest_0.77-1        \n [79] UpSetR_1.4.0           generics_0.1.3         robustbase_0.99-4-1   \n [82] class_7.3-23           graphlayouts_1.2.2     httr_1.4.7            \n [85] htmlwidgets_1.6.4      uwot_0.2.2             pkgconfig_2.0.3       \n [88] gtable_0.3.6           modeltools_0.2-23      lmtest_0.9-40         \n [91] htmltools_0.5.8.1      carData_3.0-5          dotCall64_1.2         \n [94] scales_1.3.0           png_0.1-8              spatstat.univar_3.1-1 \n [97] tzdb_0.4.0             reshape2_1.4.4         uuid_1.2-1            \n[100] nlme_3.1-167           cachem_1.1.0           zoo_1.8-12            \n[103] GlobalOptions_0.1.2    stringr_1.5.1          KernSmooth_2.23-26    \n[106] parallel_4.3.3         miniUI_0.1.1.1         pillar_1.10.1         \n[109] grid_4.3.3             vctrs_0.6.5            RANN_2.6.2            \n[112] promises_1.3.2         ggpubr_0.6.0           car_3.1-3             \n[115] xtable_1.8-4           cluster_2.1.8          readr_2.1.5           \n[118] cli_3.6.4              compiler_4.3.3         future.apply_1.11.3   \n[121] ggsignif_0.6.4         labeling_0.4.3         mclust_6.1.1          \n[124] plyr_1.8.9             stringi_1.8.4          deldir_2.0-4          \n[127] viridisLite_0.4.2      munsell_0.5.1          lazyeval_0.2.2        \n[130] spatstat.geom_3.3-5    Matrix_1.6-5           RcppHNSW_0.6.0        \n[133] hms_1.1.3              future_1.34.0          fpc_2.2-13            \n[136] shiny_1.10.0           kernlab_0.9-33         ROCR_1.0-11           \n[139] broom_1.0.7            memoise_2.0.1          phangorn_2.12.1       \n[142] fastmatch_1.1-6        DEoptimR_1.1-3-1       readxl_1.4.3          \n[145] ape_5.8-1             \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 1679.73\nuser 2269.05\nsys 21.01\npercent 136%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"SeuratClusterStats",desc:"Statistics of the clustering.",npages:1,report_toc:!0,order:17,runinfo:{session:"# Generated by pipen_runinfo v0.8.2\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 11 (bullseye)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.28.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] grid      stats     graphics  grDevices utils     datasets  methods  \n[8] base     \n\nother attached packages:\n [1] clustree_0.5.1        ggraph_2.2.1          ComplexHeatmap_2.18.0\n [4] circlize_0.4.16       tidyseurat_0.8.0      ttservice_0.4.1      \n [7] ggrepel_0.9.6         ggprism_1.0.5         tibble_3.2.1         \n[10] Seurat_5.1.0          SeuratObject_5.0.2    sp_2.2-0             \n[13] ggplot2_3.5.1         tidyr_1.3.1           dplyr_1.1.4          \n[16] tidyselect_1.2.1      jsonlite_1.8.9        rlang_1.1.5          \n[19] logger_0.4.0         \n\nloaded via a namespace (and not attached):\n  [1] RColorBrewer_1.1-3     shape_1.4.6.1          magrittr_2.0.3        \n  [4] ggbeeswarm_0.7.2       spatstat.utils_3.1-2   farver_2.1.2          \n  [7] GlobalOptions_0.1.2    vctrs_0.6.5            ROCR_1.0-11           \n [10] memoise_2.0.1          spatstat.explore_3.3-4 htmltools_0.5.8.1     \n [13] sctransform_0.4.1      parallelly_1.42.0      KernSmooth_2.23-26    \n [16] htmlwidgets_1.6.4      ica_1.0-3              plyr_1.8.9            \n [19] cachem_1.1.0           plotly_4.10.4          zoo_1.8-12            \n [22] igraph_2.0.3           mime_0.12              lifecycle_1.0.4       \n [25] iterators_1.0.14       pkgconfig_2.0.3        Matrix_1.6-5          \n [28] R6_2.6.0               fastmap_1.2.0          clue_0.3-66           \n [31] fitdistrplus_1.2-2     future_1.34.0          shiny_1.10.0          \n [34] digest_0.6.37          colorspace_2.1-1       patchwork_1.3.0       \n [37] S4Vectors_0.40.2       tensor_1.5             RSpectra_0.16-2       \n [40] irlba_2.3.5.1          labeling_0.4.3         progressr_0.15.1      \n [43] fansi_1.0.6            spatstat.sparse_3.1-0  httr_1.4.7            \n [46] polyclip_1.10-7        abind_1.4-5            compiler_4.3.3        \n [49] withr_3.0.2            doParallel_1.0.17      viridis_0.6.5         \n [52] fastDummies_1.7.5      ggforce_0.4.2          MASS_7.3-60.0.1       \n [55] rjson_0.2.23           tools_4.3.3            vipor_0.4.7           \n [58] lmtest_0.9-40          beeswarm_0.4.0         httpuv_1.6.15         \n [61] future.apply_1.11.3    goftest_1.2-3          glue_1.8.0            \n [64] nlme_3.1-167           promises_1.3.2         Rtsne_0.17            \n [67] cluster_2.1.8          reshape2_1.4.4         generics_0.1.3        \n [70] gtable_0.3.6           spatstat.data_3.1-4    data.table_1.16.4     \n [73] tidygraph_1.3.0        BiocGenerics_0.48.1    spatstat.geom_3.3-5   \n [76] RcppAnnoy_0.0.22       RANN_2.6.2             foreach_1.5.2         \n [79] pillar_1.10.1          stringr_1.5.1          spam_2.11-1           \n [82] RcppHNSW_0.6.0         later_1.4.1            splines_4.3.3         \n [85] tweenr_2.0.3           lattice_0.22-6         survival_3.8-3        \n [88] deldir_2.0-4           miniUI_0.1.1.1         pbapply_1.7-2         \n [91] gridExtra_2.3          IRanges_2.36.0         scattermore_1.2       \n [94] stats4_4.3.3           graphlayouts_1.2.2     matrixStats_1.1.0     \n [97] stringi_1.8.4          lazyeval_0.2.2         codetools_0.2-20      \n[100] cli_3.6.4              uwot_0.2.2             xtable_1.8-4          \n[103] reticulate_1.40.0      munsell_0.5.1          Rcpp_1.0.14           \n[106] globals_0.16.3         spatstat.random_3.3-2  png_0.1-8             \n[109] ggrastr_1.0.2          spatstat.univar_3.1-1  parallel_4.3.3        \n[112] ellipsis_0.3.2         dotCall64_1.2          listenv_0.9.1         \n[115] viridisLite_0.4.2      scales_1.3.0           ggridges_0.5.6        \n[118] leiden_0.4.3.1         purrr_1.0.2            crayon_1.5.3          \n[121] GetoptLong_1.0.5       cowplot_1.1.3         \n",time:"# Generated by pipen-runinfo v0.8.2\n\nreal 18.76\nuser 27.07\nsys 0.71\npercent 148%\n",device:"# Generated by pipen-runinfo v0.8.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nmforgehn4.mayo.edu\n\nCPU\n----\nArchitecture:                         x86_64\nCPU op-mode(s):                       32-bit, 64-bit\nByte Order:                           Little Endian\nAddress sizes:                        46 bits physical, 57 bits virtual\nCPU(s):                               96\nOn-line CPU(s) list:                  0-95\nThread(s) per core:                   2\nCore(s) per socket:                   24\nSocket(s):                            2\nNUMA node(s):                         2\nVendor ID:                            GenuineIntel\nCPU family:                           6\nModel:                                106\nModel name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz\nStepping:                             6\nCPU MHz:                              3500.000\nCPU max MHz:                          3500.0000\nCPU min MHz:                          800.0000\nBogoMIPS:                             5600.00\nVirtualization:                       VT-x\nL1d cache:                            2.3 MiB\nL1i cache:                            1.5 MiB\nL2 cache:                             60 MiB\nL3 cache:                             72 MiB\nNUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94\nNUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95\nVulnerability Gather data sampling:   Mitigation; Microcode\nVulnerability Itlb multihit:          Not affected\nVulnerability L1tf:                   Not affected\nVulnerability Mds:                    Not affected\nVulnerability Meltdown:               Not affected\nVulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable\nVulnerability Reg file data sampling: Not affected\nVulnerability Retbleed:               Not affected\nVulnerability Spec rstack overflow:   Not affected\nVulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                  Not affected\nVulnerability Tsx async abort:        Not affected\nFlags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities\n\nMemory\n------\n"}},{name:"Immunarch",desc:"Exploration of Single-cell and Bulk T-cell/Antibody Immune Repertoires",npages:1,report_toc:!0,order:18},{name:"RadarPlots",desc:"Radar plots for cell proportion in different clusters.",npages:1,report_toc:!0,order:19},{name:"CloneResidency",desc:"Identification of clone residency",npages:1,report_toc:!0,order:20},{name:"ScFGSEA",desc:"Gene set enrichment analysis for cells in different groups using `fgsea`",npages:1,report_toc:!0,order:21},{name:"MarkersFinder",desc:"Find markers between different groups of cells",npages:1,report_toc:!0,order:22},{name:"MetaMarkers",desc:"Find markers between three or more groups of cells, using one-way ANOVA or Kruskal-Wallis test.",npages:1,report_toc:!0,order:23},{name:"CellsDistribution",desc:"Distribution of cells (i.e. in a TCR clone) from different groups for each cluster",npages:1,report_toc:!0,order:24},{name:"CDR3AAPhyschem",desc:"CDR3 AA physicochemical feature analysis",npages:1,report_toc:!0,order:25},{name:"ScrnaMetabolicLandscape",desc:"Metabolic landscape analysis for scRNA-seq data",order:28,procs:[{name:"MetabolicPathwayActivity",desc:"This process calculates the pathway activities in different groups and subsets.",npages:1,report_toc:!0,order:28},{name:"MetabolicFeatures",desc:"This process performs enrichment analysis for the metabolic pathways for each group in each subset.",npages:1,report_toc:!0,order:29},{name:"MetabolicFeaturesIntraSubset",desc:"Intra-subset metabolic features - Enrichment analysis in details",npages:1,report_toc:!0,order:30},{name:"MetabolicPathwayHeterogeneity",desc:"Calculate Metabolic Pathway heterogeneity.",npages:1,report_toc:!0,order:31}]}]};function Xn(e){let t,n;return t=new Bt({props:{dir:"up"}}),{c(){g(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Yn(e){let t,n;return t=new Bt({props:{dir:"down"}}),{c(){g(t.$$.fragment)},m(e,a){b(t,e,a),n=!0},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){d(t.$$.fragment,e),n=!1},d(e){$(t,e)}}}function Jn(e){let t,n,a,s,r,i=e[1]>0&&Xn();n=new jn({});let m=e[1]<e[3].npages-1&&Yn();return{c(){i&&i.c(),t=h(),g(n.$$.fragment),a=h(),m&&m.c(),s=f()},m(e,l){i&&i.m(e,l),o(e,t,l),b(n,e,l),o(e,a,l),m&&m.m(e,l),o(e,s,l),r=!0},p(e,n){e[1]>0?i?2&n&&l(i,1):(i=Xn(),i.c(),l(i,1),i.m(t.parentNode,t)):i&&(c(),d(i,1,1,(()=>{i=null})),p()),e[1]<e[3].npages-1?m?10&n&&l(m,1):(m=Yn(),m.c(),l(m,1),m.m(s.parentNode,s)):m&&(c(),d(m,1,1,(()=>{m=null})),p())},i(e){r||(l(i),l(n.$$.fragment,e),l(m),r=!0)},o(e){d(i),d(n.$$.fragment,e),d(m),r=!1},d(e){e&&(_(t),_(a),_(s)),i&&i.d(e),$(n,e),m&&m.d(e)}}}function Qn(e){let t,n,a;function s(t){e[4](t)}let r={logo:e[0],logotext:e[3].desc,versions:Kn.versions,entries:Kn.entries,pipeline_name:Kn.pipeline.name,report_toc:e[3].report_toc,runinfo:e[3].runinfo,toc:Wn,page:e[1],$$slots:{default:[Jn]},$$scope:{ctx:e}};return void 0!==e[2]&&(r.isSideNavOpen=e[2]),t=new Pt({props:r}),v.push((()=>x(t,"isSideNavOpen",s))),{c(){g(t.$$.fragment)},m(e,n){b(t,e,n),a=!0},p(e,[a]){const s={};1&a&&(s.logo=e[0]),8&a&&(s.logotext=e[3].desc),8&a&&(s.report_toc=e[3].report_toc),8&a&&(s.runinfo=e[3].runinfo),2&a&&(s.page=e[1]),42&a&&(s.$$scope={dirty:a,ctx:e}),!n&&4&a&&(n=!0,s.isSideNavOpen=e[2],C((()=>n=!1))),t.$set(s)},i(e){a||(l(t.$$.fragment,e),a=!0)},o(e){d(t.$$.fragment,e),a=!1},d(e){$(t,e)}}}function Zn(e,t,n){let a,{name:s}=t,{page:r=0}=t,i=!1;for(const e of Kn.entries)if(e.procs){for(const t of e.procs)if(t.name==s){a=t;break}}else if(e.name==s){a=e;break}return u((()=>{const e=new URL(window.location.href);let t;e.hash.length>1&&(t=document.getElementById(`${e.hash.slice(1)}`)),t&&t.scrollIntoView({block:"start",behavior:"smooth"})})),e.$$set=e=>{"name"in e&&n(0,s=e.name),"page"in e&&n(1,r=e.page)},[s,r,i,a,function(e){i=e,n(2,i)}]}e("default",new class extends t{constructor(e){super(),n(this,e,Zn,Qn,a,{name:0,page:1})}}({target:document.body,props:{name:"SeuratClusterStats",page:0}}))}}}));
